<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>GeneExpression</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="GeneExpression";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":9,"i3":10,"i4":9,"i5":9,"i6":9,"i7":9,"i8":10,"i9":10,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10,"i19":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GeneExpression.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.dataset</a></div>
<h2 title="Class GeneExpression" class="title">Class GeneExpression</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li><a href="OmicsData.html" title="class in bacnet.datamodel.dataset">bacnet.datamodel.dataset.OmicsData</a></li>
<li>
<ul class="inheritance">
<li><a href="ExpressionData.html" title="class in bacnet.datamodel.dataset">bacnet.datamodel.dataset.ExpressionData</a></li>
<li>
<ul class="inheritance">
<li>bacnet.datamodel.dataset.GeneExpression</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><code>java.io.Serializable</code></dd>
</dl>
<hr>
<pre>public class <span class="typeNameLabel">GeneExpression</span>
extends <a href="ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></pre>
<div class="block">GeneExpression array data from Listeria tiling chip ArrayExpress id : A-AFFY-140
 
 Here double[] values vector has length = 5*probes.size because we put 5 columns column 0-2 = 3
 replicatas column 3 = median expression column 4 = var expression</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>UIBC</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../serialized-form.html#bacnet.datamodel.dataset.GeneExpression">Serialized Form</a></dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a id="nested.classes.inherited.from.class.bacnet.datamodel.dataset.OmicsData">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;bacnet.datamodel.dataset.<a href="OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></h3>
<code><a href="OmicsData.ColNames.html" title="enum in bacnet.datamodel.dataset">OmicsData.ColNames</a>, <a href="OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></code></li>
</ul>
</li>
</ul>
</section>
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#COL_NUMBER">COL_NUMBER</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#dataRead">dataRead</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#EXTENSION">EXTENSION</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.Integer&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#probes">probes</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#PROBES_PATH">PROBES_PATH</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#PROBES_PATH_2">PROBES_PATH_2</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private static long</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#serialVersionUID">serialVersionUID</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="fields.inherited.from.class.bacnet.datamodel.dataset.OmicsData">
<!--   -->
</a>
<h3>Fields inherited from class&nbsp;bacnet.datamodel.dataset.<a href="OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></h3>
<code><a href="OmicsData.html#GENERAL_WT_DATA">GENERAL_WT_DATA</a>, <a href="OmicsData.html#GENERAL_WT_NAME">GENERAL_WT_NAME</a>, <a href="OmicsData.html#MISSING_VALUE">MISSING_VALUE</a>, <a href="OmicsData.html#PATH_COMPARISONS">PATH_COMPARISONS</a>, <a href="OmicsData.html#PATH_EXPR_NORM">PATH_EXPR_NORM</a>, <a href="OmicsData.html#PATH_EXPR_RAW">PATH_EXPR_RAW</a>, <a href="OmicsData.html#PATH_GENEXPR_COMPLETE">PATH_GENEXPR_COMPLETE</a>, <a href="OmicsData.html#PATH_GENEXPR_NORM">PATH_GENEXPR_NORM</a>, <a href="OmicsData.html#PATH_GENEXPR_RAW">PATH_GENEXPR_RAW</a>, <a href="OmicsData.html#PATH_NGS_NORM">PATH_NGS_NORM</a>, <a href="OmicsData.html#PATH_NGS_RAW">PATH_NGS_RAW</a>, <a href="OmicsData.html#PATH_NORM">PATH_NORM</a>, <a href="OmicsData.html#PATH_PROTEOMICS_NORM">PATH_PROTEOMICS_NORM</a>, <a href="OmicsData.html#PATH_PROTEOMICS_RAW">PATH_PROTEOMICS_RAW</a>, <a href="OmicsData.html#PATH_RAW">PATH_RAW</a>, <a href="OmicsData.html#PATH_STREAMING">PATH_STREAMING</a>, <a href="OmicsData.html#PATH_TILING_NORM">PATH_TILING_NORM</a>, <a href="OmicsData.html#PATH_TILING_RAW">PATH_TILING_RAW</a></code></li>
</ul>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">GeneExpression</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.lang.String)">GeneExpression</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code><a href="GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#compare(bacnet.datamodel.dataset.GeneExpression,boolean)">compare</a></span>&#8203;(<a href="GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&nbsp;data2,
       boolean&nbsp;calcData)</code></th>
<td class="colLast">
<div class="block">Create GeneExpression data from the comparison of this one and data2</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#containProbe(java.lang.String)">containProbe</a></span>&#8203;(java.lang.String&nbsp;probe)</code></th>
<td class="colLast">
<div class="block">Check if TreeMap<String, Integer> probes contains a probe</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#convert(bacnet.datamodel.expdesign.Experiment)">convert</a></span>&#8203;(<a href="../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;experiment)</code></th>
<td class="colLast">
<div class="block">Read all GeneExpr data and save in GeneExpressionStreaming format</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#convertData(bacnet.datamodel.dataset.ExpressionMatrix)">convertData</a></span>&#8203;(<a href="ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;exprMatrix)</code></th>
<td class="colLast">
<div class="block">Convert data from an ExpressionMatrix to a GeneExpression</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createProbeList()">createProbeList</a></span>()</code></th>
<td class="colLast">
<div class="block">Read general GeneExpression data and extract the list of probes from it</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>static java.util.TreeMap&lt;java.lang.String,&#8203;<a href="GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGExpressionData(java.util.ArrayList)">getGExpressionData</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;bioConditions)</code></th>
<td class="colLast">
<div class="block">List all TilingData available by biocondName.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>static java.util.TreeMap&lt;java.lang.String,&#8203;<a href="GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGExpressionData(java.util.ArrayList,java.util.ArrayList)">getGExpressionData</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;leftBCs,
                  java.util.ArrayList&lt;java.lang.String&gt;&nbsp;rightBCs)</code></th>
<td class="colLast">
<div class="block">Load all Tiling present in leftBCs and rightBCs.</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>static <a href="GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMean(java.util.ArrayList,java.lang.String)">getMean</a></span>&#8203;(java.util.ArrayList&lt;<a href="GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;&nbsp;datas,
       java.lang.String&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Calculate the mean expression of list of Tiling</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMedianValue(java.lang.String)">getMedianValue</a></span>&#8203;(java.lang.String&nbsp;probe)</code></th>
<td class="colLast">
<div class="block">Get the median value of corresponding probe in double[][] values</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>double[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMedianVector()">getMedianVector</a></span>()</code></th>
<td class="colLast">
<div class="block">Extract the column containing the median<br></div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.Integer&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProbes()">getProbes</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>double[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTriplicateValue(java.lang.String)">getTriplicateValue</a></span>&#8203;(java.lang.String&nbsp;probe)</code></th>
<td class="colLast">
<div class="block">Get all technical replicates values of corresponding probe in double[][] values</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getVarianceValue(java.lang.String)">getVarianceValue</a></span>&#8203;(java.lang.String&nbsp;probe)</code></th>
<td class="colLast">
<div class="block">Get variance value of corresponding probe in double[][] values</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isDataRead()">isDataRead</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#load()">load</a></span>()</code></th>
<td class="colLast">
<div class="block">Load information on the data: name, date, note, stat, etc ...</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadMatrix()">loadMatrix</a></span>()</code></th>
<td class="colLast">
<div class="block">Load ExpressionMatrix found in TranscriptomeData.PATH_GENEXPR_NORM+getName()</div>
</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDataRead(boolean)">setDataRead</a></span>&#8203;(boolean&nbsp;dataRead)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setMedianValue(java.lang.String,double)">setMedianValue</a></span>&#8203;(java.lang.String&nbsp;probe,
              double&nbsp;value)</code></th>
<td class="colLast">
<div class="block">Set the median value of corresponding probe in double[][] values</div>
</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setProbes(java.util.TreeMap)">setProbes</a></span>&#8203;(java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.Integer&gt;&nbsp;probes)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code><a href="ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#toExpressionMatrix()">toExpressionMatrix</a></span>()</code></th>
<td class="colLast">
<div class="block">Transform a <code>GeneExpression</code> into an <code>ExpressionMatrix</code></div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.bacnet.datamodel.dataset.ExpressionData">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;bacnet.datamodel.dataset.<a href="ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></h3>
<code><a href="ExpressionData.html#compare(bacnet.datamodel.dataset.ExpressionData,boolean)">compare</a>, <a href="ExpressionData.html#getBioCondition()">getBioCondition</a>, <a href="ExpressionData.html#getChromosomeID()">getChromosomeID</a>, <a href="ExpressionData.html#getExtension()">getExtension</a>, <a href="ExpressionData.html#getGeneralWtName()">getGeneralWtName</a>, <a href="ExpressionData.html#getGenomeName()">getGenomeName</a>, <a href="ExpressionData.html#getLength()">getLength</a>, <a href="ExpressionData.html#getMatrix()">getMatrix</a>, <a href="ExpressionData.html#getMax()">getMax</a>, <a href="ExpressionData.html#getMean()">getMean</a>, <a href="ExpressionData.html#getMedian()">getMedian</a>, <a href="ExpressionData.html#getMin()">getMin</a>, <a href="ExpressionData.html#getPATH()">getPATH</a>, <a href="ExpressionData.html#getRead()">getRead</a>, <a href="ExpressionData.html#getSerialversionuid()">getSerialversionuid</a>, <a href="ExpressionData.html#getValue(int)">getValue</a>, <a href="ExpressionData.html#getValues()">getValues</a>, <a href="ExpressionData.html#getVariance()">getVariance</a>, <a href="ExpressionData.html#isAlreadyRead()">isAlreadyRead</a>, <a href="ExpressionData.html#isInfoRead()">isInfoRead</a>, <a href="ExpressionData.html#isNGS()">isNGS</a>, <a href="ExpressionData.html#load(boolean)">load</a>, <a href="ExpressionData.html#load(java.lang.String)">load</a>, <a href="ExpressionData.html#load(java.lang.String,boolean)">load</a>, <a href="ExpressionData.html#read()">read</a>, <a href="ExpressionData.html#read(int)">read</a>, <a href="ExpressionData.html#read(int,boolean)">read</a>, <a href="ExpressionData.html#read(int,int)">read</a>, <a href="ExpressionData.html#save()">save</a>, <a href="ExpressionData.html#setAlreadyRead(boolean)">setAlreadyRead</a>, <a href="ExpressionData.html#setChromosomeID(java.lang.String)">setChromosomeID</a>, <a href="ExpressionData.html#setGenomeName(java.lang.String)">setGenomeName</a>, <a href="ExpressionData.html#setInfoRead(boolean)">setInfoRead</a>, <a href="ExpressionData.html#setLength(int)">setLength</a>, <a href="ExpressionData.html#setMatrix(bacnet.datamodel.dataset.ExpressionMatrix)">setMatrix</a>, <a href="ExpressionData.html#setMax(double)">setMax</a>, <a href="ExpressionData.html#setMean(double)">setMean</a>, <a href="ExpressionData.html#setMedian(double)">setMedian</a>, <a href="ExpressionData.html#setMin(double)">setMin</a>, <a href="ExpressionData.html#setPATH(java.lang.String)">setPATH</a>, <a href="ExpressionData.html#setRead(boolean%5B%5D)">setRead</a>, <a href="ExpressionData.html#setStat()">setStat</a>, <a href="ExpressionData.html#setValues(double%5B%5D)">setValues</a>, <a href="ExpressionData.html#setVariance(double)">setVariance</a>, <a href="ExpressionData.html#toString()">toString</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.bacnet.datamodel.dataset.OmicsData">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;bacnet.datamodel.dataset.<a href="OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></h3>
<code><a href="OmicsData.html#getBioCondName()">getBioCondName</a>, <a href="OmicsData.html#getDate()">getDate</a>, <a href="OmicsData.html#getName()">getName</a>, <a href="OmicsData.html#getNote()">getNote</a>, <a href="OmicsData.html#getRawDatas()">getRawDatas</a>, <a href="OmicsData.html#getType()">getType</a>, <a href="OmicsData.html#initStaticVariables()">initStaticVariables</a>, <a href="OmicsData.html#setBioCondName(java.lang.String)">setBioCondName</a>, <a href="OmicsData.html#setDate(java.lang.String)">setDate</a>, <a href="OmicsData.html#setName(java.lang.String)">setName</a>, <a href="OmicsData.html#setNote(java.lang.String)">setNote</a>, <a href="OmicsData.html#setRawDatas(java.util.ArrayList)">setRawDatas</a>, <a href="OmicsData.html#setType(bacnet.datamodel.dataset.OmicsData.TypeData)">setType</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="serialVersionUID">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>serialVersionUID</h4>
<pre>private static final&nbsp;long serialVersionUID</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../constant-values.html#bacnet.datamodel.dataset.GeneExpression.serialVersionUID">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a id="PROBES_PATH">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PROBES_PATH</h4>
<pre>public static&nbsp;java.lang.String PROBES_PATH</pre>
</li>
</ul>
<a id="PROBES_PATH_2">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PROBES_PATH_2</h4>
<pre>public static&nbsp;java.lang.String PROBES_PATH_2</pre>
</li>
</ul>
<a id="EXTENSION">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>EXTENSION</h4>
<pre>public static&nbsp;java.lang.String EXTENSION</pre>
</li>
</ul>
<a id="COL_NUMBER">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>COL_NUMBER</h4>
<pre>public static&nbsp;int COL_NUMBER</pre>
</li>
</ul>
<a id="dataRead">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>dataRead</h4>
<pre>private transient&nbsp;boolean dataRead</pre>
</li>
</ul>
<a id="probes">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>probes</h4>
<pre>private transient&nbsp;java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.Integer&gt; probes</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>GeneExpression</h4>
<pre>public&nbsp;GeneExpression()</pre>
</li>
</ul>
<a id="&lt;init&gt;(java.lang.String)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>GeneExpression</h4>
<pre>public&nbsp;GeneExpression&#8203;(java.lang.String&nbsp;name)</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="load()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>load</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;load()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from class:&nbsp;<code><a href="ExpressionData.html#load()">ExpressionData</a></code></span></div>
<div class="block">Load information on the data: name, date, note, stat, etc ... The path for data loading will be:
 PATH_STREAMING+this.getName()+"Info"+this.EXTENSION;</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="ExpressionData.html#load()">load</a></code>&nbsp;in class&nbsp;<code><a href="ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></code></dd>
</dl>
</li>
</ul>
<a id="getMedianValue(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMedianValue</h4>
<pre class="methodSignature">public&nbsp;double&nbsp;getMedianValue&#8203;(java.lang.String&nbsp;probe)</pre>
<div class="block">Get the median value of corresponding probe in double[][] values</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>probe</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="setMedianValue(java.lang.String,double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMedianValue</h4>
<pre class="methodSignature">public&nbsp;double&nbsp;setMedianValue&#8203;(java.lang.String&nbsp;probe,
                             double&nbsp;value)</pre>
<div class="block">Set the median value of corresponding probe in double[][] values</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>probe</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getMedianVector()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMedianVector</h4>
<pre class="methodSignature">public&nbsp;double[]&nbsp;getMedianVector()</pre>
<div class="block">Extract the column containing the median<br></div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a vector of all median</dd>
</dl>
</li>
</ul>
<a id="getTriplicateValue(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTriplicateValue</h4>
<pre class="methodSignature">public&nbsp;double[]&nbsp;getTriplicateValue&#8203;(java.lang.String&nbsp;probe)</pre>
<div class="block">Get all technical replicates values of corresponding probe in double[][] values</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>probe</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getVarianceValue(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getVarianceValue</h4>
<pre class="methodSignature">public&nbsp;double&nbsp;getVarianceValue&#8203;(java.lang.String&nbsp;probe)</pre>
<div class="block">Get variance value of corresponding probe in double[][] values</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>probe</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="containProbe(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>containProbe</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;containProbe&#8203;(java.lang.String&nbsp;probe)</pre>
<div class="block">Check if TreeMap<String, Integer> probes contains a probe</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>probe</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="loadMatrix()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadMatrix</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;loadMatrix()</pre>
<div class="block">Load ExpressionMatrix found in TranscriptomeData.PATH_GENEXPR_NORM+getName()</div>
</li>
</ul>
<a id="toExpressionMatrix()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toExpressionMatrix</h4>
<pre class="methodSignature">public&nbsp;<a href="ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;toExpressionMatrix()</pre>
<div class="block">Transform a <code>GeneExpression</code> into an <code>ExpressionMatrix</code></div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>representation of the <code>GFeneExpression</code></dd>
</dl>
</li>
</ul>
<a id="convert(bacnet.datamodel.expdesign.Experiment)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>convert</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;convert&#8203;(<a href="../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;experiment)
                    throws java.lang.Exception</pre>
<div class="block">Read all GeneExpr data and save in GeneExpressionStreaming format</div>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="convertData(bacnet.datamodel.dataset.ExpressionMatrix)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>convertData</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;convertData&#8203;(<a href="ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;exprMatrix)</pre>
<div class="block">Convert data from an ExpressionMatrix to a GeneExpression</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>exprMatrix</code> - </dd>
</dl>
</li>
</ul>
<a id="isDataRead()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isDataRead</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;isDataRead()</pre>
</li>
</ul>
<a id="setDataRead(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDataRead</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDataRead&#8203;(boolean&nbsp;dataRead)</pre>
</li>
</ul>
<a id="createProbeList()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createProbeList</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;createProbeList()
                            throws java.io.FileNotFoundException,
                                   java.io.IOException</pre>
<div class="block">Read general GeneExpression data and extract the list of probes from it</div>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.io.FileNotFoundException</code></dd>
<dd><code>java.io.IOException</code></dd>
</dl>
</li>
</ul>
<a id="compare(bacnet.datamodel.dataset.GeneExpression,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>compare</h4>
<pre class="methodSignature">public&nbsp;<a href="GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&nbsp;compare&#8203;(<a href="GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&nbsp;data2,
                              boolean&nbsp;calcData)</pre>
<div class="block">Create GeneExpression data from the comparison of this one and data2</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>data2</code> - </dd>
<dd><code>calcData</code> - true if we want to calculate the data</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.io.IOException</code></dd>
</dl>
</li>
</ul>
<a id="getMean(java.util.ArrayList,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMean</h4>
<pre class="methodSignature">public static&nbsp;<a href="GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&nbsp;getMean&#8203;(java.util.ArrayList&lt;<a href="GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;&nbsp;datas,
                                     java.lang.String&nbsp;name)</pre>
<div class="block">Calculate the mean expression of list of Tiling</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>datas</code> - </dd>
<dd><code>name</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getGExpressionData(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGExpressionData</h4>
<pre class="methodSignature">public static&nbsp;java.util.TreeMap&lt;java.lang.String,&#8203;<a href="GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;&nbsp;getGExpressionData&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;bioConditions)</pre>
<div class="block">List all TilingData available by biocondName. Initiate all TilingDataStreaming.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>geneExp</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getGExpressionData(java.util.ArrayList,java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGExpressionData</h4>
<pre class="methodSignature">public static&nbsp;java.util.TreeMap&lt;java.lang.String,&#8203;<a href="GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;&nbsp;getGExpressionData&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;leftBCs,
                                                                                          java.util.ArrayList&lt;java.lang.String&gt;&nbsp;rightBCs)</pre>
<div class="block">Load all Tiling present in leftBCs and rightBCs.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>geneExp</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getProbes()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProbes</h4>
<pre class="methodSignature">public&nbsp;java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.Integer&gt;&nbsp;getProbes()</pre>
</li>
</ul>
<a id="setProbes(java.util.TreeMap)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setProbes</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setProbes&#8203;(java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.Integer&gt;&nbsp;probes)</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GeneExpression.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
