<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 18:10:05 CEST 2019 -->
<title>NGS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="NGS";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/NGS.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.dataset</a></div>
<h2 title="Class NGS" class="title">Class NGS</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li><a href="OmicsData.html" title="class in bacnet.datamodel.dataset">bacnet.datamodel.dataset.OmicsData</a></li>
<li>
<ul class="inheritance">
<li>bacnet.datamodel.dataset.NGS</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><code>java.io.Serializable</code></dd>
</dl>
<hr>
<pre>public class <span class="typeNameLabel">NGS</span>
extends <a href="OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a>
implements java.io.Serializable</pre>
<div class="block">Class for manipulating various RNASeq data, herited from ExpressionData <br>
 double[] value is of genome size, when ones want to see RNAseq expression at
 a certain bp position, just extract values[bpposition]<br>
 Need to be loaded first, and unlog has RNASeq will be logged for display when
 convert in the platform</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>UIBC</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../serialized-form.html#bacnet.datamodel.dataset.NGS">Serialized Form</a></dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a id="nested.classes.inherited.from.class.bacnet.datamodel.dataset.OmicsData">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;bacnet.datamodel.dataset.<a href="OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></h3>
<code><a href="OmicsData.ColNames.html" title="enum in bacnet.datamodel.dataset">OmicsData.ColNames</a>, <a href="OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></code></li>
</ul>
</li>
</ul>
</section>
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#EXTENSION">EXTENSION</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="fields.inherited.from.class.bacnet.datamodel.dataset.OmicsData">
<!--   -->
</a>
<h3>Fields inherited from class&nbsp;bacnet.datamodel.dataset.<a href="OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></h3>
<code><a href="OmicsData.html#GENERAL_WT_DATA">GENERAL_WT_DATA</a>, <a href="OmicsData.html#GENERAL_WT_NAME">GENERAL_WT_NAME</a>, <a href="OmicsData.html#MISSING_VALUE">MISSING_VALUE</a>, <a href="OmicsData.html#PATH_COMPARISONS">PATH_COMPARISONS</a>, <a href="OmicsData.html#PATH_EXPR_NORM">PATH_EXPR_NORM</a>, <a href="OmicsData.html#PATH_EXPR_RAW">PATH_EXPR_RAW</a>, <a href="OmicsData.html#PATH_GENEXPR_COMPLETE">PATH_GENEXPR_COMPLETE</a>, <a href="OmicsData.html#PATH_GENEXPR_NORM">PATH_GENEXPR_NORM</a>, <a href="OmicsData.html#PATH_GENEXPR_RAW">PATH_GENEXPR_RAW</a>, <a href="OmicsData.html#PATH_NGS_NORM">PATH_NGS_NORM</a>, <a href="OmicsData.html#PATH_NGS_RAW">PATH_NGS_RAW</a>, <a href="OmicsData.html#PATH_NORM">PATH_NORM</a>, <a href="OmicsData.html#PATH_PROTEOMICS_NORM">PATH_PROTEOMICS_NORM</a>, <a href="OmicsData.html#PATH_PROTEOMICS_RAW">PATH_PROTEOMICS_RAW</a>, <a href="OmicsData.html#PATH_RAW">PATH_RAW</a>, <a href="OmicsData.html#PATH_STREAMING">PATH_STREAMING</a>, <a href="OmicsData.html#PATH_TILING_NORM">PATH_TILING_NORM</a>, <a href="OmicsData.html#PATH_TILING_RAW">PATH_TILING_RAW</a></code></li>
</ul>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">NGS</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.lang.String,java.lang.String)">NGS</a></span>&#8203;(java.lang.String&nbsp;name,
   java.lang.String&nbsp;genomeName)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code><a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#compare(bacnet.datamodel.dataset.NGS,boolean)">compare</a></span>&#8203;(<a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&nbsp;data2,
       boolean&nbsp;calcData)</code></th>
<td class="colLast">
<div class="block">Create Tiling data from the comparison of this one and data2</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#convert(bacnet.datamodel.dataset.NGS,boolean)">convert</a></span>&#8203;(<a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&nbsp;data,
       boolean&nbsp;logTransformed)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#convertWigFile(bacnet.datamodel.dataset.NGS,boolean)">convertWigFile</a></span>&#8203;(<a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&nbsp;data,
              boolean&nbsp;logTransformed)</code></th>
<td class="colLast">
<div class="block">WIG file are common RNASeq data<br>
 Calculate first the MEDIAN value of the different replicates ad then log2
 transform and save the NGS data Two columns: <br>
 one for the base pair position
 one for the coverage value</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>java.util.HashMap&lt;java.lang.String,&#8203;<a href="ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDatasets()">getDatasets</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGenomeName()">getGenomeName</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getUnlogValue(int,java.lang.String)">getUnlogValue</a></span>&#8203;(int&nbsp;bpPosition,
             java.lang.String&nbsp;chromoID)</code></th>
<td class="colLast">
<div class="block">Get value at a certain base pair position and unlog it by calculating
 2^value<br>
 It works faster if data has been already loaded by this.read() or
 this.read(from,to)</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getValue(int,java.lang.String)">getValue</a></span>&#8203;(int&nbsp;position,
        java.lang.String&nbsp;chromoID)</code></th>
<td class="colLast">
<div class="block">Return the value at the right position in the double[] values argument<br>
 Read the data if necessary</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isInfoRead()">isInfoRead</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isLogTransformed()">isLogTransformed</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#load()">load</a></span>()</code></th>
<td class="colLast">
<div class="block">Load information on the datasets: name, date, note, stat, etc ...</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#read()">read</a></span>()</code></th>
<td class="colLast">
<div class="block">Load the entire TranscriptomeData</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#readWIGChromo(java.lang.String,java.lang.String,bacnet.datamodel.sequence.Genome)">readWIGChromo</a></span>&#8203;(java.lang.String&nbsp;fileName,
             java.lang.String&nbsp;chromoName,
             <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Will read your wig file and parse it to BACNET <br>
 ONLY WIG FILE WITH variableStep and raw format are accepted No fixedStep
 format</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#save(boolean)">save</a></span>&#8203;(boolean&nbsp;saveValues)</code></th>
<td class="colLast">
<div class="block">Save stat and values in two separate files <br>
 Saving is done via DataStream to allow streaming reading</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDatasets(java.util.HashMap)">setDatasets</a></span>&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;<a href="ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;datasets)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGenomeName(java.lang.String)">setGenomeName</a></span>&#8203;(java.lang.String&nbsp;genomeName)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setInfoRead(boolean)">setInfoRead</a></span>&#8203;(boolean&nbsp;infoRead)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setLogTransformed(boolean)">setLogTransformed</a></span>&#8203;(boolean&nbsp;logTransformed)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.bacnet.datamodel.dataset.OmicsData">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;bacnet.datamodel.dataset.<a href="OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></h3>
<code><a href="OmicsData.html#getBioCondName()">getBioCondName</a>, <a href="OmicsData.html#getDate()">getDate</a>, <a href="OmicsData.html#getExtension()">getExtension</a>, <a href="OmicsData.html#getName()">getName</a>, <a href="OmicsData.html#getNote()">getNote</a>, <a href="OmicsData.html#getRawDatas()">getRawDatas</a>, <a href="OmicsData.html#getSerialversionuid()">getSerialversionuid</a>, <a href="OmicsData.html#getType()">getType</a>, <a href="OmicsData.html#initStaticVariables()">initStaticVariables</a>, <a href="OmicsData.html#setBioCondName(java.lang.String)">setBioCondName</a>, <a href="OmicsData.html#setDate(java.lang.String)">setDate</a>, <a href="OmicsData.html#setName(java.lang.String)">setName</a>, <a href="OmicsData.html#setNote(java.lang.String)">setNote</a>, <a href="OmicsData.html#setRawDatas(java.util.ArrayList)">setRawDatas</a>, <a href="OmicsData.html#setType(bacnet.datamodel.dataset.OmicsData.TypeData)">setType</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="EXTENSION">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>EXTENSION</h4>
<pre>public static&nbsp;java.lang.String EXTENSION</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>NGS</h4>
<pre>public&nbsp;NGS()</pre>
</li>
</ul>
<a id="&lt;init&gt;(java.lang.String,java.lang.String)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>NGS</h4>
<pre>public&nbsp;NGS&#8203;(java.lang.String&nbsp;name,
           java.lang.String&nbsp;genomeName)</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="getValue(int,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getValue</h4>
<pre class="methodSignature">public&nbsp;double&nbsp;getValue&#8203;(int&nbsp;position,
                       java.lang.String&nbsp;chromoID)</pre>
<div class="block">Return the value at the right position in the double[] values argument<br>
 Read the data if necessary</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>position</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="load()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>load</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;load()</pre>
<div class="block">Load information on the datasets: name, date, note, stat, etc ... The path
 for data loading will be:
 PATH_STREAMING+this.getName()+"_"chromoID+"Info"+this.EXTENSION;</div>
</li>
</ul>
<a id="read()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>read</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;read()</pre>
<div class="block">Load the entire TranscriptomeData</div>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.io.IOException</code></dd>
</dl>
</li>
</ul>
<a id="save(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>save</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;save&#8203;(boolean&nbsp;saveValues)</pre>
<div class="block">Save stat and values in two separate files <br>
 Saving is done via DataStream to allow streaming reading</div>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.io.IOException</code></dd>
</dl>
</li>
</ul>
<a id="getUnlogValue(int,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getUnlogValue</h4>
<pre class="methodSignature">public&nbsp;double&nbsp;getUnlogValue&#8203;(int&nbsp;bpPosition,
                            java.lang.String&nbsp;chromoID)</pre>
<div class="block">Get value at a certain base pair position and unlog it by calculating
 2^value<br>
 It works faster if data has been already loaded by this.read() or
 this.read(from,to)</div>
</li>
</ul>
<a id="convert(bacnet.datamodel.dataset.NGS,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>convert</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;convert&#8203;(<a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&nbsp;data,
                    boolean&nbsp;logTransformed)</pre>
</li>
</ul>
<a id="convertWigFile(bacnet.datamodel.dataset.NGS,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>convertWigFile</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;convertWigFile&#8203;(<a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&nbsp;data,
                           boolean&nbsp;logTransformed)</pre>
<div class="block">WIG file are common RNASeq data<br>
 Calculate first the MEDIAN value of the different replicates ad then log2
 transform and save the NGS data Two columns: <br>
 <li>one for the base pair position
 <li>one for the coverage value</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>rawDataPath</code> - </dd>
<dd><code>genome</code> - </dd>
</dl>
</li>
</ul>
<a id="compare(bacnet.datamodel.dataset.NGS,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>compare</h4>
<pre class="methodSignature">public&nbsp;<a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&nbsp;compare&#8203;(<a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&nbsp;data2,
                   boolean&nbsp;calcData)</pre>
<div class="block">Create Tiling data from the comparison of this one and data2</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>data2</code> - </dd>
<dd><code>calcData</code> - true if we want to calculate the data</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.io.IOException</code></dd>
</dl>
</li>
</ul>
<a id="readWIGChromo(java.lang.String,java.lang.String,bacnet.datamodel.sequence.Genome)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readWIGChromo</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;readWIGChromo&#8203;(java.lang.String&nbsp;fileName,
                                                           java.lang.String&nbsp;chromoName,
                                                           <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</pre>
<div class="block">Will read your wig file and parse it to BACNET <br>
 ONLY WIG FILE WITH variableStep and raw format are accepted No fixedStep
 format</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>fileName</code> - </dd>
<dd><code>chromoName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>list of position and the corresponding coverage on chromoName
         chromosome</dd>
</dl>
</li>
</ul>
<a id="isInfoRead()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isInfoRead</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;isInfoRead()</pre>
</li>
</ul>
<a id="setInfoRead(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setInfoRead</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setInfoRead&#8203;(boolean&nbsp;infoRead)</pre>
</li>
</ul>
<a id="getDatasets()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDatasets</h4>
<pre class="methodSignature">public&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;<a href="ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;getDatasets()</pre>
</li>
</ul>
<a id="setDatasets(java.util.HashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDatasets</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDatasets&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;<a href="ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;datasets)</pre>
</li>
</ul>
<a id="getGenomeName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomeName</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getGenomeName()</pre>
</li>
</ul>
<a id="setGenomeName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGenomeName</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGenomeName&#8203;(java.lang.String&nbsp;genomeName)</pre>
</li>
</ul>
<a id="isLogTransformed()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isLogTransformed</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;isLogTransformed()</pre>
</li>
</ul>
<a id="setLogTransformed(boolean)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setLogTransformed</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setLogTransformed&#8203;(boolean&nbsp;logTransformed)</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/NGS.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
