<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Fri Dec 10 09:15:34 CET 2021 -->
<title>NGS</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-10">
<meta name="description" content="declaration: package: bacnet.datamodel.dataset, class: NGS">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var evenRowColor = "even-row-color";
var oddRowColor = "odd-row-color";
var tableTab = "table-tab";
var activeTableTab = "active-table-tab";
var pathtoroot = "../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/NGS.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html#class">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div>
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li><a href="#nested-class-summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field-summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor-summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li><a href="#field-detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor-detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.dataset</a></div>
<h1 title="Class NGS" class="title">Class NGS</h1>
</div>
<div class="inheritance" title="Inheritance Tree"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">java.lang.Object</a>
<div class="inheritance"><a href="OmicsData.html" title="class in bacnet.datamodel.dataset">bacnet.datamodel.dataset.OmicsData</a>
<div class="inheritance">bacnet.datamodel.dataset.NGS</div>
</div>
</div>
<section class="class-description" id="class-description">
<dl class="notes">
<dt>All Implemented Interfaces:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Serializable.html" title="class or interface in java.io" class="external-link">Serializable</a></code></dd>
</dl>
<hr>
<div class="type-signature"><span class="modifiers">public class </span><span class="element-name type-name-label">NGS</span>
<span class="extends-implements">extends <a href="OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a>
implements <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Serializable.html" title="class or interface in java.io" class="external-link">Serializable</a></span></div>
<div class="block">Class for manipulating various RNASeq data, herited from ExpressionData <br>
 double[] value is of genome size, when ones want to see RNAseq expression at a certain bp
 position, just extract values[bpposition]<br>
 Need to be loaded first, and unlog has RNASeq will be logged for display when convert in the
 platform</div>
<dl class="notes">
<dt>Author:</dt>
<dd>UIBC</dd>
<dt>See Also:</dt>
<dd>
<ul class="see-list">
<li><a href="../../../serialized-form.html#bacnet.datamodel.dataset.NGS">Serialized Form</a></li>
</ul>
</dd>
</dl>
</section>
<section class="summary">
<ul class="summary-list">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<li>
<section class="nested-class-summary" id="nested-class-summary">
<h2>Nested Class Summary</h2>
<div class="inherited-list">
<h2 id="nested-classes-inherited-from-class-bacnet.datamodel.dataset.OmicsData">Nested classes/interfaces inherited from class&nbsp;bacnet.datamodel.dataset.<a href="OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></h2>
<code><a href="OmicsData.ColNames.html" title="enum in bacnet.datamodel.dataset">OmicsData.ColNames</a>, <a href="OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></code></div>
</section>
</li>
<!-- =========== FIELD SUMMARY =========== -->
<li>
<section class="field-summary" id="field-summary">
<h2>Field Summary</h2>
<div class="caption"><span>Fields</span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Field</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><code><a href="#EXTENSION" class="member-name-link">EXTENSION</a></code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="inherited-list">
<h3 id="fields-inherited-from-class-bacnet.datamodel.dataset.OmicsData">Fields inherited from class&nbsp;bacnet.datamodel.dataset.<a href="OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></h3>
<code><a href="OmicsData.html#GENERAL_WT_DATA">GENERAL_WT_DATA</a>, <a href="OmicsData.html#GENERAL_WT_NAME">GENERAL_WT_NAME</a>, <a href="OmicsData.html#MISSING_VALUE">MISSING_VALUE</a>, <a href="OmicsData.html#PATH_COMPARISONS">PATH_COMPARISONS</a>, <a href="OmicsData.html#PATH_EXPR_NORM">PATH_EXPR_NORM</a>, <a href="OmicsData.html#PATH_EXPR_RAW">PATH_EXPR_RAW</a>, <a href="OmicsData.html#PATH_GENEXPR_COMPLETE">PATH_GENEXPR_COMPLETE</a>, <a href="OmicsData.html#PATH_GENEXPR_NORM">PATH_GENEXPR_NORM</a>, <a href="OmicsData.html#PATH_GENEXPR_RAW">PATH_GENEXPR_RAW</a>, <a href="OmicsData.html#PATH_NGS_NORM">PATH_NGS_NORM</a>, <a href="OmicsData.html#PATH_NGS_RAW">PATH_NGS_RAW</a>, <a href="OmicsData.html#PATH_NORM">PATH_NORM</a>, <a href="OmicsData.html#PATH_PROTEOMICS_NORM">PATH_PROTEOMICS_NORM</a>, <a href="OmicsData.html#PATH_PROTEOMICS_RAW">PATH_PROTEOMICS_RAW</a>, <a href="OmicsData.html#PATH_RAW">PATH_RAW</a>, <a href="OmicsData.html#PATH_STREAMING">PATH_STREAMING</a>, <a href="OmicsData.html#PATH_TILING_NORM">PATH_TILING_NORM</a>, <a href="OmicsData.html#PATH_TILING_RAW">PATH_TILING_RAW</a></code></div>
</section>
</li>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<li>
<section class="constructor-summary" id="constructor-summary">
<h2>Constructor Summary</h2>
<div class="caption"><span>Constructors</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E()" class="member-name-link">NGS</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-constructor-name odd-row-color"><code><a href="#%3Cinit%3E(java.lang.String,java.lang.String)" class="member-name-link">NGS</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab2" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab2', 3)" class="table-tab">Instance Methods</button><button id="method-summary-table-tab4" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab4', 3)" class="table-tab">Concrete Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel">
<div class="summary-table three-column-summary" aria-labelledby="method-summary-table-tab0">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#compare(bacnet.datamodel.dataset.NGS,boolean)" class="member-name-link">compare</a><wbr>(<a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&nbsp;data2,
 boolean&nbsp;calcData)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Create RNASeq data from the comparison of this one and data2</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#convert(bacnet.datamodel.dataset.NGS,boolean)" class="member-name-link">convert</a><wbr>(<a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&nbsp;data,
 boolean&nbsp;logTransformed)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Run the conversion of wigfiles</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#convertWigFile(bacnet.datamodel.dataset.NGS,boolean)" class="member-name-link">convertWigFile</a><wbr>(<a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&nbsp;data,
 boolean&nbsp;logTransformed)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">WIG file are common RNASeq data<br>
 Calculate first the MEDIAN value of the different replicates ad then log2 transform and save the
 NGS data Two columns: <br>
 one for the base pair position
 one for the coverage value</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/HashMap.html" title="class or interface in java.util" class="external-link">HashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getDatasets()" class="member-name-link">getDatasets</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getGenomeName()" class="member-name-link">getGenomeName</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>double</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getUnlogValue(int,java.lang.String)" class="member-name-link">getUnlogValue</a><wbr>(int&nbsp;bpPosition,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;chromoID)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get value at a certain base pair position and unlog it by calculating 2^value<br>
 It works faster if data has been already loaded by this.read() or this.read(from,to)</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>boolean</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#isInfoRead()" class="member-name-link">isInfoRead</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>boolean</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#isLogTransformed()" class="member-name-link">isLogTransformed</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#load()" class="member-name-link">load</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Load information on the datasets: name, date, note, stat, etc ...</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#read()" class="member-name-link">read</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Load the entire TranscriptomeData</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#readWIGChromo(java.lang.String,java.lang.String,bacnet.datamodel.sequence.Genome)" class="member-name-link">readWIGChromo</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;chromoName,
 <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Will read your wig file and parse it to BACNET <br>
 ONLY WIG FILE WITH variableStep and raw format are accepted No fixedStep format</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#save(boolean)" class="member-name-link">save</a><wbr>(boolean&nbsp;saveValues)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Save stat and values in two separate files <br>
 Saving is done via DataStream to allow streaming reading</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setDatasets(java.util.HashMap)" class="member-name-link">setDatasets</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/HashMap.html" title="class or interface in java.util" class="external-link">HashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;datasets)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setGenomeName(java.lang.String)" class="member-name-link">setGenomeName</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setInfoRead(boolean)" class="member-name-link">setInfoRead</a><wbr>(boolean&nbsp;infoRead)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setLogTransformed(boolean)" class="member-name-link">setLogTransformed</a><wbr>(boolean&nbsp;logTransformed)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
</div>
</div>
</div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-bacnet.datamodel.dataset.OmicsData">Methods inherited from class&nbsp;bacnet.datamodel.dataset.<a href="OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></h3>
<code><a href="OmicsData.html#getBioCondName()">getBioCondName</a>, <a href="OmicsData.html#getDate()">getDate</a>, <a href="OmicsData.html#getExtension()">getExtension</a>, <a href="OmicsData.html#getName()">getName</a>, <a href="OmicsData.html#getNote()">getNote</a>, <a href="OmicsData.html#getRawDatas()">getRawDatas</a>, <a href="OmicsData.html#getSerialversionuid()">getSerialversionuid</a>, <a href="OmicsData.html#getType()">getType</a>, <a href="OmicsData.html#initStaticVariables()">initStaticVariables</a>, <a href="OmicsData.html#setBioCondName(java.lang.String)">setBioCondName</a>, <a href="OmicsData.html#setDate(java.lang.String)">setDate</a>, <a href="OmicsData.html#setName(java.lang.String)">setName</a>, <a href="OmicsData.html#setNote(java.lang.String)">setNote</a>, <a href="OmicsData.html#setRawDatas(java.util.ArrayList)">setRawDatas</a>, <a href="OmicsData.html#setType(bacnet.datamodel.dataset.OmicsData.TypeData)">setType</a></code></div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Object">Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)" title="class or interface in java.lang" class="external-link">equals</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#getClass()" title="class or interface in java.lang" class="external-link">getClass</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode()" title="class or interface in java.lang" class="external-link">hashCode</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notify()" title="class or interface in java.lang" class="external-link">notify</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notifyAll()" title="class or interface in java.lang" class="external-link">notifyAll</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#toString()" title="class or interface in java.lang" class="external-link">toString</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait()" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long)" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long,int)" title="class or interface in java.lang" class="external-link">wait</a></code></div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ============ FIELD DETAIL =========== -->
<li>
<section class="field-details" id="field-detail">
<h2>Field Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="EXTENSION">
<h3>EXTENSION</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">EXTENSION</span></div>
</section>
</li>
</ul>
</section>
</li>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<li>
<section class="constructor-details" id="constructor-detail">
<h2>Constructor Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="&lt;init&gt;()">
<h3>NGS</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name">NGS</span>()</div>
</section>
</li>
<li>
<section class="detail" id="&lt;init&gt;(java.lang.String,java.lang.String)">
<h3>NGS</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name">NGS</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName)</span></div>
</section>
</li>
</ul>
</section>
</li>
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="load()">
<h3>load</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">load</span>()</div>
<div class="block">Load information on the datasets: name, date, note, stat, etc ... The path for data loading will
 be: PATH_STREAMING+this.getName()+"_"chromoID+"Info"+this.EXTENSION;</div>
</section>
</li>
<li>
<section class="detail" id="read()">
<h3>read</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">read</span>()</div>
<div class="block">Load the entire TranscriptomeData</div>
<dl class="notes">
<dt>Throws:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/IOException.html" title="class or interface in java.io" class="external-link">IOException</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="save(boolean)">
<h3>save</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">save</span><wbr><span class="parameters">(boolean&nbsp;saveValues)</span></div>
<div class="block">Save stat and values in two separate files <br>
 Saving is done via DataStream to allow streaming reading</div>
<dl class="notes">
<dt>Throws:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/IOException.html" title="class or interface in java.io" class="external-link">IOException</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getUnlogValue(int,java.lang.String)">
<h3>getUnlogValue</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">double</span>&nbsp;<span class="element-name">getUnlogValue</span><wbr><span class="parameters">(int&nbsp;bpPosition,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;chromoID)</span></div>
<div class="block">Get value at a certain base pair position and unlog it by calculating 2^value<br>
 It works faster if data has been already loaded by this.read() or this.read(from,to)</div>
</section>
</li>
<li>
<section class="detail" id="convert(bacnet.datamodel.dataset.NGS,boolean)">
<h3>convert</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">convert</span><wbr><span class="parameters">(<a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&nbsp;data,
 boolean&nbsp;logTransformed)</span></div>
<div class="block">Run the conversion of wigfiles</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>data</code> - </dd>
<dd><code>logTransformed</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="convertWigFile(bacnet.datamodel.dataset.NGS,boolean)">
<h3>convertWigFile</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">convertWigFile</span><wbr><span class="parameters">(<a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&nbsp;data,
 boolean&nbsp;logTransformed)</span></div>
<div class="block">WIG file are common RNASeq data<br>
 Calculate first the MEDIAN value of the different replicates ad then log2 transform and save the
 NGS data Two columns: <br>
 <li>one for the base pair position
 <li>one for the coverage value</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>rawDataPath</code> - </dd>
<dd><code>genome</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="compare(bacnet.datamodel.dataset.NGS,boolean)">
<h3>compare</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></span>&nbsp;<span class="element-name">compare</span><wbr><span class="parameters">(<a href="NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&nbsp;data2,
 boolean&nbsp;calcData)</span></div>
<div class="block">Create RNASeq data from the comparison of this one and data2</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>data2</code> - </dd>
<dd><code>calcData</code> - true if we want to calculate the data</dd>
<dt>Returns:</dt>
<dt>Throws:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/IOException.html" title="class or interface in java.io" class="external-link">IOException</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="readWIGChromo(java.lang.String,java.lang.String,bacnet.datamodel.sequence.Genome)">
<h3>readWIGChromo</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</span>&nbsp;<span class="element-name">readWIGChromo</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;chromoName,
 <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</span></div>
<div class="block">Will read your wig file and parse it to BACNET <br>
 ONLY WIG FILE WITH variableStep and raw format are accepted No fixedStep format</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>fileName</code> - </dd>
<dd><code>chromoName</code> - </dd>
<dt>Returns:</dt>
<dd>list of position and the corresponding coverage on chromoName chromosome</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="isInfoRead()">
<h3>isInfoRead</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">boolean</span>&nbsp;<span class="element-name">isInfoRead</span>()</div>
</section>
</li>
<li>
<section class="detail" id="setInfoRead(boolean)">
<h3>setInfoRead</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setInfoRead</span><wbr><span class="parameters">(boolean&nbsp;infoRead)</span></div>
</section>
</li>
<li>
<section class="detail" id="getDatasets()">
<h3>getDatasets</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/HashMap.html" title="class or interface in java.util" class="external-link">HashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;</span>&nbsp;<span class="element-name">getDatasets</span>()</div>
</section>
</li>
<li>
<section class="detail" id="setDatasets(java.util.HashMap)">
<h3>setDatasets</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setDatasets</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/HashMap.html" title="class or interface in java.util" class="external-link">HashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;datasets)</span></div>
</section>
</li>
<li>
<section class="detail" id="getGenomeName()">
<h3>getGenomeName</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getGenomeName</span>()</div>
</section>
</li>
<li>
<section class="detail" id="setGenomeName(java.lang.String)">
<h3>setGenomeName</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setGenomeName</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName)</span></div>
</section>
</li>
<li>
<section class="detail" id="isLogTransformed()">
<h3>isLogTransformed</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">boolean</span>&nbsp;<span class="element-name">isLogTransformed</span>()</div>
</section>
</li>
<li>
<section class="detail" id="setLogTransformed(boolean)">
<h3>setLogTransformed</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setLogTransformed</span><wbr><span class="parameters">(boolean&nbsp;logTransformed)</span></div>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
</div>
</div>
</body>
</html>
