<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>NTermData</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="NTermData";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10,"i19":10,"i20":9,"i21":10,"i22":10,"i23":10,"i24":10,"i25":10,"i26":10,"i27":10,"i28":9,"i29":10,"i30":10,"i31":10,"i32":10,"i33":10,"i34":10,"i35":10,"i36":10,"i37":10,"i38":10,"i39":10,"i40":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/NTermData.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.dataset</a></div>
<h2 title="Class NTermData" class="title">Class NTermData</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li><a href="OmicsData.html" title="class in bacnet.datamodel.dataset">bacnet.datamodel.dataset.OmicsData</a></li>
<li>
<ul class="inheritance">
<li>bacnet.datamodel.dataset.NTermData</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><code>java.io.Serializable</code></dd>
</dl>
<hr>
<pre>public class <span class="typeNameLabel">NTermData</span>
extends <a href="OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a>
implements java.io.Serializable</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../serialized-form.html#bacnet.datamodel.dataset.NTermData">Serialized Form</a></dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a id="nested.classes.inherited.from.class.bacnet.datamodel.dataset.OmicsData">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;bacnet.datamodel.dataset.<a href="OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></h3>
<code><a href="OmicsData.ColNames.html" title="enum in bacnet.datamodel.dataset">OmicsData.ColNames</a>, <a href="OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></code></li>
</ul>
</li>
</ul>
</section>
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private <a href="../annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#annotation">annotation</a></span></code></th>
<td class="colLast">
<div class="block">An <code>Annotation</code> object to order NTerm by position in the Genome</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#aTISMap">aTISMap</a></span></code></th>
<td class="colLast">
<div class="block">Link gene to they corresponding aTIS if it exists</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.HashMap&lt;java.lang.String,&#8203;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#elements">elements</a></span></code></th>
<td class="colLast">
<div class="block">Link NTerm name to <code>NTerm</code><br>
 This correspond to the list of <code>NTerm</code></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#geneModification">geneModification</a></span></code></th>
<td class="colLast">
<div class="block">Link gene to a list of all <code>TypeModif</code> encountered for aTIS</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#genes">genes</a></span></code></th>
<td class="colLast">
<div class="block">Link lmo genes to NTerm overlapping it<br>
 NTerm with no association to an lmo are associated to "unknown" key</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#modifications">modifications</a></span></code></th>
<td class="colLast">
<div class="block">Link NTerm position (coded as a <code>String</code> to <code>ArrayList(NTerm)</code><br>
 This correspond to the list of mapped sequence which are equivalent to other NTerm but with
 different modification type</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#peptides">peptides</a></span></code></th>
<td class="colLast">
<div class="block">Link peptideSequence to <code>ArrayList(NTerm)</code><br>
 This correspond to the list of <code>NTerm</code> having same peptideSequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private static long</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#serialVersionUID">serialVersionUID</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;<a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#tisList">tisList</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.HashMap&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>,&#8203;<a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#tisMap">tisMap</a></span></code></th>
<td class="colLast">
<div class="block">Link <code>Nterm</code> to a list of all <code>Nterm</code> which might be related to it (e.g.:
 almost same sequence but proteolyse cut it)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#TYPE_OVERLAPS">TYPE_OVERLAPS</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#typeOverlaps">typeOverlaps</a></span></code></th>
<td class="colLast">
<div class="block">List of Overlap Type used for TIS classification = aTIS, internal TIS, etc...</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="fields.inherited.from.class.bacnet.datamodel.dataset.OmicsData">
<!--   -->
</a>
<h3>Fields inherited from class&nbsp;bacnet.datamodel.dataset.<a href="OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></h3>
<code><a href="OmicsData.html#EXTENSION">EXTENSION</a>, <a href="OmicsData.html#GENERAL_WT_DATA">GENERAL_WT_DATA</a>, <a href="OmicsData.html#GENERAL_WT_NAME">GENERAL_WT_NAME</a>, <a href="OmicsData.html#MISSING_VALUE">MISSING_VALUE</a>, <a href="OmicsData.html#PATH_COMPARISONS">PATH_COMPARISONS</a>, <a href="OmicsData.html#PATH_EXPR_NORM">PATH_EXPR_NORM</a>, <a href="OmicsData.html#PATH_EXPR_RAW">PATH_EXPR_RAW</a>, <a href="OmicsData.html#PATH_GENEXPR_COMPLETE">PATH_GENEXPR_COMPLETE</a>, <a href="OmicsData.html#PATH_GENEXPR_NORM">PATH_GENEXPR_NORM</a>, <a href="OmicsData.html#PATH_GENEXPR_RAW">PATH_GENEXPR_RAW</a>, <a href="OmicsData.html#PATH_NGS_NORM">PATH_NGS_NORM</a>, <a href="OmicsData.html#PATH_NGS_RAW">PATH_NGS_RAW</a>, <a href="OmicsData.html#PATH_NORM">PATH_NORM</a>, <a href="OmicsData.html#PATH_PROTEOMICS_NORM">PATH_PROTEOMICS_NORM</a>, <a href="OmicsData.html#PATH_PROTEOMICS_RAW">PATH_PROTEOMICS_RAW</a>, <a href="OmicsData.html#PATH_RAW">PATH_RAW</a>, <a href="OmicsData.html#PATH_STREAMING">PATH_STREAMING</a>, <a href="OmicsData.html#PATH_TILING_NORM">PATH_TILING_NORM</a>, <a href="OmicsData.html#PATH_TILING_RAW">PATH_TILING_RAW</a></code></li>
</ul>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">NTermData</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.lang.String)">NTermData</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.lang.String,java.util.HashMap)">NTermData</a></span>&#8203;(java.lang.String&nbsp;name,
         java.util.HashMap&lt;java.lang.String,&#8203;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;elements)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addElement(bacnet.datamodel.proteomics.NTerm)">addElement</a></span>&#8203;(<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm)</code></th>
<td class="colLast">
<div class="block">Add a NTerm in elements HashMap:<br>
 key = nTerm.getName()<br>
 value = nTerm</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createAnnotation()">createAnnotation</a></span>()</code></th>
<td class="colLast">
<div class="block">Create an <code>Annotation</code> Table from the list of <code>Nterm</code></div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createGeneHashMap()">createGeneHashMap</a></span>()</code></th>
<td class="colLast">
<div class="block">Regroup NTerm using genes of EGD-e<br>
 Each EGD-e gene will be associated to a list of NTerm overlapping it<br>
 Fill: HashMap<String, ArrayList<NTerm>>() aTISMap, HashMap<String, ArrayList<NTerm>>() genes,
 HashMap<String, ArrayList<TypeModif>>() geneModification</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createHashMap()">createHashMap</a></span>()</code></th>
<td class="colLast">
<div class="block">Regroup in different HashMap the NTerm<br>
 Fill:<br>
 HashMap<String, ArrayList<NTerm>> peptides is to make a simple HashMap between NTerm sequence
 and a list of corresponding <code>NTerm</code>
 HashMap<String, ArrayList<NTerm>> modifications links Nterm position to a list of
 corresponding <code>NTerm</code></div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createProteolyseMap()">createProteolyseMap</a></span>()</code></th>
<td class="colLast">
<div class="block">Regroup peptides which overlap -> the assumption is that they come from the same protein but were
 created from proteolyse</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code><a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#findNextNterm(int,java.lang.String)">findNextNterm</a></span>&#8203;(int&nbsp;bpPosition,
             java.lang.String&nbsp;typeOverlap)</code></th>
<td class="colLast">
<div class="block">Given a position in the genome, give the NTerm just downstream to it</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code><a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#findPreviousNterm(int,java.lang.String)">findPreviousNterm</a></span>&#8203;(int&nbsp;bpPosition,
                 java.lang.String&nbsp;typeOverlap)</code></th>
<td class="colLast">
<div class="block">Given a position in the genome, give the NTerm just upstream to it</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code><a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#findSameNterm(int,java.lang.String)">findSameNterm</a></span>&#8203;(int&nbsp;bpPosition,
             java.lang.String&nbsp;typeOverlap)</code></th>
<td class="colLast">
<div class="block">Given a position in the genome, give the NTerm just upstream to it</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code><a href="../annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAnnotation()">getAnnotation</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getaTISMap()">getaTISMap</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>java.util.HashMap&lt;java.lang.String,&#8203;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getElements()">getElements</a></span>()</code></th>
<td class="colLast">
<div class="block">Link NTerm name to <code>NTerm</code><br>
 This correspond to the list of <code>NTerm</code></div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getElementsToDisplay(int,int,java.lang.String)">getElementsToDisplay</a></span>&#8203;(int&nbsp;beginDraw,
                    int&nbsp;endDraw,
                    java.lang.String&nbsp;nTermTypeOverlap)</code></th>
<td class="colLast">
<div class="block">Get the lists of NTerm between beginDraw and endDraw</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGeneModification()">getGeneModification</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGenes()">getGenes</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getModifications()">getModifications</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code><a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getNextNterm(bacnet.datamodel.proteomics.NTerm,java.lang.String)">getNextNterm</a></span>&#8203;(<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
            java.lang.String&nbsp;typeOverlap)</code></th>
<td class="colLast">
<div class="block">Given a NTerm, find the next one in the Annotation</div>
</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getNTerms()">getNTerms</a></span>()</code></th>
<td class="colLast">
<div class="block">Return a list of all NTerms available</div>
</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getPeptides()">getPeptides</a></span>()</code></th>
<td class="colLast">
<div class="block">Link peptideSequence to <code>ArrayList(NTerm)</code><br>
 This correspond to the list of <code>NTerm</code> having same peptideSequence</div>
</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code><a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getPreviousNterm(bacnet.datamodel.proteomics.NTerm,java.lang.String)">getPreviousNterm</a></span>&#8203;(<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
                java.lang.String&nbsp;typeOverlap)</code></th>
<td class="colLast">
<div class="block">Given a NTerm, find the previous one in the Annotation</div>
</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code><a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteolyseAssociated(bacnet.datamodel.proteomics.NTerm)">getProteolyseAssociated</a></span>&#8203;(<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm)</code></th>
<td class="colLast">
<div class="block">Go through <code>proteolyses</code> map, and find the list of NTerm which might be associated to
 an input <code>NTerm</code><br>
 Warning: <code>ArrayList</code> in return contains <code>key NTerm</code> of
 <code>proteolyses</code> but not the input <code>nTerm</code></div>
</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code>static long</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSerialversionuid()">getSerialversionuid</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getStat()">getStat</a></span>()</code></th>
<td class="colLast">
<div class="block">Sysout some statistics about the number of aTIS, dTISM, dTIS, and uTIS</div>
</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTisList()">getTisList</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><code>java.util.HashMap&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>,&#8203;<a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTisMap()">getTisMap</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTIStoDisplay(java.util.ArrayList)">getTIStoDisplay</a></span>&#8203;(java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;nTerms)</code></th>
<td class="colLast">
<div class="block">Return the list of from a List of NTerm</div>
</td>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><code>java.util.HashMap&lt;java.lang.String,&#8203;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTypeOfTIS(java.lang.String)">getTypeOfTIS</a></span>&#8203;(java.lang.String&nbsp;typeOverlap)</code></th>
<td class="colLast">
<div class="block">Return a HashMap of <code>NTerm</code> with <code>nTerm.getTypeOverlap() == typeOverlap</code></div>
</td>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTypeOverlaps()">getTypeOverlaps</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i27" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#load()">load</a></span>()</code></th>
<td class="colLast">
<div class="block">Load NTermData from PATH_STREAMING+ this.getName() +EXTENSION</div>
</td>
</tr>
<tr id="i28" class="altColor">
<td class="colFirst"><code>static <a href="NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#load(java.lang.String)">load</a></span>&#8203;(java.lang.String&nbsp;nTermExpName)</code></th>
<td class="colLast">
<div class="block">Load in: TranscriptomeData.PATH_STREAMING + nTermExpName + TranscriptomeData.EXTENSION
 
 Read compressed, serialized data with a FileInputStream.</div>
</td>
</tr>
<tr id="i29" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#reorderModif(java.util.ArrayList,bacnet.datamodel.proteomics.NTerm)">reorderModif</a></span>&#8203;(java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;nTermDisplay,
            <a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTermHighlight)</code></th>
<td class="colLast">
<div class="block">Reorder the Nterm to display AcD3 first, then Ace, and For.</div>
</td>
</tr>
<tr id="i30" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#save()">save</a></span>()</code></th>
<td class="colLast">
<div class="block">Save in: TranscriptomeData.PATH_STREAMING + this.getName() + TranscriptomeData.EXTENSION
 
 Serialize the vector of lines with an ObjectOutputStream.</div>
</td>
</tr>
<tr id="i31" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setAnnotation(bacnet.datamodel.annotation.Annotation)">setAnnotation</a></span>&#8203;(<a href="../annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a>&nbsp;annotation)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i32" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setaTISMap(java.util.HashMap)">setaTISMap</a></span>&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;&nbsp;aTISMap)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i33" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setElements(java.util.HashMap)">setElements</a></span>&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;elements)</code></th>
<td class="colLast">
<div class="block">Link NTerm name to <code>NTerm</code><br>
 This correspond to the list of <code>NTerm</code></div>
</td>
</tr>
<tr id="i34" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGeneModification(java.util.HashMap)">setGeneModification</a></span>&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a>&gt;&gt;&nbsp;geneModification)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i35" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGenes(java.util.HashMap)">setGenes</a></span>&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;&nbsp;genes)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i36" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setModifications(java.util.HashMap)">setModifications</a></span>&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;&nbsp;modifications)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i37" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setPeptides(java.util.HashMap)">setPeptides</a></span>&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;&nbsp;peptides)</code></th>
<td class="colLast">
<div class="block">Link peptideSequence to <code>ArrayList(NTerm)</code><br>
 This correspond to the list of <code>NTerm</code> having same peptideSequence</div>
</td>
</tr>
<tr id="i38" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setTisList(java.util.ArrayList)">setTisList</a></span>&#8203;(java.util.ArrayList&lt;<a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt;&nbsp;tisList)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i39" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setTisMap(java.util.HashMap)">setTisMap</a></span>&#8203;(java.util.HashMap&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>,&#8203;<a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt;&nbsp;tisMap)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i40" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setTypeOverlaps(java.util.ArrayList)">setTypeOverlaps</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;typeOverlaps)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.bacnet.datamodel.dataset.OmicsData">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;bacnet.datamodel.dataset.<a href="OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></h3>
<code><a href="OmicsData.html#getBioCondName()">getBioCondName</a>, <a href="OmicsData.html#getDate()">getDate</a>, <a href="OmicsData.html#getExtension()">getExtension</a>, <a href="OmicsData.html#getName()">getName</a>, <a href="OmicsData.html#getNote()">getNote</a>, <a href="OmicsData.html#getRawDatas()">getRawDatas</a>, <a href="OmicsData.html#getType()">getType</a>, <a href="OmicsData.html#initStaticVariables()">initStaticVariables</a>, <a href="OmicsData.html#setBioCondName(java.lang.String)">setBioCondName</a>, <a href="OmicsData.html#setDate(java.lang.String)">setDate</a>, <a href="OmicsData.html#setName(java.lang.String)">setName</a>, <a href="OmicsData.html#setNote(java.lang.String)">setNote</a>, <a href="OmicsData.html#setRawDatas(java.util.ArrayList)">setRawDatas</a>, <a href="OmicsData.html#setType(bacnet.datamodel.dataset.OmicsData.TypeData)">setType</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="serialVersionUID">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>serialVersionUID</h4>
<pre>private static final&nbsp;long serialVersionUID</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../constant-values.html#bacnet.datamodel.dataset.NTermData.serialVersionUID">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a id="TYPE_OVERLAPS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE_OVERLAPS</h4>
<pre>public static final&nbsp;java.lang.String[] TYPE_OVERLAPS</pre>
</li>
</ul>
<a id="elements">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>elements</h4>
<pre>private&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt; elements</pre>
<div class="block">Link NTerm name to <code>NTerm</code><br>
 This correspond to the list of <code>NTerm</code></div>
</li>
</ul>
<a id="peptides">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>peptides</h4>
<pre>private&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt; peptides</pre>
<div class="block">Link peptideSequence to <code>ArrayList(NTerm)</code><br>
 This correspond to the list of <code>NTerm</code> having same peptideSequence</div>
</li>
</ul>
<a id="modifications">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>modifications</h4>
<pre>private&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt; modifications</pre>
<div class="block">Link NTerm position (coded as a <code>String</code> to <code>ArrayList(NTerm)</code><br>
 This correspond to the list of mapped sequence which are equivalent to other NTerm but with
 different modification type</div>
</li>
</ul>
<a id="genes">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>genes</h4>
<pre>private&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt; genes</pre>
<div class="block">Link lmo genes to NTerm overlapping it<br>
 NTerm with no association to an lmo are associated to "unknown" key</div>
</li>
</ul>
<a id="aTISMap">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>aTISMap</h4>
<pre>private&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt; aTISMap</pre>
<div class="block">Link gene to they corresponding aTIS if it exists</div>
</li>
</ul>
<a id="geneModification">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>geneModification</h4>
<pre>private&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a>&gt;&gt; geneModification</pre>
<div class="block">Link gene to a list of all <code>TypeModif</code> encountered for aTIS</div>
</li>
</ul>
<a id="tisMap">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>tisMap</h4>
<pre>private&nbsp;java.util.HashMap&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>,&#8203;<a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt; tisMap</pre>
<div class="block">Link <code>Nterm</code> to a list of all <code>Nterm</code> which might be related to it (e.g.:
 almost same sequence but proteolyse cut it)</div>
</li>
</ul>
<a id="tisList">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>tisList</h4>
<pre>private&nbsp;java.util.ArrayList&lt;<a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt; tisList</pre>
</li>
</ul>
<a id="typeOverlaps">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>typeOverlaps</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.String&gt; typeOverlaps</pre>
<div class="block">List of Overlap Type used for TIS classification = aTIS, internal TIS, etc...</div>
</li>
</ul>
<a id="annotation">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>annotation</h4>
<pre>private&nbsp;<a href="../annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a> annotation</pre>
<div class="block">An <code>Annotation</code> object to order NTerm by position in the Genome</div>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>NTermData</h4>
<pre>public&nbsp;NTermData()</pre>
</li>
</ul>
<a id="&lt;init&gt;(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>NTermData</h4>
<pre>public&nbsp;NTermData&#8203;(java.lang.String&nbsp;name)</pre>
</li>
</ul>
<a id="&lt;init&gt;(java.lang.String,java.util.HashMap)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>NTermData</h4>
<pre>public&nbsp;NTermData&#8203;(java.lang.String&nbsp;name,
                 java.util.HashMap&lt;java.lang.String,&#8203;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;elements)</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="addElement(bacnet.datamodel.proteomics.NTerm)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addElement</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;addElement&#8203;(<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm)</pre>
<div class="block">Add a NTerm in elements HashMap:<br>
 key = nTerm.getName()<br>
 value = nTerm</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nTerm</code> - </dd>
</dl>
</li>
</ul>
<a id="createHashMap()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createHashMap</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;createHashMap()</pre>
<div class="block">Regroup in different HashMap the NTerm<br>
 Fill:<br>
 <li>HashMap<String, ArrayList<NTerm>> peptides is to make a simple HashMap between NTerm sequence
 and a list of corresponding <code>NTerm</code>
 <li>HashMap<String, ArrayList<NTerm>> modifications links Nterm position to a list of
 corresponding <code>NTerm</code></div>
</li>
</ul>
<a id="createGeneHashMap()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createGeneHashMap</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;createGeneHashMap()</pre>
<div class="block">Regroup NTerm using genes of EGD-e<br>
 Each EGD-e gene will be associated to a list of NTerm overlapping it<br>
 Fill: HashMap<String, ArrayList<NTerm>>() aTISMap, HashMap<String, ArrayList<NTerm>>() genes,
 HashMap<String, ArrayList<TypeModif>>() geneModification</div>
</li>
</ul>
<a id="createProteolyseMap()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createProteolyseMap</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;createProteolyseMap()</pre>
<div class="block">Regroup peptides which overlap -> the assumption is that they come from the same protein but were
 created from proteolyse</div>
</li>
</ul>
<a id="getProteolyseAssociated(bacnet.datamodel.proteomics.NTerm)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteolyseAssociated</h4>
<pre class="methodSignature">public&nbsp;<a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&nbsp;getProteolyseAssociated&#8203;(<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm)</pre>
<div class="block">Go through <code>proteolyses</code> map, and find the list of NTerm which might be associated to
 an input <code>NTerm</code><br>
 Warning: <code>ArrayList</code> in return contains <code>key NTerm</code> of
 <code>proteolyses</code> but not the input <code>nTerm</code></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nTerm</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="createAnnotation()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createAnnotation</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;createAnnotation()</pre>
<div class="block">Create an <code>Annotation</code> Table from the list of <code>Nterm</code></div>
</li>
</ul>
<a id="getNTerms()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNTerms</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;getNTerms()</pre>
<div class="block">Return a list of all NTerms available</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getTypeOfTIS(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTypeOfTIS</h4>
<pre class="methodSignature">public&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;getTypeOfTIS&#8203;(java.lang.String&nbsp;typeOverlap)</pre>
<div class="block">Return a HashMap of <code>NTerm</code> with <code>nTerm.getTypeOverlap() == typeOverlap</code></div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getElementsToDisplay(int,int,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getElementsToDisplay</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;getElementsToDisplay&#8203;(int&nbsp;beginDraw,
                                                       int&nbsp;endDraw,
                                                       java.lang.String&nbsp;nTermTypeOverlap)</pre>
<div class="block">Get the lists of NTerm between beginDraw and endDraw</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>chromosome</code> - </dd>
<dd><code>beginDraw</code> - </dd>
<dd><code>endDraw</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getTIStoDisplay(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTIStoDisplay</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt;&nbsp;getTIStoDisplay&#8203;(java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;nTerms)</pre>
<div class="block">Return the list of from a List of NTerm</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nTerms</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="reorderModif(java.util.ArrayList,bacnet.datamodel.proteomics.NTerm)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reorderModif</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;reorderModif&#8203;(java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;nTermDisplay,
                                               <a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTermHighlight)</pre>
<div class="block">Reorder the Nterm to display AcD3 first, then Ace, and For.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nTermDisplay</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="findSameNterm(int,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>findSameNterm</h4>
<pre class="methodSignature">public&nbsp;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;findSameNterm&#8203;(int&nbsp;bpPosition,
                           java.lang.String&nbsp;typeOverlap)</pre>
<div class="block">Given a position in the genome, give the NTerm just upstream to it</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bpPosition</code> - </dd>
<dd><code>typEOverlap</code> - type of NTerm to display</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getPreviousNterm(bacnet.datamodel.proteomics.NTerm,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPreviousNterm</h4>
<pre class="methodSignature">public&nbsp;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;getPreviousNterm&#8203;(<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
                              java.lang.String&nbsp;typeOverlap)</pre>
<div class="block">Given a NTerm, find the previous one in the Annotation</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nTerm</code> - </dd>
<dd><code>typEOverlap</code> - type of NTerm to display</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="findPreviousNterm(int,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>findPreviousNterm</h4>
<pre class="methodSignature">public&nbsp;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;findPreviousNterm&#8203;(int&nbsp;bpPosition,
                               java.lang.String&nbsp;typeOverlap)</pre>
<div class="block">Given a position in the genome, give the NTerm just upstream to it</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bpPosition</code> - </dd>
<dd><code>typEOverlap</code> - type of NTerm to display</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="findNextNterm(int,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>findNextNterm</h4>
<pre class="methodSignature">public&nbsp;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;findNextNterm&#8203;(int&nbsp;bpPosition,
                           java.lang.String&nbsp;typeOverlap)</pre>
<div class="block">Given a position in the genome, give the NTerm just downstream to it</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bpPosition</code> - </dd>
<dd><code>typEOverlap</code> - type of NTerm to display</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getNextNterm(bacnet.datamodel.proteomics.NTerm,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNextNterm</h4>
<pre class="methodSignature">public&nbsp;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;getNextNterm&#8203;(<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
                          java.lang.String&nbsp;typeOverlap)</pre>
<div class="block">Given a NTerm, find the next one in the Annotation</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nTerm</code> - </dd>
<dd><code>typEOverlap</code> - type of NTerm to display</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getStat()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getStat</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;getStat()</pre>
<div class="block">Sysout some statistics about the number of aTIS, dTISM, dTIS, and uTIS</div>
</li>
</ul>
<a id="load()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>load</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;load()</pre>
<div class="block">Load NTermData from PATH_STREAMING+ this.getName() +EXTENSION</div>
</li>
</ul>
<a id="load(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>load</h4>
<pre class="methodSignature">public static&nbsp;<a href="NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&nbsp;load&#8203;(java.lang.String&nbsp;nTermExpName)</pre>
<div class="block">Load in: TranscriptomeData.PATH_STREAMING + nTermExpName + TranscriptomeData.EXTENSION
 
 Read compressed, serialized data with a FileInputStream. Uncompress that data with a
 GZIPInputStream. Deserialize the vector of lines with a ObjectInputStream. Replace current data
 with new data, and redraw everything.</div>
</li>
</ul>
<a id="save()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>save</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;save()</pre>
<div class="block">Save in: TranscriptomeData.PATH_STREAMING + this.getName() + TranscriptomeData.EXTENSION
 
 Serialize the vector of lines with an ObjectOutputStream. Compress the serialized objects with a
 GZIPOutputStream. Write the compressed, serialized data to a file with a FileOutputStream. Don't
 forget to flush and close the stream.</div>
</li>
</ul>
<a id="getElements()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getElements</h4>
<pre class="methodSignature">public&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;getElements()</pre>
<div class="block">Link NTerm name to <code>NTerm</code><br>
 This correspond to the list of <code>NTerm</code></div>
</li>
</ul>
<a id="setElements(java.util.HashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setElements</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setElements&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;elements)</pre>
<div class="block">Link NTerm name to <code>NTerm</code><br>
 This correspond to the list of <code>NTerm</code></div>
</li>
</ul>
<a id="getPeptides()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPeptides</h4>
<pre class="methodSignature">public&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;&nbsp;getPeptides()</pre>
<div class="block">Link peptideSequence to <code>ArrayList(NTerm)</code><br>
 This correspond to the list of <code>NTerm</code> having same peptideSequence</div>
</li>
</ul>
<a id="setPeptides(java.util.HashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setPeptides</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setPeptides&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;&nbsp;peptides)</pre>
<div class="block">Link peptideSequence to <code>ArrayList(NTerm)</code><br>
 This correspond to the list of <code>NTerm</code> having same peptideSequence</div>
</li>
</ul>
<a id="getModifications()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getModifications</h4>
<pre class="methodSignature">public&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;&nbsp;getModifications()</pre>
</li>
</ul>
<a id="setModifications(java.util.HashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setModifications</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setModifications&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;&nbsp;modifications)</pre>
</li>
</ul>
<a id="getGenes()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenes</h4>
<pre class="methodSignature">public&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;&nbsp;getGenes()</pre>
</li>
</ul>
<a id="setGenes(java.util.HashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGenes</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGenes&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;&nbsp;genes)</pre>
</li>
</ul>
<a id="getGeneModification()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneModification</h4>
<pre class="methodSignature">public&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a>&gt;&gt;&nbsp;getGeneModification()</pre>
</li>
</ul>
<a id="setGeneModification(java.util.HashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGeneModification</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGeneModification&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a>&gt;&gt;&nbsp;geneModification)</pre>
</li>
</ul>
<a id="getSerialversionuid()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSerialversionuid</h4>
<pre class="methodSignature">public static&nbsp;long&nbsp;getSerialversionuid()</pre>
</li>
</ul>
<a id="setaTISMap(java.util.HashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setaTISMap</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setaTISMap&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;&nbsp;aTISMap)</pre>
</li>
</ul>
<a id="getaTISMap()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getaTISMap</h4>
<pre class="methodSignature">public&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt;&nbsp;getaTISMap()</pre>
</li>
</ul>
<a id="getAnnotation()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotation</h4>
<pre class="methodSignature">public&nbsp;<a href="../annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a>&nbsp;getAnnotation()</pre>
</li>
</ul>
<a id="getTypeOverlaps()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTypeOverlaps</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getTypeOverlaps()</pre>
</li>
</ul>
<a id="setTypeOverlaps(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setTypeOverlaps</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setTypeOverlaps&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;typeOverlaps)</pre>
</li>
</ul>
<a id="setAnnotation(bacnet.datamodel.annotation.Annotation)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAnnotation</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setAnnotation&#8203;(<a href="../annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a>&nbsp;annotation)</pre>
</li>
</ul>
<a id="getTisMap()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTisMap</h4>
<pre class="methodSignature">public&nbsp;java.util.HashMap&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>,&#8203;<a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt;&nbsp;getTisMap()</pre>
</li>
</ul>
<a id="setTisMap(java.util.HashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setTisMap</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setTisMap&#8203;(java.util.HashMap&lt;<a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>,&#8203;<a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt;&nbsp;tisMap)</pre>
</li>
</ul>
<a id="getTisList()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTisList</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt;&nbsp;getTisList()</pre>
</li>
</ul>
<a id="setTisList(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setTisList</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setTisList&#8203;(java.util.ArrayList&lt;<a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt;&nbsp;tisList)</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/NTermData.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
