<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>org.biojava3.core.sequence.io</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="declaration: package: org.biojava3.core.sequence.io">
<meta name="generator" content="javadoc/PackageWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="package-declaration-page">
<script type="text/javascript">var pathtoroot = "../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li class="nav-bar-cell1-rev">Package</li>
<li>Class</li>
<li><a href="package-use.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../help-doc.html#package">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div>
<ul class="sub-nav-list">
<li>Package:&nbsp;</li>
<li>Description&nbsp;|&nbsp;</li>
<li><a href="#related-package-summary">Related Packages</a>&nbsp;|&nbsp;</li>
<li><a href="#class-summary">Classes and Interfaces</a></li>
</ul>
</div>
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Package org.biojava3.core.sequence.io" class="title">Package org.biojava3.core.sequence.io</h1>
</div>
<hr>
<div class="package-signature">package <span class="element-name">org.biojava3.core.sequence.io</span></div>
<section class="summary">
<ul class="summary-list">
<li>
<div id="related-package-summary">
<div class="caption"><span>Related Packages</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="../package-summary.html">org.biojava3.core.sequence</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="template/package-summary.html">org.biojava3.core.sequence.io.template</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="util/package-summary.html">org.biojava3.core.sequence.io.util</a></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</div>
</li>
<li>
<div id="class-summary">
<div class="caption"><span>Classes</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">Need to keep track of actual bytes read and take advantage of buffered reader performance.</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">A sequence creator which preserves the case of its input string in the user collection of the
 returned ProteinSequence.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="DNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">DNASequenceCreator</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">A helper class that allows different ways to read a string and create a DNA sequence.</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="FastaGeneWriter.html" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">A Gene sequence has a Positive or Negative Strand where we want to write out to a stream the 5 to
 3 prime version.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="FastaReader.html" title="class in org.biojava3.core.sequence.io">FastaReader</a>&lt;S extends <a href="../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,<wbr>C extends <a href="../template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">Use FastaReaderHelper as an example of how to use this class where FastaReaderHelper should be
 the primary class used to read Fasta files</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab2">&nbsp;</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="FastaSequenceParser.html" title="class in org.biojava3.core.sequence.io">FastaSequenceParser</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">Used to parse a stream of a fasta file to get the sequence</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a>&lt;S extends <a href="../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,<wbr>C extends <a href="../template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">The FastaWriter writes a collection of sequences to an outputStream.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="FastaWriterHelper.html" title="class in org.biojava3.core.sequence.io">FastaWriterHelper</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">The class that should be used to write out fasta file of a sequence collection</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="FileProxyDNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyDNASequenceCreator</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the
 stream the sequence and the offset index are passed to create a Protein sequence that will be
 loaded in lazily.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="FileProxyProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyProteinSequenceCreator</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the
 stream the sequence and the offset index are passed to create a Protein sequence that will be
 loaded in lazily.</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="GenericFastaHeaderFormat.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderFormat</a>&lt;S extends <a href="../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;?&gt;,<wbr>C extends <a href="../template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">We store the original header if the sequence is parsed from a fasta file and will use that exact
 sequence if we write out the sequences to a fasta file.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="GenericFastaHeaderParser.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser</a>&lt;S extends <a href="../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">The default fasta header parser where some headers are well defined based on the source database
 which allows us to set the source of the protein sequence and the identifier that can be used in
 future implementations to load features from external sources
 
 If the user has a custom header with local data then they can create their own implementation of
 a FastaHeaderParserInterface

 GenBank gi|gi-number|gb|accession|locus ENA Data Library gi|gi-number|emb|accession|locus DDBJ,
 DNA Database of Japan gi|gi-number|dbj|accession|locus NBRF PIR pir||entry Protein Research
 Foundation prf||name SWISS-PROT sp|accession|name Brookhaven Protein Data Bank (1)
 pdb|entry|chain Brookhaven Protein Data Bank (2) entry:chain|PDBID|CHAIN|SEQUENCE PDB EBI
 PDB:1ECY_A mol:protein length:142 ECOTIN Patents pat|country|number GenInfo Backbone Id
 bbs|number General database identifier gnl|database|identifier NCBI Reference Sequence
 ref|accession|locus Local Sequence identifier lcl|identifier</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">Available translations

 
 1 - UNIVERSAL
 2 - VERTEBRATE_MITOCHONDRIAL
 3 - YEAST_MITOCHONDRIAL
 4 - MOLD_MITOCHONDRIAL
 5 - INVERTEBRATE_MITOCHONDRIAL
 6 - CILIATE_NUCLEAR
 9 - ECHINODERM_MITOCHONDRIAL
 10 - EUPLOTID_NUCLEAR
 11 - BACTERIAL
 12 - ALTERNATIVE_YEAST_NUCLEAR
 13 - ASCIDIAN_MITOCHONDRIAL
 14 - FLATWORM_MITOCHONDRIAL
 15 - BLEPHARISMA_MACRONUCLEAR
 16 - 2CHLOROPHYCEAN_MITOCHONDRIAL
 21 - TREMATODE_MITOCHONDRIAL
 23 - SCENEDESMUS_MITOCHONDRIAL
 

 Taken from <a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c">NCBI</a>
 with slight modification and put into the classpath resource.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">Holds the concept of a codon table from the IUPAC format</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="ProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">ProteinSequenceCreator</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">Used to create a ProteinSequence from a String to allow for details about the location of the
 sequence etc.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="RNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">RNASequenceCreator</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">Used to create a RNA sequence</div>
</div>
</div>
</div>
</li>
</ul>
</section>
</main>
</div>
</div>
</body>
</html>
