<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>org.biojava3.core.sequence.io</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="org.biojava3.core.sequence.io";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li class="navBarCell1Rev">Package</li>
<li>Class</li>
<li><a href="package-use.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h1 title="Package" class="title">Package&nbsp;org.biojava3.core.sequence.io</h1>
</div>
<div class="contentContainer">
<ul class="blockList">
<li class="blockList">
<table class="typeSummary">
<caption><span>Class Summary</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></th>
<td class="colLast">
<div class="block">Need to keep track of actual bytes read and take advantage of buffered reader performance.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></th>
<td class="colLast">
<div class="block">A sequence creator which preserves the case of its input string in the user collection of the
 returned ProteinSequence.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="DNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">DNASequenceCreator</a></th>
<td class="colLast">
<div class="block">A helper class that allows different ways to read a string and create a DNA sequence.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="FastaGeneWriter.html" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</a></th>
<td class="colLast">
<div class="block">A Gene sequence has a Positive or Negative Strand where we want to write out to a stream the 5 to
 3 prime version.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="FastaReader.html" title="class in org.biojava3.core.sequence.io">FastaReader</a>&lt;S extends <a href="../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</th>
<td class="colLast">
<div class="block">Use FastaReaderHelper as an example of how to use this class where FastaReaderHelper should be
 the primary class used to read Fasta files</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="FastaSequenceParser.html" title="class in org.biojava3.core.sequence.io">FastaSequenceParser</a></th>
<td class="colLast">
<div class="block">Used to parse a stream of a fasta file to get the sequence</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a>&lt;S extends <a href="../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</th>
<td class="colLast">
<div class="block">The FastaWriter writes a collection of sequences to an outputStream.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="FastaWriterHelper.html" title="class in org.biojava3.core.sequence.io">FastaWriterHelper</a></th>
<td class="colLast">
<div class="block">The class that should be used to write out fasta file of a sequence collection</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="FileProxyDNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyDNASequenceCreator</a></th>
<td class="colLast">
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the
 stream the sequence and the offset index are passed to create a Protein sequence that will be
 loaded in lazily.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="FileProxyProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyProteinSequenceCreator</a></th>
<td class="colLast">
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the
 stream the sequence and the offset index are passed to create a Protein sequence that will be
 loaded in lazily.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="GenericFastaHeaderFormat.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderFormat</a>&lt;S extends <a href="../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;?&gt;,&#8203;C extends <a href="../template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</th>
<td class="colLast">
<div class="block">We store the original header if the sequence is parsed from a fasta file and will use that exact
 sequence if we write out the sequences to a fasta file.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="GenericFastaHeaderParser.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser</a>&lt;S extends <a href="../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</th>
<td class="colLast">
<div class="block">The default fasta header parser where some headers are well defined based on the source database
 which allows us to set the source of the protein sequence and the identifier that can be used in
 future implementations to load features from external sources
 
 If the user has a custom header with local data then they can create their own implementation of
 a FastaHeaderParserInterface

 GenBank gi|gi-number|gb|accession|locus ENA Data Library gi|gi-number|emb|accession|locus DDBJ,
 DNA Database of Japan gi|gi-number|dbj|accession|locus NBRF PIR pir||entry Protein Research
 Foundation prf||name SWISS-PROT sp|accession|name Brookhaven Protein Data Bank (1)
 pdb|entry|chain Brookhaven Protein Data Bank (2) entry:chain|PDBID|CHAIN|SEQUENCE PDB EBI
 PDB:1ECY_A mol:protein length:142 ECOTIN Patents pat|country|number GenInfo Backbone Id
 bbs|number General database identifier gnl|database|identifier NCBI Reference Sequence
 ref|accession|locus Local Sequence identifier lcl|identifier</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></th>
<td class="colLast">
<div class="block">Available translations

 
 1 - UNIVERSAL
 2 - VERTEBRATE_MITOCHONDRIAL
 3 - YEAST_MITOCHONDRIAL
 4 - MOLD_MITOCHONDRIAL
 5 - INVERTEBRATE_MITOCHONDRIAL
 6 - CILIATE_NUCLEAR
 9 - ECHINODERM_MITOCHONDRIAL
 10 - EUPLOTID_NUCLEAR
 11 - BACTERIAL
 12 - ALTERNATIVE_YEAST_NUCLEAR
 13 - ASCIDIAN_MITOCHONDRIAL
 14 - FLATWORM_MITOCHONDRIAL
 15 - BLEPHARISMA_MACRONUCLEAR
 16 - 2CHLOROPHYCEAN_MITOCHONDRIAL
 21 - TREMATODE_MITOCHONDRIAL
 23 - SCENEDESMUS_MITOCHONDRIAL
 

 Taken from <a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c">NCBI</a>
 with slight modification and put into the classpath resource.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="IUPACParser.IOD.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IOD</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a></th>
<td class="colLast">
<div class="block">Holds the concept of a codon table from the IUPAC format</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="ProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">ProteinSequenceCreator</a></th>
<td class="colLast">
<div class="block">Used to create a ProteinSequence from a String to allow for details about the location of the
 sequence etc.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="RNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">RNASequenceCreator</a></th>
<td class="colLast">
<div class="block">Used to create a RNA sequence</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li class="navBarCell1Rev">Package</li>
<li>Class</li>
<li><a href="package-use.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
