<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>TrackCanvasData</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="TrackCanvasData";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10,"i19":10,"i20":10,"i21":10,"i22":10,"i23":10,"i24":10,"i25":10,"i26":10,"i27":10,"i28":10,"i29":10,"i30":10,"i31":10,"i32":10,"i33":10,"i34":10,"i35":10,"i36":10,"i37":10,"i38":10,"i39":10,"i40":10,"i41":10,"i42":10,"i43":10,"i44":10,"i45":10,"i46":10,"i47":10,"i48":10,"i49":10,"i50":10,"i51":10,"i52":10,"i53":10,"i54":10,"i55":10,"i56":10,"i57":10,"i58":10,"i59":10,"i60":10,"i61":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/TrackCanvasData.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.genomeBrowser.tracksGUI</a></div>
<h2 title="Class TrackCanvasData" class="title">Class TrackCanvasData</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>org.eclipse.swt.widgets.Widget</li>
<li>
<ul class="inheritance">
<li>org.eclipse.swt.widgets.Control</li>
<li>
<ul class="inheritance">
<li>org.eclipse.swt.widgets.Scrollable</li>
<li>
<ul class="inheritance">
<li>org.eclipse.swt.widgets.Composite</li>
<li>
<ul class="inheritance">
<li>org.eclipse.swt.widgets.Canvas</li>
<li>
<ul class="inheritance">
<li>bacnet.genomeBrowser.tracksGUI.TrackCanvasData</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><code>java.io.Serializable</code>, <code>java.util.EventListener</code>, <code>org.eclipse.rap.rwt.Adaptable</code>, <code>org.eclipse.swt.events.MouseMoveListener</code>, <code>org.eclipse.swt.graphics.Drawable</code>, <code>org.eclipse.swt.internal.SerializableCompatibility</code>, <code>org.eclipse.swt.internal.SWTEventListener</code></dd>
</dl>
<hr>
<pre>public class <span class="typeNameLabel">TrackCanvasData</span>
extends org.eclipse.swt.widgets.Canvas
implements org.eclipse.swt.events.MouseMoveListener</pre>
<div class="block">A TrackCanvasData is the Canvas in which will be displayed datasets</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>christophebecavin</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../serialized-form.html#bacnet.genomeBrowser.tracksGUI.TrackCanvasData">Serialized Form</a></dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private <a href="../../datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#annotation">annotation</a></span></code></th>
<td class="colLast">
<div class="block"><code>Annotation</code> data for all Genome information</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#bpSizeH">bpSizeH</a></span></code></th>
<td class="colLast">
<div class="block">Width in pixel of one base pair (fundamental unit for the rest of the display)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#chromoID">chromoID</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#dataSizeReference">dataSizeReference</a></span></code></th>
<td class="colLast">
<div class="block">Height of on data</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#dataWindowSize">dataWindowSize</a></span></code></th>
<td class="colLast">
<div class="block">Height of all the data window = all data combine</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#decaySliderVBar">decaySliderVBar</a></span></code></th>
<td class="colLast">
<div class="block">Vertical decay imposed by the vertical slider</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayMouseLine">displayMouseLine</a></span></code></th>
<td class="colLast">
<div class="block">If true we display a vertical line at the pointer of the mouse</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#genomeName">genomeName</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#HEIGHT_SEQUENCE">HEIGHT_SEQUENCE</a></span></code></th>
<td class="colLast">
<div class="block">Height of the sequence track (when displayed)</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#heightPix">heightPix</a></span></code></th>
<td class="colLast">
<div class="block">Height in pixels of the drawing area</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private <a href="../../datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#massSpecData">massSpecData</a></span></code></th>
<td class="colLast">
<div class="block">MassSpecData where the selected NTerm is</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#MINIMUM_DATA_SIZE">MINIMUM_DATA_SIZE</a></span></code></th>
<td class="colLast">
<div class="block">Minimal height of a data track</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#mouseXPosition">mouseXPosition</a></span></code></th>
<td class="colLast">
<div class="block">When mouse is clicked we displayed a line is displayed to show synteny: mouseYPosition gives the
 X position</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#mouseYPosition">mouseYPosition</a></span></code></th>
<td class="colLast">
<div class="block">When mouse is clicked we displayed a line is displayed to show synteny: mouseYPosition gives the
 Y position</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#Name">Name</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private <a href="../../datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#nTermHighlight">nTermHighlight</a></span></code></th>
<td class="colLast">
<div class="block">NTerm which will be highlighted</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#nTermTypeOverlap">nTermTypeOverlap</a></span></code></th>
<td class="colLast">
<div class="block">Type of NTerm to display, if = All display all types</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.String[][]&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#saveData">saveData</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#sequenceHeight">sequenceHeight</a></span></code></th>
<td class="colLast">
<div class="block">Height of the sequence track</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private static long</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#serialVersionUID">serialVersionUID</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#testData">testData</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private <a href="../core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#track">track</a></span></code></th>
<td class="colLast">
<div class="block">All information about the data to display</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private org.eclipse.swt.widgets.Composite</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#trackComposite">trackComposite</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#widthBP">widthBP</a></span></code></th>
<td class="colLast">
<div class="block">Width in Base pairs of the drawing area</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#zoomVertical">zoomVertical</a></span></code></th>
<td class="colLast">
<div class="block">Vertical zoom</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(org.eclipse.swt.widgets.Composite,int,org.eclipse.swt.widgets.Composite)">TrackCanvasData</a></span>&#8203;(org.eclipse.swt.widgets.Composite&nbsp;parent,
               int&nbsp;style,
               org.eclipse.swt.widgets.Composite&nbsp;trackComposite)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#clickedElement(org.eclipse.swt.events.MouseEvent)">clickedElement</a></span>&#8203;(org.eclipse.swt.events.MouseEvent&nbsp;e)</code></th>
<td class="colLast">
<div class="block">Get base-pair position of the clicked element</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#convertBPtoX(int)">convertBPtoX</a></span>&#8203;(int&nbsp;bpIndex)</code></th>
<td class="colLast">
<div class="block">Convert position in the genome into a position in the display<br>
 (inverse function of xImageTobasepair(int x))</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#convertXtoBP(int)">convertXtoBP</a></span>&#8203;(int&nbsp;x)</code></th>
<td class="colLast">
<div class="block">Convert position in the display to a position in the genome<br>
 (inverse function of basepairToXimage(int bpIndex))<br>
 If MassSpec data has not been register do it</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayData(org.eclipse.swt.graphics.GC)">displayData</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc)</code></th>
<td class="colLast">
<div class="block">Display every type of data in a GC (Graphic context)<br>
 THE CODE CAN BE IMPROVE HERE</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayDataBIOCOND(org.eclipse.swt.graphics.GC)">displayDataBIOCOND</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc)</code></th>
<td class="colLast">
<div class="block">Method run when track are displayed by biological conditions</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayDataDATASEPARATED(org.eclipse.swt.graphics.GC)">displayDataDATASEPARATED</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc)</code></th>
<td class="colLast">
<div class="block">Method run when track are displayed with all datasets separated</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayDataLegend(org.eclipse.swt.graphics.GC,java.lang.String,int%5B%5D,double,double,int,int,bacnet.datamodel.dataset.OmicsData.TypeData)">displayDataLegend</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
                 java.lang.String&nbsp;name,
                 int[]&nbsp;position,
                 double&nbsp;min,
                 double&nbsp;max,
                 int&nbsp;dataIndex,
                 int&nbsp;minPosition,
                 <a href="../../datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a>&nbsp;typeData)</code></th>
<td class="colLast">
<div class="block">Display all information about a current data in a GC (Graphic context)<br>
 line in the center representing 0 value
 min, max and mean value
 name of the data</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayDataOVERLAY(org.eclipse.swt.graphics.GC)">displayDataOVERLAY</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc)</code></th>
<td class="colLast">
<div class="block">Method run when track are overlayed together</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayExpressionMatrix(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">displayExpressionMatrix</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
                       java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&gt;&nbsp;matrices,
                       double&nbsp;min,
                       double&nbsp;max,
                       int&nbsp;dataIndex,
                       int&nbsp;minPosition)</code></th>
<td class="colLast">
<div class="block">Function to display a list of ExpressionMatrix data in a GC (Graphic context)<br>
 All rowNames should correspond to a locus on the genome, otherwise it will be ignore</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayGeneExpression(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">displayGeneExpression</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
                     java.util.ArrayList&lt;<a href="../../datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;&nbsp;geneExprs,
                     double&nbsp;min,
                     double&nbsp;max,
                     int&nbsp;dataIndex,
                     int&nbsp;minPosition)</code></th>
<td class="colLast">
<div class="block">Function to display a list of GeneExpressionStreaming data in a GC (Graphic context)</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayLegendAfter(org.eclipse.swt.graphics.GC)">displayLegendAfter</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc)</code></th>
<td class="colLast">
<div class="block">Display all decorative elements which will be draw at the end, so on every other elements</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayNTerm(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">displayNTerm</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
            java.util.ArrayList&lt;<a href="../../datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&gt;&nbsp;nTerms,
            double&nbsp;min,
            double&nbsp;max,
            int&nbsp;dataIndex,
            int&nbsp;minPosition)</code></th>
<td class="colLast">
<div class="block">Function to display a list of GeneExpressionStreaming data in a GC (Graphic context)</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayProteomicsData(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">displayProteomicsData</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
                     java.util.ArrayList&lt;<a href="../../datamodel/dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a>&gt;&nbsp;matrices,
                     double&nbsp;min,
                     double&nbsp;max,
                     int&nbsp;dataIndex,
                     int&nbsp;minPosition)</code></th>
<td class="colLast">
<div class="block">Function to display a list of ExpressionMatrix data in a GC (Graphic context)<br>
 All rowNames should correspond to a locus on the genome, otherwise it will be ignore</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayRNASeq(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">displayRNASeq</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
             java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;seqDatas,
             double&nbsp;min,
             double&nbsp;max,
             int&nbsp;dataIndex,
             int&nbsp;minPosition)</code></th>
<td class="colLast">
<div class="block">Function to display a list of GeneExpressionStreaming data in a GC (Graphic context)</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displaySequence(org.eclipse.swt.graphics.GC)">displaySequence</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc)</code></th>
<td class="colLast">
<div class="block">Display sequence = For each nucleotide, give the translation of the codon in which it is in the
 center</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayTermSeq(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">displayTermSeq</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
              java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;seqDatas,
              double&nbsp;min,
              double&nbsp;max,
              int&nbsp;dataIndex,
              int&nbsp;minPosition)</code></th>
<td class="colLast">
<div class="block">Function to display a list of GeneExpressionStreaming data in a GC (Graphic context)</div>
</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayTiling(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">displayTiling</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
             java.util.ArrayList&lt;<a href="../../datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a>&gt;&nbsp;tilings,
             double&nbsp;min,
             double&nbsp;max,
             int&nbsp;dataIndex,
             int&nbsp;minPosition)</code></th>
<td class="colLast">
<div class="block">Function to display a list of <code>Tiling</code> in a GC (Graphic context)</div>
</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayTSS(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">displayTSS</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
          java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;seqDatas,
          double&nbsp;min,
          double&nbsp;max,
          int&nbsp;dataIndex,
          int&nbsp;minPosition)</code></th>
<td class="colLast">
<div class="block">Function to display a list of GeneExpressionStreaming data in a GC (Graphic context)</div>
</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code><a href="../../datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAnnotation()">getAnnotation</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getBpSizeH()">getBpSizeH</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCanvasName()">getCanvasName</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getChromoID()">getChromoID</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code>private int[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDataPosition(java.lang.String,int,int,double,double,double,int)">getDataPosition</a></span>&#8203;(java.lang.String&nbsp;dataName,
               int&nbsp;bpIndex,
               int&nbsp;dataIndex,
               double&nbsp;value,
               double&nbsp;min,
               double&nbsp;max,
               int&nbsp;minPosition)</code></th>
<td class="colLast">
<div class="block">From the value of the data at a given genomic position, this function return the position in the
 <code>GC</code></div>
</td>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDataPositionMin(java.lang.String,int,double,double)">getDataPositionMin</a></span>&#8203;(java.lang.String&nbsp;dataName,
                  int&nbsp;dataIndex,
                  double&nbsp;min,
                  double&nbsp;max)</code></th>
<td class="colLast">
<div class="block">Return the position in pixel of the minimal value</div>
</td>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDataWindowSize()">getDataWindowSize</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDecaySliderVBar()">getDecaySliderVBar</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGenomeName()">getGenomeName</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i27" class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getHeightPix()">getHeightPix</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i28" class="altColor">
<td class="colFirst"><code><a href="../../datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMassSpecData()">getMassSpecData</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i29" class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMouseXPosition()">getMouseXPosition</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i30" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMouseYPosition()">getMouseYPosition</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i31" class="rowColor">
<td class="colFirst"><code><a href="../../datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getnTermHighlight()">getnTermHighlight</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i32" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getnTermTypeOverlap()">getnTermTypeOverlap</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i33" class="rowColor">
<td class="colFirst"><code><a href="../core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTrack()">getTrack</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i34" class="altColor">
<td class="colFirst"><code>org.eclipse.swt.widgets.Composite</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTrackComposite()">getTrackComposite</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i35" class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getWidthBP()">getWidthBP</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i36" class="altColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getZoomVertical()">getZoomVertical</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i37" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isDisplayMouseLine()">isDisplayMouseLine</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i38" class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isTestData()">isTestData</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i39" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#mouseMove(org.eclipse.swt.events.MouseEvent)">mouseMove</a></span>&#8203;(org.eclipse.swt.events.MouseEvent&nbsp;e)</code></th>
<td class="colLast">
<div class="block">If the mouse is over the genome viewer, detect it is on the annotation, and display the
 corresponding gene, sRNA, asRNA</div>
</td>
</tr>
<tr id="i40" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#moveVertically(int)">moveVertically</a></span>&#8203;(int&nbsp;sliderVBarIncrement)</code></th>
<td class="colLast">
<div class="block">From the position of the vertical bar draw the corresponding part of the data</div>
</td>
</tr>
<tr id="i41" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setAnnotation(bacnet.datamodel.annotation.Annotation)">setAnnotation</a></span>&#8203;(<a href="../../datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a>&nbsp;annotation)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i42" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setBpSizeH(double)">setBpSizeH</a></span>&#8203;(double&nbsp;bpSizeH)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i43" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setChromoID(java.lang.String)">setChromoID</a></span>&#8203;(java.lang.String&nbsp;chromoID)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i44" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDataWindowSize(int)">setDataWindowSize</a></span>&#8203;(int&nbsp;dataWindowSize)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i45" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDecaySliderVBar(int)">setDecaySliderVBar</a></span>&#8203;(int&nbsp;decaySliderVBar)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i46" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDisplayMouseLine(boolean)">setDisplayMouseLine</a></span>&#8203;(boolean&nbsp;displayMouseLine)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i47" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGenomeName(java.lang.String)">setGenomeName</a></span>&#8203;(java.lang.String&nbsp;genomeName)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i48" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setHeightPix(int)">setHeightPix</a></span>&#8203;(int&nbsp;heightPix)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i49" class="rowColor">
<td class="colFirst"><code>private boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setHorizontalDisplayParameter(org.eclipse.swt.events.PaintEvent)">setHorizontalDisplayParameter</a></span>&#8203;(org.eclipse.swt.events.PaintEvent&nbsp;e)</code></th>
<td class="colLast">
<div class="block">Fix the number of pixel for one base pair<br>
 bpSizeH = widthPix / (double) widthBP<br>
 bpSizeV = heightPix / number of data</div>
</td>
</tr>
<tr id="i50" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setMassSpecData(bacnet.datamodel.dataset.NTermData)">setMassSpecData</a></span>&#8203;(<a href="../../datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&nbsp;massSpecData)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i51" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setMouseXPosition(int)">setMouseXPosition</a></span>&#8203;(int&nbsp;mouseXPosition)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i52" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setMouseYPosition(int)">setMouseYPosition</a></span>&#8203;(int&nbsp;mouseYPosition)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i53" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setName(java.lang.String)">setName</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i54" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setnTermHighlight(bacnet.datamodel.proteomics.NTerm)">setnTermHighlight</a></span>&#8203;(<a href="../../datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTermHighlight)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i55" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setnTermTypeOverlap(java.lang.String)">setnTermTypeOverlap</a></span>&#8203;(java.lang.String&nbsp;nTermTypeOverlap)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i56" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setTestData(boolean)">setTestData</a></span>&#8203;(boolean&nbsp;testData)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i57" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setTrack(bacnet.genomeBrowser.core.Track)">setTrack</a></span>&#8203;(<a href="../core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a>&nbsp;track)</code></th>
<td class="colLast">
<div class="block">Set the data of the Canvas, and implement the PaintListener
 
 Execute different tasks:<br>
 Fix the number of pixel for one base pair
 Set width and height parameters for data and annotation track Display in order:<br>
 different line and ticks for the base pair position, every display needed in the background
 display all the data
 display sequence if needed
 display annotation
 different genome name and mouse line in foreground</div>
</td>
</tr>
<tr id="i58" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setTrackComposite(org.eclipse.swt.widgets.Composite)">setTrackComposite</a></span>&#8203;(org.eclipse.swt.widgets.Composite&nbsp;trackComposite)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i59" class="rowColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setVerticalDisplayParameter(org.eclipse.swt.graphics.GC)">setVerticalDisplayParameter</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc)</code></th>
<td class="colLast">
<div class="block">Set different parameters needed for displaying Genome viewer<br>
 <code>annotNumber</code>
 <code>annotSeparationNumber</code>
 <code>annotSeparationSize</code>
 <code>annotLine</code>
 <code>dataWindowSize</code>
 <code>dataSize</code>
 <code>dataSeparationSize</code></div>
</td>
</tr>
<tr id="i60" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setWidthBP(int)">setWidthBP</a></span>&#8203;(int&nbsp;widthBP)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i61" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setZoomVertical(double)">setZoomVertical</a></span>&#8203;(double&nbsp;zoomVertical)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.org.eclipse.swt.widgets.Canvas">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.eclipse.swt.widgets.Canvas</h3>
<code>addPaintListener, getAdapter, removePaintListener</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.org.eclipse.swt.widgets.Composite">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.eclipse.swt.widgets.Composite</h3>
<code>changed, computeSize, getBackgroundMode, getChildren, getLayout, getLayoutDeferred, getTabList, isLayoutDeferred, layout, layout, layout, layout, layout, setBackgroundMode, setFocus, setLayout, setLayoutDeferred, setTabList</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.org.eclipse.swt.widgets.Scrollable">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.eclipse.swt.widgets.Scrollable</h3>
<code>computeTrim, getClientArea, getHorizontalBar, getVerticalBar</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.org.eclipse.swt.widgets.Control">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.eclipse.swt.widgets.Control</h3>
<code>addControlListener, addDragDetectListener, addFocusListener, addGestureListener, addHelpListener, addKeyListener, addMenuDetectListener, addMouseListener, addTouchListener, addTraverseListener, computeSize, forceFocus, getAccessible, getBackground, getBackgroundImage, getBorderWidth, getBounds, getCursor, getEnabled, getFont, getForeground, getLayoutData, getLocation, getMenu, getMonitor, getOrientation, getParent, getShell, getSize, getToolTipText, getTouchEnabled, getVisible, isEnabled, isFocusControl, isReparentable, isVisible, moveAbove, moveBelow, pack, pack, redraw, redraw, removeControlListener, removeDragDetectListener, removeFocusListener, removeGestureListener, removeHelpListener, removeKeyListener, removeMenuDetectListener, removeMouseListener, removeTouchListener, removeTraverseListener, requestLayout, setBackground, setBackgroundImage, setBounds, setBounds, setCursor, setData, setEnabled, setFont, setForeground, setLayoutData, setLocation, setLocation, setMenu, setOrientation, setParent, setRedraw, setSize, setSize, setToolTipText, setTouchEnabled, setVisible, toControl, toControl, toDisplay, toDisplay, update</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.org.eclipse.swt.widgets.Widget">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.eclipse.swt.widgets.Widget</h3>
<code>addDisposeListener, addListener, checkSubclass, checkWidget, dispose, getData, getData, getDisplay, getListeners, getStyle, isDisposed, isListening, notifyListeners, removeDisposeListener, removeListener, removeListener, reskin, setData, toString</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="serialVersionUID">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>serialVersionUID</h4>
<pre>private static final&nbsp;long serialVersionUID</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../constant-values.html#bacnet.genomeBrowser.tracksGUI.TrackCanvasData.serialVersionUID">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a id="MINIMUM_DATA_SIZE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MINIMUM_DATA_SIZE</h4>
<pre>public&nbsp;int MINIMUM_DATA_SIZE</pre>
<div class="block">Minimal height of a data track</div>
</li>
</ul>
<a id="HEIGHT_SEQUENCE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>HEIGHT_SEQUENCE</h4>
<pre>public static&nbsp;int HEIGHT_SEQUENCE</pre>
<div class="block">Height of the sequence track (when displayed)</div>
</li>
</ul>
<a id="Name">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>Name</h4>
<pre>private&nbsp;java.lang.String Name</pre>
</li>
</ul>
<a id="trackComposite">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>trackComposite</h4>
<pre>private&nbsp;org.eclipse.swt.widgets.Composite trackComposite</pre>
</li>
</ul>
<a id="track">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>track</h4>
<pre>private&nbsp;<a href="../core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a> track</pre>
<div class="block">All information about the data to display</div>
</li>
</ul>
<a id="annotation">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>annotation</h4>
<pre>private&nbsp;<a href="../../datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a> annotation</pre>
<div class="block"><code>Annotation</code> data for all Genome information</div>
</li>
</ul>
<a id="nTermHighlight">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>nTermHighlight</h4>
<pre>private&nbsp;<a href="../../datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a> nTermHighlight</pre>
<div class="block">NTerm which will be highlighted</div>
</li>
</ul>
<a id="nTermTypeOverlap">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>nTermTypeOverlap</h4>
<pre>private&nbsp;java.lang.String nTermTypeOverlap</pre>
<div class="block">Type of NTerm to display, if = All display all types</div>
</li>
</ul>
<a id="massSpecData">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>massSpecData</h4>
<pre>private&nbsp;<a href="../../datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a> massSpecData</pre>
<div class="block">MassSpecData where the selected NTerm is</div>
</li>
</ul>
<a id="genomeName">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>genomeName</h4>
<pre>private&nbsp;java.lang.String genomeName</pre>
</li>
</ul>
<a id="chromoID">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>chromoID</h4>
<pre>private&nbsp;java.lang.String chromoID</pre>
</li>
</ul>
<a id="mouseXPosition">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>mouseXPosition</h4>
<pre>private&nbsp;int mouseXPosition</pre>
<div class="block">When mouse is clicked we displayed a line is displayed to show synteny: mouseYPosition gives the
 X position</div>
</li>
</ul>
<a id="mouseYPosition">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>mouseYPosition</h4>
<pre>private&nbsp;int mouseYPosition</pre>
<div class="block">When mouse is clicked we displayed a line is displayed to show synteny: mouseYPosition gives the
 Y position</div>
</li>
</ul>
<a id="displayMouseLine">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayMouseLine</h4>
<pre>private&nbsp;boolean displayMouseLine</pre>
<div class="block">If true we display a vertical line at the pointer of the mouse</div>
</li>
</ul>
<a id="heightPix">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>heightPix</h4>
<pre>private&nbsp;int heightPix</pre>
<div class="block">Height in pixels of the drawing area</div>
</li>
</ul>
<a id="widthBP">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>widthBP</h4>
<pre>private&nbsp;int widthBP</pre>
<div class="block">Width in Base pairs of the drawing area</div>
</li>
</ul>
<a id="dataWindowSize">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>dataWindowSize</h4>
<pre>private&nbsp;int dataWindowSize</pre>
<div class="block">Height of all the data window = all data combine</div>
</li>
</ul>
<a id="decaySliderVBar">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>decaySliderVBar</h4>
<pre>private&nbsp;int decaySliderVBar</pre>
<div class="block">Vertical decay imposed by the vertical slider</div>
</li>
</ul>
<a id="zoomVertical">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>zoomVertical</h4>
<pre>private&nbsp;double zoomVertical</pre>
<div class="block">Vertical zoom</div>
</li>
</ul>
<a id="dataSizeReference">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>dataSizeReference</h4>
<pre>private&nbsp;int dataSizeReference</pre>
<div class="block">Height of on data</div>
</li>
</ul>
<a id="sequenceHeight">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>sequenceHeight</h4>
<pre>private&nbsp;int sequenceHeight</pre>
<div class="block">Height of the sequence track</div>
</li>
</ul>
<a id="bpSizeH">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>bpSizeH</h4>
<pre>private&nbsp;double bpSizeH</pre>
<div class="block">Width in pixel of one base pair (fundamental unit for the rest of the display)</div>
</li>
</ul>
<a id="testData">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>testData</h4>
<pre>private&nbsp;boolean testData</pre>
</li>
</ul>
<a id="saveData">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>saveData</h4>
<pre>private&nbsp;java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.String[][]&gt; saveData</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;(org.eclipse.swt.widgets.Composite,int,org.eclipse.swt.widgets.Composite)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>TrackCanvasData</h4>
<pre>public&nbsp;TrackCanvasData&#8203;(org.eclipse.swt.widgets.Composite&nbsp;parent,
                       int&nbsp;style,
                       org.eclipse.swt.widgets.Composite&nbsp;trackComposite)</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="setTrack(bacnet.genomeBrowser.core.Track)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setTrack</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setTrack&#8203;(<a href="../core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a>&nbsp;track)</pre>
<div class="block">Set the data of the Canvas, and implement the PaintListener
 
 Execute different tasks:<br>
 <li>Fix the number of pixel for one base pair
 <li>Set width and height parameters for data and annotation track Display in order:<br>
 <li>different line and ticks for the base pair position, every display needed in the background
 <li>display all the data
 <li>display sequence if needed
 <li>display annotation
 <li>different genome name and mouse line in foreground</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>tracks</code> - <code>Track</code></dd>
<dd><code>data</code> - <code>TrackData</code></dd>
</dl>
</li>
</ul>
<a id="setHorizontalDisplayParameter(org.eclipse.swt.events.PaintEvent)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setHorizontalDisplayParameter</h4>
<pre class="methodSignature">private&nbsp;boolean&nbsp;setHorizontalDisplayParameter&#8203;(org.eclipse.swt.events.PaintEvent&nbsp;e)</pre>
<div class="block">Fix the number of pixel for one base pair<br>
 <li>bpSizeH = widthPix / (double) widthBP<br>
 <li>bpSizeV = heightPix / number of data</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>true if the value has been changed</dd>
</dl>
</li>
</ul>
<a id="setVerticalDisplayParameter(org.eclipse.swt.graphics.GC)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setVerticalDisplayParameter</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;setVerticalDisplayParameter&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc)</pre>
<div class="block">Set different parameters needed for displaying Genome viewer<br>
 <li><code>annotNumber</code>
 <li><code>annotSeparationNumber</code>
 <li><code>annotSeparationSize</code>
 <li><code>annotLine</code>
 <li><code>dataWindowSize</code>
 <li><code>dataSize</code>
 <li><code>dataSeparationSize</code></div>
</li>
</ul>
<a id="getDataPosition(java.lang.String,int,int,double,double,double,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDataPosition</h4>
<pre class="methodSignature">private&nbsp;int[]&nbsp;getDataPosition&#8203;(java.lang.String&nbsp;dataName,
                              int&nbsp;bpIndex,
                              int&nbsp;dataIndex,
                              double&nbsp;value,
                              double&nbsp;min,
                              double&nbsp;max,
                              int&nbsp;minPosition)</pre>
<div class="block">From the value of the data at a given genomic position, this function return the position in the
 <code>GC</code></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bpIndex</code> - position in the genome</dd>
<dd><code>dataIndex</code> - index of the current data used</dd>
<dd><code>value</code> - of the data</dd>
<dd><code>min</code> - minimum of this data</dd>
<dd><code>max</code> - maximum of this data</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>vector int[] = [x,y]</dd>
</dl>
</li>
</ul>
<a id="getDataPositionMin(java.lang.String,int,double,double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDataPositionMin</h4>
<pre class="methodSignature">private&nbsp;int&nbsp;getDataPositionMin&#8203;(java.lang.String&nbsp;dataName,
                               int&nbsp;dataIndex,
                               double&nbsp;min,
                               double&nbsp;max)</pre>
<div class="block">Return the position in pixel of the minimal value</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>dataName</code> - </dd>
<dd><code>dataIndex</code> - </dd>
<dd><code>min</code> - </dd>
<dd><code>max</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="displayData(org.eclipse.swt.graphics.GC)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayData</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;displayData&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc)</pre>
<div class="block">Display every type of data in a GC (Graphic context)<br>
 THE CODE CAN BE IMPROVE HERE</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gc</code> - </dd>
</dl>
</li>
</ul>
<a id="displayDataOVERLAY(org.eclipse.swt.graphics.GC)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayDataOVERLAY</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;displayDataOVERLAY&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc)</pre>
<div class="block">Method run when track are overlayed together</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gc</code> - </dd>
</dl>
</li>
</ul>
<a id="displayDataBIOCOND(org.eclipse.swt.graphics.GC)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayDataBIOCOND</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;displayDataBIOCOND&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc)</pre>
<div class="block">Method run when track are displayed by biological conditions</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gc</code> - </dd>
</dl>
</li>
</ul>
<a id="displayDataDATASEPARATED(org.eclipse.swt.graphics.GC)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayDataDATASEPARATED</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;displayDataDATASEPARATED&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc)</pre>
<div class="block">Method run when track are displayed with all datasets separated</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gc</code> - </dd>
</dl>
</li>
</ul>
<a id="displayDataLegend(org.eclipse.swt.graphics.GC,java.lang.String,int[],double,double,int,int,bacnet.datamodel.dataset.OmicsData.TypeData)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayDataLegend</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;displayDataLegend&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
                               java.lang.String&nbsp;name,
                               int[]&nbsp;position,
                               double&nbsp;min,
                               double&nbsp;max,
                               int&nbsp;dataIndex,
                               int&nbsp;minPosition,
                               <a href="../../datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a>&nbsp;typeData)</pre>
<div class="block">Display all information about a current data in a GC (Graphic context)<br>
 <li>line in the center representing 0 value
 <li>min, max and mean value
 <li>name of the data</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gc</code> - </dd>
<dd><code>name</code> - </dd>
<dd><code>min</code> - </dd>
<dd><code>max</code> - </dd>
<dd><code>dataIndex</code> - </dd>
</dl>
</li>
</ul>
<a id="displaySequence(org.eclipse.swt.graphics.GC)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displaySequence</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;displaySequence&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc)</pre>
<div class="block">Display sequence = For each nucleotide, give the translation of the codon in which it is in the
 center</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gc</code> - </dd>
</dl>
</li>
</ul>
<a id="displayTiling(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayTiling</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;displayTiling&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
                           java.util.ArrayList&lt;<a href="../../datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a>&gt;&nbsp;tilings,
                           double&nbsp;min,
                           double&nbsp;max,
                           int&nbsp;dataIndex,
                           int&nbsp;minPosition)</pre>
<div class="block">Function to display a list of <code>Tiling</code> in a GC (Graphic context)</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gc</code> - </dd>
<dd><code>tilings</code> - </dd>
<dd><code>min</code> - </dd>
<dd><code>max</code> - </dd>
<dd><code>dataIndex</code> - </dd>
</dl>
</li>
</ul>
<a id="displayRNASeq(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayRNASeq</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;displayRNASeq&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
                           java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;seqDatas,
                           double&nbsp;min,
                           double&nbsp;max,
                           int&nbsp;dataIndex,
                           int&nbsp;minPosition)</pre>
<div class="block">Function to display a list of GeneExpressionStreaming data in a GC (Graphic context)</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gc</code> - </dd>
<dd><code>seqDatas</code> - </dd>
<dd><code>dataIndex</code> - </dd>
</dl>
</li>
</ul>
<a id="displayTSS(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayTSS</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;displayTSS&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
                        java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;seqDatas,
                        double&nbsp;min,
                        double&nbsp;max,
                        int&nbsp;dataIndex,
                        int&nbsp;minPosition)</pre>
<div class="block">Function to display a list of GeneExpressionStreaming data in a GC (Graphic context)</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gc</code> - </dd>
<dd><code>seqDatas</code> - </dd>
<dd><code>dataIndex</code> - </dd>
</dl>
</li>
</ul>
<a id="displayTermSeq(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayTermSeq</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;displayTermSeq&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
                            java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;seqDatas,
                            double&nbsp;min,
                            double&nbsp;max,
                            int&nbsp;dataIndex,
                            int&nbsp;minPosition)</pre>
<div class="block">Function to display a list of GeneExpressionStreaming data in a GC (Graphic context)</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gc</code> - </dd>
<dd><code>seqDatas</code> - </dd>
<dd><code>dataIndex</code> - </dd>
</dl>
</li>
</ul>
<a id="displayNTerm(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayNTerm</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;displayNTerm&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
                          java.util.ArrayList&lt;<a href="../../datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&gt;&nbsp;nTerms,
                          double&nbsp;min,
                          double&nbsp;max,
                          int&nbsp;dataIndex,
                          int&nbsp;minPosition)</pre>
<div class="block">Function to display a list of GeneExpressionStreaming data in a GC (Graphic context)</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gc</code> - </dd>
<dd><code>geneExprs</code> - </dd>
<dd><code>min</code> - </dd>
<dd><code>max</code> - </dd>
<dd><code>dataIndex</code> - </dd>
</dl>
</li>
</ul>
<a id="displayGeneExpression(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayGeneExpression</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;displayGeneExpression&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
                                   java.util.ArrayList&lt;<a href="../../datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;&nbsp;geneExprs,
                                   double&nbsp;min,
                                   double&nbsp;max,
                                   int&nbsp;dataIndex,
                                   int&nbsp;minPosition)</pre>
<div class="block">Function to display a list of GeneExpressionStreaming data in a GC (Graphic context)</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gc</code> - </dd>
<dd><code>geneExprs</code> - </dd>
<dd><code>min</code> - </dd>
<dd><code>max</code> - </dd>
<dd><code>dataIndex</code> - </dd>
</dl>
</li>
</ul>
<a id="displayExpressionMatrix(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayExpressionMatrix</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;displayExpressionMatrix&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
                                     java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&gt;&nbsp;matrices,
                                     double&nbsp;min,
                                     double&nbsp;max,
                                     int&nbsp;dataIndex,
                                     int&nbsp;minPosition)</pre>
<div class="block">Function to display a list of ExpressionMatrix data in a GC (Graphic context)<br>
 All rowNames should correspond to a locus on the genome, otherwise it will be ignore</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gc</code> - </dd>
<dd><code>geneExprs</code> - </dd>
<dd><code>min</code> - </dd>
<dd><code>max</code> - </dd>
<dd><code>dataIndex</code> - </dd>
</dl>
</li>
</ul>
<a id="displayProteomicsData(org.eclipse.swt.graphics.GC,java.util.ArrayList,double,double,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayProteomicsData</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;displayProteomicsData&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
                                   java.util.ArrayList&lt;<a href="../../datamodel/dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a>&gt;&nbsp;matrices,
                                   double&nbsp;min,
                                   double&nbsp;max,
                                   int&nbsp;dataIndex,
                                   int&nbsp;minPosition)</pre>
<div class="block">Function to display a list of ExpressionMatrix data in a GC (Graphic context)<br>
 All rowNames should correspond to a locus on the genome, otherwise it will be ignore</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gc</code> - </dd>
<dd><code>geneExprs</code> - </dd>
<dd><code>min</code> - </dd>
<dd><code>max</code> - </dd>
<dd><code>dataIndex</code> - </dd>
</dl>
</li>
</ul>
<a id="displayLegendAfter(org.eclipse.swt.graphics.GC)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayLegendAfter</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;displayLegendAfter&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc)</pre>
<div class="block">Display all decorative elements which will be draw at the end, so on every other elements</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gc</code> - </dd>
</dl>
</li>
</ul>
<a id="convertBPtoX(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>convertBPtoX</h4>
<pre class="methodSignature">private&nbsp;int&nbsp;convertBPtoX&#8203;(int&nbsp;bpIndex)</pre>
<div class="block">Convert position in the genome into a position in the display<br>
 (inverse function of xImageTobasepair(int x))</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bpIndex</code> - in the genome</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>x position in the view</dd>
</dl>
</li>
</ul>
<a id="convertXtoBP(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>convertXtoBP</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;convertXtoBP&#8203;(int&nbsp;x)</pre>
<div class="block">Convert position in the display to a position in the genome<br>
 (inverse function of basepairToXimage(int bpIndex))<br>
 If MassSpec data has not been register do it</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>x</code> - position in the view</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>bpIndex in the genome</dd>
</dl>
</li>
</ul>
<a id="mouseMove(org.eclipse.swt.events.MouseEvent)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>mouseMove</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;mouseMove&#8203;(org.eclipse.swt.events.MouseEvent&nbsp;e)</pre>
<div class="block">If the mouse is over the genome viewer, detect it is on the annotation, and display the
 corresponding gene, sRNA, asRNA</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code>mouseMove</code>&nbsp;in interface&nbsp;<code>org.eclipse.swt.events.MouseMoveListener</code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>e</code> - </dd>
</dl>
</li>
</ul>
<a id="moveVertically(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>moveVertically</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;moveVertically&#8203;(int&nbsp;sliderVBarIncrement)</pre>
<div class="block">From the position of the vertical bar draw the corresponding part of the data</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sliderVBarPosition</code> - </dd>
</dl>
</li>
</ul>
<a id="clickedElement(org.eclipse.swt.events.MouseEvent)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>clickedElement</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;clickedElement&#8203;(org.eclipse.swt.events.MouseEvent&nbsp;e)</pre>
<div class="block">Get base-pair position of the clicked element</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>e</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getnTermTypeOverlap()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getnTermTypeOverlap</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getnTermTypeOverlap()</pre>
</li>
</ul>
<a id="setnTermTypeOverlap(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setnTermTypeOverlap</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setnTermTypeOverlap&#8203;(java.lang.String&nbsp;nTermTypeOverlap)</pre>
</li>
</ul>
<a id="getnTermHighlight()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getnTermHighlight</h4>
<pre class="methodSignature">public&nbsp;<a href="../../datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;getnTermHighlight()</pre>
</li>
</ul>
<a id="setnTermHighlight(bacnet.datamodel.proteomics.NTerm)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setnTermHighlight</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setnTermHighlight&#8203;(<a href="../../datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTermHighlight)</pre>
</li>
</ul>
<a id="getMouseXPosition()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMouseXPosition</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getMouseXPosition()</pre>
</li>
</ul>
<a id="setMouseXPosition(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMouseXPosition</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setMouseXPosition&#8203;(int&nbsp;mouseXPosition)</pre>
</li>
</ul>
<a id="getMouseYPosition()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMouseYPosition</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getMouseYPosition()</pre>
</li>
</ul>
<a id="setMouseYPosition(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMouseYPosition</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setMouseYPosition&#8203;(int&nbsp;mouseYPosition)</pre>
</li>
</ul>
<a id="isDisplayMouseLine()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isDisplayMouseLine</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;isDisplayMouseLine()</pre>
</li>
</ul>
<a id="setDisplayMouseLine(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDisplayMouseLine</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDisplayMouseLine&#8203;(boolean&nbsp;displayMouseLine)</pre>
</li>
</ul>
<a id="getMassSpecData()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMassSpecData</h4>
<pre class="methodSignature">public&nbsp;<a href="../../datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&nbsp;getMassSpecData()</pre>
</li>
</ul>
<a id="setMassSpecData(bacnet.datamodel.dataset.NTermData)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMassSpecData</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setMassSpecData&#8203;(<a href="../../datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&nbsp;massSpecData)</pre>
</li>
</ul>
<a id="getTrack()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTrack</h4>
<pre class="methodSignature">public&nbsp;<a href="../core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a>&nbsp;getTrack()</pre>
</li>
</ul>
<a id="getAnnotation()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotation</h4>
<pre class="methodSignature">public&nbsp;<a href="../../datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a>&nbsp;getAnnotation()</pre>
</li>
</ul>
<a id="setAnnotation(bacnet.datamodel.annotation.Annotation)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAnnotation</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setAnnotation&#8203;(<a href="../../datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a>&nbsp;annotation)</pre>
</li>
</ul>
<a id="getHeightPix()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getHeightPix</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getHeightPix()</pre>
</li>
</ul>
<a id="setHeightPix(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setHeightPix</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setHeightPix&#8203;(int&nbsp;heightPix)</pre>
</li>
</ul>
<a id="getWidthBP()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getWidthBP</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getWidthBP()</pre>
</li>
</ul>
<a id="setWidthBP(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setWidthBP</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setWidthBP&#8203;(int&nbsp;widthBP)</pre>
</li>
</ul>
<a id="getDecaySliderVBar()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDecaySliderVBar</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getDecaySliderVBar()</pre>
</li>
</ul>
<a id="setDecaySliderVBar(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDecaySliderVBar</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDecaySliderVBar&#8203;(int&nbsp;decaySliderVBar)</pre>
</li>
</ul>
<a id="getDataWindowSize()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDataWindowSize</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getDataWindowSize()</pre>
</li>
</ul>
<a id="setDataWindowSize(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDataWindowSize</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDataWindowSize&#8203;(int&nbsp;dataWindowSize)</pre>
</li>
</ul>
<a id="getBpSizeH()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBpSizeH</h4>
<pre class="methodSignature">public&nbsp;double&nbsp;getBpSizeH()</pre>
</li>
</ul>
<a id="setBpSizeH(double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setBpSizeH</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setBpSizeH&#8203;(double&nbsp;bpSizeH)</pre>
</li>
</ul>
<a id="getTrackComposite()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTrackComposite</h4>
<pre class="methodSignature">public&nbsp;org.eclipse.swt.widgets.Composite&nbsp;getTrackComposite()</pre>
</li>
</ul>
<a id="setTrackComposite(org.eclipse.swt.widgets.Composite)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setTrackComposite</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setTrackComposite&#8203;(org.eclipse.swt.widgets.Composite&nbsp;trackComposite)</pre>
</li>
</ul>
<a id="getZoomVertical()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getZoomVertical</h4>
<pre class="methodSignature">public&nbsp;double&nbsp;getZoomVertical()</pre>
</li>
</ul>
<a id="setZoomVertical(double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setZoomVertical</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setZoomVertical&#8203;(double&nbsp;zoomVertical)</pre>
</li>
</ul>
<a id="isTestData()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isTestData</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;isTestData()</pre>
</li>
</ul>
<a id="setTestData(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setTestData</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setTestData&#8203;(boolean&nbsp;testData)</pre>
</li>
</ul>
<a id="getCanvasName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCanvasName</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getCanvasName()</pre>
</li>
</ul>
<a id="setName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setName</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setName&#8203;(java.lang.String&nbsp;name)</pre>
</li>
</ul>
<a id="getGenomeName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomeName</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getGenomeName()</pre>
</li>
</ul>
<a id="setGenomeName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGenomeName</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGenomeName&#8203;(java.lang.String&nbsp;genomeName)</pre>
</li>
</ul>
<a id="getChromoID()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromoID</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getChromoID()</pre>
</li>
</ul>
<a id="setChromoID(java.lang.String)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setChromoID</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setChromoID&#8203;(java.lang.String&nbsp;chromoID)</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/TrackCanvasData.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
