<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>NTermFilter</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="NTermFilter";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":9,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":9,"i11":9,"i12":9,"i13":9,"i14":9,"i15":9,"i16":10,"i17":10,"i18":10,"i19":10,"i20":10,"i21":10,"i22":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/NTermFilter.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.proteomics</a></div>
<h2 title="Class NTermFilter" class="title">Class NTermFilter</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>bacnet.datamodel.proteomics.NTermFilter</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<pre>public class <span class="typeNameLabel">NTermFilter</span>
extends java.lang.Object</pre>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#aTISCutoff">aTISCutoff</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;java.lang.String&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#cutoffHashMap">cutoffHashMap</a></span></code></th>
<td class="colLast">
<div class="block">Create a list of cutoff to apply to the list of Nterm<br></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#DEFAULT_FILTER_PATH">DEFAULT_FILTER_PATH</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#duplicateSequenceToRemove">duplicateSequenceToRemove</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private <a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#nTermExp">nTermExp</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#typeOverlaps">typeOverlaps</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(bacnet.datamodel.dataset.NTermData,java.lang.String)">NTermFilter</a></span>&#8203;(<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&nbsp;nTermExp,
           java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Filter a NTermData using the table of filter found in fileName</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#applyCutOff(java.lang.String,bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Sequence,int%5B%5D)">applyCutOff</a></span>&#8203;(java.lang.String&nbsp;cutoff,
           <a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
           <a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;overlapSequence,
           int[]&nbsp;overlap)</code></th>
<td class="colLast">
<div class="block">Read a cutoff<br>
 Interprete each cutoff by returning true of false if the NTerm pass the condition<br>
 If the cutoff contain an "or": parse it in two separate cutoff<br>
 If the cutoff starts with an "!"</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#applyFilter()">applyFilter</a></span>()</code></th>
<td class="colLast">
<div class="block">Search for overlap with Gene, sRNAn, asRNA and ncRNA of each NTerm</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#containStopCodon(bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Sequence,int%5B%5D)">containStopCodon</a></span>&#8203;(<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
                <a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;overlapSequence,
                int[]&nbsp;overlap)</code></th>
<td class="colLast">
<div class="block">Return true if it contains a StopCodon between the peptide and the overlaping sequence</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>private <a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#findOverlap(bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Chromosome)">findOverlap</a></span>&#8203;(<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
           <a href="../sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromosomeEGDe)</code></th>
<td class="colLast">
<div class="block">Find with which element on EGDe genome and RAST genome an NTerm overlap:<br>
 region of 25bp at the beginning of EGDe gene
 region of 25bp at the beginning of EGDe RAST gene, if this overlap is smaller than with EGde
 take this one, in order to detect bona-fide aTIS
 region of 25bp at the beginning of Srna or ASrna of EGDe, if this overlap is smaller than
 with Srna take this one, in order to detect bona-fide aTIS
 overlap in the middle of EGDe gene
 overlap in the middle of EGDe RAST gene
 overlap in the middle of Srna or ASrna of EGDe<br>
 <br>
 The 25bp are given by <code>aTISCutoff</code> argument of the <code>NTermFilter</code> object</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getaTISCutoff()">getaTISCutoff</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;java.lang.String&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCutoffHashMap()">getCutoffHashMap</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDuplicateSequenceToRemove()">getDuplicateSequenceToRemove</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code><a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getnTermExp()">getnTermExp</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>private int[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getOverlap(bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Sequence)">getOverlap</a></span>&#8203;(<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
          <a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;overlapSequence)</code></th>
<td class="colLast">
<div class="block">Return the difference between the start</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTypeOverlaps()">getTypeOverlaps</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isForMandAltStart(bacnet.datamodel.proteomics.NTerm)">isForMandAltStart</a></span>&#8203;(<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm)</code></th>
<td class="colLast">
<div class="block">Find if the NTerm start with a Methyonine<br>
 if it corresponds to a regular start codon "ATG" we need a Formyl modificaition to prove its a
 real translation initiation<br>
 If the methyonine corresponds to any other type of start codon, we don't need more to prove it's
 a translation initiation</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isStart(bacnet.datamodel.proteomics.NTerm,java.lang.String)">isStart</a></span>&#8203;(<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
       java.lang.String&nbsp;typeCodon)</code></th>
<td class="colLast">
<div class="block">Find i an NTerm start with a methyonine (START), or if the amino acid just before the peptide is
 a Methyonine (START-1), or just after (START+1)</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#overlapEqual(int%5B%5D,double)">overlapEqual</a></span>&#8203;(int[]&nbsp;overlap,
            double&nbsp;value)</code></th>
<td class="colLast">
<div class="block">Return true if overlap[0] < value</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#overlapInf(int%5B%5D,double)">overlapInf</a></span>&#8203;(int[]&nbsp;overlap,
          double&nbsp;value)</code></th>
<td class="colLast">
<div class="block">Return true if overlap[0] < value</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#overlapSup(int%5B%5D,double)">overlapSup</a></span>&#8203;(int[]&nbsp;overlap,
          double&nbsp;value)</code></th>
<td class="colLast">
<div class="block">Return true if overlap[0] < value</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>static java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;java.lang.String&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#readFilter(java.lang.String)">readFilter</a></span>&#8203;(java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Create a list of cutoff to apply to the list of Nterm<br></div>
</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setaTISCutoff(int)">setaTISCutoff</a></span>&#8203;(int&nbsp;aTISCutoff)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setCutoffHashMap(java.util.HashMap)">setCutoffHashMap</a></span>&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;java.lang.String&gt;&gt;&nbsp;cutoffHashMap)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDuplicateSequenceToRemove(java.util.ArrayList)">setDuplicateSequenceToRemove</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;duplicateSequenceToRemove)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setnTermExp(bacnet.datamodel.dataset.NTermData)">setnTermExp</a></span>&#8203;(<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&nbsp;nTermExp)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setOverlapsFromCutoffs()">setOverlapsFromCutoffs</a></span>()</code></th>
<td class="colLast">
<div class="block">Get a final list of all OverlapType, from cutoffs table</div>
</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setTypeOverlaps(java.util.ArrayList)">setTypeOverlaps</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;typeOverlaps)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#verifFilter()">verifFilter</a></span>()</code></th>
<td class="colLast">
<div class="block">Run some tests to check that filters has been well applied</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="DEFAULT_FILTER_PATH">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DEFAULT_FILTER_PATH</h4>
<pre>public static&nbsp;java.lang.String DEFAULT_FILTER_PATH</pre>
</li>
</ul>
<a id="typeOverlaps">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>typeOverlaps</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.String&gt; typeOverlaps</pre>
</li>
</ul>
<a id="nTermExp">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>nTermExp</h4>
<pre>private&nbsp;<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a> nTermExp</pre>
</li>
</ul>
<a id="duplicateSequenceToRemove">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>duplicateSequenceToRemove</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.String&gt; duplicateSequenceToRemove</pre>
</li>
</ul>
<a id="aTISCutoff">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>aTISCutoff</h4>
<pre>private&nbsp;int aTISCutoff</pre>
</li>
</ul>
<a id="cutoffHashMap">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>cutoffHashMap</h4>
<pre>private&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;java.lang.String&gt;&gt; cutoffHashMap</pre>
<div class="block">Create a list of cutoff to apply to the list of Nterm<br></div>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;(bacnet.datamodel.dataset.NTermData,java.lang.String)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>NTermFilter</h4>
<pre>public&nbsp;NTermFilter&#8203;(<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&nbsp;nTermExp,
                   java.lang.String&nbsp;fileName)</pre>
<div class="block">Filter a NTermData using the table of filter found in fileName</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nTermExp</code> - </dd>
<dd><code>fileName</code> - </dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="applyFilter()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>applyFilter</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;applyFilter()</pre>
<div class="block">Search for overlap with Gene, sRNAn, asRNA and ncRNA of each NTerm</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nameNTermData</code> - </dd>
</dl>
</li>
</ul>
<a id="verifFilter()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>verifFilter</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;verifFilter()</pre>
<div class="block">Run some tests to check that filters has been well applied</div>
</li>
</ul>
<a id="applyCutOff(java.lang.String,bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Sequence,int[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>applyCutOff</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;applyCutOff&#8203;(java.lang.String&nbsp;cutoff,
                           <a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
                           <a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;overlapSequence,
                           int[]&nbsp;overlap)</pre>
<div class="block">Read a cutoff<br>
 Interprete each cutoff by returning true of false if the NTerm pass the condition<br>
 If the cutoff contain an "or": parse it in two separate cutoff<br>
 If the cutoff starts with an "!": return the opposite of the cutoff<br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>cutoff</code> - </dd>
<dd><code>nTerm</code> - </dd>
<dd><code>overlapSequence</code> - </dd>
<dd><code>overlap</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="overlapEqual(int[],double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>overlapEqual</h4>
<pre class="methodSignature">public static&nbsp;boolean&nbsp;overlapEqual&#8203;(int[]&nbsp;overlap,
                                   double&nbsp;value)</pre>
<div class="block">Return true if overlap[0] < value</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>overlap</code> - </dd>
<dd><code>value</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="overlapInf(int[],double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>overlapInf</h4>
<pre class="methodSignature">public static&nbsp;boolean&nbsp;overlapInf&#8203;(int[]&nbsp;overlap,
                                 double&nbsp;value)</pre>
<div class="block">Return true if overlap[0] < value</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>overlap</code> - </dd>
<dd><code>value</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="overlapSup(int[],double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>overlapSup</h4>
<pre class="methodSignature">public static&nbsp;boolean&nbsp;overlapSup&#8203;(int[]&nbsp;overlap,
                                 double&nbsp;value)</pre>
<div class="block">Return true if overlap[0] < value</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>overlap</code> - </dd>
<dd><code>value</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="isStart(bacnet.datamodel.proteomics.NTerm,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isStart</h4>
<pre class="methodSignature">public static&nbsp;boolean&nbsp;isStart&#8203;(<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
                              java.lang.String&nbsp;typeCodon)</pre>
<div class="block">Find i an NTerm start with a methyonine (START), or if the amino acid just before the peptide is
 a Methyonine (START-1), or just after (START+1)</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nTerm</code> - </dd>
<dd><code>typeCodon</code> - START or START-1 or START+1 or nothing</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="isForMandAltStart(bacnet.datamodel.proteomics.NTerm)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isForMandAltStart</h4>
<pre class="methodSignature">public static&nbsp;boolean&nbsp;isForMandAltStart&#8203;(<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm)</pre>
<div class="block">Find if the NTerm start with a Methyonine<br>
 if it corresponds to a regular start codon "ATG" we need a Formyl modificaition to prove its a
 real translation initiation<br>
 If the methyonine corresponds to any other type of start codon, we don't need more to prove it's
 a translation initiation</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nTerm</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="containStopCodon(bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Sequence,int[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>containStopCodon</h4>
<pre class="methodSignature">public static&nbsp;boolean&nbsp;containStopCodon&#8203;(<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
                                       <a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;overlapSequence,
                                       int[]&nbsp;overlap)</pre>
<div class="block">Return true if it contains a StopCodon between the peptide and the overlaping sequence</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nTerm</code> - </dd>
<dd><code>overlapSequence</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="readFilter(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readFilter</h4>
<pre class="methodSignature">public static&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;java.lang.String&gt;&gt;&nbsp;readFilter&#8203;(java.lang.String&nbsp;fileName)</pre>
<div class="block">Create a list of cutoff to apply to the list of Nterm<br></div>
</li>
</ul>
<a id="findOverlap(bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Chromosome)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>findOverlap</h4>
<pre class="methodSignature">private&nbsp;<a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;findOverlap&#8203;(<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
                             <a href="../sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromosomeEGDe)</pre>
<div class="block">Find with which element on EGDe genome and RAST genome an NTerm overlap:<br>
 <li>region of 25bp at the beginning of EGDe gene
 <li>region of 25bp at the beginning of EGDe RAST gene, if this overlap is smaller than with EGde
 take this one, in order to detect bona-fide aTIS
 <li>region of 25bp at the beginning of Srna or ASrna of EGDe, if this overlap is smaller than
 with Srna take this one, in order to detect bona-fide aTIS
 <li>overlap in the middle of EGDe gene
 <li>overlap in the middle of EGDe RAST gene
 <li>overlap in the middle of Srna or ASrna of EGDe<br>
 <br>
 The 25bp are given by <code>aTISCutoff</code> argument of the <code>NTermFilter</code> object</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nTerm</code> - </dd>
<dd><code>genome</code> - </dd>
<dd><code>genomeRAST</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getOverlap(bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Sequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getOverlap</h4>
<pre class="methodSignature">private&nbsp;int[]&nbsp;getOverlap&#8203;(<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
                         <a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;overlapSequence)</pre>
<div class="block">Return the difference between the start</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nTerm</code> - </dd>
<dd><code>overlapSequence</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="setOverlapsFromCutoffs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setOverlapsFromCutoffs</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;setOverlapsFromCutoffs()</pre>
<div class="block">Get a final list of all OverlapType, from cutoffs table</div>
</li>
</ul>
<a id="getnTermExp()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getnTermExp</h4>
<pre class="methodSignature">public&nbsp;<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&nbsp;getnTermExp()</pre>
</li>
</ul>
<a id="setnTermExp(bacnet.datamodel.dataset.NTermData)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setnTermExp</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setnTermExp&#8203;(<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&nbsp;nTermExp)</pre>
</li>
</ul>
<a id="getTypeOverlaps()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTypeOverlaps</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getTypeOverlaps()</pre>
</li>
</ul>
<a id="setTypeOverlaps(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setTypeOverlaps</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setTypeOverlaps&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;typeOverlaps)</pre>
</li>
</ul>
<a id="getaTISCutoff()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getaTISCutoff</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getaTISCutoff()</pre>
</li>
</ul>
<a id="setaTISCutoff(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setaTISCutoff</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setaTISCutoff&#8203;(int&nbsp;aTISCutoff)</pre>
</li>
</ul>
<a id="getCutoffHashMap()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCutoffHashMap</h4>
<pre class="methodSignature">public&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;java.lang.String&gt;&gt;&nbsp;getCutoffHashMap()</pre>
</li>
</ul>
<a id="setCutoffHashMap(java.util.HashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setCutoffHashMap</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setCutoffHashMap&#8203;(java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;java.lang.String&gt;&gt;&nbsp;cutoffHashMap)</pre>
</li>
</ul>
<a id="getDuplicateSequenceToRemove()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDuplicateSequenceToRemove</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getDuplicateSequenceToRemove()</pre>
</li>
</ul>
<a id="setDuplicateSequenceToRemove(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setDuplicateSequenceToRemove</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDuplicateSequenceToRemove&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;duplicateSequenceToRemove)</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/NTermFilter.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
