<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 14:20:51 CEST 2019 -->
<title>NTermUtils</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="NTermUtils";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":9,"i1":9,"i2":9,"i3":9,"i4":9,"i5":9,"i6":9,"i7":9,"i8":9};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/NTermUtils.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.proteomics</a></div>
<h2 title="Class NTermUtils" class="title">Class NTermUtils</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>bacnet.datamodel.proteomics.NTermUtils</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<pre>public class <span class="typeNameLabel">NTermUtils</span>
extends java.lang.Object</pre>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">NTermUtils</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#compareNTermData(java.lang.String,java.lang.String)">compareNTermData</a></span>&#8203;(java.lang.String&nbsp;NTermDataName1,
                java.lang.String&nbsp;NTermDataName2)</code></th>
<td class="colLast">
<div class="block">Return the list of NTerm common to both experiment<br>
 NTerm are equal if:<br>
 same peptide sequence
 same modification</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#countRTerminal(java.lang.String)">countRTerminal</a></span>&#8203;(java.lang.String&nbsp;massSpecName)</code></th>
<td class="colLast">
<div class="block">Count the number of peptides with an R (arginine), and finishing with an R</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#findSDSequenceforLMO()">findSDSequenceforLMO</a></span>()</code></th>
<td class="colLast">
<div class="block">Find SD binding free energy on region -20bp upstream to each lmo</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>static <a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getFullORF(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Genome)">getFullORF</a></span>&#8203;(<a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
          <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in
 frame) containing the peptide (amino acid sequence) <br></div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static <a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getFullORF(java.lang.String,int,int,boolean,bacnet.datamodel.sequence.Genome)">getFullORF</a></span>&#8203;(java.lang.String&nbsp;peptideSeq,
          int&nbsp;beginPeptide,
          int&nbsp;endPeptide,
          boolean&nbsp;strand,
          <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in
 frame) containing the peptide (amino acid sequence) <br></div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getListOfFullORFfromPeptide(java.lang.String,bacnet.datamodel.sequence.Genome)">getListOfFullORFfromPeptide</a></span>&#8203;(java.lang.String&nbsp;fileName,
                           <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">From a list of peptide position retrieve the complete sequence of the ORF (in
 frame) containing the peptide (amino acid sequence) <br>
 WARNING: Be careful that your peptide info (begin,end) are correct.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getPATH()">getPATH</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getStartCodonBefore()">getStartCodonBefore</a></span>()</code></th>
<td class="colLast">
<div class="block">NOT FINISHED <br>
 <br>
 Search for the different start codon before a peptide, and the stop codon
 after this peptide<br>
 Print the nucleotide sequence plus the codon, and identify Start, STOp and
 the reference Peptide used</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#searchStopCodon(bacnet.datamodel.sequence.Genome,int,boolean)">searchStopCodon</a></span>&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
               int&nbsp;bpPosition,
               boolean&nbsp;strand)</code></th>
<td class="colLast">
<div class="block">Get the next stop codon in frame and dowstream to a bpPosition</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>NTermUtils</h4>
<pre>public&nbsp;NTermUtils()</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="getPATH()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPATH</h4>
<pre class="methodSignature">public static&nbsp;java.lang.String&nbsp;getPATH()</pre>
</li>
</ul>
<a id="getListOfFullORFfromPeptide(java.lang.String,bacnet.datamodel.sequence.Genome)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getListOfFullORFfromPeptide</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;getListOfFullORFfromPeptide&#8203;(java.lang.String&nbsp;fileName,
                                               <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</pre>
<div class="block">From a list of peptide position retrieve the complete sequence of the ORF (in
 frame) containing the peptide (amino acid sequence) <br>
 WARNING: Be careful that your peptide info (begin,end) are correct. Test
 (begin-end+1)/3 = an integer, meaning that it is actually the sequence of a
 amino acid sequence.<br>
 If one or two nucleotides are included where they should not be, the
 algorithm will not work.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>fileName</code> - </dd>
</dl>
</li>
</ul>
<a id="getFullORF(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Genome)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFullORF</h4>
<pre class="methodSignature">public static&nbsp;<a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;getFullORF&#8203;(<a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
                                  <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</pre>
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in
 frame) containing the peptide (amino acid sequence) <br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>seq</code> - </dd>
<dd><code>genome</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getFullORF(java.lang.String,int,int,boolean,bacnet.datamodel.sequence.Genome)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFullORF</h4>
<pre class="methodSignature">public static&nbsp;<a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;getFullORF&#8203;(java.lang.String&nbsp;peptideSeq,
                                  int&nbsp;beginPeptide,
                                  int&nbsp;endPeptide,
                                  boolean&nbsp;strand,
                                  <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</pre>
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in
 frame) containing the peptide (amino acid sequence) <br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>peptideSeq</code> - </dd>
<dd><code>begin</code> - </dd>
<dd><code>strand</code> - </dd>
</dl>
</li>
</ul>
<a id="getStartCodonBefore()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getStartCodonBefore</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;getStartCodonBefore()</pre>
<div class="block">NOT FINISHED <br>
 <br>
 Search for the different start codon before a peptide, and the stop codon
 after this peptide<br>
 Print the nucleotide sequence plus the codon, and identify Start, STOp and
 the reference Peptide used</div>
</li>
</ul>
<a id="searchStopCodon(bacnet.datamodel.sequence.Genome,int,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>searchStopCodon</h4>
<pre class="methodSignature">public static&nbsp;java.lang.String&nbsp;searchStopCodon&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                                               int&nbsp;bpPosition,
                                               boolean&nbsp;strand)</pre>
<div class="block">Get the next stop codon in frame and dowstream to a bpPosition</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genome</code> - </dd>
<dd><code>bpPosition</code> - </dd>
<dd><code>strand</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="countRTerminal(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>countRTerminal</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;countRTerminal&#8203;(java.lang.String&nbsp;massSpecName)</pre>
<div class="block">Count the number of peptides with an R (arginine), and finishing with an R</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>massSpecName</code> - </dd>
</dl>
</li>
</ul>
<a id="compareNTermData(java.lang.String,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>compareNTermData</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;compareNTermData&#8203;(java.lang.String&nbsp;NTermDataName1,
                                                          java.lang.String&nbsp;NTermDataName2)</pre>
<div class="block">Return the list of NTerm common to both experiment<br>
 NTerm are equal if:<br>
 <li>same peptide sequence
 <li>same modification</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>NTermDataName1</code> - </dd>
<dd><code>NTermDataName2</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="findSDSequenceforLMO()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>findSDSequenceforLMO</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;findSDSequenceforLMO()</pre>
<div class="block">Find SD binding free energy on region -20bp upstream to each lmo</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nameNTermData</code> - </dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/NTermUtils.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
