<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>NTermCreateData</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="NTermCreateData";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":9,"i1":9,"i2":9,"i3":9,"i4":10,"i5":10,"i6":9,"i7":9,"i8":9,"i9":41,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10,"i19":10,"i20":10,"i21":10,"i22":10,"i23":9,"i24":10,"i25":10,"i26":10,"i27":10,"i28":10,"i29":10,"i30":10,"i31":10,"i32":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"],32:["t6","Deprecated Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/NTermCreateData.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.proteomics</a></div>
<h2 title="Class NTermCreateData" class="title">Class NTermCreateData</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>bacnet.datamodel.proteomics.NTermCreateData</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<pre>public class <span class="typeNameLabel">NTermCreateData</span>
extends java.lang.Object</pre>
<div class="block">Transform excel table describing all N-Terminal peptides found in Listeria<br>
 <li>Find the position of each peptide in Listeria genome
 <li>create a NGS data from it</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>christophebecavin</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private <a href="../sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#chromosome">chromosome</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#CURATEDARRAY_CURATED">CURATEDARRAY_CURATED</a></span></code></th>
<td class="colLast">
<div class="block">List of column names for the final curated table of mapping</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#genomeAA">genomeAA</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#genomeAACompl">genomeAACompl</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#genomeAAMinus">genomeAAMinus</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#genomeAAMinusCompl">genomeAAMinusCompl</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#genomeAAPlus">genomeAAPlus</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#genomeAAPlusCompl">genomeAAPlusCompl</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#nameNTermData">nameNTermData</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#nameRawData">nameRawData</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String[][]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#nTermArray">nTermArray</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#PATH_RAW_DATA">PATH_RAW_DATA</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#PATH_RESULTS">PATH_RESULTS</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#RAWARRAY_COLUMN">RAWARRAY_COLUMN</a></span></code></th>
<td class="colLast">
<div class="block">List of column names when PATHARRYA contains only one experiment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#RAWARRAY_MULTICOLUMN">RAWARRAY_MULTICOLUMN</a></span></code></th>
<td class="colLast">
<div class="block">List of column names when PATHARRYA contains multiple experiments</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(bacnet.datamodel.sequence.Genome)">NTermCreateData</a></span>&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Initialize a void NTermCreateData object and run <code>initializeGenomeSequence()</code> to have
 the genome on the six frames</div>
</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.lang.String,java.lang.String)">NTermCreateData</a></span>&#8203;(java.lang.String&nbsp;nameRawData,
               java.lang.String&nbsp;nameNTermData)</code></th>
<td class="colLast">
<div class="block">Create table summarizing the data<br>
 Curate the table by adding different information<br>
 Create streaming data for the viewer<br>
 
 Run Mapping and creation of summary tables<br>
 mapPeptide();<br>
 curateTable();<br>
 <br>
 Create all NTerm from the table<br>
 createNTermExperiment();<br>
 NTermCreation.findStart(nameNTermData);<br>
 NTermCreation.findTSS(nameNTermData);<br>
 NTermCreation.findCodonUsage(nameNTermData);<br>
 NTermCreation.findSDSequence(nameNTermData);<br>
 <br>
 Complete NTermExp by creating all HashMaps<br>
 NTermData nTermExp = NTermData.load(nameNTermData);<br>
 nTermExp.createHashMap();<br>
 nTermExp.createGeneHashMap();<br>
 nTermExp.createProteolyseMap();<br>
 nTermExp.createAnnotation();<br>
 nTermExp.save();<br></div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t6" class="tableTab"><span><a href="javascript:show(32);">Deprecated Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addMaXQuantInfo(bacnet.datamodel.dataset.NTermData,java.lang.String)">addMaXQuantInfo</a></span>&#8203;(<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&nbsp;NTermData,
               java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Compare to peptides from MaxQuant software analysis<br>
 For each NTerm search if the same peptide and the same modif has been found in MaxQuant analysis.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addMaXQuantInfo(bacnet.datamodel.dataset.NTermData,java.lang.String,boolean)">addMaXQuantInfo</a></span>&#8203;(<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&nbsp;nTermData,
               java.lang.String&nbsp;fileName,
               boolean&nbsp;multipleFiles)</code></th>
<td class="colLast">
<div class="block">Add MaxQuant information for each peptide <br>
 This function is used only when multiple files are available</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addMaXQuantInfo(bacnet.datamodel.dataset.NTermData,java.lang.String,int)">addMaXQuantInfo</a></span>&#8203;(<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&nbsp;NTermData,
               java.lang.String&nbsp;fileName,
               int&nbsp;indexTable)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>static <a href="../expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createBioCondition(java.lang.String)">createBioCondition</a></span>&#8203;(java.lang.String&nbsp;nTermName)</code></th>
<td class="colLast">
<div class="block">Create a BioCondition containing only one Nterm</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createNTermData()">createNTermData</a></span>()</code></th>
<td class="colLast">
<div class="block">Create an <code>NTermExperiment</code> from PATH_ARRAY+"_Curated.txt"</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#curateTable()">curateTable</a></span>()</code></th>
<td class="colLast">
<div class="block">Modify result table, and verify if map peptides are well mapped<br>
 For each peptide we extract position found, get the AA sequence from it, and check if it is the
 same than peptide sequence</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#findCodonUsage(java.lang.String)">findCodonUsage</a></span>&#8203;(java.lang.String&nbsp;nameNTermData)</code></th>
<td class="colLast">
<div class="block">Get Codon usage of the peptide</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#findSDSequence(java.lang.String)">findSDSequence</a></span>&#8203;(java.lang.String&nbsp;nameNTermData)</code></th>
<td class="colLast">
<div class="block">For each <code>NTerm</code> calculates anti Shine-Dalgarno sequence affinity in the region -20bp
 before peptide on the genome</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#findStart(java.lang.String)">findStart</a></span>&#8203;(java.lang.String&nbsp;nameNTermData)</code></th>
<td class="colLast">
<div class="block">Search if the first amino acid or the amino acid before is a StartCodon</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#findTSS(java.lang.String)">findTSS</a></span>&#8203;(java.lang.String&nbsp;nameNTermData)</code></th>
<td class="colLast">
<div class="block"><span class="deprecatedLabel">Deprecated.</span></div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code><a href="../sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getChromosome()">getChromosome</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGenomeAA()">getGenomeAA</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGenomeAACompl()">getGenomeAACompl</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGenomeAAMinus()">getGenomeAAMinus</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGenomeAAMinusCompl()">getGenomeAAMinusCompl</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGenomeAAPlus()">getGenomeAAPlus</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGenomeAAPlusCompl()">getGenomeAAPlusCompl</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getNameNTermData()">getNameNTermData</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getNameRawData()">getNameRawData</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#initializeGenomeSequence(bacnet.datamodel.sequence.Genome)">initializeGenomeSequence</a></span>&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Get all 6 possible amino acids sequence of the genome<br>
 <br>
 Initialize:<br>
 genomeAA - In frame sequence
 genomeAACompl - In frame sequence complementary
 genomeAAMinus - Minus -1 bp sequence
 genomeAAMinusCompl - Minus -1 bp sequence complementary
 genomeAAPlus - Plus 1 bp sequence
 genomeAAPlusCompl - Plus 1 bp sequence complementary <br>
 <br>
 Can save these genomes in NTermUtils.getPATH()+"Genome aminoacid mapping/"</div>
</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#mapPeptide()">mapPeptide</a></span>()</code></th>
<td class="colLast">
<div class="block">On each of the 6 frame genome, map the different peptide</div>
</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;<a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#mapSequence(java.lang.String)">mapSequence</a></span>&#8203;(java.lang.String&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Map an amino acide sequence to 6 frame Genomes<br>
 From initializeGenomeSequence() we get all 6 possible amino acids sequence of the genome, and map
 the peptide<br>
 The different mapping coordinates are transformed into nucleotide coordinates (= genome
 coordinates)</div>
</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#modifyPathArray()">modifyPathArray</a></span>()</code></th>
<td class="colLast">
<div class="block">Check PATHARRAY and had columns if needed<br>
 <br>
 
 Two types of array are possible:
 one with only one experiment (contains only one column with "# spectra")
 one with only multiple experiments (contains many columns containing "# spectra") <br>
 <br>
 For The first type:<br>
 Check if PATHARRAY table is complete = has 8 columns, first column is the list of peptide,
 soluble and insoluble contains "TRUE" and "FALSE" and no "yes"<br>
 Change "TRUE" and "FALSE" to "yes" and "no"
 Remove line with NH2-free peptide <br>
 <br>
 For the second type:<br></div>
</td>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#run(java.lang.String,java.lang.String)">run</a></span>&#8203;(java.lang.String&nbsp;nameRawData,
   java.lang.String&nbsp;nameNTermData)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setChromosome(bacnet.datamodel.sequence.Chromosome)">setChromosome</a></span>&#8203;(<a href="../sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromosome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGenomeAA(java.lang.String)">setGenomeAA</a></span>&#8203;(java.lang.String&nbsp;genomeAA)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGenomeAACompl(java.lang.String)">setGenomeAACompl</a></span>&#8203;(java.lang.String&nbsp;genomeAACompl)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i27" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGenomeAAMinus(java.lang.String)">setGenomeAAMinus</a></span>&#8203;(java.lang.String&nbsp;genomeAAMinus)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i28" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGenomeAAMinusCompl(java.lang.String)">setGenomeAAMinusCompl</a></span>&#8203;(java.lang.String&nbsp;genomeAAMinusCompl)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i29" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGenomeAAPlus(java.lang.String)">setGenomeAAPlus</a></span>&#8203;(java.lang.String&nbsp;genomeAAPlus)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i30" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGenomeAAPlusCompl(java.lang.String)">setGenomeAAPlusCompl</a></span>&#8203;(java.lang.String&nbsp;genomeAAPlusCompl)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i31" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setNameNTermData(java.lang.String)">setNameNTermData</a></span>&#8203;(java.lang.String&nbsp;nameNTermData)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i32" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setNameRawData(java.lang.String)">setNameRawData</a></span>&#8203;(java.lang.String&nbsp;nameRawData)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="PATH_RAW_DATA">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PATH_RAW_DATA</h4>
<pre>public static&nbsp;java.lang.String PATH_RAW_DATA</pre>
</li>
</ul>
<a id="PATH_RESULTS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PATH_RESULTS</h4>
<pre>public static&nbsp;java.lang.String PATH_RESULTS</pre>
</li>
</ul>
<a id="RAWARRAY_COLUMN">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>RAWARRAY_COLUMN</h4>
<pre>public static&nbsp;java.lang.String[] RAWARRAY_COLUMN</pre>
<div class="block">List of column names when PATHARRYA contains only one experiment</div>
</li>
</ul>
<a id="RAWARRAY_MULTICOLUMN">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>RAWARRAY_MULTICOLUMN</h4>
<pre>public static&nbsp;java.lang.String[] RAWARRAY_MULTICOLUMN</pre>
<div class="block">List of column names when PATHARRYA contains multiple experiments</div>
</li>
</ul>
<a id="CURATEDARRAY_CURATED">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>CURATEDARRAY_CURATED</h4>
<pre>public static&nbsp;java.lang.String[] CURATEDARRAY_CURATED</pre>
<div class="block">List of column names for the final curated table of mapping</div>
</li>
</ul>
<a id="nameRawData">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>nameRawData</h4>
<pre>private&nbsp;java.lang.String nameRawData</pre>
</li>
</ul>
<a id="nameNTermData">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>nameNTermData</h4>
<pre>private&nbsp;java.lang.String nameNTermData</pre>
</li>
</ul>
<a id="chromosome">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>chromosome</h4>
<pre>private&nbsp;<a href="../sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a> chromosome</pre>
</li>
</ul>
<a id="nTermArray">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>nTermArray</h4>
<pre>private&nbsp;java.lang.String[][] nTermArray</pre>
</li>
</ul>
<a id="genomeAA">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>genomeAA</h4>
<pre>private&nbsp;java.lang.String genomeAA</pre>
</li>
</ul>
<a id="genomeAACompl">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>genomeAACompl</h4>
<pre>private&nbsp;java.lang.String genomeAACompl</pre>
</li>
</ul>
<a id="genomeAAMinus">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>genomeAAMinus</h4>
<pre>private&nbsp;java.lang.String genomeAAMinus</pre>
</li>
</ul>
<a id="genomeAAMinusCompl">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>genomeAAMinusCompl</h4>
<pre>private&nbsp;java.lang.String genomeAAMinusCompl</pre>
</li>
</ul>
<a id="genomeAAPlus">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>genomeAAPlus</h4>
<pre>private&nbsp;java.lang.String genomeAAPlus</pre>
</li>
</ul>
<a id="genomeAAPlusCompl">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>genomeAAPlusCompl</h4>
<pre>private&nbsp;java.lang.String genomeAAPlusCompl</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;(bacnet.datamodel.sequence.Genome)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>NTermCreateData</h4>
<pre>public&nbsp;NTermCreateData&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</pre>
<div class="block">Initialize a void NTermCreateData object and run <code>initializeGenomeSequence()</code> to have
 the genome on the six frames</div>
</li>
</ul>
<a id="&lt;init&gt;(java.lang.String,java.lang.String)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>NTermCreateData</h4>
<pre>public&nbsp;NTermCreateData&#8203;(java.lang.String&nbsp;nameRawData,
                       java.lang.String&nbsp;nameNTermData)</pre>
<div class="block">Create table summarizing the data<br>
 Curate the table by adding different information<br>
 Create streaming data for the viewer<br>
 
 Run Mapping and creation of summary tables<br>
 mapPeptide();<br>
 curateTable();<br>
 <br>
 Create all NTerm from the table<br>
 createNTermExperiment();<br>
 NTermCreation.findStart(nameNTermData);<br>
 NTermCreation.findTSS(nameNTermData);<br>
 NTermCreation.findCodonUsage(nameNTermData);<br>
 NTermCreation.findSDSequence(nameNTermData);<br>
 <br>
 Complete NTermExp by creating all HashMaps<br>
 NTermData nTermExp = NTermData.load(nameNTermData);<br>
 nTermExp.createHashMap();<br>
 nTermExp.createGeneHashMap();<br>
 nTermExp.createProteolyseMap();<br>
 nTermExp.createAnnotation();<br>
 nTermExp.save();<br></div>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="run(java.lang.String,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>run</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;run&#8203;(java.lang.String&nbsp;nameRawData,
                       java.lang.String&nbsp;nameNTermData)</pre>
</li>
</ul>
<a id="createBioCondition(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createBioCondition</h4>
<pre class="methodSignature">public static&nbsp;<a href="../expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&nbsp;createBioCondition&#8203;(java.lang.String&nbsp;nTermName)</pre>
<div class="block">Create a BioCondition containing only one Nterm</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nTermName</code> - name of the NTerm data</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="findStart(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>findStart</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;findStart&#8203;(java.lang.String&nbsp;nameNTermData)</pre>
<div class="block">Search if the first amino acid or the amino acid before is a StartCodon</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nameNTermData</code> - </dd>
</dl>
</li>
</ul>
<a id="findTSS(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>findTSS</h4>
<pre class="methodSignature">@Deprecated
public static&nbsp;void&nbsp;findTSS&#8203;(java.lang.String&nbsp;nameNTermData)</pre>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span></div>
<div class="block">Search for upstream and downstream TSS for each <code>NTerm</code><br>
 It uses EGDe_Complete_TSS data for this search <br><br>
 IT IS NOT WORKING since we changed <code>TSS</code> to <code>NGS</code> object</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nameNTermData</code> - </dd>
</dl>
</li>
</ul>
<a id="findCodonUsage(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>findCodonUsage</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;findCodonUsage&#8203;(java.lang.String&nbsp;nameNTermData)</pre>
<div class="block">Get Codon usage of the peptide</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nameNTermData</code> - </dd>
</dl>
</li>
</ul>
<a id="findSDSequence(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>findSDSequence</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;findSDSequence&#8203;(java.lang.String&nbsp;nameNTermData)</pre>
<div class="block">For each <code>NTerm</code> calculates anti Shine-Dalgarno sequence affinity in the region -20bp
 before peptide on the genome</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nameNTermData</code> - </dd>
</dl>
</li>
</ul>
<a id="addMaXQuantInfo(bacnet.datamodel.dataset.NTermData,java.lang.String,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addMaXQuantInfo</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;addMaXQuantInfo&#8203;(<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&nbsp;nTermData,
                                   java.lang.String&nbsp;fileName,
                                   boolean&nbsp;multipleFiles)</pre>
<div class="block">Add MaxQuant information for each peptide <br>
 This function is used only when multiple files are available</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>NTermData</code> - </dd>
<dd><code>fileName</code> - </dd>
<dd><code>multipleFiles</code> - </dd>
</dl>
</li>
</ul>
<a id="addMaXQuantInfo(bacnet.datamodel.dataset.NTermData,java.lang.String,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addMaXQuantInfo</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;addMaXQuantInfo&#8203;(<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&nbsp;NTermData,
                                   java.lang.String&nbsp;fileName,
                                   int&nbsp;indexTable)</pre>
</li>
</ul>
<a id="addMaXQuantInfo(bacnet.datamodel.dataset.NTermData,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addMaXQuantInfo</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;addMaXQuantInfo&#8203;(<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&nbsp;NTermData,
                                   java.lang.String&nbsp;fileName)</pre>
<div class="block">Compare to peptides from MaxQuant software analysis<br>
 For each NTerm search if the same peptide and the same modif has been found in MaxQuant analysis.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>fileName</code> - </dd>
</dl>
</li>
</ul>
<a id="mapPeptide()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>mapPeptide</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;mapPeptide()</pre>
<div class="block">On each of the 6 frame genome, map the different peptide</div>
</li>
</ul>
<a id="mapSequence(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>mapSequence</h4>
<pre class="methodSignature">private&nbsp;java.util.ArrayList&lt;<a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;mapSequence&#8203;(java.lang.String&nbsp;sequence)</pre>
<div class="block">Map an amino acide sequence to 6 frame Genomes<br>
 From initializeGenomeSequence() we get all 6 possible amino acids sequence of the genome, and map
 the peptide<br>
 The different mapping coordinates are transformed into nucleotide coordinates (= genome
 coordinates)</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sequence</code> - amino acid sequence to map</dd>
</dl>
</li>
</ul>
<a id="curateTable()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>curateTable</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;curateTable()</pre>
<div class="block">Modify result table, and verify if map peptides are well mapped<br>
 For each peptide we extract position found, get the AA sequence from it, and check if it is the
 same than peptide sequence</div>
</li>
</ul>
<a id="createNTermData()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createNTermData</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;createNTermData()</pre>
<div class="block">Create an <code>NTermExperiment</code> from PATH_ARRAY+"_Curated.txt"</div>
</li>
</ul>
<a id="initializeGenomeSequence(bacnet.datamodel.sequence.Genome)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>initializeGenomeSequence</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;initializeGenomeSequence&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</pre>
<div class="block">Get all 6 possible amino acids sequence of the genome<br>
 <br>
 Initialize:<br>
 <li>genomeAA - In frame sequence
 <li>genomeAACompl - In frame sequence complementary
 <li>genomeAAMinus - Minus -1 bp sequence
 <li>genomeAAMinusCompl - Minus -1 bp sequence complementary
 <li>genomeAAPlus - Plus 1 bp sequence
 <li>genomeAAPlusCompl - Plus 1 bp sequence complementary <br>
 <br>
 Can save these genomes in NTermUtils.getPATH()+"Genome aminoacid mapping/"</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genome</code> - to read</dd>
</dl>
</li>
</ul>
<a id="modifyPathArray()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>modifyPathArray</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;modifyPathArray()</pre>
<div class="block">Check PATHARRAY and had columns if needed<br>
 <br>
 
 Two types of array are possible:
 <li>one with only one experiment (contains only one column with "# spectra")
 <li>one with only multiple experiments (contains many columns containing "# spectra") <br>
 <br>
 For The first type:<br>
 Check if PATHARRAY table is complete = has 8 columns, first column is the list of peptide,
 soluble and insoluble contains "TRUE" and "FALSE" and no "yes"<br>
 <li>Change "TRUE" and "FALSE" to "yes" and "no"
 <li>Remove line with NH2-free peptide <br>
 <br>
 For the second type:<br></div>
</li>
</ul>
<a id="getNameRawData()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNameRawData</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getNameRawData()</pre>
</li>
</ul>
<a id="setNameRawData(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setNameRawData</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setNameRawData&#8203;(java.lang.String&nbsp;nameRawData)</pre>
</li>
</ul>
<a id="getNameNTermData()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNameNTermData</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getNameNTermData()</pre>
</li>
</ul>
<a id="setNameNTermData(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setNameNTermData</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setNameNTermData&#8203;(java.lang.String&nbsp;nameNTermData)</pre>
</li>
</ul>
<a id="getGenomeAA()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomeAA</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getGenomeAA()</pre>
</li>
</ul>
<a id="setGenomeAA(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGenomeAA</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGenomeAA&#8203;(java.lang.String&nbsp;genomeAA)</pre>
</li>
</ul>
<a id="getGenomeAACompl()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomeAACompl</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getGenomeAACompl()</pre>
</li>
</ul>
<a id="setGenomeAACompl(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGenomeAACompl</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGenomeAACompl&#8203;(java.lang.String&nbsp;genomeAACompl)</pre>
</li>
</ul>
<a id="getGenomeAAMinus()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomeAAMinus</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getGenomeAAMinus()</pre>
</li>
</ul>
<a id="setGenomeAAMinus(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGenomeAAMinus</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGenomeAAMinus&#8203;(java.lang.String&nbsp;genomeAAMinus)</pre>
</li>
</ul>
<a id="getGenomeAAMinusCompl()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomeAAMinusCompl</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getGenomeAAMinusCompl()</pre>
</li>
</ul>
<a id="setGenomeAAMinusCompl(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGenomeAAMinusCompl</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGenomeAAMinusCompl&#8203;(java.lang.String&nbsp;genomeAAMinusCompl)</pre>
</li>
</ul>
<a id="getGenomeAAPlus()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomeAAPlus</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getGenomeAAPlus()</pre>
</li>
</ul>
<a id="setGenomeAAPlus(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGenomeAAPlus</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGenomeAAPlus&#8203;(java.lang.String&nbsp;genomeAAPlus)</pre>
</li>
</ul>
<a id="getGenomeAAPlusCompl()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomeAAPlusCompl</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getGenomeAAPlusCompl()</pre>
</li>
</ul>
<a id="setGenomeAAPlusCompl(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGenomeAAPlusCompl</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGenomeAAPlusCompl&#8203;(java.lang.String&nbsp;genomeAAPlusCompl)</pre>
</li>
</ul>
<a id="getChromosome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromosome</h4>
<pre class="methodSignature">public&nbsp;<a href="../sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;getChromosome()</pre>
</li>
</ul>
<a id="setChromosome(bacnet.datamodel.sequence.Chromosome)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setChromosome</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setChromosome&#8203;(<a href="../sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromosome)</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/NTermCreateData.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
