<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>NTermUtils</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="declaration: package: bacnet.datamodel.proteomics, class: NTermUtils">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var evenRowColor = "even-row-color";
var oddRowColor = "odd-row-color";
var tableTab = "table-tab";
var activeTableTab = "active-table-tab";
var pathtoroot = "../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/NTermUtils.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html#class">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div>
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.proteomics</a></div>
<h1 title="Class NTermUtils" class="title">Class NTermUtils</h1>
</div>
<div class="inheritance" title="Inheritance Tree"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">java.lang.Object</a>
<div class="inheritance">bacnet.datamodel.proteomics.NTermUtils</div>
</div>
<section class="class-description" id="class-description">
<hr>
<div class="type-signature"><span class="modifiers">public class </span><span class="element-name type-name-label">NTermUtils</span>
<span class="extends-implements">extends <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></span></div>
</section>
<section class="summary">
<ul class="summary-list">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<li>
<section class="constructor-summary" id="constructor-summary">
<h2>Constructor Summary</h2>
<div class="caption"><span>Constructors</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E()" class="member-name-link">NTermUtils</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab1" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab1', 3)" class="table-tab">Static Methods</button><button id="method-summary-table-tab4" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab4', 3)" class="table-tab">Concrete Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel">
<div class="summary-table three-column-summary" aria-labelledby="method-summary-table-tab0">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#compareNTermData(java.lang.String,java.lang.String)" class="member-name-link">compareNTermData</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;NTermDataName1,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;NTermDataName2)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Return the list of NTerm common to both experiment<br>
 NTerm are equal if:<br>
 same peptide sequence
 same modification</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#countRTerminal(java.lang.String)" class="member-name-link">countRTerminal</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;massSpecName)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Count the number of peptides with an R (arginine), and finishing with an R</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#findSDSequenceforLMO()" class="member-name-link">findSDSequenceforLMO</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Find SD binding free energy on region -20bp upstream to each lmo</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getFullORF(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Genome)" class="member-name-link">getFullORF</a><wbr>(<a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
 <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in frame) containing the
 peptide (amino acid sequence) <br></div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getFullORF(java.lang.String,int,int,boolean,bacnet.datamodel.sequence.Genome)" class="member-name-link">getFullORF</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;peptideSeq,
 int&nbsp;beginPeptide,
 int&nbsp;endPeptide,
 boolean&nbsp;strand,
 <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in frame) containing the
 peptide (amino acid sequence) <br></div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getListOfFullORFfromPeptide(java.lang.String,bacnet.datamodel.sequence.Genome)" class="member-name-link">getListOfFullORFfromPeptide</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName,
 <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">From a list of peptide position retrieve the complete sequence of the ORF (in frame) containing
 the peptide (amino acid sequence) <br>
 WARNING: Be careful that your peptide info (begin,end) are correct.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getPATH()" class="member-name-link">getPATH</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getStartCodonBefore()" class="member-name-link">getStartCodonBefore</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">NOT FINISHED <br>
 <br>
 Search for the different start codon before a peptide, and the stop codon after this peptide<br>
 Print the nucleotide sequence plus the codon, and identify Start, STOp and the reference Peptide
 used</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#searchStopCodon(bacnet.datamodel.sequence.Genome,int,boolean)" class="member-name-link">searchStopCodon</a><wbr>(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 int&nbsp;bpPosition,
 boolean&nbsp;strand)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Get the next stop codon in frame and dowstream to a bpPosition</div>
</div>
</div>
</div>
</div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Object">Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)" title="class or interface in java.lang" class="external-link">equals</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#getClass()" title="class or interface in java.lang" class="external-link">getClass</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode()" title="class or interface in java.lang" class="external-link">hashCode</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notify()" title="class or interface in java.lang" class="external-link">notify</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notifyAll()" title="class or interface in java.lang" class="external-link">notifyAll</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#toString()" title="class or interface in java.lang" class="external-link">toString</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait()" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long)" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long,int)" title="class or interface in java.lang" class="external-link">wait</a></code></div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<li>
<section class="constructor-details" id="constructor-detail">
<h2>Constructor Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="&lt;init&gt;()">
<h3>NTermUtils</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name">NTermUtils</span>()</div>
</section>
</li>
</ul>
</section>
</li>
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="getPATH()">
<h3>getPATH</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getPATH</span>()</div>
</section>
</li>
<li>
<section class="detail" id="getListOfFullORFfromPeptide(java.lang.String,bacnet.datamodel.sequence.Genome)">
<h3>getListOfFullORFfromPeptide</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">getListOfFullORFfromPeptide</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName,
 <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</span></div>
<div class="block">From a list of peptide position retrieve the complete sequence of the ORF (in frame) containing
 the peptide (amino acid sequence) <br>
 WARNING: Be careful that your peptide info (begin,end) are correct. Test (begin-end+1)/3 = an
 integer, meaning that it is actually the sequence of a amino acid sequence.<br>
 If one or two nucleotides are included where they should not be, the algorithm will not work.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>fileName</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getFullORF(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Genome)">
<h3>getFullORF</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span>&nbsp;<span class="element-name">getFullORF</span><wbr><span class="parameters">(<a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
 <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</span></div>
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in frame) containing the
 peptide (amino acid sequence) <br></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>seq</code> - </dd>
<dd><code>genome</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getFullORF(java.lang.String,int,int,boolean,bacnet.datamodel.sequence.Genome)">
<h3>getFullORF</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span>&nbsp;<span class="element-name">getFullORF</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;peptideSeq,
 int&nbsp;beginPeptide,
 int&nbsp;endPeptide,
 boolean&nbsp;strand,
 <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</span></div>
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in frame) containing the
 peptide (amino acid sequence) <br></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>peptideSeq</code> - </dd>
<dd><code>begin</code> - </dd>
<dd><code>strand</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getStartCodonBefore()">
<h3>getStartCodonBefore</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">getStartCodonBefore</span>()</div>
<div class="block">NOT FINISHED <br>
 <br>
 Search for the different start codon before a peptide, and the stop codon after this peptide<br>
 Print the nucleotide sequence plus the codon, and identify Start, STOp and the reference Peptide
 used</div>
</section>
</li>
<li>
<section class="detail" id="searchStopCodon(bacnet.datamodel.sequence.Genome,int,boolean)">
<h3>searchStopCodon</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">searchStopCodon</span><wbr><span class="parameters">(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 int&nbsp;bpPosition,
 boolean&nbsp;strand)</span></div>
<div class="block">Get the next stop codon in frame and dowstream to a bpPosition</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>genome</code> - </dd>
<dd><code>bpPosition</code> - </dd>
<dd><code>strand</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="countRTerminal(java.lang.String)">
<h3>countRTerminal</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">countRTerminal</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;massSpecName)</span></div>
<div class="block">Count the number of peptides with an R (arginine), and finishing with an R</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>massSpecName</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="compareNTermData(java.lang.String,java.lang.String)">
<h3>compareNTermData</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;</span>&nbsp;<span class="element-name">compareNTermData</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;NTermDataName1,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;NTermDataName2)</span></div>
<div class="block">Return the list of NTerm common to both experiment<br>
 NTerm are equal if:<br>
 <li>same peptide sequence
 <li>same modification</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>NTermDataName1</code> - </dd>
<dd><code>NTermDataName2</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="findSDSequenceforLMO()">
<h3>findSDSequenceforLMO</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">findSDSequenceforLMO</span>()</div>
<div class="block">Find SD binding free energy on region -20bp upstream to each lmo</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>nameNTermData</code> - </dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
</div>
</div>
</body>
</html>
