<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>TIS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="TIS";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10,"i19":10,"i20":9,"i21":10,"i22":10,"i23":10,"i24":10,"i25":10,"i26":10,"i27":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/TIS.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.proteomics</a></div>
<h2 title="Class TIS" class="title">Class TIS</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li><a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">bacnet.datamodel.sequence.Sequence</a></li>
<li>
<ul class="inheritance">
<li>bacnet.datamodel.proteomics.TIS</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><code>java.io.Serializable</code></dd>
</dl>
<hr>
<pre>public class <span class="typeNameLabel">TIS</span>
extends <a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>
implements java.io.Serializable</pre>
<div class="block">This class is used to regroup different Nterm. The assumption is that we regroup in a TIS every
 Nterm comes from the same peptide but are now different due to proteolysis and C-Terminal ragging</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>UIBC</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../serialized-form.html#bacnet.datamodel.proteomics.TIS">Serialized Form</a></dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a id="nested.classes.inherited.from.class.bacnet.datamodel.sequence.Sequence">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;bacnet.datamodel.sequence.<a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></h3>
<code><a href="../sequence/Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a></code></li>
</ul>
</li>
</ul>
</section>
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;<a href="NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#modifs">modifs</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private <a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#nTermRef">nTermRef</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#nTerms">nTerms</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#overlapingGene">overlapingGene</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#overlaps">overlaps</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#refSequence">refSequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private static long</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#serialVersionUID">serialVersionUID</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#totalSpectra">totalSpectra</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="fields.inherited.from.class.bacnet.datamodel.sequence.Sequence">
<!--   -->
</a>
<h3>Fields inherited from class&nbsp;bacnet.datamodel.sequence.<a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></h3>
<code><a href="../sequence/Sequence.html#ANTI_SD_SEQ">ANTI_SD_SEQ</a>, <a href="../sequence/Sequence.html#DEFAULT_LENGTH">DEFAULT_LENGTH</a></code></li>
</ul>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">TIS</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#findModifs()">findModifs</a></span>()</code></th>
<td class="colLast">
<div class="block">List the type of modification overlap of each NTerm</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#findNTermRef()">findNTermRef</a></span>()</code></th>
<td class="colLast">
<div class="block">Find the Nterm with the highest spectrum</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#findOverlaps()">findOverlaps</a></span>()</code></th>
<td class="colLast">
<div class="block">List the type of gene overlap of each NTerm</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#findPosition()">findPosition</a></span>()</code></th>
<td class="colLast">
<div class="block">From all the nTerms (supposed in the same strand), get the minimal begin and the maximum end</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#findRefSequence()">findRefSequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Find the total sequence of all peptides</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getExperiment()">getExperiment</a></span>()</code></th>
<td class="colLast">
<div class="block">Return if the NTerms of this TIS were detected in the first experiment, or the second, or both</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getFormylatedExperiment()">getFormylatedExperiment</a></span>()</code></th>
<td class="colLast">
<div class="block">Search if the TIS was found formylated in the first experiment, or the second, or both</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getModifs()">getModifs</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getModifsString()">getModifsString</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code><a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getnTermRef()">getnTermRef</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getnTerms()">getnTerms</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getnTermsSequences()">getnTermsSequences</a></span>()</code></th>
<td class="colLast">
<div class="block">Return a character string with all peptide Modif sequences</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getnTermsString()">getnTermsString</a></span>()</code></th>
<td class="colLast">
<div class="block">Return a character string with all peptide names</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getOverlapingGene()">getOverlapingGene</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getOverlaps()">getOverlaps</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getOverlapsString()">getOverlapsString</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getRefSequence()">getRefSequence</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTotalSpectra()">getTotalSpectra</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getUTRLength()">getUTRLength</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isFormylated()">isFormylated</a></span>()</code></th>
<td class="colLast">
<div class="block">Return true if the TIS includes a formylated NTerm</div>
</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code>static <a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#load(java.lang.String)">load</a></span>&#8203;(java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setModifs(java.util.ArrayList)">setModifs</a></span>&#8203;(java.util.ArrayList&lt;<a href="NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a>&gt;&nbsp;modifs)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setnTermRef(bacnet.datamodel.proteomics.NTerm)">setnTermRef</a></span>&#8203;(<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTermRef)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setnTerms(java.util.ArrayList)">setnTerms</a></span>&#8203;(java.util.ArrayList&lt;<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;nTerms)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setOverlapingGene(java.lang.String)">setOverlapingGene</a></span>&#8203;(java.lang.String&nbsp;overlapingGene)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setOverlaps(java.util.ArrayList)">setOverlaps</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;overlaps)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setRefSequence(java.lang.String)">setRefSequence</a></span>&#8203;(java.lang.String&nbsp;refSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i27" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setTotalSpectra(int)">setTotalSpectra</a></span>&#8203;(int&nbsp;totalSpectra)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.bacnet.datamodel.sequence.Sequence">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;bacnet.datamodel.sequence.<a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></h3>
<code><a href="../sequence/Sequence.html#foundSDSequence()">foundSDSequence</a>, <a href="../sequence/Sequence.html#get3UTRSequence(int)">get3UTRSequence</a>, <a href="../sequence/Sequence.html#getBegin()">getBegin</a>, <a href="../sequence/Sequence.html#getChromosome()">getChromosome</a>, <a href="../sequence/Sequence.html#getChromosomeID()">getChromosomeID</a>, <a href="../sequence/Sequence.html#getComment()">getComment</a>, <a href="../sequence/Sequence.html#getConservation()">getConservation</a>, <a href="../sequence/Sequence.html#getConservationHashMap()">getConservationHashMap</a>, <a href="../sequence/Sequence.html#getEnd()">getEnd</a>, <a href="../sequence/Sequence.html#getFeature(java.lang.String)">getFeature</a>, <a href="../sequence/Sequence.html#getFeatures()">getFeatures</a>, <a href="../sequence/Sequence.html#getFeaturesText()">getFeaturesText</a>, <a href="../sequence/Sequence.html#getFeaturesTextForTable(java.lang.String)">getFeaturesTextForTable</a>, <a href="../sequence/Sequence.html#getGenome()">getGenome</a>, <a href="../sequence/Sequence.html#getGenomeName()">getGenomeName</a>, <a href="../sequence/Sequence.html#getId()">getId</a>, <a href="../sequence/Sequence.html#getLength()">getLength</a>, <a href="../sequence/Sequence.html#getName()">getName</a>, <a href="../sequence/Sequence.html#getRef()">getRef</a>, <a href="../sequence/Sequence.html#getSDSequence()">getSDSequence</a>, <a href="../sequence/Sequence.html#getSDSequence(int)">getSDSequence</a>, <a href="../sequence/Sequence.html#getSequence()">getSequence</a>, <a href="../sequence/Sequence.html#getSequenceAA()">getSequenceAA</a>, <a href="../sequence/Sequence.html#getSequenceAACurated()">getSequenceAACurated</a>, <a href="../sequence/Sequence.html#getSequenceAACurated(java.lang.String)">getSequenceAACurated</a>, <a href="../sequence/Sequence.html#getSequenceRNA()">getSequenceRNA</a>, <a href="../sequence/Sequence.html#getSerialversionuid()">getSerialversionuid</a>, <a href="../sequence/Sequence.html#getSignatures()">getSignatures</a>, <a href="../sequence/Sequence.html#getSignaturesToString()">getSignaturesToString</a>, <a href="../sequence/Sequence.html#getStrand()">getStrand</a>, <a href="../sequence/Sequence.html#getSynonym()">getSynonym</a>, <a href="../sequence/Sequence.html#getSynonymsText()">getSynonymsText</a>, <a href="../sequence/Sequence.html#getType()">getType</a>, <a href="../sequence/Sequence.html#isEqual(bacnet.datamodel.sequence.Sequence)">isEqual</a>, <a href="../sequence/Sequence.html#isOverlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)">isOverlap</a>, <a href="../sequence/Sequence.html#isOverlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence,int)">isOverlap</a>, <a href="../sequence/Sequence.html#isOverlapStart(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence,int)">isOverlapStart</a>, <a href="../sequence/Sequence.html#isSameStrand(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)">isSameStrand</a>, <a href="../sequence/Sequence.html#isStrand()">isStrand</a>, <a href="../sequence/Sequence.html#overlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)">overlap</a>, <a href="../sequence/Sequence.html#save(java.lang.String)">save</a>, <a href="../sequence/Sequence.html#setBegin(int)">setBegin</a>, <a href="../sequence/Sequence.html#setChromosomeID(java.lang.String)">setChromosomeID</a>, <a href="../sequence/Sequence.html#setComment(java.lang.String)">setComment</a>, <a href="../sequence/Sequence.html#setConservation(int)">setConservation</a>, <a href="../sequence/Sequence.html#setConservationHashMap(java.util.LinkedHashMap)">setConservationHashMap</a>, <a href="../sequence/Sequence.html#setEnd(int)">setEnd</a>, <a href="../sequence/Sequence.html#setFeatures(java.util.LinkedHashMap)">setFeatures</a>, <a href="../sequence/Sequence.html#setGenomeName(java.lang.String)">setGenomeName</a>, <a href="../sequence/Sequence.html#setId(java.lang.String)">setId</a>, <a href="../sequence/Sequence.html#setLength(int)">setLength</a>, <a href="../sequence/Sequence.html#setName(java.lang.String)">setName</a>, <a href="../sequence/Sequence.html#setRef(java.lang.String)">setRef</a>, <a href="../sequence/Sequence.html#setSignatures(java.util.TreeSet)">setSignatures</a>, <a href="../sequence/Sequence.html#setStrand(char)">setStrand</a>, <a href="../sequence/Sequence.html#setSynonym(java.util.TreeSet)">setSynonym</a>, <a href="../sequence/Sequence.html#setType(bacnet.datamodel.sequence.Sequence.SeqType)">setType</a>, <a href="../sequence/Sequence.html#toString()">toString</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="serialVersionUID">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>serialVersionUID</h4>
<pre>private static final&nbsp;long serialVersionUID</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../constant-values.html#bacnet.datamodel.proteomics.TIS.serialVersionUID">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a id="refSequence">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>refSequence</h4>
<pre>private&nbsp;java.lang.String refSequence</pre>
</li>
</ul>
<a id="nTermRef">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>nTermRef</h4>
<pre>private&nbsp;<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a> nTermRef</pre>
</li>
</ul>
<a id="totalSpectra">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>totalSpectra</h4>
<pre>private&nbsp;int totalSpectra</pre>
</li>
</ul>
<a id="overlapingGene">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>overlapingGene</h4>
<pre>private&nbsp;java.lang.String overlapingGene</pre>
</li>
</ul>
<a id="nTerms">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>nTerms</h4>
<pre>private&nbsp;java.util.ArrayList&lt;<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt; nTerms</pre>
</li>
</ul>
<a id="overlaps">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>overlaps</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.String&gt; overlaps</pre>
</li>
</ul>
<a id="modifs">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>modifs</h4>
<pre>private&nbsp;java.util.ArrayList&lt;<a href="NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a>&gt; modifs</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>TIS</h4>
<pre>public&nbsp;TIS()</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="findPosition()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>findPosition</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;findPosition()</pre>
<div class="block">From all the nTerms (supposed in the same strand), get the minimal begin and the maximum end</div>
</li>
</ul>
<a id="findNTermRef()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>findNTermRef</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;findNTermRef()</pre>
<div class="block">Find the Nterm with the highest spectrum</div>
</li>
</ul>
<a id="findRefSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>findRefSequence</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;findRefSequence()</pre>
<div class="block">Find the total sequence of all peptides</div>
</li>
</ul>
<a id="findOverlaps()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>findOverlaps</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;findOverlaps()</pre>
<div class="block">List the type of gene overlap of each NTerm</div>
</li>
</ul>
<a id="findModifs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>findModifs</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;findModifs()</pre>
<div class="block">List the type of modification overlap of each NTerm</div>
</li>
</ul>
<a id="getExperiment()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getExperiment</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getExperiment()</pre>
<div class="block">Return if the NTerms of this TIS were detected in the first experiment, or the second, or both</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getFormylatedExperiment()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFormylatedExperiment</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getFormylatedExperiment()</pre>
<div class="block">Search if the TIS was found formylated in the first experiment, or the second, or both</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="isFormylated()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isFormylated</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;isFormylated()</pre>
<div class="block">Return true if the TIS includes a formylated NTerm</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getUTRLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getUTRLength</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getUTRLength()</pre>
</li>
</ul>
<a id="load(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>load</h4>
<pre class="methodSignature">public static&nbsp;<a href="../sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;load&#8203;(java.lang.String&nbsp;fileName)</pre>
<div class="block">Read compressed, serialized data with a FileInputStream. Uncompress that data with a
 GZIPInputStream. Deserialize the vector of lines with a ObjectInputStream. Replace current data
 with new data, and redraw everything.</div>
</li>
</ul>
<a id="getRefSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRefSequence</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getRefSequence()</pre>
</li>
</ul>
<a id="setRefSequence(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setRefSequence</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setRefSequence&#8203;(java.lang.String&nbsp;refSequence)</pre>
</li>
</ul>
<a id="getnTerms()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getnTerms</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;getnTerms()</pre>
</li>
</ul>
<a id="getnTermsString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getnTermsString</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getnTermsString()</pre>
<div class="block">Return a character string with all peptide names</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getnTermsSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getnTermsSequences</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getnTermsSequences()</pre>
<div class="block">Return a character string with all peptide Modif sequences</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="setnTerms(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setnTerms</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setnTerms&#8203;(java.util.ArrayList&lt;<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&nbsp;nTerms)</pre>
</li>
</ul>
<a id="getOverlaps()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getOverlaps</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getOverlaps()</pre>
</li>
</ul>
<a id="getOverlapsString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getOverlapsString</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getOverlapsString()</pre>
</li>
</ul>
<a id="setOverlaps(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setOverlaps</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setOverlaps&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;overlaps)</pre>
</li>
</ul>
<a id="getOverlapingGene()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getOverlapingGene</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getOverlapingGene()</pre>
</li>
</ul>
<a id="setOverlapingGene(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setOverlapingGene</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setOverlapingGene&#8203;(java.lang.String&nbsp;overlapingGene)</pre>
</li>
</ul>
<a id="getModifs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getModifs</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a>&gt;&nbsp;getModifs()</pre>
</li>
</ul>
<a id="getModifsString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getModifsString</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getModifsString()</pre>
</li>
</ul>
<a id="setModifs(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setModifs</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setModifs&#8203;(java.util.ArrayList&lt;<a href="NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a>&gt;&nbsp;modifs)</pre>
</li>
</ul>
<a id="getnTermRef()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getnTermRef</h4>
<pre class="methodSignature">public&nbsp;<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;getnTermRef()</pre>
</li>
</ul>
<a id="setnTermRef(bacnet.datamodel.proteomics.NTerm)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setnTermRef</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setnTermRef&#8203;(<a href="NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTermRef)</pre>
</li>
</ul>
<a id="getTotalSpectra()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTotalSpectra</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getTotalSpectra()</pre>
</li>
</ul>
<a id="setTotalSpectra(int)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setTotalSpectra</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setTotalSpectra&#8203;(int&nbsp;totalSpectra)</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/TIS.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
