<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>Uses of Class org.biojava3.core.sequence.template.AbstractSequence</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Class org.biojava3.core.sequence.template.AbstractSequence";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Class org.biojava3.core.sequence.template.AbstractSequence" class="title">Uses of Class<br>org.biojava3.core.sequence.template.AbstractSequence</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.sequence">bacnet.datamodel.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.reader">bacnet.reader</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence">org.biojava3.core.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.features">org.biojava3.core.sequence.features</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.io">org.biojava3.core.sequence.io</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.io.template">org.biojava3.core.sequence.io.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.location">org.biojava3.core.sequence.location</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.template">org.biojava3.core.sequence.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a> in <a href="../../../../../../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></h3>
<table class="useSummary">
<caption><span>Subclasses of <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a> in <a href="../../../../../../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></span></code></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></span></code></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.reader">
<!--   -->
</a>
<h3>Uses of <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a> in <a href="../../../../../../bacnet/reader/package-summary.html">bacnet.reader</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../../../../bacnet/reader/package-summary.html">bacnet.reader</a> with type parameters of type <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../../bacnet/reader/NCBIFastaHeaderParser.html" title="class in bacnet.reader">NCBIFastaHeaderParser</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">My own implementation of a FastaHeaderParserInterface to parse Fasta files from NCBI ftp

 NCBI fna (genome and plasmids) gi|gi-number|ref|accession|name gi|16802048|ref|NC_003210.1|
 Listeria monocytogenes EGD-e, complete genome NCBI ffn (coding) ref|refGenome
 gi-number|:start-end name ref|NC_003210.1|:318-1673 chromosomal replication initiation protein
 [Listeria monocytogenes EGD-e] NCBI frn (non-coding) ref|refGenome gi-number|:start-end| name|
 [locus-tag] ref|NC_003210|:82705-82777|Lys tRNA| [locus_tag=lmot01] NCBI faa (proteins)
 gi|gi-number|ref|accession|name gi|16802049|ref|NP_463534.1| chromosomal replication initiation
 protein [Listeria monocytogenes EGD-e]</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a> in <a href="../../package-summary.html">org.biojava3.core.sequence</a></h3>
<table class="useSummary">
<caption><span>Subclasses of <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a> in <a href="../../package-summary.html">org.biojava3.core.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../BasicSequence.html" title="class in org.biojava3.core.sequence">BasicSequence</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Bare bones version of the Sequence object to be used sparingly.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></span></code></th>
<td class="colLast">
<div class="block">Represents a exon or coding sequence in a gene.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></span></code></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span></code></th>
<td class="colLast">
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</a></span></code></th>
<td class="colLast">
<div class="block">A gene contains a collection of Exon sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></span></code></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></span></code></th>
<td class="colLast">
<div class="block">RNASequence where RNACompoundSet are the allowed values</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../StartCodonSequence.html" title="class in org.biojava3.core.sequence">StartCodonSequence</a></span></code></th>
<td class="colLast">
<div class="block">Used to map the start codon feature on a gene</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../StopCodonSequence.html" title="class in org.biojava3.core.sequence">StopCodonSequence</a></span></code></th>
<td class="colLast">
<div class="block">Used to map the stop codon sequence on a gene</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></span></code></th>
<td class="colLast">
<div class="block">This is the sequence if you want to go from a gene sequence to a protein sequence.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../package-summary.html">org.biojava3.core.sequence</a> with parameters of type <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceComparator.</span><code><span class="memberNameLink"><a href="../../SequenceComparator.html#compare(org.biojava3.core.sequence.template.AbstractSequence,org.biojava3.core.sequence.template.AbstractSequence)">compare</a></span>&#8203;(<a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;?&gt;&nbsp;o1,
       <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;?&gt;&nbsp;o2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteinSequence.</span><code><span class="memberNameLink"><a href="../../ProteinSequence.html#setParentDNASequence(org.biojava3.core.sequence.template.AbstractSequence,java.lang.Integer,java.lang.Integer)">setParentDNASequence</a></span>&#8203;(<a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&nbsp;parentDNASequence,
                    java.lang.Integer&nbsp;begin,
                    java.lang.Integer&nbsp;end)</code></th>
<td class="colLast">
<div class="block">A Protein sequence can be stand alone or loaded from a transcript sequence.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.features">
<!--   -->
</a>
<h3>Uses of <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a> in <a href="../../features/package-summary.html">org.biojava3.core.sequence.features</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../features/package-summary.html">org.biojava3.core.sequence.features</a> with type parameters of type <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s) A
 feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Interface class to handle describing arbitrary features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/QualityFeature.html" title="class in org.biojava3.core.sequence.features">QualityFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">DNA Sequences produced by modern sequencers usually have quality informaion attached to them.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/QuantityFeature.html" title="class in org.biojava3.core.sequence.features">QuantityFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">It is common to have a numerical value or values associated with a feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/TextFeature.html" title="class in org.biojava3.core.sequence.features">TextFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">A implmentation of AbstractFeature</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.io">
<!--   -->
</a>
<h3>Uses of <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a> in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> with type parameters of type <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/GenericFastaHeaderFormat.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderFormat</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">We store the original header if the sequence is parsed from a fasta file and will use that exact
 sequence if we write out the sequences to a fasta file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/GenericFastaHeaderParser.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">The default fasta header parser where some headers are well defined based on the source database
 which allows us to set the source of the protein sequence and the identifier that can be used in
 future implementations to load features from external sources
 
 If the user has a custom header with local data then they can create their own implementation of
 a FastaHeaderParserInterface

 GenBank gi|gi-number|gb|accession|locus ENA Data Library gi|gi-number|emb|accession|locus DDBJ,
 DNA Database of Japan gi|gi-number|dbj|accession|locus NBRF PIR pir||entry Protein Research
 Foundation prf||name SWISS-PROT sp|accession|name Brookhaven Protein Data Bank (1)
 pdb|entry|chain Brookhaven Protein Data Bank (2) entry:chain|PDBID|CHAIN|SEQUENCE PDB EBI
 PDB:1ECY_A mol:protein length:142 ECOTIN Patents pat|country|number GenInfo Backbone Id
 bbs|number General database identifier gnl|database|identifier NCBI Reference Sequence
 ref|accession|locus Local Sequence identifier lcl|identifier</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> that return <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CasePreservingProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(java.lang.String&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CasePreservingProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(java.util.List)">getSequence</a></span>&#8203;(java.util.List&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></th>
<td class="colLast">
<div class="block">Assumes all compounds were uppercase</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CasePreservingProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence</a></span>&#8203;(<a href="../ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/DNASequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(java.lang.String&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/DNASequenceCreator.html#getSequence(java.util.List)">getSequence</a></span>&#8203;(java.util.List&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;list)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/DNASequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence</a></span>&#8203;(<a href="../ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileProxyDNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/FileProxyDNASequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(java.lang.String&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence
 and the offset index in the fasta file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileProxyDNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/FileProxyDNASequenceCreator.html#getSequence(java.util.List)">getSequence</a></span>&#8203;(java.util.List&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;list)</code></th>
<td class="colLast">
<div class="block">Not sure of use case and currently not supported</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileProxyDNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/FileProxyDNASequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence</a></span>&#8203;(<a href="../ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code></th>
<td class="colLast">
<div class="block">Should be able to extend the same concept to a remote URL call or database connection.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileProxyProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(java.lang.String&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence
 and the offset index in the fasta file.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileProxyProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(java.util.List)">getSequence</a></span>&#8203;(java.util.List&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></th>
<td class="colLast">
<div class="block">Not sure of use case and currently not supported</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileProxyProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence</a></span>&#8203;(<a href="../ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code></th>
<td class="colLast">
<div class="block">Should be able to extend the same concept to a remote URL call or database connection.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/ProteinSequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(java.lang.String&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/ProteinSequenceCreator.html#getSequence(java.util.List)">getSequence</a></span>&#8203;(java.util.List&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/ProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence</a></span>&#8203;(<a href="../ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/RNASequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(java.lang.String&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/RNASequenceCreator.html#getSequence(java.util.List)">getSequence</a></span>&#8203;(java.util.List&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;list)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/RNASequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence</a></span>&#8203;(<a href="../ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.io.template">
<!--   -->
</a>
<h3>Uses of <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a> in <a href="../../io/template/package-summary.html">org.biojava3.core.sequence.io.template</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../io/template/package-summary.html">org.biojava3.core.sequence.io.template</a> that return <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceCreatorInterface.</span><code><span class="memberNameLink"><a href="../../io/template/SequenceCreatorInterface.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(java.lang.String&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceCreatorInterface.</span><code><span class="memberNameLink"><a href="../../io/template/SequenceCreatorInterface.html#getSequence(java.util.List)">getSequence</a></span>&#8203;(java.util.List&lt;<a href="../../io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;&nbsp;list)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../../io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceCreatorInterface.</span><code><span class="memberNameLink"><a href="../../io/template/SequenceCreatorInterface.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence</a></span>&#8203;(<a href="../ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.location">
<!--   -->
</a>
<h3>Uses of <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a> in <a href="../../location/package-summary.html">org.biojava3.core.sequence.location</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../location/package-summary.html">org.biojava3.core.sequence.location</a> with type parameters of type <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../location/SequenceLocation.html" title="class in org.biojava3.core.sequence.location">SequenceLocation</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">A location in a sequence that keeps a reference to its parent sequence</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Fields in <a href="../../location/package-summary.html">org.biojava3.core.sequence.location</a> declared as <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>private <a href="../../location/SequenceLocation.html" title="type parameter in SequenceLocation">S</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceLocation.</span><code><span class="memberNameLink"><a href="../../location/SequenceLocation.html#sequence">sequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.template">
<!--   -->
</a>
<h3>Uses of <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a> in <a href="../package-summary.html">org.biojava3.core.sequence.template</a></h3>
<table class="useSummary">
<caption><span>Fields in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> declared as <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>private <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#parentSequence">parentSequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Fields in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> with type parameters of type <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;<a href="../../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#features">features</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="../../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#groupedFeatures">groupedFeatures</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> that return <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#getParentSequence()">getParentSequence</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> that return types with arguments of type <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#getFeatures()">getFeatures</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#getFeatures(int)">getFeatures</a></span>&#8203;(int&nbsp;bioSequencePosition)</code></th>
<td class="colLast">
<div class="block">Return features at a sequence position</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#getFeatures(java.lang.String,int)">getFeatures</a></span>&#8203;(java.lang.String&nbsp;featureType,
           int&nbsp;bioSequencePosition)</code></th>
<td class="colLast">
<div class="block">Return features at a sequence position by type</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType</a></span>&#8203;(java.lang.String&nbsp;type)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> with parameters of type <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)">setParentSequence</a></span>&#8203;(<a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;abstractSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> with type arguments of type <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#addFeature(int,int,org.biojava3.core.sequence.features.FeatureInterface)">addFeature</a></span>&#8203;(int&nbsp;bioStart,
          int&nbsp;bioEnd,
          <a href="../../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code></th>
<td class="colLast">
<div class="block">Method to help set the proper details for a feature as it relates to a sequence where the feature
 needs to have a location on the sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#addFeature(org.biojava3.core.sequence.features.FeatureInterface)">addFeature</a></span>&#8203;(<a href="../../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code></th>
<td class="colLast">
<div class="block">Add a feature to this sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#removeFeature(org.biojava3.core.sequence.features.FeatureInterface)">removeFeature</a></span>&#8203;(<a href="../../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code></th>
<td class="colLast">
<div class="block">Remove a feature from the sequence</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
