<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 18:10:05 CEST 2019 -->
<title>Uses of Interface org.biojava3.core.sequence.template.CompoundSet</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Interface org.biojava3.core.sequence.template.CompoundSet";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Interface org.biojava3.core.sequence.template.CompoundSet" class="title">Uses of Interface<br>org.biojava3.core.sequence.template.CompoundSet</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.sequence">bacnet.datamodel.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.reader">bacnet.reader</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence">org.biojava3.core.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.compound">org.biojava3.core.sequence.compound</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.io">org.biojava3.core.sequence.io</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.io.util">org.biojava3.core.sequence.io.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.loader">org.biojava3.core.sequence.loader</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.storage">org.biojava3.core.sequence.storage</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.template">org.biojava3.core.sequence.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.transcription">org.biojava3.core.sequence.transcription</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.views">org.biojava3.core.sequence.views</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../../../../../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></h3>
<table class="useSummary">
<caption><span>Constructors in <a href="../../../../../../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../../bacnet/datamodel/sequence/Chromosome.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">Chromosome</a></span>&#8203;(java.lang.String&nbsp;seqString,
          <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../../bacnet/datamodel/sequence/Chromosome.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">Chromosome</a></span>&#8203;(<a href="../SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
          <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">ChromosomeBacteriaSequence</a></span>&#8203;(java.lang.String&nbsp;seqString,
                          <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">ChromosomeBacteriaSequence</a></span>&#8203;(<a href="../SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
                          <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.reader">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../../../../../bacnet/reader/package-summary.html">bacnet.reader</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../../../bacnet/reader/package-summary.html">bacnet.reader</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../../../../../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">S</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaFileReader.</span><code><span class="memberNameLink"><a href="../../../../../../bacnet/reader/FastaFileReader.html#process(org.biojava3.core.sequence.template.CompoundSet)">process</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;?&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">The parsing is done in this method
 
 ******* There is a problem when the genes in the fasta file have the same Id,
 so I modify this method accordingly</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../package-summary.html">org.biojava3.core.sequence</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../package-summary.html">org.biojava3.core.sequence</a> that return <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleSequenceAlignment.</span><code><span class="memberNameLink"><a href="../../MultipleSequenceAlignment.html#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the Compounds defined in the first sequence</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../package-summary.html">org.biojava3.core.sequence</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../BasicSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">BasicSequence</a></span>&#8203;(java.lang.String&nbsp;sequence,
             <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../BasicSequence.html" title="type parameter in BasicSequence">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../ChromosomeSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">ChromosomeSequence</a></span>&#8203;(java.lang.String&nbsp;seqString,
                  <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Allows the creation of a ChromosomeSequence using String for the sequence
 with a custom CompoundSet</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../ChromosomeSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">ChromosomeSequence</a></span>&#8203;(<a href="../SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
                  <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Allows the creation of a ChromosomeSequence using a ProxyResequenceReader for
 the sequence with a custom CompoundSet</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../DNASequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">DNASequence</a></span>&#8203;(java.lang.String&nbsp;seqString,
           <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Create a sequence from a string with user defined compound set</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../DNASequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">DNASequence</a></span>&#8203;(<a href="../SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
           <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Create a sequence from a ProxySequencereader and user defined compound set</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../ProteinSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">ProteinSequence</a></span>&#8203;(java.lang.String&nbsp;seqString,
               <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Create a protein from a string with a user defined set of amino acids</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../ProteinSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader,org.biojava3.core.sequence.template.CompoundSet)">ProteinSequence</a></span>&#8203;(<a href="../ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
               <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">A protein sequence where the storage of the sequence is somewhere else with
 user defined set of amino acids.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../RNASequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">RNASequence</a></span>&#8203;(java.lang.String&nbsp;seqString,
           <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Create a RNA sequence from a string with a user defined RNA compound set</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../RNASequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader,org.biojava3.core.sequence.template.CompoundSet)">RNASequence</a></span>&#8203;(<a href="../ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
           <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Create a RNA sequence from a proxy reader and user defined RNA compound set</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.compound">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../compound/package-summary.html">org.biojava3.core.sequence.compound</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../compound/package-summary.html">org.biojava3.core.sequence.compound</a> that implement <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../compound/AmbiguityDNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">AmbiguityDNACompoundSet</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../compound/AmbiguityRNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">AmbiguityRNACompoundSet</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></span></code></th>
<td class="colLast">
<div class="block">Set of proteinogenic amino acids.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../compound/DNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">DNACompoundSet</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../compound/RNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">RNACompoundSet</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../compound/package-summary.html">org.biojava3.core.sequence.compound</a> that return <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidCompound.</span><code><span class="memberNameLink"><a href="../../compound/AminoAcidCompound.html#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../compound/package-summary.html">org.biojava3.core.sequence.compound</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../compound/NucleotideCompound.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,java.lang.String)">NucleotideCompound</a></span>&#8203;(java.lang.String&nbsp;base,
                  <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet,
                  java.lang.String&nbsp;complementStr)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../compound/NucleotideCompound.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,java.lang.String,org.biojava3.core.sequence.compound.NucleotideCompound%5B%5D)">NucleotideCompound</a></span>&#8203;(java.lang.String&nbsp;base,
                  <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet,
                  java.lang.String&nbsp;complementStr,
                  <a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>[]&nbsp;constituents)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.io">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> that return <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IUPACParser.IUPACTable.</span><code><span class="memberNameLink"><a href="../../io/IUPACParser.IUPACTable.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)">getCodonCompoundSet</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rnaCompounds,
                   <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcidCompounds)</code></th>
<td class="colLast">
<div class="block">Returns the compound set of codons</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IUPACParser.IUPACTable.</span><code><span class="memberNameLink"><a href="../../io/IUPACParser.IUPACTable.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)">getCodonCompoundSet</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rnaCompounds,
                   <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcidCompounds)</code></th>
<td class="colLast">
<div class="block">Returns the compound set of codons</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IUPACParser.IUPACTable.</span><code><span class="memberNameLink"><a href="../../io/IUPACParser.IUPACTable.html#getCodons(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)">getCodons</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;nucelotides,
         <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcids)</code></th>
<td class="colLast">
<div class="block">Returns a list of codons where the source and target compounds are the same
 as those given by the parameters.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/CasePreservingProteinSequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)">CasePreservingProteinSequenceCreator</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/DNASequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)">DNASequenceCreator</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/FileProxyDNASequenceCreator.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.template.CompoundSet)">FileProxyDNASequenceCreator</a></span>&#8203;(java.io.File&nbsp;fastaFile,
                           <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Need File so that we can store full path name in SequenceFileProxyLoader for
 Random File access as a quick read</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/FileProxyProteinSequenceCreator.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.template.CompoundSet)">FileProxyProteinSequenceCreator</a></span>&#8203;(java.io.File&nbsp;fastaFile,
                               <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Need File so that we can store full path name in SequenceFileProxyLoader for
 Random File access as a quick read</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/ProteinSequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)">ProteinSequenceCreator</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/RNASequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)">RNASequenceCreator</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.io.util">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../io/util/package-summary.html">org.biojava3.core.sequence.io.util</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../io/util/package-summary.html">org.biojava3.core.sequence.io.util</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IOUtils.</span><code><span class="memberNameLink"><a href="../../io/util/IOUtils.html#getGCGType(org.biojava3.core.sequence.template.CompoundSet)">getGCGType</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;C&gt;&nbsp;cs)</code></th>
<td class="colLast">
<div class="block">Determines GCG type</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.loader">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../loader/package-summary.html">org.biojava3.core.sequence.loader</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../loader/package-summary.html">org.biojava3.core.sequence.loader</a> that return <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceFileProxyLoader.</span><code><span class="memberNameLink"><a href="../../loader/SequenceFileProxyLoader.html#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StringProxySequenceReader.</span><code><span class="memberNameLink"><a href="../../loader/StringProxySequenceReader.html#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">UniprotProxySequenceReader.</span><code><span class="memberNameLink"><a href="../../loader/UniprotProxySequenceReader.html#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../loader/package-summary.html">org.biojava3.core.sequence.loader</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceFileProxyLoader.</span><code><span class="memberNameLink"><a href="../../loader/SequenceFileProxyLoader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StringProxySequenceReader.</span><code><span class="memberNameLink"><a href="../../loader/StringProxySequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">UniprotProxySequenceReader.</span><code><span class="memberNameLink"><a href="../../loader/UniprotProxySequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../loader/package-summary.html">org.biojava3.core.sequence.loader</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/SequenceFileProxyLoader.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.io.template.SequenceParserInterface,long,int,org.biojava3.core.sequence.template.CompoundSet)">SequenceFileProxyLoader</a></span>&#8203;(java.io.File&nbsp;file,
                       <a href="../../io/template/SequenceParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceParserInterface</a>&nbsp;sequenceParser,
                       long&nbsp;sequenceStartIndex,
                       int&nbsp;sequenceLength,
                       <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/StringProxySequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">StringProxySequenceReader</a></span>&#8203;(java.lang.String&nbsp;sequence,
                         <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/UniprotProxySequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">UniprotProxySequenceReader</a></span>&#8203;(java.lang.String&nbsp;accession,
                          <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">The uniprot id is used to retrieve the uniprot XML which is then parsed as a
 DOM object so we know everything about the protein.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.storage">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../storage/package-summary.html">org.biojava3.core.sequence.storage</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../storage/package-summary.html">org.biojava3.core.sequence.storage</a> that return <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ArrayListSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/ArrayListSequenceReader.html#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BitSequenceReader.BitArrayWorker.</span><code><span class="memberNameLink"><a href="../../storage/BitSequenceReader.BitArrayWorker.html#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the compound set backing this store</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/BitSequenceReader.html" title="type parameter in BitSequenceReader">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BitSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/BitSequenceReader.html#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the compound set backing this store</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">JoiningSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/JoiningSequenceReader.html#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SingleCompoundSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/SingleCompoundSequenceReader.html#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the compound set given at construction</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../storage/package-summary.html">org.biojava3.core.sequence.storage</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceAsStringHelper.</span><code><span class="memberNameLink"><a href="../../storage/SequenceAsStringHelper.html#getSequenceAsString(java.util.List,org.biojava3.core.sequence.template.CompoundSet,java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)">getSequenceAsString</a></span>&#8203;(java.util.List&lt;<a href="../../storage/SequenceAsStringHelper.html" title="type parameter in SequenceAsStringHelper">C</a>&gt;&nbsp;parsedCompounds,
                   <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/SequenceAsStringHelper.html" title="type parameter in SequenceAsStringHelper">C</a>&gt;&nbsp;compoundSet,
                   java.lang.Integer&nbsp;bioBegin,
                   java.lang.Integer&nbsp;bioEnd,
                   <a href="../../Strand.html" title="enum in org.biojava3.core.sequence">Strand</a>&nbsp;strand)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ArrayListSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/ArrayListSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BitSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/BitSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/BitSequenceReader.html" title="type parameter in BitSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Class is immutable & so this is unsupported</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">JoiningSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/JoiningSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SingleCompoundSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/SingleCompoundSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Unsupported</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../storage/package-summary.html">org.biojava3.core.sequence.storage</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/ArrayListSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">ArrayListSequenceReader</a></span>&#8203;(java.lang.String&nbsp;sequence,
                       <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/ArrayListSequenceReader.html#%3Cinit%3E(java.util.List,org.biojava3.core.sequence.template.CompoundSet)">ArrayListSequenceReader</a></span>&#8203;(java.util.List&lt;<a href="../../storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>&gt;&nbsp;compounds,
                       <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/BitSequenceReader.BitArrayWorker.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">BitArrayWorker</a></span>&#8203;(java.lang.String&nbsp;sequence,
              <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/BitSequenceReader.BitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int)">BitArrayWorker</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</a>&gt;&nbsp;compoundSet,
              int&nbsp;length)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/BitSequenceReader.BitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int%5B%5D)">BitArrayWorker</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</a>&gt;&nbsp;compoundSet,
              int[]&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">FourBitArrayWorker</a></span>&#8203;(java.lang.String&nbsp;sequence,
                  <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int)">FourBitArrayWorker</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</a>&gt;&nbsp;compoundSet,
                  int&nbsp;length)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int%5B%5D)">FourBitArrayWorker</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</a>&gt;&nbsp;compoundSet,
                  int[]&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/FourBitSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">FourBitSequenceReader</a></span>&#8203;(java.lang.String&nbsp;sequence,
                     <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/FourBitSequenceReader.html" title="type parameter in FourBitSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/FourBitSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.AccessionID)">FourBitSequenceReader</a></span>&#8203;(java.lang.String&nbsp;sequence,
                     <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/FourBitSequenceReader.html" title="type parameter in FourBitSequenceReader">C</a>&gt;&nbsp;compoundSet,
                     <a href="../../AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/JoiningSequenceReader.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,java.util.List)">JoiningSequenceReader</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&nbsp;compoundSet,
                     java.util.List&lt;<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&gt;&nbsp;sequences)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/JoiningSequenceReader.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.Sequence...)">JoiningSequenceReader</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&nbsp;compoundSet,
                     <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;...&nbsp;sequences)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/SingleCompoundSequenceReader.html#%3Cinit%3E(C,org.biojava3.core.sequence.template.CompoundSet,int)">SingleCompoundSequenceReader</a></span>&#8203;(<a href="../../storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a>&nbsp;compound,
                            <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a>&gt;&nbsp;compoundSet,
                            int&nbsp;length)</code></th>
<td class="colLast">
<div class="block">Build the object with a compound rather than a String</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/SingleCompoundSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,int)">SingleCompoundSequenceReader</a></span>&#8203;(java.lang.String&nbsp;compound,
                            <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a>&gt;&nbsp;compoundSet,
                            int&nbsp;length)</code></th>
<td class="colLast">
<div class="block">Public constructor to be used with String based constructor</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/TwoBitSequenceReader.TwoBitArrayWorker.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">TwoBitArrayWorker</a></span>&#8203;(java.lang.String&nbsp;sequence,
                 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/TwoBitSequenceReader.TwoBitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int)">TwoBitArrayWorker</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</a>&gt;&nbsp;compoundSet,
                 int&nbsp;length)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/TwoBitSequenceReader.TwoBitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int%5B%5D)">TwoBitArrayWorker</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</a>&gt;&nbsp;compoundSet,
                 int[]&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/TwoBitSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">TwoBitSequenceReader</a></span>&#8203;(java.lang.String&nbsp;sequence,
                    <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/TwoBitSequenceReader.html" title="type parameter in TwoBitSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/TwoBitSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.AccessionID)">TwoBitSequenceReader</a></span>&#8203;(java.lang.String&nbsp;sequence,
                    <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/TwoBitSequenceReader.html" title="type parameter in TwoBitSequenceReader">C</a>&gt;&nbsp;compoundSet,
                    <a href="../../AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.template">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../package-summary.html">org.biojava3.core.sequence.template</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> that implement <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractNucleotideCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractNucleotideCompoundSet</a>&lt;C extends <a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> that return <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../LightweightProfile.html" title="type parameter in LightweightProfile">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">LightweightProfile.</span><code><span class="memberNameLink"><a href="../LightweightProfile.html#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template"><code>CompoundSet</code></a> of all <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a>s</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../Sequence.html" title="type parameter in Sequence">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Sequence.</span><code><span class="memberNameLink"><a href="../Sequence.html#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">
<div class="block">Gets the compound set used to back this Sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceProxyView.</span><code><span class="memberNameLink"><a href="../SequenceProxyView.html#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractCompoundTranslator.</span><code><span class="memberNameLink"><a href="../AbstractCompoundTranslator.html#getFromCompoundSet()">getFromCompoundSet</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractCompoundTranslator.</span><code><span class="memberNameLink"><a href="../AbstractCompoundTranslator.html#getToCompoundSet()">getToCompoundSet</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceReader.</span><code><span class="memberNameLink"><a href="../SequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../SequenceReader.html" title="type parameter in SequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractCompoundTranslator.html#%3Cinit%3E(org.biojava3.core.sequence.io.template.SequenceCreatorInterface,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)">AbstractCompoundTranslator</a></span>&#8203;(<a href="../../io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;&nbsp;creator,
                          <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a>&gt;&nbsp;fromCompoundSet,
                          <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;&nbsp;toCompoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">AbstractSequence</a></span>&#8203;(java.lang.String&nbsp;seqString,
                <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Create a Sequence from a simple string where the values should be found in
 compoundSet</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">AbstractSequence</a></span>&#8203;(<a href="../SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;proxyLoader,
                <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">A ProxySequenceReader allows abstraction of both the storage of the sequence
 data and the location of the sequence data.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.transcription">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> that return <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.html#getAminoAcidCompounds()">getAminoAcidCompounds</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Table.</span><code><span class="memberNameLink"><a href="../../transcription/Table.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)">getCodonCompoundSet</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rnaCompounds,
                   <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcidCompounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.html#getDnaCompounds()">getDnaCompounds</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.html#getRnaCompounds()">getRnaCompounds</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.Builder.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.Builder.html#aminoAcidsCompounds(org.biojava3.core.sequence.template.CompoundSet)">aminoAcidsCompounds</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.Builder.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.Builder.html#dnaCompounds(org.biojava3.core.sequence.template.CompoundSet)">dnaCompounds</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Table.</span><code><span class="memberNameLink"><a href="../../transcription/Table.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)">getCodonCompoundSet</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rnaCompounds,
                   <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcidCompounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Table.</span><code><span class="memberNameLink"><a href="../../transcription/Table.html#getCodons(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)">getCodons</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;nucelotides,
         <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcids)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.Builder.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.Builder.html#rnaCompounds(org.biojava3.core.sequence.template.CompoundSet)">rnaCompounds</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../transcription/DNAToRNATranslator.html#%3Cinit%3E(org.biojava3.core.sequence.io.template.SequenceCreatorInterface,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet,boolean)">DNAToRNATranslator</a></span>&#8203;(<a href="../../io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rnaCreator,
                  <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna,
                  <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rna,
                  boolean&nbsp;shortCutTranslation)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../transcription/RNAToAminoAcidTranslator.html#%3Cinit%3E(org.biojava3.core.sequence.io.template.SequenceCreatorInterface,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.transcription.Table,boolean,boolean,boolean)">RNAToAminoAcidTranslator</a></span>&#8203;(<a href="../../io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;creator,
                        <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;nucleotides,
                        <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;&nbsp;codons,
                        <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcids,
                        <a href="../../transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</a>&nbsp;table,
                        boolean&nbsp;trimStops,
                        boolean&nbsp;initMetOnly,
                        boolean&nbsp;translateNCodons)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.views">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../views/package-summary.html">org.biojava3.core.sequence.views</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../views/package-summary.html">org.biojava3.core.sequence.views</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RnaSequenceView.</span><code><span class="memberNameLink"><a href="../../views/RnaSequenceView.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../views/package-summary.html">org.biojava3.core.sequence.views</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../views/RnaSequenceView.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence,org.biojava3.core.sequence.template.CompoundSet)">RnaSequenceView</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;sourceDna,
               <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rnaCompounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
