<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Fri Dec 10 09:15:34 CET 2021 -->
<title>Uses of Interface org.biojava3.core.sequence.template.CompoundSet</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-10">
<meta name="description" content="use: package: org.biojava3.core.sequence.template, interface: CompoundSet">
<meta name="generator" content="javadoc/ClassUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-use-page">
<script type="text/javascript">var pathtoroot = "../../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">Class</a></li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Interface org.biojava3.core.sequence.template.CompoundSet" class="title">Uses of Interface<br>org.biojava3.core.sequence.template.CompoundSet</h1>
</div>
<div class="caption"><span>Packages that use <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#bacnet.datamodel.sequence">bacnet.datamodel.sequence</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.reader">bacnet.reader</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava3.core.sequence">org.biojava3.core.sequence</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava3.core.sequence.compound">org.biojava3.core.sequence.compound</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava3.core.sequence.io">org.biojava3.core.sequence.io</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava3.core.sequence.io.util">org.biojava3.core.sequence.io.util</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava3.core.sequence.loader">org.biojava3.core.sequence.loader</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava3.core.sequence.storage">org.biojava3.core.sequence.storage</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava3.core.sequence.template">org.biojava3.core.sequence.template</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava3.core.sequence.transcription">org.biojava3.core.sequence.transcription</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava3.core.sequence.views">org.biojava3.core.sequence.views</a></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<section class="class-uses">
<ul class="block-list">
<li>
<section class="detail" id="bacnet.datamodel.sequence">
<h2>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../../../../../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></h2>
<div class="caption"><span>Constructors in <a href="../../../../../../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../../../../../bacnet/datamodel/sequence/Chromosome.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">Chromosome</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../../../../../bacnet/datamodel/sequence/Chromosome.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">Chromosome</a><wbr>(<a href="../SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../../../../../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ChromosomeBacteriaSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../../../../../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ChromosomeBacteriaSequence</a><wbr>(<a href="../SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.reader">
<h2>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../../../../../bacnet/reader/package-summary.html">bacnet.reader</a></h2>
<div class="caption"><span>Methods in <a href="../../../../../../bacnet/reader/package-summary.html">bacnet.reader</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/LinkedHashMap.html" title="class or interface in java.util" class="external-link">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="../../../../../../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">S</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FastaFileReader.</span><code><a href="../../../../../../bacnet/reader/FastaFileReader.html#process(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">process</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;?&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">The parsing is done in this method
 
 ******* There is a problem when the genes in the fasta file have the same Id, so I modify this
 method accordingly</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava3.core.sequence">
<h2>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../package-summary.html">org.biojava3.core.sequence</a></h2>
<div class="caption"><span>Methods in <a href="../../package-summary.html">org.biojava3.core.sequence</a> that return <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">C</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleSequenceAlignment.</span><code><a href="../../MultipleSequenceAlignment.html#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Get the Compounds defined in the first sequence</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../../package-summary.html">org.biojava3.core.sequence</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../BasicSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">BasicSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../BasicSequence.html" title="type parameter in BasicSequence">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../ChromosomeSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ChromosomeSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">Allows the creation of a ChromosomeSequence using String for the sequence with a custom
 CompoundSet</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../ChromosomeSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ChromosomeSequence</a><wbr>(<a href="../SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Allows the creation of a ChromosomeSequence using a ProxyResequenceReader for the sequence with a
 custom CompoundSet</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../DNASequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">DNASequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create a sequence from a string with user defined compound set</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../DNASequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">DNASequence</a><wbr>(<a href="../SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Create a sequence from a ProxySequencereader and user defined compound set</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../ProteinSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ProteinSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create a protein from a string with a user defined set of amino acids</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../ProteinSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ProteinSequence</a><wbr>(<a href="../ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">A protein sequence where the storage of the sequence is somewhere else with user defined set of
 amino acids.</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../RNASequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">RNASequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create a RNA sequence from a string with a user defined RNA compound set</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../RNASequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">RNASequence</a><wbr>(<a href="../ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Create a RNA sequence from a proxy reader and user defined RNA compound set</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava3.core.sequence.compound">
<h2>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../compound/package-summary.html">org.biojava3.core.sequence.compound</a></h2>
<div class="caption"><span>Classes in <a href="../../compound/package-summary.html">org.biojava3.core.sequence.compound</a> that implement <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../compound/AmbiguityDNACompoundSet.html" class="type-name-link" title="class in org.biojava3.core.sequence.compound">AmbiguityDNACompoundSet</a></code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../compound/AmbiguityRNACompoundSet.html" class="type-name-link" title="class in org.biojava3.core.sequence.compound">AmbiguityRNACompoundSet</a></code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../compound/AminoAcidCompoundSet.html" class="type-name-link" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></code></div>
<div class="col-last even-row-color">
<div class="block">Set of proteinogenic amino acids.</div>
</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../compound/DNACompoundSet.html" class="type-name-link" title="class in org.biojava3.core.sequence.compound">DNACompoundSet</a></code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../compound/RNACompoundSet.html" class="type-name-link" title="class in org.biojava3.core.sequence.compound">RNACompoundSet</a></code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../compound/package-summary.html">org.biojava3.core.sequence.compound</a> that return <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AminoAcidCompound.</span><code><a href="../../compound/AminoAcidCompound.html#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../../compound/package-summary.html">org.biojava3.core.sequence.compound</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../compound/NucleotideCompound.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,java.lang.String)" class="member-name-link">NucleotideCompound</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;base,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;complementStr)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../compound/NucleotideCompound.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,java.lang.String,org.biojava3.core.sequence.compound.NucleotideCompound%5B%5D)" class="member-name-link">NucleotideCompound</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;base,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;complementStr,
 <a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>[]&nbsp;constituents)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava3.core.sequence.io">
<h2>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a></h2>
<div class="caption"><span>Methods in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> that return <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">IUPACParser.IUPACTable.</span><code><a href="../../io/IUPACParser.IUPACTable.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">getCodonCompoundSet</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rnaCompounds,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcidCompounds)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the compound set of codons</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">IUPACParser.IUPACTable.</span><code><a href="../../io/IUPACParser.IUPACTable.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">getCodonCompoundSet</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rnaCompounds,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcidCompounds)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the compound set of codons</div>
</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">IUPACParser.IUPACTable.</span><code><a href="../../io/IUPACParser.IUPACTable.html#getCodons(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">getCodons</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;nucelotides,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcids)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns a list of codons where the source and target compounds are the same as those given by the
 parameters.</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../io/CasePreservingProteinSequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">CasePreservingProteinSequenceCreator</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../io/DNASequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">DNASequenceCreator</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../io/FileProxyDNASequenceCreator.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">FileProxyDNASequenceCreator</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html" title="class or interface in java.io" class="external-link">File</a>&nbsp;fastaFile,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Need File so that we can store full path name in SequenceFileProxyLoader for Random File access
 as a quick read</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../io/FileProxyProteinSequenceCreator.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">FileProxyProteinSequenceCreator</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html" title="class or interface in java.io" class="external-link">File</a>&nbsp;fastaFile,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">Need File so that we can store full path name in SequenceFileProxyLoader for Random File access
 as a quick read</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../io/ProteinSequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ProteinSequenceCreator</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../io/RNASequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">RNASequenceCreator</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava3.core.sequence.io.util">
<h2>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../io/util/package-summary.html">org.biojava3.core.sequence.io.util</a></h2>
<div class="caption"><span>Methods in <a href="../../io/util/package-summary.html">org.biojava3.core.sequence.io.util</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">IOUtils.</span><code><a href="../../io/util/IOUtils.html#getGCGType(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">getGCGType</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;C&gt;&nbsp;cs)</code></div>
<div class="col-last even-row-color">
<div class="block">Determines GCG type</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava3.core.sequence.loader">
<h2>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../loader/package-summary.html">org.biojava3.core.sequence.loader</a></h2>
<div class="caption"><span>Methods in <a href="../../loader/package-summary.html">org.biojava3.core.sequence.loader</a> that return <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SequenceFileProxyLoader.</span><code><a href="../../loader/SequenceFileProxyLoader.html#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StringProxySequenceReader.</span><code><a href="../../loader/StringProxySequenceReader.html#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">UniprotProxySequenceReader.</span><code><a href="../../loader/UniprotProxySequenceReader.html#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../loader/package-summary.html">org.biojava3.core.sequence.loader</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SequenceFileProxyLoader.</span><code><a href="../../loader/SequenceFileProxyLoader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StringProxySequenceReader.</span><code><a href="../../loader/StringProxySequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">UniprotProxySequenceReader.</span><code><a href="../../loader/UniprotProxySequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../../loader/package-summary.html">org.biojava3.core.sequence.loader</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../loader/SequenceFileProxyLoader.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.io.template.SequenceParserInterface,long,int,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">SequenceFileProxyLoader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html" title="class or interface in java.io" class="external-link">File</a>&nbsp;file,
 <a href="../../io/template/SequenceParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceParserInterface</a>&nbsp;sequenceParser,
 long&nbsp;sequenceStartIndex,
 int&nbsp;sequenceLength,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../loader/StringProxySequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">StringProxySequenceReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../loader/UniprotProxySequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">UniprotProxySequenceReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">The uniprot id is used to retrieve the uniprot XML which is then parsed as a DOM object so we
 know everything about the protein.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava3.core.sequence.storage">
<h2>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../storage/package-summary.html">org.biojava3.core.sequence.storage</a></h2>
<div class="caption"><span>Methods in <a href="../../storage/package-summary.html">org.biojava3.core.sequence.storage</a> that return <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ArrayListSequenceReader.</span><code><a href="../../storage/ArrayListSequenceReader.html#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">BitSequenceReader.BitArrayWorker.</span><code><a href="../../storage/BitSequenceReader.BitArrayWorker.html#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns the compound set backing this store</div>
</div>
<div class="col-first even-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/BitSequenceReader.html" title="type parameter in BitSequenceReader">C</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">BitSequenceReader.</span><code><a href="../../storage/BitSequenceReader.html#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the compound set backing this store</div>
</div>
<div class="col-first odd-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">JoiningSequenceReader.</span><code><a href="../../storage/JoiningSequenceReader.html#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SingleCompoundSequenceReader.</span><code><a href="../../storage/SingleCompoundSequenceReader.html#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the compound set given at construction</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../storage/package-summary.html">org.biojava3.core.sequence.storage</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">SequenceAsStringHelper.</span><code><a href="../../storage/SequenceAsStringHelper.html#getSequenceAsString(java.util.List,org.biojava3.core.sequence.template.CompoundSet,java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)" class="member-name-link">getSequenceAsString</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../storage/SequenceAsStringHelper.html" title="type parameter in SequenceAsStringHelper">C</a>&gt;&nbsp;parsedCompounds,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/SequenceAsStringHelper.html" title="type parameter in SequenceAsStringHelper">C</a>&gt;&nbsp;compoundSet,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;bioBegin,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;bioEnd,
 <a href="../../Strand.html" title="enum in org.biojava3.core.sequence">Strand</a>&nbsp;strand)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">ArrayListSequenceReader.</span><code><a href="../../storage/ArrayListSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">BitSequenceReader.</span><code><a href="../../storage/BitSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/BitSequenceReader.html" title="type parameter in BitSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Class is immutable &amp; so this is unsupported</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">JoiningSequenceReader.</span><code><a href="../../storage/JoiningSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SingleCompoundSequenceReader.</span><code><a href="../../storage/SingleCompoundSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Unsupported</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../../storage/package-summary.html">org.biojava3.core.sequence.storage</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../storage/ArrayListSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ArrayListSequenceReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../storage/ArrayListSequenceReader.html#%3Cinit%3E(java.util.List,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ArrayListSequenceReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>&gt;&nbsp;compounds,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../storage/BitSequenceReader.BitArrayWorker.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">BitArrayWorker</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../storage/BitSequenceReader.BitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int)" class="member-name-link">BitArrayWorker</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</a>&gt;&nbsp;compoundSet,
 int&nbsp;length)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../storage/BitSequenceReader.BitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int%5B%5D)" class="member-name-link">BitArrayWorker</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</a>&gt;&nbsp;compoundSet,
 int[]&nbsp;sequence)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">FourBitArrayWorker</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int)" class="member-name-link">FourBitArrayWorker</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</a>&gt;&nbsp;compoundSet,
 int&nbsp;length)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int%5B%5D)" class="member-name-link">FourBitArrayWorker</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</a>&gt;&nbsp;compoundSet,
 int[]&nbsp;sequence)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../storage/FourBitSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">FourBitSequenceReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/FourBitSequenceReader.html" title="type parameter in FourBitSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../storage/FourBitSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.AccessionID)" class="member-name-link">FourBitSequenceReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/FourBitSequenceReader.html" title="type parameter in FourBitSequenceReader">C</a>&gt;&nbsp;compoundSet,
 <a href="../../AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../storage/JoiningSequenceReader.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,java.util.List)" class="member-name-link">JoiningSequenceReader</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&nbsp;compoundSet,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&gt;&nbsp;sequences)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../storage/JoiningSequenceReader.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.Sequence...)" class="member-name-link">JoiningSequenceReader</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&nbsp;compoundSet,
 <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;...&nbsp;sequences)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../storage/SingleCompoundSequenceReader.html#%3Cinit%3E(C,org.biojava3.core.sequence.template.CompoundSet,int)" class="member-name-link">SingleCompoundSequenceReader</a><wbr>(<a href="../../storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a>&nbsp;compound,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a>&gt;&nbsp;compoundSet,
 int&nbsp;length)</code></div>
<div class="col-last even-row-color">
<div class="block">Build the object with a compound rather than a String</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../storage/SingleCompoundSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,int)" class="member-name-link">SingleCompoundSequenceReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;compound,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a>&gt;&nbsp;compoundSet,
 int&nbsp;length)</code></div>
<div class="col-last odd-row-color">
<div class="block">Public constructor to be used with String based constructor</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../storage/TwoBitSequenceReader.TwoBitArrayWorker.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">TwoBitArrayWorker</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../storage/TwoBitSequenceReader.TwoBitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int)" class="member-name-link">TwoBitArrayWorker</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</a>&gt;&nbsp;compoundSet,
 int&nbsp;length)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../storage/TwoBitSequenceReader.TwoBitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int%5B%5D)" class="member-name-link">TwoBitArrayWorker</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</a>&gt;&nbsp;compoundSet,
 int[]&nbsp;sequence)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../storage/TwoBitSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">TwoBitSequenceReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/TwoBitSequenceReader.html" title="type parameter in TwoBitSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../storage/TwoBitSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.AccessionID)" class="member-name-link">TwoBitSequenceReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/TwoBitSequenceReader.html" title="type parameter in TwoBitSequenceReader">C</a>&gt;&nbsp;compoundSet,
 <a href="../../AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava3.core.sequence.template">
<h2>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../package-summary.html">org.biojava3.core.sequence.template</a></h2>
<div class="caption"><span>Classes in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> that implement <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../AbstractCompoundSet.html" class="type-name-link" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../AbstractNucleotideCompoundSet.html" class="type-name-link" title="class in org.biojava3.core.sequence.template">AbstractNucleotideCompoundSet</a>&lt;C extends <a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> that return <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AbstractSequence.</span><code><a href="../AbstractSequence.html#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../LightweightProfile.html" title="type parameter in LightweightProfile">C</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">LightweightProfile.</span><code><a href="../LightweightProfile.html#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template"><code>CompoundSet</code></a> of all <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a>s</div>
</div>
<div class="col-first even-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../Sequence.html" title="type parameter in Sequence">C</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Sequence.</span><code><a href="../Sequence.html#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Gets the compound set used to back this Sequence</div>
</div>
<div class="col-first odd-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SequenceProxyView.</span><code><a href="../SequenceProxyView.html#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AbstractCompoundTranslator.</span><code><a href="../AbstractCompoundTranslator.html#getFromCompoundSet()" class="member-name-link">getFromCompoundSet</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AbstractCompoundTranslator.</span><code><a href="../AbstractCompoundTranslator.html#getToCompoundSet()" class="member-name-link">getToCompoundSet</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AbstractSequence.</span><code><a href="../AbstractSequence.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SequenceReader.</span><code><a href="../SequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../SequenceReader.html" title="type parameter in SequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../AbstractCompoundTranslator.html#%3Cinit%3E(org.biojava3.core.sequence.io.template.SequenceCreatorInterface,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">AbstractCompoundTranslator</a><wbr>(<a href="../../io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;&nbsp;creator,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a>&gt;&nbsp;fromCompoundSet,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;&nbsp;toCompoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../AbstractSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">AbstractSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create a Sequence from a simple string where the values should be found in compoundSet</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../AbstractSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">AbstractSequence</a><wbr>(<a href="../SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;proxyLoader,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">A ProxySequenceReader allows abstraction of both the storage of the sequence data and the
 location of the sequence data.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava3.core.sequence.transcription">
<h2>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></h2>
<div class="caption"><span>Methods in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> that return <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">TranscriptionEngine.</span><code><a href="../../transcription/TranscriptionEngine.html#getAminoAcidCompounds()" class="member-name-link">getAminoAcidCompounds</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Table.</span><code><a href="../../transcription/Table.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">getCodonCompoundSet</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rnaCompounds,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcidCompounds)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">TranscriptionEngine.</span><code><a href="../../transcription/TranscriptionEngine.html#getDnaCompounds()" class="member-name-link">getDnaCompounds</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">TranscriptionEngine.</span><code><a href="../../transcription/TranscriptionEngine.html#getRnaCompounds()" class="member-name-link">getRnaCompounds</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../../transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">TranscriptionEngine.Builder.</span><code><a href="../../transcription/TranscriptionEngine.Builder.html#aminoAcidsCompounds(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">aminoAcidsCompounds</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compounds)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../../transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">TranscriptionEngine.Builder.</span><code><a href="../../transcription/TranscriptionEngine.Builder.html#dnaCompounds(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">dnaCompounds</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compounds)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Table.</span><code><a href="../../transcription/Table.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">getCodonCompoundSet</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rnaCompounds,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcidCompounds)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Table.</span><code><a href="../../transcription/Table.html#getCodons(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">getCodons</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;nucelotides,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcids)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../../transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">TranscriptionEngine.Builder.</span><code><a href="../../transcription/TranscriptionEngine.Builder.html#rnaCompounds(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">rnaCompounds</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compounds)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../transcription/DNAToRNATranslator.html#%3Cinit%3E(org.biojava3.core.sequence.io.template.SequenceCreatorInterface,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet,boolean)" class="member-name-link">DNAToRNATranslator</a><wbr>(<a href="../../io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rnaCreator,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rna,
 boolean&nbsp;shortCutTranslation)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../transcription/RNAToAminoAcidTranslator.html#%3Cinit%3E(org.biojava3.core.sequence.io.template.SequenceCreatorInterface,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.transcription.Table,boolean,boolean,boolean)" class="member-name-link">RNAToAminoAcidTranslator</a><wbr>(<a href="../../io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;creator,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;nucleotides,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;&nbsp;codons,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcids,
 <a href="../../transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</a>&nbsp;table,
 boolean&nbsp;trimStops,
 boolean&nbsp;initMetOnly,
 boolean&nbsp;translateNCodons)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava3.core.sequence.views">
<h2>Uses of <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a> in <a href="../../views/package-summary.html">org.biojava3.core.sequence.views</a></h2>
<div class="caption"><span>Methods in <a href="../../views/package-summary.html">org.biojava3.core.sequence.views</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">RnaSequenceView.</span><code><a href="../../views/RnaSequenceView.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet</a><wbr>(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../../views/package-summary.html">org.biojava3.core.sequence.views</a> with parameters of type <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../views/RnaSequenceView.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">RnaSequenceView</a><wbr>(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;sourceDna,
 <a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rnaCompounds)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
</ul>
</section>
</main>
</div>
</div>
</body>
</html>
