<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 18:10:05 CEST 2019 -->
<title>Uses of Interface org.biojava3.core.sequence.template.Compound</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Interface org.biojava3.core.sequence.template.Compound";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Interface org.biojava3.core.sequence.template.Compound" class="title">Uses of Interface<br>org.biojava3.core.sequence.template.Compound</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.reader">bacnet.reader</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence">org.biojava3.core.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.compound">org.biojava3.core.sequence.compound</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.edits">org.biojava3.core.sequence.edits</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.features">org.biojava3.core.sequence.features</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.io">org.biojava3.core.sequence.io</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.io.template">org.biojava3.core.sequence.io.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.io.util">org.biojava3.core.sequence.io.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.loader">org.biojava3.core.sequence.loader</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.location">org.biojava3.core.sequence.location</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.location.template">org.biojava3.core.sequence.location.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.storage">org.biojava3.core.sequence.storage</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.template">org.biojava3.core.sequence.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.transcription">org.biojava3.core.sequence.transcription</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.views">org.biojava3.core.sequence.views</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="bacnet.reader">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../../../../../../bacnet/reader/package-summary.html">bacnet.reader</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../../../../bacnet/reader/package-summary.html">bacnet.reader</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../../bacnet/reader/NCBIFastaHeaderParser.html" title="class in bacnet.reader">NCBIFastaHeaderParser</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">My own implementation of a FastaHeaderParserInterface to parse Fasta files
 from NCBI ftp

 NCBI fna (genome and plasmids) gi|gi-number|ref|accession|name
 gi|16802048|ref|NC_003210.1| Listeria monocytogenes EGD-e, complete genome
 NCBI ffn (coding) ref|refGenome gi-number|:start-end name
 ref|NC_003210.1|:318-1673 chromosomal replication initiation protein
 [Listeria monocytogenes EGD-e] NCBI frn (non-coding) ref|refGenome
 gi-number|:start-end| name| [locus-tag] ref|NC_003210|:82705-82777|Lys tRNA|
 [locus_tag=lmot01] NCBI faa (proteins) gi|gi-number|ref|accession|name
 gi|16802049|ref|NP_463534.1| chromosomal replication initiation protein
 [Listeria monocytogenes EGD-e]</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../../package-summary.html">org.biojava3.core.sequence</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../package-summary.html">org.biojava3.core.sequence</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../BasicSequence.html" title="class in org.biojava3.core.sequence">BasicSequence</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Bare bones version of the Sequence object to be used sparingly.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a minimal data structure for reading and writing a sequence
 alignment.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.compound">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../../compound/package-summary.html">org.biojava3.core.sequence.compound</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../compound/package-summary.html">org.biojava3.core.sequence.compound</a> that implement <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></code></th>
<td class="colLast">
<div class="block">Used to describe an Amino Acid.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></span></code></th>
<td class="colLast">
<div class="block">Define a codon</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../compound/package-summary.html">org.biojava3.core.sequence.compound</a> with parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidCompound.</span><code><span class="memberNameLink"><a href="../../compound/AminoAcidCompound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase</a></span>&#8203;(<a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CodonCompound.</span><code><span class="memberNameLink"><a href="../../compound/CodonCompound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase</a></span>&#8203;(<a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NucleotideCompound.</span><code><span class="memberNameLink"><a href="../../compound/NucleotideCompound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase</a></span>&#8203;(<a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.edits">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../../edits/package-summary.html">org.biojava3.core.sequence.edits</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../edits/package-summary.html">org.biojava3.core.sequence.edits</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../edits/Edit.html" title="interface in org.biojava3.core.sequence.edits">Edit</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Interface for carrying out edit operations on a Sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../edits/Edit.AbstractEdit.html" title="class in org.biojava3.core.sequence.edits">Edit.AbstractEdit</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Abstract class which defines all edit operations as a call to discover what
 5' and 3' ends of an editing Sequence should be joined together with a target
 Sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../edits/Edit.Delete.html" title="class in org.biojava3.core.sequence.edits">Edit.Delete</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implementation which allows for the deletion of bases from a Sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../edits/Edit.Insert.html" title="class in org.biojava3.core.sequence.edits">Edit.Insert</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Edit implementation which allows us to insert a base at any position in a
 Sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../edits/Edit.Substitute.html" title="class in org.biojava3.core.sequence.edits">Edit.Substitute</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Allows for the substitution of bases into an existing Sequence.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.features">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../../features/package-summary.html">org.biojava3.core.sequence.features</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../features/package-summary.html">org.biojava3.core.sequence.features</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">A feature is currently any descriptive item that can be associated with a
 sequence position(s) A feature has a type and a source which is currently a
 string to allow flexibility for the user Ideally well defined features should
 have a class to describe attributes of that feature</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Interface class to handle describing arbitrary features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/QualityFeature.html" title="class in org.biojava3.core.sequence.features">QualityFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">DNA Sequences produced by modern sequencers usually have quality informaion
 attached to them.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/QuantityFeature.html" title="class in org.biojava3.core.sequence.features">QuantityFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">It is common to have a numerical value or values associated with a feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/TextFeature.html" title="class in org.biojava3.core.sequence.features">TextFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">A implmentation of AbstractFeature</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.io">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/FastaReader.html" title="class in org.biojava3.core.sequence.io">FastaReader</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Use FastaReaderHelper as an example of how to use this class where
 FastaReaderHelper should be the primary class used to read Fasta files</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">The FastaWriter writes a collection of sequences to an outputStream.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/GenericFastaHeaderFormat.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderFormat</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">We store the original header if the sequence is parsed from a fasta file and
 will use that exact sequence if we write out the sequences to a fasta file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/GenericFastaHeaderParser.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">The default fasta header parser where some headers are well defined based on
 the source database which allows us to set the source of the protein sequence
 and the identifier that can be used in future implementations to load
 features from external sources
 
 If the user has a custom header with local data then they can create their
 own implementation of a FastaHeaderParserInterface

 GenBank gi|gi-number|gb|accession|locus ENA Data Library
 gi|gi-number|emb|accession|locus DDBJ, DNA Database of Japan
 gi|gi-number|dbj|accession|locus NBRF PIR pir||entry Protein Research
 Foundation prf||name SWISS-PROT sp|accession|name Brookhaven Protein Data
 Bank (1) pdb|entry|chain Brookhaven Protein Data Bank (2)
 entry:chain|PDBID|CHAIN|SEQUENCE PDB EBI PDB:1ECY_A mol:protein length:142
 ECOTIN Patents pat|country|number GenInfo Backbone Id bbs|number General
 database identifier gnl|database|identifier NCBI Reference Sequence
 ref|accession|locus Local Sequence identifier lcl|identifier</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.io.template">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../../io/template/package-summary.html">org.biojava3.core.sequence.io.template</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../io/template/package-summary.html">org.biojava3.core.sequence.io.template</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/template/FastaHeaderFormatInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderFormatInterface</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/template/FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.io.util">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../../io/util/package-summary.html">org.biojava3.core.sequence.io.util</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../io/util/package-summary.html">org.biojava3.core.sequence.io.util</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IOUtils.</span><code><span class="memberNameLink"><a href="../../io/util/IOUtils.html#getGCGChecksum(java.util.List)">getGCGChecksum</a></span>&#8203;(java.util.List&lt;S&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Calculates GCG checksum for entire list of sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IOUtils.</span><code><span class="memberNameLink"><a href="../../io/util/IOUtils.html#getGCGChecksum(S)">getGCGChecksum</a></span>&#8203;(S&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Calculates GCG checksum for a given sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IOUtils.</span><code><span class="memberNameLink"><a href="../../io/util/IOUtils.html#getGCGHeader(java.util.List)">getGCGHeader</a></span>&#8203;(java.util.List&lt;S&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Assembles a GCG file header</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IOUtils.</span><code><span class="memberNameLink"><a href="../../io/util/IOUtils.html#getGCGType(org.biojava3.core.sequence.template.CompoundSet)">getGCGType</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;C&gt;&nbsp;cs)</code></th>
<td class="colLast">
<div class="block">Determines GCG type</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IOUtils.</span><code><span class="memberNameLink"><a href="../../io/util/IOUtils.html#getIDFormat(java.util.List)">getIDFormat</a></span>&#8203;(java.util.List&lt;S&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Creates format String for accession IDs</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.loader">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../../loader/package-summary.html">org.biojava3.core.sequence.loader</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../loader/package-summary.html">org.biojava3.core.sequence.loader</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/ArrayListProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">ArrayListProxySequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">This class represents the storage container of a sequence stored in a fasta
 file where the initial parsing of the file we store the offset and length of
 the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">An example of a ProxySequenceReader that is created from a String.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Pass in a Uniprot ID and this ProxySequenceReader when passed to a
 ProteinSequence will get the sequence data and other data elements associated
 with the ProteinSequence by Uniprot.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../loader/package-summary.html">org.biojava3.core.sequence.loader</a> with parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceFileProxyLoader.</span><code><span class="memberNameLink"><a href="../../loader/SequenceFileProxyLoader.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../../loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StringProxySequenceReader.</span><code><span class="memberNameLink"><a href="../../loader/StringProxySequenceReader.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../../loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">UniprotProxySequenceReader.</span><code><span class="memberNameLink"><a href="../../loader/UniprotProxySequenceReader.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../../loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.location">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../../location/package-summary.html">org.biojava3.core.sequence.location</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../location/package-summary.html">org.biojava3.core.sequence.location</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../location/SequenceLocation.html" title="class in org.biojava3.core.sequence.location">SequenceLocation</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">A location in a sequence that keeps a reference to its parent sequence</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.location.template">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../../location/template/package-summary.html">org.biojava3.core.sequence.location.template</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../location/template/package-summary.html">org.biojava3.core.sequence.location.template</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractLocation.</span><code><span class="memberNameLink"><a href="../../location/template/AbstractLocation.html#getRelevantSubSequence(org.biojava3.core.sequence.template.Sequence)">getRelevantSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Location.</span><code><span class="memberNameLink"><a href="../../location/template/Location.html#getRelevantSubSequence(org.biojava3.core.sequence.template.Sequence)">getRelevantSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Will return a SequenceReader object which offers a view of all resolved
 locations i.e. those locations which are not complex and define the true
 Sequence represented</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractLocation.</span><code><span class="memberNameLink"><a href="../../location/template/AbstractLocation.html#getSubSequence(org.biojava3.core.sequence.template.Sequence)">getSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">If circular this will return the sequence represented by the sub locations
 joined.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Location.</span><code><span class="memberNameLink"><a href="../../location/template/Location.html#getSubSequence(org.biojava3.core.sequence.template.Sequence)">getSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Will return a SequenceReader object which represents the outer bounds of this
 Location</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../location/template/package-summary.html">org.biojava3.core.sequence.location.template</a> that return types with arguments of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;? extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AccesionedLocation.</span><code><span class="memberNameLink"><a href="../../location/template/AccesionedLocation.html#getProxySequenceReader()">getProxySequenceReader</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the proxy reader used to get sequence for this location.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.storage">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../../storage/package-summary.html">org.biojava3.core.sequence.storage</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../storage/package-summary.html">org.biojava3.core.sequence.storage</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Stores a Sequence as a collection of compounds in an ArrayList</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">An implementation of the popular bit encodings.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">The logic of working with a bit has been separated out into this class to
 help developers create the bit data structures without having to put the code
 into an intermediate format and to also use the format without the need to
 copy this code.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/FourBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Four bit encoding of the bit formats.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader.FourBitArrayWorker</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">A four bit per compound implementation of the bit array worker code.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">This reader actually proxies onto multiple types of sequence in order to
 allow a number of sequence objects to act as if they are one sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/SequenceAsStringHelper.html" title="class in org.biojava3.core.sequence.storage">SequenceAsStringHelper</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">This is a common method that can be used across multiple storage/proxy
 implementations to handle Negative strand and other interesting elements of
 sequence data.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">An implementation of the SequenceReader interface which for every call will
 return only 1 compound (given to it during construction; a String is also
 valid but will require a CompoundSet).</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../storage/package-summary.html">org.biojava3.core.sequence.storage</a> with parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ArrayListSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/ArrayListSequenceReader.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../../storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BitSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/BitSequenceReader.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../../storage/BitSequenceReader.html" title="type parameter in BitSequenceReader">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">
<div class="block">Counts the number of times a compound appears in this sequence store</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">JoiningSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/JoiningSequenceReader.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SingleCompoundSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/SingleCompoundSequenceReader.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../../storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">
<div class="block">Delegates to
 <code>SequenceMixin#countCompounds(org.biojava3.core.sequence.template.Sequence, C[])</code></div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.template">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../package-summary.html">org.biojava3.core.sequence.template</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a>&lt;F extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>,&#8203;T extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a>&lt;F extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>,&#8203;T extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">The base class for DNA, RNA and Protein sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../CompoundTranslator.html" title="interface in org.biojava3.core.sequence.template">CompoundTranslator</a>&lt;F extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>,&#8203;T extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../CompoundTranslator.html" title="interface in org.biojava3.core.sequence.template">CompoundTranslator</a>&lt;F extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>,&#8203;T extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a minimal data structure for reading and writing a sequence
 alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Main interface for defining a collection of Compounds and accessing them
 using biological indexes</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceMixin.SequenceIterator.html" title="class in org.biojava3.core.sequence.template">SequenceMixin.SequenceIterator</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">A basic sequence iterator which iterates over the given Sequence by
 biological index.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Subinterfaces of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../package-summary.html">org.biojava3.core.sequence.template</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../ComplementCompound.html" title="interface in org.biojava3.core.sequence.template">ComplementCompound</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Classes in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> that implement <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></span></code></th>
<td class="colLast">
<div class="block">The details of a Compound</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#checksum(org.biojava3.core.sequence.template.Sequence)">checksum</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Performs a simple CRC64 checksum on any given sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#countCompounds(org.biojava3.core.sequence.template.Sequence,C...)">countCompounds</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
              C...&nbsp;compounds)</code></th>
<td class="colLast">
<div class="block">For the given vargs of compounds this method counts the number of times those
 compounds appear in the given sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>java.util.Iterator&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#createIterator(org.biojava3.core.sequence.template.Sequence)">createIterator</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Creates a simple sequence iterator which moves through a sequence going from
 1 to the length of the Sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br><a href="../SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#createSubSequence(org.biojava3.core.sequence.template.Sequence,int,int)">createSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
                 int&nbsp;start,
                 int&nbsp;end)</code></th>
<td class="colLast">
<div class="block">Creates a simple sub sequence view delimited by the given start and end.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>java.util.Map&lt;C,&#8203;java.lang.Integer&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#getComposition(org.biojava3.core.sequence.template.Sequence)">getComposition</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Does a linear scan over the given Sequence and records the number of times
 each base appears.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>java.util.Map&lt;C,&#8203;java.lang.Double&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#getDistribution(org.biojava3.core.sequence.template.Sequence)">getDistribution</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Analogous to <a href="../SequenceMixin.html#getComposition(org.biojava3.core.sequence.template.Sequence)"><code>SequenceMixin.getComposition(Sequence)</code></a> but returns the distribution
 of that <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template"><code>Compound</code></a> over the given sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#indexOf(org.biojava3.core.sequence.template.Sequence,C)">indexOf</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
       C&nbsp;compound)</code></th>
<td class="colLast">
<div class="block">Performs a linear search of the given Sequence for the given compound.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br><a href="../SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#inverse(org.biojava3.core.sequence.template.Sequence)">inverse</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">A method which attempts to do the right thing when is comes to a
 reverse/reverse complement</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#lastIndexOf(org.biojava3.core.sequence.template.Sequence,C)">lastIndexOf</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
           C&nbsp;compound)</code></th>
<td class="colLast">
<div class="block">Performs a reversed linear search of the given Sequence by wrapping it in a
 <a href="../../views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views"><code>ReversedSequenceView</code></a> and passing it into
 <a href="../SequenceMixin.html#indexOf(org.biojava3.core.sequence.template.Sequence,C)"><code>SequenceMixin.indexOf(Sequence, Compound)</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>java.util.List&lt;<a href="../SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;C&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#nonOverlappingKmers(org.biojava3.core.sequence.template.Sequence,int)">nonOverlappingKmers</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
                   int&nbsp;kmer)</code></th>
<td class="colLast">
<div class="block">Produces kmers of the specified size e.g.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>java.util.List&lt;<a href="../SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;C&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#overlappingKmers(org.biojava3.core.sequence.template.Sequence,int)">overlappingKmers</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
                int&nbsp;kmer)</code></th>
<td class="colLast">
<div class="block">Used to generate overlapping k-mers such i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#sequenceEquality(org.biojava3.core.sequence.template.Sequence,org.biojava3.core.sequence.template.Sequence)">sequenceEquality</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;source,
                <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;target)</code></th>
<td class="colLast">
<div class="block">A case-sensitive manner of comparing two sequence objects together.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#sequenceEqualityIgnoreCase(org.biojava3.core.sequence.template.Sequence,org.biojava3.core.sequence.template.Sequence)">sequenceEqualityIgnoreCase</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;source,
                          <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;target)</code></th>
<td class="colLast">
<div class="block">A case-insensitive manner of comparing two sequence objects together.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#shuffle(org.biojava3.core.sequence.template.Sequence)">shuffle</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Implements sequence shuffling by first materializing the given
 <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a> into a <code>List</code>, applying
 <code>Collections.shuffle(List)</code> and then returning the shuffled elements in
 a new instance of <code>SequenceBackingStore</code> which behaves as a
 <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>java.util.List&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#toList(org.biojava3.core.sequence.template.Sequence)">toList</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">For the given <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a> this will return a <code>List</code> filled with
 the Compounds of that <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#toString(org.biojava3.core.sequence.template.Sequence)">toString</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Shortcut to
 <a href="../SequenceMixin.html#toStringBuilder(org.biojava3.core.sequence.template.Sequence)"><code>SequenceMixin.toStringBuilder(org.biojava3.core.sequence.template.Sequence)</code></a> which
 calls toString() on the resulting object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>java.lang.StringBuilder</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#toStringBuilder(org.biojava3.core.sequence.template.Sequence)">toStringBuilder</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">For the given Sequence this will return a <code>StringBuilder</code> object filled
 with the results of <code>Compound#toString()</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#write(java.lang.Appendable,org.biojava3.core.sequence.template.Sequence)">write</a></span>&#8203;(java.lang.Appendable&nbsp;appendable,
     <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Used as a way of sending a Sequence to a writer without the cost of
 converting to a full length String and then writing the data out</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence.template</a> with parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Sequence.</span><code><span class="memberNameLink"><a href="../Sequence.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../Sequence.html" title="type parameter in Sequence">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">
<div class="block">Returns the number of times we found a compound in the Sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#countCompounds(org.biojava3.core.sequence.template.Sequence,C...)">countCompounds</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
              C...&nbsp;compounds)</code></th>
<td class="colLast">
<div class="block">For the given vargs of compounds this method counts the number of times those
 compounds appear in the given sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceProxyView.</span><code><span class="memberNameLink"><a href="../SequenceProxyView.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractCompound.</span><code><span class="memberNameLink"><a href="../AbstractCompound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase</a></span>&#8203;(<a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Compound.</span><code><span class="memberNameLink"><a href="../Compound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase</a></span>&#8203;(<a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.transcription">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> that implement <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></span></code></th>
<td class="colLast">
<div class="block">Attempts to wrap compounds so it is possible to view them in a case
 insensitive manner</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></span></code></th>
<td class="colLast">
<div class="block">Instance of a Codon which is 3 <a href="../../compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound"><code>NucleotideCompound</code></a>s, its corresponding
 <a href="../../compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound"><code>AminoAcidCompound</code></a> and if it is a start or stop codon.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> with parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CaseInsensitiveCompound.</span><code><span class="memberNameLink"><a href="../../transcription/CaseInsensitiveCompound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase</a></span>&#8203;(<a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Table.Codon.</span><code><span class="memberNameLink"><a href="../../transcription/Table.Codon.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase</a></span>&#8203;(<a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.views">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a> in <a href="../../views/package-summary.html">org.biojava3.core.sequence.views</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../views/package-summary.html">org.biojava3.core.sequence.views</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views">ReversedSequenceView</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">For a given sequence this class will return the base at the reversed position
 i.e. in a sequence of size 10, if you request base 2 you will get back the
 base at position 9.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">A sliding window view of a sequence which does not implement any interfaces
 like <a href="../Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a> because they do not fit how this works.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Compound.html" title="interface in org.biojava3.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
