<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>GenomeElementAtlas</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="GenomeElementAtlas";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":9,"i1":10,"i2":10,"i3":10,"i4":10,"i5":9,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GenomeElementAtlas.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.expressionAtlas.core</a></div>
<h2 title="Class GenomeElementAtlas" class="title">Class GenomeElementAtlas</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>bacnet.expressionAtlas.core.GenomeElementAtlas</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><code>java.io.Serializable</code></dd>
</dl>
<hr>
<pre>public class <span class="typeNameLabel">GenomeElementAtlas</span>
extends java.lang.Object
implements java.io.Serializable</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../serialized-form.html#bacnet.expressionAtlas.core.GenomeElementAtlas">Serialized Form</a></dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#DEFAULT_LOGFC_CUTOFF">DEFAULT_LOGFC_CUTOFF</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#name">name</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.TreeSet&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#notDiffExpresseds">notDiffExpresseds</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.TreeSet&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#overBioConds">overBioConds</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#PATH">PATH</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private static long</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#serialVersionUID">serialVersionUID</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.TreeSet&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#underBioConds">underBioConds</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">GenomeElementAtlas</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(bacnet.datamodel.sequence.Sequence,bacnet.utils.Filter)">GenomeElementAtlas</a></span>&#8203;(<a href="../../datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
                  <a href="../../utils/Filter.html" title="class in bacnet.utils">Filter</a>&nbsp;filter)</code></th>
<td class="colLast">
<div class="block">For a specific sequence update the list of BioCondition in which the genome element is
 over-expressed, under-expressed or not differently expressed<br>
 The update will be dependent of the value of the input filter = cutoff for LogFC and p-value</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createAll(bacnet.datamodel.expdesign.Experiment)">createAll</a></span>&#8203;(<a href="../../datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;expTemp)</code></th>
<td class="colLast">
<div class="block">Read all <code>Comparison</code> results and extract which gene is over and under expressed in
 it<br>
 For each gene, sRNA,asRNA, and cisReg, create a list of all <code>BioCondition</code> where it is
 over, under, and not differentially expressed<br>
 <br>
 
 WANRING: OrganizeDatabase.organizeComparisonsData(exp) as to be run first!!!</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getName()">getName</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>java.util.TreeSet&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getNotDiffExpresseds()">getNotDiffExpresseds</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>java.util.TreeSet&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getOverBioConds()">getOverBioConds</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>java.util.TreeSet&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getUnderBioConds()">getUnderBioConds</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>static <a href="GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#load(java.lang.String)">load</a></span>&#8203;(java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#save(java.lang.String)">save</a></span>&#8203;(java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Serialize the vector of lines with an ObjectOutputStream.</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setName(java.lang.String)">setName</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setNotDiffExpresseds(java.util.TreeSet)">setNotDiffExpresseds</a></span>&#8203;(java.util.TreeSet&lt;java.lang.String&gt;&nbsp;notDiffExpresseds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setOverBioConds(java.util.TreeSet)">setOverBioConds</a></span>&#8203;(java.util.TreeSet&lt;java.lang.String&gt;&nbsp;overBioConds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setUnderBioConds(java.util.TreeSet)">setUnderBioConds</a></span>&#8203;(java.util.TreeSet&lt;java.lang.String&gt;&nbsp;underBioConds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="serialVersionUID">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>serialVersionUID</h4>
<pre>private static final&nbsp;long serialVersionUID</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../constant-values.html#bacnet.expressionAtlas.core.GenomeElementAtlas.serialVersionUID">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a id="PATH">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PATH</h4>
<pre>public static&nbsp;java.lang.String PATH</pre>
</li>
</ul>
<a id="DEFAULT_LOGFC_CUTOFF">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DEFAULT_LOGFC_CUTOFF</h4>
<pre>public static&nbsp;double DEFAULT_LOGFC_CUTOFF</pre>
</li>
</ul>
<a id="name">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>name</h4>
<pre>private&nbsp;java.lang.String name</pre>
</li>
</ul>
<a id="overBioConds">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>overBioConds</h4>
<pre>private&nbsp;java.util.TreeSet&lt;java.lang.String&gt; overBioConds</pre>
</li>
</ul>
<a id="underBioConds">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>underBioConds</h4>
<pre>private&nbsp;java.util.TreeSet&lt;java.lang.String&gt; underBioConds</pre>
</li>
</ul>
<a id="notDiffExpresseds">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>notDiffExpresseds</h4>
<pre>private&nbsp;java.util.TreeSet&lt;java.lang.String&gt; notDiffExpresseds</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>GenomeElementAtlas</h4>
<pre>public&nbsp;GenomeElementAtlas()</pre>
</li>
</ul>
<a id="&lt;init&gt;(bacnet.datamodel.sequence.Sequence,bacnet.utils.Filter)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>GenomeElementAtlas</h4>
<pre>public&nbsp;GenomeElementAtlas&#8203;(<a href="../../datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
                          <a href="../../utils/Filter.html" title="class in bacnet.utils">Filter</a>&nbsp;filter)</pre>
<div class="block">For a specific sequence update the list of BioCondition in which the genome element is
 over-expressed, under-expressed or not differently expressed<br>
 The update will be dependent of the value of the input filter = cutoff for LogFC and p-value</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>seq</code> - </dd>
<dd><code>filter</code> - object encapsulated cutoff values: cutOff1 = logFC , cutOff2 = pvalue</dd>
<dd><code>stat</code> - if the statistical value should be taken, into account or not</dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="createAll(bacnet.datamodel.expdesign.Experiment)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createAll</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;createAll&#8203;(<a href="../../datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;expTemp)</pre>
<div class="block">Read all <code>Comparison</code> results and extract which gene is over and under expressed in
 it<br>
 For each gene, sRNA,asRNA, and cisReg, create a list of all <code>BioCondition</code> where it is
 over, under, and not differentially expressed<br>
 <br>
 
 WANRING: OrganizeDatabase.organizeComparisonsData(exp) as to be run first!!!</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>expTemp</code> - </dd>
</dl>
</li>
</ul>
<a id="load(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>load</h4>
<pre class="methodSignature">public static&nbsp;<a href="GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a>&nbsp;load&#8203;(java.lang.String&nbsp;fileName)</pre>
<div class="block">Read compressed, serialized data with a FileInputStream. Uncompress that data with a
 GZIPInputStream. Deserialize the vector of lines with a ObjectInputStream. Replace current data
 with new data, and redraw everything.</div>
</li>
</ul>
<a id="save(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>save</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;save&#8203;(java.lang.String&nbsp;fileName)</pre>
<div class="block">Serialize the vector of lines with an ObjectOutputStream. Compress the serialized objects with a
 GZIPOutputStream. Write the compressed, serialized data to a file with a FileOutputStream. Don't
 forget to flush and close the stream.</div>
</li>
</ul>
<a id="getName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getName</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getName()</pre>
</li>
</ul>
<a id="setName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setName</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setName&#8203;(java.lang.String&nbsp;name)</pre>
</li>
</ul>
<a id="getOverBioConds()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getOverBioConds</h4>
<pre class="methodSignature">public&nbsp;java.util.TreeSet&lt;java.lang.String&gt;&nbsp;getOverBioConds()</pre>
</li>
</ul>
<a id="setOverBioConds(java.util.TreeSet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setOverBioConds</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setOverBioConds&#8203;(java.util.TreeSet&lt;java.lang.String&gt;&nbsp;overBioConds)</pre>
</li>
</ul>
<a id="getUnderBioConds()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getUnderBioConds</h4>
<pre class="methodSignature">public&nbsp;java.util.TreeSet&lt;java.lang.String&gt;&nbsp;getUnderBioConds()</pre>
</li>
</ul>
<a id="setUnderBioConds(java.util.TreeSet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setUnderBioConds</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setUnderBioConds&#8203;(java.util.TreeSet&lt;java.lang.String&gt;&nbsp;underBioConds)</pre>
</li>
</ul>
<a id="getNotDiffExpresseds()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNotDiffExpresseds</h4>
<pre class="methodSignature">public&nbsp;java.util.TreeSet&lt;java.lang.String&gt;&nbsp;getNotDiffExpresseds()</pre>
</li>
</ul>
<a id="setNotDiffExpresseds(java.util.TreeSet)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setNotDiffExpresseds</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setNotDiffExpresseds&#8203;(java.util.TreeSet&lt;java.lang.String&gt;&nbsp;notDiffExpresseds)</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GenomeElementAtlas.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
