<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 18:10:05 CEST 2019 -->
<title>GeneFeatureHelper</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="GeneFeatureHelper";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":9,"i1":9,"i2":9,"i3":9,"i4":9,"i5":9,"i6":9,"i7":9,"i8":9,"i9":9,"i10":9,"i11":9,"i12":9,"i13":9};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GeneFeatureHelper.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">org.biojava3.genome</a></div>
<h2 title="Class GeneFeatureHelper" class="title">Class GeneFeatureHelper</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>org.biojava3.genome.GeneFeatureHelper</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<pre>public class <span class="typeNameLabel">GeneFeatureHelper</span>
extends java.lang.Object</pre>
<div class="block">Allow to process genome elements annotation</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Scooter Willis <willishf at gmail dot com></dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">GeneFeatureHelper</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addGeneIDGFF2GeneFeatures(java.util.LinkedHashMap,org.biojava3.genome.parsers.gff.FeatureList)">addGeneIDGFF2GeneFeatures</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequenceList,
                         <a href="parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;listGenes)</code></th>
<td class="colLast">
<div class="block">Load GFF2 feature file generated from the geneid prediction algorithm and map
 features onto the chromosome sequences</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addGeneMarkGTFGeneFeatures(java.util.LinkedHashMap,org.biojava3.genome.parsers.gff.FeatureList)">addGeneMarkGTFGeneFeatures</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequenceList,
                          <a href="parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;listGenes)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addGlimmerGFF3GeneFeatures(java.util.LinkedHashMap,org.biojava3.genome.parsers.gff.FeatureList)">addGlimmerGFF3GeneFeatures</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequenceList,
                          <a href="parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;listGenes)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addGmodGFF3GeneFeatures(java.util.LinkedHashMap,org.biojava3.genome.parsers.gff.FeatureList)">addGmodGFF3GeneFeatures</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequenceList,
                       <a href="parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;listGenes)</code></th>
<td class="colLast">
<div class="block">Load GFF3 file using mRNA as the gene feature as not all GFF3 files are
 complete</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getChromosomeSequenceFromDNASequence(java.util.LinkedHashMap)">getChromosomeSequenceFromDNASequence</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequenceList)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGeneSequences(java.util.Collection)">getGeneSequences</a></span>&#8203;(java.util.Collection&lt;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequences)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteinSequences(java.util.Collection)">getProteinSequences</a></span>&#8203;(java.util.Collection&lt;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequences)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadFastaAddGeneFeaturesFromGeneIDGFF2(java.io.File,java.io.File)">loadFastaAddGeneFeaturesFromGeneIDGFF2</a></span>&#8203;(java.io.File&nbsp;fastaSequenceFile,
                                      java.io.File&nbsp;gffFile)</code></th>
<td class="colLast">
<div class="block">Loads Fasta file and GFF2 feature file generated from the geneid prediction
 algorithm</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadFastaAddGeneFeaturesFromGeneMarkGTF(java.io.File,java.io.File)">loadFastaAddGeneFeaturesFromGeneMarkGTF</a></span>&#8203;(java.io.File&nbsp;fastaSequenceFile,
                                       java.io.File&nbsp;gffFile)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadFastaAddGeneFeaturesFromGlimmerGFF3(java.io.File,java.io.File)">loadFastaAddGeneFeaturesFromGlimmerGFF3</a></span>&#8203;(java.io.File&nbsp;fastaSequenceFile,
                                       java.io.File&nbsp;gffFile)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadFastaAddGeneFeaturesFromGmodGFF3(java.io.File,java.io.File,boolean)">loadFastaAddGeneFeaturesFromGmodGFF3</a></span>&#8203;(java.io.File&nbsp;fastaSequenceFile,
                                    java.io.File&nbsp;gffFile,
                                    boolean&nbsp;lazyloadsequences)</code></th>
<td class="colLast">
<div class="block">Lots of variations in the ontology or descriptors that can be used in GFF3
 which requires writing a custom parser to handle a GFF3 generated or used by
 a specific application.</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadFastaAddGeneFeaturesFromUpperCaseExonFastaFile(java.io.File,java.io.File,boolean)">loadFastaAddGeneFeaturesFromUpperCaseExonFastaFile</a></span>&#8203;(java.io.File&nbsp;fastaSequenceFile,
                                                  java.io.File&nbsp;uppercaseFastaFile,
                                                  boolean&nbsp;throwExceptionGeneNotFound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#main(java.lang.String%5B%5D)">main</a></span>&#8203;(java.lang.String[]&nbsp;args)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#outputFastaSequenceLengthGFF3(java.io.File,java.io.File)">outputFastaSequenceLengthGFF3</a></span>&#8203;(java.io.File&nbsp;fastaSequenceFile,
                             java.io.File&nbsp;gffFile)</code></th>
<td class="colLast">
<div class="block">Output a gff3 feature file that will give the length of each
 scaffold/chromosome in the fasta file.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>GeneFeatureHelper</h4>
<pre>public&nbsp;GeneFeatureHelper()</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="loadFastaAddGeneFeaturesFromUpperCaseExonFastaFile(java.io.File,java.io.File,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadFastaAddGeneFeaturesFromUpperCaseExonFastaFile</h4>
<pre class="methodSignature">public static&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;loadFastaAddGeneFeaturesFromUpperCaseExonFastaFile&#8203;(java.io.File&nbsp;fastaSequenceFile,
                                                                                                                                    java.io.File&nbsp;uppercaseFastaFile,
                                                                                                                                    boolean&nbsp;throwExceptionGeneNotFound)
                                                                                                                             throws java.lang.Exception</pre>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="outputFastaSequenceLengthGFF3(java.io.File,java.io.File)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>outputFastaSequenceLengthGFF3</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;outputFastaSequenceLengthGFF3&#8203;(java.io.File&nbsp;fastaSequenceFile,
                                                 java.io.File&nbsp;gffFile)
                                          throws java.lang.Exception</pre>
<div class="block">Output a gff3 feature file that will give the length of each
 scaffold/chromosome in the fasta file. Used for gbrowse so it knows length.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>fastaSequenceFile</code> - </dd>
<dd><code>gffFile</code> - </dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="loadFastaAddGeneFeaturesFromGeneIDGFF2(java.io.File,java.io.File)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadFastaAddGeneFeaturesFromGeneIDGFF2</h4>
<pre class="methodSignature">public static&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;loadFastaAddGeneFeaturesFromGeneIDGFF2&#8203;(java.io.File&nbsp;fastaSequenceFile,
                                                                                                                        java.io.File&nbsp;gffFile)
                                                                                                                 throws java.lang.Exception</pre>
<div class="block">Loads Fasta file and GFF2 feature file generated from the geneid prediction
 algorithm</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>fastaSequenceFile</code> - </dd>
<dd><code>gffFile</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="addGeneIDGFF2GeneFeatures(java.util.LinkedHashMap,org.biojava3.genome.parsers.gff.FeatureList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addGeneIDGFF2GeneFeatures</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;addGeneIDGFF2GeneFeatures&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequenceList,
                                             <a href="parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;listGenes)
                                      throws java.lang.Exception</pre>
<div class="block">Load GFF2 feature file generated from the geneid prediction algorithm and map
 features onto the chromosome sequences</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>chromosomeSequenceList</code> - </dd>
<dd><code>listGenes</code> - </dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="getChromosomeSequenceFromDNASequence(java.util.LinkedHashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromosomeSequenceFromDNASequence</h4>
<pre class="methodSignature">public static&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;getChromosomeSequenceFromDNASequence&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequenceList)</pre>
</li>
</ul>
<a id="loadFastaAddGeneFeaturesFromGmodGFF3(java.io.File,java.io.File,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadFastaAddGeneFeaturesFromGmodGFF3</h4>
<pre class="methodSignature">public static&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;loadFastaAddGeneFeaturesFromGmodGFF3&#8203;(java.io.File&nbsp;fastaSequenceFile,
                                                                                                                      java.io.File&nbsp;gffFile,
                                                                                                                      boolean&nbsp;lazyloadsequences)
                                                                                                               throws java.lang.Exception</pre>
<div class="block">Lots of variations in the ontology or descriptors that can be used in GFF3
 which requires writing a custom parser to handle a GFF3 generated or used by
 a specific application. Probably could be abstracted out but for now easier
 to handle with custom code to deal with gff3 elements that are not included
 but can be extracted from other data elements.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>fastaSequenceFile</code> - </dd>
<dd><code>gffFile</code> - </dd>
<dd><code>lazyloadsequences</code> - If set to true then the fasta file will be parsed
                          for accession id but sequences will be read from
                          disk when needed to save memory</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="addGmodGFF3GeneFeatures(java.util.LinkedHashMap,org.biojava3.genome.parsers.gff.FeatureList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addGmodGFF3GeneFeatures</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;addGmodGFF3GeneFeatures&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequenceList,
                                           <a href="parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;listGenes)
                                    throws java.lang.Exception</pre>
<div class="block">Load GFF3 file using mRNA as the gene feature as not all GFF3 files are
 complete</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>chromosomeSequenceList</code> - </dd>
<dd><code>listGenes</code> - </dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="loadFastaAddGeneFeaturesFromGlimmerGFF3(java.io.File,java.io.File)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadFastaAddGeneFeaturesFromGlimmerGFF3</h4>
<pre class="methodSignature">public static&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;loadFastaAddGeneFeaturesFromGlimmerGFF3&#8203;(java.io.File&nbsp;fastaSequenceFile,
                                                                                                                         java.io.File&nbsp;gffFile)
                                                                                                                  throws java.lang.Exception</pre>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="addGlimmerGFF3GeneFeatures(java.util.LinkedHashMap,org.biojava3.genome.parsers.gff.FeatureList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addGlimmerGFF3GeneFeatures</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;addGlimmerGFF3GeneFeatures&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequenceList,
                                              <a href="parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;listGenes)
                                       throws java.lang.Exception</pre>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="loadFastaAddGeneFeaturesFromGeneMarkGTF(java.io.File,java.io.File)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadFastaAddGeneFeaturesFromGeneMarkGTF</h4>
<pre class="methodSignature">public static&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;loadFastaAddGeneFeaturesFromGeneMarkGTF&#8203;(java.io.File&nbsp;fastaSequenceFile,
                                                                                                                         java.io.File&nbsp;gffFile)
                                                                                                                  throws java.lang.Exception</pre>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="addGeneMarkGTFGeneFeatures(java.util.LinkedHashMap,org.biojava3.genome.parsers.gff.FeatureList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addGeneMarkGTFGeneFeatures</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;addGeneMarkGTFGeneFeatures&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequenceList,
                                              <a href="parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;listGenes)
                                       throws java.lang.Exception</pre>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="getProteinSequences(java.util.Collection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinSequences</h4>
<pre class="methodSignature">public static&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a>&gt;&nbsp;getProteinSequences&#8203;(java.util.Collection&lt;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequences)
                                                                                           throws java.lang.Exception</pre>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="getGeneSequences(java.util.Collection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneSequences</h4>
<pre class="methodSignature">public static&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a>&gt;&nbsp;getGeneSequences&#8203;(java.util.Collection&lt;<a href="../core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequences)
                                                                                     throws java.lang.Exception</pre>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="main(java.lang.String[])">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>main</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;main&#8203;(java.lang.String[]&nbsp;args)
                 throws java.lang.Exception</pre>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GeneFeatureHelper.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
