<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 18:10:05 CEST 2019 -->
<title>Uses of Class bacnet.datamodel.sequence.Chromosome</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Class bacnet.datamodel.sequence.Chromosome";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Class bacnet.datamodel.sequence.Chromosome" class="title">Uses of Class<br>bacnet.datamodel.sequence.Chromosome</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.annotation">bacnet.datamodel.annotation</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.proteomics">bacnet.datamodel.proteomics</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.sequence">bacnet.datamodel.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.genomeBrowser.core">bacnet.genomeBrowser.core</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.genomeBrowser.tracksGUI">bacnet.genomeBrowser.tracksGUI</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.reader">bacnet.reader</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.annotation">
<!--   -->
</a>
<h3>Uses of <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a> in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a> with parameters of type <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Annotation.</span><code><span class="memberNameLink"><a href="../../annotation/Annotation.html#getElementATbp(bacnet.datamodel.sequence.Chromosome,int)">getElementATbp</a></span>&#8203;(<a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromosome,
              int&nbsp;bpPosition)</code></th>
<td class="colLast">
<div class="block">Return the first element for which<br>
 begin < bpPosition < end</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Annotation.</span><code><span class="memberNameLink"><a href="../../annotation/Annotation.html#getElementInfoATbp(bacnet.datamodel.sequence.Chromosome,int)">getElementInfoATbp</a></span>&#8203;(<a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromosome,
                  int&nbsp;bpPosition)</code></th>
<td class="colLast">
<div class="block">Return the first element for which<br>
 begin < bpPosition < end</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Annotation.</span><code><span class="memberNameLink"><a href="../../annotation/Annotation.html#getElements(bacnet.datamodel.sequence.Chromosome,int,int)">getElements</a></span>&#8203;(<a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromosome,
           int&nbsp;beginDraw,
           int&nbsp;endDraw)</code></th>
<td class="colLast">
<div class="block">Get the lists of elements between beginDraw and endDraw</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.proteomics">
<!--   -->
</a>
<h3>Uses of <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a> in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a> that return <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermCreateData.</span><code><span class="memberNameLink"><a href="../../proteomics/NTermCreateData.html#getChromosome()">getChromosome</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a> with parameters of type <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermCreateData.</span><code><span class="memberNameLink"><a href="../../proteomics/NTermCreateData.html#setChromosome(bacnet.datamodel.sequence.Chromosome)">setChromosome</a></span>&#8203;(<a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromosome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a> in <a href="../package-summary.html">bacnet.datamodel.sequence</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> that return <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Sequence.</span><code><span class="memberNameLink"><a href="../Sequence.html#getChromosome()">getChromosome</a></span>()</code></th>
<td class="colLast">
<div class="block">Return corresponding Chromosome given by this.getChromosomeNb<br>
 Load it if necessary</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Genome.</span><code><span class="memberNameLink"><a href="../Genome.html#getFirstChromosome()">getFirstChromosome</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the first chromosome in the HashMap
 
 Is Used to replace: Genome.loadGenome(genomeName).getChromosomes().get(0)</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> that return types with arguments of type <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#getChromosomeFromDNASequence(java.util.LinkedHashMap,bacnet.datamodel.sequence.Genome,int,java.lang.String)">getChromosomeFromDNASequence</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequenceList,
                            <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                            int&nbsp;k,
                            java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Genome.</span><code><span class="memberNameLink"><a href="../Genome.html#getChromosomes()">getChromosomes</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../package-summary.html">bacnet.datamodel.sequence</a> with type arguments of type <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Genome.</span><code><span class="memberNameLink"><a href="../Genome.html#setChromosomes(java.util.LinkedHashMap)">setChromosomes</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&gt;&nbsp;chromosomes)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.genomeBrowser.core">
<!--   -->
</a>
<h3>Uses of <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a> in <a href="../../../genomeBrowser/core/package-summary.html">bacnet.genomeBrowser.core</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../genomeBrowser/core/package-summary.html">bacnet.genomeBrowser.core</a> that return <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Track.</span><code><span class="memberNameLink"><a href="../../../genomeBrowser/core/Track.html#getChromosome()">getChromosome</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../genomeBrowser/core/package-summary.html">bacnet.genomeBrowser.core</a> with parameters of type <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Track.</span><code><span class="memberNameLink"><a href="../../../genomeBrowser/core/Track.html#setChromosome(bacnet.datamodel.sequence.Chromosome)">setChromosome</a></span>&#8203;(<a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromosome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.genomeBrowser.tracksGUI">
<!--   -->
</a>
<h3>Uses of <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a> in <a href="../../../genomeBrowser/tracksGUI/package-summary.html">bacnet.genomeBrowser.tracksGUI</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../genomeBrowser/tracksGUI/package-summary.html">bacnet.genomeBrowser.tracksGUI</a> with parameters of type <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GElement.</span><code><span class="memberNameLink"><a href="../../../genomeBrowser/tracksGUI/GElement.html#displaySequence(org.eclipse.swt.graphics.GC,bacnet.datamodel.sequence.Chromosome,int,int%5B%5D,double,bacnet.genomeBrowser.core.Zoom,int)">displaySequence</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
               <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromosome,
               int&nbsp;bpIndex,
               int[]&nbsp;position,
               double&nbsp;bpSizeH,
               <a href="../../../genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a>&nbsp;zoom,
               int&nbsp;type)</code></th>
<td class="colLast">
<div class="block">Display sequence information:
 nucleotide sequence on plus strand
 nucleotide sequence on minus strand
 amino acid sequence for each nucleotide position on the plus strand (=
 the three reading frame of the plus strand)</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.reader">
<!--   -->
</a>
<h3>Uses of <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a> in <a href="../../../reader/package-summary.html">bacnet.reader</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../reader/package-summary.html">bacnet.reader</a> with parameters of type <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PTTReader.</span><code><span class="memberNameLink"><a href="../../../reader/PTTReader.html#saveProteins(java.util.ArrayList,java.lang.String,bacnet.datamodel.sequence.Chromosome)">saveProteins</a></span>&#8203;(java.util.ArrayList&lt;<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;&nbsp;sequences,
            java.lang.String&nbsp;fileName,
            <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromo)</code></th>
<td class="colLast">
<div class="block">Save list of Gene in a PTT file<br>
 Information about the resulting protein are written into different
 columns:<br>
 "Location","Strand","Length","PID","Gene","Synonym","Code","COG","Product"</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
