<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 18:10:05 CEST 2019 -->
<title>Uses of Class bacnet.datamodel.sequence.Gene</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Class bacnet.datamodel.sequence.Gene";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Gene.html" title="class in bacnet.datamodel.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Class bacnet.datamodel.sequence.Gene" class="title">Uses of Class<br>bacnet.datamodel.sequence.Gene</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.sequence">bacnet.datamodel.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.sequenceNCBI">bacnet.datamodel.sequenceNCBI</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.reader">bacnet.reader</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.genome">bacnet.scripts.genome</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.listeriomics.nterm">bacnet.scripts.listeriomics.nterm</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.listeriomics.srna">bacnet.scripts.listeriomics.srna</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.sequenceTools">bacnet.sequenceTools</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a> in <a href="../package-summary.html">bacnet.datamodel.sequence</a></h3>
<table class="useSummary">
<caption><span>Subclasses of <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a> in <a href="../package-summary.html">bacnet.datamodel.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> that return <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#getGeneDownstream(bacnet.datamodel.sequence.Gene)">getGeneDownstream</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">Return the gene downstream</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Gene.</span><code><span class="memberNameLink"><a href="../Gene.html#getGeneDownstream()">getGeneDownstream</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the gene downstream on the + strand</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Genome.</span><code><span class="memberNameLink"><a href="../Genome.html#getGeneFromName(java.lang.String)">getGeneFromName</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Go through all chromosomes and each Genes map and GeneName map to find the
 corresponding gene</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Genome.</span><code><span class="memberNameLink"><a href="../Genome.html#getGeneFromName(java.lang.String,java.lang.String)">getGeneFromName</a></span>&#8203;(java.lang.String&nbsp;name,
               java.lang.String&nbsp;accessionChromo)</code></th>
<td class="colLast">
<div class="block">Go through all chromosomes and each Genes map and GeneName map to find the
 corresponding gene</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#getGeneUpstream(bacnet.datamodel.sequence.Gene)">getGeneUpstream</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">Return the gene upstream</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Gene.</span><code><span class="memberNameLink"><a href="../Gene.html#getGeneUpstream()">getGeneUpstream</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the gene upstream on the + strand</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Gene.</span><code><span class="memberNameLink"><a href="../Gene.html#load(java.lang.String)">load</a></span>&#8203;(java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Load a serialized Gene</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> that return types with arguments of type <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#getGenes()">getGenes</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Genome.</span><code><span class="memberNameLink"><a href="../Genome.html#getGenes()">getGenes</a></span>()</code></th>
<td class="colLast">
<div class="block">Return all Genes in all chromosomes</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#getGenesAlternative()">getGenesAlternative</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> with parameters of type <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#getGeneDownstream(bacnet.datamodel.sequence.Gene)">getGeneDownstream</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">Return the gene downstream</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#getGeneUpstream(bacnet.datamodel.sequence.Gene)">getGeneUpstream</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">Return the gene upstream</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../package-summary.html">bacnet.datamodel.sequence</a> with type arguments of type <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#setGenes(java.util.LinkedHashMap)">setGenes</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;&nbsp;genes)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#setGenesAlternative(java.util.LinkedHashMap)">setGenesAlternative</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;&nbsp;genesAlternative)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.sequenceNCBI">
<!--   -->
</a>
<h3>Uses of <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a> in <a href="../../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a> that return <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeConversionElement.</span><code><span class="memberNameLink"><a href="../../sequenceNCBI/GenomeConversionElement.html#convertDNASequenceGene(org.biojava3.core.sequence.DNASequence)">convertDNASequenceGene</a></span>&#8203;(<a href="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;dnaSeq)</code></th>
<td class="colLast">
<div class="block">From a DNASequence make a Gene object</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.reader">
<!--   -->
</a>
<h3>Uses of <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a> in <a href="../../../reader/package-summary.html">bacnet.reader</a></h3>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../reader/package-summary.html">bacnet.reader</a> with type arguments of type <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaFileReader.</span><code><span class="memberNameLink"><a href="../../../reader/FastaFileReader.html#saveCDS(java.util.ArrayList,java.lang.String)">saveCDS</a></span>&#8203;(java.util.ArrayList&lt;<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;&nbsp;sequences,
       java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Save all amino acid sequence in a fasta file</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaFileReader.</span><code><span class="memberNameLink"><a href="../../../reader/FastaFileReader.html#saveProteins(java.util.ArrayList,java.lang.String)">saveProteins</a></span>&#8203;(java.util.ArrayList&lt;<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;&nbsp;sequences,
            java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Save all amino acid sequence in a fasta file</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PTTReader.</span><code><span class="memberNameLink"><a href="../../../reader/PTTReader.html#saveProteins(java.util.ArrayList,java.lang.String,bacnet.datamodel.sequence.Chromosome)">saveProteins</a></span>&#8203;(java.util.ArrayList&lt;<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;&nbsp;sequences,
            java.lang.String&nbsp;fileName,
            <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromo)</code></th>
<td class="colLast">
<div class="block">Save list of Gene in a PTT file<br>
 Information about the resulting protein are written into different
 columns:<br>
 "Location","Strand","Length","PID","Gene","Synonym","Code","COG","Product"</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.genome">
<!--   -->
</a>
<h3>Uses of <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a> in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a> with parameters of type <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SDProfile.</span><code><span class="memberNameLink"><a href="../../../scripts/genome/SDProfile.html#verifyGene(bacnet.datamodel.sequence.Gene,java.lang.String%5B%5D%5B%5D,java.lang.String%5B%5D%5B%5D)">verifyGene</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene,
          java.lang.String[][]&nbsp;arrayPlus,
          java.lang.String[][]&nbsp;arrayMinus)</code></th>
<td class="colLast">
<div class="block">For a given Gene calculate its SD binding energy, and compare to the value
 given by SDProfile data</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.listeriomics.nterm">
<!--   -->
</a>
<h3>Uses of <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a> in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a> that return types with arguments of type <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getiTIS()">getiTIS</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a> with parameters of type <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getAtisBHI(bacnet.datamodel.sequence.Gene)">getAtisBHI</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">Go through all MassSpec data and find if the lmio was detected</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getAtisBlood(bacnet.datamodel.sequence.Gene)">getAtisBlood</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getCAI(bacnet.datamodel.sequence.Gene)">getCAI</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">After generating TIRSequence.fasta in setTIRSequence() method we submit to
 http://genomes.urv.cat/CAIcal/ to calculate CAI value and expected CAI value
 The table with CAI values is saved from the website in: CAIResults.excel</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getCAIGene(bacnet.datamodel.sequence.Gene)">getCAIGene</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">After generating GeneSequence.fasta in setGeneSequence() method we submit to
 http://genomes.urv.cat/CAIcal/ to calculate CAI value and expected CAI value
 The table with CAI values is saved from the website in:
 StatGenes/CAIGeneResults.excel</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getChargedResidues(bacnet.datamodel.sequence.Gene,int)">getChargedResidues</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene,
                  int&nbsp;decay)</code></th>
<td class="colLast">
<div class="block">Sum up the different charge of the amino acids of the TIR</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getChargedResiduesGene(bacnet.datamodel.sequence.Gene)">getChargedResiduesGene</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">Sum up the different charge of the amino acids of the gene sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getComment(bacnet.datamodel.sequence.Gene)">getComment</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">Return information about gene or iTIS</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getDetectable(bacnet.datamodel.sequence.Gene)">getDetectable</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">By reading the List in: /StatGenes/EGDeDetectable.txt Say if a gene is
 detectable by the spectometer or not</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getGeneExpr(bacnet.datamodel.sequence.Gene,java.lang.String)">getGeneExpr</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene,
           java.lang.String&nbsp;dataName)</code></th>
<td class="colLast">
<div class="block">Return the median value expression (calculated on 10 BHI 37C data) in the
 Gene Expression array of the gene</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getNCRNAUpstream(bacnet.datamodel.sequence.Gene)">getNCRNAUpstream</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">For each gene search if the end of a smallRNA or cisReg element is in the
 region -200bp - start codon</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getSDEnergy(bacnet.datamodel.sequence.Gene)">getSDEnergy</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">Return the binding energy with aSD sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getStartCodon(bacnet.datamodel.sequence.Gene)">getStartCodon</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">Return start codon proportion in the genome</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getTIRStructureEnergy(bacnet.datamodel.sequence.Gene)">getTIRStructureEnergy</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">For very gene read TIRStructure.excel and extract the mean value of energy:
 We calculate the 2nd structure free energy and save it a table<br>
 And we calculate the energy for 40 different sequence: -20+40 to -20+80 and
 calculate the average energy</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getTotalSpectrum(bacnet.datamodel.sequence.Gene)">getTotalSpectrum</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">For each gene add the number of spectra of every peptide available (not only
 aTIS peptides)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getUTRLength(bacnet.datamodel.sequence.Gene)">getUTRLength</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block"><span class="deprecatedLabel">Deprecated.</span></div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.listeriomics.srna">
<!--   -->
</a>
<h3>Uses of <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a> in <a href="../../../scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a></h3>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a> with type arguments of type <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SmallPeptidesSearch.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/srna/SmallPeptidesSearch.html#multiSpeciesAlignBlastN(java.util.ArrayList)">multiSpeciesAlignBlastN</a></span>&#8203;(java.util.ArrayList&lt;<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;&nbsp;genes)</code></th>
<td class="colLast">
<div class="block"><span class="deprecatedLabel">Deprecated.</span></div>
&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.sequenceTools">
<!--   -->
</a>
<h3>Uses of <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a> in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a> that return <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneView.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/GeneView.html#getSequence()">getSequence</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a> with parameters of type <a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneView.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/GeneView.html#displayGene(bacnet.datamodel.sequence.Gene,org.eclipse.e4.ui.workbench.modeling.EPartService)">displayGene</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene,
           org.eclipse.e4.ui.workbench.modeling.EPartService&nbsp;partService)</code></th>
<td class="colLast">
<div class="block">Display a given gene</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneViewHomologTools.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/GeneViewHomologTools.html#getPhyloFigure(bacnet.datamodel.sequence.Gene,java.util.ArrayList)">getPhyloFigure</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;sequence,
              java.util.ArrayList&lt;java.lang.String&gt;&nbsp;genomes)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneViewHomologTools.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/GeneViewHomologTools.html#loadFigureHomologs(bacnet.datamodel.sequence.Gene,org.eclipse.swt.browser.Browser,java.util.ArrayList)">loadFigureHomologs</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;sequence,
                  org.eclipse.swt.browser.Browser&nbsp;browserHomolog,
                  java.util.ArrayList&lt;java.lang.String&gt;&nbsp;selectedGenomes)</code></th>
<td class="colLast">
<div class="block">Load SVG figure of phylogeny and replace all strain name by homolog
 informations<br>
 Save to JPG file and display it</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneView.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/GeneView.html#setSequence(bacnet.datamodel.sequence.Gene)">setSequence</a></span>&#8203;(<a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Gene.html" title="class in bacnet.datamodel.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
