<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Fri Dec 10 09:15:34 CET 2021 -->
<title>Uses of Class bacnet.datamodel.sequence.Genome</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-10">
<meta name="description" content="use: package: bacnet.datamodel.sequence, class: Genome">
<meta name="generator" content="javadoc/ClassUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-use-page">
<script type="text/javascript">var pathtoroot = "../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Class</a></li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Class bacnet.datamodel.sequence.Genome" class="title">Uses of Class<br>bacnet.datamodel.sequence.Genome</h1>
</div>
<div class="caption"><span>Packages that use <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#bacnet">bacnet</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.datamodel.annotation">bacnet.datamodel.annotation</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.datamodel.dataset">bacnet.datamodel.dataset</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.datamodel.expdesign">bacnet.datamodel.expdesign</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.datamodel.proteomics">bacnet.datamodel.proteomics</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.datamodel.sequence">bacnet.datamodel.sequence</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.datamodel.sequenceNCBI">bacnet.datamodel.sequenceNCBI</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.expressionAtlas.core">bacnet.expressionAtlas.core</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.genomeBrowser.core">bacnet.genomeBrowser.core</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.reader">bacnet.reader</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.scripts.core">bacnet.scripts.core</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.scripts.database">bacnet.scripts.database</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.scripts.genome">bacnet.scripts.genome</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.scripts.listeriomics.nterm">bacnet.scripts.listeriomics.nterm</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.scripts.listeriomics.srna">bacnet.scripts.listeriomics.srna</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.sequenceTools">bacnet.sequenceTools</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.table">bacnet.table</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.table.gui">bacnet.table.gui</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.views">bacnet.views</a></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<section class="class-uses">
<ul class="block-list">
<li>
<section class="detail" id="bacnet">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../package-summary.html">bacnet</a></h2>
<div class="caption"><span>Methods in <a href="../../../package-summary.html">bacnet</a> that return types with arguments of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/TreeMap.html" title="class or interface in java.util" class="external-link">TreeMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Database.</span><code><a href="../../../Database.html#getGenomes()" class="member-name-link">getGenomes</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Method parameters in <a href="../../../package-summary.html">bacnet</a> with type arguments of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Database.</span><code><a href="../../../Database.html#setGenomes(java.util.TreeMap)" class="member-name-link">setGenomes</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/TreeMap.html" title="class or interface in java.util" class="external-link">TreeMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&gt;&nbsp;genomes)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.datamodel.annotation">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a></h2>
<div class="caption"><span>Methods in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Annotation.</span><code><a href="../../annotation/Annotation.html#addAnnotation(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome)" class="member-name-link">addAnnotation</a><wbr>(<a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;exprMatrix,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Add annotation information to an ExpressionMatrix</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Annotation.</span><code><a href="../../annotation/Annotation.html#addAnnotationLite(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome)" class="member-name-link">addAnnotationLite</a><wbr>(<a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;exprMatrix,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">
<div class="block">Add annotation information to an ExpressionMatrix</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Annotation.</span><code><a href="../../annotation/Annotation.html#addAnnotationMultiChromosome(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome)" class="member-name-link">addAnnotationMultiChromosome</a><wbr>(<a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;exprMatrix,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Add annotation information to an ExpressionMatrix</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[][]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Annotation.</span><code><a href="../../annotation/Annotation.html#getAnnotationGenes(bacnet.datamodel.sequence.Genome,java.util.ArrayList)" class="member-name-link">getAnnotationGenes</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;&nbsp;elements)</code></div>
<div class="col-last odd-row-color">
<div class="block">Extract a String[][] with all gene annotation information</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Annotation.</span><code><a href="../../annotation/Annotation.html#getAnnotationMatrix(bacnet.datamodel.sequence.Genome)" class="member-name-link">getAnnotationMatrix</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Return a matrix containing different information on each element of the genome first chromosome</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Annotation.</span><code><a href="../../annotation/Annotation.html#getAnnotationMatrix(bacnet.datamodel.sequence.Genome,java.util.ArrayList)" class="member-name-link">getAnnotationMatrix</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;&nbsp;elements)</code></div>
<div class="col-last odd-row-color">
<div class="block">Return a matrix containing different information on each element of the genome first chromosome</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Annotation.</span><code><a href="../../annotation/Annotation.html#quickAddAnnotation(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome)" class="member-name-link">quickAddAnnotation</a><wbr>(<a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;exprMatrix,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Add annotation information to an ExpressionMatrix</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.datamodel.dataset">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../dataset/package-summary.html">bacnet.datamodel.dataset</a></h2>
<div class="caption"><span>Methods in <a href="../../dataset/package-summary.html">bacnet.datamodel.dataset</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Network.</span><code><a href="../../dataset/Network.html#getCoExpressionGlobalMatrix(bacnet.datamodel.sequence.Genome)" class="member-name-link">getCoExpressionGlobalMatrix</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Calculate all pearson correlation and put it in a list of network interactions<br>
 Remove correlation CORR_CUTOFF<br>
 Nedd to load: Database.COEXPR_NETWORK_TRANSCRIPTOMES_PATH+"_Temp_"+genome.getSpecies()</div>
</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">NGS.</span><code><a href="../../dataset/NGS.html#readWIGChromo(java.lang.String,java.lang.String,bacnet.datamodel.sequence.Genome)" class="member-name-link">readWIGChromo</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;chromoName,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">
<div class="block">Will read your wig file and parse it to BACNET <br>
 ONLY WIG FILE WITH variableStep and raw format are accepted No fixedStep format</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.datamodel.expdesign">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../expdesign/package-summary.html">bacnet.datamodel.expdesign</a></h2>
<div class="caption"><span>Methods in <a href="../../expdesign/package-summary.html">bacnet.datamodel.expdesign</a> that return <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">BioCondition.</span><code><a href="../../expdesign/BioCondition.html#getGenome()" class="member-name-link">getGenome</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Return corresponding Genome given by this.genomeName<br>
 Load it if necessary</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.datamodel.proteomics">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a></h2>
<div class="caption"><span>Methods in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">NTermUtils.</span><code><a href="../../proteomics/NTermUtils.html#getFullORF(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Genome)" class="member-name-link">getFullORF</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in frame) containing the
 peptide (amino acid sequence) <br></div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">NTermUtils.</span><code><a href="../../proteomics/NTermUtils.html#getFullORF(java.lang.String,int,int,boolean,bacnet.datamodel.sequence.Genome)" class="member-name-link">getFullORF</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;peptideSeq,
 int&nbsp;beginPeptide,
 int&nbsp;endPeptide,
 boolean&nbsp;strand,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in frame) containing the
 peptide (amino acid sequence) <br></div>
</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">NTermUtils.</span><code><a href="../../proteomics/NTermUtils.html#getListOfFullORFfromPeptide(java.lang.String,bacnet.datamodel.sequence.Genome)" class="member-name-link">getListOfFullORFfromPeptide</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">From a list of peptide position retrieve the complete sequence of the ORF (in frame) containing
 the peptide (amino acid sequence) <br>
 WARNING: Be careful that your peptide info (begin,end) are correct.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">NTermUtils.</span><code><a href="../../proteomics/NTermUtils.html#searchStopCodon(bacnet.datamodel.sequence.Genome,int,boolean)" class="member-name-link">searchStopCodon</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 int&nbsp;bpPosition,
 boolean&nbsp;strand)</code></div>
<div class="col-last odd-row-color">
<div class="block">Get the next stop codon in frame and dowstream to a bpPosition</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../proteomics/NTermCreateData.html#%3Cinit%3E(bacnet.datamodel.sequence.Genome)" class="member-name-link">NTermCreateData</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Initialize a void NTermCreateData object and run <code>initializeGenomeSequence()</code> to have
 the genome on the six frames</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.datamodel.sequence">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../package-summary.html">bacnet.datamodel.sequence</a></h2>
<div class="caption"><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> that return <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Sequence.</span><code><a href="../Sequence.html#getGenome()" class="member-name-link">getGenome</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Return corresponding Genome given by this.genomeName<br>
 Load it if necessary</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Chromosome.</span><code><a href="../Chromosome.html#getParentGenome()" class="member-name-link">getParentGenome</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Genome.</span><code><a href="../Genome.html#loadEgdeGenome()" class="member-name-link">loadEgdeGenome</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Find in ModelProvider if Egde Genome is already loaded If not search EGDE_NAME in GENOMEDIR</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Genome.</span><code><a href="../Genome.html#loadGenome(java.lang.String)" class="member-name-link">loadGenome</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeAccession)</code></div>
<div class="col-last odd-row-color">
<div class="block">Find in ModelProvider if Genome Accession is already loaded If not search load it</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Genome.</span><code><a href="../Genome.html#loadGenome(java.lang.String,boolean)" class="member-name-link">loadGenome</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeAccession,
 boolean&nbsp;keepInMemory)</code></div>
<div class="col-last even-row-color">
<div class="block">Find in ModelProvider if Genome Accession is already loaded If not search load it
 If keepInMemory is true : Find in ModelProvider if Genome Accession is already loaded if not load
 it <br>
 If keepInMemory is false : Only load the Genome without saving it <br>
 <br></div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Genome.</span><code><a href="../Genome.html#loadGenome(java.lang.String,java.lang.String,boolean,boolean)" class="member-name-link">loadGenome</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeAccession,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;path,
 boolean&nbsp;keepInMemory,
 boolean&nbsp;annotation)</code></div>
<div class="col-last odd-row-color">
<div class="block">Load/return a genome given by a genomeAccession ID<br>
 <br>
 
 If keepInMemory is true : Find in ModelProvider if Genome Accession is already loaded if not load
 it <br>
 If keepInMemory is false : Only load the Genome without saving it <br>
 <br>
 If annotation is true : Load annotation information of the genome <br>
 If anotation is false : Only load the sequence of the genome</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/LinkedHashMap.html" title="class or interface in java.util" class="external-link">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Chromosome.</span><code><a href="../Chromosome.html#getChromosomeFromDNASequence(java.util.LinkedHashMap,bacnet.datamodel.sequence.Genome,int,java.lang.String)" class="member-name-link">getChromosomeFromDNASequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/LinkedHashMap.html" title="class or interface in java.util" class="external-link">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequenceList,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 int&nbsp;k,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Codon.</span><code><a href="../Codon.html#getCodonProportion(bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">getCodonProportion</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName)</code></div>
<div class="col-last odd-row-color">
<div class="block">Go through genome on both strand and calculate the proportion of each codon</div>
</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Chromosome.</span><code><a href="../Chromosome.html#setParentGenome(bacnet.datamodel.sequence.Genome)" class="member-name-link">setParentGenome</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;parentGenome)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.datamodel.sequenceNCBI">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></h2>
<div class="caption"><span>Methods in <a href="../../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a> that return <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">GenomeConversion.</span><code><a href="../../sequenceNCBI/GenomeConversion.html#run(bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String,java.lang.String)" class="member-name-link">run</a><wbr>(<a href="../../sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;pathGenome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName)</code></div>
<div class="col-last even-row-color">
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GenomeConversion.</span><code><a href="../../sequenceNCBI/GenomeConversion.html#run(bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">run</a><wbr>(<a href="../../sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;pathGenome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomePath)</code></div>
<div class="col-last odd-row-color">
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">GenomeConversion.</span><code><a href="../../sequenceNCBI/GenomeConversion.html#run(java.lang.String)" class="member-name-link">run</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName)</code></div>
<div class="col-last even-row-color">
<div class="block">Method for Genome in the GenomeNCBIFolder<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">GenomeConversion.</span><code><a href="../../sequenceNCBI/GenomeConversion.html#createAnnotation(bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">createAnnotation</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;path)</code></div>
<div class="col-last even-row-color">
<div class="block">Create a Table with all elements of the Genome, which will be used for element search and display
 on the GenomeViewer</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GenomeConversion.</span><code><a href="../../sequenceNCBI/GenomeConversion.html#updateAllElements(bacnet.datamodel.sequence.Genome)" class="member-name-link">updateAllElements</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">
<div class="block">Add all elements found in the Chromosome to LinkedHashMap&lt;String, Sequence> allElements</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.expressionAtlas.core">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a></h2>
<div class="caption"><span>Constructors in <a href="../../../expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../../expressionAtlas/core/SelectGenomeElementDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.TreeSet,java.util.TreeSet,bacnet.datamodel.sequence.Genome)" class="member-name-link">SelectGenomeElementDialog</a><wbr>(org.eclipse.swt.widgets.Shell&nbsp;shell,
 org.eclipse.e4.ui.workbench.modeling.EPartService&nbsp;partService,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/TreeSet.html" title="class or interface in java.util" class="external-link">TreeSet</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;&nbsp;includeElements,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/TreeSet.html" title="class or interface in java.util" class="external-link">TreeSet</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;&nbsp;excludedElements,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Create the wizard.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.genomeBrowser.core">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../genomeBrowser/core/package-summary.html">bacnet.genomeBrowser.core</a></h2>
<div class="caption"><span>Constructors in <a href="../../../genomeBrowser/core/package-summary.html">bacnet.genomeBrowser.core</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../../genomeBrowser/core/Track.html#%3Cinit%3E(bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">Track</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;chromoID)</code></div>
<div class="col-last even-row-color">
<div class="block">Set every parameter using genome and chromosome number</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.reader">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../reader/package-summary.html">bacnet.reader</a></h2>
<div class="caption"><span>Methods in <a href="../../../reader/package-summary.html">bacnet.reader</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">GBKReader.</span><code><a href="../../../reader/GBKReader.html#createGenBankFile(bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">createGenBankFile</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName)</code></div>
<div class="col-last even-row-color">
<div class="block">Create GenBank file with first chromosome<br>
 TO BE FIXED for adding other chromosomes</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GBKReader.</span><code><a href="../../../reader/GBKReader.html#formatSequence(bacnet.datamodel.sequence.Genome)" class="member-name-link">formatSequence</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.scripts.core">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../scripts/core/package-summary.html">bacnet.scripts.core</a></h2>
<div class="caption"><span>Methods in <a href="../../../scripts/core/package-summary.html">bacnet.scripts.core</a> that return <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Expression.</span><code><a href="../../../scripts/core/Expression.html#getGenome()" class="member-name-link">getGenome</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../../scripts/core/package-summary.html">bacnet.scripts.core</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Expression.</span><code><a href="../../../scripts/core/Expression.html#createExpressionMatrix(bacnet.datamodel.sequence.Genome)" class="member-name-link">createExpressionMatrix</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Combine all Transcriptomes describe in <code>Expression</code> data, and return the combine
 <code>ExpressionMatrix</code></div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Expression.</span><code><a href="../../../scripts/core/Expression.html#setGenome(bacnet.datamodel.sequence.Genome)" class="member-name-link">setGenome</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Expression.</span><code><a href="../../../scripts/core/Expression.html#summarize(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)" class="member-name-link">summarize</a><wbr>(<a href="../../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Create 3 ExpressionMatrix: <br>
 GeneExpression of genes on all BioCondition, saved in PATH_ALLGENEXPR
 Tiling of genes on all BioCondition, saved in PATH_ALLTILING_GENE
 Tiling of Srnas on all BioCondition, saved in PATH_ALLTILING_SRNA</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../../../scripts/core/package-summary.html">bacnet.scripts.core</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../../scripts/core/Expression.html#%3Cinit%3E(bacnet.datamodel.sequence.Genome,java.util.ArrayList,java.util.ArrayList,bacnet.datamodel.dataset.OmicsData.TypeData)" class="member-name-link">Expression</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;&nbsp;genomeElements,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="../../expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&gt;&nbsp;bioCondList,
 <a href="../../dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a>&nbsp;typeData)</code></div>
<div class="col-last even-row-color">
<div class="block">Create <code>Expression</code> object to regroups multiple Expression data</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.scripts.database">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../scripts/database/package-summary.html">bacnet.scripts.database</a></h2>
<div class="caption"><span>Methods in <a href="../../../scripts/database/package-summary.html">bacnet.scripts.database</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ProteomicsCreation.</span><code><a href="../../../scripts/database/ProteomicsCreation.html#addMissingValuesToMatrices(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)" class="member-name-link">addMissingValuesToMatrices</a><wbr>(<a href="../../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">In order to represent proteomics data we need to add a -1 value in every missing row.<br>
 ATTENTION: -1 is now a value when there is no value but it is detected</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GenomesCreation.</span><code><a href="../../../scripts/database/GenomesCreation.html#createCircularGenomeView(bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">createCircularGenomeView</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;title)</code></div>
<div class="col-last odd-row-color">
<div class="block"><span class="deprecated-label">Deprecated.</span></div>
</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ComparisonsCreation.</span><code><a href="../../../scripts/database/ComparisonsCreation.html#createLogFCMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)" class="member-name-link">createLogFCMatrix</a><wbr>(<a href="../../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Create Database.logFCMatrix by regrouping every logFC values for genes, sRNAs, asRNAs, and
 cisRegs in every <code>BioCondition</code> given <code>Experiment</code><br></div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">TranscriptomesCreation.</span><code><a href="../../../scripts/database/TranscriptomesCreation.html#createLogFCMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)" class="member-name-link">createLogFCMatrix</a><wbr>(<a href="../../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create Database.logFCMatrix by regrouping every logFC values for genes, sRNAs, asRNAs, and
 cisRegs in every <code>BioCondition</code> given <code>Experiment</code><br></div>
</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ComparisonsCreation.</span><code><a href="../../../scripts/database/ComparisonsCreation.html#createStatMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)" class="member-name-link">createStatMatrix</a><wbr>(<a href="../../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Create Database.statTable by calculating stat values for genes, sRNAs, and asRNAs in every
 BioCondition<br>
 For genes we use FDRBY calculated from GebneExpression array<br>
 For sRNAs and asRNAs we use TSTUDENTTILING for Tiling array</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">TranscriptomesCreation.</span><code><a href="../../../scripts/database/TranscriptomesCreation.html#createStatMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)" class="member-name-link">createStatMatrix</a><wbr>(<a href="../../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create Database.statTable by calculating stat values for genes, sRNAs, and asRNAs in every
 BioCondition<br>
 For genes we use FDRBY calculated from GebneExpression array<br>
 For sRNAs and asRNAs we use TSTUDENTTILING for Tiling array</div>
</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ComparisonsCreation.</span><code><a href="../../../scripts/database/ComparisonsCreation.html#summarizeDiffExpr(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)" class="member-name-link">summarizeDiffExpr</a><wbr>(<a href="../../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Create an array :<br>
 Column = comparison
 Row = Gene, Srna or Asrna
 Cell = "+" if a specific gene has been detected over-expressed, and "-" if
 under-expressed<br>
 <br>
 
 Save it in : Comparison.PATH_DIFF_LISTS+".txt"</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.scripts.genome">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a></h2>
<div class="caption"><span>Fields in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a> declared as <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Field</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">CircularGenomeJPanel.</span><code><a href="../../../scripts/genome/CircularGenomeJPanel.html#genome" class="member-name-link">genome</a></code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SDProfile.</span><code><a href="../../../scripts/genome/SDProfile.html#calculateOneSDProfile(bacnet.datamodel.sequence.Genome,int,int,org.biojava3.core.sequence.Strand,java.lang.String)" class="member-name-link">calculateOneSDProfile</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 int&nbsp;begin,
 int&nbsp;end,
 <a href="../../../../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a>&nbsp;strand,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SDProfile.</span><code><a href="../../../scripts/genome/SDProfile.html#calculateSDProfile(bacnet.datamodel.sequence.Genome)" class="member-name-link">calculateSDProfile</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StartCodonProfile.</span><code><a href="../../../scripts/genome/StartCodonProfile.html#calculateStartCodonProfile(bacnet.datamodel.sequence.Genome,java.lang.String,int)" class="member-name-link">calculateStartCodonProfile</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;dataName,
 int&nbsp;typeData)</code></div>
<div class="col-last even-row-color">
<div class="block">Create an NGS data where:<br>
 Start codon represent a coverage of 10
 Stop codon represent a coverage of 5
 every other codon a coverage of 0</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CircularGenomeJPanel.</span><code><a href="../../../scripts/genome/CircularGenomeJPanel.html#RNABindingCircularGenome(int,int,bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">RNABindingCircularGenome</a><wbr>(int&nbsp;width,
 int&nbsp;height,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;title)</code></div>
<div class="col-last odd-row-color">
<div class="block">Function to create a circular genome view of rnas binding to protein lom2686</div>
</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">CircularGenomeJPanel.</span><code><a href="../../../scripts/genome/CircularGenomeJPanel.html#sRNACircularGenome(int,int,bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">sRNACircularGenome</a><wbr>(int&nbsp;width,
 int&nbsp;height,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;title)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../../scripts/genome/CircularGenomeJPanel.html#%3Cinit%3E(int,int,bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">CircularGenomeJPanel</a><wbr>(int&nbsp;width,
 int&nbsp;height,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;title)</code></div>
<div class="col-last even-row-color">
<div class="block">Run circularGenome for Listeriomics Circos network graph</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.scripts.listeriomics.nterm">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></h2>
<div class="caption"><span>Methods in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">NTermStat.</span><code><a href="../../../scripts/listeriomics/nterm/NTermStat.html#statInternalORF(bacnet.datamodel.sequence.Genome)" class="member-name-link">statInternalORF</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Given a genome extract main information such as:<br>
 number of start codons, stop codons
 number of high SD binding sites
 number of possible ORFs
 number of predicted genes<br>
 <br>
 From this metrics, predict the number of possible internal translation initiation site</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.scripts.listeriomics.srna">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a></h2>
<div class="caption"><span>Methods in <a href="../../../scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../../../scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">SrnaAlignmentBlastN.</span><code><a href="../../../scripts/listeriomics/srna/SrnaAlignmentBlastN.html#extractPosition(bacnet.datamodel.sequence.Srna,bacnet.datamodel.sequence.Genome,java.util.TreeMap)" class="member-name-link">extractPosition</a><wbr>(<a href="../Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&nbsp;sRNA,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/TreeMap.html" title="class or interface in java.util" class="external-link">TreeMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;&nbsp;accessionToName)</code></div>
<div class="col-last even-row-color">
<div class="block">From Blast Result extract position in the Genome of each sRNA<br>
 
 Srna with an identity lower than 25% are not inserted</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.sequenceTools">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a></h2>
<div class="caption"><span>Methods in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a> that return <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AnnotationView.</span><code><a href="../../../sequenceTools/AnnotationView.html#getGenome()" class="member-name-link">getGenome</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GeneView.</span><code><a href="../../../sequenceTools/GeneView.html#getGenome()" class="member-name-link">getGenome</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">SequenceDisplayDialog.</span><code><a href="../../../sequenceTools/SequenceDisplayDialog.html#getGenome()" class="member-name-link">getGenome</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SrnaView.</span><code><a href="../../../sequenceTools/SrnaView.html#getGenome()" class="member-name-link">getGenome</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">GeneViewLocalizationTools.</span><code><a href="../../../sequenceTools/GeneViewLocalizationTools.html#loadLocalizationFigure(org.eclipse.swt.browser.Browser,java.lang.String%5B%5D%5B%5D,bacnet.datamodel.sequence.Sequence,java.lang.String%5B%5D%5B%5D,org.eclipse.swt.widgets.Table,bacnet.datamodel.sequence.Genome)" class="member-name-link">loadLocalizationFigure</a><wbr>(org.eclipse.swt.browser.Browser&nbsp;browserLocalization,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[][]&nbsp;arrayGeneToLocalization,
 <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[][]&nbsp;bioCondsArray,
 org.eclipse.swt.widgets.Table&nbsp;tableLocalization,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Load SVG figure of phylogeny and replace all strain name by homolog informations<br>
 Save to JPG file and display it</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AnnotationView.</span><code><a href="../../../sequenceTools/AnnotationView.html#openAnnotationView(org.eclipse.e4.ui.workbench.modeling.EPartService,bacnet.datamodel.sequence.Genome)" class="member-name-link">openAnnotationView</a><wbr>(org.eclipse.e4.ui.workbench.modeling.EPartService&nbsp;partService,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">
<div class="block">Display a given gene</div>
</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AnnotationView.</span><code><a href="../../../sequenceTools/AnnotationView.html#setGenome(bacnet.datamodel.sequence.Genome)" class="member-name-link">setGenome</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GeneView.</span><code><a href="../../../sequenceTools/GeneView.html#setGenome(bacnet.datamodel.sequence.Genome)" class="member-name-link">setGenome</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SequenceDisplayDialog.</span><code><a href="../../../sequenceTools/SequenceDisplayDialog.html#setGenome(bacnet.datamodel.sequence.Genome)" class="member-name-link">setGenome</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SrnaView.</span><code><a href="../../../sequenceTools/SrnaView.html#setGenome(bacnet.datamodel.sequence.Genome)" class="member-name-link">setGenome</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../../sequenceTools/LocalizationDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,java.lang.String%5B%5D%5B%5D,bacnet.datamodel.sequence.Sequence,java.lang.String%5B%5D%5B%5D,bacnet.datamodel.sequence.Genome)" class="member-name-link">LocalizationDialog</a><wbr>(org.eclipse.swt.widgets.Shell&nbsp;parentShell,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[][]&nbsp;arrayGeneToLocalization,
 <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[][]&nbsp;bioCondsArray,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Create the dialog.</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../../sequenceTools/SequenceDisplayDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,org.eclipse.e4.ui.workbench.modeling.EPartService,bacnet.datamodel.sequence.Genome,java.lang.String,boolean,int,int,org.biojava3.core.sequence.Strand)" class="member-name-link">SequenceDisplayDialog</a><wbr>(org.eclipse.swt.widgets.Shell&nbsp;parentShell,
 org.eclipse.e4.ui.workbench.modeling.EPartService&nbsp;partService,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;chromoId,
 boolean&nbsp;nucleotide,
 int&nbsp;begin,
 int&nbsp;end,
 <a href="../../../../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a>&nbsp;strand)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create the dialog.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.table">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../table/package-summary.html">bacnet.table</a></h2>
<div class="caption"><span>Methods in <a href="../../../table/package-summary.html">bacnet.table</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">TableSWTView.</span><code><a href="../../../table/TableSWTView.html#displayMatrix(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome,java.lang.String,boolean,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayMatrix</a><wbr>(<a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;matrix,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;viewName,
 boolean&nbsp;saveSVG,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName,
 org.eclipse.e4.ui.workbench.modeling.EPartService&nbsp;partService)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">TableSWTView.</span><code><a href="../../../table/TableSWTView.html#displayMatrix(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayMatrix</a><wbr>(<a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;matrix,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;viewName,
 org.eclipse.e4.ui.workbench.modeling.EPartService&nbsp;partService)</code></div>
<div class="col-last odd-row-color">
<div class="block">Static function which open an HeatMapView and display corresponding ExpressionMatrix</div>
</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">TableSWTComposite.</span><code><a href="../../../table/TableSWTComposite.html#initData(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome)" class="member-name-link">initData</a><wbr>(<a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;matrix,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Init data and filter lists</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">TableSWTView.</span><code><a href="../../../table/TableSWTView.html#setData(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome)" class="member-name-link">setData</a><wbr>(<a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;matrix,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.table.gui">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../table/gui/package-summary.html">bacnet.table.gui</a></h2>
<div class="caption"><span>Constructors in <a href="../../../table/gui/package-summary.html">bacnet.table.gui</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../../table/gui/HideExpressionMatrixDialog.html#%3Cinit%3E(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome,java.util.ArrayList,java.util.ArrayList,org.eclipse.swt.widgets.Shell,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">HideExpressionMatrixDialog</a><wbr>(<a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;matrix,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;&nbsp;hideDialogExcludeRow,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;&nbsp;hideDialogExcludeColumn,
 org.eclipse.swt.widgets.Shell&nbsp;shell,
 org.eclipse.e4.ui.workbench.modeling.EPartService&nbsp;partService)</code></div>
<div class="col-last even-row-color">
<div class="block">Create the dialog.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.views">
<h2>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../views/package-summary.html">bacnet.views</a></h2>
<div class="caption"><span>Methods in <a href="../../../views/package-summary.html">bacnet.views</a> that return <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">CoExprNetworkView.</span><code><a href="../../../views/CoExprNetworkView.html#getGenome()" class="member-name-link">getGenome</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../../views/package-summary.html">bacnet.views</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">CoExprNetworkView.</span><code><a href="../../../views/CoExprNetworkView.html#setGenome(bacnet.datamodel.sequence.Genome)" class="member-name-link">setGenome</a><wbr>(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
</ul>
</section>
</main>
</div>
</div>
</body>
</html>
