<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 14:20:51 CEST 2019 -->
<title>Uses of Class bacnet.datamodel.sequence.GenomeNCBI</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Class bacnet.datamodel.sequence.GenomeNCBI";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Class bacnet.datamodel.sequence.GenomeNCBI" class="title">Uses of Class<br>bacnet.datamodel.sequence.GenomeNCBI</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet">bacnet</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.annotation">bacnet.datamodel.annotation</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.sequence">bacnet.datamodel.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.sequenceNCBI">bacnet.datamodel.sequenceNCBI</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.genome">bacnet.scripts.genome</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.listeriomics.nterm">bacnet.scripts.listeriomics.nterm</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="bacnet">
<!--   -->
</a>
<h3>Uses of <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a> in <a href="../../../package-summary.html">bacnet</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../package-summary.html">bacnet</a> that return types with arguments of type <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.TreeMap&lt;java.lang.String,&#8203;<a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Database.</span><code><span class="memberNameLink"><a href="../../../Database.html#getGenomesNCBI()">getGenomesNCBI</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../package-summary.html">bacnet</a> with type arguments of type <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Database.</span><code><span class="memberNameLink"><a href="../../../Database.html#setGenomesNCBI(java.util.TreeMap)">setGenomesNCBI</a></span>&#8203;(java.util.TreeMap&lt;java.lang.String,&#8203;<a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&gt;&nbsp;genomesNCBI)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.annotation">
<!--   -->
</a>
<h3>Uses of <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a> in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a> with parameters of type <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;java.lang.String&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">COGannotation.</span><code><span class="memberNameLink"><a href="../../annotation/COGannotation.html#getCogClassification(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.GenomeNCBI)">getCogClassification</a></span>&#8203;(<a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;matrix,
                    <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&nbsp;genome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a> in <a href="../package-summary.html">bacnet.datamodel.sequence</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> that return <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChromosomeBacteriaSequence.</span><code><span class="memberNameLink"><a href="../ChromosomeBacteriaSequence.html#getParentGenome()">getParentGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the gene based on accession.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> with parameters of type <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Operon.</span><code><span class="memberNameLink"><a href="../Operon.html#convertDNASequenceOperon(java.lang.String%5B%5D,bacnet.datamodel.sequence.GenomeNCBI)">convertDNASequenceOperon</a></span>&#8203;(java.lang.String[]&nbsp;row,
                        <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Take row from Annot_sup table and creat an Operon Object from it</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChromosomeBacteriaSequence.</span><code><span class="memberNameLink"><a href="../ChromosomeBacteriaSequence.html#setParentGenome(bacnet.datamodel.sequence.GenomeNCBI)">setParentGenome</a></span>&#8203;(<a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&nbsp;parentGenome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.sequenceNCBI">
<!--   -->
</a>
<h3>Uses of <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a> in <a href="../../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a> that return <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeNCBITools.</span><code><span class="memberNameLink"><a href="../../sequenceNCBI/GenomeNCBITools.html#loadEgdeGenome()">loadEgdeGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Find in ModelProvider if Egde Genome is already loaded If not search
 EGDE_NAME in GENOMEDIR</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeNCBITools.</span><code><span class="memberNameLink"><a href="../../sequenceNCBI/GenomeNCBITools.html#loadGenome(java.lang.String)">loadGenome</a></span>&#8203;(java.lang.String&nbsp;genomeAccession)</code></th>
<td class="colLast">
<div class="block">Find in ModelProvider if Genome Accession is already loaded If not search
 load it</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeNCBITools.</span><code><span class="memberNameLink"><a href="../../sequenceNCBI/GenomeNCBITools.html#loadGenome(java.lang.String,java.lang.String,boolean,boolean)">loadGenome</a></span>&#8203;(java.lang.String&nbsp;genomeAccession,
          java.lang.String&nbsp;path,
          boolean&nbsp;keepInMemory,
          boolean&nbsp;annotation)</code></th>
<td class="colLast">
<div class="block">Load/return a genome given by a genomeAccession ID<br>
 <br>
 
 If keepInMemory is true : Find in ModelProvider if Genome Accession is
 already loaded if not load it <br>
 If keepInMemory is false : Only load the Genome without saving it <br>
 <br>
 If annotation is true : Load annotation information of the genome <br>
 If anotation is false : Only load the sequence of the genome</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a> with parameters of type <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeConversionElement.</span><code><span class="memberNameLink"><a href="../../sequenceNCBI/GenomeConversionElement.html#convertDNASequenceOperon(java.lang.String%5B%5D,bacnet.datamodel.sequence.GenomeNCBI)">convertDNASequenceOperon</a></span>&#8203;(java.lang.String[]&nbsp;row,
                        <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&nbsp;genome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeNCBITools.</span><code><span class="memberNameLink"><a href="../../sequenceNCBI/GenomeNCBITools.html#curateGenomeFindingDoublonGeneName(bacnet.datamodel.sequence.GenomeNCBI)">curateGenomeFindingDoublonGeneName</a></span>&#8203;(<a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Check if two different locustag are not associated to the same GeneName</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeConversion.</span><code><span class="memberNameLink"><a href="../../sequenceNCBI/GenomeConversion.html#run(bacnet.datamodel.sequence.GenomeNCBI,java.lang.String,java.lang.String)">run</a></span>&#8203;(<a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&nbsp;genomeNCBI,
   java.lang.String&nbsp;pathGenome,
   java.lang.String&nbsp;genomeName)</code></th>
<td class="colLast">
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each
 ChromosomeBacteriaSequence into Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and
 serialized <it> sRNA are leaved unchanged <it> Operons, terminator and other
 elements are serialized <it> Annotation is created, saved in a table (only
 for verification) and serialized</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeConversion.</span><code><span class="memberNameLink"><a href="../../sequenceNCBI/GenomeConversion.html#run(bacnet.datamodel.sequence.GenomeNCBI,java.lang.String,java.lang.String,java.lang.String)">run</a></span>&#8203;(<a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&nbsp;genomeNCBI,
   java.lang.String&nbsp;pathGenome,
   java.lang.String&nbsp;genomeName,
   java.lang.String&nbsp;genomePath)</code></th>
<td class="colLast">
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each
 ChromosomeBacteriaSequence into Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and
 serialized <it> sRNA are leaved unchanged <it> Operons, terminator and other
 elements are serialized <it> Annotation is created, saved in a table (only
 for verification) and serialized</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.genome">
<!--   -->
</a>
<h3>Uses of <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a> in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a></h3>
<table class="useSummary">
<caption><span>Constructors in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a> with parameters of type <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../scripts/genome/CircularGenomeJPanel.html#%3Cinit%3E(int,int,bacnet.datamodel.sequence.GenomeNCBI,java.lang.String)">CircularGenomeJPanel</a></span>&#8203;(int&nbsp;width,
                    int&nbsp;height,
                    <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&nbsp;genome,
                    java.lang.String&nbsp;title)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.listeriomics.nterm">
<!--   -->
</a>
<h3>Uses of <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a> in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a> that return <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RASToverlap.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/RASToverlap.html#getEGDe()">getEGDe</a></span>()</code></th>
<td class="colLast">
<div class="block">Read RAST automoticaly annotated EGD-e genome</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RASToverlap.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/RASToverlap.html#getGenome(java.lang.String)">getGenome</a></span>&#8203;(java.lang.String&nbsp;genomeName)</code></th>
<td class="colLast">
<div class="block">Read a genome from RAST results</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
