<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 14:20:51 CEST 2019 -->
<title>Uses of Class bacnet.datamodel.sequence.Genome</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Class bacnet.datamodel.sequence.Genome";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Class bacnet.datamodel.sequence.Genome" class="title">Uses of Class<br>bacnet.datamodel.sequence.Genome</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet">bacnet</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.annotation">bacnet.datamodel.annotation</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.dataset">bacnet.datamodel.dataset</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.expdesign">bacnet.datamodel.expdesign</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.proteomics">bacnet.datamodel.proteomics</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.sequence">bacnet.datamodel.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.sequenceNCBI">bacnet.datamodel.sequenceNCBI</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.genomeBrowser.core">bacnet.genomeBrowser.core</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.reader">bacnet.reader</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.core">bacnet.scripts.core</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.database">bacnet.scripts.database</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.genome">bacnet.scripts.genome</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.listeriomics.nterm">bacnet.scripts.listeriomics.nterm</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.listeriomics.srna">bacnet.scripts.listeriomics.srna</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.sequenceTools">bacnet.sequenceTools</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.views">bacnet.views</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="bacnet">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../package-summary.html">bacnet</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../package-summary.html">bacnet</a> that return types with arguments of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.TreeMap&lt;java.lang.String,&#8203;<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Database.</span><code><span class="memberNameLink"><a href="../../../Database.html#getGenomes()">getGenomes</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../package-summary.html">bacnet</a> with type arguments of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Database.</span><code><span class="memberNameLink"><a href="../../../Database.html#setGenomes(java.util.TreeMap)">setGenomes</a></span>&#8203;(java.util.TreeMap&lt;java.lang.String,&#8203;<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&gt;&nbsp;genomes)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.annotation">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Annotation.</span><code><span class="memberNameLink"><a href="../../annotation/Annotation.html#addAnnotation(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome)">addAnnotation</a></span>&#8203;(<a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;exprMatrix,
             <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Add annotation information to an ExpressionMatrix</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Annotation.</span><code><span class="memberNameLink"><a href="../../annotation/Annotation.html#addAnnotationLite(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome)">addAnnotationLite</a></span>&#8203;(<a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;exprMatrix,
                 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Add annotation information to an ExpressionMatrix</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Annotation.</span><code><span class="memberNameLink"><a href="../../annotation/Annotation.html#addAnnotationMultiChromosome(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome)">addAnnotationMultiChromosome</a></span>&#8203;(<a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;exprMatrix,
                            <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Add annotation information to an ExpressionMatrix</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String[][]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Annotation.</span><code><span class="memberNameLink"><a href="../../annotation/Annotation.html#getAnnotationGenes(bacnet.datamodel.sequence.Genome,java.util.ArrayList)">getAnnotationGenes</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                  java.util.ArrayList&lt;java.lang.String&gt;&nbsp;elements)</code></th>
<td class="colLast">
<div class="block">Extract a String[][] with all gene annotation information</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Annotation.</span><code><span class="memberNameLink"><a href="../../annotation/Annotation.html#getAnnotationMatrix(bacnet.datamodel.sequence.Genome)">getAnnotationMatrix</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Return a matrix containing different information on each element of the
 genome first chromosome</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Annotation.</span><code><span class="memberNameLink"><a href="../../annotation/Annotation.html#getAnnotationMatrix(bacnet.datamodel.sequence.Genome,java.util.ArrayList)">getAnnotationMatrix</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                   java.util.ArrayList&lt;java.lang.String&gt;&nbsp;elements)</code></th>
<td class="colLast">
<div class="block">Return a matrix containing different information on each element of the
 genome first chromosome</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Annotation.</span><code><span class="memberNameLink"><a href="../../annotation/Annotation.html#quickAddAnnotation(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome)">quickAddAnnotation</a></span>&#8203;(<a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;exprMatrix,
                  <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Add annotation information to an ExpressionMatrix</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.dataset">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../dataset/package-summary.html">bacnet.datamodel.dataset</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../dataset/package-summary.html">bacnet.datamodel.dataset</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Network.</span><code><span class="memberNameLink"><a href="../../dataset/Network.html#getCoExpressionGlobalMatrix(bacnet.datamodel.sequence.Genome)">getCoExpressionGlobalMatrix</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Calculate all pearson correlation and put it in a list of network
 interactions<br>
 Remove correlation <0.85<br>
 Nedd to load:
 Database.COEXPR_NETWORK_TRANSCRIPTOMES_PATH+"_Temp_"+genome.getSpecies()</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NGS.</span><code><span class="memberNameLink"><a href="../../dataset/NGS.html#readWIGChromo(java.lang.String,java.lang.String,bacnet.datamodel.sequence.Genome)">readWIGChromo</a></span>&#8203;(java.lang.String&nbsp;fileName,
             java.lang.String&nbsp;chromoName,
             <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Will read your wig file and parse it to BACNET <br>
 ONLY WIG FILE WITH variableStep and raw format are accepted No fixedStep
 format</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.expdesign">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../expdesign/package-summary.html">bacnet.datamodel.expdesign</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../expdesign/package-summary.html">bacnet.datamodel.expdesign</a> that return <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BioCondition.</span><code><span class="memberNameLink"><a href="../../expdesign/BioCondition.html#getGenome()">getGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Return corresponding Genome given by this.genomeName<br>
 Load it if necessary</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.proteomics">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermUtils.</span><code><span class="memberNameLink"><a href="../../proteomics/NTermUtils.html#getFullORF(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Genome)">getFullORF</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
          <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in
 frame) containing the peptide (amino acid sequence) <br></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermUtils.</span><code><span class="memberNameLink"><a href="../../proteomics/NTermUtils.html#getFullORF(java.lang.String,int,int,boolean,bacnet.datamodel.sequence.Genome)">getFullORF</a></span>&#8203;(java.lang.String&nbsp;peptideSeq,
          int&nbsp;beginPeptide,
          int&nbsp;endPeptide,
          boolean&nbsp;strand,
          <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in
 frame) containing the peptide (amino acid sequence) <br></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermUtils.</span><code><span class="memberNameLink"><a href="../../proteomics/NTermUtils.html#getListOfFullORFfromPeptide(java.lang.String,bacnet.datamodel.sequence.Genome)">getListOfFullORFfromPeptide</a></span>&#8203;(java.lang.String&nbsp;fileName,
                           <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">From a list of peptide position retrieve the complete sequence of the ORF (in
 frame) containing the peptide (amino acid sequence) <br>
 WARNING: Be careful that your peptide info (begin,end) are correct.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermUtils.</span><code><span class="memberNameLink"><a href="../../proteomics/NTermUtils.html#searchStopCodon(bacnet.datamodel.sequence.Genome,int,boolean)">searchStopCodon</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
               int&nbsp;bpPosition,
               boolean&nbsp;strand)</code></th>
<td class="colLast">
<div class="block">Get the next stop codon in frame and dowstream to a bpPosition</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../proteomics/NTermCreateData.html#%3Cinit%3E(bacnet.datamodel.sequence.Genome)">NTermCreateData</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Initialize a void NTermCreateData object and run
 <code>initializeGenomeSequence()</code> to have the genome on the six frames</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../package-summary.html">bacnet.datamodel.sequence</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> that return <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Sequence.</span><code><span class="memberNameLink"><a href="../Sequence.html#getGenome()">getGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Return corresponding Genome given by this.genomeName<br>
 Load it if necessary</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#getParentGenome()">getParentGenome</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Genome.</span><code><span class="memberNameLink"><a href="../Genome.html#loadEgdeGenome()">loadEgdeGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Find in ModelProvider if Egde Genome is already loaded If not search
 EGDE_NAME in GENOMEDIR</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Genome.</span><code><span class="memberNameLink"><a href="../Genome.html#loadGenome(java.lang.String)">loadGenome</a></span>&#8203;(java.lang.String&nbsp;genomeAccession)</code></th>
<td class="colLast">
<div class="block">Find in ModelProvider if Genome Accession is already loaded If not search
 load it</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Genome.</span><code><span class="memberNameLink"><a href="../Genome.html#loadGenome(java.lang.String,java.lang.String,boolean,boolean)">loadGenome</a></span>&#8203;(java.lang.String&nbsp;genomeAccession,
          java.lang.String&nbsp;path,
          boolean&nbsp;keepInMemory,
          boolean&nbsp;annotation)</code></th>
<td class="colLast">
<div class="block">Load/return a genome given by a genomeAccession ID<br>
 <br>
 
 If keepInMemory is true : Find in ModelProvider if Genome Accession is
 already loaded if not load it <br>
 If keepInMemory is false : Only load the Genome without saving it <br>
 <br>
 If annotation is true : Load annotation information of the genome <br>
 If anotation is false : Only load the sequence of the genome</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#getChromosomeFromDNASequence(java.util.LinkedHashMap,bacnet.datamodel.sequence.Genome,int,java.lang.String)">getChromosomeFromDNASequence</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequenceList,
                            <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                            int&nbsp;k,
                            java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Codon.</span><code><span class="memberNameLink"><a href="../Codon.html#getCodonProportion(bacnet.datamodel.sequence.Genome,java.lang.String)">getCodonProportion</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                  java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Go through genome on both strand and calculate the proportion of each codon</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#setParentGenome(bacnet.datamodel.sequence.Genome)">setParentGenome</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;parentGenome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.sequenceNCBI">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a> that return <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeConversion.</span><code><span class="memberNameLink"><a href="../../sequenceNCBI/GenomeConversion.html#run(bacnet.datamodel.sequence.GenomeNCBI,java.lang.String,java.lang.String)">run</a></span>&#8203;(<a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&nbsp;genomeNCBI,
   java.lang.String&nbsp;pathGenome,
   java.lang.String&nbsp;genomeName)</code></th>
<td class="colLast">
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each
 ChromosomeBacteriaSequence into Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and
 serialized <it> sRNA are leaved unchanged <it> Operons, terminator and other
 elements are serialized <it> Annotation is created, saved in a table (only
 for verification) and serialized</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeConversion.</span><code><span class="memberNameLink"><a href="../../sequenceNCBI/GenomeConversion.html#run(bacnet.datamodel.sequence.GenomeNCBI,java.lang.String,java.lang.String,java.lang.String)">run</a></span>&#8203;(<a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&nbsp;genomeNCBI,
   java.lang.String&nbsp;pathGenome,
   java.lang.String&nbsp;genomeName,
   java.lang.String&nbsp;genomePath)</code></th>
<td class="colLast">
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each
 ChromosomeBacteriaSequence into Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and
 serialized <it> sRNA are leaved unchanged <it> Operons, terminator and other
 elements are serialized <it> Annotation is created, saved in a table (only
 for verification) and serialized</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeConversion.</span><code><span class="memberNameLink"><a href="../../sequenceNCBI/GenomeConversion.html#run(java.lang.String)">run</a></span>&#8203;(java.lang.String&nbsp;genomeName)</code></th>
<td class="colLast">
<div class="block">Method for Genome in the GenomeNCBIFolder<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each
 ChromosomeBacteriaSequence into Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and
 serialized <it> sRNA are leaved unchanged <it> Operons, terminator and other
 elements are serialized <it> Annotation is created, saved in a table (only
 for verification) and serialized</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeConversion.</span><code><span class="memberNameLink"><a href="../../sequenceNCBI/GenomeConversion.html#createAnnotation(bacnet.datamodel.sequence.Genome,java.lang.String)">createAnnotation</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                java.lang.String&nbsp;path)</code></th>
<td class="colLast">
<div class="block">Create a Table with all elements of the Genome, which will be used for
 element search and display on the GenomeViewer</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeConversion.</span><code><span class="memberNameLink"><a href="../../sequenceNCBI/GenomeConversion.html#updateAllElements(bacnet.datamodel.sequence.Genome)">updateAllElements</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Add all elements found in the Chromosome to LinkedHashMap<String, Sequence>
 allElements</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.genomeBrowser.core">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../genomeBrowser/core/package-summary.html">bacnet.genomeBrowser.core</a></h3>
<table class="useSummary">
<caption><span>Constructors in <a href="../../../genomeBrowser/core/package-summary.html">bacnet.genomeBrowser.core</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../genomeBrowser/core/Track.html#%3Cinit%3E(bacnet.datamodel.sequence.Genome,java.lang.String)">Track</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
     java.lang.String&nbsp;chromoID)</code></th>
<td class="colLast">
<div class="block">Set every parameter using genome and chromosome number</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.reader">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../reader/package-summary.html">bacnet.reader</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../reader/package-summary.html">bacnet.reader</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GBKReader.</span><code><span class="memberNameLink"><a href="../../../reader/GBKReader.html#createGenBankFile(bacnet.datamodel.sequence.Genome,java.lang.String)">createGenBankFile</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                 java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GBKReader.</span><code><span class="memberNameLink"><a href="../../../reader/GBKReader.html#formatSequence(bacnet.datamodel.sequence.Genome)">formatSequence</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.core">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../scripts/core/package-summary.html">bacnet.scripts.core</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/core/package-summary.html">bacnet.scripts.core</a> that return <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Expression.</span><code><span class="memberNameLink"><a href="../../../scripts/core/Expression.html#getGenome()">getGenome</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/core/package-summary.html">bacnet.scripts.core</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Expression.</span><code><span class="memberNameLink"><a href="../../../scripts/core/Expression.html#createExpressionMatrix(bacnet.datamodel.sequence.Genome)">createExpressionMatrix</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Combine all Transcriptomes describe in <code>Expression</code> data, and
 return the combine <code>ExpressionMatrix</code></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Expression.</span><code><span class="memberNameLink"><a href="../../../scripts/core/Expression.html#setGenome(bacnet.datamodel.sequence.Genome)">setGenome</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Expression.</span><code><span class="memberNameLink"><a href="../../../scripts/core/Expression.html#summarize(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">summarize</a></span>&#8203;(<a href="../../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
         <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create 3 ExpressionMatrix: <br>
 GeneExpression of genes on all BioCondition, saved in PATH_ALLGENEXPR
 Tiling of genes on all BioCondition, saved in PATH_ALLTILING_GENE
 Tiling of Srnas on all BioCondition, saved in PATH_ALLTILING_SRNA</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../../scripts/core/package-summary.html">bacnet.scripts.core</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../scripts/core/Expression.html#%3Cinit%3E(bacnet.datamodel.sequence.Genome,java.util.ArrayList,java.util.ArrayList,bacnet.datamodel.dataset.OmicsData.TypeData)">Expression</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
          java.util.ArrayList&lt;java.lang.String&gt;&nbsp;genomeElements,
          java.util.ArrayList&lt;<a href="../../expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&gt;&nbsp;bioCondList,
          <a href="../../dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a>&nbsp;typeData)</code></th>
<td class="colLast">
<div class="block">Create <code>Expression</code> object to regroups multiple Expression data</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.database">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../scripts/database/package-summary.html">bacnet.scripts.database</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/database/package-summary.html">bacnet.scripts.database</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteomicsCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/ProteomicsCreation.html#addMissingValuesToMatrices(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">addMissingValuesToMatrices</a></span>&#8203;(<a href="../../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                          <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">In order to represent proteomics data we need to add a 0 value in every
 missing row.<br></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteomicsCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/ProteomicsCreation.html#convertProteomicsData(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">convertProteomicsData</a></span>&#8203;(<a href="../../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                     <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Convert all ExpressionMatrix found for Proteomic datasets and save them in
 Streaming folder</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ComparisonsCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/ComparisonsCreation.html#createLogFCMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">createLogFCMatrix</a></span>&#8203;(<a href="../../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create Database.logFCMatrix by regrouping every logFC values for genes,
 sRNAs, asRNAs, and cisRegs in every <code>BioCondition</code> given
 <code>Experiment</code><br></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptomesCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/TranscriptomesCreation.html#createLogFCMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">createLogFCMatrix</a></span>&#8203;(<a href="../../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create Database.logFCMatrix by regrouping every logFC values for genes,
 sRNAs, asRNAs, and cisRegs in every <code>BioCondition</code> given
 <code>Experiment</code><br></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ComparisonsCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/ComparisonsCreation.html#createStatMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">createStatMatrix</a></span>&#8203;(<a href="../../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create Database.statTable by calculating stat values for genes, sRNAs, and
 asRNAs in every BioCondition<br>
 For genes we use FDRBY calculated from GebneExpression array<br>
 For sRNAs and asRNAs we use TSTUDENTTILING for Tiling array</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptomesCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/TranscriptomesCreation.html#createStatMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">createStatMatrix</a></span>&#8203;(<a href="../../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create Database.statTable by calculating stat values for genes, sRNAs, and
 asRNAs in every BioCondition<br>
 For genes we use FDRBY calculated from GebneExpression array<br>
 For sRNAs and asRNAs we use TSTUDENTTILING for Tiling array</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ComparisonsCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/ComparisonsCreation.html#summarizeDiffExpr(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">summarizeDiffExpr</a></span>&#8203;(<a href="../../expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create an array :<br>
 Column = comparison
 Row = Gene, Srna or Asrna
 Cell = "+" if a specific gene has been detected over-expressed, and "-"
 if under-expressed<br>
 <br>
 
 Save it in : Comparison.PATH_DIFF_LISTS+".txt"</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.genome">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a></h3>
<table class="useSummary">
<caption><span>Fields in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a> declared as <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CircularGenomeJPanel.</span><code><span class="memberNameLink"><a href="../../../scripts/genome/CircularGenomeJPanel.html#genome">genome</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SDProfile.</span><code><span class="memberNameLink"><a href="../../../scripts/genome/SDProfile.html#calculateOneSDProfile(bacnet.datamodel.sequence.Genome,int,int,org.biojava3.core.sequence.Strand,java.lang.String)">calculateOneSDProfile</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                     int&nbsp;begin,
                     int&nbsp;end,
                     <a href="../../../../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a>&nbsp;strand,
                     java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SDProfile.</span><code><span class="memberNameLink"><a href="../../../scripts/genome/SDProfile.html#calculateSDProfile(bacnet.datamodel.sequence.Genome)">calculateSDProfile</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StartCodonProfile.</span><code><span class="memberNameLink"><a href="../../../scripts/genome/StartCodonProfile.html#calculateStartCodonProfile(bacnet.datamodel.sequence.Genome,java.lang.String,int)">calculateStartCodonProfile</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                          java.lang.String&nbsp;dataName,
                          int&nbsp;typeData)</code></th>
<td class="colLast">
<div class="block">Create an NGS data where:<br>
 Start codon represent a coverage of 10
 Stop codon represent a coverage of 5
 every other codon a coverage of 0</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CircularGenomeJPanel.</span><code><span class="memberNameLink"><a href="../../../scripts/genome/CircularGenomeJPanel.html#RNABindingCircularGenome(int,int,bacnet.datamodel.sequence.Genome,java.lang.String)">RNABindingCircularGenome</a></span>&#8203;(int&nbsp;width,
                        int&nbsp;height,
                        <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                        java.lang.String&nbsp;title)</code></th>
<td class="colLast">
<div class="block">Function to create a circular genome view of rnas binding to protein lom2686</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CircularGenomeJPanel.</span><code><span class="memberNameLink"><a href="../../../scripts/genome/CircularGenomeJPanel.html#sRNACircularGenome(int,int,bacnet.datamodel.sequence.Genome,java.lang.String)">sRNACircularGenome</a></span>&#8203;(int&nbsp;width,
                  int&nbsp;height,
                  <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                  java.lang.String&nbsp;title)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../scripts/genome/CircularGenomeJPanel.html#%3Cinit%3E(int,int,bacnet.datamodel.sequence.Genome,java.lang.String)">CircularGenomeJPanel</a></span>&#8203;(int&nbsp;width,
                    int&nbsp;height,
                    <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                    java.lang.String&nbsp;title)</code></th>
<td class="colLast">
<div class="block">Run circularGenome for Listeriomics Circos network graph</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.listeriomics.nterm">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#statInternalORF(bacnet.datamodel.sequence.Genome)">statInternalORF</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Given a genome extract main information such as:<br>
 number of start codons, stop codons
 number of high SD binding sites
 number of possible ORFs
 number of predicted genes<br>
 <br>
 From this metrics, predict the number of possible internal translation
 initiation site</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.listeriomics.srna">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SrnaAlignmentBlastN.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/srna/SrnaAlignmentBlastN.html#extractPosition(bacnet.datamodel.sequence.Srna,bacnet.datamodel.sequence.Genome,java.util.TreeMap)">extractPosition</a></span>&#8203;(<a href="../Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&nbsp;sRNA,
               <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
               java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.String&gt;&nbsp;accessionToName)</code></th>
<td class="colLast">
<div class="block">From Blast Result extract position in the Genome of each sRNA<br>
 
 Srna with an identity lower than 25% are not inserted</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.sequenceTools">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a> that return <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AnnotationView.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/AnnotationView.html#getGenome()">getGenome</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneView.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/GeneView.html#getGenome()">getGenome</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceDisplayDialog.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/SequenceDisplayDialog.html#getGenome()">getGenome</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SrnaView.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/SrnaView.html#getGenome()">getGenome</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneViewLocalizationTools.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/GeneViewLocalizationTools.html#loadLocalizationFigure(org.eclipse.swt.browser.Browser,java.lang.String%5B%5D%5B%5D,bacnet.datamodel.sequence.Sequence,java.lang.String%5B%5D%5B%5D,org.eclipse.swt.widgets.Table,bacnet.datamodel.sequence.Genome)">loadLocalizationFigure</a></span>&#8203;(org.eclipse.swt.browser.Browser&nbsp;browserLocalization,
                      java.lang.String[][]&nbsp;arrayGeneToLocalization,
                      <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
                      java.lang.String[][]&nbsp;bioCondsArray,
                      org.eclipse.swt.widgets.Table&nbsp;tableLocalization,
                      <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Load SVG figure of phylogeny and replace all strain name by homolog
 informations<br>
 Save to JPG file and display it</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AnnotationView.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/AnnotationView.html#openAnnotationView(org.eclipse.e4.ui.workbench.modeling.EPartService,bacnet.datamodel.sequence.Genome)">openAnnotationView</a></span>&#8203;(org.eclipse.e4.ui.workbench.modeling.EPartService&nbsp;partService,
                  <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Display a given gene</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AnnotationView.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/AnnotationView.html#setGenome(bacnet.datamodel.sequence.Genome)">setGenome</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneView.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/GeneView.html#setGenome(bacnet.datamodel.sequence.Genome)">setGenome</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceDisplayDialog.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/SequenceDisplayDialog.html#setGenome(bacnet.datamodel.sequence.Genome)">setGenome</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SrnaView.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/SrnaView.html#setGenome(bacnet.datamodel.sequence.Genome)">setGenome</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../sequenceTools/LocalizationDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,java.lang.String%5B%5D%5B%5D,bacnet.datamodel.sequence.Sequence,java.lang.String%5B%5D%5B%5D,bacnet.datamodel.sequence.Genome)">LocalizationDialog</a></span>&#8203;(org.eclipse.swt.widgets.Shell&nbsp;parentShell,
                  java.lang.String[][]&nbsp;arrayGeneToLocalization,
                  <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
                  java.lang.String[][]&nbsp;bioCondsArray,
                  <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create the dialog.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../sequenceTools/SequenceDisplayDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,org.eclipse.e4.ui.workbench.modeling.EPartService,bacnet.datamodel.sequence.Genome,java.lang.String,boolean,int,int,org.biojava3.core.sequence.Strand)">SequenceDisplayDialog</a></span>&#8203;(org.eclipse.swt.widgets.Shell&nbsp;parentShell,
                     org.eclipse.e4.ui.workbench.modeling.EPartService&nbsp;partService,
                     <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                     java.lang.String&nbsp;chromoId,
                     boolean&nbsp;nucleotide,
                     int&nbsp;begin,
                     int&nbsp;end,
                     <a href="../../../../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a>&nbsp;strand)</code></th>
<td class="colLast">
<div class="block">Create the dialog.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.views">
<!--   -->
</a>
<h3>Uses of <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> in <a href="../../../views/package-summary.html">bacnet.views</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../views/package-summary.html">bacnet.views</a> that return <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CoExprNetworkView.</span><code><span class="memberNameLink"><a href="../../../views/CoExprNetworkView.html#getGenome()">getGenome</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../views/package-summary.html">bacnet.views</a> with parameters of type <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CoExprNetworkView.</span><code><span class="memberNameLink"><a href="../../../views/CoExprNetworkView.html#setGenome(bacnet.datamodel.sequence.Genome)">setGenome</a></span>&#8203;(<a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Genome.html" title="class in bacnet.datamodel.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
