<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>Uses of Class bacnet.datamodel.sequence.Sequence</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Class bacnet.datamodel.sequence.Sequence";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Class bacnet.datamodel.sequence.Sequence" class="title">Uses of Class<br>bacnet.datamodel.sequence.Sequence</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.annotation">bacnet.datamodel.annotation</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.proteomics">bacnet.datamodel.proteomics</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.sequence">bacnet.datamodel.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.sequenceNCBI">bacnet.datamodel.sequenceNCBI</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.expressionAtlas.core">bacnet.expressionAtlas.core</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.genomeBrowser.tracksGUI">bacnet.genomeBrowser.tracksGUI</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.blast">bacnet.scripts.blast</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.core">bacnet.scripts.core</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.genome">bacnet.scripts.genome</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.listeriomics.nterm">bacnet.scripts.listeriomics.nterm</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.phylogeny">bacnet.scripts.phylogeny</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.sequenceTools">bacnet.sequenceTools</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.utils">bacnet.utils</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.annotation">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a> that return <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Annotation.</span><code><span class="memberNameLink"><a href="../../annotation/Annotation.html#getElementInfoATbp(bacnet.datamodel.sequence.Chromosome,int)">getElementInfoATbp</a></span>&#8203;(<a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromosome,
                  int&nbsp;bpPosition)</code></th>
<td class="colLast">
<div class="block">Return the first element for which<br>
 begin < bpPosition < end</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a> that return types with arguments of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Annotation.</span><code><span class="memberNameLink"><a href="../../annotation/Annotation.html#getElements(bacnet.datamodel.sequence.Chromosome,int,int)">getElements</a></span>&#8203;(<a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromosome,
           int&nbsp;beginDraw,
           int&nbsp;endDraw)</code></th>
<td class="colLast">
<div class="block">Get the lists of elements between beginDraw and endDraw</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.proteomics">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a></h3>
<table class="useSummary">
<caption><span>Subclasses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></span></code></th>
<td class="colLast">
<div class="block">Object for managing every mapped sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></span></code></th>
<td class="colLast">
<div class="block">This class is used to regroup different Nterm.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a> that return <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>private <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermFilter.</span><code><span class="memberNameLink"><a href="../../proteomics/NTermFilter.html#findOverlap(bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Chromosome)">findOverlap</a></span>&#8203;(<a href="../../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
           <a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromosomeEGDe)</code></th>
<td class="colLast">
<div class="block">Find with which element on EGDe genome and RAST genome an NTerm overlap:<br>
 region of 25bp at the beginning of EGDe gene
 region of 25bp at the beginning of EGDe RAST gene, if this overlap is smaller than with EGde
 take this one, in order to detect bona-fide aTIS
 region of 25bp at the beginning of Srna or ASrna of EGDe, if this overlap is smaller than
 with Srna take this one, in order to detect bona-fide aTIS
 overlap in the middle of EGDe gene
 overlap in the middle of EGDe RAST gene
 overlap in the middle of Srna or ASrna of EGDe<br>
 <br>
 The 25bp are given by <code>aTISCutoff</code> argument of the <code>NTermFilter</code> object</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermUtils.</span><code><span class="memberNameLink"><a href="../../proteomics/NTermUtils.html#getFullORF(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Genome)">getFullORF</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
          <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in frame) containing the
 peptide (amino acid sequence) <br></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermUtils.</span><code><span class="memberNameLink"><a href="../../proteomics/NTermUtils.html#getFullORF(java.lang.String,int,int,boolean,bacnet.datamodel.sequence.Genome)">getFullORF</a></span>&#8203;(java.lang.String&nbsp;peptideSeq,
          int&nbsp;beginPeptide,
          int&nbsp;endPeptide,
          boolean&nbsp;strand,
          <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in frame) containing the
 peptide (amino acid sequence) <br></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TIS.</span><code><span class="memberNameLink"><a href="../../proteomics/TIS.html#load(java.lang.String)">load</a></span>&#8203;(java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a> that return types with arguments of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermCreateData.</span><code><span class="memberNameLink"><a href="../../proteomics/NTermCreateData.html#mapSequence(java.lang.String)">mapSequence</a></span>&#8203;(java.lang.String&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Map an amino acide sequence to 6 frame Genomes<br>
 From initializeGenomeSequence() we get all 6 possible amino acids sequence of the genome, and map
 the peptide<br>
 The different mapping coordinates are transformed into nucleotide coordinates (= genome
 coordinates)</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermFilter.</span><code><span class="memberNameLink"><a href="../../proteomics/NTermFilter.html#applyCutOff(java.lang.String,bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Sequence,int%5B%5D)">applyCutOff</a></span>&#8203;(java.lang.String&nbsp;cutoff,
           <a href="../../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
           <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;overlapSequence,
           int[]&nbsp;overlap)</code></th>
<td class="colLast">
<div class="block">Read a cutoff<br>
 Interprete each cutoff by returning true of false if the NTerm pass the condition<br>
 If the cutoff contain an "or": parse it in two separate cutoff<br>
 If the cutoff starts with an "!"</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermFilter.</span><code><span class="memberNameLink"><a href="../../proteomics/NTermFilter.html#containStopCodon(bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Sequence,int%5B%5D)">containStopCodon</a></span>&#8203;(<a href="../../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
                <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;overlapSequence,
                int[]&nbsp;overlap)</code></th>
<td class="colLast">
<div class="block">Return true if it contains a StopCodon between the peptide and the overlaping sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermUtils.</span><code><span class="memberNameLink"><a href="../../proteomics/NTermUtils.html#getFullORF(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Genome)">getFullORF</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
          <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in frame) containing the
 peptide (amino acid sequence) <br></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private int[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermFilter.</span><code><span class="memberNameLink"><a href="../../proteomics/NTermFilter.html#getOverlap(bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Sequence)">getOverlap</a></span>&#8203;(<a href="../../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
          <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;overlapSequence)</code></th>
<td class="colLast">
<div class="block">Return the difference between the start</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../package-summary.html">bacnet.datamodel.sequence</a></h3>
<table class="useSummary">
<caption><span>Subclasses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../package-summary.html">bacnet.datamodel.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Fields in <a href="../package-summary.html">bacnet.datamodel.sequence</a> with type parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>private java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#allElements">allElements</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#elements">elements</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChromosomeBacteriaSequence.</span><code><span class="memberNameLink"><a href="../ChromosomeBacteriaSequence.html#elements">elements</a></span></code></th>
<td class="colLast">
<div class="block">List of other elements : terminator, etc...</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> that return <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Genome.</span><code><span class="memberNameLink"><a href="../Genome.html#getElement(java.lang.String)">getElement</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Search an element by its name in the genome</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#getSequence(java.lang.String)">getSequence</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Given a sequence name, return the corresponding Sequence<br>
 If name is a Gene name, we search also in geneNameToLocusTagMap</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Sequence.</span><code><span class="memberNameLink"><a href="../Sequence.html#load(java.lang.String)">load</a></span>&#8203;(java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Genome.</span><code><span class="memberNameLink"><a href="../Genome.html#searchElement(java.lang.String)">searchElement</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Search in locustag names, gene name, and newlocustag if a gene is present or not</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> that return types with arguments of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#getAllElements()">getAllElements</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Genome.</span><code><span class="memberNameLink"><a href="../Genome.html#getAllElements()">getAllElements</a></span>()</code></th>
<td class="colLast">
<div class="block">Return all elements contained in all chromosomes</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#getElements()">getElements</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChromosomeBacteriaSequence.</span><code><span class="memberNameLink"><a href="../ChromosomeBacteriaSequence.html#getElements()">getElements</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Codon.</span><code><span class="memberNameLink"><a href="../Codon.html#getFirstCodon(bacnet.datamodel.sequence.Sequence)">getFirstCodon</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Return first threee nucleotide of a Sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Sequence.</span><code><span class="memberNameLink"><a href="../Sequence.html#isEqual(bacnet.datamodel.sequence.Sequence)">isEqual</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">If :
 sequence.getBegin()==this.getBegin()
 && sequence.getEnd()==this.getEnd()
 && sequence.getStrand()==this.getStrand()<br>
 return true</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Sequence.</span><code><span class="memberNameLink"><a href="../Sequence.html#isOverlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)">isOverlap</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
         <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2)</code></th>
<td class="colLast">
<div class="block">Decide if two genes are overlapping <br>
 First, calculate the overlapping size of two genes, finding the gene upstream to the other<br>
 If overlap is negative it will mean that we have no overlap (gene1.end < gene2.begin) <br></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Sequence.</span><code><span class="memberNameLink"><a href="../Sequence.html#isOverlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence,int)">isOverlap</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
         <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2,
         int&nbsp;overlapCutoff)</code></th>
<td class="colLast">
<div class="block">Decide if two genes are overlapping <br>
 First, calculate the overlapping size of two genes, finding the gene upstream to the other<br>
 If overlap is negative it will mean that we have no overlap (gene1.end < gene2.begin) <br></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Sequence.</span><code><span class="memberNameLink"><a href="../Sequence.html#isOverlapStart(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence,int)">isOverlapStart</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
              <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2,
              int&nbsp;sizeStart)</code></th>
<td class="colLast">
<div class="block">Decide if first Sequence is overlapping the start region of the second <br></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Sequence.</span><code><span class="memberNameLink"><a href="../Sequence.html#isSameStrand(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)">isSameStrand</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
            <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2)</code></th>
<td class="colLast">
<div class="block">Decide if two sequences have same strand</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Sequence.</span><code><span class="memberNameLink"><a href="../Sequence.html#overlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)">overlap</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
       <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2)</code></th>
<td class="colLast">
<div class="block">Calculate the overlapping size of two genes <br>
 First, find the gene upstream to the other<br>
 Then, as we got gene1.begin < gene2.begin, overlap = gene1.end - gene2.begin<br>
 overlap can be negative, it will mean that we have no overlap (gene1.end < gene2.begin) <br>
 or overlap can be higher than gene2.length, in that case overlap = gene2.length</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../package-summary.html">bacnet.datamodel.sequence</a> with type arguments of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#setAllElements(java.util.LinkedHashMap)">setAllElements</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;allElements)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../Chromosome.html#setElements(java.util.LinkedHashMap)">setElements</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;elements)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChromosomeBacteriaSequence.</span><code><span class="memberNameLink"><a href="../ChromosomeBacteriaSequence.html#setElements(java.util.ArrayList)">setElements</a></span>&#8203;(java.util.ArrayList&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;elements)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.sequenceNCBI">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a> that return <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeConversionElement.</span><code><span class="memberNameLink"><a href="../../sequenceNCBI/GenomeConversionElement.html#convertDNASequenceGeneral(org.biojava3.core.sequence.DNASequence)">convertDNASequenceGeneral</a></span>&#8203;(<a href="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;dnaSeq)</code></th>
<td class="colLast">
<div class="block">From a DNASequence object, create Sequence<br>
 Mainly used in GenomeConversion methods</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.expressionAtlas.core">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a></h3>
<table class="useSummary">
<caption><span>Constructors in <a href="../../../expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../expressionAtlas/core/GenomeElementAtlas.html#%3Cinit%3E(bacnet.datamodel.sequence.Sequence,bacnet.utils.Filter)">GenomeElementAtlas</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
                  <a href="../../../utils/Filter.html" title="class in bacnet.utils">Filter</a>&nbsp;filter)</code></th>
<td class="colLast">
<div class="block">For a specific sequence update the list of BioCondition in which the genome element is
 over-expressed, under-expressed or not differently expressed<br>
 The update will be dependent of the value of the input filter = cutoff for LogFC and p-value</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.genomeBrowser.tracksGUI">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../genomeBrowser/tracksGUI/package-summary.html">bacnet.genomeBrowser.tracksGUI</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../genomeBrowser/tracksGUI/package-summary.html">bacnet.genomeBrowser.tracksGUI</a> that return <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>private <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TrackCanvasGenome.</span><code><span class="memberNameLink"><a href="../../../genomeBrowser/tracksGUI/TrackCanvasGenome.html#getDisplayedSequence(bacnet.datamodel.sequence.Sequence)">getDisplayedSequence</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Return the sequence from the annotation to display<br>
 The position of this sequence is different from the input one, only if:<br>
 begin < displayRegion.X1
 end > displayRegion.X2</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../genomeBrowser/tracksGUI/package-summary.html">bacnet.genomeBrowser.tracksGUI</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GElement.</span><code><span class="memberNameLink"><a href="../../../genomeBrowser/tracksGUI/GElement.html#displayGene(org.eclipse.swt.graphics.GC,bacnet.datamodel.sequence.Sequence,org.eclipse.swt.graphics.Rectangle,bacnet.genomeBrowser.core.Track,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color,boolean)">displayGene</a></span>&#8203;(org.eclipse.swt.graphics.GC&nbsp;gc,
           <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
           org.eclipse.swt.graphics.Rectangle&nbsp;rectangle,
           <a href="../../../genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a>&nbsp;track,
           org.eclipse.swt.graphics.Color&nbsp;lightColor,
           org.eclipse.swt.graphics.Color&nbsp;darkColor,
           org.eclipse.swt.graphics.Color&nbsp;lineColor,
           org.eclipse.swt.graphics.Color&nbsp;textColor,
           boolean&nbsp;nTerm)</code></th>
<td class="colLast">
<div class="block">Display Gene annotation = square and arrow</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TrackCanvasGenome.</span><code><span class="memberNameLink"><a href="../../../genomeBrowser/tracksGUI/TrackCanvasGenome.html#getDisplayedSequence(bacnet.datamodel.sequence.Sequence)">getDisplayedSequence</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Return the sequence from the annotation to display<br>
 The position of this sequence is different from the input one, only if:<br>
 begin < displayRegion.X1
 end > displayRegion.X2</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.blast">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../scripts/blast/package-summary.html">bacnet.scripts.blast</a></h3>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../scripts/blast/package-summary.html">bacnet.scripts.blast</a> with type arguments of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PhylogenySmallORFs.</span><code><span class="memberNameLink"><a href="../../../scripts/blast/PhylogenySmallORFs.html#createPhylogenyFile(java.util.ArrayList)">createPhylogenyFile</a></span>&#8203;(java.util.ArrayList&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;smallORFs)</code></th>
<td class="colLast">
<div class="block">reading filtered table and create fasta file, assuming that Lite.excel table have been reduced
 manually by removing doublons</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Blast.</span><code><span class="memberNameLink"><a href="../../../scripts/blast/Blast.html#multiSpeciesAlignNucleotide(java.util.ArrayList,boolean)">multiSpeciesAlignNucleotide</a></span>&#8203;(java.util.ArrayList&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;seqs,
                           boolean&nbsp;smallSequence)</code></th>
<td class="colLast">
<div class="block">Run a Blast for each sRNA on a list of Genomes given in tempDatabase</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.core">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../scripts/core/package-summary.html">bacnet.scripts.core</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/core/package-summary.html">bacnet.scripts.core</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">UNAfold.</span><code><span class="memberNameLink"><a href="../../../scripts/core/UNAfold.html#foldRNA(bacnet.datamodel.sequence.Sequence,java.lang.String,boolean)">foldRNA</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
       java.lang.String&nbsp;fileName,
       boolean&nbsp;removeTemp)</code></th>
<td class="colLast">
<div class="block">For a specific sequence, get its RNA sequence and fold it<br>
 Calculate possible free energies<br>
 transform results into annotation file<br>
 Display using sir_graph</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">UNAfold.</span><code><span class="memberNameLink"><a href="../../../scripts/core/UNAfold.html#unafoldRNA(bacnet.datamodel.sequence.Sequence,java.lang.String)">unafoldRNA</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
          java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Run UNAFOld.pl Perl script</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.genome">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>private static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SDProfile.</span><code><span class="memberNameLink"><a href="../../../scripts/genome/SDProfile.html#test(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Genome)">test</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
    <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.listeriomics.nterm">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a> that return types with arguments of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RASToverlap.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/RASToverlap.html#getEGDeUnknownRAST()">getEGDeUnknownRAST</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RASToverlap.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/RASToverlap.html#getUnknownRAST(java.lang.String,java.lang.String)">getUnknownRAST</a></span>&#8203;(java.lang.String&nbsp;genome,
              java.lang.String&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Read the list of RAST elements totally unknown and create a list of Sequence from it</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermStat.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getLogo(bacnet.datamodel.sequence.Sequence)">getLogo</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">For a given Sequence, return the nucleotide sequence before and after starting position</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a> with type arguments of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NTermConservationWithRiboProfiling.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/nterm/NTermConservationWithRiboProfiling.html#multiAlign(java.util.ArrayList,java.lang.String,java.lang.String)">multiAlign</a></span>&#8203;(java.util.ArrayList&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;sequences,
          java.lang.String&nbsp;suffix,
          java.lang.String&nbsp;genomeName)</code></th>
<td class="colLast">
<div class="block">Run a Blast for each sRNA on a list of Genomes given in tempDatabase</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.phylogeny">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../scripts/phylogeny/package-summary.html">bacnet.scripts.phylogeny</a></h3>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../scripts/phylogeny/package-summary.html">bacnet.scripts.phylogeny</a> with type arguments of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PhylogenySmallORFs.</span><code><span class="memberNameLink"><a href="../../../scripts/phylogeny/PhylogenySmallORFs.html#createPhylogenyFile(java.util.ArrayList)">createPhylogenyFile</a></span>&#8203;(java.util.ArrayList&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;smallORFs)</code></th>
<td class="colLast">
<div class="block">reading filtered table and create fasta file, assuming that Lite.excel table have been reduced
 manually by removing doublons</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PhylogenySmallORFs.</span><code><span class="memberNameLink"><a href="../../../scripts/phylogeny/PhylogenySmallORFs.html#displayMultiAlign(java.util.ArrayList,org.eclipse.e4.ui.workbench.modeling.EPartService)">displayMultiAlign</a></span>&#8203;(java.util.ArrayList&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;smallORFs,
                 org.eclipse.e4.ui.workbench.modeling.EPartService&nbsp;partService)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PhylogenySmallORFs.</span><code><span class="memberNameLink"><a href="../../../scripts/phylogeny/PhylogenySmallORFs.html#displayPhylogeny(java.util.ArrayList)">displayPhylogeny</a></span>&#8203;(java.util.ArrayList&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;smallORFs)</code></th>
<td class="colLast">
<div class="block">Displaying the newly created tree with Archaeopteryx.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.sequenceTools">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a></h3>
<table class="useSummary">
<caption><span>Fields in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a> declared as <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>private <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">LocalizationDialog.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/LocalizationDialog.html#sequence">sequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String[][]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneViewHomologTools.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/GeneViewHomologTools.html#loadArrayHomologs(bacnet.datamodel.sequence.Sequence,java.lang.String%5B%5D%5B%5D,java.util.ArrayList,java.util.ArrayList)">loadArrayHomologs</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
                 java.lang.String[][]&nbsp;bioCondsArray,
                 java.util.ArrayList&lt;java.lang.String[]&gt;&nbsp;bioConds,
                 java.util.ArrayList&lt;java.lang.String[]&gt;&nbsp;bioCondsToDisplay)</code></th>
<td class="colLast">
<div class="block">Load genome info and add homolog information on the column index1 and 2</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneViewLocalizationTools.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/GeneViewLocalizationTools.html#loadLocalizationFigure(org.eclipse.swt.browser.Browser,java.lang.String%5B%5D%5B%5D,bacnet.datamodel.sequence.Sequence,java.lang.String%5B%5D%5B%5D,org.eclipse.swt.widgets.Table,bacnet.datamodel.sequence.Genome)">loadLocalizationFigure</a></span>&#8203;(org.eclipse.swt.browser.Browser&nbsp;browserLocalization,
                      java.lang.String[][]&nbsp;arrayGeneToLocalization,
                      <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
                      java.lang.String[][]&nbsp;bioCondsArray,
                      org.eclipse.swt.widgets.Table&nbsp;tableLocalization,
                      <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Load SVG figure of phylogeny and replace all strain name by homolog informations<br>
 Save to JPG file and display it</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneViewTranscriptomeTools.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/GeneViewTranscriptomeTools.html#updateExpressionAtlas(bacnet.datamodel.sequence.Sequence,org.eclipse.swt.widgets.Text,bacnet.sequenceTools.GeneView,java.lang.String%5B%5D%5B%5D)">updateExpressionAtlas</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
                     org.eclipse.swt.widgets.Text&nbsp;txtCutoffLogFC,
                     <a href="../../../sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a>&nbsp;viewer,
                     java.lang.String[][]&nbsp;arrayDataList)</code></th>
<td class="colLast">
<div class="block">Update display of expression data results</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneViewProteomeTools.</span><code><span class="memberNameLink"><a href="../../../sequenceTools/GeneViewProteomeTools.html#updateProteomesTable(bacnet.datamodel.sequence.Sequence,bacnet.sequenceTools.GeneView,java.lang.String%5B%5D%5B%5D)">updateProteomesTable</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
                    <a href="../../../sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a>&nbsp;viewer,
                    java.lang.String[][]&nbsp;arrayProteomeList)</code></th>
<td class="colLast">
<div class="block">Update proteomics table</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../sequenceTools/LocalizationDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,java.lang.String%5B%5D%5B%5D,bacnet.datamodel.sequence.Sequence,java.lang.String%5B%5D%5B%5D,bacnet.datamodel.sequence.Genome)">LocalizationDialog</a></span>&#8203;(org.eclipse.swt.widgets.Shell&nbsp;parentShell,
                  java.lang.String[][]&nbsp;arrayGeneToLocalization,
                  <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
                  java.lang.String[][]&nbsp;bioCondsArray,
                  <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create the dialog.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.utils">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../utils/package-summary.html">bacnet.utils</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../utils/package-summary.html">bacnet.utils</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">UNAfold.</span><code><span class="memberNameLink"><a href="../../../utils/UNAfold.html#foldRNA(bacnet.datamodel.sequence.Sequence,java.lang.String,boolean)">foldRNA</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
       java.lang.String&nbsp;fileName,
       boolean&nbsp;removeTemp)</code></th>
<td class="colLast">
<div class="block">For a specific sequence, get its RNA sequence and fold it<br>
 Calculate possible free energies<br>
 transform results into annotation file<br>
 Display using sir_graph</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">UNAfold.</span><code><span class="memberNameLink"><a href="../../../utils/UNAfold.html#unafoldRNA(bacnet.datamodel.sequence.Sequence,java.lang.String)">unafoldRNA</a></span>&#8203;(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
          java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Run UNAFOld.pl Perl script</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
