<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Fri Dec 10 09:15:34 CET 2021 -->
<title>Uses of Class bacnet.datamodel.sequence.Sequence</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-10">
<meta name="description" content="use: package: bacnet.datamodel.sequence, class: Sequence">
<meta name="generator" content="javadoc/ClassUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-use-page">
<script type="text/javascript">var pathtoroot = "../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Class</a></li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Class bacnet.datamodel.sequence.Sequence" class="title">Uses of Class<br>bacnet.datamodel.sequence.Sequence</h1>
</div>
<div class="caption"><span>Packages that use <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#bacnet.datamodel.annotation">bacnet.datamodel.annotation</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.datamodel.proteomics">bacnet.datamodel.proteomics</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.datamodel.sequence">bacnet.datamodel.sequence</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.datamodel.sequenceNCBI">bacnet.datamodel.sequenceNCBI</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.expressionAtlas.core">bacnet.expressionAtlas.core</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.genomeBrowser.tracksGUI">bacnet.genomeBrowser.tracksGUI</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.scripts.blast">bacnet.scripts.blast</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.scripts.core">bacnet.scripts.core</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.scripts.listeriomics.nterm">bacnet.scripts.listeriomics.nterm</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.scripts.phylogeny">bacnet.scripts.phylogeny</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.sequenceTools">bacnet.sequenceTools</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.utils">bacnet.utils</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<section class="class-uses">
<ul class="block-list">
<li>
<section class="detail" id="bacnet.datamodel.annotation">
<h2>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a></h2>
<div class="caption"><span>Methods in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a> that return <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Annotation.</span><code><a href="../../annotation/Annotation.html#getElementInfoATbp(bacnet.datamodel.sequence.Chromosome,int)" class="member-name-link">getElementInfoATbp</a><wbr>(<a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromosome,
 int&nbsp;bpPosition)</code></div>
<div class="col-last even-row-color">
<div class="block">Return the first element for which<br>
 begin &lt; bpPosition &lt; end</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a> that return types with arguments of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Annotation.</span><code><a href="../../annotation/Annotation.html#getElements(bacnet.datamodel.sequence.Chromosome,int,int)" class="member-name-link">getElements</a><wbr>(<a href="../Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;chromosome,
 int&nbsp;beginDraw,
 int&nbsp;endDraw)</code></div>
<div class="col-last even-row-color">
<div class="block">Get the lists of elements between beginDraw and endDraw</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.datamodel.proteomics">
<h2>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a></h2>
<div class="caption"><span>Subclasses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../proteomics/NTerm.html" class="type-name-link" title="class in bacnet.datamodel.proteomics">NTerm</a></code></div>
<div class="col-last even-row-color">
<div class="block">Object for managing every mapped sequences</div>
</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../proteomics/TIS.html" class="type-name-link" title="class in bacnet.datamodel.proteomics">TIS</a></code></div>
<div class="col-last odd-row-color">
<div class="block">This class is used to regroup different Nterm.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a> that return <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">NTermUtils.</span><code><a href="../../proteomics/NTermUtils.html#getFullORF(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Genome)" class="member-name-link">getFullORF</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in frame) containing the
 peptide (amino acid sequence) <br></div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">NTermUtils.</span><code><a href="../../proteomics/NTermUtils.html#getFullORF(java.lang.String,int,int,boolean,bacnet.datamodel.sequence.Genome)" class="member-name-link">getFullORF</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;peptideSeq,
 int&nbsp;beginPeptide,
 int&nbsp;endPeptide,
 boolean&nbsp;strand,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in frame) containing the
 peptide (amino acid sequence) <br></div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">TIS.</span><code><a href="../../proteomics/TIS.html#load(java.lang.String)" class="member-name-link">load</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName)</code></div>
<div class="col-last even-row-color">
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../proteomics/package-summary.html">bacnet.datamodel.proteomics</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>boolean</code></div>
<div class="col-second even-row-color"><span class="type-name-label">NTermFilter.</span><code><a href="../../proteomics/NTermFilter.html#applyCutOff(java.lang.String,bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Sequence,int%5B%5D)" class="member-name-link">applyCutOff</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;cutoff,
 <a href="../../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
 <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;overlapSequence,
 int[]&nbsp;overlap)</code></div>
<div class="col-last even-row-color">
<div class="block">Read a cutoff<br>
 Interprete each cutoff by returning true of false if the NTerm pass the condition<br>
 If the cutoff contain an "or": parse it in two separate cutoff<br>
 If the cutoff starts with an "!"</div>
</div>
<div class="col-first odd-row-color"><code>static boolean</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">NTermFilter.</span><code><a href="../../proteomics/NTermFilter.html#containStopCodon(bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Sequence,int%5B%5D)" class="member-name-link">containStopCodon</a><wbr>(<a href="../../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&nbsp;nTerm,
 <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;overlapSequence,
 int[]&nbsp;overlap)</code></div>
<div class="col-last odd-row-color">
<div class="block">Return true if it contains a StopCodon between the peptide and the overlaping sequence</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">NTermUtils.</span><code><a href="../../proteomics/NTermUtils.html#getFullORF(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Genome)" class="member-name-link">getFullORF</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in frame) containing the
 peptide (amino acid sequence) <br></div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.datamodel.sequence">
<h2>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../package-summary.html">bacnet.datamodel.sequence</a></h2>
<div class="caption"><span>Subclasses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../package-summary.html">bacnet.datamodel.sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../Gene.html" class="type-name-link" title="class in bacnet.datamodel.sequence">Gene</a></code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../NcRNA.html" class="type-name-link" title="class in bacnet.datamodel.sequence">NcRNA</a></code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../Operon.html" class="type-name-link" title="class in bacnet.datamodel.sequence">Operon</a></code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../Srna.html" class="type-name-link" title="class in bacnet.datamodel.sequence">Srna</a></code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> that return <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Genome.</span><code><a href="../Genome.html#getElement(java.lang.String)" class="member-name-link">getElement</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name)</code></div>
<div class="col-last even-row-color">
<div class="block">Search an element by its name in the genome</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Chromosome.</span><code><a href="../Chromosome.html#getSequence(java.lang.String)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name)</code></div>
<div class="col-last odd-row-color">
<div class="block">Given a sequence name, return the corresponding Sequence<br>
 If name is a Gene name, we search also in geneNameToLocusTagMap</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Sequence.</span><code><a href="../Sequence.html#load(java.lang.String)" class="member-name-link">load</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName)</code></div>
<div class="col-last even-row-color">
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Genome.</span><code><a href="../Genome.html#searchElement(java.lang.String)" class="member-name-link">searchElement</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name)</code></div>
<div class="col-last odd-row-color">
<div class="block">Search in locustag names, gene name, and newlocustag if a gene is present or not</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> that return types with arguments of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/LinkedHashMap.html" title="class or interface in java.util" class="external-link">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Chromosome.</span><code><a href="../Chromosome.html#getAllElements()" class="member-name-link">getAllElements</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Genome.</span><code><a href="../Genome.html#getAllElements()" class="member-name-link">getAllElements</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Return all elements contained in all chromosomes</div>
</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/LinkedHashMap.html" title="class or interface in java.util" class="external-link">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Chromosome.</span><code><a href="../Chromosome.html#getElements()" class="member-name-link">getElements</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">ChromosomeBacteriaSequence.</span><code><a href="../ChromosomeBacteriaSequence.html#getElements()" class="member-name-link">getElements</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequence</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Codon.</span><code><a href="../Codon.html#getFirstCodon(bacnet.datamodel.sequence.Sequence)" class="member-name-link">getFirstCodon</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence)</code></div>
<div class="col-last even-row-color">
<div class="block">Return first threee nucleotide of a Sequence</div>
</div>
<div class="col-first odd-row-color"><code>boolean</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Sequence.</span><code><a href="../Sequence.html#isEqual(bacnet.datamodel.sequence.Sequence)" class="member-name-link">isEqual</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence)</code></div>
<div class="col-last odd-row-color">
<div class="block">If :
 sequence.getBegin()==this.getBegin()
 &amp;&amp; sequence.getEnd()==this.getEnd()
 &amp;&amp; sequence.getStrand()==this.getStrand()<br>
 return true</div>
</div>
<div class="col-first even-row-color"><code>static boolean</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Sequence.</span><code><a href="../Sequence.html#isOverlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)" class="member-name-link">isOverlap</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
 <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2)</code></div>
<div class="col-last even-row-color">
<div class="block">Decide if two genes are overlapping <br>
 First, calculate the overlapping size of two genes, finding the gene upstream to the other<br>
 If overlap is negative it will mean that we have no overlap (gene1.end &lt; gene2.begin) <br></div>
</div>
<div class="col-first odd-row-color"><code>static boolean</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Sequence.</span><code><a href="../Sequence.html#isOverlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence,int)" class="member-name-link">isOverlap</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
 <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2,
 int&nbsp;overlapCutoff)</code></div>
<div class="col-last odd-row-color">
<div class="block">Decide if two genes are overlapping <br>
 First, calculate the overlapping size of two genes, finding the gene upstream to the other<br>
 If overlap is negative it will mean that we have no overlap (gene1.end &lt; gene2.begin) <br></div>
</div>
<div class="col-first even-row-color"><code>static boolean</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Sequence.</span><code><a href="../Sequence.html#isOverlapStart(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence,int)" class="member-name-link">isOverlapStart</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
 <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2,
 int&nbsp;sizeStart)</code></div>
<div class="col-last even-row-color">
<div class="block">Decide if first Sequence is overlapping the start region of the second <br></div>
</div>
<div class="col-first odd-row-color"><code>static boolean</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Sequence.</span><code><a href="../Sequence.html#isSameStrand(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)" class="member-name-link">isSameStrand</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
 <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2)</code></div>
<div class="col-last odd-row-color">
<div class="block">Decide if two sequences have same strand</div>
</div>
<div class="col-first even-row-color"><code>static int</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Sequence.</span><code><a href="../Sequence.html#overlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)" class="member-name-link">overlap</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
 <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2)</code></div>
<div class="col-last even-row-color">
<div class="block">Calculate the overlapping size of two genes <br>
 First, find the gene upstream to the other<br>
 Then, as we got gene1.begin &lt; gene2.begin, overlap = gene1.end - gene2.begin<br>
 overlap can be negative, it will mean that we have no overlap (gene1.end &lt; gene2.begin) <br>
 or overlap can be higher than gene2.length, in that case overlap = gene2.length</div>
</div>
</div>
<div class="caption"><span>Method parameters in <a href="../package-summary.html">bacnet.datamodel.sequence</a> with type arguments of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Chromosome.</span><code><a href="../Chromosome.html#setAllElements(java.util.LinkedHashMap)" class="member-name-link">setAllElements</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/LinkedHashMap.html" title="class or interface in java.util" class="external-link">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;allElements)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Chromosome.</span><code><a href="../Chromosome.html#setElements(java.util.LinkedHashMap)" class="member-name-link">setElements</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/LinkedHashMap.html" title="class or interface in java.util" class="external-link">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;elements)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ChromosomeBacteriaSequence.</span><code><a href="../ChromosomeBacteriaSequence.html#setElements(java.util.ArrayList)" class="member-name-link">setElements</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;elements)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.datamodel.sequenceNCBI">
<h2>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></h2>
<div class="caption"><span>Methods in <a href="../../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a> that return <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">GenomeConversionElement.</span><code><a href="../../sequenceNCBI/GenomeConversionElement.html#convertDNASequenceGeneral(org.biojava3.core.sequence.DNASequence)" class="member-name-link">convertDNASequenceGeneral</a><wbr>(<a href="../../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;dnaSeq)</code></div>
<div class="col-last even-row-color">
<div class="block">From a DNASequence object, create Sequence<br>
 Mainly used in GenomeConversion methods</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.expressionAtlas.core">
<h2>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a></h2>
<div class="caption"><span>Constructors in <a href="../../../expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../../expressionAtlas/core/GenomeElementAtlas.html#%3Cinit%3E(bacnet.datamodel.sequence.Sequence,bacnet.utils.Filter,boolean)" class="member-name-link">GenomeElementAtlas</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
 <a href="../../../utils/Filter.html" title="class in bacnet.utils">Filter</a>&nbsp;filter,
 boolean&nbsp;transcriptome)</code></div>
<div class="col-last even-row-color">
<div class="block">For a specific sequence update the list of BioCondition in which the genome element is
 over-expressed, under-expressed or not differently expressed<br>
 The update will be dependent of the value of the input filter = cutoff for LogFC and p-value</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.genomeBrowser.tracksGUI">
<h2>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../genomeBrowser/tracksGUI/package-summary.html">bacnet.genomeBrowser.tracksGUI</a></h2>
<div class="caption"><span>Methods in <a href="../../../genomeBrowser/tracksGUI/package-summary.html">bacnet.genomeBrowser.tracksGUI</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">GElement.</span><code><a href="../../../genomeBrowser/tracksGUI/GElement.html#displayGene(org.eclipse.swt.graphics.GC,bacnet.datamodel.sequence.Sequence,org.eclipse.swt.graphics.Rectangle,bacnet.genomeBrowser.core.Track,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color,boolean)" class="member-name-link">displayGene</a><wbr>(org.eclipse.swt.graphics.GC&nbsp;gc,
 <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
 org.eclipse.swt.graphics.Rectangle&nbsp;rectangle,
 <a href="../../../genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a>&nbsp;track,
 org.eclipse.swt.graphics.Color&nbsp;lightColor,
 org.eclipse.swt.graphics.Color&nbsp;darkColor,
 org.eclipse.swt.graphics.Color&nbsp;lineColor,
 org.eclipse.swt.graphics.Color&nbsp;textColor,
 boolean&nbsp;nTerm)</code></div>
<div class="col-last even-row-color">
<div class="block">Display Gene annotation = square and arrow</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.scripts.blast">
<h2>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../scripts/blast/package-summary.html">bacnet.scripts.blast</a></h2>
<div class="caption"><span>Method parameters in <a href="../../../scripts/blast/package-summary.html">bacnet.scripts.blast</a> with type arguments of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">PhylogenySmallORFs.</span><code><a href="../../../scripts/blast/PhylogenySmallORFs.html#createPhylogenyFile(java.util.ArrayList)" class="member-name-link">createPhylogenyFile</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;smallORFs)</code></div>
<div class="col-last even-row-color">
<div class="block">reading filtered table and create fasta file, assuming that Lite.excel table have been reduced
 manually by removing doublons</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Blast.</span><code><a href="../../../scripts/blast/Blast.html#multiSpeciesAlignNucleotide(java.util.ArrayList,boolean)" class="member-name-link">multiSpeciesAlignNucleotide</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;seqs,
 boolean&nbsp;smallSequence)</code></div>
<div class="col-last odd-row-color">
<div class="block">Run a Blast for each sRNA on a list of Genomes given in tempDatabase</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.scripts.core">
<h2>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../scripts/core/package-summary.html">bacnet.scripts.core</a></h2>
<div class="caption"><span>Methods in <a href="../../../scripts/core/package-summary.html">bacnet.scripts.core</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">UNAfold.</span><code><a href="../../../scripts/core/UNAfold.html#foldRNA(bacnet.datamodel.sequence.Sequence,java.lang.String,boolean)" class="member-name-link">foldRNA</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName,
 boolean&nbsp;removeTemp)</code></div>
<div class="col-last even-row-color">
<div class="block">For a specific sequence, get its RNA sequence and fold it<br>
 Calculate possible free energies<br>
 transform results into annotation file<br>
 Display using sir_graph</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">UNAfold.</span><code><a href="../../../scripts/core/UNAfold.html#unafoldRNA(bacnet.datamodel.sequence.Sequence,java.lang.String)" class="member-name-link">unafoldRNA</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName)</code></div>
<div class="col-last odd-row-color">
<div class="block">Run UNAFOld.pl Perl script</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.scripts.listeriomics.nterm">
<h2>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></h2>
<div class="caption"><span>Methods in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a> that return types with arguments of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">RASToverlap.</span><code><a href="../../../scripts/listeriomics/nterm/RASToverlap.html#getEGDeUnknownRAST()" class="member-name-link">getEGDeUnknownRAST</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">RASToverlap.</span><code><a href="../../../scripts/listeriomics/nterm/RASToverlap.html#getUnknownRAST(java.lang.String,java.lang.String)" class="member-name-link">getUnknownRAST</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name)</code></div>
<div class="col-last odd-row-color">
<div class="block">Read the list of RAST elements totally unknown and create a list of Sequence from it</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">NTermStat.</span><code><a href="../../../scripts/listeriomics/nterm/NTermStat.html#getLogo(bacnet.datamodel.sequence.Sequence)" class="member-name-link">getLogo</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene)</code></div>
<div class="col-last even-row-color">
<div class="block">For a given Sequence, return the nucleotide sequence before and after starting position</div>
</div>
</div>
<div class="caption"><span>Method parameters in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a> with type arguments of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">NTermConservationWithRiboProfiling.</span><code><a href="../../../scripts/listeriomics/nterm/NTermConservationWithRiboProfiling.html#multiAlign(java.util.ArrayList,java.lang.String,java.lang.String)" class="member-name-link">multiAlign</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;sequences,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;suffix,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName)</code></div>
<div class="col-last even-row-color">
<div class="block">Run a Blast for each sRNA on a list of Genomes given in tempDatabase</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.scripts.phylogeny">
<h2>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../scripts/phylogeny/package-summary.html">bacnet.scripts.phylogeny</a></h2>
<div class="caption"><span>Method parameters in <a href="../../../scripts/phylogeny/package-summary.html">bacnet.scripts.phylogeny</a> with type arguments of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">PhylogenySmallORFs.</span><code><a href="../../../scripts/phylogeny/PhylogenySmallORFs.html#createPhylogenyFile(java.util.ArrayList)" class="member-name-link">createPhylogenyFile</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;smallORFs)</code></div>
<div class="col-last even-row-color">
<div class="block">reading filtered table and create fasta file, assuming that Lite.excel table have been reduced
 manually by removing doublons</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">PhylogenySmallORFs.</span><code><a href="../../../scripts/phylogeny/PhylogenySmallORFs.html#displayMultiAlign(java.util.ArrayList,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayMultiAlign</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;smallORFs,
 org.eclipse.e4.ui.workbench.modeling.EPartService&nbsp;partService)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">PhylogenySmallORFs.</span><code><a href="../../../scripts/phylogeny/PhylogenySmallORFs.html#displayPhylogeny(java.util.ArrayList)" class="member-name-link">displayPhylogeny</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;smallORFs)</code></div>
<div class="col-last even-row-color">
<div class="block">Displaying the newly created tree with Archaeopteryx.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.sequenceTools">
<h2>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a></h2>
<div class="caption"><span>Methods in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[][]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">GeneViewHomologTools.</span><code><a href="../../../sequenceTools/GeneViewHomologTools.html#loadArrayHomologs(bacnet.datamodel.sequence.Sequence,java.lang.String%5B%5D%5B%5D,java.util.ArrayList,java.util.ArrayList)" class="member-name-link">loadArrayHomologs</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[][]&nbsp;bioCondsArray,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[]&gt;&nbsp;bioConds,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[]&gt;&nbsp;bioCondsToDisplay)</code></div>
<div class="col-last even-row-color">
<div class="block">Load genome info and add homolog information on the column index1 and 2</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GeneViewLocalizationTools.</span><code><a href="../../../sequenceTools/GeneViewLocalizationTools.html#loadLocalizationFigure(org.eclipse.swt.browser.Browser,java.lang.String%5B%5D%5B%5D,bacnet.datamodel.sequence.Sequence,java.lang.String%5B%5D%5B%5D,org.eclipse.swt.widgets.Table,bacnet.datamodel.sequence.Genome)" class="member-name-link">loadLocalizationFigure</a><wbr>(org.eclipse.swt.browser.Browser&nbsp;browserLocalization,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[][]&nbsp;arrayGeneToLocalization,
 <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[][]&nbsp;bioCondsArray,
 org.eclipse.swt.widgets.Table&nbsp;tableLocalization,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">
<div class="block">Load SVG figure of phylogeny and replace all strain name by homolog informations<br>
 Save to JPG file and display it</div>
</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">GeneViewTranscriptomeTools.</span><code><a href="../../../sequenceTools/GeneViewTranscriptomeTools.html#updateExpressionAtlas(bacnet.datamodel.sequence.Sequence,org.eclipse.swt.widgets.Text,bacnet.sequenceTools.GeneView,java.lang.String%5B%5D%5B%5D)" class="member-name-link">updateExpressionAtlas</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
 org.eclipse.swt.widgets.Text&nbsp;txtCutoffLogFC,
 <a href="../../../sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a>&nbsp;viewer,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[][]&nbsp;arrayDataList)</code></div>
<div class="col-last even-row-color">
<div class="block">Update display of expression data results</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GeneViewProteomeTools.</span><code><a href="../../../sequenceTools/GeneViewProteomeTools.html#updateProteinAtlas(bacnet.datamodel.sequence.Sequence,org.eclipse.swt.widgets.Text,bacnet.sequenceTools.GeneView,java.lang.String%5B%5D%5B%5D)" class="member-name-link">updateProteinAtlas</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
 org.eclipse.swt.widgets.Text&nbsp;txtCutoffLogFC,
 <a href="../../../sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a>&nbsp;viewer,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[][]&nbsp;arrayProteinAtlasList)</code></div>
<div class="col-last odd-row-color">
<div class="block">Update display of expression data results</div>
</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">GeneViewProteomeTools.</span><code><a href="../../../sequenceTools/GeneViewProteomeTools.html#updateProteomesTable(bacnet.datamodel.sequence.Sequence,bacnet.sequenceTools.GeneView,java.lang.String%5B%5D%5B%5D)" class="member-name-link">updateProteomesTable</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
 <a href="../../../sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a>&nbsp;viewer,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[][]&nbsp;arrayProteomeList)</code></div>
<div class="col-last even-row-color">
<div class="block">Update proteomics table = Protein expression with a value over 0</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GeneViewTranscriptomeTools.</span><code><a href="../../../sequenceTools/GeneViewTranscriptomeTools.html#updateTranscriptomesTable(bacnet.datamodel.sequence.Sequence,bacnet.sequenceTools.GeneView,java.lang.String%5B%5D%5B%5D)" class="member-name-link">updateTranscriptomesTable</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
 <a href="../../../sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a>&nbsp;viewer,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[][]&nbsp;arrayTranscriptomesList)</code></div>
<div class="col-last odd-row-color">
<div class="block">Update  transcriptomes table = transcript expression with a value over 0</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../../../sequenceTools/package-summary.html">bacnet.sequenceTools</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../../sequenceTools/LocalizationDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,java.lang.String%5B%5D%5B%5D,bacnet.datamodel.sequence.Sequence,java.lang.String%5B%5D%5B%5D,bacnet.datamodel.sequence.Genome)" class="member-name-link">LocalizationDialog</a><wbr>(org.eclipse.swt.widgets.Shell&nbsp;parentShell,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[][]&nbsp;arrayGeneToLocalization,
 <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[][]&nbsp;bioCondsArray,
 <a href="../Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Create the dialog.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.utils">
<h2>Uses of <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> in <a href="../../../utils/package-summary.html">bacnet.utils</a></h2>
<div class="caption"><span>Methods in <a href="../../../utils/package-summary.html">bacnet.utils</a> with parameters of type <a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">UNAfold.</span><code><a href="../../../utils/UNAfold.html#foldRNA(bacnet.datamodel.sequence.Sequence,java.lang.String,boolean)" class="member-name-link">foldRNA</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName,
 boolean&nbsp;removeTemp)</code></div>
<div class="col-last even-row-color">
<div class="block">For a specific sequence, get its RNA sequence and fold it<br>
 Calculate possible free energies<br>
 transform results into annotation file<br>
 Display using sir_graph</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">UNAfold.</span><code><a href="../../../utils/UNAfold.html#unafoldRNA(bacnet.datamodel.sequence.Sequence,java.lang.String)" class="member-name-link">unafoldRNA</a><wbr>(<a href="../Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;seq,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;fileName)</code></div>
<div class="col-last odd-row-color">
<div class="block">Run UNAFOld.pl Perl script</div>
</div>
</div>
</section>
</li>
</ul>
</section>
</main>
</div>
</div>
</body>
</html>
