<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>Uses of Interface org.biojava3.core.sequence.io.template.FastaHeaderParserInterface</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="use: package: org.biojava3.core.sequence.io.template, interface: FastaHeaderParserInterface">
<meta name="generator" content="javadoc/ClassUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-use-page">
<script type="text/javascript">var pathtoroot = "../../../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">Class</a></li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Interface org.biojava3.core.sequence.io.template.FastaHeaderParserInterface" class="title">Uses of Interface<br>org.biojava3.core.sequence.io.template.FastaHeaderParserInterface</h1>
</div>
<div class="caption"><span>Packages that use <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#bacnet.reader">bacnet.reader</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava3.core.sequence.io">org.biojava3.core.sequence.io</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<section class="class-uses">
<ul class="block-list">
<li>
<section class="detail" id="bacnet.reader">
<h2>Uses of <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a> in <a href="../../../../../../../bacnet/reader/package-summary.html">bacnet.reader</a></h2>
<div class="caption"><span>Classes in <a href="../../../../../../../bacnet/reader/package-summary.html">bacnet.reader</a> that implement <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../../../../../../bacnet/reader/NCBIFastaHeaderParser.html" class="type-name-link" title="class in bacnet.reader">NCBIFastaHeaderParser</a>&lt;S extends <a href="../../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../../../template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last even-row-color">
<div class="block">My own implementation of a FastaHeaderParserInterface to parse Fasta files from NCBI ftp

 NCBI fna (genome and plasmids) gi|gi-number|ref|accession|name gi|16802048|ref|NC_003210.1|
 Listeria monocytogenes EGD-e, complete genome NCBI ffn (coding) ref|refGenome
 gi-number|:start-end name ref|NC_003210.1|:318-1673 chromosomal replication initiation protein
 [Listeria monocytogenes EGD-e] NCBI frn (non-coding) ref|refGenome gi-number|:start-end| name|
 [locus-tag] ref|NC_003210|:82705-82777|Lys tRNA| [locus_tag=lmot01] NCBI faa (proteins)
 gi|gi-number|ref|accession|name gi|16802049|ref|NP_463534.1| chromosomal replication initiation
 protein [Listeria monocytogenes EGD-e]</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../../../../../../../bacnet/reader/package-summary.html">bacnet.reader</a> with parameters of type <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../../../../../../bacnet/reader/FastaFileReader.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)" class="member-name-link">FastaFileReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html" title="class or interface in java.io" class="external-link">File</a>&nbsp;file,
 <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="../../../../../../../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">S</a>,<wbr><a href="../../../../../../../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;headerParser,
 <a href="../SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../../../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;sequenceCreator)</code></div>
<div class="col-last even-row-color">
<div class="block">If you are going to use the FileProxyProteinSequenceCreator then you need to use this constructor
 because we need details about the location of the file.</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../../../../../../bacnet/reader/FastaFileReader.html#%3Cinit%3E(java.io.InputStream,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)" class="member-name-link">FastaFileReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/InputStream.html" title="class or interface in java.io" class="external-link">InputStream</a>&nbsp;is,
 <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="../../../../../../../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">S</a>,<wbr><a href="../../../../../../../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;headerParser,
 <a href="../SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../../../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;sequenceCreator)</code></div>
<div class="col-last odd-row-color">
<div class="block">If you are going to use FileProxyProteinSequenceCreator then do not use this constructor because
 we need details about local file offsets for quick reads.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava3.core.sequence.io">
<h2>Uses of <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a> in <a href="../../package-summary.html">org.biojava3.core.sequence.io</a></h2>
<div class="caption"><span>Classes in <a href="../../package-summary.html">org.biojava3.core.sequence.io</a> that implement <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../GenericFastaHeaderParser.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser</a>&lt;S extends <a href="../../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../../../template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last even-row-color">
<div class="block">The default fasta header parser where some headers are well defined based on the source database
 which allows us to set the source of the protein sequence and the identifier that can be used in
 future implementations to load features from external sources
 
 If the user has a custom header with local data then they can create their own implementation of
 a FastaHeaderParserInterface

 GenBank gi|gi-number|gb|accession|locus ENA Data Library gi|gi-number|emb|accession|locus DDBJ,
 DNA Database of Japan gi|gi-number|dbj|accession|locus NBRF PIR pir||entry Protein Research
 Foundation prf||name SWISS-PROT sp|accession|name Brookhaven Protein Data Bank (1)
 pdb|entry|chain Brookhaven Protein Data Bank (2) entry:chain|PDBID|CHAIN|SEQUENCE PDB EBI
 PDB:1ECY_A mol:protein length:142 ECOTIN Patents pat|country|number GenInfo Backbone Id
 bbs|number General database identifier gnl|database|identifier NCBI Reference Sequence
 ref|accession|locus Local Sequence identifier lcl|identifier</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../../package-summary.html">org.biojava3.core.sequence.io</a> with parameters of type <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../FastaReader.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)" class="member-name-link">FastaReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html" title="class or interface in java.io" class="external-link">File</a>&nbsp;file,
 <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="../../FastaReader.html" title="type parameter in FastaReader">S</a>,<wbr><a href="../../FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;headerParser,
 <a href="../SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;sequenceCreator)</code></div>
<div class="col-last even-row-color">
<div class="block">If you are going to use the FileProxyProteinSequenceCreator then you need to use this constructor
 because we need details about the location of the file.</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../FastaReader.html#%3Cinit%3E(java.io.InputStream,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)" class="member-name-link">FastaReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/InputStream.html" title="class or interface in java.io" class="external-link">InputStream</a>&nbsp;is,
 <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="../../FastaReader.html" title="type parameter in FastaReader">S</a>,<wbr><a href="../../FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;headerParser,
 <a href="../SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;sequenceCreator)</code></div>
<div class="col-last odd-row-color">
<div class="block">If you are going to use FileProxyProteinSequenceCreator then do not use this constructor because
 we need details about local file offsets for quick reads.</div>
</div>
</div>
</section>
</li>
</ul>
</section>
</main>
</div>
</div>
</body>
</html>
