<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 18:10:05 CEST 2019 -->
<title>Uses of Interface org.biojava3.core.sequence.io.template.FastaHeaderParserInterface</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Interface org.biojava3.core.sequence.io.template.FastaHeaderParserInterface";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Interface org.biojava3.core.sequence.io.template.FastaHeaderParserInterface" class="title">Uses of Interface<br>org.biojava3.core.sequence.io.template.FastaHeaderParserInterface</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.reader">bacnet.reader</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.io">org.biojava3.core.sequence.io</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="bacnet.reader">
<!--   -->
</a>
<h3>Uses of <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a> in <a href="../../../../../../../bacnet/reader/package-summary.html">bacnet.reader</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../../../../../bacnet/reader/package-summary.html">bacnet.reader</a> that implement <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../../../bacnet/reader/NCBIFastaHeaderParser.html" title="class in bacnet.reader">NCBIFastaHeaderParser</a>&lt;S extends <a href="../../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../../../template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">My own implementation of a FastaHeaderParserInterface to parse Fasta files
 from NCBI ftp

 NCBI fna (genome and plasmids) gi|gi-number|ref|accession|name
 gi|16802048|ref|NC_003210.1| Listeria monocytogenes EGD-e, complete genome
 NCBI ffn (coding) ref|refGenome gi-number|:start-end name
 ref|NC_003210.1|:318-1673 chromosomal replication initiation protein
 [Listeria monocytogenes EGD-e] NCBI frn (non-coding) ref|refGenome
 gi-number|:start-end| name| [locus-tag] ref|NC_003210|:82705-82777|Lys tRNA|
 [locus_tag=lmot01] NCBI faa (proteins) gi|gi-number|ref|accession|name
 gi|16802049|ref|NP_463534.1| chromosomal replication initiation protein
 [Listeria monocytogenes EGD-e]</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../../../../../../bacnet/reader/package-summary.html">bacnet.reader</a> with parameters of type <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../../../bacnet/reader/FastaFileReader.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)">FastaFileReader</a></span>&#8203;(java.io.File&nbsp;file,
               <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="../../../../../../../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">S</a>,&#8203;<a href="../../../../../../../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;headerParser,
               <a href="../SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../../../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;sequenceCreator)</code></th>
<td class="colLast">
<div class="block">If you are going to use the FileProxyProteinSequenceCreator then you need to
 use this constructor because we need details about the location of the file.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../../../bacnet/reader/FastaFileReader.html#%3Cinit%3E(java.io.InputStream,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)">FastaFileReader</a></span>&#8203;(java.io.InputStream&nbsp;is,
               <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="../../../../../../../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">S</a>,&#8203;<a href="../../../../../../../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;headerParser,
               <a href="../SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../../../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;sequenceCreator)</code></th>
<td class="colLast">
<div class="block">If you are going to use FileProxyProteinSequenceCreator then do not use this
 constructor because we need details about local file offsets for quick reads.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.io">
<!--   -->
</a>
<h3>Uses of <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a> in <a href="../../package-summary.html">org.biojava3.core.sequence.io</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../package-summary.html">org.biojava3.core.sequence.io</a> that implement <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../GenericFastaHeaderParser.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser</a>&lt;S extends <a href="../../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../../../template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">The default fasta header parser where some headers are well defined based on
 the source database which allows us to set the source of the protein sequence
 and the identifier that can be used in future implementations to load
 features from external sources
 
 If the user has a custom header with local data then they can create their
 own implementation of a FastaHeaderParserInterface

 GenBank gi|gi-number|gb|accession|locus ENA Data Library
 gi|gi-number|emb|accession|locus DDBJ, DNA Database of Japan
 gi|gi-number|dbj|accession|locus NBRF PIR pir||entry Protein Research
 Foundation prf||name SWISS-PROT sp|accession|name Brookhaven Protein Data
 Bank (1) pdb|entry|chain Brookhaven Protein Data Bank (2)
 entry:chain|PDBID|CHAIN|SEQUENCE PDB EBI PDB:1ECY_A mol:protein length:142
 ECOTIN Patents pat|country|number GenInfo Backbone Id bbs|number General
 database identifier gnl|database|identifier NCBI Reference Sequence
 ref|accession|locus Local Sequence identifier lcl|identifier</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../package-summary.html">org.biojava3.core.sequence.io</a> with parameters of type <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../FastaReader.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)">FastaReader</a></span>&#8203;(java.io.File&nbsp;file,
           <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="../../FastaReader.html" title="type parameter in FastaReader">S</a>,&#8203;<a href="../../FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;headerParser,
           <a href="../SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;sequenceCreator)</code></th>
<td class="colLast">
<div class="block">If you are going to use the FileProxyProteinSequenceCreator then you need to
 use this constructor because we need details about the location of the file.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../FastaReader.html#%3Cinit%3E(java.io.InputStream,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)">FastaReader</a></span>&#8203;(java.io.InputStream&nbsp;is,
           <a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="../../FastaReader.html" title="type parameter in FastaReader">S</a>,&#8203;<a href="../../FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;headerParser,
           <a href="../SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;sequenceCreator)</code></th>
<td class="colLast">
<div class="block">If you are going to use FileProxyProteinSequenceCreator then do not use this
 constructor because we need details about local file offsets for quick reads.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
