<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 18:10:05 CEST 2019 -->
<title>Uses of Class bacnet.datamodel.expdesign.Experiment</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Class bacnet.datamodel.expdesign.Experiment";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Class bacnet.datamodel.expdesign.Experiment" class="title">Uses of Class<br>bacnet.datamodel.expdesign.Experiment</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet">bacnet</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.dataset">bacnet.datamodel.dataset</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.expdesign">bacnet.datamodel.expdesign</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.expressionAtlas.core">bacnet.expressionAtlas.core</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.core">bacnet.scripts.core</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.core.normalization">bacnet.scripts.core.normalization</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.database">bacnet.scripts.database</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.listeriomics">bacnet.scripts.listeriomics</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="bacnet">
<!--   -->
</a>
<h3>Uses of <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a> in <a href="../../../package-summary.html">bacnet</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../package-summary.html">bacnet</a> that return <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Database.</span><code><span class="memberNameLink"><a href="../../../Database.html#getGeneralExperiment()">getGeneralExperiment</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../package-summary.html">bacnet</a> with parameters of type <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Database.</span><code><span class="memberNameLink"><a href="../../../Database.html#setGeneralExperiment(bacnet.datamodel.expdesign.Experiment)">setGeneralExperiment</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;generalExperiment)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.dataset">
<!--   -->
</a>
<h3>Uses of <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a> in <a href="../../dataset/package-summary.html">bacnet.datamodel.dataset</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../dataset/package-summary.html">bacnet.datamodel.dataset</a> with parameters of type <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneExpression.</span><code><span class="memberNameLink"><a href="../../dataset/GeneExpression.html#convert(bacnet.datamodel.expdesign.Experiment)">convert</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;experiment)</code></th>
<td class="colLast">
<div class="block">Read all GeneExpr data and save in GeneExpressionStreaming format</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Tiling.</span><code><span class="memberNameLink"><a href="../../dataset/Tiling.html#convert(bacnet.datamodel.expdesign.Experiment)">convert</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;experiment)</code></th>
<td class="colLast">
<div class="block">Read Tiling data in Experiment and save in TilingDataStreaming format</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.expdesign">
<!--   -->
</a>
<h3>Uses of <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a> in <a href="../package-summary.html">bacnet.datamodel.expdesign</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">bacnet.datamodel.expdesign</a> that return <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Experiment.</span><code><span class="memberNameLink"><a href="../Experiment.html#getGeneralExp()">getGeneralExp</a></span>()</code></th>
<td class="colLast">
<div class="block">Load all experiment<br>
 If all experiments are already loaded just get the general experiment found
 in ModelProvider.INSTANCE</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.expressionAtlas.core">
<!--   -->
</a>
<h3>Uses of <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a> in <a href="../../../expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a> with parameters of type <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ComparisonAtlas.</span><code><span class="memberNameLink"><a href="../../../expressionAtlas/core/ComparisonAtlas.html#createAll(bacnet.datamodel.expdesign.Experiment)">createAll</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</code></th>
<td class="colLast">
<div class="block">For each comparison create a <code>Comparison</code> object containing the
 list of all differentially expressed genes, sRNAs, asRNAs, and cisRegs<br>
 <br>
 
 WANRING: OrganizeDatabase.organizeComparisonsData(exp) as to be run first!!!</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeElementAtlas.</span><code><span class="memberNameLink"><a href="../../../expressionAtlas/core/GenomeElementAtlas.html#createAll(bacnet.datamodel.expdesign.Experiment)">createAll</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;expTemp)</code></th>
<td class="colLast">
<div class="block">Read all <code>Comparison</code> results and extract which gene is over and
 under expressed in it<br>
 For each gene, sRNA,asRNA, and cisReg, create a list of all
 <code>BioCondition</code> where it is over, under, and not differentially
 expressed<br>
 <br>
 
 WANRING: OrganizeDatabase.organizeComparisonsData(exp) as to be run first!!!</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.core">
<!--   -->
</a>
<h3>Uses of <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a> in <a href="../../../scripts/core/package-summary.html">bacnet.scripts.core</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/core/package-summary.html">bacnet.scripts.core</a> with parameters of type <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Expression.</span><code><span class="memberNameLink"><a href="../../../scripts/core/Expression.html#summarize(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">summarize</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
         <a href="../../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create 3 ExpressionMatrix: <br>
 GeneExpression of genes on all BioCondition, saved in PATH_ALLGENEXPR
 Tiling of genes on all BioCondition, saved in PATH_ALLTILING_GENE
 Tiling of Srnas on all BioCondition, saved in PATH_ALLTILING_SRNA</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.core.normalization">
<!--   -->
</a>
<h3>Uses of <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a> in <a href="../../../scripts/core/normalization/package-summary.html">bacnet.scripts.core.normalization</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/core/normalization/package-summary.html">bacnet.scripts.core.normalization</a> with parameters of type <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GENormalization.</span><code><span class="memberNameLink"><a href="../../../scripts/core/normalization/GENormalization.html#extractInformation(bacnet.datamodel.expdesign.Experiment)">extractInformation</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;experiment)</code></th>
<td class="colLast">
<div class="block">Extract information contained in all .ge files of the experiment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GENormalization.</span><code><span class="memberNameLink"><a href="../../../scripts/core/normalization/GENormalization.html#norm(bacnet.datamodel.expdesign.Experiment)">norm</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GRNormalization.</span><code><span class="memberNameLink"><a href="../../../scripts/core/normalization/GRNormalization.html#norm(bacnet.datamodel.expdesign.Experiment)">norm</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</code></th>
<td class="colLast">
<div class="block">Normalize the Tiling contained in the Experiment <br>
 Run one thread for each bioCondition<br>
 The number of thread is limited by the number of processor<br>
 <br>
 
 It runs DNA reference normalization (Hubert et al.), it can uses also an
 additional reference data for normalization (indicates by boolean
 normalization).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MedianNormalization.</span><code><span class="memberNameLink"><a href="../../../scripts/core/normalization/MedianNormalization.html#normGE(bacnet.datamodel.expdesign.Experiment)">normGE</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MedianNormalization.</span><code><span class="memberNameLink"><a href="../../../scripts/core/normalization/MedianNormalization.html#normTiling(bacnet.datamodel.expdesign.Experiment)">normTiling</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</code></th>
<td class="colLast">
<div class="block">Load all TilingData in this Experiment and normalize by the median, and save
 in the Tiling/MedianNorm path</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GRNormalization.</span><code><span class="memberNameLink"><a href="../../../scripts/core/normalization/GRNormalization.html#qualityControlNorm(bacnet.datamodel.expdesign.Experiment)">qualityControlNorm</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</code></th>
<td class="colLast">
<div class="block">Check temp folder, and GeneExprComplete folder to see if some GeneExpression
 are not present in GeneExprComplete</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.database">
<!--   -->
</a>
<h3>Uses of <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a> in <a href="../../../scripts/database/package-summary.html">bacnet.scripts.database</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/database/package-summary.html">bacnet.scripts.database</a> with parameters of type <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteomicsCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/ProteomicsCreation.html#addMissingValuesToMatrices(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">addMissingValuesToMatrices</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                          <a href="../../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">In order to represent proteomics data we need to add a 0 value in every
 missing row.<br></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NGSCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/NGSCreation.html#convertCoverageFiles(bacnet.datamodel.expdesign.Experiment,boolean)">convertCoverageFiles</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                    boolean&nbsp;logTransformed)</code></th>
<td class="colLast">
<div class="block">Go through all RNASeq data found in the list
 <code>BioCondition.getRNASeqBioConditions()</code> and convert all of them</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteomicsCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/ProteomicsCreation.html#convertProteomicsData(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">convertProteomicsData</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                     <a href="../../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Convert all ExpressionMatrix found for Proteomic datasets and save them in
 Streaming folder</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ComparisonsCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/ComparisonsCreation.html#createLogFCMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">createLogFCMatrix</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                 <a href="../../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create Database.logFCMatrix by regrouping every logFC values for genes,
 sRNAs, asRNAs, and cisRegs in every <code>BioCondition</code> given
 <code>Experiment</code><br></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteomicsCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/ProteomicsCreation.html#createLogFCMatrix(bacnet.datamodel.expdesign.Experiment,java.lang.String)">createLogFCMatrix</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                 java.lang.String&nbsp;genomeName)</code></th>
<td class="colLast">
<div class="block">Create a table will all protein expression for each Genome</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptomesCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/TranscriptomesCreation.html#createLogFCMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">createLogFCMatrix</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                 <a href="../../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create Database.logFCMatrix by regrouping every logFC values for genes,
 sRNAs, asRNAs, and cisRegs in every <code>BioCondition</code> given
 <code>Experiment</code><br></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ComparisonsCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/ComparisonsCreation.html#createStatMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">createStatMatrix</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                <a href="../../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create Database.statTable by calculating stat values for genes, sRNAs, and
 asRNAs in every BioCondition<br>
 For genes we use FDRBY calculated from GebneExpression array<br>
 For sRNAs and asRNAs we use TSTUDENTTILING for Tiling array</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptomesCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/TranscriptomesCreation.html#createStatMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">createStatMatrix</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                <a href="../../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create Database.statTable by calculating stat values for genes, sRNAs, and
 asRNAs in every BioCondition<br>
 For genes we use FDRBY calculated from GebneExpression array<br>
 For sRNAs and asRNAs we use TSTUDENTTILING for Tiling array</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NGSCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/NGSCreation.html#normalizeCountFiles(bacnet.datamodel.expdesign.Experiment)">normalizeCountFiles</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</code></th>
<td class="colLast">
<div class="block">ConvertCountFile and add them to the database</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ComparisonsCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/ComparisonsCreation.html#summarizeDiffExpr(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">summarizeDiffExpr</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                 <a href="../../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create an array :<br>
 Column = comparison
 Row = Gene, Srna or Asrna
 Cell = "+" if a specific gene has been detected over-expressed, and "-"
 if under-expressed<br>
 <br>
 
 Save it in : Comparison.PATH_DIFF_LISTS+".txt"</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ComparisonsCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/ComparisonsCreation.html#summarizeDiffLists(bacnet.datamodel.expdesign.Experiment)">summarizeDiffLists</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</code></th>
<td class="colLast">
<div class="block">Create an array :<br>
 Row = comparison
 Column = Number of gene, Srna Asrna over or under-expressed<br>
 <br>
 Save it in : Comparison.PATH_DIFF_LISTS+".txt"</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteomicsCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/ProteomicsCreation.html#summarizeProteomes(bacnet.datamodel.expdesign.Experiment,java.lang.String)">summarizeProteomes</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                  java.lang.String&nbsp;genomeName)</code></th>
<td class="colLast">
<div class="block">Create a table will all protein expression for each Genome</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptomesCreation.</span><code><span class="memberNameLink"><a href="../../../scripts/database/TranscriptomesCreation.html#varianceNormalization(bacnet.datamodel.expdesign.Experiment)">varianceNormalization</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</code></th>
<td class="colLast">
<div class="block">Run some scripts to analyse the statistics of the transcriptomics data and
 performed VarianceNormalization:<br>
 It will reduce the total variance of each data to 1<br>
 For each column of the ExpressionMatrix, multiply each value by:
 Math.sqr(1/var(x))<br>
 So final variance of each column is equal to zero</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.scripts.listeriomics">
<!--   -->
</a>
<h3>Uses of <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a> in <a href="../../../scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a> with parameters of type <a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TilingGeneExprDataBase.</span><code><span class="memberNameLink"><a href="../../../scripts/listeriomics/TilingGeneExprDataBase.html#calcGenExprTilingComparisons(bacnet.datamodel.expdesign.Experiment)">calcGenExprTilingComparisons</a></span>&#8203;(<a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</code></th>
<td class="colLast">
<div class="block">Calculate GeneExpr and Tiling comparisons files</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Experiment.html" title="class in bacnet.datamodel.expdesign">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
