<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Fri Dec 10 09:15:34 CET 2021 -->
<title>Uses of Class bacnet.datamodel.sequenceNCBI.GenomeNCBI</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-10">
<meta name="description" content="use: package: bacnet.datamodel.sequenceNCBI, class: GenomeNCBI">
<meta name="generator" content="javadoc/ClassUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-use-page">
<script type="text/javascript">var pathtoroot = "../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">Class</a></li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Class bacnet.datamodel.sequenceNCBI.GenomeNCBI" class="title">Uses of Class<br>bacnet.datamodel.sequenceNCBI.GenomeNCBI</h1>
</div>
<div class="caption"><span>Packages that use <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#bacnet">bacnet</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.datamodel.annotation">bacnet.datamodel.annotation</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.datamodel.sequence">bacnet.datamodel.sequence</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.datamodel.sequenceNCBI">bacnet.datamodel.sequenceNCBI</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.scripts.genome">bacnet.scripts.genome</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.scripts.listeriomics.nterm">bacnet.scripts.listeriomics.nterm</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<section class="class-uses">
<ul class="block-list">
<li>
<section class="detail" id="bacnet">
<h2>Uses of <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a> in <a href="../../../package-summary.html">bacnet</a></h2>
<div class="caption"><span>Methods in <a href="../../../package-summary.html">bacnet</a> that return types with arguments of type <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/TreeMap.html" title="class or interface in java.util" class="external-link">TreeMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Database.</span><code><a href="../../../Database.html#getGenomesNCBI()" class="member-name-link">getGenomesNCBI</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Method parameters in <a href="../../../package-summary.html">bacnet</a> with type arguments of type <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Database.</span><code><a href="../../../Database.html#setGenomesNCBI(java.util.TreeMap)" class="member-name-link">setGenomesNCBI</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/TreeMap.html" title="class or interface in java.util" class="external-link">TreeMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&gt;&nbsp;genomesNCBI)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.datamodel.annotation">
<h2>Uses of <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a> in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a></h2>
<div class="caption"><span>Methods in <a href="../../annotation/package-summary.html">bacnet.datamodel.annotation</a> with parameters of type <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/HashMap.html" title="class or interface in java.util" class="external-link">HashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">COGannotation.</span><code><a href="../../annotation/COGannotation.html#getCogClassification(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequenceNCBI.GenomeNCBI)" class="member-name-link">getCogClassification</a><wbr>(<a href="../../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;matrix,
 <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.datamodel.sequence">
<h2>Uses of <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a> in <a href="../../sequence/package-summary.html">bacnet.datamodel.sequence</a></h2>
<div class="caption"><span>Methods in <a href="../../sequence/package-summary.html">bacnet.datamodel.sequence</a> that return <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">ChromosomeBacteriaSequence.</span><code><a href="../../sequence/ChromosomeBacteriaSequence.html#getParentGenome()" class="member-name-link">getParentGenome</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Get the gene based on accession.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../sequence/package-summary.html">bacnet.datamodel.sequence</a> with parameters of type <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../../sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Operon.</span><code><a href="../../sequence/Operon.html#convertDNASequenceOperon(java.lang.String%5B%5D,bacnet.datamodel.sequenceNCBI.GenomeNCBI)" class="member-name-link">convertDNASequenceOperon</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[]&nbsp;row,
 <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">
<div class="block">Take row from Annot_sup table and creat an Operon Object from it</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">ChromosomeBacteriaSequence.</span><code><a href="../../sequence/ChromosomeBacteriaSequence.html#setParentGenome(bacnet.datamodel.sequenceNCBI.GenomeNCBI)" class="member-name-link">setParentGenome</a><wbr>(<a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;parentGenome)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.datamodel.sequenceNCBI">
<h2>Uses of <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a> in <a href="../package-summary.html">bacnet.datamodel.sequenceNCBI</a></h2>
<div class="caption"><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequenceNCBI</a> that return <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">GenomeNCBITools.</span><code><a href="../GenomeNCBITools.html#loadEgdeGenome()" class="member-name-link">loadEgdeGenome</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Find in ModelProvider if Egde Genome is already loaded If not search EGDE_NAME in GENOMEDIR</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GenomeNCBITools.</span><code><a href="../GenomeNCBITools.html#loadGenome(java.lang.String)" class="member-name-link">loadGenome</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeAccession)</code></div>
<div class="col-last odd-row-color">
<div class="block">Find in ModelProvider if Genome Accession is already loaded If not search load it</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">GenomeNCBITools.</span><code><a href="../GenomeNCBITools.html#loadGenome(java.lang.String,java.lang.String,boolean,boolean)" class="member-name-link">loadGenome</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeAccession,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;path,
 boolean&nbsp;keepInMemory,
 boolean&nbsp;annotation)</code></div>
<div class="col-last even-row-color">
<div class="block">Load/return a genome given by a genomeAccession ID<br>
 <br>
 
 If keepInMemory is true : Find in ModelProvider if Genome Accession is already loaded if not load
 it <br>
 If keepInMemory is false : Only load the Genome without saving it <br>
 <br>
 If annotation is true : Load annotation information of the genome <br>
 If anotation is false : Only load the sequence of the genome</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">bacnet.datamodel.sequenceNCBI</a> with parameters of type <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../../sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">GenomeConversionElement.</span><code><a href="../GenomeConversionElement.html#convertDNASequenceOperon(java.lang.String%5B%5D,bacnet.datamodel.sequenceNCBI.GenomeNCBI)" class="member-name-link">convertDNASequenceOperon</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[]&nbsp;row,
 <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genome)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GenomeNCBITools.</span><code><a href="../GenomeNCBITools.html#curateGenomeFindingDoublonGeneName(bacnet.datamodel.sequenceNCBI.GenomeNCBI)" class="member-name-link">curateGenomeFindingDoublonGeneName</a><wbr>(<a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color">
<div class="block">Check if two different locustag are not associated to the same GeneName</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">GenomeConversion.</span><code><a href="../GenomeConversion.html#run(bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String,java.lang.String)" class="member-name-link">run</a><wbr>(<a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;pathGenome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName)</code></div>
<div class="col-last even-row-color">
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GenomeConversion.</span><code><a href="../GenomeConversion.html#run(bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">run</a><wbr>(<a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;pathGenome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomePath)</code></div>
<div class="col-last odd-row-color">
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.scripts.genome">
<h2>Uses of <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a> in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a></h2>
<div class="caption"><span>Constructors in <a href="../../../scripts/genome/package-summary.html">bacnet.scripts.genome</a> with parameters of type <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../../scripts/genome/CircularGenomeJPanel.html#%3Cinit%3E(int,int,bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String)" class="member-name-link">CircularGenomeJPanel</a><wbr>(int&nbsp;width,
 int&nbsp;height,
 <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;title)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.scripts.listeriomics.nterm">
<h2>Uses of <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a> in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></h2>
<div class="caption"><span>Methods in <a href="../../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a> that return <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">RASToverlap.</span><code><a href="../../../scripts/listeriomics/nterm/RASToverlap.html#getEGDe()" class="member-name-link">getEGDe</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Read RAST automoticaly annotated EGD-e genome</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">RASToverlap.</span><code><a href="../../../scripts/listeriomics/nterm/RASToverlap.html#getGenome(java.lang.String)" class="member-name-link">getGenome</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName)</code></div>
<div class="col-last odd-row-color">
<div class="block">Read a genome from RAST results</div>
</div>
</div>
</section>
</li>
</ul>
</section>
</main>
</div>
</div>
</body>
</html>
