<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>AbstractSequence</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="declaration: package: org.biojava3.core.sequence.template, class: AbstractSequence">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var evenRowColor = "even-row-color";
var oddRowColor = "odd-row-color";
var tableTab = "table-tab";
var activeTableTab = "active-table-tab";
var pathtoroot = "../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/AbstractSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../help-doc.html#class">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div>
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li><a href="#nested-class-summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">org.biojava3.core.sequence.template</a></div>
<h1 title="Class AbstractSequence" class="title">Class AbstractSequence&lt;C extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</h1>
</div>
<div class="inheritance" title="Inheritance Tree"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">java.lang.Object</a>
<div class="inheritance">org.biojava3.core.sequence.template.AbstractSequence&lt;C&gt;</div>
</div>
<section class="class-description" id="class-description">
<dl class="notes">
<dt>Type Parameters:</dt>
<dd><code>C</code> - </dd>
</dl>
<dl class="notes">
<dt>All Implemented Interfaces:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html" title="class or interface in java.lang" class="external-link">Iterable</a>&lt;C&gt;</code>, <code><a href="Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</a></code>, <code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;</code></dd>
</dl>
<dl class="notes">
<dt>Direct Known Subclasses:</dt>
<dd><code><a href="../BasicSequence.html" title="class in org.biojava3.core.sequence">BasicSequence</a></code>, <code><a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code>, <code><a href="../ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></code>, <code><a href="../RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></code></dd>
</dl>
<hr>
<div class="type-signature"><span class="modifiers">public abstract class </span><span class="element-name type-name-label">AbstractSequence&lt;C extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span>
<span class="extends-implements">extends <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>
implements <a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;</span></div>
<div class="block">The base class for DNA, RNA and Protein sequences.</div>
</section>
<section class="summary">
<ul class="summary-list">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<li>
<section class="nested-class-summary" id="nested-class-summary">
<h2>Nested Class Summary</h2>
<div class="caption"><span>Nested Classes</span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static enum&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="AbstractSequence.AnnotationType.html" class="type-name-link" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<li>
<section class="constructor-summary" id="constructor-summary">
<h2>Constructor Summary</h2>
<div class="caption"><span>Constructors</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E()" class="member-name-link">AbstractSequence</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-constructor-name odd-row-color"><code><a href="#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">AbstractSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create a Sequence from a simple string where the values should be found in compoundSet</div>
</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">AbstractSequence</a><wbr>(<a href="SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;proxyLoader,
 <a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">A ProxySequenceReader allows abstraction of both the storage of the sequence data and the
 location of the sequence data.</div>
</div>
</div>
</section>
</li>
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab2" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab2', 3)" class="table-tab">Instance Methods</button><button id="method-summary-table-tab4" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab4', 3)" class="table-tab">Concrete Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel">
<div class="summary-table three-column-summary" aria-labelledby="method-summary-table-tab0">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#addFeature(int,int,org.biojava3.core.sequence.features.FeatureInterface)" class="member-name-link">addFeature</a><wbr>(int&nbsp;bioStart,
 int&nbsp;bioEnd,
 <a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Method to help set the proper details for a feature as it relates to a sequence where the feature
 needs to have a location on the sequence</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#addFeature(org.biojava3.core.sequence.features.FeatureInterface)" class="member-name-link">addFeature</a><wbr>(<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Add a feature to this sequence.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#addNote(java.lang.String)" class="member-name-link">addNote</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;note)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Add notes about this sequence that will get exported for GFF3</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>int</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#countCompounds(C...)" class="member-name-link">countCompounds</a><wbr>(<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>...&nbsp;compounds)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Returns the number of times we found a compound in the Sequence</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="../AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getAccession()" class="member-name-link">getAccession</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Returns the AccessionID this location is currently bound with</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getAnnotationType()" class="member-name-link">getAnnotationType</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getAsList()" class="member-name-link">getAsList</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Returns the Sequence as a List of compounds</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getBioBegin()" class="member-name-link">getBioBegin</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getBioEnd()" class="member-name-link">getBioEnd</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getCompoundAt(int)" class="member-name-link">getCompoundAt</a><wbr>(int&nbsp;position)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Returns the Compound at the given biological index</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Gets the compound set used to back this Sequence</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="../features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getDatabaseReferences()" class="member-name-link">getDatabaseReferences</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getDescription()" class="member-name-link">getDescription</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getFeatures()" class="member-name-link">getFeatures</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getFeatures(int)" class="member-name-link">getFeatures</a><wbr>(int&nbsp;bioSequencePosition)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Return features at a sequence position</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getFeatures(java.lang.String,int)" class="member-name-link">getFeatures</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;featureType,
 int&nbsp;bioSequencePosition)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Return features at a sequence position by type</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getFeaturesByType(java.lang.String)" class="member-name-link">getFeaturesByType</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;type)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="../features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getFeaturesKeyWord()" class="member-name-link">getFeaturesKeyWord</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>int</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getIndexOf(C)" class="member-name-link">getIndexOf</a><wbr>(<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&nbsp;compound)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Scans through the Sequence looking for the first occurrence of the given compound</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getInverse()" class="member-name-link">getInverse</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Does the <em>right thing</em> to get the inverse of the current Sequence.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>int</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getLastIndexOf(C)" class="member-name-link">getLastIndexOf</a><wbr>(<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&nbsp;compound)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Scans through the Sequence looking for the last occurrence of the given compound</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>int</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getLength()" class="member-name-link">getLength</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Returns the length of the Sequence</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getNotesList()" class="member-name-link">getNotesList</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getOriginalHeader()" class="member-name-link">getOriginalHeader</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getParentSequence()" class="member-name-link">getParentSequence</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getProxySequenceReader()" class="member-name-link">getProxySequenceReader</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getSequenceAsString()" class="member-name-link">getSequenceAsString</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Default case is to assume strand is positive because only CDSSequence can be either positive or
 negative Strand.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)" class="member-name-link">getSequenceAsString</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;bioStart,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;bioEnd,
 <a href="../Strand.html" title="enum in org.biojava3.core.sequence">Strand</a>&nbsp;strand)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Double.html" title="class or interface in java.lang" class="external-link">Double</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getSequenceScore()" class="member-name-link">getSequenceScore</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Provide place holder for a metric that indicate a score associated with the sequence</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getSource()" class="member-name-link">getSource</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Added support for the source of this sequence for GFF3 export If a sub sequence doesn't have
 source then check for parent source</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getSubSequence(java.lang.Integer,java.lang.Integer)" class="member-name-link">getSubSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;bioStart,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;bioEnd)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Returns a portion of the sequence from the different positions.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="../TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getTaxonomy()" class="member-name-link">getTaxonomy</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Collection.html" title="class or interface in java.util" class="external-link">Collection</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getUserCollection()" class="member-name-link">getUserCollection</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Provided for convience if the developer needs to associate data with a sequence</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Iterator.html" title="class or interface in java.util" class="external-link">Iterator</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#iterator()" class="member-name-link">iterator</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#removeFeature(org.biojava3.core.sequence.features.FeatureInterface)" class="member-name-link">removeFeature</a><wbr>(<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Remove a feature from the sequence</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#removeNote(java.lang.String)" class="member-name-link">removeNote</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;note)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setAccession(org.biojava3.core.sequence.AccessionID)" class="member-name-link">setAccession</a><wbr>(<a href="../AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)" class="member-name-link">setAnnotationType</a><wbr>(<a href="AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a>&nbsp;annotationType)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setBioBegin(java.lang.Integer)" class="member-name-link">setBioBegin</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;begin)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setBioEnd(java.lang.Integer)" class="member-name-link">setBioEnd</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;end)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet</a><wbr>(<a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)" class="member-name-link">setDatabaseReferences</a><wbr>(<a href="../features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</a>&nbsp;databaseReferences)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setDescription(java.lang.String)" class="member-name-link">setDescription</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;description)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)" class="member-name-link">setFeaturesKeyWord</a><wbr>(<a href="../features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</a>&nbsp;featuresKeyWord)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setNotesList(java.util.ArrayList)" class="member-name-link">setNotesList</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;&nbsp;notesList)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setOriginalHeader(java.lang.String)" class="member-name-link">setOriginalHeader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;originalHeader)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)" class="member-name-link">setParentSequence</a><wbr>(<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;abstractSequence)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)" class="member-name-link">setProxySequenceReader</a><wbr>(<a href="SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;proxyLoader)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Very important method that allows external mappings of sequence data and features.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setSequenceScore(java.lang.Double)" class="member-name-link">setSequenceScore</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Double.html" title="class or interface in java.lang" class="external-link">Double</a>&nbsp;sequenceScore)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setSource(java.lang.String)" class="member-name-link">setSource</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;source)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Added support for the source of this sequence for GFF3 export</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setTaxonomy(org.biojava3.core.sequence.TaxonomyID)" class="member-name-link">setTaxonomy</a><wbr>(<a href="../TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</a>&nbsp;taxonomy)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setUserCollection(java.util.Collection)" class="member-name-link">setUserCollection</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Collection.html" title="class or interface in java.util" class="external-link">Collection</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&gt;&nbsp;userCollection)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#toString()" class="member-name-link">toString</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
</div>
</div>
</div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Object">Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)" title="class or interface in java.lang" class="external-link">equals</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#getClass()" title="class or interface in java.lang" class="external-link">getClass</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode()" title="class or interface in java.lang" class="external-link">hashCode</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notify()" title="class or interface in java.lang" class="external-link">notify</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notifyAll()" title="class or interface in java.lang" class="external-link">notifyAll</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait()" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long)" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long,int)" title="class or interface in java.lang" class="external-link">wait</a></code></div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Iterable">Methods inherited from interface&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html" title="class or interface in java.lang" class="external-link">Iterable</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html#forEach(java.util.function.Consumer)" title="class or interface in java.lang" class="external-link">forEach</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html#spliterator()" title="class or interface in java.lang" class="external-link">spliterator</a></code></div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<li>
<section class="constructor-details" id="constructor-detail">
<h2>Constructor Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="&lt;init&gt;()">
<h3>AbstractSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name">AbstractSequence</span>()</div>
</section>
</li>
<li>
<section class="detail" id="&lt;init&gt;(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">
<h3>AbstractSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name">AbstractSequence</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</span></div>
<div class="block">Create a Sequence from a simple string where the values should be found in compoundSet</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>seqString</code> - </dd>
<dd><code>compoundSet</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="&lt;init&gt;(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">
<h3>AbstractSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name">AbstractSequence</span><wbr><span class="parameters">(<a href="SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;proxyLoader,
 <a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</span></div>
<div class="block">A ProxySequenceReader allows abstraction of both the storage of the sequence data and the
 location of the sequence data. A variety of use cases are possible. A ProxySequenceReader that
 knows the offset and of teh sequence in a large fasta file. A ProxySequenceReader that can pull
 Sequence data from Uniprot, NCBI or a custom database. If the ProxySequecneReader implements
 various interfaces then the sequence will set those interfaces so that calls to various methods
 will be valid.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>proxyLoader</code> - </dd>
<dd><code>compoundSet</code> - </dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)">
<h3>setProxySequenceReader</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setProxySequenceReader</span><wbr><span class="parameters">(<a href="SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;proxyLoader)</span></div>
<div class="block">Very important method that allows external mappings of sequence data and features. This method
 will gain additional interface inspection that allows external data sources with knowledge of
 features for a sequence to be supported.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>proxyLoader</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getProxySequenceReader()">
<h3>getProxySequenceReader</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</span>&nbsp;<span class="element-name">getProxySequenceReader</span>()</div>
</section>
</li>
<li>
<section class="detail" id="getBioBegin()">
<h3>getBioBegin</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a></span>&nbsp;<span class="element-name">getBioBegin</span>()</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the bioBegin</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setBioBegin(java.lang.Integer)">
<h3>setBioBegin</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setBioBegin</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;begin)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>bioBegin</code> - the bioBegin to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getBioEnd()">
<h3>getBioEnd</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a></span>&nbsp;<span class="element-name">getBioEnd</span>()</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the bioEnd</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setBioEnd(java.lang.Integer)">
<h3>setBioEnd</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setBioEnd</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;end)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>bioEnd</code> - the bioEnd to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getUserCollection()">
<h3>getUserCollection</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Collection.html" title="class or interface in java.util" class="external-link">Collection</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&gt;</span>&nbsp;<span class="element-name">getUserCollection</span>()</div>
<div class="block">Provided for convience if the developer needs to associate data with a sequence</div>
<dl class="notes">
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="setUserCollection(java.util.Collection)">
<h3>setUserCollection</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setUserCollection</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Collection.html" title="class or interface in java.util" class="external-link">Collection</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&gt;&nbsp;userCollection)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>userCollection</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAnnotationType()">
<h3>getAnnotationType</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></span>&nbsp;<span class="element-name">getAnnotationType</span>()</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the annotation</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)">
<h3>setAnnotationType</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setAnnotationType</span><wbr><span class="parameters">(<a href="AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a>&nbsp;annotationType)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>annotation</code> - the annotation to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getDescription()">
<h3>getDescription</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getDescription</span>()</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the description</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setDescription(java.lang.String)">
<h3>setDescription</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setDescription</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;description)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>description</code> - the description to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getOriginalHeader()">
<h3>getOriginalHeader</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getOriginalHeader</span>()</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the originalHeader</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setOriginalHeader(java.lang.String)">
<h3>setOriginalHeader</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setOriginalHeader</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;originalHeader)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>originalHeader</code> - the originalHeader to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getParentSequence()">
<h3>getParentSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</span>&nbsp;<span class="element-name">getParentSequence</span>()</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the parentSequence</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)">
<h3>setParentSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setParentSequence</span><wbr><span class="parameters">(<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;abstractSequence)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>abstractSequence</code> - the parentSequence to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getSource()">
<h3>getSource</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getSource</span>()</div>
<div class="block">Added support for the source of this sequence for GFF3 export If a sub sequence doesn't have
 source then check for parent source</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the source</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setSource(java.lang.String)">
<h3>setSource</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setSource</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;source)</span></div>
<div class="block">Added support for the source of this sequence for GFF3 export</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>source</code> - the source to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="addNote(java.lang.String)">
<h3>addNote</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">addNote</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;note)</span></div>
<div class="block">Add notes about this sequence that will get exported for GFF3</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>note</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="removeNote(java.lang.String)">
<h3>removeNote</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">removeNote</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;note)</span></div>
</section>
</li>
<li>
<section class="detail" id="getNotesList()">
<h3>getNotesList</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</span>&nbsp;<span class="element-name">getNotesList</span>()</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the notesList</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setNotesList(java.util.ArrayList)">
<h3>setNotesList</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setNotesList</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;&nbsp;notesList)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>notesList</code> - the notesList to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getSequenceScore()">
<h3>getSequenceScore</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Double.html" title="class or interface in java.lang" class="external-link">Double</a></span>&nbsp;<span class="element-name">getSequenceScore</span>()</div>
<div class="block">Provide place holder for a metric that indicate a score associated with the sequence</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the sequenceScore</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setSequenceScore(java.lang.Double)">
<h3>setSequenceScore</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setSequenceScore</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Double.html" title="class or interface in java.lang" class="external-link">Double</a>&nbsp;sequenceScore)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>sequenceScore</code> - the sequenceScore to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getFeatures(java.lang.String,int)">
<h3>getFeatures</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</span>&nbsp;<span class="element-name">getFeatures</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;featureType,
 int&nbsp;bioSequencePosition)</span></div>
<div class="block">Return features at a sequence position by type</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>featureType</code> - </dd>
<dd><code>bioSequencePosition</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getFeatures(int)">
<h3>getFeatures</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</span>&nbsp;<span class="element-name">getFeatures</span><wbr><span class="parameters">(int&nbsp;bioSequencePosition)</span></div>
<div class="block">Return features at a sequence position</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>featureType</code> - </dd>
<dd><code>bioSequencePosition</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getFeatures()">
<h3>getFeatures</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</span>&nbsp;<span class="element-name">getFeatures</span>()</div>
<dl class="notes">
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="addFeature(int,int,org.biojava3.core.sequence.features.FeatureInterface)">
<h3>addFeature</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">addFeature</span><wbr><span class="parameters">(int&nbsp;bioStart,
 int&nbsp;bioEnd,
 <a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</span></div>
<div class="block">Method to help set the proper details for a feature as it relates to a sequence where the feature
 needs to have a location on the sequence</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>bioStart</code> - </dd>
<dd><code>bioEnd</code> - </dd>
<dd><code>feature</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="addFeature(org.biojava3.core.sequence.features.FeatureInterface)">
<h3>addFeature</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">addFeature</span><wbr><span class="parameters">(<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</span></div>
<div class="block">Add a feature to this sequence. The feature will be added to the collection where the order is
 start position and if more than one feature at the same start position then longest is added
 first. This helps on doing feature layout for displaying features in SequenceFeaturePanel</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>feature</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="removeFeature(org.biojava3.core.sequence.features.FeatureInterface)">
<h3>removeFeature</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">removeFeature</span><wbr><span class="parameters">(<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</span></div>
<div class="block">Remove a feature from the sequence</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>feature</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getFeaturesByType(java.lang.String)">
<h3>getFeaturesByType</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</span>&nbsp;<span class="element-name">getFeaturesByType</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;type)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>type</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getFeaturesKeyWord()">
<h3>getFeaturesKeyWord</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="../features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</a></span>&nbsp;<span class="element-name">getFeaturesKeyWord</span>()</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the featuresKeyWord</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)">
<h3>setFeaturesKeyWord</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setFeaturesKeyWord</span><wbr><span class="parameters">(<a href="../features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</a>&nbsp;featuresKeyWord)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>featuresKeyWord</code> - the featuresKeyWord to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getDatabaseReferences()">
<h3>getDatabaseReferences</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="../features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</a></span>&nbsp;<span class="element-name">getDatabaseReferences</span>()</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the databaseReferences</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)">
<h3>setDatabaseReferences</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setDatabaseReferences</span><wbr><span class="parameters">(<a href="../features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</a>&nbsp;databaseReferences)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>databaseReferences</code> - the databaseReferences to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAccession()">
<h3>getAccession</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="../AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a></span>&nbsp;<span class="element-name">getAccession</span>()</div>
<div class="block"><span class="descfrm-type-label">Description copied from interface:&nbsp;<code><a href="Accessioned.html#getAccession()">Accessioned</a></code></span></div>
<div class="block">Returns the AccessionID this location is currently bound with</div>
<dl class="notes">
<dt>Specified by:</dt>
<dd><code><a href="Accessioned.html#getAccession()">getAccession</a></code>&nbsp;in interface&nbsp;<code><a href="Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</a></code></dd>
<dt>Returns:</dt>
<dd>the accession</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setAccession(org.biojava3.core.sequence.AccessionID)">
<h3>setAccession</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setAccession</span><wbr><span class="parameters">(<a href="../AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>accession</code> - the accession to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getTaxonomy()">
<h3>getTaxonomy</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="../TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</a></span>&nbsp;<span class="element-name">getTaxonomy</span>()</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the species</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setTaxonomy(org.biojava3.core.sequence.TaxonomyID)">
<h3>setTaxonomy</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setTaxonomy</span><wbr><span class="parameters">(<a href="../TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</a>&nbsp;taxonomy)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>species</code> - the species to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getCompoundSet()">
<h3>getCompoundSet</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</span>&nbsp;<span class="element-name">getCompoundSet</span>()</div>
<div class="block"><span class="descfrm-type-label">Description copied from interface:&nbsp;<code><a href="Sequence.html#getCompoundSet()">Sequence</a></code></span></div>
<div class="block">Gets the compound set used to back this Sequence</div>
<dl class="notes">
<dt>Specified by:</dt>
<dd><code><a href="Sequence.html#getCompoundSet()">getCompoundSet</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">
<h3>setCompoundSet</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setCompoundSet</span><wbr><span class="parameters">(<a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</span></div>
</section>
</li>
<li>
<section class="detail" id="toString()">
<h3>toString</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">toString</span>()</div>
<dl class="notes">
<dt>Overrides:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#toString()" title="class or interface in java.lang" class="external-link">toString</a></code>&nbsp;in class&nbsp;<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)">
<h3>getSequenceAsString</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getSequenceAsString</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;bioStart,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;bioEnd,
 <a href="../Strand.html" title="enum in org.biojava3.core.sequence">Strand</a>&nbsp;strand)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>begin</code> - </dd>
<dd><code>end</code> - </dd>
<dd><code>strand</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getSequenceAsString()">
<h3>getSequenceAsString</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getSequenceAsString</span>()</div>
<div class="block">Default case is to assume strand is positive because only CDSSequence can be either positive or
 negative Strand.</div>
<dl class="notes">
<dt>Specified by:</dt>
<dd><code><a href="Sequence.html#getSequenceAsString()">getSequenceAsString</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAsList()">
<h3>getAsList</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</span>&nbsp;<span class="element-name">getAsList</span>()</div>
<div class="block"><span class="descfrm-type-label">Description copied from interface:&nbsp;<code><a href="Sequence.html#getAsList()">Sequence</a></code></span></div>
<div class="block">Returns the Sequence as a List of compounds</div>
<dl class="notes">
<dt>Specified by:</dt>
<dd><code><a href="Sequence.html#getAsList()">getAsList</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getCompoundAt(int)">
<h3>getCompoundAt</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a></span>&nbsp;<span class="element-name">getCompoundAt</span><wbr><span class="parameters">(int&nbsp;position)</span></div>
<div class="block"><span class="descfrm-type-label">Description copied from interface:&nbsp;<code><a href="Sequence.html#getCompoundAt(int)">Sequence</a></code></span></div>
<div class="block">Returns the Compound at the given biological index</div>
<dl class="notes">
<dt>Specified by:</dt>
<dd><code><a href="Sequence.html#getCompoundAt(int)">getCompoundAt</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt>Parameters:</dt>
<dd><code>position</code> - The 1-indexed position of the amino acid</dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getIndexOf(C)">
<h3 id="getIndexOf(org.biojava3.core.sequence.template.Compound)">getIndexOf</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name">getIndexOf</span><wbr><span class="parameters">(<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&nbsp;compound)</span></div>
<div class="block"><span class="descfrm-type-label">Description copied from interface:&nbsp;<code><a href="Sequence.html#getIndexOf(C)">Sequence</a></code></span></div>
<div class="block">Scans through the Sequence looking for the first occurrence of the given compound</div>
<dl class="notes">
<dt>Specified by:</dt>
<dd><code><a href="Sequence.html#getIndexOf(C)">getIndexOf</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt>Parameters:</dt>
<dd><code>compound</code> - </dd>
<dt>Returns:</dt>
<dd>The first index of compound in this sequence (1-based)</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getLastIndexOf(C)">
<h3 id="getLastIndexOf(org.biojava3.core.sequence.template.Compound)">getLastIndexOf</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name">getLastIndexOf</span><wbr><span class="parameters">(<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&nbsp;compound)</span></div>
<div class="block"><span class="descfrm-type-label">Description copied from interface:&nbsp;<code><a href="Sequence.html#getLastIndexOf(C)">Sequence</a></code></span></div>
<div class="block">Scans through the Sequence looking for the last occurrence of the given compound</div>
<dl class="notes">
<dt>Specified by:</dt>
<dd><code><a href="Sequence.html#getLastIndexOf(C)">getLastIndexOf</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt>Parameters:</dt>
<dd><code>compound</code> - </dd>
<dt>Returns:</dt>
<dd>The last index of compound in this sequence (1-based)</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getLength()">
<h3>getLength</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name">getLength</span>()</div>
<div class="block"><span class="descfrm-type-label">Description copied from interface:&nbsp;<code><a href="Sequence.html#getLength()">Sequence</a></code></span></div>
<div class="block">Returns the length of the Sequence</div>
<dl class="notes">
<dt>Specified by:</dt>
<dd><code><a href="Sequence.html#getLength()">getLength</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getSubSequence(java.lang.Integer,java.lang.Integer)">
<h3>getSubSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</span>&nbsp;<span class="element-name">getSubSequence</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;bioStart,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;bioEnd)</span></div>
<div class="block"><span class="descfrm-type-label">Description copied from interface:&nbsp;<code><a href="Sequence.html#getSubSequence(java.lang.Integer,java.lang.Integer)">Sequence</a></code></span></div>
<div class="block">Returns a portion of the sequence from the different positions. This is indexed from 1</div>
<dl class="notes">
<dt>Specified by:</dt>
<dd><code><a href="Sequence.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt>Parameters:</dt>
<dd><code>bioStart</code> - </dd>
<dd><code>bioEnd</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="iterator()">
<h3>iterator</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Iterator.html" title="class or interface in java.util" class="external-link">Iterator</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</span>&nbsp;<span class="element-name">iterator</span>()</div>
<dl class="notes">
<dt>Specified by:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html#iterator()" title="class or interface in java.lang" class="external-link">iterator</a></code>&nbsp;in interface&nbsp;<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html" title="class or interface in java.lang" class="external-link">Iterable</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="countCompounds(C...)">
<h3 id="countCompounds(org.biojava3.core.sequence.template.Compound[])">countCompounds</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name">countCompounds</span><wbr><span class="parameters">(<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>...&nbsp;compounds)</span></div>
<div class="block"><span class="descfrm-type-label">Description copied from interface:&nbsp;<code><a href="Sequence.html#countCompounds(C...)">Sequence</a></code></span></div>
<div class="block">Returns the number of times we found a compound in the Sequence</div>
<dl class="notes">
<dt>Specified by:</dt>
<dd><code><a href="Sequence.html#countCompounds(C...)">countCompounds</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt>Parameters:</dt>
<dd><code>compounds</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getInverse()">
<h3>getInverse</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</span>&nbsp;<span class="element-name">getInverse</span>()</div>
<div class="block"><span class="descfrm-type-label">Description copied from interface:&nbsp;<code><a href="Sequence.html#getInverse()">Sequence</a></code></span></div>
<div class="block">Does the <em>right thing</em> to get the inverse of the current Sequence. This means either
 reversing the Sequence and optionally complementing the Sequence.</div>
<dl class="notes">
<dt>Specified by:</dt>
<dd><code><a href="Sequence.html#getInverse()">getInverse</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
</div>
</div>
</body>
</html>
