<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 14:20:51 CEST 2019 -->
<title>AbstractSequence</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="AbstractSequence";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10,"i19":10,"i20":10,"i21":10,"i22":10,"i23":10,"i24":10,"i25":10,"i26":10,"i27":10,"i28":10,"i29":10,"i30":10,"i31":10,"i32":10,"i33":10,"i34":10,"i35":10,"i36":10,"i37":10,"i38":10,"i39":10,"i40":10,"i41":10,"i42":10,"i43":10,"i44":10,"i45":10,"i46":10,"i47":10,"i48":10,"i49":10,"i50":10,"i51":10,"i52":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/AbstractSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">org.biojava3.core.sequence.template</a></div>
<h2 title="Class AbstractSequence" class="title">Class AbstractSequence&lt;C extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>org.biojava3.core.sequence.template.AbstractSequence&lt;C&gt;</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt><span class="paramLabel">Type Parameters:</span></dt>
<dd><code>C</code> - </dd>
</dl>
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><code>java.lang.Iterable&lt;C&gt;</code>, <code><a href="Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</a></code>, <code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;</code></dd>
</dl>
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><code><a href="../BasicSequence.html" title="class in org.biojava3.core.sequence">BasicSequence</a></code>, <code><a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code>, <code><a href="../ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></code>, <code><a href="../RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></code></dd>
</dl>
<hr>
<pre>public abstract class <span class="typeNameLabel">AbstractSequence&lt;C extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span>
extends java.lang.Object
implements <a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;C&gt;</pre>
<div class="block">The base class for DNA, RNA and Protein sequences.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="memberSummary">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">AbstractSequence</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">AbstractSequence</a></span>&#8203;(java.lang.String&nbsp;seqString,
                <a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Create a Sequence from a simple string where the values should be found in
 compoundSet</div>
</td>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">AbstractSequence</a></span>&#8203;(<a href="SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;proxyLoader,
                <a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">A ProxySequenceReader allows abstraction of both the storage of the sequence
 data and the location of the sequence data.</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addFeature(int,int,org.biojava3.core.sequence.features.FeatureInterface)">addFeature</a></span>&#8203;(int&nbsp;bioStart,
          int&nbsp;bioEnd,
          <a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code></th>
<td class="colLast">
<div class="block">Method to help set the proper details for a feature as it relates to a
 sequence where the feature needs to have a location on the sequence</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addFeature(org.biojava3.core.sequence.features.FeatureInterface)">addFeature</a></span>&#8203;(<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code></th>
<td class="colLast">
<div class="block">Add a feature to this sequence.</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addNote(java.lang.String)">addNote</a></span>&#8203;(java.lang.String&nbsp;note)</code></th>
<td class="colLast">
<div class="block">Add notes about this sequence that will get exported for GFF3</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">
<div class="block">Returns the number of times we found a compound in the Sequence</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code><a href="../AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAccession()">getAccession</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the AccessionID this location is currently bound with</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code><a href="AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAnnotationType()">getAnnotationType</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAsList()">getAsList</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the Sequence as a List of compounds</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>java.lang.Integer</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getBioBegin()">getBioBegin</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>java.lang.Integer</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getBioEnd()">getBioEnd</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code><a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCompoundAt(int)">getCompoundAt</a></span>&#8203;(int&nbsp;position)</code></th>
<td class="colLast">
<div class="block">Returns the Compound at the given biological index</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code><a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">
<div class="block">Gets the compound set used to back this Sequence</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code><a href="../features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDatabaseReferences()">getDatabaseReferences</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDescription()">getDescription</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getFeatures()">getFeatures</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getFeatures(int)">getFeatures</a></span>&#8203;(int&nbsp;bioSequencePosition)</code></th>
<td class="colLast">
<div class="block">Return features at a sequence position</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getFeatures(java.lang.String,int)">getFeatures</a></span>&#8203;(java.lang.String&nbsp;featureType,
           int&nbsp;bioSequencePosition)</code></th>
<td class="colLast">
<div class="block">Return features at a sequence position by type</div>
</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getFeaturesByType(java.lang.String)">getFeaturesByType</a></span>&#8203;(java.lang.String&nbsp;type)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code><a href="../features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getFeaturesKeyWord()">getFeaturesKeyWord</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getIndexOf(C)">getIndexOf</a></span>&#8203;(<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&nbsp;compound)</code></th>
<td class="colLast">
<div class="block">Scans through the Sequence looking for the first occurrence of the given
 compound</div>
</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code><a href="SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getInverse()">getInverse</a></span>()</code></th>
<td class="colLast">
<div class="block">Does the <em>right thing</em> to get the inverse of the current Sequence.</div>
</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getLastIndexOf(C)">getLastIndexOf</a></span>&#8203;(<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&nbsp;compound)</code></th>
<td class="colLast">
<div class="block">Scans through the Sequence looking for the last occurrence of the given
 compound</div>
</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getLength()">getLength</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the length of the Sequence</div>
</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getNotesList()">getNotesList</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getOriginalHeader()">getOriginalHeader</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><code><a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getParentSequence()">getParentSequence</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><code><a href="SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProxySequenceReader()">getProxySequenceReader</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSequenceAsString()">getSequenceAsString</a></span>()</code></th>
<td class="colLast">
<div class="block">Default case is to assume strand is positive because only CDSSequence can be
 either positive or negative Strand.</div>
</td>
</tr>
<tr id="i27" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)">getSequenceAsString</a></span>&#8203;(java.lang.Integer&nbsp;bioStart,
                   java.lang.Integer&nbsp;bioEnd,
                   <a href="../Strand.html" title="enum in org.biojava3.core.sequence">Strand</a>&nbsp;strand)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i28" class="altColor">
<td class="colFirst"><code>java.lang.Double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSequenceScore()">getSequenceScore</a></span>()</code></th>
<td class="colLast">
<div class="block">Provide place holder for a metric that indicate a score associated with the
 sequence</div>
</td>
</tr>
<tr id="i29" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSource()">getSource</a></span>()</code></th>
<td class="colLast">
<div class="block">Added support for the source of this sequence for GFF3 export If a sub
 sequence doesn't have source then check for parent source</div>
</td>
</tr>
<tr id="i30" class="altColor">
<td class="colFirst"><code><a href="SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence</a></span>&#8203;(java.lang.Integer&nbsp;bioStart,
              java.lang.Integer&nbsp;bioEnd)</code></th>
<td class="colLast">
<div class="block">Returns a portion of the sequence from the different positions.</div>
</td>
</tr>
<tr id="i31" class="rowColor">
<td class="colFirst"><code><a href="../TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTaxonomy()">getTaxonomy</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i32" class="altColor">
<td class="colFirst"><code>java.util.Collection&lt;java.lang.Object&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getUserCollection()">getUserCollection</a></span>()</code></th>
<td class="colLast">
<div class="block">Provided for convience if the developer needs to associate data with a
 sequence</div>
</td>
</tr>
<tr id="i33" class="rowColor">
<td class="colFirst"><code>java.util.Iterator&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#iterator()">iterator</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i34" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#removeFeature(org.biojava3.core.sequence.features.FeatureInterface)">removeFeature</a></span>&#8203;(<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code></th>
<td class="colLast">
<div class="block">Remove a feature from the sequence</div>
</td>
</tr>
<tr id="i35" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#removeNote(java.lang.String)">removeNote</a></span>&#8203;(java.lang.String&nbsp;note)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i36" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setAccession(org.biojava3.core.sequence.AccessionID)">setAccession</a></span>&#8203;(<a href="../AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i37" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType</a></span>&#8203;(<a href="AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a>&nbsp;annotationType)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i38" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setBioBegin(java.lang.Integer)">setBioBegin</a></span>&#8203;(java.lang.Integer&nbsp;begin)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i39" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setBioEnd(java.lang.Integer)">setBioEnd</a></span>&#8203;(java.lang.Integer&nbsp;end)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i40" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a></span>&#8203;(<a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i41" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences</a></span>&#8203;(<a href="../features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</a>&nbsp;databaseReferences)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i42" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDescription(java.lang.String)">setDescription</a></span>&#8203;(java.lang.String&nbsp;description)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i43" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord</a></span>&#8203;(<a href="../features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</a>&nbsp;featuresKeyWord)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i44" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setNotesList(java.util.ArrayList)">setNotesList</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;notesList)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i45" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setOriginalHeader(java.lang.String)">setOriginalHeader</a></span>&#8203;(java.lang.String&nbsp;originalHeader)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i46" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)">setParentSequence</a></span>&#8203;(<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;abstractSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i47" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)">setProxySequenceReader</a></span>&#8203;(<a href="SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;proxyLoader)</code></th>
<td class="colLast">
<div class="block">Very important method that allows external mappings of sequence data and
 features.</div>
</td>
</tr>
<tr id="i48" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setSequenceScore(java.lang.Double)">setSequenceScore</a></span>&#8203;(java.lang.Double&nbsp;sequenceScore)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i49" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setSource(java.lang.String)">setSource</a></span>&#8203;(java.lang.String&nbsp;source)</code></th>
<td class="colLast">
<div class="block">Added support for the source of this sequence for GFF3 export</div>
</td>
</tr>
<tr id="i50" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setTaxonomy(org.biojava3.core.sequence.TaxonomyID)">setTaxonomy</a></span>&#8203;(<a href="../TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</a>&nbsp;taxonomy)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i51" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setUserCollection(java.util.Collection)">setUserCollection</a></span>&#8203;(java.util.Collection&lt;java.lang.Object&gt;&nbsp;userCollection)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i52" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#toString()">toString</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>equals, getClass, hashCode, notify, notifyAll, wait, wait, wait</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Iterable">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;java.lang.Iterable</h3>
<code>forEach, spliterator</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>AbstractSequence</h4>
<pre>public&nbsp;AbstractSequence()</pre>
</li>
</ul>
<a id="&lt;init&gt;(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>AbstractSequence</h4>
<pre>public&nbsp;AbstractSequence&#8203;(java.lang.String&nbsp;seqString,
                        <a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</pre>
<div class="block">Create a Sequence from a simple string where the values should be found in
 compoundSet</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>seqString</code> - </dd>
<dd><code>compoundSet</code> - </dd>
</dl>
</li>
</ul>
<a id="&lt;init&gt;(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>AbstractSequence</h4>
<pre>public&nbsp;AbstractSequence&#8203;(<a href="SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;proxyLoader,
                        <a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</pre>
<div class="block">A ProxySequenceReader allows abstraction of both the storage of the sequence
 data and the location of the sequence data. A variety of use cases are
 possible. A ProxySequenceReader that knows the offset and of teh sequence in
 a large fasta file. A ProxySequenceReader that can pull Sequence data from
 Uniprot, NCBI or a custom database. If the ProxySequecneReader implements
 various interfaces then the sequence will set those interfaces so that calls
 to various methods will be valid.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>proxyLoader</code> - </dd>
<dd><code>compoundSet</code> - </dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setProxySequenceReader</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setProxySequenceReader&#8203;(<a href="SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;proxyLoader)</pre>
<div class="block">Very important method that allows external mappings of sequence data and
 features. This method will gain additional interface inspection that allows
 external data sources with knowledge of features for a sequence to be
 supported.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>proxyLoader</code> - </dd>
</dl>
</li>
</ul>
<a id="getProxySequenceReader()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProxySequenceReader</h4>
<pre class="methodSignature">public&nbsp;<a href="SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;getProxySequenceReader()</pre>
</li>
</ul>
<a id="getBioBegin()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBioBegin</h4>
<pre class="methodSignature">public&nbsp;java.lang.Integer&nbsp;getBioBegin()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the bioBegin</dd>
</dl>
</li>
</ul>
<a id="setBioBegin(java.lang.Integer)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setBioBegin</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setBioBegin&#8203;(java.lang.Integer&nbsp;begin)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioBegin</code> - the bioBegin to set</dd>
</dl>
</li>
</ul>
<a id="getBioEnd()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBioEnd</h4>
<pre class="methodSignature">public&nbsp;java.lang.Integer&nbsp;getBioEnd()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the bioEnd</dd>
</dl>
</li>
</ul>
<a id="setBioEnd(java.lang.Integer)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setBioEnd</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setBioEnd&#8203;(java.lang.Integer&nbsp;end)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioEnd</code> - the bioEnd to set</dd>
</dl>
</li>
</ul>
<a id="getUserCollection()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getUserCollection</h4>
<pre class="methodSignature">public&nbsp;java.util.Collection&lt;java.lang.Object&gt;&nbsp;getUserCollection()</pre>
<div class="block">Provided for convience if the developer needs to associate data with a
 sequence</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="setUserCollection(java.util.Collection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setUserCollection</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setUserCollection&#8203;(java.util.Collection&lt;java.lang.Object&gt;&nbsp;userCollection)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>userCollection</code> - </dd>
</dl>
</li>
</ul>
<a id="getAnnotationType()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotationType</h4>
<pre class="methodSignature">public&nbsp;<a href="AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a>&nbsp;getAnnotationType()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the annotation</dd>
</dl>
</li>
</ul>
<a id="setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAnnotationType</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setAnnotationType&#8203;(<a href="AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a>&nbsp;annotationType)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>annotation</code> - the annotation to set</dd>
</dl>
</li>
</ul>
<a id="getDescription()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDescription</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getDescription()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the description</dd>
</dl>
</li>
</ul>
<a id="setDescription(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDescription</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDescription&#8203;(java.lang.String&nbsp;description)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>description</code> - the description to set</dd>
</dl>
</li>
</ul>
<a id="getOriginalHeader()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getOriginalHeader</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getOriginalHeader()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the originalHeader</dd>
</dl>
</li>
</ul>
<a id="setOriginalHeader(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setOriginalHeader</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setOriginalHeader&#8203;(java.lang.String&nbsp;originalHeader)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>originalHeader</code> - the originalHeader to set</dd>
</dl>
</li>
</ul>
<a id="getParentSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getParentSequence</h4>
<pre class="methodSignature">public&nbsp;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;getParentSequence()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the parentSequence</dd>
</dl>
</li>
</ul>
<a id="setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setParentSequence</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setParentSequence&#8203;(<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;abstractSequence)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>abstractSequence</code> - the parentSequence to set</dd>
</dl>
</li>
</ul>
<a id="getSource()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSource</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getSource()</pre>
<div class="block">Added support for the source of this sequence for GFF3 export If a sub
 sequence doesn't have source then check for parent source</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the source</dd>
</dl>
</li>
</ul>
<a id="setSource(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSource</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setSource&#8203;(java.lang.String&nbsp;source)</pre>
<div class="block">Added support for the source of this sequence for GFF3 export</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>source</code> - the source to set</dd>
</dl>
</li>
</ul>
<a id="addNote(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addNote</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;addNote&#8203;(java.lang.String&nbsp;note)</pre>
<div class="block">Add notes about this sequence that will get exported for GFF3</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>note</code> - </dd>
</dl>
</li>
</ul>
<a id="removeNote(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeNote</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;removeNote&#8203;(java.lang.String&nbsp;note)</pre>
</li>
</ul>
<a id="getNotesList()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNotesList</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getNotesList()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the notesList</dd>
</dl>
</li>
</ul>
<a id="setNotesList(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setNotesList</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setNotesList&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;notesList)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>notesList</code> - the notesList to set</dd>
</dl>
</li>
</ul>
<a id="getSequenceScore()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceScore</h4>
<pre class="methodSignature">public&nbsp;java.lang.Double&nbsp;getSequenceScore()</pre>
<div class="block">Provide place holder for a metric that indicate a score associated with the
 sequence</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the sequenceScore</dd>
</dl>
</li>
</ul>
<a id="setSequenceScore(java.lang.Double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSequenceScore</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setSequenceScore&#8203;(java.lang.Double&nbsp;sequenceScore)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sequenceScore</code> - the sequenceScore to set</dd>
</dl>
</li>
</ul>
<a id="getFeatures(java.lang.String,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeatures</h4>
<pre class="methodSignature">public&nbsp;java.util.List&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;&nbsp;getFeatures&#8203;(java.lang.String&nbsp;featureType,
                                                                                 int&nbsp;bioSequencePosition)</pre>
<div class="block">Return features at a sequence position by type</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>featureType</code> - </dd>
<dd><code>bioSequencePosition</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getFeatures(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeatures</h4>
<pre class="methodSignature">public&nbsp;java.util.List&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;&nbsp;getFeatures&#8203;(int&nbsp;bioSequencePosition)</pre>
<div class="block">Return features at a sequence position</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>featureType</code> - </dd>
<dd><code>bioSequencePosition</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getFeatures()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeatures</h4>
<pre class="methodSignature">public&nbsp;java.util.List&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;&nbsp;getFeatures()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="addFeature(int,int,org.biojava3.core.sequence.features.FeatureInterface)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addFeature</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;addFeature&#8203;(int&nbsp;bioStart,
                       int&nbsp;bioEnd,
                       <a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</pre>
<div class="block">Method to help set the proper details for a feature as it relates to a
 sequence where the feature needs to have a location on the sequence</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioStart</code> - </dd>
<dd><code>bioEnd</code> - </dd>
<dd><code>feature</code> - </dd>
</dl>
</li>
</ul>
<a id="addFeature(org.biojava3.core.sequence.features.FeatureInterface)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addFeature</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;addFeature&#8203;(<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</pre>
<div class="block">Add a feature to this sequence. The feature will be added to the collection
 where the order is start position and if more than one feature at the same
 start position then longest is added first. This helps on doing feature
 layout for displaying features in SequenceFeaturePanel</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>feature</code> - </dd>
</dl>
</li>
</ul>
<a id="removeFeature(org.biojava3.core.sequence.features.FeatureInterface)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeFeature</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;removeFeature&#8203;(<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</pre>
<div class="block">Remove a feature from the sequence</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>feature</code> - </dd>
</dl>
</li>
</ul>
<a id="getFeaturesByType(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeaturesByType</h4>
<pre class="methodSignature">public&nbsp;java.util.List&lt;<a href="../features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,&#8203;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;&nbsp;getFeaturesByType&#8203;(java.lang.String&nbsp;type)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>type</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getFeaturesKeyWord()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeaturesKeyWord</h4>
<pre class="methodSignature">public&nbsp;<a href="../features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</a>&nbsp;getFeaturesKeyWord()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the featuresKeyWord</dd>
</dl>
</li>
</ul>
<a id="setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setFeaturesKeyWord</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setFeaturesKeyWord&#8203;(<a href="../features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</a>&nbsp;featuresKeyWord)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>featuresKeyWord</code> - the featuresKeyWord to set</dd>
</dl>
</li>
</ul>
<a id="getDatabaseReferences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDatabaseReferences</h4>
<pre class="methodSignature">public&nbsp;<a href="../features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</a>&nbsp;getDatabaseReferences()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the databaseReferences</dd>
</dl>
</li>
</ul>
<a id="setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDatabaseReferences</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDatabaseReferences&#8203;(<a href="../features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</a>&nbsp;databaseReferences)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>databaseReferences</code> - the databaseReferences to set</dd>
</dl>
</li>
</ul>
<a id="getAccession()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAccession</h4>
<pre class="methodSignature">public&nbsp;<a href="../AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;getAccession()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="Accessioned.html#getAccession()">Accessioned</a></code></span></div>
<div class="block">Returns the AccessionID this location is currently bound with</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="Accessioned.html#getAccession()">getAccession</a></code>&nbsp;in interface&nbsp;<code><a href="Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</a></code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the accession</dd>
</dl>
</li>
</ul>
<a id="setAccession(org.biojava3.core.sequence.AccessionID)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAccession</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setAccession&#8203;(<a href="../AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - the accession to set</dd>
</dl>
</li>
</ul>
<a id="getTaxonomy()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTaxonomy</h4>
<pre class="methodSignature">public&nbsp;<a href="../TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</a>&nbsp;getTaxonomy()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the species</dd>
</dl>
</li>
</ul>
<a id="setTaxonomy(org.biojava3.core.sequence.TaxonomyID)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setTaxonomy</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setTaxonomy&#8203;(<a href="../TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</a>&nbsp;taxonomy)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>species</code> - the species to set</dd>
</dl>
</li>
</ul>
<a id="getCompoundSet()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCompoundSet</h4>
<pre class="methodSignature">public&nbsp;<a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;getCompoundSet()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="Sequence.html#getCompoundSet()">Sequence</a></code></span></div>
<div class="block">Gets the compound set used to back this Sequence</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="Sequence.html#getCompoundSet()">getCompoundSet</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
</dl>
</li>
</ul>
<a id="setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setCompoundSet</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setCompoundSet&#8203;(<a href="CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</pre>
</li>
</ul>
<a id="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;toString()</pre>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code>toString</code>&nbsp;in class&nbsp;<code>java.lang.Object</code></dd>
</dl>
</li>
</ul>
<a id="getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAsString</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getSequenceAsString&#8203;(java.lang.Integer&nbsp;bioStart,
                                            java.lang.Integer&nbsp;bioEnd,
                                            <a href="../Strand.html" title="enum in org.biojava3.core.sequence">Strand</a>&nbsp;strand)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>begin</code> - </dd>
<dd><code>end</code> - </dd>
<dd><code>strand</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getSequenceAsString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAsString</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getSequenceAsString()</pre>
<div class="block">Default case is to assume strand is positive because only CDSSequence can be
 either positive or negative Strand.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="Sequence.html#getSequenceAsString()">getSequenceAsString</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getAsList()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAsList</h4>
<pre class="methodSignature">public&nbsp;java.util.List&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;getAsList()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="Sequence.html#getAsList()">Sequence</a></code></span></div>
<div class="block">Returns the Sequence as a List of compounds</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="Sequence.html#getAsList()">getAsList</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getCompoundAt(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCompoundAt</h4>
<pre class="methodSignature">public&nbsp;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&nbsp;getCompoundAt&#8203;(int&nbsp;position)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="Sequence.html#getCompoundAt(int)">Sequence</a></code></span></div>
<div class="block">Returns the Compound at the given biological index</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="Sequence.html#getCompoundAt(int)">getCompoundAt</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>position</code> - The 1-indexed position of the amino acid</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getIndexOf(org.biojava3.core.sequence.template.Compound)">
<!--   -->
</a><a id="getIndexOf(C)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getIndexOf</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getIndexOf&#8203;(<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&nbsp;compound)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="Sequence.html#getIndexOf(C)">Sequence</a></code></span></div>
<div class="block">Scans through the Sequence looking for the first occurrence of the given
 compound</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="Sequence.html#getIndexOf(C)">getIndexOf</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>compound</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>The first index of compound in this sequence (1-based)</dd>
</dl>
</li>
</ul>
<a id="getLastIndexOf(org.biojava3.core.sequence.template.Compound)">
<!--   -->
</a><a id="getLastIndexOf(C)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLastIndexOf</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getLastIndexOf&#8203;(<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&nbsp;compound)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="Sequence.html#getLastIndexOf(C)">Sequence</a></code></span></div>
<div class="block">Scans through the Sequence looking for the last occurrence of the given
 compound</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="Sequence.html#getLastIndexOf(C)">getLastIndexOf</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>compound</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>The last index of compound in this sequence (1-based)</dd>
</dl>
</li>
</ul>
<a id="getLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLength</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getLength()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="Sequence.html#getLength()">Sequence</a></code></span></div>
<div class="block">Returns the length of the Sequence</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="Sequence.html#getLength()">getLength</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getSubSequence(java.lang.Integer,java.lang.Integer)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSubSequence</h4>
<pre class="methodSignature">public&nbsp;<a href="SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;getSubSequence&#8203;(java.lang.Integer&nbsp;bioStart,
                                      java.lang.Integer&nbsp;bioEnd)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="Sequence.html#getSubSequence(java.lang.Integer,java.lang.Integer)">Sequence</a></code></span></div>
<div class="block">Returns a portion of the sequence from the different positions. This is
 indexed from 1</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="Sequence.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioStart</code> - </dd>
<dd><code>bioEnd</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="iterator()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>iterator</h4>
<pre class="methodSignature">public&nbsp;java.util.Iterator&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;iterator()</pre>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code>iterator</code>&nbsp;in interface&nbsp;<code>java.lang.Iterable&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="countCompounds(org.biojava3.core.sequence.template.Compound[])">
<!--   -->
</a><a id="countCompounds(C...)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>countCompounds</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;countCompounds&#8203;(<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>...&nbsp;compounds)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="Sequence.html#countCompounds(C...)">Sequence</a></code></span></div>
<div class="block">Returns the number of times we found a compound in the Sequence</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="Sequence.html#countCompounds(C...)">countCompounds</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>compounds</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getInverse()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getInverse</h4>
<pre class="methodSignature">public&nbsp;<a href="SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;getInverse()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="Sequence.html#getInverse()">Sequence</a></code></span></div>
<div class="block">Does the <em>right thing</em> to get the inverse of the current Sequence.
 This means either reversing the Sequence and optionally complementing the
 Sequence.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="Sequence.html#getInverse()">getInverse</a></code>&nbsp;in interface&nbsp;<code><a href="Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/AbstractSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
