<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>GenomeConversion</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="declaration: package: bacnet.datamodel.sequenceNCBI, class: GenomeConversion">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var evenRowColor = "even-row-color";
var oddRowColor = "odd-row-color";
var tableTab = "table-tab";
var activeTableTab = "active-table-tab";
var pathtoroot = "../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/GenomeConversion.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html#class">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div>
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.sequenceNCBI</a></div>
<h1 title="Class GenomeConversion" class="title">Class GenomeConversion</h1>
</div>
<div class="inheritance" title="Inheritance Tree"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">java.lang.Object</a>
<div class="inheritance">bacnet.datamodel.sequenceNCBI.GenomeConversion</div>
</div>
<section class="class-description" id="class-description">
<hr>
<div class="type-signature"><span class="modifiers">public class </span><span class="element-name type-name-label">GenomeConversion</span>
<span class="extends-implements">extends <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></span></div>
<div class="block">Static methods for converting GenomeNCBI to Genome<br>
 Genome is used for fast loading of all genomic information we gathered</div>
<dl class="notes">
<dt>Author:</dt>
<dd>UIBC</dd>
</dl>
</section>
<section class="summary">
<ul class="summary-list">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<li>
<section class="constructor-summary" id="constructor-summary">
<h2>Constructor Summary</h2>
<div class="caption"><span>Constructors</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E()" class="member-name-link">GenomeConversion</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab1" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab1', 3)" class="table-tab">Static Methods</button><button id="method-summary-table-tab4" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab4', 3)" class="table-tab">Concrete Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel">
<div class="summary-table three-column-summary" aria-labelledby="method-summary-table-tab0">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#createAnnotation(bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">createAnnotation</a><wbr>(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;path)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Create a Table with all elements of the Genome, which will be used for element search and display
 on the GenomeViewer</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#createFolderCopyFNA(java.lang.String,java.lang.String)" class="member-name-link">createFolderCopyFNA</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;path,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;pathNCBI)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Create a folder for each Genome, and copy all .fna in it</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#run(bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String,java.lang.String)" class="member-name-link">run</a><wbr>(<a href="GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;pathGenome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#run(bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">run</a><wbr>(<a href="GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;pathGenome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomePath)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#run(java.lang.String)" class="member-name-link">run</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Method for Genome in the GenomeNCBIFolder<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#updateAllElements(bacnet.datamodel.sequence.Genome)" class="member-name-link">updateAllElements</a><wbr>(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Add all elements found in the Chromosome to LinkedHashMap&lt;String, Sequence> allElements</div>
</div>
</div>
</div>
</div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Object">Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)" title="class or interface in java.lang" class="external-link">equals</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#getClass()" title="class or interface in java.lang" class="external-link">getClass</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode()" title="class or interface in java.lang" class="external-link">hashCode</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notify()" title="class or interface in java.lang" class="external-link">notify</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notifyAll()" title="class or interface in java.lang" class="external-link">notifyAll</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#toString()" title="class or interface in java.lang" class="external-link">toString</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait()" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long)" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long,int)" title="class or interface in java.lang" class="external-link">wait</a></code></div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<li>
<section class="constructor-details" id="constructor-detail">
<h2>Constructor Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="&lt;init&gt;()">
<h3>GenomeConversion</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name">GenomeConversion</span>()</div>
</section>
</li>
</ul>
</section>
</li>
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="run(java.lang.String)">
<h3>run</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span>&nbsp;<span class="element-name">run</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName)</span></div>
<div class="block">Method for Genome in the GenomeNCBIFolder<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>genomeNCBI</code> - </dd>
<dd><code>genomeName</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="run(bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String,java.lang.String)">
<h3>run</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span>&nbsp;<span class="element-name">run</span><wbr><span class="parameters">(<a href="GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;pathGenome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName)</span></div>
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>genomeNCBI</code> - already load GenomeNCBI</dd>
<dd><code>pathGenome</code> - path where .fna files are</dd>
<dd><code>genomeName</code> - final genome name</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="run(bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String,java.lang.String,java.lang.String)">
<h3>run</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></span>&nbsp;<span class="element-name">run</span><wbr><span class="parameters">(<a href="GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;pathGenome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomeName,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genomePath)</span></div>
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>genomeNCBI</code> - already load GenomeNCBI</dd>
<dd><code>pathGenome</code> - path where .fna files are</dd>
<dd><code>genomeName</code> - final genome name</dd>
<dd><code>genomePath</code> - should have genomePath = Project.getGENOMES()+genomeName;</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="updateAllElements(bacnet.datamodel.sequence.Genome)">
<h3>updateAllElements</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">updateAllElements</span><wbr><span class="parameters">(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</span></div>
<div class="block">Add all elements found in the Chromosome to LinkedHashMap&lt;String, Sequence> allElements</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>genome</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="createAnnotation(bacnet.datamodel.sequence.Genome,java.lang.String)">
<h3>createAnnotation</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">createAnnotation</span><wbr><span class="parameters">(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;path)</span></div>
<div class="block">Create a Table with all elements of the Genome, which will be used for element search and display
 on the GenomeViewer</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>genome</code> - </dd>
<dd><code>path</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="createFolderCopyFNA(java.lang.String,java.lang.String)">
<h3>createFolderCopyFNA</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">createFolderCopyFNA</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;path,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;pathNCBI)</span></div>
<div class="block">Create a folder for each Genome, and copy all .fna in it</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>path</code> - </dd>
<dd><code>pathNCBI</code> - </dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
</div>
</div>
</body>
</html>
