<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 14:20:51 CEST 2019 -->
<title>GenomeNCBITools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="GenomeNCBITools";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":9,"i1":9,"i2":9,"i3":9,"i4":9,"i5":9,"i6":9,"i7":9,"i8":9,"i9":9,"i10":9,"i11":9,"i12":9,"i13":9,"i14":9,"i15":9,"i16":9,"i17":9,"i18":9,"i19":9};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GenomeNCBITools.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.sequenceNCBI</a></div>
<h2 title="Class GenomeNCBITools" class="title">Class GenomeNCBITools</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>bacnet.datamodel.sequenceNCBI.GenomeNCBITools</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<pre>public class <span class="typeNameLabel">GenomeNCBITools</span>
extends java.lang.Object</pre>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#PATH_NCBI_BacGenome">PATH_NCBI_BacGenome</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">GenomeNCBITools</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#codonUsageEGDe()">codonUsageEGDe</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#curateGenomeFindingDoublonGeneName(bacnet.datamodel.sequence.GenomeNCBI)">curateGenomeFindingDoublonGeneName</a></span>&#8203;(<a href="../sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Check if two different locustag are not associated to the same GeneName</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static java.lang.String[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAvailableGenome()">getAvailableGenome</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAvailableGenome(boolean)">getAvailableGenome</a></span>&#8203;(boolean&nbsp;list)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getBacillusGenome()">getBacillusGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Return a list of all NON Listeria genomes</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>static <a href="../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCOGExpression(bacnet.datamodel.dataset.ExpressionMatrix)">getCOGExpression</a></span>&#8203;(<a href="../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;matrix)</code></th>
<td class="colLast">
<div class="block">Summarize an ExpressionMatrix by regrouping the different genes (rows) by
 their COG functional categories <br>
 Averages on all the genes for each category are calculated.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCompleteGenome(java.util.ArrayList)">getCompleteGenome</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;genomesList)</code></th>
<td class="colLast">
<div class="block">Extract all complete genomes from genomesList</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getListeriaGenomes()">getListeriaGenomes</a></span>()</code></th>
<td class="colLast">
<div class="block">Return a List of all Listeria genomes available</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getListeriaMonoGenome()">getListeriaMonoGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Return a List of all Listeria Monocytogenes genomes available</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getListeriaNonMonoGenome()">getListeriaNonMonoGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Return a List of all Listeria NOT Monocytogenes genomes available</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getNotListeriaAndBacillusGenome()">getNotListeriaAndBacillusGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Return a list of all NON Listeria genomes</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getNotListeriaGenome()">getNotListeriaGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Return a list of all NON Listeria genomes</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getPATH()">getPATH</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>static java.lang.String[][]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getRfamAvailableGenome()">getRfamAvailableGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Read RFAM_GENOME table, and extract a list of all genomes which has both
 Rfam_ID and NCBI_Id</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getWGS(java.util.ArrayList)">getWGS</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;genomesList)</code></th>
<td class="colLast">
<div class="block">Extract all WGS (Whole Genome Shotgun) from genomesList</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isCompleteGenome(java.lang.String)">isCompleteGenome</a></span>&#8203;(java.lang.String&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Detect if the genome is a Complete genome or a WGS using DATABASES_LIST_PATH
 table</div>
</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>static <a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadCorrespondingChromosome(java.lang.String)">loadCorrespondingChromosome</a></span>&#8203;(java.lang.String&nbsp;accession)</code></th>
<td class="colLast">
<div class="block">load and return the chromosome corresponding to an accession</div>
</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>static <a href="../sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadEgdeGenome()">loadEgdeGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Find in ModelProvider if Egde Genome is already loaded If not search
 EGDE_NAME in GENOMEDIR</div>
</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>static <a href="../sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadGenome(java.lang.String)">loadGenome</a></span>&#8203;(java.lang.String&nbsp;genomeAccession)</code></th>
<td class="colLast">
<div class="block">Find in ModelProvider if Genome Accession is already loaded If not search
 load it</div>
</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code>static <a href="../sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadGenome(java.lang.String,java.lang.String,boolean,boolean)">loadGenome</a></span>&#8203;(java.lang.String&nbsp;genomeAccession,
          java.lang.String&nbsp;path,
          boolean&nbsp;keepInMemory,
          boolean&nbsp;annotation)</code></th>
<td class="colLast">
<div class="block">Load/return a genome given by a genomeAccession ID<br>
 <br>
 
 If keepInMemory is true : Find in ModelProvider if Genome Accession is
 already loaded if not load it <br>
 If keepInMemory is false : Only load the Genome without saving it <br>
 <br>
 If annotation is true : Load annotation information of the genome <br>
 If anotation is false : Only load the sequence of the genome</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="PATH_NCBI_BacGenome">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>PATH_NCBI_BacGenome</h4>
<pre>public static&nbsp;java.lang.String PATH_NCBI_BacGenome</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>GenomeNCBITools</h4>
<pre>public&nbsp;GenomeNCBITools()</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="loadCorrespondingChromosome(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadCorrespondingChromosome</h4>
<pre class="methodSignature">public static&nbsp;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;loadCorrespondingChromosome&#8203;(java.lang.String&nbsp;accession)
                                               throws java.lang.Exception</pre>
<div class="block">load and return the chromosome corresponding to an accession</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="loadEgdeGenome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadEgdeGenome</h4>
<pre class="methodSignature">public static&nbsp;<a href="../sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&nbsp;loadEgdeGenome()</pre>
<div class="block">Find in ModelProvider if Egde Genome is already loaded If not search
 EGDE_NAME in GENOMEDIR</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="loadGenome(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadGenome</h4>
<pre class="methodSignature">public static&nbsp;<a href="../sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&nbsp;loadGenome&#8203;(java.lang.String&nbsp;genomeAccession)</pre>
<div class="block">Find in ModelProvider if Genome Accession is already loaded If not search
 load it</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="loadGenome(java.lang.String,java.lang.String,boolean,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadGenome</h4>
<pre class="methodSignature">public static&nbsp;<a href="../sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&nbsp;loadGenome&#8203;(java.lang.String&nbsp;genomeAccession,
                                    java.lang.String&nbsp;path,
                                    boolean&nbsp;keepInMemory,
                                    boolean&nbsp;annotation)</pre>
<div class="block">Load/return a genome given by a genomeAccession ID<br>
 <br>
 
 If keepInMemory is true : Find in ModelProvider if Genome Accession is
 already loaded if not load it <br>
 If keepInMemory is false : Only load the Genome without saving it <br>
 <br>
 If annotation is true : Load annotation information of the genome <br>
 If anotation is false : Only load the sequence of the genome</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genomeAccession</code> - </dd>
<dd><code>keepInMemory</code> - </dd>
<dd><code>annotation</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getPATH()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPATH</h4>
<pre class="methodSignature">public static&nbsp;java.lang.String&nbsp;getPATH()</pre>
</li>
</ul>
<a id="getAvailableGenome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAvailableGenome</h4>
<pre class="methodSignature">public static&nbsp;java.lang.String[]&nbsp;getAvailableGenome()</pre>
</li>
</ul>
<a id="getAvailableGenome(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAvailableGenome</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getAvailableGenome&#8203;(boolean&nbsp;list)</pre>
</li>
</ul>
<a id="getListeriaGenomes()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getListeriaGenomes</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getListeriaGenomes()</pre>
<div class="block">Return a List of all Listeria genomes available</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getListeriaMonoGenome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getListeriaMonoGenome</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getListeriaMonoGenome()</pre>
<div class="block">Return a List of all Listeria Monocytogenes genomes available</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getListeriaNonMonoGenome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getListeriaNonMonoGenome</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getListeriaNonMonoGenome()</pre>
<div class="block">Return a List of all Listeria NOT Monocytogenes genomes available</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getNotListeriaGenome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNotListeriaGenome</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getNotListeriaGenome()</pre>
<div class="block">Return a list of all NON Listeria genomes</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getBacillusGenome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBacillusGenome</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getBacillusGenome()</pre>
<div class="block">Return a list of all NON Listeria genomes</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getNotListeriaAndBacillusGenome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNotListeriaAndBacillusGenome</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getNotListeriaAndBacillusGenome()</pre>
<div class="block">Return a list of all NON Listeria genomes</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getCompleteGenome(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCompleteGenome</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getCompleteGenome&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;genomesList)
                                                               throws java.io.IOException</pre>
<div class="block">Extract all complete genomes from genomesList</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genomesList</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.io.IOException</code></dd>
</dl>
</li>
</ul>
<a id="getWGS(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getWGS</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getWGS&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;genomesList)
                                                    throws java.io.IOException</pre>
<div class="block">Extract all WGS (Whole Genome Shotgun) from genomesList</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genomesList</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.io.IOException</code></dd>
</dl>
</li>
</ul>
<a id="codonUsageEGDe()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>codonUsageEGDe</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;codonUsageEGDe()</pre>
</li>
</ul>
<a id="getCOGExpression(bacnet.datamodel.dataset.ExpressionMatrix)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCOGExpression</h4>
<pre class="methodSignature">public static&nbsp;<a href="../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;getCOGExpression&#8203;(<a href="../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&nbsp;matrix)</pre>
<div class="block">Summarize an ExpressionMatrix by regrouping the different genes (rows) by
 their COG functional categories <br>
 Averages on all the genes for each category are calculated.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>matrix</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getRfamAvailableGenome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRfamAvailableGenome</h4>
<pre class="methodSignature">public static&nbsp;java.lang.String[][]&nbsp;getRfamAvailableGenome()</pre>
<div class="block">Read RFAM_GENOME table, and extract a list of all genomes which has both
 Rfam_ID and NCBI_Id</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="isCompleteGenome(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isCompleteGenome</h4>
<pre class="methodSignature">public static&nbsp;boolean&nbsp;isCompleteGenome&#8203;(java.lang.String&nbsp;genome)
                                throws java.io.IOException</pre>
<div class="block">Detect if the genome is a Complete genome or a WGS using DATABASES_LIST_PATH
 table</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genome</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.io.IOException</code></dd>
</dl>
</li>
</ul>
<a id="curateGenomeFindingDoublonGeneName(bacnet.datamodel.sequence.GenomeNCBI)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>curateGenomeFindingDoublonGeneName</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;curateGenomeFindingDoublonGeneName&#8203;(<a href="../sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a>&nbsp;genome)</pre>
<div class="block">Check if two different locustag are not associated to the same GeneName</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genome</code> - </dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GenomeNCBITools.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
