<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>GenomeConversion</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="GenomeConversion";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":9,"i1":9,"i2":9,"i3":9,"i4":9,"i5":9,"i6":9,"i7":9,"i8":9,"i9":9};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GenomeConversion.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.sequenceNCBI</a></div>
<h2 title="Class GenomeConversion" class="title">Class GenomeConversion</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>bacnet.datamodel.sequenceNCBI.GenomeConversion</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<pre>public class <span class="typeNameLabel">GenomeConversion</span>
extends java.lang.Object</pre>
<div class="block">Static methods for converting GenomeNCBI to Genome<br>
 Genome is used for fast loading of all genomic information we gathered</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>UIBC</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">GenomeConversion</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>private static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#convertGenes(bacnet.datamodel.sequence.Genome,bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String)">convertGenes</a></span>&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
            <a href="GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
            java.lang.String&nbsp;pathSequences)</code></th>
<td class="colLast">
<div class="block">Go through all chromosomes and all Coding region of GenomeNCBI<br>
 Convert all DNASequence into Gene<br>
 Add the result to Genome</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>private static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#convertNcRNA(bacnet.datamodel.sequence.Genome,bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String)">convertNcRNA</a></span>&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
            <a href="GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
            java.lang.String&nbsp;pathSequences)</code></th>
<td class="colLast">
<div class="block">Go through all chromosomes and all NonCoding regions of GenomeNCBI<br>
 Convert all DNASequence into Gene<br>
 Add the result to Genome</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>private static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#correctGeneName(bacnet.datamodel.sequence.Gene,java.lang.String%5B%5D%5B%5D)">correctGeneName</a></span>&#8203;(<a href="../sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene,
               java.lang.String[][]&nbsp;correctGenesArray)</code></th>
<td class="colLast">
<div class="block">Some Gene name are associated to multiple lmo locus, with this function we delete gene name if
 necessary</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createAnnotation(bacnet.datamodel.sequence.Genome,java.lang.String)">createAnnotation</a></span>&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                java.lang.String&nbsp;path)</code></th>
<td class="colLast">
<div class="block">Create a Table with all elements of the Genome, which will be used for element search and display
 on the GenomeViewer</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createFolderCopyFNA(java.lang.String,java.lang.String)">createFolderCopyFNA</a></span>&#8203;(java.lang.String&nbsp;path,
                   java.lang.String&nbsp;pathNCBI)</code></th>
<td class="colLast">
<div class="block">Create a folder for each Genome, and copy all .fna in it</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>static <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#run(bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String,java.lang.String)">run</a></span>&#8203;(<a href="GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
   java.lang.String&nbsp;pathGenome,
   java.lang.String&nbsp;genomeName)</code></th>
<td class="colLast">
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>static <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#run(bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String,java.lang.String,java.lang.String)">run</a></span>&#8203;(<a href="GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
   java.lang.String&nbsp;pathGenome,
   java.lang.String&nbsp;genomeName,
   java.lang.String&nbsp;genomePath)</code></th>
<td class="colLast">
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>static <a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#run(java.lang.String)">run</a></span>&#8203;(java.lang.String&nbsp;genomeName)</code></th>
<td class="colLast">
<div class="block">Method for Genome in the GenomeNCBIFolder<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>private static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#serializeOtherElements(bacnet.datamodel.sequence.Genome,bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String)">serializeOtherElements</a></span>&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                      <a href="GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
                      java.lang.String&nbsp;pathSequences)</code></th>
<td class="colLast">
<div class="block">Get all elements except Coding and NonCoding serialize them, and put them in Chromosome</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#updateAllElements(bacnet.datamodel.sequence.Genome)">updateAllElements</a></span>&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Add all elements found in the Chromosome to LinkedHashMap<String, Sequence> allElements</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>GenomeConversion</h4>
<pre>public&nbsp;GenomeConversion()</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="run(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>run</h4>
<pre class="methodSignature">public static&nbsp;<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;run&#8203;(java.lang.String&nbsp;genomeName)</pre>
<div class="block">Method for Genome in the GenomeNCBIFolder<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genomeNCBI</code> - </dd>
<dd><code>genomeName</code> - </dd>
</dl>
</li>
</ul>
<a id="run(bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>run</h4>
<pre class="methodSignature">public static&nbsp;<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;run&#8203;(<a href="GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
                         java.lang.String&nbsp;pathGenome,
                         java.lang.String&nbsp;genomeName)</pre>
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genomeNCBI</code> - already load GenomeNCBI</dd>
<dd><code>pathGenome</code> - path where .fna files are</dd>
<dd><code>genomeName</code> - final genome name</dd>
</dl>
</li>
</ul>
<a id="run(bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String,java.lang.String,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>run</h4>
<pre class="methodSignature">public static&nbsp;<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;run&#8203;(<a href="GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
                         java.lang.String&nbsp;pathGenome,
                         java.lang.String&nbsp;genomeName,
                         java.lang.String&nbsp;genomePath)</pre>
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genomeNCBI</code> - already load GenomeNCBI</dd>
<dd><code>pathGenome</code> - path where .fna files are</dd>
<dd><code>genomeName</code> - final genome name</dd>
<dd><code>genomePath</code> - should have genomePath = Project.getGENOMES()+genomeName;</dd>
</dl>
</li>
</ul>
<a id="convertGenes(bacnet.datamodel.sequence.Genome,bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>convertGenes</h4>
<pre class="methodSignature">private static&nbsp;void&nbsp;convertGenes&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                                 <a href="GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
                                 java.lang.String&nbsp;pathSequences)</pre>
<div class="block">Go through all chromosomes and all Coding region of GenomeNCBI<br>
 Convert all DNASequence into Gene<br>
 Add the result to Genome</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genome</code> - </dd>
<dd><code>genomeNCBI</code> - </dd>
</dl>
</li>
</ul>
<a id="correctGeneName(bacnet.datamodel.sequence.Gene,java.lang.String[][])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>correctGeneName</h4>
<pre class="methodSignature">private static&nbsp;void&nbsp;correctGeneName&#8203;(<a href="../sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene,
                                    java.lang.String[][]&nbsp;correctGenesArray)</pre>
<div class="block">Some Gene name are associated to multiple lmo locus, with this function we delete gene name if
 necessary</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>correctGenesArray</code> - </dd>
</dl>
</li>
</ul>
<a id="convertNcRNA(bacnet.datamodel.sequence.Genome,bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>convertNcRNA</h4>
<pre class="methodSignature">private static&nbsp;void&nbsp;convertNcRNA&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                                 <a href="GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
                                 java.lang.String&nbsp;pathSequences)</pre>
<div class="block">Go through all chromosomes and all NonCoding regions of GenomeNCBI<br>
 Convert all DNASequence into Gene<br>
 Add the result to Genome</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genome</code> - </dd>
<dd><code>genomeNCBI</code> - </dd>
</dl>
</li>
</ul>
<a id="serializeOtherElements(bacnet.datamodel.sequence.Genome,bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>serializeOtherElements</h4>
<pre class="methodSignature">private static&nbsp;void&nbsp;serializeOtherElements&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                                           <a href="GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;genomeNCBI,
                                           java.lang.String&nbsp;pathSequences)</pre>
<div class="block">Get all elements except Coding and NonCoding serialize them, and put them in Chromosome</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genome</code> - </dd>
<dd><code>genomeNCBI</code> - </dd>
<dd><code>pathSequences</code> - </dd>
</dl>
</li>
</ul>
<a id="updateAllElements(bacnet.datamodel.sequence.Genome)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>updateAllElements</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;updateAllElements&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</pre>
<div class="block">Add all elements found in the Chromosome to LinkedHashMap<String, Sequence> allElements</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genome</code> - </dd>
</dl>
</li>
</ul>
<a id="createAnnotation(bacnet.datamodel.sequence.Genome,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createAnnotation</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;createAnnotation&#8203;(<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                                    java.lang.String&nbsp;path)</pre>
<div class="block">Create a Table with all elements of the Genome, which will be used for element search and display
 on the GenomeViewer</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genome</code> - </dd>
<dd><code>path</code> - </dd>
</dl>
</li>
</ul>
<a id="createFolderCopyFNA(java.lang.String,java.lang.String)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>createFolderCopyFNA</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;createFolderCopyFNA&#8203;(java.lang.String&nbsp;path,
                                       java.lang.String&nbsp;pathNCBI)</pre>
<div class="block">Create a folder for each Genome, and copy all .fna in it</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>path</code> - </dd>
<dd><code>pathNCBI</code> - </dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GenomeConversion.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
