<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>Uses of Package bacnet.datamodel.sequenceNCBI</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="use: package: bacnet.datamodel.sequenceNCBI">
<meta name="generator" content="javadoc/PackageUseWriter">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="package-use-page">
<script type="text/javascript">var pathtoroot = "../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Package bacnet.datamodel.sequenceNCBI" class="title">Uses of Package<br>bacnet.datamodel.sequenceNCBI</h1>
</div>
<div class="caption"><span>Packages that use <a href="package-summary.html">bacnet.datamodel.sequenceNCBI</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#bacnet">bacnet</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.datamodel.annotation">bacnet.datamodel.annotation</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.datamodel.sequence">bacnet.datamodel.sequence</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.datamodel.sequenceNCBI">bacnet.datamodel.sequenceNCBI</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#bacnet.scripts.genome">bacnet.scripts.genome</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#bacnet.scripts.listeriomics.nterm">bacnet.scripts.listeriomics.nterm</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<section class="package-uses">
<ul class="block-list">
<li>
<section class="detail" id="bacnet">
<div class="caption"><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequenceNCBI</a> used by <a href="../../package-summary.html">bacnet</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="class-use/GenomeNCBI.html#bacnet">GenomeNCBI</a></div>
<div class="col-last even-row-color">
<div class="block">Object containing an entire genome, created from NCBI RefSeq data<br>
 Data for this genome come from: <it> NCBI.fna (for the sequence) <it> NCBI.gff (for Gene, CDS,
 and ncRNA lists) If it exists some information are added: <it> Rfam.gff (other info) <it> supp
 info in a .txt file <it> supp info about RAST results in a .txt file For EGD-e only: <it>
 Srna.xml are read to add to the genome</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.datamodel.annotation">
<div class="caption"><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequenceNCBI</a> used by <a href="../annotation/package-summary.html">bacnet.datamodel.annotation</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="class-use/GenomeNCBI.html#bacnet.datamodel.annotation">GenomeNCBI</a></div>
<div class="col-last even-row-color">
<div class="block">Object containing an entire genome, created from NCBI RefSeq data<br>
 Data for this genome come from: <it> NCBI.fna (for the sequence) <it> NCBI.gff (for Gene, CDS,
 and ncRNA lists) If it exists some information are added: <it> Rfam.gff (other info) <it> supp
 info in a .txt file <it> supp info about RAST results in a .txt file For EGD-e only: <it>
 Srna.xml are read to add to the genome</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.datamodel.sequence">
<div class="caption"><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequenceNCBI</a> used by <a href="../sequence/package-summary.html">bacnet.datamodel.sequence</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="class-use/GenomeNCBI.html#bacnet.datamodel.sequence">GenomeNCBI</a></div>
<div class="col-last even-row-color">
<div class="block">Object containing an entire genome, created from NCBI RefSeq data<br>
 Data for this genome come from: <it> NCBI.fna (for the sequence) <it> NCBI.gff (for Gene, CDS,
 and ncRNA lists) If it exists some information are added: <it> Rfam.gff (other info) <it> supp
 info in a .txt file <it> supp info about RAST results in a .txt file For EGD-e only: <it>
 Srna.xml are read to add to the genome</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.datamodel.sequenceNCBI">
<div class="caption"><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequenceNCBI</a> used by <a href="package-summary.html">bacnet.datamodel.sequenceNCBI</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="class-use/GenomeNCBI.html#bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></div>
<div class="col-last even-row-color">
<div class="block">Object containing an entire genome, created from NCBI RefSeq data<br>
 Data for this genome come from: <it> NCBI.fna (for the sequence) <it> NCBI.gff (for Gene, CDS,
 and ncRNA lists) If it exists some information are added: <it> Rfam.gff (other info) <it> supp
 info in a .txt file <it> supp info about RAST results in a .txt file For EGD-e only: <it>
 Srna.xml are read to add to the genome</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.scripts.genome">
<div class="caption"><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequenceNCBI</a> used by <a href="../../scripts/genome/package-summary.html">bacnet.scripts.genome</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="class-use/GenomeNCBI.html#bacnet.scripts.genome">GenomeNCBI</a></div>
<div class="col-last even-row-color">
<div class="block">Object containing an entire genome, created from NCBI RefSeq data<br>
 Data for this genome come from: <it> NCBI.fna (for the sequence) <it> NCBI.gff (for Gene, CDS,
 and ncRNA lists) If it exists some information are added: <it> Rfam.gff (other info) <it> supp
 info in a .txt file <it> supp info about RAST results in a .txt file For EGD-e only: <it>
 Srna.xml are read to add to the genome</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="bacnet.scripts.listeriomics.nterm">
<div class="caption"><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequenceNCBI</a> used by <a href="../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="class-use/GenomeNCBI.html#bacnet.scripts.listeriomics.nterm">GenomeNCBI</a></div>
<div class="col-last even-row-color">
<div class="block">Object containing an entire genome, created from NCBI RefSeq data<br>
 Data for this genome come from: <it> NCBI.fna (for the sequence) <it> NCBI.gff (for Gene, CDS,
 and ncRNA lists) If it exists some information are added: <it> Rfam.gff (other info) <it> supp
 info in a .txt file <it> supp info about RAST results in a .txt file For EGD-e only: <it>
 Srna.xml are read to add to the genome</div>
</div>
</div>
</section>
</li>
</ul>
</section>
</main>
</div>
</div>
</body>
</html>
