<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 18:10:05 CEST 2019 -->
<title>C-Index</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="C-Index";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="contentContainer"><a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a><a id="I:C">
<!--   -->
</a>
<h2 class="title">C</h2>
<dl>
<dt><span class="memberNameLink"><a href="../bacnet/utils/MathUtils.html#C_nk(int,int)">C_nk(int, int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/MathUtils.html" title="class in bacnet.utils">MathUtils</a></dt>
<dd>
<div class="block">Calculate the combination C_nk</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/InputStreamProvider.html#CACHE_PROPERTY">CACHE_PROPERTY</a></span> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/InputStreamProvider.html" title="class in org.biojava3.core.util">InputStreamProvider</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ComparisonsCreation.html#calc(bacnet.scripts.core.Comparison)">calc(Comparison)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Run the comparison by calculating statistical parameters given in tests
 ArrayList</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/UNAfold.html#calcEnergy(java.lang.String)">calcEnergy(String)</a></span> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>
<div class="block">From a bunch of stochast folding, generate a list of Free Energies</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/UNAfold.html#calcEnergy(java.lang.String)">calcEnergy(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>
<div class="block">From a bunch of stochast folding, generate a list of Free Energies</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ComparisonsCreation.html#calcFilterAndSave(java.util.ArrayList,java.lang.String,java.lang.String,java.util.ArrayList,boolean,boolean)">calcFilterAndSave(ArrayList&lt;String&gt;, String, String, ArrayList&lt;Filter&gt;, boolean, boolean)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Calculate a comparison, filter the results, and save ExpressionMatrix and
 list of elements up-regulated and down-regulated</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DNASeqCreation.html#calcGeneCNV()">calcGeneCNV()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DNASeqCreation.html" title="class in bacnet.scripts.database">DNASeqCreation</a></dt>
<dd>
<div class="block">For each RNAseq and Sequence element, calculate the mean value of copy number
 variation THIS NEED TO BE RUN AFTER CONVERSION of RNASeq files</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/TilingGeneExprDataBase.html#calcGenExprTilingComparisons(bacnet.datamodel.expdesign.Experiment)">calcGenExprTilingComparisons(Experiment)</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TilingGeneExprDataBase.html" title="class in bacnet.scripts.listeriomics">TilingGeneExprDataBase</a></dt>
<dd>
<div class="block">Calculate GeneExpr and Tiling comparisons files</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/LPE.html#calcP(double%5B%5D,double%5B%5D,org.apache.commons.math.analysis.polynomials.PolynomialSplineFunction,org.apache.commons.math.analysis.polynomials.PolynomialSplineFunction)">calcP(double[], double[], PolynomialSplineFunction, PolynomialSplineFunction)</a></span> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/LPE.html" title="class in bacnet.scripts.core.stat">LPE</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/CodonUsage.html#calculate(java.lang.String)">calculate(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/CodonUsage.html" title="class in bacnet.datamodel.annotation">CodonUsage</a></dt>
<dd>
<div class="block">Calculate Codon usage of a sequence<br>
 Providing <code>codonTable</code> has been loaded first</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/Febit.html#calculateComparison(java.lang.String,java.lang.String)">calculateComparison(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Febit.html" title="class in bacnet.scripts.listeriomics.technology">Febit</a></dt>
<dd>
<div class="block">Calculate Log of Fold Change (mean difference) for each value, and p-value</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/LT.html#calculateComparison(java.lang.String,java.lang.String)">calculateComparison(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/LT.html" title="class in bacnet.scripts.listeriomics.technology">LT</a></dt>
<dd>
<div class="block">Calculate Log of Fold Change (mean difference) for each value, and p-value</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/Macroarray.html#calculateComparison(java.lang.String,java.lang.String)">calculateComparison(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Macroarray.html" title="class in bacnet.scripts.listeriomics.technology">Macroarray</a></dt>
<dd>
<div class="block">Calculate Log of Fold Change (mean difference) for each value, and p-value</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/NimbleGen.html#calculateComparison(java.lang.String,java.lang.String)">calculateComparison(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/NimbleGen.html" title="class in bacnet.scripts.listeriomics.technology">NimbleGen</a></dt>
<dd>
<div class="block">Calculate Log of Fold Change (mean difference) for each value, and p-value</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/Agilent.html#calculateComparison(java.lang.String,java.lang.String,boolean)">calculateComparison(String, String, boolean)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Agilent.html" title="class in bacnet.scripts.listeriomics.technology">Agilent</a></dt>
<dd>
<div class="block">Calculate Log of Fold Change (mean difference) for each value, and p-value</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/UGiessen.html#calculateComparison(java.lang.String,java.lang.String,boolean)">calculateComparison(String, String, boolean)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/UGiessen.html" title="class in bacnet.scripts.listeriomics.technology">UGiessen</a></dt>
<dd>
<div class="block">Calculate Log of Fold Change (mean difference) for each value, and p-value</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ComparisonsCreation.html#calculateCompMatrix(java.util.ArrayList,java.util.ArrayList,boolean,boolean)">calculateCompMatrix(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, boolean, boolean)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Calculate a list of comparisons</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastResult.html#calculateIdentities()">calculateIdentities()</a></span> - Method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#calculateMiddle()">calculateMiddle()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/SDProfile.html#calculateOneSDProfile(bacnet.datamodel.sequence.Genome,int,int,org.biojava3.core.sequence.Strand,java.lang.String)">calculateOneSDProfile(Genome, int, int, Strand, String)</a></span> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/SDProfile.html#calculateSDProfile(bacnet.datamodel.sequence.Genome)">calculateSDProfile(Genome)</a></span> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#calculateSize()">calculateSize()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/StartCodonProfile.html#calculateStartCodonProfile(bacnet.datamodel.sequence.Genome,java.lang.String,int)">calculateStartCodonProfile(Genome, String, int)</a></span> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/StartCodonProfile.html" title="class in bacnet.scripts.genome">StartCodonProfile</a></dt>
<dd>
<div class="block">Create an NGS data where:<br>
 Start codon represent a coverage of 10
 Stop codon represent a coverage of 5
 every other codon a coverage of 0</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/LPE.html#calcVarM(bacnet.datamodel.dataset.ExpressionMatrix,double%5B%5D)">calcVarM(ExpressionMatrix, double[])</a></span> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/LPE.html" title="class in bacnet.scripts.core.stat">LPE</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/LPE.html#calcZ(double%5B%5D,double%5B%5D,org.apache.commons.math.analysis.polynomials.PolynomialSplineFunction,org.apache.commons.math.analysis.polynomials.PolynomialSplineFunction)">calcZ(double[], double[], PolynomialSplineFunction, PolynomialSplineFunction)</a></span> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/LPE.html" title="class in bacnet.scripts.core.stat">LPE</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#calcZoomNumber()">calcZoomNumber()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#calcZoomRatio()">calcZoomRatio()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>
<div class="block">We have MINMIAL_BP_NB * zoomRatio^zoomNumber = totalBP <br>
 So: zoomRatio = (totalBP/MIN_NB_BP)^(1/zoomNumber)</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription"><span class="typeNameLink">CaseInsensitiveCompound</span></a> - Class in <a href="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></dt>
<dd>
<div class="block">Attempts to wrap compounds so it is possible to view them in a case
 insensitive manner</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#%3Cinit%3E(org.biojava3.core.sequence.compound.NucleotideCompound)">CaseInsensitiveCompound(NucleotideCompound)</a></span> - Constructor for class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#%3Cinit%3E(org.biojava3.core.sequence.compound.NucleotideCompound,org.biojava3.core.sequence.compound.NucleotideCompound,org.biojava3.core.sequence.compound.NucleotideCompound)">CaseInsensitiveTriplet(NucleotideCompound, NucleotideCompound, NucleotideCompound)</a></span> - Constructor for class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io"><span class="typeNameLink">CasePreservingProteinSequenceCreator</span></a> - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">A sequence creator which preserves the case of its input string in the user
 collection of the returned ProteinSequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)">CasePreservingProteinSequenceCreator(CompoundSet&lt;AminoAcidCompound&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/CDSComparator.html" title="class in org.biojava3.core.sequence"><span class="typeNameLink">CDSComparator</span></a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/CDSComparator.html#%3Cinit%3E()">CDSComparator()</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSComparator.html" title="class in org.biojava3.core.sequence">CDSComparator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence"><span class="typeNameLink">CDSSequence</span></a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">Represents a exon or coding sequence in a gene.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/CDSSequence.html#%3Cinit%3E(org.biojava3.core.sequence.TranscriptSequence,int,int,int)">CDSSequence(TranscriptSequence, int, int, int)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#centerColumn(double%5B%5D%5B%5D)">centerColumn(double[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Center all the column from the Matrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#centerRow(double%5B%5D%5B%5D)">centerRow(double[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Center all the row from the Matrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#checksum(org.biojava3.core.sequence.template.Sequence)">checksum(Sequence&lt;C&gt;)</a></span> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Performs a simple CRC64 checksum on any given sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/NavigationManagement.html#CHROMO">CHROMO</a></span> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence"><span class="typeNameLink">Chromosome</span></a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#%3Cinit%3E()">Chromosome()</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#%3Cinit%3E(java.lang.String)">Chromosome(String)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">Chromosome(String, CompoundSet&lt;NucleotideCompound&gt;)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader)">Chromosome(SequenceReader&lt;NucleotideCompound&gt;)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">Chromosome(SequenceReader&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.DNAType.html#CHROMOSOME">CHROMOSOME</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence"><span class="typeNameLink">ChromosomeBacteriaSequence</span></a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#%3Cinit%3E()">ChromosomeBacteriaSequence()</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#%3Cinit%3E(java.lang.String)">ChromosomeBacteriaSequence(String)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">ChromosomeBacteriaSequence(String, CompoundSet&lt;NucleotideCompound&gt;)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader)">ChromosomeBacteriaSequence(SequenceReader&lt;NucleotideCompound&gt;)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">ChromosomeBacteriaSequence(SequenceReader&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence"><span class="typeNameLink">ChromosomeSequence</span></a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#%3Cinit%3E()">ChromosomeSequence()</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>
<div class="block">Empty constructor used by tools that need a proper Bean that allows the
 actual sequence data to be set after construction.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#%3Cinit%3E(java.lang.String)">ChromosomeSequence(String)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>
<div class="block">String is king and assume DNA</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">ChromosomeSequence(String, CompoundSet&lt;NucleotideCompound&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>
<div class="block">Allows the creation of a ChromosomeSequence using String for the sequence
 with a custom CompoundSet</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader)">ChromosomeSequence(SequenceReader&lt;NucleotideCompound&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>
<div class="block">Fairly important constructor given the size of a ChromsomeSequence where the
 ProxySequenceReader could load from disk via RandomAccessFile so that the
 sequence doesn't need to be kept in memory.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">ChromosomeSequence(SequenceReader&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>
<div class="block">Allows the creation of a ChromosomeSequence using a ProxyResequenceReader for
 the sequence with a custom CompoundSet</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/HTMLUtils.html#CIRCOS">CIRCOS</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/HTMLUtils.html" title="class in bacnet.utils">HTMLUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Network.html#CIRCOS_BACK_PATH">CIRCOS_BACK_PATH</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SynTView.html#CIRCOS_NAME">CIRCOS_NAME</a></span> - Static variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#CIRCOS_NAME">CIRCOS_NAME</a></span> - Static variable in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome"><span class="typeNameLink">CircularGenomeJPanel</span></a> - Class in <a href="../bacnet/scripts/genome/package-summary.html">bacnet.scripts.genome</a></dt>
<dd>
<div class="block">Circular Genome Panel created for displaying Small RNAs in CGView</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#%3Cinit%3E(int,int,bacnet.datamodel.sequence.Genome,java.lang.String)">CircularGenomeJPanel(int, int, Genome, String)</a></span> - Constructor for class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>
<div class="block">Run circularGenome for Listeriomics Circos network graph</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#%3Cinit%3E(int,int,bacnet.datamodel.sequence.Srna.TypeSrna)">CircularGenomeJPanel(int, int, Srna.TypeSrna)</a></span> - Constructor for class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>
<div class="block">Create the panel for displaying Srna.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#%3Cinit%3E(int,int,bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String)">CircularGenomeJPanel(int, int, GenomeNCBI, String)</a></span> - Constructor for class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/LocationHelper.html#circularLocation(int,int,org.biojava3.core.sequence.Strand,int)">circularLocation(int, int, Strand, int)</a></span> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Converts a location which defines the outer bounds of a circular location and
 splits it into the required portions.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Location.Tools.html#circularLocation(int,int,org.biojava3.core.sequence.Strand,int)">circularLocation(int, int, Strand, int)</a></span> - Static method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.Tools.html" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a></dt>
<dd>
<div class="block">Converts a location which defines the outer bounds of a circular location and
 splits it into the required portions.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.TypeSrna.html#CisReg">CisReg</a></span> - bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.TypeSrna.html" title="enum in bacnet.datamodel.sequence">Srna.TypeSrna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/Equals.html#classEqual(java.lang.Object,java.lang.Object)">classEqual(Object, Object)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Equals.html" title="class in org.biojava3.core.util">Equals</a></dt>
<dd>
<div class="block">This method should be called before beginning any equals methods.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util"><span class="typeNameLink">ClasspathResource</span></a> - Class in <a href="../org/biojava3/core/sequence/io/util/package-summary.html">org.biojava3.core.sequence.io.util</a></dt>
<dd>
<div class="block">This object represents a classpath resource on the local system.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html#%3Cinit%3E(java.lang.String)">ClasspathResource(String)</a></span> - Constructor for class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a></dt>
<dd>
<div class="block">Basic constructor only allowing you to specify where to find the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html#%3Cinit%3E(java.lang.String,boolean)">ClasspathResource(String, boolean)</a></span> - Constructor for class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a></dt>
<dd>
<div class="block">Advanced constructor which allows you to optionally pre-cache the data</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html#%3Cinit%3E(java.lang.String,boolean,boolean)">ClasspathResource(String, boolean, boolean)</a></span> - Constructor for class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a></dt>
<dd>
<div class="block">Advanced constructor which lets you set the preCache variable and to force
 the type of file we are decompressing.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastDatabase.html#cleanDatabase(java.lang.String)">cleanDatabase(String)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">Remove in all genome folders a certain type of file</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#cleanDataTable()">cleanDataTable()</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>
<div class="block">Method to remove the unsued lines in the data files for BUGS and MEXP
 technologies If those lines remain present it causes failures to create the
 curated tables with the global method</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/FileUtils.html#cleanStringFromHex(java.lang.String)">cleanStringFromHex(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Search hexadecimal code in a String indicated by %<br>
 Replace them by they ASCII representation</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#cleanSVGFile()">cleanSVGFile()</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>
<div class="block">Clean SVG file to be able to replace strain name by homolog information<br>
 Names of strain are created in many different "Text table" in the SVG file
 which have to be removed</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#cleanUpDatabase()">cleanUpDatabase()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Cleanup database when closing the website</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/SDProfile.html#cleanUpTempFiles()">cleanUpTempFiles()</a></span> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/FlatFileCache.html#clear()">clear()</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/FlatFileCache.html" title="class in org.biojava3.core.util">FlatFileCache</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/SoftHashMap.html#clear()">clear()</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#clearChromosome()">clearChromosome()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>
<div class="block">Void a Chromosome when Website is quit USEFUL in Eclipse RAP</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#clearGenome()">clearGenome()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Clear current genome from JVM by setting it up to a 4 base genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#clickedElement(org.eclipse.swt.events.MouseEvent)">clickedElement(MouseEvent)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">Get base-pair position of the clicked element</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#clickedElement(org.eclipse.swt.events.MouseEvent)">clickedElement(MouseEvent)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#clone()">clone()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#clone()">clone()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>
<div class="block">Clone the object</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#clone()">clone()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>
<div class="block">Clone the object</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#clone()">clone()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperList.html#clone()">clone()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#clone()">clone()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCPList.html#clone()">clone()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#clone()">clone()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#clone()">clone()</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#clone(double%5B%5D%5B%5D)">clone(double[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#clone(java.lang.String%5B%5D%5B%5D)">clone(String[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#clone(java.util.ArrayList)">clone(ArrayList&lt;String&gt;)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#clone(java.util.TreeMap)">clone(TreeMap&lt;String, Integer&gt;)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimplePoint.html#clonePoint()">clonePoint()</a></span> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Point.html#clonePoint()">clonePoint()</a></span> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a></dt>
<dd>
<div class="block">Returns a copy of this point</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#close()">close()</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/PrettyXMLWriter.html#close()">close()</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/XMLWriter.html#close()">close()</a></span> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Close this XMLWriter, and it's underlying stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#close(java.io.Closeable)">close(Closeable)</a></span> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Closes any Object which implements the interface <code>Closeable</code> and
 sending any error to the logger but not forcing any explicit catching of
 stream errors.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SessionControl.html#closeGenomeViewer(org.eclipse.e4.ui.workbench.modeling.EPartService)">closeGenomeViewer(EPartService)</a></span> - Static method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SessionControl.html" title="class in bacnet.e4.rap">SessionControl</a></dt>
<dd>
<div class="block">Test function for closing a Genomeviewer and check if it is kept in JVM</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/PrettyXMLWriter.html#closeTag(java.lang.String)">closeTag(String)</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/XMLWriter.html#closeTag(java.lang.String)">closeTag(String)</a></span> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Closes an un-qualified element.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/PrettyXMLWriter.html#closeTag(java.lang.String,java.lang.String)">closeTag(String, String)</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/XMLWriter.html#closeTag(java.lang.String,java.lang.String)">closeTag(String, String)</a></span> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Closes an element</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#Clostridiales_NWK">Clostridiales_NWK</a></span> - Static variable in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html#CLUSTALW">CLUSTALW</a></span> - org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html" title="enum in org.biojava3.core.sequence.template">LightweightProfile.StringFormat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html#CM">CM</a></span> - bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/gui/BlastQueryPage.html#cmbFileFormat">cmbFileFormat</a></span> - Variable in class bacnet.scripts.blast.gui.<a href="../bacnet/scripts/blast/gui/BlastQueryPage.html" title="class in bacnet.scripts.blast.gui">BlastQueryPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/CMD.html" title="class in bacnet.utils"><span class="typeNameLink">CMD</span></a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/CMD.html#%3Cinit%3E()">CMD()</a></span> - Constructor for class bacnet.utils.<a href="../bacnet/utils/CMD.html" title="class in bacnet.utils">CMD</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence"><span class="typeNameLink">Codon</span></a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>
<div class="block">Class for managing triplet of nucleotides.<br>
 Allo<s detection of start and stop codons in nucleotide sequences</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Codon.html#%3Cinit%3E()">Codon()</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#%3Cinit%3E(org.biojava3.core.sequence.transcription.Table.CaseInsensitiveTriplet)">Codon(Table.CaseInsensitiveTriplet)</a></span> - Constructor for class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#%3Cinit%3E(org.biojava3.core.sequence.transcription.Table.CaseInsensitiveTriplet,org.biojava3.core.sequence.compound.AminoAcidCompound,boolean,boolean)">Codon(Table.CaseInsensitiveTriplet, AminoAcidCompound, boolean, boolean)</a></span> - Constructor for class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound"><span class="typeNameLink">CodonCompound</span></a> - Class in <a href="../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</a></dt>
<dd>
<div class="block">Define a codon</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#%3Cinit%3E(org.biojava3.core.sequence.compound.NucleotideCompound,org.biojava3.core.sequence.compound.NucleotideCompound,org.biojava3.core.sequence.compound.NucleotideCompound,boolean)">CodonCompound(NucleotideCompound, NucleotideCompound, NucleotideCompound, boolean)</a></span> - Constructor for class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#codonDisplay(java.lang.String)">codonDisplay(String)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Read a sequence and separate the codon by a space <br>
 This method return teh sequence as a StringColor, so codon might be colorized
 after</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/CodonUsage.html" title="class in bacnet.datamodel.annotation"><span class="typeNameLink">CodonUsage</span></a> - Class in <a href="../bacnet/datamodel/annotation/package-summary.html">bacnet.datamodel.annotation</a></dt>
<dd>
<div class="block">Tools for calculating codon Usage of amino acid sequences<br>
 Codon Usage tables came from : http://www.kazusa.or.jp/codon/</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/CodonUsage.html#%3Cinit%3E(java.lang.String)">CodonUsage(String)</a></span> - Constructor for class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/CodonUsage.html" title="class in bacnet.datamodel.annotation">CodonUsage</a></dt>
<dd>
<div class="block">Providing the name of codon Table, it loads the Table and create
 <code>CodonUsage</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#codonUsageEGDe()">codonUsageEGDe()</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views"><span class="typeNameLink">CoExprNetworkView</span></a> - Class in <a href="../bacnet/views/package-summary.html">bacnet.views</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#%3Cinit%3E()">CoExprNetworkView()</a></span> - Constructor for class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.InitNetworkThread.html" title="class in bacnet.views"><span class="typeNameLink">CoExprNetworkView.InitNetworkThread</span></a> - Class in <a href="../bacnet/views/package-summary.html">bacnet.views</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation"><span class="typeNameLink">COGannotation</span></a> - Class in <a href="../bacnet/datamodel/annotation/package-summary.html">bacnet.datamodel.annotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/COGannotation.html#%3Cinit%3E()">COGannotation()</a></span> - Constructor for class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/COGannotation.html#COGMAP">COGMAP</a></span> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/NavigationManagement.html#COL">COL</a></span> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#COL_NUMBER">COL_NUMBER</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/StringColor.html#colorBackground">colorBackground</a></span> - Variable in class bacnet.utils.<a href="../bacnet/utils/StringColor.html" title="class in bacnet.utils">StringColor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/StringColor.html#colorForeground">colorForeground</a></span> - Variable in class bacnet.utils.<a href="../bacnet/utils/StringColor.html" title="class in bacnet.utils">StringColor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/CodonUsage.html#colorizeSequence(java.lang.String)">colorizeSequence(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/CodonUsage.html" title="class in bacnet.datamodel.annotation">CodonUsage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core"><span class="typeNameLink">ColorMapper</span></a> - Class in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell)">ColorMapper(Shell)</a></span> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.TypeMapper.html" title="enum in bacnet.table.core"><span class="typeNameLink">ColorMapper.TypeMapper</span></a> - Enum in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core"><span class="typeNameLink">ColorMapperList</span></a> - Class in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperList.html#%3Cinit%3E()">ColorMapperList()</a></span> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperList.html#%3Cinit%3E(bacnet.datamodel.dataset.ExpressionMatrix,org.eclipse.swt.widgets.Shell)">ColorMapperList(ExpressionMatrix, Shell)</a></span> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>
<div class="block">Create a list of ColorMapper from a matrix<br>
 Each column name corresponding to a TypeMapper will be associated to a
 ColorMapper</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core"><span class="typeNameLink">ColorMapperRCP</span></a> - Class in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell)">ColorMapperRCP(Shell)</a></span> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.Type.html" title="enum in bacnet.table.core"><span class="typeNameLink">ColorMapperRCP.Type</span></a> - Enum in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core"><span class="typeNameLink">ColorMapperRCPList</span></a> - Class in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCPList.html#%3Cinit%3E()">ColorMapperRCPList()</a></span> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCPList.html#%3Cinit%3E(bacnet.datamodel.dataset.ExpressionMatrix,org.eclipse.swt.widgets.Shell)">ColorMapperRCPList(ExpressionMatrix, Shell)</a></span> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizard.html" title="class in bacnet.table.gui"><span class="typeNameLink">ColorMapperWizard</span></a> - Class in <a href="../bacnet/table/gui/package-summary.html">bacnet.table.gui</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/ColorMapperWizard.html#%3Cinit%3E(bacnet.table.core.ColorMapperList,org.eclipse.swt.widgets.Shell)">ColorMapperWizard(ColorMapperList, Shell)</a></span> - Constructor for class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizard.html" title="class in bacnet.table.gui">ColorMapperWizard</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui"><span class="typeNameLink">ColorMapperWizardPage</span></a> - Class in <a href="../bacnet/table/gui/package-summary.html">bacnet.table.gui</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/ColorMapperWizardPage.html#%3Cinit%3E(bacnet.table.core.ColorMapper,org.eclipse.swt.widgets.Shell)">ColorMapperWizardPage(ColorMapper, Shell)</a></span> - Constructor for class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/ColorMapperWizardPage.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell)">ColorMapperWizardPage(Shell)</a></span> - Constructor for class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></dt>
<dd>
<div class="block">Create the wizard.</div>
</dd>
<dt><a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core"><span class="typeNameLink">ColorSWT</span></a> - Class in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorSWT.html#%3Cinit%3E(int,int,int)">ColorSWT(int, int, int)</a></span> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorSWT.html#%3Cinit%3E(org.eclipse.swt.graphics.Color)">ColorSWT(Color)</a></span> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html" title="class in bacnet.scripts.core.vennDiagram"><span class="typeNameLink">CombinationGenerator</span></a> - Class in <a href="../bacnet/scripts/core/vennDiagram/package-summary.html">bacnet.scripts.core.vennDiagram</a></dt>
<dd>
<div class="block">Class for finding every combination of a set of integer</div>
</dd>
<dt><a href="../bacnet/utils/CombinationGenerator.html" title="class in bacnet.utils"><span class="typeNameLink">CombinationGenerator</span></a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html#%3Cinit%3E(int,int)">CombinationGenerator(int, int)</a></span> - Constructor for class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html" title="class in bacnet.scripts.core.vennDiagram">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/CombinationGenerator.html#%3Cinit%3E(int,int)">CombinationGenerator(int, int)</a></span> - Constructor for class bacnet.utils.<a href="../bacnet/utils/CombinationGenerator.html" title="class in bacnet.utils">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#combinator(java.util.ArrayList,int)">combinator(ArrayList&lt;String&gt;, int)</a></span> - Static method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>
<div class="block">Create all combinations of k elements of list1 <br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/UGiessen.html#combineMatrices(java.lang.String,java.lang.String)">combineMatrices(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/UGiessen.html" title="class in bacnet.scripts.listeriomics.technology">UGiessen</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/NavigationManagement.html#COMBO">COMBO</a></span> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html#COMMA">COMMA</a></span> - bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html" title="enum in bacnet.scripts.blast">BlastOutput.BlastOutputTYPE</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#compare(bacnet.datamodel.dataset.ExpressionData,boolean)">compare(ExpressionData, boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Create Tiling data from the comparison of this one and data2</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#compare(bacnet.datamodel.dataset.GeneExpression,boolean)">compare(GeneExpression, boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Create GeneExpression data from the comparison of this one and data2</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NGS.html#compare(bacnet.datamodel.dataset.NGS,boolean)">compare(NGS, boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>
<div class="block">Create Tiling data from the comparison of this one and data2</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#compare(bacnet.datamodel.dataset.Tiling,boolean)">compare(Tiling, boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Create Tiling data from the comparison of this one and data2</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#compare(bacnet.datamodel.expdesign.BioCondition,boolean)">compare(BioCondition, boolean)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Create a BioCondition containing the different data for comparing this
 BioCondition to bioCond2</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastResult.BlastResultCompare.html#compare(bacnet.scripts.blast.BlastResult,bacnet.scripts.blast.BlastResult)">compare(BlastResult, BlastResult)</a></span> - Method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.BlastResultCompare.html" title="class in bacnet.scripts.blast">BlastResult.BlastResultCompare</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/CDSComparator.html#compare(org.biojava3.core.sequence.CDSSequence,org.biojava3.core.sequence.CDSSequence)">compare(CDSSequence, CDSSequence)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSComparator.html" title="class in org.biojava3.core.sequence">CDSComparator</a></dt>
<dd>
<div class="block">Used to sort two CDSSequences where Negative Strand makes it tough</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ExonComparator.html#compare(org.biojava3.core.sequence.ExonSequence,org.biojava3.core.sequence.ExonSequence)">compare(ExonSequence, ExonSequence)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ExonComparator.html" title="class in org.biojava3.core.sequence">ExonComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/SequenceComparator.html#compare(org.biojava3.core.sequence.template.AbstractSequence,org.biojava3.core.sequence.template.AbstractSequence)">compare(AbstractSequence&lt;?&gt;, AbstractSequence&lt;?&gt;)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceComparator.html" title="class in org.biojava3.core.sequence">SequenceComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/BioConditionComparator.html#compare(org.eclipse.jface.viewers.Viewer,java.lang.Object,java.lang.Object)">compare(Viewer, Object, Object)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/TableViewerComparator.html#compare(org.eclipse.jface.viewers.Viewer,java.lang.Object,java.lang.Object)">compare(Viewer, Object, Object)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/TableViewerComparator.html" title="class in bacnet.table.core">TableViewerComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermUtils.html#compareNTermData(java.lang.String,java.lang.String)">compareNTermData(String, String)</a></span> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>
<div class="block">Return the list of NTerm common to both experiment<br>
 NTerm are equal if:<br>
 same peptide sequence
 same modification</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#compareTo(org.biojava3.core.sequence.location.template.Point)">compareTo(Point)</a></span> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimplePoint.html#compareTo(org.biojava3.core.sequence.location.template.Point)">compareTo(Point)</a></span> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core"><span class="typeNameLink">Comparison</span></a> - Class in <a href="../bacnet/scripts/core/package-summary.html">bacnet.scripts.core</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#%3Cinit%3E()">Comparison()</a></span> - Constructor for class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#%3Cinit%3E(java.util.ArrayList,java.util.ArrayList,java.util.ArrayList,boolean)">Comparison(ArrayList&lt;StatTest&gt;, ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, boolean)</a></span> - Constructor for class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>
<div class="block">Create a Comparison object used in Statistical comparison case</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core"><span class="typeNameLink">ComparisonAtlas</span></a> - Class in <a href="../bacnet/expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a></dt>
<dd>
<div class="block">A ComparisonAtlas is related to a Comparison object and gives the list of
 genome elements over, under and not differentially expressed</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#%3Cinit%3E()">ComparisonAtlas()</a></span> - Constructor for class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#%3Cinit%3E(bacnet.datamodel.dataset.ExpressionMatrix,java.util.ArrayList,java.lang.String,bacnet.utils.Filter)">ComparisonAtlas(ExpressionMatrix, ArrayList&lt;String&gt;, String, Filter)</a></span> - Constructor for class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>
<div class="block">For a specific biological condition comparison update the list of genome
 elements in which the genome element is over-expressed, under-expressed or
 not differently expressed<br>
 The update will be dependent of the value of the input filter = cutoff for
 LogFC and p-value</div>
</dd>
<dt><a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database"><span class="typeNameLink">ComparisonsCreation</span></a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ComparisonsCreation.html#%3Cinit%3E()">ComparisonsCreation()</a></span> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/SubCellCompartment.html#COMPARTMENT_NAMES">COMPARTMENT_NAMES</a></span> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.html" title="class in bacnet.datamodel.annotation">SubCellCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/ComplementCompound.html" title="interface in org.biojava3.core.sequence.template"><span class="typeNameLink">ComplementCompound</span></a> - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava3.core.sequence.views"><span class="typeNameLink">ComplementSequenceView</span></a>&lt;<a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html" title="type parameter in ComplementSequenceView">C</a> extends <a href="../org/biojava3/core/sequence/template/ComplementCompound.html" title="interface in org.biojava3.core.sequence.template">ComplementCompound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</a></dt>
<dd>
<div class="block">For a given sequence this class will create a view over the top of it and for
 every request the code will return the complement of the underlying base e.g.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence)">ComplementSequenceView(Sequence&lt;C&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava3.core.sequence.views">ComplementSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/LocationHelper.html#completeCircularPasses(int,int)">completeCircularPasses(int, int)</a></span> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Works in a similar way to modulateCircularLocation but returns the number of
 complete passes over a Sequence length a circular location makes i.e. if we
 have a sequence of length 10 and the location 3..52 we make 4 complete passes
 through the genome to go from position 3 to position 52.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Location.Tools.html#completeCircularPasses(int,int)">completeCircularPasses(int, int)</a></span> - Static method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.Tools.html" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a></dt>
<dd>
<div class="block">Works in a similar way to modulateCircularLocation but returns the number of
 complete passes over a Sequence length a circular location makes i.e. if we
 have a sequence of length 10 and the location 3..52 we make 4 complete passes
 through the genome to go from position 3 to position 52.</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/CompositeNTermInfo.html" title="class in bacnet.genomeBrowser"><span class="typeNameLink">CompositeNTermInfo</span></a> - Class in <a href="../bacnet/genomeBrowser/package-summary.html">bacnet.genomeBrowser</a></dt>
<dd>
<div class="block">NTerm composite which is displayed on the right of the GenomeViewer.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CompositeNTermInfo.html#%3Cinit%3E(org.eclipse.swt.widgets.Composite,int)">CompositeNTermInfo(Composite, int)</a></span> - Constructor for class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CompositeNTermInfo.html" title="class in bacnet.genomeBrowser">CompositeNTermInfo</a></dt>
<dd>
<div class="block">Initiate a CompositeNTermInfo</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template"><span class="typeNameLink">Compound</span></a> - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/CompoundColorization.html" title="class in bacnet.datamodel.sequence"><span class="typeNameLink">CompoundColorization</span></a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/CompoundColorization.html#%3Cinit%3E()">CompoundColorization()</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/CompoundColorization.html" title="class in bacnet.datamodel.sequence">CompoundColorization</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/exceptions/CompoundNotFoundError.html" title="class in org.biojava3.core.exceptions"><span class="typeNameLink">CompoundNotFoundError</span></a> - Error in <a href="../org/biojava3/core/exceptions/package-summary.html">org.biojava3.core.exceptions</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/exceptions/CompoundNotFoundError.html#%3Cinit%3E(java.lang.CharSequence)">CompoundNotFoundError(CharSequence)</a></span> - Constructor for error org.biojava3.core.exceptions.<a href="../org/biojava3/core/exceptions/CompoundNotFoundError.html" title="class in org.biojava3.core.exceptions">CompoundNotFoundError</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#compoundsEqual(C,C)">compoundsEqual(C, C)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#compoundsEquivalent(C,C)">compoundsEquivalent(C, C)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/CompoundSet.html#compoundsEquivalent(C,C)">compoundsEquivalent(C, C)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#compoundsEquivalent(org.biojava3.core.sequence.compound.AminoAcidCompound,org.biojava3.core.sequence.compound.AminoAcidCompound)">compoundsEquivalent(AminoAcidCompound, AminoAcidCompound)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template"><span class="typeNameLink">CompoundSet</span></a>&lt;<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="type parameter in CompoundSet">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#compoundToInt(org.biojava3.core.sequence.compound.NucleotideCompound)">compoundToInt(NucleotideCompound)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundTranslator.html" title="interface in org.biojava3.core.sequence.template"><span class="typeNameLink">CompoundTranslator</span></a>&lt;<a href="../org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">F</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>,&#8203;<a href="../org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">T</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#compute()">compute()</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#computeTotalList()">computeTotalList()</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util"><span class="typeNameLink">ConcurrencyTools</span></a> - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>
<div class="block">Static utility to easily share a thread pool for concurrent/parallel/lazy
 execution.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/BasicApplication.html#configure(org.eclipse.rap.rwt.application.Application)">configure(Application)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BasicApplication.html" title="class in bacnet.e4.rap">BasicApplication</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/setup/BasicApplication.html#configure(org.eclipse.rap.rwt.application.Application)">configure(Application)</a></span> - Method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/BasicApplication.html" title="class in bacnet.e4.rap.setup">BasicApplication</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/LocationHelper.html#consistentAccessions(java.util.List)">consistentAccessions(List&lt;Location&gt;)</a></span> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Scans a list of locations and returns true if all the given locations are
 linked to the same sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#containProbe(java.lang.String)">containProbe(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Check if TreeMap<String, Integer> probes contains a probe</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#contains(org.biojava3.genome.parsers.gff.Location)">contains(Location)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Check if this location contains the other.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermFilter.html#containStopCodon(bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Sequence,int%5B%5D)">containStopCodon(NTerm, Sequence, int[])</a></span> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>
<div class="block">Return true if it contains a StopCodon between the peptide and the overlaping
 sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#containTranscriptomes()">containTranscriptomes()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.DNAType.html#CONTIG">CONTIG</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NGS.html#convert(bacnet.datamodel.dataset.NGS,boolean)">convert(NGS, boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#convert(bacnet.datamodel.expdesign.Experiment)">convert(Experiment)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Read all GeneExpr data and save in GeneExpressionStreaming format</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#convert(bacnet.datamodel.expdesign.Experiment)">convert(Experiment)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Read Tiling data in Experiment and save in TilingDataStreaming format</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#convert(bacnet.datamodel.sequence.Srna)">convert(Srna)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Convert an Srna into a DNASequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#convertAll()">convertAll()</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>
<div class="block">Read all <code>Signature</code> in text files, and serialize them in
 Database.getSIGNATURES()\*<br>
 Save an array with the HashMap between Signature name and ID</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/GenomesCreation.html#convertAll()">convertAll()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Convert all genomes available in the database</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/NGSCreation.html#convertCoverageFiles(bacnet.datamodel.expdesign.Experiment,boolean)">convertCoverageFiles(Experiment, boolean)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/NGSCreation.html" title="class in bacnet.scripts.database">NGSCreation</a></dt>
<dd>
<div class="block">Go through all RNASeq data found in the list
 <code>BioCondition.getRNASeqBioConditions()</code> and convert all of them</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#convertData(bacnet.datamodel.dataset.ExpressionMatrix)">convertData(ExpressionMatrix)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Convert data from an ExpressionMatrix to a GeneExpression</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#convertData(bacnet.datamodel.dataset.ExpressionMatrix)">convertData(ExpressionMatrix)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#convertDNASequenceGene(org.biojava3.core.sequence.DNASequence)">convertDNASequenceGene(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>
<div class="block">From a DNASequence make a Gene object</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#convertDNASequenceGeneral(org.biojava3.core.sequence.DNASequence)">convertDNASequenceGeneral(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>
<div class="block">From a DNASequence object, create Sequence<br>
 Mainly used in GenomeConversion methods</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#convertDNASequenceNcRNA(org.biojava3.core.sequence.DNASequence)">convertDNASequenceNcRNA(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>
<div class="block">From a DNASequence make a Gene object</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Operon.html#convertDNASequenceOperon(java.lang.String%5B%5D,bacnet.datamodel.sequenceNCBI.GenomeNCBI)">convertDNASequenceOperon(String[], GenomeNCBI)</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></dt>
<dd>
<div class="block">Take row from Annot_sup table and creat an Operon Object from it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#convertDNASequenceOperon(java.lang.String%5B%5D,bacnet.datamodel.sequenceNCBI.GenomeNCBI)">convertDNASequenceOperon(String[], GenomeNCBI)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/FileUtils.html#convertHexToString(java.lang.String)">convertHexToString(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Convert an hexadecimal ASCII to a String</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastOutput.html#convertOuput(java.lang.String,java.lang.String,boolean,bacnet.scripts.blast.BlastOutput.BlastOutputTYPE)">convertOuput(String, String, boolean, BlastOutput.BlastOutputTYPE)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.html" title="class in bacnet.scripts.blast">BlastOutput</a></dt>
<dd>
<div class="block">Convert asn given by blastResult into out using BlastOutputType</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ProteomicsCreation.html#convertProteomicsData(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">convertProteomicsData(Experiment, Genome)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>
<div class="block">Convert all ExpressionMatrix found for Proteomic datasets and save them in
 Streaming folder</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/StringManipulationHelper.html#convertStreamToString(java.io.InputStream)">convertStreamToString(InputStream)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/StringManipulationHelper.html" title="class in org.biojava3.core.util">StringManipulationHelper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#convertToVennMasterDataList()">convertToVennMasterDataList()</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NGS.html#convertWigFile(bacnet.datamodel.dataset.NGS,boolean)">convertWigFile(NGS, boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>
<div class="block">WIG file are common RNASeq data<br>
 Calculate first the MEDIAN value of the different replicates ad then log2
 transform and save the NGS data Two columns: <br>
 one for the base pair position
 one for the coverage value</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#convertXtoBP(int)">convertXtoBP(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">Convert position in the display to a position in the genome<br>
 (inverse function of basepairToXimage(int bpIndex))<br>
 If MassSpec data has not been register do it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#convertXtoBP(int)">convertXtoBP(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>
<div class="block">Convert position in the display to a position in the genome<br>
 (inverse function of basepairToXimage(int bpIndex))</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#copy(java.io.InputStream,java.io.OutputStream)">copy(InputStream, OutputStream)</a></span> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Moves the bytes from input to output using a 4KB byte array.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/FileUtils.html#copy(java.lang.String,java.lang.String)">copy(String, String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">copy file with path=input to file with path=output</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#copyColumnAt(java.lang.String,double%5B%5D)">copyColumnAt(String, double[])</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#copyRowAt(java.lang.String,double%5B%5D)">copyRowAt(String, double[])</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#copyTranscriptomicsTable()">copyTranscriptomicsTable()</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>
<div class="block">Copy the transcriptomics curated table in the NormData>ExprMatrix folder</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#CORR">CORR</a></span> - bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.TypeMapper.html#CORR">CORR</a></span> - bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.TypeMapper.html" title="enum in bacnet.table.core">ColorMapper.TypeMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.Type.html#CORR">CORR</a></span> - bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.Type.html" title="enum in bacnet.table.core">ColorMapperRCP.Type</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Network.html#CORR_CUTOFF">CORR_CUTOFF</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/SDProfile.html#correctResults(java.lang.String)">correctResults(String)</a></span> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>
<div class="block">Data from EGDe SD profile are corrupted need to correct them<br>
 Need to inverse result of values after the 1710999 base pair<br>
 <br>
 
 IMPORTANT: Run only one time 13/01/2014</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#correspondingMassSpecData(bacnet.datamodel.proteomics.NTerm)">correspondingMassSpecData(NTerm)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>
<div class="block">Return the MassSpecData where an Nterm can be found</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#countAT(org.biojava3.core.sequence.template.Sequence)">countAT(Sequence&lt;NucleotideCompound&gt;)</a></span> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Returns the count of AT in the given sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#countCompounds(C...)">countCompounds(C...)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#countCompounds(C...)">countCompounds(C...)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#countCompounds(C...)">countCompounds(C...)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#countCompounds(C...)">countCompounds(C...)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#countCompounds(C...)">countCompounds(C...)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Counts the number of times a compound appears in this sequence store</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#countCompounds(C...)">countCompounds(C...)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#countCompounds(C...)">countCompounds(C...)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Delegates to
 <code>SequenceMixin#countCompounds(org.biojava3.core.sequence.template.Sequence, C[])</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#countCompounds(C...)">countCompounds(C...)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Sequence.html#countCompounds(C...)">countCompounds(C...)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Returns the number of times we found a compound in the Sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#countCompounds(C...)">countCompounds(C...)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#countCompounds(org.biojava3.core.sequence.template.Sequence,C...)">countCompounds(Sequence&lt;C&gt;, C...)</a></span> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">For the given vargs of compounds this method counts the number of times those
 compounds appear in the given sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#countGC(org.biojava3.core.sequence.template.Sequence)">countGC(Sequence&lt;NucleotideCompound&gt;)</a></span> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Returns the count of GC in the given sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermUtils.html#countRTerminal(java.lang.String)">countRTerminal(String)</a></span> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>
<div class="block">Count the number of peptides with an R (arginine), and finishing with an R</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#covariance(double%5B%5D,double%5B%5D)">covariance(double[], double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Calculate the covariance of x * y</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#covarianceMatrix(double%5B%5D%5B%5D)">covarianceMatrix(double[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Calcul of the covariance Matrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html#CP">CP</a></span> - bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/CRC64Checksum.html" title="class in org.biojava3.core.util"><span class="typeNameLink">CRC64Checksum</span></a> - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>
<div class="block">Utility class that calculates a CRC64 checksum on a stream of bytes.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/CRC64Checksum.html#%3Cinit%3E()">CRC64Checksum()</a></span> - Constructor for class org.biojava3.core.util.<a href="../org/biojava3/core/util/CRC64Checksum.html" title="class in org.biojava3.core.util">CRC64Checksum</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/BacnetE4PointFactory.html#create()">create()</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BacnetE4PointFactory.html" title="class in bacnet.e4.rap">BacnetE4PointFactory</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/setup/BacnetE4PointFactory.html#create()">create()</a></span> - Method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/BacnetE4PointFactory.html" title="class in bacnet.e4.rap.setup">BacnetE4PointFactory</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#createAll(bacnet.datamodel.expdesign.Experiment)">createAll(Experiment)</a></span> - Static method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>
<div class="block">For each comparison create a <code>Comparison</code> object containing the
 list of all differentially expressed genes, sRNAs, asRNAs, and cisRegs<br>
 <br>
 
 WANRING: OrganizeDatabase.organizeComparisonsData(exp) as to be run first!!!</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#createAll(bacnet.datamodel.expdesign.Experiment)">createAll(Experiment)</a></span> - Static method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>
<div class="block">Read all <code>Comparison</code> results and extract which gene is over and
 under expressed in it<br>
 For each gene, sRNA,asRNA, and cisReg, create a list of all
 <code>BioCondition</code> where it is over, under, and not differentially
 expressed<br>
 <br>
 
 WANRING: OrganizeDatabase.organizeComparisonsData(exp) as to be run first!!!</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ComparisonsCreation.html#createAllCompMatrix(java.util.ArrayList)">createAllCompMatrix(ArrayList&lt;String&gt;)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Go through all comparisons and create for each <code>Tiling</code> and
 <code>GenExpression</code> different matrices giving the number of genes,
 Srna and asRNA overexpressed and underexpressed</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#createAnnotation()">createAnnotation()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Create an <code>Annotation</code> Table from the list of <code>Nterm</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html#createAnnotation(bacnet.datamodel.sequence.Genome,java.lang.String)">createAnnotation(Genome, String)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversion</a></dt>
<dd>
<div class="block">Create a Table with all elements of the Genome, which will be used for
 element search and display on the GenomeViewer</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#createArrayExpressTable()">createArrayExpressTable()</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>
<div class="block">Create a new txt file containing all the accession data for every file to
 download</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#createArrayTable()">createArrayTable()</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>
<div class="block">Create a txt file giving the name of each array technology</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#createASrnaTSStable()">createASrnaTSStable()</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>
<div class="block">Run different function to curate the data given by Omri 15/11/2012</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#createBioCondition(java.lang.String)">createBioCondition(String)</a></span> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">Create a BioCondition containing only one Nterm</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html#createBioConditionFigure(boolean)">createBioConditionFigure(boolean)</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html" title="class in bacnet.scripts.listeriomics">SummaryListeriomics</a></dt>
<dd>
<div class="block">Create the different element for the figure representing the BioCondition</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#createBlastCommands(java.lang.String,double)">createBlastCommands(String, double)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Launch blast commands creation</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#createBlastCommands(java.lang.String,java.lang.String,java.lang.String,java.util.ArrayList,double)">createBlastCommands(String, String, String, ArrayList&lt;String&gt;, double)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Create .bat files with the command lines for each blast.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#createBlastDatabases(java.lang.String,java.lang.String%5B%5D)">createBlastDatabases(String, String[])</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Create blast database for each genome contained in genomesInput</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/Blast.html#createBlastDatabases(java.lang.String,java.lang.String%5B%5D,boolean,boolean)">createBlastDatabases(String, String[], boolean, boolean)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>
<div class="block">Create blast database for each genome contained in GenomeFolder
 
 !!!!!!!!!!</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#createBlastDB()">createBlastDB()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>
<div class="block">Create blastP databases</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#createBlastDB()">createBlastDB()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Create blastP databases</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastDatabase.html#createBlastDB(java.lang.String)">createBlastDB(String)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">Create blastP databases</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#createBlastSmallORFDB()">createBlastSmallORFDB()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>
<div class="block">Create blastP databases</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastDatabase.html#createBlastSmallORFDB(java.lang.String)">createBlastSmallORFDB(String)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">Create blastP databases</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/COGannotation.html#createCOGNumberEGDe()">createCOGNumberEGDe()</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>
<div class="block">Return a map between cogID and the number of genes associated to this cog in
 EGD-e</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#createCommandFileGeneral()">createCommandFileGeneral()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Creation of the general command file that will be used to create the ones for
 each blast</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#createComparisonsTable()">createComparisonsTable()</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>
<div class="block">Use the BioConditions and the curated sdrf files from ArrayExpressRawData
 folders to create a Comparison table</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#createConservationTables()">createConservationTables()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Read <code>Srna.PATH_CONSERVATION</code> table and display it in HeatMapView
 <br>
 Headers are reorganized by their phylogeny positions</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#createConservationTables10403S()">createConservationTables10403S()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Read sRNAConservation10403S.txt table and display it in HeatMapView <br>
 Headers are reorganized by their phylogeny positions</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/gui/BlastGenomeSelectionPage.html#createControl(org.eclipse.swt.widgets.Composite)">createControl(Composite)</a></span> - Method in class bacnet.scripts.blast.gui.<a href="../bacnet/scripts/blast/gui/BlastGenomeSelectionPage.html" title="class in bacnet.scripts.blast.gui">BlastGenomeSelectionPage</a></dt>
<dd>
<div class="block">Create contents of the wizard.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/gui/BlastParametersPage.html#createControl(org.eclipse.swt.widgets.Composite)">createControl(Composite)</a></span> - Method in class bacnet.scripts.blast.gui.<a href="../bacnet/scripts/blast/gui/BlastParametersPage.html" title="class in bacnet.scripts.blast.gui">BlastParametersPage</a></dt>
<dd>
<div class="block">Create contents of the wizard.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/gui/BlastQueryPage.html#createControl(org.eclipse.swt.widgets.Composite)">createControl(Composite)</a></span> - Method in class bacnet.scripts.blast.gui.<a href="../bacnet/scripts/blast/gui/BlastQueryPage.html" title="class in bacnet.scripts.blast.gui">BlastQueryPage</a></dt>
<dd>
<div class="block">Create contents of the wizard.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/ColorMapperWizardPage.html#createControl(org.eclipse.swt.widgets.Composite)">createControl(Composite)</a></span> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></dt>
<dd>
<div class="block">Create contents of the wizard.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/setup/SetupPart.html#createControls(org.eclipse.swt.widgets.Composite)">createControls(Composite)</a></span> - Method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SetupPart.html" title="class in bacnet.e4.rap.setup">SetupPart</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/TestPart.html#createControls(org.eclipse.swt.widgets.Composite)">createControls(Composite)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/TestPart.html" title="class in bacnet.e4.rap">TestPart</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView.html#createControls(org.eclipse.swt.widgets.Composite)">createControls(Composite)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView2.html#createControls(org.eclipse.swt.widgets.Composite)">createControls(Composite)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html#createDatabase()">createDatabase()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html" title="class in bacnet.scripts.listeriomics.srna">SmallPeptidesSearch</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span></div>
<div class="block">Create database file which contains all genome available <br>
 
 This genome list contains all the genomes from GenomeFolder and all the
 genome contained in GenomePhylogeny.PATH_TABLE (=All bacteria phylogeny)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html#createDatabase()">createDatabase()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html" title="class in bacnet.scripts.listeriomics.srna">Srna10403SOliver</a></dt>
<dd>
<div class="block">Create database file which contains all genome available <br>
 It will be used by Blast afterwards<br>
 <br>
 
 BE CAREFULL THAT THESE GENOMES HAVE BEEN ESTABLISHED FOR BLAST SEARCH
 (=database is created through Blast.createBlastDatabases();)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#createDatabaseForORF()">createDatabaseForORF()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>
<div class="block">Create a fasta file with all proteins from the different chromosomes
 Using faa file create blast
 Verify by looking at all file size</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastDatabase.html#createDatabaseForORF(java.lang.String)">createDatabaseForORF(String)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">Create a fasta file with all proteins from the different chromosomes
 Using faa file create blast
 Verify by looking at all file size</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastDatabase.html#createDatabaseForSmallORF()">createDatabaseForSmallORF()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">translate every genomes on the 6 frames
 Create a fasta file with all proteins from the different chromosomes
 Using faa file create blast
 Verify by looking at all file size</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#createDatabaseForSmallORF()">createDatabaseForSmallORF()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>
<div class="block">translate every genomes on the 6 frames
 Create a fasta file with all proteins from the different chromosomes
 Using faa file create blast
 Verify by looking at all file size</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/ProteinTools.html#createDataBases()">createDataBases()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Create:<br>
 Protein DB
 UTR DB
 fusion Protein DB and Protein DB
 Genome DB
 TIS DB</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#createDataBases()">createDataBases()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Create:<br>
 Protein DB
 UTR DB
 fusion Protein DB and Protein DB
 Genome DB
 TIS DB</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html#createDialogArea(org.eclipse.swt.widgets.Composite)">createDialogArea(Composite)</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html" title="class in bacnet.expressionAtlas.core">SelectGenomeElementDialog</a></dt>
<dd>
<div class="block">Create contents of the wizard.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/GRNormalization.html#createEdgeWT()">createEdgeWT()</a></span> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/GRNormalization.html" title="class in bacnet.scripts.core.normalization">GRNormalization</a></dt>
<dd>
<div class="block">By copying GENERAL_WT_DATA create GENERAL_WT_NAME tiling data</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/EGDeWTdata.html#createEGDeAverage()">createEGDeAverage()</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/EGDeWTdata.html" title="class in bacnet.datamodel.dataset">EGDeWTdata</a></dt>
<dd>
<div class="block">From the list of Wild Type data given by <code>EGDeWTdata.wildTypes</code>
 calculate the mean and save it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/GenomesCreation.html#createEGDeGenomeElementList()">createEGDeGenomeElementList()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Create a list of all Genes, all ASrnas, and all CisRegs of EGD-e<br>
 Srnas list is already created in EGDe_SRNA_PATH</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/LPE.html#createEstimates(java.util.TreeMap)">createEstimates(TreeMap&lt;String, GeneExpression&gt;)</a></span> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/LPE.html" title="class in bacnet.scripts.core.stat">LPE</a></dt>
<dd>
<div class="block">For each comparison calculate the LPE estimate, by creating a matrix of all
 expression values and estimates LPE using each of these matrices</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#createExpressionMatrix(bacnet.datamodel.sequence.Genome)">createExpressionMatrix(Genome)</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>
<div class="block">Combine all Transcriptomes describe in <code>Expression</code> data, and
 return the combine <code>ExpressionMatrix</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#createFAA()">createFAA()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Create a FAA file in the BLASTDB folder for each genome.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#createFastaAA()">createFastaAA()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>
<div class="block">Create blastP databases</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastDatabase.html#createFastaAA(java.lang.String)">createFastaAA(String)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">Create blastP databases</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastDatabase.html#createFastaAASmallORF()">createFastaAASmallORF()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">Extract result and create fasta files</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#createFastaAASmallORF()">createFastaAASmallORF()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>
<div class="block">Extract result and create fasta files</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/GRNormalization.html#createFilecatsTable(bacnet.datamodel.expdesign.BioCondition,java.lang.String,boolean)">createFilecatsTable(BioCondition, String, boolean)</a></span> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/GRNormalization.html" title="class in bacnet.scripts.core.normalization">GRNormalization</a></dt>
<dd>
<div class="block">Create filecats = list of data in a table, used by the R script to run Tiling
 normalization</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html#createFinalList()">createFinalList()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html" title="class in bacnet.scripts.listeriomics.srna">SrnaFinalList</a></dt>
<dd>
<div class="block">Read the different :list from different publication and combine all of them
 into an Srna final list<br>
 Every table should be found in SrnaTables.PATH</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html#createFolderCopyFNA(java.lang.String,java.lang.String)">createFolderCopyFNA(String, String)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversion</a></dt>
<dd>
<div class="block">Create a folder for each Genome, and copy all .fna in it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html#createFoldingFigures()">createFoldingFigures()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html" title="class in bacnet.scripts.listeriomics.srna">SrnaFinalList</a></dt>
<dd>
<div class="block">Run UNAFold for each sRNA to calculate folding<br>
 Copy each .png image from UNAFold folder to
 <code>Srna.PATH_SEC_STRUCTURE</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/GBKReader.html#createGenBankFile(bacnet.datamodel.sequence.Genome,java.lang.String)">createGenBankFile(Genome, String)</a></span> - Static method in class bacnet.reader.<a href="../bacnet/reader/GBKReader.html" title="class in bacnet.reader">GBKReader</a></dt>
<dd>
<div class="block">Create GenBank file with first chromosome<br>
 TO BE FIXED for adding other chromosomes</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#createGeneHashMap()">createGeneHashMap()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Regroup NTerm using genes of EGD-e<br>
 Each EGD-e gene will be associated to a list of NTerm overlapping it<br>
 Fill: HashMap<String, ArrayList<NTerm>>() aTISMap, HashMap<String,
 ArrayList<NTerm>>() genes, HashMap<String, ArrayList<TypeModif>>()
 geneModification</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/ProteinTools.html#createGenomeDB()">createGenomeDB()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">lists of small peptides contained in the genome of EGDe on 6frames.<br>
 The small peptides include in the list are the one only of size more than
 5aa, and surrounded by at least one R.<br>
 example: Start - peptide - R<br>
 Stop - peptide - R<br>
 <br>
 Element like: Start - peptide - Stop are rejected from the analysis<br>
 <br>
 frame 1 = EGDe genome, ID=genome
 frame 2 = EGDe genome 1bp on the left, ID=genomeM
 frame 3 = EGDe genome 1bp on the right, ID=genomeP
 frame 4 = EGDe reverse complement genome, ID=genomeCompl
 frame 5 = EGDe reverse complement genome 1bp on the left, ID=genomeComplM
 frame 6 = EGDe reverse complement genome 1bp on the right,
 ID=genomeComplP <br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#createGenomeDB()">createGenomeDB()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">lists of small peptides contained in the genome of EGDe on 6frames.<br>
 The small peptides include in the list are the one only of size more than
 5aa, and surrounded by at least one R.<br>
 example: Start - peptide - R<br>
 Stop - peptide - R<br>
 <br>
 Element like: Start - peptide - Stop are rejected from the analysis<br>
 <br>
 frame 1 = EGDe genome, ID=genome
 frame 2 = EGDe genome 1bp on the left, ID=genomeM
 frame 3 = EGDe genome 1bp on the right, ID=genomeP
 frame 4 = EGDe reverse complement genome, ID=genomeCompl
 frame 5 = EGDe reverse complement genome 1bp on the left, ID=genomeComplM
 frame 6 = EGDe reverse complement genome 1bp on the right,
 ID=genomeComplP <br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/GenomesCreation.html#createGenomeTable()">createGenomeTable()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Create Table 1 with the list of Genomes and different infos<br>
 Save in: Project.getPath()+"Table 1 - Genomes.txt"</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html#createGenomeTable()">createGenomeTable()</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html" title="class in bacnet.scripts.listeriomics">SummaryListeriomics</a></dt>
<dd>
<div class="block">Create Table 1 for Listeriomics paper with the list of Genomes and different infos<br>
 Save in: Project.getPath()+"Table 1 - Genomes.txt"</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#createHashMap()">createHashMap()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Regroup in different HashMap the NTerm<br>
 Fill:<br>
 HashMap<String, ArrayList<NTerm>> peptides is to make a simple HashMap
 between NTerm sequence and a list of corresponding <code>NTerm</code>
 HashMap<String, ArrayList<NTerm>> modifications links Nterm position to a
 list of corresponding <code>NTerm</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#createHTMLPageSummary()">createHTMLPageSummary()</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Read TABLE_PATH and create an HTML table from it to ease accession to
 ArrayExpress info</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/GenomesCreation.html#createHTMLPageSummary()">createHTMLPageSummary()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Read modify version GENOMENCBI_Table where "chromosome ID"column is first,
 and create an HTML summary page, allowing to acces quickly to all genome info</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ProteomicsCreation.html#createHTMLPageSummary()">createHTMLPageSummary()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>
<div class="block">DATA_TABLe has been created from a copy paste of ArrayExpress search results.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html#createHTMLPageSummary()">createHTMLPageSummary()</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressListeriomics</a></dt>
<dd>
<div class="block">DATA_TABLe has been created from a copy paste of ArrayExpress search results.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#createHTMLPageSummary()">createHTMLPageSummary()</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Read TABLE_PATH and create an HTML table from it to ease accession to
 ArrayExpress info</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html#createHTMLPageSummary()">createHTMLPageSummary()</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics">ArrayExpressListeriomics</a></dt>
<dd>
<div class="block">DATA_TABLe has been created from a copy paste of ArrayExpress search results.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#createInferiorAbsFilter(double)">createInferiorAbsFilter(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#createInferiorAbsFilter(double)">createInferiorAbsFilter(double)</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#createInferiorFilter(double)">createInferiorFilter(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#createInferiorFilter(double)">createInferiorFilter(double)</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/PhylogenyTools.html#createInfoNodes()">createInfoNodes()</a></span> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#createIterator(org.biojava3.core.sequence.template.Sequence)">createIterator(Sequence&lt;C&gt;)</a></span> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Creates a simple sequence iterator which moves through a sequence going from
 1 to the length of the Sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermMain.html#createListeriomicsTable(java.lang.String)">createListeriomicsTable(String)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermMain.html" title="class in bacnet.scripts.listeriomics.nterm">NTermMain</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#createListeriomicsTSSTermSeq()">createListeriomicsTSSTermSeq()</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>
<div class="block">Create TSS (Transcription Start Site) and TermSeq wig files</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ComparisonsCreation.html#createLogFCMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">createLogFCMatrix(Experiment, Genome)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Create Database.logFCMatrix by regrouping every logFC values for genes,
 sRNAs, asRNAs, and cisRegs in every <code>BioCondition</code> given
 <code>Experiment</code><br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/TranscriptomesCreation.html#createLogFCMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">createLogFCMatrix(Experiment, Genome)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/TranscriptomesCreation.html" title="class in bacnet.scripts.database">TranscriptomesCreation</a></dt>
<dd>
<div class="block">Create Database.logFCMatrix by regrouping every logFC values for genes,
 sRNAs, asRNAs, and cisRegs in every <code>BioCondition</code> given
 <code>Experiment</code><br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ProteomicsCreation.html#createLogFCMatrix(bacnet.datamodel.expdesign.Experiment,java.lang.String)">createLogFCMatrix(Experiment, String)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>
<div class="block">Create a table will all protein expression for each Genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/TranscriptomesCreation.html#createLogfcTranscriptomeTable(java.util.ArrayList,java.lang.String,boolean)">createLogfcTranscriptomeTable(ArrayList&lt;String&gt;, String, boolean)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/TranscriptomesCreation.html" title="class in bacnet.scripts.database">TranscriptomesCreation</a></dt>
<dd>
<div class="block">Create LogFC summary matrix for transcriptomics datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/JCVI.html#createMatrix(java.lang.String,java.lang.String)">createMatrix(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/JCVI.html" title="class in bacnet.scripts.listeriomics.technology">JCVI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ProteomicsCreation.html#createMatrixFromTISData()">createMatrixFromTISData()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>
<div class="block">Create a matrix from TIS data for which each gene has its corresponding
 number of spectra</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#createMatrixFromTISData()">createMatrixFromTISData()</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>
<div class="block">Create a matrix from TIS data for which each gene has its corresponding
 number of spectra</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/ProteinTools.html#createMDB()">createMDB()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#createMDB()">createMDB()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DataFolder.html#createMissingFolders()">createMissingFolders()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataFolder.html" title="class in bacnet.scripts.database">DataFolder</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/BannerView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BannerView.html" title="class in bacnet.e4.rap">BannerView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewBacnet.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewBacnet.html" title="class in bacnet.e4.rap">InitViewBacnet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewLeishmania.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewLeishmania.html" title="class in bacnet.e4.rap">InitViewLeishmania</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewListeria.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeria.html" title="class in bacnet.e4.rap">InitViewListeria</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewListeriaSample.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">InitViewListeriaSample</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewYersinia.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewYersinia.html" title="class in bacnet.e4.rap">InitViewYersinia</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SubmitDataView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SubmitDataView.html" title="class in bacnet.e4.rap">SubmitDataView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SynTView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastResultView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResultView.html" title="class in bacnet.scripts.blast">BlastResultView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GenomicsView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaSummaryView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaSummaryView.html" title="class in bacnet.sequenceTools">SrnaSummaryView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/TableSWTView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/BannerView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/BannerView.html" title="class in bacnet.views">BannerView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/InitView.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/InitView.html" title="class in bacnet.views">InitView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/InternalBrowser.html#createPartControl(org.eclipse.swt.widgets.Composite)">createPartControl(Composite)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/PhylogenySmallORFs.html#createPhylogenyFile(java.util.ArrayList)">createPhylogenyFile(ArrayList&lt;Sequence&gt;)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/PhylogenySmallORFs.html" title="class in bacnet.scripts.blast">PhylogenySmallORFs</a></dt>
<dd>
<div class="block">reading filtered table and create fasta file, assuming that Lite.excel table
 have been reduced manually by removing doublons</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html#createPhylogenyFile(java.util.ArrayList)">createPhylogenyFile(ArrayList&lt;Sequence&gt;)</a></span> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html" title="class in bacnet.scripts.phylogeny">PhylogenySmallORFs</a></dt>
<dd>
<div class="block">reading filtered table and create fasta file, assuming that Lite.excel table
 have been reduced manually by removing doublons</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#createProbeList()">createProbeList()</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Read general GeneExpression data and extract the list of probes from it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#createProbeList()">createProbeList()</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Read WildType tiling data and extract the list of probes from it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/ProteinTools.html#createProteinDB()">createProteinDB()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Create a fasta file which will be used has a database<br>
 This file contains:
 all genes
 all new genes described in RAST
 all Srna on 3 frames
 all ASrna on 3 frames (300bp long) lists of Genes, ID=lmo, directly taken
 from NCBI lists of "new"Gene, ID=rastlmo, from the analysis of annotation I
 just did using RAST software lists of Srna, ID=rli, rliM, rliP, we don't know
 where is the begining so we have 3 frame for each rli, we start at the first
 start codon, plus we cut when we found stop codon and peptide smaller than
 5aa. lists of ASrna, ID=anti, antiM, antiP, we don't know where is the
 begining so we have 3 frame for each asRNA, we start at the first start
 codon, plus we cut when we found stop codon and peptide smaller than 5aa.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#createProteinDB()">createProteinDB()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Create a fasta file which will be used has a database<br>
 This file contains:
 all genes
 all new genes described in RAST
 all Srna on 3 frames
 all ASrna on 3 frames (300bp long) lists of Genes, ID=lmo, directly taken
 from NCBI lists of "new"Gene, ID=rastlmo, from the analysis of annotation I
 just did using RAST software lists of Srna, ID=rli, rliM, rliP, we don't know
 where is the begining so we have 3 frame for each rli, we start at the first
 start codon, plus we cut when we found stop codon and peptide smaller than
 5aa. lists of ASrna, ID=anti, antiM, antiP, we don't know where is the
 begining so we have 3 frame for each asRNA, we start at the first start
 codon, plus we cut when we found stop codon and peptide smaller than 5aa.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#createProteolyseMap()">createProteolyseMap()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Regroup peptides which overlap -> the assumption is that they come from the
 same protein but were created from proteolyse</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ProteomicsCreation.html#createProteomeTable(java.util.ArrayList,java.lang.String)">createProteomeTable(ArrayList&lt;String&gt;, String)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>
<div class="block">Create Expression summary matrix for proteomics datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/GenomesCreation.html#createPTTandFAAFiles()">createPTTandFAAFiles()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Some genomes have been downloaded without PTT files so we create them<br>
 PTT files are used by SynTVIew software</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence)">createSequence(Sequence&lt;F&gt;)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/CompoundTranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence)">createSequence(Sequence&lt;F&gt;)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundTranslator.html" title="interface in org.biojava3.core.sequence.template">CompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence)">createSequence(Sequence&lt;NucleotideCompound&gt;)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava3.core.sequence.transcription">DNAToRNATranslator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence,org.biojava3.core.sequence.transcription.Frame)">createSequence(Sequence&lt;NucleotideCompound&gt;, Frame)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava3.core.sequence.transcription">DNAToRNATranslator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)">createSequences(Sequence&lt;F&gt;)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/CompoundTranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)">createSequences(Sequence&lt;F&gt;)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundTranslator.html" title="interface in org.biojava3.core.sequence.template">CompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)">createSequences(Sequence&lt;NucleotideCompound&gt;)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava3.core.sequence.transcription">DNAToRNATranslator</a></dt>
<dd>
<div class="block">Overloaded local version which delegates to an optional translator when told
 to (specified during construction).</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)">createSequences(Sequence&lt;NucleotideCompound&gt;)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html" title="class in org.biojava3.core.sequence.transcription">RNAToAminoAcidTranslator</a></dt>
<dd>
<div class="block">Performs the core conversion of RNA to Peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ComparisonsCreation.html#createStatMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">createStatMatrix(Experiment, Genome)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Create Database.statTable by calculating stat values for genes, sRNAs, and
 asRNAs in every BioCondition<br>
 For genes we use FDRBY calculated from GebneExpression array<br>
 For sRNAs and asRNAs we use TSTUDENTTILING for Tiling array</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/TranscriptomesCreation.html#createStatMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">createStatMatrix(Experiment, Genome)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/TranscriptomesCreation.html" title="class in bacnet.scripts.database">TranscriptomesCreation</a></dt>
<dd>
<div class="block">Create Database.statTable by calculating stat values for genes, sRNAs, and
 asRNAs in every BioCondition<br>
 For genes we use FDRBY calculated from GebneExpression array<br>
 For sRNAs and asRNAs we use TSTUDENTTILING for Tiling array</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#createStochasticMatrix(int,int)">createStochasticMatrix(int, int)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#createSubSequence(org.biojava3.core.sequence.template.Sequence,int,int)">createSubSequence(Sequence&lt;C&gt;, int, int)</a></span> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Creates a simple sub sequence view delimited by the given start and end.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/BioConditionCreation.html#createSummaryProteomesTable()">createSummaryProteomesTable()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/BioConditionCreation.html" title="class in bacnet.scripts.database">BioConditionCreation</a></dt>
<dd>
<div class="block">Create a table will all main information on the different BioCondition
 available<br>
 This table will be loaded by BioConditionView</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#createSummaryTable(java.lang.String)">createSummaryTable(String)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Create a summary table for each genome with the list of protein and the list
 of homologies</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html#createSummaryTables()">createSummaryTables()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html" title="class in bacnet.scripts.listeriomics.srna">SrnaFinalList</a></dt>
<dd>
<div class="block">Create SUmmary Tables for the website<br>
 WARNING: GenomeConversion has to be run before!</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/BioConditionCreation.html#createSummaryTranscriptomesComparisonsTable()">createSummaryTranscriptomesComparisonsTable()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/BioConditionCreation.html" title="class in bacnet.scripts.database">BioConditionCreation</a></dt>
<dd>
<div class="block">Create a table will all main information on the different BioCondition
 available<br>
 This table will be loaded by BioConditionView</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/BioConditionCreation.html#createSummaryTranscriptomesTable()">createSummaryTranscriptomesTable()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/BioConditionCreation.html" title="class in bacnet.scripts.database">BioConditionCreation</a></dt>
<dd>
<div class="block">Create a table will all main information on the different BioCondition
 available<br>
 This table will be loaded by BioConditionView</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#createSuperiorAbsFilter(double)">createSuperiorAbsFilter(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#createSuperiorAbsFilter(double)">createSuperiorAbsFilter(double)</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#createSuperiorFilter(double)">createSuperiorFilter(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#createSuperiorFilter(double)">createSuperiorFilter(double)</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#createSuperiorFilter(double,double)">createSuperiorFilter(double, double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#createSuperiorFilter(double,double)">createSuperiorFilter(double, double)</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/GENormalization.html#createTargetsTable(bacnet.datamodel.expdesign.BioCondition,java.lang.String)">createTargetsTable(BioCondition, String)</a></span> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/GENormalization.html" title="class in bacnet.scripts.core.normalization">GENormalization</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#createTechnologiesTable()">createTechnologiesTable()</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#createTechnoTable()">createTechnoTable()</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Create a table with different info about Array Design: Name, number of data
 which use it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#createTechnoTable()">createTechnoTable()</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Create a table with different info about Array Design: Name, number of data
 which use it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/ProteinTools.html#createTISDB()">createTISDB()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Create database of all peptide from a start codon (M=ATG, L=TTG, V=GTG), to
 the next stop codon or R</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#createTISDB()">createTISDB()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Create database of all peptide from a start codon (M=ATG, L=TTG, V=GTG), to
 the next stop codon or R</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/ProteinTools.html#createUTRDB()">createUTRDB()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Create a database with all UTR from Wurtzel et al. 2012<br>
 Found here : DATABASE+"table_s1_tss_table.txt"<br>
 We found the correct frame so we are sure to have the start codon of the gene
 at the exact position of NCBI annotation We start at the first start codon,
 then if stop codon are found in the UTR region we cut in different peptides,
 remove ones < 5aa, and paste to the last part of the UTR the whole gene amino
 acide sequence all gene, ID=lmo_n n being the id of the UTR peptide extracted</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#createUTRDB()">createUTRDB()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Create a database with all UTR from Wurtzel et al. 2012<br>
 Found here : DATABASE+"table_s1_tss_table.txt"<br>
 We found the correct frame so we are sure to have the start codon of the gene
 at the exact position of NCBI annotation We start at the first start codon,
 then if stop codon are found in the UTR region we cut in different peptides,
 remove ones < 5aa, and paste to the last part of the UTR the whole gene amino
 acide sequence all gene, ID=lmo_n n being the id of the UTR peptide extracted</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#createValues(int,int)">createValues(int, int)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#CRISPRGO_PROJECT">CRISPRGO_PROJECT</a></span> - Static variable in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Static variable to discriminate Listeriomics project from the others</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser"><span class="typeNameLink">CRISPRGuideView</span></a> - Class in <a href="../bacnet/genomeBrowser/package-summary.html">bacnet.genomeBrowser</a></dt>
<dd>
<div class="block">CRISPRGuideView for CRISPRGuide datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#%3Cinit%3E()">CRISPRGuideView()</a></span> - Constructor for class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views"><span class="typeNameLink">CRISPRPredictView</span></a> - Class in <a href="../bacnet/views/package-summary.html">bacnet.views</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView.html#%3Cinit%3E()">CRISPRPredictView()</a></span> - Constructor for class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views"><span class="typeNameLink">CRISPRPredictView2</span></a> - Class in <a href="../bacnet/views/package-summary.html">bacnet.views</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView2.html#%3Cinit%3E()">CRISPRPredictView2()</a></span> - Constructor for class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html#CS">CS</a></span> - bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/UNAfold.html#ct_energy">ct_energy</a></span> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/UNAfold.html#ct_energy">ct_energy</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/UNAfold.html#ct_prob">ct_prob</a></span> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/UNAfold.html#ct_prob">ct_prob</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/CUFSL.html" title="class in bacnet.scripts.listeriomics.technology"><span class="typeNameLink">CUFSL</span></a> - Class in <a href="../bacnet/scripts/listeriomics/technology/package-summary.html">bacnet.scripts.listeriomics.technology</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/CUFSL.html#%3Cinit%3E()">CUFSL()</a></span> - Constructor for class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/CUFSL.html" title="class in bacnet.scripts.listeriomics.technology">CUFSL</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html#CURATED">CURATED</a></span> - org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#CURATEDARRAY_CURATED">CURATEDARRAY_CURATED</a></span> - Static variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">List of column names for the final curated table of mapping</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#curateGenomeFindingDoublonGeneName(bacnet.datamodel.sequenceNCBI.GenomeNCBI)">curateGenomeFindingDoublonGeneName(GenomeNCBI)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Check if two different locustag are not associated to the same GeneName</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#curateMatrix(java.lang.String,boolean)">curateMatrix(String, boolean)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>
<div class="block">Read a potential ExpressionMatrix, remove all duplicate rows by calculating
 mean when 2 or more rows have same name.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html#curateMatrix(java.lang.String,boolean)">curateMatrix(String, boolean)</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressListeriomics</a></dt>
<dd>
<div class="block">Read a potential ExpressionMatrix, remove all duplicate rows by calculating
 mean when 2 or more rows have same name.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html#curateMatrix(java.lang.String,boolean)">curateMatrix(String, boolean)</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics">ArrayExpressListeriomics</a></dt>
<dd>
<div class="block">Read a potential ExpressionMatrix, remove all duplicate rows by calculating
 mean when 2 or more rows have same name.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/IHM.html#curateRawData(java.lang.String,java.lang.String)">curateRawData(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/IHM.html" title="class in bacnet.scripts.listeriomics.technology">IHM</a></dt>
<dd>
<div class="block">Raw data from E-GEOD-12151 are not well formatted, not Tab delimited, so we
 need to extacrt a proper table from it, before working on it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/UGiessen.html#curateRawData(java.lang.String,java.lang.String)">curateRawData(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/UGiessen.html" title="class in bacnet.scripts.listeriomics.technology">UGiessen</a></dt>
<dd>
<div class="block">Delete all first 4 columns of each raw data</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/Febit.html#curateTechnoTable(java.lang.String)">curateTechnoTable(String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Febit.html" title="class in bacnet.scripts.listeriomics.technology">Febit</a></dt>
<dd>
<div class="block">In some column Gene locus name is missing so we add it using EGDe genome and
 GenBank ID of each gene</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/NavigationManagement.html#CUTOFF">CUTOFF</a></span> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastResult.html#CUTOFF_EVALUE">CUTOFF_EVALUE</a></span> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastResult.html#CUTOFF_INTENSITIES">CUTOFF_INTENSITIES</a></span> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html#CW">CW</a></span> - bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/BasicColor.html#CYAN">CYAN</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a></div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
