<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 14:20:51 CEST 2019 -->
<title>E-Index</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="E-Index";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="contentContainer"><a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a><a id="I:E">
<!--   -->
</a>
<h2 class="title">E</h2>
<dl>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#ECOLI_NAME">ECOLI_NAME</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Name for Ecoli genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html#edit(org.biojava3.core.sequence.template.Sequence)">edit(Sequence&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="class in org.biojava3.core.sequence.edits">Edit.AbstractEdit</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/edits/Edit.html#edit(org.biojava3.core.sequence.template.Sequence)">edit(Sequence&lt;C&gt;)</a></span> - Method in interface org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.html" title="interface in org.biojava3.core.sequence.edits">Edit</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.html" title="interface in org.biojava3.core.sequence.edits"><span class="typeNameLink">Edit</span></a>&lt;<a href="../org/biojava3/core/sequence/edits/Edit.html" title="type parameter in Edit">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/edits/package-summary.html">org.biojava3.core.sequence.edits</a></dt>
<dd>
<div class="block">Interface for carrying out edit operations on a Sequence.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="class in org.biojava3.core.sequence.edits"><span class="typeNameLink">Edit.AbstractEdit</span></a>&lt;<a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/edits/package-summary.html">org.biojava3.core.sequence.edits</a></dt>
<dd>
<div class="block">Abstract class which defines all edit operations as a call to discover what
 5' and 3' ends of an editing Sequence should be joined together with a target
 Sequence.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.Delete.html" title="class in org.biojava3.core.sequence.edits"><span class="typeNameLink">Edit.Delete</span></a>&lt;<a href="../org/biojava3/core/sequence/edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/edits/package-summary.html">org.biojava3.core.sequence.edits</a></dt>
<dd>
<div class="block">Implementation which allows for the deletion of bases from a Sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.Insert.html" title="class in org.biojava3.core.sequence.edits"><span class="typeNameLink">Edit.Insert</span></a>&lt;<a href="../org/biojava3/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/edits/package-summary.html">org.biojava3.core.sequence.edits</a></dt>
<dd>
<div class="block">Edit implementation which allows us to insert a base at any position in a
 Sequence.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.Substitute.html" title="class in org.biojava3.core.sequence.edits"><span class="typeNameLink">Edit.Substitute</span></a>&lt;<a href="../org/biojava3/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/edits/package-summary.html">org.biojava3.core.sequence.edits</a></dt>
<dd>
<div class="block">Allows for the substitution of bases into an existing Sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#EGDC_NAME">EGDC_NAME</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Name of EGD genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#EGDE_ANNOTATION">EGDE_ANNOTATION</a></span> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">All supplementary information taken from Glaser et al., Science 2001</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#EGDE_CHROMO_NAME">EGDE_CHROMO_NAME</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Name of EGD-e chromosomegenome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#EGDE_NAME">EGDE_NAME</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Name of EGD-e genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#EGDE_RAST_NAME">EGDE_RAST_NAME</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Name of RAST EGD-e genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#EGDE_SUPPTABLE">EGDE_SUPPTABLE</a></span> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">All supplementary information taken from Toledo-Arana et al., Nature 2009</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/EGDeWTdata.html" title="class in bacnet.datamodel.dataset"><span class="typeNameLink">EGDeWTdata</span></a> - Class in <a href="../bacnet/datamodel/dataset/package-summary.html">bacnet.datamodel.dataset</a></dt>
<dd>
<div class="block">List of method for creating EGDe_Mean data: A data created from the mean of
 all Wild Type data (37C in BHI)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/EGDeWTdata.html#%3Cinit%3E()">EGDeWTdata()</a></span> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/EGDeWTdata.html" title="class in bacnet.datamodel.dataset">EGDeWTdata</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#eliminateColumn(double%5B%5D%5B%5D)">eliminateColumn(double[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html#EM">EM</a></span> - bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Location.html#EMPTY">EMPTY</a></span> - Static variable in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Basic location which is set to the minimum and maximum bounds of
 <code>Integer</code>.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DataSource.html#ENA">ENA</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/XMLUtils.html#encodeToFile(java.lang.Object,java.lang.String)">encodeToFile(Object, String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/XMLUtils.html" title="class in bacnet.utils">XMLUtils</a></dt>
<dd>
<div class="block">Serialisation d'un objet dans un fichier</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastResult.html#end">end</a></span> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#end()">end()</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Get the ending index.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastResult.html#endQuery">endQuery</a></span> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#endsAfter(org.biojava3.genome.parsers.gff.Location)">endsAfter(Location)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Check if this location ends after other location ends.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#endsBefore(org.biojava3.genome.parsers.gff.Location)">endsBefore(Location)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Check if this location ends before other location ends.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/SoftHashMap.html#entrySet()">entrySet()</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/Equals.html#equal(boolean,boolean)">equal(boolean, boolean)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Equals.html" title="class in org.biojava3.core.util">Equals</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/Equals.html#equal(int,int)">equal(int, int)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Equals.html" title="class in org.biojava3.core.util">Equals</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/Equals.html#equal(long,long)">equal(long, long)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Equals.html" title="class in org.biojava3.core.util">Equals</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/Equals.html#equal(java.lang.Object,java.lang.Object)">equal(Object, Object)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Equals.html" title="class in org.biojava3.core.util">Equals</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>
<div class="block">Override of equals method from Object
 
 To be equal two genome has to come from the same species.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/AccessionID.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/AminoAcidCompound.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimplePoint.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#equals(java.lang.Object)">equals(Object)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#equals(org.biojava3.genome.parsers.gff.Location)">equals(Location)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Compare locations for equality.</div>
</dd>
<dt><a href="../org/biojava3/core/util/Equals.html" title="class in org.biojava3.core.util"><span class="typeNameLink">Equals</span></a> - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>
<div class="block">A set of helper methods which return true if the two parameters are equal to
 each other.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/Equals.html#%3Cinit%3E()">Equals()</a></span> - Constructor for class org.biojava3.core.util.<a href="../org/biojava3/core/util/Equals.html" title="class in org.biojava3.core.util">Equals</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/AminoAcidCompound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase(Compound)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase(Compound)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase(Compound)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase(Compound)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Compound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase(Compound)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase(Compound)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#equalsIgnoreCase(org.biojava3.core.sequence.template.Compound)">equalsIgnoreCase(Compound)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/StringManipulationHelper.html#equalsToIgnoreEndline(java.lang.String,java.lang.String)">equalsToIgnoreEndline(String, String)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/StringManipulationHelper.html" title="class in org.biojava3.core.util">StringManipulationHelper</a></dt>
<dd>
<div class="block">compares two strings for equality, line by line, ignoring any difference of
 end line delimiters contained within the 2 Strings.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/StringManipulationHelper.html#equalsToXml(java.lang.String,java.lang.String)">equalsToXml(String, String)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/StringManipulationHelper.html" title="class in org.biojava3.core.util">StringManipulationHelper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#equivalent(java.lang.String,boolean)">equivalent(String, boolean)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Given a sequence name, it gives the conservation in the three different
 species, in a String[]:<br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/LPE.html#estimate(bacnet.datamodel.dataset.ExpressionMatrix)">estimate(ExpressionMatrix)</a></span> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/LPE.html" title="class in bacnet.scripts.core.stat">LPE</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#euclideanDistance(double%5B%5D%5B%5D)">euclideanDistance(double[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Calcul of euclidean distance Matrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#euclideanDistance(double%5B%5D,double%5B%5D)">euclideanDistance(double[], double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Calculate euclidean distance between vector x and vector y</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastResult.html#eValue">eValue</a></span> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/DisplayPhylogenyHandler.html#execute(org.eclipse.core.commands.ExecutionEvent)">execute(ExecutionEvent)</a></span> - Method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/DisplayPhylogenyHandler.html" title="class in bacnet.scripts.phylogeny">DisplayPhylogenyHandler</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span></div>
&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/handler/LoadExpressionMatrixHandler.html#execute(org.eclipse.e4.ui.workbench.IWorkbench,org.eclipse.swt.widgets.Shell)">execute(IWorkbench, Shell)</a></span> - Method in class bacnet.handler.<a href="../bacnet/handler/LoadExpressionMatrixHandler.html" title="class in bacnet.handler">LoadExpressionMatrixHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/FileUtils.html#exists(java.lang.String)">exists(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Test if a file exists or not</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/ExonComparator.html" title="class in org.biojava3.core.sequence"><span class="typeNameLink">ExonComparator</span></a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">Sort Exon where it is a little confusing if exons shoud always be ordered
 left to right where a negative stranded gene should go the other direction.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ExonComparator.html#%3Cinit%3E()">ExonComparator()</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ExonComparator.html" title="class in org.biojava3.core.sequence">ExonComparator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/ExonSequence.html" title="class in org.biojava3.core.sequence"><span class="typeNameLink">ExonSequence</span></a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">A gene contains a collection of Exon sequences</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ExonSequence.html#%3Cinit%3E(org.biojava3.core.sequence.GeneSequence,int,int)">ExonSequence(GeneSequence, int, int)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</a></dt>
<dd>
<div class="block">Need a parent gene sequence and the bioBegin and bioEnd.</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign"><span class="typeNameLink">Experiment</span></a> - Class in <a href="../bacnet/datamodel/expdesign/package-summary.html">bacnet.datamodel.expdesign</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/Experiment.html#%3Cinit%3E()">Experiment()</a></span> - Constructor for class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/Experiment.html#%3Cinit%3E(java.lang.String)">Experiment(String)</a></span> - Constructor for class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/TableHTMLExport.html#exportInHTML(javax.swing.JTable,java.lang.String)">exportInHTML(JTable, String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/TableHTMLExport.html" title="class in bacnet.utils">TableHTMLExport</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core"><span class="typeNameLink">Expression</span></a> - Class in <a href="../bacnet/scripts/core/package-summary.html">bacnet.scripts.core</a></dt>
<dd>
<div class="block">A list of methods to combine transcriptomics data in ExpressionMatrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#%3Cinit%3E(bacnet.datamodel.sequence.Genome,java.util.ArrayList,java.util.ArrayList,bacnet.datamodel.dataset.OmicsData.TypeData)">Expression(Genome, ArrayList&lt;String&gt;, ArrayList&lt;BioCondition&gt;, OmicsData.TypeData)</a></span> - Constructor for class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>
<div class="block">Create <code>Expression</code> object to regroups multiple Expression data</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset"><span class="typeNameLink">ExpressionData</span></a> - Class in <a href="../bacnet/datamodel/dataset/package-summary.html">bacnet.datamodel.dataset</a></dt>
<dd>
<div class="block">Abstract Class describing data related to transcriptome <br>
 This data are saved in DataStream and might be read in streaming mode <br>
 <br>
 boolean[] read is used as a "read" indicator <br>
 As soon as a position in double[] values has been read, boolean[] read is
 updated <br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#%3Cinit%3E()">ExpressionData()</a></span> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#%3Cinit%3E(java.lang.String)">ExpressionData(String)</a></span> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset"><span class="typeNameLink">ExpressionMatrix</span></a> - Class in <a href="../bacnet/datamodel/dataset/package-summary.html">bacnet.datamodel.dataset</a></dt>
<dd>
<div class="block">This is the principal data model used in the software for managing big table
 of expression</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html#ExpressionMatrix">ExpressionMatrix</a></span> - bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#%3Cinit%3E()">ExpressionMatrix()</a></span> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#%3Cinit%3E(java.lang.String,int)">ExpressionMatrix(String, int)</a></span> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#%3Cinit%3E(java.util.ArrayList)">ExpressionMatrix(ArrayList&lt;String&gt;)</a></span> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Create a void ExpressionMatrix and associate headers</div>
</dd>
<dt><a href="../bacnet/reader/ExpressionMatrixReader.html" title="class in bacnet.reader"><span class="typeNameLink">ExpressionMatrixReader</span></a> - Class in <a href="../bacnet/reader/package-summary.html">bacnet.reader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/ExpressionMatrixReader.html#%3Cinit%3E()">ExpressionMatrixReader()</a></span> - Constructor for class bacnet.reader.<a href="../bacnet/reader/ExpressionMatrixReader.html" title="class in bacnet.reader">ExpressionMatrixReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils"><span class="typeNameLink">ExpressionMatrixStat</span></a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>
<div class="block">List of method for doing Stat on ExpressionMatrices</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ExpressionMatrixStat.html#%3Cinit%3E()">ExpressionMatrixStat()</a></span> - Constructor for class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#EXTENSION">EXTENSION</a></span> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#EXTENSION">EXTENSION</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NGS.html#EXTENSION">EXTENSION</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#EXTENSION">EXTENSION</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Extension name of every OmicsData serialized</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ProteomicsData.html#EXTENSION">EXTENSION</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#EXTENSION">EXTENSION</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#EXTENSION">EXTENSION</a></span> - Static variable in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/GFFNCBIReader.html#EXTENSION">EXTENSION</a></span> - Static variable in class bacnet.reader.<a href="../bacnet/reader/GFFNCBIReader.html" title="class in bacnet.reader">GFFNCBIReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/PTTReader.html#EXTENSION">EXTENSION</a></span> - Static variable in class bacnet.reader.<a href="../bacnet/reader/PTTReader.html" title="class in bacnet.reader">PTTReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html#extractAllContigs(boolean)">extractAllContigs(boolean)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html" title="class in bacnet.scripts.blast">GenomeNCBIFolderTools</a></dt>
<dd>
<div class="block">Go through all folders of Genome directory and extract all TGZ contained in
 .faa .fna .ffn .frn .gff</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#extractAllSrnaToGff()">extractAllSrnaToGff()</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Extract to a GFF file the list of Srna from EGD-e</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/FileUtils.html#extractGZIP(java.lang.String,java.lang.String)">extractGZIP(String, String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#extractInfo(java.util.ArrayList)">extractInfo(ArrayList&lt;Srna&gt;)</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">From Blast Result extract Genome containing the sRNA<br>
 
 A filter is applied to keep results with identities above 50%</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/Blast.html#extractInfo(java.util.ArrayList,java.lang.String,java.lang.String,double,boolean)">extractInfo(ArrayList&lt;String&gt;, String, String, double, boolean)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>
<div class="block">From Blast Result extract Genome containing the sRNA<br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/GENormalization.html#extractInformation(bacnet.datamodel.expdesign.BioCondition)">extractInformation(BioCondition)</a></span> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/GENormalization.html" title="class in bacnet.scripts.core.normalization">GENormalization</a></dt>
<dd>
<div class="block">Load .ge files resulting from R script normalization and extract the
 important columns = column 0-4 = log Expression values and median expression</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/GENormalization.html#extractInformation(bacnet.datamodel.expdesign.Experiment)">extractInformation(Experiment)</a></span> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/GENormalization.html" title="class in bacnet.scripts.core.normalization">GENormalization</a></dt>
<dd>
<div class="block">Extract information contained in all .ge files of the experiment</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#extractList(java.util.ArrayList,int,int)">extractList(ArrayList&lt;String&gt;, int, int)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Extract a portion of an ArrayList into another ArrayList</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/PhylogenyTools.html#extractPhylogeny(org.forester.phylogeny.Phylogeny,java.util.HashMap)">extractPhylogeny(Phylogeny, HashMap&lt;String, String[]&gt;)</a></span> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a></dt>
<dd>
<div class="block">If a node of the Phylogeny phy, is not a key of infoNodes Map, delete the
 tree <br>
 return the subTree obtained</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#extractPhylogeny(org.forester.phylogeny.Phylogeny,java.util.TreeSet)">extractPhylogeny(Phylogeny, TreeSet&lt;Integer&gt;)</a></span> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>
<div class="block">If a node of the Phylogeny phy, is include in listIDs, delete the node <br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#extractPosition(bacnet.datamodel.sequence.Srna,bacnet.datamodel.sequence.Genome,java.util.TreeMap)">extractPosition(Srna, Genome, TreeMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">From Blast Result extract position in the Genome of each sRNA<br>
 
 Srna with an identity lower than 25% are not inserted</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/Febit.html#extractProbes(java.lang.String,java.lang.String)">extractProbes(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Febit.html" title="class in bacnet.scripts.listeriomics.technology">Febit</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/GeneralArray.html#extractProbes(java.lang.String,java.lang.String)">extractProbes(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/GeneralArray.html" title="class in bacnet.scripts.listeriomics.technology">GeneralArray</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/IHM.html#extractProbes(java.lang.String,java.lang.String)">extractProbes(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/IHM.html" title="class in bacnet.scripts.listeriomics.technology">IHM</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/JCVI.html#extractProbes(java.lang.String,java.lang.String)">extractProbes(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/JCVI.html" title="class in bacnet.scripts.listeriomics.technology">JCVI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/LT.html#extractProbes(java.lang.String,java.lang.String)">extractProbes(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/LT.html" title="class in bacnet.scripts.listeriomics.technology">LT</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/Macroarray.html#extractProbes(java.lang.String,java.lang.String)">extractProbes(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Macroarray.html" title="class in bacnet.scripts.listeriomics.technology">Macroarray</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/NimbleGen.html#extractProbes(java.lang.String,java.lang.String)">extractProbes(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/NimbleGen.html" title="class in bacnet.scripts.listeriomics.technology">NimbleGen</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/Tasmania.html#extractProbes(java.lang.String,java.lang.String)">extractProbes(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Tasmania.html" title="class in bacnet.scripts.listeriomics.technology">Tasmania</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/UGiessen.html#extractProbes(java.lang.String,java.lang.String)">extractProbes(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/UGiessen.html" title="class in bacnet.scripts.listeriomics.technology">UGiessen</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/Wuerzburg.html#extractProbes(java.lang.String,java.lang.String)">extractProbes(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Wuerzburg.html" title="class in bacnet.scripts.listeriomics.technology">Wuerzburg</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/WUSTL.html#extractProbes(java.lang.String,java.lang.String)">extractProbes(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/WUSTL.html" title="class in bacnet.scripts.listeriomics.technology">WUSTL</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/Agilent.html#extractProbes(java.lang.String,java.lang.String,java.lang.String)">extractProbes(String, String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Agilent.html" title="class in bacnet.scripts.listeriomics.technology">Agilent</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/Agilent.html#extractProbesComparison(java.lang.String,java.lang.String)">extractProbesComparison(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Agilent.html" title="class in bacnet.scripts.listeriomics.technology">Agilent</a></dt>
<dd>
<div class="block">Extract Probes from Matrix when its a Comparison</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/CUFSL.html#extractProbesComparison(java.lang.String,java.lang.String,boolean)">extractProbesComparison(String, String, boolean)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/CUFSL.html" title="class in bacnet.scripts.listeriomics.technology">CUFSL</a></dt>
<dd>
<div class="block">Extract Probes from Matrix when its a Comparison</div>
</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a></div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
