<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>R-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="index: R">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:R">R</h2>
<dl class="index">
<dt><a href="../bacnet/scripts/genome/RASTEGDeGenome.html" class="type-name-link" title="class in bacnet.scripts.genome">RASTEGDeGenome</a> - Class in <a href="../bacnet/scripts/genome/package-summary.html">bacnet.scripts.genome</a></dt>
<dd>
<div class="block">List of methods for adding RAST genes to EGD-e annotation</div>
</dd>
<dt><a href="../bacnet/scripts/genome/RASTEGDeGenome.html#%3Cinit%3E()" class="member-name-link">RASTEGDeGenome()</a> - Constructor for class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/RASTEGDeGenome.html" title="class in bacnet.scripts.genome">RASTEGDeGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a> - Class in <a href="../bacnet/scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></dt>
<dd>
<div class="block">Methods to look at overlapping elements between NCBI and RAST genomes</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#%3Cinit%3E()" class="member-name-link">RASToverlap()</a> - Constructor for class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#RATIO_ANNOT_ELEMENT_SEPARATION" class="member-name-link">RATIO_ANNOT_ELEMENT_SEPARATION</a> - Static variable in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#RAWARRAY_COLUMN" class="member-name-link">RAWARRAY_COLUMN</a> - Static variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">List of column names when PATHARRYA contains only one experiment</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#RAWARRAY_MULTICOLUMN" class="member-name-link">RAWARRAY_MULTICOLUMN</a> - Static variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">List of column names when PATHARRYA contains multiple experiments</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#read()" class="member-name-link">read()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Load the entire TranscriptomeData</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#read()" class="member-name-link">read()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>
<div class="block">Load the entire TranscriptomeData</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#read()" class="member-name-link">read()</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>
<div class="block">Reads a single character.</div>
</dd>
<dt><a href="../org/biojava3/core/util/UncompressInputStream.html#read()" class="member-name-link">read()</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/UncompressInputStream.html#read(byte%5B%5D,int,int)" class="member-name-link">read(byte[], int, int)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#read(char%5B%5D,int,int)" class="member-name-link">read(char[], int, int)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>
<div class="block">Reads characters into a portion of an array.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#read(int)" class="member-name-link">read(int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Read only one position in the data</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#read(int,boolean)" class="member-name-link">read(int, boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Read only one position in the data</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#read(int,int)" class="member-name-link">read(int, int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Read the TranscriptomeData between begin and end <br>
 Careful: begin &lt; end <br>
 It is a streaming reading so it check if the data are already read or not and act consequently</div>
</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#read(java.io.File)" class="member-name-link">read(File)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Read file with a tab separator</div>
</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#read(java.io.File,java.lang.String)" class="member-name-link">read(File, String)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Read a Tab-delimited file The number of cell in each column has not to be the same When a blank
 cell is found, it is replace by a void space.</div>
</dd>
<dt><a href="../bacnet/reader/ExpressionMatrixReader.html#read(java.lang.String)" class="member-name-link">read(String)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/ExpressionMatrixReader.html" title="class in bacnet.reader">ExpressionMatrixReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#read(java.lang.String)" class="member-name-link">read(String)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html#read(java.lang.String)" class="member-name-link">read(String)</a> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html" title="class in org.biojava3.genome.parsers.gff">GeneIDGFF2Reader</a></dt>
<dd>
<div class="block">Read a file into a FeatureList.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html#read(java.lang.String)" class="member-name-link">read(String)</a> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html" title="class in org.biojava3.genome.parsers.gff">GeneMarkGTFReader</a></dt>
<dd>
<div class="block">Read a file into a FeatureList.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GFF3Reader.html#read(java.lang.String)" class="member-name-link">read(String)</a> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GFF3Reader.html" title="class in org.biojava3.genome.parsers.gff">GFF3Reader</a></dt>
<dd>
<div class="block">Read a file into a FeatureList.</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyTools.html#readAllBacteriaPhylogeny()" class="member-name-link">readAllBacteriaPhylogeny()</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a></dt>
<dd>
<div class="block">Read PhyloXML file containing all bacteria phylogeny <br>
 read from ALLBACTERIA_XML</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#readAllBacteriaPhylogeny()" class="member-name-link">readAllBacteriaPhylogeny()</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>
<div class="block">Read Newick file containing all bacteria phylogeny <br></div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyTools.html#readAllBacteriarLMSTPhylogeny()" class="member-name-link">readAllBacteriarLMSTPhylogeny()</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a></dt>
<dd>
<div class="block">Read PhyloXML file containing all bacteria phylogeny <br>
 Read from ALLBACTERIA_XML_2</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html#readFastaDNASequence(java.io.File)" class="member-name-link">readFastaDNASequence(File)</a> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html#readFastaDNASequence(java.io.File,boolean)" class="member-name-link">readFastaDNASequence(File, boolean)</a> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</a></dt>
<dd>
<div class="block">Selecting lazySequenceLoad=true will parse the FASTA file and figure out the accessionid and
 offsets and return sequence objects that can in the future read the sequence from the disk.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html#readFastaDNASequence(java.io.InputStream)" class="member-name-link">readFastaDNASequence(InputStream)</a> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</a></dt>
<dd>
<div class="block">Read a fasta DNA sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html#readFastaProteinSequence(java.io.File)" class="member-name-link">readFastaProteinSequence(File)</a> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</a></dt>
<dd>
<div class="block">Read a fasta file containing amino acids with setup that would handle most cases.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html#readFastaProteinSequence(java.io.InputStream)" class="member-name-link">readFastaProteinSequence(InputStream)</a> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</a></dt>
<dd>
<div class="block">Read a fasta file containing amino acids with setup that would handle most cases.</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#readFilter(java.lang.String)" class="member-name-link">readFilter(String)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>
<div class="block">Create a list of cutoff to apply to the list of Nterm<br></div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#readFirmicutesPhylogeny()" class="member-name-link">readFirmicutesPhylogeny()</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>
<div class="block">Read Newick file containing all bacteria phylogeny <br>
 Remove all non firmicutes species<br>
 Remove bacteria for which no complete genome is available : no NP_ or CP id</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#readFromFileList(java.lang.String)" class="member-name-link">readFromFileList(String)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#readHashMap(java.lang.String)" class="member-name-link">readHashMap(String)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Read a tabdelimited table and extract the line as an HashMap&lt;first column, second column> of
 Strings</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#readLine()" class="member-name-link">readLine()</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>
<div class="block">Reads a line of text.</div>
</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#readList(java.lang.String)" class="member-name-link">readList(String)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Read a tabdelimited table and extract the first column as an ArrayList of String</div>
</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#readList(java.lang.String,boolean)" class="member-name-link">readList(String, boolean)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Read a tabdelimited table and extract the line as an ArrayList of String</div>
</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#readList(java.lang.String,boolean,boolean)" class="member-name-link">readList(String, boolean, boolean)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Read a tabdelimited table and extract the line as an ArrayList of String[]<br>
 Mainly used to load String[][] in SWT table graphic viewer</div>
</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#readList(java.lang.String,int)" class="member-name-link">readList(String, int)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Read a tabdelimited table and extract the line as an ArrayList of String</div>
</dd>
<dt><a href="../bacnet/Database.html#readListDatabases()" class="member-name-link">readListDatabases()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Print out JVM properties And set this.path depending on the Operating System HAS TO BE RUN AT
 STARTING OF EVERY PROJECT</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#readPhylogeny(java.lang.String)" class="member-name-link">readPhylogeny(String)</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>
<div class="block">Read Newick file containing all bacteria phylogeny <br></div>
</dd>
<dt><a href="../bacnet/utils/FileUtils.html#readText(java.lang.String)" class="member-name-link">readText(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Load text from fileName</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#readWIGChromo(java.lang.String,java.lang.String,bacnet.datamodel.sequence.Genome)" class="member-name-link">readWIGChromo(String, String, Genome)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>
<div class="block">Will read your wig file and parse it to BACNET <br>
 ONLY WIG FILE WITH variableStep and raw format are accepted No fixedStep format</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#ready()" class="member-name-link">ready()</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>
<div class="block">Tells whether this stream is ready to be read.</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#RED" class="member-name-link">RED</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#REDDARK_GENE" class="member-name-link">REDDARK_GENE</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Dark red color used for plus strand genes</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#REDLIGHT_GENE" class="member-name-link">REDLIGHT_GENE</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Light red color used for plus strand genes</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#REDLIGHT_NUCLEOTIDE" class="member-name-link">REDLIGHT_NUCLEOTIDE</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">red light color for nucleotide representation</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#REDLINE_GENE" class="member-name-link">REDLINE_GENE</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Very dark red color used for plus strand genes surrounding</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#redrawAllCanvas()" class="member-name-link">redrawAllCanvas()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html" class="type-name-link" title="class in bacnet.genomeBrowser.core">Region</a> - Class in <a href="../bacnet/genomeBrowser/core/package-summary.html">bacnet.genomeBrowser.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#%3Cinit%3E()" class="member-name-link">Region()</a> - Constructor for class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#%3Cinit%3E(int,int)" class="member-name-link">Region(int, int)</a> - Constructor for class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#regionChanged" class="member-name-link">regionChanged</a> - Variable in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SessionControl.html#registerClosingUIsession(org.eclipse.e4.ui.workbench.modeling.EPartService,org.eclipse.e4.ui.workbench.modeling.EModelService,org.eclipse.swt.widgets.Shell)" class="member-name-link">registerClosingUIsession(EPartService, EModelService, Shell)</a> - Static method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SessionControl.html" title="class in bacnet.e4.rap">SessionControl</a></dt>
<dd>
<div class="block">Register the closing of UI session and close every part manually !</div>
</dd>
<dt><a href="../bacnet/e4/rap/setup/SessionControl.html#registerClosingUIsession(org.eclipse.e4.ui.workbench.modeling.EPartService,org.eclipse.e4.ui.workbench.modeling.EModelService,org.eclipse.swt.widgets.Shell)" class="member-name-link">registerClosingUIsession(EPartService, EModelService, Shell)</a> - Static method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SessionControl.html" title="class in bacnet.e4.rap.setup">SessionControl</a></dt>
<dd>
<div class="block">Register the closing of UI session and close every part manually !</div>
</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#registerServiceAndNavigationTab(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">registerServiceAndNavigationTab(EPartService)</a> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>
<div class="block">Register service navigation management to be able to use Previous and Next buttons</div>
</dd>
<dt><a href="../bacnet/raprcp/SaveFileUtils.html#registerTextFile(java.lang.String,java.io.File)" class="member-name-link">registerTextFile(String, File)</a> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/SaveFileUtils.html" title="class in bacnet.raprcp">SaveFileUtils</a></dt>
<dd>
<div class="block">Register a text file in RWT platform</div>
</dd>
<dt><a href="../bacnet/scripts/genome/SDProfile.html#regroupResults(java.lang.String)" class="member-name-link">regroupResults(String)</a> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>
<div class="block">Read all text files created during SDProfile calculation and regroup them</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/RegulatorRegion.html" class="type-name-link" title="class in bacnet.scripts.listeriomics">RegulatorRegion</a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/RegulatorRegion.html#%3Cinit%3E()" class="member-name-link">RegulatorRegion()</a> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/RegulatorRegion.html" title="class in bacnet.scripts.listeriomics">RegulatorRegion</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ListUtils.html#relativeComplement(java.util.ArrayList,java.util.ArrayList)" class="member-name-link">relativeComplement(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ListUtils.html" title="class in bacnet.utils">ListUtils</a></dt>
<dd>
<div class="block">Calculate the relative complement of two lists L1 and L2 <br></div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#relativeTOabsoluteValue()" class="member-name-link">relativeTOabsoluteValue()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Changed all BioCondition to display when switching from "RELATIVE VALUE DISPLAY3 to "ABSOLUTE
 VALUE DISPLAY"</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/LocIterator.html#remainder()" class="member-name-link">remainder()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></dt>
<dd>
<div class="block">Get portion of bounding location that has not yet been retrieved by next() method.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html#remove()" class="member-name-link">remove()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="class in org.biojava3.core.sequence.template">SequenceMixin.SequenceIterator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/LocIterator.html#remove()" class="member-name-link">remove()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></dt>
<dd>
<div class="block">Unsupported.</div>
</dd>
<dt><a href="../org/biojava3/core/util/SoftHashMap.html#remove(java.lang.Object)" class="member-name-link">remove(Object)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#removeAlignedSequence(S)" class="member-name-link">removeAlignedSequence(S)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">Remove a sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/TranscriptSequence.html#removeCDS(java.lang.String)" class="member-name-link">removeCDS(String)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>
<div class="block">Remove a CDS or coding sequence from the transcript sequence</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#removeColumn(java.lang.String,java.lang.String)" class="member-name-link">removeColumn(String, String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#removeExon(java.lang.String)" class="member-name-link">removeExon(String)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Remove the exon sequence</div>
</dd>
<dt><a href="../bacnet/utils/FileUtils.html#removeExtension(java.lang.String)" class="member-name-link">removeExtension(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Remove extension of a fileName, by removing everything after the last '.'</div>
</dd>
<dt><a href="../bacnet/utils/FileUtils.html#removeExtensionAndPath(java.lang.String)" class="member-name-link">removeExtensionAndPath(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Remove extension of a filePath and the complete path</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#removeFeature(org.biojava3.core.sequence.features.FeatureInterface)" class="member-name-link">removeFeature(FeatureInterface&lt;AbstractSequence&lt;C&gt;, C&gt;)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Remove a feature from the sequence</div>
</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#removeFiles()" class="member-name-link">removeFiles()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>
<div class="block">Remove specific files in genome folders</div>
</dd>
<dt><a href="../bacnet/scripts/blast/PhylogenySmallORFs.html#removeFiles()" class="member-name-link">removeFiles()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/PhylogenySmallORFs.html" title="class in bacnet.scripts.blast">PhylogenySmallORFs</a></dt>
<dd>
<div class="block">Remove specific files in genome folders</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html#removeFiles()" class="member-name-link">removeFiles()</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html" title="class in bacnet.scripts.phylogeny">PhylogenySmallORFs</a></dt>
<dd>
<div class="block">Remove specific files in genome folders</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastDatabase.html#removeFiles(java.lang.String)" class="member-name-link">removeFiles(String)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">Remove specific files in genome folders</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#removeGeneSequence(java.lang.String)" class="member-name-link">removeGeneSequence(String)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#removeIntron(java.lang.String)" class="member-name-link">removeIntron(String)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Remove the intron by accession</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#removeNote(java.lang.String)" class="member-name-link">removeNote(String)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/FileUtils.html#removePath(java.lang.String)" class="member-name-link">removePath(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Remove extension of a filePath and the complete path</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#removeTranscript(java.lang.String)" class="member-name-link">removeTranscript(String)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Remove the transcript sequence from the gene</div>
</dd>
<dt><a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html#removeUnusedFile()" class="member-name-link">removeUnusedFile()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html" title="class in bacnet.scripts.blast">GenomeNCBIFolderTools</a></dt>
<dd>
<div class="block">In NCBI genome folder remove everything which is not useful for my analysis<br>
 keep fasta file, .ptt .gff and blastDB files</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#reorderHeaders(java.util.ArrayList)" class="member-name-link">reorderHeaders(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Giving a list of header we reorganize the ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#reorderModif(java.util.ArrayList,bacnet.datamodel.proteomics.NTerm)" class="member-name-link">reorderModif(ArrayList&lt;NTerm&gt;, NTerm)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Reorder the Nterm to display AcD3 first, then Ace, and For.</div>
</dd>
<dt><a href="../bacnet/scripts/database/ComparisonsCreation.html#reorganizeMatrix(bacnet.datamodel.dataset.ExpressionMatrix,java.util.ArrayList,bacnet.scripts.core.Comparison)" class="member-name-link">reorganizeMatrix(ExpressionMatrix, ArrayList&lt;StatTest&gt;, Comparison)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Reorder headers of the comparison Matrix to group GenExpression and Tiling results coming from
 the same comparison <br></div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#rescaleData(double%5B%5D%5B%5D,double)" class="member-name-link">rescaleData(double[][], double)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Rescale each Column of the array, at the end all the points as to fill in a cube of diameter
 gridSize</div>
</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html#reset()" class="member-name-link">reset()</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html" title="class in bacnet.scripts.core.vennDiagram">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/CombinationGenerator.html#reset()" class="member-name-link">reset()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/CombinationGenerator.html" title="class in bacnet.utils">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#reset()" class="member-name-link">reset()</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>
<div class="block">Resets the stream to the most recent mark.</div>
</dd>
<dt><a href="../org/biojava3/core/util/CRC64Checksum.html#reset()" class="member-name-link">reset()</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/CRC64Checksum.html" title="class in org.biojava3.core.util">CRC64Checksum</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Point.Resolver.html#resolve(T)" class="member-name-link">resolve(T)</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Point.Resolver.html" title="interface in org.biojava3.core.sequence.location.template">Point.Resolver</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/swt/ResourceManager.html" class="type-name-link" title="class in bacnet.swt">ResourceManager</a> - Class in <a href="../bacnet/swt/package-summary.html">bacnet.swt</a></dt>
<dd>
<div class="block">Utility class for managing OS resources associated with SWT/JFace controls such as colors, fonts,
 images, etc.
 
 !!!</div>
</dd>
<dt><a href="../bacnet/swt/ResourceManager.html#%3Cinit%3E()" class="member-name-link">ResourceManager()</a> - Constructor for class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/swt/ResourceManager.PluginResourceProvider.html" class="type-name-link" title="interface in bacnet.swt">ResourceManager.PluginResourceProvider</a> - Interface in <a href="../bacnet/swt/package-summary.html">bacnet.swt</a></dt>
<dd>
<div class="block">Provider for plugin resources, used by WindowBuilder at design time.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#reverse(int)" class="member-name-link">reverse(int)</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimplePoint.html#reverse(int)" class="member-name-link">reverse(int)</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Point.html#reverse(int)" class="member-name-link">reverse(int)</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a></dt>
<dd>
<div class="block">Returns the equivalent position on the reverse strand</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Frame.html#REVERSED_ONE" class="member-name-link">REVERSED_ONE</a> - Enum constant in enum org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Frame.html#REVERSED_THREE" class="member-name-link">REVERSED_THREE</a> - Enum constant in enum org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Frame.html#REVERSED_TWO" class="member-name-link">REVERSED_TWO</a> - Enum constant in enum org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/ReversedSequenceView.html" class="type-name-link" title="class in org.biojava3.core.sequence.views">ReversedSequenceView</a>&lt;<a href="../org/biojava3/core/sequence/views/ReversedSequenceView.html" title="type parameter in ReversedSequenceView">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</a></dt>
<dd>
<div class="block">For a given sequence this class will return the base at the reversed position i.e. in a sequence
 of size 10, if you request base 2 you will get back the base at position 9.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/views/ReversedSequenceView.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">ReversedSequenceView(Sequence&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views">ReversedSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Rscript.html#REXEC" class="member-name-link">REXEC</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Rscript.html" title="class in bacnet.scripts.core">Rscript</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#RFAM_GENOME" class="member-name-link">RFAM_GENOME</a> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html#rFamAnnot" class="member-name-link">rFamAnnot</a> - Variable in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html" title="class in bacnet.expressionAtlas.core">SelectGenomeElementDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html" class="type-name-link" title="class in bacnet.datamodel.annotation">RfamElement</a> - Class in <a href="../bacnet/datamodel/annotation/package-summary.html">bacnet.datamodel.annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#%3Cinit%3E(org.biojava3.genome.parsers.gff.FeatureI)" class="member-name-link">RfamElement(FeatureI)</a> - Constructor for class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html#RiboSeq" class="member-name-link">RiboSeq</a> - Enum constant in enum bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#rnaAminoAcidTranslator(org.biojava3.core.sequence.transcription.RNAToAminoAcidTranslator)" class="member-name-link">rnaAminoAcidTranslator(RNAToAminoAcidTranslator)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#RNABindingCircularGenome(int,int,bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">RNABindingCircularGenome(int, int, Genome, String)</a> - Method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>
<div class="block">Function to create a circular genome view of rnas binding to protein lom2686</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#rnaCompounds(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">rnaCompounds(CompoundSet&lt;NucleotideCompound&gt;)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/RNACompoundSet.html" class="type-name-link" title="class in org.biojava3.core.sequence.compound">RNACompoundSet</a> - Class in <a href="../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/RNACompoundSet.html#%3Cinit%3E()" class="member-name-link">RNACompoundSet()</a> - Constructor for class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/RNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">RNACompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#rnaCreator(org.biojava3.core.sequence.io.template.SequenceCreatorInterface)" class="member-name-link">rnaCreator(SequenceCreatorInterface&lt;NucleotideCompound&gt;)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html#RNASeq" class="member-name-link">RNASeq</a> - Enum constant in enum bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#RNASEQ_NOSTRAND" class="member-name-link">RNASEQ_NOSTRAND</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Color used for RNASEq when there is only one strand available = non directional RNASeq</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/RNASequence.html" class="type-name-link" title="class in org.biojava3.core.sequence">RNASequence</a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">RNASequence where RNACompoundSet are the allowed values</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/RNASequence.html#%3Cinit%3E(java.lang.String)" class="member-name-link">RNASequence(String)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>
<div class="block">Create a RNA sequence from a String</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/RNASequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">RNASequence(String, CompoundSet&lt;NucleotideCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>
<div class="block">Create a RNA sequence from a string with a user defined RNA compound set</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/RNASequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader)" class="member-name-link">RNASequence(ProxySequenceReader&lt;NucleotideCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>
<div class="block">Create a RNA aequence from a proxy reader</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/RNASequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">RNASequence(ProxySequenceReader&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>
<div class="block">Create a RNA sequence from a proxy reader and user defined RNA compound set</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/RNASequenceCreator.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">RNASequenceCreator</a> - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">Used to create a RNA sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/RNASequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">RNASequenceCreator(CompoundSet&lt;NucleotideCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/RNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">RNASequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" class="type-name-link" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a> - Class in <a href="../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</a></dt>
<dd>
<div class="block">Attempts to do on the fly translation of RNA by not requesting the compounds until asked.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">RnaSequenceView(Sequence&lt;NucleotideCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">RnaSequenceView(Sequence&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html" class="type-name-link" title="class in org.biojava3.core.sequence.transcription">RNAToAminoAcidTranslator</a> - Class in <a href="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></dt>
<dd>
<div class="block">Takes a <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a> of <a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound"><code>NucleotideCompound</code></a> which should represent an RNA sequence
 (<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence"><code>RNASequence</code></a> is good for this) and returns a list of <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a> which hold
 <a href="../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound"><code>AminoAcidCompound</code></a>.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html#%3Cinit%3E(org.biojava3.core.sequence.io.template.SequenceCreatorInterface,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.transcription.Table,boolean,boolean,boolean)" class="member-name-link">RNAToAminoAcidTranslator(SequenceCreatorInterface&lt;AminoAcidCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;, CompoundSet&lt;Table.Codon&gt;, CompoundSet&lt;AminoAcidCompound&gt;, Table, boolean, boolean, boolean)</a> - Constructor for class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html" title="class in org.biojava3.core.sequence.transcription">RNAToAminoAcidTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#ROW" class="member-name-link">ROW</a> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/NcRNA.TypeNcRNA.html#rRNA" class="member-name-link">rRNA</a> - Enum constant in enum bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.TypeNcRNA.html" title="enum in bacnet.datamodel.sequence">NcRNA.TypeNcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Rscript.html" class="type-name-link" title="class in bacnet.scripts.core">Rscript</a> - Class in <a href="../bacnet/scripts/core/package-summary.html">bacnet.scripts.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Rscript.html#%3Cinit%3E()" class="member-name-link">Rscript()</a> - Constructor for class bacnet.scripts.core.<a href="../bacnet/scripts/core/Rscript.html" title="class in bacnet.scripts.core">Rscript</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/TestGenomeBrowser.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/TestGenomeBrowser.html" title="class in bacnet.genomeBrowser">TestGenomeBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></dt>
<dd>
<div class="block">Download every related experiment from ArrayExpress</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Main method, where all other methods are ran</div>
</dd>
<dt><a href="../bacnet/scripts/blast/PhylogenySmallORFs.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/PhylogenySmallORFs.html" title="class in bacnet.scripts.blast">PhylogenySmallORFs</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/RASTEGDeGenome.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/RASTEGDeGenome.html" title="class in bacnet.scripts.genome">RASTEGDeGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/SDProfile.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/StartCodonProfile.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/StartCodonProfile.html" title="class in bacnet.scripts.genome">StartCodonProfile</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Main method, where all other methods are ran</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics">ArrayExpressListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/GenomesListeriomics.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/GenomesListeriomics.html" title="class in bacnet.scripts.listeriomics">GenomesListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>
<div class="block">This method in an example of all methods which should be run to create your multi-omics website
 <br>
 It is never run !</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermConservationWithRiboProfiling.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermConservationWithRiboProfiling.html" title="class in bacnet.scripts.listeriomics.nterm">NTermConservationWithRiboProfiling</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermMain.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermMain.html" title="class in bacnet.scripts.listeriomics.nterm">NTermMain</a></dt>
<dd>
<div class="block">General method for running all methods related to NTerm project</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/Peptidomics.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/Peptidomics.html" title="class in bacnet.scripts.listeriomics">Peptidomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/RegulatorRegion.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/RegulatorRegion.html" title="class in bacnet.scripts.listeriomics">RegulatorRegion</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html" title="class in bacnet.scripts.listeriomics.srna">SmallPeptidesSearch</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
<div class="block">List of methods to search for small unannotated peptides in reference genomes<br>
 If the peptide is not found we will try to find its surrounding genes in other strains and look
 at thze intergenic region to find the peptide.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/SrnaListeriomics.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SrnaListeriomics.html" title="class in bacnet.scripts.listeriomics">SrnaListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html" title="class in bacnet.scripts.listeriomics">SummaryListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/SystemsBiologyListeriomics.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SystemsBiologyListeriomics.html" title="class in bacnet.scripts.listeriomics">SystemsBiologyListeriomics</a></dt>
<dd>
<div class="block">Create the co-expression network</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/TilingGeneExprDataBase.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TilingGeneExprDataBase.html" title="class in bacnet.scripts.listeriomics">TilingGeneExprDataBase</a></dt>
<dd>
<div class="block">This fundamental method create all the files necessary to make Listeriomics software and website
 work!</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#run()" class="member-name-link">run()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>
<div class="block">Run createListeriomicsTSSTermSeq() to construct wig files from tab-delimited files</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.html#run(double%5B%5D,double%5B%5D)" class="member-name-link">run(double[], double[])</a> - Method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.html" title="class in bacnet.scripts.core.stat">StatTest</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html#run(bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String,java.lang.String)" class="member-name-link">run(GenomeNCBI, String, String)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversion</a></dt>
<dd>
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html#run(bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">run(GenomeNCBI, String, String, String)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversion</a></dt>
<dd>
<div class="block">Method for Genome already load<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html#run(java.lang.String)" class="member-name-link">run(String)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversion</a></dt>
<dd>
<div class="block">Method for Genome in the GenomeNCBIFolder<br>
 <br>
 
 Convert a GenomeNCBI into a Genome object -> convert each ChromosomeBacteriaSequence into
 Chromosome:<br>
 <it> each chromosome sequence are the same
 
 <it> coding and noncoding DNASequence are convert into Gene and NcRNA and serialized <it> sRNA
 are leaved unchanged <it> Operons, terminator and other elements are serialized <it> Annotation
 is created, saved in a table (only for verification) and serialized</div>
</dd>
<dt><a href="../bacnet/scripts/core/Rscript.html#run(java.lang.String)" class="member-name-link">run(String)</a> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Rscript.html" title="class in bacnet.scripts.core">Rscript</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStatProteolyse.html#run(java.lang.String)" class="member-name-link">run(String)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStatProteolyse.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStatProteolyse</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html#run(java.lang.String,boolean,double,java.lang.String,java.lang.String,boolean)" class="member-name-link">run(String, boolean, double, String, String, boolean)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html" title="class in bacnet.scripts.blast">MultiSequenceBlastProtein</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#run(java.lang.String,double,java.lang.String,java.lang.String)" class="member-name-link">run(String, double, String, String)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#run(java.lang.String,java.lang.String)" class="member-name-link">run(String, String)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html#run(java.lang.String,java.lang.String)" class="member-name-link">run(String, String)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html" title="class in bacnet.scripts.listeriomics.nterm">NTermCreateFinalData</a></dt>
<dd>
<div class="block">From the list created by Francis, update all Nterm peptides, indicating which type of TIS
 (translation initiation sites) they are: annotated TIS multiple TIS updated TIS internal TIS</div>
</dd>
<dt><a href="../bacnet/scripts/core/PredaTerm.html#run(java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">run(String, String, String)</a> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/PredaTerm.html" title="class in bacnet.scripts.core">PredaTerm</a></dt>
<dd>
<div class="block">Please enter fasta file as argument, with 800 nucleotides flanking the intergenic sequences") ;</div>
</dd>
<dt><a href="../bacnet/Database.InitDatabaseThread.html#run(org.eclipse.core.runtime.IProgressMonitor)" class="member-name-link">run(IProgressMonitor)</a> - Method in class bacnet.<a href="../bacnet/Database.InitDatabaseThread.html" title="class in bacnet">Database.InitDatabaseThread</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html#run(org.eclipse.core.runtime.IProgressMonitor)" class="member-name-link">run(IProgressMonitor)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html" title="class in bacnet.datamodel.sequence">Genome.GetMultiFastaThread</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.OpenGenomesThread.html#run(org.eclipse.core.runtime.IProgressMonitor)" class="member-name-link">run(IProgressMonitor)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.OpenGenomesThread.html" title="class in bacnet.datamodel.sequence">Genome.OpenGenomesThread</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/OpenExpressionMatrixAndComparisons.html#run(org.eclipse.core.runtime.IProgressMonitor)" class="member-name-link">run(IProgressMonitor)</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/OpenExpressionMatrixAndComparisons.html" title="class in bacnet.expressionAtlas.core">OpenExpressionMatrixAndComparisons</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.InitOmicsMatrixThread.html#run(org.eclipse.core.runtime.IProgressMonitor)" class="member-name-link">run(IProgressMonitor)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.InitOmicsMatrixThread.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView.InitOmicsMatrixThread</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.InitNetworkThread.html#run(org.eclipse.core.runtime.IProgressMonitor)" class="member-name-link">run(IProgressMonitor)</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.InitNetworkThread.html" title="class in bacnet.views">CoExprNetworkView.InitNetworkThread</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html#run(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">run(EPartService)</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html" title="class in bacnet.scripts.phylogeny">PhylogenySmallORFs</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html#runBlast(java.util.ArrayList)" class="member-name-link">runBlast(ArrayList&lt;Srna&gt;)</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html" title="class in bacnet.scripts.listeriomics.srna">Srna10403SOliver</a></dt>
<dd>
<div class="block">Run Blast of all Srna found in Oliver et al study against EGD-e and full 10403S sequence</div>
</dd>
<dt><a href="../bacnet/scripts/database/DatabaseCreation.html#runCreation()" class="member-name-link">runCreation()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DatabaseCreation.html" title="class in bacnet.scripts.database">DatabaseCreation</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
<div class="block">This method in an example of all methods which should be run to create your multi-omics website
 <br>
 It is never run !</div>
</dd>
<dt><a href="../bacnet/scripts/database/PhylogenomicsCreation.html#runFigTree(java.lang.String,java.lang.String)" class="member-name-link">runFigTree(String, String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/PhylogenomicsCreation.html" title="class in bacnet.scripts.database">PhylogenomicsCreation</a></dt>
<dd>
<div class="block">Run Tree figure construction</div>
</dd>
<dt><a href="../bacnet/scripts/database/PhylogenomicsCreation.html#runJOLYTree(java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">runJOLYTree(String, String, String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/PhylogenomicsCreation.html" title="class in bacnet.scripts.database">PhylogenomicsCreation</a></dt>
<dd>
<div class="block">Run Genomes alignment and tree reconstruciton with JolyTree
 https://gitlab.pasteur.fr/GIPhy/JolyTree</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/GenomesListeriomics.html#runPostInit()" class="member-name-link">runPostInit()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/GenomesListeriomics.html" title="class in bacnet.scripts.listeriomics">GenomesListeriomics</a></dt>
<dd>
<div class="block">Run all the methods concerning genomes in Listeriomics<br>
 Run after loading of Workbench!</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#runPostInit()" class="member-name-link">runPostInit()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>
<div class="block">Run after loading of workbench (use when display are needed)</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/SystemsBiologyListeriomics.html#runPostInit()" class="member-name-link">runPostInit()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SystemsBiologyListeriomics.html" title="class in bacnet.scripts.listeriomics">SystemsBiologyListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/Test.html#runPostTest()" class="member-name-link">runPostTest()</a> - Static method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/Test.html" title="class in bacnet.e4.rap.setup">Test</a></dt>
<dd>
<div class="block">Run post--test after interface creation</div>
</dd>
<dt><a href="../bacnet/e4/rap/TestPart.html#runPostTests(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">runPostTests(EPartService)</a> - Static method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/TestPart.html" title="class in bacnet.e4.rap">TestPart</a></dt>
<dd>
<div class="block">Ran after the ListeriomicsSample page has open</div>
</dd>
<dt><a href="../bacnet/e4/rap/setup/Test.html#runPreTest()" class="member-name-link">runPreTest()</a> - Static method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/Test.html" title="class in bacnet.e4.rap.setup">Test</a></dt>
<dd>
<div class="block">Run pre-test methods before interface creation</div>
</dd>
<dt><a href="../bacnet/utils/CMD.html#runProcess(java.lang.String)" class="member-name-link">runProcess(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/CMD.html" title="class in bacnet.utils">CMD</a></dt>
<dd>
<div class="block">Run a process</div>
</dd>
<dt><a href="../bacnet/utils/CMD.html#runProcess(java.lang.String%5B%5D)" class="member-name-link">runProcess(String[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/CMD.html" title="class in bacnet.utils">CMD</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/CMD.html#runProcess(java.lang.String%5B%5D,boolean,java.lang.String)" class="member-name-link">runProcess(String[], boolean, String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/CMD.html" title="class in bacnet.utils">CMD</a></dt>
<dd>
<div class="block">Run a process.</div>
</dd>
<dt><a href="../bacnet/utils/CMD.html#runProcess(java.lang.String,boolean,java.lang.String)" class="member-name-link">runProcess(String, boolean, String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/CMD.html" title="class in bacnet.utils">CMD</a></dt>
<dd>
<div class="block">Run a process.</div>
</dd>
<dt><a href="../bacnet/e4/rap/setup/SetupPart.html#runTests()" class="member-name-link">runTests()</a> - Static method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SetupPart.html" title="class in bacnet.e4.rap.setup">SetupPart</a></dt>
<dd>
<div class="block">Ran when the BannerView is opened</div>
</dd>
<dt><a href="../bacnet/e4/rap/TestPart.html#runTests()" class="member-name-link">runTests()</a> - Static method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/TestPart.html" title="class in bacnet.e4.rap">TestPart</a></dt>
<dd>
<div class="block">Ran when the BannerView is opened</div>
</dd>
<dt><a href="../bacnet/utils/RWTUtils.html" class="type-name-link" title="class in bacnet.utils">RWTUtils</a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>
<div class="block">Tools for saving files<br>
 Everything in bacnet.raprcp package is specific to eclipse.rap or eclipse.rcp.</div>
</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#%3Cinit%3E()" class="member-name-link">RWTUtils()</a> - Constructor for class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
