<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 14:20:51 CEST 2019 -->
<title>D-Index</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="D-Index";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="contentContainer"><a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a><a id="I:D">
<!--   -->
</a>
<h2 class="title">D</h2>
<dl>
<dt><span class="memberNameLink"><a href="../bacnet/utils/BasicColor.html#DARK_ASRNA">DARK_ASRNA</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Dark color used for ASrna</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/BasicColor.html#DARK_CISREG">DARK_CISREG</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Dark color used for NCrna</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/BasicColor.html#DARK_NCRNA">DARK_NCRNA</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Dark color used for NCrna</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/BasicColor.html#DARK_OPERON">DARK_OPERON</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Dark color used for ASrna</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/BasicColor.html#DARK_SRNA">DARK_SRNA</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Dark color used for Srna</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.DisplayType.html#DATA">DATA</a></span> - bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.DisplayType.html" title="enum in bacnet.genomeBrowser.core">Track.DisplayType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#DATA_CURATED_PATH">DATA_CURATED_PATH</a></span> - Static variable in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html#DATA_CURATED_PATH">DATA_CURATED_PATH</a></span> - Static variable in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html#DATA_CURATED_PATH">DATA_CURATED_PATH</a></span> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics">ArrayExpressListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#DATA_LIMIT">DATA_LIMIT</a></span> - Static variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#DATA_LIMIT">DATA_LIMIT</a></span> - Static variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html#DATA_PATH">DATA_PATH</a></span> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html" title="class in bacnet.scripts.listeriomics">SummaryListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#DATA_TABLE">DATA_TABLE</a></span> - Static variable in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ProteomicsCreation.html#DATA_TABLE">DATA_TABLE</a></span> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html#DATA_TABLE">DATA_TABLE</a></span> - Static variable in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html#DATA_TABLE">DATA_TABLE</a></span> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics">ArrayExpressListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html" title="class in bacnet"><span class="typeNameLink">Database</span></a> - Class in <a href="../bacnet/package-summary.html">bacnet</a></dt>
<dd>
<div class="block">A ModelProvider specifically design for the Website Database<br>
 Contains persistent variable, used through all the website.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/ProteinTools.html#DATABASE">DATABASE</a></span> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#DATABASE">DATABASE</a></span> - Static variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.InitDatabaseThread.html" title="class in bacnet"><span class="typeNameLink">Database.InitDatabaseThread</span></a> - Class in <a href="../bacnet/package-summary.html">bacnet</a></dt>
<dd>
<div class="block">Class for loading databases in a thread</div>
</dd>
<dt><a href="../bacnet/scripts/database/DatabaseCreation.html" title="class in bacnet.scripts.database"><span class="typeNameLink">DatabaseCreation</span></a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>
<div class="block">All methods for Database Creation of your multi-omics website</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DatabaseCreation.html#%3Cinit%3E()">DatabaseCreation()</a></span> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/DatabaseCreation.html" title="class in bacnet.scripts.database">DatabaseCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features"><span class="typeNameLink">DatabaseReferenceInterface</span></a> - Interface in <a href="../org/biojava3/core/sequence/features/package-summary.html">org.biojava3.core.sequence.features</a></dt>
<dd>
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list of cross reference id(s)</div>
</dd>
<dt><a href="../bacnet/scripts/database/DataFolder.html" title="class in bacnet.scripts.database"><span class="typeNameLink">DataFolder</span></a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DataFolder.html#%3Cinit%3E()">DataFolder()</a></span> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataFolder.html" title="class in bacnet.scripts.database">DataFolder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence"><span class="typeNameLink">DataSource</span></a> - Enum in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">GenBank gi|gi-number|gb|accession|locus ENA Data Library
 gi|gi-number|emb|accession|locus DDBJ, DNA Database of Japan
 gi|gi-number|dbj|accession|locus NBRF PIR pir||entry Protein Research
 Foundation prf||name SWISS-PROT UNIPROT sp|accession|name Brookhaven Protein
 Data Bank (1) pdb|entry|chain Brookhaven Protein Data Bank (2)
 entry:chain|PDBID|CHAIN|SEQUENCE Patents pat|country|number GenInfo Backbone
 Id bbs|number General database identifier gnl|database|identifier NCBI
 Reference Sequence ref|accession|locus Local Sequence identifier
 lcl|identifier</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core"><span class="typeNameLink">DataTrack</span></a> - Class in <a href="../bacnet/genomeBrowser/core/package-summary.html">bacnet.genomeBrowser.core</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#%3Cinit%3E(bacnet.genomeBrowser.core.Track)">DataTrack(Track)</a></span> - Constructor for class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database"><span class="typeNameLink">DataValidation</span></a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DataValidation.html#%3Cinit%3E()">DataValidation()</a></span> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features"><span class="typeNameLink">DBReferenceInfo</span></a> - Class in <a href="../org/biojava3/core/sequence/features/package-summary.html">org.biojava3.core.sequence.features</a></dt>
<dd>
<div class="block">If you have a uniprot ID then it is possible to get a collection of other
 id(s) that the protein is known by.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#%3Cinit%3E(java.lang.String,java.lang.String)">DBReferenceInfo(String, String)</a></span> - Constructor for class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>
<div class="block">The source database and id</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DataSource.html#DDBJ">DDBJ</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/SoftHashMap.html#DEBUG">DEBUG</a></span> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/PrettyXMLWriter.html#declareNamespace(java.lang.String,java.lang.String)">declareNamespace(String, String)</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/XMLWriter.html#declareNamespace(java.lang.String,java.lang.String)">declareNamespace(String, String)</a></span> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Hints that a namespace is going to be used in a sub-tree.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/XMLUtils.html#decodeFromFile(java.lang.String)">decodeFromFile(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/XMLUtils.html" title="class in bacnet.utils">XMLUtils</a></dt>
<dd>
<div class="block">Deserialisation d'un objet depuis un fichier</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#decorateImage(org.eclipse.swt.graphics.Image,org.eclipse.swt.graphics.Image)">decorateImage(Image, Image)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> composed of a base image decorated by another image.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#decorateImage(org.eclipse.swt.graphics.Image,org.eclipse.swt.graphics.Image)">decorateImage(Image, Image)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> composed of a base image decorated by another image.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#decorateImage(org.eclipse.swt.graphics.Image,org.eclipse.swt.graphics.Image,int)">decorateImage(Image, Image, int)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> composed of a base image decorated by another image.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#decorateImage(org.eclipse.swt.graphics.Image,org.eclipse.swt.graphics.Image,int)">decorateImage(Image, Image, int)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> composed of a base image decorated by another image.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#decorateRna(boolean)">decorateRna(boolean)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>
<div class="block">Performs an optimisation where RNASequences are not translated into their own
 objects but are views onto the base DNA sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermFilter.html#DEFAULT_FILTER_PATH">DEFAULT_FILTER_PATH</a></span> - Static variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#DEFAULT_LENGTH">DEFAULT_LENGTH</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/SoftHashMap.html#DEFAULT_LIMIT">DEFAULT_LIMIT</a></span> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#DEFAULT_LOGFC_CUTOFF">DEFAULT_LOGFC_CUTOFF</a></span> - Static variable in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView.html#DEFAULT_MODEL">DEFAULT_MODEL</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/edits/Edit.Delete.html#%3Cinit%3E(int)">Delete(int)</a></span> - Constructor for class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.Delete.html" title="class in org.biojava3.core.sequence.edits">Edit.Delete</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/edits/Edit.Delete.html#%3Cinit%3E(int,int)">Delete(int, int)</a></span> - Constructor for class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.Delete.html" title="class in org.biojava3.core.sequence.edits">Edit.Delete</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/Experiment.html#deleteBioCond(bacnet.datamodel.expdesign.BioCondition)">deleteBioCond(BioCondition)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Delete using the bioCondName as a key</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/Experiment.html#deleteBioCond(java.lang.String)">deleteBioCond(String)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Delete a BioCondition using a specific key</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#deleteColumn(double%5B%5D%5B%5D,int)">deleteColumn(double[][], int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#deleteColumn(java.lang.String)">deleteColumn(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#deleteColumn(java.lang.String%5B%5D%5B%5D,int)">deleteColumn(String[][], int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#deleteMissingValue(double%5B%5D)">deleteMissingValue(double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#deleteRow(double%5B%5D%5B%5D,int)">deleteRow(double[][], int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#deleteRow(java.lang.String%5B%5D%5B%5D,int)">deleteRow(String[][], int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/FlatFileCache.html#destroy()">destroy()</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/FlatFileCache.html" title="class in org.biojava3.core.util">FlatFileCache</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#detectable()">detectable()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Create a table where we put all lmos detectable by the mass spectrometer</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/LocationHelper.html#detectCicular(java.util.List)">detectCicular(List&lt;Location&gt;)</a></span> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Loops through the given list of locations and returns true if it looks like
 they represent a circular location.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/LocationHelper.html#detectEnd(java.util.List,boolean)">detectEnd(List&lt;Location&gt;, boolean)</a></span> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">This will attempt to find what the last point is and returns that position.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/LocationHelper.html#detectStart(java.util.List)">detectStart(List&lt;Location&gt;)</a></span> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Assumes that the first element is the start & clones it</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/LocationHelper.html#detectStrand(java.util.List)">detectStrand(List&lt;Location&gt;)</a></span> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Loops through the given list of locations and returns the consensus Strand
 class.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#deviation(double%5B%5D)">deviation(double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Calculate deviation of a vector sigma(X)=sqrt(V(X))</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ExpressionMatrixStat.html#deviation(bacnet.datamodel.dataset.ExpressionMatrix)">deviation(ExpressionMatrix)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ExpressionMatrixStat.html#deviation(bacnet.datamodel.dataset.ExpressionMatrix,int)">deviation(ExpressionMatrix, int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Calculate statistical deviation of a specific column</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#diagonalTimes(double%5B%5D%5B%5D,double%5B%5D)">diagonalTimes(double[][], double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#diagonalTimes(double%5B%5D,double%5B%5D%5B%5D)">diagonalTimes(double[], double[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Multiply a vector(=in fact a diagonal matrice) times a array X'=MX or XM with
 a diagonal = vector</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#diffBeginPosition(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence)">diffBeginPosition(DNASequence, DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Calculate difference in begin position</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#diffCenter(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence)">diffCenter(DNASequence, DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Calculate the distance between the two center of gene1 and gene2</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#diffEndPosition(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence)">diffEndPosition(DNASequence, DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Calculate difference in end position</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastResult.html#diffSize">diffSize</a></span> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#display">display</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#display">display</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/BasicColor.html#display">display</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#display()">display()</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#displayAnnotationView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)">displayAnnotationView(EPartService, HashMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayBHI37View(org.eclipse.e4.ui.workbench.modeling.EPartService)">displayBHI37View(EPartService)</a></span> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Load the minimal number of data for the simplest proteomics view</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#displayBioConditions(java.lang.String,java.util.ArrayList,org.eclipse.e4.ui.workbench.modeling.EPartService)">displayBioConditions(String, ArrayList&lt;String&gt;, EPartService)</a></span> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>
<div class="block">Run a <code>HeatMapView</code> displaying a specific bioCond</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#displayBioConditionsExpressionAtlas(java.util.HashMap,org.eclipse.e4.ui.workbench.modeling.EPartService)">displayBioConditionsExpressionAtlas(HashMap&lt;String, ArrayList&lt;String&gt;&gt;, EPartService)</a></span> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>
<div class="block">Run a <code>HeatMapView</code> displaying a specific bioCond</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayBioConditionsFromText(org.eclipse.e4.ui.workbench.modeling.EPartService,java.lang.String)">displayBioConditionsFromText(EPartService, String)</a></span> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Open a GenomeView given a genome and text file from .gview file</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#displayCoExpNetworkView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)">displayCoExpNetworkView(EPartService, HashMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#displayComparisonsAndElement(java.lang.String,java.util.ArrayList,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)">displayComparisonsAndElement(String, ArrayList&lt;String&gt;, String, EPartService)</a></span> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayCRISPROmics(org.eclipse.e4.ui.workbench.modeling.EPartService)">displayCRISPROmics(EPartService)</a></span> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Load the minimal number of data for Amastigote versus Promastigote view</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayCRISPROmics(org.eclipse.e4.ui.workbench.modeling.EPartService,int)">displayCRISPROmics(EPartService, int)</a></span> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Open a GenomeView given a genome and list of biological conditions</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#displayCRISPRTable(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)">displayCRISPRTable(EPartService, HashMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#displayedAttributes">displayedAttributes</a></span> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#displayGene(bacnet.datamodel.sequence.Gene,org.eclipse.e4.ui.workbench.modeling.EPartService)">displayGene(Gene, EPartService)</a></span> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Display a given gene</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#displayGene(org.eclipse.swt.graphics.GC,bacnet.datamodel.sequence.Sequence,org.eclipse.swt.graphics.Rectangle,bacnet.genomeBrowser.core.Track,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color,boolean)">displayGene(GC, Sequence, Rectangle, Track, Color, Color, Color, Color, boolean)</a></span> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Display Gene annotation = square and arrow</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#displayGeneView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)">displayGeneView(EPartService, HashMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayGenomeElementAndBioConditions(org.eclipse.e4.ui.workbench.modeling.EPartService,java.lang.String,java.util.ArrayList,java.lang.String)">displayGenomeElementAndBioConditions(EPartService, String, ArrayList&lt;String&gt;, String)</a></span> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Open a GenomeView given a genome and list of biological conditions</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayGenomeTranscriptomesView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)">displayGenomeTranscriptomesView(EPartService, HashMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GenomicsView.html#displayGenomeView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)">displayGenomeView(EPartService, HashMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>
<div class="block">Display a search in Genomic View</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#displayHeatMapProteomicsView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)">displayHeatMapProteomicsView(EPartService, HashMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#displayHeatMapTranscriptomicsView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)">displayHeatMapTranscriptomicsView(EPartService, HashMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayIntracellularMacrophagesView(org.eclipse.e4.ui.workbench.modeling.EPartService)">displayIntracellularMacrophagesView(EPartService)</a></span> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayLeishmaniaAmavsPro(org.eclipse.e4.ui.workbench.modeling.EPartService)">displayLeishmaniaAmavsPro(EPartService)</a></span> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Load the minimal number of data for Amastigote versus Promastigote view</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayLeishmaniaLeisheild(org.eclipse.e4.ui.workbench.modeling.EPartService)">displayLeishmaniaLeisheild(EPartService)</a></span> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Load the minimal number of data for Amastigote versus Promastigote view</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/TableSWTView.html#displayMatrix(bacnet.datamodel.dataset.ExpressionMatrix,java.lang.String,boolean,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)">displayMatrix(ExpressionMatrix, String, boolean, String, EPartService)</a></span> - Static method in class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/TableSWTView.html#displayMatrix(bacnet.datamodel.dataset.ExpressionMatrix,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)">displayMatrix(ExpressionMatrix, String, EPartService)</a></span> - Static method in class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>
<div class="block">Static function which open an HeatMapView and display corresponding
 ExpressionMatrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#displayMatrix(java.lang.String,double%5B%5D%5B%5D)">displayMatrix(String, double[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Display a Matrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#displayMatrix(java.lang.String,java.lang.Object%5B%5D%5B%5D)">displayMatrix(String, Object[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#displayMatrix(java.lang.String,java.lang.String%5B%5D%5B%5D)">displayMatrix(String, String[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/PhylogenyRli42vsRsbR.html#displayMultiAlign(java.util.ArrayList,org.eclipse.e4.ui.workbench.modeling.EPartService)">displayMultiAlign(ArrayList&lt;Sequence&gt;, EPartService)</a></span> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyRli42vsRsbR.html" title="class in bacnet.scripts.phylogeny">PhylogenyRli42vsRsbR</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html#displayMultiAlign(java.util.ArrayList,org.eclipse.e4.ui.workbench.modeling.EPartService)">displayMultiAlign(ArrayList&lt;Sequence&gt;, EPartService)</a></span> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html" title="class in bacnet.scripts.phylogeny">PhylogenySmallORFs</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html#displayMultiAlign(org.eclipse.e4.ui.workbench.modeling.EPartService)">displayMultiAlign(EPartService)</a></span> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html" title="class in bacnet.scripts.phylogeny">PhylogenySmallORFs</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayNTerminomics(org.eclipse.e4.ui.workbench.modeling.EPartService,java.lang.String)">displayNTerminomics(EPartService, String)</a></span> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Load the minimal number of data for the simplest proteomics view</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#displayNTerminomicsView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)">displayNTerminomicsView(EPartService, HashMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#displayOmicsView(org.eclipse.e4.ui.workbench.modeling.EPartService)">displayOmicsView(EPartService)</a></span> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#displayOmicsView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)">displayOmicsView(EPartService, HashMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#displayOperon(org.eclipse.swt.graphics.GC,java.lang.String,int%5B%5D,int,int,bacnet.genomeBrowser.core.Zoom)">displayOperon(GC, String, int[], int, int, Zoom)</a></span> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Display operon in orange = rectangle</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/PhylogenyRli42vsRsbR.html#displayPhylogeny(java.util.ArrayList)">displayPhylogeny(ArrayList&lt;Sequence&gt;)</a></span> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyRli42vsRsbR.html" title="class in bacnet.scripts.phylogeny">PhylogenyRli42vsRsbR</a></dt>
<dd>
<div class="block">Displaying the newly created tree with Archaeopteryx.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html#displayPhylogeny(java.util.ArrayList)">displayPhylogeny(ArrayList&lt;Sequence&gt;)</a></span> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html" title="class in bacnet.scripts.phylogeny">PhylogenySmallORFs</a></dt>
<dd>
<div class="block">Displaying the newly created tree with Archaeopteryx.</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/DisplayPhylogenyHandler.html" title="class in bacnet.scripts.phylogeny"><span class="typeNameLink">DisplayPhylogenyHandler</span></a> - Class in <a href="../bacnet/scripts/phylogeny/package-summary.html">bacnet.scripts.phylogeny</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/DisplayPhylogenyHandler.html#%3Cinit%3E()">DisplayPhylogenyHandler()</a></span> - Constructor for class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/DisplayPhylogenyHandler.html" title="class in bacnet.scripts.phylogeny">DisplayPhylogenyHandler</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span></div>
&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#displayProteomicsView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)">displayProteomicsView(EPartService, HashMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>
<div class="block">Display a search in Genomic View</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#displaySequence(org.eclipse.swt.graphics.GC,bacnet.datamodel.sequence.Chromosome,int,int%5B%5D,double,bacnet.genomeBrowser.core.Zoom,int)">displaySequence(GC, Chromosome, int, int[], double, Zoom, int)</a></span> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Display sequence information:
 nucleotide sequence on plus strand
 nucleotide sequence on minus strand
 amino acid sequence for each nucleotide position on the plus strand (=
 the three reading frame of the plus strand)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#displaySpecificRegion(java.lang.String)">displaySpecificRegion(String)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Display a specific element of the genome only by data.getDisplay() to false,
 except in the region of the genomeElement</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#displaySrna(bacnet.datamodel.sequence.Srna,org.eclipse.e4.ui.workbench.modeling.EPartService)">displaySrna(Srna, EPartService)</a></span> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>
<div class="block">Open Srna View with a specific Srna to display</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#displaySrnaView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)">displaySrnaView(EPartService, HashMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayStat37View(org.eclipse.e4.ui.workbench.modeling.EPartService)">displayStat37View(EPartService)</a></span> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#displayTerminator(org.eclipse.swt.graphics.GC,java.lang.String,int%5B%5D,int,int,bacnet.genomeBrowser.core.Zoom)">displayTerminator(GC, String, int[], int, int, Zoom)</a></span> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Display terminator if provided = dot in purple</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayTestView(org.eclipse.e4.ui.workbench.modeling.EPartService)">displayTestView(EPartService)</a></span> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Load the minimal number of data for the simplest proteomics view</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#displayTranscriptomicsView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)">displayTranscriptomicsView(EPartService, HashMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>
<div class="block">Display a search in Genomic View</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#displayValues">displayValues</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#displayValues">displayValues</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#displayVector(java.lang.String,double%5B%5D)">displayVector(String, double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#displayVector(java.lang.String,double%5B%5D)">displayVector(String, double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#displayVector(java.lang.String,java.lang.String%5B%5D)">displayVector(String, String[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#displayVector(java.lang.String,java.lang.String%5B%5D)">displayVector(String, String[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/setup/SetupPart.html#dispose()">dispose()</a></span> - Method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SetupPart.html" title="class in bacnet.e4.rap.setup">SetupPart</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/TestPart.html#dispose()">dispose()</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/TestPart.html" title="class in bacnet.e4.rap">TestPart</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#dispose()">dispose()</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Dispose of cached objects and their underlying OS resources.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#dispose()">dispose()</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Dispose of cached objects and their underlying OS resources.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView.html#dispose()">dispose()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView2.html#dispose()">dispose()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#disposeColors()">disposeColors()</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Dispose of all the cached <code>Color</code>'s.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#disposeCursors()">disposeCursors()</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Dispose all of the cached cursors.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#disposeFonts()">disposeFonts()</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Dispose all of the cached <code>Font</code>'s.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#disposeImages()">disposeImages()</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Dispose all of the cached images.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#disposeImages()">disposeImages()</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Dispose all of the cached <code>Image</code>'s.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#distance(org.biojava3.genome.parsers.gff.Location)">distance(Location)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Return distance between this location and the other location.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/ProteinTools.html#divideTISDB()">divideTISDB()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Divide TISDB into sequence starting with a M (TISMDB), and other not starting
 with an M (TISnoMDB).</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#divideTISDB()">divideTISDB()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Divide TISDB into sequence starting with a M (TISMDB), and other not starting
 with an M (TISnoMDB).</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#dnaCompounds(org.biojava3.core.sequence.template.CompoundSet)">dnaCompounds(CompoundSet&lt;NucleotideCompound&gt;)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/DNACompoundSet.html" title="class in org.biojava3.core.sequence.compound"><span class="typeNameLink">DNACompoundSet</span></a> - Class in <a href="../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/DNACompoundSet.html#%3Cinit%3E()">DNACompoundSet()</a></span> - Constructor for class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/DNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">DNACompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#dnaRnaTranslator(org.biojava3.core.sequence.transcription.DNAToRNATranslator)">dnaRnaTranslator(DNAToRNATranslator)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html#DNASeq">DNASeq</a></span> - bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DNASeqCreation.html" title="class in bacnet.scripts.database"><span class="typeNameLink">DNASeqCreation</span></a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DNASeqCreation.html#%3Cinit%3E()">DNASeqCreation()</a></span> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/DNASeqCreation.html" title="class in bacnet.scripts.database">DNASeqCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence"><span class="typeNameLink">DNASequence</span></a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#%3Cinit%3E()">DNASequence()</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Shouldn't be used but makes it bean happy</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#%3Cinit%3E(java.lang.String)">DNASequence(String)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">String is king and create a sequence from DNA with default DNA compound set</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">DNASequence(String, CompoundSet&lt;NucleotideCompound&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Create a sequence from a string with user defined compound set</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader)">DNASequence(SequenceReader&lt;NucleotideCompound&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Create a sequence where the actual storage of the sequence data is somewhere
 else</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">DNASequence(SequenceReader&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Create a sequence from a ProxySequencereader and user defined compound set</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence"><span class="typeNameLink">DNASequence.DNAType</span></a> - Enum in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">The type of DNA sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html" title="class in org.biojava3.core.sequence.io"><span class="typeNameLink">DNASequenceCreator</span></a> - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">A helper class that allows different ways to read a string and create a DNA
 sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)">DNASequenceCreator(CompoundSet&lt;NucleotideCompound&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">DNASequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava3.core.sequence.transcription"><span class="typeNameLink">DNAToRNATranslator</span></a> - Class in <a href="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></dt>
<dd>
<div class="block">Performs the first stage of transcription by going from DNA to RNA.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#%3Cinit%3E(org.biojava3.core.sequence.io.template.SequenceCreatorInterface,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet,boolean)">DNAToRNATranslator(SequenceCreatorInterface&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;, boolean)</a></span> - Constructor for class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava3.core.sequence.transcription">DNAToRNATranslator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#Donovani_NAME">Donovani_NAME</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Name of Leishmania reference genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html#downloadAdditional(java.lang.String)">downloadAdditional(String)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/GenomesCreation.html#downloadAll()">downloadAll()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Download all genomes available in the database</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#downloadAllFilesTechno()">downloadAllFilesTechno()</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html#downloadAllGenomes(java.lang.String,java.lang.String)">downloadAllGenomes(String, String)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html" title="class in bacnet.scripts.blast">GenomeNCBIFolderTools</a></dt>
<dd>
<div class="block">Take an array with a list of strain and ftp folders, and: <fi> download them:
 fna and gff file only <fi> create a folder for each <fi> verify MD5 checksum
 <fi> unzip files
 
 The strain name should be in the column with the name: strain_name The folder
 name for the genome to be donwload in should be in the column with the name:
 folder_name The ftp path should be found in the column with name : ftp_path
 The id of the assembly should be found in the column with name :
 assembly_accession</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#downloadAllTechno()">downloadAllTechno()</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Download all ADF files</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#downloadAllTechno()">downloadAllTechno()</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Download all ADF files</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#downloadFile(java.lang.String)">downloadFile(String)</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Given an accession and an extension type:
 check if the file does not already exist in the current folder
 Download it if not
 Print out a message if no file has been downloaded</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html#downloadFile(java.lang.String,java.lang.String)">downloadFile(String, String)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></dt>
<dd>
<div class="block">Given an accession and an extension type:
 check if the file does not already exist in the current folder
 Download it if not
 Print out a message if no file has been downloaded</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#downloadFile(java.lang.String,java.lang.String,java.lang.String)">downloadFile(String, String, String)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Given an accession and an extension type:
 check if the file does not already exist in the current folder
 Download it if not
 Print out a message if no file has been downloaded</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/GenomesCreation.html#downloadGenomes(java.util.ArrayList)">downloadGenomes(ArrayList&lt;String&gt;)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Download new genomes taken from a text file and open all gzip files</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html#downloadIDF(java.lang.String)">downloadIDF(String)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html#downloadProcessedFile(java.lang.String)">downloadProcessedFile(String)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html#downloadRawFile(java.lang.String)">downloadRawFile(String)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html#downloadSDRF(java.lang.String)">downloadSDRF(String)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/rap/DownloadServiceHandler.html" title="class in bacnet.rap"><span class="typeNameLink">DownloadServiceHandler</span></a> - Class in <a href="../bacnet/rap/package-summary.html">bacnet.rap</a></dt>
<dd>
<div class="block">Specific tools for downloading files from eclipse.rap instance NEED TO BE
 COMMENTED FOR Eclipse.RCP to work !!!</div>
</dd>
<dt><a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp"><span class="typeNameLink">DownloadServiceHandler</span></a> - Class in <a href="../bacnet/raprcp/package-summary.html">bacnet.raprcp</a></dt>
<dd>
<div class="block">Specific tools for downloading files from eclipse.rap instance NEED TO BE
 COMMENTED FOR Eclipse.RCP to work !!!</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/DownloadServiceHandler.html#%3Cinit%3E(java.lang.String,java.io.File)">DownloadServiceHandler(String, File)</a></span> - Constructor for class bacnet.rap.<a href="../bacnet/rap/DownloadServiceHandler.html" title="class in bacnet.rap">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/DownloadServiceHandler.html#%3Cinit%3E(java.lang.String,java.io.File)">DownloadServiceHandler(String, File)</a></span> - Constructor for class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/DownloadServiceHandler.html#%3Cinit%3E(java.lang.String,java.lang.String)">DownloadServiceHandler(String, String)</a></span> - Constructor for class bacnet.rap.<a href="../bacnet/rap/DownloadServiceHandler.html" title="class in bacnet.rap">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/DownloadServiceHandler.html#%3Cinit%3E(java.lang.String,java.lang.String)">DownloadServiceHandler(String, String)</a></span> - Constructor for class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#downstream(int)">downstream(int)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Return the adjacent location of specified length directly downstream of this
 location.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#DP10403S_NAME">DP10403S_NAME</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Name of 10403S genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#DP10403S_NAME_BIS">DP10403S_NAME_BIS</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Other name for 10403S genome</div>
</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a></div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
