<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Fri Dec 10 09:15:34 CET 2021 -->
<title>G-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-10">
<meta name="description" content="index: G">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:G">G</h2>
<dl class="index">
<dt><a href="../bacnet/scripts/blast/BlastResult.html#gaps" class="member-name-link">gaps</a> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/GBKReader.html" class="type-name-link" title="class in bacnet.reader">GBKReader</a> - Class in <a href="../bacnet/reader/package-summary.html">bacnet.reader</a></dt>
<dd>
<div class="block">GenBankl reader tools <br>
 TO BE FIXED for adding other chromosomes</div>
</dd>
<dt><a href="../bacnet/reader/GBKReader.html#%3Cinit%3E()" class="member-name-link">GBKReader()</a> - Constructor for class bacnet.reader.<a href="../bacnet/reader/GBKReader.html" title="class in bacnet.reader">GBKReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html#GCG" class="member-name-link">GCG</a> - Enum constant in enum org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html" title="enum in org.biojava3.core.sequence.template">LightweightProfile.StringFormat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GCStats.html" class="type-name-link" title="class in org.biojava3.genome.parsers.gff">GCStats</a> - Class in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GCStats.html#%3Cinit%3E()" class="member-name-link">GCStats()</a> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GCStats.html" title="class in org.biojava3.genome.parsers.gff">GCStats</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" class="type-name-link" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a> - Class in <a href="../bacnet/genomeBrowser/tracksGUI/package-summary.html">bacnet.genomeBrowser.tracksGUI</a></dt>
<dd>
<div class="block">Here are all the methods for displaying each Genome element</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#%3Cinit%3E()" class="member-name-link">GElement()</a> - Constructor for class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DataSource.html#GENBANK" class="member-name-link">GENBANK</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html" class="type-name-link" title="class in bacnet.datamodel.sequence">Gene</a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.SeqType.html#Gene" class="member-name-link">Gene</a> - Enum constant in enum bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#%3Cinit%3E()" class="member-name-link">Gene()</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#%3Cinit%3E(java.lang.String,int,int,char)" class="member-name-link">Gene(String, int, int, char)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#GENE" class="member-name-link">GENE</a> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html#GeneExpr" class="member-name-link">GeneExpr</a> - Enum constant in enum bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html" class="type-name-link" title="class in bacnet.datamodel.dataset">GeneExpression</a> - Class in <a href="../bacnet/datamodel/dataset/package-summary.html">bacnet.datamodel.dataset</a></dt>
<dd>
<div class="block">GeneExpression array data from Listeria tiling chip ArrayExpress id : A-AFFY-140
 
 Here double[] values vector has length = 5*probes.size because we put 5 columns column 0-2 = 3
 replicatas column 3 = median expression column 4 = var expression</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#%3Cinit%3E()" class="member-name-link">GeneExpression()</a> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#%3Cinit%3E(java.lang.String)" class="member-name-link">GeneExpression(String)</a> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html" class="type-name-link" title="class in org.biojava3.genome">GeneFeatureHelper</a> - Class in <a href="../org/biojava3/genome/package-summary.html">org.biojava3.genome</a></dt>
<dd>
<div class="block">Allow to process genome elements annotation</div>
</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html#%3Cinit%3E()" class="member-name-link">GeneFeatureHelper()</a> - Constructor for class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html" class="type-name-link" title="class in org.biojava3.genome.parsers.gff">GeneIDGFF2Reader</a> - Class in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>
<div class="block">http://www.bioperl.org/wiki/GTF Read and write FeatureLists as GFF/GTF formatted files.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html#%3Cinit%3E()" class="member-name-link">GeneIDGFF2Reader()</a> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html" title="class in org.biojava3.genome.parsers.gff">GeneIDGFF2Reader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html" class="type-name-link" title="class in org.biojava3.genome.parsers.geneid">GeneIDXMLReader</a> - Class in <a href="../org/biojava3/genome/parsers/geneid/package-summary.html">org.biojava3.genome.parsers.geneid</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html#%3Cinit%3E(java.lang.String)" class="member-name-link">GeneIDXMLReader(String)</a> - Constructor for class org.biojava3.genome.parsers.geneid.<a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html" title="class in org.biojava3.genome.parsers.geneid">GeneIDXMLReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html" class="type-name-link" title="class in org.biojava3.genome.parsers.gff">GeneMarkGTFReader</a> - Class in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>
<div class="block">http://www.bioperl.org/wiki/GTF Read and write FeatureLists as GFF/GTF formatted files.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html#%3Cinit%3E()" class="member-name-link">GeneMarkGTFReader()</a> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html" title="class in org.biojava3.genome.parsers.gff">GeneMarkGTFReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" class="type-name-link" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a> - Class in <a href="../bacnet/datamodel/sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></dt>
<dd>
<div class="block">List of tools useful to extract information hidden in the NoteList of a DNASequence</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#%3Cinit%3E()" class="member-name-link">GeneNCBITools()</a> - Constructor for class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DataSource.html#GENERAL" class="member-name-link">GENERAL</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#GENERAL_WT_DATA" class="member-name-link">GENERAL_WT_DATA</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Name of the Transcriptome data used as a reference data for comparisons</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#GENERAL_WT_NAME" class="member-name-link">GENERAL_WT_NAME</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Name used for "median" data created from a set of reference condition</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/GeneralArray.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.technology">GeneralArray</a> - Class in <a href="../bacnet/scripts/listeriomics/technology/package-summary.html">bacnet.scripts.listeriomics.technology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/GeneralArray.html#%3Cinit%3E()" class="member-name-link">GeneralArray()</a> - Constructor for class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/GeneralArray.html" title="class in bacnet.scripts.listeriomics.technology">GeneralArray</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderFormat</a>&lt;<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="type parameter in GenericFastaHeaderFormat">S</a> extends <a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;?&gt;,<wbr><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="type parameter in GenericFastaHeaderFormat">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">We store the original header if the sequence is parsed from a fasta file and will use that exact
 sequence if we write out the sequences to a fasta file.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html#%3Cinit%3E()" class="member-name-link">GenericFastaHeaderFormat()</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderFormat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser</a>&lt;<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="type parameter in GenericFastaHeaderParser">S</a> extends <a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="type parameter in GenericFastaHeaderParser">C</a>&gt;,<wbr><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="type parameter in GenericFastaHeaderParser">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">The default fasta header parser where some headers are well defined based on the source database
 which allows us to set the source of the protein sequence and the identifier that can be used in
 future implementations to load features from external sources
 
 If the user has a custom header with local data then they can create their own implementation of
 a FastaHeaderParserInterface

 GenBank gi|gi-number|gb|accession|locus ENA Data Library gi|gi-number|emb|accession|locus DDBJ,
 DNA Database of Japan gi|gi-number|dbj|accession|locus NBRF PIR pir||entry Protein Research
 Foundation prf||name SWISS-PROT sp|accession|name Brookhaven Protein Data Bank (1)
 pdb|entry|chain Brookhaven Protein Data Bank (2) entry:chain|PDBID|CHAIN|SEQUENCE PDB EBI
 PDB:1ECY_A mol:protein length:142 ECOTIN Patents pat|country|number GenInfo Backbone Id
 bbs|number General database identifier gnl|database|identifier NCBI Reference Sequence
 ref|accession|locus Local Sequence identifier lcl|identifier</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html#%3Cinit%3E()" class="member-name-link">GenericFastaHeaderParser()</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html" class="type-name-link" title="class in org.biojava3.core.sequence">GeneSequence</a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#%3Cinit%3E(org.biojava3.core.sequence.ChromosomeSequence,int,int,org.biojava3.core.sequence.Strand)" class="member-name-link">GeneSequence(ChromosomeSequence, int, int, Strand)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">A class that keeps track of the details of a GeneSequence which is difficult to properly model.</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html" class="type-name-link" title="class in bacnet.sequenceTools">GeneView</a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#%3Cinit%3E()" class="member-name-link">GeneView()</a> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewHomologTools.html" class="type-name-link" title="class in bacnet.sequenceTools">GeneViewHomologTools</a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>
<div class="block">All the tools to display homologs in the Gene Panel view: - It creates the homolog table - Create
 the homolog figure - Display both of them</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewHomologTools.html#%3Cinit%3E()" class="member-name-link">GeneViewHomologTools()</a> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneViewHomologTools.html" title="class in bacnet.sequenceTools">GeneViewHomologTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewLocalizationTools.html" class="type-name-link" title="class in bacnet.sequenceTools">GeneViewLocalizationTools</a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewLocalizationTools.html#%3Cinit%3E()" class="member-name-link">GeneViewLocalizationTools()</a> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneViewLocalizationTools.html" title="class in bacnet.sequenceTools">GeneViewLocalizationTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewProteomeTools.html" class="type-name-link" title="class in bacnet.sequenceTools">GeneViewProteomeTools</a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewProteomeTools.html#%3Cinit%3E()" class="member-name-link">GeneViewProteomeTools()</a> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneViewProteomeTools.html" title="class in bacnet.sequenceTools">GeneViewProteomeTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewTranscriptomeTools.html" class="type-name-link" title="class in bacnet.sequenceTools">GeneViewTranscriptomeTools</a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewTranscriptomeTools.html#%3Cinit%3E()" class="member-name-link">GeneViewTranscriptomeTools()</a> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneViewTranscriptomeTools.html" title="class in bacnet.sequenceTools">GeneViewTranscriptomeTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DataSource.html#GENINFO" class="member-name-link">GENINFO</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#genome" class="member-name-link">genome</a> - Variable in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html" class="type-name-link" title="class in bacnet.datamodel.sequence">Genome</a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../bacnet/reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#%3Cinit%3E()" class="member-name-link">Genome()</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#%3Cinit%3E(java.lang.String,boolean)" class="member-name-link">Genome(String, boolean)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Create a Genome from .fna, .annot, and all serialized Sequence object:<br>
 Read Genome sequence information in the different fna files <br>
 Load annotation information<br>
 <br>
 As we are dealing with bacteria, we are sure that the chromosome with maximum length is THE
 chromosome, other are plasmids.</div>
</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#GENOME" class="member-name-link">GENOME</a> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html#GENOME_PATH" class="member-name-link">GENOME_PATH</a> - Static variable in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html" title="class in bacnet.scripts.listeriomics.srna">SmallPeptidesSearch</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html" class="type-name-link" title="class in bacnet.datamodel.sequence">Genome.GetMultiFastaThread</a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.OpenGenomesThread.html" class="type-name-link" title="class in bacnet.datamodel.sequence">Genome.OpenGenomesThread</a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html" class="type-name-link" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversion</a> - Class in <a href="../bacnet/datamodel/sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></dt>
<dd>
<div class="block">Static methods for converting GenomeNCBI to Genome<br>
 Genome is used for fast loading of all genomic information we gathered</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html#%3Cinit%3E()" class="member-name-link">GenomeConversion()</a> - Constructor for class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversion</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" class="type-name-link" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a> - Class in <a href="../bacnet/datamodel/sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#%3Cinit%3E()" class="member-name-link">GenomeConversionElement()</a> - Constructor for class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" class="type-name-link" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a> - Class in <a href="../bacnet/expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#%3Cinit%3E()" class="member-name-link">GenomeElementAtlas()</a> - Constructor for class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#%3Cinit%3E(bacnet.datamodel.sequence.Sequence,bacnet.utils.Filter,boolean)" class="member-name-link">GenomeElementAtlas(Sequence, Filter, boolean)</a> - Constructor for class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>
<div class="block">For a specific sequence update the list of BioCondition in which the genome element is
 over-expressed, under-expressed or not differently expressed<br>
 The update will be dependent of the value of the input filter = cutoff for LogFC and p-value</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html#GenomeElements" class="member-name-link">GenomeElements</a> - Enum constant in enum bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html" title="enum in bacnet.datamodel.dataset">OmicsData.ColNames</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" class="type-name-link" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a> - Class in <a href="../bacnet/datamodel/sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></dt>
<dd>
<div class="block">Object containing an entire genome, created from NCBI RefSeq data<br>
 Data for this genome come from: <it> NCBI.fna (for the sequence) <it> NCBI.gff (for Gene, CDS,
 and ncRNA lists) If it exists some information are added: <it> Rfam.gff (other info) <it> supp
 info in a .txt file <it> supp info about RAST results in a .txt file For EGD-e only: <it>
 Srna.xml are read to add to the genome</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#%3Cinit%3E()" class="member-name-link">GenomeNCBI()</a> - Constructor for class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#%3Cinit%3E(java.lang.String,boolean)" class="member-name-link">GenomeNCBI(String, boolean)</a> - Constructor for class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Create a genome with all chromosomes found If annotation = TRUE , we had the annotation we can
 found in other fasta files annotation = FALSE, we read only the complete genome + eventual
 plasmids</div>
</dd>
<dt><a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html" class="type-name-link" title="class in bacnet.scripts.blast">GenomeNCBIFolderTools</a> - Class in <a href="../bacnet/scripts/blast/package-summary.html">bacnet.scripts.blast</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html#%3Cinit%3E()" class="member-name-link">GenomeNCBIFolderTools()</a> - Constructor for class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html" title="class in bacnet.scripts.blast">GenomeNCBIFolderTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" class="type-name-link" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a> - Class in <a href="../bacnet/datamodel/sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#%3Cinit%3E()" class="member-name-link">GenomeNCBITools()</a> - Constructor for class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html" class="type-name-link" title="class in bacnet.scripts.phylogeny">GenomePhylogeny</a> - Class in <a href="../bacnet/scripts/phylogeny/package-summary.html">bacnet.scripts.phylogeny</a></dt>
<dd>
<div class="block">Manage phylogeny from MLST scheme : Jolley and others</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html#%3Cinit%3E()" class="member-name-link">GenomePhylogeny()</a> - Constructor for class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html" title="class in bacnet.scripts.phylogeny">GenomePhylogeny</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/GenomesCreation.html" class="type-name-link" title="class in bacnet.scripts.database">GenomesCreation</a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>
<div class="block">List of tools available for genome creation</div>
</dd>
<dt><a href="../bacnet/scripts/database/GenomesCreation.html#%3Cinit%3E()" class="member-name-link">GenomesCreation()</a> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/GenomesListeriomics.html" class="type-name-link" title="class in bacnet.scripts.listeriomics">GenomesListeriomics</a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/GenomesListeriomics.html#%3Cinit%3E()" class="member-name-link">GenomesListeriomics()</a> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/GenomesListeriomics.html" title="class in bacnet.scripts.listeriomics">GenomesListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" class="type-name-link" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a> - Class in <a href="../bacnet/genomeBrowser/package-summary.html">bacnet.genomeBrowser</a></dt>
<dd>
<div class="block">Display on the genome viewer: expression of specific bioCond and the log of fold change</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#%3Cinit%3E()" class="member-name-link">GenomeTranscriptomeView()</a> - Constructor for class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html" class="type-name-link" title="class in bacnet.sequenceTools">GenomicsView</a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#%3Cinit%3E()" class="member-name-link">GenomicsView()</a> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/normalization/GENormalization.html" class="type-name-link" title="class in bacnet.scripts.core.normalization">GENormalization</a> - Class in <a href="../bacnet/scripts/core/normalization/package-summary.html">bacnet.scripts.core.normalization</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/normalization/GENormalization.html#%3Cinit%3E()" class="member-name-link">GENormalization()</a> - Constructor for class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/GENormalization.html" title="class in bacnet.scripts.core.normalization">GENormalization</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#get(int)" class="member-name-link">get(int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/WindowedSequence.html#get(int)" class="member-name-link">get(int)</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence</a></dt>
<dd>
<div class="block">Returns the window specified at the given index in offsets i.e. asking for position 2 in a moving
 window sequence of size 3 will get you the window starting at position 4.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#get(int,int,boolean)" class="member-name-link">get(int, int, boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">read probe values between two positions in the genome <br>
 important: bpbegin &lt; bpend</div>
</dd>
<dt><a href="../org/biojava3/core/util/SoftHashMap.html#get(java.lang.Object)" class="member-name-link">get(Object)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#get10403SOverlap()" class="member-name-link">get10403SOverlap()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#get10403SSrnas()" class="member-name-link">get10403SSrnas()</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Return a list of all Srna in 10403S</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#get3UTRSequence(int)" class="member-name-link">get3UTRSequence(int)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Get the nucleotide sequence n bp downstream to the <code>Sequence</code>, +3bp for including
 start codon</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/CompoundColorization.html#getAAcidColor(java.lang.String,org.eclipse.swt.widgets.Display)" class="member-name-link">getAAcidColor(String, Display)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/CompoundColorization.html" title="class in bacnet.datamodel.sequence">CompoundColorization</a></dt>
<dd>
<div class="block">Given an Amino Acid, this method return the corresponding color</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getAccession()" class="member-name-link">getAccession()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getAccession()" class="member-name-link">getAccession()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getAccession()" class="member-name-link">getAccession()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getAccession()" class="member-name-link">getAccession()</a> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getAccession()" class="member-name-link">getAccession()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getAccession()" class="member-name-link">getAccession()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getAccession()" class="member-name-link">getAccession()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getAccession()" class="member-name-link">getAccession()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Unsupoorted</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getAccession()" class="member-name-link">getAccession()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Accessioned.html#getAccession()" class="member-name-link">getAccession()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</a></dt>
<dd>
<div class="block">Returns the AccessionID this location is currently bound with</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getAccession()" class="member-name-link">getAccession()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#getAlias()" class="member-name-link">getAlias()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#getAlignedSequence(int)" class="member-name-link">getAlignedSequence(int)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">Uses bioIndex starting at 1 instead of 0</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.html#getAlignedSequence(int)" class="member-name-link">getAlignedSequence(int)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a></dt>
<dd>
<div class="block">Returns <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a> at given index.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#getAlignedSequences()" class="member-name-link">getAlignedSequences()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">Get the list of sequences</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.html#getAlignedSequences()" class="member-name-link">getAlignedSequences()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a></dt>
<dd>
<div class="block">Returns a <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link"><code>List</code></a> containing the individual <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a>s of this alignment.</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#getAllAccession(java.util.ArrayList)" class="member-name-link">getAllAccession(ArrayList&lt;BlastResult&gt;)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>
<div class="block">Return a list of all accessions contain in this list of results</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getAllBioConditionNames()" class="member-name-link">getAllBioConditionNames()</a> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Get all available <code>BioCondition</code> names by reading the text file
 <code>Experiment.INCLUDE_COND_PATH</code></div>
</dd>
<dt><a href="../bacnet/Database.html#getAllBioConditions()" class="member-name-link">getAllBioConditions()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getAllBioConditions()" class="member-name-link">getAllBioConditions()</a> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Return all BioCondition available</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getAllButtons()" class="member-name-link">getAllButtons()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#getAllButtons()" class="member-name-link">getAllButtons()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getAllButtons()" class="member-name-link">getAllButtons()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Codon.html#getAllCodons()" class="member-name-link">getAllCodons()</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>
<div class="block">Return a TreeSet containing all possible codons</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#getAllCompounds()" class="member-name-link">getAllCompounds()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#getAllCompounds()" class="member-name-link">getAllCompounds()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundSet.html#getAllCompounds()" class="member-name-link">getAllCompounds()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getAllElementNames()" class="member-name-link">getAllElementNames()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return all element names contained in all chromosomes</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getAllElements()" class="member-name-link">getAllElements()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getAllElements()" class="member-name-link">getAllElements()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return all elements contained in all chromosomes</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Frame.html#getAllFrames()" class="member-name-link">getAllFrames()</a> - Static method in enum org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></dt>
<dd>
<div class="block">Delegates to <a href="../org/biojava3/core/sequence/transcription/Frame.html#values()"><code>Frame.values()</code></a></div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#getAlllocusTag()" class="member-name-link">getAlllocusTag()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#getAlllocusTagFromGFF()" class="member-name-link">getAlllocusTagFromGFF()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>
<div class="block">They used old_locus_tag for the conservation analysis, so we need to create a HashMap to have the
 equivalent GeneId old_locus_tag ---> locus_tag in the GFF</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#getAlllocusTagINRA()" class="member-name-link">getAlllocusTagINRA()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html#getAllPeptides(java.lang.String,java.util.ArrayList,java.util.ArrayList,char,java.lang.String,int,boolean,int)" class="member-name-link">getAllPeptides(String, ArrayList&lt;Integer&gt;, ArrayList&lt;String&gt;, char, String, int, boolean, int)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Given a genome sequence, and a list of position of start codon, update a list of peptides
 accordingly<br>
 The new peptide, are the peptide from a start codon to the next R or stop codon</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#getAllPeptides(java.lang.String,java.util.ArrayList,java.util.ArrayList,char,java.lang.String,int,boolean,int)" class="member-name-link">getAllPeptides(String, ArrayList&lt;Integer&gt;, ArrayList&lt;String&gt;, char, String, int, boolean, int)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Given a genome sequence, and a list of position of start codon, update a list of peptides
 accordingly<br>
 The new peptide, are the peptide from a start codon to the next R or stop codon</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getAllPublishedBioConditions()" class="member-name-link">getAllPublishedBioConditions()</a> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Return all BioCondition available</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#getAlltechnology()" class="member-name-link">getAlltechnology()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Go through all SDRF files and read the "Array Design REF" entry, save in ListTechno.txt</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#getAlltechnology()" class="member-name-link">getAlltechnology()</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Go through all SDRF files and read the "Array Design REF" entry, save in ListTechno.txt</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getAllUnPublishedBioConditions()" class="member-name-link">getAllUnPublishedBioConditions()</a> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Return all BioCondition available</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html#getAmbiguity(org.biojava3.core.sequence.compound.NucleotideCompound...)" class="member-name-link">getAmbiguity(NucleotideCompound...)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractNucleotideCompoundSet</a></dt>
<dd>
<div class="block">Calculates the best symbol for a collection of compounds.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getAminoAcid()" class="member-name-link">getAminoAcid()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Codon.html#getAminoAcid(java.lang.String)" class="member-name-link">getAminoAcid(String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>
<div class="block">Translate a codon</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Codon.html#getAminoAcidCharge(java.lang.String)" class="member-name-link">getAminoAcidCharge(String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>
<div class="block">Using aminoacidCharge array, return the value of the charge for each amino acid aminoacidCharge =
 {{"R","1"},{"H","1"},{"K","1"},{"D","-1"},{"E","-1"}} else =0</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getAminoAcidCompounds()" class="member-name-link">getAminoAcidCompounds()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#getAminoAcidCompoundSet()" class="member-name-link">getAminoAcidCompoundSet()</a> - Static method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getANALYSIS_PATH()" class="member-name-link">getANALYSIS_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#getAnnotation()" class="member-name-link">getAnnotation()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getAnnotation()" class="member-name-link">getAnnotation()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getAnnotation()" class="member-name-link">getAnnotation()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getAnnotation()" class="member-name-link">getAnnotation()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getAnnotation()" class="member-name-link">getAnnotation()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getANNOTATIONDATA_PATH()" class="member-name-link">getANNOTATIONDATA_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#getAnnotationGenes(bacnet.datamodel.sequence.Genome,java.util.ArrayList)" class="member-name-link">getAnnotationGenes(Genome, ArrayList&lt;String&gt;)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Extract a String[][] with all gene annotation information</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#getAnnotationMatrix(bacnet.datamodel.sequence.Genome)" class="member-name-link">getAnnotationMatrix(Genome)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return a matrix containing different information on each element of the genome first chromosome</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#getAnnotationMatrix(bacnet.datamodel.sequence.Genome,java.util.ArrayList)" class="member-name-link">getAnnotationMatrix(Genome, ArrayList&lt;String&gt;)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return a matrix containing different information on each element of the genome first chromosome</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getAnnotations()" class="member-name-link">getAnnotations()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getAnnotationType()" class="member-name-link">getAnnotationType()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getAnnotElementHeight()" class="member-name-link">getAnnotElementHeight()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getAnnotNumber()" class="member-name-link">getAnnotNumber()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getAnnotSeparationHeight()" class="member-name-link">getAnnotSeparationHeight()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getAnnotSeparationNumber()" class="member-name-link">getAnnotSeparationNumber()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getAntiComparisons()" class="member-name-link">getAntiComparisons()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getAntiSDBindingFreeEnergy()" class="member-name-link">getAntiSDBindingFreeEnergy()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaSummaryView.html#getArray()" class="member-name-link">getArray()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaSummaryView.html" title="class in bacnet.sequenceTools">SrnaSummaryView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getArrayDataList()" class="member-name-link">getArrayDataList()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getArrayDataList()" class="member-name-link">getArrayDataList()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getArrayDesignRef(java.lang.String)" class="member-name-link">getArrayDesignRef(String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Read the SDRF table and search for the "Array Design REF" column</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getArrayDesignRef(java.lang.String)" class="member-name-link">getArrayDesignRef(String)</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Read the SDRF table and search for the "Array Design REF" column</div>
</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#getArrayExpressArrayLink(java.lang.String)" class="member-name-link">getArrayExpressArrayLink(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#getArrayExpressExpLink(java.lang.String)" class="member-name-link">getArrayExpressExpLink(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>
<div class="block">Get ArrayExpress database link to a specific project</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getArrayExpressId()" class="member-name-link">getArrayExpressId()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getArrayExpressTechnoId()" class="member-name-link">getArrayExpressTechnoId()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getArrayGeneToLocalization()" class="member-name-link">getArrayGeneToLocalization()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getArrayProteomeList()" class="member-name-link">getArrayProteomeList()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getArrayTranscriptomesList()" class="member-name-link">getArrayTranscriptomesList()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getAsList()" class="member-name-link">getAsList()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getAsList()" class="member-name-link">getAsList()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getAsList()" class="member-name-link">getAsList()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getAsList()" class="member-name-link">getAsList()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getAsList()" class="member-name-link">getAsList()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns this Sequence store as a List</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getAsList()" class="member-name-link">getAsList()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getAsList()" class="member-name-link">getAsList()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Delegates to <a href="../org/biojava3/core/sequence/template/SequenceMixin.html#toList(org.biojava3.core.sequence.template.Sequence)"><code>SequenceMixin.toList(org.biojava3.core.sequence.template.Sequence)</code></a></div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getAsList()" class="member-name-link">getAsList()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Sequence.html#getAsList()" class="member-name-link">getAsList()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Returns the Sequence as a List of compounds</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getAsList()" class="member-name-link">getAsList()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getAsRNAListEGDe()" class="member-name-link">getAsRNAListEGDe()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getAsRNAs()" class="member-name-link">getAsRNAs()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getAsRNAs()" class="member-name-link">getAsRNAs()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getAsRNAs()" class="member-name-link">getAsRNAs()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return all AsRNas in all chromosomes</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getAtisBHI(bacnet.datamodel.sequence.Gene)" class="member-name-link">getAtisBHI(Gene)</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Go through all MassSpec data and find if the lmio was detected</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getAtisBlood(bacnet.datamodel.sequence.Gene)" class="member-name-link">getAtisBlood(Gene)</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#getaTISCutoff()" class="member-name-link">getaTISCutoff()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getaTISMap()" class="member-name-link">getaTISMap()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#getAttribute(java.lang.String)" class="member-name-link">getAttribute(String)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Get value of specified attribute key.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureI.html#getAttribute(java.lang.String)" class="member-name-link">getAttribute(String)</a> - Method in interface org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a></dt>
<dd>
<div class="block">Get the attribute value for this key.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#getAttributeOld(java.lang.String)" class="member-name-link">getAttributeOld(String)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getAUNumber(java.lang.String)" class="member-name-link">getAUNumber(String)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Count the number of A and T nucleotides</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getAvailableGenome()" class="member-name-link">getAvailableGenome()</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getAvailableGenome(boolean)" class="member-name-link">getAvailableGenome(boolean)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getAvailableGenomes()" class="member-name-link">getAvailableGenomes()</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Read a text file containing the different genome ID available for Listeria</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#getAWTColor(org.eclipse.swt.graphics.Color)" class="member-name-link">getAWTColor(Color)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Transform SWT color to AWT color</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getBacillusGenome()" class="member-name-link">getBacillusGenome()</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Return a list of all NON Listeria genomes</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/views/WindowedSequence.html#getBackingSequence()" class="member-name-link">getBackingSequence()</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence</a></dt>
<dd>
<div class="block">Access the sequence which backs this window</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#getBase()" class="member-name-link">getBase()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#getBegin()" class="member-name-link">getBegin()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getBegin()" class="member-name-link">getBegin()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#getBegin()" class="member-name-link">getBegin()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#getBegin()" class="member-name-link">getBegin()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#getBeginColumn()" class="member-name-link">getBeginColumn()</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return the column index corresponding to Begin</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getBioBegin()" class="member-name-link">getBioBegin()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getBioCondition()" class="member-name-link">getBioCondition()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Return corresponding BioCondition</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getBioCondition(java.lang.String)" class="member-name-link">getBioCondition(String)</a> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">From a biological condition name load the corresponding BioCondition object</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getBioCondition(java.lang.String,boolean)" class="member-name-link">getBioCondition(String, boolean)</a> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">From a biological condition name, get from the ModelProvider the corresponding BioCondition
 object<br>
 If not available it will load them for the hard drive</div>
</dd>
<dt><a href="../bacnet/Database.html#getBIOCONDITION_PATH()" class="member-name-link">getBIOCONDITION_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getBioConditionHashMaps()" class="member-name-link">getBioConditionHashMaps()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DataValidation.html#getBioconditions()" class="member-name-link">getBioconditions()</a> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#getBioConditions()" class="member-name-link">getBioConditions()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#getBioConditions()" class="member-name-link">getBioConditions()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#getBioConditions()" class="member-name-link">getBioConditions()</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getBioConditionsArrayPath()" class="member-name-link">getBioConditionsArrayPath()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getBioCondMax(java.lang.String)" class="member-name-link">getBioCondMax(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getBioCondMin(java.lang.String)" class="member-name-link">getBioCondMin(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#getBioCondName()" class="member-name-link">getBioCondName()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#getBioConds()" class="member-name-link">getBioConds()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#getBioConds()" class="member-name-link">getBioConds()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#getBioConds()" class="member-name-link">getBioConds()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#getBioConds()" class="member-name-link">getBioConds()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#getBioConds()" class="member-name-link">getBioConds()</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#getBioConds()" class="member-name-link">getBioConds()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getBioConds()" class="member-name-link">getBioConds()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#getBioCondsArray()" class="member-name-link">getBioCondsArray()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#getBioCondsArray()" class="member-name-link">getBioCondsArray()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#getBioCondsArray()" class="member-name-link">getBioCondsArray()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#getBioCondsArray()" class="member-name-link">getBioCondsArray()</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#getBioCondsArray()" class="member-name-link">getBioCondsArray()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getBioCondsArray()" class="member-name-link">getBioCondsArray()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#getBioCondsArray()" class="member-name-link">getBioCondsArray()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getBioCondSize(java.lang.String)" class="member-name-link">getBioCondSize(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the size ratio that have a given BioCondition</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#getBioCondsToDisplay()" class="member-name-link">getBioCondsToDisplay()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#getBioCondsToDisplay()" class="member-name-link">getBioCondsToDisplay()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#getBioCondsToDisplay()" class="member-name-link">getBioCondsToDisplay()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#getBioCondsToDisplay()" class="member-name-link">getBioCondsToDisplay()</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#getBioCondsToDisplay()" class="member-name-link">getBioCondsToDisplay()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getBioCondsToDisplay()" class="member-name-link">getBioCondsToDisplay()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#getBioCondsToDisplay()" class="member-name-link">getBioCondsToDisplay()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getBioEnd()" class="member-name-link">getBioEnd()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getBioEnd()" class="member-name-link">getBioEnd()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceView.html#getBioEnd()" class="member-name-link">getBioEnd()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a></dt>
<dd>
<div class="block">1-indexed, inclusive.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getBioStart()" class="member-name-link">getBioStart()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceView.html#getBioStart()" class="member-name-link">getBioStart()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a></dt>
<dd>
<div class="block">1-indexed, inclusive.</div>
</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#getBLAST_RESULT_PATH()" class="member-name-link">getBLAST_RESULT_PATH()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#getBlastDBInfo(java.lang.String)" class="member-name-link">getBlastDBInfo(String)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#getBlastExtension()" class="member-name-link">getBlastExtension()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#getBlastFolder()" class="member-name-link">getBlastFolder()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#getBlastFolder()" class="member-name-link">getBlastFolder()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorSWT.html#getBlue()" class="member-name-link">getBlue()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#getBodyFont(int)" class="member-name-link">getBodyFont(int)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Get the DEFAULT Font with appropriate height and style</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#getBodyFont(int,int)" class="member-name-link">getBodyFont(int, int)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Get the DEFAULT Font with appropriate height and style</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#getBoldFont(org.eclipse.swt.graphics.Font)" class="member-name-link">getBoldFont(Font)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns a bold version of the given <code>Font</code>.</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getBpSizeH()" class="member-name-link">getBpSizeH()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getBpSizeH()" class="member-name-link">getBpSizeH()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#getBtnAllMutant()" class="member-name-link">getBtnAllMutant()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnAllMutant()" class="member-name-link">getBtnAllMutant()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#getBtnChooseOneMutant()" class="member-name-link">getBtnChooseOneMutant()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnChooseOneMutant()" class="member-name-link">getBtnChooseOneMutant()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnDeath()" class="member-name-link">getBtnDeath()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnDnDNA()" class="member-name-link">getBtnDnDNA()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnDnProt()" class="member-name-link">getBtnDnProt()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnDnRNA()" class="member-name-link">getBtnDnRNA()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnExponential()" class="member-name-link">getBtnExponential()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnGeneExpression()" class="member-name-link">getBtnGeneExpression()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnLagPhase()" class="member-name-link">getBtnLagPhase()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnNoDNA()" class="member-name-link">getBtnNoDNA()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#getBtnNoneMutant()" class="member-name-link">getBtnNoneMutant()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnNoneMutant()" class="member-name-link">getBtnNoneMutant()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnNoProt()" class="member-name-link">getBtnNoProt()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnNoRNA()" class="member-name-link">getBtnNoRNA()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getBtnOpenStructure()" class="member-name-link">getBtnOpenStructure()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnPosDNA()" class="member-name-link">getBtnPosDNA()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnPrint()" class="member-name-link">getBtnPrint()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnRegrowth()" class="member-name-link">getBtnRegrowth()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnRiboSeq()" class="member-name-link">getBtnRiboSeq()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnRnaseq()" class="member-name-link">getBtnRnaseq()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnStationnary()" class="member-name-link">getBtnStationnary()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnSurvival()" class="member-name-link">getBtnSurvival()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnTermSeq()" class="member-name-link">getBtnTermSeq()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnTiling()" class="member-name-link">getBtnTiling()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnTss()" class="member-name-link">getBtnTss()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnUnpublished()" class="member-name-link">getBtnUnpublished()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnUpdateCutoff()" class="member-name-link">getBtnUpdateCutoff()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnUpProt()" class="member-name-link">getBtnUpProt()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnUpRNA()" class="member-name-link">getBtnUpRNA()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getBtnZoomIn()" class="member-name-link">getBtnZoomIn()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getBtnZoomInVertical()" class="member-name-link">getBtnZoomInVertical()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getBtnZoomOut()" class="member-name-link">getBtnZoomOut()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getBtnZoomOutVertical()" class="member-name-link">getBtnZoomOutVertical()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html#getBufferedReader()" class="member-name-link">getBufferedReader()</a> - Method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a></dt>
<dd>
<div class="block">Returns the reader representation of this classpath resource</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#getBytesRead()" class="member-name-link">getBytesRead()</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>
<div class="block">Keep track of bytesread via ReadLine to account for CR-LF in the stream.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getCAI(bacnet.datamodel.sequence.Gene)" class="member-name-link">getCAI(Gene)</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">After generating TIRSequence.fasta in setTIRSequence() method we submit to
 http://genomes.urv.cat/CAIcal/ to calculate CAI value and expected CAI value The table with CAI
 values is saved from the website in: CAIResults.excel</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getCAIGene(bacnet.datamodel.sequence.Gene)" class="member-name-link">getCAIGene(Gene)</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">After generating GeneSequence.fasta in setGeneSequence() method we submit to
 http://genomes.urv.cat/CAIcal/ to calculate CAI value and expected CAI value The table with CAI
 values is saved from the website in: StatGenes/CAIGeneResults.excel</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getCanvasData()" class="member-name-link">getCanvasData()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getCanvasGenome()" class="member-name-link">getCanvasGenome()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getCanvasName()" class="member-name-link">getCanvasName()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getCDSSequences()" class="member-name-link">getCDSSequences()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>
<div class="block">Get the CDS sequences that have been added to the TranscriptSequences</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getCenterColor()" class="member-name-link">getCenterColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getCenterColor()" class="member-name-link">getCenterColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getCenterColorSWT()" class="member-name-link">getCenterColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getCenterColorSWT()" class="member-name-link">getCenterColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getCenterPos()" class="member-name-link">getCenterPos()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getCenterPos()" class="member-name-link">getCenterPos()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#getCgview()" class="member-name-link">getCgview()</a> - Method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getChargedResidues(bacnet.datamodel.sequence.Gene,int)" class="member-name-link">getChargedResidues(Gene, int)</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Sum up the different charge of the amino acids of the TIR</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getChargedResiduesGene(bacnet.datamodel.sequence.Gene)" class="member-name-link">getChargedResiduesGene(Gene)</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Sum up the different charge of the amino acids of the gene sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getChildrenFeatures()" class="member-name-link">getChildrenFeatures()</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">Get the children features</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getChildrenFeatures()" class="member-name-link">getChildrenFeatures()</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Get the features contained by this feature</div>
</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#getChromoId()" class="member-name-link">getChromoId()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getChromoID()" class="member-name-link">getChromoID()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getChromosome()" class="member-name-link">getChromosome()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getChromosome()" class="member-name-link">getChromosome()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Return corresponding Chromosome given by this.getChromosomeNb<br>
 Load it if necessary</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#getChromosome()" class="member-name-link">getChromosome()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getChromosomeFromDNASequence(java.util.LinkedHashMap,bacnet.datamodel.sequence.Genome,int,java.lang.String)" class="member-name-link">getChromosomeFromDNASequence(LinkedHashMap&lt;String, DNASequence&gt;, Genome, int, String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#getChromosomeID()" class="member-name-link">getChromosomeID()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getChromosomeID()" class="member-name-link">getChromosomeID()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getChromosomeID()" class="member-name-link">getChromosomeID()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getChromosomeID()" class="member-name-link">getChromosomeID()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getChromosomeID()" class="member-name-link">getChromosomeID()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#getChromosomeID()" class="member-name-link">getChromosomeID()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getChromosomeNames()" class="member-name-link">getChromosomeNames()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Get an ordered list of all chromosomes The order is fixed by the order of the chromosomes in the
 FNA (fasta) file used for reading the genome</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#getChromosomeNumber()" class="member-name-link">getChromosomeNumber()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getChromosomes()" class="member-name-link">getChromosomes()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getChromosomes()" class="member-name-link">getChromosomes()</a> - Method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html#getChromosomeSequenceFromDNASequence(java.util.LinkedHashMap)" class="member-name-link">getChromosomeSequenceFromDNASequence(LinkedHashMap&lt;String, DNASequence&gt;)</a> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getCisRegRNAListEGDe()" class="member-name-link">getCisRegRNAListEGDe()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getCisRegs()" class="member-name-link">getCisRegs()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getCisRegs()" class="member-name-link">getCisRegs()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getCisRegs()" class="member-name-link">getCisRegs()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return all CisRegs in all chromosomes</div>
</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#getCmbChromo()" class="member-name-link">getCmbChromo()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#getCmbGenome()" class="member-name-link">getCmbGenome()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getCodingRegion(java.lang.String)" class="member-name-link">getCodingRegion(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>
<div class="block">Get the gene based on accession.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/CDSSequence.html#getCodingSequence()" class="member-name-link">getCodingSequence()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></dt>
<dd>
<div class="block">A CDS sequence if negative stranded needs to be reverse complement to represent the actual coding
 sequence.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getCodingSequenceHashMap()" class="member-name-link">getCodingSequenceHashMap()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getCodingSequencesList(boolean)" class="member-name-link">getCodingSequencesList(boolean)</a> - Method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Put in a list all the CodingSequence found</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/CodonUsage.html#getCodonColor(double)" class="member-name-link">getCodonColor(double)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/CodonUsage.html" title="class in bacnet.datamodel.annotation">CodonUsage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">getCodonCompoundSet(CompoundSet&lt;NucleotideCompound&gt;, CompoundSet&lt;AminoAcidCompound&gt;)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a></dt>
<dd>
<div class="block">Returns the compound set of codons</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">getCodonCompoundSet(CompoundSet&lt;NucleotideCompound&gt;, CompoundSet&lt;AminoAcidCompound&gt;)</a> - Method in interface org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Codon.html#getCodonProportion(bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">getCodonProportion(Genome, String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>
<div class="block">Go through genome on both strand and calculate the proportion of each codon</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getCodons(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">getCodons(CompoundSet&lt;NucleotideCompound&gt;, CompoundSet&lt;AminoAcidCompound&gt;)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a></dt>
<dd>
<div class="block">Returns a list of codons where the source and target compounds are the same as those given by the
 parameters.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.html#getCodons(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">getCodons(CompoundSet&lt;NucleotideCompound&gt;, CompoundSet&lt;AminoAcidCompound&gt;)</a> - Method in interface org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getCodonUsage()" class="member-name-link">getCodonUsage()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getCodonUsage(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getCodonUsage(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getCOEXPR_NETWORK_TRANSCRIPTOMES_PATH()" class="member-name-link">getCOEXPR_NETWORK_TRANSCRIPTOMES_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Transcriptomes matrix data showing Log(Fold Change) values</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Network.html#getCoExpressionGlobalMatrix(bacnet.datamodel.sequence.Genome)" class="member-name-link">getCoExpressionGlobalMatrix(Genome)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>
<div class="block">Calculate all pearson correlation and put it in a list of network interactions<br>
 Remove correlation CORR_CUTOFF<br>
 Nedd to load: Database.COEXPR_NETWORK_TRANSCRIPTOMES_PATH+"_Temp_"+genome.getSpecies()</div>
</dd>
<dt><a href="../bacnet/Database.html#getCoExprNetworkArrayPath()" class="member-name-link">getCoExprNetworkArrayPath()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DataValidation.html#getCoExprNetworks()" class="member-name-link">getCoExprNetworks()</a> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#getCog()" class="member-name-link">getCog()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getCOG(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getCOG(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/COGannotation.html#getCogClassification(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequenceNCBI.GenomeNCBI)" class="member-name-link">getCogClassification(ExpressionMatrix, GenomeNCBI)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/COGannotation.html#getCOGDescription(java.lang.String)" class="member-name-link">getCOGDescription(String)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>
<div class="block">Return the COG description corresponding to a specific COGid</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getCOGExpression(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">getCOGExpression(ExpressionMatrix)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Summarize an ExpressionMatrix by regrouping the different genes (rows) by their COG functional
 categories <br>
 Averages on all the genes for each category are calculated.</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/COGannotation.html#getCOGMap()" class="member-name-link">getCOGMap()</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>
<div class="block">Return a map from COG ID to COG description</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/COGannotation.html#getCOGNumberEGDe()" class="member-name-link">getCOGNumberEGDe()</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>
<div class="block">Return a map between cogID and the number of genes associated to this cog in EGD-e</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#getColor(int)" class="member-name-link">getColor(int)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns the system <code>Color</code> matching the specific ID.</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#getColor(int,int,int)" class="member-name-link">getColor(int, int, int)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns a <code>Color</code> given its red, green and blue component values.</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#getColor(org.eclipse.swt.graphics.RGB)" class="member-name-link">getColor(RGB)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns a <code>Color</code> given its RGB value.</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getColorBetween(double,double,double,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color)" class="member-name-link">getColorBetween(double, double, double, Color, Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getColorBetween(double,double,double,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color)" class="member-name-link">getColorBetween(double, double, double, Color, Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizard.html#getColorMapper()" class="member-name-link">getColorMapper()</a> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizard.html" title="class in bacnet.table.gui">ColorMapperWizard</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizardPage.html#getColorMapper()" class="member-name-link">getColorMapper()</a> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#getColorMapperList()" class="member-name-link">getColorMapperList()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperList.html#getColorMappers()" class="member-name-link">getColorMappers()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCPList.html#getColorMappers()" class="member-name-link">getColorMappers()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#getColors(int)" class="member-name-link">getColors(int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#getColumn(double%5B%5D%5B%5D,int)" class="member-name-link">getColumn(double[][], int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Extract a Column from a Matrix</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getColumn(int)" class="member-name-link">getColumn(int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getColumn(java.lang.String)" class="member-name-link">getColumn(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Extract the column at this position</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#getColumn(java.lang.String%5B%5D%5B%5D,int)" class="member-name-link">getColumn(String[][], int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getColumn(java.lang.String%5B%5D%5B%5D,java.lang.String)" class="member-name-link">getColumn(String[][], String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Read the SDRF table and search for the corresponding column</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getColumn(java.lang.String%5B%5D%5B%5D,java.lang.String)" class="member-name-link">getColumn(String[][], String)</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Read the SDRF table and search for the corresponding column</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getColumnIndex(java.lang.String%5B%5D%5B%5D,java.lang.String)" class="member-name-link">getColumnIndex(String[][], String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Return column Index of the column named columnHeader</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getColumnIndex(java.lang.String%5B%5D%5B%5D,java.lang.String)" class="member-name-link">getColumnIndex(String[][], String)</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Return column Index of the column named columnHeader</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#getColumnNames()" class="member-name-link">getColumnNames()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#getColumnNames()" class="member-name-link">getColumnNames()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#getColumnNames()" class="member-name-link">getColumnNames()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#getColumnNames()" class="member-name-link">getColumnNames()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#getColumnNames()" class="member-name-link">getColumnNames()</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#getColumnNames()" class="member-name-link">getColumnNames()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#getColumnNames()" class="member-name-link">getColumnNames()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getColumnNames()" class="member-name-link">getColumnNames()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getColumnNames()" class="member-name-link">getColumnNames()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getComboAbsoluteRelative()" class="member-name-link">getComboAbsoluteRelative()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#getComboGenome()" class="member-name-link">getComboGenome()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getComboGenome()" class="member-name-link">getComboGenome()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getComboGenome()" class="member-name-link">getComboGenome()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#getComboMutant()" class="member-name-link">getComboMutant()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getComboMutant()" class="member-name-link">getComboMutant()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getComment()" class="member-name-link">getComment()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getComment()" class="member-name-link">getComment()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getComment(bacnet.datamodel.sequence.Gene)" class="member-name-link">getComment(Gene)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Return information about gene or iTIS</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getComparisonDataNames()" class="member-name-link">getComparisonDataNames()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Get every names of the Comparison which can be found:<br>
 Tiling arrays
 Gene Expression arrays
 ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getComparisonNames()" class="member-name-link">getComparisonNames()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Return a list of all Comparisons: bioCondName vs comparisons.get(i)</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getComparisons()" class="member-name-link">getComparisons()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DataValidation.html#getComparisons()" class="member-name-link">getComparisons()</a> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#getCompExpression()" class="member-name-link">getCompExpression()</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getCompFilter()" class="member-name-link">getCompFilter()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html#getComplement()" class="member-name-link">getComplement()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#getComplement()" class="member-name-link">getComplement()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Returns a Sequence which will complement every base</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/RNASequence.html#getComplement()" class="member-name-link">getComplement()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>
<div class="block">Get the complement view of the RNA sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/ComplementCompound.html#getComplement()" class="member-name-link">getComplement()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/ComplementCompound.html" title="interface in org.biojava3.core.sequence.template">ComplementCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getCompleteGenome(java.util.ArrayList)" class="member-name-link">getCompleteGenome(ArrayList&lt;String&gt;)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Extract all complete genomes from genomesList</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getComposite()" class="member-name-link">getComposite()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getComposite_2()" class="member-name-link">getComposite_2()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getComposite_6()" class="member-name-link">getComposite_6()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getComposite_7()" class="member-name-link">getComposite_7()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getCompositeAddData()" class="member-name-link">getCompositeAddData()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#getCompositeDataFilter()" class="member-name-link">getCompositeDataFilter()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#getCompositeDataFilter()" class="member-name-link">getCompositeDataFilter()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#getCompositeDataFilter()" class="member-name-link">getCompositeDataFilter()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getCompositeNterm()" class="member-name-link">getCompositeNterm()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getCompositeSummary()" class="member-name-link">getCompositeSummary()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getCompositeTypeDisplay()" class="member-name-link">getCompositeTypeDisplay()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#getComposition(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">getComposition(Sequence&lt;C&gt;)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Does a linear scan over the given Sequence and records the number of times each base appears.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getCompoundAt(int)" class="member-name-link">getCompoundAt(int)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getCompoundAt(int)" class="member-name-link">getCompoundAt(int)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getCompoundAt(int)" class="member-name-link">getCompoundAt(int)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getCompoundAt(int)" class="member-name-link">getCompoundAt(int)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#getCompoundAt(int)" class="member-name-link">getCompoundAt(int)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>
<div class="block">Returns the compound at the specified biological index</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getCompoundAt(int)" class="member-name-link">getCompoundAt(int)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns the compound at the specified biological index</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getCompoundAt(int)" class="member-name-link">getCompoundAt(int)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getCompoundAt(int)" class="member-name-link">getCompoundAt(int)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Always returns the compound given at construction</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getCompoundAt(int)" class="member-name-link">getCompoundAt(int)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Sequence.html#getCompoundAt(int)" class="member-name-link">getCompoundAt(int)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Returns the Compound at the given biological index</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getCompoundAt(int)" class="member-name-link">getCompoundAt(int)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html#getCompoundAt(int)" class="member-name-link">getCompoundAt(int)</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava3.core.sequence.views">ComplementSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/ReversedSequenceView.html#getCompoundAt(int)" class="member-name-link">getCompoundAt(int)</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views">ReversedSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#getCompoundAt(int)" class="member-name-link">getCompoundAt(int)</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#getCompoundForString(java.lang.String)" class="member-name-link">getCompoundForString(String)</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#getCompoundForString(java.lang.String)" class="member-name-link">getCompoundForString(String)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundSet.html#getCompoundForString(java.lang.String)" class="member-name-link">getCompoundForString(String)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></dt>
<dd>
<div class="block">Return null if not recognised.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/views/WindowedSequence.html#getCompounds(int)" class="member-name-link">getCompounds(int)</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence</a></dt>
<dd>
<div class="block">For a given position into the windowed view this will return those compounds we can see in the
 window. i.e. in the sequence AGGCCT requesting index 1 returns AGG and requesting index 2 return
 CCT.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#getCompoundsAt(int)" class="member-name-link">getCompoundsAt(int)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">Get a list of compounds at a sequence position</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.html#getCompoundsAt(int)" class="member-name-link">getCompoundsAt(int)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a></dt>
<dd>
<div class="block">Returns the <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template"><code>Compound</code></a> elements of the original <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a>s at the given column.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompound.html#getCompoundSet()" class="member-name-link">getCompoundSet()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getCompoundSet()" class="member-name-link">getCompoundSet()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getCompoundSet()" class="member-name-link">getCompoundSet()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getCompoundSet()" class="member-name-link">getCompoundSet()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#getCompoundSet()" class="member-name-link">getCompoundSet()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">Get the Compounds defined in the first sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getCompoundSet()" class="member-name-link">getCompoundSet()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#getCompoundSet()" class="member-name-link">getCompoundSet()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>
<div class="block">Returns the compound set backing this store</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getCompoundSet()" class="member-name-link">getCompoundSet()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns the compound set backing this store</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getCompoundSet()" class="member-name-link">getCompoundSet()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getCompoundSet()" class="member-name-link">getCompoundSet()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Returns the compound set given at construction</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getCompoundSet()" class="member-name-link">getCompoundSet()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.html#getCompoundSet()" class="member-name-link">getCompoundSet()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a></dt>
<dd>
<div class="block">Returns <a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template"><code>CompoundSet</code></a> of all <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a>s</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/Sequence.html#getCompoundSet()" class="member-name-link">getCompoundSet()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Gets the compound set used to back this Sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getCompoundSet()" class="member-name-link">getCompoundSet()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getConservation()" class="member-name-link">getConservation()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getConservationHashMap()" class="member-name-link">getConservationHashMap()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Get HashMap linking genomeName to conserved gene<br>
 Give information about the genome in which this gene is conserved</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html#getConsituents()" class="member-name-link">getConsituents()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#getContainer()" class="member-name-link">getContainer()</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ImageMagick.html#getConvertPATH()" class="member-name-link">getConvertPATH()</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ImageMagick.html" title="class in bacnet.utils">ImageMagick</a></dt>
<dd>
<div class="block">Need to convert path from local use or server utilization</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapperList.html#getCorrespondingMapper(java.lang.String)" class="member-name-link">getCorrespondingMapper(String)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCPList.html#getCorrespondingMapper(java.lang.String)" class="member-name-link">getCorrespondingMapper(String)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#getCreator()" class="member-name-link">getCreator()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getCurrentState()" class="member-name-link">getCurrentState()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#getCursor(int)" class="member-name-link">getCursor(int)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns the system cursor matching the specific ID.</div>
</dd>
<dt><a href="../bacnet/table/core/Filter.html#getCutOff1()" class="member-name-link">getCutOff1()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#getCutOff1()" class="member-name-link">getCutOff1()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#getCutOff2()" class="member-name-link">getCutOff2()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#getCutOff2()" class="member-name-link">getCutOff2()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#getCutoffHashMap()" class="member-name-link">getCutoffHashMap()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getDATA_PATH()" class="member-name-link">getDATA_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">GETTER AND SETTERS</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#getDatabase()" class="member-name-link">getDatabase()</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getDatabaseFeatures()" class="member-name-link">getDatabaseFeatures()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/DatabaseReferenceInterface.html#getDatabaseReferences()" class="member-name-link">getDatabaseReferences()</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getDatabaseReferences()" class="member-name-link">getDatabaseReferences()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">The Uniprot mappings to other database identifiers for this sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getDatabaseReferences()" class="member-name-link">getDatabaseReferences()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataColors()" class="member-name-link">getDataColors()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataCount()" class="member-name-link">getDataCount()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Count the number of data available, without duplicate and filtered</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataDisplayed()" class="member-name-link">getDataDisplayed()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Get the list of data to display</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataDisplayedNames()" class="member-name-link">getDataDisplayedNames()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the names of the data which will be displayd</div>
</dd>
<dt><a href="../bacnet/scripts/database/TranscriptomesCreation.html#getDataFromLab()" class="member-name-link">getDataFromLab()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/TranscriptomesCreation.html" title="class in bacnet.scripts.database">TranscriptomesCreation</a></dt>
<dd>
<div class="block">Get all GeneExpression from the lab</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataNames()" class="member-name-link">getDataNames()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of all the data available</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataNOTDisplayed()" class="member-name-link">getDataNOTDisplayed()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#getDataRegion()" class="member-name-link">getDataRegion()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#getDatas()" class="member-name-link">getDatas()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#getDatasets()" class="member-name-link">getDatasets()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataSize()" class="member-name-link">getDataSize()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataSizes()" class="member-name-link">getDataSizes()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/AccessionID.html#getDataSource()" class="member-name-link">getDataSource()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcParser.html#getDataSource()" class="member-name-link">getDataSource()</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/TaxonomyID.html#getDataSource()" class="member-name-link">getDataSource()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataTotalSize(bacnet.genomeBrowser.core.Track.DisplayType)" class="member-name-link">getDataTotalSize(Track.DisplayType)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Add all the size of the Tracks to know the total size of the display window</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getDataWindowSize()" class="member-name-link">getDataWindowSize()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#getDate()" class="member-name-link">getDate()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getDate()" class="member-name-link">getDate()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getDbxref(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getDbxref(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getDecaySliderVBar()" class="member-name-link">getDecaySliderVBar()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getDefault()" class="member-name-link">getDefault()</a> - Static method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>
<div class="block">Default instance to use when Transcribing from DNA -&gt; RNA -&gt; Protein.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Frame.html#getDefaultFrame()" class="member-name-link">getDefaultFrame()</a> - Static method in enum org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getDefautGenome()" class="member-name-link">getDefautGenome()</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return defaut genome for a databse <br> For the moment -> first element of getAvailableGenomes()</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#getDescription()" class="member-name-link">getDescription()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#getDescription()" class="member-name-link">getDescription()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getDescription()" class="member-name-link">getDescription()</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getDescription()" class="member-name-link">getDescription()</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Get the description that can be used to describe the feature</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#getDescription()" class="member-name-link">getDescription()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getDescription()" class="member-name-link">getDescription()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Compound.html#getDescription()" class="member-name-link">getDescription()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#getDescription()" class="member-name-link">getDescription()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getDescription()" class="member-name-link">getDescription()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getDetectable(bacnet.datamodel.sequence.Gene)" class="member-name-link">getDetectable(Gene)</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">By reading the List in: /StatGenes/EGDeDetectable.txt Say if a gene is detectable by the
 spectometer or not</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getDeviation()" class="member-name-link">getDeviation()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getDeviation()" class="member-name-link">getDeviation()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#getDeviation(java.util.ArrayList,java.lang.String)" class="member-name-link">getDeviation(ArrayList&lt;Tiling&gt;, String)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Calculate the stat deviation on each probe of a list of Tiling</div>
</dd>
<dt><a href="../bacnet/table/core/BioConditionComparator.html#getDirection()" class="member-name-link">getDirection()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/TableViewerComparator.html#getDirection()" class="member-name-link">getDirection()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/TableViewerComparator.html" title="class in bacnet.table.core">TableViewerComparator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDisplay()" class="member-name-link">getDisplay()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getDisplay()" class="member-name-link">getDisplay()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#getDisplayRegion()" class="member-name-link">getDisplayRegion()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#getDisplayType()" class="member-name-link">getDisplayType()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#getDistribution(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">getDistribution(Sequence&lt;C&gt;)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Analogous to <a href="../org/biojava3/core/sequence/template/SequenceMixin.html#getComposition(org.biojava3.core.sequence.template.Sequence)"><code>SequenceMixin.getComposition(Sequence)</code></a> but returns the distribution of that
 <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template"><code>Compound</code></a> over the given sequence.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getDNACodingSequence()" class="member-name-link">getDNACodingSequence()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>
<div class="block">Get the stitched together CDS sequences then maps to the cDNA</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html#getDNACodingSequences()" class="member-name-link">getDNACodingSequences()</a> - Method in class org.biojava3.genome.parsers.geneid.<a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html" title="class in org.biojava3.genome.parsers.geneid">GeneIDXMLReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getDnaCompounds()" class="member-name-link">getDnaCompounds()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AmbiguityDNACompoundSet.html#getDNACompoundSet()" class="member-name-link">getDNACompoundSet()</a> - Static method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AmbiguityDNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">AmbiguityDNACompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AmbiguityRNACompoundSet.html#getDNACompoundSet()" class="member-name-link">getDNACompoundSet()</a> - Static method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AmbiguityRNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">AmbiguityRNACompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/DNACompoundSet.html#getDNACompoundSet()" class="member-name-link">getDNACompoundSet()</a> - Static method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/DNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">DNACompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getDnaRnaTranslator()" class="member-name-link">getDnaRnaTranslator()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDNASeqDatas()" class="member-name-link">getDNASeqDatas()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of SeqDatas</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDNASeqDatas(java.lang.String)" class="member-name-link">getDNASeqDatas(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of RiboSeqData given a bioCondName</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#getDnaToRna()" class="member-name-link">getDnaToRna()</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#getDNAType()" class="member-name-link">getDNAType()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getDownDNAList()" class="member-name-link">getDownDNAList()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/InternalBrowser.html#getDownloadLink()" class="member-name-link">getDownloadLink()</a> - Method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/DownloadServiceHandler.html#getDownloadUrl(java.lang.String,java.io.File,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">getDownloadUrl(String, File, EPartService)</a> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/DownloadServiceHandler.html#getDownloadUrl(java.lang.String,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">getDownloadUrl(String, String, EPartService)</a> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getDownProtList()" class="member-name-link">getDownProtList()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getDownRNAList()" class="member-name-link">getDownRNAList()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getDuplicates()" class="member-name-link">getDuplicates()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#getDuplicateSequenceToRemove()" class="member-name-link">getDuplicateSequenceToRemove()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Network.html#getEdges()" class="member-name-link">getEdges()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>
<div class="block">An hashMap which map each vertex (=genome element) to the correlated vertex.<br>
 Vertex -> HashMap&lt;Vetrex,PearsonCorrelation></div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getEGDcOverlap()" class="member-name-link">getEGDcOverlap()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#getEGDcSrnas()" class="member-name-link">getEGDcSrnas()</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Return a list of all Srna in EGD-c</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getEGDe()" class="member-name-link">getEGDe()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Read RAST automoticaly annotated EGD-e genome</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#getEGDeALLSrnas()" class="member-name-link">getEGDeALLSrnas()</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Return a list of all Srna, CisReg and ASrna in EGD-e</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getEGDeAutoOverlap()" class="member-name-link">getEGDeAutoOverlap()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Go through EGDe overlap and find when two genes overlap</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#getEgdeOperonList()" class="member-name-link">getEgdeOperonList()</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#getEgdeOperonMap()" class="member-name-link">getEgdeOperonMap()</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getEGDeOverlap()" class="member-name-link">getEGDeOverlap()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getEGDeRASTAutoOverlap()" class="member-name-link">getEGDeRASTAutoOverlap()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Find RASt element which overlap another element in the genome</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#getEGDeSrnas()" class="member-name-link">getEGDeSrnas()</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Return an ordered list of all Srna in EGD-e</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getEGDeUnknownRAST()" class="member-name-link">getEGDeUnknownRAST()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getElement(java.lang.String)" class="member-name-link">getElement(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Search an element by its name in the genome</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#getElementATbp(bacnet.datamodel.sequence.Chromosome,int)" class="member-name-link">getElementATbp(Chromosome, int)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return the first element for which<br>
 begin &lt; bpPosition &lt; end</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#getElementInfoATbp(bacnet.datamodel.sequence.Chromosome,int)" class="member-name-link">getElementInfoATbp(Chromosome, int)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return the first element for which<br>
 begin &lt; bpPosition &lt; end</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#getElements()" class="member-name-link">getElements()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getElements()" class="member-name-link">getElements()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Link NTerm name to <code>NTerm</code><br>
 This correspond to the list of <code>NTerm</code></div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getElements()" class="member-name-link">getElements()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getElements()" class="member-name-link">getElements()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#getElements(bacnet.datamodel.sequence.Chromosome,int,int)" class="member-name-link">getElements(Chromosome, int, int)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Get the lists of elements between beginDraw and endDraw</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getElementsToDisplay(int,int,java.lang.String)" class="member-name-link">getElementsToDisplay(int, int, String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Get the lists of NTerm between beginDraw and endDraw</div>
</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#getENALink(java.lang.String)" class="member-name-link">getENALink(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#getEnd()" class="member-name-link">getEnd()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getEnd()" class="member-name-link">getEnd()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#getEnd()" class="member-name-link">getEnd()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html#getEnd()" class="member-name-link">getEnd()</a> - Method in class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="class in org.biojava3.core.sequence.edits">Edit.AbstractEdit</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getEnd()" class="member-name-link">getEnd()</a> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.html#getEnd()" class="member-name-link">getEnd()</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">End of the location</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#getEnd()" class="member-name-link">getEnd()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.Substitute.html#getEnd(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">getEnd(Sequence&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.Substitute.html" title="class in org.biojava3.core.sequence.edits">Edit.Substitute</a></dt>
<dd>
<div class="block">Must use this rather than the no-args getEnd as this can return -1 and the length of a sub is
 dependent on the length of the Sequence; we cannot assume 1:1 mapping between characters in a
 String and the number of compounds we will have to insert.</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#getEndColumn()" class="member-name-link">getEndColumn()</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return the column index corresponding to End</div>
</dd>
<dt><a href="../bacnet/swt/ResourceManager.PluginResourceProvider.html#getEntry(java.lang.String,java.lang.String)" class="member-name-link">getEntry(String, String)</a> - Method in interface bacnet.swt.<a href="../bacnet/swt/ResourceManager.PluginResourceProvider.html" title="interface in bacnet.swt">ResourceManager.PluginResourceProvider</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#getEquivalentCompounds(C)" class="member-name-link">getEquivalentCompounds(C)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundSet.html#getEquivalentCompounds(C)" class="member-name-link">getEquivalentCompounds(C)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#getEquivalentCompounds(org.biojava3.core.sequence.compound.AminoAcidCompound)" class="member-name-link">getEquivalentCompounds(AminoAcidCompound)</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#getExcludeColumn()" class="member-name-link">getExcludeColumn()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#getExcludeColumn()" class="member-name-link">getExcludeColumn()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/FilterList.html#getExcludeColumn()" class="member-name-link">getExcludeColumn()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#getExcludeColumn()" class="member-name-link">getExcludeColumn()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#getExcludeRow()" class="member-name-link">getExcludeRow()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#getExcludeRow()" class="member-name-link">getExcludeRow()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/FilterList.html#getExcludeRow()" class="member-name-link">getExcludeRow()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#getExcludeRow()" class="member-name-link">getExcludeRow()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getExcludeRows()" class="member-name-link">getExcludeRows()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#getExonSequences()" class="member-name-link">getExonSequences()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Get the exons as an ArrayList</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getExperienceNb()" class="member-name-link">getExperienceNb()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getExperiment()" class="member-name-link">getExperiment()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>
<div class="block">Return if the NTerm was detected in the first experiment, or the second, or both</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#getExperiment()" class="member-name-link">getExperiment()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>
<div class="block">Return if the NTerms of this TIS were detected in the first experiment, or the second, or both</div>
</dd>
<dt><a href="../bacnet/Database.html#getExperimentComparisonTablePath()" class="member-name-link">getExperimentComparisonTablePath()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#getExpression()" class="member-name-link">getExpression()</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#getExpression()" class="member-name-link">getExpression()</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getEXPRESSION_MATRIX_PROTEOMES_PATH()" class="member-name-link">getEXPRESSION_MATRIX_PROTEOMES_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Proteomes matrix data showing absolute expression values</div>
</dd>
<dt><a href="../bacnet/Database.html#getEXPRESSION_MATRIX_TRANSCRIPTOMES_PATH()" class="member-name-link">getEXPRESSION_MATRIX_TRANSCRIPTOMES_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Transcriptomes matrix data showing expression values</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getExpressionData()" class="member-name-link">getExpressionData()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Put all ExpressionData in a list</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/AMtools.html#getExpressionMatrix(double%5B%5D%5B%5D)" class="member-name-link">getExpressionMatrix(double[][])</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/AMtools.html" title="class in bacnet.scripts.core.stat">AMtools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getExprProteomesTable()" class="member-name-link">getExprProteomesTable()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getExprProteomesTable(java.lang.String)" class="member-name-link">getExprProteomesTable(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getExprTranscriptomesTable()" class="member-name-link">getExprTranscriptomesTable()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getExprTranscriptomesTable(java.lang.String)" class="member-name-link">getExprTranscriptomesTable(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getExtension()" class="member-name-link">getExtension()</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#getExtension()" class="member-name-link">getExtension()</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/FileUtils.html#getExtension(java.lang.String)" class="member-name-link">getExtension(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Get the extension in a fileName</div>
</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#getFAAPath(java.lang.String)" class="member-name-link">getFAAPath(String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getFeature(java.lang.String)" class="member-name-link">getFeature(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Get the feature in the HashMap, pointed by the key attribute<br>
 return "" if not found</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getFeatures()" class="member-name-link">getFeatures()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures()" class="member-name-link">getFeatures()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures(int)" class="member-name-link">getFeatures(int)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Return features at a sequence position</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures(java.lang.String,int)" class="member-name-link">getFeatures(String, int)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Return features at a sequence position by type</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getFeaturesByType(java.lang.String)" class="member-name-link">getFeaturesByType(String)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getFeaturesKeyWord()" class="member-name-link">getFeaturesKeyWord()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getFeaturesText()" class="member-name-link">getFeaturesText()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Create a String containing all features and their attribute</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getFeaturesTextForTable(java.lang.String)" class="member-name-link">getFeaturesTextForTable(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Used when displaying information in <code>SrnaView</code><br>
 Display only supplementary information of each Srna publication</div>
</dd>
<dt><a href="../bacnet/raprcp/DownloadServiceHandler.html#getFile()" class="member-name-link">getFile()</a> - Method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/FileUtils.html#getFileBytes(java.io.File)" class="member-name-link">getFileBytes(File)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/DownloadServiceHandler.html#getFileName()" class="member-name-link">getFileName()</a> - Method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/InternalBrowser.html#getFileName()" class="member-name-link">getFileName()</a> - Method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SynTView.html#getFilteredNetwork()" class="member-name-link">getFilteredNetwork()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#getFilteredNetwork()" class="member-name-link">getFilteredNetwork()</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/FilterList.html#getFilters()" class="member-name-link">getFilters()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getFirstChromosome()" class="member-name-link">getFirstChromosome()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Get the first chromosome in the HashMap
 
 Is Used to replace: Genome.loadGenome(genomeName).getChromosomes().get(0)</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getFirstChromosome()" class="member-name-link">getFirstChromosome()</a> - Method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Get the first chromosome in the HashMap
 
 Is Used to replace: GenomeNCBI.getChromosomes().get(0)</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Codon.html#getFirstCodon(bacnet.datamodel.sequence.Sequence)" class="member-name-link">getFirstCodon(Sequence)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>
<div class="block">Return first threee nucleotide of a Sequence</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapperList.html#getFirstMapper()" class="member-name-link">getFirstMapper()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCPList.html#getFirstMapper()" class="member-name-link">getFirstMapper()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getFirstRowName()" class="member-name-link">getFirstRowName()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#getFont(java.lang.String,int,int,boolean,boolean)" class="member-name-link">getFont(String, int, int, boolean, boolean)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns a <code>Font</code> based on its name, height and style.</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getFontDouble()" class="member-name-link">getFontDouble()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getFontDouble()" class="member-name-link">getFontDouble()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getFontRowName()" class="member-name-link">getFontRowName()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getFontString()" class="member-name-link">getFontString()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getFontText()" class="member-name-link">getFontText()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#getFormylatedExperiment()" class="member-name-link">getFormylatedExperiment()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>
<div class="block">Search if the TIS was found formylated in the first experiment, or the second, or both</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Frame.html#getForwardFrames()" class="member-name-link">getForwardFrames()</a> - Static method in enum org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></dt>
<dd>
<div class="block">Returns all frames in the forward orientation</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#getFoundIn()" class="member-name-link">getFoundIn()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#getFoundInText()" class="member-name-link">getFoundInText()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#getFromCompoundSet()" class="member-name-link">getFromCompoundSet()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermUtils.html#getFullORF(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Genome)" class="member-name-link">getFullORF(Sequence, Genome)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in frame) containing the
 peptide (amino acid sequence) <br></div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermUtils.html#getFullORF(java.lang.String,int,int,boolean,bacnet.datamodel.sequence.Genome)" class="member-name-link">getFullORF(String, int, int, boolean, Genome)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in frame) containing the
 peptide (amino acid sequence) <br></div>
</dd>
<dt><a href="../org/biojava3/core/sequence/RNASequence.html#getGC()" class="member-name-link">getGC()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#getGCCount()" class="member-name-link">getGCCount()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Get the GC count in the DNA Sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGChecksum(java.util.List)" class="member-name-link">getGCGChecksum(List&lt;S&gt;)</a> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Calculates GCG checksum for entire list of sequences</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGChecksum(S)" class="member-name-link">getGCGChecksum(S)</a> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Calculates GCG checksum for a given sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGHeader(java.util.List)" class="member-name-link">getGCGHeader(List&lt;S&gt;)</a> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Assembles a GCG file header</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGType(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">getGCGType(CompoundSet&lt;C&gt;)</a> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Determines GCG type</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getGCNumber(java.lang.String)" class="member-name-link">getGCNumber(String)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Count the number of G and C nucleotides</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GCStats.html#getGCStats(java.util.Collection)" class="member-name-link">getGCStats(Collection&lt;DNASequence&gt;)</a> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GCStats.html" title="class in org.biojava3.genome.parsers.gff">GCStats</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GCStats.html#getGCStatsString(java.util.Collection)" class="member-name-link">getGCStatsString(Collection&lt;String&gt;)</a> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GCStats.html" title="class in org.biojava3.genome.parsers.gff">GCStats</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#getGene(java.lang.String)" class="member-name-link">getGene(String)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>
<div class="block">Get the gene based on accession.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getGene(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getGene(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#getGeneDownstream()" class="member-name-link">getGeneDownstream()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>
<div class="block">Return the gene downstream on the + strand</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getGeneDownstream(bacnet.datamodel.sequence.Gene)" class="member-name-link">getGeneDownstream(Gene)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>
<div class="block">Return the gene downstream</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/EGDeWTdata.html#getGeneExpr()" class="member-name-link">getGeneExpr()</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/EGDeWTdata.html" title="class in bacnet.datamodel.dataset">EGDeWTdata</a></dt>
<dd>
<div class="block">Read GeneEpression corresponding to NAME_Mean, and return it</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getGeneExpr(bacnet.datamodel.sequence.Gene,java.lang.String)" class="member-name-link">getGeneExpr(Gene, String)</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Return the median value expression (calculated on 10 BHI 37C data) in the Gene Expression array
 of the gene</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getGeneExprs()" class="member-name-link">getGeneExprs()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#getGeneExprs()" class="member-name-link">getGeneExprs()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Return all GeneExpression data contains in all BioCondition</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getGeneExprs()" class="member-name-link">getGeneExprs()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of GeneExpressions</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getGeneExprs(java.lang.String)" class="member-name-link">getGeneExprs(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of GeneExpression given a bioCondName</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getGeneFromName(java.lang.String)" class="member-name-link">getGeneFromName(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Go through all chromosomes and each Genes map and GeneName map to find the corresponding gene</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getGeneFromName(java.lang.String)" class="member-name-link">getGeneFromName(String)</a> - Method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Go through codingSequence and nonCodingSequence searching for corresponding accession or locus</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getGeneFromName(java.lang.String,java.lang.String)" class="member-name-link">getGeneFromName(String, String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Go through all chromosomes and each Genes map and GeneName map to find the corresponding gene</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getGeneFromProteinId(java.lang.String)" class="member-name-link">getGeneFromProteinId(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Go through all chromosomes and Gene from corresponding proteinid</div>
</dd>
<dt><a href="../bacnet/Database.html#getGeneListEGDe()" class="member-name-link">getGeneListEGDe()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getGeneModification()" class="member-name-link">getGeneModification()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#getGeneName()" class="member-name-link">getGeneName()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getGeneName()" class="member-name-link">getGeneName()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">Get the gene name associated with this sequence.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getGeneName(java.lang.String)" class="member-name-link">getGeneName(String)</a> - Method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Look in LocusTagToGeneName if a geneName exists</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getGeneNameList()" class="member-name-link">getGeneNameList()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>
<div class="block">Return a List with all name of genes</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getGeneNames()" class="member-name-link">getGeneNames()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return all the Gene locus tag name available in every chromosome</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getGeneNameToLocusTagMap()" class="member-name-link">getGeneNameToLocusTagMap()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getGeneNameToLocusTagMap()" class="member-name-link">getGeneNameToLocusTagMap()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#getGeneralExp()" class="member-name-link">getGeneralExp()</a> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Load all experiment<br>
 If all experiments are already loaded just get the general experiment found in
 ModelProvider.INSTANCE</div>
</dd>
<dt><a href="../bacnet/Database.html#getGeneralExperiment()" class="member-name-link">getGeneralExperiment()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SynTView.html#getGeneralNetwork()" class="member-name-link">getGeneralNetwork()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#getGeneralNetwork()" class="member-name-link">getGeneralNetwork()</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getGeneralWtName()" class="member-name-link">getGeneralWtName()</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getGenes()" class="member-name-link">getGenes()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getGenes()" class="member-name-link">getGenes()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getGenes()" class="member-name-link">getGenes()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return all Genes in all chromosomes</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Operon.html#getGenes()" class="member-name-link">getGenes()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getGenesAlternative()" class="member-name-link">getGenesAlternative()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#getGeneSequences()" class="member-name-link">getGeneSequences()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>
<div class="block">Get the list of genes that have been added to the ChromosomeSequence where accession.toString is
 the key.</div>
</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html#getGeneSequences(java.util.Collection)" class="member-name-link">getGeneSequences(Collection&lt;ChromosomeSequence&gt;)</a> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#getGeneToOperon()" class="member-name-link">getGeneToOperon()</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#getGeneToSigMap(bacnet.datamodel.sequence.ChromosomeBacteriaSequence)" class="member-name-link">getGeneToSigMap(ChromosomeBacteriaSequence)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>
<div class="block">Create a mapping between gene to a signature <br>
 All signatures from the folder Database.getSIGNATURES() are used</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#getGeneUpstream()" class="member-name-link">getGeneUpstream()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>
<div class="block">Return the gene upstream on the + strand</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getGeneUpstream(bacnet.datamodel.sequence.Gene)" class="member-name-link">getGeneUpstream(Gene)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>
<div class="block">Return the gene upstream</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#getGenome()" class="member-name-link">getGenome()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#getGenome()" class="member-name-link">getGenome()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getGenome()" class="member-name-link">getGenome()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Return corresponding Genome given by this.genomeName<br>
 Load it if necessary</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getGenome()" class="member-name-link">getGenome()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Return corresponding Genome given by this.genomeName<br>
 Load it if necessary</div>
</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#getGenome()" class="member-name-link">getGenome()</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#getGenome()" class="member-name-link">getGenome()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getGenome()" class="member-name-link">getGenome()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#getGenome()" class="member-name-link">getGenome()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getGenome()" class="member-name-link">getGenome()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#getGenome()" class="member-name-link">getGenome()</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getGenome(java.lang.String)" class="member-name-link">getGenome(String)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Read a genome from RAST results</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getGenomeAA()" class="member-name-link">getGenomeAA()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getGenomeAACompl()" class="member-name-link">getGenomeAACompl()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getGenomeAAMinus()" class="member-name-link">getGenomeAAMinus()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getGenomeAAMinusCompl()" class="member-name-link">getGenomeAAMinusCompl()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getGenomeAAPlus()" class="member-name-link">getGenomeAAPlus()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getGenomeAAPlusCompl()" class="member-name-link">getGenomeAAPlusCompl()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getGenomeArrayPath()" class="member-name-link">getGenomeArrayPath()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#getGenomeElements()" class="member-name-link">getGenomeElements()</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#getGenomeElements()" class="member-name-link">getGenomeElements()</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getGenomeList()" class="member-name-link">getGenomeList()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getGenomeName()" class="member-name-link">getGenomeName()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#getGenomeName()" class="member-name-link">getGenomeName()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getGenomeName()" class="member-name-link">getGenomeName()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getGenomeName()" class="member-name-link">getGenomeName()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SynTView.html#getGenomeName()" class="member-name-link">getGenomeName()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getGenomeName()" class="member-name-link">getGenomeName()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#getGenomeName()" class="member-name-link">getGenomeName()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#getGenomeName()" class="member-name-link">getGenomeName()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#getGenomeName()" class="member-name-link">getGenomeName()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#getGenomeName()" class="member-name-link">getGenomeName()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#getGenomeName()" class="member-name-link">getGenomeName()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getGenomeName()" class="member-name-link">getGenomeName()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#getGenomeName()" class="member-name-link">getGenomeName()</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#getGenomeNCBILink(java.lang.String)" class="member-name-link">getGenomeNCBILink(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getGenomes()" class="member-name-link">getGenomes()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DataValidation.html#getGenomes()" class="member-name-link">getGenomes()</a> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>
<div class="block">GETTER AND SETTER</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#getGenomes()" class="member-name-link">getGenomes()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>
<div class="block">get the list of genome IDs and corresponding genome Names<br>
 And verify if all genomes are here!</div>
</dd>
<dt><a href="../bacnet/Database.html#getGENOMES_PATH()" class="member-name-link">getGENOMES_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html#getGenomesAvalaible()" class="member-name-link">getGenomesAvalaible()</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html" title="class in bacnet.scripts.phylogeny">GenomePhylogeny</a></dt>
<dd>
<div class="block">Create list of genomes which contains all genome available <br>
 
 This genome list contains all the genomes both from GenomeFolder and GenomePhylogeny.PATH_TABLE
 (=All bacteria phylogeny)</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getGenomeSelected()" class="member-name-link">getGenomeSelected()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">The comboGenome contains modified genome name so we need this method to get selected element<br>
 a '*' is add to genome name when a transcriptome data is available</div>
</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#getGenomeSelected()" class="member-name-link">getGenomeSelected()</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>
<div class="block">The comboGenome contains modified genome name so we need this method to get selected element<br>
 a '*' is add to genome name when a transcriptome data is available</div>
</dd>
<dt><a href="../bacnet/Database.html#getGenomesNCBI()" class="member-name-link">getGenomesNCBI()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getGenomeUsed()" class="member-name-link">getGenomeUsed()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getGenomeViewerColumnIndex()" class="member-name-link">getGenomeViewerColumnIndex()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Return the appropriate column index for displaying in the Genome viewer<br>
 By order of priority:
 LOGFC containing column
 FC containing column
 p-value containing column
 VALUE
 first column</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#getGExpressionData(java.util.ArrayList)" class="member-name-link">getGExpressionData(ArrayList&lt;String&gt;)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">List all TilingData available by biocondName.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#getGExpressionData(java.util.ArrayList,java.util.ArrayList)" class="member-name-link">getGExpressionData(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Load all Tiling present in leftBCs and rightBCs.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getGlaserFC(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getGlaserFC(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/GlaserFCannotation.html#getGlaserFCDescription(java.lang.String)" class="member-name-link">getGlaserFCDescription(String)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/GlaserFCannotation.html" title="class in bacnet.datamodel.annotation">GlaserFCannotation</a></dt>
<dd>
<div class="block">Return the Glaser Functional categories description corresponding to a specific glaserFCID</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/GlaserFCannotation.html#getGlaserFCMap()" class="member-name-link">getGlaserFCMap()</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/GlaserFCannotation.html" title="class in bacnet.datamodel.annotation">GlaserFCannotation</a></dt>
<dd>
<div class="block">Return a map from glaserFCID to Glaser et al. 2001 Functional categories description</div>
</dd>
<dt><a href="../bacnet/Database.html#getGoogleId()" class="member-name-link">getGoogleId()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorSWT.html#getGreen()" class="member-name-link">getGreen()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getGrowth()" class="member-name-link">getGrowth()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getHeader(int)" class="member-name-link">getHeader(int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html#getHeader(S)" class="member-name-link">getHeader(S)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderFormat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/template/FastaHeaderFormatInterface.html#getHeader(S)" class="member-name-link">getHeader(S)</a> - Method in interface org.biojava3.core.sequence.io.template.<a href="../org/biojava3/core/sequence/io/template/FastaHeaderFormatInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderFormatInterface</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getHeaderAnnotation()" class="member-name-link">getHeaderAnnotation()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getHeaderColor()" class="member-name-link">getHeaderColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getHeaderColorSWT()" class="member-name-link">getHeaderColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getHeaderColorSWT()" class="member-name-link">getHeaderColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getHeaders()" class="member-name-link">getHeaders()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getHeadersListToArray()" class="member-name-link">getHeadersListToArray()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getHeadersToArray()" class="member-name-link">getHeadersToArray()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/NCBIFastaHeaderParser.html#getHeaderValues(java.lang.String)" class="member-name-link">getHeaderValues(String)</a> - Method in class bacnet.reader.<a href="../bacnet/reader/NCBIFastaHeaderParser.html" title="class in bacnet.reader">NCBIFastaHeaderParser</a></dt>
<dd>
<div class="block">Parse out the components where some have a | and others do not</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#getHeight()" class="member-name-link">getHeight()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/MatrixSize.html#getHeight()" class="member-name-link">getHeight()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/MatrixSize.html#getHeightDefault()" class="member-name-link">getHeightDefault()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getHeightPix()" class="member-name-link">getHeightPix()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getHeightPix()" class="member-name-link">getHeightPix()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/query/BlastXMLQuery.html#getHitsQueryDef(double)" class="member-name-link">getHitsQueryDef(double)</a> - Method in class org.biojava3.genome.query.<a href="../org/biojava3/genome/query/BlastXMLQuery.html" title="class in org.biojava3.genome.query">BlastXMLQuery</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#getHTMLVersion(java.lang.Object%5B%5D%5B%5D)" class="member-name-link">getHTMLVersion(Object[][])</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Save the table in a text file containing an HTML table</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#getId()" class="member-name-link">getId()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getId()" class="member-name-link">getId()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#getId()" class="member-name-link">getId()</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getId()" class="member-name-link">getId()</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#getID()" class="member-name-link">getID()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/AccessionID.html#getID()" class="member-name-link">getID()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/TaxonomyID.html#getID()" class="member-name-link">getID()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getIDFormat(java.util.List)" class="member-name-link">getIDFormat(List&lt;S&gt;)</a> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Creates format String for accession IDs</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getIDFPath(java.lang.String)" class="member-name-link">getIDFPath(String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Get path of IDF file for a specific dataID (ex: E-MTAB-1800)</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getIDFPath(java.lang.String)" class="member-name-link">getIDFPath(String)</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Get path of IDF file for a specific dataID (ex: E-MTAB-1800)</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#getImage(java.lang.Class,java.lang.String)" class="member-name-link">getImage(Class&lt;?&gt;, String)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> stored in the file at the specified path relative to the specified
 class.</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#getImage(java.lang.String)" class="member-name-link">getImage(String)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> stored in the file at the specified path.</div>
</dd>
<dt><a href="../bacnet/swt/ResourceManager.html#getImage(org.eclipse.jface.resource.ImageDescriptor)" class="member-name-link">getImage(ImageDescriptor)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> based on the specified <code>ImageDescriptor</code>.</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getImageChecked()" class="member-name-link">getImageChecked()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/swt/ResourceManager.html#getImageDescriptor(java.lang.Class,java.lang.String)" class="member-name-link">getImageDescriptor(Class&lt;?&gt;, String)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>ImageDescriptor</code> stored in the file at the specified path relative to the
 specified class.</div>
</dd>
<dt><a href="../bacnet/swt/ResourceManager.html#getImageDescriptor(java.lang.String)" class="member-name-link">getImageDescriptor(String)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>ImageDescriptor</code> stored in the file at the specified path.</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getImageGeneExpr()" class="member-name-link">getImageGeneExpr()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getImageRNASeq()" class="member-name-link">getImageRNASeq()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getImageTilingGeneExpr()" class="member-name-link">getImageTilingGeneExpr()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getImageTSS()" class="member-name-link">getImageTSS()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getImageTSSTilingGeneExpr()" class="member-name-link">getImageTSSTilingGeneExpr()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getImageUnchecked()" class="member-name-link">getImageUnchecked()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getIndexOf(C)" class="member-name-link">getIndexOf(C)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getIndexOf(C)" class="member-name-link">getIndexOf(C)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getIndexOf(C)" class="member-name-link">getIndexOf(C)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getIndexOf(C)" class="member-name-link">getIndexOf(C)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getIndexOf(C)" class="member-name-link">getIndexOf(C)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns the first occurrence of the given compound in this store; performs a linear search</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getIndexOf(C)" class="member-name-link">getIndexOf(C)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getIndexOf(C)" class="member-name-link">getIndexOf(C)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Returns 1 if the given compound is equal to the one given during construction; otherwise will
 return -1.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getIndexOf(C)" class="member-name-link">getIndexOf(C)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Sequence.html#getIndexOf(C)" class="member-name-link">getIndexOf(C)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Scans through the Sequence looking for the first occurrence of the given compound</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getIndexOf(C)" class="member-name-link">getIndexOf(C)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html#getIndexOf(C)" class="member-name-link">getIndexOf(C)</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava3.core.sequence.views">ComplementSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#getIndexOf(org.biojava3.core.sequence.compound.NucleotideCompound)" class="member-name-link">getIndexOf(NucleotideCompound)</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#getInfo()" class="member-name-link">getInfo()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>
<div class="block">Return a String containing getProduct() and getComment() information</div>
</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#getInHTML(java.lang.String%5B%5D%5B%5D,java.lang.String)" class="member-name-link">getInHTML(String[][], String)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Return an HTML version of the table</div>
</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#getInHTML(java.util.ArrayList,java.lang.String)" class="member-name-link">getInHTML(ArrayList&lt;String&gt;, String)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Return an HTML version of the ArrayList<String></div>
</dd>
<dt><a href="../bacnet/Database.html#getInitView()" class="member-name-link">getInitView()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getInocuaColor(double)" class="member-name-link">getInocuaColor(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getInocuaColor(double)" class="member-name-link">getInocuaColor(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html#getInputStream()" class="member-name-link">getInputStream()</a> - Method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a></dt>
<dd>
<div class="block">Returns the InputStream instance of this classpath resource</div>
</dd>
<dt><a href="../org/biojava3/core/util/InputStreamProvider.html#getInputStream(java.io.File)" class="member-name-link">getInputStream(File)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/InputStreamProvider.html" title="class in org.biojava3.core.util">InputStreamProvider</a></dt>
<dd>
<div class="block">get an InputStream for the file</div>
</dd>
<dt><a href="../org/biojava3/core/util/FlatFileCache.html#getInputStream(java.lang.String)" class="member-name-link">getInputStream(String)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/FlatFileCache.html" title="class in org.biojava3.core.util">FlatFileCache</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/InputStreamProvider.html#getInputStream(java.lang.String)" class="member-name-link">getInputStream(String)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/InputStreamProvider.html" title="class in org.biojava3.core.util">InputStreamProvider</a></dt>
<dd>
<div class="block">get an InputStream for this file</div>
</dd>
<dt><a href="../org/biojava3/core/util/InputStreamProvider.html#getInputStream(java.net.URL)" class="member-name-link">getInputStream(URL)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/InputStreamProvider.html" title="class in org.biojava3.core.util">InputStreamProvider</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getInstance()" class="member-name-link">getInstance()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.html#getInstance()" class="member-name-link">getInstance()</a> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/FlatFileCache.html#getInstance()" class="member-name-link">getInstance()</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/FlatFileCache.html" title="class in org.biojava3.core.util">FlatFileCache</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#getIntronSequences()" class="member-name-link">getIntronSequences()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Get the introns as an ArrayList</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getInverse()" class="member-name-link">getInverse()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getInverse()" class="member-name-link">getInverse()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getInverse()" class="member-name-link">getInverse()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/RNASequence.html#getInverse()" class="member-name-link">getInverse()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>
<div class="block">Get the inverse view of the sequence.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getInverse()" class="member-name-link">getInverse()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getInverse()" class="member-name-link">getInverse()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getInverse()" class="member-name-link">getInverse()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getInverse()" class="member-name-link">getInverse()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getInverse()" class="member-name-link">getInverse()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Sequence.html#getInverse()" class="member-name-link">getInverse()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Does the <em>right thing</em> to get the inverse of the current Sequence.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getInverse()" class="member-name-link">getInverse()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getIsoelectricPoint(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getIsoelectricPoint(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getiTIS()" class="member-name-link">getiTIS()</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getJeffInfo()" class="member-name-link">getJeffInfo()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the different tables from Jeff summary table in 2011 and create a list of sRNAs from the
 different tables</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getJohanssonInfo()" class="member-name-link">getJohanssonInfo()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the table from Johansson 2009 data and create a list of sRNAs from the different tables</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html#getJolleyIDtoInfo()" class="member-name-link">getJolleyIDtoInfo()</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html" title="class in bacnet.scripts.phylogeny">GenomePhylogeny</a></dt>
<dd>
<div class="block">Map Jolley ID to corresponding row in PATH_INFOTABLE</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeaturesKeyWordInterface.html#getKeyWords()" class="member-name-link">getKeyWords()</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getKeyWords()" class="member-name-link">getKeyWords()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">Pull uniprot key words which is a mixed bag of words associated with this sequence</div>
</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getLabel_2()" class="member-name-link">getLabel_2()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getLabel_3()" class="member-name-link">getLabel_3()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getLastIndexOf(C)" class="member-name-link">getLastIndexOf(C)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getLastIndexOf(C)" class="member-name-link">getLastIndexOf(C)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getLastIndexOf(C)" class="member-name-link">getLastIndexOf(C)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getLastIndexOf(C)" class="member-name-link">getLastIndexOf(C)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getLastIndexOf(C)" class="member-name-link">getLastIndexOf(C)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns the last occurrence of the given compound in this store; performs a linear search</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getLastIndexOf(C)" class="member-name-link">getLastIndexOf(C)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getLastIndexOf(C)" class="member-name-link">getLastIndexOf(C)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Returns the length of the Sequence if the given compound was equal to the one given during
 construction.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getLastIndexOf(C)" class="member-name-link">getLastIndexOf(C)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Sequence.html#getLastIndexOf(C)" class="member-name-link">getLastIndexOf(C)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Scans through the Sequence looking for the last occurrence of the given compound</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getLastIndexOf(C)" class="member-name-link">getLastIndexOf(C)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html#getLastIndexOf(C)" class="member-name-link">getLastIndexOf(C)</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava3.core.sequence.views">ComplementSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#getLastIndexOf(org.biojava3.core.sequence.compound.NucleotideCompound)" class="member-name-link">getLastIndexOf(NucleotideCompound)</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getLblDescribedIn()" class="member-name-link">getLblDescribedIn()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getLblGene()" class="member-name-link">getLblGene()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getLblNodiff()" class="member-name-link">getLblNodiff()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getLblNodiffProteome()" class="member-name-link">getLblNodiffProteome()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getLblOrAnyOther()" class="member-name-link">getLblOrAnyOther()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getLblOver()" class="member-name-link">getLblOver()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getLblOverProteome()" class="member-name-link">getLblOverProteome()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getLblOverProteomes()" class="member-name-link">getLblOverProteomes()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getLblOverTranscriptomes()" class="member-name-link">getLblOverTranscriptomes()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getLblPredictedSecondaryStructure()" class="member-name-link">getLblPredictedSecondaryStructure()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getLblStructure()" class="member-name-link">getLblStructure()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getLblTranscriptomesData()" class="member-name-link">getLblTranscriptomesData()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getLblUnder()" class="member-name-link">getLblUnder()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getLblUnderProteome()" class="member-name-link">getLblUnderProteome()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#getLeftBCs()" class="member-name-link">getLeftBCs()</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getLength()" class="member-name-link">getLength()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getLength()" class="member-name-link">getLength()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/CDSSequence.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/ExonSequence.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/IntronSequence.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">The sequence length</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.html#getLength()" class="member-name-link">getLength()</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Returns the length of the outer bounds of this location</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">Get the length of the MSA where it is assumed that all sequence position</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/StartCodonSequence.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/StartCodonSequence.html" title="class in org.biojava3.core.sequence">StartCodonSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/StopCodonSequence.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/StopCodonSequence.html" title="class in org.biojava3.core.sequence">StopCodonSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns the length of the sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Returns the length given during construction</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.html#getLength()" class="member-name-link">getLength()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a></dt>
<dd>
<div class="block">Returns the number of columns in the alignment profile.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/Sequence.html#getLength()" class="member-name-link">getLength()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Returns the length of the Sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/WindowedSequence.html#getLength()" class="member-name-link">getLength()</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence</a></dt>
<dd>
<div class="block">Returns the size of the windowed sequence which is the length by the window size.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getLength(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getLength(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#getLengthAA()" class="member-name-link">getLengthAA()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getLengthGenome()" class="member-name-link">getLengthGenome()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return the total length of the genome by adding all chromosomes lengths</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getLengthProt(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getLengthProt(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html#getLineLength()" class="member-name-link">getLineLength()</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaWriter.html#getLineLength()" class="member-name-link">getLineLength()</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getLinkedBioCondition()" class="member-name-link">getLinkedBioCondition()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html#getList()" class="member-name-link">getList()</a> - Method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a></dt>
<dd>
<div class="block">Returns this resource as a list of Strings</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getList(java.io.BufferedReader)" class="member-name-link">getList(BufferedReader)</a> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Returns the contents of a buffered reader as a list of strings</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getList(java.io.File)" class="member-name-link">getList(File)</a> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Delegates to <a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getList(java.io.InputStream)"><code>IOUtils.getList(InputStream)</code></a> by wrapping the File in a valid stream.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getList(java.io.InputStream)" class="member-name-link">getList(InputStream)</a> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Delegates to <a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getList(java.io.BufferedReader)"><code>IOUtils.getList(BufferedReader)</code></a> by wrapping the InputStream in a valid reader.</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getListData()" class="member-name-link">getListData()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Read DATA_TABLe and return the list of dataset IDs</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getListData()" class="member-name-link">getListData()</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Read DATA_TABLe and return the list of dataset IDs</div>
</dd>
<dt><a href="../bacnet/Database.html#getLISTDATA_COEXPR_NETWORK_TRANSCRIPTOMES_PATH()" class="member-name-link">getLISTDATA_COEXPR_NETWORK_TRANSCRIPTOMES_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Transcriptomes matrix data showing Log(Fold Change) values</div>
</dd>
<dt><a href="../bacnet/Database.html#getListDatabases()" class="member-name-link">getListDatabases()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getListeriaGenomes()" class="member-name-link">getListeriaGenomes()</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Return a List of all Listeria genomes available</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getListeriaMonoGenome()" class="member-name-link">getListeriaMonoGenome()</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Return a List of all Listeria Monocytogenes genomes available</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getListeriaNonMonoGenome()" class="member-name-link">getListeriaNonMonoGenome()</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Return a List of all Listeria NOT Monocytogenes genomes available</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getListGenes()" class="member-name-link">getListGenes()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermUtils.html#getListOfFullORFfromPeptide(java.lang.String,bacnet.datamodel.sequence.Genome)" class="member-name-link">getListOfFullORFfromPeptide(String, Genome)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>
<div class="block">From a list of peptide position retrieve the complete sequence of the ORF (in frame) containing
 the peptide (amino acid sequence) <br>
 WARNING: Be careful that your peptide info (begin,end) are correct.</div>
</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getListSrnas()" class="member-name-link">getListSrnas()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getLocalization()" class="member-name-link">getLocalization()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getLocations()" class="member-name-link">getLocations()</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">A feature could be a single sequence position like a mutation or a post translational
 modification of an amino acid.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getLocations()" class="member-name-link">getLocations()</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">The location(s) of this feature where the location should contain a reference to parent and
 sequence etc.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getLocusTag(java.lang.String)" class="member-name-link">getLocusTag(String)</a> - Method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Look in all LocusTagToGeneName exists</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getLocusTagCodingList(boolean)" class="member-name-link">getLocusTagCodingList(boolean)</a> - Method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Put in a list all the CodingSequence found</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getLocusTagKnownCodingList(boolean)" class="member-name-link">getLocusTagKnownCodingList(boolean)</a> - Method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getLocusTagList(boolean)" class="member-name-link">getLocusTagList(boolean)</a> - Method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Put in a list all the locusTag found in the list of genes</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getLocusTagNonCodingList(boolean)" class="member-name-link">getLocusTagNonCodingList(boolean)</a> - Method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getLocusTagToGeneNameMap()" class="member-name-link">getLocusTagToGeneNameMap()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getLocusTagToGeneNameMap()" class="member-name-link">getLocusTagToGeneNameMap()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getLocusTagToOldLocusTagMap()" class="member-name-link">getLocusTagToOldLocusTagMap()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getLOGFC_MATRIX_PROTEOMES_PATH()" class="member-name-link">getLOGFC_MATRIX_PROTEOMES_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Proteomes matrix data showing relative expression values</div>
</dd>
<dt><a href="../bacnet/Database.html#getLOGFC_MATRIX_TRANSCRIPTOMES_PATH()" class="member-name-link">getLOGFC_MATRIX_TRANSCRIPTOMES_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Transcriptomes matrix data showing Log(Fold Change) values</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getLogfcOmicsMatrix()" class="member-name-link">getLogfcOmicsMatrix()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getLogFCProteomesTable(java.lang.String)" class="member-name-link">getLogFCProteomesTable(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Load Logfc proteome tables for a specific genome</div>
</dd>
<dt><a href="../bacnet/Database.html#getLogFCProteomeTable()" class="member-name-link">getLogFCProteomeTable()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getLogFCTable()" class="member-name-link">getLogFCTable()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getLogFCTranscriptomesTable(java.lang.String)" class="member-name-link">getLogFCTranscriptomesTable(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Load logFC transcriptome table for a specific genome</div>
</dd>
<dt><a href="../bacnet/Database.html#getLogo()" class="member-name-link">getLogo()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getLogo(bacnet.datamodel.sequence.Sequence)" class="member-name-link">getLogo(Sequence)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">For a given Sequence, return the nucleotide sequence before and after starting position</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#getLongName()" class="member-name-link">getLongName()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#getLongName()" class="member-name-link">getLongName()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Compound.html#getLongName()" class="member-name-link">getLongName()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#getLongName()" class="member-name-link">getLongName()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getLongName()" class="member-name-link">getLongName()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getMappingFrame()" class="member-name-link">getMappingFrame()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getMassSpecData()" class="member-name-link">getMassSpecData()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getMatrices()" class="member-name-link">getMatrices()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getMatrices()" class="member-name-link">getMatrices()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of SeqData given a bioCondName</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getMatrices(java.lang.String)" class="member-name-link">getMatrices(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of SeqData given a bioCondName</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getMatrix()" class="member-name-link">getMatrix()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#getMatrix()" class="member-name-link">getMatrix()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#getMatrix()" class="member-name-link">getMatrix()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/FilterList.html#getMatrix()" class="member-name-link">getMatrix()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#getMatrix()" class="member-name-link">getMatrix()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#getMatrixDisplayed()" class="member-name-link">getMatrixDisplayed()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getMax()" class="member-name-link">getMax()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getMax()" class="member-name-link">getMax()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMax()" class="member-name-link">getMax()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMax()" class="member-name-link">getMax()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/LocationHelper.html#getMax(java.util.List)" class="member-name-link">getMax(List&lt;Location&gt;)</a> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Scans through a list of locations to find the Location with the highest end</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.Tools.html#getMax(java.util.List)" class="member-name-link">getMax(List&lt;Location&gt;)</a> - Static method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.Tools.html" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a></dt>
<dd>
<div class="block">Scans through a list of locations to find the Location with the highest end</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMaxColor()" class="member-name-link">getMaxColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMaxColor()" class="member-name-link">getMaxColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMaxColorSWT()" class="member-name-link">getMaxColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMaxColorSWT()" class="member-name-link">getMaxColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMaxPos()" class="member-name-link">getMaxPos()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMaxPos()" class="member-name-link">getMaxPos()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getMaxQuant()" class="member-name-link">getMaxQuant()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getMaxQuantum()" class="member-name-link">getMaxQuantum()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#getMaxSingleCompoundStringLength()" class="member-name-link">getMaxSingleCompoundStringLength()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#getMaxSingleCompoundStringLength()" class="member-name-link">getMaxSingleCompoundStringLength()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundSet.html#getMaxSingleCompoundStringLength()" class="member-name-link">getMaxSingleCompoundStringLength()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></dt>
<dd>
<div class="block">Returns the maximum size of a compound String this set holds</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getMean()" class="member-name-link">getMean()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMean()" class="member-name-link">getMean()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMean()" class="member-name-link">getMean()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#getMean(java.util.ArrayList,java.lang.String)" class="member-name-link">getMean(ArrayList&lt;GeneExpression&gt;, String)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Calculate the mean expression of list of Tiling</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#getMean(java.util.ArrayList,java.lang.String)" class="member-name-link">getMean(ArrayList&lt;Tiling&gt;, String)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Calculate the mean expression of list of Tiling</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getMedia()" class="member-name-link">getMedia()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getMediaGrowthProperties()" class="member-name-link">getMediaGrowthProperties()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getMedian()" class="member-name-link">getMedian()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMedian()" class="member-name-link">getMedian()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMedian()" class="member-name-link">getMedian()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#getMedianValue(java.lang.String)" class="member-name-link">getMedianValue(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Get the median value of corresponding probe in double[][] values</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#getMedianVector()" class="member-name-link">getMedianVector()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Extract the column containing the median<br></div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#getMiddleH()" class="member-name-link">getMiddleH()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#getMiddleV()" class="member-name-link">getMiddleV()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMidLeftColor()" class="member-name-link">getMidLeftColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMidLeftColor()" class="member-name-link">getMidLeftColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMidLeftColorSWT()" class="member-name-link">getMidLeftColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMidLeftColorSWT()" class="member-name-link">getMidLeftColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMidLeftPos()" class="member-name-link">getMidLeftPos()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMidLeftPos()" class="member-name-link">getMidLeftPos()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMidRightColor()" class="member-name-link">getMidRightColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMidRightColor()" class="member-name-link">getMidRightColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMidRightColorSWT()" class="member-name-link">getMidRightColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMidRightColorSWT()" class="member-name-link">getMidRightColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMidRightPos()" class="member-name-link">getMidRightPos()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMidRightPos()" class="member-name-link">getMidRightPos()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getMin()" class="member-name-link">getMin()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getMin()" class="member-name-link">getMin()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMin()" class="member-name-link">getMin()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMin()" class="member-name-link">getMin()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/LocationHelper.html#getMin(java.util.List)" class="member-name-link">getMin(List&lt;Location&gt;)</a> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Scans through a list of locations to find the Location with the lowest start</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.Tools.html#getMin(java.util.List)" class="member-name-link">getMin(List&lt;Location&gt;)</a> - Static method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.Tools.html" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a></dt>
<dd>
<div class="block">Scans through a list of locations to find the Location with the lowest start</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMinColor()" class="member-name-link">getMinColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMinColor()" class="member-name-link">getMinColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMinColorSWT()" class="member-name-link">getMinColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMinColorSWT()" class="member-name-link">getMinColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getMinPos()" class="member-name-link">getMinPos()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getMinPos()" class="member-name-link">getMinPos()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getMissingValuesFreeMatrix()" class="member-name-link">getMissingValuesFreeMatrix()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Return a subMatrix containing only rows without Missingvalues</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getModifications()" class="member-name-link">getModifications()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#getModifs()" class="member-name-link">getModifs()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getModifSequence()" class="member-name-link">getModifSequence()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#getModifsString()" class="member-name-link">getModifsString()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#getMolecularWeight()" class="member-name-link">getMolecularWeight()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#getMolecularWeight()" class="member-name-link">getMolecularWeight()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Compound.html#getMolecularWeight()" class="member-name-link">getMolecularWeight()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#getMolecularWeight()" class="member-name-link">getMolecularWeight()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getMolecularWeight()" class="member-name-link">getMolecularWeight()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#getMolMass()" class="member-name-link">getMolMass()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getMolMass(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getMolMass(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getMouseXPosition()" class="member-name-link">getMouseXPosition()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getMouseXPosition()" class="member-name-link">getMouseXPosition()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getMouseYPosition()" class="member-name-link">getMouseYPosition()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getMouseYPosition()" class="member-name-link">getMouseYPosition()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getMraheilInfo()" class="member-name-link">getMraheilInfo()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the different tables from Mraheil et al. 2011 publication and create a list of sRNAs from
 the different tables</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html#getMultiFasta()" class="member-name-link">getMultiFasta()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html" title="class in bacnet.datamodel.sequence">Genome.GetMultiFastaThread</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html#%3Cinit%3E(java.util.HashMap)" class="member-name-link">GetMultiFastaThread(HashMap&lt;String, String&gt;)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html" title="class in bacnet.datamodel.sequence">Genome.GetMultiFastaThread</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getMULTIOMICS_MATRIX_PATH()" class="member-name-link">getMULTIOMICS_MATRIX_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Proteomes matrix data showing relative expression values</div>
</dd>
<dt><a href="../bacnet/Database.html#getMULTIOMICS_PATH()" class="member-name-link">getMULTIOMICS_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getMutant()" class="member-name-link">getMutant()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#getName()" class="member-name-link">getName()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#getName()" class="member-name-link">getName()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#getName()" class="member-name-link">getName()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getName()" class="member-name-link">getName()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#getName()" class="member-name-link">getName()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getName()" class="member-name-link">getName()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getName()" class="member-name-link">getName()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getName()" class="member-name-link">getName()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getName()" class="member-name-link">getName()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#getName()" class="member-name-link">getName()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#getName()" class="member-name-link">getName()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#getName()" class="member-name-link">getName()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getName()" class="member-name-link">getName()</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastOutput.html#getName(bacnet.scripts.blast.BlastOutput.BlastOutputTYPE)" class="member-name-link">getName(BlastOutput.BlastOutputTYPE)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.html" title="class in bacnet.scripts.blast">BlastOutput</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#getNameColumn()" class="member-name-link">getNameColumn()</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return the column index corresponding to Name</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html#getNameMassSpecDataFinal()" class="member-name-link">getNameMassSpecDataFinal()</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html" title="class in bacnet.scripts.listeriomics.nterm">NTermCreateFinalData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getNameNTermData()" class="member-name-link">getNameNTermData()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getNameRawData()" class="member-name-link">getNameRawData()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getNavigationListener()" class="member-name-link">getNavigationListener()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getNbElements()" class="member-name-link">getNbElements()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getNbElements()" class="member-name-link">getNbElements()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#getNbElementType()" class="member-name-link">getNbElementType()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#getNbSet()" class="member-name-link">getNbSet()</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyTools.html#getNCBIGenomePhylogeny()" class="member-name-link">getNCBIGenomePhylogeny()</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a></dt>
<dd>
<div class="block">From PhyloXMl of all bacteria, get only the genome which have a NCBI genome id</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getNcRNAs()" class="member-name-link">getNcRNAs()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getNcRNAs()" class="member-name-link">getNcRNAs()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return all NcRNAs in all chromosomes</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getNCRNAUpstream(bacnet.datamodel.sequence.Gene)" class="member-name-link">getNCRNAUpstream(Gene)</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">For each gene search if the end of a smallRNA or cisReg element is in the region -200bp - start
 codon</div>
</dd>
<dt><a href="../bacnet/Database.html#getNETWORK_PATH()" class="member-name-link">getNETWORK_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLHelper.html#getNewDocument()" class="member-name-link">getNewDocument()</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#getNewLocusTag()" class="member-name-link">getNewLocusTag()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/RASTEGDeGenome.html#getNewRASTids()" class="member-name-link">getNewRASTids()</a> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/RASTEGDeGenome.html" title="class in bacnet.scripts.genome">RASTEGDeGenome</a></dt>
<dd>
<div class="block">Transform a list of old RAST ids into new RAST id</div>
</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html#getNext()" class="member-name-link">getNext()</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html" title="class in bacnet.scripts.core.vennDiagram">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/CombinationGenerator.html#getNext()" class="member-name-link">getNext()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/CombinationGenerator.html" title="class in bacnet.utils">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getNextCodon()" class="member-name-link">getNextCodon()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getNextNterm(bacnet.datamodel.proteomics.NTerm,java.lang.String)" class="member-name-link">getNextNterm(NTerm, String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Given a NTerm, find the next one in the Annotation</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getNGSSeqs()" class="member-name-link">getNGSSeqs()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getNoDNAList()" class="member-name-link">getNoDNAList()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getNonCodingRegion(java.lang.String)" class="member-name-link">getNonCodingRegion(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>
<div class="block">Get the gene based on accession.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getNoncodingSequenceHashMap()" class="member-name-link">getNoncodingSequenceHashMap()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getNonCodingSequencesList(boolean)" class="member-name-link">getNonCodingSequencesList(boolean)</a> - Method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Put in a list all the NonCodingSequence found</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getNoProtList()" class="member-name-link">getNoProtList()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getNoRNAList()" class="member-name-link">getNoRNAList()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getNotAssociatedList(java.lang.String,boolean)" class="member-name-link">getNotAssociatedList(String, boolean)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Read list of gene not having been associated (RAST and NCBI association)</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getNotDiffExpressedASrnas()" class="member-name-link">getNotDiffExpressedASrnas()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getNotDiffExpressedCisRegs()" class="member-name-link">getNotDiffExpressedCisRegs()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getNotDiffExpressedGenes()" class="member-name-link">getNotDiffExpressedGenes()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#getNotDiffExpresseds()" class="member-name-link">getNotDiffExpresseds()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getNotDiffExpressedSrnas()" class="member-name-link">getNotDiffExpressedSrnas()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#getNote()" class="member-name-link">getNote()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#getNote()" class="member-name-link">getNote()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getNote(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getNote(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getNotesList()" class="member-name-link">getNotesList()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getNotListeriaAndBacillusGenome()" class="member-name-link">getNotListeriaAndBacillusGenome()</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Return a list of all NON Listeria genomes</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getNotListeriaGenome()" class="member-name-link">getNotListeriaGenome()</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Return a list of all NON Listeria genomes</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getNTermDatas()" class="member-name-link">getNTermDatas()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of NTermData given a bioCondName<br></div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getNTermDatas(java.lang.String)" class="member-name-link">getNTermDatas(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of NTermData given a bioCondName<br></div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#getnTermExp()" class="member-name-link">getnTermExp()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getnTermHighlight()" class="member-name-link">getnTermHighlight()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#getnTermRef()" class="member-name-link">getnTermRef()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getnTerms()" class="member-name-link">getnTerms()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#getnTerms()" class="member-name-link">getnTerms()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getNTerms()" class="member-name-link">getNTerms()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Return a list of all NTerms available</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#getnTermsSequences()" class="member-name-link">getnTermsSequences()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>
<div class="block">Return a character string with all peptide Modif sequences</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#getnTermsString()" class="member-name-link">getnTermsString()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>
<div class="block">Return a character string with all peptide names</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getnTermTypeOverlap()" class="member-name-link">getnTermTypeOverlap()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/CompoundColorization.html#getNucleotideColor(java.lang.String,org.eclipse.swt.widgets.Display)" class="member-name-link">getNucleotideColor(String, Display)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/CompoundColorization.html" title="class in bacnet.datamodel.sequence">CompoundColorization</a></dt>
<dd>
<div class="block">Given a Nucleotide, this method return the corresponding color</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getNumberColumn()" class="member-name-link">getNumberColumn()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Get the number of column of the <code>values</code> array</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getNumberColumnWithAnnotation()" class="member-name-link">getNumberColumnWithAnnotation()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Get the number of column of the <code>values</code> array + <code>annotations</code> array</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getNumberRow()" class="member-name-link">getNumberRow()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/Strand.html#getNumericRepresentation()" class="member-name-link">getNumericRepresentation()</a> - Method in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html#getNumLeft()" class="member-name-link">getNumLeft()</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html" title="class in bacnet.scripts.core.vennDiagram">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/CombinationGenerator.html#getNumLeft()" class="member-name-link">getNumLeft()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/CombinationGenerator.html" title="class in bacnet.utils">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#getOldLocusTag()" class="member-name-link">getOldLocusTag()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getOldLocusTag(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getOldLocusTag(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getOliverInfo()" class="member-name-link">getOliverInfo()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the different tables from Oliver et al. 2009 publication and create a list of sRNAs from the
 different tables</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getOliverInfoEGDePosition()" class="member-name-link">getOliverInfoEGDePosition()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the different tables from Oliver et al. 2009 publication and create a list of sRNAs from the
 different tables</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getOmicsData()" class="member-name-link">getOmicsData()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Put all OmicsData in a list</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#getOne()" class="member-name-link">getOne()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#getOne()" class="member-name-link">getOne()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getOne()" class="member-name-link">getOne()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#getOperon()" class="member-name-link">getOperon()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getOperon(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getOperon(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getOperons()" class="member-name-link">getOperons()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getOperons()" class="member-name-link">getOperons()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getOperons()" class="member-name-link">getOperons()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return all Operon in all chromosomes</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getOrderedRowNames()" class="member-name-link">getOrderedRowNames()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getOrganismName()" class="member-name-link">getOrganismName()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">Get the organism name assigned to this sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getOriginalHeader()" class="member-name-link">getOriginalHeader()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getOverASrnas()" class="member-name-link">getOverASrnas()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#getOverBioConds()" class="member-name-link">getOverBioConds()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getOverCisRegs()" class="member-name-link">getOverCisRegs()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getOverExpressed()" class="member-name-link">getOverExpressed()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getOverGenes()" class="member-name-link">getOverGenes()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getOverlap()" class="member-name-link">getOverlap()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getOverlap(java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">getOverlap(String, String, String)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#getOverlapingGene()" class="member-name-link">getOverlapingGene()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getOverlapList(java.util.ArrayList,java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">getOverlapList(ArrayList&lt;String&gt;, String, String, String)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Read NCBI and RAST genomes and look at overlapping elements</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#getOverlaps()" class="member-name-link">getOverlaps()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#getOverlapsString()" class="member-name-link">getOverlapsString()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getOverSrnas()" class="member-name-link">getOverSrnas()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#getParentChromosomeSequence()" class="member-name-link">getParentChromosomeSequence()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">The parent ChromosomeSequence which contains the actual DNA sequence data</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getParentFeature()" class="member-name-link">getParentFeature()</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">Get the parent Feature</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getParentFeature()" class="member-name-link">getParentFeature()</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Get the parent feature</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getParentGenome()" class="member-name-link">getParentGenome()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getParentGenome()" class="member-name-link">getParentGenome()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>
<div class="block">Get the gene based on accession.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getParentSequence()" class="member-name-link">getParentSequence()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getParentViewId()" class="member-name-link">getParentViewId()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getPartService()" class="member-name-link">getPartService()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getPath()" class="member-name-link">getPath()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/FileUtils.html#getPath(java.lang.String)" class="member-name-link">getPath(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Get only the folder where the file is contained</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getPATH()" class="member-name-link">getPATH()</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermUtils.html#getPATH()" class="member-name-link">getPATH()</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getPATH()" class="member-name-link">getPATH()</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getPDBCharacter(boolean,char,char,boolean,char)" class="member-name-link">getPDBCharacter(boolean, char, char, boolean, char)</a> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Creates formatted String for a single character of PDB output</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getPDBConservation(boolean,char,char,boolean)" class="member-name-link">getPDBConservation(boolean, char, char, boolean)</a> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Creates formatted String for displaying conservation in PDB output</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getPDBLegend()" class="member-name-link">getPDBLegend()</a> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Creates formatted String for displaying conservation legend in PDB output</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getPeptides()" class="member-name-link">getPeptides()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Link peptideSequence to <code>ArrayList(NTerm)</code><br>
 This correspond to the list of <code>NTerm</code> having same peptideSequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/CDSSequence.html#getPhase()" class="member-name-link">getPhase()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewHomologTools.html#getPhyloFigure(bacnet.datamodel.sequence.Gene,java.util.ArrayList)" class="member-name-link">getPhyloFigure(Gene, ArrayList&lt;String&gt;)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneViewHomologTools.html" title="class in bacnet.sequenceTools">GeneViewHomologTools</a></dt>
<dd>
<div class="block">Read Phylogenomic figure and modify the labels with homolog table</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#getPhyloFigure(java.util.ArrayList)" class="member-name-link">getPhyloFigure(ArrayList&lt;String&gt;)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>
<div class="block">DO NOT WORK WITH NEW VERSION OF FIGTREE Return the text file of the SVG
 figure, modified with highlighted strains</div>
</dd>
<dt><a href="../bacnet/datamodel/phylogeny/Phylogenomic.html#getPhylogenomicFigurePath()" class="member-name-link">getPhylogenomicFigurePath()</a> - Static method in class bacnet.datamodel.phylogeny.<a href="../bacnet/datamodel/phylogeny/Phylogenomic.html" title="class in bacnet.datamodel.phylogeny">Phylogenomic</a></dt>
<dd>
<div class="block">Test if the Phylogenomic figure exist and rturn its path</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/COGannotation.html#getPieChartData(java.util.HashMap)" class="member-name-link">getPieChartData(HashMap&lt;String, ArrayList&lt;String&gt;&gt;)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>
<div class="block">Transform the result of getCogClassification() method into an ExpressionMatrix which can be load
 by PieChartView for display</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/GlaserFCannotation.html#getPieChartData(java.util.HashMap)" class="member-name-link">getPieChartData(HashMap&lt;String, ArrayList&lt;String&gt;&gt;)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/GlaserFCannotation.html" title="class in bacnet.datamodel.annotation">GlaserFCannotation</a></dt>
<dd>
<div class="block">Transform the result of getGlaserFCclassification() method into an ExpressionMatrix which can be
 load by PieChartView for display</div>
</dd>
<dt><a href="../bacnet/swt/ResourceManager.html#getPluginFileURL(java.lang.String,java.lang.String)" class="member-name-link">getPluginFileURL(String, String)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/URL.html" title="class or interface in java.net" class="external-link"><code>URL</code></a> based on a <code>Bundle</code> and resource entry path.</div>
</dd>
<dt><a href="../bacnet/swt/ResourceManager.html#getPluginImage(java.lang.Object,java.lang.String)" class="member-name-link">getPluginImage(Object, String)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span>
<div class="deprecation-comment">Use <a href="../bacnet/swt/ResourceManager.html#getPluginImage(java.lang.String,java.lang.String)"><code>ResourceManager.getPluginImage(String, String)</code></a> instead.</div>
</div>
</dd>
<dt><a href="../bacnet/swt/ResourceManager.html#getPluginImage(java.lang.String,java.lang.String)" class="member-name-link">getPluginImage(String, String)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> based on a <code>Bundle</code> and resource entry path.</div>
</dd>
<dt><a href="../bacnet/swt/ResourceManager.html#getPluginImageDescriptor(java.lang.Object,java.lang.String)" class="member-name-link">getPluginImageDescriptor(Object, String)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span>
<div class="deprecation-comment">Use <a href="../bacnet/swt/ResourceManager.html#getPluginImageDescriptor(java.lang.String,java.lang.String)"><code>ResourceManager.getPluginImageDescriptor(String, String)</code></a> instead.</div>
</div>
</dd>
<dt><a href="../bacnet/swt/ResourceManager.html#getPluginImageDescriptor(java.lang.String,java.lang.String)" class="member-name-link">getPluginImageDescriptor(String, String)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>ImageDescriptor</code> based on a <code>Bundle</code> and resource entry path.</div>
</dd>
<dt><a href="../bacnet/swt/ResourceManager.html#getPluginImageURL(java.lang.Object,java.lang.String)" class="member-name-link">getPluginImageURL(Object, String)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/URL.html" title="class or interface in java.net" class="external-link"><code>URL</code></a> based on a plugin and file path.</div>
</dd>
<dt><a href="../bacnet/utils/HTMLUtils.html#getPluginTextFile(java.lang.String,java.lang.String)" class="member-name-link">getPluginTextFile(String, String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/HTMLUtils.html" title="class in bacnet.utils">HTMLUtils</a></dt>
<dd>
<div class="block">Returns an <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link"><code>String</code></a> based on a <code>Bundle</code> and resource entry path.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#getPosition()" class="member-name-link">getPosition()</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimplePoint.html#getPosition()" class="member-name-link">getPosition()</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Point.html#getPosition()" class="member-name-link">getPosition()</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a></dt>
<dd>
<div class="block">Returns the position held by this object</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#getPosition(int)" class="member-name-link">getPosition(int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Return the index position in the tiling matrix of a particular probes closest to bpPosition</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getPreviousCodon()" class="member-name-link">getPreviousCodon()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getPreviousNterm(bacnet.datamodel.proteomics.NTerm,java.lang.String)" class="member-name-link">getPreviousNterm(NTerm, String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Given a NTerm, find the previous one in the Annotation</div>
</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#getPrideLink(java.lang.String)" class="member-name-link">getPrideLink(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>
<div class="block">Get pride database link to a specific project</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#getProbes()" class="member-name-link">getProbes()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#getProbes()" class="member-name-link">getProbes()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getProbesGExpression()" class="member-name-link">getProbesGExpression()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Read probes for GExpression in GeneExpression.PROBES_PATH</div>
</dd>
<dt><a href="../bacnet/Database.html#getProbesTiling()" class="member-name-link">getProbesTiling()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Read probes for Tiling in Tiling.PROBES_PATH</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#getProduct()" class="member-name-link">getProduct()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/NcRNA.html#getProduct()" class="member-name-link">getProduct()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getProduct(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getProduct(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getProjectName()" class="member-name-link">getProjectName()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#getProperties()" class="member-name-link">getProperties()</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>
<div class="block">Get the properties</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#getProtein_id()" class="member-name-link">getProtein_id()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getProteinCDSSequences()" class="member-name-link">getProteinCDSSequences()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>
<div class="block">Return a list of protein sequences based on each CDS sequence where the phase shift between two
 CDS sequences is assigned to the CDS sequence that starts the triplet.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getProteinID(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getProteinID(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Search for protein-id in the Note section of Gene<br>
 Search first : protein_id attribute<br>
 Search then : protein_Id attribute <br>
 Finally in CDS section of Note search for: "Dbxref :" and "Genbank:" protein id</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getProteinIDTolocusTag()" class="member-name-link">getProteinIDTolocusTag()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getProteinIDToOldLocusTagMap()" class="member-name-link">getProteinIDToOldLocusTagMap()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#getProteinNCBILink(java.lang.String)" class="member-name-link">getProteinNCBILink(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/RNASequence.html#getProteinSequence()" class="member-name-link">getProteinSequence()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>
<div class="block">Get the ProteinSequence from the RNA sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getProteinSequence()" class="member-name-link">getProteinSequence()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>
<div class="block">Get the protein sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/RNASequence.html#getProteinSequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)" class="member-name-link">getProteinSequence(TranscriptionEngine)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>
<div class="block">Get the ProteinSequene from the RNA sequence with user defined transcription engine</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getProteinSequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)" class="member-name-link">getProteinSequence(TranscriptionEngine)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>
<div class="block">Get the protein sequence with user defined TranscriptEngine</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getProteinSequenceCreator()" class="member-name-link">getProteinSequenceCreator()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getProteinSequenceFromFasta(java.lang.String,java.lang.String)" class="member-name-link">getProteinSequenceFromFasta(String, String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Read all accessible FAA files to extract protein sequence of a gene</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html#getProteinSequences()" class="member-name-link">getProteinSequences()</a> - Method in class org.biojava3.genome.parsers.geneid.<a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html" title="class in org.biojava3.genome.parsers.geneid">GeneIDXMLReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html#getProteinSequences(java.util.Collection)" class="member-name-link">getProteinSequences(Collection&lt;ChromosomeSequence&gt;)</a> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getProteolyseAssociated(bacnet.datamodel.proteomics.NTerm)" class="member-name-link">getProteolyseAssociated(NTerm)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Go through <code>proteolyses</code> map, and find the list of NTerm which might be associated to
 an input <code>NTerm</code><br>
 Warning: <code>ArrayList</code> in return contains <code>key NTerm</code> of
 <code>proteolyses</code> but not the input <code>nTerm</code></div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getProteomeGenomes()" class="member-name-link">getProteomeGenomes()</a> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Get all genomes associated to a proteomics dataset</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getProteomes()" class="member-name-link">getProteomes()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getProteomes()" class="member-name-link">getProteomes()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of MassSpecData given a bioCondName<br></div>
</dd>
<dt><a href="../bacnet/scripts/database/DataValidation.html#getProteomes()" class="member-name-link">getProteomes()</a> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getProteomes(java.lang.String)" class="member-name-link">getProteomes(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of MassSpecData given a bioCondName<br></div>
</dd>
<dt><a href="../bacnet/Database.html#getPROTEOMES_PATH()" class="member-name-link">getPROTEOMES_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getProteomesArrayPath()" class="member-name-link">getProteomesArrayPath()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getProteomesComparisonsArrayPath()" class="member-name-link">getProteomesComparisonsArrayPath()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getProteomicsData()" class="member-name-link">getProteomicsData()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Put all OmicsData in a list</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AccesionedLocation.html#getProxySequenceReader()" class="member-name-link">getProxySequenceReader()</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AccesionedLocation.html" title="interface in org.biojava3.core.sequence.location.template">AccesionedLocation</a></dt>
<dd>
<div class="block">Return the proxy reader used to get sequence for this location.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getProxySequenceReader()" class="member-name-link">getProxySequenceReader()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#getPubMedLink(java.lang.String)" class="member-name-link">getPubMedLink(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/QualityFeature.html#getQualities()" class="member-name-link">getQualities()</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/QualityFeature.html" title="class in org.biojava3.core.sequence.features">QualityFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/QualityFeature.html#getQualities(int,int)" class="member-name-link">getQualities(int, int)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/QualityFeature.html" title="class in org.biojava3.core.sequence.features">QualityFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/QualityFeature.html#getQualityAt(int)" class="member-name-link">getQualityAt(int)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/QualityFeature.html" title="class in org.biojava3.core.sequence.features">QualityFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/QuantityFeature.html#getQuantities()" class="member-name-link">getQuantities()</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/QuantityFeature.html" title="class in org.biojava3.core.sequence.features">QuantityFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#getRASTinfo()" class="member-name-link">getRASTinfo()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getRASTinfo(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getRASTinfo(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#getRawDatas()" class="member-name-link">getRawDatas()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getRead()" class="member-name-link">getRead()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorSWT.html#getRed()" class="member-name-link">getRed()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getRef()" class="member-name-link">getRef()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#getReference()" class="member-name-link">getReference()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getReference()" class="member-name-link">getReference()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#getRefSequence()" class="member-name-link">getRefSequence()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermTranscriptome.html#getRegulatoryRegion()" class="member-name-link">getRegulatoryRegion()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermTranscriptome.html" title="class in bacnet.scripts.listeriomics.nterm">NTermTranscriptome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getRelevantSubLocations()" class="member-name-link">getRelevantSubLocations()</a> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>
<div class="block">Returns the normalised list of sub locations i.e. only those locations which do not have a sub
 location.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.html#getRelevantSubLocations()" class="member-name-link">getRelevantSubLocations()</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">An extension to <a href="../org/biojava3/core/sequence/location/template/Location.html#getSubLocations()"><code>Location.getSubLocations()</code></a> which returns sub-locations of sub-locations; this
 will continue until it runs out of those locations.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getRelevantSubSequence(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">getRelevantSubSequence(Sequence&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.html#getRelevantSubSequence(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">getRelevantSubSequence(Sequence&lt;C&gt;)</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Will return a SequenceReader object which offers a view of all resolved locations i.e. those
 locations which are not complex and define the true Sequence represented</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#getResultsFromXML(java.lang.String)" class="member-name-link">getResultsFromXML(String)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>
<div class="block">From an XML blast file, extract blast results</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#getReverse()" class="member-name-link">getReverse()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Returns a Sequence which runs in the current reverse order</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/Strand.html#getReverse()" class="member-name-link">getReverse()</a> - Method in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#getReverseComplement()" class="member-name-link">getReverseComplement()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Delegates to <a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getInverse()"><code>AbstractSequence.getInverse()</code></a> for the reverse complement</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/RNASequence.html#getReverseComplement()" class="member-name-link">getReverseComplement()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>
<div class="block">Get reverse complement view of the sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Frame.html#getReverseFrames()" class="member-name-link">getReverseFrames()</a> - Static method in enum org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></dt>
<dd>
<div class="block">Returns all frames which are in the reverse orientation</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#getRfamDirectory()" class="member-name-link">getRfamDirectory()</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>
<div class="block">Project.getANNOTATION_PATH()+"RfamData"+File.separator;</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#getRfamElementsFromGenome(java.lang.String)" class="member-name-link">getRfamElementsFromGenome(String)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>
<div class="block">Return the list of Rfam element in a particular Genome<br>
 <br>
 
 Search for all NCBI GFF in a particular genome folder (e.g. genomeName),<br>
 thanks to RFAM_GENOME table it found corresponding Rfam GFF file if it exists<br>
 and create HashMap&lt;String, RfamElement> rfamList</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#getRfamElementsFromGFF(java.lang.String)" class="member-name-link">getRfamElementsFromGFF(String)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>
<div class="block">Read a Gff from Rfam and get all possible information</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#getRfamElementsFromGFF(java.lang.String,java.util.HashMap)" class="member-name-link">getRfamElementsFromGFF(String, HashMap&lt;String, RfamElement&gt;)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>
<div class="block">Read a Gff from Rfam and add the information to an existing HashMap</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#getRfamGenomeId(java.lang.String)" class="member-name-link">getRfamGenomeId(String)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>
<div class="block">Given an NCBI Gff file name, this function search in RFAM_GENOME table and return Rfam Gff file
 name if it exists</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#getRfamID()" class="member-name-link">getRfamID()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getRfamInfo()" class="member-name-link">getRfamInfo()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the different tables from Rfam 2011 data and create a list of sRNAs from the different
 tables</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getRiboSeqDatas()" class="member-name-link">getRiboSeqDatas()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of SeqDatas</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getRiboSeqDatas(java.lang.String)" class="member-name-link">getRiboSeqDatas(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of RiboSeqData given a bioCondName</div>
</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#getRightBCs()" class="member-name-link">getRightBCs()</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getRnaAminoAcidTranslator()" class="member-name-link">getRnaAminoAcidTranslator()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getRnaCompounds()" class="member-name-link">getRnaCompounds()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/RNACompoundSet.html#getRNACompoundSet()" class="member-name-link">getRNACompoundSet()</a> - Static method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/RNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">RNACompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getRNASeqDatas()" class="member-name-link">getRNASeqDatas()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of SeqDatas</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getRNASeqDatas(java.lang.String)" class="member-name-link">getRNASeqDatas(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of RNASeqData given a bioCondName</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getRNASeqGenomes()" class="member-name-link">getRNASeqGenomes()</a> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Get all genomes associated to a transcriptomes dataset</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#getRNASeqs()" class="member-name-link">getRNASeqs()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Return all NGS data contains in all BioCondition</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#getRNASequence()" class="member-name-link">getRNASequence()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Return the RNASequence equivalent of the DNASequence using default Transcription Engine.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.Frame)" class="member-name-link">getRNASequence(Frame)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Allows the user to pass in the Frame shift.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)" class="member-name-link">getRNASequence(TranscriptionEngine)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Allow a user to pass in a rules engine to do the DNA to RNA translation</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine,org.biojava3.core.sequence.transcription.Frame)" class="member-name-link">getRNASequence(TranscriptionEngine, Frame)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getRnaSequenceCreator()" class="member-name-link">getRnaSequenceCreator()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#getRnaToDna()" class="member-name-link">getRnaToDna()</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#getRow(double%5B%5D%5B%5D,int)" class="member-name-link">getRow(double[][], int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#getRow(int%5B%5D%5B%5D,int)" class="member-name-link">getRow(int[][], int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getRow(java.lang.String)" class="member-name-link">getRow(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#getRow(java.lang.String%5B%5D%5B%5D,int)" class="member-name-link">getRow(String[][], int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Extract a Row from a Matrix</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getRowName(int)" class="member-name-link">getRowName(int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getRowNameCellColor()" class="member-name-link">getRowNameCellColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getRowNames()" class="member-name-link">getRowNames()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getRowNamesToList()" class="member-name-link">getRowNamesToList()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getRowNameTextColor()" class="member-name-link">getRowNameTextColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Rscript.html#getRscriptDirectory()" class="member-name-link">getRscriptDirectory()</a> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Rscript.html" title="class in bacnet.scripts.core">Rscript</a></dt>
<dd>
<div class="block">Test if the R script directory has been given, and open a dialog to update it in case</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getSameEndList(java.lang.String,boolean)" class="member-name-link">getSameEndList(String, boolean)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Read list of gene having same end position but different starting point</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#getScore()" class="member-name-link">getScore()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getScore()" class="member-name-link">getScore()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getScrolledComposite()" class="member-name-link">getScrolledComposite()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getScrolledComposite()" class="member-name-link">getScrolledComposite()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getSDEnergy(bacnet.datamodel.sequence.Gene)" class="member-name-link">getSDEnergy(Gene)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Return the binding energy with aSD sequence</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getSDRFCuratedPath(java.lang.String)" class="member-name-link">getSDRFCuratedPath(String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Get path of curated SDRF file for a specific dataID (ex: E-MTAB-1800)</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getSDRFCuratedPath(java.lang.String)" class="member-name-link">getSDRFCuratedPath(String)</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Get path of curated SDRF file for a specific dataID (ex: E-MTAB-1800)</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getSDRFPath(java.lang.String)" class="member-name-link">getSDRFPath(String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Get path of SDRF file for a specific dataID (ex: E-MTAB-1800)</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getSDRFPath(java.lang.String)" class="member-name-link">getSDRFPath(String)</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Get path of SDRF file for a specific dataID (ex: E-MTAB-1800)</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getSDSequence()" class="member-name-link">getSDSequence()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Get the nucleotide sequence 20bp upstream to the <code>Sequence</code>, +3bp for including start
 codon</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getSDSequence(int)" class="member-name-link">getSDSequence(int)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Get the nucleotide sequence n bp upstream to the <code>Sequence</code>, +3bp for including start
 codon</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getSecondRowName()" class="member-name-link">getSecondRowName()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getSelectedComparisons()" class="member-name-link">getSelectedComparisons()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Return the list of comparisons which have been selected</div>
</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getSelectedComparisons()" class="member-name-link">getSelectedComparisons()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>
<div class="block">Return the list of comparisons which have been selected</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getSelectedComparisonsProteome()" class="member-name-link">getSelectedComparisonsProteome()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#getSelectedGenes()" class="member-name-link">getSelectedGenes()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getSeq()" class="member-name-link">getSeq()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getSeqDatas(java.lang.String)" class="member-name-link">getSeqDatas(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of RNASeqData given a bioCondName</div>
</dd>
<dt><a href="../org/biojava3/core/util/SequenceTools.html#getSeqeunceFromString(java.lang.String)" class="member-name-link">getSeqeunceFromString(String)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SequenceTools.html" title="class in org.biojava3.core.util">SequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getSequence()" class="member-name-link">getSequence()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Read chromosome and extract the corresponding sequence as a String</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getSequence()" class="member-name-link">getSequence()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#getSequence()" class="member-name-link">getSequence()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#getSequence()" class="member-name-link">getSequence()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getSequence()" class="member-name-link">getSequence()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SequenceLocation.html#getSequence()" class="member-name-link">getSequence()</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SequenceLocation.html" title="class in org.biojava3.core.sequence.location">SequenceLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaSequenceParser.html#getSequence(java.io.DataInput,int)" class="member-name-link">getSequence(DataInput, int)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaSequenceParser.html" title="class in org.biojava3.core.sequence.io">FastaSequenceParser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/template/SequenceParserInterface.html#getSequence(java.io.DataInput,int)" class="member-name-link">getSequence(DataInput, int)</a> - Method in interface org.biojava3.core.sequence.io.template.<a href="../org/biojava3/core/sequence/io/template/SequenceParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceParserInterface</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getSequence(java.lang.String)" class="member-name-link">getSequence(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>
<div class="block">Given a sequence name, return the corresponding Sequence<br>
 If name is a Gene name, we search also in geneNameToLocusTagMap</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence(String, long)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence(String, long)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">DNASequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence(String, long)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyDNASequenceCreator</a></dt>
<dd>
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence
 and the offset index in the fasta file.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence(String, long)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyProteinSequenceCreator</a></dt>
<dd>
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence
 and the offset index in the fasta file.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence(String, long)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">ProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/RNASequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence(String, long)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/RNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">RNASequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence(String, long)</a> - Method in interface org.biojava3.core.sequence.io.template.<a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html#getSequence(java.util.List)" class="member-name-link">getSequence(List&lt;C&gt;)</a> - Method in interface org.biojava3.core.sequence.io.template.<a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence(List&lt;AminoAcidCompound&gt;)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></dt>
<dd>
<div class="block">Assumes all compounds were uppercase</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence(List&lt;AminoAcidCompound&gt;)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyProteinSequenceCreator</a></dt>
<dd>
<div class="block">Not sure of use case and currently not supported</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence(List&lt;AminoAcidCompound&gt;)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">ProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence(List&lt;NucleotideCompound&gt;)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">DNASequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence(List&lt;NucleotideCompound&gt;)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyDNASequenceCreator</a></dt>
<dd>
<div class="block">Not sure of use case and currently not supported</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/RNASequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence(List&lt;NucleotideCompound&gt;)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/RNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">RNASequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence(ProxySequenceReader&lt;C&gt;, long)</a> - Method in interface org.biojava3.core.sequence.io.template.<a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence(ProxySequenceReader&lt;AminoAcidCompound&gt;, long)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence(ProxySequenceReader&lt;AminoAcidCompound&gt;, long)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyProteinSequenceCreator</a></dt>
<dd>
<div class="block">Should be able to extend the same concept to a remote URL call or database connection.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence(ProxySequenceReader&lt;AminoAcidCompound&gt;, long)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">ProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence(ProxySequenceReader&lt;NucleotideCompound&gt;, long)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">DNASequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence(ProxySequenceReader&lt;NucleotideCompound&gt;, long)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyDNASequenceCreator</a></dt>
<dd>
<div class="block">Should be able to extend the same concept to a remote URL call or database connection.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/RNASequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence(ProxySequenceReader&lt;NucleotideCompound&gt;, long)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/RNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">RNASequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#getSequence5PrimeTo3Prime()" class="member-name-link">getSequence5PrimeTo3Prime()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Try to give method clarity where you want a DNASequence coding in the 5' to 3' direction Returns
 the DNASequence representative of the 5' and 3' reading based on strand</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getSequenceAA()" class="member-name-link">getSequenceAA()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Get Amino Acid sequence of a Sequence</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getSequenceAACurated()" class="member-name-link">getSequenceAACurated()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">When an Amino Acid sequence contains Stop codon ('*'), this method search the first start codon
 and remove sequence before, and then create different sequence, from one stop codon to the other.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getSequenceAACurated(java.lang.String)" class="member-name-link">getSequenceAACurated(String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">When an Amino Acid sequence contains Stop codon ('*'), this method search the first start codon
 and remove sequence before, and then create different sequence, from one stop codon to the other.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getSequenceAsString()" class="member-name-link">getSequenceAsString()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getSequenceAsString()" class="member-name-link">getSequenceAsString()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getSequenceAsString()" class="member-name-link">getSequenceAsString()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getSequenceAsString()" class="member-name-link">getSequenceAsString()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getSequenceAsString()" class="member-name-link">getSequenceAsString()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns the sequence as a String</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getSequenceAsString()" class="member-name-link">getSequenceAsString()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getSequenceAsString()" class="member-name-link">getSequenceAsString()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Delegates to <a href="../org/biojava3/core/sequence/template/SequenceMixin.html#toList(org.biojava3.core.sequence.template.Sequence)"><code>SequenceMixin.toList(org.biojava3.core.sequence.template.Sequence)</code></a></div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceAsString()" class="member-name-link">getSequenceAsString()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Default case is to assume strand is positive because only CDSSequence can be either positive or
 negative Strand.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/Sequence.html#getSequenceAsString()" class="member-name-link">getSequenceAsString()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Returns the String representation of the Sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getSequenceAsString()" class="member-name-link">getSequenceAsString()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html#getSequenceAsString()" class="member-name-link">getSequenceAsString()</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava3.core.sequence.views">ComplementSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/ReversedSequenceView.html#getSequenceAsString()" class="member-name-link">getSequenceAsString()</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views">ReversedSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#getSequenceAsString()" class="member-name-link">getSequenceAsString()</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)" class="member-name-link">getSequenceAsString(Integer, Integer, Strand)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)" class="member-name-link">getSequenceAsString(Integer, Integer, Strand)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)" class="member-name-link">getSequenceAsString(Integer, Integer, Strand)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)" class="member-name-link">getSequenceAsString(Integer, Integer, Strand)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)" class="member-name-link">getSequenceAsString(Integer, Integer, Strand)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html#getSequenceAsString(java.util.List,org.biojava3.core.sequence.template.CompoundSet,java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)" class="member-name-link">getSequenceAsString(List&lt;C&gt;, CompoundSet&lt;C&gt;, Integer, Integer, Strand)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="class in org.biojava3.core.sequence.storage">SequenceAsStringHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html#getSequenceCollection()" class="member-name-link">getSequenceCollection()</a> - Static method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getSequenceMap()" class="member-name-link">getSequenceMap()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getSequencePeptide()" class="member-name-link">getSequencePeptide()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getSequenceRNA()" class="member-name-link">getSequenceRNA()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Return RNA sequence of the <code>Sequence</code></div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceScore()" class="member-name-link">getSequenceScore()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Provide place holder for a metric that indicate a score associated with the sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html#getSequenceUsage()" class="member-name-link">getSequenceUsage()</a> - Static method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#getSerialversionuid()" class="member-name-link">getSerialversionuid()</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getSerialversionuid()" class="member-name-link">getSerialversionuid()</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getSerialversionuid()" class="member-name-link">getSerialversionuid()</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#getSerialversionuid()" class="member-name-link">getSerialversionuid()</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#getSerialversionuid()" class="member-name-link">getSerialversionuid()</a> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getSerialversionuid()" class="member-name-link">getSerialversionuid()</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#getSerialversionuid()" class="member-name-link">getSerialversionuid()</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/NcRNA.html#getSerialversionuid()" class="member-name-link">getSerialversionuid()</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getSerialversionuid()" class="member-name-link">getSerialversionuid()</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#getSerialversionuid()" class="member-name-link">getSerialversionuid()</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#getSerialversionuid()" class="member-name-link">getSerialversionuid()</a> - Static method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getSerialversionuid()" class="member-name-link">getSerialversionuid()</a> - Static method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#getSetNames()" class="member-name-link">getSetNames()</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#getSets()" class="member-name-link">getSets()</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getShell()" class="member-name-link">getShell()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getShortDescription()" class="member-name-link">getShortDescription()</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getShortDescription()" class="member-name-link">getShortDescription()</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Get the short description that can be used to describe the feature</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#getShortName()" class="member-name-link">getShortName()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html#getShortName()" class="member-name-link">getShortName()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#getShortName()" class="member-name-link">getShortName()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Compound.html#getShortName()" class="member-name-link">getShortName()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#getShortName()" class="member-name-link">getShortName()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getShortName()" class="member-name-link">getShortName()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getShowInfo()" class="member-name-link">getShowInfo()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/SignatureSelectionDialog.html#getSignature()" class="member-name-link">getSignature()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/SignatureSelectionDialog.html" title="class in bacnet.expressionAtlas.core">SignatureSelectionDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#getSignatureFromName(java.lang.String)" class="member-name-link">getSignatureFromName(String)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>
<div class="block">Load serialized signature</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getSignatures()" class="member-name-link">getSignatures()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getSIGNATURES_PATH()" class="member-name-link">getSIGNATURES_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#getSignaturesFromTextFiles()" class="member-name-link">getSignaturesFromTextFiles()</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>
<div class="block">Read all signatures in txt format from the folder Database.getSIGNATURES()</div>
</dd>
<dt><a href="../bacnet/Database.html#getSignaturesNametoID()" class="member-name-link">getSignaturesNametoID()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getSignaturesToString()" class="member-name-link">getSignaturesToString()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#getSize()" class="member-name-link">getSize()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#getSize()" class="member-name-link">getSize()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">Get the number of sequences in the MSA</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.html#getSize()" class="member-name-link">getSize()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a></dt>
<dd>
<div class="block">Returns the number of rows in this profile.</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getSliderHbar()" class="member-name-link">getSliderHbar()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getSliderVBar()" class="member-name-link">getSliderVBar()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getSource()" class="member-name-link">getSource()</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">The feature source</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getSource()" class="member-name-link">getSource()</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">The source of the feature.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getSource()" class="member-name-link">getSource()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Added support for the source of this sequence for GFF3 export If a sub sequence doesn't have
 source then check for parent source</div>
</dd>
<dt><a href="../bacnet/Database.html#getSpecies()" class="member-name-link">getSpecies()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#getSpecies()" class="member-name-link">getSpecies()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getSpecies()" class="member-name-link">getSpecies()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getSpecies()" class="member-name-link">getSpecies()</a> - Method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getSpectra()" class="member-name-link">getSpectra()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getSpectrum()" class="member-name-link">getSpectrum()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getsRNAListEGDe()" class="member-name-link">getsRNAListEGDe()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#getSrnaOrder()" class="member-name-link">getSrnaOrder()</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Return an ArrayList containing Srna names in the proper order<br>
 The list is load from PATH_LISTSRNA</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#getsRNAs()" class="member-name-link">getsRNAs()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getsRNAs()" class="member-name-link">getsRNAs()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#getsRNAs()" class="member-name-link">getsRNAs()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return all Srnas in all chromosomes</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html#getStart()" class="member-name-link">getStart()</a> - Method in class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="class in org.biojava3.core.sequence.edits">Edit.AbstractEdit</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getStart()" class="member-name-link">getStart()</a> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.html#getStart()" class="member-name-link">getStart()</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Start of the location</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getStartCode()" class="member-name-link">getStartCode()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getStartCodon()" class="member-name-link">getStartCodon()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getStartCodon(bacnet.datamodel.sequence.Gene)" class="member-name-link">getStartCodon(Gene)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Return start codon proportion in the genome</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermUtils.html#getStartCodonBefore()" class="member-name-link">getStartCodonBefore()</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>
<div class="block">NOT FINISHED <br>
 <br>
 Search for the different start codon before a peptide, and the stop codon after this peptide<br>
 Print the nucleotide sequence plus the codon, and identify Start, STOp and the reference Peptide
 used</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getStartCodonSequence()" class="member-name-link">getStartCodonSequence()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getStat()" class="member-name-link">getStat()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Sysout some statistics about the number of aTIS, dTISM, dTIS, and uTIS</div>
</dd>
<dt><a href="../bacnet/Database.html#getSTAT_MATRIX_PROTEOMES_PATH()" class="member-name-link">getSTAT_MATRIX_PROTEOMES_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Proteomes matrix data showing Fold Change values</div>
</dd>
<dt><a href="../bacnet/Database.html#getSTAT_MATRIX_TRANSCRIPTOMES_PATH()" class="member-name-link">getSTAT_MATRIX_TRANSCRIPTOMES_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Transcriptomes matrix data showing statistical p-values</div>
</dd>
<dt><a href="../bacnet/Database.html#getStatTable()" class="member-name-link">getStatTable()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getStopCodonSequence()" class="member-name-link">getStopCodonSequence()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#getStrand()" class="member-name-link">getStrand()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getStrand()" class="member-name-link">getStrand()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#getStrand()" class="member-name-link">getStrand()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/CDSSequence.html#getStrand()" class="member-name-link">getStrand()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#getStrand()" class="member-name-link">getStrand()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">A gene should have Strand</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getStrand()" class="member-name-link">getStrand()</a> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.html#getStrand()" class="member-name-link">getStrand()</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Strand which the location is located on</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getStrand()" class="member-name-link">getStrand()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getStrand(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getStrand(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getStrandHeaders(boolean)" class="member-name-link">getStrandHeaders(boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Return a list of the data contain in this matrix which correspond to a given strand</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#getStringForCompound(C)" class="member-name-link">getStringForCompound(C)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundSet.html#getStringForCompound(C)" class="member-name-link">getStringForCompound(C)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#getStringForCompound(org.biojava3.core.sequence.compound.AminoAcidCompound)" class="member-name-link">getStringForCompound(AminoAcidCompound)</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/Strand.html#getStringRepresentation()" class="member-name-link">getStringRepresentation()</a> - Method in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getStudyName()" class="member-name-link">getStudyName()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getSubCodingSequencesList(java.util.ArrayList,boolean)" class="member-name-link">getSubCodingSequencesList(ArrayList&lt;String&gt;, boolean)</a> - Method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Put in a sublist all the CodingSequence found in locuTag list</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getSubLocations()" class="member-name-link">getSubLocations()</a> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.html#getSubLocations()" class="member-name-link">getSubLocations()</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Gives access to the sub locations for this location.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getSubMatrix(java.util.ArrayList,java.util.ArrayList)" class="member-name-link">getSubMatrix(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getSubMatrixColumn(java.util.ArrayList)" class="member-name-link">getSubMatrixColumn(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Extract a matrix with only selected columns</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getSubMatrixRow(java.util.ArrayList)" class="member-name-link">getSubMatrixRow(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Provide a list of rowName and extract them from the ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getSubMatrixRow(java.util.TreeSet)" class="member-name-link">getSubMatrixRow(TreeSet&lt;String&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Provide a list of rowName and extract them from the ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getSubMatrixRowRandom(double)" class="member-name-link">getSubMatrixRowRandom(double)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Generate a subMatrix with random rows</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getSubRow(java.lang.String,java.util.ArrayList)" class="member-name-link">getSubRow(String, ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getSubSequence(int,int)" class="member-name-link">getSubSequence(int, int)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns a sub sequence view</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getSubSequence(java.lang.Integer,java.lang.Integer)" class="member-name-link">getSubSequence(Integer, Integer)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getSubSequence(java.lang.Integer,java.lang.Integer)" class="member-name-link">getSubSequence(Integer, Integer)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getSubSequence(java.lang.Integer,java.lang.Integer)" class="member-name-link">getSubSequence(Integer, Integer)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getSubSequence(java.lang.Integer,java.lang.Integer)" class="member-name-link">getSubSequence(Integer, Integer)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getSubSequence(java.lang.Integer,java.lang.Integer)" class="member-name-link">getSubSequence(Integer, Integer)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getSubSequence(java.lang.Integer,java.lang.Integer)" class="member-name-link">getSubSequence(Integer, Integer)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getSubSequence(java.lang.Integer,java.lang.Integer)" class="member-name-link">getSubSequence(Integer, Integer)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Creates a <a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template"><code>SequenceProxyView</code></a> for the given coordinates</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getSubSequence(java.lang.Integer,java.lang.Integer)" class="member-name-link">getSubSequence(Integer, Integer)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Sequence.html#getSubSequence(java.lang.Integer,java.lang.Integer)" class="member-name-link">getSubSequence(Integer, Integer)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Returns a portion of the sequence from the different positions.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getSubSequence(java.lang.Integer,java.lang.Integer)" class="member-name-link">getSubSequence(Integer, Integer)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getSubSequence(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">getSubSequence(Sequence&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>
<div class="block">If circular this will return the sequence represented by the sub locations joined.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.html#getSubSequence(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">getSubSequence(Sequence&lt;C&gt;)</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Will return a SequenceReader object which represents the outer bounds of this Location</div>
</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#getSubsetNames()" class="member-name-link">getSubsetNames()</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#getSubsets()" class="member-name-link">getSubsets()</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermSummaryTable.html#getSummaryTables(bacnet.datamodel.dataset.NTermData,java.lang.String)" class="member-name-link">getSummaryTables(NTermData, String)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermSummaryTable.html" title="class in bacnet.datamodel.proteomics">NTermSummaryTable</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getSynonym()" class="member-name-link">getSynonym()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getSynonymsText()" class="member-name-link">getSynonymsText()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getTabFolder()" class="member-name-link">getTabFolder()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getTabFolder()" class="member-name-link">getTabFolder()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getTabFolder()" class="member-name-link">getTabFolder()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#getTabFolder()" class="member-name-link">getTabFolder()</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getTable()" class="member-name-link">getTable()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getTable()" class="member-name-link">getTable()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.html#getTable(java.lang.Integer)" class="member-name-link">getTable(Integer)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></dt>
<dd>
<div class="block">Returns a table by its identifier i.e. 1 means universal codon tables</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.html#getTable(java.lang.String)" class="member-name-link">getTable(String)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></dt>
<dd>
<div class="block">Returns a table by its name</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getTableBioCondition()" class="member-name-link">getTableBioCondition()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#getTableBioCondition()" class="member-name-link">getTableBioCondition()</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#getTableBioCondition()" class="member-name-link">getTableBioCondition()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getTableComposite()" class="member-name-link">getTableComposite()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#getTableComposite()" class="member-name-link">getTableComposite()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#getTableComposite()" class="member-name-link">getTableComposite()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#getTableComposite()" class="member-name-link">getTableComposite()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/TableSWTView.html#getTableComposite()" class="member-name-link">getTableComposite()</a> - Method in class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#getTableElementName()" class="member-name-link">getTableElementName()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#getTableElementName()" class="member-name-link">getTableElementName()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getTableGenes()" class="member-name-link">getTableGenes()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#getTableGenes()" class="member-name-link">getTableGenes()</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getTableNodiff()" class="member-name-link">getTableNodiff()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getTableNodiffProteome()" class="member-name-link">getTableNodiffProteome()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getTableOtherInfo()" class="member-name-link">getTableOtherInfo()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getTableOver()" class="member-name-link">getTableOver()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getTableOverProteome()" class="member-name-link">getTableOverProteome()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getTableProteomes()" class="member-name-link">getTableProteomes()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.html#getTables()" class="member-name-link">getTables()</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></dt>
<dd>
<div class="block">Returns a list of all available IUPAC tables</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getTableSDRF(java.lang.String)" class="member-name-link">getTableSDRF(String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Return table found in DATA_TABLE</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getTableSDRF(java.lang.String)" class="member-name-link">getTableSDRF(String)</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Return table found in DATA_TABLE</div>
</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getTableSrnas()" class="member-name-link">getTableSrnas()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getTableTranscriptomes()" class="member-name-link">getTableTranscriptomes()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getTableUnder()" class="member-name-link">getTableUnder()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getTableUnderProteome()" class="member-name-link">getTableUnderProteome()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html#getTargetSequence(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">getTargetSequence(Sequence&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="class in org.biojava3.core.sequence.edits">Edit.AbstractEdit</a></dt>
<dd>
<div class="block">Returns the Sequence which is our edit.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getTaxonomy()" class="member-name-link">getTaxonomy()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#getTbtmGraphVisualization()" class="member-name-link">getTbtmGraphVisualization()</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getTbtmHeatmap()" class="member-name-link">getTbtmHeatmap()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getTbtmScatterPlot()" class="member-name-link">getTbtmScatterPlot()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#getTechnoName(java.lang.String)" class="member-name-link">getTechnoName(String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Given an ArrayDesgin ID return the full name of the microarray</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#getTechnoName(java.lang.String)" class="member-name-link">getTechnoName(String)</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Given an ArrayDesgin ID return the full name of the microarray</div>
</dd>
<dt><a href="../bacnet/Database.html#getTEMP_PATH()" class="member-name-link">getTEMP_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getTemperature()" class="member-name-link">getTemperature()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getTermSeqDatas()" class="member-name-link">getTermSeqDatas()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of SeqDatas</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getTermSeqDatas(java.lang.String)" class="member-name-link">getTermSeqDatas(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of Term given a bioCondName</div>
</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#getTests()" class="member-name-link">getTests()</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/DownloadServiceHandler.html#getText()" class="member-name-link">getText()</a> - Method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getTextColor()" class="member-name-link">getTextColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getTextColor()" class="member-name-link">getTextColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getTextColorSWT()" class="member-name-link">getTextColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getTextColorSWT()" class="member-name-link">getTextColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#getTextCutOff()" class="member-name-link">getTextCutOff()</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getTextHeaderColor()" class="member-name-link">getTextHeaderColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getTextHeaderColorSWT()" class="member-name-link">getTextHeaderColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getTextHeaderColorSWT()" class="member-name-link">getTextHeaderColorSWT()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#getTextSearch()" class="member-name-link">getTextSearch()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getTextSearch()" class="member-name-link">getTextSearch()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/ConcurrencyTools.html#getThreadPool()" class="member-name-link">getThreadPool()</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Returns current shared thread pool.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#getThree()" class="member-name-link">getThree()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#getThree()" class="member-name-link">getThree()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getThree()" class="member-name-link">getThree()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getThreshold()" class="member-name-link">getThreshold()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#getTilingData(java.util.ArrayList)" class="member-name-link">getTilingData(ArrayList&lt;String&gt;)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">List all TilingData available by biocondName.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#getTilingData(java.util.ArrayList,java.util.ArrayList)" class="member-name-link">getTilingData(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Load all Tiling present in leftBCs and rightBCs.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/EGDeWTdata.html#getTilings()" class="member-name-link">getTilings()</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/EGDeWTdata.html" title="class in bacnet.datamodel.dataset">EGDeWTdata</a></dt>
<dd>
<div class="block">Read all Tiling data corresponding to NAME_Mean, and return an ArrayList of Tiling</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getTilings()" class="member-name-link">getTilings()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#getTilings()" class="member-name-link">getTilings()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Return all Tiling data contains in all BioCondition</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getTilings()" class="member-name-link">getTilings()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of Tilings</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getTilings(java.lang.String)" class="member-name-link">getTilings(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of Tiling given a bioCondName</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getTime()" class="member-name-link">getTime()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getTIRStructureEnergy(bacnet.datamodel.sequence.Gene)" class="member-name-link">getTIRStructureEnergy(Gene)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">For very gene read TIRStructure.excel and extract the mean value of energy: We calculate the 2nd
 structure free energy and save it a table<br>
 And we calculate the energy for 40 different sequence: -20+40 to -20+80 and calculate the average
 energy</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getTisList()" class="member-name-link">getTisList()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getTisMap()" class="member-name-link">getTisMap()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getTISName()" class="member-name-link">getTISName()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getTIStoDisplay(java.util.ArrayList)" class="member-name-link">getTIStoDisplay(ArrayList&lt;NTerm&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Return the list of from a List of NTerm</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#getTitleFont()" class="member-name-link">getTitleFont()</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Get the TITLE Font with appropriate height and style</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#getTitleFont(int)" class="member-name-link">getTitleFont(int)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Get the TITLE Font with appropriate height and style</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#getToCompoundSet()" class="member-name-link">getToCompoundSet()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getToledoInfo()" class="member-name-link">getToledoInfo()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the different tables from Toledo-Arana publication and create a list of sRNAs from the
 different tables</div>
</dd>
<dt><a href="../bacnet/e4/rap/BannerView.html#getToolBar()" class="member-name-link">getToolBar()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BannerView.html" title="class in bacnet.e4.rap">BannerView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html#getTotal()" class="member-name-link">getTotal()</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html" title="class in bacnet.scripts.core.vennDiagram">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/CombinationGenerator.html#getTotal()" class="member-name-link">getTotal()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/CombinationGenerator.html" title="class in bacnet.utils">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#getTotalBP()" class="member-name-link">getTotalBP()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getTotalColumnSize(java.util.ArrayList)" class="member-name-link">getTotalColumnSize(ArrayList&lt;ExpressionMatrix&gt;)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Return the number of column of the first ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getTotalRowSize(java.util.ArrayList)" class="member-name-link">getTotalRowSize(ArrayList&lt;ExpressionMatrix&gt;)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Count the total Number of row if we concatenate all ExpressionMatrix in the ArrayList</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#getTotalSpectra()" class="member-name-link">getTotalSpectra()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getTotalSpectrum(bacnet.datamodel.sequence.Gene)" class="member-name-link">getTotalSpectrum(Gene)</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">For each gene add the number of spectra of every peptide available (not only aTIS peptides)</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getTrack()" class="member-name-link">getTrack()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html#getTrack()" class="member-name-link">getTrack()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddProteomicsDataDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html#getTrack()" class="member-name-link">getTrack()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddTranscriptomicsDataDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getTrack()" class="member-name-link">getTrack()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#getTrack()" class="member-name-link">getTrack()</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getTrack()" class="member-name-link">getTrack()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getTrack()" class="member-name-link">getTrack()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getTrack()" class="member-name-link">getTrack()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getTrackComposite()" class="member-name-link">getTrackComposite()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getTrackGenome()" class="member-name-link">getTrackGenome()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#getTracksComposite()" class="member-name-link">getTracksComposite()</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#getTranscript(java.lang.String)" class="member-name-link">getTranscript(String)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Get the transcript sequence by accession</div>
</dd>
<dt><a href="../bacnet/scripts/database/DataValidation.html#getTranscriptomes()" class="member-name-link">getTranscriptomes()</a> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermTranscriptome.html#getTranscriptomes()" class="member-name-link">getTranscriptomes()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermTranscriptome.html" title="class in bacnet.scripts.listeriomics.nterm">NTermTranscriptome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getTRANSCRIPTOMES_PATH()" class="member-name-link">getTRANSCRIPTOMES_PATH()</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getTranscriptomesArrayPath()" class="member-name-link">getTranscriptomesArrayPath()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#getTranscriptomesComparisonsArrayPath()" class="member-name-link">getTranscriptomesComparisonsArrayPath()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getTranscriptomesData()" class="member-name-link">getTranscriptomesData()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Put all TranscriptomeData in a list</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getTranscriptomesGenomes()" class="member-name-link">getTranscriptomesGenomes()</a> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Get all genomes associated to a transcriptomes dataset</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#getTranscripts()" class="member-name-link">getTranscripts()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Get the collection of transcription sequences assigned to this gene</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getTranstable(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getTranstable(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getTriplet()" class="member-name-link">getTriplet()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#getTriplicateValue(java.lang.String)" class="member-name-link">getTriplicateValue(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Get all technical replicates values of corresponding probe in double[][] values</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getTSSDatas()" class="member-name-link">getTSSDatas()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of SeqDatas</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#getTSSDatas(java.lang.String)" class="member-name-link">getTSSDatas(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of TSS given a bioCondName</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getTssDownCoverage()" class="member-name-link">getTssDownCoverage()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getTssDownDistance()" class="member-name-link">getTssDownDistance()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getTssDowntype()" class="member-name-link">getTssDowntype()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getTssUpCoverage()" class="member-name-link">getTssUpCoverage()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getTssUpDistance()" class="member-name-link">getTssUpDistance()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getTssUptype()" class="member-name-link">getTssUptype()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#getTwo()" class="member-name-link">getTwo()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#getTwo()" class="member-name-link">getTwo()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getTwo()" class="member-name-link">getTwo()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getTxtCutoffLogFC()" class="member-name-link">getTxtCutoffLogFC()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getTxtCutoffLogFC()" class="member-name-link">getTxtCutoffLogFC()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getTxtCutoffLogFCProteome()" class="member-name-link">getTxtCutoffLogFCProteome()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getTxtCutoffPvalue()" class="member-name-link">getTxtCutoffPvalue()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#getTxtSearch()" class="member-name-link">getTxtSearch()</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#getTxtSearch()" class="member-name-link">getTxtSearch()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#getTxtSearch()" class="member-name-link">getTxtSearch()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getTxtSeqnucleotide()" class="member-name-link">getTxtSeqnucleotide()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#getType()" class="member-name-link">getType()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#getType()" class="member-name-link">getType()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#getType()" class="member-name-link">getType()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#getType()" class="member-name-link">getType()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.html#getType()" class="member-name-link">getType()</a> - Method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.html" title="class in bacnet.scripts.core.stat">StatTest</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getType()" class="member-name-link">getType()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>
<div class="block">GETTERS and SETTERS</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getType()" class="member-name-link">getType()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>
<div class="block">GETTERS and SETTERS</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getType()" class="member-name-link">getType()</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">The feature type</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getType()" class="member-name-link">getType()</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">The type of the feature.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getType(org.biojava3.core.sequence.DNASequence)" class="member-name-link">getType(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#getTypeColumn()" class="member-name-link">getTypeColumn()</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return the column index corresponding to Type</div>
</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#getTypeData()" class="member-name-link">getTypeData()</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#getTypeDataContained()" class="member-name-link">getTypeDataContained()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#getTypeFilter()" class="member-name-link">getTypeFilter()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#getTypeFilter()" class="member-name-link">getTypeFilter()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getTypeModif()" class="member-name-link">getTypeModif()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/NcRNA.html#getTypeNcRNA()" class="member-name-link">getTypeNcRNA()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getTypeOfTIS(java.lang.String)" class="member-name-link">getTypeOfTIS(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Return a HashMap of <code>NTerm</code> with <code>nTerm.getTypeOverlap() == typeOverlap</code></div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#getTypeOverlap()" class="member-name-link">getTypeOverlap()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#getTypeOverlaps()" class="member-name-link">getTypeOverlaps()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#getTypeOverlaps()" class="member-name-link">getTypeOverlaps()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#getTypeSrna()" class="member-name-link">getTypeSrna()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getUnderASrnas()" class="member-name-link">getUnderASrnas()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#getUnderBioConds()" class="member-name-link">getUnderBioConds()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getUnderCisRegs()" class="member-name-link">getUnderCisRegs()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getUnderExpressed()" class="member-name-link">getUnderExpressed()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getUnderGenes()" class="member-name-link">getUnderGenes()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#getUnderlyingCompound()" class="member-name-link">getUnderlyingCompound()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getUnderSrnas()" class="member-name-link">getUnderSrnas()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getUniprotbaseURL()" class="member-name-link">getUniprotbaseURL()</a> - Static method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">The current unirpot URL to deal with caching issues. www.uniprot.org is loaded balanced but you
 can access pir.uniprot.org directly.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getUniprotDirectoryCache()" class="member-name-link">getUniprotDirectoryCache()</a> - Static method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">Local directory cache of XML that can be downloaded</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getUnknownRAST(java.lang.String,java.lang.String)" class="member-name-link">getUnknownRAST(String, String)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Read the list of RAST elements totally unknown and create a list of Sequence from it</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#getUnlogValue(int,java.lang.String)" class="member-name-link">getUnlogValue(int, String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>
<div class="block">Get value at a certain base pair position and unlog it by calculating 2^value<br>
 It works faster if data has been already loaded by this.read() or this.read(from,to)</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getUpDNAList()" class="member-name-link">getUpDNAList()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#getUpperedBase()" class="member-name-link">getUpperedBase()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getUpProtList()" class="member-name-link">getUpProtList()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getUpRNAList()" class="member-name-link">getUpRNAList()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#getURL()" class="member-name-link">getURL()</a> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>
<div class="block">Return the current URL of your RAP project</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getUserCollection()" class="member-name-link">getUserCollection()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Provided for convience if the developer needs to associate data with a sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getUserObject()" class="member-name-link">getUserObject()</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getUserObject()" class="member-name-link">getUserObject()</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#getUTRLength()" class="member-name-link">getUTRLength()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getUTRLength(bacnet.datamodel.sequence.Gene)" class="member-name-link">getUTRLength(Gene)</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
</dd>
<dt><a href="../org/biojava3/core/util/CRC64Checksum.html#getValue()" class="member-name-link">getValue()</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/CRC64Checksum.html" title="class in org.biojava3.core.util">CRC64Checksum</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getValue(int)" class="member-name-link">getValue(int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Return the value at the right position in the double[] values argument<br>
 Read the data if necessary</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getValue(int,int)" class="member-name-link">getValue(int, int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getValue(java.lang.String,java.lang.String)" class="member-name-link">getValue(String, String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Get value in the ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getValueAnnotation(int,int)" class="member-name-link">getValueAnnotation(int, int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getValueAnnotation(java.lang.String,java.lang.String)" class="member-name-link">getValueAnnotation(String, String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getValues()" class="member-name-link">getValues()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getValues()" class="member-name-link">getValues()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#getValues()" class="member-name-link">getValues()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getValues()" class="member-name-link">getValues()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getValues()" class="member-name-link">getValues()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/CodonUsage.html#getValues(java.lang.String)" class="member-name-link">getValues(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/CodonUsage.html" title="class in bacnet.datamodel.annotation">CodonUsage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#getVar()" class="member-name-link">getVar()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#getVar()" class="member-name-link">getVar()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#getVariance()" class="member-name-link">getVariance()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#getVarianceValue(java.lang.String)" class="member-name-link">getVarianceValue(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Get variance value of corresponding probe in double[][] values</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Network.html#getVertices()" class="member-name-link">getVertices()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>
<div class="block">List of vertices = genome elements + their attributes = genome position, type, annotation</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getViewedSequence()" class="member-name-link">getViewedSequence()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceView.html#getViewedSequence()" class="member-name-link">getViewedSequence()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getViewId()" class="member-name-link">getViewId()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#getViewId()" class="member-name-link">getViewId()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#getViewId()" class="member-name-link">getViewId()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#getViewId()" class="member-name-link">getViewId()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#getViewId()" class="member-name-link">getViewId()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#getViewID()" class="member-name-link">getViewID()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#getViewID()" class="member-name-link">getViewID()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getVoidMatrix()" class="member-name-link">getVoidMatrix()</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Create a matrix containing only one element = MISSING_VALUE</div>
</dd>
<dt><a href="../bacnet/Database.html#getWebpageTitle()" class="member-name-link">getWebpageTitle()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getWGS(java.util.ArrayList)" class="member-name-link">getWGS(ArrayList&lt;String&gt;)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Extract all WGS (Whole Genome Shotgun) from genomesList</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#getWidth()" class="member-name-link">getWidth()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/MatrixSize.html#getWidth()" class="member-name-link">getWidth()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getWidthBP()" class="member-name-link">getWidthBP()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getWidthBP()" class="member-name-link">getWidthBP()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/MatrixSize.html#getWidthDefault()" class="member-name-link">getWidthDefault()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/MatrixSize.html#getWidthRowNameDefault()" class="member-name-link">getWidthRowNameDefault()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/WindowedSequence.html#getWindowSize()" class="member-name-link">getWindowSize()</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence</a></dt>
<dd>
<div class="block">Access the current window size</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getWurtzelInfo()" class="member-name-link">getWurtzelInfo()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the different tables from Wurtezel, Sesto et al. 2012 publication and create a list of sRNAs
 from the different tables</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#getX1()" class="member-name-link">getX1()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#getX2()" class="member-name-link">getX2()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#getY1()" class="member-name-link">getY1()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#getY2()" class="member-name-link">getY2()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#getZoom()" class="member-name-link">getZoom()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#getZoomHeightMax()" class="member-name-link">getZoomHeightMax()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>
<div class="block">Maximal value for the zoom</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#getZoomHeightPosition()" class="member-name-link">getZoomHeightPosition()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>
<div class="block">Position of the zoom for height</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#getZoomNumber()" class="member-name-link">getZoomNumber()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#getZoomPosition()" class="member-name-link">getZoomPosition()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#getZoomRatio()" class="member-name-link">getZoomRatio()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getZoomVertical()" class="member-name-link">getZoomVertical()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/GFFNCBIReader.html#GFF_ATTRIBUTES" class="member-name-link">GFF_ATTRIBUTES</a> - Static variable in class bacnet.reader.<a href="../bacnet/reader/GFFNCBIReader.html" title="class in bacnet.reader">GFFNCBIReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GFF3Reader.html" class="type-name-link" title="class in org.biojava3.genome.parsers.gff">GFF3Reader</a> - Class in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>
<div class="block">http://www.bioperl.org/wiki/GTF Read and write FeatureLists as GFF/GTF formatted files.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GFF3Reader.html#%3Cinit%3E()" class="member-name-link">GFF3Reader()</a> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GFF3Reader.html" title="class in org.biojava3.genome.parsers.gff">GFF3Reader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GFF3Writer.html" class="type-name-link" title="class in org.biojava3.genome.parsers.gff">GFF3Writer</a> - Class in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GFF3Writer.html#%3Cinit%3E()" class="member-name-link">GFF3Writer()</a> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GFF3Writer.html" title="class in org.biojava3.genome.parsers.gff">GFF3Writer</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/GFFNCBIReader.html" class="type-name-link" title="class in bacnet.reader">GFFNCBIReader</a> - Class in <a href="../bacnet/reader/package-summary.html">bacnet.reader</a></dt>
<dd>
<div class="block">Read GFF files to create a <a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence"><code>Genome</code></a> object</div>
</dd>
<dt><a href="../bacnet/reader/GFFNCBIReader.html#%3Cinit%3E()" class="member-name-link">GFFNCBIReader()</a> - Constructor for class bacnet.reader.<a href="../bacnet/reader/GFFNCBIReader.html" title="class in bacnet.reader">GFFNCBIReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/GlaserFCannotation.html" class="type-name-link" title="class in bacnet.datamodel.annotation">GlaserFCannotation</a> - Class in <a href="../bacnet/datamodel/annotation/package-summary.html">bacnet.datamodel.annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/GlaserFCannotation.html#%3Cinit%3E()" class="member-name-link">GlaserFCannotation()</a> - Constructor for class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/GlaserFCannotation.html" title="class in bacnet.datamodel.annotation">GlaserFCannotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html#GluC_Actino_Insoluble" class="member-name-link">GluC_Actino_Insoluble</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html" title="enum in bacnet.datamodel.proteomics">NTerm.ExperimentName</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html#GluC_Actino_Soluble" class="member-name-link">GluC_Actino_Soluble</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html" title="enum in bacnet.datamodel.proteomics">NTerm.ExperimentName</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#gradientColor(int)" class="member-name-link">gradientColor(int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SynTView.html#GRAPH_NAME" class="member-name-link">GRAPH_NAME</a> - Static variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#GRAPH_NAME" class="member-name-link">GRAPH_NAME</a> - Static variable in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#GREEN" class="member-name-link">GREEN</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#GREY" class="member-name-link">GREY</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/normalization/GRNormalization.html" class="type-name-link" title="class in bacnet.scripts.core.normalization">GRNormalization</a> - Class in <a href="../bacnet/scripts/core/normalization/package-summary.html">bacnet.scripts.core.normalization</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/normalization/GRNormalization.html#%3Cinit%3E()" class="member-name-link">GRNormalization()</a> - Constructor for class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/GRNormalization.html" title="class in bacnet.scripts.core.normalization">GRNormalization</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#group()" class="member-name-link">group()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Get the first item (everything before first semicolon, if it has one) in the attribute field,
 which is assumed to be a group identifer.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureI.html#group()" class="member-name-link">group()</a> - Method in interface org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a></dt>
<dd>
<div class="block">Get the group id of the feature.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#%3Cinit%3E(int,int,org.biojava3.core.sequence.Strand,java.util.List)" class="member-name-link">GroupLocation(int, int, Strand, List&lt;Location&gt;)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#%3Cinit%3E(int,int,org.biojava3.core.sequence.Strand,org.biojava3.core.sequence.location.template.Location...)" class="member-name-link">GroupLocation(int, int, Strand, Location...)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,java.util.List)" class="member-name-link">GroupLocation(Point, Point, Strand, boolean, List&lt;Location&gt;)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,org.biojava3.core.sequence.location.template.Location...)" class="member-name-link">GroupLocation(Point, Point, Strand, boolean, Location...)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,java.util.List)" class="member-name-link">GroupLocation(Point, Point, Strand, List&lt;Location&gt;)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,org.biojava3.core.sequence.location.template.Location...)" class="member-name-link">GroupLocation(Point, Point, Strand, Location...)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#groupValues()" class="member-name-link">groupValues()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a collection of all unique group ids in the list, as defined by the group() method of the
 features.</div>
</dd>
<dt><a href="../org/biojava3/core/util/InputStreamProvider.html#GZIP_MAGIC" class="member-name-link">GZIP_MAGIC</a> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/InputStreamProvider.html" title="class in org.biojava3.core.util">InputStreamProvider</a></dt>
<dd>
<div class="block">The magic number found at the start of a GZIP stream.</div>
</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
