<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 14:20:51 CEST 2019 -->
<title>G-Index</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="G-Index";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="contentContainer"><a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a><a id="I:G">
<!--   -->
</a>
<h2 class="title">G</h2>
<dl>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastResult.html#gaps">gaps</a></span> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/GBKReader.html" title="class in bacnet.reader"><span class="typeNameLink">GBKReader</span></a> - Class in <a href="../bacnet/reader/package-summary.html">bacnet.reader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/GBKReader.html#%3Cinit%3E()">GBKReader()</a></span> - Constructor for class bacnet.reader.<a href="../bacnet/reader/GBKReader.html" title="class in bacnet.reader">GBKReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html#GCG">GCG</a></span> - org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html" title="enum in org.biojava3.core.sequence.template">LightweightProfile.StringFormat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GCStats.html" title="class in org.biojava3.genome.parsers.gff"><span class="typeNameLink">GCStats</span></a> - Class in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/GCStats.html#%3Cinit%3E()">GCStats()</a></span> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GCStats.html" title="class in org.biojava3.genome.parsers.gff">GCStats</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI"><span class="typeNameLink">GElement</span></a> - Class in <a href="../bacnet/genomeBrowser/tracksGUI/package-summary.html">bacnet.genomeBrowser.tracksGUI</a></dt>
<dd>
<div class="block">Here are all the methods for displaying each Genome element</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#%3Cinit%3E()">GElement()</a></span> - Constructor for class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DataSource.html#GENBANK">GENBANK</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence"><span class="typeNameLink">Gene</span></a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.SeqType.html#Gene">Gene</a></span> - bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#%3Cinit%3E()">Gene()</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#%3Cinit%3E(java.lang.String,int,int,char)">Gene(String, int, int, char)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/NavigationManagement.html#GENE">GENE</a></span> - Static variable in class bacnet.rap.<a href="../bacnet/rap/NavigationManagement.html" title="class in bacnet.rap">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/NavigationManagement.html#GENE">GENE</a></span> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html#GeneExpr">GeneExpr</a></span> - bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset"><span class="typeNameLink">GeneExpression</span></a> - Class in <a href="../bacnet/datamodel/dataset/package-summary.html">bacnet.datamodel.dataset</a></dt>
<dd>
<div class="block">GeneExpression array data from Listeria tiling chip ArrayExpress id :
 A-AFFY-140
 
 Here double[] values vector has length = 5*probes.size because we put 5
 columns column 0-2 = 3 replicatas column 3 = median expression column 4 = var
 expression</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#%3Cinit%3E()">GeneExpression()</a></span> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#%3Cinit%3E(java.lang.String)">GeneExpression(String)</a></span> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome"><span class="typeNameLink">GeneFeatureHelper</span></a> - Class in <a href="../org/biojava3/genome/package-summary.html">org.biojava3.genome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/GeneFeatureHelper.html#%3Cinit%3E()">GeneFeatureHelper()</a></span> - Constructor for class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html" title="class in org.biojava3.genome.parsers.gff"><span class="typeNameLink">GeneIDGFF2Reader</span></a> - Class in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>
<div class="block">http://www.bioperl.org/wiki/GTF Read and write FeatureLists as GFF/GTF
 formatted files.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html#%3Cinit%3E()">GeneIDGFF2Reader()</a></span> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html" title="class in org.biojava3.genome.parsers.gff">GeneIDGFF2Reader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html" title="class in org.biojava3.genome.parsers.geneid"><span class="typeNameLink">GeneIDXMLReader</span></a> - Class in <a href="../org/biojava3/genome/parsers/geneid/package-summary.html">org.biojava3.genome.parsers.geneid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html#%3Cinit%3E(java.lang.String)">GeneIDXMLReader(String)</a></span> - Constructor for class org.biojava3.genome.parsers.geneid.<a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html" title="class in org.biojava3.genome.parsers.geneid">GeneIDXMLReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html" title="class in org.biojava3.genome.parsers.gff"><span class="typeNameLink">GeneMarkGTFReader</span></a> - Class in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>
<div class="block">http://www.bioperl.org/wiki/GTF Read and write FeatureLists as GFF/GTF
 formatted files.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html#%3Cinit%3E()">GeneMarkGTFReader()</a></span> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html" title="class in org.biojava3.genome.parsers.gff">GeneMarkGTFReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI"><span class="typeNameLink">GeneNCBITools</span></a> - Class in <a href="../bacnet/datamodel/sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></dt>
<dd>
<div class="block">List of tools useful to extract information hidden in the NoteList of a
 DNASequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#%3Cinit%3E()">GeneNCBITools()</a></span> - Constructor for class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DataSource.html#GENERAL">GENERAL</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#GENERAL_WT_DATA">GENERAL_WT_DATA</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Name of the Transcriptome data used as a reference data for comparisons</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#GENERAL_WT_NAME">GENERAL_WT_NAME</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Name used for "median" data created from a set of reference condition</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/GeneralArray.html" title="class in bacnet.scripts.listeriomics.technology"><span class="typeNameLink">GeneralArray</span></a> - Class in <a href="../bacnet/scripts/listeriomics/technology/package-summary.html">bacnet.scripts.listeriomics.technology</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/GeneralArray.html#%3Cinit%3E()">GeneralArray()</a></span> - Constructor for class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/GeneralArray.html" title="class in bacnet.scripts.listeriomics.technology">GeneralArray</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="class in org.biojava3.core.sequence.io"><span class="typeNameLink">GenericFastaHeaderFormat</span></a>&lt;<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="type parameter in GenericFastaHeaderFormat">S</a> extends <a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;?&gt;,&#8203;<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="type parameter in GenericFastaHeaderFormat">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">We store the original header if the sequence is parsed from a fasta file and
 will use that exact sequence if we write out the sequences to a fasta file.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html#%3Cinit%3E()">GenericFastaHeaderFormat()</a></span> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderFormat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="class in org.biojava3.core.sequence.io"><span class="typeNameLink">GenericFastaHeaderParser</span></a>&lt;<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="type parameter in GenericFastaHeaderParser">S</a> extends <a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="type parameter in GenericFastaHeaderParser">C</a>&gt;,&#8203;<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="type parameter in GenericFastaHeaderParser">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">The default fasta header parser where some headers are well defined based on
 the source database which allows us to set the source of the protein sequence
 and the identifier that can be used in future implementations to load
 features from external sources
 
 If the user has a custom header with local data then they can create their
 own implementation of a FastaHeaderParserInterface

 GenBank gi|gi-number|gb|accession|locus ENA Data Library
 gi|gi-number|emb|accession|locus DDBJ, DNA Database of Japan
 gi|gi-number|dbj|accession|locus NBRF PIR pir||entry Protein Research
 Foundation prf||name SWISS-PROT sp|accession|name Brookhaven Protein Data
 Bank (1) pdb|entry|chain Brookhaven Protein Data Bank (2)
 entry:chain|PDBID|CHAIN|SEQUENCE PDB EBI PDB:1ECY_A mol:protein length:142
 ECOTIN Patents pat|country|number GenInfo Backbone Id bbs|number General
 database identifier gnl|database|identifier NCBI Reference Sequence
 ref|accession|locus Local Sequence identifier lcl|identifier</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html#%3Cinit%3E()">GenericFastaHeaderParser()</a></span> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence"><span class="typeNameLink">GeneSequence</span></a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/GeneSequence.html#%3Cinit%3E(org.biojava3.core.sequence.ChromosomeSequence,int,int,org.biojava3.core.sequence.Strand)">GeneSequence(ChromosomeSequence, int, int, Strand)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">A class that keeps track of the details of a GeneSequence which is difficult
 to properly model.</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools"><span class="typeNameLink">GeneView</span></a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#%3Cinit%3E()">GeneView()</a></span> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewHomologTools.html" title="class in bacnet.sequenceTools"><span class="typeNameLink">GeneViewHomologTools</span></a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>
<div class="block">All the tools to display homologs in the Gene Panel view: - It creates the
 homolog table - Create the homolog figure - Display both of them</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneViewHomologTools.html#%3Cinit%3E()">GeneViewHomologTools()</a></span> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneViewHomologTools.html" title="class in bacnet.sequenceTools">GeneViewHomologTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewLocalizationTools.html" title="class in bacnet.sequenceTools"><span class="typeNameLink">GeneViewLocalizationTools</span></a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneViewLocalizationTools.html#%3Cinit%3E()">GeneViewLocalizationTools()</a></span> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneViewLocalizationTools.html" title="class in bacnet.sequenceTools">GeneViewLocalizationTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewProteomeTools.html" title="class in bacnet.sequenceTools"><span class="typeNameLink">GeneViewProteomeTools</span></a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneViewProteomeTools.html#%3Cinit%3E()">GeneViewProteomeTools()</a></span> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneViewProteomeTools.html" title="class in bacnet.sequenceTools">GeneViewProteomeTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewTranscriptomeTools.html" title="class in bacnet.sequenceTools"><span class="typeNameLink">GeneViewTranscriptomeTools</span></a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneViewTranscriptomeTools.html#%3Cinit%3E()">GeneViewTranscriptomeTools()</a></span> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneViewTranscriptomeTools.html" title="class in bacnet.sequenceTools">GeneViewTranscriptomeTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DataSource.html#GENINFO">GENINFO</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#genome">genome</a></span> - Variable in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence"><span class="typeNameLink">Genome</span></a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the
 genome GFF <a href="../bacnet/reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#%3Cinit%3E()">Genome()</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#%3Cinit%3E(java.lang.String,boolean)">Genome(String, boolean)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Create a Genome from .fna, .annot, and all serialized Sequence object:<br>
 Read Genome sequence information in the different fna files <br>
 Load annotation information<br>
 <br>
 As we are dealing with bacteria, we are sure that the chromosome with maximum
 length is THE chromosome, other are plasmids.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/NavigationManagement.html#GENOME">GENOME</a></span> - Static variable in class bacnet.rap.<a href="../bacnet/rap/NavigationManagement.html" title="class in bacnet.rap">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/NavigationManagement.html#GENOME">GENOME</a></span> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html#GENOME_PATH">GENOME_PATH</a></span> - Static variable in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html" title="class in bacnet.scripts.listeriomics.srna">SmallPeptidesSearch</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span></div>
&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/GenomesCreation.html#GENOME_Table">GENOME_Table</a></span> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html" title="class in bacnet.datamodel.sequence"><span class="typeNameLink">Genome.GetMultiFastaThread</span></a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.OpenGenomesThread.html" title="class in bacnet.datamodel.sequence"><span class="typeNameLink">Genome.OpenGenomesThread</span></a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html" title="class in bacnet.datamodel.sequenceNCBI"><span class="typeNameLink">GenomeConversion</span></a> - Class in <a href="../bacnet/datamodel/sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></dt>
<dd>
<div class="block">Static methods for converting GenomeNCBI to Genome<br>
 Genome is used for fast loading of all genomic information we gathered</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html#%3Cinit%3E()">GenomeConversion()</a></span> - Constructor for class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversion</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI"><span class="typeNameLink">GenomeConversionElement</span></a> - Class in <a href="../bacnet/datamodel/sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#%3Cinit%3E()">GenomeConversionElement()</a></span> - Constructor for class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core"><span class="typeNameLink">GenomeElementAtlas</span></a> - Class in <a href="../bacnet/expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#%3Cinit%3E()">GenomeElementAtlas()</a></span> - Constructor for class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#%3Cinit%3E(bacnet.datamodel.sequence.Sequence,bacnet.utils.Filter)">GenomeElementAtlas(Sequence, Filter)</a></span> - Constructor for class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>
<div class="block">For a specific sequence update the list of BioCondition in which the genome
 element is over-expressed, under-expressed or not differently expressed<br>
 The update will be dependent of the value of the input filter = cutoff for
 LogFC and p-value</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html#GenomeElements">GenomeElements</a></span> - bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html" title="enum in bacnet.datamodel.dataset">OmicsData.ColNames</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence"><span class="typeNameLink">GenomeNCBI</span></a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>
<div class="block">Object containing an entire genome, created from NCBI RefSeq data<br>
 Data for this genome come from: <it> NCBI.fna (for the sequence) <it>
 NCBI.gff (for Gene, CDS, and ncRNA lists) If it exists some information are
 added: <it> Rfam.gff (other info) <it> supp info in a .txt file <it> supp
 info about RAST results in a .txt file For EGD-e only: <it> Srna.xml are read
 to add to the genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#%3Cinit%3E()">GenomeNCBI()</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#%3Cinit%3E(java.lang.String,boolean)">GenomeNCBI(String, boolean)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>
<div class="block">Create a genome with all chromosomes found If annotation = TRUE , we had the
 annotation we can found in other fasta files annotation = FALSE, we read only
 the complete genome + eventual plasmids</div>
</dd>
<dt><a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html" title="class in bacnet.scripts.blast"><span class="typeNameLink">GenomeNCBIFolderTools</span></a> - Class in <a href="../bacnet/scripts/blast/package-summary.html">bacnet.scripts.blast</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html#%3Cinit%3E()">GenomeNCBIFolderTools()</a></span> - Constructor for class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html" title="class in bacnet.scripts.blast">GenomeNCBIFolderTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI"><span class="typeNameLink">GenomeNCBITools</span></a> - Class in <a href="../bacnet/datamodel/sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#%3Cinit%3E()">GenomeNCBITools()</a></span> - Constructor for class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html" title="class in bacnet.scripts.phylogeny"><span class="typeNameLink">GenomePhylogeny</span></a> - Class in <a href="../bacnet/scripts/phylogeny/package-summary.html">bacnet.scripts.phylogeny</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html#%3Cinit%3E()">GenomePhylogeny()</a></span> - Constructor for class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html" title="class in bacnet.scripts.phylogeny">GenomePhylogeny</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database"><span class="typeNameLink">GenomesCreation</span></a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>
<div class="block">List of tools available for genome creation</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/GenomesCreation.html#%3Cinit%3E()">GenomesCreation()</a></span> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/GenomesListeriomics.html" title="class in bacnet.scripts.listeriomics"><span class="typeNameLink">GenomesListeriomics</span></a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/GenomesListeriomics.html#%3Cinit%3E()">GenomesListeriomics()</a></span> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/GenomesListeriomics.html" title="class in bacnet.scripts.listeriomics">GenomesListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser"><span class="typeNameLink">GenomeTranscriptomeView</span></a> - Class in <a href="../bacnet/genomeBrowser/package-summary.html">bacnet.genomeBrowser</a></dt>
<dd>
<div class="block">Display on the genome viewer: expression of specific bioCond and the log of
 fold change</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#%3Cinit%3E()">GenomeTranscriptomeView()</a></span> - Constructor for class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools"><span class="typeNameLink">GenomicsView</span></a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GenomicsView.html#%3Cinit%3E()">GenomicsView()</a></span> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/normalization/GENormalization.html" title="class in bacnet.scripts.core.normalization"><span class="typeNameLink">GENormalization</span></a> - Class in <a href="../bacnet/scripts/core/normalization/package-summary.html">bacnet.scripts.core.normalization</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/GENormalization.html#%3Cinit%3E()">GENormalization()</a></span> - Constructor for class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/GENormalization.html" title="class in bacnet.scripts.core.normalization">GENormalization</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#get(int)">get(int)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/WindowedSequence.html#get(int)">get(int)</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence</a></dt>
<dd>
<div class="block">Returns the window specified at the given index in offsets i.e. asking for
 position 2 in a moving window sequence of size 3 will get you the window
 starting at position 4.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#get(int,int,boolean)">get(int, int, boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">read probe values between two positions in the genome <br>
 important: bpbegin < bpend</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/SoftHashMap.html#get(java.lang.Object)">get(Object)</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#get10403SOverlap()">get10403SOverlap()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#get10403SSrnas()">get10403SSrnas()</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Return a list of all Srna in 10403S</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#get3UTRSequence(int)">get3UTRSequence(int)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Get the nucleotide sequence n bp downstream to the <code>Sequence</code>,
 +3bp for including start codon</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/CompoundColorization.html#getAAcidColor(java.lang.String,org.eclipse.swt.widgets.Display)">getAAcidColor(String, Display)</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/CompoundColorization.html" title="class in bacnet.datamodel.sequence">CompoundColorization</a></dt>
<dd>
<div class="block">Given an Amino Acid, this method return the corresponding color</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getAccession()">getAccession()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getAccession()">getAccession()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getAccession()">getAccession()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getAccession()">getAccession()</a></span> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getAccession()">getAccession()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getAccession()">getAccession()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getAccession()">getAccession()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getAccession()">getAccession()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Unsupoorted</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getAccession()">getAccession()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Accessioned.html#getAccession()">getAccession()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</a></dt>
<dd>
<div class="block">Returns the AccessionID this location is currently bound with</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getAccession()">getAccession()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#getAlias()">getAlias()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#getAlignedSequence(int)">getAlignedSequence(int)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">Uses bioIndex starting at 1 instead of 0</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/LightweightProfile.html#getAlignedSequence(int)">getAlignedSequence(int)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a></dt>
<dd>
<div class="block">Returns <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a> at given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#getAlignedSequences()">getAlignedSequences()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">Get the list of sequences</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/LightweightProfile.html#getAlignedSequences()">getAlignedSequences()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a></dt>
<dd>
<div class="block">Returns a <code>List</code> containing the individual <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a>s of this
 alignment.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastResult.html#getAllAccession(java.util.ArrayList)">getAllAccession(ArrayList&lt;BlastResult&gt;)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>
<div class="block">Return a list of all accessions contain in this list of results</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getAllBioConditionNames()">getAllBioConditionNames()</a></span> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Get all available <code>BioCondition</code> names by reading the text file
 <code>Experiment.INCLUDE_COND_PATH</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getAllBioConditions()">getAllBioConditions()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getAllBioConditions()">getAllBioConditions()</a></span> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Return all BioCondition available</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getAllButtons()">getAllButtons()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#getAllButtons()">getAllButtons()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getAllButtons()">getAllButtons()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Codon.html#getAllCodons()">getAllCodons()</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>
<div class="block">Return a TreeSet containing all possible codons</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#getAllCompounds()">getAllCompounds()</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#getAllCompounds()">getAllCompounds()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/CompoundSet.html#getAllCompounds()">getAllCompounds()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#getAllElementNames()">getAllElementNames()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return all element names contained in all chromosomes</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getAllElements()">getAllElements()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#getAllElements()">getAllElements()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return all elements contained in all chromosomes</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Frame.html#getAllFrames()">getAllFrames()</a></span> - Static method in enum org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></dt>
<dd>
<div class="block">Delegates to <a href="../org/biojava3/core/sequence/transcription/Frame.html#values()"><code>Frame.values()</code></a></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#getAlllocusTag()">getAlllocusTag()</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#getAlllocusTagFromGFF()">getAlllocusTagFromGFF()</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>
<div class="block">They used old_locus_tag for the conservation analysis, so we need to create a
 HashMap to have the equivalent GeneId old_locus_tag ---> locus_tag in the GFF</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#getAlllocusTagINRA()">getAlllocusTagINRA()</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/ProteinTools.html#getAllPeptides(java.lang.String,java.util.ArrayList,java.util.ArrayList,char,java.lang.String,int,boolean,int)">getAllPeptides(String, ArrayList&lt;Integer&gt;, ArrayList&lt;String&gt;, char, String, int, boolean, int)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Given a genome sequence, and a list of position of start codon, update a list
 of peptides accordingly<br>
 The new peptide, are the peptide from a start codon to the next R or stop
 codon</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#getAllPeptides(java.lang.String,java.util.ArrayList,java.util.ArrayList,char,java.lang.String,int,boolean,int)">getAllPeptides(String, ArrayList&lt;Integer&gt;, ArrayList&lt;String&gt;, char, String, int, boolean, int)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Given a genome sequence, and a list of position of start codon, update a list
 of peptides accordingly<br>
 The new peptide, are the peptide from a start codon to the next R or stop
 codon</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getAllPublishedBioConditions()">getAllPublishedBioConditions()</a></span> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Return all BioCondition available</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#getAlltechnology()">getAlltechnology()</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Go through all SDRF files and read the "Array Design REF" entry, save in
 ListTechno.txt</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#getAlltechnology()">getAlltechnology()</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Go through all SDRF files and read the "Array Design REF" entry, save in
 ListTechno.txt</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getAllUnPublishedBioConditions()">getAllUnPublishedBioConditions()</a></span> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Return all BioCondition available</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html#getAmbiguity(org.biojava3.core.sequence.compound.NucleotideCompound...)">getAmbiguity(NucleotideCompound...)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractNucleotideCompoundSet</a></dt>
<dd>
<div class="block">Calculates the best symbol for a collection of compounds.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getAminoAcid()">getAminoAcid()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Codon.html#getAminoAcid(java.lang.String)">getAminoAcid(String)</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>
<div class="block">Translate a codon</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Codon.html#getAminoAcidCharge(java.lang.String)">getAminoAcidCharge(String)</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>
<div class="block">Using aminoacidCharge array, return the value of the charge for each amino
 acid aminoacidCharge = {{"R","1"},{"H","1"},{"K","1"},{"D","-1"},{"E","-1"}}
 else =0</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getAminoAcidCompounds()">getAminoAcidCompounds()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#getAminoAcidCompoundSet()">getAminoAcidCompoundSet()</a></span> - Static method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getANALYSIS_PATH()">getANALYSIS_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#getAnnotation()">getAnnotation()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getAnnotation()">getAnnotation()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getAnnotation()">getAnnotation()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getAnnotation()">getAnnotation()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getAnnotation()">getAnnotation()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getANNOTATIONDATA_PATH()">getANNOTATIONDATA_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#getAnnotationGenes(bacnet.datamodel.sequence.Genome,java.util.ArrayList)">getAnnotationGenes(Genome, ArrayList&lt;String&gt;)</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Extract a String[][] with all gene annotation information</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#getAnnotationMatrix(bacnet.datamodel.sequence.Genome)">getAnnotationMatrix(Genome)</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return a matrix containing different information on each element of the
 genome first chromosome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#getAnnotationMatrix(bacnet.datamodel.sequence.Genome,java.util.ArrayList)">getAnnotationMatrix(Genome, ArrayList&lt;String&gt;)</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return a matrix containing different information on each element of the
 genome first chromosome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getAnnotations()">getAnnotations()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getAnnotationType()">getAnnotationType()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getAnnotElementHeight()">getAnnotElementHeight()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getAnnotNumber()">getAnnotNumber()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getAnnotSeparationHeight()">getAnnotSeparationHeight()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getAnnotSeparationNumber()">getAnnotSeparationNumber()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getAntiComparisons()">getAntiComparisons()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getAntiSDBindingFreeEnergy()">getAntiSDBindingFreeEnergy()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaSummaryView.html#getArray()">getArray()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaSummaryView.html" title="class in bacnet.sequenceTools">SrnaSummaryView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getArrayDataList()">getArrayDataList()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getArrayDataList()">getArrayDataList()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getArrayDesignRef(java.lang.String)">getArrayDesignRef(String)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Read the SDRF table and search for the "Array Design REF" column</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getArrayDesignRef(java.lang.String)">getArrayDesignRef(String)</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Read the SDRF table and search for the "Array Design REF" column</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#getArrayExpressArrayLink(java.lang.String)">getArrayExpressArrayLink(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#getArrayExpressExpLink(java.lang.String)">getArrayExpressExpLink(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getArrayExpressId()">getArrayExpressId()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getArrayExpressTechnoId()">getArrayExpressTechnoId()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getArrayGeneToLocalization()">getArrayGeneToLocalization()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getArrayProteomeList()">getArrayProteomeList()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getAsList()">getAsList()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getAsList()">getAsList()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getAsList()">getAsList()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getAsList()">getAsList()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getAsList()">getAsList()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns this Sequence store as a List</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getAsList()">getAsList()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getAsList()">getAsList()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Delegates to
 <a href="../org/biojava3/core/sequence/template/SequenceMixin.html#toList(org.biojava3.core.sequence.template.Sequence)"><code>SequenceMixin.toList(org.biojava3.core.sequence.template.Sequence)</code></a></div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getAsList()">getAsList()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Sequence.html#getAsList()">getAsList()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Returns the Sequence as a List of compounds</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getAsList()">getAsList()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getAsRNAListEGDe()">getAsRNAListEGDe()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getAsRNAs()">getAsRNAs()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getAsRNAs()">getAsRNAs()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getAtisBHI(bacnet.datamodel.sequence.Gene)">getAtisBHI(Gene)</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Go through all MassSpec data and find if the lmio was detected</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getAtisBlood(bacnet.datamodel.sequence.Gene)">getAtisBlood(Gene)</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermFilter.html#getaTISCutoff()">getaTISCutoff()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getaTISMap()">getaTISMap()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Feature.html#getAttribute(java.lang.String)">getAttribute(String)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Get value of specified attribute key.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/FeatureI.html#getAttribute(java.lang.String)">getAttribute(String)</a></span> - Method in interface org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a></dt>
<dd>
<div class="block">Get the attribute value for this key.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Feature.html#getAttributeOld(java.lang.String)">getAttributeOld(String)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getAUNumber(java.lang.String)">getAUNumber(String)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Count the number of A and T nucleotides</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getAvailableGenome()">getAvailableGenome()</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getAvailableGenome(boolean)">getAvailableGenome(boolean)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#getAvailableGenomes()">getAvailableGenomes()</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Read a text file containing the different genome ID available for Listeria</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/BasicColor.html#getAWTColor(org.eclipse.swt.graphics.Color)">getAWTColor(Color)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Transform SWT color to AWT color</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getBacillusGenome()">getBacillusGenome()</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Return a list of all NON Listeria genomes</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/WindowedSequence.html#getBackingSequence()">getBackingSequence()</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence</a></dt>
<dd>
<div class="block">Access the sequence which backs this window</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#getBase()">getBase()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#getBegin()">getBegin()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getBegin()">getBegin()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#getBegin()">getBegin()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#getBegin()">getBegin()</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#getBeginColumn()">getBeginColumn()</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return the column index corresponding to Begin</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getBioBegin()">getBioBegin()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getBioCondition()">getBioCondition()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Return corresponding BioCondition</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getBioCondition(java.lang.String)">getBioCondition(String)</a></span> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">From a biological condition name load the corresponding BioCondition object</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getBioCondition(java.lang.String,boolean)">getBioCondition(String, boolean)</a></span> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">From a biological condition name, get from the ModelProvider the
 corresponding BioCondition object<br>
 If not available it will load them for the hard drive</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getBIOCONDITION_PATH()">getBIOCONDITION_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getBioConditionHashMaps()">getBioConditionHashMaps()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DataValidation.html#getBioconditions()">getBioconditions()</a></span> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/Experiment.html#getBioConditions()">getBioConditions()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#getBioConditions()">getBioConditions()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#getBioConditions()">getBioConditions()</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getBioConditionsArrayPath()">getBioConditionsArrayPath()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getBioCondMax(java.lang.String)">getBioCondMax(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getBioCondMin(java.lang.String)">getBioCondMin(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#getBioCondName()">getBioCondName()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/Experiment.html#getBioConds()">getBioConds()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#getBioConds()">getBioConds()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#getBioConds()">getBioConds()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#getBioConds()">getBioConds()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#getBioConds()">getBioConds()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#getBioConds()">getBioConds()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getBioConds()">getBioConds()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#getBioCondsArray()">getBioCondsArray()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#getBioCondsArray()">getBioCondsArray()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#getBioCondsArray()">getBioCondsArray()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#getBioCondsArray()">getBioCondsArray()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#getBioCondsArray()">getBioCondsArray()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getBioCondsArray()">getBioCondsArray()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GenomicsView.html#getBioCondsArray()">getBioCondsArray()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getBioCondSize(java.lang.String)">getBioCondSize(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the size ratio that have a given BioCondition</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#getBioCondsToDisplay()">getBioCondsToDisplay()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#getBioCondsToDisplay()">getBioCondsToDisplay()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#getBioCondsToDisplay()">getBioCondsToDisplay()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#getBioCondsToDisplay()">getBioCondsToDisplay()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#getBioCondsToDisplay()">getBioCondsToDisplay()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getBioCondsToDisplay()">getBioCondsToDisplay()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GenomicsView.html#getBioCondsToDisplay()">getBioCondsToDisplay()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getBioEnd()">getBioEnd()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getBioEnd()">getBioEnd()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceView.html#getBioEnd()">getBioEnd()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a></dt>
<dd>
<div class="block">1-indexed, inclusive.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getBioStart()">getBioStart()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceView.html#getBioStart()">getBioStart()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a></dt>
<dd>
<div class="block">1-indexed, inclusive.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getBLAST_PATH()">getBLAST_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/Blast.html#getBLAST_RESULT_PATH()">getBLAST_RESULT_PATH()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#getBLAST_RESULT_PATH()">getBLAST_RESULT_PATH()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/Blast.html#getBlastDBInfo(java.lang.String)">getBlastDBInfo(String)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#getBlastExtension()">getBlastExtension()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/Blast.html#getBlastFolder()">getBlastFolder()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#getBlastFolder()">getBlastFolder()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorSWT.html#getBlue()">getBlue()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#getBodyFont(int)">getBodyFont(int)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Get the DEFAULT Font with appropriate height and style</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#getBodyFont(int,int)">getBodyFont(int, int)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Get the DEFAULT Font with appropriate height and style</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#getBoldFont(org.eclipse.swt.graphics.Font)">getBoldFont(Font)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns a bold version of the given <code>Font</code>.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getBpSizeH()">getBpSizeH()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getBpSizeH()">getBpSizeH()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnAllMutant()">getBtnAllMutant()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnBhiBroth()">getBtnBhiBroth()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnCacoCells()">getBtnCacoCells()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnChooseOneMutant()">getBtnChooseOneMutant()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnDeath()">getBtnDeath()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnDnDNA()">getBtnDnDNA()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnDnProt()">getBtnDnProt()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnDnRNA()">getBtnDnRNA()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnExponential()">getBtnExponential()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnGeneExpression()">getBtnGeneExpression()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnHumanBloodCells()">getBtnHumanBloodCells()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnLagPhase()">getBtnLagPhase()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnLbBroth()">getBtnLbBroth()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnMinimalMediaBroth()">getBtnMinimalMediaBroth()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnMouseIntestineCells()">getBtnMouseIntestineCells()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnMouseMacrophagesCells()">getBtnMouseMacrophagesCells()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnMouseSpleenCells()">getBtnMouseSpleenCells()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnNoDNA()">getBtnNoDNA()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnNoneMutant()">getBtnNoneMutant()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnNoProt()">getBtnNoProt()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnNoRNA()">getBtnNoRNA()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getBtnOpenStructure()">getBtnOpenStructure()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnPorcineBileCells()">getBtnPorcineBileCells()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnPosDNA()">getBtnPosDNA()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnPrint()">getBtnPrint()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnRegrowth()">getBtnRegrowth()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnRiboSeq()">getBtnRiboSeq()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnRnaseq()">getBtnRnaseq()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnStationnary()">getBtnStationnary()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnSurvival()">getBtnSurvival()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnTermSeq()">getBtnTermSeq()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnTiling()">getBtnTiling()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnTrypticSoyBroth()">getBtnTrypticSoyBroth()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnTss()">getBtnTss()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnTurkeyDeliMeat()">getBtnTurkeyDeliMeat()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getBtnUnpublished()">getBtnUnpublished()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnUpdateCutoff()">getBtnUpdateCutoff()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnUpProt()">getBtnUpProt()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getBtnUpRNA()">getBtnUpRNA()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getBtnZoomIn()">getBtnZoomIn()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getBtnZoomInVertical()">getBtnZoomInVertical()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getBtnZoomOut()">getBtnZoomOut()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getBtnZoomOutVertical()">getBtnZoomOutVertical()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html#getBufferedReader()">getBufferedReader()</a></span> - Method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a></dt>
<dd>
<div class="block">Returns the reader representation of this classpath resource</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#getBytesRead()">getBytesRead()</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>
<div class="block">Keep track of bytesread via ReadLine to account for CR-LF in the stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getCAI(bacnet.datamodel.sequence.Gene)">getCAI(Gene)</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">After generating TIRSequence.fasta in setTIRSequence() method we submit to
 http://genomes.urv.cat/CAIcal/ to calculate CAI value and expected CAI value
 The table with CAI values is saved from the website in: CAIResults.excel</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getCAIGene(bacnet.datamodel.sequence.Gene)">getCAIGene(Gene)</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">After generating GeneSequence.fasta in setGeneSequence() method we submit to
 http://genomes.urv.cat/CAIcal/ to calculate CAI value and expected CAI value
 The table with CAI values is saved from the website in:
 StatGenes/CAIGeneResults.excel</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getCanvasData()">getCanvasData()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getCanvasGenome()">getCanvasGenome()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getCDSSequences()">getCDSSequences()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>
<div class="block">Get the CDS sequences that have been added to the TranscriptSequences</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getCenterColor()">getCenterColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getCenterColor()">getCenterColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getCenterColorSWT()">getCenterColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getCenterColorSWT()">getCenterColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getCenterPos()">getCenterPos()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getCenterPos()">getCenterPos()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#getCgview()">getCgview()</a></span> - Method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getChargedResidues(bacnet.datamodel.sequence.Gene,int)">getChargedResidues(Gene, int)</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Sum up the different charge of the amino acids of the TIR</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getChargedResiduesGene(bacnet.datamodel.sequence.Gene)">getChargedResiduesGene(Gene)</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Sum up the different charge of the amino acids of the gene sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getChildrenFeatures()">getChildrenFeatures()</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">Get the children features</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getChildrenFeatures()">getChildrenFeatures()</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Get the features contained by this feature</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#getChromoId()">getChromoId()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getChromoID()">getChromoID()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getChromosome()">getChromosome()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getChromosome()">getChromosome()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Return corresponding Chromosome given by this.getChromosomeNb<br>
 Load it if necessary</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#getChromosome()">getChromosome()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getChromosomeFromDNASequence(java.util.LinkedHashMap,bacnet.datamodel.sequence.Genome,int,java.lang.String)">getChromosomeFromDNASequence(LinkedHashMap&lt;String, DNASequence&gt;, Genome, int, String)</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#getChromosomeID()">getChromosomeID()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getChromosomeID()">getChromosomeID()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getChromosomeID()">getChromosomeID()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getChromosomeID()">getChromosomeID()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getChromosomeID()">getChromosomeID()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#getChromosomeID()">getChromosomeID()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#getChromosomeNames()">getChromosomeNames()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Get an ordered list of all chromosomes The order is fixed by the order of the
 chromosomes in the FNA (fasta) file used for reading the genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#getChromosomeNumber()">getChromosomeNumber()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#getChromosomes()">getChromosomes()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#getChromosomes()">getChromosomes()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/GeneFeatureHelper.html#getChromosomeSequenceFromDNASequence(java.util.LinkedHashMap)">getChromosomeSequenceFromDNASequence(LinkedHashMap&lt;String, DNASequence&gt;)</a></span> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getCisRegRNAListEGDe()">getCisRegRNAListEGDe()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getCisRegs()">getCisRegs()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getCisRegs()">getCisRegs()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#getCmbChromo()">getCmbChromo()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#getCmbGenome()">getCmbGenome()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getCodingRegion(java.lang.String)">getCodingRegion(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>
<div class="block">Get the gene based on accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/CDSSequence.html#getCodingSequence()">getCodingSequence()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></dt>
<dd>
<div class="block">A CDS sequence if negative stranded needs to be reverse complement to
 represent the actual coding sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getCodingSequenceHashMap()">getCodingSequenceHashMap()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#getCodingSequencesList(boolean)">getCodingSequencesList(boolean)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>
<div class="block">Put in a list all the CodingSequence found</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/CodonUsage.html#getCodonColor(double)">getCodonColor(double)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/CodonUsage.html" title="class in bacnet.datamodel.annotation">CodonUsage</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)">getCodonCompoundSet(CompoundSet&lt;NucleotideCompound&gt;, CompoundSet&lt;AminoAcidCompound&gt;)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a></dt>
<dd>
<div class="block">Returns the compound set of codons</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.html#getCodonCompoundSet(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)">getCodonCompoundSet(CompoundSet&lt;NucleotideCompound&gt;, CompoundSet&lt;AminoAcidCompound&gt;)</a></span> - Method in interface org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Codon.html#getCodonProportion(bacnet.datamodel.sequence.Genome,java.lang.String)">getCodonProportion(Genome, String)</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>
<div class="block">Go through genome on both strand and calculate the proportion of each codon</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getCodons(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)">getCodons(CompoundSet&lt;NucleotideCompound&gt;, CompoundSet&lt;AminoAcidCompound&gt;)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a></dt>
<dd>
<div class="block">Returns a list of codons where the source and target compounds are the same
 as those given by the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.html#getCodons(org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)">getCodons(CompoundSet&lt;NucleotideCompound&gt;, CompoundSet&lt;AminoAcidCompound&gt;)</a></span> - Method in interface org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getCodonUsage()">getCodonUsage()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getCodonUsage(org.biojava3.core.sequence.DNASequence)">getCodonUsage(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getCOEXPR_NETWORK_TRANSCRIPTOMES_PATH()">getCOEXPR_NETWORK_TRANSCRIPTOMES_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Transcriptomes matrix data showing Log(Fold Change) values</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Network.html#getCoExpressionGlobalMatrix(bacnet.datamodel.sequence.Genome)">getCoExpressionGlobalMatrix(Genome)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>
<div class="block">Calculate all pearson correlation and put it in a list of network
 interactions<br>
 Remove correlation <0.85<br>
 Nedd to load:
 Database.COEXPR_NETWORK_TRANSCRIPTOMES_PATH+"_Temp_"+genome.getSpecies()</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#getCog()">getCog()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getCOG(org.biojava3.core.sequence.DNASequence)">getCOG(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/COGannotation.html#getCogClassification(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.GenomeNCBI)">getCogClassification(ExpressionMatrix, GenomeNCBI)</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/COGannotation.html#getCOGDescription(java.lang.String)">getCOGDescription(String)</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>
<div class="block">Return the COG description corresponding to a specific COGid</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getCOGExpression(bacnet.datamodel.dataset.ExpressionMatrix)">getCOGExpression(ExpressionMatrix)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Summarize an ExpressionMatrix by regrouping the different genes (rows) by
 their COG functional categories <br>
 Averages on all the genes for each category are calculated.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/COGannotation.html#getCOGMap()">getCOGMap()</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>
<div class="block">Return a map from COG ID to COG description</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/COGannotation.html#getCOGNumberEGDe()">getCOGNumberEGDe()</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>
<div class="block">Return a map between cogID and the number of genes associated to this cog in
 EGD-e</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#getColor(int)">getColor(int)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns the system <code>Color</code> matching the specific ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#getColor(int,int,int)">getColor(int, int, int)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns a <code>Color</code> given its red, green and blue component values.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#getColor(org.eclipse.swt.graphics.RGB)">getColor(RGB)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns a <code>Color</code> given its RGB value.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getColorBetween(double,double,double,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color)">getColorBetween(double, double, double, Color, Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getColorBetween(double,double,double,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color)">getColorBetween(double, double, double, Color, Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/ColorMapperWizard.html#getColorMapper()">getColorMapper()</a></span> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizard.html" title="class in bacnet.table.gui">ColorMapperWizard</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/ColorMapperWizardPage.html#getColorMapper()">getColorMapper()</a></span> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#getColorMapperList()">getColorMapperList()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperList.html#getColorMappers()">getColorMappers()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCPList.html#getColorMappers()">getColorMappers()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/BasicColor.html#getColors(int)">getColors(int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#getColumn(double%5B%5D%5B%5D,int)">getColumn(double[][], int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Extract a Column from a Matrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getColumn(int)">getColumn(int)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getColumn(java.lang.String)">getColumn(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Extract the column at this position</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#getColumn(java.lang.String%5B%5D%5B%5D,int)">getColumn(String[][], int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getColumn(java.lang.String%5B%5D%5B%5D,java.lang.String)">getColumn(String[][], String)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Read the SDRF table and search for the corresponding column</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getColumn(java.lang.String%5B%5D%5B%5D,java.lang.String)">getColumn(String[][], String)</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Read the SDRF table and search for the corresponding column</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getColumnIndex(java.lang.String%5B%5D%5B%5D,java.lang.String)">getColumnIndex(String[][], String)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Return column Index of the column named columnHeader</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getColumnIndex(java.lang.String%5B%5D%5B%5D,java.lang.String)">getColumnIndex(String[][], String)</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Return column Index of the column named columnHeader</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#getColumnNames()">getColumnNames()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#getColumnNames()">getColumnNames()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#getColumnNames()">getColumnNames()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#getColumnNames()">getColumnNames()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#getColumnNames()">getColumnNames()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#getColumnNames()">getColumnNames()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GenomicsView.html#getColumnNames()">getColumnNames()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getColumnNames()">getColumnNames()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getColumnNames()">getColumnNames()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getComboAbsoluteRelative()">getComboAbsoluteRelative()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#getComboGenome()">getComboGenome()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getComboGenome()">getComboGenome()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getComboGenome()">getComboGenome()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getComboMutant()">getComboMutant()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getComment()">getComment()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getComment()">getComment()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getComment(bacnet.datamodel.sequence.Gene)">getComment(Gene)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Return information about gene or iTIS</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getComparisonDataNames()">getComparisonDataNames()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Get every names of the Comparison which can be found:<br>
 Tiling arrays
 Gene Expression arrays
 ExpressionMatrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getComparisonNames()">getComparisonNames()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Return a list of all Comparisons: bioCondName vs comparisons.get(i)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getComparisons()">getComparisons()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DataValidation.html#getComparisons()">getComparisons()</a></span> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#getCompExpression()">getCompExpression()</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getCompFilter()">getCompFilter()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html#getComplement()">getComplement()</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#getComplement()">getComplement()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Returns a Sequence which will complement every base</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/RNASequence.html#getComplement()">getComplement()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>
<div class="block">Get the complement view of the RNA sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/ComplementCompound.html#getComplement()">getComplement()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/ComplementCompound.html" title="interface in org.biojava3.core.sequence.template">ComplementCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getCompleteGenome(java.util.ArrayList)">getCompleteGenome(ArrayList&lt;String&gt;)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Extract all complete genomes from genomesList</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getComposite()">getComposite()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getComposite_2()">getComposite_2()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getComposite_6()">getComposite_6()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getComposite_7()">getComposite_7()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getComposite_7()">getComposite_7()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getCompositeAddData()">getCompositeAddData()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#getCompositeDataFilter()">getCompositeDataFilter()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#getCompositeDataFilter()">getCompositeDataFilter()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getCompositeNterm()">getCompositeNterm()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getCompositeSummary()">getCompositeSummary()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getCompositeTypeDisplay()">getCompositeTypeDisplay()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#getComposition(org.biojava3.core.sequence.template.Sequence)">getComposition(Sequence&lt;C&gt;)</a></span> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Does a linear scan over the given Sequence and records the number of times
 each base appears.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getCompoundAt(int)">getCompoundAt(int)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getCompoundAt(int)">getCompoundAt(int)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getCompoundAt(int)">getCompoundAt(int)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getCompoundAt(int)">getCompoundAt(int)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#getCompoundAt(int)">getCompoundAt(int)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>
<div class="block">Returns the compound at the specified biological index</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getCompoundAt(int)">getCompoundAt(int)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns the compound at the specified biological index</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getCompoundAt(int)">getCompoundAt(int)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getCompoundAt(int)">getCompoundAt(int)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Always returns the compound given at construction</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getCompoundAt(int)">getCompoundAt(int)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Sequence.html#getCompoundAt(int)">getCompoundAt(int)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Returns the Compound at the given biological index</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getCompoundAt(int)">getCompoundAt(int)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html#getCompoundAt(int)">getCompoundAt(int)</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava3.core.sequence.views">ComplementSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/ReversedSequenceView.html#getCompoundAt(int)">getCompoundAt(int)</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views">ReversedSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#getCompoundAt(int)">getCompoundAt(int)</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#getCompoundForString(java.lang.String)">getCompoundForString(String)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#getCompoundForString(java.lang.String)">getCompoundForString(String)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/CompoundSet.html#getCompoundForString(java.lang.String)">getCompoundForString(String)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></dt>
<dd>
<div class="block">Return null if not recognised.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/WindowedSequence.html#getCompounds(int)">getCompounds(int)</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence</a></dt>
<dd>
<div class="block">For a given position into the windowed view this will return those compounds
 we can see in the window. i.e. in the sequence AGGCCT requesting index 1
 returns AGG and requesting index 2 return CCT.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#getCompoundsAt(int)">getCompoundsAt(int)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">Get a list of compounds at a sequence position</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/LightweightProfile.html#getCompoundsAt(int)">getCompoundsAt(int)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a></dt>
<dd>
<div class="block">Returns the <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template"><code>Compound</code></a> elements of the original <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a>s at
 the given column.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/AminoAcidCompound.html#getCompoundSet()">getCompoundSet()</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getCompoundSet()">getCompoundSet()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getCompoundSet()">getCompoundSet()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getCompoundSet()">getCompoundSet()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#getCompoundSet()">getCompoundSet()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">Get the Compounds defined in the first sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getCompoundSet()">getCompoundSet()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#getCompoundSet()">getCompoundSet()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>
<div class="block">Returns the compound set backing this store</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getCompoundSet()">getCompoundSet()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns the compound set backing this store</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getCompoundSet()">getCompoundSet()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getCompoundSet()">getCompoundSet()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Returns the compound set given at construction</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getCompoundSet()">getCompoundSet()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/LightweightProfile.html#getCompoundSet()">getCompoundSet()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a></dt>
<dd>
<div class="block">Returns <a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template"><code>CompoundSet</code></a> of all <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a>s</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Sequence.html#getCompoundSet()">getCompoundSet()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Gets the compound set used to back this Sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getCompoundSet()">getCompoundSet()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getConservation()">getConservation()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getConservationHashMap()">getConservationHashMap()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html#getConsituents()">getConsituents()</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#getContainer()">getContainer()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ImageMagick.html#getConvertPATH()">getConvertPATH()</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ImageMagick.html" title="class in bacnet.utils">ImageMagick</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperList.html#getCorrespondingMapper(java.lang.String)">getCorrespondingMapper(String)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCPList.html#getCorrespondingMapper(java.lang.String)">getCorrespondingMapper(String)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#getCreator()">getCreator()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getCurrentState()">getCurrentState()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#getCursor(int)">getCursor(int)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns the system cursor matching the specific ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#getCutOff1()">getCutOff1()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#getCutOff1()">getCutOff1()</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#getCutOff2()">getCutOff2()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#getCutOff2()">getCutOff2()</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermFilter.html#getCutoffHashMap()">getCutoffHashMap()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getDATA_PATH()">getDATA_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">GETTER AND SETTERS</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#getDatabase()">getDatabase()</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getDatabaseFeatures()">getDatabaseFeatures()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/DatabaseReferenceInterface.html#getDatabaseReferences()">getDatabaseReferences()</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getDatabaseReferences()">getDatabaseReferences()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">The Uniprot mappings to other database identifiers for this sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getDatabaseReferences()">getDatabaseReferences()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataColors()">getDataColors()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataCount()">getDataCount()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Count the number of data available, without duplicate and filtered</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataDisplayed()">getDataDisplayed()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Get the list of data to display</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataDisplayedNames()">getDataDisplayedNames()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the names of the data which will be displayd</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/TranscriptomesCreation.html#getDataFromLab()">getDataFromLab()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/TranscriptomesCreation.html" title="class in bacnet.scripts.database">TranscriptomesCreation</a></dt>
<dd>
<div class="block">Get all GeneExpression from the lab</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataNames()">getDataNames()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of all the data available</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataNOTDisplayed()">getDataNOTDisplayed()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#getDataRegion()">getDataRegion()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#getDatas()">getDatas()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NGS.html#getDatasets()">getDatasets()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataSize()">getDataSize()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataSizes()">getDataSizes()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/AccessionID.html#getDataSource()">getDataSource()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/InsdcParser.html#getDataSource()">getDataSource()</a></span> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/TaxonomyID.html#getDataSource()">getDataSource()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDataTotalSize(bacnet.genomeBrowser.core.Track.DisplayType)">getDataTotalSize(Track.DisplayType)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Add all the size of the Tracks to know the total size of the display window</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getDataWindowSize()">getDataWindowSize()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#getDate()">getDate()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getDate()">getDate()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getDbxref(org.biojava3.core.sequence.DNASequence)">getDbxref(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getDecaySliderVBar()">getDecaySliderVBar()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getDefault()">getDefault()</a></span> - Static method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>
<div class="block">Default instance to use when Transcribing from DNA -&gt; RNA -&gt; Protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Frame.html#getDefaultFrame()">getDefaultFrame()</a></span> - Static method in enum org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#getDefautGenome()">getDefautGenome()</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return defaut genome for a databse>br> For the moment -> first element of
 getAvailableGenomes()</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#getDescription()">getDescription()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#getDescription()">getDescription()</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getDescription()">getDescription()</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getDescription()">getDescription()</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Get the description that can be used to describe the feature</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#getDescription()">getDescription()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getDescription()">getDescription()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Compound.html#getDescription()">getDescription()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#getDescription()">getDescription()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getDescription()">getDescription()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getDetectable(bacnet.datamodel.sequence.Gene)">getDetectable(Gene)</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">By reading the List in: /StatGenes/EGDeDetectable.txt Say if a gene is
 detectable by the spectometer or not</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getDeviation()">getDeviation()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getDeviation()">getDeviation()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#getDeviation(java.util.ArrayList,java.lang.String)">getDeviation(ArrayList&lt;Tiling&gt;, String)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Calculate the stat deviation on each probe of a list of Tiling</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/BioConditionComparator.html#getDirection()">getDirection()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/TableViewerComparator.html#getDirection()">getDirection()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/TableViewerComparator.html" title="class in bacnet.table.core">TableViewerComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDisplay()">getDisplay()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getDisplay()">getDisplay()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#getDisplayRegion()">getDisplayRegion()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#getDisplayType()">getDisplayType()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#getDistribution(org.biojava3.core.sequence.template.Sequence)">getDistribution(Sequence&lt;C&gt;)</a></span> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Analogous to <a href="../org/biojava3/core/sequence/template/SequenceMixin.html#getComposition(org.biojava3.core.sequence.template.Sequence)"><code>SequenceMixin.getComposition(Sequence)</code></a> but returns the distribution
 of that <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template"><code>Compound</code></a> over the given sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getDNACodingSequence()">getDNACodingSequence()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>
<div class="block">Get the stitched together CDS sequences then maps to the cDNA</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html#getDNACodingSequences()">getDNACodingSequences()</a></span> - Method in class org.biojava3.genome.parsers.geneid.<a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html" title="class in org.biojava3.genome.parsers.geneid">GeneIDXMLReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getDnaCompounds()">getDnaCompounds()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/AmbiguityDNACompoundSet.html#getDNACompoundSet()">getDNACompoundSet()</a></span> - Static method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AmbiguityDNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">AmbiguityDNACompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/AmbiguityRNACompoundSet.html#getDNACompoundSet()">getDNACompoundSet()</a></span> - Static method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AmbiguityRNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">AmbiguityRNACompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/DNACompoundSet.html#getDNACompoundSet()">getDNACompoundSet()</a></span> - Static method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/DNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">DNACompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getDnaRnaTranslator()">getDnaRnaTranslator()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDNASeqDatas()">getDNASeqDatas()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of SeqDatas</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getDNASeqDatas(java.lang.String)">getDNASeqDatas(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of RiboSeqData given a bioCondName</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#getDnaToRna()">getDnaToRna()</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#getDNAType()">getDNAType()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getDownDNAList()">getDownDNAList()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/InternalBrowser.html#getDownloadLink()">getDownloadLink()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/DownloadServiceHandler.html#getDownloadUrl(java.lang.String,java.io.File,org.eclipse.e4.ui.workbench.modeling.EPartService)">getDownloadUrl(String, File, EPartService)</a></span> - Static method in class bacnet.rap.<a href="../bacnet/rap/DownloadServiceHandler.html" title="class in bacnet.rap">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/DownloadServiceHandler.html#getDownloadUrl(java.lang.String,java.io.File,org.eclipse.e4.ui.workbench.modeling.EPartService)">getDownloadUrl(String, File, EPartService)</a></span> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/DownloadServiceHandler.html#getDownloadUrl(java.lang.String,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)">getDownloadUrl(String, String, EPartService)</a></span> - Static method in class bacnet.rap.<a href="../bacnet/rap/DownloadServiceHandler.html" title="class in bacnet.rap">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/DownloadServiceHandler.html#getDownloadUrl(java.lang.String,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)">getDownloadUrl(String, String, EPartService)</a></span> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getDownProtList()">getDownProtList()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getDownRNAList()">getDownRNAList()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getDuplicates()">getDuplicates()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermFilter.html#getDuplicateSequenceToRemove()">getDuplicateSequenceToRemove()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Network.html#getEdges()">getEdges()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>
<div class="block">An hashMap which map each vertex (=genome element) to the correlated
 vertex.<br>
 Vertex -> HashMap<Vetrex,PearsonCorrelation></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getEGDcOverlap()">getEGDcOverlap()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#getEGDcSrnas()">getEGDcSrnas()</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Return a list of all Srna in EGD-c</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getEGDe()">getEGDe()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Read RAST automoticaly annotated EGD-e genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#getEGDeALLSrnas()">getEGDeALLSrnas()</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Return a list of all Srna, CisReg and ASrna in EGD-e</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getEGDeAutoOverlap()">getEGDeAutoOverlap()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Go through EGDe overlap and find when two genes overlap</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#getEgdeOperonList()">getEgdeOperonList()</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#getEgdeOperonMap()">getEgdeOperonMap()</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getEGDeOverlap()">getEGDeOverlap()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getEGDeRASTAutoOverlap()">getEGDeRASTAutoOverlap()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Find RASt element which overlap another element in the genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#getEGDeSrnas()">getEGDeSrnas()</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Return an ordered list of all Srna in EGD-e</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getEGDeUnknownRAST()">getEGDeUnknownRAST()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#getElement(java.lang.String)">getElement(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Search an element by its name in the genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#getElementATbp(bacnet.datamodel.sequence.Chromosome,int)">getElementATbp(Chromosome, int)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return the first element for which<br>
 begin < bpPosition < end</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#getElementInfoATbp(bacnet.datamodel.sequence.Chromosome,int)">getElementInfoATbp(Chromosome, int)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return the first element for which<br>
 begin < bpPosition < end</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#getElements()">getElements()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getElements()">getElements()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Link NTerm name to <code>NTerm</code><br>
 This correspond to the list of <code>NTerm</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getElements()">getElements()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getElements()">getElements()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#getElements(bacnet.datamodel.sequence.Chromosome,int,int)">getElements(Chromosome, int, int)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Get the lists of elements between beginDraw and endDraw</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getElementsToDisplay(int,int,java.lang.String)">getElementsToDisplay(int, int, String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Get the lists of NTerm between beginDraw and endDraw</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#getENALink(java.lang.String)">getENALink(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#getEnd()">getEnd()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getEnd()">getEnd()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#getEnd()">getEnd()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html#getEnd()">getEnd()</a></span> - Method in class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="class in org.biojava3.core.sequence.edits">Edit.AbstractEdit</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getEnd()">getEnd()</a></span> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Location.html#getEnd()">getEnd()</a></span> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">End of the location</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#getEnd()">getEnd()</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/edits/Edit.Substitute.html#getEnd(org.biojava3.core.sequence.template.Sequence)">getEnd(Sequence&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.Substitute.html" title="class in org.biojava3.core.sequence.edits">Edit.Substitute</a></dt>
<dd>
<div class="block">Must use this rather than the no-args getEnd as this can return -1 and the
 length of a sub is dependent on the length of the Sequence; we cannot assume
 1:1 mapping between characters in a String and the number of compounds we
 will have to insert.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#getEndColumn()">getEndColumn()</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return the column index corresponding to End</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.PluginResourceProvider.html#getEntry(java.lang.String,java.lang.String)">getEntry(String, String)</a></span> - Method in interface bacnet.swt.<a href="../bacnet/swt/ResourceManager.PluginResourceProvider.html" title="interface in bacnet.swt">ResourceManager.PluginResourceProvider</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#getEquivalentCompounds(C)">getEquivalentCompounds(C)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/CompoundSet.html#getEquivalentCompounds(C)">getEquivalentCompounds(C)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#getEquivalentCompounds(org.biojava3.core.sequence.compound.AminoAcidCompound)">getEquivalentCompounds(AminoAcidCompound)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#getExcludeColumn()">getExcludeColumn()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#getExcludeColumn()">getExcludeColumn()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/FilterList.html#getExcludeColumn()">getExcludeColumn()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#getExcludeColumn()">getExcludeColumn()</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#getExcludeRow()">getExcludeRow()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#getExcludeRow()">getExcludeRow()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/FilterList.html#getExcludeRow()">getExcludeRow()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#getExcludeRow()">getExcludeRow()</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getExcludeRows()">getExcludeRows()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/GeneSequence.html#getExonSequences()">getExonSequences()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Get the exons as an ArrayList</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getExperienceNb()">getExperienceNb()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getExperiment()">getExperiment()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>
<div class="block">Return if the NTerm was detected in the first experiment, or the second, or
 both</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#getExperiment()">getExperiment()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>
<div class="block">Return if the NTerms of this TIS were detected in the first experiment, or
 the second, or both</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getExperimentComparisonTablePath()">getExperimentComparisonTablePath()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#getExpression()">getExpression()</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#getExpression()">getExpression()</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getEXPRESSION_MATRIX_PROTEOMES_PATH()">getEXPRESSION_MATRIX_PROTEOMES_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Proteomes matrix data showing absolute expression values</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getExpressionData()">getExpressionData()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Put all ExpressionData in a list</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/AMtools.html#getExpressionMatrix(double%5B%5D%5B%5D)">getExpressionMatrix(double[][])</a></span> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/AMtools.html" title="class in bacnet.scripts.core.stat">AMtools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getExprProteomesTable()">getExprProteomesTable()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getExprProteomesTable(java.lang.String)">getExprProteomesTable(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getExtension()">getExtension()</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#getExtension()">getExtension()</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/FileUtils.html#getExtension(java.lang.String)">getExtension(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Get the extension in a fileName</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#getFAAPath(java.lang.String)">getFAAPath(String)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getFeature(java.lang.String)">getFeature(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Get the feature in the HashMap, pointed by the key attribute<br>
 return "" if not found</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getFeatures()">getFeatures()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures()">getFeatures()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures(int)">getFeatures(int)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Return features at a sequence position</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures(java.lang.String,int)">getFeatures(String, int)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Return features at a sequence position by type</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType(String)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getFeaturesKeyWord()">getFeaturesKeyWord()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getFeaturesText()">getFeaturesText()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Create a String containing all features and their attribute</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getFeaturesTextForTable(java.lang.String)">getFeaturesTextForTable(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Used when displaying information in <code>SrnaView</code><br>
 Display only supplementary information of each Srna publication</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/DownloadServiceHandler.html#getFile()">getFile()</a></span> - Method in class bacnet.rap.<a href="../bacnet/rap/DownloadServiceHandler.html" title="class in bacnet.rap">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/DownloadServiceHandler.html#getFile()">getFile()</a></span> - Method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/FileUtils.html#getFileBytes(java.io.File)">getFileBytes(File)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/DownloadServiceHandler.html#getFileName()">getFileName()</a></span> - Method in class bacnet.rap.<a href="../bacnet/rap/DownloadServiceHandler.html" title="class in bacnet.rap">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/DownloadServiceHandler.html#getFileName()">getFileName()</a></span> - Method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/InternalBrowser.html#getFileName()">getFileName()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SynTView.html#getFilteredNetwork()">getFilteredNetwork()</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#getFilteredNetwork()">getFilteredNetwork()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/FilterList.html#getFilters()">getFilters()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#getFirstChromosome()">getFirstChromosome()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Get the first chromosome in the HashMap
 
 Is Used to replace: Genome.loadGenome(genomeName).getChromosomes().get(0)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Codon.html#getFirstCodon(bacnet.datamodel.sequence.Sequence)">getFirstCodon(Sequence)</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>
<div class="block">Return first threee nucleotide of a Sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperList.html#getFirstMapper()">getFirstMapper()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCPList.html#getFirstMapper()">getFirstMapper()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getFirstRowName()">getFirstRowName()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#getFont(java.lang.String,int,int,boolean,boolean)">getFont(String, int, int, boolean, boolean)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns a <code>Font</code> based on its name, height and style.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getFontDouble()">getFontDouble()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getFontDouble()">getFontDouble()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getFontRowName()">getFontRowName()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getFontString()">getFontString()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getFontText()">getFontText()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#getFormylatedExperiment()">getFormylatedExperiment()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>
<div class="block">Search if the TIS was found formylated in the first experiment, or the
 second, or both</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Frame.html#getForwardFrames()">getForwardFrames()</a></span> - Static method in enum org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></dt>
<dd>
<div class="block">Returns all frames in the forward orientation</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#getFoundIn()">getFoundIn()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#getFoundInText()">getFoundInText()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#getFromCompoundSet()">getFromCompoundSet()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermUtils.html#getFullORF(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Genome)">getFullORF(Sequence, Genome)</a></span> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in
 frame) containing the peptide (amino acid sequence) <br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermUtils.html#getFullORF(java.lang.String,int,int,boolean,bacnet.datamodel.sequence.Genome)">getFullORF(String, int, int, boolean, Genome)</a></span> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>
<div class="block">Giving a position and a strand on the genome we extract the full ORF (in
 frame) containing the peptide (amino acid sequence) <br></div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/RNASequence.html#getGC()">getGC()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#getGCCount()">getGCCount()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Get the GC count in the DNA Sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGChecksum(java.util.List)">getGCGChecksum(List&lt;S&gt;)</a></span> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Calculates GCG checksum for entire list of sequences</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGChecksum(S)">getGCGChecksum(S)</a></span> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Calculates GCG checksum for a given sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGHeader(java.util.List)">getGCGHeader(List&lt;S&gt;)</a></span> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Assembles a GCG file header</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getGCGType(org.biojava3.core.sequence.template.CompoundSet)">getGCGType(CompoundSet&lt;C&gt;)</a></span> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Determines GCG type</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getGCNumber(java.lang.String)">getGCNumber(String)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Count the number of G and C nucleotides</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/GCStats.html#getGCStats(java.util.Collection)">getGCStats(Collection&lt;DNASequence&gt;)</a></span> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GCStats.html" title="class in org.biojava3.genome.parsers.gff">GCStats</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/GCStats.html#getGCStatsString(java.util.Collection)">getGCStatsString(Collection&lt;String&gt;)</a></span> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GCStats.html" title="class in org.biojava3.genome.parsers.gff">GCStats</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#getGene(java.lang.String)">getGene(String)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>
<div class="block">Get the gene based on accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getGene(org.biojava3.core.sequence.DNASequence)">getGene(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#getGeneDownstream()">getGeneDownstream()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>
<div class="block">Return the gene downstream on the + strand</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getGeneDownstream(bacnet.datamodel.sequence.Gene)">getGeneDownstream(Gene)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>
<div class="block">Return the gene downstream</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/EGDeWTdata.html#getGeneExpr()">getGeneExpr()</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/EGDeWTdata.html" title="class in bacnet.datamodel.dataset">EGDeWTdata</a></dt>
<dd>
<div class="block">Read GeneEpression corresponding to NAME_Mean, and return it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getGeneExpr(bacnet.datamodel.sequence.Gene,java.lang.String)">getGeneExpr(Gene, String)</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Return the median value expression (calculated on 10 BHI 37C data) in the
 Gene Expression array of the gene</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getGeneExprs()">getGeneExprs()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/Experiment.html#getGeneExprs()">getGeneExprs()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Return all GeneExpression data contains in all BioCondition</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getGeneExprs()">getGeneExprs()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of GeneExpressions</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getGeneExprs(java.lang.String)">getGeneExprs(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of GeneExpression given a bioCondName</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#getGeneFromName(java.lang.String)">getGeneFromName(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Go through all chromosomes and each Genes map and GeneName map to find the
 corresponding gene</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#getGeneFromName(java.lang.String)">getGeneFromName(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>
<div class="block">Go through codingSequence and nonCodingSequence searching for corresponding
 accession or locus</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#getGeneFromName(java.lang.String,java.lang.String)">getGeneFromName(String, String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Go through all chromosomes and each Genes map and GeneName map to find the
 corresponding gene</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getGeneListEGDe()">getGeneListEGDe()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getGeneModification()">getGeneModification()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#getGeneName()">getGeneName()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getGeneName()">getGeneName()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">Get the gene name associated with this sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#getGeneName(java.lang.String)">getGeneName(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>
<div class="block">Look in LocusTagToGeneName if a geneName exists</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getGeneNameList()">getGeneNameList()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>
<div class="block">Return a List with all name of genes</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#getGeneNames()">getGeneNames()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return all the Gene locus tag name available in every chromosome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getGeneNameToLocusTagMap()">getGeneNameToLocusTagMap()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getGeneNameToLocusTagMap()">getGeneNameToLocusTagMap()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/Experiment.html#getGeneralExp()">getGeneralExp()</a></span> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Load all experiment<br>
 If all experiments are already loaded just get the general experiment found
 in ModelProvider.INSTANCE</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getGeneralExperiment()">getGeneralExperiment()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SynTView.html#getGeneralNetwork()">getGeneralNetwork()</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#getGeneralNetwork()">getGeneralNetwork()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getGeneralWtName()">getGeneralWtName()</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getGenes()">getGenes()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getGenes()">getGenes()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Operon.html#getGenes()">getGenes()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getGenesAlternative()">getGenesAlternative()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#getGeneSequences()">getGeneSequences()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>
<div class="block">Get the list of genes that have been added to the ChromosomeSequence where
 accession.toString is the key.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/GeneFeatureHelper.html#getGeneSequences(java.util.Collection)">getGeneSequences(Collection&lt;ChromosomeSequence&gt;)</a></span> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#getGeneToOperon()">getGeneToOperon()</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#getGeneToSigMap(bacnet.datamodel.sequence.ChromosomeBacteriaSequence)">getGeneToSigMap(ChromosomeBacteriaSequence)</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>
<div class="block">Create a mapping between gene to a signature <br>
 All signatures from the folder Database.getSIGNATURES() are used</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#getGeneUpstream()">getGeneUpstream()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>
<div class="block">Return the gene upstream on the + strand</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getGeneUpstream(bacnet.datamodel.sequence.Gene)">getGeneUpstream(Gene)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>
<div class="block">Return the gene upstream</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#getGenome()">getGenome()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#getGenome()">getGenome()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getGenome()">getGenome()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Return corresponding Genome given by this.genomeName<br>
 Load it if necessary</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getGenome()">getGenome()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Return corresponding Genome given by this.genomeName<br>
 Load it if necessary</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#getGenome()">getGenome()</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#getGenome()">getGenome()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getGenome()">getGenome()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#getGenome()">getGenome()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getGenome()">getGenome()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#getGenome()">getGenome()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getGenome(java.lang.String)">getGenome(String)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Read a genome from RAST results</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getGenomeAA()">getGenomeAA()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getGenomeAACompl()">getGenomeAACompl()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getGenomeAAMinus()">getGenomeAAMinus()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getGenomeAAMinusCompl()">getGenomeAAMinusCompl()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getGenomeAAPlus()">getGenomeAAPlus()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getGenomeAAPlusCompl()">getGenomeAAPlusCompl()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getGenomeArrayPath()">getGenomeArrayPath()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#getGenomeElements()">getGenomeElements()</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#getGenomeElements()">getGenomeElements()</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getGenomeList()">getGenomeList()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getGenomeName()">getGenomeName()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NGS.html#getGenomeName()">getGenomeName()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getGenomeName()">getGenomeName()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getGenomeName()">getGenomeName()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SynTView.html#getGenomeName()">getGenomeName()</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getGenomeName()">getGenomeName()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#getGenomeName()">getGenomeName()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#getGenomeName()">getGenomeName()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#getGenomeName()">getGenomeName()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#getGenomeName()">getGenomeName()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getGenomeName()">getGenomeName()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#getGenomeName()">getGenomeName()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#getGenomeNCBILink(java.lang.String)">getGenomeNCBILink(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getGenomes()">getGenomes()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DataValidation.html#getGenomes()">getGenomes()</a></span> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>
<div class="block">GETTER AND SETTER</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#getGenomes()">getGenomes()</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>
<div class="block">get the list of genome IDs and corresponding genome Names<br>
 And verify if all genomes are here!</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getGENOMES_PATH()">getGENOMES_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html#getGenomesAvalaible()">getGenomesAvalaible()</a></span> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html" title="class in bacnet.scripts.phylogeny">GenomePhylogeny</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span></div>
<div class="block">Create list of genomes which contains all genome available <br>
 
 This genome list contains all the genomes both from GenomeFolder and
 GenomePhylogeny.PATH_TABLE (=All bacteria phylogeny)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getGenomeSelected()">getGenomeSelected()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">The comboGenome contains modified genome name so we need this method to get
 selected element<br>
 a '*' is add to genome name when a transcriptome data is available</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getGenomesNCBI()">getGenomesNCBI()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getGenomeUsed()">getGenomeUsed()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getGenomeViewerColumnIndex()">getGenomeViewerColumnIndex()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Return the appropriate column index for displaying in the Genome viewer<br>
 By order of priority:
 LOGFC containing column
 FC containing column
 p-value containing column
 VALUE
 first column</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#getGExpressionData(java.util.ArrayList)">getGExpressionData(ArrayList&lt;String&gt;)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">List all TilingData available by biocondName.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#getGExpressionData(java.util.ArrayList,java.util.ArrayList)">getGExpressionData(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Load all Tiling present in leftBCs and rightBCs.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getGlaserFC(org.biojava3.core.sequence.DNASequence)">getGlaserFC(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/GlaserFCannotation.html#getGlaserFCDescription(java.lang.String)">getGlaserFCDescription(String)</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/GlaserFCannotation.html" title="class in bacnet.datamodel.annotation">GlaserFCannotation</a></dt>
<dd>
<div class="block">Return the Glaser Functional categories description corresponding to a
 specific glaserFCID</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/GlaserFCannotation.html#getGlaserFCMap()">getGlaserFCMap()</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/GlaserFCannotation.html" title="class in bacnet.datamodel.annotation">GlaserFCannotation</a></dt>
<dd>
<div class="block">Return a map from glaserFCID to Glaser et al. 2001 Functional categories
 description</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getGoogleId()">getGoogleId()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorSWT.html#getGreen()">getGreen()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getGrowth()">getGrowth()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getHeader(int)">getHeader(int)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html#getHeader(S)">getHeader(S)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderFormat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/template/FastaHeaderFormatInterface.html#getHeader(S)">getHeader(S)</a></span> - Method in interface org.biojava3.core.sequence.io.template.<a href="../org/biojava3/core/sequence/io/template/FastaHeaderFormatInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderFormatInterface</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getHeaderAnnotation()">getHeaderAnnotation()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getHeaderColor()">getHeaderColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getHeaderColorSWT()">getHeaderColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getHeaderColorSWT()">getHeaderColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getHeaders()">getHeaders()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getHeadersListToArray()">getHeadersListToArray()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getHeadersToArray()">getHeadersToArray()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/NCBIFastaHeaderParser.html#getHeaderValues(java.lang.String)">getHeaderValues(String)</a></span> - Method in class bacnet.reader.<a href="../bacnet/reader/NCBIFastaHeaderParser.html" title="class in bacnet.reader">NCBIFastaHeaderParser</a></dt>
<dd>
<div class="block">Parse out the components where some have a | and others do not</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#getHeight()">getHeight()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/MatrixSize.html#getHeight()">getHeight()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/MatrixSize.html#getHeightDefault()">getHeightDefault()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getHeightPix()">getHeightPix()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getHeightPix()">getHeightPix()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/query/BlastXMLQuery.html#getHitsQueryDef(double)">getHitsQueryDef(double)</a></span> - Method in class org.biojava3.genome.query.<a href="../org/biojava3/genome/query/BlastXMLQuery.html" title="class in org.biojava3.genome.query">BlastXMLQuery</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/TabDelimitedTableReader.html#getHTMLVersion(java.lang.Object%5B%5D%5B%5D)">getHTMLVersion(Object[][])</a></span> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Save the table in a text file containing an HTML table</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#getId()">getId()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getId()">getId()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#getId()">getId()</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getId()">getId()</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#getID()">getID()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/AccessionID.html#getID()">getID()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/TaxonomyID.html#getID()">getID()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getIDFormat(java.util.List)">getIDFormat(List&lt;S&gt;)</a></span> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Creates format String for accession IDs</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getIDFPath(java.lang.String)">getIDFPath(String)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Get path of IDF file for a specific dataID (ex: E-MTAB-1800)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getIDFPath(java.lang.String)">getIDFPath(String)</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Get path of IDF file for a specific dataID (ex: E-MTAB-1800)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#getImage(java.lang.Class,java.lang.String)">getImage(Class&lt;?&gt;, String)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> stored in the file at the specified path relative to
 the specified class.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#getImage(java.lang.String)">getImage(String)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> stored in the file at the specified path.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#getImage(org.eclipse.jface.resource.ImageDescriptor)">getImage(ImageDescriptor)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> based on the specified <code>ImageDescriptor</code>.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getImageChecked()">getImageChecked()</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#getImageDescriptor(java.lang.Class,java.lang.String)">getImageDescriptor(Class&lt;?&gt;, String)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>ImageDescriptor</code> stored in the file at the specified path
 relative to the specified class.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#getImageDescriptor(java.lang.String)">getImageDescriptor(String)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>ImageDescriptor</code> stored in the file at the specified path.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getImageGeneExpr()">getImageGeneExpr()</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getImageRNASeq()">getImageRNASeq()</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getImageTilingGeneExpr()">getImageTilingGeneExpr()</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getImageTSS()">getImageTSS()</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getImageTSSTilingGeneExpr()">getImageTSSTilingGeneExpr()</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getImageUnchecked()">getImageUnchecked()</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getIndexOf(C)">getIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getIndexOf(C)">getIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getIndexOf(C)">getIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getIndexOf(C)">getIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getIndexOf(C)">getIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns the first occurrence of the given compound in this store; performs a
 linear search</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getIndexOf(C)">getIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getIndexOf(C)">getIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Returns 1 if the given compound is equal to the one given during
 construction; otherwise will return -1.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getIndexOf(C)">getIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Sequence.html#getIndexOf(C)">getIndexOf(C)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Scans through the Sequence looking for the first occurrence of the given
 compound</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getIndexOf(C)">getIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html#getIndexOf(C)">getIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava3.core.sequence.views">ComplementSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#getIndexOf(org.biojava3.core.sequence.compound.NucleotideCompound)">getIndexOf(NucleotideCompound)</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#getInfo()">getInfo()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>
<div class="block">Return a String containing getProduct() and getComment() information</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/TabDelimitedTableReader.html#getInHTML(java.lang.String%5B%5D%5B%5D,java.lang.String)">getInHTML(String[][], String)</a></span> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Return an HTML version of the table</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/TabDelimitedTableReader.html#getInHTML(java.util.ArrayList,java.lang.String)">getInHTML(ArrayList&lt;String&gt;, String)</a></span> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Return an HTML version of the ArrayList<String></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getInitView()">getInitView()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getInocuaColor(double)">getInocuaColor(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getInocuaColor(double)">getInocuaColor(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html#getInputStream()">getInputStream()</a></span> - Method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a></dt>
<dd>
<div class="block">Returns the InputStream instance of this classpath resource</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/InputStreamProvider.html#getInputStream(java.io.File)">getInputStream(File)</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/InputStreamProvider.html" title="class in org.biojava3.core.util">InputStreamProvider</a></dt>
<dd>
<div class="block">get an InputStream for the file</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/FlatFileCache.html#getInputStream(java.lang.String)">getInputStream(String)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/FlatFileCache.html" title="class in org.biojava3.core.util">FlatFileCache</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/InputStreamProvider.html#getInputStream(java.lang.String)">getInputStream(String)</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/InputStreamProvider.html" title="class in org.biojava3.core.util">InputStreamProvider</a></dt>
<dd>
<div class="block">get an InputStream for this file</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/InputStreamProvider.html#getInputStream(java.net.URL)">getInputStream(URL)</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/InputStreamProvider.html" title="class in org.biojava3.core.util">InputStreamProvider</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getInstance()">getInstance()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/IUPACParser.html#getInstance()">getInstance()</a></span> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/FlatFileCache.html#getInstance()">getInstance()</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/FlatFileCache.html" title="class in org.biojava3.core.util">FlatFileCache</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/GeneSequence.html#getIntronSequences()">getIntronSequences()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Get the introns as an ArrayList</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getInverse()">getInverse()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getInverse()">getInverse()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getInverse()">getInverse()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/RNASequence.html#getInverse()">getInverse()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>
<div class="block">Get the inverse view of the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getInverse()">getInverse()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getInverse()">getInverse()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getInverse()">getInverse()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getInverse()">getInverse()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getInverse()">getInverse()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Sequence.html#getInverse()">getInverse()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Does the <em>right thing</em> to get the inverse of the current Sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getInverse()">getInverse()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getIsoelectricPoint(org.biojava3.core.sequence.DNASequence)">getIsoelectricPoint(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getiTIS()">getiTIS()</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getJeffInfo()">getJeffInfo()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the different tables from Jeff summary table in 2011 and create a list
 of sRNAs from the different tables</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getJohanssonInfo()">getJohanssonInfo()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the table from Johansson 2009 data and create a list of sRNAs from the
 different tables</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html#getJolleyIDtoInfo()">getJolleyIDtoInfo()</a></span> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html" title="class in bacnet.scripts.phylogeny">GenomePhylogeny</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span></div>
<div class="block">Map Jolley ID to corresponding row in PATH_INFOTABLE</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeaturesKeyWordInterface.html#getKeyWords()">getKeyWords()</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getKeyWords()">getKeyWords()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">Pull uniprot key words which is a mixed bag of words associated with this
 sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getLabel_2()">getLabel_2()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getLabel_3()">getLabel_3()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getLastIndexOf(C)">getLastIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getLastIndexOf(C)">getLastIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getLastIndexOf(C)">getLastIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getLastIndexOf(C)">getLastIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getLastIndexOf(C)">getLastIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns the last occurrence of the given compound in this store; performs a
 linear search</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getLastIndexOf(C)">getLastIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getLastIndexOf(C)">getLastIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Returns the length of the Sequence if the given compound was equal to the one
 given during construction.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getLastIndexOf(C)">getLastIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Sequence.html#getLastIndexOf(C)">getLastIndexOf(C)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Scans through the Sequence looking for the last occurrence of the given
 compound</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getLastIndexOf(C)">getLastIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html#getLastIndexOf(C)">getLastIndexOf(C)</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava3.core.sequence.views">ComplementSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#getLastIndexOf(org.biojava3.core.sequence.compound.NucleotideCompound)">getLastIndexOf(NucleotideCompound)</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getLblDescribedIn()">getLblDescribedIn()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getLblGene()">getLblGene()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getLblNodiff()">getLblNodiff()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getLblOrAnyOther()">getLblOrAnyOther()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getLblOver()">getLblOver()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getLblOverProteomes()">getLblOverProteomes()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getLblPredictedSecondaryStructure()">getLblPredictedSecondaryStructure()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getLblStructure()">getLblStructure()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getLblTranscriptomesData()">getLblTranscriptomesData()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getLblUnder()">getLblUnder()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#getLeftBCs()">getLeftBCs()</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getLength()">getLength()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getLength()">getLength()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/CDSSequence.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ExonSequence.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/GeneSequence.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/IntronSequence.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">The sequence length</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Location.html#getLength()">getLength()</a></span> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Returns the length of the outer bounds of this location</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">Get the length of the MSA where it is assumed that all sequence position</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/StartCodonSequence.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/StartCodonSequence.html" title="class in org.biojava3.core.sequence">StartCodonSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/StopCodonSequence.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/StopCodonSequence.html" title="class in org.biojava3.core.sequence">StopCodonSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns the length of the sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Returns the length given during construction</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/LightweightProfile.html#getLength()">getLength()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a></dt>
<dd>
<div class="block">Returns the number of columns in the alignment profile.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Sequence.html#getLength()">getLength()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Returns the length of the Sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/WindowedSequence.html#getLength()">getLength()</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence</a></dt>
<dd>
<div class="block">Returns the size of the windowed sequence which is the length by the window
 size.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getLength(org.biojava3.core.sequence.DNASequence)">getLength(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#getLengthAA()">getLengthAA()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#getLengthGenome()">getLengthGenome()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Return the total length of the genome by adding all chromosomes lengths</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getLengthProt(org.biojava3.core.sequence.DNASequence)">getLengthProt(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html#getLineLength()">getLineLength()</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FastaWriter.html#getLineLength()">getLineLength()</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getLinkedBioCondition()">getLinkedBioCondition()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html#getList()">getList()</a></span> - Method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a></dt>
<dd>
<div class="block">Returns this resource as a list of Strings</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getList(java.io.BufferedReader)">getList(BufferedReader)</a></span> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Returns the contents of a buffered reader as a list of strings</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getList(java.io.File)">getList(File)</a></span> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Delegates to <a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getList(java.io.InputStream)"><code>IOUtils.getList(InputStream)</code></a> by wrapping the File in a valid
 stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getList(java.io.InputStream)">getList(InputStream)</a></span> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Delegates to <a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getList(java.io.BufferedReader)"><code>IOUtils.getList(BufferedReader)</code></a> by wrapping the InputStream in
 a valid reader.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getListData()">getListData()</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Read DATA_TABLe and return the list of dataset IDs</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getListData()">getListData()</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Read DATA_TABLe and return the list of dataset IDs</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getLISTDATA_COEXPR_NETWORK_TRANSCRIPTOMES_PATH()">getLISTDATA_COEXPR_NETWORK_TRANSCRIPTOMES_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Transcriptomes matrix data showing Log(Fold Change) values</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getListDatabases()">getListDatabases()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getListeriaGenomes()">getListeriaGenomes()</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Return a List of all Listeria genomes available</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getListeriaMonoGenome()">getListeriaMonoGenome()</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Return a List of all Listeria Monocytogenes genomes available</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getListeriaNonMonoGenome()">getListeriaNonMonoGenome()</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Return a List of all Listeria NOT Monocytogenes genomes available</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getListGenes()">getListGenes()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermUtils.html#getListOfFullORFfromPeptide(java.lang.String,bacnet.datamodel.sequence.Genome)">getListOfFullORFfromPeptide(String, Genome)</a></span> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>
<div class="block">From a list of peptide position retrieve the complete sequence of the ORF (in
 frame) containing the peptide (amino acid sequence) <br>
 WARNING: Be careful that your peptide info (begin,end) are correct.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getListSrnas()">getListSrnas()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getLocalization()">getLocalization()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getLocations()">getLocations()</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">A feature could be a single sequence position like a mutation or a post
 translational modification of an amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getLocations()">getLocations()</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">The location(s) of this feature where the location should contain a reference
 to parent and sequence etc.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#getLocusTag(java.lang.String)">getLocusTag(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>
<div class="block">Look in all LocusTagToGeneName exists</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#getLocusTagCodingList(boolean)">getLocusTagCodingList(boolean)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>
<div class="block">Put in a list all the CodingSequence found</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#getLocusTagKnownCodingList(boolean)">getLocusTagKnownCodingList(boolean)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#getLocusTagList(boolean)">getLocusTagList(boolean)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>
<div class="block">Put in a list all the locusTag found in the list of genes</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#getLocusTagNonCodingList(boolean)">getLocusTagNonCodingList(boolean)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getLocusTagToGeneNameMap()">getLocusTagToGeneNameMap()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getLocusTagToGeneNameMap()">getLocusTagToGeneNameMap()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getLOGFC_MATRIX_PROTEOMES_PATH()">getLOGFC_MATRIX_PROTEOMES_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Proteomes matrix data showing relative expression values</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getLOGFC_MATRIX_TRANSCRIPTOMES_PATH()">getLOGFC_MATRIX_TRANSCRIPTOMES_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Transcriptomes matrix data showing Log(Fold Change) values</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getLogfcOmicsMatrix()">getLogfcOmicsMatrix()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getLogFCTable()">getLogFCTable()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getLogFCTranscriptomesTable(java.lang.String)">getLogFCTranscriptomesTable(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getLogo()">getLogo()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getLogo(bacnet.datamodel.sequence.Sequence)">getLogo(Sequence)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">For a given Sequence, return the nucleotide sequence before and after
 starting position</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#getLongName()">getLongName()</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#getLongName()">getLongName()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Compound.html#getLongName()">getLongName()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#getLongName()">getLongName()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getLongName()">getLongName()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getMappingFrame()">getMappingFrame()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getMassSpecData()">getMassSpecData()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getMatrices()">getMatrices()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getMatrices()">getMatrices()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of SeqData given a bioCondName</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getMatrices(java.lang.String)">getMatrices(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of SeqData given a bioCondName</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getMatrix()">getMatrix()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#getMatrix()">getMatrix()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#getMatrix()">getMatrix()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/FilterList.html#getMatrix()">getMatrix()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#getMatrix()">getMatrix()</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#getMatrixDisplayed()">getMatrixDisplayed()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getMax()">getMax()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getMax()">getMax()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMax()">getMax()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMax()">getMax()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/LocationHelper.html#getMax(java.util.List)">getMax(List&lt;Location&gt;)</a></span> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Scans through a list of locations to find the Location with the highest end</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Location.Tools.html#getMax(java.util.List)">getMax(List&lt;Location&gt;)</a></span> - Static method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.Tools.html" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a></dt>
<dd>
<div class="block">Scans through a list of locations to find the Location with the highest end</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMaxColor()">getMaxColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMaxColor()">getMaxColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMaxColorSWT()">getMaxColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMaxColorSWT()">getMaxColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMaxPos()">getMaxPos()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMaxPos()">getMaxPos()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getMaxQuant()">getMaxQuant()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getMaxQuantum()">getMaxQuantum()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#getMaxSingleCompoundStringLength()">getMaxSingleCompoundStringLength()</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#getMaxSingleCompoundStringLength()">getMaxSingleCompoundStringLength()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/CompoundSet.html#getMaxSingleCompoundStringLength()">getMaxSingleCompoundStringLength()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></dt>
<dd>
<div class="block">Returns the maximum size of a compound String this set holds</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getMean()">getMean()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMean()">getMean()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMean()">getMean()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#getMean(java.util.ArrayList,java.lang.String)">getMean(ArrayList&lt;GeneExpression&gt;, String)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Calculate the mean expression of list of Tiling</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#getMean(java.util.ArrayList,java.lang.String)">getMean(ArrayList&lt;Tiling&gt;, String)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Calculate the mean expression of list of Tiling</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getMedia()">getMedia()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getMediaGrowthProperties()">getMediaGrowthProperties()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getMedian()">getMedian()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMedian()">getMedian()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMedian()">getMedian()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#getMedianValue(java.lang.String)">getMedianValue(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Get the median value of corresponding probe in double[][] values</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#getMedianVector()">getMedianVector()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Extract the column containing the median<br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#getMiddleH()">getMiddleH()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#getMiddleV()">getMiddleV()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMidLeftColor()">getMidLeftColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMidLeftColor()">getMidLeftColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMidLeftColorSWT()">getMidLeftColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMidLeftColorSWT()">getMidLeftColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMidLeftPos()">getMidLeftPos()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMidLeftPos()">getMidLeftPos()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMidRightColor()">getMidRightColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMidRightColor()">getMidRightColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMidRightColorSWT()">getMidRightColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMidRightColorSWT()">getMidRightColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMidRightPos()">getMidRightPos()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMidRightPos()">getMidRightPos()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getMin()">getMin()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getMin()">getMin()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMin()">getMin()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMin()">getMin()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/LocationHelper.html#getMin(java.util.List)">getMin(List&lt;Location&gt;)</a></span> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Scans through a list of locations to find the Location with the lowest start</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Location.Tools.html#getMin(java.util.List)">getMin(List&lt;Location&gt;)</a></span> - Static method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.Tools.html" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a></dt>
<dd>
<div class="block">Scans through a list of locations to find the Location with the lowest start</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMinColor()">getMinColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMinColor()">getMinColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMinColorSWT()">getMinColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMinColorSWT()">getMinColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getMinPos()">getMinPos()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getMinPos()">getMinPos()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getMissingValuesFreeMatrix()">getMissingValuesFreeMatrix()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Return a subMatrix containing only rows without Missingvalues</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getModifications()">getModifications()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#getModifs()">getModifs()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getModifSequence()">getModifSequence()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#getModifsString()">getModifsString()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#getMolecularWeight()">getMolecularWeight()</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#getMolecularWeight()">getMolecularWeight()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Compound.html#getMolecularWeight()">getMolecularWeight()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#getMolecularWeight()">getMolecularWeight()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getMolecularWeight()">getMolecularWeight()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#getMolMass()">getMolMass()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getMolMass(org.biojava3.core.sequence.DNASequence)">getMolMass(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getMouseXPosition()">getMouseXPosition()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getMouseXPosition()">getMouseXPosition()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getMouseYPosition()">getMouseYPosition()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getMouseYPosition()">getMouseYPosition()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getMraheilInfo()">getMraheilInfo()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the different tables from Mraheil et al. 2011 publication and create a
 list of sRNAs from the different tables</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html#getMultiFasta()">getMultiFasta()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html" title="class in bacnet.datamodel.sequence">Genome.GetMultiFastaThread</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html#%3Cinit%3E(java.util.HashMap)">GetMultiFastaThread(HashMap&lt;String, String&gt;)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html" title="class in bacnet.datamodel.sequence">Genome.GetMultiFastaThread</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getMULTIOMICS_MATRIX_PATH()">getMULTIOMICS_MATRIX_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Proteomes matrix data showing relative expression values</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getMULTIOMICS_PATH()">getMULTIOMICS_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getMutant()">getMutant()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#getName()">getName()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#getName()">getName()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#getName()">getName()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getName()">getName()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/Experiment.html#getName()">getName()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getName()">getName()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getName()">getName()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getName()">getName()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getName()">getName()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#getName()">getName()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getName()">getName()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#getName()">getName()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#getName()">getName()</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#getName()">getName()</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastOutput.html#getName(bacnet.scripts.blast.BlastOutput.BlastOutputTYPE)">getName(BlastOutput.BlastOutputTYPE)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.html" title="class in bacnet.scripts.blast">BlastOutput</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#getNameColumn()">getNameColumn()</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return the column index corresponding to Name</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getNameNTermData()">getNameNTermData()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#getNameRawData()">getNameRawData()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getNavigationListener()">getNavigationListener()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getNbElements()">getNbElements()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getNbElements()">getNbElements()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#getNbElementType()">getNbElementType()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#getNbSet()">getNbSet()</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/PhylogenyTools.html#getNCBIGenomePhylogeny()">getNCBIGenomePhylogeny()</a></span> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a></dt>
<dd>
<div class="block">From PhyloXMl of all bacteria, get only the genome which have a NCBI genome
 id</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getNcRNAs()">getNcRNAs()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getNCRNAUpstream(bacnet.datamodel.sequence.Gene)">getNCRNAUpstream(Gene)</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">For each gene search if the end of a smallRNA or cisReg element is in the
 region -200bp - start codon</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/XMLHelper.html#getNewDocument()">getNewDocument()</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#getNewLocusTag()">getNewLocusTag()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/RASTEGDeGenome.html#getNewRASTids()">getNewRASTids()</a></span> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/RASTEGDeGenome.html" title="class in bacnet.scripts.genome">RASTEGDeGenome</a></dt>
<dd>
<div class="block">Transform a list of old RAST ids into new RAST id</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html#getNext()">getNext()</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html" title="class in bacnet.scripts.core.vennDiagram">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/CombinationGenerator.html#getNext()">getNext()</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/CombinationGenerator.html" title="class in bacnet.utils">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getNextCodon()">getNextCodon()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getNextNterm(bacnet.datamodel.proteomics.NTerm,java.lang.String)">getNextNterm(NTerm, String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Given a NTerm, find the next one in the Annotation</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getNGSSeqs()">getNGSSeqs()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getNoDNAList()">getNoDNAList()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getNonCodingRegion(java.lang.String)">getNonCodingRegion(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>
<div class="block">Get the gene based on accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getNoncodingSequenceHashMap()">getNoncodingSequenceHashMap()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#getNonCodingSequencesList(boolean)">getNonCodingSequencesList(boolean)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>
<div class="block">Put in a list all the NonCodingSequence found</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getNoProtList()">getNoProtList()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getNoRNAList()">getNoRNAList()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getNotAssociatedList(java.lang.String,boolean)">getNotAssociatedList(String, boolean)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Read list of gene not having been associated (RAST and NCBI association)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getNotDiffExpressedASrnas()">getNotDiffExpressedASrnas()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getNotDiffExpressedCisRegs()">getNotDiffExpressedCisRegs()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getNotDiffExpressedGenes()">getNotDiffExpressedGenes()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#getNotDiffExpresseds()">getNotDiffExpresseds()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getNotDiffExpressedSrnas()">getNotDiffExpressedSrnas()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#getNote()">getNote()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#getNote()">getNote()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getNote(org.biojava3.core.sequence.DNASequence)">getNote(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getNotesList()">getNotesList()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getNotListeriaAndBacillusGenome()">getNotListeriaAndBacillusGenome()</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Return a list of all NON Listeria genomes</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getNotListeriaGenome()">getNotListeriaGenome()</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Return a list of all NON Listeria genomes</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getNTermDatas()">getNTermDatas()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of NTermData given a bioCondName<br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getNTermDatas(java.lang.String)">getNTermDatas(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of NTermData given a bioCondName<br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermFilter.html#getnTermExp()">getnTermExp()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getnTermHighlight()">getnTermHighlight()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#getnTermRef()">getnTermRef()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getnTerms()">getnTerms()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#getnTerms()">getnTerms()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getNTerms()">getNTerms()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Return a list of all NTerms available</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#getnTermsSequences()">getnTermsSequences()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>
<div class="block">Return a character string with all peptide Modif sequences</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#getnTermsString()">getnTermsString()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>
<div class="block">Return a character string with all peptide names</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getnTermTypeOverlap()">getnTermTypeOverlap()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/CompoundColorization.html#getNucleotideColor(java.lang.String,org.eclipse.swt.widgets.Display)">getNucleotideColor(String, Display)</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/CompoundColorization.html" title="class in bacnet.datamodel.sequence">CompoundColorization</a></dt>
<dd>
<div class="block">Given a Nucleotide, this method return the corresponding color</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getNumberColumn()">getNumberColumn()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Get the number of column of the <code>values</code> array</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getNumberColumnWithAnnotation()">getNumberColumnWithAnnotation()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Get the number of column of the <code>values</code> array +
 <code>annotations</code> array</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getNumberRow()">getNumberRow()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/Strand.html#getNumericRepresentation()">getNumericRepresentation()</a></span> - Method in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html#getNumLeft()">getNumLeft()</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html" title="class in bacnet.scripts.core.vennDiagram">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/CombinationGenerator.html#getNumLeft()">getNumLeft()</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/CombinationGenerator.html" title="class in bacnet.utils">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#getOldLocusTag()">getOldLocusTag()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getOliverInfo()">getOliverInfo()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the different tables from Oliver et al. 2009 publication and create a
 list of sRNAs from the different tables</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getOliverInfoEGDePosition()">getOliverInfoEGDePosition()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the different tables from Oliver et al. 2009 publication and create a
 list of sRNAs from the different tables</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getOmicsData()">getOmicsData()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Put all OmicsData in a list</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#getOne()">getOne()</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#getOne()">getOne()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getOne()">getOne()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#getOperon()">getOperon()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getOperon(org.biojava3.core.sequence.DNASequence)">getOperon(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getOperons()">getOperons()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getOperons()">getOperons()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getOrderedRowNames()">getOrderedRowNames()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getOrganismName()">getOrganismName()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">Get the organism name assigned to this sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getOriginalHeader()">getOriginalHeader()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getOverASrnas()">getOverASrnas()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#getOverBioConds()">getOverBioConds()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getOverCisRegs()">getOverCisRegs()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getOverExpressed()">getOverExpressed()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getOverGenes()">getOverGenes()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getOverlap()">getOverlap()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getOverlap(java.lang.String,java.lang.String,java.lang.String)">getOverlap(String, String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#getOverlapingGene()">getOverlapingGene()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getOverlapList(java.util.ArrayList,java.lang.String,java.lang.String,java.lang.String)">getOverlapList(ArrayList&lt;String&gt;, String, String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Read NCBI and RAST genomes and look at overlapping elements</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#getOverlaps()">getOverlaps()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#getOverlapsString()">getOverlapsString()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getOverSrnas()">getOverSrnas()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/GeneSequence.html#getParentChromosomeSequence()">getParentChromosomeSequence()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">The parent ChromosomeSequence which contains the actual DNA sequence data</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getParentFeature()">getParentFeature()</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">Get the parent Feature</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getParentFeature()">getParentFeature()</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Get the parent feature</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getParentGenome()">getParentGenome()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getParentGenome()">getParentGenome()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>
<div class="block">Get the gene based on accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getParentSequence()">getParentSequence()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getParentViewId()">getParentViewId()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getPartService()">getPartService()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getPath()">getPath()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/FileUtils.html#getPath(java.lang.String)">getPath(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Get only the folder where the file is contained</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getPATH()">getPATH()</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermUtils.html#getPATH()">getPATH()</a></span> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getPATH()">getPATH()</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getPDBCharacter(boolean,char,char,boolean,char)">getPDBCharacter(boolean, char, char, boolean, char)</a></span> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Creates formatted String for a single character of PDB output</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getPDBConservation(boolean,char,char,boolean)">getPDBConservation(boolean, char, char, boolean)</a></span> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Creates formatted String for displaying conservation in PDB output</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#getPDBLegend()">getPDBLegend()</a></span> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Creates formatted String for displaying conservation legend in PDB output</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getPeptides()">getPeptides()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Link peptideSequence to <code>ArrayList(NTerm)</code><br>
 This correspond to the list of <code>NTerm</code> having same peptideSequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/CDSSequence.html#getPhase()">getPhase()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneViewHomologTools.html#getPhyloFigure(bacnet.datamodel.sequence.Gene,java.util.ArrayList)">getPhyloFigure(Gene, ArrayList&lt;String&gt;)</a></span> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneViewHomologTools.html" title="class in bacnet.sequenceTools">GeneViewHomologTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GenomicsView.html#getPhyloFigure(java.util.ArrayList)">getPhyloFigure(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>
<div class="block">Return the text file of the SVG figure, modified with highlighted strains</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/COGannotation.html#getPieChartData(java.util.HashMap)">getPieChartData(HashMap&lt;String, ArrayList&lt;String&gt;&gt;)</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>
<div class="block">Transform the result of getCogClassification() method into an
 ExpressionMatrix which can be load by PieChartView for display</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/GlaserFCannotation.html#getPieChartData(java.util.HashMap)">getPieChartData(HashMap&lt;String, ArrayList&lt;String&gt;&gt;)</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/GlaserFCannotation.html" title="class in bacnet.datamodel.annotation">GlaserFCannotation</a></dt>
<dd>
<div class="block">Transform the result of getGlaserFCclassification() method into an
 ExpressionMatrix which can be load by PieChartView for display</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#getPluginFileURL(java.lang.String,java.lang.String)">getPluginFileURL(String, String)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>URL</code> based on a <code>Bundle</code> and resource entry path.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#getPluginImage(java.lang.Object,java.lang.String)">getPluginImage(Object, String)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span>
<div class="deprecationComment">Use <a href="../bacnet/swt/ResourceManager.html#getPluginImage(java.lang.String,java.lang.String)"><code>ResourceManager.getPluginImage(String, String)</code></a> instead.</div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#getPluginImage(java.lang.String,java.lang.String)">getPluginImage(String, String)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> based on a <code>Bundle</code> and resource entry path.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#getPluginImageDescriptor(java.lang.Object,java.lang.String)">getPluginImageDescriptor(Object, String)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span>
<div class="deprecationComment">Use <a href="../bacnet/swt/ResourceManager.html#getPluginImageDescriptor(java.lang.String,java.lang.String)"><code>ResourceManager.getPluginImageDescriptor(String, String)</code></a> instead.</div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#getPluginImageDescriptor(java.lang.String,java.lang.String)">getPluginImageDescriptor(String, String)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>ImageDescriptor</code> based on a <code>Bundle</code> and resource
 entry path.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#getPluginImageURL(java.lang.Object,java.lang.String)">getPluginImageURL(Object, String)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>URL</code> based on a plugin and file path.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/HTMLUtils.html#getPluginTextFile(java.lang.String,java.lang.String)">getPluginTextFile(String, String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/HTMLUtils.html" title="class in bacnet.utils">HTMLUtils</a></dt>
<dd>
<div class="block">Returns an <code>String</code> based on a <code>Bundle</code> and resource entry path.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#getPosition()">getPosition()</a></span> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimplePoint.html#getPosition()">getPosition()</a></span> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Point.html#getPosition()">getPosition()</a></span> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a></dt>
<dd>
<div class="block">Returns the position held by this object</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#getPosition(int)">getPosition(int)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Return the index position in the tiling matrix of a particular probes closest
 to bpPosition</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getPreviousCodon()">getPreviousCodon()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getPreviousNterm(bacnet.datamodel.proteomics.NTerm,java.lang.String)">getPreviousNterm(NTerm, String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Given a NTerm, find the previous one in the Annotation</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#getProbes()">getProbes()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#getProbes()">getProbes()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getProbesGExpression()">getProbesGExpression()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Read probes for GExpression in GeneExpression.PROBES_PATH</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getProbesTiling()">getProbesTiling()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Read probes for Tiling in Tiling.PROBES_PATH</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#getProduct()">getProduct()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/NcRNA.html#getProduct()">getProduct()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getProduct(org.biojava3.core.sequence.DNASequence)">getProduct(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getProjectName()">getProjectName()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#getProperties()">getProperties()</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>
<div class="block">Get the properties</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#getProtein_id()">getProtein_id()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getProteinCDSSequences()">getProteinCDSSequences()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>
<div class="block">Return a list of protein sequences based on each CDS sequence where the phase
 shift between two CDS sequences is assigned to the CDS sequence that starts
 the triplet.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getProteinID(org.biojava3.core.sequence.DNASequence)">getProteinID(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#getProteinList(java.lang.String)">getProteinList(String)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Return the list of all protein id present in the genome given</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#getProteinNCBILink(java.lang.String)">getProteinNCBILink(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/RNASequence.html#getProteinSequence()">getProteinSequence()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>
<div class="block">Get the ProteinSequence from the RNA sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getProteinSequence()">getProteinSequence()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>
<div class="block">Get the protein sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/RNASequence.html#getProteinSequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)">getProteinSequence(TranscriptionEngine)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>
<div class="block">Get the ProteinSequene from the RNA sequence with user defined transcription
 engine</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getProteinSequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)">getProteinSequence(TranscriptionEngine)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>
<div class="block">Get the protein sequence with user defined TranscriptEngine</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getProteinSequenceCreator()">getProteinSequenceCreator()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#getProteinSequenceFromFasta(java.lang.String,java.lang.String)">getProteinSequenceFromFasta(String, String)</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Read all accessible FAA files to extract protein sequence of a gene</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html#getProteinSequences()">getProteinSequences()</a></span> - Method in class org.biojava3.genome.parsers.geneid.<a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html" title="class in org.biojava3.genome.parsers.geneid">GeneIDXMLReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/GeneFeatureHelper.html#getProteinSequences(java.util.Collection)">getProteinSequences(Collection&lt;ChromosomeSequence&gt;)</a></span> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getProteolyseAssociated(bacnet.datamodel.proteomics.NTerm)">getProteolyseAssociated(NTerm)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Go through <code>proteolyses</code> map, and find the list of NTerm which
 might be associated to an input <code>NTerm</code><br>
 Warning: <code>ArrayList</code> in return contains <code>key NTerm</code> of
 <code>proteolyses</code> but not the input <code>nTerm</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getProteomeGenomes()">getProteomeGenomes()</a></span> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Get all genomes associated to a proteomics dataset</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getProteomes()">getProteomes()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getProteomes()">getProteomes()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of MassSpecData given a bioCondName<br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DataValidation.html#getProteomes()">getProteomes()</a></span> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getProteomes(java.lang.String)">getProteomes(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of MassSpecData given a bioCondName<br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getPROTEOMES_PATH()">getPROTEOMES_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getProteomesArrayPath()">getProteomesArrayPath()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getProteomicsData()">getProteomicsData()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Put all OmicsData in a list</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/AccesionedLocation.html#getProxySequenceReader()">getProxySequenceReader()</a></span> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AccesionedLocation.html" title="interface in org.biojava3.core.sequence.location.template">AccesionedLocation</a></dt>
<dd>
<div class="block">Return the proxy reader used to get sequence for this location.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getProxySequenceReader()">getProxySequenceReader()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#getPubMedLink(java.lang.String)">getPubMedLink(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/QualityFeature.html#getQualities()">getQualities()</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/QualityFeature.html" title="class in org.biojava3.core.sequence.features">QualityFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/QualityFeature.html#getQualities(int,int)">getQualities(int, int)</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/QualityFeature.html" title="class in org.biojava3.core.sequence.features">QualityFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/QualityFeature.html#getQualityAt(int)">getQualityAt(int)</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/QualityFeature.html" title="class in org.biojava3.core.sequence.features">QualityFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/QuantityFeature.html#getQuantities()">getQuantities()</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/QuantityFeature.html" title="class in org.biojava3.core.sequence.features">QuantityFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#getRASTinfo()">getRASTinfo()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getRASTinfo(org.biojava3.core.sequence.DNASequence)">getRASTinfo(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#getRawDatas()">getRawDatas()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getRead()">getRead()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorSWT.html#getRed()">getRed()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getRef()">getRef()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#getReference()">getReference()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getReference()">getReference()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#getRefSequence()">getRefSequence()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermTranscriptome.html#getRegulatoryRegion()">getRegulatoryRegion()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermTranscriptome.html" title="class in bacnet.scripts.listeriomics.nterm">NTermTranscriptome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getRelevantSubLocations()">getRelevantSubLocations()</a></span> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>
<div class="block">Returns the normalised list of sub locations i.e. only those locations which
 do not have a sub location.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Location.html#getRelevantSubLocations()">getRelevantSubLocations()</a></span> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">An extension to <a href="../org/biojava3/core/sequence/location/template/Location.html#getSubLocations()"><code>Location.getSubLocations()</code></a> which returns sub-locations of
 sub-locations; this will continue until it runs out of those locations.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getRelevantSubSequence(org.biojava3.core.sequence.template.Sequence)">getRelevantSubSequence(Sequence&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Location.html#getRelevantSubSequence(org.biojava3.core.sequence.template.Sequence)">getRelevantSubSequence(Sequence&lt;C&gt;)</a></span> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Will return a SequenceReader object which offers a view of all resolved
 locations i.e. those locations which are not complex and define the true
 Sequence represented</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastResult.html#getResultsFromXML(java.lang.String)">getResultsFromXML(String)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>
<div class="block">From an XML blast file, extract blast results</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#getReverse()">getReverse()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Returns a Sequence which runs in the current reverse order</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/Strand.html#getReverse()">getReverse()</a></span> - Method in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#getReverseComplement()">getReverseComplement()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Delegates to <a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getInverse()"><code>AbstractSequence.getInverse()</code></a> for the reverse complement</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/RNASequence.html#getReverseComplement()">getReverseComplement()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></dt>
<dd>
<div class="block">Get reverse complement view of the sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Frame.html#getReverseFrames()">getReverseFrames()</a></span> - Static method in enum org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></dt>
<dd>
<div class="block">Returns all frames which are in the reverse orientation</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getRfamAvailableGenome()">getRfamAvailableGenome()</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Read RFAM_GENOME table, and extract a list of all genomes which has both
 Rfam_ID and NCBI_Id</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#getRfamDirectory()">getRfamDirectory()</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>
<div class="block">Project.getANNOTATION_PATH()+"RfamData"+File.separator;</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#getRfamElementsFromGenome(java.lang.String)">getRfamElementsFromGenome(String)</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>
<div class="block">Return the list of Rfam element in a particular Genome<br>
 <br>
 
 Search for all NCBI GFF in a particular genome folder (e.g. genomeName),<br>
 thanks to RFAM_GENOME table it found corresponding Rfam GFF file if it
 exists<br>
 and create HashMap<String, RfamElement> rfamList</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#getRfamElementsFromGFF(java.lang.String)">getRfamElementsFromGFF(String)</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>
<div class="block">Read a Gff from Rfam and get all possible information</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#getRfamElementsFromGFF(java.lang.String,java.util.HashMap)">getRfamElementsFromGFF(String, HashMap&lt;String, RfamElement&gt;)</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>
<div class="block">Read a Gff from Rfam and add the information to an existing HashMap</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#getRfamGenomeId(java.lang.String)">getRfamGenomeId(String)</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>
<div class="block">Given an NCBI Gff file name, this function search in RFAM_GENOME table and
 return Rfam Gff file name if it exists</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#getRfamID()">getRfamID()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getRfamInfo()">getRfamInfo()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the different tables from Rfam 2011 data and create a list of sRNAs from
 the different tables</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getRiboSeqDatas()">getRiboSeqDatas()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of SeqDatas</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getRiboSeqDatas(java.lang.String)">getRiboSeqDatas(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of RiboSeqData given a bioCondName</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#getRightBCs()">getRightBCs()</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getRnaAminoAcidTranslator()">getRnaAminoAcidTranslator()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getRnaCompounds()">getRnaCompounds()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/RNACompoundSet.html#getRNACompoundSet()">getRNACompoundSet()</a></span> - Static method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/RNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">RNACompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getRNASeqDatas()">getRNASeqDatas()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of SeqDatas</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getRNASeqDatas(java.lang.String)">getRNASeqDatas(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of RNASeqData given a bioCondName</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/Experiment.html#getRNASeqs()">getRNASeqs()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Return all NGS data contains in all BioCondition</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#getRNASequence()">getRNASequence()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Return the RNASequence equivalent of the DNASequence using default
 Transcription Engine.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.Frame)">getRNASequence(Frame)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Allows the user to pass in the Frame shift.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)">getRNASequence(TranscriptionEngine)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Allow a user to pass in a rules engine to do the DNA to RNA translation</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine,org.biojava3.core.sequence.transcription.Frame)">getRNASequence(TranscriptionEngine, Frame)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getRnaSequenceCreator()">getRnaSequenceCreator()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#getRnaToDna()">getRnaToDna()</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#getRow(double%5B%5D%5B%5D,int)">getRow(double[][], int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#getRow(int%5B%5D%5B%5D,int)">getRow(int[][], int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getRow(java.lang.String)">getRow(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#getRow(java.lang.String%5B%5D%5B%5D,int)">getRow(String[][], int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Extract a Row from a Matrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getRowName(int)">getRowName(int)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getRowNameCellColor()">getRowNameCellColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getRowNames()">getRowNames()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getRowNamesToList()">getRowNamesToList()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getRowNameTextColor()">getRowNameTextColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Rscript.html#getRscriptDirectory()">getRscriptDirectory()</a></span> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Rscript.html" title="class in bacnet.scripts.core">Rscript</a></dt>
<dd>
<div class="block">Test if the R script directory has been given, and open a dialog to update it
 in case</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getSameEndList(java.lang.String,boolean)">getSameEndList(String, boolean)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Read list of gene having same end position but different starting point</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#getScore()">getScore()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getScore()">getScore()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getScrolledComposite()">getScrolledComposite()</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getScrolledComposite()">getScrolledComposite()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getSDEnergy(bacnet.datamodel.sequence.Gene)">getSDEnergy(Gene)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Return the binding energy with aSD sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getSDRFCuratedPath(java.lang.String)">getSDRFCuratedPath(String)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Get path of curated SDRF file for a specific dataID (ex: E-MTAB-1800)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getSDRFCuratedPath(java.lang.String)">getSDRFCuratedPath(String)</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Get path of curated SDRF file for a specific dataID (ex: E-MTAB-1800)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getSDRFPath(java.lang.String)">getSDRFPath(String)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Get path of SDRF file for a specific dataID (ex: E-MTAB-1800)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getSDRFPath(java.lang.String)">getSDRFPath(String)</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Get path of SDRF file for a specific dataID (ex: E-MTAB-1800)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getSDSequence()">getSDSequence()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Get the nucleotide sequence 20bp upstream to the <code>Sequence</code>, +3bp
 for including start codon</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getSDSequence(int)">getSDSequence(int)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Get the nucleotide sequence n bp upstream to the <code>Sequence</code>, +3bp
 for including start codon</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getSecondRowName()">getSecondRowName()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getSelectedComparisons()">getSelectedComparisons()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Return the list of comparisons which have been selected</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getSelectedComparisons()">getSelectedComparisons()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>
<div class="block">Return the list of comparisons which have been selected</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#getSelectedGenes()">getSelectedGenes()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/gui/BlastGenomeSelectionPage.html#getSelectedGenomes()">getSelectedGenomes()</a></span> - Method in class bacnet.scripts.blast.gui.<a href="../bacnet/scripts/blast/gui/BlastGenomeSelectionPage.html" title="class in bacnet.scripts.blast.gui">BlastGenomeSelectionPage</a></dt>
<dd>
<div class="block">Get all genomes which have been selected</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getSeq()">getSeq()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getSeqDatas(java.lang.String)">getSeqDatas(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of RNASeqData given a bioCondName</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/SequenceTools.html#getSeqeunceFromString(java.lang.String)">getSeqeunceFromString(String)</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SequenceTools.html" title="class in org.biojava3.core.util">SequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getSequence()">getSequence()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Read chromosome and extract the corresponding sequence as a String</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getSequence()">getSequence()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#getSequence()">getSequence()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getSequence()">getSequence()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SequenceLocation.html#getSequence()">getSequence()</a></span> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SequenceLocation.html" title="class in org.biojava3.core.sequence.location">SequenceLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FastaSequenceParser.html#getSequence(java.io.DataInput,int)">getSequence(DataInput, int)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaSequenceParser.html" title="class in org.biojava3.core.sequence.io">FastaSequenceParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/template/SequenceParserInterface.html#getSequence(java.io.DataInput,int)">getSequence(DataInput, int)</a></span> - Method in interface org.biojava3.core.sequence.io.template.<a href="../org/biojava3/core/sequence/io/template/SequenceParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceParserInterface</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getSequence(java.lang.String)">getSequence(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>
<div class="block">Given a sequence name, return the corresponding Sequence<br>
 If name is a Gene name, we search also in geneNameToLocusTagMap</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html#getSequence(java.lang.String,long)">getSequence(String, long)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html#getSequence(java.lang.String,long)">getSequence(String, long)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">DNASequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html#getSequence(java.lang.String,long)">getSequence(String, long)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyDNASequenceCreator</a></dt>
<dd>
<div class="block">Even though we are passing in the sequence we really only care about the
 length of the sequence and the offset index in the fasta file.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html#getSequence(java.lang.String,long)">getSequence(String, long)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyProteinSequenceCreator</a></dt>
<dd>
<div class="block">Even though we are passing in the sequence we really only care about the
 length of the sequence and the offset index in the fasta file.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html#getSequence(java.lang.String,long)">getSequence(String, long)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">ProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/RNASequenceCreator.html#getSequence(java.lang.String,long)">getSequence(String, long)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/RNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">RNASequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html#getSequence(java.lang.String,long)">getSequence(String, long)</a></span> - Method in interface org.biojava3.core.sequence.io.template.<a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html#getSequence(java.util.List)">getSequence(List&lt;C&gt;)</a></span> - Method in interface org.biojava3.core.sequence.io.template.<a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html#getSequence(java.util.List)">getSequence(List&lt;AminoAcidCompound&gt;)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></dt>
<dd>
<div class="block">Assumes all compounds were uppercase</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html#getSequence(java.util.List)">getSequence(List&lt;AminoAcidCompound&gt;)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyProteinSequenceCreator</a></dt>
<dd>
<div class="block">Not sure of use case and currently not supported</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html#getSequence(java.util.List)">getSequence(List&lt;AminoAcidCompound&gt;)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">ProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html#getSequence(java.util.List)">getSequence(List&lt;NucleotideCompound&gt;)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">DNASequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html#getSequence(java.util.List)">getSequence(List&lt;NucleotideCompound&gt;)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyDNASequenceCreator</a></dt>
<dd>
<div class="block">Not sure of use case and currently not supported</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/RNASequenceCreator.html#getSequence(java.util.List)">getSequence(List&lt;NucleotideCompound&gt;)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/RNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">RNASequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence(ProxySequenceReader&lt;C&gt;, long)</a></span> - Method in interface org.biojava3.core.sequence.io.template.<a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence(ProxySequenceReader&lt;AminoAcidCompound&gt;, long)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence(ProxySequenceReader&lt;AminoAcidCompound&gt;, long)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyProteinSequenceCreator</a></dt>
<dd>
<div class="block">Should be able to extend the same concept to a remote URL call or database
 connection.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence(ProxySequenceReader&lt;AminoAcidCompound&gt;, long)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">ProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence(ProxySequenceReader&lt;NucleotideCompound&gt;, long)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">DNASequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence(ProxySequenceReader&lt;NucleotideCompound&gt;, long)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyDNASequenceCreator</a></dt>
<dd>
<div class="block">Should be able to extend the same concept to a remote URL call or database
 connection.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/RNASequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence(ProxySequenceReader&lt;NucleotideCompound&gt;, long)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/RNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">RNASequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/GeneSequence.html#getSequence5PrimeTo3Prime()">getSequence5PrimeTo3Prime()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Try to give method clarity where you want a DNASequence coding in the 5' to
 3' direction Returns the DNASequence representative of the 5' and 3' reading
 based on strand</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getSequenceAA()">getSequenceAA()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Get Amino Acid sequence of a Sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getSequenceAACurated()">getSequenceAACurated()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">When an Amino Acid sequence contains Stop codon ('*'), this method search the
 first start codon and remove sequence before, and then create different
 sequence, from one stop codon to the other.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getSequenceAACurated(java.lang.String)">getSequenceAACurated(String)</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">When an Amino Acid sequence contains Stop codon ('*'), this method search the
 first start codon and remove sequence before, and then create different
 sequence, from one stop codon to the other.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getSequenceAsString()">getSequenceAsString()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getSequenceAsString()">getSequenceAsString()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getSequenceAsString()">getSequenceAsString()</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getSequenceAsString()">getSequenceAsString()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getSequenceAsString()">getSequenceAsString()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns the sequence as a String</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getSequenceAsString()">getSequenceAsString()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getSequenceAsString()">getSequenceAsString()</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Delegates to
 <a href="../org/biojava3/core/sequence/template/SequenceMixin.html#toList(org.biojava3.core.sequence.template.Sequence)"><code>SequenceMixin.toList(org.biojava3.core.sequence.template.Sequence)</code></a></div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceAsString()">getSequenceAsString()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Default case is to assume strand is positive because only CDSSequence can be
 either positive or negative Strand.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Sequence.html#getSequenceAsString()">getSequenceAsString()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Returns the String representation of the Sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getSequenceAsString()">getSequenceAsString()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html#getSequenceAsString()">getSequenceAsString()</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava3.core.sequence.views">ComplementSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/ReversedSequenceView.html#getSequenceAsString()">getSequenceAsString()</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views">ReversedSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#getSequenceAsString()">getSequenceAsString()</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)">getSequenceAsString(Integer, Integer, Strand)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)">getSequenceAsString(Integer, Integer, Strand)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)">getSequenceAsString(Integer, Integer, Strand)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)">getSequenceAsString(Integer, Integer, Strand)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)">getSequenceAsString(Integer, Integer, Strand)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html#getSequenceAsString(java.util.List,org.biojava3.core.sequence.template.CompoundSet,java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)">getSequenceAsString(List&lt;C&gt;, CompoundSet&lt;C&gt;, Integer, Integer, Strand)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="class in org.biojava3.core.sequence.storage">SequenceAsStringHelper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html#getSequenceCollection()">getSequenceCollection()</a></span> - Static method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getSequenceMap()">getSequenceMap()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getSequencePeptide()">getSequencePeptide()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getSequenceRNA()">getSequenceRNA()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Return RNA sequence of the <code>Sequence</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceScore()">getSequenceScore()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Provide place holder for a metric that indicate a score associated with the
 sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html#getSequenceUsage()">getSequenceUsage()</a></span> - Static method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#getSerialversionuid()">getSerialversionuid()</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getSerialversionuid()">getSerialversionuid()</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getSerialversionuid()">getSerialversionuid()</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#getSerialversionuid()">getSerialversionuid()</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/Experiment.html#getSerialversionuid()">getSerialversionuid()</a></span> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getSerialversionuid()">getSerialversionuid()</a></span> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#getSerialversionuid()">getSerialversionuid()</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/NcRNA.html#getSerialversionuid()">getSerialversionuid()</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getSerialversionuid()">getSerialversionuid()</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#getSerialversionuid()">getSerialversionuid()</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#getSerialversionuid()">getSerialversionuid()</a></span> - Static method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getSerialversionuid()">getSerialversionuid()</a></span> - Static method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#getSetNames()">getSetNames()</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#getSets()">getSets()</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getShell()">getShell()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getShortDescription()">getShortDescription()</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getShortDescription()">getShortDescription()</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Get the short description that can be used to describe the feature</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#getShortName()">getShortName()</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html#getShortName()">getShortName()</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#getShortName()">getShortName()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Compound.html#getShortName()">getShortName()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#getShortName()">getShortName()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getShortName()">getShortName()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getShowInfo()">getShowInfo()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/SignatureSelectionDialog.html#getSignature()">getSignature()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/SignatureSelectionDialog.html" title="class in bacnet.expressionAtlas.core">SignatureSelectionDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#getSignatureFromName(java.lang.String)">getSignatureFromName(String)</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>
<div class="block">Load serialized signature</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getSignatures()">getSignatures()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getSIGNATURES_PATH()">getSIGNATURES_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#getSignaturesFromTextFiles()">getSignaturesFromTextFiles()</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>
<div class="block">Read all signatures in txt format from the folder Database.getSIGNATURES()</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getSignaturesNametoID()">getSignaturesNametoID()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getSignaturesToString()">getSignaturesToString()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#getSize()">getSize()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#getSize()">getSize()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">Get the number of sequences in the MSA</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/LightweightProfile.html#getSize()">getSize()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a></dt>
<dd>
<div class="block">Returns the number of rows in this profile.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getSliderHbar()">getSliderHbar()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getSliderVBar()">getSliderVBar()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getSource()">getSource()</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">The feature source</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getSource()">getSource()</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">The source of the feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getSource()">getSource()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Added support for the source of this sequence for GFF3 export If a sub
 sequence doesn't have source then check for parent source</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getSpecies()">getSpecies()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#getSpecies()">getSpecies()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#getSpecies()">getSpecies()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#getSpecies()">getSpecies()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getSpectra()">getSpectra()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getSpectrum()">getSpectrum()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getsRNAListEGDe()">getsRNAListEGDe()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#getSrnaOrder()">getSrnaOrder()</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Return an ArrayList containing Srna names in the proper order<br>
 The list is load from PATH_LISTSRNA</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#getsRNAs()">getsRNAs()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getsRNAs()">getsRNAs()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html#getStart()">getStart()</a></span> - Method in class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="class in org.biojava3.core.sequence.edits">Edit.AbstractEdit</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getStart()">getStart()</a></span> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Location.html#getStart()">getStart()</a></span> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Start of the location</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getStartCode()">getStartCode()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getStartCodon()">getStartCodon()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getStartCodon(bacnet.datamodel.sequence.Gene)">getStartCodon(Gene)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Return start codon proportion in the genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermUtils.html#getStartCodonBefore()">getStartCodonBefore()</a></span> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>
<div class="block">NOT FINISHED <br>
 <br>
 Search for the different start codon before a peptide, and the stop codon
 after this peptide<br>
 Print the nucleotide sequence plus the codon, and identify Start, STOp and
 the reference Peptide used</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getStartCodonSequence()">getStartCodonSequence()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getStat()">getStat()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Sysout some statistics about the number of aTIS, dTISM, dTIS, and uTIS</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getSTAT_MATRIX_PROTEOMES_PATH()">getSTAT_MATRIX_PROTEOMES_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Proteomes matrix data showing Fold Change values</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getSTAT_MATRIX_TRANSCRIPTOMES_PATH()">getSTAT_MATRIX_TRANSCRIPTOMES_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for loading Transcriptomes matrix data showing statistical p-values</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getStatTable()">getStatTable()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getStopCodonSequence()">getStopCodonSequence()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#getStrand()">getStrand()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getStrand()">getStrand()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#getStrand()">getStrand()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/CDSSequence.html#getStrand()">getStrand()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/GeneSequence.html#getStrand()">getStrand()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">A gene should have Strand</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getStrand()">getStrand()</a></span> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Location.html#getStrand()">getStrand()</a></span> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Strand which the location is located on</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/TranscriptSequence.html#getStrand()">getStrand()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getStrand(org.biojava3.core.sequence.DNASequence)">getStrand(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getStrandHeaders(boolean)">getStrandHeaders(boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Return a list of the data contain in this matrix which correspond to a given
 strand</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#getStringForCompound(C)">getStringForCompound(C)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/CompoundSet.html#getStringForCompound(C)">getStringForCompound(C)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#getStringForCompound(org.biojava3.core.sequence.compound.AminoAcidCompound)">getStringForCompound(AminoAcidCompound)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/Strand.html#getStringRepresentation()">getStringRepresentation()</a></span> - Method in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getStudyName()">getStudyName()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#getSubCodingSequencesList(java.util.ArrayList,boolean)">getSubCodingSequencesList(ArrayList&lt;String&gt;, boolean)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>
<div class="block">Put in a sublist all the CodingSequence found in locuTag list</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getSubLocations()">getSubLocations()</a></span> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Location.html#getSubLocations()">getSubLocations()</a></span> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Gives access to the sub locations for this location.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getSubMatrix(java.util.ArrayList,java.util.ArrayList)">getSubMatrix(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getSubMatrixColumn(java.util.ArrayList)">getSubMatrixColumn(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Extract a matrix with only selected columns</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getSubMatrixRow(java.util.ArrayList)">getSubMatrixRow(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Provide a list of rowName and extract them from the ExpressionMatrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getSubMatrixRow(java.util.TreeSet)">getSubMatrixRow(TreeSet&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Provide a list of rowName and extract them from the ExpressionMatrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getSubMatrixRowRandom(double)">getSubMatrixRowRandom(double)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Generate a subMatrix with random rows</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getSubRow(java.lang.String,java.util.ArrayList)">getSubRow(String, ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getSubSequence(int,int)">getSubSequence(int, int)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Returns a sub sequence view</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence(Integer, Integer)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence(Integer, Integer)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence(Integer, Integer)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence(Integer, Integer)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence(Integer, Integer)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence(Integer, Integer)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence(Integer, Integer)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Creates a <a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template"><code>SequenceProxyView</code></a> for the given coordinates</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence(Integer, Integer)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Sequence.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence(Integer, Integer)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Returns a portion of the sequence from the different positions.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence(Integer, Integer)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#getSubSequence(org.biojava3.core.sequence.template.Sequence)">getSubSequence(Sequence&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>
<div class="block">If circular this will return the sequence represented by the sub locations
 joined.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Location.html#getSubSequence(org.biojava3.core.sequence.template.Sequence)">getSubSequence(Sequence&lt;C&gt;)</a></span> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Will return a SequenceReader object which represents the outer bounds of this
 Location</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#getSubsetNames()">getSubsetNames()</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#getSubsets()">getSubsets()</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermSummaryTable.html#getSummaryTables(bacnet.datamodel.dataset.NTermData,java.lang.String)">getSummaryTables(NTermData, String)</a></span> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermSummaryTable.html" title="class in bacnet.datamodel.proteomics">NTermSummaryTable</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getSynonym()">getSynonym()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getSynonymsText()">getSynonymsText()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getTabFolder()">getTabFolder()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getTabFolder()">getTabFolder()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getTabFolder()">getTabFolder()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#getTabFolder()">getTabFolder()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getTable()">getTable()</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#getTable()">getTable()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/IUPACParser.html#getTable(java.lang.Integer)">getTable(Integer)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></dt>
<dd>
<div class="block">Returns a table by its identifier i.e. 1 means universal codon tables</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/IUPACParser.html#getTable(java.lang.String)">getTable(String)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></dt>
<dd>
<div class="block">Returns a table by its name</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#getTableBioCondition()">getTableBioCondition()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getTableBioCondition()">getTableBioCondition()</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#getTableBioCondition()">getTableBioCondition()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#getTableBioCondition()">getTableBioCondition()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getTableComposite()">getTableComposite()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#getTableComposite()">getTableComposite()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#getTableComposite()">getTableComposite()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/TableSWTView.html#getTableComposite()">getTableComposite()</a></span> - Method in class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#getTableElementName()">getTableElementName()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#getTableElementName()">getTableElementName()</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getTableGenes()">getTableGenes()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#getTableGenes()">getTableGenes()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getTableNodiff()">getTableNodiff()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getTableOtherInfo()">getTableOtherInfo()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getTableOver()">getTableOver()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getTableProteomes()">getTableProteomes()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/IUPACParser.html#getTables()">getTables()</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></dt>
<dd>
<div class="block">Returns a list of all available IUPAC tables</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#getTableSDRF(java.lang.String)">getTableSDRF(String)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Return table found in DATA_TABLE</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#getTableSDRF(java.lang.String)">getTableSDRF(String)</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>
<div class="block">Return table found in DATA_TABLE</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getTableSrnas()">getTableSrnas()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getTableUnder()">getTableUnder()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html#getTargetSequence(org.biojava3.core.sequence.template.Sequence)">getTargetSequence(Sequence&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="class in org.biojava3.core.sequence.edits">Edit.AbstractEdit</a></dt>
<dd>
<div class="block">Returns the Sequence which is our edit.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getTaxonomy()">getTaxonomy()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#getTbtmGraphVisualization()">getTbtmGraphVisualization()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getTbtmHeatmap()">getTbtmHeatmap()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#getTbtmRadialNetwork()">getTbtmRadialNetwork()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getTbtmScatterPlot()">getTbtmScatterPlot()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#getTechnoName(java.lang.String)">getTechnoName(String)</a></span> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Given an ArrayDesgin ID return the full name of the microarray</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#getTechnoName(java.lang.String)">getTechnoName(String)</a></span> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Given an ArrayDesgin ID return the full name of the microarray</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getTEMP_PATH()">getTEMP_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getTemperature()">getTemperature()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getTermSeqDatas()">getTermSeqDatas()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of SeqDatas</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getTermSeqDatas(java.lang.String)">getTermSeqDatas(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of Term given a bioCondName</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#getTests()">getTests()</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/DownloadServiceHandler.html#getText()">getText()</a></span> - Method in class bacnet.rap.<a href="../bacnet/rap/DownloadServiceHandler.html" title="class in bacnet.rap">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/DownloadServiceHandler.html#getText()">getText()</a></span> - Method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getTextColor()">getTextColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getTextColor()">getTextColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getTextColorSWT()">getTextColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getTextColorSWT()">getTextColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#getTextCutOff()">getTextCutOff()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getTextHeaderColor()">getTextHeaderColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getTextHeaderColorSWT()">getTextHeaderColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getTextHeaderColorSWT()">getTextHeaderColorSWT()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/gui/BlastQueryPage.html#getTextQuery()">getTextQuery()</a></span> - Method in class bacnet.scripts.blast.gui.<a href="../bacnet/scripts/blast/gui/BlastQueryPage.html" title="class in bacnet.scripts.blast.gui">BlastQueryPage</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#getTextSearch()">getTextSearch()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#getTextSearch()">getTextSearch()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/ConcurrencyTools.html#getThreadPool()">getThreadPool()</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Returns current shared thread pool.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#getThree()">getThree()</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#getThree()">getThree()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getThree()">getThree()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getThreshold()">getThreshold()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#getTilingData(java.util.ArrayList)">getTilingData(ArrayList&lt;String&gt;)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">List all TilingData available by biocondName.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#getTilingData(java.util.ArrayList,java.util.ArrayList)">getTilingData(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Load all Tiling present in leftBCs and rightBCs.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/EGDeWTdata.html#getTilings()">getTilings()</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/EGDeWTdata.html" title="class in bacnet.datamodel.dataset">EGDeWTdata</a></dt>
<dd>
<div class="block">Read all Tiling data corresponding to NAME_Mean, and return an ArrayList of
 Tiling</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getTilings()">getTilings()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/Experiment.html#getTilings()">getTilings()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Return all Tiling data contains in all BioCondition</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getTilings()">getTilings()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of Tilings</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getTilings(java.lang.String)">getTilings(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of Tiling given a bioCondName</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getTime()">getTime()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getTIRStructureEnergy(bacnet.datamodel.sequence.Gene)">getTIRStructureEnergy(Gene)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">For very gene read TIRStructure.excel and extract the mean value of energy:
 We calculate the 2nd structure free energy and save it a table<br>
 And we calculate the energy for 40 different sequence: -20+40 to -20+80 and
 calculate the average energy</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getTisList()">getTisList()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getTisMap()">getTisMap()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getTISName()">getTISName()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getTIStoDisplay(java.util.ArrayList)">getTIStoDisplay(ArrayList&lt;NTerm&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Return the list of from a List of NTerm</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#getTitleFont()">getTitleFont()</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Get the TITLE Font with appropriate height and style</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#getTitleFont(int)">getTitleFont(int)</a></span> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Get the TITLE Font with appropriate height and style</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#getToCompoundSet()">getToCompoundSet()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getToledoInfo()">getToledoInfo()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the different tables from Toledo-Arana publication and create a list of
 sRNAs from the different tables</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/BannerView.html#getToolBar()">getToolBar()</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BannerView.html" title="class in bacnet.e4.rap">BannerView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html#getTotal()">getTotal()</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html" title="class in bacnet.scripts.core.vennDiagram">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/CombinationGenerator.html#getTotal()">getTotal()</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/CombinationGenerator.html" title="class in bacnet.utils">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#getTotalBP()">getTotalBP()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getTotalColumnSize(java.util.ArrayList)">getTotalColumnSize(ArrayList&lt;ExpressionMatrix&gt;)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Return the number of column of the first ExpressionMatrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getTotalRowSize(java.util.ArrayList)">getTotalRowSize(ArrayList&lt;ExpressionMatrix&gt;)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Count the total Number of row if we concatenate all ExpressionMatrix in the
 ArrayList</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#getTotalSpectra()">getTotalSpectra()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getTotalSpectrum(bacnet.datamodel.sequence.Gene)">getTotalSpectrum(Gene)</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">For each gene add the number of spectra of every peptide available (not only
 aTIS peptides)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getTrack()">getTrack()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html#getTrack()">getTrack()</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddProteomicsDataDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html#getTrack()">getTrack()</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddTranscriptomicsDataDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#getTrack()">getTrack()</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#getTrack()">getTrack()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getTrack()">getTrack()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getTrack()">getTrack()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#getTrack()">getTrack()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getTrackComposite()">getTrackComposite()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getTrackGenome()">getTrackGenome()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#getTracksComposite()">getTracksComposite()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/GeneSequence.html#getTranscript(java.lang.String)">getTranscript(String)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Get the transcript sequence by accession</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DataValidation.html#getTranscriptomes()">getTranscriptomes()</a></span> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermTranscriptome.html#getTranscriptomes()">getTranscriptomes()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermTranscriptome.html" title="class in bacnet.scripts.listeriomics.nterm">NTermTranscriptome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getTRANSCRIPTOMES_PATH()">getTRANSCRIPTOMES_PATH()</a></span> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getTranscriptomesArrayPath()">getTranscriptomesArrayPath()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getTranscriptomesComparisonsArrayPath()">getTranscriptomesComparisonsArrayPath()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getTranscriptomesData()">getTranscriptomesData()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Put all TranscriptomeData in a list</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getTranscriptomesGenomes()">getTranscriptomesGenomes()</a></span> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Get all genomes associated to a transcriptomes dataset</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/GeneSequence.html#getTranscripts()">getTranscripts()</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Get the collection of transcription sequences assigned to this gene</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getTranstable(org.biojava3.core.sequence.DNASequence)">getTranstable(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getTriplet()">getTriplet()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#getTriplicateValue(java.lang.String)">getTriplicateValue(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Get all technical replicates values of corresponding probe in double[][]
 values</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getTSSDatas()">getTSSDatas()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of SeqDatas</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#getTSSDatas(java.lang.String)">getTSSDatas(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Return the list of TSS given a bioCondName</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getTssDownCoverage()">getTssDownCoverage()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getTssDownDistance()">getTssDownDistance()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getTssDowntype()">getTssDowntype()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getTssUpCoverage()">getTssUpCoverage()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getTssUpDistance()">getTssUpDistance()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getTssUptype()">getTssUptype()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#getTwo()">getTwo()</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#getTwo()">getTwo()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#getTwo()">getTwo()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getTxtCutoffLogFC()">getTxtCutoffLogFC()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getTxtCutoffLogFC()">getTxtCutoffLogFC()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getTxtCutoffPvalue()">getTxtCutoffPvalue()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#getTxtSearch()">getTxtSearch()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#getTxtSearch()">getTxtSearch()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#getTxtSearch()">getTxtSearch()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GenomicsView.html#getTxtSearch()">getTxtSearch()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getTxtSeqnucleotide()">getTxtSeqnucleotide()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#getType()">getType()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#getType()">getType()</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#getType()">getType()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#getType()">getType()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/StatTest.html#getType()">getType()</a></span> - Method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.html" title="class in bacnet.scripts.core.stat">StatTest</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getType()">getType()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>
<div class="block">GETTERS and SETTERS</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getType()">getType()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>
<div class="block">GETTERS and SETTERS</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getType()">getType()</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">The feature type</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getType()">getType()</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">The type of the feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getType(org.biojava3.core.sequence.DNASequence)">getType(DNASequence)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#getTypeColumn()">getTypeColumn()</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Return the column index corresponding to Type</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#getTypeData()">getTypeData()</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#getTypeDataContained()">getTypeDataContained()</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#getTypeFilter()">getTypeFilter()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#getTypeFilter()">getTypeFilter()</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getTypeModif()">getTypeModif()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/NcRNA.html#getTypeNcRNA()">getTypeNcRNA()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getTypeOfTIS(java.lang.String)">getTypeOfTIS(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Return a HashMap of <code>NTerm</code> with
 <code>nTerm.getTypeOverlap() == typeOverlap</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#getTypeOverlap()">getTypeOverlap()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#getTypeOverlaps()">getTypeOverlaps()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermFilter.html#getTypeOverlaps()">getTypeOverlaps()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#getTypeSrna()">getTypeSrna()</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getUnderASrnas()">getUnderASrnas()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#getUnderBioConds()">getUnderBioConds()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getUnderCisRegs()">getUnderCisRegs()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getUnderExpressed()">getUnderExpressed()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getUnderGenes()">getUnderGenes()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#getUnderlyingCompound()">getUnderlyingCompound()</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#getUnderSrnas()">getUnderSrnas()</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getUniprotbaseURL()">getUniprotbaseURL()</a></span> - Static method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">The current unirpot URL to deal with caching issues. www.uniprot.org is
 loaded balanced but you can access pir.uniprot.org directly.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#getUniprotDirectoryCache()">getUniprotDirectoryCache()</a></span> - Static method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">Local directory cache of XML that can be downloaded</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#getUnknownRAST(java.lang.String,java.lang.String)">getUnknownRAST(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Read the list of RAST elements totally unknown and create a list of Sequence
 from it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NGS.html#getUnlogValue(int,java.lang.String)">getUnlogValue(int, String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>
<div class="block">Get value at a certain base pair position and unlog it by calculating
 2^value<br>
 It works faster if data has been already loaded by this.read() or
 this.read(from,to)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getUpDNAList()">getUpDNAList()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#getUpperedBase()">getUpperedBase()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getUpProtList()">getUpProtList()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getUpRNAList()">getUpRNAList()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/NavigationManagement.html#getURL()">getURL()</a></span> - Static method in class bacnet.rap.<a href="../bacnet/rap/NavigationManagement.html" title="class in bacnet.rap">NavigationManagement</a></dt>
<dd>
<div class="block">Return the current URL of your RAP project</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/NavigationManagement.html#getURL()">getURL()</a></span> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>
<div class="block">Return the current URL of your RAP project</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#getUserCollection()">getUserCollection()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Provided for convience if the developer needs to associate data with a
 sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#getUserObject()">getUserObject()</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#getUserObject()">getUserObject()</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#getUTRLength()">getUTRLength()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#getUTRLength(bacnet.datamodel.sequence.Gene)">getUTRLength(Gene)</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span></div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/CRC64Checksum.html#getValue()">getValue()</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/CRC64Checksum.html" title="class in org.biojava3.core.util">CRC64Checksum</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getValue(int)">getValue(int)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Return the value at the right position in the double[] values argument<br>
 Read the data if necessary</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getValue(int,int)">getValue(int, int)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NGS.html#getValue(int,java.lang.String)">getValue(int, String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>
<div class="block">Return the value at the right position in the double[] values argument<br>
 Read the data if necessary</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getValue(java.lang.String,java.lang.String)">getValue(String, String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Get value in the ExpressionMatrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getValueAnnotation(int,int)">getValueAnnotation(int, int)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getValueAnnotation(java.lang.String,java.lang.String)">getValueAnnotation(String, String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getValues()">getValues()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getValues()">getValues()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getValues()">getValues()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getValues()">getValues()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/CodonUsage.html#getValues(java.lang.String)">getValues(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/CodonUsage.html" title="class in bacnet.datamodel.annotation">CodonUsage</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#getVar()">getVar()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#getVar()">getVar()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#getVariance()">getVariance()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#getVarianceValue(java.lang.String)">getVarianceValue(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Get variance value of corresponding probe in double[][] values</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Network.html#getVertices()">getVertices()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>
<div class="block">List of vertices = genome elements + their attributes = genome position,
 type, annotation</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#getViewedSequence()">getViewedSequence()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceView.html#getViewedSequence()">getViewedSequence()</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#getViewId()">getViewId()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#getViewId()">getViewId()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#getViewId()">getViewId()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#getViewId()">getViewId()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#getViewID()">getViewID()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#getViewID()">getViewID()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#getVoidMatrix()">getVoidMatrix()</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Create a matrix containing only one element = MISSING_VALUE</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#getWebpageTitle()">getWebpageTitle()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#getWGS(java.util.ArrayList)">getWGS(ArrayList&lt;String&gt;)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Extract all WGS (Whole Genome Shotgun) from genomesList</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#getWidth()">getWidth()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/MatrixSize.html#getWidth()">getWidth()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getWidthBP()">getWidthBP()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#getWidthBP()">getWidthBP()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/MatrixSize.html#getWidthDefault()">getWidthDefault()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/MatrixSize.html#getWidthRowNameDefault()">getWidthRowNameDefault()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/WindowedSequence.html#getWindowSize()">getWindowSize()</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence</a></dt>
<dd>
<div class="block">Access the current window size</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#getWurtzelInfo()">getWurtzelInfo()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>
<div class="block">Read the different tables from Wurtezel, Sesto et al. 2012 publication and
 create a list of sRNAs from the different tables</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#getX1()">getX1()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#getX2()">getX2()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#getY1()">getY1()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#getY2()">getY2()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#getZoom()">getZoom()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#getZoomHeightMax()">getZoomHeightMax()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>
<div class="block">Maximal value for the zoom</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#getZoomHeightPosition()">getZoomHeightPosition()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>
<div class="block">Position of the zoom for height</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#getZoomNumber()">getZoomNumber()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#getZoomPosition()">getZoomPosition()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#getZoomRatio()">getZoomRatio()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#getZoomVertical()">getZoomVertical()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/GFFNCBIReader.html#GFF_ATTRIBUTES">GFF_ATTRIBUTES</a></span> - Static variable in class bacnet.reader.<a href="../bacnet/reader/GFFNCBIReader.html" title="class in bacnet.reader">GFFNCBIReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GFF3Reader.html" title="class in org.biojava3.genome.parsers.gff"><span class="typeNameLink">GFF3Reader</span></a> - Class in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>
<div class="block">http://www.bioperl.org/wiki/GTF Read and write FeatureLists as GFF/GTF
 formatted files.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/GFF3Reader.html#%3Cinit%3E()">GFF3Reader()</a></span> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GFF3Reader.html" title="class in org.biojava3.genome.parsers.gff">GFF3Reader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GFF3Writer.html" title="class in org.biojava3.genome.parsers.gff"><span class="typeNameLink">GFF3Writer</span></a> - Class in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/GFF3Writer.html#%3Cinit%3E()">GFF3Writer()</a></span> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GFF3Writer.html" title="class in org.biojava3.genome.parsers.gff">GFF3Writer</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/GFFNCBIReader.html" title="class in bacnet.reader"><span class="typeNameLink">GFFNCBIReader</span></a> - Class in <a href="../bacnet/reader/package-summary.html">bacnet.reader</a></dt>
<dd>
<div class="block">Read GFF files to create a <a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence"><code>Genome</code></a> object</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/GFFNCBIReader.html#%3Cinit%3E()">GFFNCBIReader()</a></span> - Constructor for class bacnet.reader.<a href="../bacnet/reader/GFFNCBIReader.html" title="class in bacnet.reader">GFFNCBIReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/GlaserFCannotation.html" title="class in bacnet.datamodel.annotation"><span class="typeNameLink">GlaserFCannotation</span></a> - Class in <a href="../bacnet/datamodel/annotation/package-summary.html">bacnet.datamodel.annotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/GlaserFCannotation.html#%3Cinit%3E()">GlaserFCannotation()</a></span> - Constructor for class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/GlaserFCannotation.html" title="class in bacnet.datamodel.annotation">GlaserFCannotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html#GluC_Actino_Insoluble">GluC_Actino_Insoluble</a></span> - bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html" title="enum in bacnet.datamodel.proteomics">NTerm.ExperimentName</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html#GluC_Actino_Soluble">GluC_Actino_Soluble</a></span> - bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html" title="enum in bacnet.datamodel.proteomics">NTerm.ExperimentName</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/BasicColor.html#gradientColor(int)">gradientColor(int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SynTView.html#GRAPH_NAME">GRAPH_NAME</a></span> - Static variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#GRAPH_NAME">GRAPH_NAME</a></span> - Static variable in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/BasicColor.html#GREEN">GREEN</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/BasicColor.html#GREY">GREY</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/normalization/GRNormalization.html" title="class in bacnet.scripts.core.normalization"><span class="typeNameLink">GRNormalization</span></a> - Class in <a href="../bacnet/scripts/core/normalization/package-summary.html">bacnet.scripts.core.normalization</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/GRNormalization.html#%3Cinit%3E()">GRNormalization()</a></span> - Constructor for class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/GRNormalization.html" title="class in bacnet.scripts.core.normalization">GRNormalization</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Feature.html#group()">group()</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Get the first item (everything before first semicolon, if it has one) in the
 attribute field, which is assumed to be a group identifer.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/FeatureI.html#group()">group()</a></span> - Method in interface org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a></dt>
<dd>
<div class="block">Get the group id of the feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#%3Cinit%3E(int,int,org.biojava3.core.sequence.Strand,java.util.List)">GroupLocation(int, int, Strand, List&lt;Location&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#%3Cinit%3E(int,int,org.biojava3.core.sequence.Strand,org.biojava3.core.sequence.location.template.Location...)">GroupLocation(int, int, Strand, Location...)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,java.util.List)">GroupLocation(Point, Point, Strand, boolean, List&lt;Location&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,org.biojava3.core.sequence.location.template.Location...)">GroupLocation(Point, Point, Strand, boolean, Location...)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,java.util.List)">GroupLocation(Point, Point, Strand, List&lt;Location&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,org.biojava3.core.sequence.location.template.Location...)">GroupLocation(Point, Point, Strand, Location...)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#groupValues()">groupValues()</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a collection of all unique group ids in the list, as defined by the
 group() method of the features.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/InputStreamProvider.html#GZIP_MAGIC">GZIP_MAGIC</a></span> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/InputStreamProvider.html" title="class in org.biojava3.core.util">InputStreamProvider</a></dt>
<dd>
<div class="block">The magic number found at the start of a GZIP stream.</div>
</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a></div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
