<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>I-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="index: I">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:I">I</h2>
<dl class="index">
<dt><a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html#iCM" class="member-name-link">iCM</a> - Enum constant in enum bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/BannerView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BannerView.html" title="class in bacnet.e4.rap">BannerView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewBacnet.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewBacnet.html" title="class in bacnet.e4.rap">InitViewBacnet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeria.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeria.html" title="class in bacnet.e4.rap">InitViewListeria</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeriaSample.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">InitViewListeriaSample</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewYersinia.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewYersinia.html" title="class in bacnet.e4.rap">InitViewYersinia</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SynTView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaSummaryView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaSummaryView.html" title="class in bacnet.sequenceTools">SrnaSummaryView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/TableSWTView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/BannerView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.views.<a href="../bacnet/views/BannerView.html" title="class in bacnet.views">BannerView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView2.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/InitView.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.views.<a href="../bacnet/views/InitView.html" title="class in bacnet.views">InitView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/InternalBrowser.html#ID" class="member-name-link">ID</a> - Static variable in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#ident" class="member-name-link">ident</a> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#identities" class="member-name-link">identities</a> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#IDENTITY_CUTOFF" class="member-name-link">IDENTITY_CUTOFF</a> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Cutoff for indeityt value</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/IHM.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.technology">IHM</a> - Class in <a href="../bacnet/scripts/listeriomics/technology/package-summary.html">bacnet.scripts.listeriomics.technology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/IHM.html#%3Cinit%3E()" class="member-name-link">IHM()</a> - Constructor for class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/IHM.html" title="class in bacnet.scripts.listeriomics.technology">IHM</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ImageMagick.html" class="type-name-link" title="class in bacnet.utils">ImageMagick</a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>
<div class="block">Methods link to image mlanipulation with ImageMagick</div>
</dd>
<dt><a href="../bacnet/utils/ImageMagick.html#%3Cinit%3E()" class="member-name-link">ImageMagick()</a> - Constructor for class bacnet.utils.<a href="../bacnet/utils/ImageMagick.html" title="class in bacnet.utils">ImageMagick</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#indexOf(org.biojava3.core.sequence.template.Sequence,C)" class="member-name-link">indexOf(Sequence&lt;C&gt;, C)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Performs a linear search of the given Sequence for the given compound.</div>
</dd>
<dt><a href="../bacnet/table/core/Filter.TypeFilter.html#INFERIOR" class="member-name-link">INFERIOR</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/Filter.TypeFilter.html" title="enum in bacnet.table.core">Filter.TypeFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.TypeFilter.html#INFERIOR" class="member-name-link">INFERIOR</a> - Enum constant in enum bacnet.utils.<a href="../bacnet/utils/Filter.TypeFilter.html" title="enum in bacnet.utils">Filter.TypeFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.TypeFilter.html#INFERIOR_ABS" class="member-name-link">INFERIOR_ABS</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/Filter.TypeFilter.html" title="enum in bacnet.table.core">Filter.TypeFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.TypeFilter.html#INFERIOR_ABS" class="member-name-link">INFERIOR_ABS</a> - Enum constant in enum bacnet.utils.<a href="../bacnet/utils/Filter.TypeFilter.html" title="enum in bacnet.utils">Filter.TypeFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SessionControl.html#initBacnetApp(org.eclipse.e4.ui.workbench.modeling.EPartService,org.eclipse.e4.ui.workbench.modeling.EModelService,org.eclipse.swt.widgets.Shell)" class="member-name-link">initBacnetApp(EPartService, EModelService, Shell)</a> - Static method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SessionControl.html" title="class in bacnet.e4.rap">SessionControl</a></dt>
<dd>
<div class="block">Performed all initialisation necessary at the running of the webapp<br>
 Is runned in BannerView.createPartControl()</div>
</dd>
<dt><a href="../bacnet/e4/rap/setup/SessionControl.html#initBacnetApp(org.eclipse.e4.ui.workbench.modeling.EPartService,org.eclipse.e4.ui.workbench.modeling.EModelService,org.eclipse.swt.widgets.Shell)" class="member-name-link">initBacnetApp(EPartService, EModelService, Shell)</a> - Static method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SessionControl.html" title="class in bacnet.e4.rap.setup">SessionControl</a></dt>
<dd>
<div class="block">Performed all initialisation necessary at the running of the webapp<br>
 Is runned in BannerView.createPartControl()</div>
</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#initComboGenome()" class="member-name-link">initComboGenome()</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#initData(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">initData(ExpressionMatrix)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>
<div class="block">Init data and filter lists</div>
</dd>
<dt><a href="../bacnet/table/TableSWTComposite.html#initData(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome)" class="member-name-link">initData(ExpressionMatrix, Genome)</a> - Method in class bacnet.table.<a href="../bacnet/table/TableSWTComposite.html" title="class in bacnet.table">TableSWTComposite</a></dt>
<dd>
<div class="block">Init data and filter lists</div>
</dd>
<dt><a href="../bacnet/Database.html#initDatabase(java.lang.String)" class="member-name-link">initDatabase(String)</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Initialize appropriate database with right files</div>
</dd>
<dt><a href="../bacnet/Database.html#initDatabase(org.eclipse.swt.widgets.Shell)" class="member-name-link">initDatabase(Shell)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Run at the initialization of the workbench in Eclipse RAP website</div>
</dd>
<dt><a href="../bacnet/Database.html#initDatabaseMonitor(bacnet.Database,org.eclipse.core.runtime.IProgressMonitor)" class="member-name-link">initDatabaseMonitor(Database, IProgressMonitor)</a> - Static method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Initialize appropriate database with right files</div>
</dd>
<dt><a href="../bacnet/Database.InitDatabaseThread.html#%3Cinit%3E(bacnet.Database)" class="member-name-link">InitDatabaseThread(Database)</a> - Constructor for class bacnet.<a href="../bacnet/Database.InitDatabaseThread.html" title="class in bacnet">Database.InitDatabaseThread</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/FilterDialog.html#initDisplay()" class="member-name-link">initDisplay()</a> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/FilterDialog.html" title="class in bacnet.table.gui">FilterDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#initDisplayBoolean()" class="member-name-link">initDisplayBoolean()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Initiate Display boolean Vector when the first is added</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#initMet(boolean)" class="member-name-link">initMet(boolean)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.InitNetworkThread.html#%3Cinit%3E(bacnet.views.CoExprNetworkView)" class="member-name-link">InitNetworkThread(CoExprNetworkView)</a> - Constructor for class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.InitNetworkThread.html" title="class in bacnet.views">CoExprNetworkView.InitNetworkThread</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.InitOmicsMatrixThread.html#%3Cinit%3E(bacnet.expressionAtlas.HeatMapMultiOmicsView)" class="member-name-link">InitOmicsMatrixThread(HeatMapMultiOmicsView)</a> - Constructor for class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.InitOmicsMatrixThread.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView.InitOmicsMatrixThread</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#initStaticVariables()" class="member-name-link">initStaticVariables()</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Reinit static variables of OmicsData when database path is changed</div>
</dd>
<dt><a href="../bacnet/datamodel/phylogeny/Phylogenomic.html#initStaticVariables()" class="member-name-link">initStaticVariables()</a> - Static method in class bacnet.datamodel.phylogeny.<a href="../bacnet/datamodel/phylogeny/Phylogenomic.html" title="class in bacnet.datamodel.phylogeny">Phylogenomic</a></dt>
<dd>
<div class="block">Init static variables after Database change</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#initStaticVariables()" class="member-name-link">initStaticVariables()</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Init static variables after Database change</div>
</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#initView()" class="member-name-link">initView()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/InitView.html" class="type-name-link" title="class in bacnet.views">InitView</a> - Class in <a href="../bacnet/views/package-summary.html">bacnet.views</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/InitView.html#%3Cinit%3E()" class="member-name-link">InitView()</a> - Constructor for class bacnet.views.<a href="../bacnet/views/InitView.html" title="class in bacnet.views">InitView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewBacnet.html" class="type-name-link" title="class in bacnet.e4.rap">InitViewBacnet</a> - Class in <a href="../bacnet/e4/rap/package-summary.html">bacnet.e4.rap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewBacnet.html#%3Cinit%3E()" class="member-name-link">InitViewBacnet()</a> - Constructor for class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewBacnet.html" title="class in bacnet.e4.rap">InitViewBacnet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeria.html" class="type-name-link" title="class in bacnet.e4.rap">InitViewListeria</a> - Class in <a href="../bacnet/e4/rap/package-summary.html">bacnet.e4.rap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeria.html#%3Cinit%3E()" class="member-name-link">InitViewListeria()</a> - Constructor for class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeria.html" title="class in bacnet.e4.rap">InitViewListeria</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeriaSample.html" class="type-name-link" title="class in bacnet.e4.rap">InitViewListeriaSample</a> - Class in <a href="../bacnet/e4/rap/package-summary.html">bacnet.e4.rap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeriaSample.html#%3Cinit%3E()" class="member-name-link">InitViewListeriaSample()</a> - Constructor for class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">InitViewListeriaSample</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewYersinia.html" class="type-name-link" title="class in bacnet.e4.rap">InitViewYersinia</a> - Class in <a href="../bacnet/e4/rap/package-summary.html">bacnet.e4.rap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewYersinia.html#%3Cinit%3E()" class="member-name-link">InitViewYersinia()</a> - Constructor for class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewYersinia.html" title="class in bacnet.e4.rap">InitViewYersinia</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#innerProduct(double%5B%5D%5B%5D)" class="member-name-link">innerProduct(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Calcul of the Inner Product Matrix</div>
</dd>
<dt><a href="../org/biojava3/core/util/InputStreamProvider.html" class="type-name-link" title="class in org.biojava3.core.util">InputStreamProvider</a> - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>
<div class="block">A class that provides an InputStream from a File.</div>
</dd>
<dt><a href="../org/biojava3/core/util/InputStreamProvider.html#%3Cinit%3E()" class="member-name-link">InputStreamProvider()</a> - Constructor for class org.biojava3.core.util.<a href="../org/biojava3/core/util/InputStreamProvider.html" title="class in org.biojava3.core.util">InputStreamProvider</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLHelper.html#inputStreamToDocument(java.io.InputStream)" class="member-name-link">inputStreamToDocument(InputStream)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.html" class="type-name-link" title="class in org.biojava3.core.sequence.location">InsdcLocations</a> - Class in <a href="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</a></dt>
<dd>
<div class="block">A collection of locations which are used whenever we work with INSDC; some of which could be
 deprecated (from INSDC's point of view) yet appear in records.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.html#%3Cinit%3E()" class="member-name-link">InsdcLocations()</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.html" title="class in org.biojava3.core.sequence.location">InsdcLocations</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.BondLocation.html" class="type-name-link" title="class in org.biojava3.core.sequence.location">InsdcLocations.BondLocation</a> - Class in <a href="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</a></dt>
<dd>
<div class="block">Used to represent bond locations equivalent to bond(7,8) or bond(7).</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" class="type-name-link" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</a> - Class in <a href="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</a></dt>
<dd>
<div class="block">Deprecated in INSDC yet still appears; equivalent to the order() directive except no 5' to 3'
 ordering is defined.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.OneOfLocation.html" class="type-name-link" title="class in org.biojava3.core.sequence.location">InsdcLocations.OneOfLocation</a> - Class in <a href="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</a></dt>
<dd>
<div class="block">Deprecated in INSDC; refers to a set of locations of which one location could be valid e.g.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" class="type-name-link" title="class in org.biojava3.core.sequence.location">InsdcLocations.OrderLocation</a> - Class in <a href="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</a></dt>
<dd>
<div class="block">Used to describe a 5' to 3' ordering but no firm assurance it is correct</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcParser.html" class="type-name-link" title="class in org.biojava3.core.sequence.location">InsdcParser</a> - Class in <a href="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</a></dt>
<dd>
<div class="block">Parser for working with INSDC style locations.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcParser.html#%3Cinit%3E()" class="member-name-link">InsdcParser()</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcParser.html#%3Cinit%3E(org.biojava3.core.sequence.DataSource)" class="member-name-link">InsdcParser(DataSource)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.Insert.html#%3Cinit%3E(java.lang.String,int)" class="member-name-link">Insert(String, int)</a> - Constructor for class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.Insert.html" title="class in org.biojava3.core.sequence.edits">Edit.Insert</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.Insert.html#%3Cinit%3E(java.lang.String,int,int)" class="member-name-link">Insert(String, int, int)</a> - Constructor for class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.Insert.html" title="class in org.biojava3.core.sequence.edits">Edit.Insert</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.Insert.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence,int)" class="member-name-link">Insert(Sequence&lt;C&gt;, int)</a> - Constructor for class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.Insert.html" title="class in org.biojava3.core.sequence.edits">Edit.Insert</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.Insert.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence,int,int)" class="member-name-link">Insert(Sequence&lt;C&gt;, int, int)</a> - Constructor for class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.Insert.html" title="class in org.biojava3.core.sequence.edits">Edit.Insert</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/InternalBrowser.html" class="type-name-link" title="class in bacnet.views">InternalBrowser</a> - Class in <a href="../bacnet/views/package-summary.html">bacnet.views</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/InternalBrowser.html#%3Cinit%3E()" class="member-name-link">InternalBrowser()</a> - Constructor for class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#intersect(java.lang.String%5B%5D,java.lang.String%5B%5D)" class="member-name-link">intersect(String[], String[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Calculate intersection of two vectors L1 and L2 <br></div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#intersect(java.util.ArrayList)" class="member-name-link">intersect(ArrayList&lt;String[]&gt;)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Calculate the intersection of a list of Vector</div>
</dd>
<dt><a href="../bacnet/utils/ListUtils.html#intersect(java.util.ArrayList,java.util.ArrayList)" class="member-name-link">intersect(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ListUtils.html" title="class in bacnet.utils">ListUtils</a></dt>
<dd>
<div class="block">Calculate intersection of two lists L1 and L2 <br></div>
</dd>
<dt><a href="../bacnet/utils/ListUtils.html#intersect(java.util.ArrayList)" class="member-name-link">intersect(ArrayList&lt;ArrayList&lt;String&gt;&gt;)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ListUtils.html" title="class in bacnet.utils">ListUtils</a></dt>
<dd>
<div class="block">Calculate the intersection of a list of lists</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#intersection(org.biojava3.genome.parsers.gff.Location)" class="member-name-link">intersection(Location)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Return the intersection, or null if no overlap.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/IntronSequence.html" class="type-name-link" title="class in org.biojava3.core.sequence">IntronSequence</a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/IntronSequence.html#%3Cinit%3E(org.biojava3.core.sequence.GeneSequence,int,int)" class="member-name-link">IntronSequence(GeneSequence, int, int)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</a></dt>
<dd>
<div class="block">Place holder for Intron sequence features</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#intValue()" class="member-name-link">intValue()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></dt>
<dd>
<div class="block">Attempts to provide an int version of this codon which multiplies each position by</div>
</dd>
<dt><a href="../bacnet/utils/ListUtils.html#inverse(java.util.ArrayList,java.util.ArrayList)" class="member-name-link">inverse(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ListUtils.html" title="class in bacnet.utils">ListUtils</a></dt>
<dd>
<div class="block">Return the invert list of List1 inside the total universe</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#inverse(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">inverse(Sequence&lt;C&gt;)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">A method which attempts to do the right thing when is comes to a reverse/reverse complement</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html" class="type-name-link" title="class in org.biojava3.core.sequence.io.util">IOUtils</a> - Class in <a href="../org/biojava3/core/sequence/io/util/package-summary.html">org.biojava3.core.sequence.io.util</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#%3Cinit%3E()" class="member-name-link">IOUtils()</a> - Constructor for class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.ReaderProcessor.html" class="type-name-link" title="interface in org.biojava3.core.sequence.io.util">IOUtils.ReaderProcessor</a> - Interface in <a href="../org/biojava3/core/sequence/io/util/package-summary.html">org.biojava3.core.sequence.io.util</a></dt>
<dd>
<div class="block">Closure interface used when working with <code>IOUtils#processReader(String)</code>.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#isAfter(org.biojava3.genome.parsers.gff.Location)" class="member-name-link">isAfter(Location)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Check if this location is entirely after the other location (no overlap).</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#isAlreadyRead()" class="member-name-link">isAlreadyRead()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html#isAmbiguous()" class="member-name-link">isAmbiguous()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#isBefore(org.biojava3.genome.parsers.gff.Location)" class="member-name-link">isBefore(Location)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Check if this location is entirely before other location (no overlap).</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#isBetweenCompounds()" class="member-name-link">isBetweenCompounds()</a> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.html#isBetweenCompounds()" class="member-name-link">isBetweenCompounds()</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Returns true if the position is meant to represent a point between two points such as 78^79.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#isCircular()" class="member-name-link">isCircular()</a> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.html#isCircular()" class="member-name-link">isCircular()</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Indicates if this location is circular.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#isComplementable()" class="member-name-link">isComplementable()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#isComplementable()" class="member-name-link">isComplementable()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html#isComplementable()" class="member-name-link">isComplementable()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractNucleotideCompoundSet</a></dt>
<dd>
<div class="block">NucleotideCompounds can always complement</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundSet.html#isComplementable()" class="member-name-link">isComplementable()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#isCompleteGenome(java.lang.String)" class="member-name-link">isCompleteGenome(String)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Detect if the genome is a Complete genome or a WGS using DATABASES_LIST_PATH table</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#isComplex()" class="member-name-link">isComplex()</a> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.html#isComplex()" class="member-name-link">isComplex()</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></dt>
<dd>
<div class="block">Returns true if the location is considered to be complex meaning the location is actually
 composed of sub-locations.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#isCompoundStringLengthEqual()" class="member-name-link">isCompoundStringLengthEqual()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#isCompoundStringLengthEqual()" class="member-name-link">isCompoundStringLengthEqual()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundSet.html#isCompoundStringLengthEqual()" class="member-name-link">isCompoundStringLengthEqual()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></dt>
<dd>
<div class="block">Returns true if all String representations of Compounds are of the same length.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#isDataRead()" class="member-name-link">isDataRead()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#isDisplayAbsoluteValue()" class="member-name-link">isDisplayAbsoluteValue()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#isDisplayMouseLine()" class="member-name-link">isDisplayMouseLine()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#isDisplayMouseLine()" class="member-name-link">isDisplayMouseLine()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#isDisplaySequence()" class="member-name-link">isDisplaySequence()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#isEqual(bacnet.datamodel.sequence.Sequence)" class="member-name-link">isEqual(Sequence)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">If :
 sequence.getBegin()==this.getBegin()
 &amp;&amp; sequence.getEnd()==this.getEnd()
 &amp;&amp; sequence.getStrand()==this.getStrand()<br>
 return true</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#isEqual(bacnet.datamodel.sequence.Srna,int)" class="member-name-link">isEqual(Srna, int)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#isEqual(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence,int)" class="member-name-link">isEqual(DNASequence, DNASequence, int)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Calculates the distance between the two center fo gene1 and gene2<br>
 If this distance is under scale, in absolute value, and both gene are on the same strand, True is
 return</div>
</dd>
<dt><a href="../bacnet/table/core/Filter.html#isFilterColumn()" class="member-name-link">isFilterColumn()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#isFilterColumn()" class="member-name-link">isFilterColumn()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewBacnet.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewBacnet.html" title="class in bacnet.e4.rap">InitViewBacnet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeria.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeria.html" title="class in bacnet.e4.rap">InitViewListeria</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeriaSample.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">InitViewListeriaSample</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewYersinia.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewYersinia.html" title="class in bacnet.e4.rap">InitViewYersinia</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#isFocused()" class="member-name-link">isFocused()</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#isForMandAltStart(bacnet.datamodel.proteomics.NTerm)" class="member-name-link">isForMandAltStart(NTerm)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>
<div class="block">Find if the NTerm start with a Methyonine<br>
 if it corresponds to a regular start codon "ATG" we need a Formyl modificaition to prove its a
 real translation initiation<br>
 If the methyonine corresponds to any other type of start codon, we don't need more to prove it's
 a translation initiation</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#isFormylated()" class="member-name-link">isFormylated()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>
<div class="block">Return true if the TIS includes a formylated NTerm</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimplePoint.html#isHigher(org.biojava3.core.sequence.location.template.Point)" class="member-name-link">isHigher(Point)</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Point.html#isHigher(org.biojava3.core.sequence.location.template.Point)" class="member-name-link">isHigher(Point)</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a></dt>
<dd>
<div class="block">Returns true if the point is higher in value to the current point</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#isInfoRead()" class="member-name-link">isInfoRead()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#isInfoRead()" class="member-name-link">isInfoRead()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#isInitialized()" class="member-name-link">isInitialized()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#isInsoluble()" class="member-name-link">isInsoluble()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/MathUtils.html#isInteger(double)" class="member-name-link">isInteger(double)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/MathUtils.html" title="class in bacnet.utils">MathUtils</a></dt>
<dd>
<div class="block">Test if a double is an integer or not</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#isIntersect()" class="member-name-link">isIntersect()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#isLoaded()" class="member-name-link">isLoaded()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#isLogTransformed()" class="member-name-link">isLogTransformed()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimplePoint.html#isLower(org.biojava3.core.sequence.location.template.Point)" class="member-name-link">isLower(Point)</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Point.html#isLower(org.biojava3.core.sequence.location.template.Point)" class="member-name-link">isLower(Point)</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a></dt>
<dd>
<div class="block">Returns true if the current point is at a lower position than the point given.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#isNegative()" class="member-name-link">isNegative()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Check if location is on negative strand.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#isNGS()" class="member-name-link">isNGS()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#isNoData()" class="member-name-link">isNoData()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#isNucleotide()" class="member-name-link">isNucleotide()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/SequenceTools.html#isNucleotideSequence(java.lang.String)" class="member-name-link">isNucleotideSequence(String)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SequenceTools.html" title="class in org.biojava3.core.util">SequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#isOrdered()" class="member-name-link">isOrdered()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#isOverlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)" class="member-name-link">isOverlap(Sequence, Sequence)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Decide if two genes are overlapping <br>
 First, calculate the overlapping size of two genes, finding the gene upstream to the other<br>
 If overlap is negative it will mean that we have no overlap (gene1.end &lt; gene2.begin) <br></div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#isOverlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence,int)" class="member-name-link">isOverlap(Sequence, Sequence, int)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Decide if two genes are overlapping <br>
 First, calculate the overlapping size of two genes, finding the gene upstream to the other<br>
 If overlap is negative it will mean that we have no overlap (gene1.end &lt; gene2.begin) <br></div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#isOverlap(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence)" class="member-name-link">isOverlap(DNASequence, DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Decide if two genes are overlapping <br>
 First, calculate the overlapping size of two genes, finding the gene upstream to the other<br>
 If overlap is negative it will mean that we have no overlap (gene1.end &lt; gene2.begin) <br></div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#isOverlapInFrame()" class="member-name-link">isOverlapInFrame()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#isOverlapStart(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence,int)" class="member-name-link">isOverlapStart(Sequence, Sequence, int)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Decide if first Sequence is overlapping the start region of the second <br></div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#isPseudogene()" class="member-name-link">isPseudogene()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#isSameStrand(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)" class="member-name-link">isSameStrand(Sequence, Sequence)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Decide if two sequences have same strand</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#isSameStrand(org.biojava3.genome.parsers.gff.Location)" class="member-name-link">isSameStrand(Location)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Check if this location is on same strand as other location.</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#isSoluble()" class="member-name-link">isSoluble()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#isStart()" class="member-name-link">isStart()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#isStart()" class="member-name-link">isStart()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#isStart(bacnet.datamodel.proteomics.NTerm,java.lang.String)" class="member-name-link">isStart(NTerm, String)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>
<div class="block">Find i an NTerm start with a methyonine (START), or if the amino acid just before the peptide is
 a Methyonine (START-1), or just after (START+1)</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Codon.html#isStart(java.lang.String)" class="member-name-link">isStart(String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>
<div class="block">Test if a codon is a start codon <br></div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#isStart(org.biojava3.core.sequence.compound.AminoAcidCompound)" class="member-name-link">isStart(AminoAcidCompound)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a></dt>
<dd>
<div class="block">Returns true if the given compound was a start codon in this codon table.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.html#isStart(org.biojava3.core.sequence.compound.AminoAcidCompound)" class="member-name-link">isStart(AminoAcidCompound)</a> - Method in interface org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</a></dt>
<dd>
<div class="block">Returns true if the given compound could have been a start amino acid; this does not assert if
 the codon that actually coded for the amino acid was a start codon.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#isStop()" class="member-name-link">isStop()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Codon.html#isStop(java.lang.String)" class="member-name-link">isStop(String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>
<div class="block">Test if a codon is a stop codon <br></div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#isStrand()" class="member-name-link">isStrand()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Return True if strand=='+'</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#isTestData()" class="member-name-link">isTestData()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#isTiling()" class="member-name-link">isTiling()</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#isTranscriptomics()" class="member-name-link">isTranscriptomics()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimplePoint.html#isUncertain()" class="member-name-link">isUncertain()</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Point.html#isUncertain()" class="member-name-link">isUncertain()</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a></dt>
<dd>
<div class="block">Returns a true if the exact point is unknown.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimplePoint.html#isUnknown()" class="member-name-link">isUnknown()</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Point.html#isUnknown()" class="member-name-link">isUnknown()</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a></dt>
<dd>
<div class="block">Returns true if the current position is unknown but is beyond the position encoded for.</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#isWildType()" class="member-name-link">isWildType()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#isZoomChangedFlag()" class="member-name-link">isZoomChangedFlag()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#ITEM" class="member-name-link">ITEM</a> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#iterator()" class="member-name-link">iterator()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#iterator()" class="member-name-link">iterator()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#iterator()" class="member-name-link">iterator()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#iterator()" class="member-name-link">iterator()</a> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>
<div class="block">Iterates through all known sub-locations for this location but does not descend</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#iterator()" class="member-name-link">iterator()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#iterator()" class="member-name-link">iterator()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Provides basic iterable access to this class</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#iterator()" class="member-name-link">iterator()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>
<div class="block">Iterator implementation which attempts to move through the 2D structure attempting to skip onto
 the next sequence as &amp; when it is asked to</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#iterator()" class="member-name-link">iterator()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Returns an instance of <a href="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="class in org.biojava3.core.sequence.template"><code>SequenceMixin.SequenceIterator</code></a></div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#iterator()" class="member-name-link">iterator()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#iterator()" class="member-name-link">iterator()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/WindowedSequence.html#iterator()" class="member-name-link">iterator()</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence</a></dt>
<dd>
<div class="block">Returns an iterator which will return the windows in a sequence in sequential order.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#iterator()" class="member-name-link">iterator()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Create a location iterator over this location with a window size of 1 and an increment of +1
 (successive symbols from start to end).</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#iterator(int,int)" class="member-name-link">iterator(int, int)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Create a location iterator over this location, using specified window size and increment.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.html#IUPAC_LOCATION" class="member-name-link">IUPAC_LOCATION</a> - Static variable in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">IUPACParser</a> - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">Available translations

 
 1 - UNIVERSAL
 2 - VERTEBRATE_MITOCHONDRIAL
 3 - YEAST_MITOCHONDRIAL
 4 - MOLD_MITOCHONDRIAL
 5 - INVERTEBRATE_MITOCHONDRIAL
 6 - CILIATE_NUCLEAR
 9 - ECHINODERM_MITOCHONDRIAL
 10 - EUPLOTID_NUCLEAR
 11 - BACTERIAL
 12 - ALTERNATIVE_YEAST_NUCLEAR
 13 - ASCIDIAN_MITOCHONDRIAL
 14 - FLATWORM_MITOCHONDRIAL
 15 - BLEPHARISMA_MACRONUCLEAR
 16 - 2CHLOROPHYCEAN_MITOCHONDRIAL
 21 - TREMATODE_MITOCHONDRIAL
 23 - SCENEDESMUS_MITOCHONDRIAL
 

 Taken from <a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c">NCBI</a>
 with slight modification and put into the classpath resource.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.html#%3Cinit%3E()" class="member-name-link">IUPACParser()</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></dt>
<dd>
<div class="block">Default version and uses the classpath based IUPAC table</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.html#%3Cinit%3E(java.io.InputStream)" class="member-name-link">IUPACParser(InputStream)</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></dt>
<dd>
<div class="block">Allows you to specify a different IUPAC table.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a> - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">Holds the concept of a codon table from the IUPAC format</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#%3Cinit%3E(java.lang.String,int,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">IUPACTable(String, int, String, String, String, String, String)</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html#%3Cinit%3E(java.lang.String,java.lang.Integer,java.lang.String,java.lang.String)" class="member-name-link">IUPACTable(String, Integer, String, String)</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a></dt>
<dd>
<div class="block">Constructor which uses the basic IUPAC codon table format.</div>
</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
