<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>M-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="index: M">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:M">M</h2>
<dl class="index">
<dt><a href="../bacnet/scripts/core/stat/AMtools.html#M(double,double)" class="member-name-link">M(double, double)</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/AMtools.html" title="class in bacnet.scripts.core.stat">AMtools</a></dt>
<dd>
<div class="block">Calculate M = x1-x2<br>
 Represents the variability within expression value x1 and x2</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/Macroarray.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.technology">Macroarray</a> - Class in <a href="../bacnet/scripts/listeriomics/technology/package-summary.html">bacnet.scripts.listeriomics.technology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/Macroarray.html#%3Cinit%3E()" class="member-name-link">Macroarray()</a> - Constructor for class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Macroarray.html" title="class in bacnet.scripts.listeriomics.technology">Macroarray</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaReader.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaReader.html" title="class in org.biojava3.core.sequence.io">FastaReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaWriter.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/ProteinSequence.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/UncompressInputStream.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/App.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/App.html" title="class in org.biojava3.genome">App</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.genome.parsers.geneid.<a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html" title="class in org.biojava3.genome.parsers.geneid">GeneIDXMLReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span>&nbsp;</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span>&nbsp;</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html" title="class in org.biojava3.genome.parsers.gff">GeneIDGFF2Reader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html" title="class in org.biojava3.genome.parsers.gff">GeneMarkGTFReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GFF3Reader.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GFF3Reader.html" title="class in org.biojava3.genome.parsers.gff">GFF3Reader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/GFF3Writer.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GFF3Writer.html" title="class in org.biojava3.genome.parsers.gff">GFF3Writer</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span>&nbsp;</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/LocIterator.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span>&nbsp;</div>
</dd>
<dt><a href="../org/biojava3/genome/query/BlastXMLQuery.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.genome.query.<a href="../org/biojava3/genome/query/BlastXMLQuery.html" title="class in org.biojava3.genome.query">BlastXMLQuery</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/query/OutputHitsGFF.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.genome.query.<a href="../org/biojava3/genome/query/OutputHitsGFF.html" title="class in org.biojava3.genome.query">OutputHitsGFF</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/uniprot/UniprotToFasta.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.genome.uniprot.<a href="../org/biojava3/genome/uniprot/UniprotToFasta.html" title="class in org.biojava3.genome.uniprot">UniprotToFasta</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/util/SplitFasta.html#main(java.lang.String%5B%5D)" class="member-name-link">main(String[])</a> - Static method in class org.biojava3.genome.util.<a href="../org/biojava3/genome/util/SplitFasta.html" title="class in org.biojava3.genome.util">SplitFasta</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/BasicApplication.html#MAIN_ENTRYPOINT" class="member-name-link">MAIN_ENTRYPOINT</a> - Static variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BasicApplication.html" title="class in bacnet.e4.rap">BasicApplication</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/BasicApplication.html#MAIN_ENTRYPOINT" class="member-name-link">MAIN_ENTRYPOINT</a> - Static variable in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/BasicApplication.html" title="class in bacnet.e4.rap.setup">BasicApplication</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/MainListeriomics.html" class="type-name-link" title="class in bacnet.scripts.listeriomics">MainListeriomics</a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#%3Cinit%3E()" class="member-name-link">MainListeriomics()</a> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#makeblastdb" class="member-name-link">makeblastdb</a> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#makeblastdb" class="member-name-link">makeblastdb</a> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Shortcut for running makeblastdb</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/GlaserFCannotation.html#MANDINFC_MAP" class="member-name-link">MANDINFC_MAP</a> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/GlaserFCannotation.html" title="class in bacnet.datamodel.annotation">GlaserFCannotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#MANNWHIT" class="member-name-link">MANNWHIT</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#mark(int)" class="member-name-link">mark(int)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>
<div class="block">Marks the present position in the stream.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#markSupported()" class="member-name-link">markSupported()</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>
<div class="block">Tells whether this stream supports the mark() operation, which it does.</div>
</dd>
<dt><a href="../org/biojava3/core/util/UncompressInputStream.html#markSupported()" class="member-name-link">markSupported()</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></dt>
<dd>
<div class="block">This stream does not support mark/reset on the stream.</div>
</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#MARKUP_ENABLED" class="member-name-link">MARKUP_ENABLED</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/MathUtils.html" class="type-name-link" title="class in bacnet.utils">MathUtils</a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/MathUtils.html#%3Cinit%3E()" class="member-name-link">MathUtils()</a> - Constructor for class bacnet.utils.<a href="../bacnet/utils/MathUtils.html" title="class in bacnet.utils">MathUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/MatrixSize.html" class="type-name-link" title="class in bacnet.table.core">MatrixSize</a> - Class in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/MatrixSize.html#%3Cinit%3E()" class="member-name-link">MatrixSize()</a> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#max(double%5B%5D)" class="member-name-link">max(double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Find maximum value of a vector</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#max(double%5B%5D%5B%5D)" class="member-name-link">max(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#max(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">max(ExpressionMatrix)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Search for the maximum of all the values</div>
</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#max(bacnet.datamodel.dataset.ExpressionMatrix,int)" class="member-name-link">max(ExpressionMatrix, int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Search for the maximum on one column</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#MAX_RESOLVER" class="member-name-link">MAX_RESOLVER</a> - Static variable in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>
<div class="block">Always uses the max point to resolve a location</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#mean(double%5B%5D)" class="member-name-link">mean(double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Calculate mean of a vector</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#mean(double%5B%5D%5B%5D)" class="member-name-link">mean(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Return the mean of the whole array</div>
</dd>
<dt><a href="../bacnet/utils/MathUtils.html#mean(double,double)" class="member-name-link">mean(double, double)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/MathUtils.html" title="class in bacnet.utils">MathUtils</a></dt>
<dd>
<div class="block">Calculate the mean between two numbers</div>
</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#mean(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">mean(ExpressionMatrix)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Calculate the mean of the ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#mean(bacnet.datamodel.dataset.ExpressionMatrix,int)" class="member-name-link">mean(ExpressionMatrix, int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Calculate the mean of the ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#mean(java.lang.Double%5B%5D)" class="member-name-link">mean(Double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#MEAN_RESOLVER" class="member-name-link">MEAN_RESOLVER</a> - Static variable in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>
<div class="block">Combines min and max and then gets the mean of it</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#median(double%5B%5D)" class="member-name-link">median(double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Calculate median of a vector</div>
</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#median(bacnet.datamodel.dataset.ExpressionMatrix,int)" class="member-name-link">median(ExpressionMatrix, int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Calculate the median of the ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#median(java.lang.Double%5B%5D)" class="member-name-link">median(Double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Calculate median of a vector</div>
</dd>
<dt><a href="../bacnet/scripts/core/normalization/MedianNormalization.html" class="type-name-link" title="class in bacnet.scripts.core.normalization">MedianNormalization</a> - Class in <a href="../bacnet/scripts/core/normalization/package-summary.html">bacnet.scripts.core.normalization</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/normalization/MedianNormalization.html#%3Cinit%3E()" class="member-name-link">MedianNormalization()</a> - Constructor for class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/MedianNormalization.html" title="class in bacnet.scripts.core.normalization">MedianNormalization</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#merge(java.util.ArrayList,boolean)" class="member-name-link">merge(ArrayList&lt;ExpressionMatrix&gt;, boolean)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Create a fusion of a list of ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#midline" class="member-name-link">midline</a> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#min(double%5B%5D)" class="member-name-link">min(double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Find minimum value of a vector</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#min(double%5B%5D%5B%5D)" class="member-name-link">min(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#min(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">min(ExpressionMatrix)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Search for the minimum on all the values</div>
</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#min(bacnet.datamodel.dataset.ExpressionMatrix,int)" class="member-name-link">min(ExpressionMatrix, int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Search for the minimum on one column</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#MIN_RESOLVER" class="member-name-link">MIN_RESOLVER</a> - Static variable in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>
<div class="block">Always uses the min point to resolve a location</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#MINIMAL_BP_NB" class="member-name-link">MINIMAL_BP_NB</a> - Static variable in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceUsage.html#MINIMAL_SEQUENCE_DATA" class="member-name-link">MINIMAL_SEQUENCE_DATA</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceUsage.html" title="enum in org.biojava3.core.sequence">SequenceOptimizationHints.SequenceUsage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceCollection.html#MINIMINAL_SEQUENCES" class="member-name-link">MINIMINAL_SEQUENCES</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceCollection.html" title="enum in org.biojava3.core.sequence">SequenceOptimizationHints.SequenceCollection</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#MINIMUM_DATA_SIZE" class="member-name-link">MINIMUM_DATA_SIZE</a> - Variable in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">Minimal height of a data track</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html#minus" class="member-name-link">minus</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#minus()" class="member-name-link">minus()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Return location that is in same position on negative strand.</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#minus(double%5B%5D,double%5B%5D)" class="member-name-link">minus(double[], double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#minus(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">minus(ExpressionMatrix, ExpressionMatrix)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Perform a soustraction between two ExpressionMatrix <br>
 First the intersection in term of RowNames and headers is found, then addition is done for each
 row</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html#minusM" class="member-name-link">minusM</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html#minusP" class="member-name-link">minusP</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#MISSING_VALUE" class="member-name-link">MISSING_VALUE</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Every Found missing values are replaced by this number</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.DNAType.html#MITOCHONDRIAL" class="member-name-link">MITOCHONDRIAL</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView.html#MODEL_CAS9" class="member-name-link">MODEL_CAS9</a> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#modifHeader(java.lang.String,java.lang.String)" class="member-name-link">modifHeader(String, String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/AppSpecificMethods.html#modifyApp(java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">modifyApp(String, EPartService)</a> - Static method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/AppSpecificMethods.html" title="class in bacnet.e4.rap">AppSpecificMethods</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/SaveFileUtils.html#modifyHTMLwithFile(java.lang.String,java.lang.String)" class="member-name-link">modifyHTMLwithFile(String, String)</a> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/SaveFileUtils.html" title="class in bacnet.raprcp">SaveFileUtils</a></dt>
<dd>
<div class="block">Replace "_FileName" by fileName in the htmlFile found in bacnet/html/ folder</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#modifyID(java.lang.String,java.lang.String)" class="member-name-link">modifyID(String, String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Modify id of gene, to be sure that the suffix number is always of 4 numbers<br>
 ex: replace lmpc1 by lmpc0001, lmpc12 by lmpc0012<br></div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#modifyPathArray()" class="member-name-link">modifyPathArray()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">Check PATHARRAY and had columns if needed<br>
 <br>
 
 Two types of array are possible:
 one with only one experiment (contains only one column with "# spectra")
 one with only multiple experiments (contains many columns containing "# spectra") <br>
 <br>
 For The first type:<br>
 Check if PATHARRAY table is complete = has 8 columns, first column is the list of peptide,
 soluble and insoluble contains "TRUE" and "FALSE" and no "yes"<br>
 Change "TRUE" and "FALSE" to "yes" and "no"
 Remove line with NH2-free peptide <br>
 <br>
 For the second type:<br></div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/LocationHelper.html#modulateCircularIndex(int,int)" class="member-name-link">modulateCircularIndex(int, int)</a> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Takes a point on a circular location and moves it left until it falls at the earliest possible
 point that represents the same base.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.Tools.html#modulateCircularIndex(int,int)" class="member-name-link">modulateCircularIndex(int, int)</a> - Static method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.Tools.html" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a></dt>
<dd>
<div class="block">Takes a point on a circular location and moves it left until it falls at the earliest possible
 point that represents the same base.</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#MOUSE_ANNOTTABLE" class="member-name-link">MOUSE_ANNOTTABLE</a> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#mouseDoubleClick(org.eclipse.swt.events.MouseEvent)" class="member-name-link">mouseDoubleClick(MouseEvent)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#mouseDoubleClick(org.eclipse.swt.events.MouseEvent)" class="member-name-link">mouseDoubleClick(MouseEvent)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#mouseDoubleClick(org.eclipse.swt.events.MouseEvent)" class="member-name-link">mouseDoubleClick(MouseEvent)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#mouseDown(org.eclipse.swt.events.MouseEvent)" class="member-name-link">mouseDown(MouseEvent)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#mouseDown(org.eclipse.swt.events.MouseEvent)" class="member-name-link">mouseDown(MouseEvent)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#mouseDown(org.eclipse.swt.events.MouseEvent)" class="member-name-link">mouseDown(MouseEvent)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#mouseMove(org.eclipse.swt.events.MouseEvent)" class="member-name-link">mouseMove(MouseEvent)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">If the mouse is over the genome viewer, detect it is on the annotation, and display the
 corresponding gene, sRNA, asRNA</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#mouseMove(org.eclipse.swt.events.MouseEvent)" class="member-name-link">mouseMove(MouseEvent)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>
<div class="block">If the mouse is over the genome viewer, detect it is on the annotation, and display the
 corresponding gene, sRNA, asRNA</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#mouseUp(org.eclipse.swt.events.MouseEvent)" class="member-name-link">mouseUp(MouseEvent)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#mouseUp(org.eclipse.swt.events.MouseEvent)" class="member-name-link">mouseUp(MouseEvent)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#mouseUp(org.eclipse.swt.events.MouseEvent)" class="member-name-link">mouseUp(MouseEvent)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#moveHorizontally(int)" class="member-name-link">moveHorizontally(int)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#moveHorizontally(int)" class="member-name-link">moveHorizontally(int)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#moveHorizontally(int)" class="member-name-link">moveHorizontally(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Move horizontally the tracks</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#moveVertically(int)" class="member-name-link">moveVertically(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">From the position of the vertical bar draw the corresponding part of the data</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html#MSF" class="member-name-link">MSF</a> - Enum constant in enum org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html" title="enum in org.biojava3.core.sequence.template">LightweightProfile.StringFormat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermConservationWithRiboProfiling.html#multiAlign(java.util.ArrayList,java.lang.String,java.lang.String)" class="member-name-link">multiAlign(ArrayList&lt;Sequence&gt;, String, String)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermConservationWithRiboProfiling.html" title="class in bacnet.scripts.listeriomics.nterm">NTermConservationWithRiboProfiling</a></dt>
<dd>
<div class="block">Run a Blast for each sRNA on a list of Genomes given in tempDatabase</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#multipleFrameTranslation(org.biojava3.core.sequence.template.Sequence,org.biojava3.core.sequence.transcription.Frame...)" class="member-name-link">multipleFrameTranslation(Sequence&lt;NucleotideCompound&gt;, Frame...)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>
<div class="block">A way of translating DNA in a number of frames</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" class="type-name-link" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a>&lt;<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">S</a> extends <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">C</a>&gt;,<wbr><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">Implements a minimal data structure for reading and writing a sequence alignment.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#%3Cinit%3E()" class="member-name-link">MultipleSequenceAlignment()</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html" class="type-name-link" title="class in bacnet.scripts.blast">MultiSequenceBlastProtein</a> - Class in <a href="../bacnet/scripts/blast/package-summary.html">bacnet.scripts.blast</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html#%3Cinit%3E()" class="member-name-link">MultiSequenceBlastProtein()</a> - Constructor for class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html" title="class in bacnet.scripts.blast">MultiSequenceBlastProtein</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" class="type-name-link" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a> - Class in <a href="../bacnet/scripts/blast/package-summary.html">bacnet.scripts.blast</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#%3Cinit%3E()" class="member-name-link">MultiSequenceTBlastNProtein()</a> - Constructor for class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#multiSpeciesAlign()" class="member-name-link">multiSpeciesAlign()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Run a Blast for each sRNA on a list of Genomes given in tempDatabase</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#multiSpeciesAlign(java.util.ArrayList)" class="member-name-link">multiSpeciesAlign(ArrayList&lt;Srna&gt;)</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Run a Blast for each sRNA on a list of Genomes given in tempDatabase</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#multiSpeciesAlign10403S()" class="member-name-link">multiSpeciesAlign10403S()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Run a Blast for each sRNA (found in Oliver et al. in 10403S genome) on a list of Genomes given in
 tempDatabase</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html#multiSpeciesAlignBlastN(java.util.ArrayList)" class="member-name-link">multiSpeciesAlignBlastN(ArrayList&lt;Gene&gt;)</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html" title="class in bacnet.scripts.listeriomics.srna">SmallPeptidesSearch</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#multiSpeciesAlignNucleotide(java.util.ArrayList,boolean)" class="member-name-link">multiSpeciesAlignNucleotide(ArrayList&lt;Sequence&gt;, boolean)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>
<div class="block">Run a Blast for each sRNA on a list of Genomes given in tempDatabase</div>
</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#multiSpeciesAlignProtein(java.util.ArrayList,java.util.ArrayList,java.lang.String,double)" class="member-name-link">multiSpeciesAlignProtein(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, String, double)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>
<div class="block">Run a BlastP given a a list of sequence name, amino acid sequences, and a database name</div>
</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#multiSpeciesAlignTblastN(java.util.ArrayList,java.util.ArrayList,java.lang.String,double)" class="member-name-link">multiSpeciesAlignTblastN(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, String, double)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>
<div class="block">Run TblastN on multy amino acid sequence against nucleotide database that will be translated in
 the 6 frame</div>
</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
