<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 18:10:05 CEST 2019 -->
<title>N-Index</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="N-Index";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="contentContainer"><a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a><a id="I:N">
<!--   -->
</a>
<h2 class="title">N</h2>
<dl>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/EGDeWTdata.html#NAME_Deviation">NAME_Deviation</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/EGDeWTdata.html" title="class in bacnet.datamodel.dataset">EGDeWTdata</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/EGDeWTdata.html#NAME_Mean">NAME_Mean</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/EGDeWTdata.html" title="class in bacnet.datamodel.dataset">EGDeWTdata</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp"><span class="typeNameLink">NavigationManagement</span></a> - Class in <a href="../bacnet/raprcp/package-summary.html">bacnet.raprcp</a></dt>
<dd>
<div class="block">Every lmethods for managing navigation state in eclipse.rap application.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/NavigationManagement.html#%3Cinit%3E()">NavigationManagement()</a></span> - Constructor for class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DataSource.html#NBRF">NBRF</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DataSource.html#NCBI">NCBI</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/NCBIFastaHeaderParser.html" title="class in bacnet.reader"><span class="typeNameLink">NCBIFastaHeaderParser</span></a>&lt;<a href="../bacnet/reader/NCBIFastaHeaderParser.html" title="type parameter in NCBIFastaHeaderParser">S</a> extends <a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../bacnet/reader/NCBIFastaHeaderParser.html" title="type parameter in NCBIFastaHeaderParser">C</a>&gt;,&#8203;<a href="../bacnet/reader/NCBIFastaHeaderParser.html" title="type parameter in NCBIFastaHeaderParser">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../bacnet/reader/package-summary.html">bacnet.reader</a></dt>
<dd>
<div class="block">My own implementation of a FastaHeaderParserInterface to parse Fasta files
 from NCBI ftp

 NCBI fna (genome and plasmids) gi|gi-number|ref|accession|name
 gi|16802048|ref|NC_003210.1| Listeria monocytogenes EGD-e, complete genome
 NCBI ffn (coding) ref|refGenome gi-number|:start-end name
 ref|NC_003210.1|:318-1673 chromosomal replication initiation protein
 [Listeria monocytogenes EGD-e] NCBI frn (non-coding) ref|refGenome
 gi-number|:start-end| name| [locus-tag] ref|NC_003210|:82705-82777|Lys tRNA|
 [locus_tag=lmot01] NCBI faa (proteins) gi|gi-number|ref|accession|name
 gi|16802049|ref|NP_463534.1| chromosomal replication initiation protein
 [Listeria monocytogenes EGD-e]</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/NCBIFastaHeaderParser.html#%3Cinit%3E()">NCBIFastaHeaderParser()</a></span> - Constructor for class bacnet.reader.<a href="../bacnet/reader/NCBIFastaHeaderParser.html" title="class in bacnet.reader">NCBIFastaHeaderParser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence"><span class="typeNameLink">NcRNA</span></a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.SeqType.html#NcRNA">NcRNA</a></span> - bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/NcRNA.html#%3Cinit%3E()">NcRNA()</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/NcRNA.html#%3Cinit%3E(java.lang.String,int,int,char)">NcRNA(String, int, int, char)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/NcRNA.TypeNcRNA.html" title="enum in bacnet.datamodel.sequence"><span class="typeNameLink">NcRNA.TypeNcRNA</span></a> - Enum in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/Strand.html#NEGATIVE">NEGATIVE</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset"><span class="typeNameLink">Network</span></a> - Class in <a href="../bacnet/datamodel/dataset/package-summary.html">bacnet.datamodel.dataset</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Network.html#%3Cinit%3E()">Network()</a></span> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/HTMLUtils.html#NETWORK">NETWORK</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/HTMLUtils.html" title="class in bacnet.utils">HTMLUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/LocusTag.html#NewLocusTagArray">NewLocusTagArray</a></span> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/LocusTag.html" title="class in bacnet.datamodel.annotation">LocusTag</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/DNACompoundSet.html#newNucleotideCompound(java.lang.String,java.lang.String,java.lang.String...)">newNucleotideCompound(String, String, String...)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/DNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">DNACompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/RNACompoundSet.html#newNucleotideCompound(java.lang.String,java.lang.String,java.lang.String...)">newNucleotideCompound(String, String, String...)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/RNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">RNACompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html#next()">next()</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="class in org.biojava3.core.sequence.template">SequenceMixin.SequenceIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/LocIterator.html#next()">next()</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></dt>
<dd>
<div class="block">Get next window of default size, then increment position by default amount.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/LocIterator.html#next(int,int)">next(int, int)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></dt>
<dd>
<div class="block">Get next window of specified size, then increment position by specified
 amount.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset"><span class="typeNameLink">NGS</span></a> - Class in <a href="../bacnet/datamodel/dataset/package-summary.html">bacnet.datamodel.dataset</a></dt>
<dd>
<div class="block">Class for manipulating various RNASeq data, herited from ExpressionData <br>
 double[] value is of genome size, when ones want to see RNAseq expression at
 a certain bp position, just extract values[bpposition]<br>
 Need to be loaded first, and unlog has RNASeq will be logged for display when
 convert in the platform</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NGS.html#%3Cinit%3E()">NGS()</a></span> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NGS.html#%3Cinit%3E(java.lang.String,java.lang.String)">NGS(String, String)</a></span> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/NGSCreation.html" title="class in bacnet.scripts.database"><span class="typeNameLink">NGSCreation</span></a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/NGSCreation.html#%3Cinit%3E()">NGSCreation()</a></span> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/NGSCreation.html" title="class in bacnet.scripts.database">NGSCreation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.TypeModif.html#NH2">NH2</a></span> - bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/NimbleGen.html" title="class in bacnet.scripts.listeriomics.technology"><span class="typeNameLink">NimbleGen</span></a> - Class in <a href="../bacnet/scripts/listeriomics/technology/package-summary.html">bacnet.scripts.listeriomics.technology</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/NimbleGen.html#%3Cinit%3E()">NimbleGen()</a></span> - Constructor for class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/NimbleGen.html" title="class in bacnet.scripts.listeriomics.technology">NimbleGen</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#nonOverlappingKmers(org.biojava3.core.sequence.template.Sequence,int)">nonOverlappingKmers(Sequence&lt;C&gt;, int)</a></span> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Produces kmers of the specified size e.g.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/MedianNormalization.html#norm(bacnet.datamodel.dataset.ExpressionMatrix)">norm(ExpressionMatrix)</a></span> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/MedianNormalization.html" title="class in bacnet.scripts.core.normalization">MedianNormalization</a></dt>
<dd>
<div class="block">For each column of the ExpressionMatrix, delete the median.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/MedianNormalization.html#norm(bacnet.datamodel.dataset.ExpressionMatrix,java.lang.String)">norm(ExpressionMatrix, String)</a></span> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/MedianNormalization.html" title="class in bacnet.scripts.core.normalization">MedianNormalization</a></dt>
<dd>
<div class="block">Normalize by median the column with a specific name</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/VarianceNormalization.html#norm(bacnet.datamodel.dataset.ExpressionMatrix,java.lang.String)">norm(ExpressionMatrix, String)</a></span> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/VarianceNormalization.html" title="class in bacnet.scripts.core.normalization">VarianceNormalization</a></dt>
<dd>
<div class="block">Normalize by variance the column with a specific name<br>
 Multiply each value by Math.sqr(1/var(x))</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/GENormalization.html#norm(bacnet.datamodel.expdesign.Experiment)">norm(Experiment)</a></span> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/GENormalization.html" title="class in bacnet.scripts.core.normalization">GENormalization</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/GRNormalization.html#norm(bacnet.datamodel.expdesign.Experiment)">norm(Experiment)</a></span> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/GRNormalization.html" title="class in bacnet.scripts.core.normalization">GRNormalization</a></dt>
<dd>
<div class="block">Normalize the Tiling contained in the Experiment <br>
 Run one thread for each bioCondition<br>
 The number of thread is limited by the number of processor<br>
 <br>
 
 It runs DNA reference normalization (Hubert et al.), it can uses also an
 additional reference data for normalization (indicates by boolean
 normalization).</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/Macroarray.html#normalize(java.lang.String,java.lang.String)">normalize(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Macroarray.html" title="class in bacnet.scripts.listeriomics.technology">Macroarray</a></dt>
<dd>
<div class="block">Data are not Log transformed:<br>
 add the min value to be sure to have only positive values
 log2 transform</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/NimbleGen.html#normalize(java.lang.String,java.lang.String)">normalize(String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/NimbleGen.html" title="class in bacnet.scripts.listeriomics.technology">NimbleGen</a></dt>
<dd>
<div class="block">Data are not Log transformed:<br>
 add the min value to be sure to have only positive values
 log2 transform</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/Agilent.html#normalize(java.lang.String,java.lang.String,java.lang.String)">normalize(String, String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Agilent.html" title="class in bacnet.scripts.listeriomics.technology">Agilent</a></dt>
<dd>
<div class="block">Data are not Log transformed:<br>
 add the min value to be sure to have only positive values
 log2 transform</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/NGSCreation.html#normalizeCountFiles(bacnet.datamodel.expdesign.Experiment)">normalizeCountFiles(Experiment)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/NGSCreation.html" title="class in bacnet.scripts.database">NGSCreation</a></dt>
<dd>
<div class="block">ConvertCountFile and add them to the database</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/MedianNormalization.html#normGE(bacnet.datamodel.dataset.ExpressionMatrix)">normGE(ExpressionMatrix)</a></span> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/MedianNormalization.html" title="class in bacnet.scripts.core.normalization">MedianNormalization</a></dt>
<dd>
<div class="block">From a GE ExpressionMatrix containing 5 columns (3replicates exp + median +
 stdev) Apply median normalization to the first 3 columns, and recalculate
 median and stdev</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/MedianNormalization.html#normGE(bacnet.datamodel.expdesign.Experiment)">normGE(Experiment)</a></span> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/MedianNormalization.html" title="class in bacnet.scripts.core.normalization">MedianNormalization</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/MedianNormalization.html#normTiling(bacnet.datamodel.expdesign.Experiment)">normTiling(Experiment)</a></span> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/MedianNormalization.html" title="class in bacnet.scripts.core.normalization">MedianNormalization</a></dt>
<dd>
<div class="block">Load all TilingData in this Experiment and normalize by the median, and save
 in the Tiling/MedianNorm path</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#NOTFOUND_LOCUSTAG_CHANGE_PATH">NOTFOUND_LOCUSTAG_CHANGE_PATH</a></span> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics"><span class="typeNameLink">NTerm</span></a> - Class in <a href="../bacnet/datamodel/proteomics/package-summary.html">bacnet.datamodel.proteomics</a></dt>
<dd>
<div class="block">Object for managing every mapped sequences</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html#NTerm">NTerm</a></span> - bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#%3Cinit%3E()">NTerm()</a></span> - Constructor for class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#%3Cinit%3E(java.lang.String,int,int,char)">NTerm(String, int, int, char)</a></span> - Constructor for class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#NTERM_PROJECT_BHI">NTERM_PROJECT_BHI</a></span> - Static variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html" title="enum in bacnet.datamodel.proteomics"><span class="typeNameLink">NTerm.ExperimentName</span></a> - Enum in <a href="../bacnet/datamodel/proteomics/package-summary.html">bacnet.datamodel.proteomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics"><span class="typeNameLink">NTerm.MappingFrame</span></a> - Enum in <a href="../bacnet/datamodel/proteomics/package-summary.html">bacnet.datamodel.proteomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics"><span class="typeNameLink">NTerm.TypeModif</span></a> - Enum in <a href="../bacnet/datamodel/proteomics/package-summary.html">bacnet.datamodel.proteomics</a></dt>
<dd>
<div class="block">Type of modification at the start of a peptide</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermConservationWithRiboProfiling.html" title="class in bacnet.scripts.listeriomics.nterm"><span class="typeNameLink">NTermConservationWithRiboProfiling</span></a> - Class in <a href="../bacnet/scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></dt>
<dd>
<div class="block">Run different method to compare iTIS position (internal translation) to data
 of Ribosome profiling in Bacillus and Ecoli:<br>
 compare the gene by BlastN
 find internal translation in the ribosome profiling
 find internal SD sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermConservationWithRiboProfiling.html#%3Cinit%3E()">NTermConservationWithRiboProfiling()</a></span> - Constructor for class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermConservationWithRiboProfiling.html" title="class in bacnet.scripts.listeriomics.nterm">NTermConservationWithRiboProfiling</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics"><span class="typeNameLink">NTermCreateData</span></a> - Class in <a href="../bacnet/datamodel/proteomics/package-summary.html">bacnet.datamodel.proteomics</a></dt>
<dd>
<div class="block">Transform excel table describing all N-Terminal peptides found in
 Listeria<br>
 Find the position of each peptide in Listeria genome
 create a NGS data from it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#%3Cinit%3E(bacnet.datamodel.sequence.Genome)">NTermCreateData(Genome)</a></span> - Constructor for class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">Initialize a void NTermCreateData object and run
 <code>initializeGenomeSequence()</code> to have the genome on the six frames</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#%3Cinit%3E(java.lang.String,java.lang.String)">NTermCreateData(String, String)</a></span> - Constructor for class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">Create table summarizing the data<br>
 Curate the table by adding different information<br>
 Create streaming data for the viewer<br>
 
 Run Mapping and creation of summary tables<br>
 mapPeptide();<br>
 curateTable();<br>
 <br>
 Create all NTerm from the table<br>
 createNTermExperiment();<br>
 NTermCreation.findStart(nameNTermData);<br>
 NTermCreation.findTSS(nameNTermData);<br>
 NTermCreation.findCodonUsage(nameNTermData);<br>
 NTermCreation.findSDSequence(nameNTermData);<br>
 <br>
 Complete NTermExp by creating all HashMaps<br>
 NTermData nTermExp = NTermData.load(nameNTermData);<br>
 nTermExp.createHashMap();<br>
 nTermExp.createGeneHashMap();<br>
 nTermExp.createProteolyseMap();<br>
 nTermExp.createAnnotation();<br>
 nTermExp.save();<br></div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html" title="class in bacnet.scripts.listeriomics.nterm"><span class="typeNameLink">NTermCreateFinalData</span></a> - Class in <a href="../bacnet/scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></dt>
<dd>
<div class="block">Genertal class for creation of NTerminomics tables</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html#%3Cinit%3E(java.lang.String,java.lang.String)">NTermCreateFinalData(String, String)</a></span> - Constructor for class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html" title="class in bacnet.scripts.listeriomics.nterm">NTermCreateFinalData</a></dt>
<dd>
<div class="block">From the list created by Francis, update all Nterm peptides, indicating which
 type of TIS (translation initiation sites) they are: annotated TIS multiple
 TIS updated TIS internal TIS</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset"><span class="typeNameLink">NTermData</span></a> - Class in <a href="../bacnet/datamodel/dataset/package-summary.html">bacnet.datamodel.dataset</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#%3Cinit%3E()">NTermData()</a></span> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#%3Cinit%3E(java.lang.String)">NTermData(String)</a></span> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#%3Cinit%3E(java.lang.String,java.util.HashMap)">NTermData(String, HashMap&lt;String, NTerm&gt;)</a></span> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm"><span class="typeNameLink">NTermDatabase</span></a> - Class in <a href="../bacnet/scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#%3Cinit%3E()">NTermDatabase()</a></span> - Constructor for class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics"><span class="typeNameLink">NTermFilter</span></a> - Class in <a href="../bacnet/datamodel/proteomics/package-summary.html">bacnet.datamodel.proteomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermFilter.html#%3Cinit%3E(bacnet.datamodel.dataset.NTermData,java.lang.String)">NTermFilter(NTermData, String)</a></span> - Constructor for class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>
<div class="block">Filter a NTermData using the table of filter found in fileName</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser"><span class="typeNameLink">NTerminomicsView</span></a> - Class in <a href="../bacnet/genomeBrowser/package-summary.html">bacnet.genomeBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#%3Cinit%3E()">NTerminomicsView()</a></span> - Constructor for class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermMain.html" title="class in bacnet.scripts.listeriomics.nterm"><span class="typeNameLink">NTermMain</span></a> - Class in <a href="../bacnet/scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></dt>
<dd>
<div class="block">General method for running all methods related to NTerm project</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermMain.html#%3Cinit%3E()">NTermMain()</a></span> - Constructor for class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermMain.html" title="class in bacnet.scripts.listeriomics.nterm">NTermMain</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm"><span class="typeNameLink">NTermStat</span></a> - Class in <a href="../bacnet/scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></dt>
<dd>
<div class="block">List of method to do statistics about common features in N-Terminus of genes:
 - start codon - SD binding - AU proportion - Folding energy</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#%3Cinit%3E()">NTermStat()</a></span> - Constructor for class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStatProteolyse.html" title="class in bacnet.scripts.listeriomics.nterm"><span class="typeNameLink">NTermStatProteolyse</span></a> - Class in <a href="../bacnet/scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></dt>
<dd>
<div class="block">A list of methods to assess the quantity of proteolysis on the peptides</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStatProteolyse.html#%3Cinit%3E()">NTermStatProteolyse()</a></span> - Constructor for class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStatProteolyse.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStatProteolyse</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermSummaryTable.html" title="class in bacnet.datamodel.proteomics"><span class="typeNameLink">NTermSummaryTable</span></a> - Class in <a href="../bacnet/datamodel/proteomics/package-summary.html">bacnet.datamodel.proteomics</a></dt>
<dd>
<div class="block">List of methods for creating Summary Table for TIS <br>
 Using NTermFilter we have regroup TIS in different overlap groups, which we
 use now to create the Summary Tables.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermSummaryTable.html#%3Cinit%3E()">NTermSummaryTable()</a></span> - Constructor for class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermSummaryTable.html" title="class in bacnet.datamodel.proteomics">NTermSummaryTable</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermTranscriptome.html" title="class in bacnet.scripts.listeriomics.nterm"><span class="typeNameLink">NTermTranscriptome</span></a> - Class in <a href="../bacnet/scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermTranscriptome.html#%3Cinit%3E()">NTermTranscriptome()</a></span> - Constructor for class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermTranscriptome.html" title="class in bacnet.scripts.listeriomics.nterm">NTermTranscriptome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics"><span class="typeNameLink">NTermUtils</span></a> - Class in <a href="../bacnet/datamodel/proteomics/package-summary.html">bacnet.datamodel.proteomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermUtils.html#%3Cinit%3E()">NTermUtils()</a></span> - Constructor for class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound"><span class="typeNameLink">NucleotideCompound</span></a> - Class in <a href="../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,java.lang.String)">NucleotideCompound(String, CompoundSet&lt;NucleotideCompound&gt;, String)</a></span> - Constructor for class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,java.lang.String,org.biojava3.core.sequence.compound.NucleotideCompound%5B%5D)">NucleotideCompound(String, CompoundSet&lt;NucleotideCompound&gt;, String, NucleotideCompound[])</a></span> - Constructor for class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a></div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
