<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Fri Dec 10 09:15:34 CET 2021 -->
<title>L-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-10">
<meta name="description" content="index: L">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:L">L</h2>
<dl class="index">
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#Lactobacillales_NWK" class="member-name-link">Lactobacillales_NWK</a> - Static variable in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#lastIndexOf(org.biojava3.core.sequence.template.Sequence,C)" class="member-name-link">lastIndexOf(Sequence&lt;C&gt;, C)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Performs a reversed linear search of the given Sequence by wrapping it in a
 <a href="../org/biojava3/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views"><code>ReversedSequenceView</code></a> and passing it into <a href="../org/biojava3/core/sequence/template/SequenceMixin.html#indexOf(org.biojava3.core.sequence.template.Sequence,C)"><code>SequenceMixin.indexOf(Sequence, Compound)</code></a>.</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" class="type-name-link" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a> - Class in <a href="../bacnet/genomeBrowser/dialog/package-summary.html">bacnet.genomeBrowser.dialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,bacnet.genomeBrowser.core.Track)" class="member-name-link">LegendDialog(Shell, Track)</a> - Constructor for class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>
<div class="block">Create the dialog.</div>
</dd>
<dt><a href="../bacnet/Database.html#LEISHOMICS_PROJECT" class="member-name-link">LEISHOMICS_PROJECT</a> - Static variable in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Static variable to discriminate Listeriomics project from the others</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#length()" class="member-name-link">length()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Get length of range.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#LENGTH" class="member-name-link">LENGTH</a> - Static variable in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">Sort features by length.</div>
</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html#LENGTH_CUTOFF" class="member-name-link">LENGTH_CUTOFF</a> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#LENGTH_CUTOFF" class="member-name-link">LENGTH_CUTOFF</a> - Static variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LIGHT_ASRNA" class="member-name-link">LIGHT_ASRNA</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Light color used for ASrna</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LIGHT_CISREG" class="member-name-link">LIGHT_CISREG</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Light color used for NCrna</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LIGHT_NCRNA" class="member-name-link">LIGHT_NCRNA</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Light color used for NCrna</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LIGHT_ONE" class="member-name-link">LIGHT_ONE</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LIGHT_OPERON" class="member-name-link">LIGHT_OPERON</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Light color used for ASrna</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LIGHT_SRNA" class="member-name-link">LIGHT_SRNA</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Light color used for Srna</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LIGHT_THREE" class="member-name-link">LIGHT_THREE</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LIGHT_TWO" class="member-name-link">LIGHT_TWO</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LIGHTBLUE" class="member-name-link">LIGHTBLUE</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LIGHTERGREY" class="member-name-link">LIGHTERGREY</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LIGHTGREY" class="member-name-link">LIGHTGREY</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.html" class="type-name-link" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a>&lt;<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="type parameter in LightweightProfile">S</a> extends <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="type parameter in LightweightProfile">C</a>&gt;,<wbr><a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="type parameter in LightweightProfile">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>
<div class="block">Defines a minimal data structure for reading and writing a sequence alignment.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html" class="type-name-link" title="enum in org.biojava3.core.sequence.template">LightweightProfile.StringFormat</a> - Enum in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>
<div class="block">List of output formats.</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LINE_ASRNA" class="member-name-link">LINE_ASRNA</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Very dark color used for ASrna surrounding</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LINE_CISREG" class="member-name-link">LINE_CISREG</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Very dark color used for NCrna surrounding</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LINE_NCRNA" class="member-name-link">LINE_NCRNA</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Very dark color used for NCrna surrounding</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LINE_OPERON" class="member-name-link">LINE_OPERON</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Very dark color used for NCrna surrounding</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#LINE_SRNA" class="member-name-link">LINE_SRNA</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Very dark color used for Srna surrounding</div>
</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#LIST" class="member-name-link">LIST</a> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#LIST_PATH" class="member-name-link">LIST_PATH</a> - Static variable in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#LIST_PATH" class="member-name-link">LIST_PATH</a> - Static variable in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/HideExpressionMatrixDialog.html#listAddColumn" class="member-name-link">listAddColumn</a> - Variable in class bacnet.table.gui.<a href="../bacnet/table/gui/HideExpressionMatrixDialog.html" title="class in bacnet.table.gui">HideExpressionMatrixDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/HideExpressionMatrixDialog.html#listAddRow" class="member-name-link">listAddRow</a> - Variable in class bacnet.table.gui.<a href="../bacnet/table/gui/HideExpressionMatrixDialog.html" title="class in bacnet.table.gui">HideExpressionMatrixDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#listAllDB(java.lang.String)" class="member-name-link">listAllDB(String)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#LISTERIOMICS_PROJECT" class="member-name-link">LISTERIOMICS_PROJECT</a> - Static variable in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Static variable to discriminate Listeriomics project from the others</div>
</dd>
<dt><a href="../bacnet/table/gui/HideExpressionMatrixDialog.html#listHideColumn" class="member-name-link">listHideColumn</a> - Variable in class bacnet.table.gui.<a href="../bacnet/table/gui/HideExpressionMatrixDialog.html" title="class in bacnet.table.gui">HideExpressionMatrixDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/HideExpressionMatrixDialog.html#listHideRow" class="member-name-link">listHideRow</a> - Variable in class bacnet.table.gui.<a href="../bacnet/table/gui/HideExpressionMatrixDialog.html" title="class in bacnet.table.gui">HideExpressionMatrixDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#LISTNOTINCLUDE" class="member-name-link">LISTNOTINCLUDE</a> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ListUtils.html" class="type-name-link" title="class in bacnet.utils">ListUtils</a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ListUtils.html#%3Cinit%3E()" class="member-name-link">ListUtils()</a> - Constructor for class bacnet.utils.<a href="../bacnet/utils/ListUtils.html" title="class in bacnet.utils">ListUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#LNFCWT" class="member-name-link">LNFCWT</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#load()" class="member-name-link">load()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Load information on the data: name, date, note, stat, etc ...</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#load()" class="member-name-link">load()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#load()" class="member-name-link">load()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#load()" class="member-name-link">load()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>
<div class="block">Load information on the datasets: name, date, note, stat, etc ...</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#load()" class="member-name-link">load()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Load NTermData from PATH_STREAMING+ this.getName() +EXTENSION</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ProteomicsData.html#load()" class="member-name-link">load()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a></dt>
<dd>
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#load()" class="member-name-link">load()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#load(boolean)" class="member-name-link">load(boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Load information on the data: name, date, note, stat, etc ...</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Load information on the data: name, date, note, stat, etc ...</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Network.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>
<div class="block">Load information of the data</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Load in: TranscriptomeData.PATH_STREAMING + nTermExpName + TranscriptomeData.EXTENSION
 
 Read compressed, serialized data with a FileInputStream.</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>
<div class="block">Load a serialized Gene</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/NcRNA.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Operon.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Static method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Static method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>
<div class="block">Load Track from a serialized file</div>
</dd>
<dt><a href="../bacnet/table/core/Filter.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Static method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>
<div class="block">Prompt for a filename, and load a scribble from that file.</div>
</dd>
<dt><a href="../bacnet/utils/Filter.html#load(java.lang.String)" class="member-name-link">load(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>
<div class="block">Prompt for a filename, and load a scribble from that file.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#load(java.lang.String,boolean)" class="member-name-link">load(String, boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Load information on the data: name, date, note, stat, etc ...</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#load(java.lang.String,org.eclipse.swt.widgets.Shell)" class="member-name-link">load(String, Shell)</a> - Static method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#load(java.lang.String,org.eclipse.swt.widgets.Shell)" class="member-name-link">load(String, Shell)</a> - Static method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewHomologTools.html#loadArrayHomologs(bacnet.datamodel.sequence.Sequence,java.lang.String%5B%5D%5B%5D,java.util.ArrayList,java.util.ArrayList)" class="member-name-link">loadArrayHomologs(Sequence, String[][], ArrayList&lt;String[]&gt;, ArrayList&lt;String[]&gt;)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneViewHomologTools.html" title="class in bacnet.sequenceTools">GeneViewHomologTools</a></dt>
<dd>
<div class="block">Load genome info and add homolog information on the column index1 and 2</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#loadCorrespondingChromosome(java.lang.String)" class="member-name-link">loadCorrespondingChromosome(String)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">load and return the chromosome corresponding to an accession</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#loadEgdeGenome()" class="member-name-link">loadEgdeGenome()</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Find in ModelProvider if Egde Genome is already loaded If not search EGDE_NAME in GENOMEDIR</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#loadEgdeGenome()" class="member-name-link">loadEgdeGenome()</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Find in ModelProvider if Egde Genome is already loaded If not search EGDE_NAME in GENOMEDIR</div>
</dd>
<dt><a href="../bacnet/handler/LoadExpressionMatrixHandler.html" class="type-name-link" title="class in bacnet.handler">LoadExpressionMatrixHandler</a> - Class in <a href="../bacnet/handler/package-summary.html">bacnet.handler</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/handler/LoadExpressionMatrixHandler.html#%3Cinit%3E()" class="member-name-link">LoadExpressionMatrixHandler()</a> - Constructor for class bacnet.handler.<a href="../bacnet/handler/LoadExpressionMatrixHandler.html" title="class in bacnet.handler">LoadExpressionMatrixHandler</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html#loadFastaAddGeneFeaturesFromGeneIDGFF2(java.io.File,java.io.File)" class="member-name-link">loadFastaAddGeneFeaturesFromGeneIDGFF2(File, File)</a> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>
<div class="block">Loads Fasta file and GFF2 feature file generated from the geneid prediction algorithm</div>
</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html#loadFastaAddGeneFeaturesFromGeneMarkGTF(java.io.File,java.io.File)" class="member-name-link">loadFastaAddGeneFeaturesFromGeneMarkGTF(File, File)</a> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html#loadFastaAddGeneFeaturesFromGlimmerGFF3(java.io.File,java.io.File)" class="member-name-link">loadFastaAddGeneFeaturesFromGlimmerGFF3(File, File)</a> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html#loadFastaAddGeneFeaturesFromGmodGFF3(java.io.File,java.io.File,boolean)" class="member-name-link">loadFastaAddGeneFeaturesFromGmodGFF3(File, File, boolean)</a> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>
<div class="block">Lots of variations in the ontology or descriptors that can be used in GFF3 which requires writing
 a custom parser to handle a GFF3 generated or used by a specific application.</div>
</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html#loadFastaAddGeneFeaturesFromUpperCaseExonFastaFile(java.io.File,java.io.File,boolean)" class="member-name-link">loadFastaAddGeneFeaturesFromUpperCaseExonFastaFile(File, File, boolean)</a> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewHomologTools.html#loadFigureHomologs(bacnet.datamodel.sequence.Gene,org.eclipse.swt.browser.Browser,java.util.ArrayList)" class="member-name-link">loadFigureHomologs(Gene, Browser, ArrayList&lt;String&gt;)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneViewHomologTools.html" title="class in bacnet.sequenceTools">GeneViewHomologTools</a></dt>
<dd>
<div class="block">Load SVG figure of phylogeny and replace all strain name by homolog informations<br>
 Save to JPG file and display it</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#loadFromText(java.lang.String)" class="member-name-link">loadFromText(String)</a> - Static method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>
<div class="block">Load a Track from a text file</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#loadGeneFromProteinId(java.lang.String)" class="member-name-link">loadGeneFromProteinId(String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Load proteinIdnToLocusTag hashmap for HomologCreation</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#loadGenome(java.lang.String)" class="member-name-link">loadGenome(String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Find in ModelProvider if Genome Accession is already loaded If not search load it</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#loadGenome(java.lang.String)" class="member-name-link">loadGenome(String)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Find in ModelProvider if Genome Accession is already loaded If not search load it</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#loadGenome(java.lang.String,boolean)" class="member-name-link">loadGenome(String, boolean)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Find in ModelProvider if Genome Accession is already loaded If not search load it
 If keepInMemory is true : Find in ModelProvider if Genome Accession is already loaded if not load
 it <br>
 If keepInMemory is false : Only load the Genome without saving it <br>
 <br></div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#loadGenome(java.lang.String,java.lang.String,boolean,boolean)" class="member-name-link">loadGenome(String, String, boolean, boolean)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Load/return a genome given by a genomeAccession ID<br>
 <br>
 
 If keepInMemory is true : Find in ModelProvider if Genome Accession is already loaded if not load
 it <br>
 If keepInMemory is false : Only load the Genome without saving it <br>
 <br>
 If annotation is true : Load annotation information of the genome <br>
 If anotation is false : Only load the sequence of the genome</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#loadGenome(java.lang.String,java.lang.String,boolean,boolean)" class="member-name-link">loadGenome(String, String, boolean, boolean)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Load/return a genome given by a genomeAccession ID<br>
 <br>
 
 If keepInMemory is true : Find in ModelProvider if Genome Accession is already loaded if not load
 it <br>
 If keepInMemory is false : Only load the Genome without saving it <br>
 <br>
 If annotation is true : Load annotation information of the genome <br>
 If anotation is false : Only load the sequence of the genome</div>
</dd>
<dt><a href="../bacnet/views/InternalBrowser.html#loadHTML(java.lang.String)" class="member-name-link">loadHTML(String)</a> - Method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneViewLocalizationTools.html#loadLocalizationFigure(org.eclipse.swt.browser.Browser,java.lang.String%5B%5D%5B%5D,bacnet.datamodel.sequence.Sequence,java.lang.String%5B%5D%5B%5D,org.eclipse.swt.widgets.Table,bacnet.datamodel.sequence.Genome)" class="member-name-link">loadLocalizationFigure(Browser, String[][], Sequence, String[][], Table, Genome)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneViewLocalizationTools.html" title="class in bacnet.sequenceTools">GeneViewLocalizationTools</a></dt>
<dd>
<div class="block">Load SVG figure of phylogeny and replace all strain name by homolog informations<br>
 Save to JPG file and display it</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#loadMatrix()" class="member-name-link">loadMatrix()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Load ExpressionMatrix found in TranscriptomeData.PATH_GENEXPR_NORM+getName()</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#loadMatrix()" class="member-name-link">loadMatrix()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Load ExpressionMatrix found in TranscriptomeData.PATH_TILING_NORM + getName()</div>
</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#loadNetwork()" class="member-name-link">loadNetwork()</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>
<div class="block">Load the general Network</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#loadOldLocusTagFromLocusTag(java.lang.String)" class="member-name-link">loadOldLocusTagFromLocusTag(String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Load locusTagToOldLocusTag hashmap for HomologCreation</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#loadOldLocusTagFromProteinId(java.lang.String)" class="member-name-link">loadOldLocusTagFromProteinId(String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Load proteinIDToOldLocusTag hashmap for HomologCreation</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#loadOperonsAndTerminator()" class="member-name-link">loadOperonsAndTerminator()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>
<div class="block">Add Operons and Terminator from Annotation.EGDE_SUPPTABLE <br>
 This table comes directly from Wurtzel et al. annotation table</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#loadPhylogenomicFigure(java.util.ArrayList)" class="member-name-link">loadPhylogenomicFigure(ArrayList&lt;String&gt;)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>
<div class="block">Load SVG figure of phylogeny and replace all strain name by homolog
 informations<br>
 Save to JPG file and display it</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#loadSignaturesNametoID(java.lang.String)" class="member-name-link">loadSignaturesNametoID(String)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>
<div class="block">Load <code>TreeMap&lt;String,String> signaturesNametoID</code> from text file in
 <code>LIST_PATH</code><br>
 Used for initiating DATABASE.INSTANCE data provider</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#loadSrnas()" class="member-name-link">loadSrnas()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>
<div class="block">Load all serialized Srna object found in Srna.PATHSerialize</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#loadTab(java.lang.String,boolean)" class="member-name-link">loadTab(String, boolean)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Load an ExpressionMatrix from a Tabdelimited text file</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#loadText(java.lang.String)" class="member-name-link">loadText(String)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>
<div class="block">Read a formatted text file, and create the corresponding signature object <br>
 Each field of this text file, has to be separated by "#"</div>
</dd>
<dt><a href="../bacnet/views/InternalBrowser.html#loadURL(java.lang.String)" class="member-name-link">loadURL(String)</a> - Method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLHelper.html#loadXML(java.lang.String)" class="member-name-link">loadXML(String)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DataSource.html#LOCAL" class="member-name-link">LOCAL</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/SubCellCompartment.html#LOCALIZATION_LEGEND_PATH" class="member-name-link">LOCALIZATION_LEGEND_PATH</a> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.html" title="class in bacnet.datamodel.annotation">SubCellCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/SubCellCompartment.html#LOCALIZATION_PATH" class="member-name-link">LOCALIZATION_PATH</a> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.html" title="class in bacnet.datamodel.annotation">SubCellCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/LocalizationDialog.html" class="type-name-link" title="class in bacnet.sequenceTools">LocalizationDialog</a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/LocalizationDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,java.lang.String%5B%5D%5B%5D,bacnet.datamodel.sequence.Sequence,java.lang.String%5B%5D%5B%5D,bacnet.datamodel.sequence.Genome)" class="member-name-link">LocalizationDialog(Shell, String[][], Sequence, String[][], Genome)</a> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/LocalizationDialog.html" title="class in bacnet.sequenceTools">LocalizationDialog</a></dt>
<dd>
<div class="block">Create the dialog.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#location()" class="member-name-link">location()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Get location of feature.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureI.html#location()" class="member-name-link">location()</a> - Method in interface org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a></dt>
<dd>
<div class="block">Get the location of the feature.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/LocationHelper.html#location(int,int,org.biojava3.core.sequence.Strand,int)" class="member-name-link">location(int, int, Strand, int)</a> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Returns a location object which unlike the location constructors allows you to input reverse
 coordinates and will convert these into the right location on the positive strand.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.Tools.html#location(int,int,org.biojava3.core.sequence.Strand,int)" class="member-name-link">location(int, int, Strand, int)</a> - Static method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.Tools.html" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a></dt>
<dd>
<div class="block">Returns a location object which unlike the location constructors allows you to input reverse
 coordinates and will convert these into the right location on the positive strand.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/LocationHelper.html#location(java.util.List)" class="member-name-link">location(List&lt;Location&gt;)</a> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Used as a thin wrapper to the <a href="../org/biojava3/core/sequence/location/LocationHelper.html#location(java.util.List,java.lang.String)"><code>LocationHelper.location(java.util.List, java.lang.String)</code></a> method to
 bring the given location list together as a join (the default type)</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.Tools.html#location(java.util.List,java.lang.Integer,java.lang.String)" class="member-name-link">location(List&lt;Location&gt;, Integer, String)</a> - Static method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.Tools.html" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a></dt>
<dd>
<div class="block">Used for building a location from a series of sub-locations</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/LocationHelper.html#location(java.util.List,java.lang.String)" class="member-name-link">location(List&lt;Location&gt;, String)</a> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Builds a location from a List of locations; this can be circular or linear joins.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html" class="type-name-link" title="class in org.biojava3.genome.parsers.gff">Location</a> - Class in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>
<div class="block">A location on a sequence.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.html" class="type-name-link" title="interface in org.biojava3.core.sequence.location.template">Location</a> - Interface in <a href="../org/biojava3/core/sequence/location/template/package-summary.html">org.biojava3.core.sequence.location.template</a></dt>
<dd>
<div class="block">Sets of integers used to represent the location of features on sequence.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#%3Cinit%3E(int,int)" class="member-name-link">Location(int, int)</a> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Construct new location from coordinates.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#%3Cinit%3E(org.biojava3.genome.parsers.gff.Location)" class="member-name-link">Location(Location)</a> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Clone other location.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#LOCATION_LENGTH" class="member-name-link">LOCATION_LENGTH</a> - Static variable in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">Sort features by start position and then longest length.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.Tools.html" class="type-name-link" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a> - Class in <a href="../org/biojava3/core/sequence/location/template/package-summary.html">org.biojava3.core.sequence.location.template</a></dt>
<dd>
<div class="block">Helper methods for use with the Location classes.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/LocationHelper.html" class="type-name-link" title="class in org.biojava3.core.sequence.location">LocationHelper</a> - Class in <a href="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</a></dt>
<dd>
<div class="block">Helper methods for use with the Location classes.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/LocationHelper.html#%3Cinit%3E()" class="member-name-link">LocationHelper()</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/LocIterator.html" class="type-name-link" title="class in org.biojava3.genome.parsers.gff">LocIterator</a> - Class in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>
<div class="block">Move a sliding window over a Location.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/LocIterator.html#%3Cinit%3E(org.biojava3.genome.parsers.gff.Location,int,int)" class="member-name-link">LocIterator(Location, int, int)</a> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></dt>
<dd>
<div class="block">Construct an iterator that slides a window over a Location.</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/LocusTag.html" class="type-name-link" title="class in bacnet.datamodel.annotation">LocusTag</a> - Class in <a href="../bacnet/datamodel/annotation/package-summary.html">bacnet.datamodel.annotation</a></dt>
<dd>
<div class="block">Since the update of RefSeq database all the locustag have been changed <br>
 Here is some methods to deal with this change</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/LocusTag.html#%3Cinit%3E()" class="member-name-link">LocusTag()</a> - Constructor for class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/LocusTag.html" title="class in bacnet.datamodel.annotation">LocusTag</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#LOCUSTAG_CHANGE_PATH" class="member-name-link">LOCUSTAG_CHANGE_PATH</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#log10(double%5B%5D)" class="member-name-link">log10(double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">log2(value)</div>
</dd>
<dt><a href="../bacnet/utils/MathUtils.html#log2(double)" class="member-name-link">log2(double)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/MathUtils.html" title="class in bacnet.utils">MathUtils</a></dt>
<dd>
<div class="block">Calculate log with base 2</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#log2(double%5B%5D)" class="member-name-link">log2(double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">log2(value)</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#log2(double%5B%5D%5B%5D)" class="member-name-link">log2(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">log2(value)</div>
</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#log2(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">log2(ExpressionMatrix)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">log2(values)</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/StatUtils.html#logFC(double%5B%5D,double%5B%5D)" class="member-name-link">logFC(double[], double[])</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatUtils.html" title="class in bacnet.scripts.core.stat">StatUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html#LOGFC" class="member-name-link">LOGFC</a> - Enum constant in enum bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html" title="enum in bacnet.datamodel.dataset">OmicsData.ColNames</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#LOGFC" class="member-name-link">LOGFC</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.TypeMapper.html#LOGFC" class="member-name-link">LOGFC</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.TypeMapper.html" title="enum in bacnet.table.core">ColorMapper.TypeMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.Type.html#LOGFC" class="member-name-link">LOGFC</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.Type.html" title="enum in bacnet.table.core">ColorMapperRCP.Type</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ListUtils.html#longestCommonSub(java.lang.String,java.lang.String)" class="member-name-link">longestCommonSub(String, String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ListUtils.html" title="class in bacnet.utils">ListUtils</a></dt>
<dd>
<div class="block">The longest common subsequence (or LCS) of groups A and B is the longest group of elements from A
 and B that are common between the two groups and in the same order in each group.<br>
 For example, the sequences "1234" and "1224533324" have an LCS of "1234"<br>
 <code>Code extract from http://rosettacode.org/wiki/Rosetta_Code</code></div>
</dd>
<dt><a href="../bacnet/utils/ListUtils.html#longestCommonSub(java.util.ArrayList)" class="member-name-link">longestCommonSub(ArrayList&lt;String&gt;)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ListUtils.html" title="class in bacnet.utils">ListUtils</a></dt>
<dd>
<div class="block">From a list of String extract the longest common subsequence</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/LPE.html" class="type-name-link" title="class in bacnet.scripts.core.stat">LPE</a> - Class in <a href="../bacnet/scripts/core/stat/package-summary.html">bacnet.scripts.core.stat</a></dt>
<dd>
<div class="block">Class for calculating LPE statistics <br>
 Inspired from R library LPE<br></div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#LPE" class="member-name-link">LPE</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/LPE.html#%3Cinit%3E()" class="member-name-link">LPE()</a> - Constructor for class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/LPE.html" title="class in bacnet.scripts.core.stat">LPE</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/LT.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.technology">LT</a> - Class in <a href="../bacnet/scripts/listeriomics/technology/package-summary.html">bacnet.scripts.listeriomics.technology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/LT.html#%3Cinit%3E()" class="member-name-link">LT()</a> - Constructor for class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/LT.html" title="class in bacnet.scripts.listeriomics.technology">LT</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
