<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>T-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="index: T">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:T">T</h2>
<dl class="index">
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html" class="type-name-link" title="class in bacnet.reader">TabDelimitedTableReader</a> - Class in <a href="../bacnet/reader/package-summary.html">bacnet.reader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#%3Cinit%3E()" class="member-name-link">TabDelimitedTableReader()</a> - Constructor for class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#table(java.lang.Integer)" class="member-name-link">table(Integer)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>
<div class="block">Uses the static instance of <a href="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io"><code>IUPACParser</code></a> to find instances of <a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io"><code>IUPACParser.IUPACTable</code></a>s by ID.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#table(java.lang.String)" class="member-name-link">table(String)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>
<div class="block">Uses the static instance of <a href="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io"><code>IUPACParser</code></a> to find instances of <a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io"><code>IUPACParser.IUPACTable</code></a>s by its
 String name</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#table(org.biojava3.core.sequence.transcription.Table)" class="member-name-link">table(Table)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.html" class="type-name-link" title="interface in org.biojava3.core.sequence.transcription">Table</a> - Interface in <a href="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></dt>
<dd>
<div class="block">Provides a way of separating us from the specific <a href="../org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io"><code>IUPACParser.IUPACTable</code></a> even though this is the only
 implementing class for the interface.</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html#TABLE" class="member-name-link">TABLE</a> - Enum constant in enum bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html" title="enum in bacnet.scripts.blast">BlastOutput.BlastOutputTYPE</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html#TABLE_COMMENT" class="member-name-link">TABLE_COMMENT</a> - Enum constant in enum bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html" title="enum in bacnet.scripts.blast">BlastOutput.BlastOutputTYPE</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#TABLE_PATH" class="member-name-link">TABLE_PATH</a> - Static variable in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#TABLE_PATH" class="member-name-link">TABLE_PATH</a> - Static variable in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" class="type-name-link" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a> - Class in <a href="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></dt>
<dd>
<div class="block">Class used to hold three nucleotides together and allow for equality to be assessed in a case
 insensitive manner.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" class="type-name-link" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a> - Class in <a href="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></dt>
<dd>
<div class="block">Instance of a Codon which is 3 <a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound"><code>NucleotideCompound</code></a>s, its corresponding
 <a href="../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound"><code>AminoAcidCompound</code></a> and if it is a start or stop codon.</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" class="type-name-link" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a> - Class in <a href="../bacnet/expressionAtlas/package-summary.html">bacnet.expressionAtlas</a></dt>
<dd>
<div class="block">The Composite in which is displayed the heatmap</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#%3Cinit%3E(org.eclipse.swt.widgets.Composite,int,java.lang.String,boolean,org.eclipse.e4.ui.workbench.modeling.EPartService,org.eclipse.swt.widgets.Shell)" class="member-name-link">TableCompositeHeatMap(Composite, int, String, boolean, EPartService, Shell)</a> - Constructor for class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>
<div class="block">Create the composite.</div>
</dd>
<dt><a href="../bacnet/table/gui/TableFilter.html" class="type-name-link" title="class in bacnet.table.gui">TableFilter</a> - Class in <a href="../bacnet/table/gui/package-summary.html">bacnet.table.gui</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/TableFilter.html#%3Cinit%3E()" class="member-name-link">TableFilter()</a> - Constructor for class bacnet.table.gui.<a href="../bacnet/table/gui/TableFilter.html" title="class in bacnet.table.gui">TableFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#tableGeneEGDe()" class="member-name-link">tableGeneEGDe()</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Get all lmos from EGD-e, and extract different information: <br>
 
 The Translation Iinitiation region here is always: -20bp ---- 60bp around start codon
 Start codon
 SD energy
 Structure energy ....</div>
</dd>
<dt><a href="../bacnet/utils/TableHTMLExport.html" class="type-name-link" title="class in bacnet.utils">TableHTMLExport</a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/TableHTMLExport.html#%3Cinit%3E()" class="member-name-link">TableHTMLExport()</a> - Constructor for class bacnet.utils.<a href="../bacnet/utils/TableHTMLExport.html" title="class in bacnet.utils">TableHTMLExport</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/TableSWTComposite.html" class="type-name-link" title="class in bacnet.table">TableSWTComposite</a> - Class in <a href="../bacnet/table/package-summary.html">bacnet.table</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/TableSWTComposite.html#%3Cinit%3E(org.eclipse.swt.widgets.Composite,int,org.eclipse.swt.widgets.Shell,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">TableSWTComposite(Composite, int, Shell, EPartService)</a> - Constructor for class bacnet.table.<a href="../bacnet/table/TableSWTComposite.html" title="class in bacnet.table">TableSWTComposite</a></dt>
<dd>
<div class="block">Create the composite.</div>
</dd>
<dt><a href="../bacnet/table/TableSWTView.html" class="type-name-link" title="class in bacnet.table">TableSWTView</a> - Class in <a href="../bacnet/table/package-summary.html">bacnet.table</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/TableSWTView.html#%3Cinit%3E()" class="member-name-link">TableSWTView()</a> - Constructor for class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/TableViewerComparator.html" class="type-name-link" title="class in bacnet.table.core">TableViewerComparator</a> - Class in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/TableViewerComparator.html#%3Cinit%3E()" class="member-name-link">TableViewerComparator()</a> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/TableViewerComparator.html" title="class in bacnet.table.core">TableViewerComparator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/Tasmania.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.technology">Tasmania</a> - Class in <a href="../bacnet/scripts/listeriomics/technology/package-summary.html">bacnet.scripts.listeriomics.technology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/Tasmania.html#%3Cinit%3E()" class="member-name-link">Tasmania()</a> - Constructor for class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Tasmania.html" title="class in bacnet.scripts.listeriomics.technology">Tasmania</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/TaxonomyID.html" class="type-name-link" title="class in org.biojava3.core.sequence">TaxonomyID</a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">A sequence can be associated with a species or Taxonomy ID</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/TaxonomyID.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.DataSource)" class="member-name-link">TaxonomyID(String, DataSource)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#tblastN" class="member-name-link">tblastN</a> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.SeqType.html#terminator" class="member-name-link">terminator</a> - Enum constant in enum bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html#TermSeq" class="member-name-link">TermSeq</a> - Enum constant in enum bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#TermSeq_ARRAY" class="member-name-link">TermSeq_ARRAY</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#TermSeq_HEADERS" class="member-name-link">TermSeq_HEADERS</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#test()" class="member-name-link">test()</a> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>
<div class="block">Test ShowExpression.showExpression();</div>
</dd>
<dt><a href="../bacnet/scripts/genome/SDProfile.html#test()" class="member-name-link">test()</a> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/Test.html" class="type-name-link" title="class in bacnet.e4.rap.setup">Test</a> - Class in <a href="../bacnet/e4/rap/setup/package-summary.html">bacnet.e4.rap.setup</a></dt>
<dd>
<div class="block">All methods which will be executed before or after interface run</div>
</dd>
<dt><a href="../bacnet/e4/rap/setup/Test.html#%3Cinit%3E()" class="member-name-link">Test()</a> - Constructor for class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/Test.html" title="class in bacnet.e4.rap.setup">Test</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SessionControl.html#testApplicationContext(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">testApplicationContext(EPartService)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SessionControl.html" title="class in bacnet.e4.rap">SessionControl</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/SessionControl.html#testApplicationContext(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">testApplicationContext(EPartService)</a> - Method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SessionControl.html" title="class in bacnet.e4.rap.setup">SessionControl</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/TestGenomeBrowser.html" class="type-name-link" title="class in bacnet.genomeBrowser">TestGenomeBrowser</a> - Class in <a href="../bacnet/genomeBrowser/package-summary.html">bacnet.genomeBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/TestGenomeBrowser.html#%3Cinit%3E()" class="member-name-link">TestGenomeBrowser()</a> - Constructor for class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/TestGenomeBrowser.html" title="class in bacnet.genomeBrowser">TestGenomeBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/TestGenomeBrowser.html#%3Cinit%3E(bacnet.genomeBrowser.tracksGUI.TracksComposite,bacnet.genomeBrowser.core.Track)" class="member-name-link">TestGenomeBrowser(TracksComposite, Track)</a> - Constructor for class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/TestGenomeBrowser.html" title="class in bacnet.genomeBrowser">TestGenomeBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/LPE.html#testLPE()" class="member-name-link">testLPE()</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/LPE.html" title="class in bacnet.scripts.core.stat">LPE</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/TestPart.html" class="type-name-link" title="class in bacnet.e4.rap">TestPart</a> - Class in <a href="../bacnet/e4/rap/package-summary.html">bacnet.e4.rap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/TestPart.html#%3Cinit%3E()" class="member-name-link">TestPart()</a> - Constructor for class bacnet.e4.rap.<a href="../bacnet/e4/rap/TestPart.html" title="class in bacnet.e4.rap">TestPart</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizardPage.html#text" class="member-name-link">text</a> - Variable in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/TextFeature.html" class="type-name-link" title="class in org.biojava3.core.sequence.features">TextFeature</a>&lt;<a href="../org/biojava3/core/sequence/features/TextFeature.html" title="type parameter in TextFeature">S</a> extends <a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../org/biojava3/core/sequence/features/TextFeature.html" title="type parameter in TextFeature">C</a>&gt;,<wbr><a href="../org/biojava3/core/sequence/features/TextFeature.html" title="type parameter in TextFeature">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/features/package-summary.html">org.biojava3.core.sequence.features</a></dt>
<dd>
<div class="block">A implmentation of AbstractFeature</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/TextFeature.html#%3Cinit%3E(java.lang.String,java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">TextFeature(String, String, String, String)</a> - Constructor for class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/TextFeature.html" title="class in org.biojava3.core.sequence.features">TextFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Frame.html#THREE" class="member-name-link">THREE</a> - Enum constant in enum org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html" class="type-name-link" title="class in bacnet.datamodel.dataset">Tiling</a> - Class in <a href="../bacnet/datamodel/dataset/package-summary.html">bacnet.datamodel.dataset</a></dt>
<dd>
<div class="block">Tiling array data from Listeria tiling chip ArrayExpress id : A-MEXP-1572 *</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html#Tiling" class="member-name-link">Tiling</a> - Enum constant in enum bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#%3Cinit%3E()" class="member-name-link">Tiling()</a> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#%3Cinit%3E(java.lang.String)" class="member-name-link">Tiling(String)</a> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/TilingGeneExprDataBase.html" class="type-name-link" title="class in bacnet.scripts.listeriomics">TilingGeneExprDataBase</a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/TilingGeneExprDataBase.html#%3Cinit%3E()" class="member-name-link">TilingGeneExprDataBase()</a> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TilingGeneExprDataBase.html" title="class in bacnet.scripts.listeriomics">TilingGeneExprDataBase</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html" class="type-name-link" title="class in bacnet.datamodel.proteomics">TIS</a> - Class in <a href="../bacnet/datamodel/proteomics/package-summary.html">bacnet.datamodel.proteomics</a></dt>
<dd>
<div class="block">This class is used to regroup different Nterm.</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#%3Cinit%3E()" class="member-name-link">TIS()</a> - Constructor for class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#toArray(java.lang.String)" class="member-name-link">toArray(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Transform an ExpressionMatrix to a String[][]</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#toArray(java.lang.String,boolean,boolean)" class="member-name-link">toArray(String, boolean, boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Transform an ExpressionMatrix to a String[][]</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#toArray(java.util.ArrayList)" class="member-name-link">toArray(ArrayList&lt;String[]&gt;)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Network.html#toArrayList()" class="member-name-link">toArrayList()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>
<div class="block">Export to ArrayList to send to d3.js viewer</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#toArrayList(java.lang.String,java.lang.String,java.util.ArrayList)" class="member-name-link">toArrayList(String, String, ArrayList&lt;ExpressionMatrix&gt;)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Export in an array a List of ExpressionMatrix All ExpressionMatrix from the list are supposed to
 have the same headers, and thus the same number of column
 
 The first column = the annotation of each ExpressionMatrix The second column = the RowNames of
 each ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/table/core/ColorSWT.html#toColor(org.eclipse.swt.graphics.Device)" class="member-name-link">toColor(Device)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#toExpressionMatrix()" class="member-name-link">toExpressionMatrix()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Transform a <code>GeneExpression</code> into an <code>ExpressionMatrix</code></div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#toExpressionMatrix()" class="member-name-link">toExpressionMatrix()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Transform a <code>Tiling</code> into an <code>ExpressionMatrix</code></div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#toList(java.lang.String%5B%5D)" class="member-name-link">toList(String[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Return a <code>ArrayList<String></code> representation of the input vector</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#toList(java.lang.String%5B%5D%5B%5D)" class="member-name-link">toList(String[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#toList(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">toList(Sequence&lt;C&gt;)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">For the given <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a> this will return a <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link"><code>List</code></a> filled with the Compounds of that
 <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a>.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.Tools.html#%3Cinit%3E()" class="member-name-link">Tools()</a> - Constructor for class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.Tools.html" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#TOP_LEFT" class="member-name-link">TOP_LEFT</a> - Static variable in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Style constant for placing decorator image in top left corner of base image.</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#TOP_RIGHT" class="member-name-link">TOP_RIGHT</a> - Static variable in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Style constant for placing decorator image in top right corner of base image.</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#toString()" class="member-name-link">toString()</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#toString()" class="member-name-link">toString()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#toString()" class="member-name-link">toString()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Operon.html#toString()" class="member-name-link">toString()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#toString()" class="member-name-link">toString()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Print different <br></div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#toString()" class="member-name-link">toString()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#toString()" class="member-name-link">toString()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#toString()" class="member-name-link">toString()</a> - Method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#toString()" class="member-name-link">toString()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#toString()" class="member-name-link">toString()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/AccessionID.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimplePoint.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">String representation of the MSA</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.html#toString()" class="member-name-link">toString()</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a></dt>
<dd>
<div class="block">Returns a simple view of the alignment profile.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/CRC64Checksum.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/CRC64Checksum.html" title="class in org.biojava3.core.util">CRC64Checksum</a></dt>
<dd>
<div class="block">Returns a zero-padded 16 character wide string containing the current value of this checksum in
 uppercase hexadecimal format.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureI.html#toString()" class="member-name-link">toString()</a> - Method in interface org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a></dt>
<dd>
<div class="block">A string representation of the feature.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Return a string representation of all features in this list.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Return a string representation of location.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/LocIterator.html#toString()" class="member-name-link">toString()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></dt>
<dd>
<div class="block">Get string representation of iterator.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#toString(int)" class="member-name-link">toString(int)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">Get a string representation of the MSA with a fixed width</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.html#toString(int)" class="member-name-link">toString(int)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a></dt>
<dd>
<div class="block">Returns a formatted view of the alignment profile.</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#toString(java.lang.String%5B%5D)" class="member-name-link">toString(String[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Return a <code>String</code> representation of the input vector Tab delimited</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#toString(java.lang.String%5B%5D%5B%5D)" class="member-name-link">toString(String[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#toString(java.util.ArrayList)" class="member-name-link">toString(ArrayList&lt;String[]&gt;)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ListUtils.html#toString(java.util.ArrayList,java.lang.String)" class="member-name-link">toString(ArrayList&lt;String&gt;, String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ListUtils.html" title="class in bacnet.utils">ListUtils</a></dt>
<dd>
<div class="block">Return a String version of the list</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#toString(java.util.TreeSet)" class="member-name-link">toString(TreeSet&lt;String&gt;)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Return a <code>String</code> representation of the input TreeSet Tab delimited</div>
</dd>
<dt><a href="../bacnet/utils/ListUtils.html#toString(java.util.TreeSet,java.lang.String)" class="member-name-link">toString(TreeSet&lt;String&gt;, String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ListUtils.html" title="class in bacnet.utils">ListUtils</a></dt>
<dd>
<div class="block">Return a String version of the TreeSet</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#toString(org.biojava3.core.sequence.template.LightweightProfile.StringFormat)" class="member-name-link">toString(LightweightProfile.StringFormat)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">Support for different MSA formats</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.html#toString(org.biojava3.core.sequence.template.LightweightProfile.StringFormat)" class="member-name-link">toString(LightweightProfile.StringFormat)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a></dt>
<dd>
<div class="block">Returns a formatted view of the alignment profile.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#toString(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">toString(Sequence&lt;C&gt;)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Shortcut to <a href="../org/biojava3/core/sequence/template/SequenceMixin.html#toStringBuilder(org.biojava3.core.sequence.template.Sequence)"><code>SequenceMixin.toStringBuilder(org.biojava3.core.sequence.template.Sequence)</code></a> which calls
 toString() on the resulting object.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#toStringBuilder(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">toStringBuilder(Sequence&lt;C&gt;)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">For the given Sequence this will return a <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/StringBuilder.html" title="class or interface in java.lang" class="external-link"><code>StringBuilder</code></a> object filled with the results of
 <code>Compound#toString()</code>.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#toStringCodingInfo(org.biojava3.core.sequence.DNASequence)" class="member-name-link">toStringCodingInfo(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Display the info contain in a coding DNASequence</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Operon.html#toStringGenes()" class="member-name-link">toStringGenes()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></dt>
<dd>
<div class="block">Return a list of all genes part of this Operon</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Operon.html#toStringInfo()" class="member-name-link">toStringInfo()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></dt>
<dd>
<div class="block">Return a String containing a description of Operon properties</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#toStringMatrix(double%5B%5D%5B%5D)" class="member-name-link">toStringMatrix(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#toStringNonCodingInfo(org.biojava3.core.sequence.DNASequence)" class="member-name-link">toStringNonCodingInfo(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Display info contain in a non coding DNASequence</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#toVector(double%5B%5D%5B%5D)" class="member-name-link">toVector(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Transform to a Vector an array</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html" class="type-name-link" title="class in bacnet.genomeBrowser.core">Track</a> - Class in <a href="../bacnet/genomeBrowser/core/package-summary.html">bacnet.genomeBrowser.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#%3Cinit%3E()" class="member-name-link">Track()</a> - Constructor for class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#%3Cinit%3E(bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">Track(Genome, String)</a> - Constructor for class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>
<div class="block">Set every parameter using genome and chromosome number</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.DisplayType.html" class="type-name-link" title="enum in bacnet.genomeBrowser.core">Track.DisplayType</a> - Enum in <a href="../bacnet/genomeBrowser/core/package-summary.html">bacnet.genomeBrowser.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" class="type-name-link" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a> - Class in <a href="../bacnet/genomeBrowser/tracksGUI/package-summary.html">bacnet.genomeBrowser.tracksGUI</a></dt>
<dd>
<div class="block">A TrackCanvasData is the Canvas in which will be displayed datasets</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#%3Cinit%3E(org.eclipse.swt.widgets.Composite,int,org.eclipse.swt.widgets.Composite)" class="member-name-link">TrackCanvasData(Composite, int, Composite)</a> - Constructor for class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" class="type-name-link" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a> - Class in <a href="../bacnet/genomeBrowser/tracksGUI/package-summary.html">bacnet.genomeBrowser.tracksGUI</a></dt>
<dd>
<div class="block">A TrackCanvasGenome is the Canvas in which will be displayed genome annotation</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#%3Cinit%3E(org.eclipse.swt.widgets.Composite,int)" class="member-name-link">TrackCanvasGenome(Composite, int)</a> - Constructor for class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>
<div class="block">Initiate a canvas genome to display genome annotation</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#trackChangedFlag" class="member-name-link">trackChangedFlag</a> - Variable in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" class="type-name-link" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a> - Class in <a href="../bacnet/genomeBrowser/tracksGUI/package-summary.html">bacnet.genomeBrowser.tracksGUI</a></dt>
<dd>
<div class="block">In TracksComposite both <code>TrackCanvasData</code> and <code>TrackCanvasGenome</code> will be
 displayed.</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#%3Cinit%3E(org.eclipse.swt.widgets.Composite,int,boolean,org.eclipse.e4.ui.workbench.modeling.EPartService,org.eclipse.swt.widgets.Shell)" class="member-name-link">TracksComposite(Composite, int, boolean, EPartService, Shell)</a> - Constructor for class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Create the composite.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" class="type-name-link" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a> - Class in <a href="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></dt>
<dd>
<div class="block">Used as a way of encapsulating the data structures required to parse DNA to a Protein sequence.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" class="type-name-link" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a> - Class in <a href="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></dt>
<dd>
<div class="block">This class is the way to create a <code>TranslationEngine</code>.</div>
</dd>
<dt><a href="../bacnet/scripts/database/TranscriptomesCreation.html" class="type-name-link" title="class in bacnet.scripts.database">TranscriptomesCreation</a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>
<div class="block">List of methods to add the different transcriptomic datasetsto your multi-omics website</div>
</dd>
<dt><a href="../bacnet/scripts/database/TranscriptomesCreation.html#%3Cinit%3E()" class="member-name-link">TranscriptomesCreation()</a> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/TranscriptomesCreation.html" title="class in bacnet.scripts.database">TranscriptomesCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" class="type-name-link" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a> - Class in <a href="../bacnet/expressionAtlas/package-summary.html">bacnet.expressionAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#%3Cinit%3E(org.eclipse.swt.widgets.Composite,int,bacnet.expressionAtlas.TranscriptomicsView)" class="member-name-link">TranscriptomicsDataFilterComposite(Composite, int, TranscriptomicsView)</a> - Constructor for class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html" class="type-name-link" title="class in bacnet.expressionAtlas">TranscriptomicsView</a> - Class in <a href="../bacnet/expressionAtlas/package-summary.html">bacnet.expressionAtlas</a></dt>
<dd>
<div class="block">View in which is displayed the table with all transcriptomics data available</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#%3Cinit%3E()" class="member-name-link">TranscriptomicsView()</a> - Constructor for class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/TranscriptSequence.html" class="type-name-link" title="class in org.biojava3.core.sequence">TranscriptSequence</a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">This is the sequence if you want to go from a gene sequence to a protein sequence.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/TranscriptSequence.html#%3Cinit%3E(org.biojava3.core.sequence.GeneSequence,int,int)" class="member-name-link">TranscriptSequence(GeneSequence, int, int)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#transform(double%5B%5D%5B%5D,int)" class="member-name-link">transform(double[][], int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Transform all value of this array, using MathUtils.transform</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#transform(double%5B%5D,int)" class="member-name-link">transform(double[], int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Transform all value of this vector, using MathUtils.transform</div>
</dd>
<dt><a href="../bacnet/utils/MathUtils.html#transform(double,int)" class="member-name-link">transform(double, int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/MathUtils.html" title="class in bacnet.utils">MathUtils</a></dt>
<dd>
<div class="block">Transform a value, according to the type of transformation<br>
 type = 0 -> log2(value)<br>
 type = 1 -> 2^value<br>
 type = 2 -> log10(value)<br></div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#transformData()" class="member-name-link">transformData()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>
<div class="block">For each genome save a list of genes and their homologs</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#translate(F)" class="member-name-link">translate(F)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundTranslator.html#translate(F)" class="member-name-link">translate(F)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundTranslator.html" title="interface in org.biojava3.core.sequence.template">CompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#translate(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">translate(Sequence&lt;NucleotideCompound&gt;)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>
<div class="block">Quick method to let you go from a CDS to a Peptide quickly.</div>
</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#translateGenomes(int)" class="member-name-link">translateGenomes(int)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>
<div class="block">Read all fasta file from each genome and create a list of every possible small peptides<br>
 Extract from NTermDatabase.createTISDB()</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastDatabase.html#translateGenomes(int,java.lang.String)" class="member-name-link">translateGenomes(int, String)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">Read all fasta file from each genome and create a list of every possible small peptides<br>
 Extract from NTermDatabase.createTISDB()</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#translateMany(F)" class="member-name-link">translateMany(F)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundTranslator.html#translateMany(F)" class="member-name-link">translateMany(F)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundTranslator.html" title="interface in org.biojava3.core.sequence.template">CompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html#translateNCodons()" class="member-name-link">translateNCodons()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html" title="class in org.biojava3.core.sequence.transcription">RNAToAminoAcidTranslator</a></dt>
<dd>
<div class="block">Indicates if we want to force exact translation of compounds or not i.e. those with internal N
 RNA bases.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#translateNCodons(boolean)" class="member-name-link">translateNCodons(boolean)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/exceptions/TranslationException.html" class="type-name-link" title="class in org.biojava3.core.exceptions">TranslationException</a> - Exception in <a href="../org/biojava3/core/exceptions/package-summary.html">org.biojava3.core.exceptions</a></dt>
<dd>
<div class="block">Thrown from AbstractCompundTranslator</div>
</dd>
<dt><a href="../org/biojava3/core/exceptions/TranslationException.html#%3Cinit%3E(java.lang.Exception)" class="member-name-link">TranslationException(Exception)</a> - Constructor for exception org.biojava3.core.exceptions.<a href="../org/biojava3/core/exceptions/TranslationException.html" title="class in org.biojava3.core.exceptions">TranslationException</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/exceptions/TranslationException.html#%3Cinit%3E(java.lang.String)" class="member-name-link">TranslationException(String)</a> - Constructor for exception org.biojava3.core.exceptions.<a href="../org/biojava3/core/exceptions/TranslationException.html" title="class in org.biojava3.core.exceptions">TranslationException</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/exceptions/TranslationException.html#%3Cinit%3E(java.lang.String,java.lang.Exception)" class="member-name-link">TranslationException(String, Exception)</a> - Constructor for exception org.biojava3.core.exceptions.<a href="../org/biojava3/core/exceptions/TranslationException.html" title="class in org.biojava3.core.exceptions">TranslationException</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#transpose()" class="member-name-link">transpose()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#transpose(double%5B%5D%5B%5D)" class="member-name-link">transpose(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Return transposed array of input</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#transpose(java.lang.String%5B%5D%5B%5D)" class="member-name-link">transpose(String[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#trimStop(boolean)" class="member-name-link">trimStop(boolean)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/NcRNA.TypeNcRNA.html#tRNA" class="member-name-link">tRNA</a> - Enum constant in enum bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.TypeNcRNA.html" title="enum in bacnet.datamodel.sequence">NcRNA.TypeNcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html#Trypsin_Actino_Insoluble" class="member-name-link">Trypsin_Actino_Insoluble</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html" title="enum in bacnet.datamodel.proteomics">NTerm.ExperimentName</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html#Trypsin_Actino_Soluble" class="member-name-link">Trypsin_Actino_Soluble</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html" title="enum in bacnet.datamodel.proteomics">NTerm.ExperimentName</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html#Trypsin_Insoluble" class="member-name-link">Trypsin_Insoluble</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html" title="enum in bacnet.datamodel.proteomics">NTerm.ExperimentName</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html#Trypsin_Soluble" class="member-name-link">Trypsin_Soluble</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.ExperimentName.html" title="enum in bacnet.datamodel.proteomics">NTerm.ExperimentName</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html#TSS" class="member-name-link">TSS</a> - Enum constant in enum bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#TSS_ARRAY" class="member-name-link">TSS_ARRAY</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#TSS_HEADERS" class="member-name-link">TSS_HEADERS</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#TSS_Innocua_HEADERS" class="member-name-link">TSS_Innocua_HEADERS</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" class="type-name-link" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>
<div class="block">List of methods to process TSS, TermSeq, and RiboSeq of Listeriomics Some TSS and NTerm lists
 were provided in tab-delimited format.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#%3Cinit%3E()" class="member-name-link">TSSNTermRiboSeqListeriomics()</a> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#TSTUDENT" class="member-name-link">TSTUDENT</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#TSTUDENTTILING" class="member-name-link">TSTUDENTTILING</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Frame.html#TWO" class="member-name-link">TWO</a> - Enum constant in enum org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">TwoBitArrayWorker(String, CompoundSet&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader.TwoBitArrayWorker</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int)" class="member-name-link">TwoBitArrayWorker(CompoundSet&lt;C&gt;, int)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader.TwoBitArrayWorker</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int%5B%5D)" class="member-name-link">TwoBitArrayWorker(CompoundSet&lt;C&gt;, int[])</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader.TwoBitArrayWorker</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">TwoBitArrayWorker(Sequence&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader.TwoBitArrayWorker</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" class="type-name-link" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader</a>&lt;<a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="type parameter in TwoBitSequenceReader">C</a> extends <a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</a></dt>
<dd>
<div class="block">Implementation of the 2bit encoding.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">TwoBitSequenceReader(String, CompoundSet&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.AccessionID)" class="member-name-link">TwoBitSequenceReader(String, CompoundSet&lt;C&gt;, AccessionID)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html#%3Cinit%3E(org.biojava3.core.sequence.storage.TwoBitSequenceReader.TwoBitArrayWorker)" class="member-name-link">TwoBitSequenceReader(TwoBitSequenceReader.TwoBitArrayWorker&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html#%3Cinit%3E(org.biojava3.core.sequence.storage.TwoBitSequenceReader.TwoBitArrayWorker,org.biojava3.core.sequence.AccessionID)" class="member-name-link">TwoBitSequenceReader(TwoBitSequenceReader.TwoBitArrayWorker&lt;C&gt;, AccessionID)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">TwoBitSequenceReader(Sequence&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" class="type-name-link" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader.TwoBitArrayWorker</a>&lt;<a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</a> extends <a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</a></dt>
<dd>
<div class="block">Extension of the BitArrayWorker which provides the 2bit implementation code.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#type()" class="member-name-link">type()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Get feature type, such as "exon" or "CDS".</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureI.html#type()" class="member-name-link">type()</a> - Method in interface org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a></dt>
<dd>
<div class="block">Get the feature type, for example, "exon", "CDS", etc.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#TYPE_OVERLAPS" class="member-name-link">TYPE_OVERLAPS</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#TYPE_REPRESENTATION" class="member-name-link">TYPE_REPRESENTATION</a> - Static variable in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#TYPE_REPRESENTATION" class="member-name-link">TYPE_REPRESENTATION</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
