<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Fri Dec 10 09:15:34 CET 2021 -->
<title>F-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-10">
<meta name="description" content="index: F">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:F">F</h2>
<dl class="index">
<dt><a href="../bacnet/utils/MathUtils.html#factorial(int)" class="member-name-link">factorial(int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/MathUtils.html" title="class in bacnet.utils">MathUtils</a></dt>
<dd>
<div class="block">Calculate the factorial of n</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html#FASTA" class="member-name-link">FASTA</a> - Enum constant in enum org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html" title="enum in org.biojava3.core.sequence.template">LightweightProfile.StringFormat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/FastaFileReader.html" class="type-name-link" title="class in bacnet.reader">FastaFileReader</a>&lt;<a href="../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">S</a> extends <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,<wbr><a href="../bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../bacnet/reader/package-summary.html">bacnet.reader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/FastaFileReader.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)" class="member-name-link">FastaFileReader(File, FastaHeaderParserInterface&lt;S, C&gt;, SequenceCreatorInterface&lt;C&gt;)</a> - Constructor for class bacnet.reader.<a href="../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a></dt>
<dd>
<div class="block">If you are going to use the FileProxyProteinSequenceCreator then you need to use this constructor
 because we need details about the location of the file.</div>
</dd>
<dt><a href="../bacnet/reader/FastaFileReader.html#%3Cinit%3E(java.io.InputStream,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)" class="member-name-link">FastaFileReader(InputStream, FastaHeaderParserInterface&lt;S, C&gt;, SequenceCreatorInterface&lt;C&gt;)</a> - Constructor for class bacnet.reader.<a href="../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a></dt>
<dd>
<div class="block">If you are going to use FileProxyProteinSequenceCreator then do not use this constructor because
 we need details about local file offsets for quick reads.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</a> - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">A Gene sequence has a Positive or Negative Strand where we want to write out to a stream the 5 to
 3 prime version.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html#%3Cinit%3E(java.io.OutputStream,java.util.Collection,org.biojava3.core.sequence.io.template.FastaHeaderFormatInterface,boolean)" class="member-name-link">FastaGeneWriter(OutputStream, Collection&lt;GeneSequence&gt;, FastaHeaderFormatInterface&lt;GeneSequence, NucleotideCompound&gt;, boolean)</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html#%3Cinit%3E(java.io.OutputStream,java.util.Collection,org.biojava3.core.sequence.io.template.FastaHeaderFormatInterface,boolean,int)" class="member-name-link">FastaGeneWriter(OutputStream, Collection&lt;GeneSequence&gt;, FastaHeaderFormatInterface&lt;GeneSequence, NucleotideCompound&gt;, boolean, int)</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/template/FastaHeaderFormatInterface.html" class="type-name-link" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderFormatInterface</a>&lt;<a href="../org/biojava3/core/sequence/io/template/FastaHeaderFormatInterface.html" title="type parameter in FastaHeaderFormatInterface">S</a> extends <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,<wbr><a href="../org/biojava3/core/sequence/io/template/FastaHeaderFormatInterface.html" title="type parameter in FastaHeaderFormatInterface">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/io/template/package-summary.html">org.biojava3.core.sequence.io.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" class="type-name-link" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="../org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="type parameter in FastaHeaderParserInterface">S</a> extends <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,<wbr><a href="../org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="type parameter in FastaHeaderParserInterface">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/io/template/package-summary.html">org.biojava3.core.sequence.io.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaReader.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">FastaReader</a>&lt;<a href="../org/biojava3/core/sequence/io/FastaReader.html" title="type parameter in FastaReader">S</a> extends <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,<wbr><a href="../org/biojava3/core/sequence/io/FastaReader.html" title="type parameter in FastaReader">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">Use FastaReaderHelper as an example of how to use this class where FastaReaderHelper should be
 the primary class used to read Fasta files</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaReader.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)" class="member-name-link">FastaReader(File, FastaHeaderParserInterface&lt;S, C&gt;, SequenceCreatorInterface&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaReader.html" title="class in org.biojava3.core.sequence.io">FastaReader</a></dt>
<dd>
<div class="block">If you are going to use the FileProxyProteinSequenceCreator then you need to use this constructor
 because we need details about the location of the file.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaReader.html#%3Cinit%3E(java.io.InputStream,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)" class="member-name-link">FastaReader(InputStream, FastaHeaderParserInterface&lt;S, C&gt;, SequenceCreatorInterface&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaReader.html" title="class in org.biojava3.core.sequence.io">FastaReader</a></dt>
<dd>
<div class="block">If you are going to use FileProxyProteinSequenceCreator then do not use this constructor because
 we need details about local file offsets for quick reads.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</a> - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html#%3Cinit%3E()" class="member-name-link">FastaReaderHelper()</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaSequenceParser.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">FastaSequenceParser</a> - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">Used to parse a stream of a fasta file to get the sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaSequenceParser.html#%3Cinit%3E()" class="member-name-link">FastaSequenceParser()</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaSequenceParser.html" title="class in org.biojava3.core.sequence.io">FastaSequenceParser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaWriter.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">FastaWriter</a>&lt;<a href="../org/biojava3/core/sequence/io/FastaWriter.html" title="type parameter in FastaWriter">S</a> extends <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,<wbr><a href="../org/biojava3/core/sequence/io/FastaWriter.html" title="type parameter in FastaWriter">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">The FastaWriter writes a collection of sequences to an outputStream.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaWriter.html#%3Cinit%3E(java.io.OutputStream,java.util.Collection,org.biojava3.core.sequence.io.template.FastaHeaderFormatInterface)" class="member-name-link">FastaWriter(OutputStream, Collection&lt;S&gt;, FastaHeaderFormatInterface&lt;S, C&gt;)</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a></dt>
<dd>
<div class="block">Use default line length of 60</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaWriter.html#%3Cinit%3E(java.io.OutputStream,java.util.Collection,org.biojava3.core.sequence.io.template.FastaHeaderFormatInterface,int)" class="member-name-link">FastaWriter(OutputStream, Collection&lt;S&gt;, FastaHeaderFormatInterface&lt;S, C&gt;, int)</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a></dt>
<dd>
<div class="block">Set custom lineLength</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaWriterHelper.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">FastaWriterHelper</a> - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">The class that should be used to write out fasta file of a sequence collection</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaWriterHelper.html#%3Cinit%3E()" class="member-name-link">FastaWriterHelper()</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaWriterHelper.html" title="class in org.biojava3.core.sequence.io">FastaWriterHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html#FC" class="member-name-link">FC</a> - Enum constant in enum bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html" title="enum in bacnet.datamodel.dataset">OmicsData.ColNames</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#FC" class="member-name-link">FC</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.TypeMapper.html#FC" class="member-name-link">FC</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.TypeMapper.html" title="enum in bacnet.table.core">ColorMapper.TypeMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.Type.html#FC" class="member-name-link">FC</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.Type.html" title="enum in bacnet.table.core">ColorMapperRCP.Type</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/FDR.html" class="type-name-link" title="class in bacnet.scripts.core.stat">FDR</a> - Class in <a href="../bacnet/scripts/core/stat/package-summary.html">bacnet.scripts.core.stat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/FDR.html#%3Cinit%3E()" class="member-name-link">FDR()</a> - Constructor for class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/FDR.html" title="class in bacnet.scripts.core.stat">FDR</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#FDRBH" class="member-name-link">FDRBH</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#FDRBONF" class="member-name-link">FDRBONF</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#FDRBY" class="member-name-link">FDRBY</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html" class="type-name-link" title="class in org.biojava3.genome.parsers.gff">Feature</a> - Class in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>
<div class="block">A Feature corresponds to a single row in a GFF file.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#%3Cinit%3E(java.lang.String,java.lang.String,java.lang.String,org.biojava3.genome.parsers.gff.Location,java.lang.Double,int,java.lang.String)" class="member-name-link">Feature(String, String, String, Location, Double, int, String)</a> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Construct a new Feature from raw data (usually a GFF row).</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#%3Cinit%3E(org.biojava3.genome.parsers.gff.Feature)" class="member-name-link">Feature(Feature)</a> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Make a copy of the specified feature.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureHelper.html" class="type-name-link" title="class in org.biojava3.genome.parsers.gff">FeatureHelper</a> - Class in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureHelper.html#%3Cinit%3E()" class="member-name-link">FeatureHelper()</a> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureHelper.html" title="class in org.biojava3.genome.parsers.gff">FeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureI.html" class="type-name-link" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a> - Interface in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>
<div class="block">A feature on a sequence (for example, an exon or a gene), defined by a location and a set of
 attributes encoded as key/value pairs.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html" class="type-name-link" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="type parameter in FeatureInterface">S</a> extends <a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="type parameter in FeatureInterface">C</a>&gt;,<wbr><a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="type parameter in FeatureInterface">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/features/package-summary.html">org.biojava3.core.sequence.features</a></dt>
<dd>
<div class="block">Interface class to handle describing arbitrary features.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html" class="type-name-link" title="class in org.biojava3.genome.parsers.gff">FeatureList</a> - Class in <a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></dt>
<dd>
<div class="block">A list of FeatureI objects implemented using a Java ArrayList; corresponds to a GFF file.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#%3Cinit%3E()" class="member-name-link">FeatureList()</a> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Construct an empty list.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#%3Cinit%3E(java.util.Collection)" class="member-name-link">FeatureList(Collection&lt;FeatureI&gt;)</a> - Constructor for class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Construct a new list containing the same features as the specified list.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeaturesKeyWordInterface.html" class="type-name-link" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</a> - Interface in <a href="../org/biojava3/core/sequence/features/package-summary.html">org.biojava3.core.sequence.features</a></dt>
<dd>
<div class="block">Models the keywords that are annotated for a protein sequence at Uniprot.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/Febit.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.technology">Febit</a> - Class in <a href="../bacnet/scripts/listeriomics/technology/package-summary.html">bacnet.scripts.listeriomics.technology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/Febit.html#%3Cinit%3E()" class="member-name-link">Febit()</a> - Constructor for class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Febit.html" title="class in bacnet.scripts.listeriomics.technology">Febit</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#FILE_SEPARATOR" class="member-name-link">FILE_SEPARATOR</a> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">We added manually this variable because when running .bat file on windows the file separator can get tricky.</div>
</dd>
<dt><a href="../org/biojava3/core/exceptions/FileAccessError.html" class="type-name-link" title="class in org.biojava3.core.exceptions">FileAccessError</a> - Error in <a href="../org/biojava3/core/exceptions/package-summary.html">org.biojava3.core.exceptions</a></dt>
<dd>
<div class="block">Thrown from SequenceFileProxyLoader if error opening file using RandomnAccessFile</div>
</dd>
<dt><a href="../org/biojava3/core/exceptions/FileAccessError.html#%3Cinit%3E(java.lang.String)" class="member-name-link">FileAccessError(String)</a> - Constructor for error org.biojava3.core.exceptions.<a href="../org/biojava3/core/exceptions/FileAccessError.html" title="class in org.biojava3.core.exceptions">FileAccessError</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastOutput.html#fileExtension(bacnet.scripts.blast.BlastOutput.BlastOutputTYPE)" class="member-name-link">fileExtension(BlastOutput.BlastOutputTYPE)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.html" title="class in bacnet.scripts.blast">BlastOutput</a></dt>
<dd>
<div class="block">For a specific type, return the extension of the corresponding file</div>
</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#fileExtension(bacnet.scripts.blast.BlastOutput.BlastOutputTYPE)" class="member-name-link">fileExtension(BlastOutput.BlastOutputTYPE)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">For a specific type, return the extension of the corresponding file</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">FileProxyDNASequenceCreator</a> - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the
 stream the sequence and the offset index are passed to create a Protein sequence that will be
 loaded in lazily.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">FileProxyDNASequenceCreator(File, CompoundSet&lt;NucleotideCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyDNASequenceCreator</a></dt>
<dd>
<div class="block">Need File so that we can store full path name in SequenceFileProxyLoader for Random File access
 as a quick read</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">FileProxyProteinSequenceCreator</a> - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the
 stream the sequence and the offset index are passed to create a Protein sequence that will be
 loaded in lazily.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">FileProxyProteinSequenceCreator(File, CompoundSet&lt;AminoAcidCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyProteinSequenceCreator</a></dt>
<dd>
<div class="block">Need File so that we can store full path name in SequenceFileProxyLoader for Random File access
 as a quick read</div>
</dd>
<dt><a href="../bacnet/utils/FileUtils.html" class="type-name-link" title="class in bacnet.utils">FileUtils</a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>
<div class="block">Utils for file manipulation.</div>
</dd>
<dt><a href="../bacnet/utils/FileUtils.html#%3Cinit%3E()" class="member-name-link">FileUtils()</a> - Constructor for class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#filter(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.utils.Filter)" class="member-name-link">filter(ExpressionMatrix, Filter)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Apply a filter to an ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/table/core/Filter.html" class="type-name-link" title="class in bacnet.table.core">Filter</a> - Class in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html" class="type-name-link" title="class in bacnet.utils">Filter</a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#%3Cinit%3E()" class="member-name-link">Filter()</a> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#%3Cinit%3E()" class="member-name-link">Filter()</a> - Constructor for class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#%3Cinit%3E(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">Filter(ExpressionMatrix)</a> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#%3Cinit%3E(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">Filter(ExpressionMatrix)</a> - Constructor for class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.TypeFilter.html" class="type-name-link" title="enum in bacnet.table.core">Filter.TypeFilter</a> - Enum in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.TypeFilter.html" class="type-name-link" title="enum in bacnet.utils">Filter.TypeFilter</a> - Enum in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/LPE.html#filterAM(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">filterAM(ExpressionMatrix)</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/LPE.html" title="class in bacnet.scripts.core.stat">LPE</a></dt>
<dd>
<div class="block">Delete values inferior to 0 in am matrix which as been sorted by A values</div>
</dd>
<dt><a href="../bacnet/scripts/genome/SDProfile.html#filterCopyResults(java.lang.String,java.lang.String)" class="member-name-link">filterCopyResults(String, String)</a> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>
<div class="block">Filters all files by deleting all positive results, and inverting energies.</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#filterData(java.util.ArrayList)" class="member-name-link">filterData(ArrayList&lt;String&gt;)</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">We filter data, and add all comparisons from the data</div>
</dd>
<dt><a href="../bacnet/table/gui/FilterDialog.html" class="type-name-link" title="class in bacnet.table.gui">FilterDialog</a> - Class in <a href="../bacnet/table/gui/package-summary.html">bacnet.table.gui</a></dt>
<dd>
<div class="block">Dialog for managing <code>Filter</code> for <code>ExpressionMatrix</code></div>
</dd>
<dt><a href="../bacnet/table/gui/FilterDialog.html#%3Cinit%3E(bacnet.table.core.Filter,org.eclipse.swt.widgets.Shell)" class="member-name-link">FilterDialog(Filter, Shell)</a> - Constructor for class bacnet.table.gui.<a href="../bacnet/table/gui/FilterDialog.html" title="class in bacnet.table.gui">FilterDialog</a></dt>
<dd>
<div class="block">Create the dialog.</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#filterDiffSize(java.util.ArrayList,int)" class="member-name-link">filterDiffSize(ArrayList&lt;BlastResult&gt;, int)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/GeneralArray.html#filterEGDe(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">filterEGDe(ExpressionMatrix)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/GeneralArray.html" title="class in bacnet.scripts.listeriomics.technology">GeneralArray</a></dt>
<dd>
<div class="block">Keep only the row containing EGD-e genes: "lmo" gene</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/JCVI.html#filterEGDe(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">filterEGDe(ExpressionMatrix)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/JCVI.html" title="class in bacnet.scripts.listeriomics.technology">JCVI</a></dt>
<dd>
<div class="block">Keep only the row containing EGD-e genes: "lmo" gene</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/WUSTL.html#filterEGDe(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">filterEGDe(ExpressionMatrix)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/WUSTL.html" title="class in bacnet.scripts.listeriomics.technology">WUSTL</a></dt>
<dd>
<div class="block">Keep only the row containing EGD-e genes: "lmo" gene</div>
</dd>
<dt><a href="../bacnet/table/gui/FilterExpressionMatrixDialog.html" class="type-name-link" title="class in bacnet.table.gui">FilterExpressionMatrixDialog</a> - Class in <a href="../bacnet/table/gui/package-summary.html">bacnet.table.gui</a></dt>
<dd>
<div class="block">Dialog displaying the list of <code>Filters</code> currently used on the
 <code>ExpressionMatrix</code></div>
</dd>
<dt><a href="../bacnet/table/gui/FilterExpressionMatrixDialog.html#%3Cinit%3E(bacnet.table.core.FilterList,org.eclipse.swt.widgets.Shell)" class="member-name-link">FilterExpressionMatrixDialog(FilterList, Shell)</a> - Constructor for class bacnet.table.gui.<a href="../bacnet/table/gui/FilterExpressionMatrixDialog.html" title="class in bacnet.table.gui">FilterExpressionMatrixDialog</a></dt>
<dd>
<div class="block">Create the dialog.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/GeneralArray.html#filterF2365(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">filterF2365(ExpressionMatrix)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/GeneralArray.html" title="class in bacnet.scripts.listeriomics.technology">GeneralArray</a></dt>
<dd>
<div class="block">Keep only the row containing F2365 genes: "LMOf2365" gene</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/JCVI.html#filterF2365(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">filterF2365(ExpressionMatrix)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/JCVI.html" title="class in bacnet.scripts.listeriomics.technology">JCVI</a></dt>
<dd>
<div class="block">Keep only the row containing F2365 genes: "LMOf2365" gene</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/WUSTL.html#filterF2365(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">filterF2365(ExpressionMatrix)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/WUSTL.html" title="class in bacnet.scripts.listeriomics.technology">WUSTL</a></dt>
<dd>
<div class="block">Keep only the row containing F2365 genes: "LMOf2365" gene</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/JCVI.html#filterHCC23(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">filterHCC23(ExpressionMatrix)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/JCVI.html" title="class in bacnet.scripts.listeriomics.technology">JCVI</a></dt>
<dd>
<div class="block">Keep only the row containing HCC23 genes: "LMHCC" gene</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#filterIdentities(java.util.ArrayList,int)" class="member-name-link">filterIdentities(ArrayList&lt;BlastResult&gt;, int)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>
<div class="block">Use identities to filter a list of BlastResult</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#filterLength(java.util.ArrayList,int)" class="member-name-link">filterLength(ArrayList&lt;BlastResult&gt;, int)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/FilterList.html" class="type-name-link" title="class in bacnet.table.core">FilterList</a> - Class in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/FilterList.html#%3Cinit%3E()" class="member-name-link">FilterList()</a> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/FilterList.html#%3Cinit%3E(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">FilterList(ExpressionMatrix)</a> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#filterNetwork()" class="member-name-link">filterNetwork()</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>
<div class="block">Using genome elements selected and current cut-off.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Network.html#filterNetwork(java.util.ArrayList,double)" class="member-name-link">filterNetwork(ArrayList&lt;String&gt;, double)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>
<div class="block">Given a list of vertice and a cutoff, filter the network</div>
</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html#filterSmallORF(java.util.ArrayList,double)" class="member-name-link">filterSmallORF(ArrayList&lt;String&gt;, double)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html" title="class in bacnet.scripts.blast">MultiSequenceBlastProtein</a></dt>
<dd>
<div class="block">Filtering table of results by e-value</div>
</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#filterSmallORF(java.util.ArrayList,double)" class="member-name-link">filterSmallORF(ArrayList&lt;String&gt;, double)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>
<div class="block">Filtering table of results by e-value</div>
</dd>
<dt><a href="../bacnet/table/core/Filter.html#filterValue(double)" class="member-name-link">filterValue(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#filterValue(double)" class="member-name-link">filterValue(double)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#findCodonUsage(java.lang.String)" class="member-name-link">findCodonUsage(String)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">Get Codon usage of the peptide</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#findColumn(java.lang.String%5B%5D%5B%5D,java.lang.String)" class="member-name-link">findColumn(String[][], String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Find the index of a column, given its header</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html#findEGDePosition()" class="member-name-link">findEGDePosition()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html" title="class in bacnet.scripts.listeriomics.srna">Srna10403SOliver</a></dt>
<dd>
<div class="block">Get all Srna from Oliver et al. 2009 list, get Sequence in
 <code>10403S pseudo_chromosome</code><br>
 Run BlastN on EGD-e.<br>
 When Blast are all performed, summarize information in a Table.<br>
 Choose which rli to keep</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermMain.html#findLeaderLess()" class="member-name-link">findLeaderLess()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermMain.html" title="class in bacnet.scripts.listeriomics.nterm">NTermMain</a></dt>
<dd>
<div class="block">Search for upstream TSS for each <code>NTerm</code><br>
 It uses list of aTIS</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#findLocusTag(java.lang.String)" class="member-name-link">findLocusTag(String)</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#findLocusTag(java.lang.String,java.lang.String)" class="member-name-link">findLocusTag(String, String)</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>
<div class="block">Search a locustag in all GFF present for a genomeName</div>
</dd>
<dt><a href="../bacnet/scripts/database/BioConditionCreation.html#findMissingCELfilesListeria()" class="member-name-link">findMissingCELfilesListeria()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/BioConditionCreation.html" title="class in bacnet.scripts.database">BioConditionCreation</a></dt>
<dd>
<div class="block">Read Array in expDesignTableFile and search which Raw data is missing<br>
 Save missing .CEL and .txt files in three lists:
 NotFoundCEl GeneExpr.txt
 NotFoundCEl Tiling.txt
 NotFoundCEl RNASeq.txt</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#findModifs()" class="member-name-link">findModifs()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>
<div class="block">List the type of modification overlap of each NTerm</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#findNextNterm(int,java.lang.String)" class="member-name-link">findNextNterm(int, String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Given a position in the genome, give the NTerm just downstream to it</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#findNTermRef()" class="member-name-link">findNTermRef()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>
<div class="block">Find the Nterm with the highest spectrum</div>
</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html#findORFSequence(java.lang.String,java.lang.String)" class="member-name-link">findORFSequence(String, String)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html" title="class in bacnet.scripts.blast">MultiSequenceBlastProtein</a></dt>
<dd>
<div class="block">Go through fasta file and extract sequence</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#findOverlaps()" class="member-name-link">findOverlaps()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>
<div class="block">List the type of gene overlap of each NTerm</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#findPosition()" class="member-name-link">findPosition()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>
<div class="block">From all the nTerms (supposed in the same strand), get the minimal begin and the maximum end</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#findPreviousNterm(int,java.lang.String)" class="member-name-link">findPreviousNterm(int, String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Given a position in the genome, give the NTerm just upstream to it</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#findProbes()" class="member-name-link">findProbes()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">From array probe sequence try to retrieve the gene name</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#findProbes()" class="member-name-link">findProbes()</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">From array probe sequence try to retrieve the gene name</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#findRefSequence()" class="member-name-link">findRefSequence()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>
<div class="block">Find the total sequence of all peptides</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#findSameNterm(int,java.lang.String)" class="member-name-link">findSameNterm(int, String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Given a position in the genome, give the NTerm just upstream to it</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#findSDSequence(java.lang.String)" class="member-name-link">findSDSequence(String)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">For each <code>NTerm</code> calculates anti Shine-Dalgarno sequence affinity in the region -20bp
 before peptide on the genome</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermUtils.html#findSDSequenceforLMO()" class="member-name-link">findSDSequenceforLMO()</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>
<div class="block">Find SD binding free energy on region -20bp upstream to each lmo</div>
</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html#findSmallORFSequence(java.lang.String,java.lang.String)" class="member-name-link">findSmallORFSequence(String, String)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html" title="class in bacnet.scripts.blast">MultiSequenceBlastProtein</a></dt>
<dd>
<div class="block">Go through fasta file and extract sequence</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#findStart(java.lang.String)" class="member-name-link">findStart(String)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">Search if the first amino acid or the amino acid before is a StartCodon</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#findTSS(java.lang.String)" class="member-name-link">findTSS(String)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html#FLAT" class="member-name-link">FLAT</a> - Enum constant in enum bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html" title="enum in bacnet.scripts.blast">BlastOutput.BlastOutputTYPE</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html#FLAT_SHOW" class="member-name-link">FLAT_SHOW</a> - Enum constant in enum bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html" title="enum in bacnet.scripts.blast">BlastOutput.BlastOutputTYPE</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/FlatFileCache.html" class="type-name-link" title="class in org.biojava3.core.util">FlatFileCache</a> - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>
<div class="block">Provides a cache for storing multiple small files in memory.</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/StatUtils.html#foldChange(double%5B%5D,double%5B%5D)" class="member-name-link">foldChange(double[], double[])</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatUtils.html" title="class in bacnet.scripts.core.stat">StatUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatUtils.html#foldChange(double,double)" class="member-name-link">foldChange(double, double)</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatUtils.html" title="class in bacnet.scripts.core.stat">StatUtils</a></dt>
<dd>
<div class="block">Calculate ratio value1/value2</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/StatUtils.html#foldChangeGeneExpression(double%5B%5D,double%5B%5D,boolean)" class="member-name-link">foldChangeGeneExpression(double[], double[], boolean)</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatUtils.html" title="class in bacnet.scripts.core.stat">StatUtils</a></dt>
<dd>
<div class="block">Vector1 and Vectior2 correspond to three technical replicates, so:<br>
 We calculate the fold change by measuring the relative expression (or ratio) of the median
 expression of vector1 and vector2</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/StatUtils.html#foldChangeGeneExpressionWT(double%5B%5D,double,boolean)" class="member-name-link">foldChangeGeneExpressionWT(double[], double, boolean)</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatUtils.html" title="class in bacnet.scripts.core.stat">StatUtils</a></dt>
<dd>
<div class="block">Use this fonction when comparing a vector to a median value already calculated <br>
 For example when vector2Median is equal to WT average expression</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/StatUtils.html#foldChangeTiling(double%5B%5D,double%5B%5D,boolean)" class="member-name-link">foldChangeTiling(double[], double[], boolean)</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatUtils.html" title="class in bacnet.scripts.core.stat">StatUtils</a></dt>
<dd>
<div class="block">Vector1 and vector2 are the values of all the probes contained in a specific genome element,
 so:<br>
 We calculate the difference within vector1-vector2 (probe by probe) And then calculate the median
 the resulting vector</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/StatUtils.html#foldChangeTilingWT(double%5B%5D,double,boolean)" class="member-name-link">foldChangeTilingWT(double[], double, boolean)</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatUtils.html" title="class in bacnet.scripts.core.stat">StatUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#folderCreation(java.lang.String)" class="member-name-link">folderCreation(String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#foldRNA(bacnet.datamodel.sequence.Sequence,java.lang.String,boolean)" class="member-name-link">foldRNA(Sequence, String, boolean)</a> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>
<div class="block">For a specific sequence, get its RNA sequence and fold it<br>
 Calculate possible free energies<br>
 transform results into annotation file<br>
 Display using sir_graph</div>
</dd>
<dt><a href="../bacnet/utils/UNAfold.html#foldRNA(bacnet.datamodel.sequence.Sequence,java.lang.String,boolean)" class="member-name-link">foldRNA(Sequence, String, boolean)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>
<div class="block">For a specific sequence, get its RNA sequence and fold it<br>
 Calculate possible free energies<br>
 transform results into annotation file<br>
 Display using sir_graph</div>
</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#foldRNA(java.lang.String,java.lang.String,java.lang.String,boolean)" class="member-name-link">foldRNA(String, String, String, boolean)</a> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>
<div class="block">For a specific sequence, get its RNA sequence and fold it<br>
 Calculate possible free energies<br>
 transform results into annotation file<br>
 Display using sir_graph</div>
</dd>
<dt><a href="../bacnet/utils/UNAfold.html#foldRNA(java.lang.String,java.lang.String,java.lang.String,boolean)" class="member-name-link">foldRNA(String, String, String, boolean)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>
<div class="block">For a specific sequence, get its RNA sequence and fold it<br>
 Calculate possible free energies<br>
 transform results into annotation file<br>
 Display using sir_graph</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.TypeModif.html#For" class="member-name-link">For</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/GBKReader.html#formatSequence(bacnet.datamodel.sequence.Genome)" class="member-name-link">formatSequence(Genome)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/GBKReader.html" title="class in bacnet.reader">GBKReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#foundSDSequence()" class="member-name-link">foundSDSequence()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Search the consensus SD sequence in the 20bp upstream to a <code>Sequence</code></div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">FourBitArrayWorker(String, CompoundSet&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader.FourBitArrayWorker</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int)" class="member-name-link">FourBitArrayWorker(CompoundSet&lt;C&gt;, int)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader.FourBitArrayWorker</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet,int%5B%5D)" class="member-name-link">FourBitArrayWorker(CompoundSet&lt;C&gt;, int[])</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader.FourBitArrayWorker</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">FourBitArrayWorker(Sequence&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader.FourBitArrayWorker</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.html" class="type-name-link" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader</a>&lt;<a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.html" title="type parameter in FourBitSequenceReader">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</a></dt>
<dd>
<div class="block">Four bit encoding of the bit formats.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">FourBitSequenceReader(String, CompoundSet&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.AccessionID)" class="member-name-link">FourBitSequenceReader(String, CompoundSet&lt;C&gt;, AccessionID)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.html#%3Cinit%3E(org.biojava3.core.sequence.storage.FourBitSequenceReader.FourBitArrayWorker)" class="member-name-link">FourBitSequenceReader(FourBitSequenceReader.FourBitArrayWorker&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.html#%3Cinit%3E(org.biojava3.core.sequence.storage.FourBitSequenceReader.FourBitArrayWorker,org.biojava3.core.sequence.AccessionID)" class="member-name-link">FourBitSequenceReader(FourBitSequenceReader.FourBitArrayWorker&lt;C&gt;, AccessionID)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">FourBitSequenceReader(Sequence&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" class="type-name-link" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader.FourBitArrayWorker</a>&lt;<a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</a></dt>
<dd>
<div class="block">A four bit per compound implementation of the bit array worker code.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#frame()" class="member-name-link">frame()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Get frame (aka phase).</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Frame.html" class="type-name-link" title="enum in org.biojava3.core.sequence.transcription">Frame</a> - Enum in <a href="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></dt>
<dd>
<div class="block">Indicates a way of translating a sequence.</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#fromArraytoList(java.lang.String%5B%5D%5B%5D)" class="member-name-link">fromArraytoList(String[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Transform an array into a List of rows</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#fromBio(int,int,char)" class="member-name-link">fromBio(int, int, char)</a> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Create location from "biocoordinates", as in GFF file.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#fromBioExt(int,int,char,int)" class="member-name-link">fromBioExt(int, int, char, int)</a> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Create a location from MAF file coordinates, which represent negative strand locations as the
 distance from the end of the sequence.</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html#FTP_Path" class="member-name-link">FTP_Path</a> - Static variable in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#FTP_URL" class="member-name-link">FTP_URL</a> - Static variable in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#FTP_URL" class="member-name-link">FTP_URL</a> - Static variable in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceUsage.html#FULL_SEQUENCE_DATA" class="member-name-link">FULL_SEQUENCE_DATA</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceUsage.html" title="enum in org.biojava3.core.sequence">SequenceOptimizationHints.SequenceUsage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#fusion(java.lang.String%5B%5D%5B%5D,java.lang.String%5B%5D%5B%5D,boolean)" class="member-name-link">fusion(String[][], String[][], boolean)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Return a new Array of String which is the fusion of both array If header = true; the first row of
 array2 is conserved</div>
</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html#fusionProteinUTRDB()" class="member-name-link">fusionProteinUTRDB()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Fusion ProteinDB and UTRDB in one file</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#fusionProteinUTRDB()" class="member-name-link">fusionProteinUTRDB()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Fusion ProteinDB and UTRDB in one file</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" class="type-name-link" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a> - Class in <a href="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</a></dt>
<dd>
<div class="block">Implementation for resolving fuzzy locations.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#%3Cinit%3E(int,int)" class="member-name-link">FuzzyPoint(int, int)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#%3Cinit%3E(int,int,org.biojava3.core.sequence.location.template.Point.Resolver)" class="member-name-link">FuzzyPoint(int, int, Point.Resolver&lt;FuzzyPoint&gt;)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#%3Cinit%3E(int,int,org.biojava3.core.sequence.location.template.Point.Resolver,boolean,boolean)" class="member-name-link">FuzzyPoint(int, int, Point.Resolver&lt;FuzzyPoint&gt;, boolean, boolean)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
