<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>A-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="index: A">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:A">A</h2>
<dl class="index">
<dt><a href="../bacnet/scripts/core/stat/AMtools.html#A(double,double)" class="member-name-link">A(double, double)</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/AMtools.html" title="class in bacnet.scripts.core.stat">AMtools</a></dt>
<dd>
<div class="block">Calculate A = (x1+x2)/2<br>
 Represents mean value of x1+x2</div>
</dd>
<dt><a href="../bacnet/utils/MathUtils.html#A_nk(int,int)" class="member-name-link">A_nk(int, int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/MathUtils.html" title="class in bacnet.utils">MathUtils</a></dt>
<dd>
<div class="block">Calculate the argument A_nk</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#absoluteTOrelativeValue()" class="member-name-link">absoluteTOrelativeValue()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Changed all BioCondition to display when switching from "ABSOLUTE VALUE DISPLAY3 to "RELATIVE
 VALUE DISPLAY"</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompound.html" class="type-name-link" title="class in org.biojava3.core.sequence.template">AbstractCompound</a> - Class in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>
<div class="block">The details of a Compound</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#%3Cinit%3E(java.lang.String)" class="member-name-link">AbstractCompound(String)</a> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" class="type-name-link" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a>&lt;<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="type parameter in AbstractCompoundSet">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#%3Cinit%3E()" class="member-name-link">AbstractCompoundSet()</a> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" class="type-name-link" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a>&lt;<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>,<wbr><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#%3Cinit%3E(org.biojava3.core.sequence.io.template.SequenceCreatorInterface,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">AbstractCompoundTranslator(SequenceCreatorInterface&lt;T&gt;, CompoundSet&lt;F&gt;, CompoundSet&lt;T&gt;)</a> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html#%3Cinit%3E(int)" class="member-name-link">AbstractEdit(int)</a> - Constructor for class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="class in org.biojava3.core.sequence.edits">Edit.AbstractEdit</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html#%3Cinit%3E(int,int)" class="member-name-link">AbstractEdit(int, int)</a> - Constructor for class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="class in org.biojava3.core.sequence.edits">Edit.AbstractEdit</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html" class="type-name-link" title="class in org.biojava3.core.sequence.features">AbstractFeature</a>&lt;<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="type parameter in AbstractFeature">S</a> extends <a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="type parameter in AbstractFeature">C</a>&gt;,<wbr><a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="type parameter in AbstractFeature">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/features/package-summary.html">org.biojava3.core.sequence.features</a></dt>
<dd>
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s) A
 feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#%3Cinit%3E(java.lang.String,java.lang.String)" class="member-name-link">AbstractFeature(String, String)</a> - Constructor for class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">A feature has a type and a source</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" class="type-name-link" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a> - Class in <a href="../org/biojava3/core/sequence/location/template/package-summary.html">org.biojava3.core.sequence.location.template</a></dt>
<dd>
<div class="block">Base abstraction of a location which encodes for the majority of important features about a
 location such as the start, end and strand</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,boolean,java.util.List)" class="member-name-link">AbstractLocation(Point, Point, Strand, boolean, boolean, List&lt;Location&gt;)</a> - Constructor for class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>
<div class="block">Default constructor</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,boolean,org.biojava3.core.sequence.AccessionID,java.util.List)" class="member-name-link">AbstractLocation(Point, Point, Strand, boolean, boolean, AccessionID, List&lt;Location&gt;)</a> - Constructor for class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>
<div class="block">Default constructor</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html" class="type-name-link" title="class in org.biojava3.core.sequence.template">AbstractNucleotideCompoundSet</a>&lt;<a href="../org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html" title="type parameter in AbstractNucleotideCompoundSet">C</a> extends <a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html#%3Cinit%3E()" class="member-name-link">AbstractNucleotideCompoundSet()</a> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractNucleotideCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html" class="type-name-link" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>
<div class="block">The base class for DNA, RNA and Protein sequences.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#%3Cinit%3E()" class="member-name-link">AbstractSequence()</a> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">AbstractSequence(String, CompoundSet&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Create a Sequence from a simple string where the values should be found in compoundSet</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">AbstractSequence(SequenceReader&lt;C&gt;, CompoundSet&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">A ProxySequenceReader allows abstraction of both the storage of the sequence data and the
 location of the sequence data.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" class="type-name-link" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a> - Enum in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AccesionedLocation.html" class="type-name-link" title="interface in org.biojava3.core.sequence.location.template">AccesionedLocation</a> - Interface in <a href="../org/biojava3/core/sequence/location/template/package-summary.html">org.biojava3.core.sequence.location.template</a></dt>
<dd>
<div class="block">A location which is bound to an AccessionID.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/Accessioned.html" class="type-name-link" title="interface in org.biojava3.core.sequence.template">Accessioned</a> - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>
<div class="block">Indicates an entity is accessioned</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/AccessionID.html" class="type-name-link" title="class in org.biojava3.core.sequence">AccessionID</a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">Used in Sequences as the unique indentifier.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/AccessionID.html#%3Cinit%3E()" class="member-name-link">AccessionID()</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/AccessionID.html#%3Cinit%3E(java.lang.String)" class="member-name-link">AccessionID(String)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/AccessionID.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.DataSource)" class="member-name-link">AccessionID(String, DataSource)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.TypeModif.html#AcD3" class="member-name-link">AcD3</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.TypeModif.html#Ace" class="member-name-link">Ace</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html#aCM" class="member-name-link">aCM</a> - Enum constant in enum bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/Activator.html" class="type-name-link" title="class in bacnet.e4.rap.setup">Activator</a> - Class in <a href="../bacnet/e4/rap/setup/package-summary.html">bacnet.e4.rap.setup</a></dt>
<dd>
<div class="block">OSGI Activator of bacnet.e4.rap.setup</div>
</dd>
<dt><a href="../bacnet/scripts/Activator.html" class="type-name-link" title="class in bacnet.scripts">Activator</a> - Class in <a href="../bacnet/scripts/package-summary.html">bacnet.scripts</a></dt>
<dd>
<div class="block">OSGI Activator of bacnet.scripts</div>
</dd>
<dt><a href="../bacnet/e4/rap/setup/Activator.html#%3Cinit%3E()" class="member-name-link">Activator()</a> - Constructor for class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/Activator.html" title="class in bacnet.e4.rap.setup">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/Activator.html#%3Cinit%3E()" class="member-name-link">Activator()</a> - Constructor for class bacnet.scripts.<a href="../bacnet/scripts/Activator.html" title="class in bacnet.scripts">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/StringColor.html#add(java.lang.String)" class="member-name-link">add(String)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/StringColor.html" title="class in bacnet.utils">StringColor</a></dt>
<dd>
<div class="block">Add text Default colorization will be applied BLACK = Foreground WHITE = Background</div>
</dd>
<dt><a href="../bacnet/utils/StringColor.html#add(java.lang.String,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color)" class="member-name-link">add(String, Color, Color)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/StringColor.html" title="class in bacnet.utils">StringColor</a></dt>
<dd>
<div class="block">Add text colorF will be applied to the foreground colorB will be applied to the background</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#add(java.util.Collection)" class="member-name-link">add(Collection&lt;FeatureI&gt;)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Add all features in the specified list or collection to this list.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#add(org.biojava3.genome.parsers.gff.FeatureI)" class="member-name-link">add(FeatureI)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Add specified feature to the end of the list.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#addAlignedSequence(S)" class="member-name-link">addAlignedSequence(S)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>
<div class="block">A sequence that has been aligned to other sequences will have inserts.</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#addAnnotation(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome)" class="member-name-link">addAnnotation(ExpressionMatrix, Genome)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Add annotation information to an ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/reader/GFFNCBIReader.html#addAnnotationFromGFF(java.io.File,java.util.LinkedHashMap)" class="member-name-link">addAnnotationFromGFF(File, LinkedHashMap&lt;String, ChromosomeBacteriaSequence&gt;)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/GFFNCBIReader.html" title="class in bacnet.reader">GFFNCBIReader</a></dt>
<dd>
<div class="block">Read a GFF and for each Gene associate all the corresponding information</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#addAnnotationLite(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome)" class="member-name-link">addAnnotationLite(ExpressionMatrix, Genome)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Add annotation information to an ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#addAnnotationMultiChromosome(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome)" class="member-name-link">addAnnotationMultiChromosome(ExpressionMatrix, Genome)</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Add annotation information to an ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/utils/StringColor.html#addB(java.lang.String,org.eclipse.swt.graphics.Color)" class="member-name-link">addB(String, Color)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/StringColor.html" title="class in bacnet.utils">StringColor</a></dt>
<dd>
<div class="block">Add text The colorization will be applied on the BACKGROUND</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#addBioCond(bacnet.datamodel.expdesign.BioCondition)" class="member-name-link">addBioCond(BioCondition)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Add using the bioCondName as a key</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#addBioCond(java.lang.String,bacnet.datamodel.expdesign.BioCondition)" class="member-name-link">addBioCond(String, BioCondition)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Add using a specific Key</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#addBioCondition(java.lang.String)" class="member-name-link">addBioCondition(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">Read every ExpressionData contained in bioCondName</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/TranscriptSequence.html#addCDS(org.biojava3.core.sequence.AccessionID,int,int,int)" class="member-name-link">addCDS(AccessionID, int, int, int)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>
<div class="block">Add a Coding Sequence region with phase to the transcript sequence</div>
</dd>
<dt><a href="../org/biojava3/core/util/XMLHelper.html#addChildElement(org.w3c.dom.Element,java.lang.String)" class="member-name-link">addChildElement(Element, String)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/FastaFileReader.html#addCodingRegionFeaturesFromFFN(java.io.File,java.io.File,java.util.LinkedHashMap)" class="member-name-link">addCodingRegionFeaturesFromFFN(File, File, LinkedHashMap&lt;String, ChromosomeBacteriaSequence&gt;)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a></dt>
<dd>
<div class="block">From a .ffn file in root/file.faa We had all CodingInfo into the ChromosomeSequence
 
 This method use my personal NCBIFastaHeaderParser</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#addColumn(double%5B%5D%5B%5D,double%5B%5D)" class="member-name-link">addColumn(double[][], double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#addColumn(java.lang.String%5B%5D%5B%5D,java.lang.String%5B%5D)" class="member-name-link">addColumn(String[][], String[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#addColumn(java.lang.String,double%5B%5D)" class="member-name-link">addColumn(String, double[])</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Add a column at the end of Matrix</div>
</dd>
<dt><a href="../bacnet/scripts/database/BioConditionCreation.html#addComparisonFromTable(java.lang.String)" class="member-name-link">addComparisonFromTable(String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/BioConditionCreation.html" title="class in bacnet.scripts.database">BioConditionCreation</a></dt>
<dd>
<div class="block">Read a Table and construct the list of Comparisons from it</div>
</dd>
<dt><a href="../bacnet/scripts/database/BioConditionCreation.html#addComparisonFromTable(java.util.ArrayList,java.lang.String)" class="member-name-link">addComparisonFromTable(ArrayList&lt;String&gt;, String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/BioConditionCreation.html" title="class in bacnet.scripts.database">BioConditionCreation</a></dt>
<dd>
<div class="block">Read a Table and construct the list of Comparisons from it</div>
</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#addDeviation(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">addDeviation(ExpressionMatrix, ExpressionMatrix)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#addEGDeFilter()" class="member-name-link">addEGDeFilter()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>
<div class="block">Read "10403S-noMatch-overlap.txt" and add a column indicating if the corresponding lmo gene is
 kept or not in "EGD-e-noMatch-decision.txt"</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#addElement(bacnet.datamodel.proteomics.NTerm)" class="member-name-link">addElement(NTerm)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Add a NTerm in elements HashMap:<br>
 key = nTerm.getName()<br>
 value = nTerm</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#addElement(java.lang.Object%5B%5D,java.lang.String)" class="member-name-link">addElement(Object[], String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Add an element to the end of the vector</div>
</dd>
<dt><a href="../bacnet/utils/StringColor.html#addElement(org.eclipse.swt.graphics.Color%5B%5D,org.eclipse.swt.graphics.Color)" class="member-name-link">addElement(Color[], Color)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/StringColor.html" title="class in bacnet.utils">StringColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#addExon(org.biojava3.core.sequence.AccessionID,int,int)" class="member-name-link">addExon(AccessionID, int, int)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Add an ExonSequence mainly used to mark as a feature</div>
</dd>
<dt><a href="../bacnet/scripts/database/BioConditionCreation.html#addExperimentFromTable(java.lang.String)" class="member-name-link">addExperimentFromTable(String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/BioConditionCreation.html" title="class in bacnet.scripts.database">BioConditionCreation</a></dt>
<dd>
<div class="block">Read a Table summarizing all BioCondition and parse every information available<br>
 Save in BioCondition.PATH every <code>BioCondition</code> created</div>
</dd>
<dt><a href="../bacnet/scripts/database/BioConditionCreation.html#addExperimentFromTable(java.lang.String,java.util.ArrayList,java.lang.String)" class="member-name-link">addExperimentFromTable(String, ArrayList&lt;String&gt;, String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/BioConditionCreation.html" title="class in bacnet.scripts.database">BioConditionCreation</a></dt>
<dd>
<div class="block">Read a Table summarizing all BioCondition and parse every information available<br>
 Save in BioCondition.PATH every <code>BioCondition</code> created</div>
</dd>
<dt><a href="../bacnet/utils/StringColor.html#addF(java.lang.String,org.eclipse.swt.graphics.Color)" class="member-name-link">addF(String, Color)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/StringColor.html" title="class in bacnet.utils">StringColor</a></dt>
<dd>
<div class="block">Add text The colorization will be applied on the FOREGROUND</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#addFeature(int,int,org.biojava3.core.sequence.features.FeatureInterface)" class="member-name-link">addFeature(int, int, FeatureInterface&lt;AbstractSequence&lt;C&gt;, C&gt;)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Method to help set the proper details for a feature as it relates to a sequence where the feature
 needs to have a location on the sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#addFeature(org.biojava3.core.sequence.features.FeatureInterface)" class="member-name-link">addFeature(FeatureInterface&lt;AbstractSequence&lt;C&gt;, C&gt;)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Add a feature to this sequence.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#addGene(org.biojava3.core.sequence.AccessionID,int,int,org.biojava3.core.sequence.Strand)" class="member-name-link">addGene(AccessionID, int, int, Strand)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>
<div class="block">Add a gene to the chromosome sequence using bioIndexing starts at 1 instead of 0.</div>
</dd>
<dt><a href="../bacnet/reader/PTTReader.html#addGeneFeaturesFromPTT(java.io.File,java.io.File,java.util.LinkedHashMap)" class="member-name-link">addGeneFeaturesFromPTT(File, File, LinkedHashMap&lt;String, ChromosomeBacteriaSequence&gt;)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/PTTReader.html" title="class in bacnet.reader">PTTReader</a></dt>
<dd>
<div class="block">Add feature from a PTT file<br>
 IMPORTANT: Need to be modified!</div>
</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html#addGeneIDGFF2GeneFeatures(java.util.LinkedHashMap,org.biojava3.genome.parsers.gff.FeatureList)" class="member-name-link">addGeneIDGFF2GeneFeatures(LinkedHashMap&lt;String, ChromosomeSequence&gt;, FeatureList)</a> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>
<div class="block">Load GFF2 feature file generated from the geneid prediction algorithm and map features onto the
 chromosome sequences</div>
</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html#addGeneMarkGTFGeneFeatures(java.util.LinkedHashMap,org.biojava3.genome.parsers.gff.FeatureList)" class="member-name-link">addGeneMarkGTFGeneFeatures(LinkedHashMap&lt;String, ChromosomeSequence&gt;, FeatureList)</a> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html#addGlimmerGFF3GeneFeatures(java.util.LinkedHashMap,org.biojava3.genome.parsers.gff.FeatureList)" class="member-name-link">addGlimmerGFF3GeneFeatures(LinkedHashMap&lt;String, ChromosomeSequence&gt;, FeatureList)</a> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html#addGmodGFF3GeneFeatures(java.util.LinkedHashMap,org.biojava3.genome.parsers.gff.FeatureList)" class="member-name-link">addGmodGFF3GeneFeatures(LinkedHashMap&lt;String, ChromosomeSequence&gt;, FeatureList)</a> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>
<div class="block">Load GFF3 file using mRNA as the gene feature as not all GFF3 files are complete</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#addHeader(java.lang.String)" class="member-name-link">addHeader(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#addHomologToGene(java.lang.String)" class="member-name-link">addHomologToGene(String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Add homologs to every gene by parsing all homolog tables</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#addInfoTOGenomePhylogeny(org.forester.phylogeny.Phylogeny)" class="member-name-link">addInfoTOGenomePhylogeny(Phylogeny)</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>
<div class="block">Get a Phylogeny where nodeName contain ID from Jolley and add all taxonomy information contained
 in it</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyTools.html#addInfoToPhylogeny(org.forester.phylogeny.Phylogeny,java.util.HashMap)" class="member-name-link">addInfoToPhylogeny(Phylogeny, HashMap&lt;String, String[]&gt;)</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#addInfoToPhylogeny(org.forester.phylogeny.Phylogeny,java.util.HashMap)" class="member-name-link">addInfoToPhylogeny(Phylogeny, HashMap&lt;String, String[]&gt;)</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#addIntron(org.biojava3.core.sequence.AccessionID,int,int)" class="member-name-link">addIntron(AccessionID, int, int)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Add an Intron Currently used to mark an IntronSequence as a feature</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#addIntronsUsingExons()" class="member-name-link">addIntronsUsingExons()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Once everything has been added to the gene sequence where you might have added exon sequences
 only then you can infer the intron sequences and add them.</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#addition(double,double%5B%5D%5B%5D,double,double%5B%5D%5B%5D)" class="member-name-link">addition(double, double[][], double, double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">return=(coeff1)*array1 + (coeff2)*array2</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#addition(double,double%5B%5D,double,double%5B%5D)" class="member-name-link">addition(double, double[], double, double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">return=(coeff1)*array1 + (coeff2)*array2</div>
</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#addition(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.dataset.ExpressionMatrix,boolean)" class="member-name-link">addition(ExpressionMatrix, ExpressionMatrix, boolean)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Perform an addition or soustraction between two ExpressionMatrix <br>
 First the intersection in term of RowNames and headers is found, then addition is done for each
 row</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/LocusTag.html#addLocusChange()" class="member-name-link">addLocusChange()</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/LocusTag.html" title="class in bacnet.datamodel.annotation">LocusTag</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#addMaXQuantInfo(bacnet.datamodel.dataset.NTermData,java.lang.String)" class="member-name-link">addMaXQuantInfo(NTermData, String)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">Compare to peptides from MaxQuant software analysis<br>
 For each NTerm search if the same peptide and the same modif has been found in MaxQuant analysis.</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#addMaXQuantInfo(bacnet.datamodel.dataset.NTermData,java.lang.String,boolean)" class="member-name-link">addMaXQuantInfo(NTermData, String, boolean)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">Add MaxQuant information for each peptide <br>
 This function is used only when multiple files are available</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#addMaXQuantInfo(bacnet.datamodel.dataset.NTermData,java.lang.String,int)" class="member-name-link">addMaXQuantInfo(NTermData, String, int)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#addMean(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">addMean(ExpressionMatrix, ExpressionMatrix)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#addMedian(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">addMedian(ExpressionMatrix, ExpressionMatrix)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/ProteomicsCreation.html#addMissingValuesToMatrices(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)" class="member-name-link">addMissingValuesToMatrices(Experiment, Genome)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>
<div class="block">In order to represent proteomics data we need to add a -1 value in every missing row.<br>
 ATTENTION: -1 is now a value when there is no value but it is detected</div>
</dd>
<dt><a href="../bacnet/reader/FastaFileReader.html#addNonCodingRegionFeaturesFromFRN(java.io.File,java.io.File,java.util.LinkedHashMap)" class="member-name-link">addNonCodingRegionFeaturesFromFRN(File, File, LinkedHashMap&lt;String, ChromosomeBacteriaSequence&gt;)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a></dt>
<dd>
<div class="block">From a .frn file in root/file.frn We add all Noncoding Info into the ChromosomeSequence
 
 This method use my personal NCBIFastaHeaderParser</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#addNote(java.lang.String)" class="member-name-link">addNote(String)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Add notes about this sequence that will get exported for GFF3</div>
</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizard.html#addPages()" class="member-name-link">addPages()</a> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizard.html" title="class in bacnet.table.gui">ColorMapperWizard</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#addPhylogenyToGenes()" class="member-name-link">addPhylogenyToGenes()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>
<div class="block">Go through all result files and create Conservation HadhMap for each Gene</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#addProperty(java.lang.String,java.lang.String)" class="member-name-link">addProperty(String, String)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>
<div class="block">Add a property and type to associate with this DBReferenceInfo</div>
</dd>
<dt><a href="../bacnet/scripts/database/ProteomicsCreation.html#addProteomesToDatabase(java.util.ArrayList)" class="member-name-link">addProteomesToDatabase(ArrayList&lt;String&gt;)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>
<div class="block">Clean proteomics tables, and convert them for the database</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html" class="type-name-link" title="class in bacnet.genomeBrowser.dialog">AddProteomicsDataDialog</a> - Class in <a href="../bacnet/genomeBrowser/dialog/package-summary.html">bacnet.genomeBrowser.dialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,bacnet.genomeBrowser.core.Track)" class="member-name-link">AddProteomicsDataDialog(Shell, Track)</a> - Constructor for class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddProteomicsDataDialog</a></dt>
<dd>
<div class="block">Create the dialog.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/QuantityFeature.html#addQuantity(java.lang.Number)" class="member-name-link">addQuantity(Number)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/QuantityFeature.html" title="class in org.biojava3.core.sequence.features">QuantityFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermMain.html#addRASTInfo()" class="member-name-link">addRASTInfo()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermMain.html" title="class in bacnet.scripts.listeriomics.nterm">NTermMain</a></dt>
<dd>
<div class="block">Add RAST info to final supplementary tables</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#addRow(double%5B%5D%5B%5D,double%5B%5D)" class="member-name-link">addRow(double[][], double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#addRow(java.lang.String%5B%5D%5B%5D,java.lang.String%5B%5D,int)" class="member-name-link">addRow(String[][], String[], int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Add the input row in the String[][] array, at the postiion given by index integer</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#addRow(java.lang.String,double%5B%5D)" class="member-name-link">addRow(String, double[])</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Add a row at the end of values</div>
</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#addSet(java.lang.String,java.util.ArrayList)" class="member-name-link">addSet(String, ArrayList&lt;String&gt;)</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/TranscriptSequence.html#addStartCodonSequence(org.biojava3.core.sequence.AccessionID,int,int)" class="member-name-link">addStartCodonSequence(AccessionID, int, int)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/TranscriptSequence.html#addStopCodonSequence(org.biojava3.core.sequence.AccessionID,int,int)" class="member-name-link">addStopCodonSequence(AccessionID, int, int)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/FlatFileCache.html#addToCache(java.lang.String,java.io.File)" class="member-name-link">addToCache(String, File)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/FlatFileCache.html" title="class in org.biojava3.core.util">FlatFileCache</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#addToGenome(java.lang.String)" class="member-name-link">addToGenome(String)</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Add Srna to any Genome by reading BlastResults and extracting the lower e-value result<br>
 Srna with an identities less than 25% are not kept!!!</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermMain.html#addTotalSpectraCount()" class="member-name-link">addTotalSpectraCount()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermMain.html" title="class in bacnet.scripts.listeriomics.nterm">NTermMain</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#addTranscript(org.biojava3.core.sequence.AccessionID,int,int)" class="member-name-link">addTranscript(AccessionID, int, int)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>
<div class="block">Add a transcription sequence to a gene which describes a ProteinSequence</div>
</dd>
<dt><a href="../bacnet/scripts/database/TranscriptomesCreation.html#addTranscriptomeToDatabase(java.util.ArrayList,java.lang.String)" class="member-name-link">addTranscriptomeToDatabase(ArrayList&lt;String&gt;, String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/TranscriptomesCreation.html" title="class in bacnet.scripts.database">TranscriptomesCreation</a></dt>
<dd>
<div class="block">Convert all transcriptomics files<br>
 <br>
 - Convert all Seq data: RNASeq, RiboSeq, TSS, TermSeq<br>
 - Convert all GeneExpression and Tiling<br>
 - Convert all ExpressionMatrix<br></div>
</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html" class="type-name-link" title="class in bacnet.genomeBrowser.dialog">AddTranscriptomicsDataDialog</a> - Class in <a href="../bacnet/genomeBrowser/dialog/package-summary.html">bacnet.genomeBrowser.dialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,bacnet.genomeBrowser.core.Track)" class="member-name-link">AddTranscriptomicsDataDialog(Shell, Track)</a> - Constructor for class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddTranscriptomicsDataDialog</a></dt>
<dd>
<div class="block">Create the dialog.</div>
</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#addVariance(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">addVariance(ExpressionMatrix, ExpressionMatrix)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#adfCleaning()" class="member-name-link">adfCleaning()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/FDR.html#adjust(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.scripts.core.stat.StatTest,boolean)" class="member-name-link">adjust(ExpressionMatrix, StatTest, boolean)</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/FDR.html" title="class in bacnet.scripts.core.stat">FDR</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/Agilent.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.technology">Agilent</a> - Class in <a href="../bacnet/scripts/listeriomics/technology/package-summary.html">bacnet.scripts.listeriomics.technology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/Agilent.html#%3Cinit%3E()" class="member-name-link">Agilent()</a> - Constructor for class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Agilent.html" title="class in bacnet.scripts.listeriomics.technology">Agilent</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#alignblastN(java.lang.String,java.lang.String,java.lang.String,double)" class="member-name-link">alignblastN(String, String, String, double)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>
<div class="block">Run a BlastN given a sequence name, an amino acid sequence and a database name</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html#alignBlastP(java.lang.String,java.lang.String,java.util.ArrayList)" class="member-name-link">alignBlastP(String, String, ArrayList&lt;String&gt;)</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html" title="class in bacnet.scripts.listeriomics.srna">SmallPeptidesSearch</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#alignLength" class="member-name-link">alignLength</a> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#alignProtein(java.lang.String,java.lang.String,java.lang.String,double)" class="member-name-link">alignProtein(String, String, String, double)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>
<div class="block">Run a BlastP given a sequence name, an amino acid sequence and a database name</div>
</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#alignTblastN(java.lang.String,java.lang.String,java.lang.String,double)" class="member-name-link">alignTblastN(String, String, String, double)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>
<div class="block">Run a TBlastN given a sequence name, an amino acid sequence and a database name</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceCollection.html#ALL_SEQUENCES" class="member-name-link">ALL_SEQUENCES</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceCollection.html" title="enum in org.biojava3.core.sequence">SequenceOptimizationHints.SequenceCollection</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#ALLBACTERIA_NWK_1" class="member-name-link">ALLBACTERIA_NWK_1</a> - Static variable in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#ALLBACTERIA_NWK_2" class="member-name-link">ALLBACTERIA_NWK_2</a> - Static variable in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyTools.html#ALLBACTERIA_XML" class="member-name-link">ALLBACTERIA_XML</a> - Static variable in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyTools.html#ALLBACTERIA_XML_1" class="member-name-link">ALLBACTERIA_XML_1</a> - Static variable in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyTools.html#ALLBACTERIA_XML_2" class="member-name-link">ALLBACTERIA_XML_2</a> - Static variable in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/FastaFileReader.html#allfileExt" class="member-name-link">allfileExt</a> - Static variable in class bacnet.reader.<a href="../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#ALLFIRMICUTES_AVAILABLE" class="member-name-link">ALLFIRMICUTES_AVAILABLE</a> - Static variable in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#ALLFIRMICUTES_NWK" class="member-name-link">ALLFIRMICUTES_NWK</a> - Static variable in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html#ALN" class="member-name-link">ALN</a> - Enum constant in enum org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html" title="enum in org.biojava3.core.sequence.template">LightweightProfile.StringFormat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/AMtools.html#AM(double%5B%5D%5B%5D)" class="member-name-link">AM(double[][])</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/AMtools.html" title="class in bacnet.scripts.core.stat">AMtools</a></dt>
<dd>
<div class="block">Calculate AM matrix</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/AMtools.html#AM(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">AM(ExpressionMatrix)</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/AMtools.html" title="class in bacnet.scripts.core.stat">AMtools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AmbiguityDNACompoundSet.html" class="type-name-link" title="class in org.biojava3.core.sequence.compound">AmbiguityDNACompoundSet</a> - Class in <a href="../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AmbiguityDNACompoundSet.html#%3Cinit%3E()" class="member-name-link">AmbiguityDNACompoundSet()</a> - Constructor for class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AmbiguityDNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">AmbiguityDNACompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AmbiguityRNACompoundSet.html" class="type-name-link" title="class in org.biojava3.core.sequence.compound">AmbiguityRNACompoundSet</a> - Class in <a href="../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AmbiguityRNACompoundSet.html#%3Cinit%3E()" class="member-name-link">AmbiguityRNACompoundSet()</a> - Constructor for class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AmbiguityRNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">AmbiguityRNACompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Codon.html#aminoacidCharge" class="member-name-link">aminoacidCharge</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompound.html" class="type-name-link" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a> - Class in <a href="../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</a></dt>
<dd>
<div class="block">Used to describe an Amino Acid.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompound.html#%3Cinit%3E(org.biojava3.core.sequence.compound.AminoAcidCompoundSet,java.lang.String,java.lang.String,java.lang.String,java.lang.Float)" class="member-name-link">AminoAcidCompound(AminoAcidCompoundSet, String, String, String, Float)</a> - Constructor for class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" class="type-name-link" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a> - Class in <a href="../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</a></dt>
<dd>
<div class="block">Set of proteinogenic amino acids.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#%3Cinit%3E()" class="member-name-link">AminoAcidCompoundSet()</a> - Constructor for class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#aminoAcidsCompounds(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">aminoAcidsCompounds(CompoundSet&lt;AminoAcidCompound&gt;)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/AMtools.html" class="type-name-link" title="class in bacnet.scripts.core.stat">AMtools</a> - Class in <a href="../bacnet/scripts/core/stat/package-summary.html">bacnet.scripts.core.stat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/AMtools.html#%3Cinit%3E()" class="member-name-link">AMtools()</a> - Constructor for class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/AMtools.html" title="class in bacnet.scripts.core.stat">AMtools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html" class="type-name-link" title="class in bacnet.datamodel.annotation">Annotation</a> - Class in <a href="../bacnet/datamodel/annotation/package-summary.html">bacnet.datamodel.annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#%3Cinit%3E()" class="member-name-link">Annotation()</a> - Constructor for class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#%3Cinit%3E(java.lang.String%5B%5D%5B%5D)" class="member-name-link">Annotation(String[][])</a> - Constructor for class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html" class="type-name-link" title="class in bacnet.sequenceTools">AnnotationView</a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#%3Cinit%3E()" class="member-name-link">AnnotationView()</a> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#ANTI_SD_SEQ" class="member-name-link">ANTI_SD_SEQ</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Anti-SD sequence taken from Correlations between Shine-Dalgarno Sequences and Gene Features Such
 as Predicted Expression Levels and Operon Structures Jiong Ma, Allan Campbell, and Samuel Karlin
 J.</div>
</dd>
<dt><a href="../org/biojava3/genome/App.html" class="type-name-link" title="class in org.biojava3.genome">App</a> - Class in <a href="../org/biojava3/genome/package-summary.html">org.biojava3.genome</a></dt>
<dd>
<div class="block">Hello world!</div>
</dd>
<dt><a href="../org/biojava3/genome/App.html#%3Cinit%3E()" class="member-name-link">App()</a> - Constructor for class org.biojava3.genome.<a href="../org/biojava3/genome/App.html" title="class in org.biojava3.genome">App</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#applyCutOff(java.lang.String,bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Sequence,int%5B%5D)" class="member-name-link">applyCutOff(String, NTerm, Sequence, int[])</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>
<div class="block">Read a cutoff<br>
 Interprete each cutoff by returning true of false if the NTerm pass the condition<br>
 If the cutoff contain an "or": parse it in two separate cutoff<br>
 If the cutoff starts with an "!"</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#applyFilter()" class="member-name-link">applyFilter()</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>
<div class="block">Search for overlap with Gene, sRNAn, asRNA and ncRNA of each NTerm</div>
</dd>
<dt><a href="../bacnet/e4/rap/AppSpecificMethods.html" class="type-name-link" title="class in bacnet.e4.rap">AppSpecificMethods</a> - Class in <a href="../bacnet/e4/rap/package-summary.html">bacnet.e4.rap</a></dt>
<dd>
<div class="block">All the methods which are specific to a webpage</div>
</dd>
<dt><a href="../bacnet/e4/rap/AppSpecificMethods.html#%3Cinit%3E()" class="member-name-link">AppSpecificMethods()</a> - Constructor for class bacnet.e4.rap.<a href="../bacnet/e4/rap/AppSpecificMethods.html" title="class in bacnet.e4.rap">AppSpecificMethods</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" class="type-name-link" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a> - Class in <a href="../bacnet/scripts/arrayexpress/package-summary.html">bacnet.scripts.arrayexpress</a></dt>
<dd>
<div class="block">Different methods to curate and parse the different data for Listeriomics website <br>
 <br>
 
 ONLY METHODS BEFORE FINAL DATA PROCESSING ARE PRESENT HERE</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#%3Cinit%3E()" class="member-name-link">ArrayExpress()</a> - Constructor for class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#ARRAYEXPRESS_PATH" class="member-name-link">ARRAYEXPRESS_PATH</a> - Static variable in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" class="type-name-link" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a> - Class in <a href="../bacnet/scripts/arrayexpress/package-summary.html">bacnet.scripts.arrayexpress</a></dt>
<dd>
<div class="block">List of methods to download all ArrayExpress needed for Listeriomics website</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html#%3Cinit%3E()" class="member-name-link">ArrayExpressDataImport()</a> - Constructor for class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" class="type-name-link" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a> - Class in <a href="../bacnet/scripts/arrayexpress/package-summary.html">bacnet.scripts.arrayexpress</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a> - Class in <a href="../bacnet/scripts/listeriomics/ArrayExpress/package-summary.html">bacnet.scripts.listeriomics.ArrayExpress</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#%3Cinit%3E()" class="member-name-link">ArrayExpressDataUtils()</a> - Constructor for class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#%3Cinit%3E()" class="member-name-link">ArrayExpressDataUtils()</a> - Constructor for class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressListeriomics</a> - Class in <a href="../bacnet/scripts/listeriomics/ArrayExpress/package-summary.html">bacnet.scripts.listeriomics.ArrayExpress</a></dt>
<dd>
<div class="block">Different methods to curate and parse the different data for Listeriomics website <br>
 <br>
 
 ONLY METHODS BEFORE FINAL DATA PROCESSING ARE PRESENT HERE</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html" class="type-name-link" title="class in bacnet.scripts.listeriomics">ArrayExpressListeriomics</a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>
<div class="block">Different methods to curate and parse the different data for Listeriomics website <br>
 <br>
 
 ONLY METHODS BEFORE FINAL DATA PROCESSING ARE PRESENT HERE</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html#%3Cinit%3E()" class="member-name-link">ArrayExpressListeriomics()</a> - Constructor for class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html#%3Cinit%3E()" class="member-name-link">ArrayExpressListeriomics()</a> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics">ArrayExpressListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" class="type-name-link" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a> - Class in <a href="../bacnet/scripts/arrayexpress/package-summary.html">bacnet.scripts.arrayexpress</a></dt>
<dd>
<div class="block">List of methods to determine the different technology used</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a> - Class in <a href="../bacnet/scripts/listeriomics/ArrayExpress/package-summary.html">bacnet.scripts.listeriomics.ArrayExpress</a></dt>
<dd>
<div class="block">List of methods to determine the different technology used</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#%3Cinit%3E()" class="member-name-link">ArrayExpressTechnology()</a> - Constructor for class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#%3Cinit%3E()" class="member-name-link">ArrayExpressTechnology()</a> - Constructor for class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/ArrayListProxySequenceReader.html" class="type-name-link" title="class in org.biojava3.core.sequence.loader">ArrayListProxySequenceReader</a>&lt;<a href="../org/biojava3/core/sequence/loader/ArrayListProxySequenceReader.html" title="type parameter in ArrayListProxySequenceReader">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/ArrayListProxySequenceReader.html#%3Cinit%3E()" class="member-name-link">ArrayListProxySequenceReader()</a> - Constructor for class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/ArrayListProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">ArrayListProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" class="type-name-link" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a>&lt;<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</a></dt>
<dd>
<div class="block">Stores a Sequence as a collection of compounds in an ArrayList</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#%3Cinit%3E()" class="member-name-link">ArrayListSequenceReader()</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ArrayListSequenceReader(String, CompoundSet&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#%3Cinit%3E(java.util.List,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ArrayListSequenceReader(List&lt;C&gt;, CompoundSet&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#ARRAYSCORR" class="member-name-link">ARRAYSCORR</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#arrayToExpressionMatrix(java.lang.String%5B%5D%5B%5D,boolean)" class="member-name-link">arrayToExpressionMatrix(String[][], boolean)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#arrayToExpressionMatrix(java.lang.String%5B%5D%5B%5D,boolean,java.lang.String)" class="member-name-link">arrayToExpressionMatrix(String[][], boolean, String)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#arrayToExpressionMatrixList(java.lang.String%5B%5D%5B%5D)" class="member-name-link">arrayToExpressionMatrixList(String[][])</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html" class="type-name-link" title="class in bacnet.utils">ArrayUtils</a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>
<div class="block">Public class with general array manipulation utilities, not specific to PROJECTNAME.</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#%3Cinit%3E()" class="member-name-link">ArrayUtils()</a> - Constructor for class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#ARROW_SIZE" class="member-name-link">ARROW_SIZE</a> - Static variable in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Size of the arrow drawn in each rectangle to indicate strand</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html#ASN" class="member-name-link">ASN</a> - Enum constant in enum bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html" title="enum in bacnet.scripts.blast">BlastOutput.BlastOutputTYPE</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html#ASN_Bin" class="member-name-link">ASN_Bin</a> - Enum constant in enum bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html" title="enum in bacnet.scripts.blast">BlastOutput.BlastOutputTYPE</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html#ASN_TxT" class="member-name-link">ASN_TxT</a> - Enum constant in enum bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html" title="enum in bacnet.scripts.blast">BlastOutput.BlastOutputTYPE</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.SeqType.html#ASrna" class="member-name-link">ASrna</a> - Enum constant in enum bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.TypeSrna.html#ASrna" class="member-name-link">ASrna</a> - Enum constant in enum bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.TypeSrna.html" title="enum in bacnet.datamodel.sequence">Srna.TypeSrna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/PrettyXMLWriter.html#attribute(java.lang.String,java.lang.String)" class="member-name-link">attribute(String, String)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLWriter.html#attribute(java.lang.String,java.lang.String)" class="member-name-link">attribute(String, String)</a> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Add an un-qualified attribute to an element.</div>
</dd>
<dt><a href="../org/biojava3/core/util/PrettyXMLWriter.html#attribute(java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">attribute(String, String, String)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLWriter.html#attribute(java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">attribute(String, String, String)</a> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Add an attribute to an element.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#attributes()" class="member-name-link">attributes()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Get the string of key/value attributes.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#attributeValues(java.lang.String)" class="member-name-link">attributeValues(String)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a collection of the unique values for the specified key.</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#AUTO_BOTH" class="member-name-link">AUTO_BOTH</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#AUTO_GENEXPR" class="member-name-link">AUTO_GENEXPR</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#AUTO_TILING" class="member-name-link">AUTO_TILING</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/UncompressInputStream.html#available()" class="member-name-link">available()</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
