<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>C-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="index: C">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:C">C</h2>
<dl class="index">
<dt><a href="../bacnet/utils/MathUtils.html#C_nk(int,int)" class="member-name-link">C_nk(int, int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/MathUtils.html" title="class in bacnet.utils">MathUtils</a></dt>
<dd>
<div class="block">Calculate the combination C_nk</div>
</dd>
<dt><a href="../org/biojava3/core/util/InputStreamProvider.html#CACHE_PROPERTY" class="member-name-link">CACHE_PROPERTY</a> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/InputStreamProvider.html" title="class in org.biojava3.core.util">InputStreamProvider</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/ComparisonsCreation.html#calc(bacnet.scripts.core.Comparison)" class="member-name-link">calc(Comparison)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Run the comparison by calculating statistical parameters given in tests ArrayList</div>
</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#calcEnergy(java.lang.String)" class="member-name-link">calcEnergy(String)</a> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>
<div class="block">From a bunch of stochast folding, generate a list of Free Energies</div>
</dd>
<dt><a href="../bacnet/utils/UNAfold.html#calcEnergy(java.lang.String)" class="member-name-link">calcEnergy(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>
<div class="block">From a bunch of stochast folding, generate a list of Free Energies</div>
</dd>
<dt><a href="../bacnet/scripts/database/ComparisonsCreation.html#calcFilterAndSave(java.util.ArrayList,java.lang.String,java.lang.String,java.util.ArrayList,boolean,boolean)" class="member-name-link">calcFilterAndSave(ArrayList&lt;String&gt;, String, String, ArrayList&lt;Filter&gt;, boolean, boolean)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Calculate a comparison, filter the results, and save ExpressionMatrix and list of elements
 up-regulated and down-regulated</div>
</dd>
<dt><a href="../bacnet/scripts/database/DNASeqCreation.html#calcGeneCNV()" class="member-name-link">calcGeneCNV()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DNASeqCreation.html" title="class in bacnet.scripts.database">DNASeqCreation</a></dt>
<dd>
<div class="block">For each RNAseq and Sequence element, calculate the mean value of copy number variation THIS NEED
 TO BE RUN AFTER CONVERSION of RNASeq files</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/TilingGeneExprDataBase.html#calcGenExprTilingComparisons(bacnet.datamodel.expdesign.Experiment)" class="member-name-link">calcGenExprTilingComparisons(Experiment)</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TilingGeneExprDataBase.html" title="class in bacnet.scripts.listeriomics">TilingGeneExprDataBase</a></dt>
<dd>
<div class="block">Calculate GeneExpr and Tiling comparisons files</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/LPE.html#calcP(double%5B%5D,double%5B%5D,org.apache.commons.math.analysis.polynomials.PolynomialSplineFunction,org.apache.commons.math.analysis.polynomials.PolynomialSplineFunction)" class="member-name-link">calcP(double[], double[], PolynomialSplineFunction, PolynomialSplineFunction)</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/LPE.html" title="class in bacnet.scripts.core.stat">LPE</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/CodonUsage.html#calculate(java.lang.String)" class="member-name-link">calculate(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/CodonUsage.html" title="class in bacnet.datamodel.annotation">CodonUsage</a></dt>
<dd>
<div class="block">Calculate Codon usage of a sequence<br>
 Providing <code>codonTable</code> has been loaded first</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/Febit.html#calculateComparison(java.lang.String,java.lang.String)" class="member-name-link">calculateComparison(String, String)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Febit.html" title="class in bacnet.scripts.listeriomics.technology">Febit</a></dt>
<dd>
<div class="block">Calculate Log of Fold Change (mean difference) for each value, and p-value</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/LT.html#calculateComparison(java.lang.String,java.lang.String)" class="member-name-link">calculateComparison(String, String)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/LT.html" title="class in bacnet.scripts.listeriomics.technology">LT</a></dt>
<dd>
<div class="block">Calculate Log of Fold Change (mean difference) for each value, and p-value</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/Macroarray.html#calculateComparison(java.lang.String,java.lang.String)" class="member-name-link">calculateComparison(String, String)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Macroarray.html" title="class in bacnet.scripts.listeriomics.technology">Macroarray</a></dt>
<dd>
<div class="block">Calculate Log of Fold Change (mean difference) for each value, and p-value</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/NimbleGen.html#calculateComparison(java.lang.String,java.lang.String)" class="member-name-link">calculateComparison(String, String)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/NimbleGen.html" title="class in bacnet.scripts.listeriomics.technology">NimbleGen</a></dt>
<dd>
<div class="block">Calculate Log of Fold Change (mean difference) for each value, and p-value</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/Agilent.html#calculateComparison(java.lang.String,java.lang.String,boolean)" class="member-name-link">calculateComparison(String, String, boolean)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Agilent.html" title="class in bacnet.scripts.listeriomics.technology">Agilent</a></dt>
<dd>
<div class="block">Calculate Log of Fold Change (mean difference) for each value, and p-value</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/UGiessen.html#calculateComparison(java.lang.String,java.lang.String,boolean)" class="member-name-link">calculateComparison(String, String, boolean)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/UGiessen.html" title="class in bacnet.scripts.listeriomics.technology">UGiessen</a></dt>
<dd>
<div class="block">Calculate Log of Fold Change (mean difference) for each value, and p-value</div>
</dd>
<dt><a href="../bacnet/scripts/database/ComparisonsCreation.html#calculateCompMatrix(java.util.ArrayList,java.util.ArrayList,boolean,boolean)" class="member-name-link">calculateCompMatrix(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, boolean, boolean)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Calculate a list of comparisons</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#calculateIdentities()" class="member-name-link">calculateIdentities()</a> - Method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#calculateMiddle()" class="member-name-link">calculateMiddle()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/SDProfile.html#calculateOneSDProfile(bacnet.datamodel.sequence.Genome,int,int,org.biojava3.core.sequence.Strand,java.lang.String)" class="member-name-link">calculateOneSDProfile(Genome, int, int, Strand, String)</a> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/SDProfile.html#calculateSDProfile(bacnet.datamodel.sequence.Genome)" class="member-name-link">calculateSDProfile(Genome)</a> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#calculateSize()" class="member-name-link">calculateSize()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/StartCodonProfile.html#calculateStartCodonProfile(bacnet.datamodel.sequence.Genome,java.lang.String,int)" class="member-name-link">calculateStartCodonProfile(Genome, String, int)</a> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/StartCodonProfile.html" title="class in bacnet.scripts.genome">StartCodonProfile</a></dt>
<dd>
<div class="block">Create an NGS data where:<br>
 Start codon represent a coverage of 10
 Stop codon represent a coverage of 5
 every other codon a coverage of 0</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/LPE.html#calcVarM(bacnet.datamodel.dataset.ExpressionMatrix,double%5B%5D)" class="member-name-link">calcVarM(ExpressionMatrix, double[])</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/LPE.html" title="class in bacnet.scripts.core.stat">LPE</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/LPE.html#calcZ(double%5B%5D,double%5B%5D,org.apache.commons.math.analysis.polynomials.PolynomialSplineFunction,org.apache.commons.math.analysis.polynomials.PolynomialSplineFunction)" class="member-name-link">calcZ(double[], double[], PolynomialSplineFunction, PolynomialSplineFunction)</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/LPE.html" title="class in bacnet.scripts.core.stat">LPE</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#calcZoomNumber()" class="member-name-link">calcZoomNumber()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#calcZoomRatio()" class="member-name-link">calcZoomRatio()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>
<div class="block">We have MINMIAL_BP_NB * zoomRatio^zoomNumber = totalBP <br>
 So: zoomRatio = (totalBP/MIN_NB_BP)^(1/zoomNumber)</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" class="type-name-link" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a> - Class in <a href="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></dt>
<dd>
<div class="block">Attempts to wrap compounds so it is possible to view them in a case insensitive manner</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#%3Cinit%3E(org.biojava3.core.sequence.compound.NucleotideCompound)" class="member-name-link">CaseInsensitiveCompound(NucleotideCompound)</a> - Constructor for class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#%3Cinit%3E(org.biojava3.core.sequence.compound.NucleotideCompound,org.biojava3.core.sequence.compound.NucleotideCompound,org.biojava3.core.sequence.compound.NucleotideCompound)" class="member-name-link">CaseInsensitiveTriplet(NucleotideCompound, NucleotideCompound, NucleotideCompound)</a> - Constructor for class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a> - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">A sequence creator which preserves the case of its input string in the user collection of the
 returned ProteinSequence.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">CasePreservingProteinSequenceCreator(CompoundSet&lt;AminoAcidCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/CDSComparator.html" class="type-name-link" title="class in org.biojava3.core.sequence">CDSComparator</a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/CDSComparator.html#%3Cinit%3E()" class="member-name-link">CDSComparator()</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSComparator.html" title="class in org.biojava3.core.sequence">CDSComparator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/CDSSequence.html" class="type-name-link" title="class in org.biojava3.core.sequence">CDSSequence</a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">Represents a exon or coding sequence in a gene.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/CDSSequence.html#%3Cinit%3E(org.biojava3.core.sequence.TranscriptSequence,int,int,int)" class="member-name-link">CDSSequence(TranscriptSequence, int, int, int)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#centerColumn(double%5B%5D%5B%5D)" class="member-name-link">centerColumn(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Center all the column from the Matrix</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#centerRow(double%5B%5D%5B%5D)" class="member-name-link">centerRow(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Center all the row from the Matrix</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#checksum(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">checksum(Sequence&lt;C&gt;)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Performs a simple CRC64 checksum on any given sequence.</div>
</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#CHROMO" class="member-name-link">CHROMO</a> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html" class="type-name-link" title="class in bacnet.datamodel.sequence">Chromosome</a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#%3Cinit%3E()" class="member-name-link">Chromosome()</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#%3Cinit%3E(java.lang.String)" class="member-name-link">Chromosome(String)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">Chromosome(String, CompoundSet&lt;NucleotideCompound&gt;)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader)" class="member-name-link">Chromosome(SequenceReader&lt;NucleotideCompound&gt;)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">Chromosome(SequenceReader&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.DNAType.html#CHROMOSOME" class="member-name-link">CHROMOSOME</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" class="type-name-link" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#%3Cinit%3E()" class="member-name-link">ChromosomeBacteriaSequence()</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#%3Cinit%3E(java.lang.String)" class="member-name-link">ChromosomeBacteriaSequence(String)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ChromosomeBacteriaSequence(String, CompoundSet&lt;NucleotideCompound&gt;)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader)" class="member-name-link">ChromosomeBacteriaSequence(SequenceReader&lt;NucleotideCompound&gt;)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ChromosomeBacteriaSequence(SequenceReader&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/ChromosomeSequence.html" class="type-name-link" title="class in org.biojava3.core.sequence">ChromosomeSequence</a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#%3Cinit%3E()" class="member-name-link">ChromosomeSequence()</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>
<div class="block">Empty constructor used by tools that need a proper Bean that allows the actual sequence data to
 be set after construction.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#%3Cinit%3E(java.lang.String)" class="member-name-link">ChromosomeSequence(String)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>
<div class="block">String is king and assume DNA</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ChromosomeSequence(String, CompoundSet&lt;NucleotideCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>
<div class="block">Allows the creation of a ChromosomeSequence using String for the sequence with a custom
 CompoundSet</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader)" class="member-name-link">ChromosomeSequence(SequenceReader&lt;NucleotideCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>
<div class="block">Fairly important constructor given the size of a ChromsomeSequence where the ProxySequenceReader
 could load from disk via RandomAccessFile so that the sequence doesn't need to be kept in memory.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ChromosomeSequence(SequenceReader&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>
<div class="block">Allows the creation of a ChromosomeSequence using a ProxyResequenceReader for the sequence with a
 custom CompoundSet</div>
</dd>
<dt><a href="../bacnet/utils/HTMLUtils.html#CIRCOS" class="member-name-link">CIRCOS</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/HTMLUtils.html" title="class in bacnet.utils">HTMLUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Network.html#CIRCOS_BACK_PATH" class="member-name-link">CIRCOS_BACK_PATH</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SynTView.html#CIRCOS_NAME" class="member-name-link">CIRCOS_NAME</a> - Static variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#CIRCOS_NAME" class="member-name-link">CIRCOS_NAME</a> - Static variable in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" class="type-name-link" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a> - Class in <a href="../bacnet/scripts/genome/package-summary.html">bacnet.scripts.genome</a></dt>
<dd>
<div class="block">Circular Genome Panel created for displaying Small RNAs in CGView</div>
</dd>
<dt><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#%3Cinit%3E(int,int,bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">CircularGenomeJPanel(int, int, Genome, String)</a> - Constructor for class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>
<div class="block">Run circularGenome for Listeriomics Circos network graph</div>
</dd>
<dt><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#%3Cinit%3E(int,int,bacnet.datamodel.sequence.Srna.TypeSrna)" class="member-name-link">CircularGenomeJPanel(int, int, Srna.TypeSrna)</a> - Constructor for class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>
<div class="block">Create the panel for displaying Srna.</div>
</dd>
<dt><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#%3Cinit%3E(int,int,bacnet.datamodel.sequenceNCBI.GenomeNCBI,java.lang.String)" class="member-name-link">CircularGenomeJPanel(int, int, GenomeNCBI, String)</a> - Constructor for class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/LocationHelper.html#circularLocation(int,int,org.biojava3.core.sequence.Strand,int)" class="member-name-link">circularLocation(int, int, Strand, int)</a> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Converts a location which defines the outer bounds of a circular location and splits it into the
 required portions.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.Tools.html#circularLocation(int,int,org.biojava3.core.sequence.Strand,int)" class="member-name-link">circularLocation(int, int, Strand, int)</a> - Static method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.Tools.html" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a></dt>
<dd>
<div class="block">Converts a location which defines the outer bounds of a circular location and splits it into the
 required portions.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.TypeSrna.html#CisReg" class="member-name-link">CisReg</a> - Enum constant in enum bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.TypeSrna.html" title="enum in bacnet.datamodel.sequence">Srna.TypeSrna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/Equals.html#classEqual(java.lang.Object,java.lang.Object)" class="member-name-link">classEqual(Object, Object)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Equals.html" title="class in org.biojava3.core.util">Equals</a></dt>
<dd>
<div class="block">This method should be called before beginning any equals methods.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html" class="type-name-link" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a> - Class in <a href="../org/biojava3/core/sequence/io/util/package-summary.html">org.biojava3.core.sequence.io.util</a></dt>
<dd>
<div class="block">This object represents a classpath resource on the local system.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html#%3Cinit%3E(java.lang.String)" class="member-name-link">ClasspathResource(String)</a> - Constructor for class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a></dt>
<dd>
<div class="block">Basic constructor only allowing you to specify where to find the file.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html#%3Cinit%3E(java.lang.String,boolean)" class="member-name-link">ClasspathResource(String, boolean)</a> - Constructor for class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a></dt>
<dd>
<div class="block">Advanced constructor which allows you to optionally pre-cache the data</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html#%3Cinit%3E(java.lang.String,boolean,boolean)" class="member-name-link">ClasspathResource(String, boolean, boolean)</a> - Constructor for class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a></dt>
<dd>
<div class="block">Advanced constructor which lets you set the preCache variable and to force the type of file we
 are decompressing.</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastDatabase.html#cleanDatabase(java.lang.String)" class="member-name-link">cleanDatabase(String)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">Remove in all genome folders a certain type of file</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#cleanDataTable()" class="member-name-link">cleanDataTable()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>
<div class="block">Method to remove the unsued lines in the data files for BUGS and MEXP technologies If those lines
 remain present it causes failures to create the curated tables with the global method</div>
</dd>
<dt><a href="../bacnet/utils/FileUtils.html#cleanStringFromHex(java.lang.String)" class="member-name-link">cleanStringFromHex(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Search hexadecimal code in a String indicated by %<br>
 Replace them by they ASCII representation</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#cleanSVGFile()" class="member-name-link">cleanSVGFile()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>
<div class="block">Clean SVG file to be able to replace strain name by homolog information<br>
 Names of strain are created in many different "Text table" in the SVG file which have to be
 removed</div>
</dd>
<dt><a href="../bacnet/Database.html#cleanUpDatabase()" class="member-name-link">cleanUpDatabase()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Cleanup database when closing the website</div>
</dd>
<dt><a href="../bacnet/scripts/genome/SDProfile.html#cleanUpTempFiles()" class="member-name-link">cleanUpTempFiles()</a> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/FlatFileCache.html#clear()" class="member-name-link">clear()</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/FlatFileCache.html" title="class in org.biojava3.core.util">FlatFileCache</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/SoftHashMap.html#clear()" class="member-name-link">clear()</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#clearChromosome()" class="member-name-link">clearChromosome()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>
<div class="block">Void a Chromosome when Website is quit USEFUL in Eclipse RAP</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#clearGenome()" class="member-name-link">clearGenome()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Clear current genome from JVM by setting it up to a 4 base genome</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#clickedElement(org.eclipse.swt.events.MouseEvent)" class="member-name-link">clickedElement(MouseEvent)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">Get base-pair position of the clicked element</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#clickedElement(org.eclipse.swt.events.MouseEvent)" class="member-name-link">clickedElement(MouseEvent)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#clone()" class="member-name-link">clone()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#clone()" class="member-name-link">clone()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>
<div class="block">Clone the object</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#clone()" class="member-name-link">clone()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>
<div class="block">Clone the object</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#clone()" class="member-name-link">clone()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperList.html#clone()" class="member-name-link">clone()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#clone()" class="member-name-link">clone()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCPList.html#clone()" class="member-name-link">clone()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#clone()" class="member-name-link">clone()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#clone()" class="member-name-link">clone()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#clone(double%5B%5D%5B%5D)" class="member-name-link">clone(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#clone(java.lang.String%5B%5D%5B%5D)" class="member-name-link">clone(String[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#clone(java.util.ArrayList)" class="member-name-link">clone(ArrayList&lt;String&gt;)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#clone(java.util.TreeMap)" class="member-name-link">clone(TreeMap&lt;String, Integer&gt;)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimplePoint.html#clonePoint()" class="member-name-link">clonePoint()</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Point.html#clonePoint()" class="member-name-link">clonePoint()</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a></dt>
<dd>
<div class="block">Returns a copy of this point</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#close()" class="member-name-link">close()</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/PrettyXMLWriter.html#close()" class="member-name-link">close()</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLWriter.html#close()" class="member-name-link">close()</a> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Close this XMLWriter, and it's underlying stream.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#close(java.io.Closeable)" class="member-name-link">close(Closeable)</a> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Closes any Object which implements the interface <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Closeable.html" title="class or interface in java.io" class="external-link"><code>Closeable</code></a> and sending any error to the
 logger but not forcing any explicit catching of stream errors.</div>
</dd>
<dt><a href="../bacnet/e4/rap/SessionControl.html#closeGenomeViewer(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">closeGenomeViewer(EPartService)</a> - Static method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SessionControl.html" title="class in bacnet.e4.rap">SessionControl</a></dt>
<dd>
<div class="block">Test function for closing a Genomeviewer and check if it is kept in JVM</div>
</dd>
<dt><a href="../org/biojava3/core/util/PrettyXMLWriter.html#closeTag(java.lang.String)" class="member-name-link">closeTag(String)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLWriter.html#closeTag(java.lang.String)" class="member-name-link">closeTag(String)</a> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Closes an un-qualified element.</div>
</dd>
<dt><a href="../org/biojava3/core/util/PrettyXMLWriter.html#closeTag(java.lang.String,java.lang.String)" class="member-name-link">closeTag(String, String)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLWriter.html#closeTag(java.lang.String,java.lang.String)" class="member-name-link">closeTag(String, String)</a> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Closes an element</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#Clostridiales_NWK" class="member-name-link">Clostridiales_NWK</a> - Static variable in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html#CLUSTALW" class="member-name-link">CLUSTALW</a> - Enum constant in enum org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html" title="enum in org.biojava3.core.sequence.template">LightweightProfile.StringFormat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html#CM" class="member-name-link">CM</a> - Enum constant in enum bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/CMD.html" class="type-name-link" title="class in bacnet.utils">CMD</a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/CMD.html#%3Cinit%3E()" class="member-name-link">CMD()</a> - Constructor for class bacnet.utils.<a href="../bacnet/utils/CMD.html" title="class in bacnet.utils">CMD</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Codon.html" class="type-name-link" title="class in bacnet.datamodel.sequence">Codon</a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>
<div class="block">Class for managing triplet of nucleotides.<br>
 Allo&lt;s detection of start and stop codons in nucleotide sequences</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Codon.html#%3Cinit%3E()" class="member-name-link">Codon()</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#%3Cinit%3E(org.biojava3.core.sequence.transcription.Table.CaseInsensitiveTriplet)" class="member-name-link">Codon(Table.CaseInsensitiveTriplet)</a> - Constructor for class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#%3Cinit%3E(org.biojava3.core.sequence.transcription.Table.CaseInsensitiveTriplet,org.biojava3.core.sequence.compound.AminoAcidCompound,boolean,boolean)" class="member-name-link">Codon(Table.CaseInsensitiveTriplet, AminoAcidCompound, boolean, boolean)</a> - Constructor for class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html" class="type-name-link" title="class in org.biojava3.core.sequence.compound">CodonCompound</a> - Class in <a href="../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</a></dt>
<dd>
<div class="block">Define a codon</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#%3Cinit%3E(org.biojava3.core.sequence.compound.NucleotideCompound,org.biojava3.core.sequence.compound.NucleotideCompound,org.biojava3.core.sequence.compound.NucleotideCompound,boolean)" class="member-name-link">CodonCompound(NucleotideCompound, NucleotideCompound, NucleotideCompound, boolean)</a> - Constructor for class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#codonDisplay(java.lang.String)" class="member-name-link">codonDisplay(String)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Read a sequence and separate the codon by a space <br>
 This method return teh sequence as a StringColor, so codon might be colorized after</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/CodonUsage.html" class="type-name-link" title="class in bacnet.datamodel.annotation">CodonUsage</a> - Class in <a href="../bacnet/datamodel/annotation/package-summary.html">bacnet.datamodel.annotation</a></dt>
<dd>
<div class="block">Tools for calculating codon Usage of amino acid sequences<br>
 Codon Usage tables came from : http://www.kazusa.or.jp/codon/</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/CodonUsage.html#%3Cinit%3E(java.lang.String)" class="member-name-link">CodonUsage(String)</a> - Constructor for class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/CodonUsage.html" title="class in bacnet.datamodel.annotation">CodonUsage</a></dt>
<dd>
<div class="block">Providing the name of codon Table, it loads the Table and create <code>CodonUsage</code></div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#codonUsageEGDe()" class="member-name-link">codonUsageEGDe()</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html" class="type-name-link" title="class in bacnet.views">CoExprNetworkView</a> - Class in <a href="../bacnet/views/package-summary.html">bacnet.views</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#%3Cinit%3E()" class="member-name-link">CoExprNetworkView()</a> - Constructor for class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.InitNetworkThread.html" class="type-name-link" title="class in bacnet.views">CoExprNetworkView.InitNetworkThread</a> - Class in <a href="../bacnet/views/package-summary.html">bacnet.views</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/COGannotation.html" class="type-name-link" title="class in bacnet.datamodel.annotation">COGannotation</a> - Class in <a href="../bacnet/datamodel/annotation/package-summary.html">bacnet.datamodel.annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/COGannotation.html#%3Cinit%3E()" class="member-name-link">COGannotation()</a> - Constructor for class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/COGannotation.html#COGMAP" class="member-name-link">COGMAP</a> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#COL" class="member-name-link">COL</a> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#COL_NUMBER" class="member-name-link">COL_NUMBER</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/GenomesCreation.html#COLNAME_REFSEQ" class="member-name-link">COLNAME_REFSEQ</a> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Name fo the column in which RefSeq FTP url is found</div>
</dd>
<dt><a href="../bacnet/utils/StringColor.html#colorBackground" class="member-name-link">colorBackground</a> - Variable in class bacnet.utils.<a href="../bacnet/utils/StringColor.html" title="class in bacnet.utils">StringColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/StringColor.html#colorForeground" class="member-name-link">colorForeground</a> - Variable in class bacnet.utils.<a href="../bacnet/utils/StringColor.html" title="class in bacnet.utils">StringColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/CodonUsage.html#colorizeSequence(java.lang.String)" class="member-name-link">colorizeSequence(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/CodonUsage.html" title="class in bacnet.datamodel.annotation">CodonUsage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html" class="type-name-link" title="class in bacnet.table.core">ColorMapper</a> - Class in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell)" class="member-name-link">ColorMapper(Shell)</a> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.TypeMapper.html" class="type-name-link" title="enum in bacnet.table.core">ColorMapper.TypeMapper</a> - Enum in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperList.html" class="type-name-link" title="class in bacnet.table.core">ColorMapperList</a> - Class in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperList.html#%3Cinit%3E()" class="member-name-link">ColorMapperList()</a> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperList.html#%3Cinit%3E(bacnet.datamodel.dataset.ExpressionMatrix,org.eclipse.swt.widgets.Shell)" class="member-name-link">ColorMapperList(ExpressionMatrix, Shell)</a> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>
<div class="block">Create a list of ColorMapper from a matrix<br>
 Each column name corresponding to a TypeMapper will be associated to a ColorMapper</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html" class="type-name-link" title="class in bacnet.table.core">ColorMapperRCP</a> - Class in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell)" class="member-name-link">ColorMapperRCP(Shell)</a> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.Type.html" class="type-name-link" title="enum in bacnet.table.core">ColorMapperRCP.Type</a> - Enum in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCPList.html" class="type-name-link" title="class in bacnet.table.core">ColorMapperRCPList</a> - Class in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCPList.html#%3Cinit%3E()" class="member-name-link">ColorMapperRCPList()</a> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCPList.html#%3Cinit%3E(bacnet.datamodel.dataset.ExpressionMatrix,org.eclipse.swt.widgets.Shell)" class="member-name-link">ColorMapperRCPList(ExpressionMatrix, Shell)</a> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizard.html" class="type-name-link" title="class in bacnet.table.gui">ColorMapperWizard</a> - Class in <a href="../bacnet/table/gui/package-summary.html">bacnet.table.gui</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizard.html#%3Cinit%3E(bacnet.table.core.ColorMapperList,org.eclipse.swt.widgets.Shell)" class="member-name-link">ColorMapperWizard(ColorMapperList, Shell)</a> - Constructor for class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizard.html" title="class in bacnet.table.gui">ColorMapperWizard</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizardPage.html" class="type-name-link" title="class in bacnet.table.gui">ColorMapperWizardPage</a> - Class in <a href="../bacnet/table/gui/package-summary.html">bacnet.table.gui</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizardPage.html#%3Cinit%3E(bacnet.table.core.ColorMapper,org.eclipse.swt.widgets.Shell)" class="member-name-link">ColorMapperWizardPage(ColorMapper, Shell)</a> - Constructor for class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizardPage.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell)" class="member-name-link">ColorMapperWizardPage(Shell)</a> - Constructor for class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></dt>
<dd>
<div class="block">Create the wizard.</div>
</dd>
<dt><a href="../bacnet/table/core/ColorSWT.html" class="type-name-link" title="class in bacnet.table.core">ColorSWT</a> - Class in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorSWT.html#%3Cinit%3E(int,int,int)" class="member-name-link">ColorSWT(int, int, int)</a> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorSWT.html#%3Cinit%3E(org.eclipse.swt.graphics.Color)" class="member-name-link">ColorSWT(Color)</a> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html" class="type-name-link" title="class in bacnet.scripts.core.vennDiagram">CombinationGenerator</a> - Class in <a href="../bacnet/scripts/core/vennDiagram/package-summary.html">bacnet.scripts.core.vennDiagram</a></dt>
<dd>
<div class="block">Class for finding every combination of a set of integer</div>
</dd>
<dt><a href="../bacnet/utils/CombinationGenerator.html" class="type-name-link" title="class in bacnet.utils">CombinationGenerator</a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html#%3Cinit%3E(int,int)" class="member-name-link">CombinationGenerator(int, int)</a> - Constructor for class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html" title="class in bacnet.scripts.core.vennDiagram">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/CombinationGenerator.html#%3Cinit%3E(int,int)" class="member-name-link">CombinationGenerator(int, int)</a> - Constructor for class bacnet.utils.<a href="../bacnet/utils/CombinationGenerator.html" title="class in bacnet.utils">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#combinator(java.util.ArrayList,int)" class="member-name-link">combinator(ArrayList&lt;String&gt;, int)</a> - Static method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>
<div class="block">Create all combinations of k elements of list1 <br></div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/UGiessen.html#combineMatrices(java.lang.String,java.lang.String)" class="member-name-link">combineMatrices(String, String)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/UGiessen.html" title="class in bacnet.scripts.listeriomics.technology">UGiessen</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#COMBO" class="member-name-link">COMBO</a> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html#COMMA" class="member-name-link">COMMA</a> - Enum constant in enum bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html" title="enum in bacnet.scripts.blast">BlastOutput.BlastOutputTYPE</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#compare(bacnet.datamodel.dataset.ExpressionData,boolean)" class="member-name-link">compare(ExpressionData, boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Create Tiling data from the comparison of this one and data2</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#compare(bacnet.datamodel.dataset.GeneExpression,boolean)" class="member-name-link">compare(GeneExpression, boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Create GeneExpression data from the comparison of this one and data2</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#compare(bacnet.datamodel.dataset.NGS,boolean)" class="member-name-link">compare(NGS, boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>
<div class="block">Create RNASeq data from the comparison of this one and data2</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#compare(bacnet.datamodel.dataset.Tiling,boolean)" class="member-name-link">compare(Tiling, boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Create Tiling data from the comparison of this one and data2</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#compare(bacnet.datamodel.expdesign.BioCondition,boolean)" class="member-name-link">compare(BioCondition, boolean)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Create a BioCondition containing the different data for comparing this BioCondition to bioCond2</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.BlastResultCompare.html#compare(bacnet.scripts.blast.BlastResult,bacnet.scripts.blast.BlastResult)" class="member-name-link">compare(BlastResult, BlastResult)</a> - Method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.BlastResultCompare.html" title="class in bacnet.scripts.blast">BlastResult.BlastResultCompare</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/CDSComparator.html#compare(org.biojava3.core.sequence.CDSSequence,org.biojava3.core.sequence.CDSSequence)" class="member-name-link">compare(CDSSequence, CDSSequence)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSComparator.html" title="class in org.biojava3.core.sequence">CDSComparator</a></dt>
<dd>
<div class="block">Used to sort two CDSSequences where Negative Strand makes it tough</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/ExonComparator.html#compare(org.biojava3.core.sequence.ExonSequence,org.biojava3.core.sequence.ExonSequence)" class="member-name-link">compare(ExonSequence, ExonSequence)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ExonComparator.html" title="class in org.biojava3.core.sequence">ExonComparator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceComparator.html#compare(org.biojava3.core.sequence.template.AbstractSequence,org.biojava3.core.sequence.template.AbstractSequence)" class="member-name-link">compare(AbstractSequence&lt;?&gt;, AbstractSequence&lt;?&gt;)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceComparator.html" title="class in org.biojava3.core.sequence">SequenceComparator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/BioConditionComparator.html#compare(org.eclipse.jface.viewers.Viewer,java.lang.Object,java.lang.Object)" class="member-name-link">compare(Viewer, Object, Object)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/TableViewerComparator.html#compare(org.eclipse.jface.viewers.Viewer,java.lang.Object,java.lang.Object)" class="member-name-link">compare(Viewer, Object, Object)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/TableViewerComparator.html" title="class in bacnet.table.core">TableViewerComparator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermUtils.html#compareNTermData(java.lang.String,java.lang.String)" class="member-name-link">compareNTermData(String, String)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>
<div class="block">Return the list of NTerm common to both experiment<br>
 NTerm are equal if:<br>
 same peptide sequence
 same modification</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#compareTo(org.biojava3.core.sequence.location.template.Point)" class="member-name-link">compareTo(Point)</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimplePoint.html#compareTo(org.biojava3.core.sequence.location.template.Point)" class="member-name-link">compareTo(Point)</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html" class="type-name-link" title="class in bacnet.scripts.core">Comparison</a> - Class in <a href="../bacnet/scripts/core/package-summary.html">bacnet.scripts.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#%3Cinit%3E()" class="member-name-link">Comparison()</a> - Constructor for class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#%3Cinit%3E(java.util.ArrayList,java.util.ArrayList,java.util.ArrayList,boolean)" class="member-name-link">Comparison(ArrayList&lt;StatTest&gt;, ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, boolean)</a> - Constructor for class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>
<div class="block">Create a Comparison object used in Statistical comparison case</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" class="type-name-link" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a> - Class in <a href="../bacnet/expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a></dt>
<dd>
<div class="block">A ComparisonAtlas is related to a Comparison object and gives the list of genome elements over,
 under and not differentially expressed</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#%3Cinit%3E()" class="member-name-link">ComparisonAtlas()</a> - Constructor for class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#%3Cinit%3E(bacnet.datamodel.dataset.ExpressionMatrix,java.util.ArrayList,java.lang.String,bacnet.utils.Filter)" class="member-name-link">ComparisonAtlas(ExpressionMatrix, ArrayList&lt;String&gt;, String, Filter)</a> - Constructor for class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>
<div class="block">For a specific biological condition comparison update the list of genome elements in which the
 genome element is over-expressed, under-expressed or not differently expressed<br>
 The update will be dependent of the value of the input filter = cutoff for LogFC and p-value</div>
</dd>
<dt><a href="../bacnet/scripts/database/ComparisonsCreation.html" class="type-name-link" title="class in bacnet.scripts.database">ComparisonsCreation</a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/ComparisonsCreation.html#%3Cinit%3E()" class="member-name-link">ComparisonsCreation()</a> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/SubCellCompartment.html#COMPARTMENT_NAMES" class="member-name-link">COMPARTMENT_NAMES</a> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.html" title="class in bacnet.datamodel.annotation">SubCellCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/ComplementCompound.html" class="type-name-link" title="interface in org.biojava3.core.sequence.template">ComplementCompound</a> - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html" class="type-name-link" title="class in org.biojava3.core.sequence.views">ComplementSequenceView</a>&lt;<a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html" title="type parameter in ComplementSequenceView">C</a> extends <a href="../org/biojava3/core/sequence/template/ComplementCompound.html" title="interface in org.biojava3.core.sequence.template">ComplementCompound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</a></dt>
<dd>
<div class="block">For a given sequence this class will create a view over the top of it and for every request the
 code will return the complement of the underlying base e.g. base A will become base T</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">ComplementSequenceView(Sequence&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava3.core.sequence.views">ComplementSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/LocationHelper.html#completeCircularPasses(int,int)" class="member-name-link">completeCircularPasses(int, int)</a> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Works in a similar way to modulateCircularLocation but returns the number of complete passes over
 a Sequence length a circular location makes i.e. if we have a sequence of length 10 and the
 location 3..52 we make 4 complete passes through the genome to go from position 3 to position 52.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Location.Tools.html#completeCircularPasses(int,int)" class="member-name-link">completeCircularPasses(int, int)</a> - Static method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.Tools.html" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a></dt>
<dd>
<div class="block">Works in a similar way to modulateCircularLocation but returns the number of complete passes over
 a Sequence length a circular location makes i.e. if we have a sequence of length 10 and the
 location 3..52 we make 4 complete passes through the genome to go from position 3 to position 52.</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/CompositeNTermInfo.html" class="type-name-link" title="class in bacnet.genomeBrowser">CompositeNTermInfo</a> - Class in <a href="../bacnet/genomeBrowser/package-summary.html">bacnet.genomeBrowser</a></dt>
<dd>
<div class="block">NTerm composite which is displayed on the right of the GenomeViewer.</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/CompositeNTermInfo.html#%3Cinit%3E(org.eclipse.swt.widgets.Composite,int)" class="member-name-link">CompositeNTermInfo(Composite, int)</a> - Constructor for class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CompositeNTermInfo.html" title="class in bacnet.genomeBrowser">CompositeNTermInfo</a></dt>
<dd>
<div class="block">Initiate a CompositeNTermInfo</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/Compound.html" class="type-name-link" title="interface in org.biojava3.core.sequence.template">Compound</a> - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/CompoundColorization.html" class="type-name-link" title="class in bacnet.datamodel.sequence">CompoundColorization</a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/CompoundColorization.html#%3Cinit%3E()" class="member-name-link">CompoundColorization()</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/CompoundColorization.html" title="class in bacnet.datamodel.sequence">CompoundColorization</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/exceptions/CompoundNotFoundError.html" class="type-name-link" title="class in org.biojava3.core.exceptions">CompoundNotFoundError</a> - Error in <a href="../org/biojava3/core/exceptions/package-summary.html">org.biojava3.core.exceptions</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/exceptions/CompoundNotFoundError.html#%3Cinit%3E(java.lang.CharSequence)" class="member-name-link">CompoundNotFoundError(CharSequence)</a> - Constructor for error org.biojava3.core.exceptions.<a href="../org/biojava3/core/exceptions/CompoundNotFoundError.html" title="class in org.biojava3.core.exceptions">CompoundNotFoundError</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#compoundsEqual(C,C)" class="member-name-link">compoundsEqual(C, C)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#compoundsEquivalent(C,C)" class="member-name-link">compoundsEquivalent(C, C)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundSet.html#compoundsEquivalent(C,C)" class="member-name-link">compoundsEquivalent(C, C)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#compoundsEquivalent(org.biojava3.core.sequence.compound.AminoAcidCompound,org.biojava3.core.sequence.compound.AminoAcidCompound)" class="member-name-link">compoundsEquivalent(AminoAcidCompound, AminoAcidCompound)</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundSet.html" class="type-name-link" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="type parameter in CompoundSet">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#compoundToInt(org.biojava3.core.sequence.compound.NucleotideCompound)" class="member-name-link">compoundToInt(NucleotideCompound)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundTranslator.html" class="type-name-link" title="interface in org.biojava3.core.sequence.template">CompoundTranslator</a>&lt;<a href="../org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">F</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>,<wbr><a href="../org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">T</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#compute()" class="member-name-link">compute()</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#computeTotalList()" class="member-name-link">computeTotalList()</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/ConcurrencyTools.html" class="type-name-link" title="class in org.biojava3.core.util">ConcurrencyTools</a> - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>
<div class="block">Static utility to easily share a thread pool for concurrent/parallel/lazy execution.</div>
</dd>
<dt><a href="../bacnet/e4/rap/BasicApplication.html#configure(org.eclipse.rap.rwt.application.Application)" class="member-name-link">configure(Application)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BasicApplication.html" title="class in bacnet.e4.rap">BasicApplication</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/BasicApplication.html#configure(org.eclipse.rap.rwt.application.Application)" class="member-name-link">configure(Application)</a> - Method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/BasicApplication.html" title="class in bacnet.e4.rap.setup">BasicApplication</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/LocationHelper.html#consistentAccessions(java.util.List)" class="member-name-link">consistentAccessions(List&lt;Location&gt;)</a> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Scans a list of locations and returns true if all the given locations are linked to the same
 sequence.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#containProbe(java.lang.String)" class="member-name-link">containProbe(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Check if TreeMap&lt;String, Integer> probes contains a probe</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#containProteomes()" class="member-name-link">containProteomes()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#containRNASeq()" class="member-name-link">containRNASeq()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#contains(org.biojava3.genome.parsers.gff.Location)" class="member-name-link">contains(Location)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Check if this location contains the other.</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#containStopCodon(bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.sequence.Sequence,int%5B%5D)" class="member-name-link">containStopCodon(NTerm, Sequence, int[])</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>
<div class="block">Return true if it contains a StopCodon between the peptide and the overlaping sequence</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#containTranscriptomes()" class="member-name-link">containTranscriptomes()</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.DNAType.html#CONTIG" class="member-name-link">CONTIG</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#convert(bacnet.datamodel.dataset.NGS,boolean)" class="member-name-link">convert(NGS, boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>
<div class="block">Run the conversion of wigfiles</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#convert(bacnet.datamodel.expdesign.Experiment)" class="member-name-link">convert(Experiment)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Read all GeneExpr data and save in GeneExpressionStreaming format</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#convert(bacnet.datamodel.expdesign.Experiment)" class="member-name-link">convert(Experiment)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Read Tiling data in Experiment and save in TilingDataStreaming format</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#convert(bacnet.datamodel.sequence.Srna)" class="member-name-link">convert(Srna)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Convert an Srna into a DNASequence</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#convertAll()" class="member-name-link">convertAll()</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>
<div class="block">Read all <code>Signature</code> in text files, and serialize them in
 Database.getSIGNATURES()\*<br>
 Save an array with the HashMap between Signature name and ID</div>
</dd>
<dt><a href="../bacnet/scripts/database/GenomesCreation.html#convertAll()" class="member-name-link">convertAll()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Convert all genomes available in the database</div>
</dd>
<dt><a href="../bacnet/scripts/database/NGSCreation.html#convertCoverageFiles(bacnet.datamodel.expdesign.Experiment,boolean)" class="member-name-link">convertCoverageFiles(Experiment, boolean)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/NGSCreation.html" title="class in bacnet.scripts.database">NGSCreation</a></dt>
<dd>
<div class="block">Go through all RNASeq data found in the list <code>BioCondition.getRNASeqBioConditions()</code>
 and convert all of them</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#convertData(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">convertData(ExpressionMatrix)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Convert data from an ExpressionMatrix to a GeneExpression</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#convertData(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">convertData(ExpressionMatrix)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#convertDNASequenceGene(org.biojava3.core.sequence.DNASequence)" class="member-name-link">convertDNASequenceGene(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>
<div class="block">From a DNASequence make a Gene object</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#convertDNASequenceGeneral(org.biojava3.core.sequence.DNASequence)" class="member-name-link">convertDNASequenceGeneral(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>
<div class="block">From a DNASequence object, create Sequence<br>
 Mainly used in GenomeConversion methods</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#convertDNASequenceNcRNA(org.biojava3.core.sequence.DNASequence)" class="member-name-link">convertDNASequenceNcRNA(DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>
<div class="block">From a DNASequence make a Gene object</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Operon.html#convertDNASequenceOperon(java.lang.String%5B%5D,bacnet.datamodel.sequenceNCBI.GenomeNCBI)" class="member-name-link">convertDNASequenceOperon(String[], GenomeNCBI)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></dt>
<dd>
<div class="block">Take row from Annot_sup table and creat an Operon Object from it</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#convertDNASequenceOperon(java.lang.String%5B%5D,bacnet.datamodel.sequenceNCBI.GenomeNCBI)" class="member-name-link">convertDNASequenceOperon(String[], GenomeNCBI)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/FileUtils.html#convertHexToString(java.lang.String)" class="member-name-link">convertHexToString(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Convert an hexadecimal ASCII to a String</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastOutput.html#convertOuput(java.lang.String,java.lang.String,boolean,bacnet.scripts.blast.BlastOutput.BlastOutputTYPE)" class="member-name-link">convertOuput(String, String, boolean, BlastOutput.BlastOutputTYPE)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.html" title="class in bacnet.scripts.blast">BlastOutput</a></dt>
<dd>
<div class="block">Convert asn given by blastResult into out using BlastOutputType</div>
</dd>
<dt><a href="../bacnet/scripts/database/ProteomicsCreation.html#convertProteomicsData(bacnet.datamodel.expdesign.Experiment)" class="member-name-link">convertProteomicsData(Experiment)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>
<div class="block">Convert all ExpressionMatrix found for Proteomic datasets and save them in Streaming folder</div>
</dd>
<dt><a href="../org/biojava3/core/util/StringManipulationHelper.html#convertStreamToString(java.io.InputStream)" class="member-name-link">convertStreamToString(InputStream)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/StringManipulationHelper.html" title="class in org.biojava3.core.util">StringManipulationHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#convertToVennMasterDataList()" class="member-name-link">convertToVennMasterDataList()</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#convertWigFile(bacnet.datamodel.dataset.NGS,boolean)" class="member-name-link">convertWigFile(NGS, boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>
<div class="block">WIG file are common RNASeq data<br>
 Calculate first the MEDIAN value of the different replicates ad then log2 transform and save the
 NGS data Two columns: <br>
 one for the base pair position
 one for the coverage value</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#convertXtoBP(int)" class="member-name-link">convertXtoBP(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">Convert position in the display to a position in the genome<br>
 (inverse function of basepairToXimage(int bpIndex))<br>
 If MassSpec data has not been register do it</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#convertXtoBP(int)" class="member-name-link">convertXtoBP(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>
<div class="block">Convert position in the display to a position in the genome<br>
 (inverse function of basepairToXimage(int bpIndex))</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#copy(java.io.InputStream,java.io.OutputStream)" class="member-name-link">copy(InputStream, OutputStream)</a> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Moves the bytes from input to output using a 4KB byte array.</div>
</dd>
<dt><a href="../bacnet/utils/FileUtils.html#copy(java.lang.String,java.lang.String)" class="member-name-link">copy(String, String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">copy file with path=input to file with path=output</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#copyColumnAt(java.lang.String,double%5B%5D)" class="member-name-link">copyColumnAt(String, double[])</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/PhylogenomicsCreation.html#copyFastafile(java.lang.String,java.lang.String)" class="member-name-link">copyFastafile(String, String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/PhylogenomicsCreation.html" title="class in bacnet.scripts.database">PhylogenomicsCreation</a></dt>
<dd>
<div class="block">Copy All 1st chromosome fasta files in fastaFolder and rename them with
 genome name</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#copyRowAt(java.lang.String,double%5B%5D)" class="member-name-link">copyRowAt(String, double[])</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#copyTranscriptomicsTable()" class="member-name-link">copyTranscriptomicsTable()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>
<div class="block">Copy the transcriptomics curated table in the NormData>ExprMatrix folder</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#CORR" class="member-name-link">CORR</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.TypeMapper.html#CORR" class="member-name-link">CORR</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.TypeMapper.html" title="enum in bacnet.table.core">ColorMapper.TypeMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.Type.html#CORR" class="member-name-link">CORR</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.Type.html" title="enum in bacnet.table.core">ColorMapperRCP.Type</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Network.html#CORR_CUTOFF" class="member-name-link">CORR_CUTOFF</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/SDProfile.html#correctResults(java.lang.String)" class="member-name-link">correctResults(String)</a> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>
<div class="block">Data from EGDe SD profile are corrupted need to correct them<br>
 Need to inverse result of values after the 1710999 base pair<br>
 <br>
 
 IMPORTANT: Run only one time 13/01/2014</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#correspondingMassSpecData(bacnet.datamodel.proteomics.NTerm)" class="member-name-link">correspondingMassSpecData(NTerm)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>
<div class="block">Return the MassSpecData where an Nterm can be found</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#countAT(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">countAT(Sequence&lt;NucleotideCompound&gt;)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Returns the count of AT in the given sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#countCompounds(C...)" class="member-name-link">countCompounds(C...)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#countCompounds(C...)" class="member-name-link">countCompounds(C...)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#countCompounds(C...)" class="member-name-link">countCompounds(C...)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#countCompounds(C...)" class="member-name-link">countCompounds(C...)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#countCompounds(C...)" class="member-name-link">countCompounds(C...)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Counts the number of times a compound appears in this sequence store</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#countCompounds(C...)" class="member-name-link">countCompounds(C...)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#countCompounds(C...)" class="member-name-link">countCompounds(C...)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Delegates to
 <code>SequenceMixin#countCompounds(org.biojava3.core.sequence.template.Sequence, C[])</code></div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#countCompounds(C...)" class="member-name-link">countCompounds(C...)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Sequence.html#countCompounds(C...)" class="member-name-link">countCompounds(C...)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a></dt>
<dd>
<div class="block">Returns the number of times we found a compound in the Sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#countCompounds(C...)" class="member-name-link">countCompounds(C...)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#countCompounds(org.biojava3.core.sequence.template.Sequence,C...)" class="member-name-link">countCompounds(Sequence&lt;C&gt;, C...)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">For the given vargs of compounds this method counts the number of times those compounds appear in
 the given sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#countGC(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">countGC(Sequence&lt;NucleotideCompound&gt;)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Returns the count of GC in the given sequence</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermUtils.html#countRTerminal(java.lang.String)" class="member-name-link">countRTerminal(String)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>
<div class="block">Count the number of peptides with an R (arginine), and finishing with an R</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#covariance(double%5B%5D,double%5B%5D)" class="member-name-link">covariance(double[], double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Calculate the covariance of x * y</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#covarianceMatrix(double%5B%5D%5B%5D)" class="member-name-link">covarianceMatrix(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Calcul of the covariance Matrix</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html#CP" class="member-name-link">CP</a> - Enum constant in enum bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/CRC64Checksum.html" class="type-name-link" title="class in org.biojava3.core.util">CRC64Checksum</a> - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>
<div class="block">Utility class that calculates a CRC64 checksum on a stream of bytes.</div>
</dd>
<dt><a href="../org/biojava3/core/util/CRC64Checksum.html#%3Cinit%3E()" class="member-name-link">CRC64Checksum()</a> - Constructor for class org.biojava3.core.util.<a href="../org/biojava3/core/util/CRC64Checksum.html" title="class in org.biojava3.core.util">CRC64Checksum</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/BacnetE4PointFactory.html#create()" class="member-name-link">create()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BacnetE4PointFactory.html" title="class in bacnet.e4.rap">BacnetE4PointFactory</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/BacnetE4PointFactory.html#create()" class="member-name-link">create()</a> - Method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/BacnetE4PointFactory.html" title="class in bacnet.e4.rap.setup">BacnetE4PointFactory</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#createAll(bacnet.datamodel.expdesign.Experiment)" class="member-name-link">createAll(Experiment)</a> - Static method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>
<div class="block">For each comparison create a <code>Comparison</code> object containing the list of all
 differentially expressed genes, sRNAs, asRNAs, and cisRegs<br>
 <br>
 
 WANRING: OrganizeDatabase.organizeComparisonsData(exp) as to be run first!!!</div>
</dd>
<dt><a href="../bacnet/scripts/database/ComparisonsCreation.html#createAllCompMatrix(java.util.ArrayList)" class="member-name-link">createAllCompMatrix(ArrayList&lt;String&gt;)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Go through all comparisons and create for each <code>Tiling</code> and <code>GenExpression</code>
 different matrices giving the number of genes, Srna and asRNA overexpressed and underexpressed</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#createAnnotation()" class="member-name-link">createAnnotation()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Create an <code>Annotation</code> Table from the list of <code>Nterm</code></div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html#createAnnotation(bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">createAnnotation(Genome, String)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversion</a></dt>
<dd>
<div class="block">Create a Table with all elements of the Genome, which will be used for element search and display
 on the GenomeViewer</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#createArrayExpressTable()" class="member-name-link">createArrayExpressTable()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>
<div class="block">Create a new txt file containing all the accession data for every file to download</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#createArrayTable()" class="member-name-link">createArrayTable()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>
<div class="block">Create a txt file giving the name of each array technology</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#createASrnaTSStable()" class="member-name-link">createASrnaTSStable()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>
<div class="block">Run different function to curate the data given by Omri 15/11/2012</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#createBioCondition(java.lang.String)" class="member-name-link">createBioCondition(String)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">Create a BioCondition containing only one Nterm</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html#createBioConditionFigure(boolean)" class="member-name-link">createBioConditionFigure(boolean)</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html" title="class in bacnet.scripts.listeriomics">SummaryListeriomics</a></dt>
<dd>
<div class="block">Create the different element for the figure representing the BioCondition</div>
</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#createBlastCommands(java.lang.String,double)" class="member-name-link">createBlastCommands(String, double)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Launch blast commands creation</div>
</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#createBlastCommands(java.lang.String,java.lang.String,java.lang.String,java.util.ArrayList,double)" class="member-name-link">createBlastCommands(String, String, String, ArrayList&lt;String&gt;, double)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Create .bat files with the command lines for each blast.</div>
</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#createBlastDatabases(java.lang.String,java.lang.String%5B%5D,boolean,boolean)" class="member-name-link">createBlastDatabases(String, String[], boolean, boolean)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>
<div class="block">Create blast database for each genome contained in GenomeFolder
 
 !!!!!!!!!!</div>
</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#createBlastDatabases(java.lang.String,java.lang.String,java.util.ArrayList)" class="member-name-link">createBlastDatabases(String, String, ArrayList&lt;String&gt;)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Create blast database for each genome contained in genomesInput</div>
</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#createBlastDB()" class="member-name-link">createBlastDB()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>
<div class="block">Create blastP databases</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastDatabase.html#createBlastDB(java.lang.String)" class="member-name-link">createBlastDB(String)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">Create blastP databases</div>
</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#createBlastDB(java.lang.String)" class="member-name-link">createBlastDB(String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Create blastP databases</div>
</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#createBlastScript(java.lang.String)" class="member-name-link">createBlastScript(String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Creation of the general command file that will be used to create the ones for
 each blast</div>
</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#createBlastSmallORFDB()" class="member-name-link">createBlastSmallORFDB()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>
<div class="block">Create blastP databases</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastDatabase.html#createBlastSmallORFDB(java.lang.String)" class="member-name-link">createBlastSmallORFDB(String)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">Create blastP databases</div>
</dd>
<dt><a href="../bacnet/scripts/database/GenomesCreation.html#createCircularGenomeView(bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">createCircularGenomeView(Genome, String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
</dd>
<dt><a href="../bacnet/scripts/database/NetworkCreation.html#createCoExpressionNetwork(java.util.ArrayList,java.lang.String)" class="member-name-link">createCoExpressionNetwork(ArrayList&lt;String&gt;, String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/NetworkCreation.html" title="class in bacnet.scripts.database">NetworkCreation</a></dt>
<dd>
<div class="block">Create Co-Expression network for all absolute expression dataset of a specific genome</div>
</dd>
<dt><a href="../bacnet/scripts/database/NetworkCreation.html#createCoExprTable(java.lang.String)" class="member-name-link">createCoExprTable(String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/NetworkCreation.html" title="class in bacnet.scripts.database">NetworkCreation</a></dt>
<dd>
<div class="block">Create Database.getInstance().getCoExprNetworkArrayPath() with all co-expression network available <br>
 Then for each genome available create a file with list of datasets available for co-expression network creation. this will be saved in
 Database.getLISTDATA_COEXPR_NETWORK_TRANSCRIPTOMES_PATH() + "_" + genomeName + ".txt"</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/COGannotation.html#createCOGNumberEGDe()" class="member-name-link">createCOGNumberEGDe()</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></dt>
<dd>
<div class="block">Return a map between cogID and the number of genes associated to this cog in EGD-e</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#createComparisonsTable()" class="member-name-link">createComparisonsTable()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>
<div class="block">Use the BioConditions and the curated sdrf files from ArrayExpressRawData folders to create a
 Comparison table</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#createConservationTables()" class="member-name-link">createConservationTables()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Read <code>Srna.PATH_CONSERVATION</code> table and display it in HeatMapView <br>
 Headers are reorganized by their phylogeny positions</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#createConservationTables10403S()" class="member-name-link">createConservationTables10403S()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Read sRNAConservation10403S.txt table and display it in HeatMapView <br>
 Headers are reorganized by their phylogeny positions</div>
</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizardPage.html#createControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createControl(Composite)</a> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></dt>
<dd>
<div class="block">Create contents of the wizard.</div>
</dd>
<dt><a href="../bacnet/e4/rap/setup/SetupPart.html#createControls(org.eclipse.swt.widgets.Composite)" class="member-name-link">createControls(Composite)</a> - Method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SetupPart.html" title="class in bacnet.e4.rap.setup">SetupPart</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/e4/rap/TestPart.html#createControls(org.eclipse.swt.widgets.Composite)" class="member-name-link">createControls(Composite)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/TestPart.html" title="class in bacnet.e4.rap">TestPart</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/views/CRISPRPredictView.html#createControls(org.eclipse.swt.widgets.Composite)" class="member-name-link">createControls(Composite)</a> - Method in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/views/CRISPRPredictView2.html#createControls(org.eclipse.swt.widgets.Composite)" class="member-name-link">createControls(Composite)</a> - Method in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html#createDatabase()" class="member-name-link">createDatabase()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html" title="class in bacnet.scripts.listeriomics.srna">SmallPeptidesSearch</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
<div class="block">Create database file which contains all genome available <br>
 
 This genome list contains all the genomes from GenomeFolder and all the genome contained in
 GenomePhylogeny.PATH_TABLE (=All bacteria phylogeny)</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html#createDatabase()" class="member-name-link">createDatabase()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html" title="class in bacnet.scripts.listeriomics.srna">Srna10403SOliver</a></dt>
<dd>
<div class="block">Create database file which contains all genome available <br>
 It will be used by Blast afterwards<br>
 <br>
 
 BE CAREFULL THAT THESE GENOMES HAVE BEEN ESTABLISHED FOR BLAST SEARCH (=database is created
 through Blast.createBlastDatabases();)</div>
</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#createDatabaseForORF()" class="member-name-link">createDatabaseForORF()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>
<div class="block">Create a fasta file with all proteins from the different chromosomes
 Using faa file create blast
 Verify by looking at all file size</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastDatabase.html#createDatabaseForORF(java.lang.String)" class="member-name-link">createDatabaseForORF(String)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">Create a fasta file with all proteins from the different chromosomes
 Using faa file create blast
 Verify by looking at all file size</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastDatabase.html#createDatabaseForSmallORF()" class="member-name-link">createDatabaseForSmallORF()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">translate every genomes on the 6 frames
 Create a fasta file with all proteins from the different chromosomes
 Using faa file create blast
 Verify by looking at all file size</div>
</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#createDatabaseForSmallORF()" class="member-name-link">createDatabaseForSmallORF()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>
<div class="block">translate every genomes on the 6 frames
 Create a fasta file with all proteins from the different chromosomes
 Using faa file create blast
 Verify by looking at all file size</div>
</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html#createDataBases()" class="member-name-link">createDataBases()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Create:<br>
 Protein DB
 UTR DB
 fusion Protein DB and Protein DB
 Genome DB
 TIS DB</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#createDataBases()" class="member-name-link">createDataBases()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Create:<br>
 Protein DB
 UTR DB
 fusion Protein DB and Protein DB
 Genome DB
 TIS DB</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html#createDialogArea(org.eclipse.swt.widgets.Composite)" class="member-name-link">createDialogArea(Composite)</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html" title="class in bacnet.expressionAtlas.core">SelectGenomeElementDialog</a></dt>
<dd>
<div class="block">Create contents of the wizard.</div>
</dd>
<dt><a href="../bacnet/scripts/core/normalization/GRNormalization.html#createEdgeWT()" class="member-name-link">createEdgeWT()</a> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/GRNormalization.html" title="class in bacnet.scripts.core.normalization">GRNormalization</a></dt>
<dd>
<div class="block">By copying GENERAL_WT_DATA create GENERAL_WT_NAME tiling data</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/EGDeWTdata.html#createEGDeAverage()" class="member-name-link">createEGDeAverage()</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/EGDeWTdata.html" title="class in bacnet.datamodel.dataset">EGDeWTdata</a></dt>
<dd>
<div class="block">From the list of Wild Type data given by <code>EGDeWTdata.wildTypes</code> calculate the mean and
 save it</div>
</dd>
<dt><a href="../bacnet/scripts/database/GenomesCreation.html#createEGDeGenomeElementList()" class="member-name-link">createEGDeGenomeElementList()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Create a list of all Genes, all ASrnas, and all CisRegs of EGD-e<br>
 Srnas list is already created in EGDe_SRNA_PATH</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/LPE.html#createEstimates(java.util.TreeMap)" class="member-name-link">createEstimates(TreeMap&lt;String, GeneExpression&gt;)</a> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/LPE.html" title="class in bacnet.scripts.core.stat">LPE</a></dt>
<dd>
<div class="block">For each comparison calculate the LPE estimate, by creating a matrix of all expression values and
 estimates LPE using each of these matrices</div>
</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#createExpressionMatrix(bacnet.datamodel.sequence.Genome)" class="member-name-link">createExpressionMatrix(Genome)</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>
<div class="block">Combine all Transcriptomes describe in <code>Expression</code> data, and return the combine
 <code>ExpressionMatrix</code></div>
</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#createFAA(java.lang.String)" class="member-name-link">createFAA(String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Create a FAA file in the BLASTDB folder for each genome.</div>
</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#createFastaAA()" class="member-name-link">createFastaAA()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>
<div class="block">Create blastP databases</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastDatabase.html#createFastaAA(java.lang.String)" class="member-name-link">createFastaAA(String)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">Create blastP databases</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastDatabase.html#createFastaAASmallORF()" class="member-name-link">createFastaAASmallORF()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></dt>
<dd>
<div class="block">Extract result and create fasta files</div>
</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#createFastaAASmallORF()" class="member-name-link">createFastaAASmallORF()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>
<div class="block">Extract result and create fasta files</div>
</dd>
<dt><a href="../bacnet/scripts/core/normalization/GRNormalization.html#createFilecatsTable(bacnet.datamodel.expdesign.BioCondition,java.lang.String,boolean)" class="member-name-link">createFilecatsTable(BioCondition, String, boolean)</a> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/GRNormalization.html" title="class in bacnet.scripts.core.normalization">GRNormalization</a></dt>
<dd>
<div class="block">Create filecats = list of data in a table, used by the R script to run Tiling normalization</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html#createFinalList()" class="member-name-link">createFinalList()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html" title="class in bacnet.scripts.listeriomics.srna">SrnaFinalList</a></dt>
<dd>
<div class="block">Read the different :list from different publication and combine all of them into an Srna final
 list<br>
 Every table should be found in SrnaTables.PATH</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html#createFolderCopyFNA(java.lang.String,java.lang.String)" class="member-name-link">createFolderCopyFNA(String, String)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeConversion.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversion</a></dt>
<dd>
<div class="block">Create a folder for each Genome, and copy all .fna in it</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html#createFoldingFigures()" class="member-name-link">createFoldingFigures()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html" title="class in bacnet.scripts.listeriomics.srna">SrnaFinalList</a></dt>
<dd>
<div class="block">Run UNAFold for each sRNA to calculate folding<br>
 Copy each .png image from UNAFold folder to <code>Srna.PATH_SEC_STRUCTURE</code></div>
</dd>
<dt><a href="../bacnet/reader/GBKReader.html#createGenBankFile(bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">createGenBankFile(Genome, String)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/GBKReader.html" title="class in bacnet.reader">GBKReader</a></dt>
<dd>
<div class="block">Create GenBank file with first chromosome<br>
 TO BE FIXED for adding other chromosomes</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#createGeneHashMap()" class="member-name-link">createGeneHashMap()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Regroup NTerm using genes of EGD-e<br>
 Each EGD-e gene will be associated to a list of NTerm overlapping it<br>
 Fill: HashMap&lt;String, ArrayList<NTerm>>() aTISMap, HashMap&lt;String, ArrayList<NTerm>>() genes,
 HashMap&lt;String, ArrayList<TypeModif>>() geneModification</div>
</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html#createGenomeDB()" class="member-name-link">createGenomeDB()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">lists of small peptides contained in the genome of EGDe on 6frames.<br>
 The small peptides include in the list are the one only of size more than 5aa, and surrounded by
 at least one R.<br>
 example: Start - peptide - R<br>
 Stop - peptide - R<br>
 <br>
 Element like: Start - peptide - Stop are rejected from the analysis<br>
 <br>
 frame 1 = EGDe genome, ID=genome
 frame 2 = EGDe genome 1bp on the left, ID=genomeM
 frame 3 = EGDe genome 1bp on the right, ID=genomeP
 frame 4 = EGDe reverse complement genome, ID=genomeCompl
 frame 5 = EGDe reverse complement genome 1bp on the left, ID=genomeComplM
 frame 6 = EGDe reverse complement genome 1bp on the right, ID=genomeComplP <br></div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#createGenomeDB()" class="member-name-link">createGenomeDB()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">lists of small peptides contained in the genome of EGDe on 6frames.<br>
 The small peptides include in the list are the one only of size more than 5aa, and surrounded by
 at least one R.<br>
 example: Start - peptide - R<br>
 Stop - peptide - R<br>
 <br>
 Element like: Start - peptide - Stop are rejected from the analysis<br>
 <br>
 frame 1 = EGDe genome, ID=genome
 frame 2 = EGDe genome 1bp on the left, ID=genomeM
 frame 3 = EGDe genome 1bp on the right, ID=genomeP
 frame 4 = EGDe reverse complement genome, ID=genomeCompl
 frame 5 = EGDe reverse complement genome 1bp on the left, ID=genomeComplM
 frame 6 = EGDe reverse complement genome 1bp on the right, ID=genomeComplP <br></div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html#createGenomeTable()" class="member-name-link">createGenomeTable()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html" title="class in bacnet.scripts.listeriomics">SummaryListeriomics</a></dt>
<dd>
<div class="block">Create Table 1 for Listeriomics paper with the list of Genomes and different infos<br>
 Save in: Project.getPath()+"Table 1 - Genomes.txt"</div>
</dd>
<dt><a href="../bacnet/scripts/database/GenomesCreation.html#createGenomeTable(java.lang.String)" class="member-name-link">createGenomeTable(String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Create Table 1 with the list of Genomes and different infos<br>
 Save in: Project.getPath()+"Table 1 - Genomes.txt"</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#createHashMap()" class="member-name-link">createHashMap()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Regroup in different HashMap the NTerm<br>
 Fill:<br>
 HashMap&lt;String, ArrayList<NTerm>> peptides is to make a simple HashMap between NTerm sequence
 and a list of corresponding <code>NTerm</code>
 HashMap&lt;String, ArrayList<NTerm>> modifications links Nterm position to a list of
 corresponding <code>NTerm</code></div>
</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#createHomologTable(java.lang.String)" class="member-name-link">createHomologTable(String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Create all homolog tables for each genome<br>
 3 different log files are saved<br>
 TabDelimitedTableReader.saveTreeSet(proteinNotFound, GenomeNCBI.PATH_HOMOLOGS + "ProteinNotfFound.homologs.txt");<br>
 TabDelimitedTableReader.saveTreeSet(noHomologs, GenomeNCBI.PATH_HOMOLOGS + "NoHomologsFound.homologs.txt");<br>
 TabDelimitedTableReader.saveTreeSet(lowHomologs, GenomeNCBI.PATH_HOMOLOGS + "LowHomologsFound.homologs.txt");<br><br></div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#createHTMLPageSummary()" class="member-name-link">createHTMLPageSummary()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Read TABLE_PATH and create an HTML table from it to ease accession to ArrayExpress info</div>
</dd>
<dt><a href="../bacnet/scripts/database/ProteomicsCreation.html#createHTMLPageSummary()" class="member-name-link">createHTMLPageSummary()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>
<div class="block">DATA_TABLe has been created from a copy paste of ArrayExpress search results.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html#createHTMLPageSummary()" class="member-name-link">createHTMLPageSummary()</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressListeriomics</a></dt>
<dd>
<div class="block">DATA_TABLe has been created from a copy paste of ArrayExpress search results.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#createHTMLPageSummary()" class="member-name-link">createHTMLPageSummary()</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Read TABLE_PATH and create an HTML table from it to ease accession to ArrayExpress info</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html#createHTMLPageSummary()" class="member-name-link">createHTMLPageSummary()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics">ArrayExpressListeriomics</a></dt>
<dd>
<div class="block">DATA_TABLe has been created from a copy paste of ArrayExpress search results.</div>
</dd>
<dt><a href="../bacnet/table/core/Filter.html#createInferiorAbsFilter(double)" class="member-name-link">createInferiorAbsFilter(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#createInferiorAbsFilter(double)" class="member-name-link">createInferiorAbsFilter(double)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#createInferiorFilter(double)" class="member-name-link">createInferiorFilter(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#createInferiorFilter(double)" class="member-name-link">createInferiorFilter(double)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyTools.html#createInfoNodes()" class="member-name-link">createInfoNodes()</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#createIterator(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">createIterator(Sequence&lt;C&gt;)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Creates a simple sequence iterator which moves through a sequence going from 1 to the length of
 the Sequence.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermMain.html#createListeriomicsTable(java.lang.String)" class="member-name-link">createListeriomicsTable(String)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermMain.html" title="class in bacnet.scripts.listeriomics.nterm">NTermMain</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#createListeriomicsTSSTermSeq()" class="member-name-link">createListeriomicsTSSTermSeq()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>
<div class="block">Create TSS (Transcription Start Site) and TermSeq wig files</div>
</dd>
<dt><a href="../bacnet/scripts/database/ComparisonsCreation.html#createLogFCMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)" class="member-name-link">createLogFCMatrix(Experiment, Genome)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Create Database.logFCMatrix by regrouping every logFC values for genes, sRNAs, asRNAs, and
 cisRegs in every <code>BioCondition</code> given <code>Experiment</code><br></div>
</dd>
<dt><a href="../bacnet/scripts/database/TranscriptomesCreation.html#createLogFCMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)" class="member-name-link">createLogFCMatrix(Experiment, Genome)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/TranscriptomesCreation.html" title="class in bacnet.scripts.database">TranscriptomesCreation</a></dt>
<dd>
<div class="block">Create Database.logFCMatrix by regrouping every logFC values for genes, sRNAs, asRNAs, and
 cisRegs in every <code>BioCondition</code> given <code>Experiment</code><br></div>
</dd>
<dt><a href="../bacnet/scripts/database/ProteomicsCreation.html#createLogFCMatrix(bacnet.datamodel.expdesign.Experiment,java.lang.String)" class="member-name-link">createLogFCMatrix(Experiment, String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>
<div class="block">Create a table will all protein expression for each Genome</div>
</dd>
<dt><a href="../bacnet/scripts/database/TranscriptomesCreation.html#createLogfcTranscriptomeTable(java.util.ArrayList,java.lang.String,boolean)" class="member-name-link">createLogfcTranscriptomeTable(ArrayList&lt;String&gt;, String, boolean)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/TranscriptomesCreation.html" title="class in bacnet.scripts.database">TranscriptomesCreation</a></dt>
<dd>
<div class="block">Create LogFC summary matrix for transcriptomics datasets</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/JCVI.html#createMatrix(java.lang.String,java.lang.String)" class="member-name-link">createMatrix(String, String)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/JCVI.html" title="class in bacnet.scripts.listeriomics.technology">JCVI</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/ProteomicsCreation.html#createMatrixFromTISData()" class="member-name-link">createMatrixFromTISData()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>
<div class="block">Create a matrix from TIS data for which each gene has its corresponding number of spectra</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#createMatrixFromTISData()" class="member-name-link">createMatrixFromTISData()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>
<div class="block">Create a matrix from TIS data for which each gene has its corresponding number of spectra</div>
</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html#createMDB()" class="member-name-link">createMDB()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#createMDB()" class="member-name-link">createMDB()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DataFolder.html#createMissingFolders()" class="member-name-link">createMissingFolders()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataFolder.html" title="class in bacnet.scripts.database">DataFolder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/BannerView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BannerView.html" title="class in bacnet.e4.rap">BannerView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/e4/rap/InitViewBacnet.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewBacnet.html" title="class in bacnet.e4.rap">InitViewBacnet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeria.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeria.html" title="class in bacnet.e4.rap">InitViewListeria</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeriaSample.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">InitViewListeriaSample</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewYersinia.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewYersinia.html" title="class in bacnet.e4.rap">InitViewYersinia</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SynTView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/sequenceTools/SrnaSummaryView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaSummaryView.html" title="class in bacnet.sequenceTools">SrnaSummaryView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/table/TableSWTView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/views/BannerView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.views.<a href="../bacnet/views/BannerView.html" title="class in bacnet.views">BannerView</a></dt>
<dd>
<div class="block">Create contents of the view part.</div>
</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/InitView.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.views.<a href="../bacnet/views/InitView.html" title="class in bacnet.views">InitView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/InternalBrowser.html#createPartControl(org.eclipse.swt.widgets.Composite)" class="member-name-link">createPartControl(Composite)</a> - Method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/PhylogenomicsCreation.html#createPhylogenomicFigure(java.lang.String)" class="member-name-link">createPhylogenomicFigure(String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/PhylogenomicsCreation.html" title="class in bacnet.scripts.database">PhylogenomicsCreation</a></dt>
<dd>
<div class="block">Method to create a Phylogenomic figure using:<br>
 JolyTree - https://gitlab.pasteur.fr/GIPhy/JolyTree for alignment and tree
 construction<br>
 FigTree for Drawing Phylogenomic tree</div>
</dd>
<dt><a href="../bacnet/scripts/blast/PhylogenySmallORFs.html#createPhylogenyFile(java.util.ArrayList)" class="member-name-link">createPhylogenyFile(ArrayList&lt;Sequence&gt;)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/PhylogenySmallORFs.html" title="class in bacnet.scripts.blast">PhylogenySmallORFs</a></dt>
<dd>
<div class="block">reading filtered table and create fasta file, assuming that Lite.excel table have been reduced
 manually by removing doublons</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html#createPhylogenyFile(java.util.ArrayList)" class="member-name-link">createPhylogenyFile(ArrayList&lt;Sequence&gt;)</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html" title="class in bacnet.scripts.phylogeny">PhylogenySmallORFs</a></dt>
<dd>
<div class="block">reading filtered table and create fasta file, assuming that Lite.excel table have been reduced
 manually by removing doublons</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#createProbeList()" class="member-name-link">createProbeList()</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Read general GeneExpression data and extract the list of probes from it</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#createProbeList()" class="member-name-link">createProbeList()</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>
<div class="block">Read WildType tiling data and extract the list of probes from it</div>
</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html#createProteinDB()" class="member-name-link">createProteinDB()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Create a fasta file which will be used has a database<br>
 This file contains:
 all genes
 all new genes described in RAST
 all Srna on 3 frames
 all ASrna on 3 frames (300bp long) lists of Genes, ID=lmo, directly taken from NCBI lists of
 "new"Gene, ID=rastlmo, from the analysis of annotation I just did using RAST software lists of
 Srna, ID=rli, rliM, rliP, we don't know where is the begining so we have 3 frame for each rli, we
 start at the first start codon, plus we cut when we found stop codon and peptide smaller than
 5aa. lists of ASrna, ID=anti, antiM, antiP, we don't know where is the begining so we have 3
 frame for each asRNA, we start at the first start codon, plus we cut when we found stop codon and
 peptide smaller than 5aa.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#createProteinDB()" class="member-name-link">createProteinDB()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Create a fasta file which will be used has a database<br>
 This file contains:
 all genes
 all new genes described in RAST
 all Srna on 3 frames
 all ASrna on 3 frames (300bp long) lists of Genes, ID=lmo, directly taken from NCBI lists of
 "new"Gene, ID=rastlmo, from the analysis of annotation I just did using RAST software lists of
 Srna, ID=rli, rliM, rliP, we don't know where is the begining so we have 3 frame for each rli, we
 start at the first start codon, plus we cut when we found stop codon and peptide smaller than
 5aa. lists of ASrna, ID=anti, antiM, antiP, we don't know where is the begining so we have 3
 frame for each asRNA, we start at the first start codon, plus we cut when we found stop codon and
 peptide smaller than 5aa.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#createProteolyseMap()" class="member-name-link">createProteolyseMap()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Regroup peptides which overlap -> the assumption is that they come from the same protein but were
 created from proteolyse</div>
</dd>
<dt><a href="../bacnet/scripts/database/ProteomicsCreation.html#createProteomeTable(java.util.ArrayList,java.lang.String)" class="member-name-link">createProteomeTable(ArrayList&lt;String&gt;, String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>
<div class="block">Create Expression summary matrix for proteomics datasets</div>
</dd>
<dt><a href="../bacnet/scripts/database/GenomesCreation.html#createPTTandFAAFiles()" class="member-name-link">createPTTandFAAFiles()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Some genomes have been downloaded without PTT files so we create them<br>
 PTT files are used by SynTVIew software</div>
</dd>
<dt><a href="../bacnet/e4/rap/BasicApplication.html#createResourceLoader(java.lang.String)" class="member-name-link">createResourceLoader(String)</a> - Static method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BasicApplication.html" title="class in bacnet.e4.rap">BasicApplication</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">createSequence(Sequence&lt;F&gt;)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundTranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">createSequence(Sequence&lt;F&gt;)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundTranslator.html" title="interface in org.biojava3.core.sequence.template">CompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">createSequence(Sequence&lt;NucleotideCompound&gt;)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava3.core.sequence.transcription">DNAToRNATranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#createSequence(org.biojava3.core.sequence.template.Sequence,org.biojava3.core.sequence.transcription.Frame)" class="member-name-link">createSequence(Sequence&lt;NucleotideCompound&gt;, Frame)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava3.core.sequence.transcription">DNAToRNATranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">createSequences(Sequence&lt;F&gt;)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundTranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">createSequences(Sequence&lt;F&gt;)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundTranslator.html" title="interface in org.biojava3.core.sequence.template">CompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">createSequences(Sequence&lt;NucleotideCompound&gt;)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava3.core.sequence.transcription">DNAToRNATranslator</a></dt>
<dd>
<div class="block">Overloaded local version which delegates to an optional translator when told to (specified during
 construction).</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">createSequences(Sequence&lt;NucleotideCompound&gt;)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html" title="class in org.biojava3.core.sequence.transcription">RNAToAminoAcidTranslator</a></dt>
<dd>
<div class="block">Performs the core conversion of RNA to Peptide.</div>
</dd>
<dt><a href="../bacnet/scripts/database/ComparisonsCreation.html#createStatMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)" class="member-name-link">createStatMatrix(Experiment, Genome)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Create Database.statTable by calculating stat values for genes, sRNAs, and asRNAs in every
 BioCondition<br>
 For genes we use FDRBY calculated from GebneExpression array<br>
 For sRNAs and asRNAs we use TSTUDENTTILING for Tiling array</div>
</dd>
<dt><a href="../bacnet/scripts/database/TranscriptomesCreation.html#createStatMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)" class="member-name-link">createStatMatrix(Experiment, Genome)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/TranscriptomesCreation.html" title="class in bacnet.scripts.database">TranscriptomesCreation</a></dt>
<dd>
<div class="block">Create Database.statTable by calculating stat values for genes, sRNAs, and asRNAs in every
 BioCondition<br>
 For genes we use FDRBY calculated from GebneExpression array<br>
 For sRNAs and asRNAs we use TSTUDENTTILING for Tiling array</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#createStochasticMatrix(int,int)" class="member-name-link">createStochasticMatrix(int, int)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#createSubSequence(org.biojava3.core.sequence.template.Sequence,int,int)" class="member-name-link">createSubSequence(Sequence&lt;C&gt;, int, int)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Creates a simple sub sequence view delimited by the given start and end.</div>
</dd>
<dt><a href="../bacnet/scripts/database/BioConditionCreation.html#createSummaryProteomesComparisonsTable()" class="member-name-link">createSummaryProteomesComparisonsTable()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/BioConditionCreation.html" title="class in bacnet.scripts.database">BioConditionCreation</a></dt>
<dd>
<div class="block">Create a table will all main information on the different Comparison of proteomic available<br>
 This table will be loaded by BioConditionView</div>
</dd>
<dt><a href="../bacnet/scripts/database/BioConditionCreation.html#createSummaryProteomesTable()" class="member-name-link">createSummaryProteomesTable()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/BioConditionCreation.html" title="class in bacnet.scripts.database">BioConditionCreation</a></dt>
<dd>
<div class="block">Create a table will all main information on the different BioCondition available<br>
 This table will be loaded by BioConditionView</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html#createSummaryTables()" class="member-name-link">createSummaryTables()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html" title="class in bacnet.scripts.listeriomics.srna">SrnaFinalList</a></dt>
<dd>
<div class="block">Create SUmmary Tables for the website<br>
 WARNING: GenomeConversion has to be run before!</div>
</dd>
<dt><a href="../bacnet/scripts/database/BioConditionCreation.html#createSummaryTranscriptomesComparisonsTable()" class="member-name-link">createSummaryTranscriptomesComparisonsTable()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/BioConditionCreation.html" title="class in bacnet.scripts.database">BioConditionCreation</a></dt>
<dd>
<div class="block">Create a table will all main information on the different BioCondition available<br>
 This table will be loaded by BioConditionView</div>
</dd>
<dt><a href="../bacnet/scripts/database/BioConditionCreation.html#createSummaryTranscriptomesTable()" class="member-name-link">createSummaryTranscriptomesTable()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/BioConditionCreation.html" title="class in bacnet.scripts.database">BioConditionCreation</a></dt>
<dd>
<div class="block">Create a table will all main information on the different BioCondition available<br>
 This table will be loaded by BioConditionView</div>
</dd>
<dt><a href="../bacnet/table/core/Filter.html#createSuperiorAbsFilter(double)" class="member-name-link">createSuperiorAbsFilter(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#createSuperiorAbsFilter(double)" class="member-name-link">createSuperiorAbsFilter(double)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#createSuperiorFilter(double)" class="member-name-link">createSuperiorFilter(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#createSuperiorFilter(double)" class="member-name-link">createSuperiorFilter(double)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#createSuperiorFilter(double,double)" class="member-name-link">createSuperiorFilter(double, double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#createSuperiorFilter(double,double)" class="member-name-link">createSuperiorFilter(double, double)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/normalization/GENormalization.html#createTargetsTable(bacnet.datamodel.expdesign.BioCondition,java.lang.String)" class="member-name-link">createTargetsTable(BioCondition, String)</a> - Static method in class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/GENormalization.html" title="class in bacnet.scripts.core.normalization">GENormalization</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#createTechnologiesTable()" class="member-name-link">createTechnologiesTable()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#createTechnoTable()" class="member-name-link">createTechnoTable()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Create a table with different info about Array Design: Name, number of data which use it</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#createTechnoTable()" class="member-name-link">createTechnoTable()</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Create a table with different info about Array Design: Name, number of data which use it</div>
</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html#createTISDB()" class="member-name-link">createTISDB()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Create database of all peptide from a start codon (M=ATG, L=TTG, V=GTG), to the next stop codon
 or R</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#createTISDB()" class="member-name-link">createTISDB()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Create database of all peptide from a start codon (M=ATG, L=TTG, V=GTG), to the next stop codon
 or R</div>
</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html#createUTRDB()" class="member-name-link">createUTRDB()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Create a database with all UTR from Wurtzel et al. 2012<br>
 Found here : DATABASE+"table_s1_tss_table.txt"<br>
 We found the correct frame so we are sure to have the start codon of the gene at the exact
 position of NCBI annotation We start at the first start codon, then if stop codon are found in
 the UTR region we cut in different peptides, remove ones &lt; 5aa, and paste to the last part of the
 UTR the whole gene amino acide sequence all gene, ID=lmo_n n being the id of the UTR peptide
 extracted</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#createUTRDB()" class="member-name-link">createUTRDB()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Create a database with all UTR from Wurtzel et al. 2012<br>
 Found here : DATABASE+"table_s1_tss_table.txt"<br>
 We found the correct frame so we are sure to have the start codon of the gene at the exact
 position of NCBI annotation We start at the first start codon, then if stop codon are found in
 the UTR region we cut in different peptides, remove ones &lt; 5aa, and paste to the last part of the
 UTR the whole gene amino acide sequence all gene, ID=lmo_n n being the id of the UTR peptide
 extracted</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#createValues(int,int)" class="member-name-link">createValues(int, int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#CRISPRGO_PROJECT" class="member-name-link">CRISPRGO_PROJECT</a> - Static variable in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Static variable to discriminate Listeriomics project from the others</div>
</dd>
<dt><a href="../bacnet/views/CRISPRPredictView.html" class="type-name-link" title="class in bacnet.views">CRISPRPredictView</a> - Class in <a href="../bacnet/views/package-summary.html">bacnet.views</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView.html#%3Cinit%3E()" class="member-name-link">CRISPRPredictView()</a> - Constructor for class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView2.html" class="type-name-link" title="class in bacnet.views">CRISPRPredictView2</a> - Class in <a href="../bacnet/views/package-summary.html">bacnet.views</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView2.html#%3Cinit%3E()" class="member-name-link">CRISPRPredictView2()</a> - Constructor for class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html#CS" class="member-name-link">CS</a> - Enum constant in enum bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#ct_energy" class="member-name-link">ct_energy</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/UNAfold.html#ct_energy" class="member-name-link">ct_energy</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#ct_prob" class="member-name-link">ct_prob</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/UNAfold.html#ct_prob" class="member-name-link">ct_prob</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/CUFSL.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.technology">CUFSL</a> - Class in <a href="../bacnet/scripts/listeriomics/technology/package-summary.html">bacnet.scripts.listeriomics.technology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/CUFSL.html#%3Cinit%3E()" class="member-name-link">CUFSL()</a> - Constructor for class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/CUFSL.html" title="class in bacnet.scripts.listeriomics.technology">CUFSL</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html#CURATED" class="member-name-link">CURATED</a> - Enum constant in enum org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#CURATEDARRAY_CURATED" class="member-name-link">CURATEDARRAY_CURATED</a> - Static variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">List of column names for the final curated table of mapping</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#curateGenomeFindingDoublonGeneName(bacnet.datamodel.sequenceNCBI.GenomeNCBI)" class="member-name-link">curateGenomeFindingDoublonGeneName(GenomeNCBI)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>
<div class="block">Check if two different locustag are not associated to the same GeneName</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#curateMatrix(java.lang.String,boolean)" class="member-name-link">curateMatrix(String, boolean)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>
<div class="block">Read a potential ExpressionMatrix, remove all duplicate rows by calculating mean when 2 or more
 rows have same name.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html#curateMatrix(java.lang.String,boolean)" class="member-name-link">curateMatrix(String, boolean)</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressListeriomics</a></dt>
<dd>
<div class="block">Read a potential ExpressionMatrix, remove all duplicate rows by calculating mean when 2 or more
 rows have same name.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html#curateMatrix(java.lang.String,boolean)" class="member-name-link">curateMatrix(String, boolean)</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics">ArrayExpressListeriomics</a></dt>
<dd>
<div class="block">Read a potential ExpressionMatrix, remove all duplicate rows by calculating mean when 2 or more
 rows have same name.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/IHM.html#curateRawData(java.lang.String,java.lang.String)" class="member-name-link">curateRawData(String, String)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/IHM.html" title="class in bacnet.scripts.listeriomics.technology">IHM</a></dt>
<dd>
<div class="block">Raw data from E-GEOD-12151 are not well formatted, not Tab delimited, so we need to extacrt a
 proper table from it, before working on it</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/UGiessen.html#curateRawData(java.lang.String,java.lang.String)" class="member-name-link">curateRawData(String, String)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/UGiessen.html" title="class in bacnet.scripts.listeriomics.technology">UGiessen</a></dt>
<dd>
<div class="block">Delete all first 4 columns of each raw data</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/Febit.html#curateTechnoTable(java.lang.String)" class="member-name-link">curateTechnoTable(String)</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Febit.html" title="class in bacnet.scripts.listeriomics.technology">Febit</a></dt>
<dd>
<div class="block">In some column Gene locus name is missing so we add it using EGDe genome and GenBank ID of each
 gene</div>
</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#CUTOFF" class="member-name-link">CUTOFF</a> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#CUTOFF_EVALUE" class="member-name-link">CUTOFF_EVALUE</a> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#CUTOFF_INTENSITIES" class="member-name-link">CUTOFF_INTENSITIES</a> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html#CW" class="member-name-link">CW</a> - Enum constant in enum bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#CYAN" class="member-name-link">CYAN</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
