<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 14:20:51 CEST 2019 -->
<title>S-Index</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="S-Index";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="contentContainer"><a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a><a id="I:S">
<!--   -->
</a>
<h2 class="title">S</h2>
<dl>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#sameEnd(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence,int)">sameEnd(DNASequence, DNASequence, int)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Determine that to gene have the same end if their ending piosition are the
 same +- scale bp</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#save()">save()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Save stat and values in two separate files <br>
 Saving is done via DataStream to allow streaming reading</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#save()">save()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Serialize the vector of lines with an ObjectOutputStream.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#save()">save()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Save in: TranscriptomeData.PATH_STREAMING + this.getName() +
 TranscriptomeData.EXTENSION
 
 Serialize the vector of lines with an ObjectOutputStream.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NGS.html#save(boolean)">save(boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>
<div class="block">Save stat and values in two separate files <br>
 Saving is done via DataStream to allow streaming reading</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/TabDelimitedTableReader.html#save(java.lang.Object%5B%5D%5B%5D,java.lang.String)">save(Object[][], String)</a></span> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/TabDelimitedTableReader.html#save(java.lang.Object%5B%5D%5B%5D,java.lang.String,boolean,boolean)">save(Object[][], String, boolean, boolean)</a></span> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Save the array to the destination file parameter</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#save(java.lang.String)">save(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Serialize the vector of lines with an ObjectOutputStream.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#save(java.lang.String)">save(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>
<div class="block">Serialize the vector of lines with an ObjectOutputStream.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#save(java.lang.String)">save(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Serialize the vector of lines with an ObjectOutputStream.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Network.html#save(java.lang.String)">save(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>
<div class="block">Save stat and values in two separate files <br>
 Saving is done via DataStream to allow streaming reading</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#save(java.lang.String)">save(String)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Prompt the user for a filename, and save the scribble in that file.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/Experiment.html#save(java.lang.String)">save(String)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Prompt the user for a filename, and save the scribble in that file.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#save(java.lang.String)">save(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Serialize the vector of lines with an ObjectOutputStream.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#save(java.lang.String)">save(String)</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>
<div class="block">Serialize the vector of lines with an ObjectOutputStream.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#save(java.lang.String)">save(String)</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>
<div class="block">Serialize the vector of lines with an ObjectOutputStream.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#save(java.lang.String)">save(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#save(java.lang.String)">save(String)</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#save(java.lang.String)">save(String)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#save(java.lang.String)">save(String)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#save(java.lang.String)">save(String)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>
<div class="block">Prompt the user for a filename, and save the scribble in that file.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#save(java.lang.String)">save(String)</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>
<div class="block">Prompt the user for a filename, and save the scribble in that file.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/FastaFileReader.html#saveCDS(java.util.ArrayList,java.lang.String)">saveCDS(ArrayList&lt;Gene&gt;, String)</a></span> - Static method in class bacnet.reader.<a href="../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a></dt>
<dd>
<div class="block">Save all amino acid sequence in a fasta file</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/SaveFileUtils.html#saveControltoPNG(java.lang.String,java.lang.String,org.eclipse.swt.widgets.Shell)">saveControltoPNG(String, String, Shell)</a></span> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/SaveFileUtils.html" title="class in bacnet.raprcp">SaveFileUtils</a></dt>
<dd>
<div class="block">Same as saveTextFile but with no Preview</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/SaveFileUtils.html#saveFile(java.lang.String,java.io.File,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService,org.eclipse.swt.widgets.Shell)">saveFile(String, File, String, EPartService, Shell)</a></span> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/SaveFileUtils.html" title="class in bacnet.raprcp">SaveFileUtils</a></dt>
<dd>
<div class="block">Same as saveTextFile but with no Preview</div>
</dd>
<dt><a href="../bacnet/raprcp/SaveFileUtils.html" title="class in bacnet.raprcp"><span class="typeNameLink">SaveFileUtils</span></a> - Class in <a href="../bacnet/raprcp/package-summary.html">bacnet.raprcp</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/SaveFileUtils.html#%3Cinit%3E()">SaveFileUtils()</a></span> - Constructor for class bacnet.raprcp.<a href="../bacnet/raprcp/SaveFileUtils.html" title="class in bacnet.raprcp">SaveFileUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/TabDelimitedTableReader.html#saveHashMap(java.util.HashMap,java.lang.String)">saveHashMap(HashMap&lt;String, String&gt;, String)</a></span> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Save HashMap<String, String> to a 2 columns tab-delimited table</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/TabDelimitedTableReader.html#saveInHTML(java.lang.Object%5B%5D%5B%5D,java.lang.String,java.lang.String)">saveInHTML(Object[][], String, String)</a></span> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Save the table in a text file containing an HTML table</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/TabDelimitedTableReader.html#saveList(java.util.ArrayList,java.lang.String)">saveList(ArrayList&lt;String&gt;, String)</a></span> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Save an ArrayList of String in a tabdelimited table</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/FastaFileReader.html#saveProteins(java.util.ArrayList,java.lang.String)">saveProteins(ArrayList&lt;Gene&gt;, String)</a></span> - Static method in class bacnet.reader.<a href="../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a></dt>
<dd>
<div class="block">Save all amino acid sequence in a fasta file</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/PTTReader.html#saveProteins(java.util.ArrayList,java.lang.String,bacnet.datamodel.sequence.Chromosome)">saveProteins(ArrayList&lt;Gene&gt;, String, Chromosome)</a></span> - Static method in class bacnet.reader.<a href="../bacnet/reader/PTTReader.html" title="class in bacnet.reader">PTTReader</a></dt>
<dd>
<div class="block">Save list of Gene in a PTT file<br>
 Information about the resulting protein are written into different
 columns:<br>
 "Location","Strand","Length","PID","Gene","Synonym","Code","COG","Product"</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStatProteolyse.html#saveProteolyses(java.util.ArrayList,java.lang.String)">saveProteolyses(ArrayList&lt;TIS&gt;, String)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStatProteolyse.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStatProteolyse</a></dt>
<dd>
<div class="block">Save list of protoelyses hashmap</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#saveTab(java.lang.String,java.lang.String)">saveTab(String, String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Save in a TabDelimited text file</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#saveText(java.lang.String)">saveText(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>
<div class="block">Save Signature in a readable text file</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/FileUtils.html#saveText(java.lang.String,java.lang.String)">saveText(String, String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/FileUtils.html#saveText(java.lang.String,java.lang.String,boolean,boolean)">saveText(String, String, boolean, boolean)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Save text from "text" into fileName</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/SaveFileUtils.html#saveTextFile(java.lang.String,java.lang.String,boolean,java.lang.String,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService,org.eclipse.swt.widgets.Shell)">saveTextFile(String, String, boolean, String, String, EPartService, Shell)</a></span> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/SaveFileUtils.html" title="class in bacnet.raprcp">SaveFileUtils</a></dt>
<dd>
<div class="block">Save a text file</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#saveToFileList(java.lang.String,java.lang.String,java.lang.String,java.util.ArrayList)">saveToFileList(String, String, String, ArrayList&lt;ExpressionMatrix&gt;)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#saveToFileList(java.lang.String,java.lang.String,java.lang.String,java.util.ArrayList,boolean,boolean)">saveToFileList(String, String, String, ArrayList&lt;ExpressionMatrix&gt;, boolean, boolean)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#savetoTextFile()">savetoTextFile()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>
<div class="block">Save Track information to a text file</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/TabDelimitedTableReader.html#saveTreeSet(java.util.TreeSet,java.lang.String)">saveTreeSet(TreeSet&lt;String&gt;, String)</a></span> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Save a TreeSet of String in a Tabdelimited table</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/HTMLUtils.html#SCATTER">SCATTER</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/HTMLUtils.html" title="class in bacnet.utils">HTMLUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#SCATTER_NAME">SCATTER_NAME</a></span> - Static variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Feature.html#score()">score()</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Get score.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView.html#SCRIPT_PATH">SCRIPT_PATH</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView2.html#SCRIPT_PATH">SCRIPT_PATH</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView.html#SCRIPT_SERVER">SCRIPT_SERVER</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/StartCodonProfile.html#SD_ENERGY_CUTOFF">SD_ENERGY_CUTOFF</a></span> - Static variable in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/StartCodonProfile.html" title="class in bacnet.scripts.genome">StartCodonProfile</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome"><span class="typeNameLink">SDProfile</span></a> - Class in <a href="../bacnet/scripts/genome/package-summary.html">bacnet.scripts.genome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/SDProfile.html#%3Cinit%3E()">SDProfile()</a></span> - Constructor for class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SynTView.html#search(java.lang.String)">search(String)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>
<div class="block">Search a text file in the annotation<br>
 Search if it is a position and go
 Search if it is contains in
 <code>getChromosome().getAllElements().keySet()</code>
 Search if it is contains in <code>gene.getName()</code><br>
 All the search are case insensitive using <code>text.toUpperCase();</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#search(java.lang.String)">search(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>
<div class="block">Search a text file in the annotation<br>
 Search if it is a position and go
 Search if it is contains in
 <code>getChromosome().getAllElements().keySet()</code>
 Search if it is contains in <code>gene.getName()</code><br>
 All the search are case insensitive using <code>text.toUpperCase();</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/TestGenomeBrowser.html#search(java.lang.String)">search(String)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/TestGenomeBrowser.html" title="class in bacnet.genomeBrowser">TestGenomeBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#search(java.lang.String)">search(String)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Search a specific element and center the view on it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#search(java.lang.String)">search(String)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Search a text file in the annotation<br>
 Search if it is a position and go
 Search if it is contains in
 <code>getChromosome().getAllElements().keySet()</code>
 Search if it is contains in <code>gene.getName()</code><br>
 All the search are case insensitive using <code>text.toUpperCase();</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#search(java.lang.String)">search(String)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>
<div class="block">Search a text file in the annotation<br>
 Search if it is a position and go
 Search if it is contains in
 <code>getChromosome().getAllElements().keySet()</code>
 Search if it is contains in <code>gene.getName()</code><br>
 All the search are case insensitive using <code>text.toUpperCase();</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#search(java.lang.String)">search(String)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>
<div class="block">Search a text file in the annotation<br>
 Search if it is a position and go
 Search if it is contains in
 <code>getChromosome().getAllElements().keySet()</code>
 Search if it is contains in <code>gene.getName()</code><br>
 All the search are case insensitive using <code>text.toUpperCase();</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/NavigationManagement.html#SEARCH">SEARCH</a></span> - Static variable in class bacnet.rap.<a href="../bacnet/rap/NavigationManagement.html" title="class in bacnet.rap">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/NavigationManagement.html#SEARCH">SEARCH</a></span> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#searchElement(java.lang.String)">searchElement(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Search in locustag names, gene name, and newlocustag if a gene is present or
 not</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#searchElement(org.biojava3.core.sequence.DNASequence,java.lang.String)">searchElement(DNASequence, String)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Search an element in the Note part of a DNASequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/GenomesCreation.html#searchForSTandCC()">searchForSTandCC()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Concatenate all genomes together and create a huge fasta file.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#searchMissingLocus()">searchMissingLocus()</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>
<div class="block">By going through all GFF available search for the corresponding locustag</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#searchNterm(java.lang.String)">searchNterm(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>
<div class="block">Search an NTerm among all MassSpecData</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/FileUtils.html#searchPosition(java.lang.String,java.lang.String)">searchPosition(String, String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Given a query sequence, find its different occurences on a database sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermUtils.html#searchStopCodon(bacnet.datamodel.sequence.Genome,int,boolean)">searchStopCodon(Genome, int, boolean)</a></span> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>
<div class="block">Get the next stop codon in frame and dowstream to a bpPosition</div>
</dd>
<dt><a href="../bacnet/sequenceTools/SecondStructureRNADialog.html" title="class in bacnet.sequenceTools"><span class="typeNameLink">SecondStructureRNADialog</span></a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SecondStructureRNADialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,java.lang.String)">SecondStructureRNADialog(Shell, String)</a></span> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SecondStructureRNADialog.html" title="class in bacnet.sequenceTools">SecondStructureRNADialog</a></dt>
<dd>
<div class="block">Create the dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/Hashcoder.html#SEED">SEED</a></span> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Hashcoder.html" title="class in org.biojava3.core.util">Hashcoder</a></dt>
<dd>
<div class="block">An initial value for a <code>hashCode</code>, to which we add contributions
 from fields.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/TableFilter.html#select(org.eclipse.jface.viewers.Viewer,java.lang.Object,java.lang.Object)">select(Viewer, Object, Object)</a></span> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/TableFilter.html" title="class in bacnet.table.gui">TableFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#selectByAttribute(java.lang.String)">selectByAttribute(String)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a list of all features that include the specified attribute key.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#selectByAttribute(java.lang.String,java.lang.String)">selectByAttribute(String, String)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a list of all features that include the specified attribute key/value
 pair.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#selectByGroup(java.lang.String)">selectByGroup(String)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a list of all features that have the specified group id, as defined by
 the group() method of the features.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#selectByType(java.lang.String)">selectByType(String)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a list of all features that are of the specified type, as defined by
 the type() method of the features.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#selectByUserData(java.lang.String)">selectByUserData(String)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a list of all features that include the specified key in their
 userMap().</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#selectByUserData(java.lang.String,java.lang.Object)">selectByUserData(String, Object)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a list of all features that include the specified key/value pair in
 their userMap().</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/XMLHelper.html#selectElements(org.w3c.dom.Element,java.lang.String)">selectElements(Element, String)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/UNAfold.html#selectFirstResult(java.lang.String)">selectFirstResult(String)</a></span> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>
<div class="block">Remove duplicates results, keep only first result</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/UNAfold.html#selectFirstResult(java.lang.String)">selectFirstResult(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>
<div class="block">Remove duplicates results, keep only first result</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/UNAfold.html#selectFirstResult(java.lang.String,int)">selectFirstResult(String, int)</a></span> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>
<div class="block">Remove duplicates results, keep only first result</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/UNAfold.html#selectFirstResult(java.lang.String,int)">selectFirstResult(String, int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>
<div class="block">Remove duplicates results, keep only first result</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html" title="class in bacnet.expressionAtlas.core"><span class="typeNameLink">SelectGenomeElementDialog</span></a> - Class in <a href="../bacnet/expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.TreeSet,java.util.TreeSet,java.lang.String)">SelectGenomeElementDialog(Shell, EPartService, TreeSet&lt;String&gt;, TreeSet&lt;String&gt;, String)</a></span> - Constructor for class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html" title="class in bacnet.expressionAtlas.core">SelectGenomeElementDialog</a></dt>
<dd>
<div class="block">Create the wizard.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#selectOverlapping(java.lang.String,org.biojava3.genome.parsers.gff.Location,boolean)">selectOverlapping(String, Location, boolean)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a list of all features that overlap the specified location on the
 specified sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/XMLHelper.html#selectParentElement(org.w3c.dom.Element,java.lang.String)">selectParentElement(Element, String)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/XMLHelper.html#selectSingleElement(org.w3c.dom.Element,java.lang.String)">selectSingleElement(Element, String)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SubmitDataView.html#sendEmail(java.lang.String%5B%5D,java.lang.String,java.lang.String)">sendEmail(String[], String, String)</a></span> - Static method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SubmitDataView.html" title="class in bacnet.e4.rap">SubmitDataView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#SEPARATOR">SEPARATOR</a></span> - Static variable in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">BioConditoon separator for comparisons<br>
 equal to " vs "</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#seqArraySize(int)">seqArraySize(int)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/FastaFileReader.html#seqfileExt">seqfileExt</a></span> - Static variable in class bacnet.reader.<a href="../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Feature.html#seqname()">seqname()</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Get the sequence name.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/FeatureI.html#seqname()">seqname()</a></span> - Method in interface org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a></dt>
<dd>
<div class="block">Get the sequence name.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence"><span class="typeNameLink">Sequence</span></a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are
 <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><span class="typeNameLink">Sequence</span></a>&lt;<a href="../org/biojava3/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>
<div class="block">Main interface for defining a collection of Compounds and accessing them
 using biological indexes</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#%3Cinit%3E()">Sequence()</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#%3Cinit%3E(java.lang.String,int,int)">Sequence(String, int, int)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Create a Sequence object<br>
 if(begin>end) <br>
 this.end = begin <br>
 this.begin = end <br>
 this.strand = '-' <br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#%3Cinit%3E(java.lang.String,int,int,char)">Sequence(String, int, int, char)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Create a Sequence with a specific name, from, to and strand <br>
 CAREFUL: <br>
 from < to ALWAYS<br>
 sense is given by strand<br>
 If to < from it will correct automatically the position and strand, to have
 from < to</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence"><span class="typeNameLink">Sequence.SeqType</span></a> - Enum in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="class in org.biojava3.core.sequence.storage"><span class="typeNameLink">SequenceAsStringHelper</span></a>&lt;<a href="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="type parameter in SequenceAsStringHelper">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</a></dt>
<dd>
<div class="block">This is a common method that can be used across multiple storage/proxy
 implementations to handle Negative strand and other interesting elements of
 sequence data.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html#%3Cinit%3E()">SequenceAsStringHelper()</a></span> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="class in org.biojava3.core.sequence.storage">SequenceAsStringHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceComparator.html" title="class in org.biojava3.core.sequence"><span class="typeNameLink">SequenceComparator</span></a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">Used to sort sequences</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/SequenceComparator.html#%3Cinit%3E()">SequenceComparator()</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceComparator.html" title="class in org.biojava3.core.sequence">SequenceComparator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template"><span class="typeNameLink">SequenceCreatorInterface</span></a>&lt;<a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/io/template/package-summary.html">org.biojava3.core.sequence.io.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools"><span class="typeNameLink">SequenceDisplayDialog</span></a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,org.eclipse.e4.ui.workbench.modeling.EPartService,bacnet.datamodel.sequence.Genome,java.lang.String,boolean,int,int,org.biojava3.core.sequence.Strand)">SequenceDisplayDialog(Shell, EPartService, Genome, String, boolean, int, int, Strand)</a></span> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>
<div class="block">Create the dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#sequenceEquality(org.biojava3.core.sequence.template.Sequence,org.biojava3.core.sequence.template.Sequence)">sequenceEquality(Sequence&lt;C&gt;, Sequence&lt;C&gt;)</a></span> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">A case-sensitive manner of comparing two sequence objects together.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#sequenceEqualityIgnoreCase(org.biojava3.core.sequence.template.Sequence,org.biojava3.core.sequence.template.Sequence)">sequenceEqualityIgnoreCase(Sequence&lt;C&gt;, Sequence&lt;C&gt;)</a></span> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">A case-insensitive manner of comparing two sequence objects together.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader"><span class="typeNameLink">SequenceFileProxyLoader</span></a>&lt;<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</a></dt>
<dd>
<div class="block">This class represents the storage container of a sequence stored in a fasta
 file where the initial parsing of the file we store the offset and length of
 the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.io.template.SequenceParserInterface,long,int,org.biojava3.core.sequence.template.CompoundSet)">SequenceFileProxyLoader(File, SequenceParserInterface, long, int, CompoundSet&lt;C&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence)">SequenceIterator(Sequence&lt;C&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="class in org.biojava3.core.sequence.template">SequenceMixin.SequenceIterator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/exceptions/SequenceLengthError.html" title="class in org.biojava3.core.exceptions"><span class="typeNameLink">SequenceLengthError</span></a> - Error in <a href="../org/biojava3/core/exceptions/package-summary.html">org.biojava3.core.exceptions</a></dt>
<dd>
<div class="block">Currently not used</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/exceptions/SequenceLengthError.html#%3Cinit%3E(java.lang.String)">SequenceLengthError(String)</a></span> - Constructor for error org.biojava3.core.exceptions.<a href="../org/biojava3/core/exceptions/SequenceLengthError.html" title="class in org.biojava3.core.exceptions">SequenceLengthError</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SequenceLocation.html" title="class in org.biojava3.core.sequence.location"><span class="typeNameLink">SequenceLocation</span></a>&lt;<a href="../org/biojava3/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">S</a> extends <a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../org/biojava3/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">C</a>&gt;,&#8203;<a href="../org/biojava3/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</a></dt>
<dd>
<div class="block">A location in a sequence that keeps a reference to its parent sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SequenceLocation.html#%3Cinit%3E(int,int,S)">SequenceLocation(int, int, S)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SequenceLocation.html" title="class in org.biojava3.core.sequence.location">SequenceLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template"><span class="typeNameLink">SequenceMixin</span></a> - Class in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>
<div class="block">Provides a set of static methods to be used as static imports when needed
 across multiple Sequence implementations but inheritance gets in the way.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#%3Cinit%3E()">SequenceMixin()</a></span> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="class in org.biojava3.core.sequence.template"><span class="typeNameLink">SequenceMixin.SequenceIterator</span></a>&lt;<a href="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="type parameter in SequenceMixin.SequenceIterator">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>
<div class="block">A basic sequence iterator which iterates over the given Sequence by
 biological index.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence"><span class="typeNameLink">SequenceOptimizationHints</span></a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">A static class that provides optimization hints for memory or performance
 handling of sequence data.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html#%3Cinit%3E()">SequenceOptimizationHints()</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceCollection.html" title="enum in org.biojava3.core.sequence"><span class="typeNameLink">SequenceOptimizationHints.SequenceCollection</span></a> - Enum in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceUsage.html" title="enum in org.biojava3.core.sequence"><span class="typeNameLink">SequenceOptimizationHints.SequenceUsage</span></a> - Enum in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/template/SequenceParserInterface.html" title="interface in org.biojava3.core.sequence.io.template"><span class="typeNameLink">SequenceParserInterface</span></a> - Interface in <a href="../org/biojava3/core/sequence/io/template/package-summary.html">org.biojava3.core.sequence.io.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template"><span class="typeNameLink">SequenceProxyView</span></a>&lt;<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#%3Cinit%3E()">SequenceProxyView()</a></span> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence)">SequenceProxyView(Sequence&lt;C&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence,java.lang.Integer,java.lang.Integer)">SequenceProxyView(Sequence&lt;C&gt;, Integer, Integer)</a></span> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>
<div class="block">Main constructor for working with SequenceProxyViews</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template"><span class="typeNameLink">SequenceReader</span></a>&lt;<a href="../org/biojava3/core/sequence/template/SequenceReader.html" title="type parameter in SequenceReader">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/SequenceTools.html" title="class in org.biojava3.core.util"><span class="typeNameLink">SequenceTools</span></a> - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/SequenceTools.html#%3Cinit%3E()">SequenceTools()</a></span> - Constructor for class org.biojava3.core.util.<a href="../org/biojava3/core/util/SequenceTools.html" title="class in org.biojava3.core.util">SequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template"><span class="typeNameLink">SequenceView</span></a>&lt;<a href="../org/biojava3/core/sequence/template/SequenceView.html" title="type parameter in SequenceView">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/DownloadServiceHandler.html#service(javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse)">service(HttpServletRequest, HttpServletResponse)</a></span> - Method in class bacnet.rap.<a href="../bacnet/rap/DownloadServiceHandler.html" title="class in bacnet.rap">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/DownloadServiceHandler.html#service(javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse)">service(HttpServletRequest, HttpServletResponse)</a></span> - Method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SessionControl.html" title="class in bacnet.e4.rap"><span class="typeNameLink">SessionControl</span></a> - Class in <a href="../bacnet/e4/rap/package-summary.html">bacnet.e4.rap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/SessionControl.html" title="class in bacnet.e4.rap.setup"><span class="typeNameLink">SessionControl</span></a> - Class in <a href="../bacnet/e4/rap/setup/package-summary.html">bacnet.e4.rap.setup</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SessionControl.html#%3Cinit%3E()">SessionControl()</a></span> - Constructor for class bacnet.e4.rap.<a href="../bacnet/e4/rap/SessionControl.html" title="class in bacnet.e4.rap">SessionControl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/setup/SessionControl.html#%3Cinit%3E()">SessionControl()</a></span> - Constructor for class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SessionControl.html" title="class in bacnet.e4.rap.setup">SessionControl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setAccession(org.biojava3.core.sequence.AccessionID)">setAccession(AccessionID)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#setAlias(java.lang.String)">setAlias(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setAllBioConditions(java.util.ArrayList)">setAllBioConditions(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setAllButtons(java.util.ArrayList)">setAllButtons(ArrayList&lt;Button&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#setAllButtons(java.util.ArrayList)">setAllButtons(ArrayList&lt;Button&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setAllButtons(java.util.ArrayList)">setAllButtons(ArrayList&lt;Button&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ComparisonsCreation.html#setAllComparisonStreamData(java.util.ArrayList)">setAllComparisonStreamData(ArrayList&lt;String&gt;)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">For each Comparison calculates corresponding TranscriptomeData, and save it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setAllElements(java.util.LinkedHashMap)">setAllElements(LinkedHashMap&lt;String, Sequence&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCPList.html#setAllFonts(java.awt.Font)">setAllFonts(Font)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperList.html#setAllFonts(org.eclipse.swt.graphics.Font)">setAllFonts(Font)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#setAlreadyRead(boolean)">setAlreadyRead(boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#setAnnotation(bacnet.datamodel.annotation.Annotation)">setAnnotation(Annotation)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setAnnotation(bacnet.datamodel.annotation.Annotation)">setAnnotation(Annotation)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setAnnotation(bacnet.datamodel.annotation.Annotation)">setAnnotation(Annotation)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setAnnotation(bacnet.datamodel.annotation.Annotation)">setAnnotation(Annotation)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#setAnnotation(java.lang.String%5B%5D%5B%5D)">setAnnotation(String[][])</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#setAnnotationFont(org.eclipse.swt.graphics.GC)">setAnnotationFont(GC)</a></span> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Set Font to : "Arial",9,SWT.BOLD</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setAnnotations(java.lang.String%5B%5D%5B%5D)">setAnnotations(String[][])</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType(AbstractSequence.AnnotationType)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setAnnotElementHeight(int)">setAnnotElementHeight(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setAnnotNumber(float)">setAnnotNumber(float)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setAnnotSeparationHeight(int)">setAnnotSeparationHeight(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setAnnotSeparationNumber(int)">setAnnotSeparationNumber(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setAntiComparisons(java.util.ArrayList)">setAntiComparisons(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setAntiSDBindingFreeEnergy(double)">setAntiSDBindingFreeEnergy(double)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaSummaryView.html#setArray(java.lang.String%5B%5D%5B%5D)">setArray(String[][])</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaSummaryView.html" title="class in bacnet.sequenceTools">SrnaSummaryView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setArrayDataList(java.lang.String%5B%5D%5B%5D)">setArrayDataList(String[][])</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setArrayDataList(java.lang.String%5B%5D%5B%5D)">setArrayDataList(String[][])</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#setArrayExpressArrayLink(java.lang.String)">setArrayExpressArrayLink(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#setArrayExpressExpLink(java.lang.String)">setArrayExpressExpLink(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setArrayExpressId(java.lang.String)">setArrayExpressId(String)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setArrayExpressTechnoId(java.lang.String)">setArrayExpressTechnoId(String)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setArrayGeneToLocalization(java.lang.String%5B%5D%5B%5D)">setArrayGeneToLocalization(String[][])</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setArrayListWithAnnotation(java.lang.String)">setArrayListWithAnnotation(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>
<div class="block">Read Annotation and from the list of sequences obtained, create the different
 LinkedHashMap<br>
 This method is run at the construction of a Genome object</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setArrayProteomeList(java.lang.String%5B%5D%5B%5D)">setArrayProteomeList(String[][])</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setAsRNAListEGDe(java.util.ArrayList)">setAsRNAListEGDe(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setAsRNAs(java.util.ArrayList)">setAsRNAs(ArrayList&lt;Srna&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setAsRNAs(java.util.LinkedHashMap)">setAsRNAs(LinkedHashMap&lt;String, Srna&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermFilter.html#setaTISCutoff(int)">setaTISCutoff(int)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#setaTISMap(java.util.HashMap)">setaTISMap(HashMap&lt;String, ArrayList&lt;NTerm&gt;&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#setBegin(int)">setBegin(int)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setBegin(int)">setBegin(int)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setBegin(int)">setBegin(int)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setBioBegin(java.lang.Integer)">setBioBegin(Integer)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#setBioCondition()">setBioCondition()</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#setBioConditionHashMaps(java.util.LinkedHashMap)">setBioConditionHashMaps(LinkedHashMap&lt;String, BioCondition&gt;)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DataValidation.html#setBioconditions(java.util.HashMap)">setBioconditions(HashMap&lt;String, Boolean&gt;)</a></span> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#setBioConditions(java.util.ArrayList)">setBioConditions(ArrayList&lt;BioCondition&gt;)</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#setBioConditions(java.util.ArrayList)">setBioConditions(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setBioConditionsArrayPath(java.lang.String)">setBioConditionsArrayPath(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#setBioCondName(java.lang.String)">setBioCondName(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#setBioConds(java.util.ArrayList)">setBioConds(ArrayList&lt;String[]&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#setBioConds(java.util.ArrayList)">setBioConds(ArrayList&lt;String[]&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#setBioConds(java.util.ArrayList)">setBioConds(ArrayList&lt;String[]&gt;)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#setBioConds(java.util.ArrayList)">setBioConds(ArrayList&lt;String[]&gt;)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#setBioConds(java.util.ArrayList)">setBioConds(ArrayList&lt;String[]&gt;)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setBioConds(java.util.ArrayList)">setBioConds(ArrayList&lt;String[]&gt;)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/Experiment.html#setBioConds(java.util.LinkedHashMap)">setBioConds(LinkedHashMap&lt;String, BioCondition&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#setBioCondsArray(java.lang.String%5B%5D%5B%5D)">setBioCondsArray(String[][])</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#setBioCondsArray(java.lang.String%5B%5D%5B%5D)">setBioCondsArray(String[][])</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#setBioCondsArray(java.lang.String%5B%5D%5B%5D)">setBioCondsArray(String[][])</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#setBioCondsArray(java.lang.String%5B%5D%5B%5D)">setBioCondsArray(String[][])</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#setBioCondsArray(java.lang.String%5B%5D%5B%5D)">setBioCondsArray(String[][])</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setBioCondsArray(java.lang.String%5B%5D%5B%5D)">setBioCondsArray(String[][])</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GenomicsView.html#setBioCondsArray(java.lang.String%5B%5D%5B%5D)">setBioCondsArray(String[][])</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#setBioCondsToDisplay(java.util.ArrayList)">setBioCondsToDisplay(ArrayList&lt;String[]&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#setBioCondsToDisplay(java.util.ArrayList)">setBioCondsToDisplay(ArrayList&lt;String[]&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#setBioCondsToDisplay(java.util.ArrayList)">setBioCondsToDisplay(ArrayList&lt;String[]&gt;)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#setBioCondsToDisplay(java.util.ArrayList)">setBioCondsToDisplay(ArrayList&lt;String[]&gt;)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#setBioCondsToDisplay(java.util.ArrayList)">setBioCondsToDisplay(ArrayList&lt;String[]&gt;)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setBioCondsToDisplay(java.util.ArrayList)">setBioCondsToDisplay(ArrayList&lt;String[]&gt;)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GenomicsView.html#setBioCondsToDisplay(java.util.ArrayList)">setBioCondsToDisplay(ArrayList&lt;String[]&gt;)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setBioEnd(java.lang.Integer)">setBioEnd(Integer)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#setBioEnd(java.lang.Integer)">setBioEnd(Integer)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#setBioStart(java.lang.Integer)">setBioStart(Integer)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorSWT.html#setBlue(int)">setBlue(int)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setBpSizeH(double)">setBpSizeH(double)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setBpSizeH(double)">setBpSizeH(double)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnAllMutant(org.eclipse.swt.widgets.Button)">setBtnAllMutant(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnBhiBroth(org.eclipse.swt.widgets.Button)">setBtnBhiBroth(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnCacoCells(org.eclipse.swt.widgets.Button)">setBtnCacoCells(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnChooseOneMutant(org.eclipse.swt.widgets.Button)">setBtnChooseOneMutant(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnDeath(org.eclipse.swt.widgets.Button)">setBtnDeath(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnDnDNA(org.eclipse.swt.widgets.Button)">setBtnDnDNA(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnDnProt(org.eclipse.swt.widgets.Button)">setBtnDnProt(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnDnRNA(org.eclipse.swt.widgets.Button)">setBtnDnRNA(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnExponential(org.eclipse.swt.widgets.Button)">setBtnExponential(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnGeneExpression(org.eclipse.swt.widgets.Button)">setBtnGeneExpression(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnHumanBloodCells(org.eclipse.swt.widgets.Button)">setBtnHumanBloodCells(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnLagPhase(org.eclipse.swt.widgets.Button)">setBtnLagPhase(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnLbBroth(org.eclipse.swt.widgets.Button)">setBtnLbBroth(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnMinimalMediaBroth(org.eclipse.swt.widgets.Button)">setBtnMinimalMediaBroth(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnMouseIntestineCells(org.eclipse.swt.widgets.Button)">setBtnMouseIntestineCells(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnMouseMacrophagesCells(org.eclipse.swt.widgets.Button)">setBtnMouseMacrophagesCells(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnMouseSpleenCells(org.eclipse.swt.widgets.Button)">setBtnMouseSpleenCells(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnNoDNA(org.eclipse.swt.widgets.Button)">setBtnNoDNA(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnNoneMutant(org.eclipse.swt.widgets.Button)">setBtnNoneMutant(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnNoProt(org.eclipse.swt.widgets.Button)">setBtnNoProt(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnNoRNA(org.eclipse.swt.widgets.Button)">setBtnNoRNA(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setBtnOpenStructure(org.eclipse.swt.widgets.Button)">setBtnOpenStructure(Button)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnPorcineBileCells(org.eclipse.swt.widgets.Button)">setBtnPorcineBileCells(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnPosDNA(org.eclipse.swt.widgets.Button)">setBtnPosDNA(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnPrint(org.eclipse.swt.widgets.Button)">setBtnPrint(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnRegrowth(org.eclipse.swt.widgets.Button)">setBtnRegrowth(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnRiboSeq(org.eclipse.swt.widgets.Button)">setBtnRiboSeq(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnRnaseq(org.eclipse.swt.widgets.Button)">setBtnRnaseq(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnStationnary(org.eclipse.swt.widgets.Button)">setBtnStationnary(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnSurvival(org.eclipse.swt.widgets.Button)">setBtnSurvival(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnTermSeq(org.eclipse.swt.widgets.Button)">setBtnTermSeq(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnTiling(org.eclipse.swt.widgets.Button)">setBtnTiling(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnTrypticSoyBroth(org.eclipse.swt.widgets.Button)">setBtnTrypticSoyBroth(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnTss(org.eclipse.swt.widgets.Button)">setBtnTss(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnTurkeyDeliMeat(org.eclipse.swt.widgets.Button)">setBtnTurkeyDeliMeat(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnUnpublished(org.eclipse.swt.widgets.Button)">setBtnUnpublished(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnUpdateCutoff(org.eclipse.swt.widgets.Button)">setBtnUpdateCutoff(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnUpProt(org.eclipse.swt.widgets.Button)">setBtnUpProt(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnUpRNA(org.eclipse.swt.widgets.Button)">setBtnUpRNA(Button)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setCenterColor(org.eclipse.swt.graphics.Color)">setCenterColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setCenterColor(org.eclipse.swt.graphics.Color)">setCenterColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setCenterColorSWT(bacnet.table.core.ColorSWT)">setCenterColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setCenterColorSWT(bacnet.table.core.ColorSWT)">setCenterColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setCenterPos(double)">setCenterPos(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setCenterPos(double)">setCenterPos(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#setCgview(ca.ualberta.stothard.cgview.Cgview)">setCgview(Cgview)</a></span> - Method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setChildrenFeatures(java.util.List)">setChildrenFeatures(List&lt;FeatureInterface&lt;S, C&gt;&gt;)</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">Set the children features</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setChildrenFeatures(java.util.List)">setChildrenFeatures(List&lt;FeatureInterface&lt;S, C&gt;&gt;)</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Set the children features</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setChromoId(java.lang.String)">setChromoId(String)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setChromoID(java.lang.String)">setChromoID(String)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setChromosome(bacnet.datamodel.sequence.Chromosome)">setChromosome(Chromosome)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#setChromosome(bacnet.datamodel.sequence.Chromosome)">setChromosome(Chromosome)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#setChromosomeID(java.lang.String)">setChromosomeID(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#setChromosomeID(java.lang.String)">setChromosomeID(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setChromosomeID(java.lang.String)">setChromosomeID(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setChromosomeID(java.lang.String)">setChromosomeID(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setChromosomeID(java.lang.String)">setChromosomeID(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#setChromosomeID(java.lang.String)">setChromosomeID(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#setChromosomeNumber(int)">setChromosomeNumber(int)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#setChromosomes(java.util.LinkedHashMap)">setChromosomes(LinkedHashMap&lt;String, Chromosome&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#setChromosomes(java.util.LinkedHashMap)">setChromosomes(LinkedHashMap&lt;String, ChromosomeBacteriaSequence&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setCisRegRNAListEGDe(java.util.ArrayList)">setCisRegRNAListEGDe(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setCisRegs(java.util.ArrayList)">setCisRegs(ArrayList&lt;Srna&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setCisRegs(java.util.LinkedHashMap)">setCisRegs(LinkedHashMap&lt;String, Srna&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setCmbChromo(org.eclipse.swt.widgets.Combo)">setCmbChromo(Combo)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setCmbGenome(org.eclipse.swt.widgets.Combo)">setCmbGenome(Combo)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setCodingSequenceHashMap(java.util.LinkedHashMap)">setCodingSequenceHashMap(LinkedHashMap&lt;String, DNASequence&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setCodonUsage(double)">setCodonUsage(double)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#setCog(java.lang.String)">setCog(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setColorMapperList(bacnet.table.core.ColorMapperList)">setColorMapperList(ColorMapperList)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperList.html#setColorMappers(java.util.TreeMap)">setColorMappers(TreeMap&lt;ColorMapper.TypeMapper, ColorMapper&gt;)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCPList.html#setColorMappers(java.util.TreeMap)">setColorMappers(TreeMap&lt;ColorMapperRCP.Type, ColorMapperRCP&gt;)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setColorMapperValues(java.util.ArrayList)">setColorMapperValues(ArrayList&lt;ExpressionMatrix&gt;)</a></span> - Static method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setColorMapperValues(java.util.ArrayList)">setColorMapperValues(ArrayList&lt;ExpressionMatrix&gt;)</a></span> - Static method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/BioConditionComparator.html#setColumn(int)">setColumn(int)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/TableViewerComparator.html#setColumn(int)">setColumn(int)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/TableViewerComparator.html" title="class in bacnet.table.core">TableViewerComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#setColumnNames(java.util.ArrayList)">setColumnNames(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setColumnNames(java.util.ArrayList)">setColumnNames(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#setColumnNames(java.util.ArrayList)">setColumnNames(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#setColumnNames(java.util.ArrayList)">setColumnNames(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#setColumnNames(java.util.ArrayList)">setColumnNames(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#setColumnNames(java.util.ArrayList)">setColumnNames(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GenomicsView.html#setColumnNames(java.util.ArrayList)">setColumnNames(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setColumnNames(java.util.TreeSet)">setColumnNames(TreeSet&lt;String&gt;)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setColumnNames(java.util.TreeSet)">setColumnNames(TreeSet&lt;String&gt;)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setComboAbsoluteRelative(org.eclipse.swt.widgets.Combo)">setComboAbsoluteRelative(Combo)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#setComboGenome(org.eclipse.swt.widgets.Combo)">setComboGenome(Combo)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setComboGenome(org.eclipse.swt.widgets.Combo)">setComboGenome(Combo)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setComboGenome(org.eclipse.swt.widgets.Combo)">setComboGenome(Combo)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setComboMutant(org.eclipse.swt.widgets.Combo)">setComboMutant(Combo)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setComment(java.lang.String)">setComment(String)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setComment(java.lang.String)">setComment(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setComparisons(java.util.ArrayList)">setComparisons(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DataValidation.html#setComparisons(java.util.HashMap)">setComparisons(HashMap&lt;String, Boolean&gt;)</a></span> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#setCompExpression(bacnet.datamodel.dataset.ExpressionMatrix)">setCompExpression(ExpressionMatrix)</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setCompFilter(org.eclipse.swt.widgets.Composite)">setCompFilter(Composite)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setComposite(org.eclipse.swt.widgets.Composite)">setComposite(Composite)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setComposite_2(org.eclipse.swt.widgets.Composite)">setComposite_2(Composite)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setComposite_6(org.eclipse.swt.widgets.Composite)">setComposite_6(Composite)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setComposite_7(org.eclipse.swt.widgets.Composite)">setComposite_7(Composite)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setComposite_7(org.eclipse.swt.widgets.Composite)">setComposite_7(Composite)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setCompositeAddData(org.eclipse.swt.widgets.Composite)">setCompositeAddData(Composite)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#setCompositeDataFilter(bacnet.expressionAtlas.ProteomicsDataFilterComposite)">setCompositeDataFilter(ProteomicsDataFilterComposite)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#setCompositeDataFilter(bacnet.expressionAtlas.TranscriptomicsDataFilterComposite)">setCompositeDataFilter(TranscriptomicsDataFilterComposite)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setCompositeNterm(bacnet.genomeBrowser.CompositeNTermInfo)">setCompositeNterm(CompositeNTermInfo)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setCompositeSummary(org.eclipse.swt.widgets.Composite)">setCompositeSummary(Composite)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setCompositeTypeDisplay(org.eclipse.swt.widgets.Composite)">setCompositeTypeDisplay(Composite)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#setCompoundAt(char,int)">setCompoundAt(char, int)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>
<div class="block">Converts from char to Compound and sets it at the given biological index</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#setCompoundAt(C,int)">setCompoundAt(C, int)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>
<div class="block">Sets the compound at the specified biological index</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet(CompoundSet&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet(CompoundSet&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet(CompoundSet&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet(CompoundSet&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet(CompoundSet&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Class is immutable & so this is unsupported</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet(CompoundSet&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet(CompoundSet&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Unsupported</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet(CompoundSet&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet(CompoundSet&lt;C&gt;)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet(CompoundSet&lt;NucleotideCompound&gt;)</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setConservation(int)">setConservation(int)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setConservationHashMap(java.util.LinkedHashMap)">setConservationHashMap(LinkedHashMap&lt;String, String&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#setContainer(org.eclipse.swt.widgets.Composite)">setContainer(Composite)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#setContents(java.lang.String)">setContents(String)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#setContents(java.lang.String)">setContents(String)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#setContents(java.lang.String)">setContents(String)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">Once the sequence is retrieved set the contents and make sure everything this
 is valid</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#setContents(java.lang.String)">setContents(String)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#setContents(java.lang.String)">setContents(String)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Class is immutable & so this is unsupported</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#setContents(java.lang.String)">setContents(String)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#setContents(java.lang.String)">setContents(String)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Unsupported</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceReader.html#setContents(java.lang.String)">setContents(String)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#setContents(java.lang.String)">setContents(String)</a></span> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#setContents(java.util.List)">setContents(List&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setCurrentState(java.lang.String)">setCurrentState(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#setCutOff1(double)">setCutOff1(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#setCutOff1(double)">setCutOff1(double)</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#setCutOff2(double)">setCutOff2(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#setCutOff2(double)">setCutOff2(double)</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermFilter.html#setCutoffHashMap(java.util.HashMap)">setCutoffHashMap(HashMap&lt;String, ArrayList&lt;String&gt;&gt;)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/TableSWTView.html#setData(bacnet.datamodel.dataset.ExpressionMatrix)">setData(ExpressionMatrix)</a></span> - Method in class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#setData(java.lang.String,java.util.ArrayList)">setData(String, ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">set the Data for the genomeViewer</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#setDatabase(java.lang.String)">setDatabase(String)</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setDatabaseFeatures(java.util.HashMap)">setDatabaseFeatures(HashMap&lt;String, String&gt;)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences(DatabaseReferenceInterface)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#setDataColors()">setDataColors()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">For each data assign a Color <br>
 <br>
 OVERLAYBC : Color are always the same, blue and red for tiling, black for
 GeneExpr, other color for the rest <br>
 SUPERIMPOSED, OVERLAYDATA : Color.getColors(index) is used</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#setDataColors(java.util.TreeMap)">setDataColors(TreeMap&lt;String, Color&gt;)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#setDataNameFont(org.eclipse.swt.graphics.GC)">setDataNameFont(GC)</a></span> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Set Font to : "Arial",11,SWT.BOLD</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#setDataNOTDisplayed(java.util.ArrayList)">setDataNOTDisplayed(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#setDataRead(boolean)">setDataRead(boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#setDataRegion(bacnet.genomeBrowser.core.Region)">setDataRegion(Region)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#setDatas(bacnet.genomeBrowser.core.DataTrack)">setDatas(DataTrack)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NGS.html#setDatasets(java.util.HashMap)">setDatasets(HashMap&lt;String, ExpressionData&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#setDataSizes()">setDataSizes()</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">For each data assign a Size <br>
 <br>
 1.5 for Tiling array, GeneExpression array and RNASeq
 1 for TSS
 0.5 for NTerm and Proteomics data</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setDataWindowSize(int)">setDataWindowSize(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#setDate(java.lang.String)">setDate(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setDate(java.lang.String)">setDate(String)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setDecaySliderVBar(int)">setDecaySliderVBar(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setDefault()">setDefault()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setDefault()">setDefault()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#setDefault()">setDefault()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#setDefault()">setDefault()</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#setDefaultAlpha(org.eclipse.swt.graphics.GC)">setDefaultAlpha(GC)</a></span> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Set Alpha to 255</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setDefaultColor()">setDefaultColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setDefaultColor()">setDefaultColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setDefaultFont()">setDefaultFont()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#setDefaultFont(org.eclipse.swt.graphics.GC)">setDefaultFont(GC)</a></span> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Set Font to : "Arial",12,SWT.NORMAL</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setDefaultPos()">setDefaultPos()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setDefaultPos()">setDefaultPos()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#setDescription(java.lang.String)">setDescription(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#setDescription(java.lang.String)">setDescription(String)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setDescription(java.lang.String)">setDescription(String)</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setDescription(java.lang.String)">setDescription(String)</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Set the description that can be used to describe the feature</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#setDescription(java.lang.String)">setDescription(String)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setDescription(java.lang.String)">setDescription(String)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Compound.html#setDescription(java.lang.String)">setDescription(String)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#setDescription(java.lang.String)">setDescription(String)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#setDescription(java.lang.String)">setDescription(String)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setDeviation(double)">setDeviation(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setDeviation(double)">setDeviation(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#setDisplay(boolean%5B%5D)">setDisplay(boolean[])</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setDisplay(org.eclipse.swt.widgets.Display)">setDisplay(Display)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#setDisplayAbsoluteValue(boolean)">setDisplayAbsoluteValue(boolean)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setDisplayMouseLine(boolean)">setDisplayMouseLine(boolean)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setDisplayMouseLine(boolean)">setDisplayMouseLine(boolean)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#setDisplayRegion(bacnet.genomeBrowser.core.Region)">setDisplayRegion(Region)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#setDisplaySequence(boolean)">setDisplaySequence(boolean)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#setDisplayType(bacnet.genomeBrowser.core.Track.DisplayType)">setDisplayType(Track.DisplayType)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#setDNAType(org.biojava3.core.sequence.DNASequence.DNAType)">setDNAType(DNASequence.DNAType)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setDownDNAList(java.util.ArrayList)">setDownDNAList(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/InternalBrowser.html#setDownloadLink(java.lang.String)">setDownloadLink(String)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setDownProtList(java.util.ArrayList)">setDownProtList(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setDownRNAList(java.util.ArrayList)">setDownRNAList(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setDuplicates(java.util.ArrayList)">setDuplicates(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermFilter.html#setDuplicateSequenceToRemove(java.util.ArrayList)">setDuplicateSequenceToRemove(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Network.html#setEdges(java.util.HashMap)">setEdges(HashMap&lt;String, HashMap&lt;String, Double&gt;&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>
<div class="block">An hashMap which map each vertex (=genome element) to the correlated
 vertex.<br>
 Vertex -> HashMap<Vetrex,PearsonCorrelation></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setElements(java.util.ArrayList)">setElements(ArrayList&lt;Sequence&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#setElements(java.util.ArrayList)">setElements(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#setElements(java.util.HashMap)">setElements(HashMap&lt;String, NTerm&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Link NTerm name to <code>NTerm</code><br>
 This correspond to the list of <code>NTerm</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setElements(java.util.LinkedHashMap)">setElements(LinkedHashMap&lt;String, Sequence&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#setEnd(int)">setEnd(int)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setEnd(int)">setEnd(int)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setEnd(int)">setEnd(int)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setExcludeColumn(java.util.ArrayList)">setExcludeColumn(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#setExcludeColumn(java.util.ArrayList)">setExcludeColumn(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/FilterList.html#setExcludeColumn(java.util.ArrayList)">setExcludeColumn(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#setExcludeColumn(java.util.ArrayList)">setExcludeColumn(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setExcludeRow(java.util.ArrayList)">setExcludeRow(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#setExcludeRow(java.util.ArrayList)">setExcludeRow(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/FilterList.html#setExcludeRow(java.util.ArrayList)">setExcludeRow(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#setExcludeRow(java.util.ArrayList)">setExcludeRow(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SessionControl.html#setExitConfirmation()">setExitConfirmation()</a></span> - Static method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SessionControl.html" title="class in bacnet.e4.rap">SessionControl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/setup/SessionControl.html#setExitConfirmation()">setExitConfirmation()</a></span> - Static method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SessionControl.html" title="class in bacnet.e4.rap.setup">SessionControl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setExperienceNb(int)">setExperienceNb(int)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setExperimentComparisonTablePath(java.lang.String)">setExperimentComparisonTablePath(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#setExpression(bacnet.datamodel.dataset.ExpressionMatrix)">setExpression(ExpressionMatrix)</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#setExpression(bacnet.datamodel.dataset.ExpressionMatrix)">setExpression(ExpressionMatrix)</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#setExpressionAlpha(org.eclipse.swt.graphics.GC)">setExpressionAlpha(GC)</a></span> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Set Alpha to 150</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setExprProteomesTable(java.util.HashMap)">setExprProteomesTable(HashMap&lt;String, ExpressionMatrix&gt;)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setFeatures(java.util.LinkedHashMap)">setFeatures(LinkedHashMap&lt;String, String&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord(FeaturesKeyWordInterface)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/DownloadServiceHandler.html#setFile(java.io.File)">setFile(File)</a></span> - Method in class bacnet.rap.<a href="../bacnet/rap/DownloadServiceHandler.html" title="class in bacnet.rap">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/DownloadServiceHandler.html#setFile(java.io.File)">setFile(File)</a></span> - Method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/DownloadServiceHandler.html#setFileName(java.lang.String)">setFileName(String)</a></span> - Method in class bacnet.rap.<a href="../bacnet/rap/DownloadServiceHandler.html" title="class in bacnet.rap">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/DownloadServiceHandler.html#setFileName(java.lang.String)">setFileName(String)</a></span> - Method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/InternalBrowser.html#setFileName(java.lang.String)">setFileName(String)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#setFilterColumn(boolean)">setFilterColumn(boolean)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#setFilterColumn(boolean)">setFilterColumn(boolean)</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SynTView.html#setFilteredNetwork(bacnet.datamodel.dataset.Network)">setFilteredNetwork(Network)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#setFilteredNetwork(bacnet.datamodel.dataset.Network)">setFilteredNetwork(Network)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/FilterList.html#setFilters(java.util.TreeMap)">setFilters(TreeMap&lt;String, Filter&gt;)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setFirstandSecondRowNameInList(java.util.ArrayList,java.lang.String,java.lang.String)">setFirstandSecondRowNameInList(ArrayList&lt;ExpressionMatrix&gt;, String, String)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setFirstRowName(java.lang.String)">setFirstRowName(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/setup/SetupPart.html#setFocus()">setFocus()</a></span> - Method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SetupPart.html" title="class in bacnet.e4.rap.setup">SetupPart</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/TestPart.html#setFocus()">setFocus()</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/TestPart.html" title="class in bacnet.e4.rap">TestPart</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView.html#setFocus()">setFocus()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView2.html#setFocus()">setFocus()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewBacnet.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewBacnet.html" title="class in bacnet.e4.rap">InitViewBacnet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewLeishmania.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewLeishmania.html" title="class in bacnet.e4.rap">InitViewLeishmania</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewListeria.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeria.html" title="class in bacnet.e4.rap">InitViewListeria</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewListeriaSample.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">InitViewListeriaSample</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewYersinia.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewYersinia.html" title="class in bacnet.e4.rap">InitViewYersinia</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SubmitDataView.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SubmitDataView.html" title="class in bacnet.e4.rap">SubmitDataView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GenomicsView.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#setFocused(boolean)">setFocused(boolean)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setFontDouble(java.awt.Font)">setFontDouble(Font)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setFontDouble(org.eclipse.swt.graphics.Font)">setFontDouble(Font)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setFontRowName(org.eclipse.swt.graphics.Font)">setFontRowName(Font)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCPList.html#setFontsDouble(java.awt.Font)">setFontsDouble(Font)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperList.html#setFontsDouble(org.eclipse.swt.graphics.Font)">setFontsDouble(Font)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCPList.html#setFontsString(java.awt.Font)">setFontsString(Font)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperList.html#setFontsString(org.eclipse.swt.graphics.Font)">setFontsString(Font)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setFontString(java.awt.Font)">setFontString(Font)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setFontText(org.eclipse.swt.graphics.Font)">setFontText(Font)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#setFoundIn(java.util.ArrayList)">setFoundIn(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setGeneExprs(java.util.ArrayList)">setGeneExprs(ArrayList&lt;GeneExpression&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setGeneListEGDe(java.util.ArrayList)">setGeneListEGDe(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#setGeneModification(java.util.HashMap)">setGeneModification(HashMap&lt;String, ArrayList&lt;NTerm.TypeModif&gt;&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#setGeneName(java.lang.String)">setGeneName(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setGeneNameToLocusTagMap(java.util.LinkedHashMap)">setGeneNameToLocusTagMap(LinkedHashMap&lt;String, String&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setGeneNameToLocusTagMap(java.util.LinkedHashMap)">setGeneNameToLocusTagMap(LinkedHashMap&lt;String, String&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setGeneralExperiment(bacnet.datamodel.expdesign.Experiment)">setGeneralExperiment(Experiment)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SynTView.html#setGeneralNetwork(bacnet.datamodel.dataset.Network)">setGeneralNetwork(Network)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#setGeneralNetwork(bacnet.datamodel.dataset.Network)">setGeneralNetwork(Network)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#setGenes(java.util.HashMap)">setGenes(HashMap&lt;String, ArrayList&lt;NTerm&gt;&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setGenes(java.util.LinkedHashMap)">setGenes(LinkedHashMap&lt;String, Gene&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Operon.html#setGenes(java.util.TreeSet)">setGenes(TreeSet&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setGenesAlternative(java.util.LinkedHashMap)">setGenesAlternative(LinkedHashMap&lt;String, Gene&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#setGeneSequence()">setGeneSequence()</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Create a fasta file with all the gene sequences</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#setGenome(bacnet.datamodel.sequence.Genome)">setGenome(Genome)</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#setGenome(bacnet.datamodel.sequence.Genome)">setGenome(Genome)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setGenome(bacnet.datamodel.sequence.Genome)">setGenome(Genome)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setGenome(bacnet.datamodel.sequence.Genome)">setGenome(Genome)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setGenome(bacnet.datamodel.sequence.Genome)">setGenome(Genome)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#setGenome(bacnet.datamodel.sequence.Genome)">setGenome(Genome)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#setGenome(java.lang.String)">setGenome(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#setGenome(java.lang.String)">setGenome(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setGenomeAA(java.lang.String)">setGenomeAA(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setGenomeAACompl(java.lang.String)">setGenomeAACompl(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setGenomeAAMinus(java.lang.String)">setGenomeAAMinus(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setGenomeAAMinusCompl(java.lang.String)">setGenomeAAMinusCompl(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setGenomeAAPlus(java.lang.String)">setGenomeAAPlus(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setGenomeAAPlusCompl(java.lang.String)">setGenomeAAPlusCompl(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setGenomeArrayPath(java.lang.String)">setGenomeArrayPath(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#setGenomeElements(java.util.ArrayList)">setGenomeElements(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#setGenomeElements(java.util.ArrayList)">setGenomeElements(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setGenomeList(java.util.ArrayList)">setGenomeList(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#setGenomeName(java.lang.String)">setGenomeName(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NGS.html#setGenomeName(java.lang.String)">setGenomeName(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setGenomeName(java.lang.String)">setGenomeName(String)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setGenomeName(java.lang.String)">setGenomeName(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SynTView.html#setGenomeName(java.lang.String)">setGenomeName(String)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setGenomeName(java.lang.String)">setGenomeName(String)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#setGenomeName(java.lang.String)">setGenomeName(String)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#setGenomeName(java.lang.String)">setGenomeName(String)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setGenomeName(java.lang.String)">setGenomeName(String)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#setGenomeName(java.lang.String)">setGenomeName(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setGenomeName(java.lang.String)">setGenomeName(String)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#setGenomeName(java.lang.String)">setGenomeName(String)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#setGenomeNCBILink(java.lang.String)">setGenomeNCBILink(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DataValidation.html#setGenomes(java.util.HashMap)">setGenomes(HashMap&lt;String, Boolean&gt;)</a></span> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setGenomes(java.util.TreeMap)">setGenomes(TreeMap&lt;String, Genome&gt;)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setGenomeSelected(java.lang.String)">setGenomeSelected(String)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">The comboGenome contains modified genome name so we need this method to
 select an element<br>
 a '*' is add to genome name when a transcriptome data is available</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setGenomesNCBI(java.util.TreeMap)">setGenomesNCBI(TreeMap&lt;String, GenomeNCBI&gt;)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setGenomeUsed(java.lang.String)">setGenomeUsed(String)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setGoogleId(java.lang.String)">setGoogleId(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorSWT.html#setGreen(int)">setGreen(int)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setGrowth(java.util.ArrayList)">setGrowth(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setHeader(java.lang.String,int)">setHeader(String, int)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setHeaderAnnotation(java.util.ArrayList)">setHeaderAnnotation(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setHeaderColor(org.eclipse.swt.graphics.Color)">setHeaderColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setHeaderColorSWT(bacnet.table.core.ColorSWT)">setHeaderColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setHeaderColorSWT(bacnet.table.core.ColorSWT)">setHeaderColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setHeaders(java.util.ArrayList)">setHeaders(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#setHeight(int)">setHeight(int)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/MatrixSize.html#setHeight(java.util.HashMap)">setHeight(HashMap&lt;String, Integer&gt;)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/MatrixSize.html#setHeightDefault(int)">setHeightDefault(int)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setHeightPix(int)">setHeightPix(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setHeightPix(int)">setHeightPix(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setHorizontalBarProperties()">setHorizontalBarProperties()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Depending on <code>Track</code> parameters (<code>Region</code> and
 <code>Zoom</code>) fix properties of HorizontalBar</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#setId(java.lang.String)">setId(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setId(java.lang.String)">setId(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#setId(java.lang.String)">setId(String)</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#setID(java.lang.String)">setID(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/ColorMapperWizardPage.html#setInfo()">setInfo()</a></span> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#setInfoRead(boolean)">setInfoRead(boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NGS.html#setInfoRead(boolean)">setInfoRead(boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#setInitialized(boolean)">setInitialized(boolean)</a></span> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setInitView(java.lang.String)">setInitView(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setInsoluble(boolean)">setInsoluble(boolean)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setIntersect(boolean)">setIntersect(boolean)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setLabel_2(org.eclipse.swt.widgets.Label)">setLabel_2(Label)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setLabel_3(org.eclipse.swt.widgets.Label)">setLabel_3(Label)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setLblDescribedIn(org.eclipse.swt.widgets.Label)">setLblDescribedIn(Label)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setLblGene(org.eclipse.swt.widgets.Label)">setLblGene(Label)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setLblNodiff(org.eclipse.swt.widgets.Label)">setLblNodiff(Label)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setLblOrAnyOther(org.eclipse.swt.widgets.Label)">setLblOrAnyOther(Label)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setLblOver(org.eclipse.swt.widgets.Label)">setLblOver(Label)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setLblOverProteomes(org.eclipse.swt.widgets.Label)">setLblOverProteomes(Label)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setLblPredictedSecondaryStructure(org.eclipse.swt.widgets.Label)">setLblPredictedSecondaryStructure(Label)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setLblStructure(org.eclipse.swt.widgets.Label)">setLblStructure(Label)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setLblTranscriptomesData(org.eclipse.swt.widgets.Label)">setLblTranscriptomesData(Label)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setLblUnder(org.eclipse.swt.widgets.Label)">setLblUnder(Label)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#setLeftBCs(java.util.ArrayList)">setLeftBCs(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#setLegend()">setLegend()</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#setLength(int)">setLength(int)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setLength(int)">setLength(int)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#setLengthAA(int)">setLengthAA(int)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html#setLineLength(int)">setLineLength(int)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FastaWriter.html#setLineLength(int)">setLineLength(int)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FastaWriter.html#setLineSeparator(java.lang.String)">setLineSeparator(String)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a></dt>
<dd>
<div class="block">Allow an override of operating system line separator for programs that needs
 a specific CRLF or CR or LF option</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setLinkedBioCondition(java.util.ArrayList)">setLinkedBioCondition(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setListDatabases(java.util.HashMap)">setListDatabases(HashMap&lt;String, String&gt;)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setListGenes(java.util.ArrayList)">setListGenes(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setListSrnas(java.util.ArrayList)">setListSrnas(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setLoaded(boolean)">setLoaded(boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setLocalization(java.util.ArrayList)">setLocalization(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setLocation(org.biojava3.core.sequence.location.SequenceLocation)">setLocation(SequenceLocation&lt;S, C&gt;)</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">A feature could be a single sequence position like a mutation or a post
 translational modification of an amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setLocation(org.biojava3.core.sequence.location.SequenceLocation)">setLocation(SequenceLocation&lt;S, C&gt;)</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">The new location for this feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setLocusTagToGeneNameMap(java.util.LinkedHashMap)">setLocusTagToGeneNameMap(LinkedHashMap&lt;String, String&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setLocusTagToGeneNameMap(java.util.LinkedHashMap)">setLocusTagToGeneNameMap(LinkedHashMap&lt;String, String&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setLogfcOmicsMatrix(bacnet.datamodel.dataset.ExpressionMatrix)">setLogfcOmicsMatrix(ExpressionMatrix)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setLogFCTable(java.util.HashMap)">setLogFCTable(HashMap&lt;String, ExpressionMatrix&gt;)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setLogo(java.lang.String)">setLogo(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NGS.html#setLogTransformed(boolean)">setLogTransformed(boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#setLongName(java.lang.String)">setLongName(String)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#setLongName(java.lang.String)">setLongName(String)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Compound.html#setLongName(java.lang.String)">setLongName(String)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#setLongName(java.lang.String)">setLongName(String)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#setLongName(java.lang.String)">setLongName(String)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html#setLowercaseToNull(org.biojava3.core.sequence.ProteinSequence,java.lang.Object%5B%5D)">setLowercaseToNull(ProteinSequence, Object[])</a></span> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></dt>
<dd>
<div class="block">Takes a <a href="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence"><code>ProteinSequence</code></a> which was created by a
 <a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io"><code>CasePreservingProteinSequenceCreator</code></a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setMappingFrame(bacnet.datamodel.proteomics.NTerm.MappingFrame)">setMappingFrame(NTerm.MappingFrame)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#setMarkup(org.eclipse.swt.widgets.Composite)">setMarkup(Composite)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#setMarkup(org.eclipse.swt.widgets.Label)">setMarkup(Label)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setMassSpecData(bacnet.datamodel.dataset.NTermData)">setMassSpecData(NTermData)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setMatrices(java.util.ArrayList)">setMatrices(ArrayList&lt;ExpressionMatrix&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#setMatrix(bacnet.datamodel.dataset.ExpressionMatrix)">setMatrix(ExpressionMatrix)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setMatrix(bacnet.datamodel.dataset.ExpressionMatrix)">setMatrix(ExpressionMatrix)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#setMatrix(bacnet.datamodel.dataset.ExpressionMatrix)">setMatrix(ExpressionMatrix)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/FilterList.html#setMatrix(bacnet.datamodel.dataset.ExpressionMatrix)">setMatrix(ExpressionMatrix)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#setMatrix(bacnet.datamodel.dataset.ExpressionMatrix)">setMatrix(ExpressionMatrix)</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setMatrixDisplayed(bacnet.datamodel.dataset.ExpressionMatrix)">setMatrixDisplayed(ExpressionMatrix)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#setMax(double)">setMax(double)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMax(double)">setMax(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMax(double)">setMax(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMaxColor(org.eclipse.swt.graphics.Color)">setMaxColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMaxColor(org.eclipse.swt.graphics.Color)">setMaxColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMaxColorSWT(bacnet.table.core.ColorSWT)">setMaxColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMaxColorSWT(bacnet.table.core.ColorSWT)">setMaxColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMaxPos(double)">setMaxPos(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMaxPos(double)">setMaxPos(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setMaxQuant(java.lang.String)">setMaxQuant(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setMaxQuantum(java.lang.String%5B%5D)">setMaxQuantum(String[])</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#setMean(double)">setMean(double)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMean(double)">setMean(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMean(double)">setMean(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setMedia(java.util.ArrayList)">setMedia(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setMediaGrowthProperties(java.util.ArrayList)">setMediaGrowthProperties(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#setMedian(double)">setMedian(double)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMedian(double)">setMedian(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMedian(double)">setMedian(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#setMedianValue(java.lang.String,double)">setMedianValue(String, double)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Set the median value of corresponding probe in double[][] values</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#setMiddleH(int)">setMiddleH(int)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#setMiddleV(int)">setMiddleV(int)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMidLeftColor(org.eclipse.swt.graphics.Color)">setMidLeftColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMidLeftColor(org.eclipse.swt.graphics.Color)">setMidLeftColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMidLeftColorSWT(bacnet.table.core.ColorSWT)">setMidLeftColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMidLeftColorSWT(bacnet.table.core.ColorSWT)">setMidLeftColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMidLeftPos(double)">setMidLeftPos(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMidLeftPos(double)">setMidLeftPos(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMidRightColor(org.eclipse.swt.graphics.Color)">setMidRightColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMidRightColor(org.eclipse.swt.graphics.Color)">setMidRightColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMidRightColorSWT(bacnet.table.core.ColorSWT)">setMidRightColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMidRightColorSWT(bacnet.table.core.ColorSWT)">setMidRightColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMidRightPos(double)">setMidRightPos(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMidRightPos(double)">setMidRightPos(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#setMin(double)">setMin(double)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMin(double)">setMin(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMin(double)">setMin(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMinColor(org.eclipse.swt.graphics.Color)">setMinColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMinColor(org.eclipse.swt.graphics.Color)">setMinColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMinColorSWT(bacnet.table.core.ColorSWT)">setMinColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMinColorSWT(bacnet.table.core.ColorSWT)">setMinColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#setMinimumAlpha(org.eclipse.swt.graphics.GC)">setMinimumAlpha(GC)</a></span> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Set Alpha to 15</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setMinPos(double)">setMinPos(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setMinPos(double)">setMinPos(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#setModifications(java.util.HashMap)">setModifications(HashMap&lt;String, ArrayList&lt;NTerm&gt;&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#setModifs(java.util.ArrayList)">setModifs(ArrayList&lt;NTerm.TypeModif&gt;)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setModifSequence(java.lang.String)">setModifSequence(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#setMolecularWeight(java.lang.Float)">setMolecularWeight(Float)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#setMolecularWeight(java.lang.Float)">setMolecularWeight(Float)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Compound.html#setMolecularWeight(java.lang.Float)">setMolecularWeight(Float)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#setMolecularWeight(java.lang.Float)">setMolecularWeight(Float)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#setMolecularWeight(java.lang.Float)">setMolecularWeight(Float)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#setMolMass(double)">setMolMass(double)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setMouseXPosition(int)">setMouseXPosition(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setMouseXPosition(int)">setMouseXPosition(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setMouseYPosition(int)">setMouseYPosition(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setMouseYPosition(int)">setMouseYPosition(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html#setMultiFasta(java.util.ArrayList)">setMultiFasta(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html" title="class in bacnet.datamodel.sequence">Genome.GetMultiFastaThread</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setMutant(java.util.ArrayList)">setMutant(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#setName(java.lang.String)">setName(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#setName(java.lang.String)">setName(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#setName(java.lang.String)">setName(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setName(java.lang.String)">setName(String)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/Experiment.html#setName(java.lang.String)">setName(String)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setName(java.lang.String)">setName(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setName(java.lang.String)">setName(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setName(java.lang.String)">setName(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setName(java.lang.String)">setName(String)</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#setName(java.lang.String)">setName(String)</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setName(java.lang.String)">setName(String)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#setName(java.lang.String)">setName(String)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#setName(java.lang.String)">setName(String)</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setNameNTermData(java.lang.String)">setNameNTermData(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setNameRawData(java.lang.String)">setNameRawData(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setNavigationListener(org.eclipse.rap.rwt.client.service.BrowserNavigationListener)">setNavigationListener(BrowserNavigationListener)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setNbElements(int)">setNbElements(int)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setNbElements(int)">setNbElements(int)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#setNbElementType(int)">setNbElementType(int)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#setNbSet(int)">setNbSet(int)</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setNcRNAs(java.util.LinkedHashMap)">setNcRNAs(LinkedHashMap&lt;String, NcRNA&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#setNewChromosome(java.lang.String)">setNewChromosome(String)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#setNewLocusTag(java.lang.String)">setNewLocusTag(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setNextCodon(java.lang.String)">setNextCodon(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setNGSSeqs(java.util.ArrayList)">setNGSSeqs(ArrayList&lt;NGS&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setNoData(boolean)">setNoData(boolean)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setNoDNAList(java.util.ArrayList)">setNoDNAList(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setNoncodingSequenceHashMap(java.util.LinkedHashMap)">setNoncodingSequenceHashMap(LinkedHashMap&lt;String, DNASequence&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setNoProtList(java.util.ArrayList)">setNoProtList(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setNoRNAList(java.util.ArrayList)">setNoRNAList(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setNotDiffExpressedASrnas(java.lang.String%5B%5D)">setNotDiffExpressedASrnas(String[])</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setNotDiffExpressedCisRegs(java.lang.String%5B%5D)">setNotDiffExpressedCisRegs(String[])</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setNotDiffExpressedGenes(java.lang.String%5B%5D)">setNotDiffExpressedGenes(String[])</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#setNotDiffExpresseds(java.util.TreeSet)">setNotDiffExpresseds(TreeSet&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setNotDiffExpressedSrnas(java.lang.String%5B%5D)">setNotDiffExpressedSrnas(String[])</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#setNote(java.lang.String)">setNote(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#setNote(java.lang.String)">setNote(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setNotesList(java.util.ArrayList)">setNotesList(ArrayList&lt;String&gt;)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermFilter.html#setnTermExp(bacnet.datamodel.dataset.NTermData)">setnTermExp(NTermData)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setnTermHighlight(bacnet.datamodel.proteomics.NTerm)">setnTermHighlight(NTerm)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CompositeNTermInfo.html#setNTermInfo(bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.dataset.NTermData)">setNTermInfo(NTerm, NTermData)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CompositeNTermInfo.html" title="class in bacnet.genomeBrowser">CompositeNTermInfo</a></dt>
<dd>
<div class="block">Set all Labels for NTerm info</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#setnTermRef(bacnet.datamodel.proteomics.NTerm)">setnTermRef(NTerm)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setnTerms(java.util.ArrayList)">setnTerms(ArrayList&lt;NTermData&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#setnTerms(java.util.ArrayList)">setnTerms(ArrayList&lt;NTerm&gt;)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setnTermTypeOverlap(java.lang.String)">setnTermTypeOverlap(String)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setNucleotide(boolean)">setNucleotide(boolean)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#setOldLocusTag(java.lang.String)">setOldLocusTag(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#setOperon(java.lang.String)">setOperon(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setOperons(java.util.ArrayList)">setOperons(ArrayList&lt;Operon&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setOperons(java.util.LinkedHashMap)">setOperons(LinkedHashMap&lt;String, Operon&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setOrdered(boolean)">setOrdered(boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setOrdered(boolean,java.util.ArrayList)">setOrdered(boolean, ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setOrderedRowNames(java.util.ArrayList)">setOrderedRowNames(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setOriginalHeader(java.lang.String)">setOriginalHeader(String)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setOverASrnas(java.lang.String%5B%5D)">setOverASrnas(String[])</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#setOverBioConds(java.util.TreeSet)">setOverBioConds(TreeSet&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setOverCisRegs(java.lang.String%5B%5D)">setOverCisRegs(String[])</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setOverGenes(java.lang.String%5B%5D)">setOverGenes(String[])</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setOverlap(java.lang.String)">setOverlap(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setOverlapInFrame(boolean)">setOverlapInFrame(boolean)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#setOverlapingGene(java.lang.String)">setOverlapingGene(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#setOverlaps(java.util.ArrayList)">setOverlaps(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setOverSrnas(java.lang.String%5B%5D)">setOverSrnas(String[])</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ProteinSequence.html#setParentDNASequence(org.biojava3.core.sequence.template.AbstractSequence,java.lang.Integer,java.lang.Integer)">setParentDNASequence(AbstractSequence, Integer, Integer)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></dt>
<dd>
<div class="block">A Protein sequence can be stand alone or loaded from a transcript sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setParentFeature(org.biojava3.core.sequence.features.FeatureInterface)">setParentFeature(FeatureInterface&lt;S, C&gt;)</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">A feature can be the child or contained by a parent feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setParentFeature(org.biojava3.core.sequence.features.FeatureInterface)">setParentFeature(FeatureInterface&lt;S, C&gt;)</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Set the parent feature</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setParentGenome(bacnet.datamodel.sequence.Genome)">setParentGenome(Genome)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setParentGenome(bacnet.datamodel.sequence.GenomeNCBI)">setParentGenome(GenomeNCBI)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)">setParentSequence(AbstractSequence&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setParentViewId(java.lang.String)">setParentViewId(String)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setPartService(org.eclipse.e4.ui.workbench.modeling.EPartService)">setPartService(EPartService)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setPath(java.lang.String)">setPath(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#setPATH(java.lang.String)">setPATH(String)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#setPeptides(java.util.HashMap)">setPeptides(HashMap&lt;String, ArrayList&lt;NTerm&gt;&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Link peptideSequence to <code>ArrayList(NTerm)</code><br>
 This correspond to the list of <code>NTerm</code> having same peptideSequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#setPreloadedItems(org.eclipse.swt.widgets.Composite)">setPreloadedItems(Composite)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setPreviousCodon(java.lang.String)">setPreviousCodon(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#setProbes(java.util.TreeMap)">setProbes(TreeMap&lt;Integer, Integer&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#setProbes(java.util.TreeMap)">setProbes(TreeMap&lt;String, Integer&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setProbesGExpression(java.util.TreeMap)">setProbesGExpression(TreeMap&lt;String, Integer&gt;)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setProbesTiling(java.util.TreeMap)">setProbesTiling(TreeMap&lt;Integer, Integer&gt;)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#setProduct(java.lang.String)">setProduct(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/NcRNA.html#setProduct(java.lang.String)">setProduct(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setProjectName(java.lang.String)">setProjectName(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#setProperties(java.util.LinkedHashMap)">setProperties(LinkedHashMap&lt;String, String&gt;)</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#setProtein_id(java.lang.String)">setProtein_id(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#setProteinNCBILink(java.lang.String)">setProteinNCBILink(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setProteomes(java.util.ArrayList)">setProteomes(ArrayList&lt;ProteomicsData&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DataValidation.html#setProteomes(java.util.HashMap)">setProteomes(HashMap&lt;String, Boolean&gt;)</a></span> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setProteomesArrayPath(java.lang.String)">setProteomesArrayPath(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)">setProxySequenceReader(SequenceReader&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Very important method that allows external mappings of sequence data and
 features.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#setPseudogene(boolean)">setPseudogene(boolean)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#setPubMedLink(java.lang.String)">setPubMedLink(String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/QualityFeature.html#setQualities(java.util.List)">setQualities(List&lt;Number&gt;)</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/QualityFeature.html" title="class in org.biojava3.core.sequence.features">QualityFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/QuantityFeature.html#setQuantities(java.util.List)">setQuantities(List&lt;Number&gt;)</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/QuantityFeature.html" title="class in org.biojava3.core.sequence.features">QuantityFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#setRawDatas(java.util.ArrayList)">setRawDatas(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#setRead(boolean%5B%5D)">setRead(boolean[])</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorSWT.html#setRed(int)">setRed(int)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setRef(java.lang.String)">setRef(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#setReference(java.lang.String)">setReference(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setReference(java.lang.String)">setReference(String)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#setRefSequence(java.lang.String)">setRefSequence(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/TestGenomeBrowser.html#setRelativeValueDisplay()">setRelativeValueDisplay()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/TestGenomeBrowser.html" title="class in bacnet.genomeBrowser">TestGenomeBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#setRightBCs(java.util.ArrayList)">setRightBCs(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setRowNameCellColor(org.eclipse.swt.graphics.Color)">setRowNameCellColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setRowNames(java.util.TreeMap)">setRowNames(TreeMap&lt;String, Integer&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setRowNameTextColor(org.eclipse.swt.graphics.Color)">setRowNameTextColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#setScore(double)">setScore(double)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setScore(int)">setScore(int)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#setScrolledComposite(org.eclipse.swt.custom.ScrolledComposite)">setScrolledComposite(ScrolledComposite)</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setScrolledComposite(org.eclipse.swt.custom.ScrolledComposite)">setScrolledComposite(ScrolledComposite)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#setSDBindingEnergy()">setSDBindingEnergy()</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">For every gene calculate the SD-aSD hybridization affinity and save it in:
 /StatGenes/aSDBindingEneryg.excel</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/TableFilter.html#setSearchText(java.lang.String)">setSearchText(String)</a></span> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/TableFilter.html" title="class in bacnet.table.gui">TableFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setSecondRowName(java.lang.String)">setSecondRowName(String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#setSelectedGenes(java.util.ArrayList)">setSelectedGenes(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setSeq(bacnet.datamodel.sequence.Srna)">setSeq(Srna)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setSequence(bacnet.datamodel.sequence.Gene)">setSequence(Gene)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setSequence(java.lang.String)">setSequence(String)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#setSequence(java.lang.String)">setSequence(String)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html#setSequenceCollection(org.biojava3.core.sequence.SequenceOptimizationHints.SequenceCollection)">setSequenceCollection(SequenceOptimizationHints.SequenceCollection)</a></span> - Static method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setSequenceMap(java.lang.String)">setSequenceMap(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setSequencePeptide(java.lang.String)">setSequencePeptide(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setSequenceScore(java.lang.Double)">setSequenceScore(Double)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html#setSequenceUsage(org.biojava3.core.sequence.SequenceOptimizationHints.SequenceUsage)">setSequenceUsage(SequenceOptimizationHints.SequenceUsage)</a></span> - Static method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#setSetNames(java.util.ArrayList)">setSetNames(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#setSets(java.util.TreeMap)">setSets(TreeMap&lt;String, ArrayList&lt;String&gt;&gt;)</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setShell(org.eclipse.swt.widgets.Shell)">setShell(Shell)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setShortDescription(java.lang.String)">setShortDescription(String)</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setShortDescription(java.lang.String)">setShortDescription(String)</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Set the short description that can be used to describe the feature</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#setShortName(java.lang.String)">setShortName(String)</a></span> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#setShortName(java.lang.String)">setShortName(String)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/Compound.html#setShortName(java.lang.String)">setShortName(String)</a></span> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#setShortName(java.lang.String)">setShortName(String)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#setShortName(java.lang.String)">setShortName(String)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/SignatureSelectionDialog.html#setSignature(bacnet.datamodel.annotation.Signature)">setSignature(Signature)</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/SignatureSelectionDialog.html" title="class in bacnet.expressionAtlas.core">SignatureSelectionDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setSignatures(java.util.TreeSet)">setSignatures(TreeSet&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setSignaturesNametoID(java.util.TreeMap)">setSignaturesNametoID(TreeMap&lt;String, String&gt;)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#setSize(int)">setSize(int)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setSoluble(boolean)">setSoluble(boolean)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setSource(java.lang.String)">setSource(String)</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">Set the feature source</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setSource(java.lang.String)">setSource(String)</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Change the source of the FeatureInterface.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setSource(java.lang.String)">setSource(String)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Added support for the source of this sequence for GFF3 export</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setSpecies(java.lang.String)">setSpecies(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#setSpecies(java.lang.String)">setSpecies(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#setSpecies(java.lang.String)">setSpecies(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/GenomeNCBI.html#setSpecies(java.lang.String)">setSpecies(String)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/GenomeNCBI.html" title="class in bacnet.datamodel.sequence">GenomeNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setSpectra(int)">setSpectra(int)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setSpectrum(int%5B%5D)">setSpectrum(int[])</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#setSrnaLink(java.lang.String,java.util.HashMap)">setSrnaLink(String, HashMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setsRNAListEGDe(java.util.ArrayList)">setsRNAListEGDe(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#setSrnaOrder()">setSrnaOrder()</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Classify Srna par IDs and save it in PATH_LISTSRNA</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setsRNAs(java.util.ArrayList)">setsRNAs(ArrayList&lt;Srna&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#setsRNAs(java.util.LinkedHashMap)">setsRNAs(LinkedHashMap&lt;String, Srna&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setStartCode(java.lang.String)">setStartCode(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setStartCodon(java.lang.String)">setStartCodon(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#setStat()">setStat()</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Set statistical parameters using values array</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setStat()">setStat()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setStat()">setStat()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setStatDataColor()">setStatDataColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setStatDataColor()">setStatDataColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setStatDataValues()">setStatDataValues()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setStatDataValues()">setStatDataValues()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setStatMapper()">setStatMapper()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>
<div class="block">Set parameters corresponding to stat data i.e. value from 0 to 1</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setStatMapper()">setStatMapper()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>
<div class="block">Set parameters corresponding to stat data i.e. value from 0 to 1</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setStatTable(bacnet.datamodel.dataset.ExpressionMatrix)">setStatTable(ExpressionMatrix)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setStdevDataColor()">setStdevDataColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setStdevDataColor()">setStdevDataColor()</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#setStrand(char)">setStrand(char)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setStrand(char)">setStrand(char)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setStrand(org.biojava3.core.sequence.Strand)">setStrand(Strand)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/GeneSequence.html#setStrand(org.biojava3.core.sequence.Strand)">setStrand(Strand)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setStudyName(java.lang.String)">setStudyName(String)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#setSubsetNames(java.util.ArrayList)">setSubsetNames(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#setSubsets(java.util.TreeMap)">setSubsets(TreeMap&lt;String, ArrayList&lt;String&gt;&gt;)</a></span> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setSynonym(java.util.TreeSet)">setSynonym(TreeSet&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setTabFolder(org.eclipse.swt.widgets.TabFolder)">setTabFolder(TabFolder)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setTabFolder(org.eclipse.swt.widgets.TabFolder)">setTabFolder(TabFolder)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setTabFolder(org.eclipse.swt.widgets.TabFolder)">setTabFolder(TabFolder)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#setTabFolder(org.eclipse.swt.widgets.TabFolder)">setTabFolder(TabFolder)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#setTable(org.eclipse.swt.widgets.Table)">setTable(Table)</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#setTableBioCondition(org.eclipse.jface.viewers.TableViewer)">setTableBioCondition(TableViewer)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#setTableBioCondition(org.eclipse.jface.viewers.TableViewer)">setTableBioCondition(TableViewer)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#setTableBioCondition(org.eclipse.jface.viewers.TableViewer)">setTableBioCondition(TableViewer)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#setTableBioCondition(org.eclipse.swt.widgets.Table)">setTableBioCondition(Table)</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setTableComposite(bacnet.expressionAtlas.TableCompositeHeatMap)">setTableComposite(TableCompositeHeatMap)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#setTableComposite(bacnet.expressionAtlas.TableCompositeHeatMap)">setTableComposite(TableCompositeHeatMap)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#setTableComposite(bacnet.expressionAtlas.TableCompositeHeatMap)">setTableComposite(TableCompositeHeatMap)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/TableSWTView.html#setTableComposite(bacnet.table.TableSWTComposite)">setTableComposite(TableSWTComposite)</a></span> - Method in class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#setTableElementName(java.lang.String)">setTableElementName(String)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#setTableElementName(java.lang.String)">setTableElementName(String)</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setTableGenes(org.eclipse.swt.widgets.Table)">setTableGenes(Table)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#setTableGenes(org.eclipse.swt.widgets.Table)">setTableGenes(Table)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setTableNodiff(org.eclipse.swt.widgets.Table)">setTableNodiff(Table)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setTableOtherInfo(org.eclipse.swt.widgets.Table)">setTableOtherInfo(Table)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setTableOver(org.eclipse.swt.widgets.Table)">setTableOver(Table)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setTableProteomes(org.eclipse.swt.widgets.Table)">setTableProteomes(Table)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setTableSrnas(org.eclipse.swt.widgets.Table)">setTableSrnas(Table)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setTableUnder(org.eclipse.swt.widgets.Table)">setTableUnder(Table)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setTaxonomy(org.biojava3.core.sequence.TaxonomyID)">setTaxonomy(TaxonomyID)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#setTbtmGraphVisualization(org.eclipse.swt.widgets.TabItem)">setTbtmGraphVisualization(TabItem)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setTbtmHeatmap(org.eclipse.swt.widgets.TabItem)">setTbtmHeatmap(TabItem)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#setTbtmRadialNetwork(org.eclipse.swt.widgets.TabItem)">setTbtmRadialNetwork(TabItem)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setTbtmScatterPlot(org.eclipse.swt.widgets.TabItem)">setTbtmScatterPlot(TabItem)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setTemperature(java.lang.String)">setTemperature(String)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setTestData(boolean)">setTestData(boolean)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#setTests(java.util.ArrayList)">setTests(ArrayList&lt;StatTest&gt;)</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/DownloadServiceHandler.html#setText(java.lang.String)">setText(String)</a></span> - Method in class bacnet.rap.<a href="../bacnet/rap/DownloadServiceHandler.html" title="class in bacnet.rap">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/DownloadServiceHandler.html#setText(java.lang.String)">setText(String)</a></span> - Method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setTextColor(org.eclipse.swt.graphics.Color)">setTextColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setTextColor(org.eclipse.swt.graphics.Color)">setTextColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setTextColorSWT(bacnet.table.core.ColorSWT)">setTextColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setTextColorSWT(bacnet.table.core.ColorSWT)">setTextColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#setTextCutOff(org.eclipse.swt.widgets.Text)">setTextCutOff(Text)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setTextHeaderColor(org.eclipse.swt.graphics.Color)">setTextHeaderColor(Color)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setTextHeaderColorSWT(bacnet.table.core.ColorSWT)">setTextHeaderColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setTextHeaderColorSWT(bacnet.table.core.ColorSWT)">setTextHeaderColorSWT(ColorSWT)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/gui/BlastQueryPage.html#setTextQuery(org.eclipse.swt.widgets.Text)">setTextQuery(Text)</a></span> - Method in class bacnet.scripts.blast.gui.<a href="../bacnet/scripts/blast/gui/BlastQueryPage.html" title="class in bacnet.scripts.blast.gui">BlastQueryPage</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#setTextSearch(org.eclipse.swt.widgets.Text)">setTextSearch(Text)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setTextSearch(org.eclipse.swt.widgets.Text)">setTextSearch(Text)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPool(java.util.concurrent.ThreadPoolExecutor)">setThreadPool(ThreadPoolExecutor)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Sets thread pool to any given <code>ThreadPoolExecutor</code> to allow use of an
 alternative execution style.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPoolCPUsAvailable(int)">setThreadPoolCPUsAvailable(int)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Sets thread pool to reserve a given number of processor cores for foreground
 or other use.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPoolCPUsFraction(float)">setThreadPoolCPUsFraction(float)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Sets thread pool to a given fraction of the available processors.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPoolDefault()">setThreadPoolDefault()</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Sets thread pool to default of 1 background thread for each processor core.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPoolSingle()">setThreadPoolSingle()</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Sets thread pool to a single background thread.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPoolSize(int)">setThreadPoolSize(int)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Sets thread pool to given size.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setThreshold(int)">setThreshold(int)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#setTiling(boolean)">setTiling(boolean)</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setTilings(java.util.ArrayList)">setTilings(ArrayList&lt;Tiling&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setTime(java.lang.String)">setTime(String)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#setTIRSequence()">setTIRSequence()</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">We create a fasta file with all the TIR DNA sequence (-18 +63)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#setTIRStructureEnergy()">setTIRStructureEnergy()</a></span> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">For every gene calculate the 2nd structure free energy and save it a
 table<br>
 We calculate the energy for 40 different sequence: -20+40 to -20+80 and
 calculate the average energy</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#setTisList(java.util.ArrayList)">setTisList(ArrayList&lt;TIS&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#setTisMap(java.util.HashMap)">setTisMap(HashMap&lt;NTerm, TIS&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setTISName(java.lang.String)">setTISName(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/InternalBrowser.html#setTitle(java.lang.String)">setTitle(String)</a></span> - Method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/BannerView.html#setToolBar(org.eclipse.swt.widgets.ToolBar)">setToolBar(ToolBar)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BannerView.html" title="class in bacnet.e4.rap">BannerView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#setTotalBP(int)">setTotalBP(int)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#setTotalSpectra(int)">setTotalSpectra(int)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/DataTrack.html#setTrack(bacnet.genomeBrowser.core.Track)">setTrack(Track)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html#setTrack(bacnet.genomeBrowser.core.Track)">setTrack(Track)</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddProteomicsDataDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html#setTrack(bacnet.genomeBrowser.core.Track)">setTrack(Track)</a></span> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddTranscriptomicsDataDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#setTrack(bacnet.genomeBrowser.core.Track)">setTrack(Track)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setTrack(bacnet.genomeBrowser.core.Track)">setTrack(Track)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">Set the data of the Canvas, and implement the PaintListener
 
 Execute different tasks:<br>
 Fix the number of pixel for one base pair
 Set width and height parameters for data and annotation track Display in
 order:<br>
 different line and ticks for the base pair position, every display needed
 in the background
 display all the data
 display sequence if needed
 display annotation
 different genome name and mouse line in foreground</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setTrack(bacnet.genomeBrowser.core.Track)">setTrack(Track)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>
<div class="block">Set the data of the Canvas, and implement the PaintListener
 
 Execute different tasks:<br>
 Fix the number of pixel for one base pair
 Set width and height parameters for data and annotation track Display in
 order:<br>
 different line and ticks for the base pair position, every display needed
 in the background
 display all the data
 display sequence if needed
 display annotation
 different genome name and mouse line in foreground</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setTrack(bacnet.genomeBrowser.core.Track)">setTrack(Track)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setTrackComposite(org.eclipse.swt.widgets.Composite)">setTrackComposite(Composite)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setTrackGenome(bacnet.genomeBrowser.core.Track)">setTrackGenome(Track)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#setTracksComposite(bacnet.genomeBrowser.tracksGUI.TracksComposite)">setTracksComposite(TracksComposite)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DataValidation.html#setTranscriptomes(java.util.HashMap)">setTranscriptomes(HashMap&lt;String, Boolean&gt;)</a></span> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setTranscriptomesArrayPath(java.lang.String)">setTranscriptomesArrayPath(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setTranscriptomesComparisonsArrayPath(java.lang.String)">setTranscriptomesComparisonsArrayPath(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setTranscriptomics(boolean)">setTranscriptomics(boolean)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setTssDownCoverage(int)">setTssDownCoverage(int)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setTssDownDistance(int)">setTssDownDistance(int)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setTssDowntype(java.lang.String)">setTssDowntype(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setTssUpCoverage(int)">setTssUpCoverage(int)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setTssUpDistance(int)">setTssUpDistance(int)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setTssUptype(java.lang.String)">setTssUptype(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setTxtCutoffLogFC(org.eclipse.swt.widgets.Text)">setTxtCutoffLogFC(Text)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setTxtCutoffLogFC(org.eclipse.swt.widgets.Text)">setTxtCutoffLogFC(Text)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setTxtCutoffPvalue(org.eclipse.swt.widgets.Text)">setTxtCutoffPvalue(Text)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#setTxtSearch(org.eclipse.swt.widgets.Text)">setTxtSearch(Text)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#setTxtSearch(org.eclipse.swt.widgets.Text)">setTxtSearch(Text)</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#setTxtSearch(org.eclipse.swt.widgets.Text)">setTxtSearch(Text)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GenomicsView.html#setTxtSearch(org.eclipse.swt.widgets.Text)">setTxtSearch(Text)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setTxtSeqnucleotide(org.eclipse.swt.widgets.Text)">setTxtSeqnucleotide(Text)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#setType(bacnet.datamodel.dataset.OmicsData.TypeData)">setType(OmicsData.TypeData)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.html#setType(bacnet.datamodel.sequence.Sequence.SeqType)">setType(Sequence.SeqType)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/StatTest.html#setType(bacnet.scripts.core.stat.StatTest.TypeStat)">setType(StatTest.TypeStat)</a></span> - Method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.html" title="class in bacnet.scripts.core.stat">StatTest</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setType(bacnet.table.core.ColorMapper.TypeMapper)">setType(ColorMapper.TypeMapper)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setType(bacnet.table.core.ColorMapperRCP.Type)">setType(ColorMapperRCP.Type)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/RfamElement.html#setType(java.lang.String)">setType(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#setType(java.lang.String)">setType(String)</a></span> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setType(java.lang.String)">setType(String)</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">Set the feature type</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setType(java.lang.String)">setType(String)</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Change the type of this feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#setTypeData(bacnet.datamodel.dataset.OmicsData.TypeData)">setTypeData(OmicsData.TypeData)</a></span> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setTypeDataContained(java.util.TreeSet)">setTypeDataContained(TreeSet&lt;OmicsData.TypeData&gt;)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#setTypeFilter(bacnet.table.core.Filter.TypeFilter)">setTypeFilter(Filter.TypeFilter)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#setTypeFilter(bacnet.utils.Filter.TypeFilter)">setTypeFilter(Filter.TypeFilter)</a></span> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setTypeModif(bacnet.datamodel.proteomics.NTerm.TypeModif)">setTypeModif(NTerm.TypeModif)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/NcRNA.html#setTypeNcRNA(bacnet.datamodel.sequence.NcRNA.TypeNcRNA)">setTypeNcRNA(NcRNA.TypeNcRNA)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#setTypeOverlap(java.lang.String)">setTypeOverlap(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#setTypeOverlaps(java.util.ArrayList)">setTypeOverlaps(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermFilter.html#setTypeOverlaps(java.util.ArrayList)">setTypeOverlaps(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#setTypeSrna(bacnet.datamodel.sequence.Srna.TypeSrna)">setTypeSrna(Srna.TypeSrna)</a></span> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setUnderASrnas(java.lang.String%5B%5D)">setUnderASrnas(String[])</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#setUnderBioConds(java.util.TreeSet)">setUnderBioConds(TreeSet&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setUnderCisRegs(java.lang.String%5B%5D)">setUnderCisRegs(String[])</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setUnderGenes(java.lang.String%5B%5D)">setUnderGenes(String[])</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setUnderSrnas(java.lang.String%5B%5D)">setUnderSrnas(String[])</a></span> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#setUniprotbaseURL(java.lang.String)">setUniprotbaseURL(String)</a></span> - Static method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#setUniprotDirectoryCache(java.lang.String)">setUniprotDirectoryCache(String)</a></span> - Static method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setUpDNAList(java.util.ArrayList)">setUpDNAList(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/SetupPart.html" title="class in bacnet.e4.rap.setup"><span class="typeNameLink">SetupPart</span></a> - Class in <a href="../bacnet/e4/rap/setup/package-summary.html">bacnet.e4.rap.setup</a></dt>
<dd>
<div class="block">SetupPart for initialisation and creation of the Database</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/setup/SetupPart.html#%3Cinit%3E()">SetupPart()</a></span> - Constructor for class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SetupPart.html" title="class in bacnet.e4.rap.setup">SetupPart</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setUpProtList(java.util.ArrayList)">setUpProtList(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setUpRNAList(java.util.ArrayList)">setUpRNAList(ArrayList&lt;String&gt;)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setUserCollection(java.util.Collection)">setUserCollection(Collection&lt;Object&gt;)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setUserObject(java.lang.Object)">setUserObject(Object)</a></span> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">Allow the user to associate an object with the feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setUserObject(java.lang.Object)">setUserObject(Object)</a></span> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setValue(double,int,int)">setValue(double, int, int)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setValue(double,java.lang.String,java.lang.String)">setValue(double, String, String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#setValue(int,double)">setValue(int, double)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setValueAnnotation(java.lang.String,java.lang.String,java.lang.String)">setValueAnnotation(String, String, String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#setValues(double%5B%5D)">setValues(double[])</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setValues(double%5B%5D)">setValues(double[])</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setValues(double%5B%5D)">setValues(double[])</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setValues(double%5B%5D%5B%5D)">setValues(double[][])</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setValues(int,int)">setValues(int, int)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Create new double[][] values with nbRow and nbColumn All value will then be
 equal to 0</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setValues(bacnet.datamodel.dataset.ExpressionMatrix)">setValues(ExpressionMatrix)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setValues(bacnet.datamodel.dataset.ExpressionMatrix)">setValues(ExpressionMatrix)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#setVar(double)">setVar(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#setVar(double)">setVar(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setVariables()">setVariables()</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">MOST IMPORTANT INIT METHOD : Set all variables for the website</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#setVariance(double)">setVariance(double)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setVerticalBarProperties()">setVerticalBarProperties()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Depending on <code>CanvasData</code> parameters fix properties of VerticalBar</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Network.html#setVertices(java.util.HashMap)">setVertices(HashMap&lt;String, String&gt;)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>
<div class="block">List of vertices = genome elements + their attributes = genome position,
 type, annotation</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setViewId(java.lang.String)">setViewId(String)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#setViewId(java.lang.String)">setViewId(String)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#setViewId(java.lang.String)">setViewId(String)</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#setViewId(java.lang.String)">setViewId(String)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#setViewID(java.lang.String)">setViewID(String)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#setViewID(java.lang.String)">setViewID(String)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#setWebpageTitle(java.lang.String)">setWebpageTitle(String)</a></span> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#setWidth(int)">setWidth(int)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/MatrixSize.html#setWidth(java.util.HashMap)">setWidth(HashMap&lt;String, Integer&gt;)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setWidthBP(int)">setWidthBP(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setWidthBP(int)">setWidthBP(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/MatrixSize.html#setWidthDefault(int)">setWidthDefault(int)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/MatrixSize.html#setWidthRowNameDefault(int)">setWidthRowNameDefault(int)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#setWildType(boolean)">setWildType(boolean)</a></span> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#setX1(int)">setX1(int)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#setX2(int)">setX2(int)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#setY1(int)">setY1(int)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#setY2(int)">setY2(int)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#setZoom(bacnet.genomeBrowser.core.Zoom)">setZoom(Zoom)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#setZoomChangedFlag(boolean)">setZoomChangedFlag(boolean)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#setZoomHeightMax(int)">setZoomHeightMax(int)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#setZoomHeightPosition(int)">setZoomHeightPosition(int)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#setZoomNumber(int)">setZoomNumber(int)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#setZoomPosition(int)">setZoomPosition(int)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#setZoomPosition(bacnet.genomeBrowser.core.Region)">setZoomPosition(Region)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>
<div class="block">Search the zoom position which correspond the best to displayRegion
 DisplayRegion should be inferior to display range given by the zoom</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#setZoomRatio(double)">setZoomRatio(double)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setZoomVertical(double)">setZoomVertical(double)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/GenomesCreation.html#showAllLocusID()">showAllLocusID()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/HelpPage.html#showHelp(java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)">showHelp(String, EPartService)</a></span> - Static method in class bacnet.views.<a href="../bacnet/views/HelpPage.html" title="class in bacnet.views">HelpPage</a></dt>
<dd>
<div class="block">Show the help page in an external browser</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#showInfoPanel()">showInfoPanel()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Show the info Panel showing NTerm data</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/Blast.html#showSearchPath()">showSearchPath()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#shuffle(org.biojava3.core.sequence.template.Sequence)">shuffle(Sequence&lt;C&gt;)</a></span> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Implements sequence shuffling by first materializing the given
 <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a> into a <code>List</code>, applying
 <code>Collections.shuffle(List)</code> and then returning the shuffled elements in
 a new instance of <code>SequenceBackingStore</code> which behaves as a
 <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a>.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/ConcurrencyTools.html#shutdown()">shutdown()</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Disables new tasks from being submitted and closes the thread pool cleanly.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/ConcurrencyTools.html#shutdownAndAwaitTermination()">shutdownAndAwaitTermination()</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Closes the thread pool.</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation"><span class="typeNameLink">Signature</span></a> - Class in <a href="../bacnet/datamodel/annotation/package-summary.html">bacnet.datamodel.annotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Signature.html#%3Cinit%3E()">Signature()</a></span> - Constructor for class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/SignatureSelectionDialog.html" title="class in bacnet.expressionAtlas.core"><span class="typeNameLink">SignatureSelectionDialog</span></a> - Class in <a href="../bacnet/expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/SignatureSelectionDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell)">SignatureSelectionDialog(Shell)</a></span> - Constructor for class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/SignatureSelectionDialog.html" title="class in bacnet.expressionAtlas.core">SignatureSelectionDialog</a></dt>
<dd>
<div class="block">Create the dialog.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location"><span class="typeNameLink">SimpleLocation</span></a> - Class in <a href="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</a></dt>
<dd>
<div class="block">Very basic implementation of the Location interface which defines a series of
 simple constructors.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(int,int)">SimpleLocation(int, int)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(int,int,org.biojava3.core.sequence.Strand)">SimpleLocation(int, int, Strand)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(int,int,org.biojava3.core.sequence.Strand,org.biojava3.core.sequence.location.template.Location...)">SimpleLocation(int, int, Strand, Location...)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point)">SimpleLocation(Point, Point)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand)">SimpleLocation(Point, Point, Strand)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,boolean)">SimpleLocation(Point, Point, Strand, boolean, boolean)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,boolean,java.util.List)">SimpleLocation(Point, Point, Strand, boolean, boolean, List&lt;Location&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,java.util.List)">SimpleLocation(Point, Point, Strand, boolean, List&lt;Location&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,org.biojava3.core.sequence.AccessionID)">SimpleLocation(Point, Point, Strand, boolean, AccessionID)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,org.biojava3.core.sequence.location.template.Location...)">SimpleLocation(Point, Point, Strand, boolean, Location...)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,org.biojava3.core.sequence.AccessionID)">SimpleLocation(Point, Point, Strand, AccessionID)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,org.biojava3.core.sequence.location.template.Location...)">SimpleLocation(Point, Point, Strand, Location...)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location"><span class="typeNameLink">SimplePoint</span></a> - Class in <a href="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</a></dt>
<dd>
<div class="block">Basic implementation of the Point interface.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimplePoint.html#%3Cinit%3E(int)">SimplePoint(int)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimplePoint.html#%3Cinit%3E(int,boolean,boolean)">SimplePoint(int, boolean, boolean)</a></span> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage"><span class="typeNameLink">SingleCompoundSequenceReader</span></a>&lt;<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</a></dt>
<dd>
<div class="block">An implementation of the SequenceReader interface which for every call will
 return only 1 compound (given to it during construction; a String is also
 valid but will require a CompoundSet).</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#%3Cinit%3E(C,org.biojava3.core.sequence.template.CompoundSet,int)">SingleCompoundSequenceReader(C, CompoundSet&lt;C&gt;, int)</a></span> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Build the object with a compound rather than a String</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,int)">SingleCompoundSequenceReader(String, CompoundSet&lt;C&gt;, int)</a></span> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Public constructor to be used with String based constructor</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView.html#SINGULARITY_PATH">SINGULARITY_PATH</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView2.html#SINGULARITY_PATH">SINGULARITY_PATH</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/UNAfold.html#sir_graph">sir_graph</a></span> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/UNAfold.html#sir_graph">sir_graph</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/FlatFileCache.html#size()">size()</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/FlatFileCache.html" title="class in org.biojava3.core.util">FlatFileCache</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/SoftHashMap.html#size()">size()</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#skip(long)">skip(long)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>
<div class="block">Skips characters.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/UncompressInputStream.html#skip(long)">skip(long)</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html" title="class in bacnet.scripts.listeriomics.srna"><span class="typeNameLink">SmallPeptidesSearch</span></a> - Class in <a href="../bacnet/scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html#%3Cinit%3E()">SmallPeptidesSearch()</a></span> - Constructor for class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html" title="class in bacnet.scripts.listeriomics.srna">SmallPeptidesSearch</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span></div>
&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#SMIRNOV">SMIRNOV</a></span> - bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util"><span class="typeNameLink">SoftHashMap</span></a>&lt;<a href="../org/biojava3/core/util/SoftHashMap.html" title="type parameter in SoftHashMap">K</a>,&#8203;<a href="../org/biojava3/core/util/SoftHashMap.html" title="type parameter in SoftHashMap">V</a>&gt; - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>
<div class="block">A in memory cache using soft references.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/SoftHashMap.html#%3Cinit%3E()">SoftHashMap()</a></span> - Constructor for class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/SoftHashMap.html#%3Cinit%3E(int)">SoftHashMap(int)</a></span> - Constructor for class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#sort(int)">sort(int)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#sort(int,boolean)">sort(int, boolean)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">TO IMPROVE<br>
 Sort double[][] values
 Using sorted array, reorder rows through OrderedRowNames</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#sortByStart()">sortByStart()</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a new list that is ordered by the starting index of the features'
 locations.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#sortColumn(double%5B%5D%5B%5D,int)">sortColumn(double[][], int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Sort an array using column given by the index</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#sortColumn(java.lang.String%5B%5D%5B%5D,int,boolean)">sortColumn(String[][], int, boolean)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Sort a column inside a String[][] array <br>
 <br>
 WARNING: if two values are the same in the column, it will display only the
 first one with this value in the sorted array</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Feature.html#source()">source()</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Get source (aka method).</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#SPEARCORR">SPEARCORR</a></span> - bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#splice(org.biojava3.core.sequence.DNASequence)">splice(DNASequence)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Concatenate successive portions of the specified sequence using the feature
 locations in the list.</div>
</dd>
<dt><a href="../org/biojava3/genome/util/SplitFasta.html" title="class in org.biojava3.genome.util"><span class="typeNameLink">SplitFasta</span></a> - Class in <a href="../org/biojava3/genome/util/package-summary.html">org.biojava3.genome.util</a></dt>
<dd>
<div class="block">Utility to write each Fasta entry to a unique file</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/util/SplitFasta.html#%3Cinit%3E()">SplitFasta()</a></span> - Constructor for class org.biojava3.genome.util.<a href="../org/biojava3/genome/util/SplitFasta.html" title="class in org.biojava3.genome.util">SplitFasta</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#square(double%5B%5D%5B%5D)">square(double[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Return array Y=(x_ij ^ 2)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#squareVector(double%5B%5D)">squareVector(double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Calcul of the square of a vector</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence"><span class="typeNameLink">Srna</span></a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Sequence.SeqType.html#Srna">Srna</a></span> - bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.TypeSrna.html#Srna">Srna</a></span> - bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.TypeSrna.html" title="enum in bacnet.datamodel.sequence">Srna.TypeSrna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#%3Cinit%3E()">Srna()</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#%3Cinit%3E(java.lang.String,int,int)">Srna(String, int, int)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#%3Cinit%3E(java.lang.String,int,int,char)">Srna(String, int, int, char)</a></span> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#SRNA_NUMBER">SRNA_NUMBER</a></span> - Static variable in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.TypeSrna.html" title="enum in bacnet.datamodel.sequence"><span class="typeNameLink">Srna.TypeSrna</span></a> - Enum in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html" title="class in bacnet.scripts.listeriomics.srna"><span class="typeNameLink">Srna10403SOliver</span></a> - Class in <a href="../bacnet/scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html#%3Cinit%3E()">Srna10403SOliver()</a></span> - Constructor for class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html" title="class in bacnet.scripts.listeriomics.srna">Srna10403SOliver</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna"><span class="typeNameLink">SrnaAlignmentBlastN</span></a> - Class in <a href="../bacnet/scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#%3Cinit%3E()">SrnaAlignmentBlastN()</a></span> - Constructor for class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#sRNACircularGenome(int,int,bacnet.datamodel.sequence.Genome,java.lang.String)">sRNACircularGenome(int, int, Genome, String)</a></span> - Method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html" title="class in bacnet.scripts.listeriomics.srna"><span class="typeNameLink">SrnaFinalList</span></a> - Class in <a href="../bacnet/scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html#%3Cinit%3E()">SrnaFinalList()</a></span> - Constructor for class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html" title="class in bacnet.scripts.listeriomics.srna">SrnaFinalList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/SrnaListeriomics.html" title="class in bacnet.scripts.listeriomics"><span class="typeNameLink">SrnaListeriomics</span></a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>
<div class="block">List of methods to manage Srna data for Listeriomics</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/SrnaListeriomics.html#%3Cinit%3E()">SrnaListeriomics()</a></span> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SrnaListeriomics.html" title="class in bacnet.scripts.listeriomics">SrnaListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#sRNAs">sRNAs</a></span> - Variable in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaSummaryView.html" title="class in bacnet.sequenceTools"><span class="typeNameLink">SrnaSummaryView</span></a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaSummaryView.html#%3Cinit%3E()">SrnaSummaryView()</a></span> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaSummaryView.html" title="class in bacnet.sequenceTools">SrnaSummaryView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna"><span class="typeNameLink">SrnaTables</span></a> - Class in <a href="../bacnet/scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a></dt>
<dd>
<div class="block">Include a list of Methods to read every Table containing Small RNAs in
 Listeria<br>
 For each it reads the <code>Table</code> and create a <code>List</code> of
 <code>Srna</code></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#%3Cinit%3E()">SrnaTables()</a></span> - Constructor for class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools"><span class="typeNameLink">SrnaView</span></a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#%3Cinit%3E()">SrnaView()</a></span> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#start()">start()</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Get starting index (origin 0).</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Activator.html#start(org.osgi.framework.BundleContext)">start(BundleContext)</a></span> - Method in class bacnet.<a href="../bacnet/Activator.html" title="class in bacnet">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/Activator.html#start(org.osgi.framework.BundleContext)">start(BundleContext)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/Activator.html" title="class in bacnet.e4.rap">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/setup/Activator.html#start(org.osgi.framework.BundleContext)">start(BundleContext)</a></span> - Method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/Activator.html" title="class in bacnet.e4.rap.setup">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/Activator.html#start(org.osgi.framework.BundleContext)">start(BundleContext)</a></span> - Method in class bacnet.scripts.<a href="../bacnet/scripts/Activator.html" title="class in bacnet.scripts">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Codon.html#startCodon">startCodon</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Codon.html#startCodon(java.lang.String)">startCodon(String)</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>
<div class="block">If it is a start codon return the corresponding codon, an empty
 <code>String</code> otherwise</div>
</dd>
<dt><a href="../bacnet/scripts/genome/StartCodonProfile.html" title="class in bacnet.scripts.genome"><span class="typeNameLink">StartCodonProfile</span></a> - Class in <a href="../bacnet/scripts/genome/package-summary.html">bacnet.scripts.genome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/StartCodonProfile.html#%3Cinit%3E()">StartCodonProfile()</a></span> - Constructor for class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/StartCodonProfile.html" title="class in bacnet.scripts.genome">StartCodonProfile</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/StartCodonSequence.html" title="class in org.biojava3.core.sequence"><span class="typeNameLink">StartCodonSequence</span></a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">Used to map the start codon feature on a gene</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/StartCodonSequence.html#%3Cinit%3E(org.biojava3.core.sequence.TranscriptSequence,int,int)">StartCodonSequence(TranscriptSequence, int, int)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/StartCodonSequence.html" title="class in org.biojava3.core.sequence">StartCodonSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#startsAfter(org.biojava3.genome.parsers.gff.Location)">startsAfter(Location)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Check if this location starts after the other location starts.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#startsBefore(org.biojava3.genome.parsers.gff.Location)">startsBefore(Location)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Check if this location starts before other location starts.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.TypeMapper.html#STAT">STAT</a></span> - bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.TypeMapper.html" title="enum in bacnet.table.core">ColorMapper.TypeMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.Type.html#STAT">STAT</a></span> - bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.Type.html" title="enum in bacnet.table.core">ColorMapperRCP.Type</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/ProteinTools.html#statDatabase()">statDatabase()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Extract statistics from existing databases, number of lmos, sRNA, utr...</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#statDatabase()">statDatabase()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Extract statistics from existing databases, number of lmos, sRNA, utr...</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#statInternalORF(bacnet.datamodel.sequence.Genome)">statInternalORF(Genome)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Given a genome extract main information such as:<br>
 number of start codons, stop codons
 number of high SD binding sites
 number of possible ORFs
 number of predicted genes<br>
 <br>
 From this metrics, predict the number of possible internal translation
 initiation site</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.html" title="class in bacnet.scripts.core.stat"><span class="typeNameLink">StatTest</span></a> - Class in <a href="../bacnet/scripts/core/stat/package-summary.html">bacnet.scripts.core.stat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/StatTest.html#%3Cinit%3E()">StatTest()</a></span> - Constructor for class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.html" title="class in bacnet.scripts.core.stat">StatTest</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/StatTest.html#%3Cinit%3E(bacnet.scripts.core.stat.StatTest.TypeStat)">StatTest(StatTest.TypeStat)</a></span> - Constructor for class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.html" title="class in bacnet.scripts.core.stat">StatTest</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat"><span class="typeNameLink">StatTest.TypeStat</span></a> - Enum in <a href="../bacnet/scripts/core/stat/package-summary.html">bacnet.scripts.core.stat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatUtils.html" title="class in bacnet.scripts.core.stat"><span class="typeNameLink">StatUtils</span></a> - Class in <a href="../bacnet/scripts/core/stat/package-summary.html">bacnet.scripts.core.stat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/StatUtils.html#%3Cinit%3E()">StatUtils()</a></span> - Constructor for class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatUtils.html" title="class in bacnet.scripts.core.stat">StatUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.TypeMapper.html#STDEV">STDEV</a></span> - bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.TypeMapper.html" title="enum in bacnet.table.core">ColorMapper.TypeMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.Type.html#STDEV">STDEV</a></span> - bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.Type.html" title="enum in bacnet.table.core">ColorMapperRCP.Type</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#STDEVWT">STDEVWT</a></span> - bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Activator.html#stop(org.osgi.framework.BundleContext)">stop(BundleContext)</a></span> - Method in class bacnet.<a href="../bacnet/Activator.html" title="class in bacnet">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/Activator.html#stop(org.osgi.framework.BundleContext)">stop(BundleContext)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/Activator.html" title="class in bacnet.e4.rap">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/setup/Activator.html#stop(org.osgi.framework.BundleContext)">stop(BundleContext)</a></span> - Method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/Activator.html" title="class in bacnet.e4.rap.setup">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/Activator.html#stop(org.osgi.framework.BundleContext)">stop(BundleContext)</a></span> - Method in class bacnet.scripts.<a href="../bacnet/scripts/Activator.html" title="class in bacnet.scripts">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/StopCodonSequence.html" title="class in org.biojava3.core.sequence"><span class="typeNameLink">StopCodonSequence</span></a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">Used to map the stop codon sequence on a gene</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/StopCodonSequence.html#%3Cinit%3E(org.biojava3.core.sequence.TranscriptSequence,int,int)">StopCodonSequence(TranscriptSequence, int, int)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/StopCodonSequence.html" title="class in org.biojava3.core.sequence">StopCodonSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/StringColor.html#str">str</a></span> - Variable in class bacnet.utils.<a href="../bacnet/utils/StringColor.html" title="class in bacnet.utils">StringColor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastResult.html#strand">strand</a></span> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence"><span class="typeNameLink">Strand</span></a> - Enum in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">Provides a way of representing the strand of a sequence, location hit or
 feature.</div>
</dd>
<dt><a href="../bacnet/utils/StringColor.html" title="class in bacnet.utils"><span class="typeNameLink">StringColor</span></a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/StringColor.html#%3Cinit%3E()">StringColor()</a></span> - Constructor for class bacnet.utils.<a href="../bacnet/utils/StringColor.html" title="class in bacnet.utils">StringColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/StringManipulationHelper.html" title="class in org.biojava3.core.util"><span class="typeNameLink">StringManipulationHelper</span></a> - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>
<div class="block">A utility class for common <code>String</code> manipulation tasks.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader"><span class="typeNameLink">StringProxySequenceReader</span></a>&lt;<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</a></dt>
<dd>
<div class="block">An example of a ProxySequenceReader that is created from a String.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">StringProxySequenceReader(String, CompoundSet&lt;C&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/rap/NavigationManagement.html#STYLE">STYLE</a></span> - Static variable in class bacnet.rap.<a href="../bacnet/rap/NavigationManagement.html" title="class in bacnet.rap">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/NavigationManagement.html#STYLE">STYLE</a></span> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceUsage.html#SUB_SEQUENCE_DATA">SUB_SEQUENCE_DATA</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceUsage.html" title="enum in org.biojava3.core.sequence">SequenceOptimizationHints.SequenceUsage</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#subArray(java.lang.String%5B%5D%5B%5D,int,int)">subArray(String[][], int, int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#subArray(java.lang.String%5B%5D%5B%5D,java.util.ArrayList,java.util.ArrayList)">subArray(String[][], ArrayList&lt;Integer&gt;, ArrayList&lt;Integer&gt;)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Return an array which correspond to the row and colum</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/SubCellCompartment.html" title="class in bacnet.datamodel.annotation"><span class="typeNameLink">SubCellCompartment</span></a> - Class in <a href="../bacnet/datamodel/annotation/package-summary.html">bacnet.datamodel.annotation</a></dt>
<dd>
<div class="block">All data and methods for Subcellular compartment information are done here</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/SubCellCompartment.html#%3Cinit%3E()">SubCellCompartment()</a></span> - Constructor for class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.html" title="class in bacnet.datamodel.annotation">SubCellCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation"><span class="typeNameLink">SubCellCompartment.TypeCompartment</span></a> - Enum in <a href="../bacnet/datamodel/annotation/package-summary.html">bacnet.datamodel.annotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/ConcurrencyTools.html#submit(java.util.concurrent.Callable)">submit(Callable&lt;T&gt;)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Queues up a task and adds a default log entry.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/ConcurrencyTools.html#submit(java.util.concurrent.Callable,java.lang.String)">submit(Callable&lt;T&gt;, String)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Queues up a task and adds a log entry.</div>
</dd>
<dt><a href="../bacnet/e4/rap/SubmitDataView.html" title="class in bacnet.e4.rap"><span class="typeNameLink">SubmitDataView</span></a> - Class in <a href="../bacnet/e4/rap/package-summary.html">bacnet.e4.rap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SubmitDataView.html#%3Cinit%3E()">SubmitDataView()</a></span> - Constructor for class bacnet.e4.rap.<a href="../bacnet/e4/rap/SubmitDataView.html" title="class in bacnet.e4.rap">SubmitDataView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/edits/Edit.Substitute.html#%3Cinit%3E(java.lang.String,int)">Substitute(String, int)</a></span> - Constructor for class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.Substitute.html" title="class in org.biojava3.core.sequence.edits">Edit.Substitute</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/edits/Edit.Substitute.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence,int)">Substitute(Sequence&lt;C&gt;, int)</a></span> - Constructor for class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.Substitute.html" title="class in org.biojava3.core.sequence.edits">Edit.Substitute</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#subVector(double%5B%5D,int,int)">subVector(double[], int, int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Return the subVector of vector between index begin and index end</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#suffix(int)">suffix(int)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">The part of this location after the specified position.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#suffix(org.biojava3.genome.parsers.gff.Location)">suffix(Location)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">The part of this location after the other location (not inclusive).</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#sum(double%5B%5D)">sum(double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Sum of all term in the vector</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#sum(double%5B%5D%5B%5D)">sum(double[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#summarize(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">summarize(Experiment, Genome)</a></span> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>
<div class="block">Create 3 ExpressionMatrix: <br>
 GeneExpression of genes on all BioCondition, saved in PATH_ALLGENEXPR
 Tiling of genes on all BioCondition, saved in PATH_ALLTILING_GENE
 Tiling of Srnas on all BioCondition, saved in PATH_ALLTILING_SRNA</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ComparisonsCreation.html#summarizeDiffExpr(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">summarizeDiffExpr(Experiment, Genome)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Create an array :<br>
 Column = comparison
 Row = Gene, Srna or Asrna
 Cell = "+" if a specific gene has been detected over-expressed, and "-"
 if under-expressed<br>
 <br>
 
 Save it in : Comparison.PATH_DIFF_LISTS+".txt"</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ComparisonsCreation.html#summarizeDiffLists(bacnet.datamodel.expdesign.Experiment)">summarizeDiffLists(Experiment)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Create an array :<br>
 Row = comparison
 Column = Number of gene, Srna Asrna over or under-expressed<br>
 <br>
 Save it in : Comparison.PATH_DIFF_LISTS+".txt"</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ProteomicsCreation.html#summarizeProteomes(bacnet.datamodel.expdesign.Experiment,java.lang.String)">summarizeProteomes(Experiment, String)</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>
<div class="block">Create a table will all protein expression for each Genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#summarizeResultsInTable()">summarizeResultsInTable()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Read all the XML results of Blast and create two PhyloXML files and a sRNA
 conservation table<br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html#summarizeResultsInTable(java.util.ArrayList)">summarizeResultsInTable(ArrayList&lt;Srna&gt;)</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html" title="class in bacnet.scripts.listeriomics.srna">Srna10403SOliver</a></dt>
<dd>
<div class="block">Read all the XML results of Blast and create a sRNA conservation table<br>
 header = {"Srna","EGD-e identites","EGD-e begin","EGD-e end","10403S
 identities","10403S begin","10403S end","10403S Average GEI","Delta_sigB
 Average GEI","FC range (10403S/Delta_sigB)"};</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#summarizeResultsInTable(java.util.ArrayList)">summarizeResultsInTable(ArrayList&lt;Srna&gt;)</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Read all the XML results of Blast and create two PhyloXML files and a sRNA
 conservation table<br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#summarizeResultsInTable10403S()">summarizeResultsInTable10403S()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Read all the XML results of Blast and create two PhyloXML files and a sRNA
 conservation table<br></div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html" title="class in bacnet.scripts.listeriomics"><span class="typeNameLink">SummaryListeriomics</span></a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>
<div class="block">Methods for creating the different figures and Tables of the paper</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html#%3Cinit%3E()">SummaryListeriomics()</a></span> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html" title="class in bacnet.scripts.listeriomics">SummaryListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.TypeFilter.html#SUPERIOR">SUPERIOR</a></span> - bacnet.table.core.<a href="../bacnet/table/core/Filter.TypeFilter.html" title="enum in bacnet.table.core">Filter.TypeFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.TypeFilter.html#SUPERIOR">SUPERIOR</a></span> - bacnet.utils.<a href="../bacnet/utils/Filter.TypeFilter.html" title="enum in bacnet.utils">Filter.TypeFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.TypeFilter.html#SUPERIOR_ABS">SUPERIOR_ABS</a></span> - bacnet.table.core.<a href="../bacnet/table/core/Filter.TypeFilter.html" title="enum in bacnet.table.core">Filter.TypeFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.TypeFilter.html#SUPERIOR_ABS">SUPERIOR_ABS</a></span> - bacnet.utils.<a href="../bacnet/utils/Filter.TypeFilter.html" title="enum in bacnet.utils">Filter.TypeFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/HTMLUtils.html#SVG">SVG</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/HTMLUtils.html" title="class in bacnet.utils">HTMLUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#swtColorToAwt(org.eclipse.swt.graphics.Color)">swtColorToAwt(Color)</a></span> - Static method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt"><span class="typeNameLink">SWTResourceManager</span></a> - Class in <a href="../bacnet/swt/package-summary.html">bacnet.swt</a></dt>
<dd>
<div class="block">Utility class for managing OS resources associated with SWT controls such as
 colors, fonts, images, etc.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#%3Cinit%3E()">SWTResourceManager()</a></span> - Constructor for class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ListUtils.html#symDifference(java.util.ArrayList,java.util.ArrayList)">symDifference(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ListUtils.html" title="class in bacnet.utils">ListUtils</a></dt>
<dd>
<div class="block">Calculate symmetrical difference of two lists L1 and L2 <br></div>
</dd>
<dt><a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap"><span class="typeNameLink">SynTView</span></a> - Class in <a href="../bacnet/e4/rap/package-summary.html">bacnet.e4.rap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SynTView.html#%3Cinit%3E()">SynTView()</a></span> - Constructor for class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/SystemsBiologyListeriomics.html" title="class in bacnet.scripts.listeriomics"><span class="typeNameLink">SystemsBiologyListeriomics</span></a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>
<div class="block">List of method to create Co-expression network and analyse it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/SystemsBiologyListeriomics.html#%3Cinit%3E()">SystemsBiologyListeriomics()</a></span> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SystemsBiologyListeriomics.html" title="class in bacnet.scripts.listeriomics">SystemsBiologyListeriomics</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a></div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
