<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>H-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="index: H">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:H">H</h2>
<dl class="index">
<dt><a href="../bacnet/table/gui/ColorMapperWizardPage.html#handleEvent(org.eclipse.swt.widgets.Event)" class="member-name-link">handleEvent(Event)</a> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#hasAttribute(java.lang.String)" class="member-name-link">hasAttribute(String)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureI.html#hasAttribute(java.lang.String)" class="member-name-link">hasAttribute(String)</a> - Method in interface org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a></dt>
<dd>
<div class="block">Check if the feature has a value defined for the specified key.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#hasAttribute(java.lang.String)" class="member-name-link">hasAttribute(String)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Check if any feature in list has the specified attribute key.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#hasAttribute(java.lang.String,java.lang.String)" class="member-name-link">hasAttribute(String, String)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureI.html#hasAttribute(java.lang.String,java.lang.String)" class="member-name-link">hasAttribute(String, String)</a> - Method in interface org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a></dt>
<dd>
<div class="block">Check if the feature attributes include the specified key/value pair.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#hasAttribute(java.lang.String,java.lang.String)" class="member-name-link">hasAttribute(String, String)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Check if any feature in list has the specified attribute key/value pair.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#hasCompound(C)" class="member-name-link">hasCompound(C)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/CompoundSet.html#hasCompound(C)" class="member-name-link">hasCompound(C)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html#hasCompound(org.biojava3.core.sequence.compound.AminoAcidCompound)" class="member-name-link">hasCompound(AminoAcidCompound)</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#hasGaps(int)" class="member-name-link">hasGaps(int)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Check size of gaps between successive features in list.</div>
</dd>
<dt><a href="../org/biojava3/core/util/Hashcoder.html#hash(int,boolean)" class="member-name-link">hash(int, boolean)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Hashcoder.html" title="class in org.biojava3.core.util">Hashcoder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/Hashcoder.html#hash(int,char)" class="member-name-link">hash(int, char)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Hashcoder.html" title="class in org.biojava3.core.util">Hashcoder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/Hashcoder.html#hash(int,double)" class="member-name-link">hash(int, double)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Hashcoder.html" title="class in org.biojava3.core.util">Hashcoder</a></dt>
<dd>
<div class="block">double support which is done using the same techinque as float hashing except we convert to a
 long not to an int.</div>
</dd>
<dt><a href="../org/biojava3/core/util/Hashcoder.html#hash(int,float)" class="member-name-link">hash(int, float)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Hashcoder.html" title="class in org.biojava3.core.util">Hashcoder</a></dt>
<dd>
<div class="block">float support done via <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Float.html#floatToIntBits(float)" title="class or interface in java.lang" class="external-link"><code>Float.floatToIntBits(float)</code></a> which allows the encoding of a float
 as an int.</div>
</dd>
<dt><a href="../org/biojava3/core/util/Hashcoder.html#hash(int,int)" class="member-name-link">hash(int, int)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Hashcoder.html" title="class in org.biojava3.core.util">Hashcoder</a></dt>
<dd>
<div class="block">Used for ints, bytes and shorts</div>
</dd>
<dt><a href="../org/biojava3/core/util/Hashcoder.html#hash(int,long)" class="member-name-link">hash(int, long)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Hashcoder.html" title="class in org.biojava3.core.util">Hashcoder</a></dt>
<dd>
<div class="block">long support done by shifting by 32 (using unsigned shift)</div>
</dd>
<dt><a href="../org/biojava3/core/util/Hashcoder.html#hash(int,java.lang.Object)" class="member-name-link">hash(int, Object)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Hashcoder.html" title="class in org.biojava3.core.util">Hashcoder</a></dt>
<dd>
<div class="block"><code>o</code> is a possibly-null object field, and possibly an array.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/AccessionID.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/AminoAcidCompound.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimplePoint.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#hashCode()" class="member-name-link">hashCode()</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/Hashcoder.html" class="type-name-link" title="class in org.biojava3.core.util">Hashcoder</a> - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>
<div class="block">Contains helper methods for generating a HashCode without having to resort to the commons lang
 hashcode builders.</div>
</dd>
<dt><a href="../org/biojava3/core/util/Hashcoder.html#%3Cinit%3E()" class="member-name-link">Hashcoder()</a> - Constructor for class org.biojava3.core.util.<a href="../org/biojava3/core/util/Hashcoder.html" title="class in org.biojava3.core.util">Hashcoder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html#hasNext()" class="member-name-link">hasNext()</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/CombinationGenerator.html" title="class in bacnet.scripts.core.vennDiagram">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/CombinationGenerator.html#hasNext()" class="member-name-link">hasNext()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/CombinationGenerator.html" title="class in bacnet.utils">CombinationGenerator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html#hasNext()" class="member-name-link">hasNext()</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="class in org.biojava3.core.sequence.template">SequenceMixin.SequenceIterator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/LocIterator.html#hasNext()" class="member-name-link">hasNext()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></dt>
<dd>
<div class="block">Check if next window of default size is available.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/LocIterator.html#hasNext(int,int)" class="member-name-link">hasNext(int, int)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></dt>
<dd>
<div class="block">Check if next window of specified size is available.</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#HEADER" class="member-name-link">HEADER</a> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/exceptions/HeaderParseException.html" class="type-name-link" title="class in org.biojava3.core.exceptions">HeaderParseException</a> - Error in <a href="../org/biojava3/core/exceptions/package-summary.html">org.biojava3.core.exceptions</a></dt>
<dd>
<div class="block">Currently not used</div>
</dd>
<dt><a href="../org/biojava3/core/exceptions/HeaderParseException.html#%3Cinit%3E(java.lang.String)" class="member-name-link">HeaderParseException(String)</a> - Constructor for error org.biojava3.core.exceptions.<a href="../org/biojava3/core/exceptions/HeaderParseException.html" title="class in org.biojava3.core.exceptions">HeaderParseException</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#HEATMAP_NAME" class="member-name-link">HEATMAP_NAME</a> - Static variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" class="type-name-link" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a> - Class in <a href="../bacnet/expressionAtlas/package-summary.html">bacnet.expressionAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#%3Cinit%3E()" class="member-name-link">HeatMapMultiOmicsView()</a> - Constructor for class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.InitOmicsMatrixThread.html" class="type-name-link" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView.InitOmicsMatrixThread</a> - Class in <a href="../bacnet/expressionAtlas/package-summary.html">bacnet.expressionAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" class="type-name-link" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a> - Class in <a href="../bacnet/expressionAtlas/package-summary.html">bacnet.expressionAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#%3Cinit%3E()" class="member-name-link">HeatMapProteomicsExpressionView()</a> - Constructor for class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" class="type-name-link" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a> - Class in <a href="../bacnet/expressionAtlas/package-summary.html">bacnet.expressionAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#%3Cinit%3E()" class="member-name-link">HeatMapProteomicsView()</a> - Constructor for class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" class="type-name-link" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a> - Class in <a href="../bacnet/expressionAtlas/package-summary.html">bacnet.expressionAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#%3Cinit%3E()" class="member-name-link">HeatMapTranscriptomicsView()</a> - Constructor for class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#HEIGHT_BPPOSITION" class="member-name-link">HEIGHT_BPPOSITION</a> - Static variable in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>
<div class="block">Height of the bp position track</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#HEIGHT_SEQUENCE" class="member-name-link">HEIGHT_SEQUENCE</a> - Static variable in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">Height of the sequence track (when displayed)</div>
</dd>
<dt><a href="../bacnet/views/HelpPage.html#helpGeneView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">helpGeneView(EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/HelpPage.html" title="class in bacnet.views">HelpPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/HelpPage.html#helpGenomeView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">helpGenomeView(EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/HelpPage.html" title="class in bacnet.views">HelpPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/HelpPage.html#helpGenomeViewer(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">helpGenomeViewer(EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/HelpPage.html" title="class in bacnet.views">HelpPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/HelpPage.html#helpNetworkView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">helpNetworkView(EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/HelpPage.html" title="class in bacnet.views">HelpPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/HelpPage.html" class="type-name-link" title="class in bacnet.views">HelpPage</a> - Class in <a href="../bacnet/views/package-summary.html">bacnet.views</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/HelpPage.html#%3Cinit%3E()" class="member-name-link">HelpPage()</a> - Constructor for class bacnet.views.<a href="../bacnet/views/HelpPage.html" title="class in bacnet.views">HelpPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/HelpPage.html#helpProteomHeatMapViewer(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">helpProteomHeatMapViewer(EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/HelpPage.html" title="class in bacnet.views">HelpPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/HelpPage.html#helpProteomicView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">helpProteomicView(EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/HelpPage.html" title="class in bacnet.views">HelpPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/HelpPage.html#helpSrnaView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">helpSrnaView(EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/HelpPage.html" title="class in bacnet.views">HelpPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/HelpPage.html#helpSubmit(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">helpSubmit(EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/HelpPage.html" title="class in bacnet.views">HelpPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/HelpPage.html#helpTrancriptomicView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">helpTrancriptomicView(EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/HelpPage.html" title="class in bacnet.views">HelpPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/HelpPage.html#helpTranscriptHeatMapViewer(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">helpTranscriptHeatMapViewer(EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/HelpPage.html" title="class in bacnet.views">HelpPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/HideExpressionMatrixDialog.html" class="type-name-link" title="class in bacnet.table.gui">HideExpressionMatrixDialog</a> - Class in <a href="../bacnet/table/gui/package-summary.html">bacnet.table.gui</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/HideExpressionMatrixDialog.html#%3Cinit%3E(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome,java.util.ArrayList,java.util.ArrayList,org.eclipse.swt.widgets.Shell,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">HideExpressionMatrixDialog(ExpressionMatrix, Genome, ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, Shell, EPartService)</a> - Constructor for class bacnet.table.gui.<a href="../bacnet/table/gui/HideExpressionMatrixDialog.html" title="class in bacnet.table.gui">HideExpressionMatrixDialog</a></dt>
<dd>
<div class="block">Create the dialog.</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#hideInfoPanel()" class="member-name-link">hideInfoPanel()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Hide the info Panel showing NTerm data</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#hidePhylogeny(org.forester.phylogeny.Phylogeny,java.util.TreeSet)" class="member-name-link">hidePhylogeny(Phylogeny, TreeSet&lt;Integer&gt;)</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>
<div class="block">If a node of the Phylogeny phy, is include in listIDs, hide it by setting the name to void <br></div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#hit_accession" class="member-name-link">hit_accession</a> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#hit_def" class="member-name-link">hit_def</a> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#hit_id" class="member-name-link">hit_id</a> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/phylogeny/Phylogenomic.html#HOMOLOG_SUMMARY" class="member-name-link">HOMOLOG_SUMMARY</a> - Static variable in class bacnet.datamodel.phylogeny.<a href="../bacnet/datamodel/phylogeny/Phylogenomic.html" title="class in bacnet.datamodel.phylogeny">Phylogenomic</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html" class="type-name-link" title="class in bacnet.scripts.database">HomologCreation</a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#%3Cinit%3E()" class="member-name-link">HomologCreation()</a> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#hSequence" class="member-name-link">hSequence</a> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#HTML_URL_ARRAY" class="member-name-link">HTML_URL_ARRAY</a> - Static variable in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#HTML_URL_ARRAY" class="member-name-link">HTML_URL_ARRAY</a> - Static variable in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#HTML_URL_EXP" class="member-name-link">HTML_URL_EXP</a> - Static variable in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#HTML_URL_EXP" class="member-name-link">HTML_URL_EXP</a> - Static variable in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/HTMLUtils.html" class="type-name-link" title="class in bacnet.utils">HTMLUtils</a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>
<div class="block">List of methods to manage D3js visualization in BACNET</div>
</dd>
<dt><a href="../bacnet/utils/HTMLUtils.html#%3Cinit%3E()" class="member-name-link">HTMLUtils()</a> - Constructor for class bacnet.utils.<a href="../bacnet/utils/HTMLUtils.html" title="class in bacnet.utils">HTMLUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#hybrid_min" class="member-name-link">hybrid_min</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/UNAfold.html#hybrid_min" class="member-name-link">hybrid_min</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#hybrid_ss" class="member-name-link">hybrid_ss</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/UNAfold.html#hybrid_ss" class="member-name-link">hybrid_ss</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#hybrid_ss_min" class="member-name-link">hybrid_ss_min</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/UNAfold.html#hybrid_ss_min" class="member-name-link">hybrid_ss_min</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#hybridRNA(java.lang.String,java.lang.String,java.lang.String,java.lang.String,boolean)" class="member-name-link">hybridRNA(String, String, String, String, boolean)</a> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/UNAfold.html#hybridRNA(java.lang.String,java.lang.String,java.lang.String,java.lang.String,boolean)" class="member-name-link">hybridRNA(String, String, String, String, boolean)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
