<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>S-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="index: S">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:S">S</h2>
<dl class="index">
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#sameEnd(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence,int)" class="member-name-link">sameEnd(DNASequence, DNASequence, int)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Determine that to gene have the same end if their ending piosition are the same +- scale bp</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#save()" class="member-name-link">save()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Save stat and values in two separate files <br>
 Saving is done via DataStream to allow streaming reading</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#save()" class="member-name-link">save()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Serialize the vector of lines with an ObjectOutputStream.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#save()" class="member-name-link">save()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Save in: TranscriptomeData.PATH_STREAMING + this.getName() + TranscriptomeData.EXTENSION
 
 Serialize the vector of lines with an ObjectOutputStream.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#save(boolean)" class="member-name-link">save(boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>
<div class="block">Save stat and values in two separate files <br>
 Saving is done via DataStream to allow streaming reading</div>
</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#save(java.lang.Object%5B%5D%5B%5D,java.lang.String)" class="member-name-link">save(Object[][], String)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#save(java.lang.Object%5B%5D%5B%5D,java.lang.String,boolean,boolean)" class="member-name-link">save(Object[][], String, boolean, boolean)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Save the array to the destination file parameter</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#save(java.lang.String)" class="member-name-link">save(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Serialize the vector of lines with an ObjectOutputStream.</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#save(java.lang.String)" class="member-name-link">save(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>
<div class="block">Serialize the vector of lines with an ObjectOutputStream.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#save(java.lang.String)" class="member-name-link">save(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Serialize the vector of lines with an ObjectOutputStream.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Network.html#save(java.lang.String)" class="member-name-link">save(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>
<div class="block">Save stat and values in two separate files <br>
 Saving is done via DataStream to allow streaming reading</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#save(java.lang.String)" class="member-name-link">save(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Prompt the user for a filename, and save the scribble in that file.</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#save(java.lang.String)" class="member-name-link">save(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Prompt the user for a filename, and save the scribble in that file.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#save(java.lang.String)" class="member-name-link">save(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Serialize the vector of lines with an ObjectOutputStream.</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#save(java.lang.String)" class="member-name-link">save(String)</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>
<div class="block">Serialize the vector of lines with an ObjectOutputStream.</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#save(java.lang.String)" class="member-name-link">save(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#save(java.lang.String)" class="member-name-link">save(String)</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#save(java.lang.String)" class="member-name-link">save(String)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#save(java.lang.String)" class="member-name-link">save(String)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#save(java.lang.String)" class="member-name-link">save(String)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>
<div class="block">Prompt the user for a filename, and save the scribble in that file.</div>
</dd>
<dt><a href="../bacnet/utils/Filter.html#save(java.lang.String)" class="member-name-link">save(String)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>
<div class="block">Prompt the user for a filename, and save the scribble in that file.</div>
</dd>
<dt><a href="../bacnet/reader/FastaFileReader.html#saveCDS(java.util.ArrayList,java.lang.String)" class="member-name-link">saveCDS(ArrayList&lt;Gene&gt;, String)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a></dt>
<dd>
<div class="block">Save all amino acid sequence in a fasta file</div>
</dd>
<dt><a href="../bacnet/raprcp/SaveFileUtils.html#saveFile(java.lang.String,java.io.File,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService,org.eclipse.swt.widgets.Shell)" class="member-name-link">saveFile(String, File, String, EPartService, Shell)</a> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/SaveFileUtils.html" title="class in bacnet.raprcp">SaveFileUtils</a></dt>
<dd>
<div class="block">Same as saveTextFile but with no Preview</div>
</dd>
<dt><a href="../bacnet/raprcp/SaveFileUtils.html" class="type-name-link" title="class in bacnet.raprcp">SaveFileUtils</a> - Class in <a href="../bacnet/raprcp/package-summary.html">bacnet.raprcp</a></dt>
<dd>
<div class="block">Utils methods for managinf files.</div>
</dd>
<dt><a href="../bacnet/raprcp/SaveFileUtils.html#%3Cinit%3E()" class="member-name-link">SaveFileUtils()</a> - Constructor for class bacnet.raprcp.<a href="../bacnet/raprcp/SaveFileUtils.html" title="class in bacnet.raprcp">SaveFileUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#saveHashMap(java.util.HashMap,java.lang.String)" class="member-name-link">saveHashMap(HashMap&lt;String, String&gt;, String)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Save HashMap&lt;String, String> to a 2 columns tab-delimited table</div>
</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#saveInHTML(java.lang.Object%5B%5D%5B%5D,java.lang.String,java.lang.String)" class="member-name-link">saveInHTML(Object[][], String, String)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Save the table in a text file containing an HTML table</div>
</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#saveList(java.util.ArrayList,java.lang.String)" class="member-name-link">saveList(ArrayList&lt;String&gt;, String)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Save an ArrayList of String in a tabdelimited table</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#saveLocusTagToOldLocusTag()" class="member-name-link">saveLocusTagToOldLocusTag()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Save locusTagToOldLocusTag in a file for HomologCreation</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#saveProteinIdToLocusTag()" class="member-name-link">saveProteinIdToLocusTag()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Save proteinIdnToLocusTag in a file for HomologCreation</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#saveProteinIdToOldLocusTag()" class="member-name-link">saveProteinIdToOldLocusTag()</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Save locusTagToOldLocusTag in a file for HomologCreation</div>
</dd>
<dt><a href="../bacnet/reader/FastaFileReader.html#saveProteins(java.util.ArrayList,java.lang.String)" class="member-name-link">saveProteins(ArrayList&lt;Gene&gt;, String)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a></dt>
<dd>
<div class="block">Save all amino acid sequence in a fasta file</div>
</dd>
<dt><a href="../bacnet/reader/PTTReader.html#saveProteins(java.util.ArrayList,java.lang.String,bacnet.datamodel.sequence.Chromosome)" class="member-name-link">saveProteins(ArrayList&lt;Gene&gt;, String, Chromosome)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/PTTReader.html" title="class in bacnet.reader">PTTReader</a></dt>
<dd>
<div class="block">Save list of Gene in a PTT file<br>
 Information about the resulting protein are written into different columns:<br>
 "Location","Strand","Length","PID","Gene","Synonym","Code","COG","Product"</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStatProteolyse.html#saveProteolyses(java.util.ArrayList,java.lang.String)" class="member-name-link">saveProteolyses(ArrayList&lt;TIS&gt;, String)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStatProteolyse.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStatProteolyse</a></dt>
<dd>
<div class="block">Save list of protoelyses hashmap</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#saveTab(java.lang.String,java.lang.String)" class="member-name-link">saveTab(String, String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Save in a TabDelimited text file</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#saveText(java.lang.String)" class="member-name-link">saveText(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>
<div class="block">Save Signature in a readable text file</div>
</dd>
<dt><a href="../bacnet/utils/FileUtils.html#saveText(java.lang.String,java.lang.String)" class="member-name-link">saveText(String, String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/FileUtils.html#saveText(java.lang.String,java.lang.String,boolean,boolean)" class="member-name-link">saveText(String, String, boolean, boolean)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Save text from "text" into fileName</div>
</dd>
<dt><a href="../bacnet/raprcp/SaveFileUtils.html#saveTextFile(java.lang.String,java.lang.String,boolean,java.lang.String,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService,org.eclipse.swt.widgets.Shell)" class="member-name-link">saveTextFile(String, String, boolean, String, String, EPartService, Shell)</a> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/SaveFileUtils.html" title="class in bacnet.raprcp">SaveFileUtils</a></dt>
<dd>
<div class="block">Save a text file</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#saveToFileList(java.lang.String,java.lang.String,java.lang.String,java.util.ArrayList)" class="member-name-link">saveToFileList(String, String, String, ArrayList&lt;ExpressionMatrix&gt;)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#saveToFileList(java.lang.String,java.lang.String,java.lang.String,java.util.ArrayList,boolean,boolean)" class="member-name-link">saveToFileList(String, String, String, ArrayList&lt;ExpressionMatrix&gt;, boolean, boolean)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#savetoTextFile()" class="member-name-link">savetoTextFile()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>
<div class="block">Save Track information to a text file</div>
</dd>
<dt><a href="../bacnet/reader/TabDelimitedTableReader.html#saveTreeSet(java.util.TreeSet,java.lang.String)" class="member-name-link">saveTreeSet(TreeSet&lt;String&gt;, String)</a> - Static method in class bacnet.reader.<a href="../bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></dt>
<dd>
<div class="block">Save a TreeSet of String in a Tabdelimited table</div>
</dd>
<dt><a href="../bacnet/utils/HTMLUtils.html#SCATTER" class="member-name-link">SCATTER</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/HTMLUtils.html" title="class in bacnet.utils">HTMLUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#SCATTER_NAME" class="member-name-link">SCATTER_NAME</a> - Static variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#score()" class="member-name-link">score()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Get score.</div>
</dd>
<dt><a href="../bacnet/views/CRISPRPredictView.html#SCRIPT_PATH" class="member-name-link">SCRIPT_PATH</a> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView2.html#SCRIPT_PATH" class="member-name-link">SCRIPT_PATH</a> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView.html#SCRIPT_SERVER" class="member-name-link">SCRIPT_SERVER</a> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/StartCodonProfile.html#SD_ENERGY_CUTOFF" class="member-name-link">SD_ENERGY_CUTOFF</a> - Static variable in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/StartCodonProfile.html" title="class in bacnet.scripts.genome">StartCodonProfile</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/SDProfile.html" class="type-name-link" title="class in bacnet.scripts.genome">SDProfile</a> - Class in <a href="../bacnet/scripts/genome/package-summary.html">bacnet.scripts.genome</a></dt>
<dd>
<div class="block">Calculate Shine-Dalgarno profile for bacterial genome.</div>
</dd>
<dt><a href="../bacnet/scripts/genome/SDProfile.html#%3Cinit%3E()" class="member-name-link">SDProfile()</a> - Constructor for class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SynTView.html#search(java.lang.String)" class="member-name-link">search(String)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>
<div class="block">Search a text file in the annotation<br>
 Search if it is a position and go
 Search if it is contains in <code>getChromosome().getAllElements().keySet()</code>
 Search if it is contains in <code>gene.getName()</code><br>
 All the search are case insensitive using <code>text.toUpperCase();</code></div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#search(java.lang.String)" class="member-name-link">search(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>
<div class="block">Search a text file in the annotation<br>
 Search if it is a position and go
 Search if it is contains in <code>getChromosome().getAllElements().keySet()</code>
 Search if it is contains in <code>gene.getName()</code><br>
 All the search are case insensitive using <code>text.toUpperCase();</code></div>
</dd>
<dt><a href="../bacnet/genomeBrowser/TestGenomeBrowser.html#search(java.lang.String)" class="member-name-link">search(String)</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/TestGenomeBrowser.html" title="class in bacnet.genomeBrowser">TestGenomeBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#search(java.lang.String)" class="member-name-link">search(String)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Search a specific element and center the view on it</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#search(java.lang.String)" class="member-name-link">search(String)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Search a text file in the annotation<br>
 Search if it is a position and go
 Search if it is contains in <code>getChromosome().getAllElements().keySet()</code>
 Search if it is contains in <code>gene.getName()</code><br>
 All the search are case insensitive using <code>text.toUpperCase();</code></div>
</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#search(java.lang.String)" class="member-name-link">search(String)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>
<div class="block">Search a text file in the annotation<br>
 Search if it is a position and go
 Search if it is contains in <code>getChromosome().getAllElements().keySet()</code>
 Search if it is contains in <code>gene.getName()</code><br>
 All the search are case insensitive using <code>text.toUpperCase();</code></div>
</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#search(java.lang.String)" class="member-name-link">search(String)</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>
<div class="block">Search a text file in the annotation<br>
 Search if it is a position and go
 Search if it is contains in <code>getChromosome().getAllElements().keySet()</code>
 Search if it is contains in <code>gene.getName()</code><br>
 All the search are case insensitive using <code>text.toUpperCase();</code></div>
</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#SEARCH" class="member-name-link">SEARCH</a> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#searchElement(java.lang.String)" class="member-name-link">searchElement(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Search in locustag names, gene name, and newlocustag if a gene is present or not</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#searchElement(org.biojava3.core.sequence.DNASequence,java.lang.String)" class="member-name-link">searchElement(DNASequence, String)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Search an element in the Note part of a DNASequence</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#searchElement(org.biojava3.core.sequence.DNASequence,java.lang.String,java.lang.String)" class="member-name-link">searchElement(DNASequence, String, String)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Search an element in the Note part of a DNASequence in a sectio of the note :
 Section = gene, CDS, etc...</div>
</dd>
<dt><a href="../bacnet/scripts/database/GenomesCreation.html#searchForSTandCC()" class="member-name-link">searchForSTandCC()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Concatenate all genomes together and create a huge fasta file.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#searchMissingLocus()" class="member-name-link">searchMissingLocus()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>
<div class="block">By going through all GFF available search for the corresponding locustag</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#searchNterm(java.lang.String)" class="member-name-link">searchNterm(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>
<div class="block">Search an NTerm among all MassSpecData</div>
</dd>
<dt><a href="../bacnet/utils/FileUtils.html#searchPosition(java.lang.String,java.lang.String)" class="member-name-link">searchPosition(String, String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></dt>
<dd>
<div class="block">Given a query sequence, find its different occurences on a database sequence</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermUtils.html#searchStopCodon(bacnet.datamodel.sequence.Genome,int,boolean)" class="member-name-link">searchStopCodon(Genome, int, boolean)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></dt>
<dd>
<div class="block">Get the next stop codon in frame and dowstream to a bpPosition</div>
</dd>
<dt><a href="../bacnet/sequenceTools/SecondStructureRNADialog.html" class="type-name-link" title="class in bacnet.sequenceTools">SecondStructureRNADialog</a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SecondStructureRNADialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,java.lang.String)" class="member-name-link">SecondStructureRNADialog(Shell, String)</a> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SecondStructureRNADialog.html" title="class in bacnet.sequenceTools">SecondStructureRNADialog</a></dt>
<dd>
<div class="block">Create the dialog.</div>
</dd>
<dt><a href="../org/biojava3/core/util/Hashcoder.html#SEED" class="member-name-link">SEED</a> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Hashcoder.html" title="class in org.biojava3.core.util">Hashcoder</a></dt>
<dd>
<div class="block">An initial value for a <code>hashCode</code>, to which we add contributions from fields.</div>
</dd>
<dt><a href="../bacnet/table/gui/TableFilter.html#select(org.eclipse.jface.viewers.Viewer,java.lang.Object,java.lang.Object)" class="member-name-link">select(Viewer, Object, Object)</a> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/TableFilter.html" title="class in bacnet.table.gui">TableFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#selectByAttribute(java.lang.String)" class="member-name-link">selectByAttribute(String)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a list of all features that include the specified attribute key.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#selectByAttribute(java.lang.String,java.lang.String)" class="member-name-link">selectByAttribute(String, String)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a list of all features that include the specified attribute key/value pair.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#selectByGroup(java.lang.String)" class="member-name-link">selectByGroup(String)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a list of all features that have the specified group id, as defined by the group() method
 of the features.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#selectByType(java.lang.String)" class="member-name-link">selectByType(String)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a list of all features that are of the specified type, as defined by the type() method of
 the features.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#selectByUserData(java.lang.String)" class="member-name-link">selectByUserData(String)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a list of all features that include the specified key in their userMap().</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#selectByUserData(java.lang.String,java.lang.Object)" class="member-name-link">selectByUserData(String, Object)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a list of all features that include the specified key/value pair in their userMap().</div>
</dd>
<dt><a href="../org/biojava3/core/util/XMLHelper.html#selectElements(org.w3c.dom.Element,java.lang.String)" class="member-name-link">selectElements(Element, String)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#selectFirstResult(java.lang.String)" class="member-name-link">selectFirstResult(String)</a> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>
<div class="block">Remove duplicates results, keep only first result</div>
</dd>
<dt><a href="../bacnet/utils/UNAfold.html#selectFirstResult(java.lang.String)" class="member-name-link">selectFirstResult(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>
<div class="block">Remove duplicates results, keep only first result</div>
</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#selectFirstResult(java.lang.String,int)" class="member-name-link">selectFirstResult(String, int)</a> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>
<div class="block">Remove duplicates results, keep only first result</div>
</dd>
<dt><a href="../bacnet/utils/UNAfold.html#selectFirstResult(java.lang.String,int)" class="member-name-link">selectFirstResult(String, int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>
<div class="block">Remove duplicates results, keep only first result</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html" class="type-name-link" title="class in bacnet.expressionAtlas.core">SelectGenomeElementDialog</a> - Class in <a href="../bacnet/expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a></dt>
<dd>
<div class="block">Dialog which allow specific selection of genes, smallRNAs and cisRegs</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.TreeSet,java.util.TreeSet,bacnet.datamodel.sequence.Genome)" class="member-name-link">SelectGenomeElementDialog(Shell, EPartService, TreeSet&lt;String&gt;, TreeSet&lt;String&gt;, Genome)</a> - Constructor for class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html" title="class in bacnet.expressionAtlas.core">SelectGenomeElementDialog</a></dt>
<dd>
<div class="block">Create the wizard.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#selectOverlapping(java.lang.String,org.biojava3.genome.parsers.gff.Location,boolean)" class="member-name-link">selectOverlapping(String, Location, boolean)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a list of all features that overlap the specified location on the specified sequence.</div>
</dd>
<dt><a href="../org/biojava3/core/util/XMLHelper.html#selectParentElement(org.w3c.dom.Element,java.lang.String)" class="member-name-link">selectParentElement(Element, String)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLHelper.html#selectSingleElement(org.w3c.dom.Element,java.lang.String)" class="member-name-link">selectSingleElement(Element, String)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#SEPARATOR" class="member-name-link">SEPARATOR</a> - Static variable in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">BioConditoon separator for comparisons<br>
 equal to " vs "</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#seqArraySize(int)" class="member-name-link">seqArraySize(int)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/FastaFileReader.html#seqfileExt" class="member-name-link">seqfileExt</a> - Static variable in class bacnet.reader.<a href="../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#seqname()" class="member-name-link">seqname()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Get the sequence name.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureI.html#seqname()" class="member-name-link">seqname()</a> - Method in interface org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a></dt>
<dd>
<div class="block">Get the sequence name.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html" class="type-name-link" title="class in bacnet.datamodel.sequence">Sequence</a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/Sequence.html" class="type-name-link" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../org/biojava3/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>
<div class="block">Main interface for defining a collection of Compounds and accessing them using biological indexes</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#%3Cinit%3E()" class="member-name-link">Sequence()</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#%3Cinit%3E(java.lang.String,int,int)" class="member-name-link">Sequence(String, int, int)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Create a Sequence object<br>
 if(begin>end) <br>
 this.end = begin <br>
 this.begin = end <br>
 this.strand = '-' <br></div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#%3Cinit%3E(java.lang.String,int,int,char)" class="member-name-link">Sequence(String, int, int, char)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Create a Sequence with a specific name, from, to and strand <br>
 CAREFUL: <br>
 from &lt; to ALWAYS<br>
 sense is given by strand<br>
 If to &lt; from it will correct automatically the position and strand, to have from &lt; to</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.SeqType.html" class="type-name-link" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a> - Enum in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" class="type-name-link" title="class in org.biojava3.core.sequence.storage">SequenceAsStringHelper</a>&lt;<a href="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="type parameter in SequenceAsStringHelper">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</a></dt>
<dd>
<div class="block">This is a common method that can be used across multiple storage/proxy implementations to handle
 Negative strand and other interesting elements of sequence data.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html#%3Cinit%3E()" class="member-name-link">SequenceAsStringHelper()</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="class in org.biojava3.core.sequence.storage">SequenceAsStringHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceComparator.html" class="type-name-link" title="class in org.biojava3.core.sequence">SequenceComparator</a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">Used to sort sequences</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceComparator.html#%3Cinit%3E()" class="member-name-link">SequenceComparator()</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceComparator.html" title="class in org.biojava3.core.sequence">SequenceComparator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" class="type-name-link" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/io/template/package-summary.html">org.biojava3.core.sequence.io.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" class="type-name-link" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell,org.eclipse.e4.ui.workbench.modeling.EPartService,bacnet.datamodel.sequence.Genome,java.lang.String,boolean,int,int,org.biojava3.core.sequence.Strand)" class="member-name-link">SequenceDisplayDialog(Shell, EPartService, Genome, String, boolean, int, int, Strand)</a> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>
<div class="block">Create the dialog.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#sequenceEquality(org.biojava3.core.sequence.template.Sequence,org.biojava3.core.sequence.template.Sequence)" class="member-name-link">sequenceEquality(Sequence&lt;C&gt;, Sequence&lt;C&gt;)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">A case-sensitive manner of comparing two sequence objects together.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#sequenceEqualityIgnoreCase(org.biojava3.core.sequence.template.Sequence,org.biojava3.core.sequence.template.Sequence)" class="member-name-link">sequenceEqualityIgnoreCase(Sequence&lt;C&gt;, Sequence&lt;C&gt;)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">A case-insensitive manner of comparing two sequence objects together.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" class="type-name-link" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a>&lt;<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</a></dt>
<dd>
<div class="block">This class represents the storage container of a sequence stored in a fasta file where the
 initial parsing of the file we store the offset and length of the sequence.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.io.template.SequenceParserInterface,long,int,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">SequenceFileProxyLoader(File, SequenceParserInterface, long, int, CompoundSet&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">SequenceIterator(Sequence&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="class in org.biojava3.core.sequence.template">SequenceMixin.SequenceIterator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/exceptions/SequenceLengthError.html" class="type-name-link" title="class in org.biojava3.core.exceptions">SequenceLengthError</a> - Error in <a href="../org/biojava3/core/exceptions/package-summary.html">org.biojava3.core.exceptions</a></dt>
<dd>
<div class="block">Currently not used</div>
</dd>
<dt><a href="../org/biojava3/core/exceptions/SequenceLengthError.html#%3Cinit%3E(java.lang.String)" class="member-name-link">SequenceLengthError(String)</a> - Constructor for error org.biojava3.core.exceptions.<a href="../org/biojava3/core/exceptions/SequenceLengthError.html" title="class in org.biojava3.core.exceptions">SequenceLengthError</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SequenceLocation.html" class="type-name-link" title="class in org.biojava3.core.sequence.location">SequenceLocation</a>&lt;<a href="../org/biojava3/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">S</a> extends <a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../org/biojava3/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">C</a>&gt;,<wbr><a href="../org/biojava3/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</a></dt>
<dd>
<div class="block">A location in a sequence that keeps a reference to its parent sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/SequenceLocation.html#%3Cinit%3E(int,int,S)" class="member-name-link">SequenceLocation(int, int, S)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SequenceLocation.html" title="class in org.biojava3.core.sequence.location">SequenceLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html" class="type-name-link" title="class in org.biojava3.core.sequence.template">SequenceMixin</a> - Class in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>
<div class="block">Provides a set of static methods to be used as static imports when needed across multiple
 Sequence implementations but inheritance gets in the way.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#%3Cinit%3E()" class="member-name-link">SequenceMixin()</a> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" class="type-name-link" title="class in org.biojava3.core.sequence.template">SequenceMixin.SequenceIterator</a>&lt;<a href="../org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="type parameter in SequenceMixin.SequenceIterator">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>
<div class="block">A basic sequence iterator which iterates over the given Sequence by biological index.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html" class="type-name-link" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">A static class that provides optimization hints for memory or performance handling of sequence
 data.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html#%3Cinit%3E()" class="member-name-link">SequenceOptimizationHints()</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceCollection.html" class="type-name-link" title="enum in org.biojava3.core.sequence">SequenceOptimizationHints.SequenceCollection</a> - Enum in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceUsage.html" class="type-name-link" title="enum in org.biojava3.core.sequence">SequenceOptimizationHints.SequenceUsage</a> - Enum in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/template/SequenceParserInterface.html" class="type-name-link" title="interface in org.biojava3.core.sequence.io.template">SequenceParserInterface</a> - Interface in <a href="../org/biojava3/core/sequence/io/template/package-summary.html">org.biojava3.core.sequence.io.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" class="type-name-link" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a>&lt;<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#%3Cinit%3E()" class="member-name-link">SequenceProxyView()</a> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">SequenceProxyView(Sequence&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence,java.lang.Integer,java.lang.Integer)" class="member-name-link">SequenceProxyView(Sequence&lt;C&gt;, Integer, Integer)</a> - Constructor for class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>
<div class="block">Main constructor for working with SequenceProxyViews</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceReader.html" class="type-name-link" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="../org/biojava3/core/sequence/template/SequenceReader.html" title="type parameter in SequenceReader">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/SequenceTools.html" class="type-name-link" title="class in org.biojava3.core.util">SequenceTools</a> - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/SequenceTools.html#%3Cinit%3E()" class="member-name-link">SequenceTools()</a> - Constructor for class org.biojava3.core.util.<a href="../org/biojava3/core/util/SequenceTools.html" title="class in org.biojava3.core.util">SequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceView.html" class="type-name-link" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="../org/biojava3/core/sequence/template/SequenceView.html" title="type parameter in SequenceView">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/DownloadServiceHandler.html#service(javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse)" class="member-name-link">service(HttpServletRequest, HttpServletResponse)</a> - Method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SessionControl.html" class="type-name-link" title="class in bacnet.e4.rap">SessionControl</a> - Class in <a href="../bacnet/e4/rap/package-summary.html">bacnet.e4.rap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/SessionControl.html" class="type-name-link" title="class in bacnet.e4.rap.setup">SessionControl</a> - Class in <a href="../bacnet/e4/rap/setup/package-summary.html">bacnet.e4.rap.setup</a></dt>
<dd>
<div class="block">General methods for session control of an eclipse.rap application</div>
</dd>
<dt><a href="../bacnet/e4/rap/SessionControl.html#%3Cinit%3E()" class="member-name-link">SessionControl()</a> - Constructor for class bacnet.e4.rap.<a href="../bacnet/e4/rap/SessionControl.html" title="class in bacnet.e4.rap">SessionControl</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/SessionControl.html#%3Cinit%3E()" class="member-name-link">SessionControl()</a> - Constructor for class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SessionControl.html" title="class in bacnet.e4.rap.setup">SessionControl</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setAccession(org.biojava3.core.sequence.AccessionID)" class="member-name-link">setAccession(AccessionID)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#setAlias(java.lang.String)" class="member-name-link">setAlias(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setAllBioConditions(java.util.ArrayList)" class="member-name-link">setAllBioConditions(ArrayList&lt;String&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setAllButtons(java.util.ArrayList)" class="member-name-link">setAllButtons(ArrayList&lt;Button&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#setAllButtons(java.util.ArrayList)" class="member-name-link">setAllButtons(ArrayList&lt;Button&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setAllButtons(java.util.ArrayList)" class="member-name-link">setAllButtons(ArrayList&lt;Button&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/ComparisonsCreation.html#setAllComparisonStreamData(java.util.ArrayList)" class="member-name-link">setAllComparisonStreamData(ArrayList&lt;String&gt;)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">For each Comparison calculates corresponding TranscriptomeData, and save it</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setAllElements(java.util.LinkedHashMap)" class="member-name-link">setAllElements(LinkedHashMap&lt;String, Sequence&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCPList.html#setAllFonts(java.awt.Font)" class="member-name-link">setAllFonts(Font)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperList.html#setAllFonts(org.eclipse.swt.graphics.Font)" class="member-name-link">setAllFonts(Font)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#setAlreadyRead(boolean)" class="member-name-link">setAlreadyRead(boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#setAnnotation(bacnet.datamodel.annotation.Annotation)" class="member-name-link">setAnnotation(Annotation)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setAnnotation(bacnet.datamodel.annotation.Annotation)" class="member-name-link">setAnnotation(Annotation)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setAnnotation(bacnet.datamodel.annotation.Annotation)" class="member-name-link">setAnnotation(Annotation)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setAnnotation(bacnet.datamodel.annotation.Annotation)" class="member-name-link">setAnnotation(Annotation)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#setAnnotation(java.lang.String%5B%5D%5B%5D)" class="member-name-link">setAnnotation(String[][])</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#setAnnotationFont(org.eclipse.swt.graphics.GC)" class="member-name-link">setAnnotationFont(GC)</a> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Set Font to : "Arial",9,SWT.BOLD</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setAnnotations(java.lang.String%5B%5D%5B%5D)" class="member-name-link">setAnnotations(String[][])</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)" class="member-name-link">setAnnotationType(AbstractSequence.AnnotationType)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setAnnotElementHeight(int)" class="member-name-link">setAnnotElementHeight(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setAnnotNumber(float)" class="member-name-link">setAnnotNumber(float)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setAnnotSeparationHeight(int)" class="member-name-link">setAnnotSeparationHeight(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setAnnotSeparationNumber(int)" class="member-name-link">setAnnotSeparationNumber(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setAntiComparisons(java.util.ArrayList)" class="member-name-link">setAntiComparisons(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setAntiSDBindingFreeEnergy(double)" class="member-name-link">setAntiSDBindingFreeEnergy(double)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaSummaryView.html#setArray(java.lang.String%5B%5D%5B%5D)" class="member-name-link">setArray(String[][])</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaSummaryView.html" title="class in bacnet.sequenceTools">SrnaSummaryView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setArrayDataList(java.lang.String%5B%5D%5B%5D)" class="member-name-link">setArrayDataList(String[][])</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setArrayDataList(java.lang.String%5B%5D%5B%5D)" class="member-name-link">setArrayDataList(String[][])</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#setArrayExpressArrayLink(java.lang.String)" class="member-name-link">setArrayExpressArrayLink(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#setArrayExpressExpLink(java.lang.String)" class="member-name-link">setArrayExpressExpLink(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setArrayExpressId(java.lang.String)" class="member-name-link">setArrayExpressId(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setArrayExpressTechnoId(java.lang.String)" class="member-name-link">setArrayExpressTechnoId(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setArrayGeneToLocalization(java.lang.String%5B%5D%5B%5D)" class="member-name-link">setArrayGeneToLocalization(String[][])</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setArrayListWithAnnotation(java.lang.String)" class="member-name-link">setArrayListWithAnnotation(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>
<div class="block">Read Annotation and from the list of sequences obtained, create the different LinkedHashMap<br>
 This method is run at the construction of a Genome object</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setArrayProteomeList(java.lang.String%5B%5D%5B%5D)" class="member-name-link">setArrayProteomeList(String[][])</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setArrayTranscriptomesList(java.lang.String%5B%5D%5B%5D)" class="member-name-link">setArrayTranscriptomesList(String[][])</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setAsRNAListEGDe(java.util.ArrayList)" class="member-name-link">setAsRNAListEGDe(ArrayList&lt;String&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setAsRNAs(java.util.ArrayList)" class="member-name-link">setAsRNAs(ArrayList&lt;Srna&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setAsRNAs(java.util.LinkedHashMap)" class="member-name-link">setAsRNAs(LinkedHashMap&lt;String, Srna&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#setaTISCutoff(int)" class="member-name-link">setaTISCutoff(int)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#setaTISMap(java.util.HashMap)" class="member-name-link">setaTISMap(HashMap&lt;String, ArrayList&lt;NTerm&gt;&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#setBegin(int)" class="member-name-link">setBegin(int)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setBegin(int)" class="member-name-link">setBegin(int)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setBegin(int)" class="member-name-link">setBegin(int)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setBioBegin(java.lang.Integer)" class="member-name-link">setBioBegin(Integer)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#setBioCondition()" class="member-name-link">setBioCondition()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#setBioConditionHashMaps(java.util.LinkedHashMap)" class="member-name-link">setBioConditionHashMaps(LinkedHashMap&lt;String, BioCondition&gt;)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DataValidation.html#setBioconditions(java.util.HashMap)" class="member-name-link">setBioconditions(HashMap&lt;String, Boolean&gt;)</a> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#setBioConditions(java.util.ArrayList)" class="member-name-link">setBioConditions(ArrayList&lt;BioCondition&gt;)</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#setBioConditions(java.util.ArrayList)" class="member-name-link">setBioConditions(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setBioConditionsArrayPath(java.lang.String)" class="member-name-link">setBioConditionsArrayPath(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#setBioCondName(java.lang.String)" class="member-name-link">setBioCondName(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#setBioConds(java.util.ArrayList)" class="member-name-link">setBioConds(ArrayList&lt;String[]&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#setBioConds(java.util.ArrayList)" class="member-name-link">setBioConds(ArrayList&lt;String[]&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#setBioConds(java.util.ArrayList)" class="member-name-link">setBioConds(ArrayList&lt;String[]&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#setBioConds(java.util.ArrayList)" class="member-name-link">setBioConds(ArrayList&lt;String[]&gt;)</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#setBioConds(java.util.ArrayList)" class="member-name-link">setBioConds(ArrayList&lt;String[]&gt;)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setBioConds(java.util.ArrayList)" class="member-name-link">setBioConds(ArrayList&lt;String[]&gt;)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#setBioConds(java.util.LinkedHashMap)" class="member-name-link">setBioConds(LinkedHashMap&lt;String, BioCondition&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#setBioCondsArray(java.lang.String%5B%5D%5B%5D)" class="member-name-link">setBioCondsArray(String[][])</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#setBioCondsArray(java.lang.String%5B%5D%5B%5D)" class="member-name-link">setBioCondsArray(String[][])</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#setBioCondsArray(java.lang.String%5B%5D%5B%5D)" class="member-name-link">setBioCondsArray(String[][])</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#setBioCondsArray(java.lang.String%5B%5D%5B%5D)" class="member-name-link">setBioCondsArray(String[][])</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#setBioCondsArray(java.lang.String%5B%5D%5B%5D)" class="member-name-link">setBioCondsArray(String[][])</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setBioCondsArray(java.lang.String%5B%5D%5B%5D)" class="member-name-link">setBioCondsArray(String[][])</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#setBioCondsArray(java.lang.String%5B%5D%5B%5D)" class="member-name-link">setBioCondsArray(String[][])</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#setBioCondsToDisplay(java.util.ArrayList)" class="member-name-link">setBioCondsToDisplay(ArrayList&lt;String[]&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#setBioCondsToDisplay(java.util.ArrayList)" class="member-name-link">setBioCondsToDisplay(ArrayList&lt;String[]&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#setBioCondsToDisplay(java.util.ArrayList)" class="member-name-link">setBioCondsToDisplay(ArrayList&lt;String[]&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#setBioCondsToDisplay(java.util.ArrayList)" class="member-name-link">setBioCondsToDisplay(ArrayList&lt;String[]&gt;)</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#setBioCondsToDisplay(java.util.ArrayList)" class="member-name-link">setBioCondsToDisplay(ArrayList&lt;String[]&gt;)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setBioCondsToDisplay(java.util.ArrayList)" class="member-name-link">setBioCondsToDisplay(ArrayList&lt;String[]&gt;)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#setBioCondsToDisplay(java.util.ArrayList)" class="member-name-link">setBioCondsToDisplay(ArrayList&lt;String[]&gt;)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setBioEnd(java.lang.Integer)" class="member-name-link">setBioEnd(Integer)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#setBioEnd(java.lang.Integer)" class="member-name-link">setBioEnd(Integer)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceProxyView.html#setBioStart(java.lang.Integer)" class="member-name-link">setBioStart(Integer)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorSWT.html#setBlue(int)" class="member-name-link">setBlue(int)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setBpSizeH(double)" class="member-name-link">setBpSizeH(double)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setBpSizeH(double)" class="member-name-link">setBpSizeH(double)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#setBtnAllMutant(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnAllMutant(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnAllMutant(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnAllMutant(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#setBtnChooseOneMutant(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnChooseOneMutant(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnChooseOneMutant(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnChooseOneMutant(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnDeath(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnDeath(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnDnDNA(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnDnDNA(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnDnProt(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnDnProt(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnDnRNA(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnDnRNA(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnExponential(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnExponential(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnGeneExpression(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnGeneExpression(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnLagPhase(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnLagPhase(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnNoDNA(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnNoDNA(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#setBtnNoneMutant(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnNoneMutant(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnNoneMutant(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnNoneMutant(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnNoProt(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnNoProt(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnNoRNA(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnNoRNA(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setBtnOpenStructure(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnOpenStructure(Button)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnPosDNA(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnPosDNA(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnPrint(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnPrint(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnRegrowth(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnRegrowth(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnRiboSeq(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnRiboSeq(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnRnaseq(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnRnaseq(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnStationnary(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnStationnary(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnSurvival(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnSurvival(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnTermSeq(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnTermSeq(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnTiling(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnTiling(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnTss(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnTss(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setBtnUnpublished(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnUnpublished(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnUpdateCutoff(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnUpdateCutoff(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnUpProt(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnUpProt(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setBtnUpRNA(org.eclipse.swt.widgets.Button)" class="member-name-link">setBtnUpRNA(Button)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setCenterColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setCenterColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setCenterColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setCenterColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setCenterColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setCenterColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setCenterColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setCenterColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setCenterPos(double)" class="member-name-link">setCenterPos(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setCenterPos(double)" class="member-name-link">setCenterPos(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#setCgview(ca.ualberta.stothard.cgview.Cgview)" class="member-name-link">setCgview(Cgview)</a> - Method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setChildrenFeatures(java.util.List)" class="member-name-link">setChildrenFeatures(List&lt;FeatureInterface&lt;S, C&gt;&gt;)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">Set the children features</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setChildrenFeatures(java.util.List)" class="member-name-link">setChildrenFeatures(List&lt;FeatureInterface&lt;S, C&gt;&gt;)</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Set the children features</div>
</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setChromoId(java.lang.String)" class="member-name-link">setChromoId(String)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setChromoID(java.lang.String)" class="member-name-link">setChromoID(String)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setChromosome(bacnet.datamodel.sequence.Chromosome)" class="member-name-link">setChromosome(Chromosome)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#setChromosome(bacnet.datamodel.sequence.Chromosome)" class="member-name-link">setChromosome(Chromosome)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#setChromosomeID(java.lang.String)" class="member-name-link">setChromosomeID(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#setChromosomeID(java.lang.String)" class="member-name-link">setChromosomeID(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setChromosomeID(java.lang.String)" class="member-name-link">setChromosomeID(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setChromosomeID(java.lang.String)" class="member-name-link">setChromosomeID(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setChromosomeID(java.lang.String)" class="member-name-link">setChromosomeID(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#setChromosomeID(java.lang.String)" class="member-name-link">setChromosomeID(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/ChromosomeSequence.html#setChromosomeNumber(int)" class="member-name-link">setChromosomeNumber(int)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#setChromosomes(java.util.LinkedHashMap)" class="member-name-link">setChromosomes(LinkedHashMap&lt;String, Chromosome&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#setChromosomes(java.util.LinkedHashMap)" class="member-name-link">setChromosomes(LinkedHashMap&lt;String, ChromosomeBacteriaSequence&gt;)</a> - Method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setCisRegRNAListEGDe(java.util.ArrayList)" class="member-name-link">setCisRegRNAListEGDe(ArrayList&lt;String&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setCisRegs(java.util.ArrayList)" class="member-name-link">setCisRegs(ArrayList&lt;Srna&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setCisRegs(java.util.LinkedHashMap)" class="member-name-link">setCisRegs(LinkedHashMap&lt;String, Srna&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setCmbChromo(org.eclipse.swt.widgets.Combo)" class="member-name-link">setCmbChromo(Combo)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setCmbGenome(org.eclipse.swt.widgets.Combo)" class="member-name-link">setCmbGenome(Combo)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setCodingSequenceHashMap(java.util.LinkedHashMap)" class="member-name-link">setCodingSequenceHashMap(LinkedHashMap&lt;String, DNASequence&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setCodonUsage(double)" class="member-name-link">setCodonUsage(double)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setCoExprNetworkArrayPath(java.lang.String)" class="member-name-link">setCoExprNetworkArrayPath(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DataValidation.html#setCoExprNetworks(java.util.HashMap)" class="member-name-link">setCoExprNetworks(HashMap&lt;String, Boolean&gt;)</a> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#setCog(java.lang.String)" class="member-name-link">setCog(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setColorMapperList(bacnet.table.core.ColorMapperList)" class="member-name-link">setColorMapperList(ColorMapperList)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperList.html#setColorMappers(java.util.TreeMap)" class="member-name-link">setColorMappers(TreeMap&lt;ColorMapper.TypeMapper, ColorMapper&gt;)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCPList.html#setColorMappers(java.util.TreeMap)" class="member-name-link">setColorMappers(TreeMap&lt;ColorMapperRCP.Type, ColorMapperRCP&gt;)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setColorMapperValues(java.util.ArrayList)" class="member-name-link">setColorMapperValues(ArrayList&lt;ExpressionMatrix&gt;)</a> - Static method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setColorMapperValues(java.util.ArrayList)" class="member-name-link">setColorMapperValues(ArrayList&lt;ExpressionMatrix&gt;)</a> - Static method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/BioConditionComparator.html#setColumn(int)" class="member-name-link">setColumn(int)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/TableViewerComparator.html#setColumn(int)" class="member-name-link">setColumn(int)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/TableViewerComparator.html" title="class in bacnet.table.core">TableViewerComparator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#setColumnNames(java.util.ArrayList)" class="member-name-link">setColumnNames(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#setColumnNames(java.util.ArrayList)" class="member-name-link">setColumnNames(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setColumnNames(java.util.ArrayList)" class="member-name-link">setColumnNames(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#setColumnNames(java.util.ArrayList)" class="member-name-link">setColumnNames(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#setColumnNames(java.util.ArrayList)" class="member-name-link">setColumnNames(ArrayList&lt;String&gt;)</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#setColumnNames(java.util.ArrayList)" class="member-name-link">setColumnNames(ArrayList&lt;String&gt;)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#setColumnNames(java.util.ArrayList)" class="member-name-link">setColumnNames(ArrayList&lt;String&gt;)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setColumnNames(java.util.TreeSet)" class="member-name-link">setColumnNames(TreeSet&lt;String&gt;)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setColumnNames(java.util.TreeSet)" class="member-name-link">setColumnNames(TreeSet&lt;String&gt;)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setComboAbsoluteRelative(org.eclipse.swt.widgets.Combo)" class="member-name-link">setComboAbsoluteRelative(Combo)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#setComboGenome(org.eclipse.swt.widgets.Combo)" class="member-name-link">setComboGenome(Combo)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setComboGenome(org.eclipse.swt.widgets.Combo)" class="member-name-link">setComboGenome(Combo)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setComboGenome(org.eclipse.swt.widgets.Combo)" class="member-name-link">setComboGenome(Combo)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#setComboMutant(org.eclipse.swt.widgets.Combo)" class="member-name-link">setComboMutant(Combo)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setComboMutant(org.eclipse.swt.widgets.Combo)" class="member-name-link">setComboMutant(Combo)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setComment(java.lang.String)" class="member-name-link">setComment(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setComment(java.lang.String)" class="member-name-link">setComment(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setComparisons(java.util.ArrayList)" class="member-name-link">setComparisons(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DataValidation.html#setComparisons(java.util.HashMap)" class="member-name-link">setComparisons(HashMap&lt;String, Boolean&gt;)</a> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#setCompExpression(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">setCompExpression(ExpressionMatrix)</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setCompFilter(org.eclipse.swt.widgets.Composite)" class="member-name-link">setCompFilter(Composite)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setComposite(org.eclipse.swt.widgets.Composite)" class="member-name-link">setComposite(Composite)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setComposite_2(org.eclipse.swt.widgets.Composite)" class="member-name-link">setComposite_2(Composite)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setComposite_6(org.eclipse.swt.widgets.Composite)" class="member-name-link">setComposite_6(Composite)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setComposite_7(org.eclipse.swt.widgets.Composite)" class="member-name-link">setComposite_7(Composite)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setCompositeAddData(org.eclipse.swt.widgets.Composite)" class="member-name-link">setCompositeAddData(Composite)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#setCompositeDataFilter(bacnet.expressionAtlas.ProteomicsDataFilterComposite)" class="member-name-link">setCompositeDataFilter(ProteomicsDataFilterComposite)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#setCompositeDataFilter(bacnet.expressionAtlas.ProteomicsDataFilterComposite)" class="member-name-link">setCompositeDataFilter(ProteomicsDataFilterComposite)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#setCompositeDataFilter(bacnet.expressionAtlas.TranscriptomicsDataFilterComposite)" class="member-name-link">setCompositeDataFilter(TranscriptomicsDataFilterComposite)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setCompositeNterm(bacnet.genomeBrowser.CompositeNTermInfo)" class="member-name-link">setCompositeNterm(CompositeNTermInfo)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setCompositeSummary(org.eclipse.swt.widgets.Composite)" class="member-name-link">setCompositeSummary(Composite)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setCompositeTypeDisplay(org.eclipse.swt.widgets.Composite)" class="member-name-link">setCompositeTypeDisplay(Composite)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#setCompoundAt(char,int)" class="member-name-link">setCompoundAt(char, int)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>
<div class="block">Converts from char to Compound and sets it at the given biological index</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#setCompoundAt(C,int)" class="member-name-link">setCompoundAt(C, int)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>
<div class="block">Sets the compound at the specified biological index</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet(CompoundSet&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet(CompoundSet&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet(CompoundSet&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet(CompoundSet&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet(CompoundSet&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Class is immutable &amp; so this is unsupported</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet(CompoundSet&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet(CompoundSet&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Unsupported</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet(CompoundSet&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceReader.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet(CompoundSet&lt;C&gt;)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">setCompoundSet(CompoundSet&lt;NucleotideCompound&gt;)</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setConservation(int)" class="member-name-link">setConservation(int)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setConservationHashMap(java.util.LinkedHashMap)" class="member-name-link">setConservationHashMap(LinkedHashMap&lt;String, String&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#setContainer(org.eclipse.swt.widgets.Composite)" class="member-name-link">setContainer(Composite)</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#setContents(java.lang.String)" class="member-name-link">setContents(String)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#setContents(java.lang.String)" class="member-name-link">setContents(String)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#setContents(java.lang.String)" class="member-name-link">setContents(String)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>
<div class="block">Once the sequence is retrieved set the contents and make sure everything this is valid</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#setContents(java.lang.String)" class="member-name-link">setContents(String)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html#setContents(java.lang.String)" class="member-name-link">setContents(String)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a></dt>
<dd>
<div class="block">Class is immutable &amp; so this is unsupported</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#setContents(java.lang.String)" class="member-name-link">setContents(String)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#setContents(java.lang.String)" class="member-name-link">setContents(String)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Unsupported</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceReader.html#setContents(java.lang.String)" class="member-name-link">setContents(String)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/RnaSequenceView.html#setContents(java.lang.String)" class="member-name-link">setContents(String)</a> - Method in class org.biojava3.core.sequence.views.<a href="../org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#setContents(java.util.List)" class="member-name-link">setContents(List&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setCurrentState(java.lang.String)" class="member-name-link">setCurrentState(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#setCutOff1(double)" class="member-name-link">setCutOff1(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#setCutOff1(double)" class="member-name-link">setCutOff1(double)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#setCutOff2(double)" class="member-name-link">setCutOff2(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#setCutOff2(double)" class="member-name-link">setCutOff2(double)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#setCutoffHashMap(java.util.HashMap)" class="member-name-link">setCutoffHashMap(HashMap&lt;String, ArrayList&lt;String&gt;&gt;)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/TableSWTView.html#setData(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome)" class="member-name-link">setData(ExpressionMatrix, Genome)</a> - Method in class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#setData(java.lang.String,java.util.ArrayList)" class="member-name-link">setData(String, ArrayList&lt;String&gt;)</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">set the Data for the genomeViewer</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#setDatabase(java.lang.String)" class="member-name-link">setDatabase(String)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setDatabaseFeatures(java.util.HashMap)" class="member-name-link">setDatabaseFeatures(HashMap&lt;String, String&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)" class="member-name-link">setDatabaseReferences(DatabaseReferenceInterface)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#setDataColors()" class="member-name-link">setDataColors()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">For each data assign a Color <br>
 <br>
 OVERLAYBC : Color are always the same, blue and red for tiling, black for GeneExpr, other color
 for the rest <br>
 SUPERIMPOSED, OVERLAYDATA : Color.getColors(index) is used</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#setDataColors(java.util.TreeMap)" class="member-name-link">setDataColors(TreeMap&lt;String, Color&gt;)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#setDataNameFont(org.eclipse.swt.graphics.GC)" class="member-name-link">setDataNameFont(GC)</a> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Set Font to : "Arial",11,SWT.BOLD</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#setDataNOTDisplayed(java.util.ArrayList)" class="member-name-link">setDataNOTDisplayed(ArrayList&lt;String&gt;)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#setDataRead(boolean)" class="member-name-link">setDataRead(boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#setDataRegion(bacnet.genomeBrowser.core.Region)" class="member-name-link">setDataRegion(Region)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#setDatas(bacnet.genomeBrowser.core.DataTrack)" class="member-name-link">setDatas(DataTrack)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#setDatasets(java.util.HashMap)" class="member-name-link">setDatasets(HashMap&lt;String, ExpressionData&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#setDataSizes()" class="member-name-link">setDataSizes()</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>
<div class="block">For each data assign a Size <br>
 <br>
 1.5 for Tiling array, GeneExpression array and RNASeq
 1 for TSS
 0.5 for NTerm and Proteomics data</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setDataWindowSize(int)" class="member-name-link">setDataWindowSize(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#setDate(java.lang.String)" class="member-name-link">setDate(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setDate(java.lang.String)" class="member-name-link">setDate(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setDecaySliderVBar(int)" class="member-name-link">setDecaySliderVBar(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setDefault()" class="member-name-link">setDefault()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setDefault()" class="member-name-link">setDefault()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#setDefault()" class="member-name-link">setDefault()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#setDefault()" class="member-name-link">setDefault()</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#setDefaultAlpha(org.eclipse.swt.graphics.GC)" class="member-name-link">setDefaultAlpha(GC)</a> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Set Alpha to 255</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setDefaultColor()" class="member-name-link">setDefaultColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setDefaultColor()" class="member-name-link">setDefaultColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setDefaultFont()" class="member-name-link">setDefaultFont()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#setDefaultFont(org.eclipse.swt.graphics.GC)" class="member-name-link">setDefaultFont(GC)</a> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Set Font to : "Arial",12,SWT.NORMAL</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setDefaultPos()" class="member-name-link">setDefaultPos()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setDefaultPos()" class="member-name-link">setDefaultPos()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#setDescription(java.lang.String)" class="member-name-link">setDescription(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#setDescription(java.lang.String)" class="member-name-link">setDescription(String)</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setDescription(java.lang.String)" class="member-name-link">setDescription(String)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setDescription(java.lang.String)" class="member-name-link">setDescription(String)</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Set the description that can be used to describe the feature</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#setDescription(java.lang.String)" class="member-name-link">setDescription(String)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setDescription(java.lang.String)" class="member-name-link">setDescription(String)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Compound.html#setDescription(java.lang.String)" class="member-name-link">setDescription(String)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#setDescription(java.lang.String)" class="member-name-link">setDescription(String)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#setDescription(java.lang.String)" class="member-name-link">setDescription(String)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setDeviation(double)" class="member-name-link">setDeviation(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setDeviation(double)" class="member-name-link">setDeviation(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#setDisplay(boolean%5B%5D)" class="member-name-link">setDisplay(boolean[])</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setDisplay(org.eclipse.swt.widgets.Display)" class="member-name-link">setDisplay(Display)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#setDisplayAbsoluteValue(boolean)" class="member-name-link">setDisplayAbsoluteValue(boolean)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setDisplayMouseLine(boolean)" class="member-name-link">setDisplayMouseLine(boolean)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setDisplayMouseLine(boolean)" class="member-name-link">setDisplayMouseLine(boolean)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#setDisplayRegion(bacnet.genomeBrowser.core.Region)" class="member-name-link">setDisplayRegion(Region)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#setDisplaySequence(boolean)" class="member-name-link">setDisplaySequence(boolean)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#setDisplayType(bacnet.genomeBrowser.core.Track.DisplayType)" class="member-name-link">setDisplayType(Track.DisplayType)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#setDNAType(org.biojava3.core.sequence.DNASequence.DNAType)" class="member-name-link">setDNAType(DNASequence.DNAType)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setDownDNAList(java.util.ArrayList)" class="member-name-link">setDownDNAList(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/InternalBrowser.html#setDownloadLink(java.lang.String)" class="member-name-link">setDownloadLink(String)</a> - Method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setDownProtList(java.util.ArrayList)" class="member-name-link">setDownProtList(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setDownRNAList(java.util.ArrayList)" class="member-name-link">setDownRNAList(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setDuplicates(java.util.ArrayList)" class="member-name-link">setDuplicates(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#setDuplicateSequenceToRemove(java.util.ArrayList)" class="member-name-link">setDuplicateSequenceToRemove(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Network.html#setEdges(java.util.HashMap)" class="member-name-link">setEdges(HashMap&lt;String, HashMap&lt;String, Double&gt;&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>
<div class="block">An hashMap which map each vertex (=genome element) to the correlated vertex.<br>
 Vertex -> HashMap&lt;Vetrex,PearsonCorrelation></div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setElements(java.util.ArrayList)" class="member-name-link">setElements(ArrayList&lt;Sequence&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#setElements(java.util.ArrayList)" class="member-name-link">setElements(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#setElements(java.util.HashMap)" class="member-name-link">setElements(HashMap&lt;String, NTerm&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Link NTerm name to <code>NTerm</code><br>
 This correspond to the list of <code>NTerm</code></div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setElements(java.util.LinkedHashMap)" class="member-name-link">setElements(LinkedHashMap&lt;String, Sequence&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#setEnd(int)" class="member-name-link">setEnd(int)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setEnd(int)" class="member-name-link">setEnd(int)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setEnd(int)" class="member-name-link">setEnd(int)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setExcludeColumn(java.util.ArrayList)" class="member-name-link">setExcludeColumn(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#setExcludeColumn(java.util.ArrayList)" class="member-name-link">setExcludeColumn(ArrayList&lt;String&gt;)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/FilterList.html#setExcludeColumn(java.util.ArrayList)" class="member-name-link">setExcludeColumn(ArrayList&lt;String&gt;)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#setExcludeColumn(java.util.ArrayList)" class="member-name-link">setExcludeColumn(ArrayList&lt;String&gt;)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setExcludeRow(java.util.ArrayList)" class="member-name-link">setExcludeRow(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#setExcludeRow(java.util.ArrayList)" class="member-name-link">setExcludeRow(ArrayList&lt;String&gt;)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/FilterList.html#setExcludeRow(java.util.ArrayList)" class="member-name-link">setExcludeRow(ArrayList&lt;String&gt;)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#setExcludeRow(java.util.ArrayList)" class="member-name-link">setExcludeRow(ArrayList&lt;String&gt;)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SessionControl.html#setExitConfirmation()" class="member-name-link">setExitConfirmation()</a> - Static method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SessionControl.html" title="class in bacnet.e4.rap">SessionControl</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/SessionControl.html#setExitConfirmation()" class="member-name-link">setExitConfirmation()</a> - Static method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SessionControl.html" title="class in bacnet.e4.rap.setup">SessionControl</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setExperienceNb(int)" class="member-name-link">setExperienceNb(int)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setExperimentComparisonTablePath(java.lang.String)" class="member-name-link">setExperimentComparisonTablePath(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#setExpression(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">setExpression(ExpressionMatrix)</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#setExpression(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">setExpression(ExpressionMatrix)</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#setExpressionAlpha(org.eclipse.swt.graphics.GC)" class="member-name-link">setExpressionAlpha(GC)</a> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Set Alpha to 150</div>
</dd>
<dt><a href="../bacnet/Database.html#setExprProteomesTable(java.util.HashMap)" class="member-name-link">setExprProteomesTable(HashMap&lt;String, ExpressionMatrix&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setExprTranscriptomesTable(java.util.HashMap)" class="member-name-link">setExprTranscriptomesTable(HashMap&lt;String, ExpressionMatrix&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setFeatures(java.util.LinkedHashMap)" class="member-name-link">setFeatures(LinkedHashMap&lt;String, String&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)" class="member-name-link">setFeaturesKeyWord(FeaturesKeyWordInterface)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/DownloadServiceHandler.html#setFile(java.io.File)" class="member-name-link">setFile(File)</a> - Method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/DownloadServiceHandler.html#setFileName(java.lang.String)" class="member-name-link">setFileName(String)</a> - Method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/InternalBrowser.html#setFileName(java.lang.String)" class="member-name-link">setFileName(String)</a> - Method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#setFilterColumn(boolean)" class="member-name-link">setFilterColumn(boolean)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#setFilterColumn(boolean)" class="member-name-link">setFilterColumn(boolean)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SynTView.html#setFilteredNetwork(bacnet.datamodel.dataset.Network)" class="member-name-link">setFilteredNetwork(Network)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#setFilteredNetwork(bacnet.datamodel.dataset.Network)" class="member-name-link">setFilteredNetwork(Network)</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/FilterList.html#setFilters(java.util.TreeMap)" class="member-name-link">setFilters(TreeMap&lt;String, Filter&gt;)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setFirstandSecondRowNameInList(java.util.ArrayList,java.lang.String,java.lang.String)" class="member-name-link">setFirstandSecondRowNameInList(ArrayList&lt;ExpressionMatrix&gt;, String, String)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setFirstRowName(java.lang.String)" class="member-name-link">setFirstRowName(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/SetupPart.html#setFocus()" class="member-name-link">setFocus()</a> - Method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SetupPart.html" title="class in bacnet.e4.rap.setup">SetupPart</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/TestPart.html#setFocus()" class="member-name-link">setFocus()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/TestPart.html" title="class in bacnet.e4.rap">TestPart</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView.html#setFocus()" class="member-name-link">setFocus()</a> - Method in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView2.html#setFocus()" class="member-name-link">setFocus()</a> - Method in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewBacnet.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewBacnet.html" title="class in bacnet.e4.rap">InitViewBacnet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeria.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeria.html" title="class in bacnet.e4.rap">InitViewListeria</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeriaSample.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">InitViewListeriaSample</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewYersinia.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewYersinia.html" title="class in bacnet.e4.rap">InitViewYersinia</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#setFocused(boolean)" class="member-name-link">setFocused(boolean)</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setFontDouble(java.awt.Font)" class="member-name-link">setFontDouble(Font)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setFontDouble(org.eclipse.swt.graphics.Font)" class="member-name-link">setFontDouble(Font)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setFontRowName(org.eclipse.swt.graphics.Font)" class="member-name-link">setFontRowName(Font)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCPList.html#setFontsDouble(java.awt.Font)" class="member-name-link">setFontsDouble(Font)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperList.html#setFontsDouble(org.eclipse.swt.graphics.Font)" class="member-name-link">setFontsDouble(Font)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCPList.html#setFontsString(java.awt.Font)" class="member-name-link">setFontsString(Font)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperList.html#setFontsString(org.eclipse.swt.graphics.Font)" class="member-name-link">setFontsString(Font)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setFontString(java.awt.Font)" class="member-name-link">setFontString(Font)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setFontText(org.eclipse.swt.graphics.Font)" class="member-name-link">setFontText(Font)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#setFoundIn(java.util.ArrayList)" class="member-name-link">setFoundIn(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setGeneExprs(java.util.ArrayList)" class="member-name-link">setGeneExprs(ArrayList&lt;GeneExpression&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setGeneListEGDe(java.util.ArrayList)" class="member-name-link">setGeneListEGDe(ArrayList&lt;String&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#setGeneModification(java.util.HashMap)" class="member-name-link">setGeneModification(HashMap&lt;String, ArrayList&lt;NTerm.TypeModif&gt;&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#setGeneName(java.lang.String)" class="member-name-link">setGeneName(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setGeneNameToLocusTagMap(java.util.LinkedHashMap)" class="member-name-link">setGeneNameToLocusTagMap(LinkedHashMap&lt;String, String&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setGeneNameToLocusTagMap(java.util.LinkedHashMap)" class="member-name-link">setGeneNameToLocusTagMap(LinkedHashMap&lt;String, String&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setGeneralExperiment(bacnet.datamodel.expdesign.Experiment)" class="member-name-link">setGeneralExperiment(Experiment)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SynTView.html#setGeneralNetwork(bacnet.datamodel.dataset.Network)" class="member-name-link">setGeneralNetwork(Network)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#setGeneralNetwork(bacnet.datamodel.dataset.Network)" class="member-name-link">setGeneralNetwork(Network)</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#setGenes(java.util.HashMap)" class="member-name-link">setGenes(HashMap&lt;String, ArrayList&lt;NTerm&gt;&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setGenes(java.util.LinkedHashMap)" class="member-name-link">setGenes(LinkedHashMap&lt;String, Gene&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Operon.html#setGenes(java.util.TreeSet)" class="member-name-link">setGenes(TreeSet&lt;String&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setGenesAlternative(java.util.LinkedHashMap)" class="member-name-link">setGenesAlternative(LinkedHashMap&lt;String, Gene&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#setGeneSequence()" class="member-name-link">setGeneSequence()</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Create a fasta file with all the gene sequences</div>
</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#setGenome(bacnet.datamodel.sequence.Genome)" class="member-name-link">setGenome(Genome)</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#setGenome(bacnet.datamodel.sequence.Genome)" class="member-name-link">setGenome(Genome)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setGenome(bacnet.datamodel.sequence.Genome)" class="member-name-link">setGenome(Genome)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setGenome(bacnet.datamodel.sequence.Genome)" class="member-name-link">setGenome(Genome)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setGenome(bacnet.datamodel.sequence.Genome)" class="member-name-link">setGenome(Genome)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#setGenome(bacnet.datamodel.sequence.Genome)" class="member-name-link">setGenome(Genome)</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#setGenome(java.lang.String)" class="member-name-link">setGenome(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#setGenome(java.lang.String)" class="member-name-link">setGenome(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setGenomeAA(java.lang.String)" class="member-name-link">setGenomeAA(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setGenomeAACompl(java.lang.String)" class="member-name-link">setGenomeAACompl(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setGenomeAAMinus(java.lang.String)" class="member-name-link">setGenomeAAMinus(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setGenomeAAMinusCompl(java.lang.String)" class="member-name-link">setGenomeAAMinusCompl(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setGenomeAAPlus(java.lang.String)" class="member-name-link">setGenomeAAPlus(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setGenomeAAPlusCompl(java.lang.String)" class="member-name-link">setGenomeAAPlusCompl(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setGenomeArrayPath(java.lang.String)" class="member-name-link">setGenomeArrayPath(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#setGenomeElements(java.util.ArrayList)" class="member-name-link">setGenomeElements(ArrayList&lt;String&gt;)</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#setGenomeElements(java.util.ArrayList)" class="member-name-link">setGenomeElements(ArrayList&lt;String&gt;)</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setGenomeList(java.util.ArrayList)" class="member-name-link">setGenomeList(ArrayList&lt;String&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#setGenomeName(java.lang.String)" class="member-name-link">setGenomeName(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#setGenomeName(java.lang.String)" class="member-name-link">setGenomeName(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setGenomeName(java.lang.String)" class="member-name-link">setGenomeName(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setGenomeName(java.lang.String)" class="member-name-link">setGenomeName(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SynTView.html#setGenomeName(java.lang.String)" class="member-name-link">setGenomeName(String)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setGenomeName(java.lang.String)" class="member-name-link">setGenomeName(String)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#setGenomeName(java.lang.String)" class="member-name-link">setGenomeName(String)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#setGenomeName(java.lang.String)" class="member-name-link">setGenomeName(String)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#setGenomeName(java.lang.String)" class="member-name-link">setGenomeName(String)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setGenomeName(java.lang.String)" class="member-name-link">setGenomeName(String)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#setGenomeName(java.lang.String)" class="member-name-link">setGenomeName(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setGenomeName(java.lang.String)" class="member-name-link">setGenomeName(String)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#setGenomeName(java.lang.String)" class="member-name-link">setGenomeName(String)</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#setGenomeNCBILink(java.lang.String)" class="member-name-link">setGenomeNCBILink(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DataValidation.html#setGenomes(java.util.HashMap)" class="member-name-link">setGenomes(HashMap&lt;String, Boolean&gt;)</a> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setGenomes(java.util.TreeMap)" class="member-name-link">setGenomes(TreeMap&lt;String, Genome&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setGenomeSelected(java.lang.String)" class="member-name-link">setGenomeSelected(String)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">The comboGenome contains modified genome name so we need this method to select an element<br>
 a '*' is add to genome name when a transcriptome or proteome data is available</div>
</dd>
<dt><a href="../bacnet/Database.html#setGenomesNCBI(java.util.TreeMap)" class="member-name-link">setGenomesNCBI(TreeMap&lt;String, GenomeNCBI&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setGenomeUsed(java.lang.String)" class="member-name-link">setGenomeUsed(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setGoogleId(java.lang.String)" class="member-name-link">setGoogleId(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorSWT.html#setGreen(int)" class="member-name-link">setGreen(int)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setGrowth(java.util.ArrayList)" class="member-name-link">setGrowth(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setHeader(java.lang.String,int)" class="member-name-link">setHeader(String, int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setHeaderAnnotation(java.util.ArrayList)" class="member-name-link">setHeaderAnnotation(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setHeaderColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setHeaderColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setHeaderColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setHeaderColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setHeaderColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setHeaderColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setHeaders(java.util.ArrayList)" class="member-name-link">setHeaders(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#setHeight(int)" class="member-name-link">setHeight(int)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/MatrixSize.html#setHeight(java.util.HashMap)" class="member-name-link">setHeight(HashMap&lt;String, Integer&gt;)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/MatrixSize.html#setHeightDefault(int)" class="member-name-link">setHeightDefault(int)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setHeightPix(int)" class="member-name-link">setHeightPix(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setHeightPix(int)" class="member-name-link">setHeightPix(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setHorizontalBarProperties()" class="member-name-link">setHorizontalBarProperties()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Depending on <code>Track</code> parameters (<code>Region</code> and <code>Zoom</code>) fix
 properties of HorizontalBar</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#setId(java.lang.String)" class="member-name-link">setId(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setId(java.lang.String)" class="member-name-link">setId(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#setId(java.lang.String)" class="member-name-link">setId(String)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#setID(java.lang.String)" class="member-name-link">setID(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizardPage.html#setInfo()" class="member-name-link">setInfo()</a> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#setInfoRead(boolean)" class="member-name-link">setInfoRead(boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#setInfoRead(boolean)" class="member-name-link">setInfoRead(boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#setInitialized(boolean)" class="member-name-link">setInitialized(boolean)</a> - Method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setInitView(java.lang.String)" class="member-name-link">setInitView(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setInsoluble(boolean)" class="member-name-link">setInsoluble(boolean)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setIntersect(boolean)" class="member-name-link">setIntersect(boolean)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setLabel_2(org.eclipse.swt.widgets.Label)" class="member-name-link">setLabel_2(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setLabel_3(org.eclipse.swt.widgets.Label)" class="member-name-link">setLabel_3(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setLblDescribedIn(org.eclipse.swt.widgets.Label)" class="member-name-link">setLblDescribedIn(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setLblGene(org.eclipse.swt.widgets.Label)" class="member-name-link">setLblGene(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setLblNodiff(org.eclipse.swt.widgets.Label)" class="member-name-link">setLblNodiff(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setLblNodiffProteome(org.eclipse.swt.widgets.Label)" class="member-name-link">setLblNodiffProteome(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setLblOrAnyOther(org.eclipse.swt.widgets.Label)" class="member-name-link">setLblOrAnyOther(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setLblOveProteome(org.eclipse.swt.widgets.Label)" class="member-name-link">setLblOveProteome(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setLblOver(org.eclipse.swt.widgets.Label)" class="member-name-link">setLblOver(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setLblOverProteomes(org.eclipse.swt.widgets.Label)" class="member-name-link">setLblOverProteomes(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setLblOverTranscriptomes(org.eclipse.swt.widgets.Label)" class="member-name-link">setLblOverTranscriptomes(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setLblPredictedSecondaryStructure(org.eclipse.swt.widgets.Label)" class="member-name-link">setLblPredictedSecondaryStructure(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setLblStructure(org.eclipse.swt.widgets.Label)" class="member-name-link">setLblStructure(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setLblTranscriptomesData(org.eclipse.swt.widgets.Label)" class="member-name-link">setLblTranscriptomesData(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setLblUnder(org.eclipse.swt.widgets.Label)" class="member-name-link">setLblUnder(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setLblUnderProteome(org.eclipse.swt.widgets.Label)" class="member-name-link">setLblUnderProteome(Label)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#setLeftBCs(java.util.ArrayList)" class="member-name-link">setLeftBCs(ArrayList&lt;String&gt;)</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#setLegend()" class="member-name-link">setLegend()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#setLength(int)" class="member-name-link">setLength(int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setLength(int)" class="member-name-link">setLength(int)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#setLengthAA(int)" class="member-name-link">setLengthAA(int)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html#setLineLength(int)" class="member-name-link">setLineLength(int)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaWriter.html#setLineLength(int)" class="member-name-link">setLineLength(int)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaWriter.html#setLineSeparator(java.lang.String)" class="member-name-link">setLineSeparator(String)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a></dt>
<dd>
<div class="block">Allow an override of operating system line separator for programs that needs a specific CRLF or
 CR or LF option</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setLinkedBioCondition(java.util.ArrayList)" class="member-name-link">setLinkedBioCondition(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setListDatabases(java.util.HashMap)" class="member-name-link">setListDatabases(HashMap&lt;String, String&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setListGenes(java.util.ArrayList)" class="member-name-link">setListGenes(ArrayList&lt;String&gt;)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setListSrnas(java.util.ArrayList)" class="member-name-link">setListSrnas(ArrayList&lt;String&gt;)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setLoaded(boolean)" class="member-name-link">setLoaded(boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setLocalization(java.util.ArrayList)" class="member-name-link">setLocalization(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setLocation(org.biojava3.core.sequence.location.SequenceLocation)" class="member-name-link">setLocation(SequenceLocation&lt;S, C&gt;)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">A feature could be a single sequence position like a mutation or a post translational
 modification of an amino acid.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setLocation(org.biojava3.core.sequence.location.SequenceLocation)" class="member-name-link">setLocation(SequenceLocation&lt;S, C&gt;)</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">The new location for this feature.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setLocusTagToGeneNameMap(java.util.LinkedHashMap)" class="member-name-link">setLocusTagToGeneNameMap(LinkedHashMap&lt;String, String&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setLocusTagToGeneNameMap(java.util.LinkedHashMap)" class="member-name-link">setLocusTagToGeneNameMap(LinkedHashMap&lt;String, String&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setLocusTagToOldLocusTagMap(java.util.LinkedHashMap)" class="member-name-link">setLocusTagToOldLocusTagMap(LinkedHashMap&lt;String, String&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setLogfcOmicsMatrix(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">setLogfcOmicsMatrix(ExpressionMatrix)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setLogFCProteomeTable(java.util.HashMap)" class="member-name-link">setLogFCProteomeTable(HashMap&lt;String, ExpressionMatrix&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setLogFCTable(java.util.HashMap)" class="member-name-link">setLogFCTable(HashMap&lt;String, ExpressionMatrix&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setLogo(java.lang.String)" class="member-name-link">setLogo(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NGS.html#setLogTransformed(boolean)" class="member-name-link">setLogTransformed(boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#setLongName(java.lang.String)" class="member-name-link">setLongName(String)</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#setLongName(java.lang.String)" class="member-name-link">setLongName(String)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Compound.html#setLongName(java.lang.String)" class="member-name-link">setLongName(String)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#setLongName(java.lang.String)" class="member-name-link">setLongName(String)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#setLongName(java.lang.String)" class="member-name-link">setLongName(String)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html#setLowercaseToNull(org.biojava3.core.sequence.ProteinSequence,java.lang.Object%5B%5D)" class="member-name-link">setLowercaseToNull(ProteinSequence, Object[])</a> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></dt>
<dd>
<div class="block">Takes a <a href="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence"><code>ProteinSequence</code></a> which was created by a
 <a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io"><code>CasePreservingProteinSequenceCreator</code></a>.</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setMappingFrame(bacnet.datamodel.proteomics.NTerm.MappingFrame)" class="member-name-link">setMappingFrame(NTerm.MappingFrame)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#setMarkup(org.eclipse.swt.widgets.Composite)" class="member-name-link">setMarkup(Composite)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#setMarkup(org.eclipse.swt.widgets.Label)" class="member-name-link">setMarkup(Label)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setMassSpecData(bacnet.datamodel.dataset.NTermData)" class="member-name-link">setMassSpecData(NTermData)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setMatrices(java.util.ArrayList)" class="member-name-link">setMatrices(ArrayList&lt;ExpressionMatrix&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#setMatrix(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">setMatrix(ExpressionMatrix)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setMatrix(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">setMatrix(ExpressionMatrix)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#setMatrix(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">setMatrix(ExpressionMatrix)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/FilterList.html#setMatrix(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">setMatrix(ExpressionMatrix)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#setMatrix(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">setMatrix(ExpressionMatrix)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setMatrixDisplayed(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">setMatrixDisplayed(ExpressionMatrix)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#setMax(double)" class="member-name-link">setMax(double)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMax(double)" class="member-name-link">setMax(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMax(double)" class="member-name-link">setMax(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMaxColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setMaxColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMaxColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setMaxColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMaxColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setMaxColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMaxColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setMaxColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMaxPos(double)" class="member-name-link">setMaxPos(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMaxPos(double)" class="member-name-link">setMaxPos(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setMaxQuant(java.lang.String)" class="member-name-link">setMaxQuant(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setMaxQuantum(java.lang.String%5B%5D)" class="member-name-link">setMaxQuantum(String[])</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#setMean(double)" class="member-name-link">setMean(double)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMean(double)" class="member-name-link">setMean(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMean(double)" class="member-name-link">setMean(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setMedia(java.util.ArrayList)" class="member-name-link">setMedia(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setMediaGrowthProperties(java.util.ArrayList)" class="member-name-link">setMediaGrowthProperties(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#setMedian(double)" class="member-name-link">setMedian(double)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMedian(double)" class="member-name-link">setMedian(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMedian(double)" class="member-name-link">setMedian(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#setMedianValue(java.lang.String,double)" class="member-name-link">setMedianValue(String, double)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>
<div class="block">Set the median value of corresponding probe in double[][] values</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#setMiddleH(int)" class="member-name-link">setMiddleH(int)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#setMiddleV(int)" class="member-name-link">setMiddleV(int)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMidLeftColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setMidLeftColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMidLeftColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setMidLeftColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMidLeftColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setMidLeftColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMidLeftColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setMidLeftColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMidLeftPos(double)" class="member-name-link">setMidLeftPos(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMidLeftPos(double)" class="member-name-link">setMidLeftPos(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMidRightColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setMidRightColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMidRightColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setMidRightColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMidRightColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setMidRightColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMidRightColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setMidRightColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMidRightPos(double)" class="member-name-link">setMidRightPos(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMidRightPos(double)" class="member-name-link">setMidRightPos(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#setMin(double)" class="member-name-link">setMin(double)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMin(double)" class="member-name-link">setMin(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMin(double)" class="member-name-link">setMin(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMinColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setMinColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMinColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setMinColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMinColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setMinColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMinColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setMinColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#setMinimumAlpha(org.eclipse.swt.graphics.GC)" class="member-name-link">setMinimumAlpha(GC)</a> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Set Alpha to 15</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setMinPos(double)" class="member-name-link">setMinPos(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setMinPos(double)" class="member-name-link">setMinPos(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#setModifications(java.util.HashMap)" class="member-name-link">setModifications(HashMap&lt;String, ArrayList&lt;NTerm&gt;&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#setModifs(java.util.ArrayList)" class="member-name-link">setModifs(ArrayList&lt;NTerm.TypeModif&gt;)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setModifSequence(java.lang.String)" class="member-name-link">setModifSequence(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#setMolecularWeight(java.lang.Float)" class="member-name-link">setMolecularWeight(Float)</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#setMolecularWeight(java.lang.Float)" class="member-name-link">setMolecularWeight(Float)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Compound.html#setMolecularWeight(java.lang.Float)" class="member-name-link">setMolecularWeight(Float)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#setMolecularWeight(java.lang.Float)" class="member-name-link">setMolecularWeight(Float)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#setMolecularWeight(java.lang.Float)" class="member-name-link">setMolecularWeight(Float)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#setMolMass(double)" class="member-name-link">setMolMass(double)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setMouseXPosition(int)" class="member-name-link">setMouseXPosition(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setMouseXPosition(int)" class="member-name-link">setMouseXPosition(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setMouseYPosition(int)" class="member-name-link">setMouseYPosition(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setMouseYPosition(int)" class="member-name-link">setMouseYPosition(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html#setMultiFasta(java.util.ArrayList)" class="member-name-link">setMultiFasta(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html" title="class in bacnet.datamodel.sequence">Genome.GetMultiFastaThread</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setMutant(java.util.ArrayList)" class="member-name-link">setMutant(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#setName(java.lang.String)" class="member-name-link">setName(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#setName(java.lang.String)" class="member-name-link">setName(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#setName(java.lang.String)" class="member-name-link">setName(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setName(java.lang.String)" class="member-name-link">setName(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#setName(java.lang.String)" class="member-name-link">setName(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setName(java.lang.String)" class="member-name-link">setName(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setName(java.lang.String)" class="member-name-link">setName(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setName(java.lang.String)" class="member-name-link">setName(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setName(java.lang.String)" class="member-name-link">setName(String)</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#setName(java.lang.String)" class="member-name-link">setName(String)</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setName(java.lang.String)" class="member-name-link">setName(String)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#setName(java.lang.String)" class="member-name-link">setName(String)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#setName(java.lang.String)" class="member-name-link">setName(String)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html#setNameMassSpecDataFinal(java.lang.String)" class="member-name-link">setNameMassSpecDataFinal(String)</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html" title="class in bacnet.scripts.listeriomics.nterm">NTermCreateFinalData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setNameNTermData(java.lang.String)" class="member-name-link">setNameNTermData(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#setNameRawData(java.lang.String)" class="member-name-link">setNameRawData(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setNavigationListener(org.eclipse.rap.rwt.client.service.BrowserNavigationListener)" class="member-name-link">setNavigationListener(BrowserNavigationListener)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setNbElements(int)" class="member-name-link">setNbElements(int)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setNbElements(int)" class="member-name-link">setNbElements(int)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#setNbElementType(int)" class="member-name-link">setNbElementType(int)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#setNbSet(int)" class="member-name-link">setNbSet(int)</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setNcRNAs(java.util.LinkedHashMap)" class="member-name-link">setNcRNAs(LinkedHashMap&lt;String, NcRNA&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#setNewChromosome(java.lang.String)" class="member-name-link">setNewChromosome(String)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#setNewLocusTag(java.lang.String)" class="member-name-link">setNewLocusTag(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setNextCodon(java.lang.String)" class="member-name-link">setNextCodon(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setNGSSeqs(java.util.ArrayList)" class="member-name-link">setNGSSeqs(ArrayList&lt;NGS&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setNoData(boolean)" class="member-name-link">setNoData(boolean)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setNoDNAList(java.util.ArrayList)" class="member-name-link">setNoDNAList(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setNoncodingSequenceHashMap(java.util.LinkedHashMap)" class="member-name-link">setNoncodingSequenceHashMap(LinkedHashMap&lt;String, DNASequence&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setNoProtList(java.util.ArrayList)" class="member-name-link">setNoProtList(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setNoRNAList(java.util.ArrayList)" class="member-name-link">setNoRNAList(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setNotDiffExpressedASrnas(java.lang.String%5B%5D)" class="member-name-link">setNotDiffExpressedASrnas(String[])</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setNotDiffExpressedCisRegs(java.lang.String%5B%5D)" class="member-name-link">setNotDiffExpressedCisRegs(String[])</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setNotDiffExpressedGenes(java.lang.String%5B%5D)" class="member-name-link">setNotDiffExpressedGenes(String[])</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#setNotDiffExpresseds(java.util.TreeSet)" class="member-name-link">setNotDiffExpresseds(TreeSet&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setNotDiffExpressedSrnas(java.lang.String%5B%5D)" class="member-name-link">setNotDiffExpressedSrnas(String[])</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#setNote(java.lang.String)" class="member-name-link">setNote(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#setNote(java.lang.String)" class="member-name-link">setNote(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setNotesList(java.util.ArrayList)" class="member-name-link">setNotesList(ArrayList&lt;String&gt;)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#setnTermExp(bacnet.datamodel.dataset.NTermData)" class="member-name-link">setnTermExp(NTermData)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setnTermHighlight(bacnet.datamodel.proteomics.NTerm)" class="member-name-link">setnTermHighlight(NTerm)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/CompositeNTermInfo.html#setNTermInfo(bacnet.datamodel.proteomics.NTerm,bacnet.datamodel.dataset.NTermData)" class="member-name-link">setNTermInfo(NTerm, NTermData)</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CompositeNTermInfo.html" title="class in bacnet.genomeBrowser">CompositeNTermInfo</a></dt>
<dd>
<div class="block">Set all Labels for NTerm info</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#setnTermRef(bacnet.datamodel.proteomics.NTerm)" class="member-name-link">setnTermRef(NTerm)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setnTerms(java.util.ArrayList)" class="member-name-link">setnTerms(ArrayList&lt;NTermData&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#setnTerms(java.util.ArrayList)" class="member-name-link">setnTerms(ArrayList&lt;NTerm&gt;)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setnTermTypeOverlap(java.lang.String)" class="member-name-link">setnTermTypeOverlap(String)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setNucleotide(boolean)" class="member-name-link">setNucleotide(boolean)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#setOldLocusTag(java.lang.String)" class="member-name-link">setOldLocusTag(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#setOperon(java.lang.String)" class="member-name-link">setOperon(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setOperons(java.util.ArrayList)" class="member-name-link">setOperons(ArrayList&lt;Operon&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setOperons(java.util.LinkedHashMap)" class="member-name-link">setOperons(LinkedHashMap&lt;String, Operon&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setOrdered(boolean)" class="member-name-link">setOrdered(boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setOrdered(boolean,java.util.ArrayList)" class="member-name-link">setOrdered(boolean, ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setOrderedRowNames(java.util.ArrayList)" class="member-name-link">setOrderedRowNames(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setOriginalHeader(java.lang.String)" class="member-name-link">setOriginalHeader(String)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setOverASrnas(java.lang.String%5B%5D)" class="member-name-link">setOverASrnas(String[])</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#setOverBioConds(java.util.TreeSet)" class="member-name-link">setOverBioConds(TreeSet&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setOverCisRegs(java.lang.String%5B%5D)" class="member-name-link">setOverCisRegs(String[])</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setOverGenes(java.lang.String%5B%5D)" class="member-name-link">setOverGenes(String[])</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setOverlap(java.lang.String)" class="member-name-link">setOverlap(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setOverlapInFrame(boolean)" class="member-name-link">setOverlapInFrame(boolean)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#setOverlapingGene(java.lang.String)" class="member-name-link">setOverlapingGene(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#setOverlaps(java.util.ArrayList)" class="member-name-link">setOverlaps(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setOverSrnas(java.lang.String%5B%5D)" class="member-name-link">setOverSrnas(String[])</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/ProteinSequence.html#setParentDNASequence(org.biojava3.core.sequence.template.AbstractSequence,java.lang.Integer,java.lang.Integer)" class="member-name-link">setParentDNASequence(AbstractSequence, Integer, Integer)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></dt>
<dd>
<div class="block">A Protein sequence can be stand alone or loaded from a transcript sequence.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setParentFeature(org.biojava3.core.sequence.features.FeatureInterface)" class="member-name-link">setParentFeature(FeatureInterface&lt;S, C&gt;)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">A feature can be the child or contained by a parent feature.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setParentFeature(org.biojava3.core.sequence.features.FeatureInterface)" class="member-name-link">setParentFeature(FeatureInterface&lt;S, C&gt;)</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Set the parent feature</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setParentGenome(bacnet.datamodel.sequence.Genome)" class="member-name-link">setParentGenome(Genome)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setParentGenome(bacnet.datamodel.sequenceNCBI.GenomeNCBI)" class="member-name-link">setParentGenome(GenomeNCBI)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)" class="member-name-link">setParentSequence(AbstractSequence&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setParentViewId(java.lang.String)" class="member-name-link">setParentViewId(String)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setPartService(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">setPartService(EPartService)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setPath(java.lang.String)" class="member-name-link">setPath(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#setPATH(java.lang.String)" class="member-name-link">setPATH(String)</a> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#setPeptides(java.util.HashMap)" class="member-name-link">setPeptides(HashMap&lt;String, ArrayList&lt;NTerm&gt;&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Link peptideSequence to <code>ArrayList(NTerm)</code><br>
 This correspond to the list of <code>NTerm</code> having same peptideSequence</div>
</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#setPreloadedItems(org.eclipse.swt.widgets.Composite)" class="member-name-link">setPreloadedItems(Composite)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setPreviousCodon(java.lang.String)" class="member-name-link">setPreviousCodon(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#setPrideLink(java.lang.String)" class="member-name-link">setPrideLink(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#setProbes(java.util.TreeMap)" class="member-name-link">setProbes(TreeMap&lt;Integer, Integer&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#setProbes(java.util.TreeMap)" class="member-name-link">setProbes(TreeMap&lt;String, Integer&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setProbesGExpression(java.util.TreeMap)" class="member-name-link">setProbesGExpression(TreeMap&lt;String, Integer&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setProbesTiling(java.util.TreeMap)" class="member-name-link">setProbesTiling(TreeMap&lt;Integer, Integer&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#setProduct(java.lang.String)" class="member-name-link">setProduct(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/NcRNA.html#setProduct(java.lang.String)" class="member-name-link">setProduct(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setProjectName(java.lang.String)" class="member-name-link">setProjectName(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#setProperties(java.util.LinkedHashMap)" class="member-name-link">setProperties(LinkedHashMap&lt;String, String&gt;)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#setProtein_id(java.lang.String)" class="member-name-link">setProtein_id(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setProteinIDTolocusTag(java.util.LinkedHashMap)" class="member-name-link">setProteinIDTolocusTag(LinkedHashMap&lt;String, String&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setProteinIDToOldLocusTagMap(java.util.LinkedHashMap)" class="member-name-link">setProteinIDToOldLocusTagMap(LinkedHashMap&lt;String, String&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#setProteinNCBILink(java.lang.String)" class="member-name-link">setProteinNCBILink(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setProteomes(java.util.ArrayList)" class="member-name-link">setProteomes(ArrayList&lt;ProteomicsData&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DataValidation.html#setProteomes(java.util.HashMap)" class="member-name-link">setProteomes(HashMap&lt;String, Boolean&gt;)</a> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setProteomesArrayPath(java.lang.String)" class="member-name-link">setProteomesArrayPath(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setProteomesComparisonsArrayPath(java.lang.String)" class="member-name-link">setProteomesComparisonsArrayPath(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)" class="member-name-link">setProxySequenceReader(SequenceReader&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Very important method that allows external mappings of sequence data and features.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Gene.html#setPseudogene(boolean)" class="member-name-link">setPseudogene(boolean)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#setPubMedLink(java.lang.String)" class="member-name-link">setPubMedLink(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/QualityFeature.html#setQualities(java.util.List)" class="member-name-link">setQualities(List&lt;Number&gt;)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/QualityFeature.html" title="class in org.biojava3.core.sequence.features">QualityFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/QuantityFeature.html#setQuantities(java.util.List)" class="member-name-link">setQuantities(List&lt;Number&gt;)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/QuantityFeature.html" title="class in org.biojava3.core.sequence.features">QuantityFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#setRawDatas(java.util.ArrayList)" class="member-name-link">setRawDatas(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#setRead(boolean%5B%5D)" class="member-name-link">setRead(boolean[])</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorSWT.html#setRed(int)" class="member-name-link">setRed(int)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setRef(java.lang.String)" class="member-name-link">setRef(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#setReference(java.lang.String)" class="member-name-link">setReference(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setReference(java.lang.String)" class="member-name-link">setReference(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#setRefSequence(java.lang.String)" class="member-name-link">setRefSequence(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/TestGenomeBrowser.html#setRelativeValueDisplay()" class="member-name-link">setRelativeValueDisplay()</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/TestGenomeBrowser.html" title="class in bacnet.genomeBrowser">TestGenomeBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#setRightBCs(java.util.ArrayList)" class="member-name-link">setRightBCs(ArrayList&lt;String&gt;)</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setRowNameCellColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setRowNameCellColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setRowNames(java.util.TreeMap)" class="member-name-link">setRowNames(TreeMap&lt;String, Integer&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setRowNameTextColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setRowNameTextColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#setScore(double)" class="member-name-link">setScore(double)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setScore(int)" class="member-name-link">setScore(int)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#setScrolledComposite(org.eclipse.swt.custom.ScrolledComposite)" class="member-name-link">setScrolledComposite(ScrolledComposite)</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setScrolledComposite(org.eclipse.swt.custom.ScrolledComposite)" class="member-name-link">setScrolledComposite(ScrolledComposite)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#setSDBindingEnergy()" class="member-name-link">setSDBindingEnergy()</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">For every gene calculate the SD-aSD hybridization affinity and save it in:
 /StatGenes/aSDBindingEneryg.excel</div>
</dd>
<dt><a href="../bacnet/table/gui/TableFilter.html#setSearchText(java.lang.String)" class="member-name-link">setSearchText(String)</a> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/TableFilter.html" title="class in bacnet.table.gui">TableFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setSecondRowName(java.lang.String)" class="member-name-link">setSecondRowName(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#setSelectedGenes(java.util.ArrayList)" class="member-name-link">setSelectedGenes(ArrayList&lt;String&gt;)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setSeq(bacnet.datamodel.sequence.Srna)" class="member-name-link">setSeq(Srna)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setSequence(bacnet.datamodel.sequence.Gene)" class="member-name-link">setSequence(Gene)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setSequence(java.lang.String)" class="member-name-link">setSequence(String)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#setSequence(java.lang.String)" class="member-name-link">setSequence(String)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#setSequence(java.lang.String)" class="member-name-link">setSequence(String)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html#setSequenceCollection(org.biojava3.core.sequence.SequenceOptimizationHints.SequenceCollection)" class="member-name-link">setSequenceCollection(SequenceOptimizationHints.SequenceCollection)</a> - Static method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setSequenceMap(java.lang.String)" class="member-name-link">setSequenceMap(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setSequencePeptide(java.lang.String)" class="member-name-link">setSequencePeptide(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setSequenceScore(java.lang.Double)" class="member-name-link">setSequenceScore(Double)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html#setSequenceUsage(org.biojava3.core.sequence.SequenceOptimizationHints.SequenceUsage)" class="member-name-link">setSequenceUsage(SequenceOptimizationHints.SequenceUsage)</a> - Static method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#setSetNames(java.util.ArrayList)" class="member-name-link">setSetNames(ArrayList&lt;String&gt;)</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#setSets(java.util.TreeMap)" class="member-name-link">setSets(TreeMap&lt;String, ArrayList&lt;String&gt;&gt;)</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setShell(org.eclipse.swt.widgets.Shell)" class="member-name-link">setShell(Shell)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setShortDescription(java.lang.String)" class="member-name-link">setShortDescription(String)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setShortDescription(java.lang.String)" class="member-name-link">setShortDescription(String)</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Set the short description that can be used to describe the feature</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/compound/CodonCompound.html#setShortName(java.lang.String)" class="member-name-link">setShortName(String)</a> - Method in class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#setShortName(java.lang.String)" class="member-name-link">setShortName(String)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/Compound.html#setShortName(java.lang.String)" class="member-name-link">setShortName(String)</a> - Method in interface org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html#setShortName(java.lang.String)" class="member-name-link">setShortName(String)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Table.Codon.html#setShortName(java.lang.String)" class="member-name-link">setShortName(String)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/SignatureSelectionDialog.html#setSignature(bacnet.datamodel.annotation.Signature)" class="member-name-link">setSignature(Signature)</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/SignatureSelectionDialog.html" title="class in bacnet.expressionAtlas.core">SignatureSelectionDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setSignatures(java.util.TreeSet)" class="member-name-link">setSignatures(TreeSet&lt;String&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setSignaturesNametoID(java.util.TreeMap)" class="member-name-link">setSignaturesNametoID(TreeMap&lt;String, String&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#setSize(int)" class="member-name-link">setSize(int)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setSoluble(boolean)" class="member-name-link">setSoluble(boolean)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setSource(java.lang.String)" class="member-name-link">setSource(String)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">Set the feature source</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setSource(java.lang.String)" class="member-name-link">setSource(String)</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Change the source of the FeatureInterface.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setSource(java.lang.String)" class="member-name-link">setSource(String)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>
<div class="block">Added support for the source of this sequence for GFF3 export</div>
</dd>
<dt><a href="../bacnet/Database.html#setSpecies(java.lang.String)" class="member-name-link">setSpecies(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#setSpecies(java.lang.String)" class="member-name-link">setSpecies(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#setSpecies(java.lang.String)" class="member-name-link">setSpecies(String)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#setSpecies(java.lang.String)" class="member-name-link">setSpecies(String)</a> - Method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setSpectra(int)" class="member-name-link">setSpectra(int)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setSpectrum(int%5B%5D)" class="member-name-link">setSpectrum(int[])</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#setSrnaLink(java.lang.String,java.util.HashMap)" class="member-name-link">setSrnaLink(String, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setsRNAListEGDe(java.util.ArrayList)" class="member-name-link">setsRNAListEGDe(ArrayList&lt;String&gt;)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#setSrnaOrder()" class="member-name-link">setSrnaOrder()</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Classify Srna par IDs and save it in PATH_LISTSRNA</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setsRNAs(java.util.ArrayList)" class="member-name-link">setsRNAs(ArrayList&lt;Srna&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Chromosome.html#setsRNAs(java.util.LinkedHashMap)" class="member-name-link">setsRNAs(LinkedHashMap&lt;String, Srna&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setStartCode(java.lang.String)" class="member-name-link">setStartCode(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setStartCodon(java.lang.String)" class="member-name-link">setStartCodon(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#setStat()" class="member-name-link">setStat()</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">Set statistical parameters using values array</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setStat()" class="member-name-link">setStat()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setStat()" class="member-name-link">setStat()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setStatDataColor()" class="member-name-link">setStatDataColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setStatDataColor()" class="member-name-link">setStatDataColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setStatDataValues()" class="member-name-link">setStatDataValues()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setStatDataValues()" class="member-name-link">setStatDataValues()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setStatMapper()" class="member-name-link">setStatMapper()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>
<div class="block">Set parameters corresponding to stat data i.e. value from 0 to 1</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setStatMapper()" class="member-name-link">setStatMapper()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>
<div class="block">Set parameters corresponding to stat data i.e. value from 0 to 1</div>
</dd>
<dt><a href="../bacnet/Database.html#setStatTable(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">setStatTable(ExpressionMatrix)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setStdevDataColor()" class="member-name-link">setStdevDataColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setStdevDataColor()" class="member-name-link">setStdevDataColor()</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#setStrand(char)" class="member-name-link">setStrand(char)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setStrand(char)" class="member-name-link">setStrand(char)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#setStrand(org.biojava3.core.sequence.Strand)" class="member-name-link">setStrand(Strand)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/GeneSequence.html#setStrand(org.biojava3.core.sequence.Strand)" class="member-name-link">setStrand(Strand)</a> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setStudyName(java.lang.String)" class="member-name-link">setStudyName(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#setSubsetNames(java.util.ArrayList)" class="member-name-link">setSubsetNames(ArrayList&lt;String&gt;)</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html#setSubsets(java.util.TreeMap)" class="member-name-link">setSubsets(TreeMap&lt;String, ArrayList&lt;String&gt;&gt;)</a> - Method in class bacnet.scripts.core.vennDiagram.<a href="../bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setSynonym(java.util.TreeSet)" class="member-name-link">setSynonym(TreeSet&lt;String&gt;)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setTabFolder(org.eclipse.swt.widgets.TabFolder)" class="member-name-link">setTabFolder(TabFolder)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setTabFolder(org.eclipse.swt.widgets.TabFolder)" class="member-name-link">setTabFolder(TabFolder)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setTabFolder(org.eclipse.swt.widgets.TabFolder)" class="member-name-link">setTabFolder(TabFolder)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#setTabFolder(org.eclipse.swt.widgets.TabFolder)" class="member-name-link">setTabFolder(TabFolder)</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#setTable(org.eclipse.swt.widgets.Table)" class="member-name-link">setTable(Table)</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#setTableBioCondition(org.eclipse.jface.viewers.TableViewer)" class="member-name-link">setTableBioCondition(TableViewer)</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#setTableBioCondition(org.eclipse.jface.viewers.TableViewer)" class="member-name-link">setTableBioCondition(TableViewer)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#setTableBioCondition(org.eclipse.swt.widgets.Table)" class="member-name-link">setTableBioCondition(Table)</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setTableComposite(bacnet.expressionAtlas.TableCompositeHeatMap)" class="member-name-link">setTableComposite(TableCompositeHeatMap)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#setTableComposite(bacnet.expressionAtlas.TableCompositeHeatMap)" class="member-name-link">setTableComposite(TableCompositeHeatMap)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#setTableComposite(bacnet.expressionAtlas.TableCompositeHeatMap)" class="member-name-link">setTableComposite(TableCompositeHeatMap)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#setTableComposite(bacnet.expressionAtlas.TableCompositeHeatMap)" class="member-name-link">setTableComposite(TableCompositeHeatMap)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/TableSWTView.html#setTableComposite(bacnet.table.TableSWTComposite)" class="member-name-link">setTableComposite(TableSWTComposite)</a> - Method in class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#setTableElementName(java.lang.String)" class="member-name-link">setTableElementName(String)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#setTableElementName(java.lang.String)" class="member-name-link">setTableElementName(String)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setTableGenes(org.eclipse.swt.widgets.Table)" class="member-name-link">setTableGenes(Table)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#setTableGenes(org.eclipse.swt.widgets.Table)" class="member-name-link">setTableGenes(Table)</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setTableNodiff(org.eclipse.swt.widgets.Table)" class="member-name-link">setTableNodiff(Table)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setTableNodiffProteome(org.eclipse.swt.widgets.Table)" class="member-name-link">setTableNodiffProteome(Table)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setTableOtherInfo(org.eclipse.swt.widgets.Table)" class="member-name-link">setTableOtherInfo(Table)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setTableOver(org.eclipse.swt.widgets.Table)" class="member-name-link">setTableOver(Table)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setTableOverProteome(org.eclipse.swt.widgets.Table)" class="member-name-link">setTableOverProteome(Table)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setTableProteomes(org.eclipse.swt.widgets.Table)" class="member-name-link">setTableProteomes(Table)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setTableSrnas(org.eclipse.swt.widgets.Table)" class="member-name-link">setTableSrnas(Table)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setTableTranscriptomes(org.eclipse.swt.widgets.Table)" class="member-name-link">setTableTranscriptomes(Table)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setTableUnder(org.eclipse.swt.widgets.Table)" class="member-name-link">setTableUnder(Table)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setTableUnderProteome(org.eclipse.swt.widgets.Table)" class="member-name-link">setTableUnderProteome(Table)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setTaxonomy(org.biojava3.core.sequence.TaxonomyID)" class="member-name-link">setTaxonomy(TaxonomyID)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#setTbtmGraphVisualization(org.eclipse.swt.widgets.TabItem)" class="member-name-link">setTbtmGraphVisualization(TabItem)</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setTbtmHeatmap(org.eclipse.swt.widgets.TabItem)" class="member-name-link">setTbtmHeatmap(TabItem)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setTbtmScatterPlot(org.eclipse.swt.widgets.TabItem)" class="member-name-link">setTbtmScatterPlot(TabItem)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setTemperature(java.lang.String)" class="member-name-link">setTemperature(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setTestData(boolean)" class="member-name-link">setTestData(boolean)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#setTests(java.util.ArrayList)" class="member-name-link">setTests(ArrayList&lt;StatTest&gt;)</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/DownloadServiceHandler.html#setText(java.lang.String)" class="member-name-link">setText(String)</a> - Method in class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setTextColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setTextColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setTextColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setTextColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setTextColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setTextColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setTextColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setTextColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#setTextCutOff(org.eclipse.swt.widgets.Text)" class="member-name-link">setTextCutOff(Text)</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setTextHeaderColor(org.eclipse.swt.graphics.Color)" class="member-name-link">setTextHeaderColor(Color)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setTextHeaderColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setTextHeaderColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setTextHeaderColorSWT(bacnet.table.core.ColorSWT)" class="member-name-link">setTextHeaderColorSWT(ColorSWT)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#setTextSearch(org.eclipse.swt.widgets.Text)" class="member-name-link">setTextSearch(Text)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#setTextSearch(org.eclipse.swt.widgets.Text)" class="member-name-link">setTextSearch(Text)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPool(java.util.concurrent.ThreadPoolExecutor)" class="member-name-link">setThreadPool(ThreadPoolExecutor)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Sets thread pool to any given <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/ThreadPoolExecutor.html" title="class or interface in java.util.concurrent" class="external-link"><code>ThreadPoolExecutor</code></a> to allow use of an alternative execution
 style.</div>
</dd>
<dt><a href="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPoolCPUsAvailable(int)" class="member-name-link">setThreadPoolCPUsAvailable(int)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Sets thread pool to reserve a given number of processor cores for foreground or other use.</div>
</dd>
<dt><a href="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPoolCPUsFraction(float)" class="member-name-link">setThreadPoolCPUsFraction(float)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Sets thread pool to a given fraction of the available processors.</div>
</dd>
<dt><a href="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPoolDefault()" class="member-name-link">setThreadPoolDefault()</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Sets thread pool to default of 1 background thread for each processor core.</div>
</dd>
<dt><a href="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPoolSingle()" class="member-name-link">setThreadPoolSingle()</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Sets thread pool to a single background thread.</div>
</dd>
<dt><a href="../org/biojava3/core/util/ConcurrencyTools.html#setThreadPoolSize(int)" class="member-name-link">setThreadPoolSize(int)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Sets thread pool to given size.</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setThreshold(int)" class="member-name-link">setThreshold(int)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#setTiling(boolean)" class="member-name-link">setTiling(boolean)</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setTilings(java.util.ArrayList)" class="member-name-link">setTilings(ArrayList&lt;Tiling&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setTime(java.lang.String)" class="member-name-link">setTime(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#setTIRSequence()" class="member-name-link">setTIRSequence()</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">We create a fasta file with all the TIR DNA sequence (-18 +63)</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#setTIRStructureEnergy()" class="member-name-link">setTIRStructureEnergy()</a> - Method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">For every gene calculate the 2nd structure free energy and save it a table<br>
 We calculate the energy for 40 different sequence: -20+40 to -20+80 and calculate the average
 energy</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#setTisList(java.util.ArrayList)" class="member-name-link">setTisList(ArrayList&lt;TIS&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#setTisMap(java.util.HashMap)" class="member-name-link">setTisMap(HashMap&lt;NTerm, TIS&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setTISName(java.lang.String)" class="member-name-link">setTISName(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/InternalBrowser.html#setTitle(java.lang.String)" class="member-name-link">setTitle(String)</a> - Method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/BannerView.html#setToolBar(org.eclipse.swt.widgets.ToolBar)" class="member-name-link">setToolBar(ToolBar)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BannerView.html" title="class in bacnet.e4.rap">BannerView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#setTotalBP(int)" class="member-name-link">setTotalBP(int)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/TIS.html#setTotalSpectra(int)" class="member-name-link">setTotalSpectra(int)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#setTrack(bacnet.genomeBrowser.core.Track)" class="member-name-link">setTrack(Track)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html#setTrack(bacnet.genomeBrowser.core.Track)" class="member-name-link">setTrack(Track)</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddProteomicsDataDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html#setTrack(bacnet.genomeBrowser.core.Track)" class="member-name-link">setTrack(Track)</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddTranscriptomicsDataDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#setTrack(bacnet.genomeBrowser.core.Track)" class="member-name-link">setTrack(Track)</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setTrack(bacnet.genomeBrowser.core.Track)" class="member-name-link">setTrack(Track)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">Set the data of the Canvas, and implement the PaintListener
 
 Execute different tasks:<br>
 Fix the number of pixel for one base pair
 Set width and height parameters for data and annotation track Display in order:<br>
 different line and ticks for the base pair position, every display needed in the background
 display all the data
 display sequence if needed
 display annotation
 different genome name and mouse line in foreground</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setTrack(bacnet.genomeBrowser.core.Track)" class="member-name-link">setTrack(Track)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>
<div class="block">Set the data of the Canvas, and implement the PaintListener
 
 Execute different tasks:<br>
 Fix the number of pixel for one base pair
 Set width and height parameters for data and annotation track Display in order:<br>
 different line and ticks for the base pair position, every display needed in the background
 display all the data
 display sequence if needed
 display annotation
 different genome name and mouse line in foreground</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setTrack(bacnet.genomeBrowser.core.Track)" class="member-name-link">setTrack(Track)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setTrackComposite(org.eclipse.swt.widgets.Composite)" class="member-name-link">setTrackComposite(Composite)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setTrackGenome(bacnet.genomeBrowser.core.Track)" class="member-name-link">setTrackGenome(Track)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#setTracksComposite(bacnet.genomeBrowser.tracksGUI.TracksComposite)" class="member-name-link">setTracksComposite(TracksComposite)</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DataValidation.html#setTranscriptomes(java.util.HashMap)" class="member-name-link">setTranscriptomes(HashMap&lt;String, Boolean&gt;)</a> - Method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setTranscriptomesArrayPath(java.lang.String)" class="member-name-link">setTranscriptomesArrayPath(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setTranscriptomesComparisonsArrayPath(java.lang.String)" class="member-name-link">setTranscriptomesComparisonsArrayPath(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#setTranscriptomics(boolean)" class="member-name-link">setTranscriptomics(boolean)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setTssDownCoverage(int)" class="member-name-link">setTssDownCoverage(int)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setTssDownDistance(int)" class="member-name-link">setTssDownDistance(int)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setTssDowntype(java.lang.String)" class="member-name-link">setTssDowntype(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setTssUpCoverage(int)" class="member-name-link">setTssUpCoverage(int)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setTssUpDistance(int)" class="member-name-link">setTssUpDistance(int)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setTssUptype(java.lang.String)" class="member-name-link">setTssUptype(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setTxtCutoffLogFC(org.eclipse.swt.widgets.Text)" class="member-name-link">setTxtCutoffLogFC(Text)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setTxtCutoffLogFC(org.eclipse.swt.widgets.Text)" class="member-name-link">setTxtCutoffLogFC(Text)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setTxtCutoffLogFCProteome(org.eclipse.swt.widgets.Text)" class="member-name-link">setTxtCutoffLogFCProteome(Text)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setTxtCutoffPvalue(org.eclipse.swt.widgets.Text)" class="member-name-link">setTxtCutoffPvalue(Text)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#setTxtSearch(org.eclipse.swt.widgets.Text)" class="member-name-link">setTxtSearch(Text)</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#setTxtSearch(org.eclipse.swt.widgets.Text)" class="member-name-link">setTxtSearch(Text)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#setTxtSearch(org.eclipse.swt.widgets.Text)" class="member-name-link">setTxtSearch(Text)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setTxtSeqnucleotide(org.eclipse.swt.widgets.Text)" class="member-name-link">setTxtSeqnucleotide(Text)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#setType(bacnet.datamodel.dataset.OmicsData.TypeData)" class="member-name-link">setType(OmicsData.TypeData)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#setType(bacnet.datamodel.sequence.Sequence.SeqType)" class="member-name-link">setType(Sequence.SeqType)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.html#setType(bacnet.scripts.core.stat.StatTest.TypeStat)" class="member-name-link">setType(StatTest.TypeStat)</a> - Method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.html" title="class in bacnet.scripts.core.stat">StatTest</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setType(bacnet.table.core.ColorMapper.TypeMapper)" class="member-name-link">setType(ColorMapper.TypeMapper)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setType(bacnet.table.core.ColorMapperRCP.Type)" class="member-name-link">setType(ColorMapperRCP.Type)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/RfamElement.html#setType(java.lang.String)" class="member-name-link">setType(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#setType(java.lang.String)" class="member-name-link">setType(String)</a> - Method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setType(java.lang.String)" class="member-name-link">setType(String)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">Set the feature type</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setType(java.lang.String)" class="member-name-link">setType(String)</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>
<div class="block">Change the type of this feature.</div>
</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#setTypeData(bacnet.datamodel.dataset.OmicsData.TypeData)" class="member-name-link">setTypeData(OmicsData.TypeData)</a> - Method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setTypeDataContained(java.util.TreeSet)" class="member-name-link">setTypeDataContained(TreeSet&lt;OmicsData.TypeData&gt;)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.html#setTypeFilter(bacnet.table.core.Filter.TypeFilter)" class="member-name-link">setTypeFilter(Filter.TypeFilter)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.html#setTypeFilter(bacnet.utils.Filter.TypeFilter)" class="member-name-link">setTypeFilter(Filter.TypeFilter)</a> - Method in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setTypeModif(bacnet.datamodel.proteomics.NTerm.TypeModif)" class="member-name-link">setTypeModif(NTerm.TypeModif)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/NcRNA.html#setTypeNcRNA(bacnet.datamodel.sequence.NcRNA.TypeNcRNA)" class="member-name-link">setTypeNcRNA(NcRNA.TypeNcRNA)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.html#setTypeOverlap(java.lang.String)" class="member-name-link">setTypeOverlap(String)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/NTermData.html#setTypeOverlaps(java.util.ArrayList)" class="member-name-link">setTypeOverlaps(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#setTypeOverlaps(java.util.ArrayList)" class="member-name-link">setTypeOverlaps(ArrayList&lt;String&gt;)</a> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#setTypeSrna(bacnet.datamodel.sequence.Srna.TypeSrna)" class="member-name-link">setTypeSrna(Srna.TypeSrna)</a> - Method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setUnderASrnas(java.lang.String%5B%5D)" class="member-name-link">setUnderASrnas(String[])</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#setUnderBioConds(java.util.TreeSet)" class="member-name-link">setUnderBioConds(TreeSet&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setUnderCisRegs(java.lang.String%5B%5D)" class="member-name-link">setUnderCisRegs(String[])</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setUnderGenes(java.lang.String%5B%5D)" class="member-name-link">setUnderGenes(String[])</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#setUnderSrnas(java.lang.String%5B%5D)" class="member-name-link">setUnderSrnas(String[])</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#setUniprotbaseURL(java.lang.String)" class="member-name-link">setUniprotbaseURL(String)</a> - Static method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#setUniprotDirectoryCache(java.lang.String)" class="member-name-link">setUniprotDirectoryCache(String)</a> - Static method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setUpDNAList(java.util.ArrayList)" class="member-name-link">setUpDNAList(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/SetupPart.html" class="type-name-link" title="class in bacnet.e4.rap.setup">SetupPart</a> - Class in <a href="../bacnet/e4/rap/setup/package-summary.html">bacnet.e4.rap.setup</a></dt>
<dd>
<div class="block">SetupPart for initialisation and creation of the Database</div>
</dd>
<dt><a href="../bacnet/e4/rap/setup/SetupPart.html#%3Cinit%3E()" class="member-name-link">SetupPart()</a> - Constructor for class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SetupPart.html" title="class in bacnet.e4.rap.setup">SetupPart</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setUpProtList(java.util.ArrayList)" class="member-name-link">setUpProtList(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setUpRNAList(java.util.ArrayList)" class="member-name-link">setUpRNAList(ArrayList&lt;String&gt;)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#setUserCollection(java.util.Collection)" class="member-name-link">setUserCollection(Collection&lt;Object&gt;)</a> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#setUserObject(java.lang.Object)" class="member-name-link">setUserObject(Object)</a> - Method in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>
<div class="block">Allow the user to associate an object with the feature.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/FeatureInterface.html#setUserObject(java.lang.Object)" class="member-name-link">setUserObject(Object)</a> - Method in interface org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setValue(double,int,int)" class="member-name-link">setValue(double, int, int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setValue(double,java.lang.String,java.lang.String)" class="member-name-link">setValue(double, String, String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#setValue(int,double)" class="member-name-link">setValue(int, double)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setValueAnnotation(java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">setValueAnnotation(String, String, String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#setValues(double%5B%5D)" class="member-name-link">setValues(double[])</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setValues(double%5B%5D)" class="member-name-link">setValues(double[])</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setValues(double%5B%5D)" class="member-name-link">setValues(double[])</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setValues(double%5B%5D%5B%5D)" class="member-name-link">setValues(double[][])</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#setValues(int,int)" class="member-name-link">setValues(int, int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Create new double[][] values with nbRow and nbColumn All value will then be equal to 0</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setValues(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">setValues(ExpressionMatrix)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setValues(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">setValues(ExpressionMatrix)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#setVar(double)" class="member-name-link">setVar(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#setVar(double)" class="member-name-link">setVar(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setVariables()" class="member-name-link">setVariables()</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">MOST IMPORTANT INIT METHOD : Set all variables for the website</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionData.html#setVariance(double)" class="member-name-link">setVariance(double)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#setVerticalBarProperties()" class="member-name-link">setVerticalBarProperties()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Depending on <code>CanvasData</code> parameters fix properties of VerticalBar</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/Network.html#setVertices(java.util.HashMap)" class="member-name-link">setVertices(HashMap&lt;String, String&gt;)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></dt>
<dd>
<div class="block">List of vertices = genome elements + their attributes = genome position, type, annotation</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#setViewId(java.lang.String)" class="member-name-link">setViewId(String)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#setViewId(java.lang.String)" class="member-name-link">setViewId(String)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#setViewId(java.lang.String)" class="member-name-link">setViewId(String)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#setViewId(java.lang.String)" class="member-name-link">setViewId(String)</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#setViewId(java.lang.String)" class="member-name-link">setViewId(String)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#setViewID(java.lang.String)" class="member-name-link">setViewID(String)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#setViewID(java.lang.String)" class="member-name-link">setViewID(String)</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#setWebpageTitle(java.lang.String)" class="member-name-link">setWebpageTitle(String)</a> - Method in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#setWidth(int)" class="member-name-link">setWidth(int)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/MatrixSize.html#setWidth(java.util.HashMap)" class="member-name-link">setWidth(HashMap&lt;String, Integer&gt;)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setWidthBP(int)" class="member-name-link">setWidthBP(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#setWidthBP(int)" class="member-name-link">setWidthBP(int)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/MatrixSize.html#setWidthDefault(int)" class="member-name-link">setWidthDefault(int)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/MatrixSize.html#setWidthRowNameDefault(int)" class="member-name-link">setWidthRowNameDefault(int)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#setWildType(boolean)" class="member-name-link">setWildType(boolean)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#setX1(int)" class="member-name-link">setX1(int)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#setX2(int)" class="member-name-link">setX2(int)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#setY1(int)" class="member-name-link">setY1(int)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Region.html#setY2(int)" class="member-name-link">setY2(int)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.html#setZoom(bacnet.genomeBrowser.core.Zoom)" class="member-name-link">setZoom(Zoom)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#setZoomChangedFlag(boolean)" class="member-name-link">setZoomChangedFlag(boolean)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#setZoomHeightMax(int)" class="member-name-link">setZoomHeightMax(int)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#setZoomHeightPosition(int)" class="member-name-link">setZoomHeightPosition(int)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#setZoomNumber(int)" class="member-name-link">setZoomNumber(int)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#setZoomPosition(int)" class="member-name-link">setZoomPosition(int)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#setZoomPosition(bacnet.genomeBrowser.core.Region)" class="member-name-link">setZoomPosition(Region)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>
<div class="block">Search the zoom position which correspond the best to displayRegion DisplayRegion should be
 inferior to display range given by the zoom</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Zoom.html#setZoomRatio(double)" class="member-name-link">setZoomRatio(double)</a> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#setZoomVertical(double)" class="member-name-link">setZoomVertical(double)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/GenomesCreation.html#showAllLocusID()" class="member-name-link">showAllLocusID()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/HelpPage.html#showHelp(java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">showHelp(String, EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/HelpPage.html" title="class in bacnet.views">HelpPage</a></dt>
<dd>
<div class="block">Show the help page in an external browser</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#showInfoPanel()" class="member-name-link">showInfoPanel()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Show the info Panel showing NTerm data</div>
</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#showSearchPath()" class="member-name-link">showSearchPath()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#shuffle(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">shuffle(Sequence&lt;C&gt;)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Implements sequence shuffling by first materializing the given <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a> into a
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link"><code>List</code></a>, applying <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Collections.html#shuffle(java.util.List)" title="class or interface in java.util" class="external-link"><code>Collections.shuffle(List)</code></a> and then returning the shuffled elements
 in a new instance of <code>SequenceBackingStore</code> which behaves as a <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a>.</div>
</dd>
<dt><a href="../org/biojava3/core/util/ConcurrencyTools.html#shutdown()" class="member-name-link">shutdown()</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Disables new tasks from being submitted and closes the thread pool cleanly.</div>
</dd>
<dt><a href="../org/biojava3/core/util/ConcurrencyTools.html#shutdownAndAwaitTermination()" class="member-name-link">shutdownAndAwaitTermination()</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Closes the thread pool.</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html" class="type-name-link" title="class in bacnet.datamodel.annotation">Signature</a> - Class in <a href="../bacnet/datamodel/annotation/package-summary.html">bacnet.datamodel.annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Signature.html#%3Cinit%3E()" class="member-name-link">Signature()</a> - Constructor for class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/SignatureSelectionDialog.html" class="type-name-link" title="class in bacnet.expressionAtlas.core">SignatureSelectionDialog</a> - Class in <a href="../bacnet/expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/SignatureSelectionDialog.html#%3Cinit%3E(org.eclipse.swt.widgets.Shell)" class="member-name-link">SignatureSelectionDialog(Shell)</a> - Constructor for class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/SignatureSelectionDialog.html" title="class in bacnet.expressionAtlas.core">SignatureSelectionDialog</a></dt>
<dd>
<div class="block">Create the dialog.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimpleLocation.html" class="type-name-link" title="class in org.biojava3.core.sequence.location">SimpleLocation</a> - Class in <a href="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</a></dt>
<dd>
<div class="block">Very basic implementation of the Location interface which defines a series of simple
 constructors.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(int,int)" class="member-name-link">SimpleLocation(int, int)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(int,int,org.biojava3.core.sequence.Strand)" class="member-name-link">SimpleLocation(int, int, Strand)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(int,int,org.biojava3.core.sequence.Strand,org.biojava3.core.sequence.location.template.Location...)" class="member-name-link">SimpleLocation(int, int, Strand, Location...)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point)" class="member-name-link">SimpleLocation(Point, Point)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand)" class="member-name-link">SimpleLocation(Point, Point, Strand)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,boolean)" class="member-name-link">SimpleLocation(Point, Point, Strand, boolean, boolean)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,boolean,java.util.List)" class="member-name-link">SimpleLocation(Point, Point, Strand, boolean, boolean, List&lt;Location&gt;)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,java.util.List)" class="member-name-link">SimpleLocation(Point, Point, Strand, boolean, List&lt;Location&gt;)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,org.biojava3.core.sequence.AccessionID)" class="member-name-link">SimpleLocation(Point, Point, Strand, boolean, AccessionID)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,org.biojava3.core.sequence.location.template.Location...)" class="member-name-link">SimpleLocation(Point, Point, Strand, boolean, Location...)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,org.biojava3.core.sequence.AccessionID)" class="member-name-link">SimpleLocation(Point, Point, Strand, AccessionID)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimpleLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,org.biojava3.core.sequence.location.template.Location...)" class="member-name-link">SimpleLocation(Point, Point, Strand, Location...)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimplePoint.html" class="type-name-link" title="class in org.biojava3.core.sequence.location">SimplePoint</a> - Class in <a href="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</a></dt>
<dd>
<div class="block">Basic implementation of the Point interface.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimplePoint.html#%3Cinit%3E(int)" class="member-name-link">SimplePoint(int)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimplePoint.html#%3Cinit%3E(int,boolean,boolean)" class="member-name-link">SimplePoint(int, boolean, boolean)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" class="type-name-link" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a>&lt;<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</a></dt>
<dd>
<div class="block">An implementation of the SequenceReader interface which for every call will return only 1
 compound (given to it during construction; a String is also valid but will require a
 CompoundSet).</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#%3Cinit%3E(C,org.biojava3.core.sequence.template.CompoundSet,int)" class="member-name-link">SingleCompoundSequenceReader(C, CompoundSet&lt;C&gt;, int)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Build the object with a compound rather than a String</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet,int)" class="member-name-link">SingleCompoundSequenceReader(String, CompoundSet&lt;C&gt;, int)</a> - Constructor for class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a></dt>
<dd>
<div class="block">Public constructor to be used with String based constructor</div>
</dd>
<dt><a href="../bacnet/views/CRISPRPredictView.html#SINGULARITY_PATH" class="member-name-link">SINGULARITY_PATH</a> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView2.html#SINGULARITY_PATH" class="member-name-link">SINGULARITY_PATH</a> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#sir_graph" class="member-name-link">sir_graph</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/UNAfold.html#sir_graph" class="member-name-link">sir_graph</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/FlatFileCache.html#size()" class="member-name-link">size()</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/FlatFileCache.html" title="class in org.biojava3.core.util">FlatFileCache</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/SoftHashMap.html#size()" class="member-name-link">size()</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#skip(long)" class="member-name-link">skip(long)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>
<div class="block">Skips characters.</div>
</dd>
<dt><a href="../org/biojava3/core/util/UncompressInputStream.html#skip(long)" class="member-name-link">skip(long)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.srna">SmallPeptidesSearch</a> - Class in <a href="../bacnet/scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html#%3Cinit%3E()" class="member-name-link">SmallPeptidesSearch()</a> - Constructor for class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html" title="class in bacnet.scripts.listeriomics.srna">SmallPeptidesSearch</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#SMIRNOV" class="member-name-link">SMIRNOV</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/SoftHashMap.html" class="type-name-link" title="class in org.biojava3.core.util">SoftHashMap</a>&lt;<a href="../org/biojava3/core/util/SoftHashMap.html" title="type parameter in SoftHashMap">K</a>,<wbr><a href="../org/biojava3/core/util/SoftHashMap.html" title="type parameter in SoftHashMap">V</a>&gt; - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>
<div class="block">A in memory cache using soft references.</div>
</dd>
<dt><a href="../org/biojava3/core/util/SoftHashMap.html#%3Cinit%3E()" class="member-name-link">SoftHashMap()</a> - Constructor for class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/SoftHashMap.html#%3Cinit%3E(int)" class="member-name-link">SoftHashMap(int)</a> - Constructor for class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#sort(int)" class="member-name-link">sort(int)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#sort(int,boolean)" class="member-name-link">sort(int, boolean)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">TO IMPROVE<br>
 Sort double[][] values
 Using sorted array, reorder rows through OrderedRowNames</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#sortByStart()" class="member-name-link">sortByStart()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a new list that is ordered by the starting index of the features' locations.</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#sortColumn(double%5B%5D%5B%5D,int)" class="member-name-link">sortColumn(double[][], int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Sort an array using column given by the index</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#sortColumn(java.lang.String%5B%5D%5B%5D,int,boolean)" class="member-name-link">sortColumn(String[][], int, boolean)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Sort a column inside a String[][] array <br>
 <br>
 WARNING: if two values are the same in the column, it will display only the first one with this
 value in the sorted array</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Feature.html#source()" class="member-name-link">source()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="block">Get source (aka method).</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#SPEARCORR" class="member-name-link">SPEARCORR</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#splice(org.biojava3.core.sequence.DNASequence)" class="member-name-link">splice(DNASequence)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Concatenate successive portions of the specified sequence using the feature locations in the
 list.</div>
</dd>
<dt><a href="../org/biojava3/genome/util/SplitFasta.html" class="type-name-link" title="class in org.biojava3.genome.util">SplitFasta</a> - Class in <a href="../org/biojava3/genome/util/package-summary.html">org.biojava3.genome.util</a></dt>
<dd>
<div class="block">Utility to write each Fasta entry to a unique file</div>
</dd>
<dt><a href="../org/biojava3/genome/util/SplitFasta.html#%3Cinit%3E()" class="member-name-link">SplitFasta()</a> - Constructor for class org.biojava3.genome.util.<a href="../org/biojava3/genome/util/SplitFasta.html" title="class in org.biojava3.genome.util">SplitFasta</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#square(double%5B%5D%5B%5D)" class="member-name-link">square(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Return array Y=(x_ij ^ 2)</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#squareVector(double%5B%5D)" class="member-name-link">squareVector(double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Calcul of the square of a vector</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html" class="type-name-link" title="class in bacnet.datamodel.sequence">Srna</a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.SeqType.html#Srna" class="member-name-link">Srna</a> - Enum constant in enum bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.TypeSrna.html#Srna" class="member-name-link">Srna</a> - Enum constant in enum bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.TypeSrna.html" title="enum in bacnet.datamodel.sequence">Srna.TypeSrna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#%3Cinit%3E()" class="member-name-link">Srna()</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#%3Cinit%3E(java.lang.String,int,int)" class="member-name-link">Srna(String, int, int)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#%3Cinit%3E(java.lang.String,int,int,char)" class="member-name-link">Srna(String, int, int, char)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#SRNA_NUMBER" class="member-name-link">SRNA_NUMBER</a> - Static variable in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.TypeSrna.html" class="type-name-link" title="enum in bacnet.datamodel.sequence">Srna.TypeSrna</a> - Enum in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.srna">Srna10403SOliver</a> - Class in <a href="../bacnet/scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html#%3Cinit%3E()" class="member-name-link">Srna10403SOliver()</a> - Constructor for class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html" title="class in bacnet.scripts.listeriomics.srna">Srna10403SOliver</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a> - Class in <a href="../bacnet/scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#%3Cinit%3E()" class="member-name-link">SrnaAlignmentBlastN()</a> - Constructor for class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#sRNACircularGenome(int,int,bacnet.datamodel.sequence.Genome,java.lang.String)" class="member-name-link">sRNACircularGenome(int, int, Genome, String)</a> - Method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.srna">SrnaFinalList</a> - Class in <a href="../bacnet/scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html#%3Cinit%3E()" class="member-name-link">SrnaFinalList()</a> - Constructor for class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaFinalList.html" title="class in bacnet.scripts.listeriomics.srna">SrnaFinalList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/SrnaListeriomics.html" class="type-name-link" title="class in bacnet.scripts.listeriomics">SrnaListeriomics</a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>
<div class="block">List of methods to manage Srna data for Listeriomics</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/SrnaListeriomics.html#%3Cinit%3E()" class="member-name-link">SrnaListeriomics()</a> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SrnaListeriomics.html" title="class in bacnet.scripts.listeriomics">SrnaListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#sRNAs" class="member-name-link">sRNAs</a> - Variable in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaSummaryView.html" class="type-name-link" title="class in bacnet.sequenceTools">SrnaSummaryView</a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaSummaryView.html#%3Cinit%3E()" class="member-name-link">SrnaSummaryView()</a> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaSummaryView.html" title="class in bacnet.sequenceTools">SrnaSummaryView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" class="type-name-link" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a> - Class in <a href="../bacnet/scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a></dt>
<dd>
<div class="block">Include a list of Methods to read every Table containing Small RNAs in Listeria<br>
 For each it reads the <code>Table</code> and create a <code>List</code> of <code>Srna</code></div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#%3Cinit%3E()" class="member-name-link">SrnaTables()</a> - Constructor for class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html" class="type-name-link" title="class in bacnet.sequenceTools">SrnaView</a> - Class in <a href="../bacnet/sequenceTools/package-summary.html">bacnet.sequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#%3Cinit%3E()" class="member-name-link">SrnaView()</a> - Constructor for class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#start()" class="member-name-link">start()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Get starting index (origin 0).</div>
</dd>
<dt><a href="../bacnet/e4/rap/setup/Activator.html#start(org.osgi.framework.BundleContext)" class="member-name-link">start(BundleContext)</a> - Method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/Activator.html" title="class in bacnet.e4.rap.setup">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/Activator.html#start(org.osgi.framework.BundleContext)" class="member-name-link">start(BundleContext)</a> - Method in class bacnet.scripts.<a href="../bacnet/scripts/Activator.html" title="class in bacnet.scripts">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Codon.html#startCodon" class="member-name-link">startCodon</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Codon.html#startCodon(java.lang.String)" class="member-name-link">startCodon(String)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></dt>
<dd>
<div class="block">If it is a start codon return the corresponding codon, an empty <code>String</code> otherwise</div>
</dd>
<dt><a href="../bacnet/scripts/genome/StartCodonProfile.html" class="type-name-link" title="class in bacnet.scripts.genome">StartCodonProfile</a> - Class in <a href="../bacnet/scripts/genome/package-summary.html">bacnet.scripts.genome</a></dt>
<dd>
<div class="block">Calculate presence of start codon on firrst chromosome of bacterial genomes</div>
</dd>
<dt><a href="../bacnet/scripts/genome/StartCodonProfile.html#%3Cinit%3E()" class="member-name-link">StartCodonProfile()</a> - Constructor for class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/StartCodonProfile.html" title="class in bacnet.scripts.genome">StartCodonProfile</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/StartCodonSequence.html" class="type-name-link" title="class in org.biojava3.core.sequence">StartCodonSequence</a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">Used to map the start codon feature on a gene</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/StartCodonSequence.html#%3Cinit%3E(org.biojava3.core.sequence.TranscriptSequence,int,int)" class="member-name-link">StartCodonSequence(TranscriptSequence, int, int)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/StartCodonSequence.html" title="class in org.biojava3.core.sequence">StartCodonSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#startsAfter(org.biojava3.genome.parsers.gff.Location)" class="member-name-link">startsAfter(Location)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Check if this location starts after the other location starts.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#startsBefore(org.biojava3.genome.parsers.gff.Location)" class="member-name-link">startsBefore(Location)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Check if this location starts before other location starts.</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.TypeMapper.html#STAT" class="member-name-link">STAT</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.TypeMapper.html" title="enum in bacnet.table.core">ColorMapper.TypeMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.Type.html#STAT" class="member-name-link">STAT</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.Type.html" title="enum in bacnet.table.core">ColorMapperRCP.Type</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html#statDatabase()" class="member-name-link">statDatabase()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Extract statistics from existing databases, number of lmos, sRNA, utr...</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#statDatabase()" class="member-name-link">statDatabase()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Extract statistics from existing databases, number of lmos, sRNA, utr...</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#statInternalORF(bacnet.datamodel.sequence.Genome)" class="member-name-link">statInternalORF(Genome)</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Given a genome extract main information such as:<br>
 number of start codons, stop codons
 number of high SD binding sites
 number of possible ORFs
 number of predicted genes<br>
 <br>
 From this metrics, predict the number of possible internal translation initiation site</div>
</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.html" class="type-name-link" title="class in bacnet.scripts.core.stat">StatTest</a> - Class in <a href="../bacnet/scripts/core/stat/package-summary.html">bacnet.scripts.core.stat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.html#%3Cinit%3E()" class="member-name-link">StatTest()</a> - Constructor for class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.html" title="class in bacnet.scripts.core.stat">StatTest</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.html#%3Cinit%3E(bacnet.scripts.core.stat.StatTest.TypeStat)" class="member-name-link">StatTest(StatTest.TypeStat)</a> - Constructor for class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.html" title="class in bacnet.scripts.core.stat">StatTest</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" class="type-name-link" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a> - Enum in <a href="../bacnet/scripts/core/stat/package-summary.html">bacnet.scripts.core.stat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatUtils.html" class="type-name-link" title="class in bacnet.scripts.core.stat">StatUtils</a> - Class in <a href="../bacnet/scripts/core/stat/package-summary.html">bacnet.scripts.core.stat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatUtils.html#%3Cinit%3E()" class="member-name-link">StatUtils()</a> - Constructor for class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatUtils.html" title="class in bacnet.scripts.core.stat">StatUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.TypeMapper.html#STDEV" class="member-name-link">STDEV</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.TypeMapper.html" title="enum in bacnet.table.core">ColorMapper.TypeMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.Type.html#STDEV" class="member-name-link">STDEV</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.Type.html" title="enum in bacnet.table.core">ColorMapperRCP.Type</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#STDEVWT" class="member-name-link">STDEVWT</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/Activator.html#stop(org.osgi.framework.BundleContext)" class="member-name-link">stop(BundleContext)</a> - Method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/Activator.html" title="class in bacnet.e4.rap.setup">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/Activator.html#stop(org.osgi.framework.BundleContext)" class="member-name-link">stop(BundleContext)</a> - Method in class bacnet.scripts.<a href="../bacnet/scripts/Activator.html" title="class in bacnet.scripts">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/StopCodonSequence.html" class="type-name-link" title="class in org.biojava3.core.sequence">StopCodonSequence</a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">Used to map the stop codon sequence on a gene</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/StopCodonSequence.html#%3Cinit%3E(org.biojava3.core.sequence.TranscriptSequence,int,int)" class="member-name-link">StopCodonSequence(TranscriptSequence, int, int)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/StopCodonSequence.html" title="class in org.biojava3.core.sequence">StopCodonSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/StringColor.html#str" class="member-name-link">str</a> - Variable in class bacnet.utils.<a href="../bacnet/utils/StringColor.html" title="class in bacnet.utils">StringColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#strand" class="member-name-link">strand</a> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/Strand.html" class="type-name-link" title="enum in org.biojava3.core.sequence">Strand</a> - Enum in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">Provides a way of representing the strand of a sequence, location hit or feature.</div>
</dd>
<dt><a href="../bacnet/utils/StringColor.html" class="type-name-link" title="class in bacnet.utils">StringColor</a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/StringColor.html#%3Cinit%3E()" class="member-name-link">StringColor()</a> - Constructor for class bacnet.utils.<a href="../bacnet/utils/StringColor.html" title="class in bacnet.utils">StringColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/StringManipulationHelper.html" class="type-name-link" title="class in org.biojava3.core.util">StringManipulationHelper</a> - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>
<div class="block">A utility class for common <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link"><code>String</code></a> manipulation tasks.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" class="type-name-link" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a>&lt;<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</a></dt>
<dd>
<div class="block">An example of a ProxySequenceReader that is created from a String.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">StringProxySequenceReader(String, CompoundSet&lt;C&gt;)</a> - Constructor for class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#STYLE" class="member-name-link">STYLE</a> - Static variable in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceUsage.html#SUB_SEQUENCE_DATA" class="member-name-link">SUB_SEQUENCE_DATA</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceUsage.html" title="enum in org.biojava3.core.sequence">SequenceOptimizationHints.SequenceUsage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#subArray(java.lang.String%5B%5D%5B%5D,int,int)" class="member-name-link">subArray(String[][], int, int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#subArray(java.lang.String%5B%5D%5B%5D,java.util.ArrayList,java.util.ArrayList)" class="member-name-link">subArray(String[][], ArrayList&lt;Integer&gt;, ArrayList&lt;Integer&gt;)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Return an array which correspond to the row and colum</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/SubCellCompartment.html" class="type-name-link" title="class in bacnet.datamodel.annotation">SubCellCompartment</a> - Class in <a href="../bacnet/datamodel/annotation/package-summary.html">bacnet.datamodel.annotation</a></dt>
<dd>
<div class="block">All data and methods for Subcellular compartment information are done here</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/SubCellCompartment.html#%3Cinit%3E()" class="member-name-link">SubCellCompartment()</a> - Constructor for class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.html" title="class in bacnet.datamodel.annotation">SubCellCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" class="type-name-link" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a> - Enum in <a href="../bacnet/datamodel/annotation/package-summary.html">bacnet.datamodel.annotation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/ConcurrencyTools.html#submit(java.util.concurrent.Callable)" class="member-name-link">submit(Callable&lt;T&gt;)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Queues up a task and adds a default log entry.</div>
</dd>
<dt><a href="../org/biojava3/core/util/ConcurrencyTools.html#submit(java.util.concurrent.Callable,java.lang.String)" class="member-name-link">submit(Callable&lt;T&gt;, String)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>
<div class="block">Queues up a task and adds a log entry.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.Substitute.html#%3Cinit%3E(java.lang.String,int)" class="member-name-link">Substitute(String, int)</a> - Constructor for class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.Substitute.html" title="class in org.biojava3.core.sequence.edits">Edit.Substitute</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.Substitute.html#%3Cinit%3E(org.biojava3.core.sequence.template.Sequence,int)" class="member-name-link">Substitute(Sequence&lt;C&gt;, int)</a> - Constructor for class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.Substitute.html" title="class in org.biojava3.core.sequence.edits">Edit.Substitute</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#subVector(double%5B%5D,int,int)" class="member-name-link">subVector(double[], int, int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Return the subVector of vector between index begin and index end</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#suffix(int)" class="member-name-link">suffix(int)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">The part of this location after the specified position.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#suffix(org.biojava3.genome.parsers.gff.Location)" class="member-name-link">suffix(Location)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">The part of this location after the other location (not inclusive).</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#sum(double%5B%5D)" class="member-name-link">sum(double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Sum of all term in the vector</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#sum(double%5B%5D%5B%5D)" class="member-name-link">sum(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#summarize(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)" class="member-name-link">summarize(Experiment, Genome)</a> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>
<div class="block">Create 3 ExpressionMatrix: <br>
 GeneExpression of genes on all BioCondition, saved in PATH_ALLGENEXPR
 Tiling of genes on all BioCondition, saved in PATH_ALLTILING_GENE
 Tiling of Srnas on all BioCondition, saved in PATH_ALLTILING_SRNA</div>
</dd>
<dt><a href="../bacnet/scripts/database/ComparisonsCreation.html#summarizeDiffExpr(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)" class="member-name-link">summarizeDiffExpr(Experiment, Genome)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Create an array :<br>
 Column = comparison
 Row = Gene, Srna or Asrna
 Cell = "+" if a specific gene has been detected over-expressed, and "-" if
 under-expressed<br>
 <br>
 
 Save it in : Comparison.PATH_DIFF_LISTS+".txt"</div>
</dd>
<dt><a href="../bacnet/scripts/database/ComparisonsCreation.html#summarizeDiffLists(bacnet.datamodel.expdesign.Experiment)" class="member-name-link">summarizeDiffLists(Experiment)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></dt>
<dd>
<div class="block">Create an array :<br>
 Row = comparison
 Column = Number of gene, Srna Asrna over or under-expressed<br>
 <br>
 Save it in : Comparison.PATH_DIFF_LISTS+".txt"</div>
</dd>
<dt><a href="../bacnet/scripts/database/ProteomicsCreation.html#summarizeProteomes(bacnet.datamodel.expdesign.Experiment,java.lang.String)" class="member-name-link">summarizeProteomes(Experiment, String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>
<div class="block">Create a table will all protein expression for each Genome</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#summarizeResultsInTable()" class="member-name-link">summarizeResultsInTable()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Read all the XML results of Blast and create two PhyloXML files and a sRNA conservation table<br></div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html#summarizeResultsInTable(java.util.ArrayList)" class="member-name-link">summarizeResultsInTable(ArrayList&lt;Srna&gt;)</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/Srna10403SOliver.html" title="class in bacnet.scripts.listeriomics.srna">Srna10403SOliver</a></dt>
<dd>
<div class="block">Read all the XML results of Blast and create a sRNA conservation table<br>
 header = {"Srna","EGD-e identites","EGD-e begin","EGD-e end","10403S identities","10403S
 begin","10403S end","10403S Average GEI","Delta_sigB Average GEI","FC range
 (10403S/Delta_sigB)"};</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#summarizeResultsInTable(java.util.ArrayList)" class="member-name-link">summarizeResultsInTable(ArrayList&lt;Srna&gt;)</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Read all the XML results of Blast and create two PhyloXML files and a sRNA conservation table<br></div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#summarizeResultsInTable10403S()" class="member-name-link">summarizeResultsInTable10403S()</a> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Read all the XML results of Blast and create two PhyloXML files and a sRNA conservation table<br></div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html" class="type-name-link" title="class in bacnet.scripts.listeriomics">SummaryListeriomics</a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>
<div class="block">Methods for creating the different figures and Tables of the paper</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html#%3Cinit%3E()" class="member-name-link">SummaryListeriomics()</a> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html" title="class in bacnet.scripts.listeriomics">SummaryListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.TypeFilter.html#SUPERIOR" class="member-name-link">SUPERIOR</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/Filter.TypeFilter.html" title="enum in bacnet.table.core">Filter.TypeFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.TypeFilter.html#SUPERIOR" class="member-name-link">SUPERIOR</a> - Enum constant in enum bacnet.utils.<a href="../bacnet/utils/Filter.TypeFilter.html" title="enum in bacnet.utils">Filter.TypeFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/Filter.TypeFilter.html#SUPERIOR_ABS" class="member-name-link">SUPERIOR_ABS</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/Filter.TypeFilter.html" title="enum in bacnet.table.core">Filter.TypeFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/Filter.TypeFilter.html#SUPERIOR_ABS" class="member-name-link">SUPERIOR_ABS</a> - Enum constant in enum bacnet.utils.<a href="../bacnet/utils/Filter.TypeFilter.html" title="enum in bacnet.utils">Filter.TypeFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/HTMLUtils.html#SVG" class="member-name-link">SVG</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/HTMLUtils.html" title="class in bacnet.utils">HTMLUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#swtColorToAwt(org.eclipse.swt.graphics.Color)" class="member-name-link">swtColorToAwt(Color)</a> - Static method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html" class="type-name-link" title="class in bacnet.swt">SWTResourceManager</a> - Class in <a href="../bacnet/swt/package-summary.html">bacnet.swt</a></dt>
<dd>
<div class="block">Utility class for managing OS resources associated with SWT controls such as colors, fonts,
 images, etc.</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#%3Cinit%3E()" class="member-name-link">SWTResourceManager()</a> - Constructor for class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ListUtils.html#symDifference(java.util.ArrayList,java.util.ArrayList)" class="member-name-link">symDifference(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ListUtils.html" title="class in bacnet.utils">ListUtils</a></dt>
<dd>
<div class="block">Calculate symmetrical difference of two lists L1 and L2 <br></div>
</dd>
<dt><a href="../bacnet/e4/rap/SynTView.html" class="type-name-link" title="class in bacnet.e4.rap">SynTView</a> - Class in <a href="../bacnet/e4/rap/package-summary.html">bacnet.e4.rap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/SynTView.html#%3Cinit%3E()" class="member-name-link">SynTView()</a> - Constructor for class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/SystemsBiologyListeriomics.html" class="type-name-link" title="class in bacnet.scripts.listeriomics">SystemsBiologyListeriomics</a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>
<div class="block">List of method to create Co-expression network and analyse it</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/SystemsBiologyListeriomics.html#%3Cinit%3E()" class="member-name-link">SystemsBiologyListeriomics()</a> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SystemsBiologyListeriomics.html" title="class in bacnet.scripts.listeriomics">SystemsBiologyListeriomics</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
