<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>P-Index</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="P-Index";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="contentContainer"><a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">K</a>&nbsp;<a href="index-12.html">L</a>&nbsp;<a href="index-13.html">M</a>&nbsp;<a href="index-14.html">N</a>&nbsp;<a href="index-15.html">O</a>&nbsp;<a href="index-16.html">P</a>&nbsp;<a href="index-17.html">Q</a>&nbsp;<a href="index-18.html">R</a>&nbsp;<a href="index-19.html">S</a>&nbsp;<a href="index-20.html">T</a>&nbsp;<a href="index-21.html">U</a>&nbsp;<a href="index-22.html">V</a>&nbsp;<a href="index-23.html">W</a>&nbsp;<a href="index-24.html">X</a>&nbsp;<a href="index-25.html">Y</a>&nbsp;<a href="index-26.html">Z</a>&nbsp;<a href="index-27.html">_</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a><a id="I:P">
<!--   -->
</a>
<h2 class="title">P</h2>
<dl>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#packColumns">packColumns</a></span> - Variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html#PADJ">PADJ</a></span> - bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html" title="enum in bacnet.datamodel.dataset">OmicsData.ColNames</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/gui/BlastWizard.html#page">page</a></span> - Variable in class bacnet.scripts.blast.gui.<a href="../bacnet/scripts/blast/gui/BlastWizard.html" title="class in bacnet.scripts.blast.gui">BlastWizard</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/gui/BlastWizard.html#page2">page2</a></span> - Variable in class bacnet.scripts.blast.gui.<a href="../bacnet/scripts/blast/gui/BlastWizard.html" title="class in bacnet.scripts.blast.gui">BlastWizard</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#paintComponent(java.awt.Graphics)">paintComponent(Graphics)</a></span> - Method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html#PAIRWISE">PAIRWISE</a></span> - bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html" title="enum in bacnet.scripts.blast">BlastOutput.BlastOutputTYPE</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/AbstractFeature.html#parentFeature">parentFeature</a></span> - Variable in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/StartCodonSequence.html#parentGeneSequence">parentGeneSequence</a></span> - Variable in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/StartCodonSequence.html" title="class in org.biojava3.core.sequence">StartCodonSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/StopCodonSequence.html#parentGeneSequence">parentGeneSequence</a></span> - Variable in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/StopCodonSequence.html" title="class in org.biojava3.core.sequence">StopCodonSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/TranscriptSequence.html#parentGeneSequence">parentGeneSequence</a></span> - Variable in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#parentGenome">parentGenome</a></span> - Variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>
<div class="block">Parent Genome where this chromosome is placed</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#parentGenome">parentGenome</a></span> - Variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.html#parentSequence">parentSequence</a></span> - Variable in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/CDSSequence.html#parentTranscriptSequence">parentTranscriptSequence</a></span> - Variable in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#parentViewId">parentViewId</a></span> - Variable in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/InsdcParser.html#parse(java.io.Reader)">parse(Reader)</a></span> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</a></dt>
<dd>
<div class="block">Reader based version of the parse methods.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/InsdcParser.html#parse(java.io.Reader,org.biojava3.core.sequence.Strand)">parse(Reader, Strand)</a></span> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/InsdcParser.html#parse(java.lang.String)">parse(String)</a></span> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</a></dt>
<dd>
<div class="block">Main method for parsing a location from a String instance</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/UncompressInputStream.html#parse_header()">parse_header()</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#parseColor(double)">parseColor(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#parseColor(double)">parseColor(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#parseColorAWT(double)">parseColorAWT(double)</a></span> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#parseData()">parseData()</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>
<div class="block">Using LOCUSTAG_CHANGE_PATH modify every table by changin locutag ID or removing the row if the
 gene is not found</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html#parsedCompounds">parsedCompounds</a></span> - Variable in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html#parsedCompounds">parsedCompounds</a></span> - Variable in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#parsedCompounds">parsedCompounds</a></span> - Variable in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html#parsedCompounds">parsedCompounds</a></span> - Variable in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/NCBIFastaHeaderParser.html#parseHeader(java.lang.String,S)">parseHeader(String, S)</a></span> - Method in class bacnet.reader.<a href="../bacnet/reader/NCBIFastaHeaderParser.html" title="class in bacnet.reader">NCBIFastaHeaderParser</a></dt>
<dd>
<div class="block">Parse the header and set the values in the sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html#parseHeader(java.lang.String,S)">parseHeader(String, S)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser</a></dt>
<dd>
<div class="block">Parse the header and set the values in the sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html#parseHeader(java.lang.String,S)">parseHeader(String, S)</a></span> - Method in interface org.biojava3.core.sequence.io.template.<a href="../org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/NavigationManagement.html#parseInitURL(org.eclipse.e4.ui.workbench.modeling.EPartService)">parseInitURL(EPartService)</a></span> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>
<div class="block">Parse init URL and display the correct View with the correct view parameters</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html#parseLine(java.lang.String)">parseLine(String)</a></span> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html" title="class in org.biojava3.genome.parsers.gff">GeneIDGFF2Reader</a></dt>
<dd>
<div class="block">create Feature from line of GFF file</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html#parseLine(java.lang.String)">parseLine(String)</a></span> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html" title="class in org.biojava3.genome.parsers.gff">GeneMarkGTFReader</a></dt>
<dd>
<div class="block">create Feature from line of GFF file</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/GFF3Reader.html#parseLine(java.lang.String)">parseLine(String)</a></span> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GFF3Reader.html" title="class in org.biojava3.genome.parsers.gff">GFF3Reader</a></dt>
<dd>
<div class="block">create Feature from line of GFF file</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#parseList(java.lang.String)">parseList(String)</a></span> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>
<div class="block">Curate list given by Omri, by parsing each element</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/InsdcParser.html#parseLocation(java.lang.String,org.biojava3.core.sequence.Strand)">parseLocation(String, Strand)</a></span> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#parseName(java.lang.String)">parseName(String)</a></span> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Parse a Comparison name, extracting leftBioCondition and rightBiocondition</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/LocusTag.html#parseNewLocusTagChange()">parseNewLocusTagChange()</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/LocusTag.html" title="class in bacnet.datamodel.annotation">LocusTag</a></dt>
<dd>
<div class="block">Create a HashMap with all "old_locus_tag" and "locus_tag" from the new GFF folder
 Listeriomics/NCBIGenomesNew/</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/LocusTag.html#parseOldLocusTagChange()">parseOldLocusTagChange()</a></span> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/LocusTag.html" title="class in bacnet.datamodel.annotation">LocusTag</a></dt>
<dd>
<div class="block">Create a HashMap with all "old_locus_tag" and "locus_tag" from the current GFF folder
 Listeriomics/NCBIGenomes/</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/phylogeny/Phylogenomic.html#parsePhylogenomicFigure(java.lang.String)">parsePhylogenomicFigure(String)</a></span> - Static method in class bacnet.datamodel.phylogeny.<a href="../bacnet/datamodel/phylogeny/Phylogenomic.html" title="class in bacnet.datamodel.phylogeny">Phylogenomic</a></dt>
<dd>
<div class="block">Parse phylogenomic SVG figure and extract attributes corresponding to each genome</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/InsdcParser.html#parseRange(java.util.regex.Matcher,org.biojava3.core.sequence.Strand)">parseRange(Matcher, Strand)</a></span> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/exceptions/ParserException.html" title="class in org.biojava3.core.exceptions"><span class="typeNameLink">ParserException</span></a> - Exception in <a href="../org/biojava3/core/exceptions/package-summary.html">org.biojava3.core.exceptions</a></dt>
<dd>
<div class="block">General abstraction of different parsing errors</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/exceptions/ParserException.html#%3Cinit%3E(java.lang.Exception)">ParserException(Exception)</a></span> - Constructor for exception org.biojava3.core.exceptions.<a href="../org/biojava3/core/exceptions/ParserException.html" title="class in org.biojava3.core.exceptions">ParserException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/exceptions/ParserException.html#%3Cinit%3E(java.lang.String)">ParserException(String)</a></span> - Constructor for exception org.biojava3.core.exceptions.<a href="../org/biojava3/core/exceptions/ParserException.html" title="class in org.biojava3.core.exceptions">ParserException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/exceptions/ParserException.html#%3Cinit%3E(java.lang.String,java.lang.Exception)">ParserException(String, Exception)</a></span> - Constructor for exception org.biojava3.core.exceptions.<a href="../org/biojava3/core/exceptions/ParserException.html" title="class in org.biojava3.core.exceptions">ParserException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/InsdcParser.html#parseSingle(java.util.regex.Matcher,org.biojava3.core.sequence.Strand)">parseSingle(Matcher, Strand)</a></span> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/NavigationManagement.html#parseStateParemeters(java.lang.String)">parseStateParemeters(String)</a></span> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/IUPACParser.html#parseTables()">parseTables()</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DNASeqCreation.html#parseWIGFile()">parseWIGFile()</a></span> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DNASeqCreation.html" title="class in bacnet.scripts.database">DNASeqCreation</a></dt>
<dd>
<div class="block">Parse WIG file and calculate Copy Number Variation = raw coverage / mean coverage<br>
 The WIG has to be Raw number of reads per basepair</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#part">part</a></span> - Variable in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/BannerView.html#partService">partService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BannerView.html" title="class in bacnet.e4.rap">BannerView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/BasicApplication.html#partService">partService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BasicApplication.html" title="class in bacnet.e4.rap">BasicApplication</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewBacnet.html#partService">partService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewBacnet.html" title="class in bacnet.e4.rap">InitViewBacnet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewCRISPR.html#partService">partService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewCRISPR.html" title="class in bacnet.e4.rap">InitViewCRISPR</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewLeishmania.html#partService">partService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewLeishmania.html" title="class in bacnet.e4.rap">InitViewLeishmania</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewListeria.html#partService">partService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeria.html" title="class in bacnet.e4.rap">InitViewListeria</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewListeriaSample.html#partService">partService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">InitViewListeriaSample</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewYersinia.html#partService">partService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewYersinia.html" title="class in bacnet.e4.rap">InitViewYersinia</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/setup/BasicApplication.html#partService">partService</a></span> - Variable in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/BasicApplication.html" title="class in bacnet.e4.rap.setup">BasicApplication</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/setup/SetupPart.html#partService">partService</a></span> - Variable in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SetupPart.html" title="class in bacnet.e4.rap.setup">SetupPart</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/SynTView.html#partService">partService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/TestPart.html#partService">partService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/TestPart.html" title="class in bacnet.e4.rap">TestPart</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html#partService">partService</a></span> - Variable in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html" title="class in bacnet.expressionAtlas.core">SelectGenomeElementDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#partService">partService</a></span> - Variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#partService">partService</a></span> - Variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#partService">partService</a></span> - Variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#partService">partService</a></span> - Variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#partService">partService</a></span> - Variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#partService">partService</a></span> - Variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#partService">partService</a></span> - Variable in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#partService">partService</a></span> - Variable in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#partService">partService</a></span> - Variable in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/TestGenomeBrowser.html#partService">partService</a></span> - Variable in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/TestGenomeBrowser.html" title="class in bacnet.genomeBrowser">TestGenomeBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#partService">partService</a></span> - Variable in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/handler/LoadExpressionMatrixHandler.html#partService">partService</a></span> - Variable in class bacnet.handler.<a href="../bacnet/handler/LoadExpressionMatrixHandler.html" title="class in bacnet.handler">LoadExpressionMatrixHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastResultView.html#partService">partService</a></span> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResultView.html" title="class in bacnet.scripts.blast">BlastResultView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/gui/BlastWizard.html#partService">partService</a></span> - Variable in class bacnet.scripts.blast.gui.<a href="../bacnet/scripts/blast/gui/BlastWizard.html" title="class in bacnet.scripts.blast.gui">BlastWizard</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#partService">partService</a></span> - Variable in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#partService">partService</a></span> - Variable in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GenomicsView.html#partService">partService</a></span> - Variable in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SequenceDisplayDialog.html#partService">partService</a></span> - Variable in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaSummaryView.html#partService">partService</a></span> - Variable in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaSummaryView.html" title="class in bacnet.sequenceTools">SrnaSummaryView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#partService">partService</a></span> - Variable in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/HideExpressionMatrixDialog.html#partService">partService</a></span> - Variable in class bacnet.table.gui.<a href="../bacnet/table/gui/HideExpressionMatrixDialog.html" title="class in bacnet.table.gui">HideExpressionMatrixDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/TableSWTComposite.html#partService">partService</a></span> - Variable in class bacnet.table.<a href="../bacnet/table/TableSWTComposite.html" title="class in bacnet.table">TableSWTComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/TableSWTView.html#partService">partService</a></span> - Variable in class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#partService">partService</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView.html#partService">partService</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView2.html#partService">partService</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/InitView.html#partService">partService</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/InitView.html" title="class in bacnet.views">InitView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/InternalBrowser.html#partService">partService</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DataSource.html#PATENTS">PATENTS</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#path">path</a></span> - Variable in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">The absolute path on the database folder</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#PATH">PATH</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Path for Srna raw data</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#PATH">PATH</a></span> - Static variable in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#PATH">PATH</a></span> - Static variable in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#PATH">PATH</a></span> - Static variable in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#PATH">PATH</a></span> - Static variable in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/SDProfile.html#PATH">PATH</a></span> - Static variable in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#PATH">PATH</a></span> - Static variable in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#PATH">PATH</a></span> - Static variable in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStatProteolyse.html#PATH">PATH</a></span> - Static variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStatProteolyse.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStatProteolyse</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#PATH">PATH</a></span> - Static variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#PATH">PATH</a></span> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#PATH">PATH</a></span> - Static variable in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#PATH_ADD_IDENTITY">PATH_ADD_IDENTITY</a></span> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#PATH_ALLDataType">PATH_ALLDataType</a></span> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#PATH_ALLExpMatrix">PATH_ALLExpMatrix</a></span> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#PATH_ALLGENEXPR">PATH_ALLGENEXPR</a></span> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#PATH_ALLRNASEQ">PATH_ALLRNASEQ</a></span> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Expression.html#PATH_ALLTILING">PATH_ALLTILING</a></span> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_ANNOTATION">PATH_ANNOTATION</a></span> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for all supplementary information for genome annotation : Database.getInstance().getPath() +
 "/GenomeNCBI/Annotation/";</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/UNAfold.html#PATH_BIN">PATH_BIN</a></span> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/UNAfold.html#PATH_BIN">PATH_BIN</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#PATH_BLAST">PATH_BLAST</a></span> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Path for Blast+</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/Blast.html#PATH_BLAST_MAC">PATH_BLAST_MAC</a></span> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/Blast.html#PATH_BLAST_WIN">PATH_BLAST_WIN</a></span> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_BLASTDB">PATH_BLASTDB</a></span> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for BlastDB folder</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#PATH_BLASTDB">PATH_BLASTDB</a></span> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Path for BlastDB folder</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#PATH_COMP">PATH_COMP</a></span> - Static variable in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_COMPARISONS">PATH_COMPARISONS</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for saving all comparisons datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#PATH_Conservation">PATH_Conservation</a></span> - Static variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#PATH_CONSERVATION">PATH_CONSERVATION</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#PATH_Conservation_Cutoff">PATH_Conservation_Cutoff</a></span> - Static variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#PATH_CORRECTGENE">PATH_CORRECTGENE</a></span> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Some EGD-e genes have same name, this file is used to correct gene name</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/UNAfold.html#PATH_DATA">PATH_DATA</a></span> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/UNAfold.html#PATH_DATA">PATH_DATA</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#PATH_DIFF_EXPR">PATH_DIFF_EXPR</a></span> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Comparison.html#PATH_DIFF_LISTS">PATH_DIFF_LISTS</a></span> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_EXPR_NORM">PATH_EXPR_NORM</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all ExpressionMatrix RAW datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_EXPR_RAW">PATH_EXPR_RAW</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all ExpressionMatrix RAW datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_GENEXPR_COMPLETE">PATH_GENEXPR_COMPLETE</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all GeneExpression COMPLETE datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_GENEXPR_NORM">PATH_GENEXPR_NORM</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all GeneExpression NORM datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_GENEXPR_RAW">PATH_GENEXPR_RAW</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all ExpressionMatrix NORM datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_GENOMES">PATH_GENOMES</a></span> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for all Genomes : Database.getInstance().getPath() + "/GenomeNCBI/Genomes/";</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html#PATH_GENOMES_LIST">PATH_GENOMES_LIST</a></span> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html" title="class in bacnet.scripts.blast">GenomeNCBIFolderTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html#PATH_GENOMES_LIST">PATH_GENOMES_LIST</a></span> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html" title="class in bacnet.scripts.blast">MultiSequenceBlastProtein</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#PATH_GENOMES_LIST">PATH_GENOMES_LIST</a></span> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_GENOMES_NEW">PATH_GENOMES_NEW</a></span> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span>
<div class="deprecationComment">only used in Old Listeriomics database</div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_HOMOLOGS">PATH_HOMOLOGS</a></span> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for all homologs calculation file : Database.getInstance().getPath() +
 "/GenomeNCBI/Homologs/";</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html#PATH_INFOTABLE">PATH_INFOTABLE</a></span> - Static variable in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html" title="class in bacnet.scripts.phylogeny">GenomePhylogeny</a></dt>
<dd>
<div class="block">This table taken from Jolley et al.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#PATH_iTIS">PATH_iTIS</a></span> - Variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html#PATH_LIST">PATH_LIST</a></span> - Static variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html" title="class in bacnet.scripts.listeriomics.nterm">NTermCreateFinalData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#PATH_MAC">PATH_MAC</a></span> - Static variable in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path when using MAC OX X</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#PATH_NCBI_BacGenome">PATH_NCBI_BacGenome</a></span> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_NGS_NORM">PATH_NGS_NORM</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all RNASeq NORM datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_NGS_RAW">PATH_NGS_RAW</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all RNASeq RAW datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_NORM">PATH_NORM</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all normalized datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_PROTEINID">PATH_PROTEINID</a></span> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for all homologs calculation file : Database.getInstance().getPath() +
 "/GenomeNCBI/Homologs/";</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_PROTEOMICS_NORM">PATH_PROTEOMICS_NORM</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all Proteome NORM datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_PROTEOMICS_RAW">PATH_PROTEOMICS_RAW</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all Proteome RAW datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_RAW">PATH_RAW</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all raw datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_RAW">PATH_RAW</a></span> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for all raw data for Genomes : Database.getInstance().getPath() + "/GenomeNCBI/";</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#PATH_RAW_DATA">PATH_RAW_DATA</a></span> - Static variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#PATH_RESULTS">PATH_RESULTS</a></span> - Static variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#PATH_RESULTS">PATH_RESULTS</a></span> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Path for BlastDB folder</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#PATH_SCRIPT">PATH_SCRIPT</a></span> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Path for data on the server when running homolog search blasts</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#PATH_SEC_STRUCTURE">PATH_SEC_STRUCTURE</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_STREAMING">PATH_STREAMING</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Path for all streaming datasets = serialized data for BACNET platform</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#PATH_TermSeq_EGDe">PATH_TermSeq_EGDe</a></span> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#PATH_TermSeq_EGDe">PATH_TermSeq_EGDe</a></span> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_THREADS">PATH_THREADS</a></span> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for all homologs calculation file : Database.getInstance().getPath() +
 "/GenomeNCBI/Homologs/";</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#PATH_THREADS">PATH_THREADS</a></span> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Path for BlastDB folder</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_TILING_NORM">PATH_TILING_NORM</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all Tiling NORM datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_TILING_RAW">PATH_TILING_RAW</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all Tiling RAW datasets</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#PATH_TSS_EGDe">PATH_TSS_EGDe</a></span> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#PATH_TSS_EGDe">PATH_TSS_EGDe</a></span> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#PATH_TSS_Innocua">PATH_TSS_Innocua</a></span> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#PATH_TSS_Innocua">PATH_TSS_Innocua</a></span> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#PATH_TSSasRNA">PATH_TSSasRNA</a></span> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#PATH_TSSasRNA">PATH_TSSasRNA</a></span> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#PATH_WEBSITE">PATH_WEBSITE</a></span> - Static variable in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for website database on Pasteur Virtual Machine</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#PATH_WIN">PATH_WIN</a></span> - Static variable in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for website database on Windows</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/Rscript.html#PATH_WIN7">PATH_WIN7</a></span> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Rscript.html" title="class in bacnet.scripts.core">Rscript</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#PATHFigure_ASrna">PATHFigure_ASrna</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#PATHFigure_CISReg">PATHFigure_CISReg</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#PATHFigure_Srna">PATHFigure_Srna</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#PATHSerialize">PATHSerialize</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Path for serialized Srna before transferring data to Database</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#PATHTABLE_ASrna">PATHTABLE_ASrna</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#PATHTABLE_CISReg">PATHTABLE_CISReg</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#PATHTABLE_Srna">PATHTABLE_Srna</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#PATHTABLE_SrnaReference">PATHTABLE_SrnaReference</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#PATHXML">PATHXML</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Path for Srna XML serialization for control</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Srna.html#PATHXML_10403S">PATHXML_10403S</a></span> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Path for XML serialization for 10403S (Which have a specific annotation)</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html#PC">PC</a></span> - bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DataSource.html#PDB1">PDB1</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DataSource.html#PDB2">PDB2</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DataSource.html#PDBe">PDBe</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html#PDBWEB">PDBWEB</a></span> - org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html" title="enum in org.biojava3.core.sequence.template">LightweightProfile.StringFormat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#pearsonCorrelation(double%5B%5D,double%5B%5D)">pearsonCorrelation(double[], double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#peptideLogo()">peptideLogo()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Create a fasta file with all sequences to create a logo</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#peptides">peptides</a></span> - Variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Link peptideSequence to <code>ArrayList(NTerm)</code><br>
 This correspond to the list of <code>NTerm</code> having same peptideSequence</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/Peptidomics.html" title="class in bacnet.scripts.listeriomics"><span class="typeNameLink">Peptidomics</span></a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/Peptidomics.html#%3Cinit%3E()">Peptidomics()</a></span> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/Peptidomics.html" title="class in bacnet.scripts.listeriomics">Peptidomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/SequenceTools.html#percentNucleotideSequence(java.lang.String)">percentNucleotideSequence(String)</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SequenceTools.html" title="class in org.biojava3.core.util">SequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#percentOverlap(org.biojava3.genome.parsers.gff.Location)">percentOverlap(Location)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Return percent overlap of two locations.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/ColorMapperWizard.html#performCancel()">performCancel()</a></span> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizard.html" title="class in bacnet.table.gui">ColorMapperWizard</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/ColorMapperWizardPage.html#performCancel()">performCancel()</a></span> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/gui/BlastWizard.html#performFinish()">performFinish()</a></span> - Method in class bacnet.scripts.blast.gui.<a href="../bacnet/scripts/blast/gui/BlastWizard.html" title="class in bacnet.scripts.blast.gui">BlastWizard</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/ColorMapperWizard.html#performFinish()">performFinish()</a></span> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizard.html" title="class in bacnet.table.gui">ColorMapperWizard</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#PERMUT">PERMUT</a></span> - bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DataSource.html#PFAM">PFAM</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/CDSSequence.html#phase">phase</a></span> - Variable in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/phylogeny/Phylogenomic.html#PHYLO_GENOME_SVG">PHYLO_GENOME_SVG</a></span> - Static variable in class bacnet.datamodel.phylogeny.<a href="../bacnet/datamodel/phylogeny/Phylogenomic.html" title="class in bacnet.datamodel.phylogeny">Phylogenomic</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/phylogeny/Phylogenomic.html" title="class in bacnet.datamodel.phylogeny"><span class="typeNameLink">Phylogenomic</span></a> - Class in <a href="../bacnet/datamodel/phylogeny/package-summary.html">bacnet.datamodel.phylogeny</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/phylogeny/Phylogenomic.html#%3Cinit%3E()">Phylogenomic()</a></span> - Constructor for class bacnet.datamodel.phylogeny.<a href="../bacnet/datamodel/phylogeny/Phylogenomic.html" title="class in bacnet.datamodel.phylogeny">Phylogenomic</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/PhylogenomicsCreation.html" title="class in bacnet.scripts.database"><span class="typeNameLink">PhylogenomicsCreation</span></a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/PhylogenomicsCreation.html#%3Cinit%3E()">PhylogenomicsCreation()</a></span> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/PhylogenomicsCreation.html" title="class in bacnet.scripts.database">PhylogenomicsCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics"><span class="typeNameLink">PhylogenyListeriomics</span></a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>
<div class="block">Different methods to create phylogeny tree, and tables for each genome elements</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#%3Cinit%3E()">PhylogenyListeriomics()</a></span> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/PhylogenySmallORFs.html" title="class in bacnet.scripts.blast"><span class="typeNameLink">PhylogenySmallORFs</span></a> - Class in <a href="../bacnet/scripts/blast/package-summary.html">bacnet.scripts.blast</a></dt>
<dd>
<div class="block">Tools to blast a small ORF on other bacterial genomes, and display results on a phylogeny tree</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html" title="class in bacnet.scripts.phylogeny"><span class="typeNameLink">PhylogenySmallORFs</span></a> - Class in <a href="../bacnet/scripts/phylogeny/package-summary.html">bacnet.scripts.phylogeny</a></dt>
<dd>
<div class="block">Tools to blast a small ORF on other bacterial genomes, and display results on a phylogeny tree</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/PhylogenySmallORFs.html#%3Cinit%3E()">PhylogenySmallORFs()</a></span> - Constructor for class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/PhylogenySmallORFs.html" title="class in bacnet.scripts.blast">PhylogenySmallORFs</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html#%3Cinit%3E()">PhylogenySmallORFs()</a></span> - Constructor for class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html" title="class in bacnet.scripts.phylogeny">PhylogenySmallORFs</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny"><span class="typeNameLink">PhylogenyTools</span></a> - Class in <a href="../bacnet/scripts/phylogeny/package-summary.html">bacnet.scripts.phylogeny</a></dt>
<dd>
<div class="block">Different methods to manipulate and display Newick and other phylogeny files</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/PhylogenyTools.html#%3Cinit%3E()">PhylogenyTools()</a></span> - Constructor for class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny"><span class="typeNameLink">PhylogenyToolsJolley</span></a> - Class in <a href="../bacnet/scripts/phylogeny/package-summary.html">bacnet.scripts.phylogeny</a></dt>
<dd>
<div class="block">This list of function is useful to manipulate Newick files given by Jolley in May 2012, from its
 publication Jolley et al.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#%3Cinit%3E()">PhylogenyToolsJolley()</a></span> - Constructor for class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.DNAType.html#PLASMID">PLASMID</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.DNAType.html#PLASTID">PLASTID</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/UNAfold.html#plot2ann">plot2ann</a></span> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/UNAfold.html#plot2ann">plot2ann</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/UNAfold.html#plot2Ann(java.lang.String,java.lang.String,java.lang.String)">plot2Ann(String, String, String)</a></span> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>
<div class="block">Convert a .ct file and .plot file into an .ann file<br>
 The output of plot2ann script is save into a .ann file</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/UNAfold.html#plot2Ann(java.lang.String,java.lang.String,java.lang.String)">plot2Ann(String, String, String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>
<div class="block">Convert a .ct file and .plot file into an .ann file<br>
 The output of plot2ann script is save into a .ann file</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/UNAfold.html#plotFolding(java.lang.String,int,java.lang.String)">plotFolding(String, int, String)</a></span> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>
<div class="block">Run sir_graph software on a particular file</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/UNAfold.html#plotFolding(java.lang.String,int,java.lang.String)">plotFolding(String, int, String)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>
<div class="block">Run sir_graph software on a particular file</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Activator.html#PLUGIN_ID">PLUGIN_ID</a></span> - Static variable in class bacnet.<a href="../bacnet/Activator.html" title="class in bacnet">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/Activator.html#PLUGIN_ID">PLUGIN_ID</a></span> - Static variable in class bacnet.scripts.<a href="../bacnet/scripts/Activator.html" title="class in bacnet.scripts">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html#plus">plus</a></span> - bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#plus()">plus()</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Return location that is in same position on plus strand.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ExpressionMatrixStat.html#plus(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.dataset.ExpressionMatrix)">plus(ExpressionMatrix, ExpressionMatrix)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Perform an addition between two ExpressionMatrix <br>
 First the intersection in term of RowNames and headers is found, then addition is done for each
 row</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html#plusM">plusM</a></span> - bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html#plusP">plusP</a></span> - bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template"><span class="typeNameLink">Point</span></a> - Interface in <a href="../org/biojava3/core/sequence/location/template/package-summary.html">org.biojava3.core.sequence.location.template</a></dt>
<dd>
<div class="block">Holds a single point part of a location</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Point.Resolver.html" title="interface in org.biojava3.core.sequence.location.template"><span class="typeNameLink">Point.Resolver</span></a>&lt;<a href="../org/biojava3/core/sequence/location/template/Point.Resolver.html" title="type parameter in Point.Resolver">T</a> extends <a href="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/location/template/package-summary.html">org.biojava3.core.sequence.location.template</a></dt>
<dd>
<div class="block">Used to resolve a position about a point</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/CRC64Checksum.html#POLY64">POLY64</a></span> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/CRC64Checksum.html" title="class in org.biojava3.core.util">CRC64Checksum</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/ConcurrencyTools.html#pool">pool</a></span> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#populate(java.lang.String)">populate(String)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>
<div class="block">Loops through the chars in a String and passes them onto <a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#setCompoundAt(char,int)"><code>BitSequenceReader.BitArrayWorker.setCompoundAt(char, int)</code></a></div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#populate(org.biojava3.core.sequence.template.Sequence)">populate(Sequence&lt;C&gt;)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>
<div class="block">Loops through the Compounds in a Sequence and passes them onto
 <a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#setCompoundAt(C,int)"><code>BitSequenceReader.BitArrayWorker.setCompoundAt(Compound, int)</code></a></div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/IUPACParser.html#populateLookups()">populateLookups()</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/SimplePoint.html#position">position</a></span> - Variable in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/Strand.html#POSITIVE">POSITIVE</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/ProteinTools.html#possibleSmallRNAPeptides()">possibleSmallRNAPeptides()</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Get all possible peptides from EGD-e: <br>
 get all proteins sequence from Small RNAs
 get peptide sequence from M start to first R, or first E
 cut-off every peptide of size under 5aa
 export in a table
 use <code>http://web.expasy.org/compute_pi/</code> to calculate mass and isoelectric point
 save the different mass in NTermUtils.getPATH()+EGDePeptideMass.txt</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#possibleSmallRNAPeptides()">possibleSmallRNAPeptides()</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Get all possible peptides from EGD-e: <br>
 get all proteins sequence from Small RNAs
 get peptide sequence from M start to first R, or first E
 cut-off every peptide of size under 5aa
 export in a table
 use <code>http://web.expasy.org/compute_pi/</code> to calculate mass and isoelectric point
 save the different mass in NTermUtils.getPATH()+EGDePeptideMass.txt</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewCRISPR.html#postConstruct(org.eclipse.swt.widgets.Composite)">postConstruct(Composite)</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewCRISPR.html" title="class in bacnet.e4.rap">InitViewCRISPR</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html#postProcessCompoundLists(java.util.List)">postProcessCompoundLists(List&lt;List&lt;AminoAcidCompound&gt;&gt;)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html" title="class in org.biojava3.core.sequence.transcription">RNAToAminoAcidTranslator</a></dt>
<dd>
<div class="block">Performs the trimming of stop codons and the conversion of a valid start amino acid to M</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#postProcessCompoundLists(java.util.List)">postProcessCompoundLists(List&lt;List&lt;NucleotideCompound&gt;&gt;)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava3.core.sequence.transcription">DNAToRNATranslator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#postProcessCompoundLists(java.util.List)">postProcessCompoundLists(List&lt;List&lt;T&gt;&gt;)</a></span> - Method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#pow2(double%5B%5D)">pow2(double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">2^value</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#pow2(double%5B%5D%5B%5D)">pow2(double[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">2^value</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ExpressionMatrixStat.html#pow2(bacnet.datamodel.dataset.ExpressionMatrix)">pow2(ExpressionMatrix)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">2^values</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html#preCache">preCache</a></span> - Variable in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/PredaTerm.html" title="class in bacnet.scripts.core"><span class="typeNameLink">PredaTerm</span></a> - Class in <a href="../bacnet/scripts/core/package-summary.html">bacnet.scripts.core</a></dt>
<dd>
<div class="block">PREDicting poly-A region and TERMinal splice acceptor for Leishmania Please read manuscript for
 details : Smith, M., Blanchette, M., Papadopoulou, B.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/PredaTerm.html#%3Cinit%3E()">PredaTerm()</a></span> - Constructor for class bacnet.scripts.core.<a href="../bacnet/scripts/core/PredaTerm.html" title="class in bacnet.scripts.core">PredaTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#preDestroy()">preDestroy()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html#PREDICTED">PREDICTED</a></span> - org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/PredaTerm.html#PredictSJ(java.lang.String%5B%5D%5B%5D)">PredictSJ(String[][])</a></span> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/PredaTerm.html" title="class in bacnet.scripts.core">PredaTerm</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#prefix(int)">prefix(int)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">The part of this location before the specified position.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#prefix(org.biojava3.genome.parsers.gff.Location)">prefix(Location)</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">The part of this location before the other location (not inclusive).</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#PRELOADED_ITEMS">PRELOADED_ITEMS</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util"><span class="typeNameLink">PrettyXMLWriter</span></a> - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>
<div class="block">Implementation of XMLWriter which emits nicely formatted documents to a PrintWriter.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/PrettyXMLWriter.html#%3Cinit%3E(java.io.PrintWriter)">PrettyXMLWriter(PrintWriter)</a></span> - Constructor for class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#previousCodon">previousCodon</a></span> - Variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>
<div class="block">Codon upstream to mapped sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/ColorMapperWizardPage.html#previousColorMapper">previousColorMapper</a></span> - Variable in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DataSource.html#PRF">PRF</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/Hashcoder.html#PRIME">PRIME</a></span> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Hashcoder.html" title="class in org.biojava3.core.util">Hashcoder</a></dt>
<dd>
<div class="block">The prime number used to multiply any calculated hashcode seed by

 i.e. result = PRIME*result + c

 Where result is the result of the previous calculation (at first this will be seed) and c is the
 calculated int to add to result</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/PrettyXMLWriter.html#print(java.lang.String)">print(String)</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/XMLWriter.html#print(java.lang.String)">print(String)</a></span> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Prints some textual content in an element.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/PrettyXMLWriter.html#printAttributeValue(java.lang.String)">printAttributeValue(String)</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/PrettyXMLWriter.html#printChars(java.lang.String)">printChars(String)</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#printConservation(java.lang.StringBuilder,java.lang.String,int,int,boolean)">printConservation(StringBuilder, String, int, int, boolean)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/PrettyXMLWriter.html#println(java.lang.String)">println(String)</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/XMLWriter.html#println(java.lang.String)">println(String)</a></span> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Prints some textual content, terminated with a newline character.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/PrettyXMLWriter.html#printRaw(java.lang.String)">printRaw(String)</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/XMLWriter.html#printRaw(java.lang.String)">printRaw(String)</a></span> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Send raw data to the stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#printSequenceAlignmentWeb(java.lang.StringBuilder,int,java.lang.String,int,int)">printSequenceAlignmentWeb(StringBuilder, int, String, int, int)</a></span> - Method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#probes">probes</a></span> - Variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#probes">probes</a></span> - Variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#PROBES_PATH">PROBES_PATH</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#PROBES_PATH">PROBES_PATH</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/GeneExpression.html#PROBES_PATH_2">PROBES_PATH_2</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/Tiling.html#PROBES_PATH_2">PROBES_PATH_2</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#probesGExpression">probesGExpression</a></span> - Variable in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">ProbeInformation for GeneExpression</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#probesTiling">probesTiling</a></span> - Variable in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">ProbeInformation for Tiling and GeneExpression</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html#process()">process()</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FastaReader.html#process()">process()</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaReader.html" title="class in org.biojava3.core.sequence.io">FastaReader</a></dt>
<dd>
<div class="block">The parsing is done in this method</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FastaWriter.html#process()">process()</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/query/OutputHitsGFF.html#process(java.io.File,java.io.File,java.io.File,double,double,boolean,boolean)">process(File, File, File, double, double, boolean, boolean)</a></span> - Method in class org.biojava3.genome.query.<a href="../org/biojava3/genome/query/OutputHitsGFF.html" title="class in org.biojava3.genome.query">OutputHitsGFF</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/IOUtils.ReaderProcessor.html#process(java.lang.String)">process(String)</a></span> - Method in interface org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.ReaderProcessor.html" title="interface in org.biojava3.core.sequence.io.util">IOUtils.ReaderProcessor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/uniprot/UniprotToFasta.html#process(java.lang.String,java.lang.String)">process(String, String)</a></span> - Method in class org.biojava3.genome.uniprot.<a href="../org/biojava3/genome/uniprot/UniprotToFasta.html" title="class in org.biojava3.genome.uniprot">UniprotToFasta</a></dt>
<dd>
<div class="block">Convert a Uniprot sequence file to a fasta file.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/FastaFileReader.html#process(org.biojava3.core.sequence.template.CompoundSet)">process(CompoundSet&lt;?&gt;)</a></span> - Method in class bacnet.reader.<a href="../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a></dt>
<dd>
<div class="block">The parsing is done in this method
 
 ******* There is a problem when the genes in the fasta file have the same Id, so I modify this
 method accordingly</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/UGiessen.html#processAllData()">processAllData()</a></span> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/UGiessen.html" title="class in bacnet.scripts.listeriomics.technology">UGiessen</a></dt>
<dd>
<div class="block">Open every processed files and extract the different signal from it.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#processDanielRiboSeq()">processDanielRiboSeq()</a></span> - Method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>
<div class="block">Process Daniel Dar TermSeq</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#processGenomeName(java.lang.String)">processGenomeName(String)</a></span> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Replace all spaces in genomeName by "_" to have folder name compatible for Blast search<br>
 All "_" already present are replaced by "--"</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/util/SplitFasta.html#processNucleotides(java.io.File,java.lang.String,java.io.File)">processNucleotides(File, String, File)</a></span> - Method in class org.biojava3.genome.util.<a href="../org/biojava3/genome/util/SplitFasta.html" title="class in org.biojava3.genome.util">SplitFasta</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#processReader(java.io.BufferedReader,org.biojava3.core.sequence.io.util.IOUtils.ReaderProcessor)">processReader(BufferedReader, IOUtils.ReaderProcessor)</a></span> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Takes in a reader and a processor, reads every line from the given file and then invokes the
 processor.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#processUnknownCompound(C,int)">processUnknownCompound(C, int)</a></span> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>
<div class="block">Since bit encoding only supports a finite number of bases it is more than likely when processing
 sequence you will encounter a compound which is not covered by the encoding e.g.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#product">product</a></span> - Variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/NcRNA.html#product">product</a></span> - Variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#projectName">projectName</a></span> - Variable in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Name of the current project loaded</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/BasicApplication.html#projectName">projectName</a></span> - Static variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BasicApplication.html" title="class in bacnet.e4.rap">BasicApplication</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/setup/BasicApplication.html#projectName">projectName</a></span> - Static variable in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/BasicApplication.html" title="class in bacnet.e4.rap.setup">BasicApplication</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#properties">properties</a></span> - Variable in class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/BioConditionComparator.html#propertyIndex">propertyIndex</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/TableViewerComparator.html#propertyIndex">propertyIndex</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/TableViewerComparator.html" title="class in bacnet.table.core">TableViewerComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#protein_id">protein_id</a></span> - Variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#proteinCreator(org.biojava3.core.sequence.io.template.SequenceCreatorInterface)">proteinCreator(SequenceCreatorInterface&lt;AminoAcidCompound&gt;)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/FastaFileReader.html#proteinfileExt">proteinfileExt</a></span> - Static variable in class bacnet.reader.<a href="../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Chromosome.html#proteinIDTolocusTag">proteinIDTolocusTag</a></span> - Variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></dt>
<dd>
<div class="block">Map proteinId to locustag</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence"><span class="typeNameLink">ProteinSequence</span></a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">The representation of a ProteinSequence</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ProteinSequence.html#%3Cinit%3E(java.lang.String)">ProteinSequence(String)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></dt>
<dd>
<div class="block">Create a protein from a string</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ProteinSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">ProteinSequence(String, CompoundSet&lt;AminoAcidCompound&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></dt>
<dd>
<div class="block">Create a protein from a string with a user defined set of amino acids</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ProteinSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader)">ProteinSequence(ProxySequenceReader&lt;AminoAcidCompound&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></dt>
<dd>
<div class="block">A protein sequence where the storage of the sequence is somewhere else.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ProteinSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader,org.biojava3.core.sequence.template.CompoundSet)">ProteinSequence(ProxySequenceReader&lt;AminoAcidCompound&gt;, CompoundSet&lt;AminoAcidCompound&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></dt>
<dd>
<div class="block">A protein sequence where the storage of the sequence is somewhere else with user defined set of
 amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#proteinSequenceCreator">proteinSequenceCreator</a></span> - Variable in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#proteinSequenceCreator">proteinSequenceCreator</a></span> - Variable in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io"><span class="typeNameLink">ProteinSequenceCreator</span></a> - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">Used to create a ProteinSequence from a String to allow for details about the location of the
 sequence etc.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)">ProteinSequenceCreator(CompoundSet&lt;AminoAcidCompound&gt;)</a></span> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">ProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast"><span class="typeNameLink">ProteinTools</span></a> - Class in <a href="../bacnet/scripts/blast/package-summary.html">bacnet.scripts.blast</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/ProteinTools.html#%3Cinit%3E()">ProteinTools()</a></span> - Constructor for class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html#Proteome">Proteome</a></span> - bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#proteomes">proteomes</a></span> - Variable in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">All ProteomicsData data associated to this BioCondition</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/DataValidation.html#proteomes">proteomes</a></span> - Variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>
<div class="block">HashMap for validation of proteomes</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/Database.html#proteomesArrayPath">proteomesArrayPath</a></span> - Variable in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for the proteome table</div>
</dd>
<dt><a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database"><span class="typeNameLink">ProteomicsCreation</span></a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/ProteomicsCreation.html#%3Cinit%3E()">ProteomicsCreation()</a></span> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset"><span class="typeNameLink">ProteomicsData</span></a> - Class in <a href="../bacnet/datamodel/dataset/package-summary.html">bacnet.datamodel.dataset</a></dt>
<dd>
<div class="block">OmicsData class inherited everything from ExpressionMatrix, the only difference is the TypeDate
 value, it a Proteome data.<br>
 It is not very useful for the moment but will be in the futur</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ProteomicsData.html#%3Cinit%3E()">ProteomicsData()</a></span> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas"><span class="typeNameLink">ProteomicsDataFilterComposite</span></a> - Class in <a href="../bacnet/expressionAtlas/package-summary.html">bacnet.expressionAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#%3Cinit%3E(org.eclipse.swt.widgets.Composite,int,bacnet.expressionAtlas.ProteomicsView)">ProteomicsDataFilterComposite(Composite, int, ProteomicsView)</a></span> - Constructor for class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas"><span class="typeNameLink">ProteomicsView</span></a> - Class in <a href="../bacnet/expressionAtlas/package-summary.html">bacnet.expressionAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsView.html#%3Cinit%3E()">ProteomicsView()</a></span> - Constructor for class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template"><span class="typeNameLink">ProxySequenceReader</span></a>&lt;<a href="../org/biojava3/core/sequence/template/ProxySequenceReader.html" title="type parameter in ProxySequenceReader">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#pseudogene">pseudogene</a></span> - Variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/NcRNA.TypeNcRNA.html#pseudoGene">pseudoGene</a></span> - bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.TypeNcRNA.html" title="enum in bacnet.datamodel.sequence">NcRNA.TypeNcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/PTTReader.html" title="class in bacnet.reader"><span class="typeNameLink">PTTReader</span></a> - Class in <a href="../bacnet/reader/package-summary.html">bacnet.reader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/reader/PTTReader.html#%3Cinit%3E()">PTTReader()</a></span> - Constructor for class bacnet.reader.<a href="../bacnet/reader/PTTReader.html" title="class in bacnet.reader">PTTReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/BasicColor.html#PURPLE">PURPLE</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewBacnet.html#pushState()">pushState()</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewBacnet.html" title="class in bacnet.e4.rap">InitViewBacnet</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewLeishmania.html#pushState()">pushState()</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewLeishmania.html" title="class in bacnet.e4.rap">InitViewLeishmania</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewListeria.html#pushState()">pushState()</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeria.html" title="class in bacnet.e4.rap">InitViewListeria</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewListeriaSample.html#pushState()">pushState()</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">InitViewListeriaSample</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewYersinia.html#pushState()">pushState()</a></span> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewYersinia.html" title="class in bacnet.e4.rap">InitViewYersinia</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#pushState()">pushState()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>
<div class="block">Push states to change url</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#pushState()">pushState()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>
<div class="block">Push states to change url</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#pushState()">pushState()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>
<div class="block">Push states to change url</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#pushState()">pushState()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#pushState()">pushState()</a></span> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/CRISPRGuideView.html#pushState()">pushState()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/NTerminomicsView.html#pushState()">pushState()</a></span> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#pushState()">pushState()</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Push states to change url</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/AnnotationView.html#pushState()">pushState()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>
<div class="block">Push states to change url</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#pushState()">pushState()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GenomicsView.html#pushState()">pushState()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#pushState()">pushState()</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CoExprNetworkView.html#pushState()">pushState()</a></span> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>
<div class="block">Push Naviagtion state</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/NavigationManagement.html#pushStateView(java.lang.String)">pushStateView(String)</a></span> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/NavigationManagement.html#pushStateView(java.lang.String,java.util.HashMap)">pushStateView(String, HashMap&lt;String, String&gt;)</a></span> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>
<div class="block">Push state puttin viewId and state parameters <br>
 stateName = viewID + key + "=" + parameters.get(key) + ";"</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/SoftHashMap.html#put(K,V)">put(K, V)</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>
<div class="block">Here we put the key, value pair into the HashMap using a SoftValue
 
 object.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html#PVALUE">PVALUE</a></span> - bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html" title="enum in bacnet.datamodel.dataset">OmicsData.ColNames</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.TypeModif.html#PyroGlu">PyroGlu</a></span> - bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView.html#PYTHON_PATH">PYTHON_PATH</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView2.html#PYTHON_PATH">PYTHON_PATH</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView.html#PYTHON_SERVER">PYTHON_SERVER</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView2.html#PYTHON_SERVER">PYTHON_SERVER</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">K</a>&nbsp;<a href="index-12.html">L</a>&nbsp;<a href="index-13.html">M</a>&nbsp;<a href="index-14.html">N</a>&nbsp;<a href="index-15.html">O</a>&nbsp;<a href="index-16.html">P</a>&nbsp;<a href="index-17.html">Q</a>&nbsp;<a href="index-18.html">R</a>&nbsp;<a href="index-19.html">S</a>&nbsp;<a href="index-20.html">T</a>&nbsp;<a href="index-21.html">U</a>&nbsp;<a href="index-22.html">V</a>&nbsp;<a href="index-23.html">W</a>&nbsp;<a href="index-24.html">X</a>&nbsp;<a href="index-25.html">Y</a>&nbsp;<a href="index-26.html">Z</a>&nbsp;<a href="index-27.html">_</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a></div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
