<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>M-Index</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="M-Index";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="contentContainer"><a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">K</a>&nbsp;<a href="index-12.html">L</a>&nbsp;<a href="index-13.html">M</a>&nbsp;<a href="index-14.html">N</a>&nbsp;<a href="index-15.html">O</a>&nbsp;<a href="index-16.html">P</a>&nbsp;<a href="index-17.html">Q</a>&nbsp;<a href="index-18.html">R</a>&nbsp;<a href="index-19.html">S</a>&nbsp;<a href="index-20.html">T</a>&nbsp;<a href="index-21.html">U</a>&nbsp;<a href="index-22.html">V</a>&nbsp;<a href="index-23.html">W</a>&nbsp;<a href="index-24.html">X</a>&nbsp;<a href="index-25.html">Y</a>&nbsp;<a href="index-26.html">Z</a>&nbsp;<a href="index-27.html">_</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a><a id="I:M">
<!--   -->
</a>
<h2 class="title">M</h2>
<dl>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/AMtools.html#M(double,double)">M(double, double)</a></span> - Static method in class bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/AMtools.html" title="class in bacnet.scripts.core.stat">AMtools</a></dt>
<dd>
<div class="block">Calculate M = x1-x2<br>
 Represents the variability within expression value x1 and x2</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#m_colorMap">m_colorMap</a></span> - Static variable in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#m_decoratedImageMap">m_decoratedImageMap</a></span> - Static variable in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Maps images to decorated images.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#m_decoratedImageMap">m_decoratedImageMap</a></span> - Static variable in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Maps images to decorated images.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#m_descriptorImageMap">m_descriptorImageMap</a></span> - Static variable in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#m_designTimePluginResourceProvider">m_designTimePluginResourceProvider</a></span> - Static variable in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Instance of <a href="../bacnet/swt/ResourceManager.PluginResourceProvider.html" title="interface in bacnet.swt"><code>ResourceManager.PluginResourceProvider</code></a>, used by WindowBuilder at design time.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#m_fontMap">m_fontMap</a></span> - Static variable in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Maps font names to fonts.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#m_fontToBoldFontMap">m_fontToBoldFontMap</a></span> - Static variable in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Maps fonts to their bold versions.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#m_idToCursorMap">m_idToCursorMap</a></span> - Static variable in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Maps IDs to cursors.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#m_imageMap">m_imageMap</a></span> - Static variable in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Maps image paths to images.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/ResourceManager.html#m_URLImageMap">m_URLImageMap</a></span> - Static variable in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Maps URL to images.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/Macroarray.html" title="class in bacnet.scripts.listeriomics.technology"><span class="typeNameLink">Macroarray</span></a> - Class in <a href="../bacnet/scripts/listeriomics/technology/package-summary.html">bacnet.scripts.listeriomics.technology</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/technology/Macroarray.html#%3Cinit%3E()">Macroarray()</a></span> - Constructor for class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/Macroarray.html" title="class in bacnet.scripts.listeriomics.technology">Macroarray</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FastaReader.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaReader.html" title="class in org.biojava3.core.sequence.io">FastaReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/FastaWriter.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.core.sequence.loader.<a href="../org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/ProteinSequence.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/UncompressInputStream.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/App.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/App.html" title="class in org.biojava3.genome">App</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/GeneFeatureHelper.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.genome.parsers.geneid.<a href="../org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html" title="class in org.biojava3.genome.parsers.geneid">GeneIDXMLReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Feature.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html" title="class in org.biojava3.genome.parsers.gff">GeneIDGFF2Reader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html" title="class in org.biojava3.genome.parsers.gff">GeneMarkGTFReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/GFF3Reader.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GFF3Reader.html" title="class in org.biojava3.genome.parsers.gff">GFF3Reader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/GFF3Writer.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/GFF3Writer.html" title="class in org.biojava3.genome.parsers.gff">GFF3Writer</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/LocIterator.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/query/BlastXMLQuery.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.genome.query.<a href="../org/biojava3/genome/query/BlastXMLQuery.html" title="class in org.biojava3.genome.query">BlastXMLQuery</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/query/OutputHitsGFF.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.genome.query.<a href="../org/biojava3/genome/query/OutputHitsGFF.html" title="class in org.biojava3.genome.query">OutputHitsGFF</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/uniprot/UniprotToFasta.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.genome.uniprot.<a href="../org/biojava3/genome/uniprot/UniprotToFasta.html" title="class in org.biojava3.genome.uniprot">UniprotToFasta</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/util/SplitFasta.html#main(java.lang.String%5B%5D)">main(String[])</a></span> - Static method in class org.biojava3.genome.util.<a href="../org/biojava3/genome/util/SplitFasta.html" title="class in org.biojava3.genome.util">SplitFasta</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/BasicApplication.html#MAIN_ENTRYPOINT">MAIN_ENTRYPOINT</a></span> - Static variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BasicApplication.html" title="class in bacnet.e4.rap">BasicApplication</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/setup/BasicApplication.html#MAIN_ENTRYPOINT">MAIN_ENTRYPOINT</a></span> - Static variable in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/BasicApplication.html" title="class in bacnet.e4.rap.setup">BasicApplication</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics"><span class="typeNameLink">MainListeriomics</span></a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#%3Cinit%3E()">MainListeriomics()</a></span> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/Blast.html#makeblastdb">makeblastdb</a></span> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/database/HomologCreation.html#makeblastdb">makeblastdb</a></span> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Shortcut for running makeblastdb</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/GlaserFCannotation.html#MANDINFC_MAP">MANDINFC_MAP</a></span> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/GlaserFCannotation.html" title="class in bacnet.datamodel.annotation">GlaserFCannotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#MANNWHIT">MANNWHIT</a></span> - bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/SoftHashMap.html#map">map</a></span> - Variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>
<div class="block">The internal HashMap that stores SoftReference to actual data.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#mapPeptide()">mapPeptide()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">On each of the 6 frame genome, map the different peptide</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html#mapper">mapper</a></span> - Variable in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#mappingFrame">mappingFrame</a></span> - Variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>
<div class="block">Frame where the peptide has been mapped</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html#%3Cinit%3E()">MappingFrame()</a></span> - Constructor for enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#mapSequence(java.lang.String)">mapSequence(String)</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">Map an amino acide sequence to 6 frame Genomes<br>
 From initializeGenomeSequence() we get all 6 possible amino acids sequence of the genome, and map
 the peptide<br>
 The different mapping coordinates are transformed into nucleotide coordinates (= genome
 coordinates)</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#mark(int)">mark(int)</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>
<div class="block">Marks the present position in the stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#markedChar">markedChar</a></span> - Variable in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#markedSkipLF">markedSkipLF</a></span> - Variable in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>
<div class="block">The skipLF flag when the mark was set</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html#markSupported()">markSupported()</a></span> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></dt>
<dd>
<div class="block">Tells whether this stream supports the mark() operation, which it does.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/UncompressInputStream.html#markSupported()">markSupported()</a></span> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></dt>
<dd>
<div class="block">This stream does not support mark/reset on the stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/RWTUtils.html#MARKUP_ENABLED">MARKUP_ENABLED</a></span> - Static variable in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html#MASK">MASK</a></span> - Static variable in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader.FourBitArrayWorker</a></dt>
<dd>
<div class="block">Masking value used for extracting the right most 2 bits from a byte</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html#MASK">MASK</a></span> - Static variable in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader.TwoBitArrayWorker</a></dt>
<dd>
<div class="block">Masking value used for extracting the right most 2 bits from a byte</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#massSpec">massSpec</a></span> - Variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#massSpec2">massSpec2</a></span> - Variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#massSpecData">massSpecData</a></span> - Variable in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">MassSpecData where the selected NTerm is</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#massSpecData">massSpecData</a></span> - Variable in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">MassSpecData where the selected NTerm is</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html#massSpecData">massSpecData</a></span> - Variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html" title="class in bacnet.scripts.listeriomics.nterm">NTermCreateFinalData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html#massSpecDataNew">massSpecDataNew</a></span> - Variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html" title="class in bacnet.scripts.listeriomics.nterm">NTermCreateFinalData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/MathUtils.html" title="class in bacnet.utils"><span class="typeNameLink">MathUtils</span></a> - Class in <a href="../bacnet/utils/package-summary.html">bacnet.utils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/MathUtils.html#%3Cinit%3E()">MathUtils()</a></span> - Constructor for class bacnet.utils.<a href="../bacnet/utils/MathUtils.html" title="class in bacnet.utils">MathUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#matrices">matrices</a></span> - Variable in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">All data in <code>ExpressioMatrix</code> associated to this BioCondition</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#matrix">matrix</a></span> - Variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>
<div class="block">ExpressionMatrix used only at the creation of the data</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#matrix">matrix</a></span> - Variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/Filter.html#matrix">matrix</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/FilterList.html#matrix">matrix</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/FilterDialog.html#matrix">matrix</a></span> - Variable in class bacnet.table.gui.<a href="../bacnet/table/gui/FilterDialog.html" title="class in bacnet.table.gui">FilterDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/gui/HideExpressionMatrixDialog.html#matrix">matrix</a></span> - Variable in class bacnet.table.gui.<a href="../bacnet/table/gui/HideExpressionMatrixDialog.html" title="class in bacnet.table.gui">HideExpressionMatrixDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/TableSWTComposite.html#matrix">matrix</a></span> - Variable in class bacnet.table.<a href="../bacnet/table/TableSWTComposite.html" title="class in bacnet.table">TableSWTComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/Filter.html#matrix">matrix</a></span> - Variable in class bacnet.utils.<a href="../bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#matrixDisplayed">matrixDisplayed</a></span> - Variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/TableSWTComposite.html#matrixDisplayed">matrixDisplayed</a></span> - Variable in class bacnet.table.<a href="../bacnet/table/TableSWTComposite.html" title="class in bacnet.table">TableSWTComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html#matrixSize">matrixSize</a></span> - Variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/TableSWTComposite.html#matrixSize">matrixSize</a></span> - Variable in class bacnet.table.<a href="../bacnet/table/TableSWTComposite.html" title="class in bacnet.table">TableSWTComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core"><span class="typeNameLink">MatrixSize</span></a> - Class in <a href="../bacnet/table/core/package-summary.html">bacnet.table.core</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/MatrixSize.html#%3Cinit%3E()">MatrixSize()</a></span> - Constructor for class bacnet.table.core.<a href="../bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Feature.html#mAttributes">mAttributes</a></span> - Variable in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#max">max</a></span> - Variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#max">max</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#max">max</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#max">max</a></span> - Variable in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#max(double%5B%5D)">max(double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Find maximum value of a vector</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#max(double%5B%5D%5B%5D)">max(double[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ExpressionMatrixStat.html#max(bacnet.datamodel.dataset.ExpressionMatrix)">max(ExpressionMatrix)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Search for the maximum of all the values</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ExpressionMatrixStat.html#max(bacnet.datamodel.dataset.ExpressionMatrix,int)">max(ExpressionMatrix, int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Search for the maximum on one column</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/UncompressInputStream.html#MAX_BITS">MAX_BITS</a></span> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/SoftHashMap.html#max_limit">max_limit</a></span> - Variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>
<div class="block">Maximum Number of references you dont want GC to collect.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#MAX_RESOLVER">MAX_RESOLVER</a></span> - Static variable in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>
<div class="block">Always uses the max point to resolve a location</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/UncompressInputStream.html#maxbits">maxbits</a></span> - Variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/UncompressInputStream.html#maxcode">maxcode</a></span> - Variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#maxColor">maxColor</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#maxColor">maxColor</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#maxColorSWT">maxColorSWT</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#maxColorSWT">maxColorSWT</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html#maxCompoundCharSequenceLength">maxCompoundCharSequenceLength</a></span> - Variable in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/UncompressInputStream.html#maxmaxcode">maxmaxcode</a></span> - Variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#maxPos">maxPos</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#maxPos">maxPos</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#maxQuant">maxQuant</a></span> - Variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>
<div class="block">If the peptide has also been found with MaxQuant analysis software, we write something in this
 attribute, otherwise its blank</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#maxQuantum">maxQuantum</a></span> - Variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>
<div class="block">If the peptide has also been found with MaxQuant analysis software, we write something in this
 attribute, otherwise its blank, each column is for one type of experiments.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#maxSequenceIndex">maxSequenceIndex</a></span> - Variable in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/LocIterator.html#mBounds">mBounds</a></span> - Variable in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/util/FlatFileCache.html#me">me</a></span> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/FlatFileCache.html" title="class in org.biojava3.core.util">FlatFileCache</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#mean">mean</a></span> - Variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#mean">mean</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#mean">mean</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#mean(double%5B%5D)">mean(double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Calculate mean of a vector</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#mean(double%5B%5D%5B%5D)">mean(double[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Return the mean of the whole array</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/MathUtils.html#mean(double,double)">mean(double, double)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/MathUtils.html" title="class in bacnet.utils">MathUtils</a></dt>
<dd>
<div class="block">Calculate the mean between two numbers</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ExpressionMatrixStat.html#mean(bacnet.datamodel.dataset.ExpressionMatrix)">mean(ExpressionMatrix)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Calculate the mean of the ExpressionMatrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ExpressionMatrixStat.html#mean(bacnet.datamodel.dataset.ExpressionMatrix,int)">mean(ExpressionMatrix, int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Calculate the mean of the ExpressionMatrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#mean(java.lang.Double%5B%5D)">mean(Double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#MEAN_RESOLVER">MEAN_RESOLVER</a></span> - Static variable in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>
<div class="block">Combines min and max and then gets the mean of it</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#media">media</a></span> - Variable in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Growth media is specified here</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#mediaGrowthProperties">mediaGrowthProperties</a></span> - Variable in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Quantity/properties of drugs added to the media if specified: quantity of glucose, type of Broth</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#median">median</a></span> - Variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#median">median</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#median">median</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#median(double%5B%5D)">median(double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Calculate median of a vector</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ExpressionMatrixStat.html#median(bacnet.datamodel.dataset.ExpressionMatrix,int)">median(ExpressionMatrix, int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Calculate the median of the ExpressionMatrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#median(java.lang.Double%5B%5D)">median(Double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Calculate median of a vector</div>
</dd>
<dt><a href="../bacnet/scripts/core/normalization/MedianNormalization.html" title="class in bacnet.scripts.core.normalization"><span class="typeNameLink">MedianNormalization</span></a> - Class in <a href="../bacnet/scripts/core/normalization/package-summary.html">bacnet.scripts.core.normalization</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/core/normalization/MedianNormalization.html#%3Cinit%3E()">MedianNormalization()</a></span> - Constructor for class bacnet.scripts.core.normalization.<a href="../bacnet/scripts/core/normalization/MedianNormalization.html" title="class in bacnet.scripts.core.normalization">MedianNormalization</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#mEnd">mEnd</a></span> - Variable in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#merge(java.util.ArrayList,boolean)">merge(ArrayList&lt;ExpressionMatrix&gt;, boolean)</a></span> - Static method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>
<div class="block">Create a fusion of a list of ExpressionMatrix</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html#methionineAminoAcidCompound">methionineAminoAcidCompound</a></span> - Variable in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html" title="class in org.biojava3.core.sequence.transcription">RNAToAminoAcidTranslator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Feature.html#mFrame">mFrame</a></span> - Variable in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#middleH">middleH</a></span> - Variable in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#middleV">middleV</a></span> - Variable in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#midLeftColor">midLeftColor</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#midLeftColor">midLeftColor</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#midLeftColorSWT">midLeftColorSWT</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#midLeftColorSWT">midLeftColorSWT</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#midLeftPos">midLeftPos</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#midLeftPos">midLeftPos</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/BlastResult.html#midline">midline</a></span> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#midRightColor">midRightColor</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#midRightColor">midRightColor</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#midRightColorSWT">midRightColorSWT</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#midRightColorSWT">midRightColorSWT</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#midRightPos">midRightPos</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#midRightPos">midRightPos</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionData.html#min">min</a></span> - Variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#min">min</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#min">min</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#min">min</a></span> - Variable in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#min(double%5B%5D)">min(double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Find minimum value of a vector</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ArrayUtils.html#min(double%5B%5D%5B%5D)">min(double[][])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ExpressionMatrixStat.html#min(bacnet.datamodel.dataset.ExpressionMatrix)">min(ExpressionMatrix)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Search for the minimum on all the values</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ExpressionMatrixStat.html#min(bacnet.datamodel.dataset.ExpressionMatrix,int)">min(ExpressionMatrix, int)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Search for the minimum on one column</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#MIN_RESOLVER">MIN_RESOLVER</a></span> - Static variable in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>
<div class="block">Always uses the min point to resolve a location</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#minColor">minColor</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#minColor">minColor</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#minColorSWT">minColorSWT</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#minColorSWT">minColorSWT</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/LocIterator.html#mIncrement">mIncrement</a></span> - Variable in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Zoom.html#MINIMAL_BP_NB">MINIMAL_BP_NB</a></span> - Static variable in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceUsage.html#MINIMAL_SEQUENCE_DATA">MINIMAL_SEQUENCE_DATA</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceUsage.html" title="enum in org.biojava3.core.sequence">SequenceOptimizationHints.SequenceUsage</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceCollection.html#MINIMINAL_SEQUENCES">MINIMINAL_SEQUENCES</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/SequenceOptimizationHints.SequenceCollection.html" title="enum in org.biojava3.core.sequence">SequenceOptimizationHints.SequenceCollection</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#MINIMUM_DATA_SIZE">MINIMUM_DATA_SIZE</a></span> - Variable in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">Minimal height of a data track</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapper.html#minPos">minPos</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/table/core/ColorMapperRCP.html#minPos">minPos</a></span> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html#minSequenceIndex">minSequenceIndex</a></span> - Variable in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html#minus">minus</a></span> - bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#minus()">minus()</a></span> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Return location that is in same position on negative strand.</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/VectorUtils.html#minus(double%5B%5D,double%5B%5D)">minus(double[], double[])</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/utils/ExpressionMatrixStat.html#minus(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.dataset.ExpressionMatrix)">minus(ExpressionMatrix, ExpressionMatrix)</a></span> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Perform a soustraction between two ExpressionMatrix <br>
 First the intersection in term of RowNames and headers is found, then addition is done for each
 row</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html#minusM">minusM</a></span> - bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html#minusP">minusP</a></span> - bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/swt/SWTResourceManager.html#MISSING_IMAGE_SIZE">MISSING_IMAGE_SIZE</a></span> - Static variable in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/OmicsData.html#MISSING_VALUE">MISSING_VALUE</a></span> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Every Found missing values are replaced by this number</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/DNASequence.DNAType.html#MITOCHONDRIAL">MITOCHONDRIAL</a></span> - org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Feature.html#mLocation">mLocation</a></span> - Variable in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#mLocation">mLocation</a></span> - Variable in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/views/CRISPRPredictView.html#MODEL_CAS9">MODEL_CAS9</a></span> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/BannerView.html#modelService">modelService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BannerView.html" title="class in bacnet.e4.rap">BannerView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewBacnet.html#modelService">modelService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewBacnet.html" title="class in bacnet.e4.rap">InitViewBacnet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewLeishmania.html#modelService">modelService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewLeishmania.html" title="class in bacnet.e4.rap">InitViewLeishmania</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewListeria.html#modelService">modelService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeria.html" title="class in bacnet.e4.rap">InitViewListeria</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewListeriaSample.html#modelService">modelService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">InitViewListeriaSample</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/InitViewYersinia.html#modelService">modelService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewYersinia.html" title="class in bacnet.e4.rap">InitViewYersinia</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/setup/SetupPart.html#modelService">modelService</a></span> - Variable in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SetupPart.html" title="class in bacnet.e4.rap.setup">SetupPart</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/TestPart.html#modelService">modelService</a></span> - Variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/TestPart.html" title="class in bacnet.e4.rap">TestPart</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#modifHeader(java.lang.String,java.lang.String)">modifHeader(String, String)</a></span> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/dataset/NTermData.html#modifications">modifications</a></span> - Variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></dt>
<dd>
<div class="block">Link NTerm position (coded as a <code>String</code> to <code>ArrayList(NTerm)</code><br>
 This correspond to the list of mapped sequence which are equivalent to other NTerm but with
 different modification type</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/TIS.html#modifs">modifs</a></span> - Variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTerm.html#modifSequence">modifSequence</a></span> - Variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></dt>
<dd>
<div class="block">Modify sequence provided by Francis</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/e4/rap/AppSpecificMethods.html#modifyApp(java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)">modifyApp(String, EPartService)</a></span> - Static method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/AppSpecificMethods.html" title="class in bacnet.e4.rap">AppSpecificMethods</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/genome/RASTEGDeGenome.html#modifyGenome(bacnet.datamodel.sequence.Genome,java.util.ArrayList,java.lang.String)">modifyGenome(Genome, ArrayList&lt;Gene&gt;, String)</a></span> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/RASTEGDeGenome.html" title="class in bacnet.scripts.genome">RASTEGDeGenome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/raprcp/SaveFileUtils.html#modifyHTMLwithFile(java.lang.String,java.lang.String)">modifyHTMLwithFile(String, String)</a></span> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/SaveFileUtils.html" title="class in bacnet.raprcp">SaveFileUtils</a></dt>
<dd>
<div class="block">Replace "_FileName" by fileName in the htmlFile found in bacnet/html/ folder</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Genome.html#modifyID(java.lang.String,java.lang.String)">modifyID(String, String)</a></span> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Modify id of gene, to be sure that the suffix number is always of 4 numbers<br>
 ex: replace lmpc1 by lmpc0001, lmpc12 by lmpc0012<br></div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#modifyPathArray()">modifyPathArray()</a></span> - Method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>
<div class="block">Check PATHARRAY and had columns if needed<br>
 <br>
 
 Two types of array are possible:
 one with only one experiment (contains only one column with "# spectra")
 one with only multiple experiments (contains many columns containing "# spectra") <br>
 <br>
 For The first type:<br>
 Check if PATHARRAY table is complete = has 8 columns, first column is the list of peptide,
 soluble and insoluble contains "TRUE" and "FALSE" and no "yes"<br>
 Change "TRUE" and "FALSE" to "yes" and "no"
 Remove line with NH2-free peptide <br>
 <br>
 For the second type:<br></div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/LocationHelper.html#modulateCircularIndex(int,int)">modulateCircularIndex(int, int)</a></span> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Takes a point on a circular location and moves it left until it falls at the earliest possible
 point that represents the same base.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/location/template/Location.Tools.html#modulateCircularIndex(int,int)">modulateCircularIndex(int, int)</a></span> - Static method in class org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Location.Tools.html" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a></dt>
<dd>
<div class="block">Takes a point on a circular location and moves it left until it falls at the earliest possible
 point that represents the same base.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/AbstractCompound.html#molecularWeight">molecularWeight</a></span> - Variable in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/sequence/Gene.html#molMass">molMass</a></span> - Variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/annotation/Annotation.html#MOUSE_ANNOTTABLE">MOUSE_ANNOTTABLE</a></span> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#mouseDoubleClick(org.eclipse.swt.events.MouseEvent)">mouseDoubleClick(MouseEvent)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#mouseDoubleClick(org.eclipse.swt.events.MouseEvent)">mouseDoubleClick(MouseEvent)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#mouseDoubleClick(org.eclipse.swt.events.MouseEvent)">mouseDoubleClick(MouseEvent)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#mouseDown(org.eclipse.swt.events.MouseEvent)">mouseDown(MouseEvent)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#mouseDown(org.eclipse.swt.events.MouseEvent)">mouseDown(MouseEvent)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#mouseDown(org.eclipse.swt.events.MouseEvent)">mouseDown(MouseEvent)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#mouseMove(org.eclipse.swt.events.MouseEvent)">mouseMove(MouseEvent)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">If the mouse is over the genome viewer, detect it is on the annotation, and display the
 corresponding gene, sRNA, asRNA</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#mouseMove(org.eclipse.swt.events.MouseEvent)">mouseMove(MouseEvent)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>
<div class="block">If the mouse is over the genome viewer, detect it is on the annotation, and display the
 corresponding gene, sRNA, asRNA</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#mouseUp(org.eclipse.swt.events.MouseEvent)">mouseUp(MouseEvent)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/GeneView.html#mouseUp(org.eclipse.swt.events.MouseEvent)">mouseUp(MouseEvent)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/sequenceTools/SrnaView.html#mouseUp(org.eclipse.swt.events.MouseEvent)">mouseUp(MouseEvent)</a></span> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#mouseXPosition">mouseXPosition</a></span> - Variable in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">When mouse is clicked we displayed a line is displayed to show synteny: mouseYPosition gives the
 X position</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#mouseXPosition">mouseXPosition</a></span> - Variable in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>
<div class="block">When mouse is clicked we displayed a line is displayed to show synteny: mouseYPosition gives the
 X position</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#mouseYPosition">mouseYPosition</a></span> - Variable in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">When mouse is clicked we displayed a line is displayed to show synteny: mouseYPosition gives the
 Y position</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html#mouseYPosition">mouseYPosition</a></span> - Variable in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></dt>
<dd>
<div class="block">When mouse is clicked we displayed a line is displayed to show synteny: mouseYPosition gives the
 Y position</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Region.html#moveHorizontally(int)">moveHorizontally(int)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/core/Track.html#moveHorizontally(int)">moveHorizontally(int)</a></span> - Method in class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#moveHorizontally(int)">moveHorizontally(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Move horizontally the tracks</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html#moveVertically(int)">moveVertically(int)</a></span> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></dt>
<dd>
<div class="block">From the position of the vertical bar draw the corresponding part of the data</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/LocIterator.html#mPosition">mPosition</a></span> - Variable in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Feature.html#mScore">mScore</a></span> - Variable in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Feature.html#mSeqname">mSeqname</a></span> - Variable in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html#MSF">MSF</a></span> - org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html" title="enum in org.biojava3.core.sequence.template">LightweightProfile.StringFormat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Feature.html#mSource">mSource</a></span> - Variable in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Location.html#mStart">mStart</a></span> - Variable in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Feature.html#mType">mType</a></span> - Variable in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/nterm/NTermConservationWithRiboProfiling.html#multiAlign(java.util.ArrayList,java.lang.String,java.lang.String)">multiAlign(ArrayList&lt;Sequence&gt;, String, String)</a></span> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermConservationWithRiboProfiling.html" title="class in bacnet.scripts.listeriomics.nterm">NTermConservationWithRiboProfiling</a></dt>
<dd>
<div class="block">Run a Blast for each sRNA on a list of Genomes given in tempDatabase</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html#multipleFrameTranslation(org.biojava3.core.sequence.template.Sequence,org.biojava3.core.sequence.transcription.Frame...)">multipleFrameTranslation(Sequence&lt;NucleotideCompound&gt;, Frame...)</a></span> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></dt>
<dd>
<div class="block">A way of translating DNA in a number of frames</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence"><span class="typeNameLink">MultipleSequenceAlignment</span></a>&lt;<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">S</a> extends <a href="../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">C</a>&gt;,&#8203;<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">Implements a minimal data structure for reading and writing a sequence alignment.</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html#%3Cinit%3E()">MultipleSequenceAlignment()</a></span> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html" title="class in bacnet.scripts.blast"><span class="typeNameLink">MultiSequenceBlastProtein</span></a> - Class in <a href="../bacnet/scripts/blast/package-summary.html">bacnet.scripts.blast</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html#%3Cinit%3E()">MultiSequenceBlastProtein()</a></span> - Constructor for class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html" title="class in bacnet.scripts.blast">MultiSequenceBlastProtein</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast"><span class="typeNameLink">MultiSequenceTBlastNProtein</span></a> - Class in <a href="../bacnet/scripts/blast/package-summary.html">bacnet.scripts.blast</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#%3Cinit%3E()">MultiSequenceTBlastNProtein()</a></span> - Constructor for class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#multiSpeciesAlign()">multiSpeciesAlign()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Run a Blast for each sRNA on a list of Genomes given in tempDatabase</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#multiSpeciesAlign(java.util.ArrayList)">multiSpeciesAlign(ArrayList&lt;Srna&gt;)</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Run a Blast for each sRNA on a list of Genomes given in tempDatabase</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html#multiSpeciesAlign10403S()">multiSpeciesAlign10403S()</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></dt>
<dd>
<div class="block">Run a Blast for each sRNA (found in Oliver et al. in 10403S genome) on a list of Genomes given in
 tempDatabase</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html#multiSpeciesAlignBlastN(java.util.ArrayList)">multiSpeciesAlignBlastN(ArrayList&lt;Gene&gt;)</a></span> - Static method in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html" title="class in bacnet.scripts.listeriomics.srna">SmallPeptidesSearch</a></dt>
<dd>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span></div>
&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/Blast.html#multiSpeciesAlignNucleotide(java.util.ArrayList,boolean)">multiSpeciesAlignNucleotide(ArrayList&lt;Sequence&gt;, boolean)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>
<div class="block">Run a Blast for each sRNA on a list of Genomes given in tempDatabase</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/Blast.html#multiSpeciesAlignProtein(java.util.ArrayList,java.util.ArrayList,java.lang.String,double)">multiSpeciesAlignProtein(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, String, double)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>
<div class="block">Run a BlastP given a a list of sequence name, amino acid sequences, and a database name</div>
</dd>
<dt><span class="memberNameLink"><a href="../bacnet/scripts/blast/Blast.html#multiSpeciesAlignTblastN(java.util.ArrayList,java.util.ArrayList,java.lang.String,double)">multiSpeciesAlignTblastN(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, String, double)</a></span> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>
<div class="block">Run TblastN on multy amino acid sequence against nucleotide database that will be translated in
 the 6 frame</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/Feature.html#mUserMap">mUserMap</a></span> - Variable in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="../bacnet/datamodel/expdesign/BioCondition.html#mutant">mutant</a></span> - Variable in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">If it is a mutant, the locus is indicated (example: prfA, lmo0200 or rliA), leave empty otherwise</div>
</dd>
<dt><span class="memberNameLink"><a href="../org/biojava3/genome/parsers/gff/LocIterator.html#mWindowSize">mWindowSize</a></span> - Variable in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">K</a>&nbsp;<a href="index-12.html">L</a>&nbsp;<a href="index-13.html">M</a>&nbsp;<a href="index-14.html">N</a>&nbsp;<a href="index-15.html">O</a>&nbsp;<a href="index-16.html">P</a>&nbsp;<a href="index-17.html">Q</a>&nbsp;<a href="index-18.html">R</a>&nbsp;<a href="index-19.html">S</a>&nbsp;<a href="index-20.html">T</a>&nbsp;<a href="index-21.html">U</a>&nbsp;<a href="index-22.html">V</a>&nbsp;<a href="index-23.html">W</a>&nbsp;<a href="index-24.html">X</a>&nbsp;<a href="index-25.html">Y</a>&nbsp;<a href="index-26.html">Z</a>&nbsp;<a href="index-27.html">_</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes</a>&nbsp;<a href="../allpackages-index.html">All&nbsp;Packages</a></div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
