<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Fri Dec 10 09:15:34 CET 2021 -->
<title>O-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-10">
<meta name="description" content="index: O">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:O">O</h2>
<dl class="index">
<dt><a href="../org/biojava3/core/sequence/location/FuzzyPoint.html#offset(int)" class="member-name-link">offset(int)</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/SimplePoint.html#offset(int)" class="member-name-link">offset(int)</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Point.html#offset(int)" class="member-name-link">offset(int)</a> - Method in interface org.biojava3.core.sequence.location.template.<a href="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a></dt>
<dd>
<div class="block">Returns a new point offset by the given distance</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html#okPressed()" class="member-name-link">okPressed()</a> - Method in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/SelectGenomeElementDialog.html" title="class in bacnet.expressionAtlas.core">SelectGenomeElementDialog</a></dt>
<dd>
<div class="block">When Ok is pressed, update the different genome elements sleected</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html#okPressed()" class="member-name-link">okPressed()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddProteomicsDataDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html#okPressed()" class="member-name-link">okPressed()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddTranscriptomicsDataDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/dialog/LegendDialog.html#okPressed()" class="member-name-link">okPressed()</a> - Method in class bacnet.genomeBrowser.dialog.<a href="../bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/FilterDialog.html#okPressed()" class="member-name-link">okPressed()</a> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/FilterDialog.html" title="class in bacnet.table.gui">FilterDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/FilterExpressionMatrixDialog.html#okPressed()" class="member-name-link">okPressed()</a> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/FilterExpressionMatrixDialog.html" title="class in bacnet.table.gui">FilterExpressionMatrixDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/HideExpressionMatrixDialog.html#okPressed()" class="member-name-link">okPressed()</a> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/HideExpressionMatrixDialog.html" title="class in bacnet.table.gui">HideExpressionMatrixDialog</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/LocusTag.html#OldLocusTagArray" class="member-name-link">OldLocusTagArray</a> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/LocusTag.html" title="class in bacnet.datamodel.annotation">LocusTag</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html" class="type-name-link" title="class in bacnet.datamodel.dataset">OmicsData</a> - Class in <a href="../bacnet/datamodel/dataset/package-summary.html">bacnet.datamodel.dataset</a></dt>
<dd>
<div class="block">Abstract Class describing data related to omics data <br></div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#%3Cinit%3E()" class="member-name-link">OmicsData()</a> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html" class="type-name-link" title="enum in bacnet.datamodel.dataset">OmicsData.ColNames</a> - Enum in <a href="../bacnet/datamodel/dataset/package-summary.html">bacnet.datamodel.dataset</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html" class="type-name-link" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a> - Enum in <a href="../bacnet/datamodel/dataset/package-summary.html">bacnet.datamodel.dataset</a></dt>
<dd>
<div class="block">Enum referencing the different type of OmicsData possible:
 GeneExpr
 Tiling
 RNASeq
 TSS (Transcription Initiation Site)
 NTerm (Translation Initiation Site)
 Proteome (Matrix of protein found)
 MassSpec (List of peptides found by mass spectrometry)
 ExpressionMatrix (General matrix format)
 unknown</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/FeatureList.html#omitOverlapping(java.lang.String,org.biojava3.genome.parsers.gff.Location,boolean)" class="member-name-link">omitOverlapping(String, Location, boolean)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></dt>
<dd>
<div class="block">Create a list of all features that do not overlap the specified location on the specified
 sequence.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/Frame.html#ONE" class="member-name-link">ONE</a> - Enum constant in enum org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.OneOfLocation.html#%3Cinit%3E(java.util.List)" class="member-name-link">OneOfLocation(List&lt;Location&gt;)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.OneOfLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OneOfLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.OneOfLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Location...)" class="member-name-link">OneOfLocation(Location...)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.OneOfLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OneOfLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewBacnet.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewBacnet.html" title="class in bacnet.e4.rap">InitViewBacnet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeria.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeria.html" title="class in bacnet.e4.rap">InitViewListeria</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeriaSample.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">InitViewListeriaSample</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewYersinia.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewYersinia.html" title="class in bacnet.e4.rap">InitViewYersinia</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#onFocus()" class="member-name-link">onFocus()</a> - Method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#open8081GeneView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">open8081GeneView(EPartService)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Display GeneView <br>
 It will display 8081</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#open91001GeneView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">open91001GeneView(EPartService)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Display GeneView <br>
 It will display 91001</div>
</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#openAnnotationView(org.eclipse.e4.ui.workbench.modeling.EPartService,bacnet.datamodel.sequence.Genome)" class="member-name-link">openAnnotationView(EPartService, Genome)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>
<div class="block">Display a given gene</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/core/OpenExpressionMatrixAndComparisons.html" class="type-name-link" title="class in bacnet.expressionAtlas.core">OpenExpressionMatrixAndComparisons</a> - Class in <a href="../bacnet/expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a></dt>
<dd>
<div class="block">Run thread for opening BioCondition, Comparisons, detect Genome and open corresponding LogFC
 ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/core/OpenExpressionMatrixAndComparisons.html#%3Cinit%3E(java.util.ArrayList,java.util.HashMap,boolean)" class="member-name-link">OpenExpressionMatrixAndComparisons(ArrayList&lt;String&gt;, HashMap&lt;String, ArrayList&lt;String&gt;&gt;, boolean)</a> - Constructor for class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/OpenExpressionMatrixAndComparisons.html" title="class in bacnet.expressionAtlas.core">OpenExpressionMatrixAndComparisons</a></dt>
<dd>
<div class="block">Open BioCondition, Comparisons, detect Genome and open corresponding LogFC ExpressionMatrix</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#openFile(java.io.File)" class="member-name-link">openFile(File)</a> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">For a filename this code will check the extension of the file for a .gz extension.</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#openGeneView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">openGeneView(EPartService)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Display GeneView <br>
 It will display Genome.getDefautGenome(); CO92</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.OpenGenomesThread.html#%3Cinit%3E(java.util.ArrayList)" class="member-name-link">OpenGenomesThread(ArrayList&lt;String&gt;)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.OpenGenomesThread.html" title="class in bacnet.datamodel.sequence">Genome.OpenGenomesThread</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/InternalBrowser.html#openHTML(java.lang.String,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">openHTML(String, String, EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>
<div class="block">Open an HTML file</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#openIP32953GeneView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">openIP32953GeneView(EPartService)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Display GeneView <br>
 It will display IP32953</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#openKIMGeneView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">openKIMGeneView(EPartService)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Display GeneView <br>
 It will display KIM</div>
</dd>
<dt><a href="../bacnet/views/InternalBrowser.html#openList(java.util.ArrayList,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">openList(ArrayList&lt;String&gt;, String, EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>
<div class="block">Open an HTML version of the String ArrayList</div>
</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeriaSample.html#openPasswordDialog()" class="member-name-link">openPasswordDialog()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">InitViewListeriaSample</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/AppSpecificMethods.html#openPasswordDialog(org.eclipse.swt.widgets.Shell)" class="member-name-link">openPasswordDialog(Shell)</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/AppSpecificMethods.html" title="class in bacnet.e4.rap">AppSpecificMethods</a></dt>
<dd>
<div class="block">Push states to change url</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#openPB1GeneView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">openPB1GeneView(EPartService)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#openPestoidesGeneView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">openPestoidesGeneView(EPartService)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#openQMA0440GeneView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">openQMA0440GeneView(EPartService)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Display GeneView <br>
 It will display QMA0440</div>
</dd>
<dt><a href="../bacnet/views/InternalBrowser.html#openTable(java.lang.String%5B%5D%5B%5D,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">openTable(String[][], String, EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>
<div class="block">Open an HTML version of the String array</div>
</dd>
<dt><a href="../org/biojava3/core/util/PrettyXMLWriter.html#openTag(java.lang.String)" class="member-name-link">openTag(String)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLWriter.html#openTag(java.lang.String)" class="member-name-link">openTag(String)</a> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Open a new unqualified XML tag.</div>
</dd>
<dt><a href="../org/biojava3/core/util/PrettyXMLWriter.html#openTag(java.lang.String,java.lang.String)" class="member-name-link">openTag(String, String)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLWriter.html#openTag(java.lang.String,java.lang.String)" class="member-name-link">openTag(String, String)</a> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Open a new namespace-qualified XML tag.</div>
</dd>
<dt><a href="../bacnet/views/InternalBrowser.html#openText(java.lang.String,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">openText(String, String, EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>
<div class="block">Open an HTML version of the String</div>
</dd>
<dt><a href="../bacnet/views/InternalBrowser.html#openURL(java.lang.String,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">openURL(String, String, EPartService)</a> - Static method in class bacnet.views.<a href="../bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></dt>
<dd>
<div class="block">Open an HTML file</div>
</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#openURLInExternalBrowser(java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">openURLInExternalBrowser(String, EPartService)</a> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>
<div class="block">Show the help page in an external browser</div>
</dd>
<dt><a href="../bacnet/table/TableSWTView.html#openView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.lang.String)" class="member-name-link">openView(EPartService, String)</a> - Static method in class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>
<div class="block">Open a HeatMapTranscriptomicsView using EPartService and an id</div>
</dd>
<dt><a href="../bacnet/swt/ResourceManager.html#openView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.lang.String,java.lang.String)" class="member-name-link">openView(EPartService, String, String)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Open a HeatMapTranscriptomicsView using EPartService and an id !!!!!!</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#openWAGeneView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">openWAGeneView(EPartService)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#openY11GeneView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">openY11GeneView(EPartService)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Display GeneView <br>
 It will display Y11</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#openYPIIIGeneView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">openYPIIIGeneView(EPartService)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Display GeneView <br>
 It will display YPIII</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Operon.html" class="type-name-link" title="class in bacnet.datamodel.sequence">Operon</a> - Class in <a href="../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.SeqType.html#Operon" class="member-name-link">Operon</a> - Enum constant in enum bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Operon.html#%3Cinit%3E()" class="member-name-link">Operon()</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Operon.html#%3Cinit%3E(java.lang.String,int,int,char)" class="member-name-link">Operon(String, int, int, char)</a> - Constructor for class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#operons" class="member-name-link">operons</a> - Variable in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#opposite()" class="member-name-link">opposite()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Return location that is in same position on opposite strand.</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#ORANGE" class="member-name-link">ORANGE</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html#%3Cinit%3E(int,int,org.biojava3.core.sequence.Strand,java.util.List)" class="member-name-link">OrderLocation(int, int, Strand, List&lt;Location&gt;)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OrderLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html#%3Cinit%3E(int,int,org.biojava3.core.sequence.Strand,org.biojava3.core.sequence.location.template.Location...)" class="member-name-link">OrderLocation(int, int, Strand, Location...)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OrderLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,java.util.List)" class="member-name-link">OrderLocation(Point, Point, Strand, boolean, List&lt;Location&gt;)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OrderLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,boolean,org.biojava3.core.sequence.location.template.Location...)" class="member-name-link">OrderLocation(Point, Point, Strand, boolean, Location...)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OrderLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,java.util.List)" class="member-name-link">OrderLocation(Point, Point, Strand, List&lt;Location&gt;)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OrderLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html#%3Cinit%3E(org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.location.template.Point,org.biojava3.core.sequence.Strand,org.biojava3.core.sequence.location.template.Location...)" class="member-name-link">OrderLocation(Point, Point, Strand, Location...)</a> - Constructor for class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OrderLocation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/exceptions/package-summary.html">org.biojava3.core.exceptions</a> - package org.biojava3.core.exceptions</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a> - package org.biojava3.core.sequence</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</a> - package org.biojava3.core.sequence.compound</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/edits/package-summary.html">org.biojava3.core.sequence.edits</a> - package org.biojava3.core.sequence.edits</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/package-summary.html">org.biojava3.core.sequence.features</a> - package org.biojava3.core.sequence.features</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a> - package org.biojava3.core.sequence.io</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/template/package-summary.html">org.biojava3.core.sequence.io.template</a> - package org.biojava3.core.sequence.io.template</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/package-summary.html">org.biojava3.core.sequence.io.util</a> - package org.biojava3.core.sequence.io.util</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/loader/package-summary.html">org.biojava3.core.sequence.loader</a> - package org.biojava3.core.sequence.loader</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/package-summary.html">org.biojava3.core.sequence.location</a> - package org.biojava3.core.sequence.location</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/package-summary.html">org.biojava3.core.sequence.location.template</a> - package org.biojava3.core.sequence.location.template</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/storage/package-summary.html">org.biojava3.core.sequence.storage</a> - package org.biojava3.core.sequence.storage</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a> - package org.biojava3.core.sequence.template</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> - package org.biojava3.core.sequence.transcription</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/views/package-summary.html">org.biojava3.core.sequence.views</a> - package org.biojava3.core.sequence.views</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a> - package org.biojava3.core.util</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/package-summary.html">org.biojava3.genome</a> - package org.biojava3.genome</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/geneid/package-summary.html">org.biojava3.genome.parsers.geneid</a> - package org.biojava3.genome.parsers.geneid</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a> - package org.biojava3.genome.parsers.gff</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/query/package-summary.html">org.biojava3.genome.query</a> - package org.biojava3.genome.query</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/uniprot/package-summary.html">org.biojava3.genome.uniprot</a> - package org.biojava3.genome.uniprot</dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/util/package-summary.html">org.biojava3.genome.util</a> - package org.biojava3.genome.util</dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/PhylogenomicsCreation.html#organizePhyloTable(java.lang.String)" class="member-name-link">organizePhyloTable(String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/PhylogenomicsCreation.html" title="class in bacnet.scripts.database">PhylogenomicsCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/SDProfile.html#organizeResults(java.lang.String)" class="member-name-link">organizeResults(String)</a> - Static method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>
<div class="block">Go through the tables created, reorganize the different indexes to order Base pairs.<br>
 For random values, verify that the SD calculation is good.</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.TypeMapper.html#OTHER" class="member-name-link">OTHER</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.TypeMapper.html" title="enum in bacnet.table.core">ColorMapper.TypeMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.Type.html#OTHER" class="member-name-link">OTHER</a> - Enum constant in enum bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.Type.html" title="enum in bacnet.table.core">ColorMapperRCP.Type</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#outerProduct(double%5B%5D%5B%5D)" class="member-name-link">outerProduct(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Calcul of the Outer Product Matrix</div>
</dd>
<dt><a href="../org/biojava3/genome/GeneFeatureHelper.html#outputFastaSequenceLengthGFF3(java.io.File,java.io.File)" class="member-name-link">outputFastaSequenceLengthGFF3(File, File)</a> - Static method in class org.biojava3.genome.<a href="../org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></dt>
<dd>
<div class="block">Output a gff3 feature file that will give the length of each scaffold/chromosome in the fasta
 file.</div>
</dd>
<dt><a href="../org/biojava3/genome/query/OutputHitsGFF.html" class="type-name-link" title="class in org.biojava3.genome.query">OutputHitsGFF</a> - Class in <a href="../org/biojava3/genome/query/package-summary.html">org.biojava3.genome.query</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/query/OutputHitsGFF.html#%3Cinit%3E()" class="member-name-link">OutputHitsGFF()</a> - Constructor for class org.biojava3.genome.query.<a href="../org/biojava3/genome/query/OutputHitsGFF.html" title="class in org.biojava3.genome.query">OutputHitsGFF</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastOutput.html#outputName" class="member-name-link">outputName</a> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.html" title="class in bacnet.scripts.blast">BlastOutput</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLHelper.html#outputToStream(org.w3c.dom.Document,java.io.OutputStream)" class="member-name-link">outputToStream(Document, OutputStream)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLHelper.html#outputToStream(org.w3c.dom.Element,java.io.OutputStream)" class="member-name-link">outputToStream(Element, OutputStream)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#overlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)" class="member-name-link">overlap(Sequence, Sequence)</a> - Static method in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>
<div class="block">Calculate the overlapping size of two genes <br>
 First, find the gene upstream to the other<br>
 Then, as we got gene1.begin &lt; gene2.begin, overlap = gene1.end - gene2.begin<br>
 overlap can be negative, it will mean that we have no overlap (gene1.end &lt; gene2.begin) <br>
 or overlap can be higher than gene2.length, in that case overlap = gene2.length</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#overlap(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence)" class="member-name-link">overlap(DNASequence, DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Calculate the overlapping size of two genes <br>
 First, find the gene upstream to the other<br>
 Then, as we got gene1.begin &lt; gene2.begin, overlap = gene1.end - gene2.begin<br>
 overlap can be negative, it will mean that we have no overlap (gene1.end &lt; gene2.begin) <br>
 or overlap can be higher than gene2.length, in that case overlap = gene2.length</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#overlapEqual(int%5B%5D,double)" class="member-name-link">overlapEqual(int[], double)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>
<div class="block">Return true if overlap[0] &lt; value</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#overlapInf(int%5B%5D,double)" class="member-name-link">overlapInf(int[], double)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>
<div class="block">Return true if overlap[0] &lt; value</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/template/SequenceMixin.html#overlappingKmers(org.biojava3.core.sequence.template.Sequence,int)" class="member-name-link">overlappingKmers(Sequence&lt;C&gt;, int)</a> - Static method in class org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></dt>
<dd>
<div class="block">Used to generate overlapping k-mers such i.e.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#overlaps(org.biojava3.genome.parsers.gff.Location)" class="member-name-link">overlaps(Location)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Check if this location and other location overlap.</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#overlapSup(int%5B%5D,double)" class="member-name-link">overlapSup(int[], double)</a> - Static method in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>
<div class="block">Return true if overlap[0] &lt; value</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.DisplayType.html#OVERLAY" class="member-name-link">OVERLAY</a> - Enum constant in enum bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.DisplayType.html" title="enum in bacnet.genomeBrowser.core">Track.DisplayType</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
