<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Fri Dec 10 09:15:34 CET 2021 -->
<title>D-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-10">
<meta name="description" content="index: D">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:D">D</h2>
<dl class="index">
<dt><a href="../bacnet/utils/BasicColor.html#DARK_ASRNA" class="member-name-link">DARK_ASRNA</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Dark color used for ASrna</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#DARK_CISREG" class="member-name-link">DARK_CISREG</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Dark color used for NCrna</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#DARK_NCRNA" class="member-name-link">DARK_NCRNA</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Dark color used for NCrna</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#DARK_ONE" class="member-name-link">DARK_ONE</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#DARK_OPERON" class="member-name-link">DARK_OPERON</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Dark color used for ASrna</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#DARK_SRNA" class="member-name-link">DARK_SRNA</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>
<div class="block">Dark color used for Srna</div>
</dd>
<dt><a href="../bacnet/utils/BasicColor.html#DARK_THREE" class="member-name-link">DARK_THREE</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#DARK_TWO" class="member-name-link">DARK_TWO</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/Track.DisplayType.html#DATA" class="member-name-link">DATA</a> - Enum constant in enum bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/Track.DisplayType.html" title="enum in bacnet.genomeBrowser.core">Track.DisplayType</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#DATA_CURATED_PATH" class="member-name-link">DATA_CURATED_PATH</a> - Static variable in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html#DATA_CURATED_PATH" class="member-name-link">DATA_CURATED_PATH</a> - Static variable in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html#DATA_CURATED_PATH" class="member-name-link">DATA_CURATED_PATH</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics">ArrayExpressListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#DATA_LIMIT" class="member-name-link">DATA_LIMIT</a> - Static variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#DATA_LIMIT" class="member-name-link">DATA_LIMIT</a> - Static variable in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html#DATA_PATH" class="member-name-link">DATA_PATH</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/SummaryListeriomics.html" title="class in bacnet.scripts.listeriomics">SummaryListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpress.html#DATA_TABLE" class="member-name-link">DATA_TABLE</a> - Static variable in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/ProteomicsCreation.html#DATA_TABLE" class="member-name-link">DATA_TABLE</a> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html#DATA_TABLE" class="member-name-link">DATA_TABLE</a> - Static variable in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html#DATA_TABLE" class="member-name-link">DATA_TABLE</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics">ArrayExpressListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html" class="type-name-link" title="class in bacnet">Database</a> - Class in <a href="../bacnet/package-summary.html">bacnet</a></dt>
<dd>
<div class="block">A ModelProvider specifically design for the Website Database<br>
 Contains persistent variable, used through all the website.</div>
</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html#DATABASE" class="member-name-link">DATABASE</a> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#DATABASE" class="member-name-link">DATABASE</a> - Static variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.InitDatabaseThread.html" class="type-name-link" title="class in bacnet">Database.InitDatabaseThread</a> - Class in <a href="../bacnet/package-summary.html">bacnet</a></dt>
<dd>
<div class="block">Class for loading databases in a thread</div>
</dd>
<dt><a href="../bacnet/scripts/database/DatabaseCreation.html" class="type-name-link" title="class in bacnet.scripts.database">DatabaseCreation</a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
</dd>
<dt><a href="../bacnet/scripts/database/DatabaseCreation.html#%3Cinit%3E()" class="member-name-link">DatabaseCreation()</a> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/DatabaseCreation.html" title="class in bacnet.scripts.database">DatabaseCreation</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/DatabaseReferenceInterface.html" class="type-name-link" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</a> - Interface in <a href="../org/biojava3/core/sequence/features/package-summary.html">org.biojava3.core.sequence.features</a></dt>
<dd>
<div class="block">If a SequenceProxyReader implements this interface then that external source has a list of cross
 reference id(s)</div>
</dd>
<dt><a href="../bacnet/scripts/database/DataFolder.html" class="type-name-link" title="class in bacnet.scripts.database">DataFolder</a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DataFolder.html#%3Cinit%3E()" class="member-name-link">DataFolder()</a> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataFolder.html" title="class in bacnet.scripts.database">DataFolder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DataSource.html" class="type-name-link" title="enum in org.biojava3.core.sequence">DataSource</a> - Enum in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">GenBank gi|gi-number|gb|accession|locus ENA Data Library gi|gi-number|emb|accession|locus DDBJ,
 DNA Database of Japan gi|gi-number|dbj|accession|locus NBRF PIR pir||entry Protein Research
 Foundation prf||name SWISS-PROT UNIPROT sp|accession|name Brookhaven Protein Data Bank (1)
 pdb|entry|chain Brookhaven Protein Data Bank (2) entry:chain|PDBID|CHAIN|SEQUENCE Patents
 pat|country|number GenInfo Backbone Id bbs|number General database identifier
 gnl|database|identifier NCBI Reference Sequence ref|accession|locus Local Sequence identifier
 lcl|identifier</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html" class="type-name-link" title="class in bacnet.genomeBrowser.core">DataTrack</a> - Class in <a href="../bacnet/genomeBrowser/core/package-summary.html">bacnet.genomeBrowser.core</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/core/DataTrack.html#%3Cinit%3E(bacnet.genomeBrowser.core.Track)" class="member-name-link">DataTrack(Track)</a> - Constructor for class bacnet.genomeBrowser.core.<a href="../bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DataValidation.html" class="type-name-link" title="class in bacnet.scripts.database">DataValidation</a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DataValidation.html#%3Cinit%3E()" class="member-name-link">DataValidation()</a> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" class="type-name-link" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a> - Class in <a href="../org/biojava3/core/sequence/features/package-summary.html">org.biojava3.core.sequence.features</a></dt>
<dd>
<div class="block">If you have a uniprot ID then it is possible to get a collection of other id(s) that the protein
 is known by.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html#%3Cinit%3E(java.lang.String,java.lang.String)" class="member-name-link">DBReferenceInfo(String, String)</a> - Constructor for class org.biojava3.core.sequence.features.<a href="../org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></dt>
<dd>
<div class="block">The source database and id</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/DataSource.html#DDBJ" class="member-name-link">DDBJ</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/SoftHashMap.html#DEBUG" class="member-name-link">DEBUG</a> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/PrettyXMLWriter.html#declareNamespace(java.lang.String,java.lang.String)" class="member-name-link">declareNamespace(String, String)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLWriter.html#declareNamespace(java.lang.String,java.lang.String)" class="member-name-link">declareNamespace(String, String)</a> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Hints that a namespace is going to be used in a sub-tree.</div>
</dd>
<dt><a href="../bacnet/utils/XMLUtils.html#decodeFromFile(java.lang.String)" class="member-name-link">decodeFromFile(String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/XMLUtils.html" title="class in bacnet.utils">XMLUtils</a></dt>
<dd>
<div class="block">Deserialisation d'un objet depuis un fichier</div>
</dd>
<dt><a href="../bacnet/swt/ResourceManager.html#decorateImage(org.eclipse.swt.graphics.Image,org.eclipse.swt.graphics.Image)" class="member-name-link">decorateImage(Image, Image)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> composed of a base image decorated by another image.</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#decorateImage(org.eclipse.swt.graphics.Image,org.eclipse.swt.graphics.Image)" class="member-name-link">decorateImage(Image, Image)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> composed of a base image decorated by another image.</div>
</dd>
<dt><a href="../bacnet/swt/ResourceManager.html#decorateImage(org.eclipse.swt.graphics.Image,org.eclipse.swt.graphics.Image,int)" class="member-name-link">decorateImage(Image, Image, int)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> composed of a base image decorated by another image.</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#decorateImage(org.eclipse.swt.graphics.Image,org.eclipse.swt.graphics.Image,int)" class="member-name-link">decorateImage(Image, Image, int)</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Returns an <code>Image</code> composed of a base image decorated by another image.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#decorateRna(boolean)" class="member-name-link">decorateRna(boolean)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>
<div class="block">Performs an optimisation where RNASequences are not translated into their own objects but are
 views onto the base DNA sequence.</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermFilter.html#DEFAULT_FILTER_PATH" class="member-name-link">DEFAULT_FILTER_PATH</a> - Static variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Sequence.html#DEFAULT_LENGTH" class="member-name-link">DEFAULT_LENGTH</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/SoftHashMap.html#DEFAULT_LIMIT" class="member-name-link">DEFAULT_LIMIT</a> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#DEFAULT_LOGFC_CUTOFF" class="member-name-link">DEFAULT_LOGFC_CUTOFF</a> - Static variable in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#DEFAULT_LOGFC_PROTEOMIC_CUTOFF" class="member-name-link">DEFAULT_LOGFC_PROTEOMIC_CUTOFF</a> - Static variable in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView.html#DEFAULT_MODEL" class="member-name-link">DEFAULT_MODEL</a> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#DEFAULT_PVAL_CUTOFF" class="member-name-link">DEFAULT_PVAL_CUTOFF</a> - Static variable in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#DEFAULT_PVAL_PROTEOMIC_CUTOFF" class="member-name-link">DEFAULT_PVAL_PROTEOMIC_CUTOFF</a> - Static variable in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.Delete.html#%3Cinit%3E(int)" class="member-name-link">Delete(int)</a> - Constructor for class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.Delete.html" title="class in org.biojava3.core.sequence.edits">Edit.Delete</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/edits/Edit.Delete.html#%3Cinit%3E(int,int)" class="member-name-link">Delete(int, int)</a> - Constructor for class org.biojava3.core.sequence.edits.<a href="../org/biojava3/core/sequence/edits/Edit.Delete.html" title="class in org.biojava3.core.sequence.edits">Edit.Delete</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#deleteBioCond(bacnet.datamodel.expdesign.BioCondition)" class="member-name-link">deleteBioCond(BioCondition)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Delete using the bioCondName as a key</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/Experiment.html#deleteBioCond(java.lang.String)" class="member-name-link">deleteBioCond(String)</a> - Method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></dt>
<dd>
<div class="block">Delete a BioCondition using a specific key</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#deleteColumn(double%5B%5D%5B%5D,int)" class="member-name-link">deleteColumn(double[][], int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ExpressionMatrix.html#deleteColumn(java.lang.String)" class="member-name-link">deleteColumn(String)</a> - Method in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#deleteColumn(java.lang.String%5B%5D%5B%5D,int)" class="member-name-link">deleteColumn(String[][], int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#deleteMissingValue(double%5B%5D)" class="member-name-link">deleteMissingValue(double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#deleteRow(double%5B%5D%5B%5D,int)" class="member-name-link">deleteRow(double[][], int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#deleteRow(java.lang.String%5B%5D%5B%5D,int)" class="member-name-link">deleteRow(String[][], int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/FlatFileCache.html#destroy()" class="member-name-link">destroy()</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/FlatFileCache.html" title="class in org.biojava3.core.util">FlatFileCache</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#detectable()" class="member-name-link">detectable()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Create a table where we put all lmos detectable by the mass spectrometer</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/LocationHelper.html#detectCicular(java.util.List)" class="member-name-link">detectCicular(List&lt;Location&gt;)</a> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Loops through the given list of locations and returns true if it looks like they represent a
 circular location.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/LocationHelper.html#detectEnd(java.util.List,boolean)" class="member-name-link">detectEnd(List&lt;Location&gt;, boolean)</a> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">This will attempt to find what the last point is and returns that position.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/LocationHelper.html#detectStart(java.util.List)" class="member-name-link">detectStart(List&lt;Location&gt;)</a> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Assumes that the first element is the start &amp; clones it</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/LocationHelper.html#detectStrand(java.util.List)" class="member-name-link">detectStrand(List&lt;Location&gt;)</a> - Static method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></dt>
<dd>
<div class="block">Loops through the given list of locations and returns the consensus Strand class.</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#deviation(double%5B%5D)" class="member-name-link">deviation(double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">Calculate deviation of a vector sigma(X)=sqrt(V(X))</div>
</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#deviation(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">deviation(ExpressionMatrix)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#deviation(bacnet.datamodel.dataset.ExpressionMatrix,int)" class="member-name-link">deviation(ExpressionMatrix, int)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Calculate statistical deviation of a specific column</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#diagonalTimes(double%5B%5D%5B%5D,double%5B%5D)" class="member-name-link">diagonalTimes(double[][], double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#diagonalTimes(double%5B%5D,double%5B%5D%5B%5D)" class="member-name-link">diagonalTimes(double[], double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Multiply a vector(=in fact a diagonal matrice) times a array X'=MX or XM with a diagonal = vector</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#diffBeginPosition(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence)" class="member-name-link">diffBeginPosition(DNASequence, DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Calculate difference in begin position</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#diffCenter(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence)" class="member-name-link">diffCenter(DNASequence, DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Calculate the distance between the two center of gene1 and gene2</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#diffEndPosition(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence)" class="member-name-link">diffEndPosition(DNASequence, DNASequence)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>
<div class="block">Calculate difference in end position</div>
</dd>
<dt><a href="../bacnet/scripts/blast/BlastResult.html#diffSize" class="member-name-link">diffSize</a> - Variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#display" class="member-name-link">display</a> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#display" class="member-name-link">display</a> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#display" class="member-name-link">display</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/AnnotationView.html#displayAnnotationView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)" class="member-name-link">displayAnnotationView(EPartService, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayBHI37View(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayBHI37View(EPartService)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Load the minimal number of data for the simplest proteomics view</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#displayBioConditions(java.lang.String,java.util.ArrayList,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayBioConditions(String, ArrayList&lt;String&gt;, EPartService)</a> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>
<div class="block">Run a <code>HeatMapView</code> displaying a specific bioCond</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#displayBioConditionsExpressionAtlas(java.util.HashMap,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayBioConditionsExpressionAtlas(HashMap&lt;String, ArrayList&lt;String&gt;&gt;, EPartService)</a> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>
<div class="block">Run a <code>HeatMapProteomicsView</code> displaying a specific comparison of bioCond</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#displayBioConditionsExpressionAtlas(java.util.HashMap,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayBioConditionsExpressionAtlas(HashMap&lt;String, ArrayList&lt;String&gt;&gt;, EPartService)</a> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>
<div class="block">Run a <code>HeatMapTranscriptomicsView</code> displaying a specific comparison of bioCond</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayBioConditionsFromText(org.eclipse.e4.ui.workbench.modeling.EPartService,java.lang.String)" class="member-name-link">displayBioConditionsFromText(EPartService, String)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Open a GenomeView given a genome and text file from .gview file</div>
</dd>
<dt><a href="../bacnet/views/CoExprNetworkView.html#displayCoExpNetworkView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)" class="member-name-link">displayCoExpNetworkView(EPartService, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.views.<a href="../bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#displayComparisonsAndElement(java.lang.String,java.util.ArrayList,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayComparisonsAndElement(String, ArrayList&lt;String&gt;, String, EPartService)</a> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#displayComparisonsAndElement(java.lang.String,java.util.ArrayList,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayComparisonsAndElement(String, ArrayList&lt;String&gt;, String, EPartService)</a> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayCRISPROmics(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayCRISPROmics(EPartService)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Load the minimal number of data for Amastigote versus Promastigote view</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayCRISPROmics(org.eclipse.e4.ui.workbench.modeling.EPartService,int)" class="member-name-link">displayCRISPROmics(EPartService, int)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Open a GenomeView given a genome and list of biological conditions</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#displayedAttributes" class="member-name-link">displayedAttributes</a> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#displayGene(bacnet.datamodel.sequence.Gene,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayGene(Gene, EPartService)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Display a given gene</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#displayGene(org.eclipse.swt.graphics.GC,bacnet.datamodel.sequence.Sequence,org.eclipse.swt.graphics.Rectangle,bacnet.genomeBrowser.core.Track,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color,org.eclipse.swt.graphics.Color,boolean)" class="member-name-link">displayGene(GC, Sequence, Rectangle, Track, Color, Color, Color, Color, boolean)</a> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Display Gene annotation = square and arrow</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#displayGeneView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)" class="member-name-link">displayGeneView(EPartService, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayGenomeElementAndBioConditions(org.eclipse.e4.ui.workbench.modeling.EPartService,java.lang.String,java.util.ArrayList,java.lang.String)" class="member-name-link">displayGenomeElementAndBioConditions(EPartService, String, ArrayList&lt;String&gt;, String)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Open a GenomeView given a genome and list of biological conditions</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayGenomeTranscriptomesView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)" class="member-name-link">displayGenomeTranscriptomesView(EPartService, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#displayGenomeView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)" class="member-name-link">displayGenomeView(EPartService, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>
<div class="block">Display a search in Genomic View</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html#displayHeatMapProteomicsView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)" class="member-name-link">displayHeatMapProteomicsView(EPartService, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html#displayHeatMapProteomicsView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)" class="member-name-link">displayHeatMapProteomicsView(EPartService, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html#displayHeatMapTranscriptomicsView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)" class="member-name-link">displayHeatMapTranscriptomicsView(EPartService, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayIntracellularMacrophagesView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayIntracellularMacrophagesView(EPartService)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayLeishmaniaAmavsPro(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayLeishmaniaAmavsPro(EPartService)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Load the minimal number of data for Amastigote versus Promastigote view</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayLeishmaniaLeisheild(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayLeishmaniaLeisheild(EPartService)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Load the minimal number of data for Amastigote versus Promastigote view</div>
</dd>
<dt><a href="../bacnet/table/TableSWTView.html#displayMatrix(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome,java.lang.String,boolean,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayMatrix(ExpressionMatrix, Genome, String, boolean, String, EPartService)</a> - Static method in class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/TableSWTView.html#displayMatrix(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.sequence.Genome,java.lang.String,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayMatrix(ExpressionMatrix, Genome, String, EPartService)</a> - Static method in class bacnet.table.<a href="../bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></dt>
<dd>
<div class="block">Static function which open an HeatMapView and display corresponding ExpressionMatrix</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#displayMatrix(java.lang.String,double%5B%5D%5B%5D)" class="member-name-link">displayMatrix(String, double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">Display a Matrix</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#displayMatrix(java.lang.String,java.lang.Object%5B%5D%5B%5D)" class="member-name-link">displayMatrix(String, Object[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#displayMatrix(java.lang.String,java.lang.String%5B%5D%5B%5D)" class="member-name-link">displayMatrix(String, String[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html#displayMultiAlign(java.util.ArrayList,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayMultiAlign(ArrayList&lt;Sequence&gt;, EPartService)</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html" title="class in bacnet.scripts.phylogeny">PhylogenySmallORFs</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html#displayMultiAlign(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayMultiAlign(EPartService)</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html" title="class in bacnet.scripts.phylogeny">PhylogenySmallORFs</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayNTerminomics(org.eclipse.e4.ui.workbench.modeling.EPartService,java.lang.String)" class="member-name-link">displayNTerminomics(EPartService, String)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Load the minimal number of data for the simplest proteomics view</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#displayNTerminomicsView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)" class="member-name-link">displayNTerminomicsView(EPartService, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#displayOmicsView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayOmicsView(EPartService)</a> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html#displayOmicsView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)" class="member-name-link">displayOmicsView(EPartService, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#displayOperon(org.eclipse.swt.graphics.GC,java.lang.String,int%5B%5D,int,int,bacnet.genomeBrowser.core.Zoom)" class="member-name-link">displayOperon(GC, String, int[], int, int, Zoom)</a> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Display operon in orange = rectangle</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html#displayPhylogeny(java.util.ArrayList)" class="member-name-link">displayPhylogeny(ArrayList&lt;Sequence&gt;)</a> - Static method in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html" title="class in bacnet.scripts.phylogeny">PhylogenySmallORFs</a></dt>
<dd>
<div class="block">Displaying the newly created tree with Archaeopteryx.</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#displayProteomicsView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)" class="member-name-link">displayProteomicsView(EPartService, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>
<div class="block">Display a search in Genomic View</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#displayProteomicsView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)" class="member-name-link">displayProteomicsView(EPartService, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>
<div class="block">Display a search in Genomic View</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#displaySequence(org.eclipse.swt.graphics.GC,bacnet.datamodel.sequence.Chromosome,int,int%5B%5D,double,bacnet.genomeBrowser.core.Zoom,int)" class="member-name-link">displaySequence(GC, Chromosome, int, int[], double, Zoom, int)</a> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Display sequence information:
 nucleotide sequence on plus strand
 nucleotide sequence on minus strand
 amino acid sequence for each nucleotide position on the plus strand (= the three reading
 frame of the plus strand)</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#displaySpecificRegion(java.lang.String)" class="member-name-link">displaySpecificRegion(String)</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Display a specific element of the genome only by data.getDisplay() to false, except in the region
 of the genomeElement</div>
</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#displaySrna(bacnet.datamodel.sequence.Srna,org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displaySrna(Srna, EPartService)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>
<div class="block">Open Srna View with a specific Srna to display</div>
</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#displaySrnaView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)" class="member-name-link">displaySrnaView(EPartService, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>
<div class="block">Display the view with saved parameters</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayStat37View(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayStat37View(EPartService)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/GElement.html#displayTerminator(org.eclipse.swt.graphics.GC,java.lang.String,int%5B%5D,int,int,bacnet.genomeBrowser.core.Zoom)" class="member-name-link">displayTerminator(GC, String, int[], int, int, Zoom)</a> - Static method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></dt>
<dd>
<div class="block">Display terminator if provided = dot in purple</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayTestView(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayTestView(EPartService)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Load the minimal number of data for the simplest proteomics view</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsView.html#displayTranscriptomicsView(org.eclipse.e4.ui.workbench.modeling.EPartService,java.util.HashMap)" class="member-name-link">displayTranscriptomicsView(EPartService, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></dt>
<dd>
<div class="block">Display a search in Genomic View</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#displayValues" class="member-name-link">displayValues</a> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#displayValues" class="member-name-link">displayValues</a> - Variable in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#displayVector(java.lang.String,double%5B%5D)" class="member-name-link">displayVector(String, double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#displayVector(java.lang.String,double%5B%5D)" class="member-name-link">displayVector(String, double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#displayVector(java.lang.String,java.lang.String%5B%5D)" class="member-name-link">displayVector(String, String[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#displayVector(java.lang.String,java.lang.String%5B%5D)" class="member-name-link">displayVector(String, String[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayYersi8081(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayYersi8081(EPartService)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Display omics data for Yersinia enterocolitica 8081</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayYersiCO92(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayYersiCO92(EPartService)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Display omics data for Yersinia pestis CO92</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayYersiCSF007(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayYersiCSF007(EPartService)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Display omics data for Yersinia ruckeri CSF007</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayYersiIP32953(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayYersiIP32953(EPartService)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Display omics data for Yersinia  pseudotuberculosis IP32953</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayYersiKIM(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayYersiKIM(EPartService)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Display omics data for Yersinia pestis KIM</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayYersiY11(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayYersiY11(EPartService)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Display omics data for Yersinia enterocolitica Y11</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#displayYersiYPIII(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">displayYersiYPIII(EPartService)</a> - Static method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>
<div class="block">Display omics data for Yersinia pseudotuberculosis YPIII</div>
</dd>
<dt><a href="../bacnet/e4/rap/setup/SetupPart.html#dispose()" class="member-name-link">dispose()</a> - Method in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/SetupPart.html" title="class in bacnet.e4.rap.setup">SetupPart</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/TestPart.html#dispose()" class="member-name-link">dispose()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/TestPart.html" title="class in bacnet.e4.rap">TestPart</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/swt/ResourceManager.html#dispose()" class="member-name-link">dispose()</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Dispose of cached objects and their underlying OS resources.</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#dispose()" class="member-name-link">dispose()</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Dispose of cached objects and their underlying OS resources.</div>
</dd>
<dt><a href="../bacnet/views/CRISPRPredictView.html#dispose()" class="member-name-link">dispose()</a> - Method in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView2.html#dispose()" class="member-name-link">dispose()</a> - Method in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#disposeColors()" class="member-name-link">disposeColors()</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Dispose of all the cached <code>Color</code>'s.</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#disposeCursors()" class="member-name-link">disposeCursors()</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Dispose all of the cached cursors.</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#disposeFonts()" class="member-name-link">disposeFonts()</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Dispose all of the cached <code>Font</code>'s.</div>
</dd>
<dt><a href="../bacnet/swt/ResourceManager.html#disposeImages()" class="member-name-link">disposeImages()</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></dt>
<dd>
<div class="block">Dispose all of the cached images.</div>
</dd>
<dt><a href="../bacnet/swt/SWTResourceManager.html#disposeImages()" class="member-name-link">disposeImages()</a> - Static method in class bacnet.swt.<a href="../bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></dt>
<dd>
<div class="block">Dispose all of the cached <code>Image</code>'s.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#distance(org.biojava3.genome.parsers.gff.Location)" class="member-name-link">distance(Location)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Return distance between this location and the other location.</div>
</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html#divideTISDB()" class="member-name-link">divideTISDB()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Divide TISDB into sequence starting with a M (TISMDB), and other not starting with an M
 (TISnoMDB).</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#divideTISDB()" class="member-name-link">divideTISDB()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Divide TISDB into sequence starting with a M (TISMDB), and other not starting with an M
 (TISnoMDB).</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#dnaCompounds(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">dnaCompounds(CompoundSet&lt;NucleotideCompound&gt;)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/DNACompoundSet.html" class="type-name-link" title="class in org.biojava3.core.sequence.compound">DNACompoundSet</a> - Class in <a href="../org/biojava3/core/sequence/compound/package-summary.html">org.biojava3.core.sequence.compound</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/compound/DNACompoundSet.html#%3Cinit%3E()" class="member-name-link">DNACompoundSet()</a> - Constructor for class org.biojava3.core.sequence.compound.<a href="../org/biojava3/core/sequence/compound/DNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">DNACompoundSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#dnaRnaTranslator(org.biojava3.core.sequence.transcription.DNAToRNATranslator)" class="member-name-link">dnaRnaTranslator(DNAToRNATranslator)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html#DNASeq" class="member-name-link">DNASeq</a> - Enum constant in enum bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DNASeqCreation.html" class="type-name-link" title="class in bacnet.scripts.database">DNASeqCreation</a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DNASeqCreation.html#%3Cinit%3E()" class="member-name-link">DNASeqCreation()</a> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/DNASeqCreation.html" title="class in bacnet.scripts.database">DNASeqCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html" class="type-name-link" title="class in org.biojava3.core.sequence">DNASequence</a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#%3Cinit%3E()" class="member-name-link">DNASequence()</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Shouldn't be used but makes it bean happy</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#%3Cinit%3E(java.lang.String)" class="member-name-link">DNASequence(String)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">String is king and create a sequence from DNA with default DNA compound set</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">DNASequence(String, CompoundSet&lt;NucleotideCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Create a sequence from a string with user defined compound set</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader)" class="member-name-link">DNASequence(SequenceReader&lt;NucleotideCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Create a sequence where the actual storage of the sequence data is somewhere else</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">DNASequence(SequenceReader&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></dt>
<dd>
<div class="block">Create a sequence from a ProxySequencereader and user defined compound set</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.DNAType.html" class="type-name-link" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a> - Enum in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">The type of DNA sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">DNASequenceCreator</a> - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">A helper class that allows different ways to read a string and create a DNA sequence.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">DNASequenceCreator(CompoundSet&lt;NucleotideCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/DNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">DNASequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html" class="type-name-link" title="class in org.biojava3.core.sequence.transcription">DNAToRNATranslator</a> - Class in <a href="../org/biojava3/core/sequence/transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></dt>
<dd>
<div class="block">Performs the first stage of transcription by going from DNA to RNA.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html#%3Cinit%3E(org.biojava3.core.sequence.io.template.SequenceCreatorInterface,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet,boolean)" class="member-name-link">DNAToRNATranslator(SequenceCreatorInterface&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;, CompoundSet&lt;NucleotideCompound&gt;, boolean)</a> - Constructor for class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava3.core.sequence.transcription">DNAToRNATranslator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#Donovani_NAME" class="member-name-link">Donovani_NAME</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Name of Leishmania reference genome</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html#downloadAdditional(java.lang.String)" class="member-name-link">downloadAdditional(String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/GenomesCreation.html#downloadAll()" class="member-name-link">downloadAll()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Download all genomes available in the database</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#downloadAllFilesTechno()" class="member-name-link">downloadAllFilesTechno()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html#downloadAllGenomes(java.lang.String,java.lang.String)" class="member-name-link">downloadAllGenomes(String, String)</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html" title="class in bacnet.scripts.blast">GenomeNCBIFolderTools</a></dt>
<dd>
<div class="block">Take an array with a list of strain and ftp folders, and: <fi> download them: fna and gff file
 only <fi> create a folder for each <fi> verify MD5 checksum <fi> unzip files
 
 The strain name should be in the column with the name: strain_name The folder name for the genome
 to be donwload in should be in the column with the name: folder_name The ftp path should be found
 in the column with name : ftp_path The id of the assembly should be found in the column with name
 : assembly_accession</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#downloadAllTechno()" class="member-name-link">downloadAllTechno()</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Download all ADF files</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#downloadAllTechno()" class="member-name-link">downloadAllTechno()</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Download all ADF files</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#downloadFile(java.lang.String)" class="member-name-link">downloadFile(String)</a> - Static method in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Given an accession and an extension type:
 check if the file does not already exist in the current folder
 Download it if not
 Print out a message if no file has been downloaded</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html#downloadFile(java.lang.String,java.lang.String)" class="member-name-link">downloadFile(String, String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></dt>
<dd>
<div class="block">Given an accession and an extension type:
 check if the file does not already exist in the current folder
 Download it if not
 Print out a message if no file has been downloaded</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#downloadFile(java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">downloadFile(String, String, String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>
<div class="block">Given an accession and an extension type:
 check if the file does not already exist in the current folder
 Download it if not
 Print out a message if no file has been downloaded</div>
</dd>
<dt><a href="../bacnet/scripts/database/GenomesCreation.html#downloadGenomes(java.util.ArrayList,java.lang.String)" class="member-name-link">downloadGenomes(ArrayList&lt;String&gt;, String)</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></dt>
<dd>
<div class="block">Download new genomes taken from a text file and open all gzip files</div>
</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html#downloadIDF(java.lang.String)" class="member-name-link">downloadIDF(String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html#downloadProcessedFile(java.lang.String)" class="member-name-link">downloadProcessedFile(String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html#downloadRawFile(java.lang.String)" class="member-name-link">downloadRawFile(String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html#downloadSDRF(java.lang.String)" class="member-name-link">downloadSDRF(String)</a> - Static method in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/DownloadServiceHandler.html" class="type-name-link" title="class in bacnet.raprcp">DownloadServiceHandler</a> - Class in <a href="../bacnet/raprcp/package-summary.html">bacnet.raprcp</a></dt>
<dd>
<div class="block">Specific tools for downloading files from eclipse.rap instance NEED TO BE COMMENTED FOR
 Eclipse.RCP to work !!!</div>
</dd>
<dt><a href="../bacnet/raprcp/DownloadServiceHandler.html#%3Cinit%3E(java.lang.String,java.io.File)" class="member-name-link">DownloadServiceHandler(String, File)</a> - Constructor for class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/DownloadServiceHandler.html#%3Cinit%3E(java.lang.String,java.lang.String)" class="member-name-link">DownloadServiceHandler(String, String)</a> - Constructor for class bacnet.raprcp.<a href="../bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#downstream(int)" class="member-name-link">downstream(int)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Return the adjacent location of specified length directly downstream of this location.</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#DP10403S_NAME" class="member-name-link">DP10403S_NAME</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Name of 10403S genome</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Genome.html#DP10403S_NAME_BIS" class="member-name-link">DP10403S_NAME_BIS</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></dt>
<dd>
<div class="block">Other name for 10403S genome</div>
</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
