<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Fri Dec 10 09:15:34 CET 2021 -->
<title>P-Index</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-10">
<meta name="description" content="index: P">
<meta name="generator" content="javadoc/IndexWriter">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
<script type="text/javascript" src="../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../script-dir/jquery-ui.min.js"></script>
</head>
<body class="index-page">
<script type="text/javascript">var pathtoroot = "../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="nav-bar-cell1-rev">Index</li>
<li><a href="../help-doc.html#index">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1>Index</h1>
</div>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a>
<h2 class="title" id="I:P">P</h2>
<dl class="index">
<dt><a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html#PADJ" class="member-name-link">PADJ</a> - Enum constant in enum bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html" title="enum in bacnet.datamodel.dataset">OmicsData.ColNames</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/CircularGenomeJPanel.html#paintComponent(java.awt.Graphics)" class="member-name-link">paintComponent(Graphics)</a> - Method in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html#PAIRWISE" class="member-name-link">PAIRWISE</a> - Enum constant in enum bacnet.scripts.blast.<a href="../bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html" title="enum in bacnet.scripts.blast">BlastOutput.BlastOutputTYPE</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/StartCodonSequence.html#parentGeneSequence" class="member-name-link">parentGeneSequence</a> - Variable in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/StartCodonSequence.html" title="class in org.biojava3.core.sequence">StartCodonSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/StopCodonSequence.html#parentGeneSequence" class="member-name-link">parentGeneSequence</a> - Variable in class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/StopCodonSequence.html" title="class in org.biojava3.core.sequence">StopCodonSequence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcParser.html#parse(java.io.Reader)" class="member-name-link">parse(Reader)</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</a></dt>
<dd>
<div class="block">Reader based version of the parse methods.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/InsdcParser.html#parse(java.lang.String)" class="member-name-link">parse(String)</a> - Method in class org.biojava3.core.sequence.location.<a href="../org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</a></dt>
<dd>
<div class="block">Main method for parsing a location from a String instance</div>
</dd>
<dt><a href="../bacnet/table/core/ColorMapper.html#parseColor(double)" class="member-name-link">parseColor(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#parseColor(double)" class="member-name-link">parseColor(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/core/ColorMapperRCP.html#parseColorAWT(double)" class="member-name-link">parseColorAWT(double)</a> - Method in class bacnet.table.core.<a href="../bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#parseData()" class="member-name-link">parseData()</a> - Static method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>
<div class="block">Using LOCUSTAG_CHANGE_PATH modify every table by changin locutag ID or removing the row if the
 gene is not found</div>
</dd>
<dt><a href="../bacnet/reader/NCBIFastaHeaderParser.html#parseHeader(java.lang.String,S)" class="member-name-link">parseHeader(String, S)</a> - Method in class bacnet.reader.<a href="../bacnet/reader/NCBIFastaHeaderParser.html" title="class in bacnet.reader">NCBIFastaHeaderParser</a></dt>
<dd>
<div class="block">Parse the header and set the values in the sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html#parseHeader(java.lang.String,S)" class="member-name-link">parseHeader(String, S)</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser</a></dt>
<dd>
<div class="block">Parse the header and set the values in the sequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html#parseHeader(java.lang.String,S)" class="member-name-link">parseHeader(String, S)</a> - Method in interface org.biojava3.core.sequence.io.template.<a href="../org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#parseInitURL(org.eclipse.e4.ui.workbench.modeling.EPartService)" class="member-name-link">parseInitURL(EPartService)</a> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>
<div class="block">Parse init URL and display the correct View with the correct view parameters</div>
</dd>
<dt><a href="../bacnet/datamodel/expdesign/BioCondition.html#parseName(java.lang.String)" class="member-name-link">parseName(String)</a> - Static method in class bacnet.datamodel.expdesign.<a href="../bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></dt>
<dd>
<div class="block">Parse a Comparison name, extracting leftBioCondition and rightBiocondition</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/LocusTag.html#parseNewLocusTagChange()" class="member-name-link">parseNewLocusTagChange()</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/LocusTag.html" title="class in bacnet.datamodel.annotation">LocusTag</a></dt>
<dd>
<div class="block">Create a HashMap with all "old_locus_tag" and "locus_tag" from the new GFF folder
 Listeriomics/NCBIGenomesNew/</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/LocusTag.html#parseOldLocusTagChange()" class="member-name-link">parseOldLocusTagChange()</a> - Static method in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/LocusTag.html" title="class in bacnet.datamodel.annotation">LocusTag</a></dt>
<dd>
<div class="block">Create a HashMap with all "old_locus_tag" and "locus_tag" from the current GFF folder
 Listeriomics/NCBIGenomes/</div>
</dd>
<dt><a href="../bacnet/datamodel/phylogeny/Phylogenomic.html#parsePhylogenomicFigure(java.lang.String)" class="member-name-link">parsePhylogenomicFigure(String)</a> - Static method in class bacnet.datamodel.phylogeny.<a href="../bacnet/datamodel/phylogeny/Phylogenomic.html" title="class in bacnet.datamodel.phylogeny">Phylogenomic</a></dt>
<dd>
<div class="block">Parse phylogenomic SVG figure and extract attributes corresponding to each genome</div>
</dd>
<dt><a href="../org/biojava3/core/exceptions/ParserException.html" class="type-name-link" title="class in org.biojava3.core.exceptions">ParserException</a> - Exception in <a href="../org/biojava3/core/exceptions/package-summary.html">org.biojava3.core.exceptions</a></dt>
<dd>
<div class="block">General abstraction of different parsing errors</div>
</dd>
<dt><a href="../org/biojava3/core/exceptions/ParserException.html#%3Cinit%3E(java.lang.Exception)" class="member-name-link">ParserException(Exception)</a> - Constructor for exception org.biojava3.core.exceptions.<a href="../org/biojava3/core/exceptions/ParserException.html" title="class in org.biojava3.core.exceptions">ParserException</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/exceptions/ParserException.html#%3Cinit%3E(java.lang.String)" class="member-name-link">ParserException(String)</a> - Constructor for exception org.biojava3.core.exceptions.<a href="../org/biojava3/core/exceptions/ParserException.html" title="class in org.biojava3.core.exceptions">ParserException</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/exceptions/ParserException.html#%3Cinit%3E(java.lang.String,java.lang.Exception)" class="member-name-link">ParserException(String, Exception)</a> - Constructor for exception org.biojava3.core.exceptions.<a href="../org/biojava3/core/exceptions/ParserException.html" title="class in org.biojava3.core.exceptions">ParserException</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#parseStateParemeters(java.lang.String)" class="member-name-link">parseStateParemeters(String)</a> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/DNASeqCreation.html#parseWIGFile()" class="member-name-link">parseWIGFile()</a> - Static method in class bacnet.scripts.database.<a href="../bacnet/scripts/database/DNASeqCreation.html" title="class in bacnet.scripts.database">DNASeqCreation</a></dt>
<dd>
<div class="block">Parse WIG file and calculate Copy Number Variation = raw coverage / mean coverage<br>
 The WIG has to be Raw number of reads per basepair</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/DataSource.html#PATENTS" class="member-name-link">PATENTS</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#PATH" class="member-name-link">PATH</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Path for Srna raw data</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#PATH" class="member-name-link">PATH</a> - Static variable in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html#PATH" class="member-name-link">PATH</a> - Static variable in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html#PATH" class="member-name-link">PATH</a> - Static variable in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html#PATH" class="member-name-link">PATH</a> - Static variable in class bacnet.scripts.arrayexpress.<a href="../bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/genome/SDProfile.html#PATH" class="member-name-link">PATH</a> - Static variable in class bacnet.scripts.genome.<a href="../bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html#PATH" class="member-name-link">PATH</a> - Static variable in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html#PATH" class="member-name-link">PATH</a> - Static variable in class bacnet.scripts.listeriomics.ArrayExpress.<a href="../bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStatProteolyse.html#PATH" class="member-name-link">PATH</a> - Static variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStatProteolyse.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStatProteolyse</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#PATH" class="member-name-link">PATH</a> - Static variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#PATH" class="member-name-link">PATH</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html#PATH" class="member-name-link">PATH</a> - Static variable in class bacnet.scripts.listeriomics.srna.<a href="../bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#PATH_ADD_IDENTITY" class="member-name-link">PATH_ADD_IDENTITY</a> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#PATH_ALLDataType" class="member-name-link">PATH_ALLDataType</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#PATH_ALLExpMatrix" class="member-name-link">PATH_ALLExpMatrix</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#PATH_ALLGENEXPR" class="member-name-link">PATH_ALLGENEXPR</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#PATH_ALLRNASEQ" class="member-name-link">PATH_ALLRNASEQ</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Expression.html#PATH_ALLTILING" class="member-name-link">PATH_ALLTILING</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_ANNOTATION" class="member-name-link">PATH_ANNOTATION</a> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for all supplementary information for genome annotation : Database.getInstance().getPath() +
 "/GenomeNCBI/Annotation/";</div>
</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#PATH_BLAST" class="member-name-link">PATH_BLAST</a> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Path for Blast+</div>
</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#PATH_BLAST_MAC" class="member-name-link">PATH_BLAST_MAC</a> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/Blast.html#PATH_BLAST_WIN" class="member-name-link">PATH_BLAST_WIN</a> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_BLASTDB" class="member-name-link">PATH_BLASTDB</a> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for BlastDB folder</div>
</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#PATH_BLASTDB" class="member-name-link">PATH_BLASTDB</a> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Path for BlastDB folder</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html#PATH_COMP" class="member-name-link">PATH_COMP</a> - Static variable in class bacnet.expressionAtlas.core.<a href="../bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_COMPARISONS" class="member-name-link">PATH_COMPARISONS</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for saving all comparisons datasets</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#PATH_Conservation" class="member-name-link">PATH_Conservation</a> - Static variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#PATH_CONSERVATION" class="member-name-link">PATH_CONSERVATION</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html#PATH_Conservation_Cutoff" class="member-name-link">PATH_Conservation_Cutoff</a> - Static variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/annotation/Annotation.html#PATH_CORRECTGENE" class="member-name-link">PATH_CORRECTGENE</a> - Static variable in class bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></dt>
<dd>
<div class="block">Some EGD-e genes have same name, this file is used to correct gene name</div>
</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#PATH_DATA" class="member-name-link">PATH_DATA</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/UNAfold.html#PATH_DATA" class="member-name-link">PATH_DATA</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#PATH_DIFF_EXPR" class="member-name-link">PATH_DIFF_EXPR</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/Comparison.html#PATH_DIFF_LISTS" class="member-name-link">PATH_DIFF_LISTS</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_EXPR_NORM" class="member-name-link">PATH_EXPR_NORM</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all ExpressionMatrix RAW datasets</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_EXPR_RAW" class="member-name-link">PATH_EXPR_RAW</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all ExpressionMatrix RAW datasets</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_GENEXPR_COMPLETE" class="member-name-link">PATH_GENEXPR_COMPLETE</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all GeneExpression COMPLETE datasets</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_GENEXPR_NORM" class="member-name-link">PATH_GENEXPR_NORM</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all GeneExpression NORM datasets</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_GENEXPR_RAW" class="member-name-link">PATH_GENEXPR_RAW</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all ExpressionMatrix NORM datasets</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_GENOMES" class="member-name-link">PATH_GENOMES</a> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for all Genomes : Database.getInstance().getPath() + "/GenomeNCBI/Genomes/";</div>
</dd>
<dt><a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html#PATH_GENOMES_LIST" class="member-name-link">PATH_GENOMES_LIST</a> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/GenomeNCBIFolderTools.html" title="class in bacnet.scripts.blast">GenomeNCBIFolderTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html#PATH_GENOMES_LIST" class="member-name-link">PATH_GENOMES_LIST</a> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceBlastProtein.html" title="class in bacnet.scripts.blast">MultiSequenceBlastProtein</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html#PATH_GENOMES_LIST" class="member-name-link">PATH_GENOMES_LIST</a> - Static variable in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_GENOMES_NEW" class="member-name-link">PATH_GENOMES_NEW</a> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span>
<div class="deprecation-comment">only used in Old Listeriomics database</div>
</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_HOMOLOGS" class="member-name-link">PATH_HOMOLOGS</a> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for all homologs calculation file : Database.getInstance().getPath() +
 "/GenomeNCBI/Homologs/";</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html#PATH_INFOTABLE" class="member-name-link">PATH_INFOTABLE</a> - Static variable in class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/GenomePhylogeny.html" title="class in bacnet.scripts.phylogeny">GenomePhylogeny</a></dt>
<dd>
<div class="block">This table taken from Jolley et al.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html#PATH_LIST" class="member-name-link">PATH_LIST</a> - Static variable in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html" title="class in bacnet.scripts.listeriomics.nterm">NTermCreateFinalData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#PATH_MAC" class="member-name-link">PATH_MAC</a> - Static variable in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path when using MAC OX X</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#PATH_NCBI_BacGenome" class="member-name-link">PATH_NCBI_BacGenome</a> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_NGS_NORM" class="member-name-link">PATH_NGS_NORM</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all RNASeq NORM datasets</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_NGS_RAW" class="member-name-link">PATH_NGS_RAW</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all RNASeq RAW datasets</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_NORM" class="member-name-link">PATH_NORM</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all normalized datasets</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_OLDLOCUSTAG" class="member-name-link">PATH_OLDLOCUSTAG</a> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for all homologs calculation file : Database.getInstance().getPath() +
 "/GenomeNCBI/Homologs/";</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_PROTEINID" class="member-name-link">PATH_PROTEINID</a> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for all homologs calculation file : Database.getInstance().getPath() +
 "/GenomeNCBI/Homologs/";</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_PROTEINIDTOOLDLOCUSTAG" class="member-name-link">PATH_PROTEINIDTOOLDLOCUSTAG</a> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for all homologs calculation file : Database.getInstance().getPath() +
 "/GenomeNCBI/Homologs/";</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_PROTEOMICS_NORM" class="member-name-link">PATH_PROTEOMICS_NORM</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all Proteome NORM datasets</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_PROTEOMICS_RAW" class="member-name-link">PATH_PROTEOMICS_RAW</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all Proteome RAW datasets</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_RAW" class="member-name-link">PATH_RAW</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all raw datasets</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_RAW" class="member-name-link">PATH_RAW</a> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for all raw data for Genomes : Database.getInstance().getPath() + "/GenomeNCBI/";</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#PATH_RAW_DATA" class="member-name-link">PATH_RAW_DATA</a> - Static variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTermCreateData.html#PATH_RESULTS" class="member-name-link">PATH_RESULTS</a> - Static variable in class bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#PATH_RESULTS" class="member-name-link">PATH_RESULTS</a> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Path for BlastDB folder</div>
</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#PATH_SCRIPT" class="member-name-link">PATH_SCRIPT</a> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Path for data on the server when running homolog search blasts</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#PATH_SEC_STRUCTURE" class="member-name-link">PATH_SEC_STRUCTURE</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_STREAMING" class="member-name-link">PATH_STREAMING</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Path for all streaming datasets = serialized data for BACNET platform</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#PATH_TermSeq_EGDe" class="member-name-link">PATH_TermSeq_EGDe</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#PATH_TermSeq_EGDe" class="member-name-link">PATH_TermSeq_EGDe</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#PATH_THREADS" class="member-name-link">PATH_THREADS</a> - Static variable in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Path for all homologs calculation file : Database.getInstance().getPath() +
 "/GenomeNCBI/Homologs/";</div>
</dd>
<dt><a href="../bacnet/scripts/database/HomologCreation.html#PATH_THREADS" class="member-name-link">PATH_THREADS</a> - Static variable in class bacnet.scripts.database.<a href="../bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></dt>
<dd>
<div class="block">Path for BlastDB folder</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_TILING_NORM" class="member-name-link">PATH_TILING_NORM</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all Tiling NORM datasets</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.html#PATH_TILING_RAW" class="member-name-link">PATH_TILING_RAW</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></dt>
<dd>
<div class="block">Folder for all Tiling RAW datasets</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#PATH_TSS_EGDe" class="member-name-link">PATH_TSS_EGDe</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#PATH_TSS_EGDe" class="member-name-link">PATH_TSS_EGDe</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#PATH_TSS_Innocua" class="member-name-link">PATH_TSS_Innocua</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#PATH_TSS_Innocua" class="member-name-link">PATH_TSS_Innocua</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/MainListeriomics.html#PATH_TSSasRNA" class="member-name-link">PATH_TSSasRNA</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#PATH_TSSasRNA" class="member-name-link">PATH_TSSasRNA</a> - Static variable in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/Database.html#PATH_WEBSITE" class="member-name-link">PATH_WEBSITE</a> - Static variable in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for website database on Pasteur Virtual Machine</div>
</dd>
<dt><a href="../bacnet/Database.html#PATH_WIN" class="member-name-link">PATH_WIN</a> - Static variable in class bacnet.<a href="../bacnet/Database.html" title="class in bacnet">Database</a></dt>
<dd>
<div class="block">Path for website database on Windows</div>
</dd>
<dt><a href="../bacnet/scripts/core/Rscript.html#PATH_WIN7" class="member-name-link">PATH_WIN7</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/Rscript.html" title="class in bacnet.scripts.core">Rscript</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#PATHFigure_ASrna" class="member-name-link">PATHFigure_ASrna</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#PATHFigure_CISReg" class="member-name-link">PATHFigure_CISReg</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#PATHFigure_Srna" class="member-name-link">PATHFigure_Srna</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#PATHSerialize" class="member-name-link">PATHSerialize</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Path for serialized Srna before transferring data to Database</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#PATHTABLE_ASrna" class="member-name-link">PATHTABLE_ASrna</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#PATHTABLE_CISReg" class="member-name-link">PATHTABLE_CISReg</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#PATHTABLE_Srna" class="member-name-link">PATHTABLE_Srna</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#PATHTABLE_SrnaReference" class="member-name-link">PATHTABLE_SrnaReference</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#PATHXML" class="member-name-link">PATHXML</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Path for Srna XML serialization for control</div>
</dd>
<dt><a href="../bacnet/datamodel/sequence/Srna.html#PATHXML_10403S" class="member-name-link">PATHXML_10403S</a> - Static variable in class bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></dt>
<dd>
<div class="block">Path for XML serialization for 10403S (Which have a specific annotation)</div>
</dd>
<dt><a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html#PC" class="member-name-link">PC</a> - Enum constant in enum bacnet.datamodel.annotation.<a href="../bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DataSource.html#PDB1" class="member-name-link">PDB1</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DataSource.html#PDB2" class="member-name-link">PDB2</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DataSource.html#PDBe" class="member-name-link">PDBe</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html#PDBWEB" class="member-name-link">PDBWEB</a> - Enum constant in enum org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html" title="enum in org.biojava3.core.sequence.template">LightweightProfile.StringFormat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#pearsonCorrelation(double%5B%5D,double%5B%5D)" class="member-name-link">pearsonCorrelation(double[], double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html#peptideLogo()" class="member-name-link">peptideLogo()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></dt>
<dd>
<div class="block">Create a fasta file with all sequences to create a logo</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/Peptidomics.html" class="type-name-link" title="class in bacnet.scripts.listeriomics">Peptidomics</a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/Peptidomics.html#%3Cinit%3E()" class="member-name-link">Peptidomics()</a> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/Peptidomics.html" title="class in bacnet.scripts.listeriomics">Peptidomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/SequenceTools.html#percentNucleotideSequence(java.lang.String)" class="member-name-link">percentNucleotideSequence(String)</a> - Static method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SequenceTools.html" title="class in org.biojava3.core.util">SequenceTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#percentOverlap(org.biojava3.genome.parsers.gff.Location)" class="member-name-link">percentOverlap(Location)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Return percent overlap of two locations.</div>
</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizard.html#performCancel()" class="member-name-link">performCancel()</a> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizard.html" title="class in bacnet.table.gui">ColorMapperWizard</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizardPage.html#performCancel()" class="member-name-link">performCancel()</a> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/table/gui/ColorMapperWizard.html#performFinish()" class="member-name-link">performFinish()</a> - Method in class bacnet.table.gui.<a href="../bacnet/table/gui/ColorMapperWizard.html" title="class in bacnet.table.gui">ColorMapperWizard</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html#PERMUT" class="member-name-link">PERMUT</a> - Enum constant in enum bacnet.scripts.core.stat.<a href="../bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DataSource.html#PFAM" class="member-name-link">PFAM</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/phylogeny/Phylogenomic.html#PHYLO_GENOME_SVG" class="member-name-link">PHYLO_GENOME_SVG</a> - Static variable in class bacnet.datamodel.phylogeny.<a href="../bacnet/datamodel/phylogeny/Phylogenomic.html" title="class in bacnet.datamodel.phylogeny">Phylogenomic</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/phylogeny/Phylogenomic.html" class="type-name-link" title="class in bacnet.datamodel.phylogeny">Phylogenomic</a> - Class in <a href="../bacnet/datamodel/phylogeny/package-summary.html">bacnet.datamodel.phylogeny</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/phylogeny/Phylogenomic.html#%3Cinit%3E()" class="member-name-link">Phylogenomic()</a> - Constructor for class bacnet.datamodel.phylogeny.<a href="../bacnet/datamodel/phylogeny/Phylogenomic.html" title="class in bacnet.datamodel.phylogeny">Phylogenomic</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/PhylogenomicsCreation.html" class="type-name-link" title="class in bacnet.scripts.database">PhylogenomicsCreation</a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/PhylogenomicsCreation.html#%3Cinit%3E()" class="member-name-link">PhylogenomicsCreation()</a> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/PhylogenomicsCreation.html" title="class in bacnet.scripts.database">PhylogenomicsCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" class="type-name-link" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a> - Class in <a href="../bacnet/scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></dt>
<dd>
<div class="block">Different methods to create phylogeny tree, and tables for each genome elements</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html#%3Cinit%3E()" class="member-name-link">PhylogenyListeriomics()</a> - Constructor for class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/PhylogenySmallORFs.html" class="type-name-link" title="class in bacnet.scripts.blast">PhylogenySmallORFs</a> - Class in <a href="../bacnet/scripts/blast/package-summary.html">bacnet.scripts.blast</a></dt>
<dd>
<div class="block">Tools to blast a small ORF on other bacterial genomes, and display results on a phylogeny tree</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html" class="type-name-link" title="class in bacnet.scripts.phylogeny">PhylogenySmallORFs</a> - Class in <a href="../bacnet/scripts/phylogeny/package-summary.html">bacnet.scripts.phylogeny</a></dt>
<dd>
<div class="block">Tools to blast a small ORF on other bacterial genomes, and display results on a phylogeny tree</div>
</dd>
<dt><a href="../bacnet/scripts/blast/PhylogenySmallORFs.html#%3Cinit%3E()" class="member-name-link">PhylogenySmallORFs()</a> - Constructor for class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/PhylogenySmallORFs.html" title="class in bacnet.scripts.blast">PhylogenySmallORFs</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html#%3Cinit%3E()" class="member-name-link">PhylogenySmallORFs()</a> - Constructor for class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenySmallORFs.html" title="class in bacnet.scripts.phylogeny">PhylogenySmallORFs</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyTools.html" class="type-name-link" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a> - Class in <a href="../bacnet/scripts/phylogeny/package-summary.html">bacnet.scripts.phylogeny</a></dt>
<dd>
<div class="block">Different methods to manipulate and display Newick and other phylogeny files</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyTools.html#%3Cinit%3E()" class="member-name-link">PhylogenyTools()</a> - Constructor for class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" class="type-name-link" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a> - Class in <a href="../bacnet/scripts/phylogeny/package-summary.html">bacnet.scripts.phylogeny</a></dt>
<dd>
<div class="block">This list of function is useful to manipulate Newick files given by Jolley in May 2012, from its
 publication Jolley et al.</div>
</dd>
<dt><a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html#%3Cinit%3E()" class="member-name-link">PhylogenyToolsJolley()</a> - Constructor for class bacnet.scripts.phylogeny.<a href="../bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.DNAType.html#PLASMID" class="member-name-link">PLASMID</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DNASequence.DNAType.html#PLASTID" class="member-name-link">PLASTID</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#plot2ann" class="member-name-link">plot2ann</a> - Static variable in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/UNAfold.html#plot2ann" class="member-name-link">plot2ann</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#plot2Ann(java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">plot2Ann(String, String, String)</a> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>
<div class="block">Convert a .ct file and .plot file into an .ann file<br>
 The output of plot2ann script is save into a .ann file</div>
</dd>
<dt><a href="../bacnet/utils/UNAfold.html#plot2Ann(java.lang.String,java.lang.String,java.lang.String)" class="member-name-link">plot2Ann(String, String, String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>
<div class="block">Convert a .ct file and .plot file into an .ann file<br>
 The output of plot2ann script is save into a .ann file</div>
</dd>
<dt><a href="../bacnet/scripts/core/UNAfold.html#plotFolding(java.lang.String,int,java.lang.String)" class="member-name-link">plotFolding(String, int, String)</a> - Static method in class bacnet.scripts.core.<a href="../bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></dt>
<dd>
<div class="block">Run sir_graph software on a particular file</div>
</dd>
<dt><a href="../bacnet/utils/UNAfold.html#plotFolding(java.lang.String,int,java.lang.String)" class="member-name-link">plotFolding(String, int, String)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></dt>
<dd>
<div class="block">Run sir_graph software on a particular file</div>
</dd>
<dt><a href="../bacnet/scripts/Activator.html#PLUGIN_ID" class="member-name-link">PLUGIN_ID</a> - Static variable in class bacnet.scripts.<a href="../bacnet/scripts/Activator.html" title="class in bacnet.scripts">Activator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html#plus" class="member-name-link">plus</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#plus()" class="member-name-link">plus()</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">Return location that is in same position on plus strand.</div>
</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#plus(bacnet.datamodel.dataset.ExpressionMatrix,bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">plus(ExpressionMatrix, ExpressionMatrix)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">Perform an addition between two ExpressionMatrix <br>
 First the intersection in term of RowNames and headers is found, then addition is done for each
 row</div>
</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html#plusM" class="member-name-link">plusM</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html#plusP" class="member-name-link">plusP</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Point.html" class="type-name-link" title="interface in org.biojava3.core.sequence.location.template">Point</a> - Interface in <a href="../org/biojava3/core/sequence/location/template/package-summary.html">org.biojava3.core.sequence.location.template</a></dt>
<dd>
<div class="block">Holds a single point part of a location</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/location/template/Point.Resolver.html" class="type-name-link" title="interface in org.biojava3.core.sequence.location.template">Point.Resolver</a>&lt;<a href="../org/biojava3/core/sequence/location/template/Point.Resolver.html" title="type parameter in Point.Resolver">T</a> extends <a href="../org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/location/template/package-summary.html">org.biojava3.core.sequence.location.template</a></dt>
<dd>
<div class="block">Used to resolve a position about a point</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#populate(java.lang.String)" class="member-name-link">populate(String)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>
<div class="block">Loops through the chars in a String and passes them onto <a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#setCompoundAt(char,int)"><code>BitSequenceReader.BitArrayWorker.setCompoundAt(char, int)</code></a></div>
</dd>
<dt><a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#populate(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">populate(Sequence&lt;C&gt;)</a> - Method in class org.biojava3.core.sequence.storage.<a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a></dt>
<dd>
<div class="block">Loops through the Compounds in a Sequence and passes them onto
 <a href="../org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#setCompoundAt(C,int)"><code>BitSequenceReader.BitArrayWorker.setCompoundAt(Compound, int)</code></a></div>
</dd>
<dt><a href="../org/biojava3/core/sequence/Strand.html#POSITIVE" class="member-name-link">POSITIVE</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html#possibleSmallRNAPeptides()" class="member-name-link">possibleSmallRNAPeptides()</a> - Static method in class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>
<div class="block">Get all possible peptides from EGD-e: <br>
 get all proteins sequence from Small RNAs
 get peptide sequence from M start to first R, or first E
 cut-off every peptide of size under 5aa
 export in a table
 use <code>http://web.expasy.org/compute_pi/</code> to calculate mass and isoelectric point
 save the different mass in NTermUtils.getPATH()+EGDePeptideMass.txt</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html#possibleSmallRNAPeptides()" class="member-name-link">possibleSmallRNAPeptides()</a> - Static method in class bacnet.scripts.listeriomics.nterm.<a href="../bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></dt>
<dd>
<div class="block">Get all possible peptides from EGD-e: <br>
 get all proteins sequence from Small RNAs
 get peptide sequence from M start to first R, or first E
 cut-off every peptide of size under 5aa
 export in a table
 use <code>http://web.expasy.org/compute_pi/</code> to calculate mass and isoelectric point
 save the different mass in NTermUtils.getPATH()+EGDePeptideMass.txt</div>
</dd>
<dt><a href="../bacnet/utils/VectorUtils.html#pow2(double%5B%5D)" class="member-name-link">pow2(double[])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></dt>
<dd>
<div class="block">2^value</div>
</dd>
<dt><a href="../bacnet/utils/ArrayUtils.html#pow2(double%5B%5D%5B%5D)" class="member-name-link">pow2(double[][])</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></dt>
<dd>
<div class="block">2^value</div>
</dd>
<dt><a href="../bacnet/utils/ExpressionMatrixStat.html#pow2(bacnet.datamodel.dataset.ExpressionMatrix)" class="member-name-link">pow2(ExpressionMatrix)</a> - Static method in class bacnet.utils.<a href="../bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></dt>
<dd>
<div class="block">2^values</div>
</dd>
<dt><a href="../bacnet/scripts/core/PredaTerm.html" class="type-name-link" title="class in bacnet.scripts.core">PredaTerm</a> - Class in <a href="../bacnet/scripts/core/package-summary.html">bacnet.scripts.core</a></dt>
<dd>
<div class="block">PREDicting poly-A region and TERMinal splice acceptor for Leishmania Please read manuscript for
 details : Smith, M., Blanchette, M., Papadopoulou, B.</div>
</dd>
<dt><a href="../bacnet/scripts/core/PredaTerm.html#%3Cinit%3E()" class="member-name-link">PredaTerm()</a> - Constructor for class bacnet.scripts.core.<a href="../bacnet/scripts/core/PredaTerm.html" title="class in bacnet.scripts.core">PredaTerm</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html#preDestroy()" class="member-name-link">preDestroy()</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html#PREDICTED" class="member-name-link">PREDICTED</a> - Enum constant in enum org.biojava3.core.sequence.template.<a href="../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#prefix(int)" class="member-name-link">prefix(int)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">The part of this location before the specified position.</div>
</dd>
<dt><a href="../org/biojava3/genome/parsers/gff/Location.html#prefix(org.biojava3.genome.parsers.gff.Location)" class="member-name-link">prefix(Location)</a> - Method in class org.biojava3.genome.parsers.gff.<a href="../org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></dt>
<dd>
<div class="block">The part of this location before the other location (not inclusive).</div>
</dd>
<dt><a href="../bacnet/utils/RWTUtils.html#PRELOADED_ITEMS" class="member-name-link">PRELOADED_ITEMS</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/PrettyXMLWriter.html" class="type-name-link" title="class in org.biojava3.core.util">PrettyXMLWriter</a> - Class in <a href="../org/biojava3/core/util/package-summary.html">org.biojava3.core.util</a></dt>
<dd>
<div class="block">Implementation of XMLWriter which emits nicely formatted documents to a PrintWriter.</div>
</dd>
<dt><a href="../org/biojava3/core/util/PrettyXMLWriter.html#%3Cinit%3E(java.io.PrintWriter)" class="member-name-link">PrettyXMLWriter(PrintWriter)</a> - Constructor for class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/DataSource.html#PRF" class="member-name-link">PRF</a> - Enum constant in enum org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/Hashcoder.html#PRIME" class="member-name-link">PRIME</a> - Static variable in class org.biojava3.core.util.<a href="../org/biojava3/core/util/Hashcoder.html" title="class in org.biojava3.core.util">Hashcoder</a></dt>
<dd>
<div class="block">The prime number used to multiply any calculated hashcode seed by

 i.e. result = PRIME*result + c

 Where result is the result of the previous calculation (at first this will be seed) and c is the
 calculated int to add to result</div>
</dd>
<dt><a href="../org/biojava3/core/util/PrettyXMLWriter.html#print(java.lang.String)" class="member-name-link">print(String)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLWriter.html#print(java.lang.String)" class="member-name-link">print(String)</a> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Prints some textual content in an element.</div>
</dd>
<dt><a href="../org/biojava3/core/util/PrettyXMLWriter.html#println(java.lang.String)" class="member-name-link">println(String)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLWriter.html#println(java.lang.String)" class="member-name-link">println(String)</a> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Prints some textual content, terminated with a newline character.</div>
</dd>
<dt><a href="../org/biojava3/core/util/PrettyXMLWriter.html#printRaw(java.lang.String)" class="member-name-link">printRaw(String)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/util/XMLWriter.html#printRaw(java.lang.String)" class="member-name-link">printRaw(String)</a> - Method in interface org.biojava3.core.util.<a href="../org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></dt>
<dd>
<div class="block">Send raw data to the stream.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#PROBES_PATH" class="member-name-link">PROBES_PATH</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#PROBES_PATH" class="member-name-link">PROBES_PATH</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/GeneExpression.html#PROBES_PATH_2" class="member-name-link">PROBES_PATH_2</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/Tiling.html#PROBES_PATH_2" class="member-name-link">PROBES_PATH_2</a> - Static variable in class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html#process()" class="member-name-link">process()</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaGeneWriter.html" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaReader.html#process()" class="member-name-link">process()</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaReader.html" title="class in org.biojava3.core.sequence.io">FastaReader</a></dt>
<dd>
<div class="block">The parsing is done in this method</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/FastaWriter.html#process()" class="member-name-link">process()</a> - Method in class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/query/OutputHitsGFF.html#process(java.io.File,java.io.File,java.io.File,double,double,boolean,boolean)" class="member-name-link">process(File, File, File, double, double, boolean, boolean)</a> - Method in class org.biojava3.genome.query.<a href="../org/biojava3/genome/query/OutputHitsGFF.html" title="class in org.biojava3.genome.query">OutputHitsGFF</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.ReaderProcessor.html#process(java.lang.String)" class="member-name-link">process(String)</a> - Method in interface org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.ReaderProcessor.html" title="interface in org.biojava3.core.sequence.io.util">IOUtils.ReaderProcessor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/genome/uniprot/UniprotToFasta.html#process(java.lang.String,java.lang.String)" class="member-name-link">process(String, String)</a> - Method in class org.biojava3.genome.uniprot.<a href="../org/biojava3/genome/uniprot/UniprotToFasta.html" title="class in org.biojava3.genome.uniprot">UniprotToFasta</a></dt>
<dd>
<div class="block">Convert a Uniprot sequence file to a fasta file.</div>
</dd>
<dt><a href="../bacnet/reader/FastaFileReader.html#process(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">process(CompoundSet&lt;?&gt;)</a> - Method in class bacnet.reader.<a href="../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a></dt>
<dd>
<div class="block">The parsing is done in this method
 
 ******* There is a problem when the genes in the fasta file have the same Id, so I modify this
 method accordingly</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/technology/UGiessen.html#processAllData()" class="member-name-link">processAllData()</a> - Static method in class bacnet.scripts.listeriomics.technology.<a href="../bacnet/scripts/listeriomics/technology/UGiessen.html" title="class in bacnet.scripts.listeriomics.technology">UGiessen</a></dt>
<dd>
<div class="block">Open every processed files and extract the different signal from it.</div>
</dd>
<dt><a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html#processDanielRiboSeq()" class="member-name-link">processDanielRiboSeq()</a> - Method in class bacnet.scripts.listeriomics.<a href="../bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></dt>
<dd>
<div class="block">Process Daniel Dar TermSeq</div>
</dd>
<dt><a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#processGenomeName(java.lang.String)" class="member-name-link">processGenomeName(String)</a> - Static method in class bacnet.datamodel.sequenceNCBI.<a href="../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></dt>
<dd>
<div class="block">Replace all spaces in genomeName by "_" to have folder name compatible for Blast search<br>
 All "_" already present are replaced by "--"</div>
</dd>
<dt><a href="../org/biojava3/genome/util/SplitFasta.html#processNucleotides(java.io.File,java.lang.String,java.io.File)" class="member-name-link">processNucleotides(File, String, File)</a> - Method in class org.biojava3.genome.util.<a href="../org/biojava3/genome/util/SplitFasta.html" title="class in org.biojava3.genome.util">SplitFasta</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/io/util/IOUtils.html#processReader(java.io.BufferedReader,org.biojava3.core.sequence.io.util.IOUtils.ReaderProcessor)" class="member-name-link">processReader(BufferedReader, IOUtils.ReaderProcessor)</a> - Static method in class org.biojava3.core.sequence.io.util.<a href="../org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></dt>
<dd>
<div class="block">Takes in a reader and a processor, reads every line from the given file and then invokes the
 processor.</div>
</dd>
<dt><a href="../bacnet/e4/rap/BasicApplication.html#projectName" class="member-name-link">projectName</a> - Static variable in class bacnet.e4.rap.<a href="../bacnet/e4/rap/BasicApplication.html" title="class in bacnet.e4.rap">BasicApplication</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/setup/BasicApplication.html#projectName" class="member-name-link">projectName</a> - Static variable in class bacnet.e4.rap.setup.<a href="../bacnet/e4/rap/setup/BasicApplication.html" title="class in bacnet.e4.rap.setup">BasicApplication</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html#proteinCreator(org.biojava3.core.sequence.io.template.SequenceCreatorInterface)" class="member-name-link">proteinCreator(SequenceCreatorInterface&lt;AminoAcidCompound&gt;)</a> - Method in class org.biojava3.core.sequence.transcription.<a href="../org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/FastaFileReader.html#proteinfileExt" class="member-name-link">proteinfileExt</a> - Static variable in class bacnet.reader.<a href="../bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/ProteinSequence.html" class="type-name-link" title="class in org.biojava3.core.sequence">ProteinSequence</a> - Class in <a href="../org/biojava3/core/sequence/package-summary.html">org.biojava3.core.sequence</a></dt>
<dd>
<div class="block">The representation of a ProteinSequence</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/ProteinSequence.html#%3Cinit%3E(java.lang.String)" class="member-name-link">ProteinSequence(String)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></dt>
<dd>
<div class="block">Create a protein from a string</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/ProteinSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ProteinSequence(String, CompoundSet&lt;AminoAcidCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></dt>
<dd>
<div class="block">Create a protein from a string with a user defined set of amino acids</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/ProteinSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader)" class="member-name-link">ProteinSequence(ProxySequenceReader&lt;AminoAcidCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></dt>
<dd>
<div class="block">A protein sequence where the storage of the sequence is somewhere else.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/ProteinSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ProteinSequence(ProxySequenceReader&lt;AminoAcidCompound&gt;, CompoundSet&lt;AminoAcidCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.<a href="../org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></dt>
<dd>
<div class="block">A protein sequence where the storage of the sequence is somewhere else with user defined set of
 amino acids.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html" class="type-name-link" title="class in org.biojava3.core.sequence.io">ProteinSequenceCreator</a> - Class in <a href="../org/biojava3/core/sequence/io/package-summary.html">org.biojava3.core.sequence.io</a></dt>
<dd>
<div class="block">Used to create a ProteinSequence from a String to allow for details about the location of the
 sequence etc.</div>
</dd>
<dt><a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ProteinSequenceCreator(CompoundSet&lt;AminoAcidCompound&gt;)</a> - Constructor for class org.biojava3.core.sequence.io.<a href="../org/biojava3/core/sequence/io/ProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">ProteinSequenceCreator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html" class="type-name-link" title="class in bacnet.scripts.blast">ProteinTools</a> - Class in <a href="../bacnet/scripts/blast/package-summary.html">bacnet.scripts.blast</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/blast/ProteinTools.html#%3Cinit%3E()" class="member-name-link">ProteinTools()</a> - Constructor for class bacnet.scripts.blast.<a href="../bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html#Proteome" class="member-name-link">Proteome</a> - Enum constant in enum bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/ProteomicsCreation.html" class="type-name-link" title="class in bacnet.scripts.database">ProteomicsCreation</a> - Class in <a href="../bacnet/scripts/database/package-summary.html">bacnet.scripts.database</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/scripts/database/ProteomicsCreation.html#%3Cinit%3E()" class="member-name-link">ProteomicsCreation()</a> - Constructor for class bacnet.scripts.database.<a href="../bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/dataset/ProteomicsData.html" class="type-name-link" title="class in bacnet.datamodel.dataset">ProteomicsData</a> - Class in <a href="../bacnet/datamodel/dataset/package-summary.html">bacnet.datamodel.dataset</a></dt>
<dd>
<div class="block">OmicsData class inherited everything from ExpressionMatrix, the only difference is the TypeDate
 value, it a Proteome data.<br>
 It is not very useful for the moment but will be in the futur</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/ProteomicsData.html#%3Cinit%3E()" class="member-name-link">ProteomicsData()</a> - Constructor for class bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" class="type-name-link" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a> - Class in <a href="../bacnet/expressionAtlas/package-summary.html">bacnet.expressionAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#%3Cinit%3E(org.eclipse.swt.widgets.Composite,int,bacnet.expressionAtlas.ProteomicsExpressionView)" class="member-name-link">ProteomicsDataFilterComposite(Composite, int, ProteomicsExpressionView)</a> - Constructor for class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#%3Cinit%3E(org.eclipse.swt.widgets.Composite,int,bacnet.expressionAtlas.ProteomicsView)" class="member-name-link">ProteomicsDataFilterComposite(Composite, int, ProteomicsView)</a> - Constructor for class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" class="type-name-link" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a> - Class in <a href="../bacnet/expressionAtlas/package-summary.html">bacnet.expressionAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html#%3Cinit%3E()" class="member-name-link">ProteomicsExpressionView()</a> - Constructor for class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html" class="type-name-link" title="class in bacnet.expressionAtlas">ProteomicsView</a> - Class in <a href="../bacnet/expressionAtlas/package-summary.html">bacnet.expressionAtlas</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsView.html#%3Cinit%3E()" class="member-name-link">ProteomicsView()</a> - Constructor for class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/biojava3/core/sequence/template/ProxySequenceReader.html" class="type-name-link" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../org/biojava3/core/sequence/template/ProxySequenceReader.html" title="type parameter in ProxySequenceReader">C</a> extends <a href="../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; - Interface in <a href="../org/biojava3/core/sequence/template/package-summary.html">org.biojava3.core.sequence.template</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/sequence/NcRNA.TypeNcRNA.html#pseudoGene" class="member-name-link">pseudoGene</a> - Enum constant in enum bacnet.datamodel.sequence.<a href="../bacnet/datamodel/sequence/NcRNA.TypeNcRNA.html" title="enum in bacnet.datamodel.sequence">NcRNA.TypeNcRNA</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/PTTReader.html" class="type-name-link" title="class in bacnet.reader">PTTReader</a> - Class in <a href="../bacnet/reader/package-summary.html">bacnet.reader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/reader/PTTReader.html#%3Cinit%3E()" class="member-name-link">PTTReader()</a> - Constructor for class bacnet.reader.<a href="../bacnet/reader/PTTReader.html" title="class in bacnet.reader">PTTReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/utils/BasicColor.html#PURPLE" class="member-name-link">PURPLE</a> - Static variable in class bacnet.utils.<a href="../bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/e4/rap/InitViewBacnet.html#pushState()" class="member-name-link">pushState()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewBacnet.html" title="class in bacnet.e4.rap">InitViewBacnet</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeria.html#pushState()" class="member-name-link">pushState()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeria.html" title="class in bacnet.e4.rap">InitViewListeria</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><a href="../bacnet/e4/rap/InitViewListeriaSample.html#pushState()" class="member-name-link">pushState()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">InitViewListeriaSample</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><a href="../bacnet/e4/rap/InitViewYersinia.html#pushState()" class="member-name-link">pushState()</a> - Method in class bacnet.e4.rap.<a href="../bacnet/e4/rap/InitViewYersinia.html" title="class in bacnet.e4.rap">InitViewYersinia</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html#pushState()" class="member-name-link">pushState()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html#pushState()" class="member-name-link">pushState()</a> - Method in class bacnet.expressionAtlas.<a href="../bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/genomeBrowser/NTerminomicsView.html#pushState()" class="member-name-link">pushState()</a> - Method in class bacnet.genomeBrowser.<a href="../bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html#pushState()" class="member-name-link">pushState()</a> - Method in class bacnet.genomeBrowser.tracksGUI.<a href="../bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></dt>
<dd>
<div class="block">Push states to change url</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GeneView.html#pushState()" class="member-name-link">pushState()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><a href="../bacnet/sequenceTools/GenomicsView.html#pushState()" class="member-name-link">pushState()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><a href="../bacnet/sequenceTools/SrnaView.html#pushState()" class="member-name-link">pushState()</a> - Method in class bacnet.sequenceTools.<a href="../bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></dt>
<dd>
<div class="block">Push genome, chromosome, gene and Tabitem state</div>
</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#pushStateView(java.lang.String)" class="member-name-link">pushStateView(String)</a> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/raprcp/NavigationManagement.html#pushStateView(java.lang.String,java.util.HashMap)" class="member-name-link">pushStateView(String, HashMap&lt;String, String&gt;)</a> - Static method in class bacnet.raprcp.<a href="../bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></dt>
<dd>
<div class="block">Push state puttin viewId and state parameters <br>
 stateName = viewID + key + "=" + parameters.get(key) + ";"</div>
</dd>
<dt><a href="../org/biojava3/core/util/SoftHashMap.html#put(K,V)" class="member-name-link">put(K, V)</a> - Method in class org.biojava3.core.util.<a href="../org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a></dt>
<dd>
<div class="block">Here we put the key, value pair into the HashMap using a SoftValue
 
 object.</div>
</dd>
<dt><a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html#PVALUE" class="member-name-link">PVALUE</a> - Enum constant in enum bacnet.datamodel.dataset.<a href="../bacnet/datamodel/dataset/OmicsData.ColNames.html" title="enum in bacnet.datamodel.dataset">OmicsData.ColNames</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/datamodel/proteomics/NTerm.TypeModif.html#PyroGlu" class="member-name-link">PyroGlu</a> - Enum constant in enum bacnet.datamodel.proteomics.<a href="../bacnet/datamodel/proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView.html#PYTHON_PATH" class="member-name-link">PYTHON_PATH</a> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView2.html#PYTHON_PATH" class="member-name-link">PYTHON_PATH</a> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView.html#PYTHON_SERVER" class="member-name-link">PYTHON_SERVER</a> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../bacnet/views/CRISPRPredictView2.html#PYTHON_SERVER" class="member-name-link">PYTHON_SERVER</a> - Variable in class bacnet.views.<a href="../bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<br><a href="../allclasses-index.html">All&nbsp;Classes&nbsp;and&nbsp;Interfaces</a><span class="vertical-separator">|</span><a href="../allpackages-index.html">All&nbsp;Packages</a><span class="vertical-separator">|</span><a href="../constant-values.html">Constant&nbsp;Field&nbsp;Values</a><span class="vertical-separator">|</span><a href="../serialized-form.html">Serialized&nbsp;Form</a></main>
</div>
</div>
</body>
</html>
