<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>FastaFileReader</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
<script type="text/javascript" src="../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="FastaFileReader";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":9,"i1":9,"i2":10,"i3":9,"i4":9,"i5":9};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FastaFileReader.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-files/index-1.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.reader</a></div>
<h2 title="Class FastaFileReader" class="title">Class FastaFileReader&lt;S extends <a href="../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>bacnet.reader.FastaFileReader&lt;S,&#8203;C&gt;</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<pre>public class <span class="typeNameLabel">FastaFileReader&lt;S extends <a href="../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</span>
extends java.lang.Object</pre>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#allfileExt">allfileExt</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>(package private) <a href="../../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#br">br</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>(package private) java.io.FileInputStream</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#fi">fi</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>(package private) <a href="../../org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">S</a>,&#8203;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#headerParser">headerParser</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>(package private) java.io.InputStreamReader</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isr">isr</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#proteinfileExt">proteinfileExt</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#seqfileExt">seqfileExt</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>(package private) <a href="../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#sequenceCreator">sequenceCreator</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)">FastaFileReader</a></span>&#8203;(java.io.File&nbsp;file,
               <a href="../../org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">S</a>,&#8203;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;headerParser,
               <a href="../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;sequenceCreator)</code></th>
<td class="colLast">
<div class="block">If you are going to use the FileProxyProteinSequenceCreator then you need to use this constructor
 because we need details about the location of the file.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.io.InputStream,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)">FastaFileReader</a></span>&#8203;(java.io.InputStream&nbsp;is,
               <a href="../../org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">S</a>,&#8203;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;headerParser,
               <a href="../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;sequenceCreator)</code></th>
<td class="colLast">
<div class="block">If you are going to use FileProxyProteinSequenceCreator then do not use this constructor because
 we need details about local file offsets for quick reads.</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addCodingRegionFeaturesFromFFN(java.io.File,java.io.File,java.util.LinkedHashMap)">addCodingRegionFeaturesFromFFN</a></span>&#8203;(java.io.File&nbsp;root,
                              java.io.File&nbsp;file,
                              java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;&nbsp;chromosomes)</code></th>
<td class="colLast">
<div class="block">From a .ffn file in root/file.faa We had all CodingInfo into the ChromosomeSequence
 
 This method use my personal NCBIFastaHeaderParser</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addNonCodingRegionFeaturesFromFRN(java.io.File,java.io.File,java.util.LinkedHashMap)">addNonCodingRegionFeaturesFromFRN</a></span>&#8203;(java.io.File&nbsp;root,
                                 java.io.File&nbsp;file,
                                 java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;&nbsp;chromosomes)</code></th>
<td class="colLast">
<div class="block">From a .frn file in root/file.frn We add all Noncoding Info into the ChromosomeSequence
 
 This method use my personal NCBIFastaHeaderParser</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">S</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#process(org.biojava3.core.sequence.template.CompoundSet)">process</a></span>&#8203;(<a href="../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;?&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">The parsing is done in this method
 
 ******* There is a problem when the genes in the fasta file have the same Id, so I modify this
 method accordingly</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#saveCDS(java.util.ArrayList,java.lang.String)">saveCDS</a></span>&#8203;(java.util.ArrayList&lt;<a href="../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;&nbsp;sequences,
       java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Save all amino acid sequence in a fasta file</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#saveProteins(java.util.ArrayList,java.lang.String)">saveProteins</a></span>&#8203;(java.util.ArrayList&lt;<a href="../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;&nbsp;sequences,
            java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Save all amino acid sequence in a fasta file</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#updateInfoFromChromosome(java.util.LinkedHashMap)">updateInfoFromChromosome</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;&nbsp;chromosomes)</code></th>
<td class="colLast">
<div class="block">According to the info we found in Chromosomes, we update the info of the different components
 (coding, ncoding, proteins)</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="allfileExt">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>allfileExt</h4>
<pre>public static&nbsp;java.lang.String[] allfileExt</pre>
</li>
</ul>
<a id="seqfileExt">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>seqfileExt</h4>
<pre>public static&nbsp;java.lang.String[] seqfileExt</pre>
</li>
</ul>
<a id="proteinfileExt">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>proteinfileExt</h4>
<pre>public static&nbsp;java.lang.String[] proteinfileExt</pre>
</li>
</ul>
<a id="sequenceCreator">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>sequenceCreator</h4>
<pre><a href="../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">C</a> extends <a href="../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; sequenceCreator</pre>
</li>
</ul>
<a id="headerParser">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>headerParser</h4>
<pre><a href="../../org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">S</a> extends <a href="../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">C</a> extends <a href="../../org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt; headerParser</pre>
</li>
</ul>
<a id="br">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>br</h4>
<pre><a href="../../org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a> br</pre>
</li>
</ul>
<a id="isr">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isr</h4>
<pre>java.io.InputStreamReader isr</pre>
</li>
</ul>
<a id="fi">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>fi</h4>
<pre>java.io.FileInputStream fi</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;(java.io.InputStream,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>FastaFileReader</h4>
<pre>public&nbsp;FastaFileReader&#8203;(java.io.InputStream&nbsp;is,
                       <a href="../../org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">S</a>,&#8203;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;headerParser,
                       <a href="../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;sequenceCreator)</pre>
<div class="block">If you are going to use FileProxyProteinSequenceCreator then do not use this constructor because
 we need details about local file offsets for quick reads. InputStreams does not give you the name
 of the stream to access quickly via file seek. A seek in an inputstream is forced to read all the
 data so you don't gain anything.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>br</code> - </dd>
<dd><code>headerParser</code> - </dd>
<dd><code>sequenceCreator</code> - </dd>
</dl>
</li>
</ul>
<a id="&lt;init&gt;(java.io.File,org.biojava3.core.sequence.io.template.FastaHeaderParserInterface,org.biojava3.core.sequence.io.template.SequenceCreatorInterface)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>FastaFileReader</h4>
<pre>public&nbsp;FastaFileReader&#8203;(java.io.File&nbsp;file,
                       <a href="../../org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">S</a>,&#8203;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;headerParser,
                       <a href="../../org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">C</a>&gt;&nbsp;sequenceCreator)
                throws java.lang.Exception</pre>
<div class="block">If you are going to use the FileProxyProteinSequenceCreator then you need to use this constructor
 because we need details about the location of the file.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>file</code> - </dd>
<dd><code>headerParser</code> - </dd>
<dd><code>sequenceCreator</code> - </dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="addCodingRegionFeaturesFromFFN(java.io.File,java.io.File,java.util.LinkedHashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addCodingRegionFeaturesFromFFN</h4>
<pre class="methodSignature">public static&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;&nbsp;addCodingRegionFeaturesFromFFN&#8203;(java.io.File&nbsp;root,
                                                                                                                        java.io.File&nbsp;file,
                                                                                                                        java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;&nbsp;chromosomes)
                                                                                                                 throws java.lang.Exception</pre>
<div class="block">From a .ffn file in root/file.faa We had all CodingInfo into the ChromosomeSequence
 
 This method use my personal NCBIFastaHeaderParser</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>root</code> - </dd>
<dd><code>file</code> - </dd>
<dd><code>chromosomes</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="addNonCodingRegionFeaturesFromFRN(java.io.File,java.io.File,java.util.LinkedHashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addNonCodingRegionFeaturesFromFRN</h4>
<pre class="methodSignature">public static&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;&nbsp;addNonCodingRegionFeaturesFromFRN&#8203;(java.io.File&nbsp;root,
                                                                                                                           java.io.File&nbsp;file,
                                                                                                                           java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;&nbsp;chromosomes)
                                                                                                                    throws java.lang.Exception</pre>
<div class="block">From a .frn file in root/file.frn We add all Noncoding Info into the ChromosomeSequence
 
 This method use my personal NCBIFastaHeaderParser</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>root</code> - </dd>
<dd><code>file</code> - </dd>
<dd><code>chromosomes</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="updateInfoFromChromosome(java.util.LinkedHashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>updateInfoFromChromosome</h4>
<pre class="methodSignature">public static&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;&nbsp;updateInfoFromChromosome&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;&nbsp;chromosomes)</pre>
<div class="block">According to the info we found in Chromosomes, we update the info of the different components
 (coding, ncoding, proteins)</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>chromosomes</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="process(org.biojava3.core.sequence.template.CompoundSet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>process</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="FastaFileReader.html" title="type parameter in FastaFileReader">S</a>&gt;&nbsp;process&#8203;(<a href="../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;?&gt;&nbsp;compoundSet)
                                                          throws java.lang.Exception</pre>
<div class="block">The parsing is done in this method
 
 ******* There is a problem when the genes in the fasta file have the same Id, so I modify this
 method accordingly</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="saveProteins(java.util.ArrayList,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>saveProteins</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;saveProteins&#8203;(java.util.ArrayList&lt;<a href="../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;&nbsp;sequences,
                                java.lang.String&nbsp;fileName)</pre>
<div class="block">Save all amino acid sequence in a fasta file</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sequences</code> - </dd>
<dd><code>fileName</code> - </dd>
</dl>
</li>
</ul>
<a id="saveCDS(java.util.ArrayList,java.lang.String)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>saveCDS</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;saveCDS&#8203;(java.util.ArrayList&lt;<a href="../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;&nbsp;sequences,
                           java.lang.String&nbsp;fileName)</pre>
<div class="block">Save all amino acid sequence in a fasta file</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sequences</code> - </dd>
<dd><code>fileName</code> - </dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FastaFileReader.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-files/index-1.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
