<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 18:10:05 CEST 2019 -->
<title>FeatureList</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="FeatureList";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":41,"i9":10,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10,"i19":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"],32:["t6","Deprecated Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FeatureList.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">org.biojava3.genome.parsers.gff</a></div>
<h2 title="Class FeatureList" class="title">Class FeatureList</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>java.util.AbstractCollection&lt;E&gt;</li>
<li>
<ul class="inheritance">
<li>java.util.AbstractList&lt;E&gt;</li>
<li>
<ul class="inheritance">
<li>java.util.ArrayList&lt;<a href="FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&gt;</li>
<li>
<ul class="inheritance">
<li>org.biojava3.genome.parsers.gff.FeatureList</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><code>java.io.Serializable</code>, <code>java.lang.Cloneable</code>, <code>java.lang.Iterable&lt;<a href="FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&gt;</code>, <code>java.util.Collection&lt;<a href="FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&gt;</code>, <code>java.util.List&lt;<a href="FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&gt;</code>, <code>java.util.RandomAccess</code></dd>
</dl>
<hr>
<pre>public class <span class="typeNameLabel">FeatureList</span>
extends java.util.ArrayList&lt;<a href="FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&gt;</pre>
<div class="block">A list of FeatureI objects implemented using a Java ArrayList; corresponds to
 a GFF file. This class is implemented entirely using FeatureI objects, so
 everything here will work correctly if you choose to implement your own
 feature class -- there are no dependencies on JavaGene's native Feature
 class.</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Hanno Hinsch</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../../../serialized-form.html#org.biojava3.genome.parsers.gff.FeatureList">Serialized Form</a></dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">FeatureList</a></span>()</code></th>
<td class="colLast">
<div class="block">Construct an empty list.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.util.Collection)">FeatureList</a></span>&#8203;(java.util.Collection&lt;<a href="FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&gt;&nbsp;features)</code></th>
<td class="colLast">
<div class="block">Construct a new list containing the same features as the specified list.</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t6" class="tableTab"><span><a href="javascript:show(32);">Deprecated Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#add(java.util.Collection)">add</a></span>&#8203;(java.util.Collection&lt;<a href="FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&gt;&nbsp;list)</code></th>
<td class="colLast">
<div class="block">Add all features in the specified list or collection to this list.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#add(org.biojava3.genome.parsers.gff.FeatureI)">add</a></span>&#8203;(<a href="FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&nbsp;feature)</code></th>
<td class="colLast">
<div class="block">Add specified feature to the end of the list.</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>java.util.Collection&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#attributeValues(java.lang.String)">attributeValues</a></span>&#8203;(java.lang.String&nbsp;key)</code></th>
<td class="colLast">
<div class="block">Create a collection of the unique values for the specified key.</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code><a href="Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#bounds()">bounds</a></span>()</code></th>
<td class="colLast">
<div class="block">The union of all locations of all features in this list, mapped to the
 positive strand.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>java.util.Collection&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#groupValues()">groupValues</a></span>()</code></th>
<td class="colLast">
<div class="block">Create a collection of all unique group ids in the list, as defined by the
 group() method of the features.</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#hasAttribute(java.lang.String)">hasAttribute</a></span>&#8203;(java.lang.String&nbsp;key)</code></th>
<td class="colLast">
<div class="block">Check if any feature in list has the specified attribute key.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#hasAttribute(java.lang.String,java.lang.String)">hasAttribute</a></span>&#8203;(java.lang.String&nbsp;key,
            java.lang.String&nbsp;value)</code></th>
<td class="colLast">
<div class="block">Check if any feature in list has the specified attribute key/value pair.</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#hasGaps(int)">hasGaps</a></span>&#8203;(int&nbsp;gapLength)</code></th>
<td class="colLast">
<div class="block">Check size of gaps between successive features in list.</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#main(java.lang.String%5B%5D)">main</a></span>&#8203;(java.lang.String[]&nbsp;args)</code></th>
<td class="colLast">
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code><a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#omitOverlapping(java.lang.String,org.biojava3.genome.parsers.gff.Location,boolean)">omitOverlapping</a></span>&#8203;(java.lang.String&nbsp;seqname,
               <a href="Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a>&nbsp;location,
               boolean&nbsp;useBothStrands)</code></th>
<td class="colLast">
<div class="block">Create a list of all features that do not overlap the specified location on
 the specified sequence.</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code><a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#selectByAttribute(java.lang.String)">selectByAttribute</a></span>&#8203;(java.lang.String&nbsp;key)</code></th>
<td class="colLast">
<div class="block">Create a list of all features that include the specified attribute key.</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code><a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#selectByAttribute(java.lang.String,java.lang.String)">selectByAttribute</a></span>&#8203;(java.lang.String&nbsp;key,
                 java.lang.String&nbsp;value)</code></th>
<td class="colLast">
<div class="block">Create a list of all features that include the specified attribute key/value
 pair.</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code><a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#selectByGroup(java.lang.String)">selectByGroup</a></span>&#8203;(java.lang.String&nbsp;groupid)</code></th>
<td class="colLast">
<div class="block">Create a list of all features that have the specified group id, as defined by
 the group() method of the features.</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code><a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#selectByType(java.lang.String)">selectByType</a></span>&#8203;(java.lang.String&nbsp;type)</code></th>
<td class="colLast">
<div class="block">Create a list of all features that are of the specified type, as defined by
 the type() method of the features.</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code><a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#selectByUserData(java.lang.String)">selectByUserData</a></span>&#8203;(java.lang.String&nbsp;key)</code></th>
<td class="colLast">
<div class="block">Create a list of all features that include the specified key in their
 userMap().</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code><a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#selectByUserData(java.lang.String,java.lang.Object)">selectByUserData</a></span>&#8203;(java.lang.String&nbsp;key,
                java.lang.Object&nbsp;value)</code></th>
<td class="colLast">
<div class="block">Create a list of all features that include the specified key/value pair in
 their userMap().</div>
</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code><a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#selectOverlapping(java.lang.String,org.biojava3.genome.parsers.gff.Location,boolean)">selectOverlapping</a></span>&#8203;(java.lang.String&nbsp;seqname,
                 <a href="Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a>&nbsp;location,
                 boolean&nbsp;useBothStrands)</code></th>
<td class="colLast">
<div class="block">Create a list of all features that overlap the specified location on the
 specified sequence.</div>
</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code><a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#sortByStart()">sortByStart</a></span>()</code></th>
<td class="colLast">
<div class="block">Create a new list that is ordered by the starting index of the features'
 locations.</div>
</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#splice(org.biojava3.core.sequence.DNASequence)">splice</a></span>&#8203;(<a href="../../../core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Concatenate successive portions of the specified sequence using the feature
 locations in the list.</div>
</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#toString()">toString</a></span>()</code></th>
<td class="colLast">
<div class="block">Return a string representation of all features in this list.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.util.ArrayList">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.util.ArrayList</h3>
<code>add, addAll, addAll, clear, clone, contains, ensureCapacity, equals, forEach, get, hashCode, indexOf, isEmpty, iterator, lastIndexOf, listIterator, listIterator, remove, remove, removeAll, removeIf, replaceAll, retainAll, set, size, sort, spliterator, subList, toArray, toArray, trimToSize</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.util.AbstractCollection">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.util.AbstractCollection</h3>
<code>containsAll</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>getClass, notify, notifyAll, wait, wait, wait</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.util.Collection">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;java.util.Collection</h3>
<code>parallelStream, stream, toArray</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.util.List">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;java.util.List</h3>
<code>containsAll</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>FeatureList</h4>
<pre>public&nbsp;FeatureList()</pre>
<div class="block">Construct an empty list.</div>
</li>
</ul>
<a id="&lt;init&gt;(java.util.Collection)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>FeatureList</h4>
<pre>public&nbsp;FeatureList&#8203;(java.util.Collection&lt;<a href="FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&gt;&nbsp;features)</pre>
<div class="block">Construct a new list containing the same features as the specified list.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>features</code> - An existing list or collection of FeatureI objects.</dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="add(org.biojava3.genome.parsers.gff.FeatureI)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>add</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;add&#8203;(<a href="FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&nbsp;feature)</pre>
<div class="block">Add specified feature to the end of the list. Updates the bounding location
 of the feature list, if needed.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code>add</code>&nbsp;in interface&nbsp;<code>java.util.Collection&lt;<a href="FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&gt;</code></dd>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code>add</code>&nbsp;in interface&nbsp;<code>java.util.List&lt;<a href="FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&gt;</code></dd>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code>add</code>&nbsp;in class&nbsp;<code>java.util.ArrayList&lt;<a href="FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&gt;</code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>feature</code> - The FeatureI object to add.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>True if the feature was added.</dd>
</dl>
</li>
</ul>
<a id="add(java.util.Collection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>add</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;add&#8203;(java.util.Collection&lt;<a href="FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&gt;&nbsp;list)</pre>
<div class="block">Add all features in the specified list or collection to this list.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>list</code> - The collection of FeatureI objects.</dd>
</dl>
</li>
</ul>
<a id="bounds()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>bounds</h4>
<pre class="methodSignature">public&nbsp;<a href="Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a>&nbsp;bounds()</pre>
<div class="block">The union of all locations of all features in this list, mapped to the
 positive strand. If an added feature is on the negative strand, its positive
 strand image is added to the union. The bounding location is not updated when
 a feature is removed from the list, so it is not guaranteed to be the minimal
 bounding location.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>A location that is the union of all feature locations in the list.</dd>
</dl>
</li>
</ul>
<a id="hasGaps(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hasGaps</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;hasGaps&#8203;(int&nbsp;gapLength)</pre>
<div class="block">Check size of gaps between successive features in list. The features in the
 list are assumed to be appropriately ordered.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gapLength</code> - The minimum gap length to consider. Use a gapLength of 0 to
                  check if features are contiguous.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>True if list has any gaps equal to or greater than gapLength.</dd>
</dl>
</li>
</ul>
<a id="splice(org.biojava3.core.sequence.DNASequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>splice</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;splice&#8203;(<a href="../../../core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;sequence)</pre>
<div class="block">Concatenate successive portions of the specified sequence using the feature
 locations in the list. The list is assumed to be appropriately ordered.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sequence</code> - The source sequence from which portions should be selected.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>The spliced data.</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.IllegalStateException</code> - Out of order or overlapping FeatureI locations
                               detected.</dd>
</dl>
</li>
</ul>
<a id="groupValues()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>groupValues</h4>
<pre class="methodSignature">public&nbsp;java.util.Collection&lt;java.lang.String&gt;&nbsp;groupValues()</pre>
<div class="block">Create a collection of all unique group ids in the list, as defined by the
 group() method of the features. For example, if the features are from a GFF1
 file, then each group id identifies a particular gene, and this method
 returns a collection of all gene ids.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>A collection (suitable for iteration using Java's "for" loop) of all
         the group ids found in this list. The order of the values is
         undefined; it will not match the order of features in the list.</dd>
</dl>
</li>
</ul>
<a id="attributeValues(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>attributeValues</h4>
<pre class="methodSignature">public&nbsp;java.util.Collection&lt;java.lang.String&gt;&nbsp;attributeValues&#8203;(java.lang.String&nbsp;key)</pre>
<div class="block">Create a collection of the unique values for the specified key. Example: For
 GTF files, using the "gene_id" key will give the names of all the genes in
 this list.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>A collection (suitable for iteration using Java's "for" loop) of all
         the values found for this key. The order of the values is undefined;
         it will not match the order of features in the list.</dd>
</dl>
</li>
</ul>
<a id="selectByGroup(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>selectByGroup</h4>
<pre class="methodSignature">public&nbsp;<a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;selectByGroup&#8203;(java.lang.String&nbsp;groupid)</pre>
<div class="block">Create a list of all features that have the specified group id, as defined by
 the group() method of the features.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>groupid</code> - The group to match.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>A list of features having the specified group id.</dd>
</dl>
</li>
</ul>
<a id="selectByType(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>selectByType</h4>
<pre class="methodSignature">public&nbsp;<a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;selectByType&#8203;(java.lang.String&nbsp;type)</pre>
<div class="block">Create a list of all features that are of the specified type, as defined by
 the type() method of the features. This might be, for example, "exon" or
 "CDS".</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>type</code> - The type to match.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>A list of features of the specified type.</dd>
</dl>
</li>
</ul>
<a id="selectByAttribute(java.lang.String,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>selectByAttribute</h4>
<pre class="methodSignature">public&nbsp;<a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;selectByAttribute&#8203;(java.lang.String&nbsp;key,
                                     java.lang.String&nbsp;value)</pre>
<div class="block">Create a list of all features that include the specified attribute key/value
 pair.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>key</code> - The key to consider.</dd>
<dd><code>value</code> - The value to consider.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>A list of features that include the key/value pair.</dd>
</dl>
</li>
</ul>
<a id="selectByAttribute(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>selectByAttribute</h4>
<pre class="methodSignature">public&nbsp;<a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;selectByAttribute&#8203;(java.lang.String&nbsp;key)</pre>
<div class="block">Create a list of all features that include the specified attribute key.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>key</code> - The key to consider.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>A list of features that include the key.</dd>
</dl>
</li>
</ul>
<a id="selectByUserData(java.lang.String,java.lang.Object)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>selectByUserData</h4>
<pre class="methodSignature">public&nbsp;<a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;selectByUserData&#8203;(java.lang.String&nbsp;key,
                                    java.lang.Object&nbsp;value)</pre>
<div class="block">Create a list of all features that include the specified key/value pair in
 their userMap().</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>key</code> - The key to consider.</dd>
<dd><code>value</code> - The value to consider.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>A list of features that include the key/value pair.</dd>
</dl>
</li>
</ul>
<a id="selectByUserData(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>selectByUserData</h4>
<pre class="methodSignature">public&nbsp;<a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;selectByUserData&#8203;(java.lang.String&nbsp;key)</pre>
<div class="block">Create a list of all features that include the specified key in their
 userMap().</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>key</code> - The key to consider.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>A list of features that include the key.</dd>
</dl>
</li>
</ul>
<a id="selectOverlapping(java.lang.String,org.biojava3.genome.parsers.gff.Location,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>selectOverlapping</h4>
<pre class="methodSignature">public&nbsp;<a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;selectOverlapping&#8203;(java.lang.String&nbsp;seqname,
                                     <a href="Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a>&nbsp;location,
                                     boolean&nbsp;useBothStrands)
                              throws java.lang.Exception</pre>
<div class="block">Create a list of all features that overlap the specified location on the
 specified sequence.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>seqname</code> - The sequence name. Only features with this sequence
                       name will be checked for overlap.</dd>
<dd><code>location</code> - The location to check.</dd>
<dd><code>useBothStrands</code> - If true, locations are mapped to their positive strand
                       image before being checked for overlap. If false, only
                       features whose locations are on the same strand as the
                       specified location will be considered for inclusion.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>The new list of features that overlap the location.</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="omitOverlapping(java.lang.String,org.biojava3.genome.parsers.gff.Location,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>omitOverlapping</h4>
<pre class="methodSignature">public&nbsp;<a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;omitOverlapping&#8203;(java.lang.String&nbsp;seqname,
                                   <a href="Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a>&nbsp;location,
                                   boolean&nbsp;useBothStrands)</pre>
<div class="block">Create a list of all features that do not overlap the specified location on
 the specified sequence.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>seqname</code> - The sequence name. Only features with this sequence
                       name will be checked for overlap.</dd>
<dd><code>location</code> - The location to check.</dd>
<dd><code>useBothStrands</code> - If true, locations are mapped to their positive strand
                       image before being checked for overlap. If false, all
                       features whose locations are on the opposite strand
                       from the specified location will be considered
                       non-overlapping.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>The new list of features that do not overlap the location.</dd>
</dl>
</li>
</ul>
<a id="hasAttribute(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hasAttribute</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;hasAttribute&#8203;(java.lang.String&nbsp;key)</pre>
<div class="block">Check if any feature in list has the specified attribute key.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>key</code> - The attribute key to consider.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>True if at least one feature has the attribute key.</dd>
</dl>
</li>
</ul>
<a id="hasAttribute(java.lang.String,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hasAttribute</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;hasAttribute&#8203;(java.lang.String&nbsp;key,
                            java.lang.String&nbsp;value)</pre>
<div class="block">Check if any feature in list has the specified attribute key/value pair.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>key</code> - The attribute key to consider.</dd>
<dd><code>value</code> - The attribute value to consider.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>True if at least one feature has the key/value pair.</dd>
</dl>
</li>
</ul>
<a id="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;toString()</pre>
<div class="block">Return a string representation of all features in this list.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code>toString</code>&nbsp;in class&nbsp;<code>java.util.AbstractCollection&lt;<a href="FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&gt;</code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>A string.</dd>
</dl>
</li>
</ul>
<a id="sortByStart()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>sortByStart</h4>
<pre class="methodSignature">public&nbsp;<a href="FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a>&nbsp;sortByStart()</pre>
<div class="block">Create a new list that is ordered by the starting index of the features'
 locations. All locations must be on the same strand of the same sequence.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>An ordered list.</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.IndexOutOfBoundsException</code> - Cannot compare/sort features whose
                                   locations are on opposite strands, or whose
                                   seqnames differ.</dd>
</dl>
</li>
</ul>
<a id="main(java.lang.String[])">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>main</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;main&#8203;(java.lang.String[]&nbsp;args)</pre>
<div class="deprecationBlock"><span class="deprecatedLabel">Deprecated.</span></div>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FeatureList.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
