<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>TranscriptomesCreation</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="TranscriptomesCreation";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":9,"i1":9,"i2":9,"i3":9,"i4":9,"i5":9,"i6":9,"i7":9};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/TranscriptomesCreation.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.scripts.database</a></div>
<h2 title="Class TranscriptomesCreation" class="title">Class TranscriptomesCreation</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>bacnet.scripts.database.TranscriptomesCreation</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<pre>public class <span class="typeNameLabel">TranscriptomesCreation</span>
extends java.lang.Object</pre>
<div class="block">List of methods to add the different transcriptomic datasetsto your multi-omics website</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>UIBC</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">TranscriptomesCreation</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addTranscriptomeToDatabase(java.util.ArrayList,java.lang.String)">addTranscriptomeToDatabase</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;bioConds,
                          java.lang.String&nbsp;logs)</code></th>
<td class="colLast">
<div class="block">Convert all transcriptomics files<br>
 <br>
 - Convert all Seq data: RNASeq, RiboSeq, TSS, TermSeq<br>
 - Convert all GeneExpression and Tiling<br>
 - Convert all ExpressionMatrix<br></div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>private static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#convertExpressionMatrix(bacnet.datamodel.expdesign.Experiment)">convertExpressionMatrix</a></span>&#8203;(<a href="../../datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</code></th>
<td class="colLast">
<div class="block">Process unprocessed data and save them in Streaming folder, ExpressionMatrix of comparisons
 mainly</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createLogFCMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">createLogFCMatrix</a></span>&#8203;(<a href="../../datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                 <a href="../../datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create Database.logFCMatrix by regrouping every logFC values for genes, sRNAs, asRNAs, and
 cisRegs in every <code>BioCondition</code> given <code>Experiment</code><br></div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createLogfcTranscriptomeTable(java.util.ArrayList,java.lang.String,boolean)">createLogfcTranscriptomeTable</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;bioConds,
                             java.lang.String&nbsp;logs,
                             boolean&nbsp;varianceNorm)</code></th>
<td class="colLast">
<div class="block">Create LogFC summary matrix for transcriptomics datasets</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createStatMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">createStatMatrix</a></span>&#8203;(<a href="../../datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                <a href="../../datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></th>
<td class="colLast">
<div class="block">Create Database.statTable by calculating stat values for genes, sRNAs, and asRNAs in every
 BioCondition<br>
 For genes we use FDRBY calculated from GebneExpression array<br>
 For sRNAs and asRNAs we use TSTUDENTTILING for Tiling array</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;<a href="../../datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDataFromLab()">getDataFromLab</a></span>()</code></th>
<td class="colLast">
<div class="block">Get all GeneExpression from the lab</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>private static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#qualityControl(bacnet.datamodel.expdesign.Experiment)">qualityControl</a></span>&#8203;(<a href="../../datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</code></th>
<td class="colLast">
<div class="block">Look at the Expression of HouseKeeping genes<br></div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#varianceNormalization(bacnet.datamodel.expdesign.Experiment)">varianceNormalization</a></span>&#8203;(<a href="../../datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</code></th>
<td class="colLast">
<div class="block">Run some scripts to analyse the statistics of the transcriptomics data and performed
 VarianceNormalization:<br>
 It will reduce the total variance of each data to 1<br>
 For each column of the ExpressionMatrix, multiply each value by: Math.sqr(1/var(x))<br>
 So final variance of each column is equal to zero</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>TranscriptomesCreation</h4>
<pre>public&nbsp;TranscriptomesCreation()</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="addTranscriptomeToDatabase(java.util.ArrayList,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addTranscriptomeToDatabase</h4>
<pre class="methodSignature">public static&nbsp;java.lang.String&nbsp;addTranscriptomeToDatabase&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;bioConds,
                                                          java.lang.String&nbsp;logs)</pre>
<div class="block">Convert all transcriptomics files<br>
 <br>
 - Convert all Seq data: RNASeq, RiboSeq, TSS, TermSeq<br>
 - Convert all GeneExpression and Tiling<br>
 - Convert all ExpressionMatrix<br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioConds</code> - </dd>
<dd><code>logs</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>logs of the process</dd>
</dl>
</li>
</ul>
<a id="convertExpressionMatrix(bacnet.datamodel.expdesign.Experiment)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>convertExpressionMatrix</h4>
<pre class="methodSignature">private static&nbsp;void&nbsp;convertExpressionMatrix&#8203;(<a href="../../datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</pre>
<div class="block">Process unprocessed data and save them in Streaming folder, ExpressionMatrix of comparisons
 mainly</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>exp</code> - </dd>
</dl>
</li>
</ul>
<a id="createLogfcTranscriptomeTable(java.util.ArrayList,java.lang.String,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createLogfcTranscriptomeTable</h4>
<pre class="methodSignature">public static&nbsp;java.lang.String&nbsp;createLogfcTranscriptomeTable&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;bioConds,
                                                             java.lang.String&nbsp;logs,
                                                             boolean&nbsp;varianceNorm)</pre>
<div class="block">Create LogFC summary matrix for transcriptomics datasets</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioConds</code> - </dd>
<dd><code>logs</code> - </dd>
<dd><code>varianceNorm</code> - if variance normalization is needed</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the logs of the creation</dd>
</dl>
</li>
</ul>
<a id="varianceNormalization(bacnet.datamodel.expdesign.Experiment)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>varianceNormalization</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;varianceNormalization&#8203;(<a href="../../datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</pre>
<div class="block">Run some scripts to analyse the statistics of the transcriptomics data and performed
 VarianceNormalization:<br>
 It will reduce the total variance of each data to 1<br>
 For each column of the ExpressionMatrix, multiply each value by: Math.sqr(1/var(x))<br>
 So final variance of each column is equal to zero</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>exp</code> - </dd>
</dl>
</li>
</ul>
<a id="createLogFCMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createLogFCMatrix</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;createLogFCMatrix&#8203;(<a href="../../datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                                     <a href="../../datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</pre>
<div class="block">Create Database.logFCMatrix by regrouping every logFC values for genes, sRNAs, asRNAs, and
 cisRegs in every <code>BioCondition</code> given <code>Experiment</code><br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>exp</code> - list of BioCondition</dd>
<dd><code>genome</code> - current genome with list of genes, sRNAs, asRNAs and cisRegs</dd>
</dl>
</li>
</ul>
<a id="createStatMatrix(bacnet.datamodel.expdesign.Experiment,bacnet.datamodel.sequence.Genome)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createStatMatrix</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;createStatMatrix&#8203;(<a href="../../datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp,
                                    <a href="../../datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</pre>
<div class="block">Create Database.statTable by calculating stat values for genes, sRNAs, and asRNAs in every
 BioCondition<br>
 For genes we use FDRBY calculated from GebneExpression array<br>
 For sRNAs and asRNAs we use TSTUDENTTILING for Tiling array</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>exp</code> - </dd>
<dd><code>genome</code> - </dd>
</dl>
</li>
</ul>
<a id="getDataFromLab()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDataFromLab</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;&nbsp;getDataFromLab()</pre>
<div class="block">Get all GeneExpression from the lab</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>list of <a href="../../datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset"><code>GeneExpression</code></a></dd>
</dl>
</li>
</ul>
<a id="qualityControl(bacnet.datamodel.expdesign.Experiment)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>qualityControl</h4>
<pre class="methodSignature">private static&nbsp;void&nbsp;qualityControl&#8203;(<a href="../../datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a>&nbsp;exp)</pre>
<div class="block">Look at the Expression of HouseKeeping genes<br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>exp</code> - </dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/TranscriptomesCreation.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
