<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>BioCondition</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="BioCondition";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":9,"i3":9,"i4":9,"i5":9,"i6":10,"i7":10,"i8":10,"i9":9,"i10":9,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10,"i19":10,"i20":10,"i21":10,"i22":10,"i23":10,"i24":10,"i25":10,"i26":10,"i27":10,"i28":10,"i29":10,"i30":10,"i31":10,"i32":10,"i33":9,"i34":10,"i35":10,"i36":10,"i37":10,"i38":10,"i39":10,"i40":10,"i41":10,"i42":9,"i43":10,"i44":10,"i45":10,"i46":9,"i47":9,"i48":10,"i49":10,"i50":10,"i51":10,"i52":10,"i53":10,"i54":10,"i55":10,"i56":10,"i57":10,"i58":10,"i59":10,"i60":10,"i61":10,"i62":10,"i63":10,"i64":10,"i65":10,"i66":10,"i67":10,"i68":10,"i69":10,"i70":10,"i71":10,"i72":10,"i73":10,"i74":10,"i75":10,"i76":10,"i77":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/BioCondition.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.expdesign</a></div>
<h2 title="Class BioCondition" class="title">Class BioCondition</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>bacnet.datamodel.expdesign.BioCondition</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><code>java.io.Serializable</code></dd>
</dl>
<hr>
<pre>public class <span class="typeNameLabel">BioCondition</span>
extends java.lang.Object
implements java.io.Serializable</pre>
<div class="block">This class is used for the different bioconds declared in the experiment
 
 It contains the biocond name, and all the related data.
 
 IMPORTANT, MAJ 04102011 : WT data are only used as indication on experimental design, in all
 analysis if we want WT we need to add it in the Experiment</div>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../serialized-form.html#bacnet.datamodel.expdesign.BioCondition">Serialized Form</a></dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#antiComparisons">antiComparisons</a></span></code></th>
<td class="colLast">
<div class="block">List of Data for which the BioCondition is the reference bioCodntion for the comparison:
 BioCondComparison vs This Biocondition</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#arrayExpressId">arrayExpressId</a></span></code></th>
<td class="colLast">
<div class="block">ArrayExpressId if this data is public</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#arrayExpressTechnoId">arrayExpressTechnoId</a></span></code></th>
<td class="colLast">
<div class="block">ArrayExpressId of the technology used</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#comment">comment</a></span></code></th>
<td class="colLast">
<div class="block">Different comment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#comparisons">comparisons</a></span></code></th>
<td class="colLast">
<div class="block">List of Data for which we have comparison data: This Biocondition vs BioCondComparison</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#date">date</a></span></code></th>
<td class="colLast">
<div class="block">Date of creation or data of publication</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#experienceNb">experienceNb</a></span></code></th>
<td class="colLast">
<div class="block">For experiment from Cossart lab, ID of the experiment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;<a href="../dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#geneExprs">geneExprs</a></span></code></th>
<td class="colLast">
<div class="block">All Listeria GeneExpression arrays associated to this BioCondition</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#genomeName">genomeName</a></span></code></th>
<td class="colLast">
<div class="block">Reference genome ID to used for this BioCondition</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#genomeUsed">genomeUsed</a></span></code></th>
<td class="colLast">
<div class="block">Actual strain used to create this BioCondition, might be different from genomeName</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#growth">growth</a></span></code></th>
<td class="colLast">
<div class="block">Growth conditions are specified here</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#linkedBioCondition">linkedBioCondition</a></span></code></th>
<td class="colLast">
<div class="block">Data associated to this BioCondition to use in the case of timepoint experiment, or same mutant
 in different condition</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#localization">localization</a></span></code></th>
<td class="colLast">
<div class="block">If it is a proteome, this parameters tells you where in the bacteria cell the proteins have been
 extracted</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;<a href="../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#matrices">matrices</a></span></code></th>
<td class="colLast">
<div class="block">All data in <code>ExpressioMatrix</code> associated to this BioCondition</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#media">media</a></span></code></th>
<td class="colLast">
<div class="block">Growth media is specified here</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#mediaGrowthProperties">mediaGrowthProperties</a></span></code></th>
<td class="colLast">
<div class="block">Quantity/properties of drugs added to the media if specified: quantity of glucose, type of Broth</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#mutant">mutant</a></span></code></th>
<td class="colLast">
<div class="block">If it is a mutant, the locus is indicated (example: prfA, lmo0200 or rliA), leave empty otherwise</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#name">name</a></span></code></th>
<td class="colLast">
<div class="block">Name of the BioCondition</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;<a href="../dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#ngss">ngss</a></span></code></th>
<td class="colLast">
<div class="block">All RNASeq data associated to this BioCondition</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#noData">noData</a></span></code></th>
<td class="colLast">
<div class="block">True if no data are associated to this BioCondition<br>
 It is used principaly to indicate that the BioCondition is a reference condition for another
 BioCondition, and thus no data are available for it, as the data are in the other BioCondition</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#nTerms">nTerms</a></span></code></th>
<td class="colLast">
<div class="block">All NTermData data associated to this BioCondition</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;<a href="../dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#proteomes">proteomes</a></span></code></th>
<td class="colLast">
<div class="block">All ProteomicsData data associated to this BioCondition</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#reference">reference</a></span></code></th>
<td class="colLast">
<div class="block">Article in which data is published</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#SEPARATOR">SEPARATOR</a></span></code></th>
<td class="colLast">
<div class="block">BioConditoon separator for comparisons<br>
 equal to " vs "</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private static long</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#serialVersionUID">serialVersionUID</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#studyName">studyName</a></span></code></th>
<td class="colLast">
<div class="block">Name of the study from which the data is extracted</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#temperature">temperature</a></span></code></th>
<td class="colLast">
<div class="block">Temperature is specified here</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;<a href="../dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#tilings">tilings</a></span></code></th>
<td class="colLast">
<div class="block">All Listeria Tiling arrays associated to this BioCondition</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#time">time</a></span></code></th>
<td class="colLast">
<div class="block">time point if necessary</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.TreeSet&lt;<a href="../dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#typeDataContained">typeDataContained</a></span></code></th>
<td class="colLast">
<div class="block">Type of bioCondition : TSS only, TIS only, RNASeq + Tiling + GeneExpr, RNASeq only ...</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#wildType">wildType</a></span></code></th>
<td class="colLast">
<div class="block">True if this BioCondition is a wild-type strain, to be used as a reference condition</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">BioCondition</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.lang.String)">BioCondition</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code><a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#compare(bacnet.datamodel.expdesign.BioCondition,boolean)">compare</a></span>&#8203;(<a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&nbsp;bioCond2,
       boolean&nbsp;calcData)</code></th>
<td class="colLast">
<div class="block">Create a BioCondition containing the different data for comparing this BioCondition to bioCond2</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#containTranscriptomes()">containTranscriptomes</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAllBioConditionNames()">getAllBioConditionNames</a></span>()</code></th>
<td class="colLast">
<div class="block">Get all available <code>BioCondition</code> names by reading the text file
 <code>Experiment.INCLUDE_COND_PATH</code></div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;<a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAllBioConditions()">getAllBioConditions</a></span>()</code></th>
<td class="colLast">
<div class="block">Return all BioCondition available</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;<a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAllPublishedBioConditions()">getAllPublishedBioConditions</a></span>()</code></th>
<td class="colLast">
<div class="block">Return all BioCondition available</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;<a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAllUnPublishedBioConditions()">getAllUnPublishedBioConditions</a></span>()</code></th>
<td class="colLast">
<div class="block">Return all BioCondition available</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAntiComparisons()">getAntiComparisons</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getArrayExpressId()">getArrayExpressId</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getArrayExpressTechnoId()">getArrayExpressTechnoId</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>static <a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getBioCondition(java.lang.String)">getBioCondition</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">
<div class="block">From a biological condition name load the corresponding BioCondition object</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>static <a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getBioCondition(java.lang.String,boolean)">getBioCondition</a></span>&#8203;(java.lang.String&nbsp;bioCondName,
               boolean&nbsp;modelProvider)</code></th>
<td class="colLast">
<div class="block">From a biological condition name, get from the ModelProvider the corresponding BioCondition
 object<br>
 If not available it will load them for the hard drive</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getComment()">getComment</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getComparisonDataNames()">getComparisonDataNames</a></span>()</code></th>
<td class="colLast">
<div class="block">Get every names of the Comparison which can be found:<br>
 Tiling arrays
 Gene Expression arrays
 ExpressionMatrix</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getComparisonNames()">getComparisonNames</a></span>()</code></th>
<td class="colLast">
<div class="block">Return a list of all Comparisons: bioCondName vs comparisons.get(i)</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getComparisons()">getComparisons</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDate()">getDate</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getExperienceNb()">getExperienceNb</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getExpressionData()">getExpressionData</a></span>()</code></th>
<td class="colLast">
<div class="block">Put all ExpressionData in a list</div>
</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGeneExprs()">getGeneExprs</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code><a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGenome()">getGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Return corresponding Genome given by this.genomeName<br>
 Load it if necessary</div>
</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGenomeName()">getGenomeName</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGenomeUsed()">getGenomeUsed</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGrowth()">getGrowth</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getLinkedBioCondition()">getLinkedBioCondition</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getLocalization()">getLocalization</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMatrices()">getMatrices</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMedia()">getMedia</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i27" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMediaGrowthProperties()">getMediaGrowthProperties</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i28" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMutant()">getMutant</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i29" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getName()">getName</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i30" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getNGSSeqs()">getNGSSeqs</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i31" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getnTerms()">getnTerms</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i32" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getOmicsData()">getOmicsData</a></span>()</code></th>
<td class="colLast">
<div class="block">Put all OmicsData in a list</div>
</td>
</tr>
<tr id="i33" class="rowColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteomeGenomes()">getProteomeGenomes</a></span>()</code></th>
<td class="colLast">
<div class="block">Get all genomes associated to a proteomics dataset</div>
</td>
</tr>
<tr id="i34" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteomes()">getProteomes</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i35" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteomicsData()">getProteomicsData</a></span>()</code></th>
<td class="colLast">
<div class="block">Put all OmicsData in a list</div>
</td>
</tr>
<tr id="i36" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getReference()">getReference</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i37" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getStudyName()">getStudyName</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i38" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTemperature()">getTemperature</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i39" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTilings()">getTilings</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i40" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTime()">getTime</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i41" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTranscriptomesData()">getTranscriptomesData</a></span>()</code></th>
<td class="colLast">
<div class="block">Put all TranscriptomeData in a list</div>
</td>
</tr>
<tr id="i42" class="altColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTranscriptomesGenomes()">getTranscriptomesGenomes</a></span>()</code></th>
<td class="colLast">
<div class="block">Get all genomes associated to a transcriptomes dataset</div>
</td>
</tr>
<tr id="i43" class="rowColor">
<td class="colFirst"><code>java.util.TreeSet&lt;<a href="../dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTypeDataContained()">getTypeDataContained</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i44" class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isNoData()">isNoData</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i45" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isWildType()">isWildType</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i46" class="altColor">
<td class="colFirst"><code>static <a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#load(java.lang.String)">load</a></span>&#8203;(java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</td>
</tr>
<tr id="i47" class="rowColor">
<td class="colFirst"><code>static java.lang.String[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#parseName(java.lang.String)">parseName</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Parse a Comparison name, extracting leftBioCondition and rightBiocondition</div>
</td>
</tr>
<tr id="i48" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#save(java.lang.String)">save</a></span>&#8203;(java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Prompt the user for a filename, and save the scribble in that file.</div>
</td>
</tr>
<tr id="i49" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setAntiComparisons(java.util.ArrayList)">setAntiComparisons</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;antiComparisons)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i50" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setArrayExpressId(java.lang.String)">setArrayExpressId</a></span>&#8203;(java.lang.String&nbsp;arrayExpressId)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i51" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setArrayExpressTechnoId(java.lang.String)">setArrayExpressTechnoId</a></span>&#8203;(java.lang.String&nbsp;arrayExpressTechnoId)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i52" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setComment(java.lang.String)">setComment</a></span>&#8203;(java.lang.String&nbsp;comment)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i53" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setComparisons(java.util.ArrayList)">setComparisons</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;comparisons)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i54" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDate(java.lang.String)">setDate</a></span>&#8203;(java.lang.String&nbsp;date)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i55" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setExperienceNb(int)">setExperienceNb</a></span>&#8203;(int&nbsp;experienceNb)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i56" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGeneExprs(java.util.ArrayList)">setGeneExprs</a></span>&#8203;(java.util.ArrayList&lt;<a href="../dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;&nbsp;geneExprs)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i57" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGenomeName(java.lang.String)">setGenomeName</a></span>&#8203;(java.lang.String&nbsp;genomeName)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i58" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGenomeUsed(java.lang.String)">setGenomeUsed</a></span>&#8203;(java.lang.String&nbsp;genomeUsed)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i59" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGrowth(java.util.ArrayList)">setGrowth</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;growth)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i60" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setLinkedBioCondition(java.util.ArrayList)">setLinkedBioCondition</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;linkedBioCondition)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i61" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setLocalization(java.util.ArrayList)">setLocalization</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;localization)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i62" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setMatrices(java.util.ArrayList)">setMatrices</a></span>&#8203;(java.util.ArrayList&lt;<a href="../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&gt;&nbsp;matrices)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i63" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setMedia(java.util.ArrayList)">setMedia</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;media)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i64" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setMediaGrowthProperties(java.util.ArrayList)">setMediaGrowthProperties</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;mediaGrowthProperties)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i65" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setMutant(java.util.ArrayList)">setMutant</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;mutant)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i66" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setName(java.lang.String)">setName</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i67" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setNGSSeqs(java.util.ArrayList)">setNGSSeqs</a></span>&#8203;(java.util.ArrayList&lt;<a href="../dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&gt;&nbsp;rnaSeqs)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i68" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setNoData(boolean)">setNoData</a></span>&#8203;(boolean&nbsp;noData)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i69" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setnTerms(java.util.ArrayList)">setnTerms</a></span>&#8203;(java.util.ArrayList&lt;<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&gt;&nbsp;nTerms)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i70" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setProteomes(java.util.ArrayList)">setProteomes</a></span>&#8203;(java.util.ArrayList&lt;<a href="../dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a>&gt;&nbsp;proteomes)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i71" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setReference(java.lang.String)">setReference</a></span>&#8203;(java.lang.String&nbsp;reference)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i72" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setStudyName(java.lang.String)">setStudyName</a></span>&#8203;(java.lang.String&nbsp;studyName)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i73" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setTemperature(java.lang.String)">setTemperature</a></span>&#8203;(java.lang.String&nbsp;temperature)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i74" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setTilings(java.util.ArrayList)">setTilings</a></span>&#8203;(java.util.ArrayList&lt;<a href="../dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a>&gt;&nbsp;tilings)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i75" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setTime(java.lang.String)">setTime</a></span>&#8203;(java.lang.String&nbsp;time)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i76" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setTypeDataContained(java.util.TreeSet)">setTypeDataContained</a></span>&#8203;(java.util.TreeSet&lt;<a href="../dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a>&gt;&nbsp;typeDataContained)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i77" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setWildType(boolean)">setWildType</a></span>&#8203;(boolean&nbsp;wildType)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="serialVersionUID">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>serialVersionUID</h4>
<pre>private static final&nbsp;long serialVersionUID</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../constant-values.html#bacnet.datamodel.expdesign.BioCondition.serialVersionUID">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a id="SEPARATOR">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SEPARATOR</h4>
<pre>public static&nbsp;java.lang.String SEPARATOR</pre>
<div class="block">BioConditoon separator for comparisons<br>
 equal to " vs "</div>
</li>
</ul>
<a id="name">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>name</h4>
<pre>private&nbsp;java.lang.String name</pre>
<div class="block">Name of the BioCondition</div>
</li>
</ul>
<a id="genomeName">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>genomeName</h4>
<pre>private&nbsp;java.lang.String genomeName</pre>
<div class="block">Reference genome ID to used for this BioCondition</div>
</li>
</ul>
<a id="genomeUsed">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>genomeUsed</h4>
<pre>private&nbsp;java.lang.String genomeUsed</pre>
<div class="block">Actual strain used to create this BioCondition, might be different from genomeName</div>
</li>
</ul>
<a id="typeDataContained">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>typeDataContained</h4>
<pre>private&nbsp;java.util.TreeSet&lt;<a href="../dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a>&gt; typeDataContained</pre>
<div class="block">Type of bioCondition : TSS only, TIS only, RNASeq + Tiling + GeneExpr, RNASeq only ...</div>
</li>
</ul>
<a id="date">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>date</h4>
<pre>private&nbsp;java.lang.String date</pre>
<div class="block">Date of creation or data of publication</div>
</li>
</ul>
<a id="experienceNb">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>experienceNb</h4>
<pre>private&nbsp;int experienceNb</pre>
<div class="block">For experiment from Cossart lab, ID of the experiment</div>
</li>
</ul>
<a id="wildType">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>wildType</h4>
<pre>private&nbsp;boolean wildType</pre>
<div class="block">True if this BioCondition is a wild-type strain, to be used as a reference condition</div>
</li>
</ul>
<a id="comment">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>comment</h4>
<pre>private&nbsp;java.lang.String comment</pre>
<div class="block">Different comment</div>
</li>
</ul>
<a id="geneExprs">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>geneExprs</h4>
<pre>private&nbsp;java.util.ArrayList&lt;<a href="../dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt; geneExprs</pre>
<div class="block">All Listeria GeneExpression arrays associated to this BioCondition</div>
</li>
</ul>
<a id="tilings">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>tilings</h4>
<pre>private&nbsp;java.util.ArrayList&lt;<a href="../dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a>&gt; tilings</pre>
<div class="block">All Listeria Tiling arrays associated to this BioCondition</div>
</li>
</ul>
<a id="ngss">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>ngss</h4>
<pre>private&nbsp;java.util.ArrayList&lt;<a href="../dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&gt; ngss</pre>
<div class="block">All RNASeq data associated to this BioCondition</div>
</li>
</ul>
<a id="nTerms">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>nTerms</h4>
<pre>private&nbsp;java.util.ArrayList&lt;<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&gt; nTerms</pre>
<div class="block">All NTermData data associated to this BioCondition</div>
</li>
</ul>
<a id="proteomes">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>proteomes</h4>
<pre>private&nbsp;java.util.ArrayList&lt;<a href="../dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a>&gt; proteomes</pre>
<div class="block">All ProteomicsData data associated to this BioCondition</div>
</li>
</ul>
<a id="matrices">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>matrices</h4>
<pre>private&nbsp;java.util.ArrayList&lt;<a href="../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&gt; matrices</pre>
<div class="block">All data in <code>ExpressioMatrix</code> associated to this BioCondition</div>
</li>
</ul>
<a id="linkedBioCondition">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>linkedBioCondition</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.String&gt; linkedBioCondition</pre>
<div class="block">Data associated to this BioCondition to use in the case of timepoint experiment, or same mutant
 in different condition</div>
</li>
</ul>
<a id="arrayExpressId">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>arrayExpressId</h4>
<pre>private&nbsp;java.lang.String arrayExpressId</pre>
<div class="block">ArrayExpressId if this data is public</div>
</li>
</ul>
<a id="studyName">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>studyName</h4>
<pre>private&nbsp;java.lang.String studyName</pre>
<div class="block">Name of the study from which the data is extracted</div>
</li>
</ul>
<a id="arrayExpressTechnoId">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>arrayExpressTechnoId</h4>
<pre>private&nbsp;java.lang.String arrayExpressTechnoId</pre>
<div class="block">ArrayExpressId of the technology used</div>
</li>
</ul>
<a id="mutant">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>mutant</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.String&gt; mutant</pre>
<div class="block">If it is a mutant, the locus is indicated (example: prfA, lmo0200 or rliA), leave empty otherwise</div>
</li>
</ul>
<a id="growth">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>growth</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.String&gt; growth</pre>
<div class="block">Growth conditions are specified here</div>
</li>
</ul>
<a id="media">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>media</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.String&gt; media</pre>
<div class="block">Growth media is specified here</div>
</li>
</ul>
<a id="temperature">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>temperature</h4>
<pre>private&nbsp;java.lang.String temperature</pre>
<div class="block">Temperature is specified here</div>
</li>
</ul>
<a id="time">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>time</h4>
<pre>private&nbsp;java.lang.String time</pre>
<div class="block">time point if necessary</div>
</li>
</ul>
<a id="reference">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reference</h4>
<pre>private&nbsp;java.lang.String reference</pre>
<div class="block">Article in which data is published</div>
</li>
</ul>
<a id="mediaGrowthProperties">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>mediaGrowthProperties</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.String&gt; mediaGrowthProperties</pre>
<div class="block">Quantity/properties of drugs added to the media if specified: quantity of glucose, type of Broth</div>
</li>
</ul>
<a id="localization">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>localization</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.String&gt; localization</pre>
<div class="block">If it is a proteome, this parameters tells you where in the bacteria cell the proteins have been
 extracted</div>
</li>
</ul>
<a id="comparisons">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>comparisons</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.String&gt; comparisons</pre>
<div class="block">List of Data for which we have comparison data: This Biocondition vs BioCondComparison</div>
</li>
</ul>
<a id="antiComparisons">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>antiComparisons</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.String&gt; antiComparisons</pre>
<div class="block">List of Data for which the BioCondition is the reference bioCodntion for the comparison:
 BioCondComparison vs This Biocondition</div>
</li>
</ul>
<a id="noData">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>noData</h4>
<pre>private&nbsp;boolean noData</pre>
<div class="block">True if no data are associated to this BioCondition<br>
 It is used principaly to indicate that the BioCondition is a reference condition for another
 BioCondition, and thus no data are available for it, as the data are in the other BioCondition</div>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>BioCondition</h4>
<pre>public&nbsp;BioCondition()</pre>
</li>
</ul>
<a id="&lt;init&gt;(java.lang.String)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>BioCondition</h4>
<pre>public&nbsp;BioCondition&#8203;(java.lang.String&nbsp;bioCondName)</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="containTranscriptomes()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>containTranscriptomes</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;containTranscriptomes()</pre>
</li>
</ul>
<a id="getExpressionData()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getExpressionData</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;getExpressionData()</pre>
<div class="block">Put all ExpressionData in a list</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getTranscriptomesData()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTranscriptomesData</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a>&gt;&nbsp;getTranscriptomesData()</pre>
<div class="block">Put all TranscriptomeData in a list</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getTranscriptomesGenomes()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTranscriptomesGenomes</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getTranscriptomesGenomes()</pre>
<div class="block">Get all genomes associated to a transcriptomes dataset</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getOmicsData()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getOmicsData</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a>&gt;&nbsp;getOmicsData()</pre>
<div class="block">Put all OmicsData in a list</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getProteomicsData()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteomicsData</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a>&gt;&nbsp;getProteomicsData()</pre>
<div class="block">Put all OmicsData in a list</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getProteomeGenomes()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteomeGenomes</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getProteomeGenomes()</pre>
<div class="block">Get all genomes associated to a proteomics dataset</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getComparisonDataNames()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getComparisonDataNames</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getComparisonDataNames()</pre>
<div class="block">Get every names of the Comparison which can be found:<br>
 <li>Tiling arrays
 <li>Gene Expression arrays
 <li>ExpressionMatrix</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a list of all the data</dd>
</dl>
</li>
</ul>
<a id="getGenome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenome</h4>
<pre class="methodSignature">public&nbsp;<a href="../sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;getGenome()</pre>
<div class="block">Return corresponding Genome given by this.genomeName<br>
 Load it if necessary</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getComparisonNames()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getComparisonNames</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getComparisonNames()</pre>
<div class="block">Return a list of all Comparisons: bioCondName vs comparisons.get(i)</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="compare(bacnet.datamodel.expdesign.BioCondition,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>compare</h4>
<pre class="methodSignature">public&nbsp;<a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&nbsp;compare&#8203;(<a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&nbsp;bioCond2,
                            boolean&nbsp;calcData)</pre>
<div class="block">Create a BioCondition containing the different data for comparing this BioCondition to bioCond2</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCond2</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.io.IOException</code></dd>
</dl>
</li>
</ul>
<a id="getBioCondition(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBioCondition</h4>
<pre class="methodSignature">public static&nbsp;<a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&nbsp;getBioCondition&#8203;(java.lang.String&nbsp;bioCondName)</pre>
<div class="block">From a biological condition name load the corresponding BioCondition object</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getBioCondition(java.lang.String,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBioCondition</h4>
<pre class="methodSignature">public static&nbsp;<a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&nbsp;getBioCondition&#8203;(java.lang.String&nbsp;bioCondName,
                                           boolean&nbsp;modelProvider)</pre>
<div class="block">From a biological condition name, get from the ModelProvider the corresponding BioCondition
 object<br>
 If not available it will load them for the hard drive</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dd><code>true</code> - if you want to get BioCondition from pre-loaded BioCondition</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="parseName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>parseName</h4>
<pre class="methodSignature">public static&nbsp;java.lang.String[]&nbsp;parseName&#8203;(java.lang.String&nbsp;name)</pre>
<div class="block">Parse a Comparison name, extracting leftBioCondition and rightBiocondition</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>name</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>{leftBioCond,rightBioCond}</dd>
</dl>
</li>
</ul>
<a id="getAllBioConditionNames()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAllBioConditionNames</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getAllBioConditionNames()</pre>
<div class="block">Get all available <code>BioCondition</code> names by reading the text file
 <code>Experiment.INCLUDE_COND_PATH</code></div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getAllBioConditions()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAllBioConditions</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;<a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&gt;&nbsp;getAllBioConditions()</pre>
<div class="block">Return all BioCondition available</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getAllPublishedBioConditions()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAllPublishedBioConditions</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;<a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&gt;&nbsp;getAllPublishedBioConditions()</pre>
<div class="block">Return all BioCondition available</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getAllUnPublishedBioConditions()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAllUnPublishedBioConditions</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;<a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&gt;&nbsp;getAllUnPublishedBioConditions()</pre>
<div class="block">Return all BioCondition available</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="load(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>load</h4>
<pre class="methodSignature">public static&nbsp;<a href="BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&nbsp;load&#8203;(java.lang.String&nbsp;fileName)</pre>
<div class="block">Read compressed, serialized data with a FileInputStream. Uncompress that data with a
 GZIPInputStream. Deserialize the vector of lines with a ObjectInputStream. Replace current data
 with new data, and redraw everything.</div>
</li>
</ul>
<a id="save(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>save</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;save&#8203;(java.lang.String&nbsp;fileName)</pre>
<div class="block">Prompt the user for a filename, and save the scribble in that file. Serialize the vector of lines
 with an ObjectOutputStream. Compress the serialized objects with a GZIPOutputStream. Write the
 compressed, serialized data to a file with a FileOutputStream. Don't forget to flush and close
 the stream.</div>
</li>
</ul>
<a id="getName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getName</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getName()</pre>
</li>
</ul>
<a id="setName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setName</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setName&#8203;(java.lang.String&nbsp;name)</pre>
</li>
</ul>
<a id="getGenomeName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomeName</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getGenomeName()</pre>
</li>
</ul>
<a id="setGenomeName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGenomeName</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGenomeName&#8203;(java.lang.String&nbsp;genomeName)</pre>
</li>
</ul>
<a id="getDate()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDate</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getDate()</pre>
</li>
</ul>
<a id="setDate(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDate</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDate&#8203;(java.lang.String&nbsp;date)</pre>
</li>
</ul>
<a id="getExperienceNb()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getExperienceNb</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getExperienceNb()</pre>
</li>
</ul>
<a id="setExperienceNb(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setExperienceNb</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setExperienceNb&#8203;(int&nbsp;experienceNb)</pre>
</li>
</ul>
<a id="isWildType()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isWildType</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;isWildType()</pre>
</li>
</ul>
<a id="setWildType(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setWildType</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setWildType&#8203;(boolean&nbsp;wildType)</pre>
</li>
</ul>
<a id="getComment()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getComment</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getComment()</pre>
</li>
</ul>
<a id="setComment(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setComment</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setComment&#8203;(java.lang.String&nbsp;comment)</pre>
</li>
</ul>
<a id="getGeneExprs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneExprs</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;&nbsp;getGeneExprs()</pre>
</li>
</ul>
<a id="setGeneExprs(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGeneExprs</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGeneExprs&#8203;(java.util.ArrayList&lt;<a href="../dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;&nbsp;geneExprs)</pre>
</li>
</ul>
<a id="getTilings()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTilings</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a>&gt;&nbsp;getTilings()</pre>
</li>
</ul>
<a id="setTilings(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setTilings</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setTilings&#8203;(java.util.ArrayList&lt;<a href="../dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a>&gt;&nbsp;tilings)</pre>
</li>
</ul>
<a id="getMatrices()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMatrices</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&gt;&nbsp;getMatrices()</pre>
</li>
</ul>
<a id="setMatrices(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMatrices</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setMatrices&#8203;(java.util.ArrayList&lt;<a href="../dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&gt;&nbsp;matrices)</pre>
</li>
</ul>
<a id="getArrayExpressId()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getArrayExpressId</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getArrayExpressId()</pre>
</li>
</ul>
<a id="setArrayExpressId(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setArrayExpressId</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setArrayExpressId&#8203;(java.lang.String&nbsp;arrayExpressId)</pre>
</li>
</ul>
<a id="getArrayExpressTechnoId()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getArrayExpressTechnoId</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getArrayExpressTechnoId()</pre>
</li>
</ul>
<a id="isNoData()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isNoData</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;isNoData()</pre>
</li>
</ul>
<a id="setNoData(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setNoData</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setNoData&#8203;(boolean&nbsp;noData)</pre>
</li>
</ul>
<a id="setArrayExpressTechnoId(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setArrayExpressTechnoId</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setArrayExpressTechnoId&#8203;(java.lang.String&nbsp;arrayExpressTechnoId)</pre>
</li>
</ul>
<a id="getGenomeUsed()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomeUsed</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getGenomeUsed()</pre>
</li>
</ul>
<a id="setGenomeUsed(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGenomeUsed</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGenomeUsed&#8203;(java.lang.String&nbsp;genomeUsed)</pre>
</li>
</ul>
<a id="getMutant()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMutant</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getMutant()</pre>
</li>
</ul>
<a id="setMutant(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMutant</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setMutant&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;mutant)</pre>
</li>
</ul>
<a id="getMediaGrowthProperties()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMediaGrowthProperties</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getMediaGrowthProperties()</pre>
</li>
</ul>
<a id="setMediaGrowthProperties(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMediaGrowthProperties</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setMediaGrowthProperties&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;mediaGrowthProperties)</pre>
</li>
</ul>
<a id="getGrowth()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGrowth</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getGrowth()</pre>
</li>
</ul>
<a id="setGrowth(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGrowth</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGrowth&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;growth)</pre>
</li>
</ul>
<a id="getTemperature()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTemperature</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getTemperature()</pre>
</li>
</ul>
<a id="setTemperature(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setTemperature</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setTemperature&#8203;(java.lang.String&nbsp;temperature)</pre>
</li>
</ul>
<a id="getTime()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTime</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getTime()</pre>
</li>
</ul>
<a id="setTime(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setTime</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setTime&#8203;(java.lang.String&nbsp;time)</pre>
</li>
</ul>
<a id="getMedia()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMedia</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getMedia()</pre>
</li>
</ul>
<a id="setMedia(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMedia</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setMedia&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;media)</pre>
</li>
</ul>
<a id="getLinkedBioCondition()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLinkedBioCondition</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getLinkedBioCondition()</pre>
</li>
</ul>
<a id="setLinkedBioCondition(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setLinkedBioCondition</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setLinkedBioCondition&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;linkedBioCondition)</pre>
</li>
</ul>
<a id="getComparisons()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getComparisons</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getComparisons()</pre>
</li>
</ul>
<a id="setComparisons(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setComparisons</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setComparisons&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;comparisons)</pre>
</li>
</ul>
<a id="getTypeDataContained()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTypeDataContained</h4>
<pre class="methodSignature">public&nbsp;java.util.TreeSet&lt;<a href="../dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a>&gt;&nbsp;getTypeDataContained()</pre>
</li>
</ul>
<a id="getnTerms()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getnTerms</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&gt;&nbsp;getnTerms()</pre>
</li>
</ul>
<a id="getNGSSeqs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNGSSeqs</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&gt;&nbsp;getNGSSeqs()</pre>
</li>
</ul>
<a id="setNGSSeqs(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setNGSSeqs</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setNGSSeqs&#8203;(java.util.ArrayList&lt;<a href="../dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&gt;&nbsp;rnaSeqs)</pre>
</li>
</ul>
<a id="setnTerms(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setnTerms</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setnTerms&#8203;(java.util.ArrayList&lt;<a href="../dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&gt;&nbsp;nTerms)</pre>
</li>
</ul>
<a id="getProteomes()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteomes</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a>&gt;&nbsp;getProteomes()</pre>
</li>
</ul>
<a id="setProteomes(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setProteomes</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setProteomes&#8203;(java.util.ArrayList&lt;<a href="../dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a>&gt;&nbsp;proteomes)</pre>
</li>
</ul>
<a id="setTypeDataContained(java.util.TreeSet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setTypeDataContained</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setTypeDataContained&#8203;(java.util.TreeSet&lt;<a href="../dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a>&gt;&nbsp;typeDataContained)</pre>
</li>
</ul>
<a id="getReference()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReference</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getReference()</pre>
</li>
</ul>
<a id="setReference(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setReference</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setReference&#8203;(java.lang.String&nbsp;reference)</pre>
</li>
</ul>
<a id="getStudyName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getStudyName</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getStudyName()</pre>
</li>
</ul>
<a id="setStudyName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setStudyName</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setStudyName&#8203;(java.lang.String&nbsp;studyName)</pre>
</li>
</ul>
<a id="getAntiComparisons()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAntiComparisons</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getAntiComparisons()</pre>
</li>
</ul>
<a id="setAntiComparisons(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAntiComparisons</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setAntiComparisons&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;antiComparisons)</pre>
</li>
</ul>
<a id="getLocalization()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLocalization</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getLocalization()</pre>
</li>
</ul>
<a id="setLocalization(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setLocalization</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setLocalization&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;localization)</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/BioCondition.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
