<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 18:10:05 CEST 2019 -->
<title>Uses of Class org.biojava3.core.sequence.DNASequence</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Class org.biojava3.core.sequence.DNASequence";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../DNASequence.html" title="class in org.biojava3.core.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Class org.biojava3.core.sequence.DNASequence" class="title">Uses of Class<br>org.biojava3.core.sequence.DNASequence</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.sequence">bacnet.datamodel.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.sequenceNCBI">bacnet.datamodel.sequenceNCBI</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence">org.biojava3.core.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.io">org.biojava3.core.sequence.io</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.genome">org.biojava3.genome</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.genome.parsers.geneid">org.biojava3.genome.parsers.geneid</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.genome.parsers.gff">org.biojava3.genome.parsers.gff</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a> in <a href="../../../../../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></h3>
<table class="useSummary">
<caption><span>Subclasses of <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a> in <a href="../../../../../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></span></code></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></span></code></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a> that return <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChromosomeBacteriaSequence.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getCodingRegion(java.lang.String)">getCodingRegion</a></span>&#8203;(java.lang.String&nbsp;accession)</code></th>
<td class="colLast">
<div class="block">Get the gene based on accession.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChromosomeBacteriaSequence.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getNonCodingRegion(java.lang.String)">getNonCodingRegion</a></span>&#8203;(java.lang.String&nbsp;accession)</code></th>
<td class="colLast">
<div class="block">Get the gene based on accession.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a> that return types with arguments of type <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChromosomeBacteriaSequence.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getCodingSequenceHashMap()">getCodingSequenceHashMap</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChromosomeBacteriaSequence.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#getNoncodingSequenceHashMap()">getNoncodingSequenceHashMap</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../../../bacnet/datamodel/sequence/package-summary.html">bacnet.datamodel.sequence</a> with type arguments of type <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../../../../bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chromosome.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequence/Chromosome.html#getChromosomeFromDNASequence(java.util.LinkedHashMap,bacnet.datamodel.sequence.Genome,int,java.lang.String)">getChromosomeFromDNASequence</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequenceList,
                            <a href="../../../../../bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome,
                            int&nbsp;k,
                            java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChromosomeBacteriaSequence.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setCodingSequenceHashMap(java.util.LinkedHashMap)">setCodingSequenceHashMap</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;codingSequenceHashMap)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChromosomeBacteriaSequence.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html#setNoncodingSequenceHashMap(java.util.LinkedHashMap)">setNoncodingSequenceHashMap</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;noncodingSequenceHashMap)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="bacnet.datamodel.sequenceNCBI">
<!--   -->
</a>
<h3>Uses of <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a> in <a href="../../../../../bacnet/datamodel/sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../../bacnet/datamodel/sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a> that return <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#convert(bacnet.datamodel.sequence.Srna)">convert</a></span>&#8203;(<a href="../../../../../bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&nbsp;sRNA)</code></th>
<td class="colLast">
<div class="block">Convert an Srna into a DNASequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeNCBI.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getGeneFromName(java.lang.String)">getGeneFromName</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Go through codingSequence and nonCodingSequence searching for corresponding
 accession or locus</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html#loadCorrespondingChromosome(java.lang.String)">loadCorrespondingChromosome</a></span>&#8203;(java.lang.String&nbsp;accession)</code></th>
<td class="colLast">
<div class="block">load and return the chromosome corresponding to an accession</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../../bacnet/datamodel/sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a> that return types with arguments of type <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeNCBI.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getCodingSequencesList(boolean)">getCodingSequencesList</a></span>&#8203;(boolean&nbsp;allChromosomes)</code></th>
<td class="colLast">
<div class="block">Put in a list all the CodingSequence found</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeNCBI.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getNonCodingSequencesList(boolean)">getNonCodingSequencesList</a></span>&#8203;(boolean&nbsp;allChromosomes)</code></th>
<td class="colLast">
<div class="block">Put in a list all the NonCodingSequence found</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeNCBI.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GenomeNCBI.html#getSubCodingSequencesList(java.util.ArrayList,boolean)">getSubCodingSequencesList</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;locusTag,
                         boolean&nbsp;allChromosomes)</code></th>
<td class="colLast">
<div class="block">Put in a sublist all the CodingSequence found in locuTag list</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../../bacnet/datamodel/sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a> with parameters of type <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeConversionElement.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#convertDNASequenceGene(org.biojava3.core.sequence.DNASequence)">convertDNASequenceGene</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;dnaSeq)</code></th>
<td class="colLast">
<div class="block">From a DNASequence make a Gene object</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeConversionElement.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#convertDNASequenceGeneral(org.biojava3.core.sequence.DNASequence)">convertDNASequenceGeneral</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;dnaSeq)</code></th>
<td class="colLast">
<div class="block">From a DNASequence object, create Sequence<br>
 Mainly used in GenomeConversion methods</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenomeConversionElement.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html#convertDNASequenceNcRNA(org.biojava3.core.sequence.DNASequence)">convertDNASequenceNcRNA</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;dnaSeq)</code></th>
<td class="colLast">
<div class="block">From a DNASequence make a Gene object</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#diffBeginPosition(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence)">diffBeginPosition</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;gene1,
                 <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;gene2)</code></th>
<td class="colLast">
<div class="block">Calculate difference in begin position</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#diffCenter(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence)">diffCenter</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;gene1,
          <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;gene2)</code></th>
<td class="colLast">
<div class="block">Calculate the distance between the two center of gene1 and gene2</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#diffEndPosition(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence)">diffEndPosition</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;gene1,
               <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;gene2)</code></th>
<td class="colLast">
<div class="block">Calculate difference in end position</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getCodonUsage(org.biojava3.core.sequence.DNASequence)">getCodonUsage</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getCOG(org.biojava3.core.sequence.DNASequence)">getCOG</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getDbxref(org.biojava3.core.sequence.DNASequence)">getDbxref</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getGene(org.biojava3.core.sequence.DNASequence)">getGene</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getGlaserFC(org.biojava3.core.sequence.DNASequence)">getGlaserFC</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getIsoelectricPoint(org.biojava3.core.sequence.DNASequence)">getIsoelectricPoint</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getLength(org.biojava3.core.sequence.DNASequence)">getLength</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getLengthProt(org.biojava3.core.sequence.DNASequence)">getLengthProt</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getMolMass(org.biojava3.core.sequence.DNASequence)">getMolMass</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getNote(org.biojava3.core.sequence.DNASequence)">getNote</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getOperon(org.biojava3.core.sequence.DNASequence)">getOperon</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getProduct(org.biojava3.core.sequence.DNASequence)">getProduct</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getProteinID(org.biojava3.core.sequence.DNASequence)">getProteinID</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getRASTinfo(org.biojava3.core.sequence.DNASequence)">getRASTinfo</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getStrand(org.biojava3.core.sequence.DNASequence)">getStrand</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getTranstable(org.biojava3.core.sequence.DNASequence)">getTranstable</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#getType(org.biojava3.core.sequence.DNASequence)">getType</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#isEqual(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence,int)">isEqual</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;gene1,
       <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;gene2,
       int&nbsp;scale)</code></th>
<td class="colLast">
<div class="block">Calculates the distance between the two center fo gene1 and gene2<br>
 If this distance is under scale, in absolute value, and both gene are on the
 same strand, True is return</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#isOverlap(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence)">isOverlap</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;gene1,
         <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;gene2)</code></th>
<td class="colLast">
<div class="block">Decide if two genes are overlapping <br>
 First, calculate the overlapping size of two genes, finding the gene upstream
 to the other<br>
 If overlap is negative it will mean that we have no overlap (gene1.end <
 gene2.begin) <br></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#overlap(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence)">overlap</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;gene1,
       <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;gene2)</code></th>
<td class="colLast">
<div class="block">Calculate the overlapping size of two genes <br>
 First, find the gene upstream to the other<br>
 Then, as we got gene1.begin < gene2.begin, overlap = gene1.end -
 gene2.begin<br>
 overlap can be negative, it will mean that we have no overlap (gene1.end <
 gene2.begin) <br>
 or overlap can be higher than gene2.length, in that case overlap =
 gene2.length</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#sameEnd(org.biojava3.core.sequence.DNASequence,org.biojava3.core.sequence.DNASequence,int)">sameEnd</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;gene1,
       <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;gene2,
       int&nbsp;scale)</code></th>
<td class="colLast">
<div class="block">Determine that to gene have the same end if their ending piosition are the
 same +- scale bp</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#searchElement(org.biojava3.core.sequence.DNASequence,java.lang.String)">searchElement</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;seq,
             java.lang.String&nbsp;attribute)</code></th>
<td class="colLast">
<div class="block">Search an element in the Note part of a DNASequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#toStringCodingInfo(org.biojava3.core.sequence.DNASequence)">toStringCodingInfo</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;coding)</code></th>
<td class="colLast">
<div class="block">Display the info contain in a coding DNASequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneNCBITools.</span><code><span class="memberNameLink"><a href="../../../../../bacnet/datamodel/sequenceNCBI/GeneNCBITools.html#toStringNonCodingInfo(org.biojava3.core.sequence.DNASequence)">toStringNonCodingInfo</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;coding)</code></th>
<td class="colLast">
<div class="block">Display info contain in a non coding DNASequence</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a> in <a href="../package-summary.html">org.biojava3.core.sequence</a></h3>
<table class="useSummary">
<caption><span>Subclasses of <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a> in <a href="../package-summary.html">org.biojava3.core.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></span></code></th>
<td class="colLast">
<div class="block">Represents a exon or coding sequence in a gene.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></span></code></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</a></span></code></th>
<td class="colLast">
<div class="block">A gene contains a collection of Exon sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../StartCodonSequence.html" title="class in org.biojava3.core.sequence">StartCodonSequence</a></span></code></th>
<td class="colLast">
<div class="block">Used to map the start codon feature on a gene</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../StopCodonSequence.html" title="class in org.biojava3.core.sequence">StopCodonSequence</a></span></code></th>
<td class="colLast">
<div class="block">Used to map the stop codon sequence on a gene</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></span></code></th>
<td class="colLast">
<div class="block">This is the sequence if you want to go from a gene sequence to a protein
 sequence.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Fields in <a href="../package-summary.html">org.biojava3.core.sequence</a> declared as <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StartCodonSequence.</span><code><span class="memberNameLink"><a href="../StartCodonSequence.html#parentGeneSequence">parentGeneSequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StopCodonSequence.</span><code><span class="memberNameLink"><a href="../StopCodonSequence.html#parentGeneSequence">parentGeneSequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence</a> that return <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptSequence.</span><code><span class="memberNameLink"><a href="../TranscriptSequence.html#getDNACodingSequence()">getDNACodingSequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the stitched together CDS sequences then maps to the cDNA</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneSequence.</span><code><span class="memberNameLink"><a href="../GeneSequence.html#getSequence5PrimeTo3Prime()">getSequence5PrimeTo3Prime</a></span>()</code></th>
<td class="colLast">
<div class="block">Try to give method clarity where you want a DNASequence coding in the 5' to
 3' direction Returns the DNASequence representative of the 5' and 3' reading
 based on strand</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.io">
<!--   -->
</a>
<h3>Uses of <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a> in <a href="../io/package-summary.html">org.biojava3.core.sequence.io</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../io/package-summary.html">org.biojava3.core.sequence.io</a> that return types with arguments of type <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaReaderHelper.</span><code><span class="memberNameLink"><a href="../io/FastaReaderHelper.html#readFastaDNASequence(java.io.File)">readFastaDNASequence</a></span>&#8203;(java.io.File&nbsp;file)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaReaderHelper.</span><code><span class="memberNameLink"><a href="../io/FastaReaderHelper.html#readFastaDNASequence(java.io.File,boolean)">readFastaDNASequence</a></span>&#8203;(java.io.File&nbsp;file,
                    boolean&nbsp;lazySequenceLoad)</code></th>
<td class="colLast">
<div class="block">Selecting lazySequenceLoad=true will parse the FASTA file and figure out the
 accessionid and offsets and return sequence objects that can in the future
 read the sequence from the disk.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaReaderHelper.</span><code><span class="memberNameLink"><a href="../io/FastaReaderHelper.html#readFastaDNASequence(java.io.InputStream)">readFastaDNASequence</a></span>&#8203;(java.io.InputStream&nbsp;inStream)</code></th>
<td class="colLast">
<div class="block">Read a fasta DNA sequence</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../io/package-summary.html">org.biojava3.core.sequence.io</a> with type arguments of type <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaWriterHelper.</span><code><span class="memberNameLink"><a href="../io/FastaWriterHelper.html#writeNucleotideSequence(java.io.File,java.util.Collection)">writeNucleotideSequence</a></span>&#8203;(java.io.File&nbsp;file,
                       java.util.Collection&lt;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequences)</code></th>
<td class="colLast">
<div class="block">Write a collection of NucleotideSequences to a file</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaWriterHelper.</span><code><span class="memberNameLink"><a href="../io/FastaWriterHelper.html#writeNucleotideSequence(java.io.OutputStream,java.util.Collection)">writeNucleotideSequence</a></span>&#8203;(java.io.OutputStream&nbsp;outputStream,
                       java.util.Collection&lt;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequences)</code></th>
<td class="colLast">
<div class="block">Write a collection of NucleotideSequences to a file</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.genome">
<!--   -->
</a>
<h3>Uses of <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a> in <a href="../../../genome/package-summary.html">org.biojava3.genome</a></h3>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../genome/package-summary.html">org.biojava3.genome</a> with type arguments of type <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneFeatureHelper.</span><code><span class="memberNameLink"><a href="../../../genome/GeneFeatureHelper.html#getChromosomeSequenceFromDNASequence(java.util.LinkedHashMap)">getChromosomeSequenceFromDNASequence</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequenceList)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.genome.parsers.geneid">
<!--   -->
</a>
<h3>Uses of <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a> in <a href="../../../genome/parsers/geneid/package-summary.html">org.biojava3.genome.parsers.geneid</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../genome/parsers/geneid/package-summary.html">org.biojava3.genome.parsers.geneid</a> that return types with arguments of type <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneIDXMLReader.</span><code><span class="memberNameLink"><a href="../../../genome/parsers/geneid/GeneIDXMLReader.html#getDNACodingSequences()">getDNACodingSequences</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.genome.parsers.gff">
<!--   -->
</a>
<h3>Uses of <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a> in <a href="../../../genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a> with parameters of type <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FeatureList.</span><code><span class="memberNameLink"><a href="../../../genome/parsers/gff/FeatureList.html#splice(org.biojava3.core.sequence.DNASequence)">splice</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Concatenate successive portions of the specified sequence using the feature
 locations in the list.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../genome/parsers/gff/package-summary.html">org.biojava3.genome.parsers.gff</a> with type arguments of type <a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GCStats.</span><code><span class="memberNameLink"><a href="../../../genome/parsers/gff/GCStats.html#getGCStats(java.util.Collection)">getGCStats</a></span>&#8203;(java.util.Collection&lt;<a href="../DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;sequences)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../DNASequence.html" title="class in org.biojava3.core.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
