<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>NTermStat</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="declaration: package: bacnet.scripts.listeriomics.nterm, class: NTermStat">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var evenRowColor = "even-row-color";
var oddRowColor = "odd-row-color";
var tableTab = "table-tab";
var activeTableTab = "active-table-tab";
var pathtoroot = "../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/NTermStat.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html#class">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div>
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">bacnet.scripts.listeriomics.nterm</a></div>
<h1 title="Class NTermStat" class="title">Class NTermStat</h1>
</div>
<div class="inheritance" title="Inheritance Tree"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">java.lang.Object</a>
<div class="inheritance">bacnet.scripts.listeriomics.nterm.NTermStat</div>
</div>
<section class="class-description" id="class-description">
<hr>
<div class="type-signature"><span class="modifiers">public class </span><span class="element-name type-name-label">NTermStat</span>
<span class="extends-implements">extends <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></span></div>
<div class="block">List of method to do statistics about common features in N-Terminus of genes: - start codon - SD
 binding - AU proportion - Folding energy</div>
<dl class="notes">
<dt>Author:</dt>
<dd>UIBC</dd>
</dl>
</section>
<section class="summary">
<ul class="summary-list">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<li>
<section class="constructor-summary" id="constructor-summary">
<h2>Constructor Summary</h2>
<div class="caption"><span>Constructors</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E()" class="member-name-link">NTermStat</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab1" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab1', 3)" class="table-tab">Static Methods</button><button id="method-summary-table-tab2" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab2', 3)" class="table-tab">Instance Methods</button><button id="method-summary-table-tab4" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab4', 3)" class="table-tab">Concrete Methods</button><button id="method-summary-table-tab6" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab6', 3)" class="table-tab">Deprecated Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel">
<div class="summary-table three-column-summary" aria-labelledby="method-summary-table-tab0">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#detectable()" class="member-name-link">detectable</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Create a table where we put all lmos detectable by the mass spectrometer</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[]</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getAtisBHI(bacnet.datamodel.sequence.Gene)" class="member-name-link">getAtisBHI</a><wbr>(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Go through all MassSpec data and find if the lmio was detected</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getAtisBlood(bacnet.datamodel.sequence.Gene)" class="member-name-link">getAtisBlood</a><wbr>(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static double</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getAUNumber(java.lang.String)" class="member-name-link">getAUNumber</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Count the number of A and T nucleotides</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getCAI(bacnet.datamodel.sequence.Gene)" class="member-name-link">getCAI</a><wbr>(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">After generating TIRSequence.fasta in setTIRSequence() method we submit to
 http://genomes.urv.cat/CAIcal/ to calculate CAI value and expected CAI value The table with CAI
 values is saved from the website in: CAIResults.excel</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getCAIGene(bacnet.datamodel.sequence.Gene)" class="member-name-link">getCAIGene</a><wbr>(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">After generating GeneSequence.fasta in setGeneSequence() method we submit to
 http://genomes.urv.cat/CAIcal/ to calculate CAI value and expected CAI value The table with CAI
 values is saved from the website in: StatGenes/CAIGeneResults.excel</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>int</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getChargedResidues(bacnet.datamodel.sequence.Gene,int)" class="member-name-link">getChargedResidues</a><wbr>(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene,
 int&nbsp;decay)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Sum up the different charge of the amino acids of the TIR</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>int</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getChargedResiduesGene(bacnet.datamodel.sequence.Gene)" class="member-name-link">getChargedResiduesGene</a><wbr>(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Sum up the different charge of the amino acids of the gene sequence</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getComment(bacnet.datamodel.sequence.Gene)" class="member-name-link">getComment</a><wbr>(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Return information about gene or iTIS</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getDetectable(bacnet.datamodel.sequence.Gene)" class="member-name-link">getDetectable</a><wbr>(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">By reading the List in: /StatGenes/EGDeDetectable.txt Say if a gene is detectable by the
 spectometer or not</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static double</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getGCNumber(java.lang.String)" class="member-name-link">getGCNumber</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Count the number of G and C nucleotides</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>double</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getGeneExpr(bacnet.datamodel.sequence.Gene,java.lang.String)" class="member-name-link">getGeneExpr</a><wbr>(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;dataName)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Return the median value expression (calculated on 10 BHI 37C data) in the Gene Expression array
 of the gene</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getiTIS()" class="member-name-link">getiTIS</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getLogo(bacnet.datamodel.sequence.Sequence)" class="member-name-link">getLogo</a><wbr>(<a href="../../../datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">For a given Sequence, return the nucleotide sequence before and after starting position</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getNCRNAUpstream(bacnet.datamodel.sequence.Gene)" class="member-name-link">getNCRNAUpstream</a><wbr>(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">For each gene search if the end of a smallRNA or cisReg element is in the region -200bp - start
 codon</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getSDEnergy(bacnet.datamodel.sequence.Gene)" class="member-name-link">getSDEnergy</a><wbr>(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Return the binding energy with aSD sequence</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static double</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getStartCodon(bacnet.datamodel.sequence.Gene)" class="member-name-link">getStartCodon</a><wbr>(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Return start codon proportion in the genome</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getTIRStructureEnergy(bacnet.datamodel.sequence.Gene)" class="member-name-link">getTIRStructureEnergy</a><wbr>(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">For very gene read TIRStructure.excel and extract the mean value of energy: We calculate the 2nd
 structure free energy and save it a table<br>
 And we calculate the energy for 40 different sequence: -20+40 to -20+80 and calculate the average
 energy</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>int</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getTotalSpectrum(bacnet.datamodel.sequence.Gene)" class="member-name-link">getTotalSpectrum</a><wbr>(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">For each gene add the number of spectra of every peptide available (not only aTIS peptides)</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4 method-summary-table-tab6"><code>int</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4 method-summary-table-tab6"><code><a href="#getUTRLength(bacnet.datamodel.sequence.Gene)" class="member-name-link">getUTRLength</a><wbr>(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4 method-summary-table-tab6">
<div class="block"><span class="deprecated-label">Deprecated.</span></div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#peptideLogo()" class="member-name-link">peptideLogo</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Create a fasta file with all sequences to create a logo</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setGeneSequence()" class="member-name-link">setGeneSequence</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Create a fasta file with all the gene sequences</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setSDBindingEnergy()" class="member-name-link">setSDBindingEnergy</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">For every gene calculate the SD-aSD hybridization affinity and save it in:
 /StatGenes/aSDBindingEneryg.excel</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setTIRSequence()" class="member-name-link">setTIRSequence</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">We create a fasta file with all the TIR DNA sequence (-18 +63)</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setTIRStructureEnergy()" class="member-name-link">setTIRStructureEnergy</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">For every gene calculate the 2nd structure free energy and save it a table<br>
 We calculate the energy for 40 different sequence: -20+40 to -20+80 and calculate the average
 energy</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#statInternalORF(bacnet.datamodel.sequence.Genome)" class="member-name-link">statInternalORF</a><wbr>(<a href="../../../datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Given a genome extract main information such as:<br>
 number of start codons, stop codons
 number of high SD binding sites
 number of possible ORFs
 number of predicted genes<br>
 <br>
 From this metrics, predict the number of possible internal translation initiation site</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#tableGeneEGDe()" class="member-name-link">tableGeneEGDe</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get all lmos from EGD-e, and extract different information: <br>
 
 The Translation Iinitiation region here is always: -20bp ---- 60bp around start codon
 Start codon
 SD energy
 Structure energy ....</div>
</div>
</div>
</div>
</div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Object">Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)" title="class or interface in java.lang" class="external-link">equals</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#getClass()" title="class or interface in java.lang" class="external-link">getClass</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode()" title="class or interface in java.lang" class="external-link">hashCode</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notify()" title="class or interface in java.lang" class="external-link">notify</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notifyAll()" title="class or interface in java.lang" class="external-link">notifyAll</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#toString()" title="class or interface in java.lang" class="external-link">toString</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait()" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long)" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long,int)" title="class or interface in java.lang" class="external-link">wait</a></code></div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<li>
<section class="constructor-details" id="constructor-detail">
<h2>Constructor Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="&lt;init&gt;()">
<h3>NTermStat</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name">NTermStat</span>()</div>
</section>
</li>
</ul>
</section>
</li>
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="getiTIS()">
<h3>getiTIS</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;</span>&nbsp;<span class="element-name">getiTIS</span>()</div>
</section>
</li>
<li>
<section class="detail" id="tableGeneEGDe()">
<h3>tableGeneEGDe</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">tableGeneEGDe</span>()</div>
<div class="block">Get all lmos from EGD-e, and extract different information: <br>
 
 The Translation Iinitiation region here is always: -20bp ---- 60bp around start codon
 <li>Start codon
 <li>SD energy
 <li>Structure energy ....</div>
</section>
</li>
<li>
<section class="detail" id="getAUNumber(java.lang.String)">
<h3>getAUNumber</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">double</span>&nbsp;<span class="element-name">getAUNumber</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence)</span></div>
<div class="block">Count the number of A and T nucleotides</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>sequence</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getGCNumber(java.lang.String)">
<h3>getGCNumber</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">double</span>&nbsp;<span class="element-name">getGCNumber</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence)</span></div>
<div class="block">Count the number of G and C nucleotides</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>sequence</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getComment(bacnet.datamodel.sequence.Gene)">
<h3>getComment</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getComment</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</span></div>
<div class="block">Return information about gene or iTIS</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>gene</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getStartCodon(bacnet.datamodel.sequence.Gene)">
<h3>getStartCodon</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">double</span>&nbsp;<span class="element-name">getStartCodon</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</span></div>
<div class="block">Return start codon proportion in the genome</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>gene</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getSDEnergy(bacnet.datamodel.sequence.Gene)">
<h3>getSDEnergy</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getSDEnergy</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</span></div>
<div class="block">Return the binding energy with aSD sequence</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>gene</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="setSDBindingEnergy()">
<h3>setSDBindingEnergy</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setSDBindingEnergy</span>()</div>
<div class="block">For every gene calculate the SD-aSD hybridization affinity and save it in:
 /StatGenes/aSDBindingEneryg.excel</div>
</section>
</li>
<li>
<section class="detail" id="getTIRStructureEnergy(bacnet.datamodel.sequence.Gene)">
<h3>getTIRStructureEnergy</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getTIRStructureEnergy</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</span></div>
<div class="block">For very gene read TIRStructure.excel and extract the mean value of energy: We calculate the 2nd
 structure free energy and save it a table<br>
 And we calculate the energy for 40 different sequence: -20+40 to -20+80 and calculate the average
 energy</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>gene</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="setTIRStructureEnergy()">
<h3>setTIRStructureEnergy</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setTIRStructureEnergy</span>()</div>
<div class="block">For every gene calculate the 2nd structure free energy and save it a table<br>
 We calculate the energy for 40 different sequence: -20+40 to -20+80 and calculate the average
 energy</div>
</section>
</li>
<li>
<section class="detail" id="setTIRSequence()">
<h3>setTIRSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setTIRSequence</span>()</div>
<div class="block">We create a fasta file with all the TIR DNA sequence (-18 +63)</div>
</section>
</li>
<li>
<section class="detail" id="setGeneSequence()">
<h3>setGeneSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setGeneSequence</span>()</div>
<div class="block">Create a fasta file with all the gene sequences</div>
</section>
</li>
<li>
<section class="detail" id="getCAI(bacnet.datamodel.sequence.Gene)">
<h3>getCAI</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getCAI</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</span></div>
<div class="block">After generating TIRSequence.fasta in setTIRSequence() method we submit to
 http://genomes.urv.cat/CAIcal/ to calculate CAI value and expected CAI value The table with CAI
 values is saved from the website in: CAIResults.excel</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>gene</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getCAIGene(bacnet.datamodel.sequence.Gene)">
<h3>getCAIGene</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getCAIGene</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</span></div>
<div class="block">After generating GeneSequence.fasta in setGeneSequence() method we submit to
 http://genomes.urv.cat/CAIcal/ to calculate CAI value and expected CAI value The table with CAI
 values is saved from the website in: StatGenes/CAIGeneResults.excel</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>gene</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getDetectable(bacnet.datamodel.sequence.Gene)">
<h3>getDetectable</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getDetectable</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</span></div>
<div class="block">By reading the List in: /StatGenes/EGDeDetectable.txt Say if a gene is detectable by the
 spectometer or not</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>gene</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAtisBHI(bacnet.datamodel.sequence.Gene)">
<h3>getAtisBHI</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[]</span>&nbsp;<span class="element-name">getAtisBHI</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</span></div>
<div class="block">Go through all MassSpec data and find if the lmio was detected</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>gene</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAtisBlood(bacnet.datamodel.sequence.Gene)">
<h3>getAtisBlood</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getAtisBlood</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</span></div>
</section>
</li>
<li>
<section class="detail" id="getChargedResidues(bacnet.datamodel.sequence.Gene,int)">
<h3>getChargedResidues</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name">getChargedResidues</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene,
 int&nbsp;decay)</span></div>
<div class="block">Sum up the different charge of the amino acids of the TIR</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>gene</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getChargedResiduesGene(bacnet.datamodel.sequence.Gene)">
<h3>getChargedResiduesGene</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name">getChargedResiduesGene</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</span></div>
<div class="block">Sum up the different charge of the amino acids of the gene sequence</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>gene</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getUTRLength(bacnet.datamodel.sequence.Gene)">
<h3>getUTRLength</h3>
<div class="member-signature"><span class="annotations"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Deprecated.html" title="class or interface in java.lang" class="external-link">@Deprecated</a>
</span><span class="modifiers">public</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name">getUTRLength</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</span></div>
<div class="deprecation-block"><span class="deprecated-label">Deprecated.</span></div>
<div class="block">For each gene search the closest TSS downstream, and return the UTR length (=distance between
 gene and TSS) We don't start at the start codon but -20bp, in case there is a TSS in this region
 indicating a possible short UTR with alternative aTIS</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>gene</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getGeneExpr(bacnet.datamodel.sequence.Gene,java.lang.String)">
<h3>getGeneExpr</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">double</span>&nbsp;<span class="element-name">getGeneExpr</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;dataName)</span></div>
<div class="block">Return the median value expression (calculated on 10 BHI 37C data) in the Gene Expression array
 of the gene</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>gene</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getTotalSpectrum(bacnet.datamodel.sequence.Gene)">
<h3>getTotalSpectrum</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name">getTotalSpectrum</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</span></div>
<div class="block">For each gene add the number of spectra of every peptide available (not only aTIS peptides)</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>gene</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getNCRNAUpstream(bacnet.datamodel.sequence.Gene)">
<h3>getNCRNAUpstream</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getNCRNAUpstream</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;gene)</span></div>
<div class="block">For each gene search if the end of a smallRNA or cisReg element is in the region -200bp - start
 codon</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>gene</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="detectable()">
<h3>detectable</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">detectable</span>()</div>
<div class="block">Create a table where we put all lmos detectable by the mass spectrometer</div>
</section>
</li>
<li>
<section class="detail" id="statInternalORF(bacnet.datamodel.sequence.Genome)">
<h3>statInternalORF</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">statInternalORF</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;genome)</span></div>
<div class="block">Given a genome extract main information such as:<br>
 <li>number of start codons, stop codons
 <li>number of high SD binding sites
 <li>number of possible ORFs
 <li>number of predicted genes<br>
 <br>
 From this metrics, predict the number of possible internal translation initiation site</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>genome</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="peptideLogo()">
<h3>peptideLogo</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">peptideLogo</span>()</div>
<div class="block">Create a fasta file with all sequences to create a logo</div>
</section>
</li>
<li>
<section class="detail" id="getLogo(bacnet.datamodel.sequence.Sequence)">
<h3>getLogo</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">getLogo</span><wbr><span class="parameters">(<a href="../../../datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene)</span></div>
<div class="block">For a given Sequence, return the nucleotide sequence before and after starting position</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>gene</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
</div>
</div>
</body>
</html>
