<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 18:10:05 CEST 2019 -->
<title>Uses of Class org.biojava3.core.sequence.compound.AminoAcidCompound</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Class org.biojava3.core.sequence.compound.AminoAcidCompound";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Class org.biojava3.core.sequence.compound.AminoAcidCompound" class="title">Uses of Class<br>org.biojava3.core.sequence.compound.AminoAcidCompound</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence">org.biojava3.core.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.compound">org.biojava3.core.sequence.compound</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.io">org.biojava3.core.sequence.io</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava3.core.sequence.transcription">org.biojava3.core.sequence.transcription</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a> in <a href="../../package-summary.html">org.biojava3.core.sequence</a></h3>
<table class="useSummary">
<caption><span>Constructor parameters in <a href="../../package-summary.html">org.biojava3.core.sequence</a> with type arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../ProteinSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">ProteinSequence</a></span>&#8203;(java.lang.String&nbsp;seqString,
               <a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Create a protein from a string with a user defined set of amino acids</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../ProteinSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader)">ProteinSequence</a></span>&#8203;(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader)</code></th>
<td class="colLast">
<div class="block">A protein sequence where the storage of the sequence is somewhere else.</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../ProteinSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader,org.biojava3.core.sequence.template.CompoundSet)">ProteinSequence</a></span>&#8203;(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
               <a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">A protein sequence where the storage of the sequence is somewhere else with
 user defined set of amino acids.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../ProteinSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader,org.biojava3.core.sequence.template.CompoundSet)">ProteinSequence</a></span>&#8203;(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
               <a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">A protein sequence where the storage of the sequence is somewhere else with
 user defined set of amino acids.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.compound">
<!--   -->
</a>
<h3>Uses of <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a> in <a href="../package-summary.html">org.biojava3.core.sequence.compound</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence.compound</a> that return <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidCompoundSet.</span><code><span class="memberNameLink"><a href="../AminoAcidCompoundSet.html#getCompoundForString(java.lang.String)">getCompoundForString</a></span>&#8203;(java.lang.String&nbsp;string)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence.compound</a> that return types with arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidCompoundSet.</span><code><span class="memberNameLink"><a href="../AminoAcidCompoundSet.html#getAllCompounds()">getAllCompounds</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidCompound.</span><code><span class="memberNameLink"><a href="../AminoAcidCompound.html#getCompoundSet()">getCompoundSet</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.Set&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidCompoundSet.</span><code><span class="memberNameLink"><a href="../AminoAcidCompoundSet.html#getEquivalentCompounds(org.biojava3.core.sequence.compound.AminoAcidCompound)">getEquivalentCompounds</a></span>&#8203;(<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence.compound</a> with parameters of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidCompoundSet.</span><code><span class="memberNameLink"><a href="../AminoAcidCompoundSet.html#compoundsEquivalent(org.biojava3.core.sequence.compound.AminoAcidCompound,org.biojava3.core.sequence.compound.AminoAcidCompound)">compoundsEquivalent</a></span>&#8203;(<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compoundOne,
                   <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compoundTwo)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.Set&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidCompoundSet.</span><code><span class="memberNameLink"><a href="../AminoAcidCompoundSet.html#getEquivalentCompounds(org.biojava3.core.sequence.compound.AminoAcidCompound)">getEquivalentCompounds</a></span>&#8203;(<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidCompoundSet.</span><code><span class="memberNameLink"><a href="../AminoAcidCompoundSet.html#getStringForCompound(org.biojava3.core.sequence.compound.AminoAcidCompound)">getStringForCompound</a></span>&#8203;(<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidCompoundSet.</span><code><span class="memberNameLink"><a href="../AminoAcidCompoundSet.html#hasCompound(org.biojava3.core.sequence.compound.AminoAcidCompound)">hasCompound</a></span>&#8203;(<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../package-summary.html">org.biojava3.core.sequence.compound</a> with type arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidCompoundSet.</span><code><span class="memberNameLink"><a href="../AminoAcidCompoundSet.html#verifySequence(org.biojava3.core.sequence.template.Sequence)">verifySequence</a></span>&#8203;(<a href="../../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.io">
<!--   -->
</a>
<h3>Uses of <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a> in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> that return types with arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CasePreservingProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(java.lang.String&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CasePreservingProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(java.util.List)">getSequence</a></span>&#8203;(java.util.List&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></th>
<td class="colLast">
<div class="block">Assumes all compounds were uppercase</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CasePreservingProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence</a></span>&#8203;(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileProxyProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(java.lang.String&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">
<div class="block">Even though we are passing in the sequence we really only care about the
 length of the sequence and the offset index in the fasta file.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileProxyProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(java.util.List)">getSequence</a></span>&#8203;(java.util.List&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></th>
<td class="colLast">
<div class="block">Not sure of use case and currently not supported</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileProxyProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence</a></span>&#8203;(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code></th>
<td class="colLast">
<div class="block">Should be able to extend the same concept to a remote URL call or database
 connection.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/ProteinSequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(java.lang.String&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/ProteinSequenceCreator.html#getSequence(java.util.List)">getSequence</a></span>&#8203;(java.util.List&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/ProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence</a></span>&#8203;(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> with parameters of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IUPACParser.IUPACTable.</span><code><span class="memberNameLink"><a href="../../io/IUPACParser.IUPACTable.html#isStart(org.biojava3.core.sequence.compound.AminoAcidCompound)">isStart</a></span>&#8203;(<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compound)</code></th>
<td class="colLast">
<div class="block">Returns true if the given compound was a start codon in this codon table.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> with type arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CasePreservingProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(java.util.List)">getSequence</a></span>&#8203;(java.util.List&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></th>
<td class="colLast">
<div class="block">Assumes all compounds were uppercase</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CasePreservingProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence</a></span>&#8203;(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileProxyProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(java.util.List)">getSequence</a></span>&#8203;(java.util.List&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></th>
<td class="colLast">
<div class="block">Not sure of use case and currently not supported</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileProxyProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence</a></span>&#8203;(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code></th>
<td class="colLast">
<div class="block">Should be able to extend the same concept to a remote URL call or database
 connection.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/ProteinSequenceCreator.html#getSequence(java.util.List)">getSequence</a></span>&#8203;(java.util.List&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../io/ProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)">getSequence</a></span>&#8203;(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructor parameters in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> with type arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/CasePreservingProteinSequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)">CasePreservingProteinSequenceCreator</a></span>&#8203;(<a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/FileProxyProteinSequenceCreator.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.template.CompoundSet)">FileProxyProteinSequenceCreator</a></span>&#8203;(java.io.File&nbsp;fastaFile,
                               <a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Need File so that we can store full path name in SequenceFileProxyLoader for
 Random File access as a quick read</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/ProteinSequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)">ProteinSequenceCreator</a></span>&#8203;(<a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava3.core.sequence.transcription">
<!--   -->
</a>
<h3>Uses of <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a> in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> that return <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Table.Codon.</span><code><span class="memberNameLink"><a href="../../transcription/Table.Codon.html#getAminoAcid()">getAminoAcid</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> that return types with arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RNAToAminoAcidTranslator.</span><code><span class="memberNameLink"><a href="../../transcription/RNAToAminoAcidTranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)">createSequences</a></span>&#8203;(<a href="../../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence)</code></th>
<td class="colLast">
<div class="block">Performs the core conversion of RNA to Peptide.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.html#getAminoAcidCompounds()">getAminoAcidCompounds</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.html#getProteinSequenceCreator()">getProteinSequenceCreator</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.Map&lt;<a href="../../transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a>,&#8203;<a href="../../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.html#multipleFrameTranslation(org.biojava3.core.sequence.template.Sequence,org.biojava3.core.sequence.transcription.Frame...)">multipleFrameTranslation</a></span>&#8203;(<a href="../../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna,
                        <a href="../../transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a>...&nbsp;frames)</code></th>
<td class="colLast">
<div class="block">A way of translating DNA in a number of frames</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.html#translate(org.biojava3.core.sequence.template.Sequence)">translate</a></span>&#8203;(<a href="../../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna)</code></th>
<td class="colLast">
<div class="block">Quick method to let you go from a CDS to a Peptide quickly.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> with parameters of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Table.</span><code><span class="memberNameLink"><a href="../../transcription/Table.html#isStart(org.biojava3.core.sequence.compound.AminoAcidCompound)">isStart</a></span>&#8203;(<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compound)</code></th>
<td class="colLast">
<div class="block">Returns true if the given compound could have been a start amino acid; this
 does not assert if the codon that actually coded for the amino acid was a
 start codon.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> with type arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.Builder.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.Builder.html#aminoAcidsCompounds(org.biojava3.core.sequence.template.CompoundSet)">aminoAcidsCompounds</a></span>&#8203;(<a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.Builder.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.Builder.html#proteinCreator(org.biojava3.core.sequence.io.template.SequenceCreatorInterface)">proteinCreator</a></span>&#8203;(<a href="../../io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;creator)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> with parameters of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../transcription/Table.Codon.html#%3Cinit%3E(org.biojava3.core.sequence.transcription.Table.CaseInsensitiveTriplet,org.biojava3.core.sequence.compound.AminoAcidCompound,boolean,boolean)">Codon</a></span>&#8203;(<a href="../../transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a>&nbsp;triplet,
     <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;aminoAcid,
     boolean&nbsp;start,
     boolean&nbsp;stop)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructor parameters in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> with type arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../transcription/RNAToAminoAcidTranslator.html#%3Cinit%3E(org.biojava3.core.sequence.io.template.SequenceCreatorInterface,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.transcription.Table,boolean,boolean,boolean)">RNAToAminoAcidTranslator</a></span>&#8203;(<a href="../../io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;creator,
                        <a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;nucleotides,
                        <a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;&nbsp;codons,
                        <a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcids,
                        <a href="../../transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</a>&nbsp;table,
                        boolean&nbsp;trimStops,
                        boolean&nbsp;initMetOnly,
                        boolean&nbsp;translateNCodons)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
