<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Fri Dec 10 09:15:34 CET 2021 -->
<title>Uses of Class org.biojava3.core.sequence.compound.AminoAcidCompound</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-10">
<meta name="description" content="use: package: org.biojava3.core.sequence.compound, class: AminoAcidCompound">
<meta name="generator" content="javadoc/ClassUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-use-page">
<script type="text/javascript">var pathtoroot = "../../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">Class</a></li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Class org.biojava3.core.sequence.compound.AminoAcidCompound" class="title">Uses of Class<br>org.biojava3.core.sequence.compound.AminoAcidCompound</h1>
</div>
<div class="caption"><span>Packages that use <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#org.biojava3.core.sequence">org.biojava3.core.sequence</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava3.core.sequence.compound">org.biojava3.core.sequence.compound</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava3.core.sequence.io">org.biojava3.core.sequence.io</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava3.core.sequence.transcription">org.biojava3.core.sequence.transcription</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<section class="class-uses">
<ul class="block-list">
<li>
<section class="detail" id="org.biojava3.core.sequence">
<h2>Uses of <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a> in <a href="../../package-summary.html">org.biojava3.core.sequence</a></h2>
<div class="caption"><span>Constructor parameters in <a href="../../package-summary.html">org.biojava3.core.sequence</a> with type arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../ProteinSequence.html#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ProteinSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Create a protein from a string with a user defined set of amino acids</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../ProteinSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader)" class="member-name-link">ProteinSequence</a><wbr>(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader)</code></div>
<div class="col-last odd-row-color">
<div class="block">A protein sequence where the storage of the sequence is somewhere else.</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../ProteinSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ProteinSequence</a><wbr>(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
 <a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">A protein sequence where the storage of the sequence is somewhere else with user defined set of
 amino acids.</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../ProteinSequence.html#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ProteinSequence</a><wbr>(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
 <a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">A protein sequence where the storage of the sequence is somewhere else with user defined set of
 amino acids.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava3.core.sequence.compound">
<h2>Uses of <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a> in <a href="../package-summary.html">org.biojava3.core.sequence.compound</a></h2>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence.compound</a> that return <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AminoAcidCompoundSet.</span><code><a href="../AminoAcidCompoundSet.html#getCompoundForString(java.lang.String)" class="member-name-link">getCompoundForString</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;string)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence.compound</a> that return types with arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AminoAcidCompoundSet.</span><code><a href="../AminoAcidCompoundSet.html#getAllCompounds()" class="member-name-link">getAllCompounds</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AminoAcidCompound.</span><code><a href="../AminoAcidCompound.html#getCompoundSet()" class="member-name-link">getCompoundSet</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Set.html" title="class or interface in java.util" class="external-link">Set</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AminoAcidCompoundSet.</span><code><a href="../AminoAcidCompoundSet.html#getEquivalentCompounds(org.biojava3.core.sequence.compound.AminoAcidCompound)" class="member-name-link">getEquivalentCompounds</a><wbr>(<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compound)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava3.core.sequence.compound</a> with parameters of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>boolean</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AminoAcidCompoundSet.</span><code><a href="../AminoAcidCompoundSet.html#compoundsEquivalent(org.biojava3.core.sequence.compound.AminoAcidCompound,org.biojava3.core.sequence.compound.AminoAcidCompound)" class="member-name-link">compoundsEquivalent</a><wbr>(<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compoundOne,
 <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compoundTwo)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Set.html" title="class or interface in java.util" class="external-link">Set</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AminoAcidCompoundSet.</span><code><a href="../AminoAcidCompoundSet.html#getEquivalentCompounds(org.biojava3.core.sequence.compound.AminoAcidCompound)" class="member-name-link">getEquivalentCompounds</a><wbr>(<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compound)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AminoAcidCompoundSet.</span><code><a href="../AminoAcidCompoundSet.html#getStringForCompound(org.biojava3.core.sequence.compound.AminoAcidCompound)" class="member-name-link">getStringForCompound</a><wbr>(<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compound)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>boolean</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AminoAcidCompoundSet.</span><code><a href="../AminoAcidCompoundSet.html#hasCompound(org.biojava3.core.sequence.compound.AminoAcidCompound)" class="member-name-link">hasCompound</a><wbr>(<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compound)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Method parameters in <a href="../package-summary.html">org.biojava3.core.sequence.compound</a> with type arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AminoAcidCompoundSet.</span><code><a href="../AminoAcidCompoundSet.html#verifySequence(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">verifySequence</a><wbr>(<a href="../../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequence)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava3.core.sequence.io">
<h2>Uses of <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a> in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a></h2>
<div class="caption"><span>Methods in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> that return types with arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">CasePreservingProteinSequenceCreator.</span><code><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CasePreservingProteinSequenceCreator.</span><code><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></div>
<div class="col-last odd-row-color">
<div class="block">Assumes all compounds were uppercase</div>
</div>
<div class="col-first even-row-color"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">CasePreservingProteinSequenceCreator.</span><code><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence</a><wbr>(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FileProxyProteinSequenceCreator.</span><code><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence
 and the offset index in the fasta file.</div>
</div>
<div class="col-first even-row-color"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FileProxyProteinSequenceCreator.</span><code><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></div>
<div class="col-last even-row-color">
<div class="block">Not sure of use case and currently not supported</div>
</div>
<div class="col-first odd-row-color"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FileProxyProteinSequenceCreator.</span><code><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence</a><wbr>(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">
<div class="block">Should be able to extend the same concept to a remote URL call or database connection.</div>
</div>
<div class="col-first even-row-color"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ProteinSequenceCreator.</span><code><a href="../../io/ProteinSequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">ProteinSequenceCreator.</span><code><a href="../../io/ProteinSequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ProteinSequenceCreator.</span><code><a href="../../io/ProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence</a><wbr>(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> with parameters of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>boolean</code></div>
<div class="col-second even-row-color"><span class="type-name-label">IUPACParser.IUPACTable.</span><code><a href="../../io/IUPACParser.IUPACTable.html#isStart(org.biojava3.core.sequence.compound.AminoAcidCompound)" class="member-name-link">isStart</a><wbr>(<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compound)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns true if the given compound was a start codon in this codon table.</div>
</div>
</div>
<div class="caption"><span>Method parameters in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> with type arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">CasePreservingProteinSequenceCreator.</span><code><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></div>
<div class="col-last even-row-color">
<div class="block">Assumes all compounds were uppercase</div>
</div>
<div class="col-first odd-row-color"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CasePreservingProteinSequenceCreator.</span><code><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence</a><wbr>(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FileProxyProteinSequenceCreator.</span><code><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></div>
<div class="col-last even-row-color">
<div class="block">Not sure of use case and currently not supported</div>
</div>
<div class="col-first odd-row-color"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FileProxyProteinSequenceCreator.</span><code><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence</a><wbr>(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">
<div class="block">Should be able to extend the same concept to a remote URL call or database connection.</div>
</div>
<div class="col-first even-row-color"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ProteinSequenceCreator.</span><code><a href="../../io/ProteinSequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../../template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">ProteinSequenceCreator.</span><code><a href="../../io/ProteinSequenceCreator.html#getSequence(org.biojava3.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence</a><wbr>(<a href="../../template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructor parameters in <a href="../../io/package-summary.html">org.biojava3.core.sequence.io</a> with type arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../io/CasePreservingProteinSequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">CasePreservingProteinSequenceCreator</a><wbr>(<a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../io/FileProxyProteinSequenceCreator.html#%3Cinit%3E(java.io.File,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">FileProxyProteinSequenceCreator</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html" title="class or interface in java.io" class="external-link">File</a>&nbsp;fastaFile,
 <a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">Need File so that we can store full path name in SequenceFileProxyLoader for Random File access
 as a quick read</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../io/ProteinSequenceCreator.html#%3Cinit%3E(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">ProteinSequenceCreator</a><wbr>(<a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava3.core.sequence.transcription">
<h2>Uses of <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a> in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a></h2>
<div class="caption"><span>Methods in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> that return <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Table.Codon.</span><code><a href="../../transcription/Table.Codon.html#getAminoAcid()" class="member-name-link">getAminoAcid</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> that return types with arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">RNAToAminoAcidTranslator.</span><code><a href="../../transcription/RNAToAminoAcidTranslator.html#createSequences(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">createSequences</a><wbr>(<a href="../../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence)</code></div>
<div class="col-last even-row-color">
<div class="block">Performs the core conversion of RNA to Peptide.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">TranscriptionEngine.</span><code><a href="../../transcription/TranscriptionEngine.html#getAminoAcidCompounds()" class="member-name-link">getAminoAcidCompounds</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../../io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">TranscriptionEngine.</span><code><a href="../../transcription/TranscriptionEngine.html#getProteinSequenceCreator()" class="member-name-link">getProteinSequenceCreator</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html" title="class or interface in java.util" class="external-link">Map</a>&lt;<a href="../../transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a>,<wbr><a href="../../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">TranscriptionEngine.</span><code><a href="../../transcription/TranscriptionEngine.html#multipleFrameTranslation(org.biojava3.core.sequence.template.Sequence,org.biojava3.core.sequence.transcription.Frame...)" class="member-name-link">multipleFrameTranslation</a><wbr>(<a href="../../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna,
 <a href="../../transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a>...&nbsp;frames)</code></div>
<div class="col-last odd-row-color">
<div class="block">A way of translating DNA in a number of frames</div>
</div>
<div class="col-first even-row-color"><code><a href="../../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">TranscriptionEngine.</span><code><a href="../../transcription/TranscriptionEngine.html#translate(org.biojava3.core.sequence.template.Sequence)" class="member-name-link">translate</a><wbr>(<a href="../../template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna)</code></div>
<div class="col-last even-row-color">
<div class="block">Quick method to let you go from a CDS to a Peptide quickly.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> with parameters of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>boolean</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Table.</span><code><a href="../../transcription/Table.html#isStart(org.biojava3.core.sequence.compound.AminoAcidCompound)" class="member-name-link">isStart</a><wbr>(<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;compound)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns true if the given compound could have been a start amino acid; this does not assert if
 the codon that actually coded for the amino acid was a start codon.</div>
</div>
</div>
<div class="caption"><span>Method parameters in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> with type arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../../transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">TranscriptionEngine.Builder.</span><code><a href="../../transcription/TranscriptionEngine.Builder.html#aminoAcidsCompounds(org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">aminoAcidsCompounds</a><wbr>(<a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compounds)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../../transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">TranscriptionEngine.Builder.</span><code><a href="../../transcription/TranscriptionEngine.Builder.html#proteinCreator(org.biojava3.core.sequence.io.template.SequenceCreatorInterface)" class="member-name-link">proteinCreator</a><wbr>(<a href="../../io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;creator)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> with parameters of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../transcription/Table.Codon.html#%3Cinit%3E(org.biojava3.core.sequence.transcription.Table.CaseInsensitiveTriplet,org.biojava3.core.sequence.compound.AminoAcidCompound,boolean,boolean)" class="member-name-link">Codon</a><wbr>(<a href="../../transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a>&nbsp;triplet,
 <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&nbsp;aminoAcid,
 boolean&nbsp;start,
 boolean&nbsp;stop)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructor parameters in <a href="../../transcription/package-summary.html">org.biojava3.core.sequence.transcription</a> with type arguments of type <a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../transcription/RNAToAminoAcidTranslator.html#%3Cinit%3E(org.biojava3.core.sequence.io.template.SequenceCreatorInterface,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.template.CompoundSet,org.biojava3.core.sequence.transcription.Table,boolean,boolean,boolean)" class="member-name-link">RNAToAminoAcidTranslator</a><wbr>(<a href="../../io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;creator,
 <a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;nucleotides,
 <a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a>&gt;&nbsp;codons,
 <a href="../../template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;aminoAcids,
 <a href="../../transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</a>&nbsp;table,
 boolean&nbsp;trimStops,
 boolean&nbsp;initMetOnly,
 boolean&nbsp;translateNCodons)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
</ul>
</section>
</main>
</div>
</div>
</body>
</html>
