<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Fri Dec 10 09:15:34 CET 2021 -->
<title>All Classes and Interfaces</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-10">
<meta name="description" content="class index">
<meta name="generator" content="javadoc/AllClassesIndexWriter">
<link rel="stylesheet" type="text/css" href="stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="script.js"></script>
<script type="text/javascript" src="script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="script-dir/jquery-ui.min.js"></script>
</head>
<body class="all-classes-index-page">
<script type="text/javascript">var evenRowColor = "even-row-color";
var oddRowColor = "odd-row-color";
var tableTab = "table-tab";
var activeTableTab = "active-table-tab";
var pathtoroot = "./";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="overview-tree.html">Tree</a></li>
<li><a href="deprecated-list.html">Deprecated</a></li>
<li><a href="index-files/index-1.html">Index</a></li>
<li><a href="help-doc.html#all-classes">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="All Classes and Interfaces" class="title">All Classes and Interfaces</h1>
</div>
<div id="all-classes-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="all-classes-table-tab0" role="tab" aria-selected="true" aria-controls="all-classes-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('all-classes-table', 'all-classes-table', 2)" class="active-table-tab">All Classes and Interfaces</button><button id="all-classes-table-tab1" role="tab" aria-selected="false" aria-controls="all-classes-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('all-classes-table', 'all-classes-table-tab1', 2)" class="table-tab">Interfaces</button><button id="all-classes-table-tab2" role="tab" aria-selected="false" aria-controls="all-classes-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('all-classes-table', 'all-classes-table-tab2', 2)" class="table-tab">Classes</button><button id="all-classes-table-tab3" role="tab" aria-selected="false" aria-controls="all-classes-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('all-classes-table', 'all-classes-table-tab3', 2)" class="table-tab">Enums</button><button id="all-classes-table-tab5" role="tab" aria-selected="false" aria-controls="all-classes-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('all-classes-table', 'all-classes-table-tab5', 2)" class="table-tab">Exceptions</button><button id="all-classes-table-tab6" role="tab" aria-selected="false" aria-controls="all-classes-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('all-classes-table', 'all-classes-table-tab6', 2)" class="table-tab">Errors</button></div>
<div id="all-classes-table.tabpanel" role="tabpanel">
<div class="summary-table two-column-summary" aria-labelledby="all-classes-table-tab0">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The details of a Compound</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a>&lt;<a href="org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="type parameter in AbstractCompoundSet">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a>&lt;<a href="org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>,<wbr><a href="org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a>&lt;<a href="org/biojava3/core/sequence/features/AbstractFeature.html" title="type parameter in AbstractFeature">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/features/AbstractFeature.html" title="type parameter in AbstractFeature">C</a>&gt;,<wbr><a href="org/biojava3/core/sequence/features/AbstractFeature.html" title="type parameter in AbstractFeature">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s) A
 feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Base abstraction of a location which encodes for the majority of important features about a
 location such as the start, end and strand</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractNucleotideCompoundSet</a>&lt;<a href="org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html" title="type parameter in AbstractNucleotideCompoundSet">C</a> extends <a href="org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The base class for DNA, RNA and Protein sequences.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/location/template/AccesionedLocation.html" title="interface in org.biojava3.core.sequence.location.template">AccesionedLocation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">A location which is bound to an AccessionID.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/template/Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Indicates an entity is accessioned</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used in Sequences as the unique indentifier.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/setup/Activator.html" title="class in bacnet.e4.rap.setup">Activator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">OSGI Activator of bacnet.e4.rap.setup</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/Activator.html" title="class in bacnet.scripts">Activator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">OSGI Activator of bacnet.scripts</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddProteomicsDataDialog</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddTranscriptomicsDataDialog</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/technology/Agilent.html" title="class in bacnet.scripts.listeriomics.technology">Agilent</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/compound/AmbiguityDNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">AmbiguityDNACompoundSet</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/compound/AmbiguityRNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">AmbiguityRNACompoundSet</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to describe an Amino Acid.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Set of proteinogenic amino acids.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/stat/AMtools.html" title="class in bacnet.scripts.core.stat">AMtools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/App.html" title="class in org.biojava3.genome">App</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Hello world!</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/AppSpecificMethods.html" title="class in bacnet.e4.rap">AppSpecificMethods</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">All the methods which are specific to a webpage</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Different methods to curate and parse the different data for Listeriomics website <br>
 <br>
 
 ONLY METHODS BEFORE FINAL DATA PROCESSING ARE PRESENT HERE</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">List of methods to download all ArrayExpress needed for Listeriomics website</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressListeriomics</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Different methods to curate and parse the different data for Listeriomics website <br>
 <br>
 
 ONLY METHODS BEFORE FINAL DATA PROCESSING ARE PRESENT HERE</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics">ArrayExpressListeriomics</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Different methods to curate and parse the different data for Listeriomics website <br>
 <br>
 
 ONLY METHODS BEFORE FINAL DATA PROCESSING ARE PRESENT HERE</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">List of methods to determine the different technology used</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">List of methods to determine the different technology used</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/loader/ArrayListProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">ArrayListProxySequenceReader</a>&lt;<a href="org/biojava3/core/sequence/loader/ArrayListProxySequenceReader.html" title="type parameter in ArrayListProxySequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a>&lt;<a href="org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores a Sequence as a collection of compounds in an ArrayList</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Public class with general array manipulation utilities, not specific to PROJECTNAME.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/BacnetE4PointFactory.html" title="class in bacnet.e4.rap">BacnetE4PointFactory</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/setup/BacnetE4PointFactory.html" title="class in bacnet.e4.rap.setup">BacnetE4PointFactory</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/BannerView.html" title="class in bacnet.e4.rap">BannerView</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/views/BannerView.html" title="class in bacnet.views">BannerView</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/BasicApplication.html" title="class in bacnet.e4.rap">BasicApplication</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/setup/BasicApplication.html" title="class in bacnet.e4.rap.setup">BasicApplication</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/BasicSequence.html" title="class in org.biojava3.core.sequence">BasicSequence</a>&lt;<a href="org/biojava3/core/sequence/BasicSequence.html" title="type parameter in BasicSequence">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Bare bones version of the Sequence object to be used sparingly.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class is used for the different bioconds declared in the experiment
 
 It contains the biocond name, and all the related data.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/database/BioConditionCreation.html" title="class in bacnet.scripts.database">BioConditionCreation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a>&lt;<a href="org/biojava3/core/sequence/storage/BitSequenceReader.html" title="type parameter in BitSequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">An implementation of the popular bit encodings.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a>&lt;<a href="org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The logic of working with a bit has been separated out into this class to help developers create
 the bit data structures without having to put the code into an intermediate format and to also
 use the format without the need to copy this code.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Different methods to create blast database for the genome, the ORF, and the smallORF</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/blast/BlastOutput.html" title="class in bacnet.scripts.blast">BlastOutput</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html" title="enum in bacnet.scripts.blast">BlastOutput.BlastOutputTYPE</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/blast/BlastResult.BlastResultCompare.html" title="class in bacnet.scripts.blast">BlastResult.BlastResultCompare</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This comparator classify the BlastResult by their position on the genome</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/query/BlastXMLQuery.html" title="class in org.biojava3.genome.query">BlastXMLQuery</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Need to keep track of actual bytes read and take advantage of buffered reader performance.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Attempts to wrap compounds so it is possible to view them in a case insensitive manner</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A sequence creator which preserves the case of its input string in the user collection of the
 returned ProteinSequence.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/CDSComparator.html" title="class in org.biojava3.core.sequence">CDSComparator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Represents a exon or coding sequence in a gene.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Circular Genome Panel created for displaying Small RNAs in CGView</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This object represents a classpath resource on the local system.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/CMD.html" title="class in bacnet.utils">CMD</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Class for managing triplet of nucleotides.<br>
 Allo&lt;s detection of start and stop codons in nucleotide sequences</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Define a codon</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/annotation/CodonUsage.html" title="class in bacnet.datamodel.annotation">CodonUsage</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Tools for calculating codon Usage of amino acid sequences<br>
 Codon Usage tables came from : http://www.kazusa.or.jp/codon/</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/views/CoExprNetworkView.InitNetworkThread.html" title="class in bacnet.views">CoExprNetworkView.InitNetworkThread</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/table/core/ColorMapper.TypeMapper.html" title="enum in bacnet.table.core">ColorMapper.TypeMapper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/table/core/ColorMapperRCP.Type.html" title="enum in bacnet.table.core">ColorMapperRCP.Type</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/gui/ColorMapperWizard.html" title="class in bacnet.table.gui">ColorMapperWizard</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/vennDiagram/CombinationGenerator.html" title="class in bacnet.scripts.core.vennDiagram">CombinationGenerator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Class for finding every combination of a set of integer</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/CombinationGenerator.html" title="class in bacnet.utils">CombinationGenerator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A ComparisonAtlas is related to a Comparison object and gives the list of genome elements over,
 under and not differentially expressed</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/template/ComplementCompound.html" title="interface in org.biojava3.core.sequence.template">ComplementCompound</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava3.core.sequence.views">ComplementSequenceView</a>&lt;<a href="org/biojava3/core/sequence/views/ComplementSequenceView.html" title="type parameter in ComplementSequenceView">C</a> extends <a href="org/biojava3/core/sequence/template/ComplementCompound.html" title="interface in org.biojava3.core.sequence.template">ComplementCompound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">For a given sequence this class will create a view over the top of it and for every request the
 code will return the complement of the underlying base e.g. base A will become base T</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/genomeBrowser/CompositeNTermInfo.html" title="class in bacnet.genomeBrowser">CompositeNTermInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">NTerm composite which is displayed on the right of the GenomeViewer.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequence/CompoundColorization.html" title="class in bacnet.datamodel.sequence">CompoundColorization</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab6"><a href="org/biojava3/core/exceptions/CompoundNotFoundError.html" title="class in org.biojava3.core.exceptions">CompoundNotFoundError</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab6">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="org/biojava3/core/sequence/template/CompoundSet.html" title="type parameter in CompoundSet">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/template/CompoundTranslator.html" title="interface in org.biojava3.core.sequence.template">CompoundTranslator</a>&lt;<a href="org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">F</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>,<wbr><a href="org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">T</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Static utility to easily share a thread pool for concurrent/parallel/lazy execution.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/util/CRC64Checksum.html" title="class in org.biojava3.core.util">CRC64Checksum</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utility class that calculates a CRC64 checksum on a stream of bytes.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/technology/CUFSL.html" title="class in bacnet.scripts.listeriomics.technology">CUFSL</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/Database.html" title="class in bacnet">Database</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A ModelProvider specifically design for the Website Database<br>
 Contains persistent variable, used through all the website.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/Database.InitDatabaseThread.html" title="class in bacnet">Database.InitDatabaseThread</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Class for loading databases in a thread</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/database/DatabaseCreation.html" title="class in bacnet.scripts.database">DatabaseCreation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">Deprecated.</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">If a SequenceProxyReader implements this interface then that external source has a list of cross
 reference id(s)</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/database/DataFolder.html" title="class in bacnet.scripts.database">DataFolder</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">GenBank gi|gi-number|gb|accession|locus ENA Data Library gi|gi-number|emb|accession|locus DDBJ,
 DNA Database of Japan gi|gi-number|dbj|accession|locus NBRF PIR pir||entry Protein Research
 Foundation prf||name SWISS-PROT UNIPROT sp|accession|name Brookhaven Protein Data Bank (1)
 pdb|entry|chain Brookhaven Protein Data Bank (2) entry:chain|PDBID|CHAIN|SEQUENCE Patents
 pat|country|number GenInfo Backbone Id bbs|number General database identifier
 gnl|database|identifier NCBI Reference Sequence ref|accession|locus Local Sequence identifier
 lcl|identifier</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">If you have a uniprot ID then it is possible to get a collection of other id(s) that the protein
 is known by.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/compound/DNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">DNACompoundSet</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/database/DNASeqCreation.html" title="class in bacnet.scripts.database">DNASeqCreation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">The type of DNA sequence</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/DNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">DNASequenceCreator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A helper class that allows different ways to read a string and create a DNA sequence.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava3.core.sequence.transcription">DNAToRNATranslator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Performs the first stage of transcription by going from DNA to RNA.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Specific tools for downloading files from eclipse.rap instance NEED TO BE COMMENTED FOR
 Eclipse.RCP to work !!!</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/edits/Edit.html" title="interface in org.biojava3.core.sequence.edits">Edit</a>&lt;<a href="org/biojava3/core/sequence/edits/Edit.html" title="type parameter in Edit">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Interface for carrying out edit operations on a Sequence.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="class in org.biojava3.core.sequence.edits">Edit.AbstractEdit</a>&lt;<a href="org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Abstract class which defines all edit operations as a call to discover what 5' and 3' ends of an
 editing Sequence should be joined together with a target Sequence.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/edits/Edit.Delete.html" title="class in org.biojava3.core.sequence.edits">Edit.Delete</a>&lt;<a href="org/biojava3/core/sequence/edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implementation which allows for the deletion of bases from a Sequence</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/edits/Edit.Insert.html" title="class in org.biojava3.core.sequence.edits">Edit.Insert</a>&lt;<a href="org/biojava3/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Edit implementation which allows us to insert a base at any position in a Sequence.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/edits/Edit.Substitute.html" title="class in org.biojava3.core.sequence.edits">Edit.Substitute</a>&lt;<a href="org/biojava3/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Allows for the substitution of bases into an existing Sequence.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/dataset/EGDeWTdata.html" title="class in bacnet.datamodel.dataset">EGDeWTdata</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">List of method for creating EGDe_Mean data: A data created from the mean of all Wild Type data
 (37C in BHI)</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/util/Equals.html" title="class in org.biojava3.core.util">Equals</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A set of helper methods which return true if the two parameters are equal to each other.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/ExonComparator.html" title="class in org.biojava3.core.sequence">ExonComparator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Sort Exon where it is a little confusing if exons shoud always be ordered left to right where a
 negative stranded gene should go the other direction.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A gene contains a collection of Exon sequences</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A list of methods to combine transcriptomics data in ExpressionMatrix</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Abstract Class describing data related to transcriptome <br>
 This data are saved in DataStream and might be read in streaming mode <br>
 <br>
 boolean[] read is used as a "read" indicator <br>
 As soon as a position in double[] values has been read, boolean[] read is updated <br></div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This is the principal data model used in the software for managing big table of expression</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/reader/ExpressionMatrixReader.html" title="class in bacnet.reader">ExpressionMatrixReader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">List of method for doing Stat on ExpressionMatrices</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a>&lt;<a href="bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">S</a> extends <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,<wbr><a href="bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/FastaGeneWriter.html" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A Gene sequence has a Positive or Negative Strand where we want to write out to a stream the 5 to
 3 prime version.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/io/template/FastaHeaderFormatInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderFormatInterface</a>&lt;<a href="org/biojava3/core/sequence/io/template/FastaHeaderFormatInterface.html" title="type parameter in FastaHeaderFormatInterface">S</a> extends <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,<wbr><a href="org/biojava3/core/sequence/io/template/FastaHeaderFormatInterface.html" title="type parameter in FastaHeaderFormatInterface">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="type parameter in FastaHeaderParserInterface">S</a> extends <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,<wbr><a href="org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="type parameter in FastaHeaderParserInterface">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/FastaReader.html" title="class in org.biojava3.core.sequence.io">FastaReader</a>&lt;<a href="org/biojava3/core/sequence/io/FastaReader.html" title="type parameter in FastaReader">S</a> extends <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,<wbr><a href="org/biojava3/core/sequence/io/FastaReader.html" title="type parameter in FastaReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Use FastaReaderHelper as an example of how to use this class where FastaReaderHelper should be
 the primary class used to read Fasta files</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/FastaSequenceParser.html" title="class in org.biojava3.core.sequence.io">FastaSequenceParser</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to parse a stream of a fasta file to get the sequence</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a>&lt;<a href="org/biojava3/core/sequence/io/FastaWriter.html" title="type parameter in FastaWriter">S</a> extends <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,<wbr><a href="org/biojava3/core/sequence/io/FastaWriter.html" title="type parameter in FastaWriter">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The FastaWriter writes a collection of sequences to an outputStream.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/FastaWriterHelper.html" title="class in org.biojava3.core.sequence.io">FastaWriterHelper</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">The class that should be used to write out fasta file of a sequence collection</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/stat/FDR.html" title="class in bacnet.scripts.core.stat">FDR</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A Feature corresponds to a single row in a GFF file.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/parsers/gff/FeatureHelper.html" title="class in org.biojava3.genome.parsers.gff">FeatureHelper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/genome/parsers/gff/FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">A feature on a sequence (for example, an exon or a gene), defined by a location and a set of
 attributes encoded as key/value pairs.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="org/biojava3/core/sequence/features/FeatureInterface.html" title="type parameter in FeatureInterface">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/features/FeatureInterface.html" title="type parameter in FeatureInterface">C</a>&gt;,<wbr><a href="org/biojava3/core/sequence/features/FeatureInterface.html" title="type parameter in FeatureInterface">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Interface class to handle describing arbitrary features.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A list of FeatureI objects implemented using a Java ArrayList; corresponds to a GFF file.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Models the keywords that are annotated for a protein sequence at Uniprot.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/technology/Febit.html" title="class in bacnet.scripts.listeriomics.technology">Febit</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab6"><a href="org/biojava3/core/exceptions/FileAccessError.html" title="class in org.biojava3.core.exceptions">FileAccessError</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab6">
<div class="block">Thrown from SequenceFileProxyLoader if error opening file using RandomnAccessFile</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyDNASequenceCreator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the
 stream the sequence and the offset index are passed to create a Protein sequence that will be
 loaded in lazily.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyProteinSequenceCreator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the
 stream the sequence and the offset index are passed to create a Protein sequence that will be
 loaded in lazily.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utils for file manipulation.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/table/core/Filter.TypeFilter.html" title="enum in bacnet.table.core">Filter.TypeFilter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/utils/Filter.TypeFilter.html" title="enum in bacnet.utils">Filter.TypeFilter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/gui/FilterDialog.html" title="class in bacnet.table.gui">FilterDialog</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Dialog for managing <code>Filter</code> for <code>ExpressionMatrix</code></div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/gui/FilterExpressionMatrixDialog.html" title="class in bacnet.table.gui">FilterExpressionMatrixDialog</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Dialog displaying the list of <code>Filters</code> currently used on the
 <code>ExpressionMatrix</code></div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/util/FlatFileCache.html" title="class in org.biojava3.core.util">FlatFileCache</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Provides a cache for storing multiple small files in memory.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/storage/FourBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader</a>&lt;<a href="org/biojava3/core/sequence/storage/FourBitSequenceReader.html" title="type parameter in FourBitSequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Four bit encoding of the bit formats.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader.FourBitArrayWorker</a>&lt;<a href="org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A four bit per compound implementation of the bit array worker code.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">Indicates a way of translating a sequence.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implementation for resolving fuzzy locations.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/reader/GBKReader.html" title="class in bacnet.reader">GBKReader</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">GenBankl reader tools <br>
 TO BE FIXED for adding other chromosomes</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/parsers/gff/GCStats.html" title="class in org.biojava3.genome.parsers.gff">GCStats</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Here are all the methods for displaying each Genome element</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">GeneExpression array data from Listeria tiling chip ArrayExpress id : A-AFFY-140
 
 Here double[] values vector has length = 5*probes.size because we put 5 columns column 0-2 = 3
 replicatas column 3 = median expression column 4 = var expression</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Allow to process genome elements annotation</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html" title="class in org.biojava3.genome.parsers.gff">GeneIDGFF2Reader</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">http://www.bioperl.org/wiki/GTF Read and write FeatureLists as GFF/GTF formatted files.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html" title="class in org.biojava3.genome.parsers.geneid">GeneIDXMLReader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html" title="class in org.biojava3.genome.parsers.gff">GeneMarkGTFReader</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">http://www.bioperl.org/wiki/GTF Read and write FeatureLists as GFF/GTF formatted files.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">List of tools useful to extract information hidden in the NoteList of a DNASequence</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/technology/GeneralArray.html" title="class in bacnet.scripts.listeriomics.technology">GeneralArray</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderFormat</a>&lt;<a href="org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="type parameter in GenericFastaHeaderFormat">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;?&gt;,<wbr><a href="org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="type parameter in GenericFastaHeaderFormat">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">We store the original header if the sequence is parsed from a fasta file and will use that exact
 sequence if we write out the sequences to a fasta file.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser</a>&lt;<a href="org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="type parameter in GenericFastaHeaderParser">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="type parameter in GenericFastaHeaderParser">C</a>&gt;,<wbr><a href="org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="type parameter in GenericFastaHeaderParser">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The default fasta header parser where some headers are well defined based on the source database
 which allows us to set the source of the protein sequence and the identifier that can be used in
 future implementations to load features from external sources
 
 If the user has a custom header with local data then they can create their own implementation of
 a FastaHeaderParserInterface

 GenBank gi|gi-number|gb|accession|locus ENA Data Library gi|gi-number|emb|accession|locus DDBJ,
 DNA Database of Japan gi|gi-number|dbj|accession|locus NBRF PIR pir||entry Protein Research
 Foundation prf||name SWISS-PROT sp|accession|name Brookhaven Protein Data Bank (1)
 pdb|entry|chain Brookhaven Protein Data Bank (2) entry:chain|PDBID|CHAIN|SEQUENCE PDB EBI
 PDB:1ECY_A mol:protein length:142 ECOTIN Patents pat|country|number GenInfo Backbone Id
 bbs|number General database identifier gnl|database|identifier NCBI Reference Sequence
 ref|accession|locus Local Sequence identifier lcl|identifier</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/sequenceTools/GeneViewHomologTools.html" title="class in bacnet.sequenceTools">GeneViewHomologTools</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">All the tools to display homologs in the Gene Panel view: - It creates the homolog table - Create
 the homolog figure - Display both of them</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/sequenceTools/GeneViewLocalizationTools.html" title="class in bacnet.sequenceTools">GeneViewLocalizationTools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/sequenceTools/GeneViewProteomeTools.html" title="class in bacnet.sequenceTools">GeneViewProteomeTools</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/sequenceTools/GeneViewTranscriptomeTools.html" title="class in bacnet.sequenceTools">GeneViewTranscriptomeTools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="bacnet/reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html" title="class in bacnet.datamodel.sequence">Genome.GetMultiFastaThread</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequence/Genome.OpenGenomesThread.html" title="class in bacnet.datamodel.sequence">Genome.OpenGenomesThread</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequenceNCBI/GenomeConversion.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversion</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Static methods for converting GenomeNCBI to Genome<br>
 Genome is used for fast loading of all genomic information we gathered</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Object containing an entire genome, created from NCBI RefSeq data<br>
 Data for this genome come from: <it> NCBI.fna (for the sequence) <it> NCBI.gff (for Gene, CDS,
 and ncRNA lists) If it exists some information are added: <it> Rfam.gff (other info) <it> supp
 info in a .txt file <it> supp info about RAST results in a .txt file For EGD-e only: <it>
 Srna.xml are read to add to the genome</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/blast/GenomeNCBIFolderTools.html" title="class in bacnet.scripts.blast">GenomeNCBIFolderTools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/phylogeny/GenomePhylogeny.html" title="class in bacnet.scripts.phylogeny">GenomePhylogeny</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Manage phylogeny from MLST scheme : Jolley and others</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">List of tools available for genome creation</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/GenomesListeriomics.html" title="class in bacnet.scripts.listeriomics">GenomesListeriomics</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Display on the genome viewer: expression of specific bioCond and the log of fold change</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/normalization/GENormalization.html" title="class in bacnet.scripts.core.normalization">GENormalization</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/parsers/gff/GFF3Reader.html" title="class in org.biojava3.genome.parsers.gff">GFF3Reader</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">http://www.bioperl.org/wiki/GTF Read and write FeatureLists as GFF/GTF formatted files.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/parsers/gff/GFF3Writer.html" title="class in org.biojava3.genome.parsers.gff">GFF3Writer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/reader/GFFNCBIReader.html" title="class in bacnet.reader">GFFNCBIReader</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Read GFF files to create a <a href="bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence"><code>Genome</code></a> object</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/annotation/GlaserFCannotation.html" title="class in bacnet.datamodel.annotation">GlaserFCannotation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/normalization/GRNormalization.html" title="class in bacnet.scripts.core.normalization">GRNormalization</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/util/Hashcoder.html" title="class in org.biojava3.core.util">Hashcoder</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Contains helper methods for generating a HashCode without having to resort to the commons lang
 hashcode builders.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab6"><a href="org/biojava3/core/exceptions/HeaderParseException.html" title="class in org.biojava3.core.exceptions">HeaderParseException</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab6">
<div class="block">Currently not used</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/HeatMapMultiOmicsView.InitOmicsMatrixThread.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView.InitOmicsMatrixThread</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/HeatMapProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsExpressionView</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/views/HelpPage.html" title="class in bacnet.views">HelpPage</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/gui/HideExpressionMatrixDialog.html" title="class in bacnet.table.gui">HideExpressionMatrixDialog</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/HTMLUtils.html" title="class in bacnet.utils">HTMLUtils</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">List of methods to manage D3js visualization in BACNET</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/technology/IHM.html" title="class in bacnet.scripts.listeriomics.technology">IHM</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/ImageMagick.html" title="class in bacnet.utils">ImageMagick</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Methods link to image mlanipulation with ImageMagick</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/views/InitView.html" title="class in bacnet.views">InitView</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/InitViewBacnet.html" title="class in bacnet.e4.rap">InitViewBacnet</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/InitViewListeria.html" title="class in bacnet.e4.rap">InitViewListeria</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">InitViewListeriaSample</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/InitViewYersinia.html" title="class in bacnet.e4.rap">InitViewYersinia</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/util/InputStreamProvider.html" title="class in org.biojava3.core.util">InputStreamProvider</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that provides an InputStream from a File.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/location/InsdcLocations.html" title="class in org.biojava3.core.sequence.location">InsdcLocations</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A collection of locations which are used whenever we work with INSDC; some of which could be
 deprecated (from INSDC's point of view) yet appear in records.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/location/InsdcLocations.BondLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.BondLocation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to represent bond locations equivalent to bond(7,8) or bond(7).</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Deprecated in INSDC yet still appears; equivalent to the order() directive except no 5' to 3'
 ordering is defined.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/location/InsdcLocations.OneOfLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OneOfLocation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Deprecated in INSDC; refers to a set of locations of which one location could be valid e.g.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OrderLocation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to describe a 5' to 3' ordering but no firm assurance it is correct</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Parser for working with INSDC style locations.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/io/util/IOUtils.ReaderProcessor.html" title="interface in org.biojava3.core.sequence.io.util">IOUtils.ReaderProcessor</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Closure interface used when working with <code>IOUtils#processReader(String)</code>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Available translations

 
 1 - UNIVERSAL
 2 - VERTEBRATE_MITOCHONDRIAL
 3 - YEAST_MITOCHONDRIAL
 4 - MOLD_MITOCHONDRIAL
 5 - INVERTEBRATE_MITOCHONDRIAL
 6 - CILIATE_NUCLEAR
 9 - ECHINODERM_MITOCHONDRIAL
 10 - EUPLOTID_NUCLEAR
 11 - BACTERIAL
 12 - ALTERNATIVE_YEAST_NUCLEAR
 13 - ASCIDIAN_MITOCHONDRIAL
 14 - FLATWORM_MITOCHONDRIAL
 15 - BLEPHARISMA_MACRONUCLEAR
 16 - 2CHLOROPHYCEAN_MITOCHONDRIAL
 21 - TREMATODE_MITOCHONDRIAL
 23 - SCENEDESMUS_MITOCHONDRIAL
 

 Taken from <a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c">NCBI</a>
 with slight modification and put into the classpath resource.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Holds the concept of a codon table from the IUPAC format</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/technology/JCVI.html" title="class in bacnet.scripts.listeriomics.technology">JCVI</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a>&lt;<a href="org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This reader actually proxies onto multiple types of sequence in order to allow a number of
 sequence objects to act as if they are one sequence.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a>&lt;<a href="org/biojava3/core/sequence/template/LightweightProfile.html" title="type parameter in LightweightProfile">S</a> extends <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="org/biojava3/core/sequence/template/LightweightProfile.html" title="type parameter in LightweightProfile">C</a>&gt;,<wbr><a href="org/biojava3/core/sequence/template/LightweightProfile.html" title="type parameter in LightweightProfile">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a minimal data structure for reading and writing a sequence alignment.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html" title="enum in org.biojava3.core.sequence.template">LightweightProfile.StringFormat</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">List of output formats.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/ListUtils.html" title="class in bacnet.utils">ListUtils</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/handler/LoadExpressionMatrixHandler.html" title="class in bacnet.handler">LoadExpressionMatrixHandler</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/sequenceTools/LocalizationDialog.html" title="class in bacnet.sequenceTools">LocalizationDialog</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Sets of integers used to represent the location of features on sequence.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A location on a sequence.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/location/template/Location.Tools.html" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Helper methods for use with the Location classes.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Helper methods for use with the Location classes.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Move a sliding window over a Location.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/annotation/LocusTag.html" title="class in bacnet.datamodel.annotation">LocusTag</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Since the update of RefSeq database all the locustag have been changed <br>
 Here is some methods to deal with this change</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/stat/LPE.html" title="class in bacnet.scripts.core.stat">LPE</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Class for calculating LPE statistics <br>
 Inspired from R library LPE<br></div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/technology/LT.html" title="class in bacnet.scripts.listeriomics.technology">LT</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/technology/Macroarray.html" title="class in bacnet.scripts.listeriomics.technology">Macroarray</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/MathUtils.html" title="class in bacnet.utils">MathUtils</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/normalization/MedianNormalization.html" title="class in bacnet.scripts.core.normalization">MedianNormalization</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a>&lt;<a href="org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">S</a> extends <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">C</a>&gt;,<wbr><a href="org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements a minimal data structure for reading and writing a sequence alignment.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/blast/MultiSequenceBlastProtein.html" title="class in bacnet.scripts.blast">MultiSequenceBlastProtein</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Every lmethods for managing navigation state in eclipse.rap application.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/reader/NCBIFastaHeaderParser.html" title="class in bacnet.reader">NCBIFastaHeaderParser</a>&lt;<a href="bacnet/reader/NCBIFastaHeaderParser.html" title="type parameter in NCBIFastaHeaderParser">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="bacnet/reader/NCBIFastaHeaderParser.html" title="type parameter in NCBIFastaHeaderParser">C</a>&gt;,<wbr><a href="bacnet/reader/NCBIFastaHeaderParser.html" title="type parameter in NCBIFastaHeaderParser">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">My own implementation of a FastaHeaderParserInterface to parse Fasta files from NCBI ftp

 NCBI fna (genome and plasmids) gi|gi-number|ref|accession|name gi|16802048|ref|NC_003210.1|
 Listeria monocytogenes EGD-e, complete genome NCBI ffn (coding) ref|refGenome
 gi-number|:start-end name ref|NC_003210.1|:318-1673 chromosomal replication initiation protein
 [Listeria monocytogenes EGD-e] NCBI frn (non-coding) ref|refGenome gi-number|:start-end| name|
 [locus-tag] ref|NC_003210|:82705-82777|Lys tRNA| [locus_tag=lmot01] NCBI faa (proteins)
 gi|gi-number|ref|accession|name gi|16802049|ref|NP_463534.1| chromosomal replication initiation
 protein [Listeria monocytogenes EGD-e]</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/datamodel/sequence/NcRNA.TypeNcRNA.html" title="enum in bacnet.datamodel.sequence">NcRNA.TypeNcRNA</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Network object for co-expression network analysis and visualisation</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/database/NetworkCreation.html" title="class in bacnet.scripts.database">NetworkCreation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Class for manipulating various RNASeq data, herited from ExpressionData <br>
 double[] value is of genome size, when ones want to see RNAseq expression at a certain bp
 position, just extract values[bpposition]<br>
 Need to be loaded first, and unlog has RNASeq will be logged for display when convert in the
 platform</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/database/NGSCreation.html" title="class in bacnet.scripts.database">NGSCreation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/technology/NimbleGen.html" title="class in bacnet.scripts.listeriomics.technology">NimbleGen</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Object for managing every mapped sequences</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/datamodel/proteomics/NTerm.ExperimentName.html" title="enum in bacnet.datamodel.proteomics">NTerm.ExperimentName</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/datamodel/proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">Type of modification at the start of a peptide</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/nterm/NTermConservationWithRiboProfiling.html" title="class in bacnet.scripts.listeriomics.nterm">NTermConservationWithRiboProfiling</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Run different method to compare iTIS position (internal translation) to data of Ribosome
 profiling in Bacillus and Ecoli:<br>
 compare the gene by BlastN
 find internal translation in the ribosome profiling
 find internal SD sequence</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Transform excel table describing all N-Terminal peptides found in Listeria<br>
 Find the position of each peptide in Listeria genome
 create a NGS data from it</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html" title="class in bacnet.scripts.listeriomics.nterm">NTermCreateFinalData</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Genertal class for creation of NTerminomics tables</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/nterm/NTermMain.html" title="class in bacnet.scripts.listeriomics.nterm">NTermMain</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">General method for running all methods related to NTerm project</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">List of method to do statistics about common features in N-Terminus of genes: - start codon - SD
 binding - AU proportion - Folding energy</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/nterm/NTermStatProteolyse.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStatProteolyse</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A list of methods to assess the quantity of proteolysis on the peptides</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/proteomics/NTermSummaryTable.html" title="class in bacnet.datamodel.proteomics">NTermSummaryTable</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">List of methods for creating Summary Table for TIS <br>
 Using NTermFilter we have regroup TIS in different overlap groups, which we use now to create the
 Summary Tables.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/nterm/NTermTranscriptome.html" title="class in bacnet.scripts.listeriomics.nterm">NTermTranscriptome</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Abstract Class describing data related to omics data <br></div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/datamodel/dataset/OmicsData.ColNames.html" title="enum in bacnet.datamodel.dataset">OmicsData.ColNames</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Enum referencing the different type of OmicsData possible:
 GeneExpr
 Tiling
 RNASeq
 TSS (Transcription Initiation Site)
 NTerm (Translation Initiation Site)
 Proteome (Matrix of protein found)
 MassSpec (List of peptides found by mass spectrometry)
 ExpressionMatrix (General matrix format)
 unknown</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/core/OpenExpressionMatrixAndComparisons.html" title="class in bacnet.expressionAtlas.core">OpenExpressionMatrixAndComparisons</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Run thread for opening BioCondition, Comparisons, detect Genome and open corresponding LogFC
 ExpressionMatrix</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/query/OutputHitsGFF.html" title="class in org.biojava3.genome.query">OutputHitsGFF</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab5"><a href="org/biojava3/core/exceptions/ParserException.html" title="class in org.biojava3.core.exceptions">ParserException</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab5">
<div class="block">General abstraction of different parsing errors</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/Peptidomics.html" title="class in bacnet.scripts.listeriomics">Peptidomics</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/phylogeny/Phylogenomic.html" title="class in bacnet.datamodel.phylogeny">Phylogenomic</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/database/PhylogenomicsCreation.html" title="class in bacnet.scripts.database">PhylogenomicsCreation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Different methods to create phylogeny tree, and tables for each genome elements</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/blast/PhylogenySmallORFs.html" title="class in bacnet.scripts.blast">PhylogenySmallORFs</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Tools to blast a small ORF on other bacterial genomes, and display results on a phylogeny tree</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/phylogeny/PhylogenySmallORFs.html" title="class in bacnet.scripts.phylogeny">PhylogenySmallORFs</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Tools to blast a small ORF on other bacterial genomes, and display results on a phylogeny tree</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Different methods to manipulate and display Newick and other phylogeny files</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This list of function is useful to manipulate Newick files given by Jolley in May 2012, from its
 publication Jolley et al.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Holds a single point part of a location</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/location/template/Point.Resolver.html" title="interface in org.biojava3.core.sequence.location.template">Point.Resolver</a>&lt;<a href="org/biojava3/core/sequence/location/template/Point.Resolver.html" title="type parameter in Point.Resolver">T</a> extends <a href="org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Used to resolve a position about a point</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/PredaTerm.html" title="class in bacnet.scripts.core">PredaTerm</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">PREDicting poly-A region and TERMinal splice acceptor for Leishmania Please read manuscript for
 details : Smith, M., Blanchette, M., Papadopoulou, B.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implementation of XMLWriter which emits nicely formatted documents to a PrintWriter.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The representation of a ProteinSequence</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/ProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">ProteinSequenceCreator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to create a ProteinSequence from a String to allow for details about the location of the
 sequence etc.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">OmicsData class inherited everything from ExpressionMatrix, the only difference is the TypeDate
 value, it a Proteome data.<br>
 It is not very useful for the moment but will be in the futur</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/ProteomicsExpressionView.html" title="class in bacnet.expressionAtlas">ProteomicsExpressionView</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="org/biojava3/core/sequence/template/ProxySequenceReader.html" title="type parameter in ProxySequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/reader/PTTReader.html" title="class in bacnet.reader">PTTReader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/features/QualityFeature.html" title="class in org.biojava3.core.sequence.features">QualityFeature</a>&lt;<a href="org/biojava3/core/sequence/features/QualityFeature.html" title="type parameter in QualityFeature">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/features/QualityFeature.html" title="type parameter in QualityFeature">C</a>&gt;,<wbr><a href="org/biojava3/core/sequence/features/QualityFeature.html" title="type parameter in QualityFeature">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">DNA Sequences produced by modern sequencers usually have quality informaion attached to them.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/features/QuantityFeature.html" title="class in org.biojava3.core.sequence.features">QuantityFeature</a>&lt;<a href="org/biojava3/core/sequence/features/QuantityFeature.html" title="type parameter in QuantityFeature">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/features/QuantityFeature.html" title="type parameter in QuantityFeature">C</a>&gt;,<wbr><a href="org/biojava3/core/sequence/features/QuantityFeature.html" title="type parameter in QuantityFeature">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">It is common to have a numerical value or values associated with a feature.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/genome/RASTEGDeGenome.html" title="class in bacnet.scripts.genome">RASTEGDeGenome</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">List of methods for adding RAST genes to EGD-e annotation</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Methods to look at overlapping elements between NCBI and RAST genomes</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/RegulatorRegion.html" title="class in bacnet.scripts.listeriomics">RegulatorRegion</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utility class for managing OS resources associated with SWT/JFace controls such as colors, fonts,
 images, etc.
 
 !!!</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="bacnet/swt/ResourceManager.PluginResourceProvider.html" title="interface in bacnet.swt">ResourceManager.PluginResourceProvider</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Provider for plugin resources, used by WindowBuilder at design time.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views">ReversedSequenceView</a>&lt;<a href="org/biojava3/core/sequence/views/ReversedSequenceView.html" title="type parameter in ReversedSequenceView">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">For a given sequence this class will return the base at the reversed position i.e. in a sequence
 of size 10, if you request base 2 you will get back the base at position 9.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/compound/RNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">RNACompoundSet</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">RNASequence where RNACompoundSet are the allowed values</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/io/RNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">RNASequenceCreator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to create a RNA sequence</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Attempts to do on the fly translation of RNA by not requesting the compounds until asked.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html" title="class in org.biojava3.core.sequence.transcription">RNAToAminoAcidTranslator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Takes a <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a> of <a href="org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound"><code>NucleotideCompound</code></a> which should represent an RNA sequence
 (<a href="org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence"><code>RNASequence</code></a> is good for this) and returns a list of <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a> which hold
 <a href="org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound"><code>AminoAcidCompound</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/Rscript.html" title="class in bacnet.scripts.core">Rscript</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Tools for saving files<br>
 Everything in bacnet.raprcp package is specific to eclipse.rap or eclipse.rcp.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/raprcp/SaveFileUtils.html" title="class in bacnet.raprcp">SaveFileUtils</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utils methods for managinf files.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Calculate Shine-Dalgarno profile for bacterial genome.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/sequenceTools/SecondStructureRNADialog.html" title="class in bacnet.sequenceTools">SecondStructureRNADialog</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/core/SelectGenomeElementDialog.html" title="class in bacnet.expressionAtlas.core">SelectGenomeElementDialog</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Dialog which allow specific selection of genes, smallRNAs and cisRegs</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="org/biojava3/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Main interface for defining a collection of Compounds and accessing them using biological indexes</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/datamodel/sequence/Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="class in org.biojava3.core.sequence.storage">SequenceAsStringHelper</a>&lt;<a href="org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="type parameter in SequenceAsStringHelper">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This is a common method that can be used across multiple storage/proxy implementations to handle
 Negative strand and other interesting elements of sequence data.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/SequenceComparator.html" title="class in org.biojava3.core.sequence">SequenceComparator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to sort sequences</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a>&lt;<a href="org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class represents the storage container of a sequence stored in a fasta file where the
 initial parsing of the file we store the offset and length of the sequence.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab6"><a href="org/biojava3/core/exceptions/SequenceLengthError.html" title="class in org.biojava3.core.exceptions">SequenceLengthError</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab6">
<div class="block">Currently not used</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/location/SequenceLocation.html" title="class in org.biojava3.core.sequence.location">SequenceLocation</a>&lt;<a href="org/biojava3/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">C</a>&gt;,<wbr><a href="org/biojava3/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A location in a sequence that keeps a reference to its parent sequence</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Provides a set of static methods to be used as static imports when needed across multiple
 Sequence implementations but inheritance gets in the way.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="class in org.biojava3.core.sequence.template">SequenceMixin.SequenceIterator</a>&lt;<a href="org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="type parameter in SequenceMixin.SequenceIterator">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A basic sequence iterator which iterates over the given Sequence by biological index.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A static class that provides optimization hints for memory or performance handling of sequence
 data.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava3/core/sequence/SequenceOptimizationHints.SequenceCollection.html" title="enum in org.biojava3.core.sequence">SequenceOptimizationHints.SequenceCollection</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava3/core/sequence/SequenceOptimizationHints.SequenceUsage.html" title="enum in org.biojava3.core.sequence">SequenceOptimizationHints.SequenceUsage</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/io/template/SequenceParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceParserInterface</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a>&lt;<a href="org/biojava3/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="org/biojava3/core/sequence/template/SequenceReader.html" title="type parameter in SequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/util/SequenceTools.html" title="class in org.biojava3.core.util">SequenceTools</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="org/biojava3/core/sequence/template/SequenceView.html" title="type parameter in SequenceView">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/SessionControl.html" title="class in bacnet.e4.rap">SessionControl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/setup/SessionControl.html" title="class in bacnet.e4.rap.setup">SessionControl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">General methods for session control of an eclipse.rap application</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/setup/SetupPart.html" title="class in bacnet.e4.rap.setup">SetupPart</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">SetupPart for initialisation and creation of the Database</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/core/SignatureSelectionDialog.html" title="class in bacnet.expressionAtlas.core">SignatureSelectionDialog</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Very basic implementation of the Location interface which defines a series of simple
 constructors.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Basic implementation of the Point interface.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a>&lt;<a href="org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">An implementation of the SequenceReader interface which for every call will return only 1
 compound (given to it during construction; a String is also valid but will require a
 CompoundSet).</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html" title="class in bacnet.scripts.listeriomics.srna">SmallPeptidesSearch</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">Deprecated.</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a>&lt;<a href="org/biojava3/core/util/SoftHashMap.html" title="type parameter in SoftHashMap">K</a>,<wbr><a href="org/biojava3/core/util/SoftHashMap.html" title="type parameter in SoftHashMap">V</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A in memory cache using soft references.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/util/SplitFasta.html" title="class in org.biojava3.genome.util">SplitFasta</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utility to write each Fasta entry to a unique file</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/datamodel/sequence/Srna.TypeSrna.html" title="enum in bacnet.datamodel.sequence">Srna.TypeSrna</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/srna/Srna10403SOliver.html" title="class in bacnet.scripts.listeriomics.srna">Srna10403SOliver</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/srna/SrnaFinalList.html" title="class in bacnet.scripts.listeriomics.srna">SrnaFinalList</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/SrnaListeriomics.html" title="class in bacnet.scripts.listeriomics">SrnaListeriomics</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">List of methods to manage Srna data for Listeriomics</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/sequenceTools/SrnaSummaryView.html" title="class in bacnet.sequenceTools">SrnaSummaryView</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Include a list of Methods to read every Table containing Small RNAs in Listeria<br>
 For each it reads the <code>Table</code> and create a <code>List</code> of <code>Srna</code></div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/genome/StartCodonProfile.html" title="class in bacnet.scripts.genome">StartCodonProfile</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Calculate presence of start codon on firrst chromosome of bacterial genomes</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/StartCodonSequence.html" title="class in org.biojava3.core.sequence">StartCodonSequence</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to map the start codon feature on a gene</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/stat/StatTest.html" title="class in bacnet.scripts.core.stat">StatTest</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/stat/StatUtils.html" title="class in bacnet.scripts.core.stat">StatUtils</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/StopCodonSequence.html" title="class in org.biojava3.core.sequence">StopCodonSequence</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to map the stop codon sequence on a gene</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Provides a way of representing the strand of a sequence, location hit or feature.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/StringColor.html" title="class in bacnet.utils">StringColor</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/util/StringManipulationHelper.html" title="class in org.biojava3.core.util">StringManipulationHelper</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A utility class for common <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link"><code>String</code></a> manipulation tasks.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a>&lt;<a href="org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">An example of a ProxySequenceReader that is created from a String.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/annotation/SubCellCompartment.html" title="class in bacnet.datamodel.annotation">SubCellCompartment</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">All data and methods for Subcellular compartment information are done here</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/SummaryListeriomics.html" title="class in bacnet.scripts.listeriomics">SummaryListeriomics</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Methods for creating the different figures and Tables of the paper</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utility class for managing OS resources associated with SWT controls such as colors, fonts,
 images, etc.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/SystemsBiologyListeriomics.html" title="class in bacnet.scripts.listeriomics">SystemsBiologyListeriomics</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">List of method to create Co-expression network and analyse it</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Provides a way of separating us from the specific <a href="org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io"><code>IUPACParser.IUPACTable</code></a> even though this is the only
 implementing class for the interface.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Class used to hold three nucleotides together and allow for equality to be assessed in a case
 insensitive manner.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Instance of a Codon which is 3 <a href="org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound"><code>NucleotideCompound</code></a>s, its corresponding
 <a href="org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound"><code>AminoAcidCompound</code></a> and if it is a start or stop codon.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">The Composite in which is displayed the heatmap</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/gui/TableFilter.html" title="class in bacnet.table.gui">TableFilter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/TableHTMLExport.html" title="class in bacnet.utils">TableHTMLExport</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/TableSWTComposite.html" title="class in bacnet.table">TableSWTComposite</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/table/core/TableViewerComparator.html" title="class in bacnet.table.core">TableViewerComparator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/technology/Tasmania.html" title="class in bacnet.scripts.listeriomics.technology">Tasmania</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A sequence can be associated with a species or Taxonomy ID</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/setup/Test.html" title="class in bacnet.e4.rap.setup">Test</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">All methods which will be executed before or after interface run</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/genomeBrowser/TestGenomeBrowser.html" title="class in bacnet.genomeBrowser">TestGenomeBrowser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/e4/rap/TestPart.html" title="class in bacnet.e4.rap">TestPart</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/features/TextFeature.html" title="class in org.biojava3.core.sequence.features">TextFeature</a>&lt;<a href="org/biojava3/core/sequence/features/TextFeature.html" title="type parameter in TextFeature">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/features/TextFeature.html" title="type parameter in TextFeature">C</a>&gt;,<wbr><a href="org/biojava3/core/sequence/features/TextFeature.html" title="type parameter in TextFeature">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A implmentation of AbstractFeature</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Tiling array data from Listeria tiling chip ArrayExpress id : A-MEXP-1572 *</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/TilingGeneExprDataBase.html" title="class in bacnet.scripts.listeriomics">TilingGeneExprDataBase</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class is used to regroup different Nterm.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="bacnet/genomeBrowser/core/Track.DisplayType.html" title="enum in bacnet.genomeBrowser.core">Track.DisplayType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A TrackCanvasData is the Canvas in which will be displayed datasets</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A TrackCanvasGenome is the Canvas in which will be displayed genome annotation</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">In TracksComposite both <code>TrackCanvasData</code> and <code>TrackCanvasGenome</code> will be
 displayed.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used as a way of encapsulating the data structures required to parse DNA to a Protein sequence.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class is the way to create a <code>TranslationEngine</code>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/database/TranscriptomesCreation.html" title="class in bacnet.scripts.database">TranscriptomesCreation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">List of methods to add the different transcriptomic datasetsto your multi-omics website</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">View in which is displayed the table with all transcriptomics data available</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This is the sequence if you want to go from a gene sequence to a protein sequence.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab5"><a href="org/biojava3/core/exceptions/TranslationException.html" title="class in org.biojava3.core.exceptions">TranslationException</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab5">
<div class="block">Thrown from AbstractCompundTranslator</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">List of methods to process TSS, TermSeq, and RiboSeq of Listeriomics Some TSS and NTerm lists
 were provided in tab-delimited format.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader</a>&lt;<a href="org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="type parameter in TwoBitSequenceReader">C</a> extends <a href="org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implementation of the 2bit encoding.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader.TwoBitArrayWorker</a>&lt;<a href="org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</a> extends <a href="org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Extension of the BitArrayWorker which provides the 2bit implementation code.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/technology/UGiessen.html" title="class in bacnet.scripts.listeriomics.technology">UGiessen</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class decompresses an input stream containing data compressed with the unix "compress"
 utility (LZC, a LZW variant).</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a>&lt;<a href="org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Pass in a Uniprot ID and this ProxySequenceReader when passed to a ProteinSequence will get the
 sequence data and other data elements associated with the ProteinSequence by Uniprot.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/genome/uniprot/UniprotToFasta.html" title="class in org.biojava3.genome.uniprot">UniprotToFasta</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/normalization/VarianceNormalization.html" title="class in bacnet.scripts.core.normalization">VarianceNormalization</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Variance normalization methods which will reduce the total variance of each data to 1<br>
 For each column of the ExpressionMatrix, multiply each value by: Math.sqr(1/var(x))<br>
 So final variance of each column is equal to zero</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Vector are array with one dimension like : <code>String[]</code> or <code>double[]</code></div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Class for creating sets necessary in Venn diagrams<br>
 Providing a list of sets (through <code>ArrayList<String></code>) it will find intersections,
 unions and differences between all sets.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/sequence/views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence</a>&lt;<a href="org/biojava3/core/sequence/views/WindowedSequence.html" title="type parameter in WindowedSequence">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A sliding window view of a sequence which does not implement any interfaces like <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a>
 because they do not fit how this works.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/technology/Wuerzburg.html" title="class in bacnet.scripts.listeriomics.technology">Wuerzburg</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/scripts/listeriomics/technology/WUSTL.html" title="class in bacnet.scripts.listeriomics.technology">WUSTL</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/utils/XMLUtils.html" title="class in bacnet.utils">XMLUtils</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Simple interface for building XML documents.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
</div>
</div>
</div>
</main>
</div>
</div>
</body>
</html>
