<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>All Classes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="script.js"></script>
<script type="text/javascript" src="jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="All Classes";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":2,"i1":2,"i2":2,"i3":2,"i4":2,"i5":2,"i6":2,"i7":4,"i8":1,"i9":1,"i10":2,"i11":2,"i12":2,"i13":2,"i14":2,"i15":2,"i16":2,"i17":2,"i18":2,"i19":2,"i20":2,"i21":2,"i22":2,"i23":2,"i24":2,"i25":2,"i26":2,"i27":2,"i28":2,"i29":2,"i30":2,"i31":2,"i32":2,"i33":2,"i34":2,"i35":2,"i36":2,"i37":2,"i38":2,"i39":2,"i40":2,"i41":2,"i42":2,"i43":2,"i44":2,"i45":2,"i46":2,"i47":2,"i48":2,"i49":2,"i50":2,"i51":2,"i52":2,"i53":2,"i54":2,"i55":2,"i56":2,"i57":4,"i58":2,"i59":2,"i60":2,"i61":2,"i62":2,"i63":2,"i64":2,"i65":2,"i66":2,"i67":2,"i68":2,"i69":2,"i70":2,"i71":2,"i72":2,"i73":2,"i74":2,"i75":2,"i76":2,"i77":2,"i78":2,"i79":2,"i80":2,"i81":2,"i82":2,"i83":4,"i84":2,"i85":2,"i86":4,"i87":2,"i88":2,"i89":2,"i90":2,"i91":2,"i92":2,"i93":2,"i94":2,"i95":2,"i96":1,"i97":2,"i98":2,"i99":1,"i100":2,"i101":16,"i102":1,"i103":1,"i104":2,"i105":2,"i106":2,"i107":2,"i108":2,"i109":2,"i110":2,"i111":2,"i112":2,"i113":1,"i114":2,"i115":4,"i116":2,"i117":2,"i118":2,"i119":2,"i120":2,"i121":2,"i122":2,"i123":2,"i124":4,"i125":2,"i126":2,"i127":2,"i128":1,"i129":2,"i130":2,"i131":2,"i132":2,"i133":2,"i134":2,"i135":2,"i136":2,"i137":2,"i138":2,"i139":2,"i140":2,"i141":2,"i142":2,"i143":2,"i144":2,"i145":1,"i146":1,"i147":2,"i148":2,"i149":2,"i150":2,"i151":2,"i152":2,"i153":2,"i154":2,"i155":1,"i156":1,"i157":2,"i158":1,"i159":2,"i160":16,"i161":2,"i162":2,"i163":2,"i164":2,"i165":2,"i166":4,"i167":4,"i168":2,"i169":2,"i170":2,"i171":2,"i172":2,"i173":2,"i174":4,"i175":2,"i176":2,"i177":2,"i178":2,"i179":2,"i180":2,"i181":2,"i182":2,"i183":2,"i184":2,"i185":2,"i186":2,"i187":2,"i188":2,"i189":2,"i190":2,"i191":2,"i192":2,"i193":2,"i194":2,"i195":2,"i196":2,"i197":2,"i198":2,"i199":2,"i200":2,"i201":2,"i202":2,"i203":2,"i204":2,"i205":2,"i206":2,"i207":2,"i208":2,"i209":2,"i210":2,"i211":2,"i212":2,"i213":2,"i214":2,"i215":2,"i216":2,"i217":16,"i218":2,"i219":2,"i220":2,"i221":2,"i222":2,"i223":2,"i224":2,"i225":2,"i226":2,"i227":2,"i228":2,"i229":2,"i230":2,"i231":2,"i232":2,"i233":2,"i234":2,"i235":2,"i236":2,"i237":2,"i238":2,"i239":2,"i240":2,"i241":2,"i242":2,"i243":2,"i244":2,"i245":1,"i246":2,"i247":2,"i248":2,"i249":2,"i250":2,"i251":2,"i252":1,"i253":4,"i254":2,"i255":2,"i256":2,"i257":1,"i258":2,"i259":2,"i260":1,"i261":2,"i262":1,"i263":2,"i264":2,"i265":2,"i266":2,"i267":2,"i268":2,"i269":2,"i270":2,"i271":2,"i272":2,"i273":2,"i274":2,"i275":2,"i276":2,"i277":2,"i278":4,"i279":2,"i280":2,"i281":2,"i282":2,"i283":2,"i284":2,"i285":4,"i286":4,"i287":4,"i288":2,"i289":2,"i290":2,"i291":2,"i292":2,"i293":2,"i294":2,"i295":2,"i296":2,"i297":2,"i298":2,"i299":2,"i300":2,"i301":2,"i302":2,"i303":4,"i304":4,"i305":2,"i306":2,"i307":8,"i308":2,"i309":2,"i310":2,"i311":2,"i312":2,"i313":2,"i314":2,"i315":2,"i316":1,"i317":1,"i318":2,"i319":2,"i320":2,"i321":2,"i322":2,"i323":2,"i324":2,"i325":2,"i326":2,"i327":1,"i328":2,"i329":2,"i330":2,"i331":2,"i332":2,"i333":2,"i334":2,"i335":2,"i336":1,"i337":2,"i338":2,"i339":2,"i340":2,"i341":2,"i342":2,"i343":2,"i344":2,"i345":2,"i346":2,"i347":2,"i348":2,"i349":2,"i350":2,"i351":2,"i352":1,"i353":4,"i354":2,"i355":2,"i356":1,"i357":2,"i358":2,"i359":16,"i360":2,"i361":2,"i362":2,"i363":2,"i364":4,"i365":4,"i366":1,"i367":2,"i368":1,"i369":2,"i370":1,"i371":2,"i372":2,"i373":2,"i374":2,"i375":2,"i376":2,"i377":2,"i378":2,"i379":2,"i380":2,"i381":2,"i382":2,"i383":2,"i384":4,"i385":2,"i386":2,"i387":2,"i388":2,"i389":2,"i390":2,"i391":2,"i392":2,"i393":2,"i394":2,"i395":4,"i396":2,"i397":2,"i398":4,"i399":2,"i400":2,"i401":2,"i402":2,"i403":4,"i404":2,"i405":2,"i406":2,"i407":2,"i408":2,"i409":1,"i410":2,"i411":2,"i412":2,"i413":2,"i414":2,"i415":2,"i416":2,"i417":2,"i418":2,"i419":2,"i420":2,"i421":2,"i422":2,"i423":2,"i424":2,"i425":2,"i426":2,"i427":2,"i428":4,"i429":2,"i430":2,"i431":2,"i432":2,"i433":2,"i434":2,"i435":2,"i436":2,"i437":2,"i438":2,"i439":2,"i440":8,"i441":2,"i442":2,"i443":2,"i444":2,"i445":2,"i446":2,"i447":2,"i448":2,"i449":2,"i450":2,"i451":2,"i452":2,"i453":2,"i454":2,"i455":2,"i456":2,"i457":2,"i458":2,"i459":1,"i460":2};
var tabs = {65535:["t0","All Classes"],1:["t1","Interface Summary"],2:["t2","Class Summary"],4:["t3","Enum Summary"],8:["t4","Exception Summary"],16:["t5","Error Summary"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "./";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="overview-tree.html">Tree</a></li>
<li><a href="deprecated-list.html">Deprecated</a></li>
<li><a href="index-files/index-1.html">Index</a></li>
<li><a href="help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h1 title="All&amp;nbsp;Classes" class="title">All&nbsp;Classes</h1>
</div>
<div class="allClassesContainer">
<ul class="blockList">
<li class="blockList">
<table class="typeSummary">
<caption><span id="t0" class="activeTableTab"><span>All Classes</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Interface Summary</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Class Summary</a></span><span class="tabEnd">&nbsp;</span></span><span id="t3" class="tableTab"><span><a href="javascript:show(4);">Enum Summary</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Exception Summary</a></span><span class="tabEnd">&nbsp;</span></span><span id="t5" class="tableTab"><span><a href="javascript:show(16);">Error Summary</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/AbstractCompound.html" title="class in org.biojava3.core.sequence.template">AbstractCompound</a></td>
<th class="colLast" scope="row">
<div class="block">The details of a Compound</div>
</th>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundSet</a>&lt;<a href="org/biojava3/core/sequence/template/AbstractCompoundSet.html" title="type parameter in AbstractCompoundSet">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava3.core.sequence.template">AbstractCompoundTranslator</a>&lt;<a href="org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>,&#8203;<a href="org/biojava3/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/features/AbstractFeature.html" title="class in org.biojava3.core.sequence.features">AbstractFeature</a>&lt;<a href="org/biojava3/core/sequence/features/AbstractFeature.html" title="type parameter in AbstractFeature">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/features/AbstractFeature.html" title="type parameter in AbstractFeature">C</a>&gt;,&#8203;<a href="org/biojava3/core/sequence/features/AbstractFeature.html" title="type parameter in AbstractFeature">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s) A
 feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</th>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava3.core.sequence.location.template">AbstractLocation</a></td>
<th class="colLast" scope="row">
<div class="block">Base abstraction of a location which encodes for the majority of important features about a
 location such as the start, end and strand</div>
</th>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html" title="class in org.biojava3.core.sequence.template">AbstractNucleotideCompoundSet</a>&lt;<a href="org/biojava3/core/sequence/template/AbstractNucleotideCompoundSet.html" title="type parameter in AbstractNucleotideCompoundSet">C</a> extends <a href="org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">The base class for DNA, RNA and Protein sequences.</div>
</th>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/template/AccesionedLocation.html" title="interface in org.biojava3.core.sequence.location.template">AccesionedLocation</a></td>
<th class="colLast" scope="row">
<div class="block">A location which is bound to an AccessionID.</div>
</th>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</a></td>
<th class="colLast" scope="row">
<div class="block">Indicates an entity is accessioned</div>
</th>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a></td>
<th class="colLast" scope="row">
<div class="block">Used in Sequences as the unique indentifier.</div>
</th>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><a href="bacnet/Activator.html" title="class in bacnet">Activator</a></td>
<th class="colLast" scope="row">
<div class="block">OSGI Activator of bacnet plugin</div>
</th>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><a href="bacnet/e4/rap/Activator.html" title="class in bacnet.e4.rap">Activator</a></td>
<th class="colLast" scope="row">
<div class="block">OSGI Activator of bacnet.e4.rap</div>
</th>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><a href="bacnet/e4/rap/setup/Activator.html" title="class in bacnet.e4.rap.setup">Activator</a></td>
<th class="colLast" scope="row">
<div class="block">OSGI Activator of bacnet.e4.rap.setup</div>
</th>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/Activator.html" title="class in bacnet.scripts">Activator</a></td>
<th class="colLast" scope="row">
<div class="block">OSGI Activator of bacnet.scripts</div>
</th>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddProteomicsDataDialog</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">AddTranscriptomicsDataDialog</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/technology/Agilent.html" title="class in bacnet.scripts.listeriomics.technology">Agilent</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/compound/AmbiguityDNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">AmbiguityDNACompoundSet</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/compound/AmbiguityDNACompoundSet.InitaliseOnDemand.html" title="class in org.biojava3.core.sequence.compound">AmbiguityDNACompoundSet.InitaliseOnDemand</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/compound/AmbiguityRNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">AmbiguityRNACompoundSet</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/compound/AmbiguityRNACompoundSet.InitaliseOnDemand.html" title="class in org.biojava3.core.sequence.compound">AmbiguityRNACompoundSet.InitaliseOnDemand</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompound</a></td>
<th class="colLast" scope="row">
<div class="block">Used to describe an Amino Acid.</div>
</th>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava3.core.sequence.compound">AminoAcidCompoundSet</a></td>
<th class="colLast" scope="row">
<div class="block">Set of proteinogenic amino acids.</div>
</th>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/core/stat/AMtools.html" title="class in bacnet.scripts.core.stat">AMtools</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><a href="bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">AnnotationView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i27" class="rowColor">
<td class="colFirst"><a href="org/biojava3/genome/App.html" title="class in org.biojava3.genome">App</a></td>
<th class="colLast" scope="row">
<div class="block">Hello world!</div>
</th>
</tr>
<tr id="i28" class="altColor">
<td class="colFirst"><a href="bacnet/e4/rap/AppSpecificMethods.html" title="class in bacnet.e4.rap">AppSpecificMethods</a></td>
<th class="colLast" scope="row">
<div class="block">All the methods which are specific to a webpage</div>
</th>
</tr>
<tr id="i29" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/arrayexpress/ArrayExpress.html" title="class in bacnet.scripts.arrayexpress">ArrayExpress</a></td>
<th class="colLast" scope="row">
<div class="block">Different methods to curate and parse the different data for Listeriomics website <br>
 <br>
 
 ONLY METHODS BEFORE FINAL DATA PROCESSING ARE PRESENT HERE</div>
</th>
</tr>
<tr id="i30" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/arrayexpress/ArrayExpressDataImport.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataImport</a></td>
<th class="colLast" scope="row">
<div class="block">List of methods to download all ArrayExpress needed for Listeriomics website</div>
</th>
</tr>
<tr id="i31" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/arrayexpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressDataUtils</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i32" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressDataUtils.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressDataUtils</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i33" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressListeriomics</a></td>
<th class="colLast" scope="row">
<div class="block">Different methods to curate and parse the different data for Listeriomics website <br>
 <br>
 
 ONLY METHODS BEFORE FINAL DATA PROCESSING ARE PRESENT HERE</div>
</th>
</tr>
<tr id="i34" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/ArrayExpressListeriomics.html" title="class in bacnet.scripts.listeriomics">ArrayExpressListeriomics</a></td>
<th class="colLast" scope="row">
<div class="block">Different methods to curate and parse the different data for Listeriomics website <br>
 <br>
 
 ONLY METHODS BEFORE FINAL DATA PROCESSING ARE PRESENT HERE</div>
</th>
</tr>
<tr id="i35" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/arrayexpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.arrayexpress">ArrayExpressTechnology</a></td>
<th class="colLast" scope="row">
<div class="block">List of methods to determine the different technology used</div>
</th>
</tr>
<tr id="i36" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/ArrayExpress/ArrayExpressTechnology.html" title="class in bacnet.scripts.listeriomics.ArrayExpress">ArrayExpressTechnology</a></td>
<th class="colLast" scope="row">
<div class="block">List of methods to determine the different technology used</div>
</th>
</tr>
<tr id="i37" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/loader/ArrayListProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">ArrayListProxySequenceReader</a>&lt;<a href="org/biojava3/core/sequence/loader/ArrayListProxySequenceReader.html" title="type parameter in ArrayListProxySequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i38" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava3.core.sequence.storage">ArrayListSequenceReader</a>&lt;<a href="org/biojava3/core/sequence/storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Stores a Sequence as a collection of compounds in an ArrayList</div>
</th>
</tr>
<tr id="i39" class="rowColor">
<td class="colFirst"><a href="bacnet/utils/ArrayUtils.html" title="class in bacnet.utils">ArrayUtils</a></td>
<th class="colLast" scope="row">
<div class="block">Public class with general array manipulation utilities, not specific to PROJECTNAME.</div>
</th>
</tr>
<tr id="i40" class="altColor">
<td class="colFirst"><a href="bacnet/e4/rap/BacnetE4PointFactory.html" title="class in bacnet.e4.rap">BacnetE4PointFactory</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i41" class="rowColor">
<td class="colFirst"><a href="bacnet/e4/rap/setup/BacnetE4PointFactory.html" title="class in bacnet.e4.rap.setup">BacnetE4PointFactory</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i42" class="altColor">
<td class="colFirst"><a href="bacnet/e4/rap/BannerView.html" title="class in bacnet.e4.rap">BannerView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i43" class="rowColor">
<td class="colFirst"><a href="bacnet/views/BannerView.html" title="class in bacnet.views">BannerView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i44" class="altColor">
<td class="colFirst"><a href="bacnet/e4/rap/BasicApplication.html" title="class in bacnet.e4.rap">BasicApplication</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i45" class="rowColor">
<td class="colFirst"><a href="bacnet/e4/rap/setup/BasicApplication.html" title="class in bacnet.e4.rap.setup">BasicApplication</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i46" class="altColor">
<td class="colFirst"><a href="bacnet/utils/BasicColor.html" title="class in bacnet.utils">BasicColor</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i47" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/BasicSequence.html" title="class in org.biojava3.core.sequence">BasicSequence</a>&lt;<a href="org/biojava3/core/sequence/BasicSequence.html" title="type parameter in BasicSequence">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Bare bones version of the Sequence object to be used sparingly.</div>
</th>
</tr>
<tr id="i48" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a></td>
<th class="colLast" scope="row">
<div class="block">This class is used for the different bioconds declared in the experiment
 
 It contains the biocond name, and all the related data.</div>
</th>
</tr>
<tr id="i49" class="rowColor">
<td class="colFirst"><a href="bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i50" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/database/BioConditionCreation.html" title="class in bacnet.scripts.database">BioConditionCreation</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i51" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader</a>&lt;<a href="org/biojava3/core/sequence/storage/BitSequenceReader.html" title="type parameter in BitSequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">An implementation of the popular bit encodings.</div>
</th>
</tr>
<tr id="i52" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">BitSequenceReader.BitArrayWorker</a>&lt;<a href="org/biojava3/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">The logic of working with a bit has been separated out into this class to help developers create
 the bit data structures without having to put the code into an intermediate format and to also
 use the format without the need to copy this code.</div>
</th>
</tr>
<tr id="i53" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/blast/Blast.html" title="class in bacnet.scripts.blast">Blast</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i54" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/blast/BlastDatabase.html" title="class in bacnet.scripts.blast">BlastDatabase</a></td>
<th class="colLast" scope="row">
<div class="block">Different methods to create blast database for the genome, the ORF, and the smallORF</div>
</th>
</tr>
<tr id="i55" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/blast/gui/BlastGenomeSelectionPage.html" title="class in bacnet.scripts.blast.gui">BlastGenomeSelectionPage</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i56" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/blast/BlastOutput.html" title="class in bacnet.scripts.blast">BlastOutput</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i57" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/blast/BlastOutput.BlastOutputTYPE.html" title="enum in bacnet.scripts.blast">BlastOutput.BlastOutputTYPE</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i58" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/blast/gui/BlastParametersPage.html" title="class in bacnet.scripts.blast.gui">BlastParametersPage</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i59" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/blast/gui/BlastQueryPage.html" title="class in bacnet.scripts.blast.gui">BlastQueryPage</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i60" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/blast/BlastResult.html" title="class in bacnet.scripts.blast">BlastResult</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i61" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/blast/BlastResult.BlastResultCompare.html" title="class in bacnet.scripts.blast">BlastResult.BlastResultCompare</a></td>
<th class="colLast" scope="row">
<div class="block">This comparator classify the BlastResult by their position on the genome</div>
</th>
</tr>
<tr id="i62" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/blast/BlastResultView.html" title="class in bacnet.scripts.blast">BlastResultView</a></td>
<th class="colLast" scope="row">
<div class="block">Old classes to display Blast Results in Eclipse.rcp</div>
</th>
</tr>
<tr id="i63" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/blast/gui/BlastWizard.html" title="class in bacnet.scripts.blast.gui">BlastWizard</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i64" class="altColor">
<td class="colFirst"><a href="org/biojava3/genome/query/BlastXMLQuery.html" title="class in org.biojava3.genome.query">BlastXMLQuery</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i65" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava3.core.sequence.io">BufferedReaderBytesRead</a></td>
<th class="colLast" scope="row">
<div class="block">Need to keep track of actual bytes read and take advantage of buffered reader performance.</div>
</th>
</tr>
<tr id="i66" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava3.core.sequence.transcription">CaseInsensitiveCompound</a></td>
<th class="colLast" scope="row">
<div class="block">Attempts to wrap compounds so it is possible to view them in a case insensitive manner</div>
</th>
</tr>
<tr id="i67" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">CasePreservingProteinSequenceCreator</a></td>
<th class="colLast" scope="row">
<div class="block">A sequence creator which preserves the case of its input string in the user collection of the
 returned ProteinSequence.</div>
</th>
</tr>
<tr id="i68" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/CDSComparator.html" title="class in org.biojava3.core.sequence">CDSComparator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i69" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></td>
<th class="colLast" scope="row">
<div class="block">Represents a exon or coding sequence in a gene.</div>
</th>
</tr>
<tr id="i70" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/sequence/Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></td>
<th class="colLast" scope="row">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</th>
</tr>
<tr id="i71" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/sequence/ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a></td>
<th class="colLast" scope="row">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</th>
</tr>
<tr id="i72" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></td>
<th class="colLast" scope="row">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</th>
</tr>
<tr id="i73" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">CircularGenomeJPanel</a></td>
<th class="colLast" scope="row">
<div class="block">Circular Genome Panel created for displaying Small RNAs in CGView</div>
</th>
</tr>
<tr id="i74" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava3.core.sequence.io.util">ClasspathResource</a></td>
<th class="colLast" scope="row">
<div class="block">This object represents a classpath resource on the local system.</div>
</th>
</tr>
<tr id="i75" class="rowColor">
<td class="colFirst"><a href="bacnet/utils/CMD.html" title="class in bacnet.utils">CMD</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i76" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/sequence/Codon.html" title="class in bacnet.datamodel.sequence">Codon</a></td>
<th class="colLast" scope="row">
<div class="block">Class for managing triplet of nucleotides.<br>
 Allo<s detection of start and stop codons in nucleotide sequences</div>
</th>
</tr>
<tr id="i77" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/compound/CodonCompound.html" title="class in org.biojava3.core.sequence.compound">CodonCompound</a></td>
<th class="colLast" scope="row">
<div class="block">Define a codon</div>
</th>
</tr>
<tr id="i78" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/annotation/CodonUsage.html" title="class in bacnet.datamodel.annotation">CodonUsage</a></td>
<th class="colLast" scope="row">
<div class="block">Tools for calculating codon Usage of amino acid sequences<br>
 Codon Usage tables came from : http://www.kazusa.or.jp/codon/</div>
</th>
</tr>
<tr id="i79" class="rowColor">
<td class="colFirst"><a href="bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">CoExprNetworkView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i80" class="altColor">
<td class="colFirst"><a href="bacnet/views/CoExprNetworkView.InitNetworkThread.html" title="class in bacnet.views">CoExprNetworkView.InitNetworkThread</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i81" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/annotation/COGannotation.html" title="class in bacnet.datamodel.annotation">COGannotation</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i82" class="altColor">
<td class="colFirst"><a href="bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i83" class="rowColor">
<td class="colFirst"><a href="bacnet/table/core/ColorMapper.TypeMapper.html" title="enum in bacnet.table.core">ColorMapper.TypeMapper</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i84" class="altColor">
<td class="colFirst"><a href="bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i85" class="rowColor">
<td class="colFirst"><a href="bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">ColorMapperRCP</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i86" class="altColor">
<td class="colFirst"><a href="bacnet/table/core/ColorMapperRCP.Type.html" title="enum in bacnet.table.core">ColorMapperRCP.Type</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i87" class="rowColor">
<td class="colFirst"><a href="bacnet/table/core/ColorMapperRCPList.html" title="class in bacnet.table.core">ColorMapperRCPList</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i88" class="altColor">
<td class="colFirst"><a href="bacnet/table/gui/ColorMapperWizard.html" title="class in bacnet.table.gui">ColorMapperWizard</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i89" class="rowColor">
<td class="colFirst"><a href="bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">ColorMapperWizardPage</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i90" class="altColor">
<td class="colFirst"><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i91" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/core/vennDiagram/CombinationGenerator.html" title="class in bacnet.scripts.core.vennDiagram">CombinationGenerator</a></td>
<th class="colLast" scope="row">
<div class="block">Class for finding every combination of a set of integer</div>
</th>
</tr>
<tr id="i92" class="altColor">
<td class="colFirst"><a href="bacnet/utils/CombinationGenerator.html" title="class in bacnet.utils">CombinationGenerator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i93" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/core/Comparison.html" title="class in bacnet.scripts.core">Comparison</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i94" class="altColor">
<td class="colFirst"><a href="bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">ComparisonAtlas</a></td>
<th class="colLast" scope="row">
<div class="block">A ComparisonAtlas is related to a Comparison object and gives the list of genome elements over,
 under and not differentially expressed</div>
</th>
</tr>
<tr id="i95" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/database/ComparisonsCreation.html" title="class in bacnet.scripts.database">ComparisonsCreation</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i96" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/ComplementCompound.html" title="interface in org.biojava3.core.sequence.template">ComplementCompound</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i97" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava3.core.sequence.views">ComplementSequenceView</a>&lt;<a href="org/biojava3/core/sequence/views/ComplementSequenceView.html" title="type parameter in ComplementSequenceView">C</a> extends <a href="org/biojava3/core/sequence/template/ComplementCompound.html" title="interface in org.biojava3.core.sequence.template">ComplementCompound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">For a given sequence this class will create a view over the top of it and for every request the
 code will return the complement of the underlying base e.g. base A will become base T</div>
</th>
</tr>
<tr id="i98" class="altColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/CompositeNTermInfo.html" title="class in bacnet.genomeBrowser">CompositeNTermInfo</a></td>
<th class="colLast" scope="row">
<div class="block">NTerm composite which is displayed on the right of the GenomeViewer.</div>
</th>
</tr>
<tr id="i99" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i100" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/sequence/CompoundColorization.html" title="class in bacnet.datamodel.sequence">CompoundColorization</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i101" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/exceptions/CompoundNotFoundError.html" title="class in org.biojava3.core.exceptions">CompoundNotFoundError</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i102" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="org/biojava3/core/sequence/template/CompoundSet.html" title="type parameter in CompoundSet">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i103" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/CompoundTranslator.html" title="interface in org.biojava3.core.sequence.template">CompoundTranslator</a>&lt;<a href="org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">F</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>,&#8203;<a href="org/biojava3/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">T</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i104" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/util/ConcurrencyTools.html" title="class in org.biojava3.core.util">ConcurrencyTools</a></td>
<th class="colLast" scope="row">
<div class="block">Static utility to easily share a thread pool for concurrent/parallel/lazy execution.</div>
</th>
</tr>
<tr id="i105" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/util/CRC64Checksum.html" title="class in org.biojava3.core.util">CRC64Checksum</a></td>
<th class="colLast" scope="row">
<div class="block">Utility class that calculates a CRC64 checksum on a stream of bytes.</div>
</th>
</tr>
<tr id="i106" class="altColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">CRISPRGuideView</a></td>
<th class="colLast" scope="row">
<div class="block">CRISPRGuideView for CRISPRGuide datasets</div>
</th>
</tr>
<tr id="i107" class="rowColor">
<td class="colFirst"><a href="bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">CRISPRPredictView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i108" class="altColor">
<td class="colFirst"><a href="bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">CRISPRPredictView2</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i109" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/technology/CUFSL.html" title="class in bacnet.scripts.listeriomics.technology">CUFSL</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i110" class="altColor">
<td class="colFirst"><a href="bacnet/Database.html" title="class in bacnet">Database</a></td>
<th class="colLast" scope="row">
<div class="block">A ModelProvider specifically design for the Website Database<br>
 Contains persistent variable, used through all the website.</div>
</th>
</tr>
<tr id="i111" class="rowColor">
<td class="colFirst"><a href="bacnet/Database.InitDatabaseThread.html" title="class in bacnet">Database.InitDatabaseThread</a></td>
<th class="colLast" scope="row">
<div class="block">Class for loading databases in a thread</div>
</th>
</tr>
<tr id="i112" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/database/DatabaseCreation.html" title="class in bacnet.scripts.database">DatabaseCreation</a></td>
<th class="colLast" scope="row">Deprecated.</th>
</tr>
<tr id="i113" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/features/DatabaseReferenceInterface.html" title="interface in org.biojava3.core.sequence.features">DatabaseReferenceInterface</a></td>
<th class="colLast" scope="row">
<div class="block">If a SequenceProxyReader implements this interface then that external source has a list of cross
 reference id(s)</div>
</th>
</tr>
<tr id="i114" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/database/DataFolder.html" title="class in bacnet.scripts.database">DataFolder</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i115" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/DataSource.html" title="enum in org.biojava3.core.sequence">DataSource</a></td>
<th class="colLast" scope="row">
<div class="block">GenBank gi|gi-number|gb|accession|locus ENA Data Library gi|gi-number|emb|accession|locus DDBJ,
 DNA Database of Japan gi|gi-number|dbj|accession|locus NBRF PIR pir||entry Protein Research
 Foundation prf||name SWISS-PROT UNIPROT sp|accession|name Brookhaven Protein Data Bank (1)
 pdb|entry|chain Brookhaven Protein Data Bank (2) entry:chain|PDBID|CHAIN|SEQUENCE Patents
 pat|country|number GenInfo Backbone Id bbs|number General database identifier
 gnl|database|identifier NCBI Reference Sequence ref|accession|locus Local Sequence identifier
 lcl|identifier</div>
</th>
</tr>
<tr id="i116" class="altColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i117" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i118" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava3.core.sequence.features">DBReferenceInfo</a></td>
<th class="colLast" scope="row">
<div class="block">If you have a uniprot ID then it is possible to get a collection of other id(s) that the protein
 is known by.</div>
</th>
</tr>
<tr id="i119" class="rowColor">
<td class="colFirst"><a href="bacnet/utils/DisplayFlux.html" title="class in bacnet.utils">DisplayFlux</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i120" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/compound/DNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">DNACompoundSet</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i121" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/compound/DNACompoundSet.InitaliseOnDemand.html" title="class in org.biojava3.core.sequence.compound">DNACompoundSet.InitaliseOnDemand</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i122" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/database/DNASeqCreation.html" title="class in bacnet.scripts.database">DNASeqCreation</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i123" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></td>
<th class="colLast" scope="row">
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</th>
</tr>
<tr id="i124" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></td>
<th class="colLast" scope="row">
<div class="block">The type of DNA sequence</div>
</th>
</tr>
<tr id="i125" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/DNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">DNASequenceCreator</a></td>
<th class="colLast" scope="row">
<div class="block">A helper class that allows different ways to read a string and create a DNA sequence.</div>
</th>
</tr>
<tr id="i126" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava3.core.sequence.transcription">DNAToRNATranslator</a></td>
<th class="colLast" scope="row">
<div class="block">Performs the first stage of transcription by going from DNA to RNA.</div>
</th>
</tr>
<tr id="i127" class="rowColor">
<td class="colFirst"><a href="bacnet/raprcp/DownloadServiceHandler.html" title="class in bacnet.raprcp">DownloadServiceHandler</a></td>
<th class="colLast" scope="row">
<div class="block">Specific tools for downloading files from eclipse.rap instance NEED TO BE COMMENTED FOR
 Eclipse.RCP to work !!!</div>
</th>
</tr>
<tr id="i128" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/edits/Edit.html" title="interface in org.biojava3.core.sequence.edits">Edit</a>&lt;<a href="org/biojava3/core/sequence/edits/Edit.html" title="type parameter in Edit">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Interface for carrying out edit operations on a Sequence.</div>
</th>
</tr>
<tr id="i129" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="class in org.biojava3.core.sequence.edits">Edit.AbstractEdit</a>&lt;<a href="org/biojava3/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Abstract class which defines all edit operations as a call to discover what 5' and 3' ends of an
 editing Sequence should be joined together with a target Sequence.</div>
</th>
</tr>
<tr id="i130" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/edits/Edit.Delete.html" title="class in org.biojava3.core.sequence.edits">Edit.Delete</a>&lt;<a href="org/biojava3/core/sequence/edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implementation which allows for the deletion of bases from a Sequence</div>
</th>
</tr>
<tr id="i131" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/edits/Edit.Insert.html" title="class in org.biojava3.core.sequence.edits">Edit.Insert</a>&lt;<a href="org/biojava3/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Edit implementation which allows us to insert a base at any position in a Sequence.</div>
</th>
</tr>
<tr id="i132" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/edits/Edit.Substitute.html" title="class in org.biojava3.core.sequence.edits">Edit.Substitute</a>&lt;<a href="org/biojava3/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Allows for the substitution of bases into an existing Sequence.</div>
</th>
</tr>
<tr id="i133" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/dataset/EGDeWTdata.html" title="class in bacnet.datamodel.dataset">EGDeWTdata</a></td>
<th class="colLast" scope="row">
<div class="block">List of method for creating EGDe_Mean data: A data created from the mean of all Wild Type data
 (37C in BHI)</div>
</th>
</tr>
<tr id="i134" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/util/Equals.html" title="class in org.biojava3.core.util">Equals</a></td>
<th class="colLast" scope="row">
<div class="block">A set of helper methods which return true if the two parameters are equal to each other.</div>
</th>
</tr>
<tr id="i135" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/ExonComparator.html" title="class in org.biojava3.core.sequence">ExonComparator</a></td>
<th class="colLast" scope="row">
<div class="block">Sort Exon where it is a little confusing if exons shoud always be ordered left to right where a
 negative stranded gene should go the other direction.</div>
</th>
</tr>
<tr id="i136" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</a></td>
<th class="colLast" scope="row">
<div class="block">A gene contains a collection of Exon sequences</div>
</th>
</tr>
<tr id="i137" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">Experiment</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i138" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/core/Expression.html" title="class in bacnet.scripts.core">Expression</a></td>
<th class="colLast" scope="row">
<div class="block">A list of methods to combine transcriptomics data in ExpressionMatrix</div>
</th>
</tr>
<tr id="i139" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a></td>
<th class="colLast" scope="row">
<div class="block">Abstract Class describing data related to transcriptome <br>
 This data are saved in DataStream and might be read in streaming mode <br>
 <br>
 boolean[] read is used as a "read" indicator <br>
 As soon as a position in double[] values has been read, boolean[] read is updated <br></div>
</th>
</tr>
<tr id="i140" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a></td>
<th class="colLast" scope="row">
<div class="block">This is the principal data model used in the software for managing big table of expression</div>
</th>
</tr>
<tr id="i141" class="rowColor">
<td class="colFirst"><a href="bacnet/reader/ExpressionMatrixReader.html" title="class in bacnet.reader">ExpressionMatrixReader</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i142" class="altColor">
<td class="colFirst"><a href="bacnet/utils/ExpressionMatrixStat.html" title="class in bacnet.utils">ExpressionMatrixStat</a></td>
<th class="colLast" scope="row">
<div class="block">List of method for doing Stat on ExpressionMatrices</div>
</th>
</tr>
<tr id="i143" class="rowColor">
<td class="colFirst"><a href="bacnet/reader/FastaFileReader.html" title="class in bacnet.reader">FastaFileReader</a>&lt;<a href="bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">S</a> extends <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;<a href="bacnet/reader/FastaFileReader.html" title="type parameter in FastaFileReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i144" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/FastaGeneWriter.html" title="class in org.biojava3.core.sequence.io">FastaGeneWriter</a></td>
<th class="colLast" scope="row">
<div class="block">A Gene sequence has a Positive or Negative Strand where we want to write out to a stream the 5 to
 3 prime version.</div>
</th>
</tr>
<tr id="i145" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/template/FastaHeaderFormatInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderFormatInterface</a>&lt;<a href="org/biojava3/core/sequence/io/template/FastaHeaderFormatInterface.html" title="type parameter in FastaHeaderFormatInterface">S</a> extends <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;<a href="org/biojava3/core/sequence/io/template/FastaHeaderFormatInterface.html" title="type parameter in FastaHeaderFormatInterface">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i146" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">FastaHeaderParserInterface</a>&lt;<a href="org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="type parameter in FastaHeaderParserInterface">S</a> extends <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;<a href="org/biojava3/core/sequence/io/template/FastaHeaderParserInterface.html" title="type parameter in FastaHeaderParserInterface">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i147" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/FastaReader.html" title="class in org.biojava3.core.sequence.io">FastaReader</a>&lt;<a href="org/biojava3/core/sequence/io/FastaReader.html" title="type parameter in FastaReader">S</a> extends <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;<a href="org/biojava3/core/sequence/io/FastaReader.html" title="type parameter in FastaReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Use FastaReaderHelper as an example of how to use this class where FastaReaderHelper should be
 the primary class used to read Fasta files</div>
</th>
</tr>
<tr id="i148" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava3.core.sequence.io">FastaReaderHelper</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i149" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/FastaSequenceParser.html" title="class in org.biojava3.core.sequence.io">FastaSequenceParser</a></td>
<th class="colLast" scope="row">
<div class="block">Used to parse a stream of a fasta file to get the sequence</div>
</th>
</tr>
<tr id="i150" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/FastaWriter.html" title="class in org.biojava3.core.sequence.io">FastaWriter</a>&lt;<a href="org/biojava3/core/sequence/io/FastaWriter.html" title="type parameter in FastaWriter">S</a> extends <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;<a href="org/biojava3/core/sequence/io/FastaWriter.html" title="type parameter in FastaWriter">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">The FastaWriter writes a collection of sequences to an outputStream.</div>
</th>
</tr>
<tr id="i151" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/FastaWriterHelper.html" title="class in org.biojava3.core.sequence.io">FastaWriterHelper</a></td>
<th class="colLast" scope="row">
<div class="block">The class that should be used to write out fasta file of a sequence collection</div>
</th>
</tr>
<tr id="i152" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/core/stat/FDR.html" title="class in bacnet.scripts.core.stat">FDR</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i153" class="rowColor">
<td class="colFirst"><a href="org/biojava3/genome/parsers/gff/Feature.html" title="class in org.biojava3.genome.parsers.gff">Feature</a></td>
<th class="colLast" scope="row">
<div class="block">A Feature corresponds to a single row in a GFF file.</div>
</th>
</tr>
<tr id="i154" class="altColor">
<td class="colFirst"><a href="org/biojava3/genome/parsers/gff/FeatureHelper.html" title="class in org.biojava3.genome.parsers.gff">FeatureHelper</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i155" class="rowColor">
<td class="colFirst"><a href="org/biojava3/genome/parsers/gff/FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a></td>
<th class="colLast" scope="row">
<div class="block">A feature on a sequence (for example, an exon or a gene), defined by a location and a set of
 attributes encoded as key/value pairs.</div>
</th>
</tr>
<tr id="i156" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/features/FeatureInterface.html" title="interface in org.biojava3.core.sequence.features">FeatureInterface</a>&lt;<a href="org/biojava3/core/sequence/features/FeatureInterface.html" title="type parameter in FeatureInterface">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/features/FeatureInterface.html" title="type parameter in FeatureInterface">C</a>&gt;,&#8203;<a href="org/biojava3/core/sequence/features/FeatureInterface.html" title="type parameter in FeatureInterface">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Interface class to handle describing arbitrary features.</div>
</th>
</tr>
<tr id="i157" class="rowColor">
<td class="colFirst"><a href="org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">FeatureList</a></td>
<th class="colLast" scope="row">
<div class="block">A list of FeatureI objects implemented using a Java ArrayList; corresponds to a GFF file.</div>
</th>
</tr>
<tr id="i158" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/features/FeaturesKeyWordInterface.html" title="interface in org.biojava3.core.sequence.features">FeaturesKeyWordInterface</a></td>
<th class="colLast" scope="row">
<div class="block">Models the keywords that are annotated for a protein sequence at Uniprot.</div>
</th>
</tr>
<tr id="i159" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/technology/Febit.html" title="class in bacnet.scripts.listeriomics.technology">Febit</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i160" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/exceptions/FileAccessError.html" title="class in org.biojava3.core.exceptions">FileAccessError</a></td>
<th class="colLast" scope="row">
<div class="block">Thrown from SequenceFileProxyLoader if error opening file using RandomnAccessFile</div>
</th>
</tr>
<tr id="i161" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/FileProxyDNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyDNASequenceCreator</a></td>
<th class="colLast" scope="row">
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the
 stream the sequence and the offset index are passed to create a Protein sequence that will be
 loaded in lazily.</div>
</th>
</tr>
<tr id="i162" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/FileProxyProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">FileProxyProteinSequenceCreator</a></td>
<th class="colLast" scope="row">
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the
 stream the sequence and the offset index are passed to create a Protein sequence that will be
 loaded in lazily.</div>
</th>
</tr>
<tr id="i163" class="rowColor">
<td class="colFirst"><a href="bacnet/utils/FileUtils.html" title="class in bacnet.utils">FileUtils</a></td>
<th class="colLast" scope="row">
<div class="block">Utils for file manipulation.</div>
</th>
</tr>
<tr id="i164" class="altColor">
<td class="colFirst"><a href="bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i165" class="rowColor">
<td class="colFirst"><a href="bacnet/utils/Filter.html" title="class in bacnet.utils">Filter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i166" class="altColor">
<td class="colFirst"><a href="bacnet/table/core/Filter.TypeFilter.html" title="enum in bacnet.table.core">Filter.TypeFilter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i167" class="rowColor">
<td class="colFirst"><a href="bacnet/utils/Filter.TypeFilter.html" title="enum in bacnet.utils">Filter.TypeFilter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i168" class="altColor">
<td class="colFirst"><a href="bacnet/table/gui/FilterDialog.html" title="class in bacnet.table.gui">FilterDialog</a></td>
<th class="colLast" scope="row">
<div class="block">Dialog for managing <code>Filter</code> for <code>ExpressionMatrix</code></div>
</th>
</tr>
<tr id="i169" class="rowColor">
<td class="colFirst"><a href="bacnet/table/gui/FilterExpressionMatrixDialog.html" title="class in bacnet.table.gui">FilterExpressionMatrixDialog</a></td>
<th class="colLast" scope="row">
<div class="block">Dialog displaying the list of <code>Filters</code> currently used on the
 <code>ExpressionMatrix</code></div>
</th>
</tr>
<tr id="i170" class="altColor">
<td class="colFirst"><a href="bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i171" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/util/FlatFileCache.html" title="class in org.biojava3.core.util">FlatFileCache</a></td>
<th class="colLast" scope="row">
<div class="block">Provides a cache for storing multiple small files in memory.</div>
</th>
</tr>
<tr id="i172" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/storage/FourBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader</a>&lt;<a href="org/biojava3/core/sequence/storage/FourBitSequenceReader.html" title="type parameter in FourBitSequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Four bit encoding of the bit formats.</div>
</th>
</tr>
<tr id="i173" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">FourBitSequenceReader.FourBitArrayWorker</a>&lt;<a href="org/biojava3/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A four bit per compound implementation of the bit array worker code.</div>
</th>
</tr>
<tr id="i174" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a></td>
<th class="colLast" scope="row">
<div class="block">Indicates a way of translating a sequence.</div>
</th>
</tr>
<tr id="i175" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/FuzzyPoint.html" title="class in org.biojava3.core.sequence.location">FuzzyPoint</a></td>
<th class="colLast" scope="row">
<div class="block">Implementation for resolving fuzzy locations.</div>
</th>
</tr>
<tr id="i176" class="altColor">
<td class="colFirst"><a href="bacnet/reader/GBKReader.html" title="class in bacnet.reader">GBKReader</a></td>
<th class="colLast" scope="row">
<div class="block">GenBankl reader tools <br>
 TO BE FIXED for adding other chromosomes</div>
</th>
</tr>
<tr id="i177" class="rowColor">
<td class="colFirst"><a href="org/biojava3/genome/parsers/gff/GCStats.html" title="class in org.biojava3.genome.parsers.gff">GCStats</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i178" class="altColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/tracksGUI/GElement.html" title="class in bacnet.genomeBrowser.tracksGUI">GElement</a></td>
<th class="colLast" scope="row">
<div class="block">Here are all the methods for displaying each Genome element</div>
</th>
</tr>
<tr id="i179" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i180" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a></td>
<th class="colLast" scope="row">
<div class="block">GeneExpression array data from Listeria tiling chip ArrayExpress id : A-AFFY-140
 
 Here double[] values vector has length = 5*probes.size because we put 5 columns column 0-2 = 3
 replicatas column 3 = median expression column 4 = var expression</div>
</th>
</tr>
<tr id="i181" class="rowColor">
<td class="colFirst"><a href="org/biojava3/genome/GeneFeatureHelper.html" title="class in org.biojava3.genome">GeneFeatureHelper</a></td>
<th class="colLast" scope="row">
<div class="block">Allow to process genome elements annotation</div>
</th>
</tr>
<tr id="i182" class="altColor">
<td class="colFirst"><a href="org/biojava3/genome/parsers/gff/GeneIDGFF2Reader.html" title="class in org.biojava3.genome.parsers.gff">GeneIDGFF2Reader</a></td>
<th class="colLast" scope="row">
<div class="block">http://www.bioperl.org/wiki/GTF Read and write FeatureLists as GFF/GTF formatted files.</div>
</th>
</tr>
<tr id="i183" class="rowColor">
<td class="colFirst"><a href="org/biojava3/genome/parsers/geneid/GeneIDXMLReader.html" title="class in org.biojava3.genome.parsers.geneid">GeneIDXMLReader</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i184" class="altColor">
<td class="colFirst"><a href="org/biojava3/genome/parsers/gff/GeneMarkGTFReader.html" title="class in org.biojava3.genome.parsers.gff">GeneMarkGTFReader</a></td>
<th class="colLast" scope="row">
<div class="block">http://www.bioperl.org/wiki/GTF Read and write FeatureLists as GFF/GTF formatted files.</div>
</th>
</tr>
<tr id="i185" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/sequenceNCBI/GeneNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GeneNCBITools</a></td>
<th class="colLast" scope="row">
<div class="block">List of tools useful to extract information hidden in the NoteList of a DNASequence</div>
</th>
</tr>
<tr id="i186" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/technology/GeneralArray.html" title="class in bacnet.scripts.listeriomics.technology">GeneralArray</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i187" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderFormat</a>&lt;<a href="org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="type parameter in GenericFastaHeaderFormat">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;?&gt;,&#8203;<a href="org/biojava3/core/sequence/io/GenericFastaHeaderFormat.html" title="type parameter in GenericFastaHeaderFormat">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">We store the original header if the sequence is parsed from a fasta file and will use that exact
 sequence if we write out the sequences to a fasta file.</div>
</th>
</tr>
<tr id="i188" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="class in org.biojava3.core.sequence.io">GenericFastaHeaderParser</a>&lt;<a href="org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="type parameter in GenericFastaHeaderParser">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="type parameter in GenericFastaHeaderParser">C</a>&gt;,&#8203;<a href="org/biojava3/core/sequence/io/GenericFastaHeaderParser.html" title="type parameter in GenericFastaHeaderParser">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">The default fasta header parser where some headers are well defined based on the source database
 which allows us to set the source of the protein sequence and the identifier that can be used in
 future implementations to load features from external sources
 
 If the user has a custom header with local data then they can create their own implementation of
 a FastaHeaderParserInterface

 GenBank gi|gi-number|gb|accession|locus ENA Data Library gi|gi-number|emb|accession|locus DDBJ,
 DNA Database of Japan gi|gi-number|dbj|accession|locus NBRF PIR pir||entry Protein Research
 Foundation prf||name SWISS-PROT sp|accession|name Brookhaven Protein Data Bank (1)
 pdb|entry|chain Brookhaven Protein Data Bank (2) entry:chain|PDBID|CHAIN|SEQUENCE PDB EBI
 PDB:1ECY_A mol:protein length:142 ECOTIN Patents pat|country|number GenInfo Backbone Id
 bbs|number General database identifier gnl|database|identifier NCBI Reference Sequence
 ref|accession|locus Local Sequence identifier lcl|identifier</div>
</th>
</tr>
<tr id="i189" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i190" class="altColor">
<td class="colFirst"><a href="bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">GeneView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i191" class="rowColor">
<td class="colFirst"><a href="bacnet/sequenceTools/GeneViewHomologTools.html" title="class in bacnet.sequenceTools">GeneViewHomologTools</a></td>
<th class="colLast" scope="row">
<div class="block">All the tools to display homologs in the Gene Panel view: - It creates the homolog table - Create
 the homolog figure - Display both of them</div>
</th>
</tr>
<tr id="i192" class="altColor">
<td class="colFirst"><a href="bacnet/sequenceTools/GeneViewLocalizationTools.html" title="class in bacnet.sequenceTools">GeneViewLocalizationTools</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i193" class="rowColor">
<td class="colFirst"><a href="bacnet/sequenceTools/GeneViewProteomeTools.html" title="class in bacnet.sequenceTools">GeneViewProteomeTools</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i194" class="altColor">
<td class="colFirst"><a href="bacnet/sequenceTools/GeneViewTranscriptomeTools.html" title="class in bacnet.sequenceTools">GeneViewTranscriptomeTools</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i195" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></td>
<th class="colLast" scope="row">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="bacnet/reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</th>
</tr>
<tr id="i196" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/sequence/Genome.GetMultiFastaThread.html" title="class in bacnet.datamodel.sequence">Genome.GetMultiFastaThread</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i197" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/sequence/Genome.OpenGenomesThread.html" title="class in bacnet.datamodel.sequence">Genome.OpenGenomesThread</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i198" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/sequenceNCBI/GenomeConversion.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversion</a></td>
<th class="colLast" scope="row">
<div class="block">Static methods for converting GenomeNCBI to Genome<br>
 Genome is used for fast loading of all genomic information we gathered</div>
</th>
</tr>
<tr id="i199" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/sequenceNCBI/GenomeConversionElement.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeConversionElement</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i200" class="altColor">
<td class="colFirst"><a href="bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">GenomeElementAtlas</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i201" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></td>
<th class="colLast" scope="row">
<div class="block">Object containing an entire genome, created from NCBI RefSeq data<br>
 Data for this genome come from: <it> NCBI.fna (for the sequence) <it> NCBI.gff (for Gene, CDS,
 and ncRNA lists) If it exists some information are added: <it> Rfam.gff (other info) <it> supp
 info in a .txt file <it> supp info about RAST results in a .txt file For EGD-e only: <it>
 Srna.xml are read to add to the genome</div>
</th>
</tr>
<tr id="i202" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/blast/GenomeNCBIFolderTools.html" title="class in bacnet.scripts.blast">GenomeNCBIFolderTools</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i203" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/sequenceNCBI/GenomeNCBITools.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBITools</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i204" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/phylogeny/GenomePhylogeny.html" title="class in bacnet.scripts.phylogeny">GenomePhylogeny</a></td>
<th class="colLast" scope="row">
<div class="block">Manage phylogeny from MLST scheme : Jolley and others</div>
</th>
</tr>
<tr id="i205" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/database/GenomesCreation.html" title="class in bacnet.scripts.database">GenomesCreation</a></td>
<th class="colLast" scope="row">
<div class="block">List of tools available for genome creation</div>
</th>
</tr>
<tr id="i206" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/GenomesListeriomics.html" title="class in bacnet.scripts.listeriomics">GenomesListeriomics</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i207" class="rowColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/GenomeTranscriptomeView.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView</a></td>
<th class="colLast" scope="row">
<div class="block">Display on the genome viewer: expression of specific bioCond and the log of fold change</div>
</th>
</tr>
<tr id="i208" class="altColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/GenomeTranscriptomeView.OpenBioConditionAndGenomeElementThread.html" title="class in bacnet.genomeBrowser">GenomeTranscriptomeView.OpenBioConditionAndGenomeElementThread</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i209" class="rowColor">
<td class="colFirst"><a href="bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">GenomicsView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i210" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/core/normalization/GENormalization.html" title="class in bacnet.scripts.core.normalization">GENormalization</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i211" class="rowColor">
<td class="colFirst"><a href="org/biojava3/genome/parsers/gff/GFF3Reader.html" title="class in org.biojava3.genome.parsers.gff">GFF3Reader</a></td>
<th class="colLast" scope="row">
<div class="block">http://www.bioperl.org/wiki/GTF Read and write FeatureLists as GFF/GTF formatted files.</div>
</th>
</tr>
<tr id="i212" class="altColor">
<td class="colFirst"><a href="org/biojava3/genome/parsers/gff/GFF3Writer.html" title="class in org.biojava3.genome.parsers.gff">GFF3Writer</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i213" class="rowColor">
<td class="colFirst"><a href="bacnet/reader/GFFNCBIReader.html" title="class in bacnet.reader">GFFNCBIReader</a></td>
<th class="colLast" scope="row">
<div class="block">Read GFF files to create a <a href="bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence"><code>Genome</code></a> object</div>
</th>
</tr>
<tr id="i214" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/annotation/GlaserFCannotation.html" title="class in bacnet.datamodel.annotation">GlaserFCannotation</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i215" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/core/normalization/GRNormalization.html" title="class in bacnet.scripts.core.normalization">GRNormalization</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i216" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/util/Hashcoder.html" title="class in org.biojava3.core.util">Hashcoder</a></td>
<th class="colLast" scope="row">
<div class="block">Contains helper methods for generating a HashCode without having to resort to the commons lang
 hashcode builders.</div>
</th>
</tr>
<tr id="i217" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/exceptions/HeaderParseException.html" title="class in org.biojava3.core.exceptions">HeaderParseException</a></td>
<th class="colLast" scope="row">
<div class="block">Currently not used</div>
</th>
</tr>
<tr id="i218" class="altColor">
<td class="colFirst"><a href="bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i219" class="rowColor">
<td class="colFirst"><a href="bacnet/expressionAtlas/HeatMapMultiOmicsView.InitOmicsMatrixThread.html" title="class in bacnet.expressionAtlas">HeatMapMultiOmicsView.InitOmicsMatrixThread</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i220" class="altColor">
<td class="colFirst"><a href="bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">HeatMapProteomicsView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i221" class="rowColor">
<td class="colFirst"><a href="bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">HeatMapTranscriptomicsView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i222" class="altColor">
<td class="colFirst"><a href="bacnet/views/HelpPage.html" title="class in bacnet.views">HelpPage</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i223" class="rowColor">
<td class="colFirst"><a href="bacnet/table/gui/HideExpressionMatrixDialog.html" title="class in bacnet.table.gui">HideExpressionMatrixDialog</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i224" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/database/HomologCreation.html" title="class in bacnet.scripts.database">HomologCreation</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i225" class="rowColor">
<td class="colFirst"><a href="bacnet/utils/HTMLUtils.html" title="class in bacnet.utils">HTMLUtils</a></td>
<th class="colLast" scope="row">
<div class="block">List of methods to manage D3js visualization in BACNET</div>
</th>
</tr>
<tr id="i226" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/technology/IHM.html" title="class in bacnet.scripts.listeriomics.technology">IHM</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i227" class="rowColor">
<td class="colFirst"><a href="bacnet/utils/ImageMagick.html" title="class in bacnet.utils">ImageMagick</a></td>
<th class="colLast" scope="row">
<div class="block">Methods link to image mlanipulation with ImageMagick</div>
</th>
</tr>
<tr id="i228" class="altColor">
<td class="colFirst"><a href="bacnet/views/InitView.html" title="class in bacnet.views">InitView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i229" class="rowColor">
<td class="colFirst"><a href="bacnet/e4/rap/InitViewBacnet.html" title="class in bacnet.e4.rap">InitViewBacnet</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i230" class="altColor">
<td class="colFirst"><a href="bacnet/e4/rap/InitViewCRISPR.html" title="class in bacnet.e4.rap">InitViewCRISPR</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i231" class="rowColor">
<td class="colFirst"><a href="bacnet/e4/rap/InitViewLeishmania.html" title="class in bacnet.e4.rap">InitViewLeishmania</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i232" class="altColor">
<td class="colFirst"><a href="bacnet/e4/rap/InitViewListeria.html" title="class in bacnet.e4.rap">InitViewListeria</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i233" class="rowColor">
<td class="colFirst"><a href="bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">InitViewListeriaSample</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i234" class="altColor">
<td class="colFirst"><a href="bacnet/e4/rap/InitViewYersinia.html" title="class in bacnet.e4.rap">InitViewYersinia</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i235" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/util/InputStreamProvider.html" title="class in org.biojava3.core.util">InputStreamProvider</a></td>
<th class="colLast" scope="row">
<div class="block">A class that provides an InputStream from a File.</div>
</th>
</tr>
<tr id="i236" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/InsdcLocations.html" title="class in org.biojava3.core.sequence.location">InsdcLocations</a></td>
<th class="colLast" scope="row">
<div class="block">A collection of locations which are used whenever we work with INSDC; some of which could be
 deprecated (from INSDC's point of view) yet appear in records.</div>
</th>
</tr>
<tr id="i237" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/InsdcLocations.BondLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.BondLocation</a></td>
<th class="colLast" scope="row">
<div class="block">Used to represent bond locations equivalent to bond(7,8) or bond(7).</div>
</th>
</tr>
<tr id="i238" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.GroupLocation</a></td>
<th class="colLast" scope="row">
<div class="block">Deprecated in INSDC yet still appears; equivalent to the order() directive except no 5' to 3'
 ordering is defined.</div>
</th>
</tr>
<tr id="i239" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/InsdcLocations.OneOfLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OneOfLocation</a></td>
<th class="colLast" scope="row">
<div class="block">Deprecated in INSDC; refers to a set of locations of which one location could be valid e.g.</div>
</th>
</tr>
<tr id="i240" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava3.core.sequence.location">InsdcLocations.OrderLocation</a></td>
<th class="colLast" scope="row">
<div class="block">Used to describe a 5' to 3' ordering but no firm assurance it is correct</div>
</th>
</tr>
<tr id="i241" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/InsdcParser.html" title="class in org.biojava3.core.sequence.location">InsdcParser</a></td>
<th class="colLast" scope="row">
<div class="block">Parser for working with INSDC style locations.</div>
</th>
</tr>
<tr id="i242" class="altColor">
<td class="colFirst"><a href="bacnet/views/InternalBrowser.html" title="class in bacnet.views">InternalBrowser</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i243" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i244" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/util/IOUtils.html" title="class in org.biojava3.core.sequence.io.util">IOUtils</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i245" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/util/IOUtils.ReaderProcessor.html" title="interface in org.biojava3.core.sequence.io.util">IOUtils.ReaderProcessor</a></td>
<th class="colLast" scope="row">
<div class="block">Closure interface used when working with <code>IOUtils#processReader(String)</code>.</div>
</th>
</tr>
<tr id="i246" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/IUPACParser.html" title="class in org.biojava3.core.sequence.io">IUPACParser</a></td>
<th class="colLast" scope="row">
<div class="block">Available translations

 
 1 - UNIVERSAL
 2 - VERTEBRATE_MITOCHONDRIAL
 3 - YEAST_MITOCHONDRIAL
 4 - MOLD_MITOCHONDRIAL
 5 - INVERTEBRATE_MITOCHONDRIAL
 6 - CILIATE_NUCLEAR
 9 - ECHINODERM_MITOCHONDRIAL
 10 - EUPLOTID_NUCLEAR
 11 - BACTERIAL
 12 - ALTERNATIVE_YEAST_NUCLEAR
 13 - ASCIDIAN_MITOCHONDRIAL
 14 - FLATWORM_MITOCHONDRIAL
 15 - BLEPHARISMA_MACRONUCLEAR
 16 - 2CHLOROPHYCEAN_MITOCHONDRIAL
 21 - TREMATODE_MITOCHONDRIAL
 23 - SCENEDESMUS_MITOCHONDRIAL
 

 Taken from <a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c">NCBI</a>
 with slight modification and put into the classpath resource.</div>
</th>
</tr>
<tr id="i247" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/IUPACParser.IOD.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IOD</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i248" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io">IUPACParser.IUPACTable</a></td>
<th class="colLast" scope="row">
<div class="block">Holds the concept of a codon table from the IUPAC format</div>
</th>
</tr>
<tr id="i249" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/technology/JCVI.html" title="class in bacnet.scripts.listeriomics.technology">JCVI</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i250" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava3.core.sequence.storage">JoiningSequenceReader</a>&lt;<a href="org/biojava3/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">This reader actually proxies onto multiple types of sequence in order to allow a number of
 sequence objects to act as if they are one sequence.</div>
</th>
</tr>
<tr id="i251" class="rowColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">LegendDialog</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i252" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/LightweightProfile.html" title="interface in org.biojava3.core.sequence.template">LightweightProfile</a>&lt;<a href="org/biojava3/core/sequence/template/LightweightProfile.html" title="type parameter in LightweightProfile">S</a> extends <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="org/biojava3/core/sequence/template/LightweightProfile.html" title="type parameter in LightweightProfile">C</a>&gt;,&#8203;<a href="org/biojava3/core/sequence/template/LightweightProfile.html" title="type parameter in LightweightProfile">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines a minimal data structure for reading and writing a sequence alignment.</div>
</th>
</tr>
<tr id="i253" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/LightweightProfile.StringFormat.html" title="enum in org.biojava3.core.sequence.template">LightweightProfile.StringFormat</a></td>
<th class="colLast" scope="row">
<div class="block">List of output formats.</div>
</th>
</tr>
<tr id="i254" class="altColor">
<td class="colFirst"><a href="bacnet/utils/ListUtils.html" title="class in bacnet.utils">ListUtils</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i255" class="rowColor">
<td class="colFirst"><a href="bacnet/handler/LoadExpressionMatrixHandler.html" title="class in bacnet.handler">LoadExpressionMatrixHandler</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i256" class="altColor">
<td class="colFirst"><a href="bacnet/sequenceTools/LocalizationDialog.html" title="class in bacnet.sequenceTools">LocalizationDialog</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i257" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/template/Location.html" title="interface in org.biojava3.core.sequence.location.template">Location</a></td>
<th class="colLast" scope="row">
<div class="block">Sets of integers used to represent the location of features on sequence.</div>
</th>
</tr>
<tr id="i258" class="altColor">
<td class="colFirst"><a href="org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a></td>
<th class="colLast" scope="row">
<div class="block">A location on a sequence.</div>
</th>
</tr>
<tr id="i259" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/template/Location.Tools.html" title="class in org.biojava3.core.sequence.location.template">Location.Tools</a></td>
<th class="colLast" scope="row">
<div class="block">Helper methods for use with the Location classes.</div>
</th>
</tr>
<tr id="i260" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/template/Location.Tools.LocationPredicate.html" title="interface in org.biojava3.core.sequence.location.template">Location.Tools.LocationPredicate</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i261" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/LocationHelper.html" title="class in org.biojava3.core.sequence.location">LocationHelper</a></td>
<th class="colLast" scope="row">
<div class="block">Helper methods for use with the Location classes.</div>
</th>
</tr>
<tr id="i262" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/LocationHelper.LocationPredicate.html" title="interface in org.biojava3.core.sequence.location">LocationHelper.LocationPredicate</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i263" class="rowColor">
<td class="colFirst"><a href="org/biojava3/genome/parsers/gff/LocIterator.html" title="class in org.biojava3.genome.parsers.gff">LocIterator</a></td>
<th class="colLast" scope="row">
<div class="block">Move a sliding window over a Location.</div>
</th>
</tr>
<tr id="i264" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/annotation/LocusTag.html" title="class in bacnet.datamodel.annotation">LocusTag</a></td>
<th class="colLast" scope="row">
<div class="block">Since the update of RefSeq database all the locustag have been changed <br>
 Here is some methods to deal with this change</div>
</th>
</tr>
<tr id="i265" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/core/stat/LPE.html" title="class in bacnet.scripts.core.stat">LPE</a></td>
<th class="colLast" scope="row">
<div class="block">Class for calculating LPE statistics <br>
 Inspired from R library LPE<br></div>
</th>
</tr>
<tr id="i266" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/technology/LT.html" title="class in bacnet.scripts.listeriomics.technology">LT</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i267" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/technology/Macroarray.html" title="class in bacnet.scripts.listeriomics.technology">Macroarray</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i268" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/MainListeriomics.html" title="class in bacnet.scripts.listeriomics">MainListeriomics</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i269" class="rowColor">
<td class="colFirst"><a href="bacnet/utils/MathUtils.html" title="class in bacnet.utils">MathUtils</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i270" class="altColor">
<td class="colFirst"><a href="bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i271" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/core/normalization/MedianNormalization.html" title="class in bacnet.scripts.core.normalization">MedianNormalization</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i272" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava3.core.sequence">MultipleSequenceAlignment</a>&lt;<a href="org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">S</a> extends <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">C</a>&gt;,&#8203;<a href="org/biojava3/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements a minimal data structure for reading and writing a sequence alignment.</div>
</th>
</tr>
<tr id="i273" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/blast/MultiSequenceBlastProtein.html" title="class in bacnet.scripts.blast">MultiSequenceBlastProtein</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i274" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/blast/MultiSequenceTBlastNProtein.html" title="class in bacnet.scripts.blast">MultiSequenceTBlastNProtein</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i275" class="rowColor">
<td class="colFirst"><a href="bacnet/raprcp/NavigationManagement.html" title="class in bacnet.raprcp">NavigationManagement</a></td>
<th class="colLast" scope="row">
<div class="block">Every lmethods for managing navigation state in eclipse.rap application.</div>
</th>
</tr>
<tr id="i276" class="altColor">
<td class="colFirst"><a href="bacnet/reader/NCBIFastaHeaderParser.html" title="class in bacnet.reader">NCBIFastaHeaderParser</a>&lt;<a href="bacnet/reader/NCBIFastaHeaderParser.html" title="type parameter in NCBIFastaHeaderParser">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="bacnet/reader/NCBIFastaHeaderParser.html" title="type parameter in NCBIFastaHeaderParser">C</a>&gt;,&#8203;<a href="bacnet/reader/NCBIFastaHeaderParser.html" title="type parameter in NCBIFastaHeaderParser">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">My own implementation of a FastaHeaderParserInterface to parse Fasta files from NCBI ftp

 NCBI fna (genome and plasmids) gi|gi-number|ref|accession|name gi|16802048|ref|NC_003210.1|
 Listeria monocytogenes EGD-e, complete genome NCBI ffn (coding) ref|refGenome
 gi-number|:start-end name ref|NC_003210.1|:318-1673 chromosomal replication initiation protein
 [Listeria monocytogenes EGD-e] NCBI frn (non-coding) ref|refGenome gi-number|:start-end| name|
 [locus-tag] ref|NC_003210|:82705-82777|Lys tRNA| [locus_tag=lmot01] NCBI faa (proteins)
 gi|gi-number|ref|accession|name gi|16802049|ref|NP_463534.1| chromosomal replication initiation
 protein [Listeria monocytogenes EGD-e]</div>
</th>
</tr>
<tr id="i277" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i278" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/sequence/NcRNA.TypeNcRNA.html" title="enum in bacnet.datamodel.sequence">NcRNA.TypeNcRNA</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i279" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a></td>
<th class="colLast" scope="row">
<div class="block">Network object for co-expression network analysis and visualisation</div>
</th>
</tr>
<tr id="i280" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/database/NetworkCreation.html" title="class in bacnet.scripts.database">NetworkCreation</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i281" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a></td>
<th class="colLast" scope="row">
<div class="block">Class for manipulating various RNASeq data, herited from ExpressionData <br>
 double[] value is of genome size, when ones want to see RNAseq expression at a certain bp
 position, just extract values[bpposition]<br>
 Need to be loaded first, and unlog has RNASeq will be logged for display when convert in the
 platform</div>
</th>
</tr>
<tr id="i282" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/database/NGSCreation.html" title="class in bacnet.scripts.database">NGSCreation</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i283" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/technology/NimbleGen.html" title="class in bacnet.scripts.listeriomics.technology">NimbleGen</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i284" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></td>
<th class="colLast" scope="row">
<div class="block">Object for managing every mapped sequences</div>
</th>
</tr>
<tr id="i285" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/proteomics/NTerm.ExperimentName.html" title="enum in bacnet.datamodel.proteomics">NTerm.ExperimentName</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i286" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i287" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a></td>
<th class="colLast" scope="row">
<div class="block">Type of modification at the start of a peptide</div>
</th>
</tr>
<tr id="i288" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/nterm/NTermConservationWithRiboProfiling.html" title="class in bacnet.scripts.listeriomics.nterm">NTermConservationWithRiboProfiling</a></td>
<th class="colLast" scope="row">
<div class="block">Run different method to compare iTIS position (internal translation) to data of Ribosome
 profiling in Bacillus and Ecoli:<br>
 compare the gene by BlastN
 find internal translation in the ribosome profiling
 find internal SD sequence</div>
</th>
</tr>
<tr id="i289" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/proteomics/NTermCreateData.html" title="class in bacnet.datamodel.proteomics">NTermCreateData</a></td>
<th class="colLast" scope="row">
<div class="block">Transform excel table describing all N-Terminal peptides found in Listeria<br>
 Find the position of each peptide in Listeria genome
 create a NGS data from it</div>
</th>
</tr>
<tr id="i290" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/nterm/NTermCreateFinalData.html" title="class in bacnet.scripts.listeriomics.nterm">NTermCreateFinalData</a></td>
<th class="colLast" scope="row">
<div class="block">Genertal class for creation of NTerminomics tables</div>
</th>
</tr>
<tr id="i291" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i292" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/nterm/NTermDatabase.html" title="class in bacnet.scripts.listeriomics.nterm">NTermDatabase</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i293" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/proteomics/NTermFilter.html" title="class in bacnet.datamodel.proteomics">NTermFilter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i294" class="altColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">NTerminomicsView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i295" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/nterm/NTermMain.html" title="class in bacnet.scripts.listeriomics.nterm">NTermMain</a></td>
<th class="colLast" scope="row">
<div class="block">General method for running all methods related to NTerm project</div>
</th>
</tr>
<tr id="i296" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/nterm/NTermStat.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStat</a></td>
<th class="colLast" scope="row">
<div class="block">List of method to do statistics about common features in N-Terminus of genes: - start codon - SD
 binding - AU proportion - Folding energy</div>
</th>
</tr>
<tr id="i297" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/nterm/NTermStatProteolyse.html" title="class in bacnet.scripts.listeriomics.nterm">NTermStatProteolyse</a></td>
<th class="colLast" scope="row">
<div class="block">A list of methods to assess the quantity of proteolysis on the peptides</div>
</th>
</tr>
<tr id="i298" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/proteomics/NTermSummaryTable.html" title="class in bacnet.datamodel.proteomics">NTermSummaryTable</a></td>
<th class="colLast" scope="row">
<div class="block">List of methods for creating Summary Table for TIS <br>
 Using NTermFilter we have regroup TIS in different overlap groups, which we use now to create the
 Summary Tables.</div>
</th>
</tr>
<tr id="i299" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/nterm/NTermTranscriptome.html" title="class in bacnet.scripts.listeriomics.nterm">NTermTranscriptome</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i300" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/proteomics/NTermUtils.html" title="class in bacnet.datamodel.proteomics">NTermUtils</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i301" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i302" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a></td>
<th class="colLast" scope="row">
<div class="block">Abstract Class describing data related to omics data <br></div>
</th>
</tr>
<tr id="i303" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/dataset/OmicsData.ColNames.html" title="enum in bacnet.datamodel.dataset">OmicsData.ColNames</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i304" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a></td>
<th class="colLast" scope="row">
<div class="block">Enum referencing the different type of OmicsData possible:
 GeneExpr
 Tiling
 RNASeq
 TSS (Transcription Initiation Site)
 NTerm (Translation Initiation Site)
 Proteome (Matrix of protein found)
 MassSpec (List of peptides found by mass spectrometry)
 ExpressionMatrix (General matrix format)
 unknown</div>
</th>
</tr>
<tr id="i305" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i306" class="altColor">
<td class="colFirst"><a href="org/biojava3/genome/query/OutputHitsGFF.html" title="class in org.biojava3.genome.query">OutputHitsGFF</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i307" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/exceptions/ParserException.html" title="class in org.biojava3.core.exceptions">ParserException</a></td>
<th class="colLast" scope="row">
<div class="block">General abstraction of different parsing errors</div>
</th>
</tr>
<tr id="i308" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/Peptidomics.html" title="class in bacnet.scripts.listeriomics">Peptidomics</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i309" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/phylogeny/Phylogenomic.html" title="class in bacnet.datamodel.phylogeny">Phylogenomic</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i310" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/database/PhylogenomicsCreation.html" title="class in bacnet.scripts.database">PhylogenomicsCreation</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i311" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/PhylogenyListeriomics.html" title="class in bacnet.scripts.listeriomics">PhylogenyListeriomics</a></td>
<th class="colLast" scope="row">
<div class="block">Different methods to create phylogeny tree, and tables for each genome elements</div>
</th>
</tr>
<tr id="i312" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/blast/PhylogenySmallORFs.html" title="class in bacnet.scripts.blast">PhylogenySmallORFs</a></td>
<th class="colLast" scope="row">
<div class="block">Tools to blast a small ORF on other bacterial genomes, and display results on a phylogeny tree</div>
</th>
</tr>
<tr id="i313" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/phylogeny/PhylogenySmallORFs.html" title="class in bacnet.scripts.phylogeny">PhylogenySmallORFs</a></td>
<th class="colLast" scope="row">
<div class="block">Tools to blast a small ORF on other bacterial genomes, and display results on a phylogeny tree</div>
</th>
</tr>
<tr id="i314" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/phylogeny/PhylogenyTools.html" title="class in bacnet.scripts.phylogeny">PhylogenyTools</a></td>
<th class="colLast" scope="row">
<div class="block">Different methods to manipulate and display Newick and other phylogeny files</div>
</th>
</tr>
<tr id="i315" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/phylogeny/PhylogenyToolsJolley.html" title="class in bacnet.scripts.phylogeny">PhylogenyToolsJolley</a></td>
<th class="colLast" scope="row">
<div class="block">This list of function is useful to manipulate Newick files given by Jolley in May 2012, from its
 publication Jolley et al.</div>
</th>
</tr>
<tr id="i316" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a></td>
<th class="colLast" scope="row">
<div class="block">Holds a single point part of a location</div>
</th>
</tr>
<tr id="i317" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/template/Point.Resolver.html" title="interface in org.biojava3.core.sequence.location.template">Point.Resolver</a>&lt;<a href="org/biojava3/core/sequence/location/template/Point.Resolver.html" title="type parameter in Point.Resolver">T</a> extends <a href="org/biojava3/core/sequence/location/template/Point.html" title="interface in org.biojava3.core.sequence.location.template">Point</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Used to resolve a position about a point</div>
</th>
</tr>
<tr id="i318" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/core/PredaTerm.html" title="class in bacnet.scripts.core">PredaTerm</a></td>
<th class="colLast" scope="row">
<div class="block">PREDicting poly-A region and TERMinal splice acceptor for Leishmania Please read manuscript for
 details : Smith, M., Blanchette, M., Papadopoulou, B.</div>
</th>
</tr>
<tr id="i319" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/util/PrettyXMLWriter.html" title="class in org.biojava3.core.util">PrettyXMLWriter</a></td>
<th class="colLast" scope="row">
<div class="block">Implementation of XMLWriter which emits nicely formatted documents to a PrintWriter.</div>
</th>
</tr>
<tr id="i320" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></td>
<th class="colLast" scope="row">
<div class="block">The representation of a ProteinSequence</div>
</th>
</tr>
<tr id="i321" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/ProteinSequenceCreator.html" title="class in org.biojava3.core.sequence.io">ProteinSequenceCreator</a></td>
<th class="colLast" scope="row">
<div class="block">Used to create a ProteinSequence from a String to allow for details about the location of the
 sequence etc.</div>
</th>
</tr>
<tr id="i322" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/blast/ProteinTools.html" title="class in bacnet.scripts.blast">ProteinTools</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i323" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/database/ProteomicsCreation.html" title="class in bacnet.scripts.database">ProteomicsCreation</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i324" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a></td>
<th class="colLast" scope="row">
<div class="block">OmicsData class inherited everything from ExpressionMatrix, the only difference is the TypeDate
 value, it a Proteome data.<br>
 It is not very useful for the moment but will be in the futur</div>
</th>
</tr>
<tr id="i325" class="rowColor">
<td class="colFirst"><a href="bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i326" class="altColor">
<td class="colFirst"><a href="bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i327" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="org/biojava3/core/sequence/template/ProxySequenceReader.html" title="type parameter in ProxySequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i328" class="altColor">
<td class="colFirst"><a href="bacnet/reader/PTTReader.html" title="class in bacnet.reader">PTTReader</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i329" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/features/QualityFeature.html" title="class in org.biojava3.core.sequence.features">QualityFeature</a>&lt;<a href="org/biojava3/core/sequence/features/QualityFeature.html" title="type parameter in QualityFeature">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/features/QualityFeature.html" title="type parameter in QualityFeature">C</a>&gt;,&#8203;<a href="org/biojava3/core/sequence/features/QualityFeature.html" title="type parameter in QualityFeature">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">DNA Sequences produced by modern sequencers usually have quality informaion attached to them.</div>
</th>
</tr>
<tr id="i330" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/features/QuantityFeature.html" title="class in org.biojava3.core.sequence.features">QuantityFeature</a>&lt;<a href="org/biojava3/core/sequence/features/QuantityFeature.html" title="type parameter in QuantityFeature">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/features/QuantityFeature.html" title="type parameter in QuantityFeature">C</a>&gt;,&#8203;<a href="org/biojava3/core/sequence/features/QuantityFeature.html" title="type parameter in QuantityFeature">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">It is common to have a numerical value or values associated with a feature.</div>
</th>
</tr>
<tr id="i331" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/genome/RASTEGDeGenome.html" title="class in bacnet.scripts.genome">RASTEGDeGenome</a></td>
<th class="colLast" scope="row">
<div class="block">List of methods for adding RAST genes to EGD-e annotation</div>
</th>
</tr>
<tr id="i332" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/nterm/RASToverlap.html" title="class in bacnet.scripts.listeriomics.nterm">RASToverlap</a></td>
<th class="colLast" scope="row">
<div class="block">Methods to look at overlapping elements between NCBI and RAST genomes</div>
</th>
</tr>
<tr id="i333" class="rowColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i334" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/RegulatorRegion.html" title="class in bacnet.scripts.listeriomics">RegulatorRegion</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i335" class="rowColor">
<td class="colFirst"><a href="bacnet/swt/ResourceManager.html" title="class in bacnet.swt">ResourceManager</a></td>
<th class="colLast" scope="row">
<div class="block">Utility class for managing OS resources associated with SWT/JFace controls such as colors, fonts,
 images, etc.
 
 !!!</div>
</th>
</tr>
<tr id="i336" class="altColor">
<td class="colFirst"><a href="bacnet/swt/ResourceManager.PluginResourceProvider.html" title="interface in bacnet.swt">ResourceManager.PluginResourceProvider</a></td>
<th class="colLast" scope="row">
<div class="block">Provider for plugin resources, used by WindowBuilder at design time.</div>
</th>
</tr>
<tr id="i337" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava3.core.sequence.views">ReversedSequenceView</a>&lt;<a href="org/biojava3/core/sequence/views/ReversedSequenceView.html" title="type parameter in ReversedSequenceView">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">For a given sequence this class will return the base at the reversed position i.e. in a sequence
 of size 10, if you request base 2 you will get back the base at position 9.</div>
</th>
</tr>
<tr id="i338" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/annotation/RfamElement.html" title="class in bacnet.datamodel.annotation">RfamElement</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i339" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/compound/RNACompoundSet.html" title="class in org.biojava3.core.sequence.compound">RNACompoundSet</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i340" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/compound/RNACompoundSet.InitaliseOnDemand.html" title="class in org.biojava3.core.sequence.compound">RNACompoundSet.InitaliseOnDemand</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i341" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></td>
<th class="colLast" scope="row">
<div class="block">RNASequence where RNACompoundSet are the allowed values</div>
</th>
</tr>
<tr id="i342" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/RNASequenceCreator.html" title="class in org.biojava3.core.sequence.io">RNASequenceCreator</a></td>
<th class="colLast" scope="row">
<div class="block">Used to create a RNA sequence</div>
</th>
</tr>
<tr id="i343" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/views/RnaSequenceView.html" title="class in org.biojava3.core.sequence.views">RnaSequenceView</a></td>
<th class="colLast" scope="row">
<div class="block">Attempts to do on the fly translation of RNA by not requesting the compounds until asked.</div>
</th>
</tr>
<tr id="i344" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/transcription/RNAToAminoAcidTranslator.html" title="class in org.biojava3.core.sequence.transcription">RNAToAminoAcidTranslator</a></td>
<th class="colLast" scope="row">
<div class="block">Takes a <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a> of <a href="org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound"><code>NucleotideCompound</code></a> which should represent an RNA sequence
 (<a href="org/biojava3/core/sequence/RNASequence.html" title="class in org.biojava3.core.sequence"><code>RNASequence</code></a> is good for this) and returns a list of <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a> which hold
 <a href="org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound"><code>AminoAcidCompound</code></a>.</div>
</th>
</tr>
<tr id="i345" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/core/Rscript.html" title="class in bacnet.scripts.core">Rscript</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i346" class="altColor">
<td class="colFirst"><a href="bacnet/utils/RWTUtils.html" title="class in bacnet.utils">RWTUtils</a></td>
<th class="colLast" scope="row">
<div class="block">Tools for saving files<br>
 Everything in bacnet.raprcp package is specific to eclipse.rap or eclipse.rcp.</div>
</th>
</tr>
<tr id="i347" class="rowColor">
<td class="colFirst"><a href="bacnet/raprcp/SaveFileUtils.html" title="class in bacnet.raprcp">SaveFileUtils</a></td>
<th class="colLast" scope="row">
<div class="block">Utils methods for managinf files.</div>
</th>
</tr>
<tr id="i348" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/genome/SDProfile.html" title="class in bacnet.scripts.genome">SDProfile</a></td>
<th class="colLast" scope="row">
<div class="block">Calculate Shine-Dalgarno profile for bacterial genome.</div>
</th>
</tr>
<tr id="i349" class="rowColor">
<td class="colFirst"><a href="bacnet/sequenceTools/SecondStructureRNADialog.html" title="class in bacnet.sequenceTools">SecondStructureRNADialog</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i350" class="altColor">
<td class="colFirst"><a href="bacnet/expressionAtlas/core/SelectGenomeElementDialog.html" title="class in bacnet.expressionAtlas.core">SelectGenomeElementDialog</a></td>
<th class="colLast" scope="row">
<div class="block">Dialog which allow specific selection of genes, smallRNAs and cisRegs</div>
</th>
</tr>
<tr id="i351" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></td>
<th class="colLast" scope="row">
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</th>
</tr>
<tr id="i352" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="org/biojava3/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Main interface for defining a collection of Compounds and accessing them using biological indexes</div>
</th>
</tr>
<tr id="i353" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/sequence/Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i354" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="class in org.biojava3.core.sequence.storage">SequenceAsStringHelper</a>&lt;<a href="org/biojava3/core/sequence/storage/SequenceAsStringHelper.html" title="type parameter in SequenceAsStringHelper">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">This is a common method that can be used across multiple storage/proxy implementations to handle
 Negative strand and other interesting elements of sequence data.</div>
</th>
</tr>
<tr id="i355" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/SequenceComparator.html" title="class in org.biojava3.core.sequence">SequenceComparator</a></td>
<th class="colLast" scope="row">
<div class="block">Used to sort sequences</div>
</th>
</tr>
<tr id="i356" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="org/biojava3/core/sequence/io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i357" class="rowColor">
<td class="colFirst"><a href="bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">SequenceDisplayDialog</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i358" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava3.core.sequence.loader">SequenceFileProxyLoader</a>&lt;<a href="org/biojava3/core/sequence/loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">This class represents the storage container of a sequence stored in a fasta file where the
 initial parsing of the file we store the offset and length of the sequence.</div>
</th>
</tr>
<tr id="i359" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/exceptions/SequenceLengthError.html" title="class in org.biojava3.core.exceptions">SequenceLengthError</a></td>
<th class="colLast" scope="row">
<div class="block">Currently not used</div>
</th>
</tr>
<tr id="i360" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/SequenceLocation.html" title="class in org.biojava3.core.sequence.location">SequenceLocation</a>&lt;<a href="org/biojava3/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">C</a>&gt;,&#8203;<a href="org/biojava3/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A location in a sequence that keeps a reference to its parent sequence</div>
</th>
</tr>
<tr id="i361" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/SequenceMixin.html" title="class in org.biojava3.core.sequence.template">SequenceMixin</a></td>
<th class="colLast" scope="row">
<div class="block">Provides a set of static methods to be used as static imports when needed across multiple
 Sequence implementations but inheritance gets in the way.</div>
</th>
</tr>
<tr id="i362" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="class in org.biojava3.core.sequence.template">SequenceMixin.SequenceIterator</a>&lt;<a href="org/biojava3/core/sequence/template/SequenceMixin.SequenceIterator.html" title="type parameter in SequenceMixin.SequenceIterator">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A basic sequence iterator which iterates over the given Sequence by biological index.</div>
</th>
</tr>
<tr id="i363" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava3.core.sequence">SequenceOptimizationHints</a></td>
<th class="colLast" scope="row">
<div class="block">A static class that provides optimization hints for memory or performance handling of sequence
 data.</div>
</th>
</tr>
<tr id="i364" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/SequenceOptimizationHints.SequenceCollection.html" title="enum in org.biojava3.core.sequence">SequenceOptimizationHints.SequenceCollection</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i365" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/SequenceOptimizationHints.SequenceUsage.html" title="enum in org.biojava3.core.sequence">SequenceOptimizationHints.SequenceUsage</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i366" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/io/template/SequenceParserInterface.html" title="interface in org.biojava3.core.sequence.io.template">SequenceParserInterface</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i367" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/SequenceProxyView.html" title="class in org.biojava3.core.sequence.template">SequenceProxyView</a>&lt;<a href="org/biojava3/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i368" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="org/biojava3/core/sequence/template/SequenceReader.html" title="type parameter in SequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i369" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/util/SequenceTools.html" title="class in org.biojava3.core.util">SequenceTools</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i370" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="org/biojava3/core/sequence/template/SequenceView.html" title="type parameter in SequenceView">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i371" class="rowColor">
<td class="colFirst"><a href="bacnet/e4/rap/SessionControl.html" title="class in bacnet.e4.rap">SessionControl</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i372" class="altColor">
<td class="colFirst"><a href="bacnet/e4/rap/setup/SessionControl.html" title="class in bacnet.e4.rap.setup">SessionControl</a></td>
<th class="colLast" scope="row">
<div class="block">General methods for session control of an eclipse.rap application</div>
</th>
</tr>
<tr id="i373" class="rowColor">
<td class="colFirst"><a href="bacnet/e4/rap/setup/SetupPart.html" title="class in bacnet.e4.rap.setup">SetupPart</a></td>
<th class="colLast" scope="row">
<div class="block">SetupPart for initialisation and creation of the Database</div>
</th>
</tr>
<tr id="i374" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i375" class="rowColor">
<td class="colFirst"><a href="bacnet/expressionAtlas/core/SignatureSelectionDialog.html" title="class in bacnet.expressionAtlas.core">SignatureSelectionDialog</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i376" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/SimpleLocation.html" title="class in org.biojava3.core.sequence.location">SimpleLocation</a></td>
<th class="colLast" scope="row">
<div class="block">Very basic implementation of the Location interface which defines a series of simple
 constructors.</div>
</th>
</tr>
<tr id="i377" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/location/SimplePoint.html" title="class in org.biojava3.core.sequence.location">SimplePoint</a></td>
<th class="colLast" scope="row">
<div class="block">Basic implementation of the Point interface.</div>
</th>
</tr>
<tr id="i378" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava3.core.sequence.storage">SingleCompoundSequenceReader</a>&lt;<a href="org/biojava3/core/sequence/storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">An implementation of the SequenceReader interface which for every call will return only 1
 compound (given to it during construction; a String is also valid but will require a
 CompoundSet).</div>
</th>
</tr>
<tr id="i379" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/srna/SmallPeptidesSearch.html" title="class in bacnet.scripts.listeriomics.srna">SmallPeptidesSearch</a></td>
<th class="colLast" scope="row">Deprecated.</th>
</tr>
<tr id="i380" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/util/SoftHashMap.html" title="class in org.biojava3.core.util">SoftHashMap</a>&lt;<a href="org/biojava3/core/util/SoftHashMap.html" title="type parameter in SoftHashMap">K</a>,&#8203;<a href="org/biojava3/core/util/SoftHashMap.html" title="type parameter in SoftHashMap">V</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A in memory cache using soft references.</div>
</th>
</tr>
<tr id="i381" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/util/SoftHashMap.SoftValue.html" title="class in org.biojava3.core.util">SoftHashMap.SoftValue</a>&lt;<a href="org/biojava3/core/util/SoftHashMap.SoftValue.html" title="type parameter in SoftHashMap.SoftValue">K</a>,&#8203;<a href="org/biojava3/core/util/SoftHashMap.SoftValue.html" title="type parameter in SoftHashMap.SoftValue">V</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">We define our own subclass of SoftReference which contains not only the
 
 value but also the key to make it easier to find the entry in the HashMap
 
 after it's been garbage collected.</div>
</th>
</tr>
<tr id="i382" class="altColor">
<td class="colFirst"><a href="org/biojava3/genome/util/SplitFasta.html" title="class in org.biojava3.genome.util">SplitFasta</a></td>
<th class="colLast" scope="row">
<div class="block">Utility to write each Fasta entry to a unique file</div>
</th>
</tr>
<tr id="i383" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i384" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/sequence/Srna.TypeSrna.html" title="enum in bacnet.datamodel.sequence">Srna.TypeSrna</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i385" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/srna/Srna10403SOliver.html" title="class in bacnet.scripts.listeriomics.srna">Srna10403SOliver</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i386" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/srna/SrnaAlignmentBlastN.html" title="class in bacnet.scripts.listeriomics.srna">SrnaAlignmentBlastN</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i387" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/srna/SrnaFinalList.html" title="class in bacnet.scripts.listeriomics.srna">SrnaFinalList</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i388" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/SrnaListeriomics.html" title="class in bacnet.scripts.listeriomics">SrnaListeriomics</a></td>
<th class="colLast" scope="row">
<div class="block">List of methods to manage Srna data for Listeriomics</div>
</th>
</tr>
<tr id="i389" class="rowColor">
<td class="colFirst"><a href="bacnet/sequenceTools/SrnaSummaryView.html" title="class in bacnet.sequenceTools">SrnaSummaryView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i390" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/srna/SrnaTables.html" title="class in bacnet.scripts.listeriomics.srna">SrnaTables</a></td>
<th class="colLast" scope="row">
<div class="block">Include a list of Methods to read every Table containing Small RNAs in Listeria<br>
 For each it reads the <code>Table</code> and create a <code>List</code> of <code>Srna</code></div>
</th>
</tr>
<tr id="i391" class="rowColor">
<td class="colFirst"><a href="bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">SrnaView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i392" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/genome/StartCodonProfile.html" title="class in bacnet.scripts.genome">StartCodonProfile</a></td>
<th class="colLast" scope="row">
<div class="block">Calculate presence of start codon on firrst chromosome of bacterial genomes</div>
</th>
</tr>
<tr id="i393" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/StartCodonSequence.html" title="class in org.biojava3.core.sequence">StartCodonSequence</a></td>
<th class="colLast" scope="row">
<div class="block">Used to map the start codon feature on a gene</div>
</th>
</tr>
<tr id="i394" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/core/stat/StatTest.html" title="class in bacnet.scripts.core.stat">StatTest</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i395" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/core/stat/StatTest.TypeStat.html" title="enum in bacnet.scripts.core.stat">StatTest.TypeStat</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i396" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/core/stat/StatUtils.html" title="class in bacnet.scripts.core.stat">StatUtils</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i397" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/StopCodonSequence.html" title="class in org.biojava3.core.sequence">StopCodonSequence</a></td>
<th class="colLast" scope="row">
<div class="block">Used to map the stop codon sequence on a gene</div>
</th>
</tr>
<tr id="i398" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a></td>
<th class="colLast" scope="row">
<div class="block">Provides a way of representing the strand of a sequence, location hit or feature.</div>
</th>
</tr>
<tr id="i399" class="rowColor">
<td class="colFirst"><a href="bacnet/utils/StringColor.html" title="class in bacnet.utils">StringColor</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i400" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/util/StringManipulationHelper.html" title="class in org.biojava3.core.util">StringManipulationHelper</a></td>
<th class="colLast" scope="row">
<div class="block">A utility class for common <code>String</code> manipulation tasks.</div>
</th>
</tr>
<tr id="i401" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">StringProxySequenceReader</a>&lt;<a href="org/biojava3/core/sequence/loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">An example of a ProxySequenceReader that is created from a String.</div>
</th>
</tr>
<tr id="i402" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/annotation/SubCellCompartment.html" title="class in bacnet.datamodel.annotation">SubCellCompartment</a></td>
<th class="colLast" scope="row">
<div class="block">All data and methods for Subcellular compartment information are done here</div>
</th>
</tr>
<tr id="i403" class="rowColor">
<td class="colFirst"><a href="bacnet/datamodel/annotation/SubCellCompartment.TypeCompartment.html" title="enum in bacnet.datamodel.annotation">SubCellCompartment.TypeCompartment</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i404" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/SummaryListeriomics.html" title="class in bacnet.scripts.listeriomics">SummaryListeriomics</a></td>
<th class="colLast" scope="row">
<div class="block">Methods for creating the different figures and Tables of the paper</div>
</th>
</tr>
<tr id="i405" class="rowColor">
<td class="colFirst"><a href="bacnet/swt/SWTResourceManager.html" title="class in bacnet.swt">SWTResourceManager</a></td>
<th class="colLast" scope="row">
<div class="block">Utility class for managing OS resources associated with SWT controls such as colors, fonts,
 images, etc.</div>
</th>
</tr>
<tr id="i406" class="altColor">
<td class="colFirst"><a href="bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">SynTView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i407" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/SystemsBiologyListeriomics.html" title="class in bacnet.scripts.listeriomics">SystemsBiologyListeriomics</a></td>
<th class="colLast" scope="row">
<div class="block">List of method to create Co-expression network and analyse it</div>
</th>
</tr>
<tr id="i408" class="altColor">
<td class="colFirst"><a href="bacnet/reader/TabDelimitedTableReader.html" title="class in bacnet.reader">TabDelimitedTableReader</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i409" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/transcription/Table.html" title="interface in org.biojava3.core.sequence.transcription">Table</a></td>
<th class="colLast" scope="row">
<div class="block">Provides a way of separating us from the specific <a href="org/biojava3/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava3.core.sequence.io"><code>IUPACParser.IUPACTable</code></a> even though this is the only
 implementing class for the interface.</div>
</th>
</tr>
<tr id="i410" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava3.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></td>
<th class="colLast" scope="row">
<div class="block">Class used to hold three nucleotides together and allow for equality to be assessed in a case
 insensitive manner.</div>
</th>
</tr>
<tr id="i411" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/transcription/Table.Codon.html" title="class in org.biojava3.core.sequence.transcription">Table.Codon</a></td>
<th class="colLast" scope="row">
<div class="block">Instance of a Codon which is 3 <a href="org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound"><code>NucleotideCompound</code></a>s, its corresponding
 <a href="org/biojava3/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava3.core.sequence.compound"><code>AminoAcidCompound</code></a> and if it is a start or stop codon.</div>
</th>
</tr>
<tr id="i412" class="altColor">
<td class="colFirst"><a href="bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a></td>
<th class="colLast" scope="row">
<div class="block">The Composite in which is displayed the heatmap</div>
</th>
</tr>
<tr id="i413" class="rowColor">
<td class="colFirst"><a href="bacnet/table/gui/TableFilter.html" title="class in bacnet.table.gui">TableFilter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i414" class="altColor">
<td class="colFirst"><a href="bacnet/utils/TableHTMLExport.html" title="class in bacnet.utils">TableHTMLExport</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i415" class="rowColor">
<td class="colFirst"><a href="bacnet/table/TableSWTComposite.html" title="class in bacnet.table">TableSWTComposite</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i416" class="altColor">
<td class="colFirst"><a href="bacnet/table/TableSWTView.html" title="class in bacnet.table">TableSWTView</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i417" class="rowColor">
<td class="colFirst"><a href="bacnet/table/core/TableViewerComparator.html" title="class in bacnet.table.core">TableViewerComparator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i418" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/technology/Tasmania.html" title="class in bacnet.scripts.listeriomics.technology">Tasmania</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i419" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/TaxonomyID.html" title="class in org.biojava3.core.sequence">TaxonomyID</a></td>
<th class="colLast" scope="row">
<div class="block">A sequence can be associated with a species or Taxonomy ID</div>
</th>
</tr>
<tr id="i420" class="altColor">
<td class="colFirst"><a href="bacnet/e4/rap/setup/Test.html" title="class in bacnet.e4.rap.setup">Test</a></td>
<th class="colLast" scope="row">
<div class="block">All methods which will be executed before or after interface run</div>
</th>
</tr>
<tr id="i421" class="rowColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/TestGenomeBrowser.html" title="class in bacnet.genomeBrowser">TestGenomeBrowser</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i422" class="altColor">
<td class="colFirst"><a href="bacnet/e4/rap/TestPart.html" title="class in bacnet.e4.rap">TestPart</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i423" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/features/TextFeature.html" title="class in org.biojava3.core.sequence.features">TextFeature</a>&lt;<a href="org/biojava3/core/sequence/features/TextFeature.html" title="type parameter in TextFeature">S</a> extends <a href="org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava3/core/sequence/features/TextFeature.html" title="type parameter in TextFeature">C</a>&gt;,&#8203;<a href="org/biojava3/core/sequence/features/TextFeature.html" title="type parameter in TextFeature">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A implmentation of AbstractFeature</div>
</th>
</tr>
<tr id="i424" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a></td>
<th class="colLast" scope="row">
<div class="block">Tiling array data from Listeria tiling chip ArrayExpress id : A-MEXP-1572 *</div>
</th>
</tr>
<tr id="i425" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/TilingGeneExprDataBase.html" title="class in bacnet.scripts.listeriomics">TilingGeneExprDataBase</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i426" class="altColor">
<td class="colFirst"><a href="bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></td>
<th class="colLast" scope="row">
<div class="block">This class is used to regroup different Nterm.</div>
</th>
</tr>
<tr id="i427" class="rowColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i428" class="altColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/core/Track.DisplayType.html" title="enum in bacnet.genomeBrowser.core">Track.DisplayType</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i429" class="rowColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a></td>
<th class="colLast" scope="row">
<div class="block">A TrackCanvasData is the Canvas in which will be displayed datasets</div>
</th>
</tr>
<tr id="i430" class="altColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a></td>
<th class="colLast" scope="row">
<div class="block">A TrackCanvasGenome is the Canvas in which will be displayed genome annotation</div>
</th>
</tr>
<tr id="i431" class="rowColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">TracksComposite</a></td>
<th class="colLast" scope="row">
<div class="block">In TracksComposite both <code>TrackCanvasData</code> and <code>TrackCanvasGenome</code> will be
 displayed.</div>
</th>
</tr>
<tr id="i432" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a></td>
<th class="colLast" scope="row">
<div class="block">Used as a way of encapsulating the data structures required to parse DNA to a Protein sequence.</div>
</th>
</tr>
<tr id="i433" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.Builder</a></td>
<th class="colLast" scope="row">
<div class="block">This class is the way to create a <code>TranslationEngine</code>.</div>
</th>
</tr>
<tr id="i434" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/transcription/TranscriptionEngine.IOD.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine.IOD</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i435" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/database/TranscriptomesCreation.html" title="class in bacnet.scripts.database">TranscriptomesCreation</a></td>
<th class="colLast" scope="row">
<div class="block">List of methods to add the different transcriptomic datasetsto your multi-omics website</div>
</th>
</tr>
<tr id="i436" class="altColor">
<td class="colFirst"><a href="bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i437" class="rowColor">
<td class="colFirst"><a href="bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a></td>
<th class="colLast" scope="row">
<div class="block">View in which is displayed the table with all transcriptomics data available</div>
</th>
</tr>
<tr id="i438" class="altColor">
<td class="colFirst"><a href="bacnet/expressionAtlas/TranscriptomicsView.OpenExpressionMatrixAndComparisons.html" title="class in bacnet.expressionAtlas">TranscriptomicsView.OpenExpressionMatrixAndComparisons</a></td>
<th class="colLast" scope="row">
<div class="block">Run thread for opening BioCondition, Comparisons, detect Genome and open corresponding LogFC
 ExpressionMatrix</div>
</th>
</tr>
<tr id="i439" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></td>
<th class="colLast" scope="row">
<div class="block">This is the sequence if you want to go from a gene sequence to a protein sequence.</div>
</th>
</tr>
<tr id="i440" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/exceptions/TranslationException.html" title="class in org.biojava3.core.exceptions">TranslationException</a></td>
<th class="colLast" scope="row">
<div class="block">Thrown from AbstractCompundTranslator</div>
</th>
</tr>
<tr id="i441" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/TSSNTermRiboSeqListeriomics.html" title="class in bacnet.scripts.listeriomics">TSSNTermRiboSeqListeriomics</a></td>
<th class="colLast" scope="row">
<div class="block">List of methods to process TSS, TermSeq, and RiboSeq of Listeriomics Some TSS and NTerm lists
 were provided in tab-delimited format.</div>
</th>
</tr>
<tr id="i442" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader</a>&lt;<a href="org/biojava3/core/sequence/storage/TwoBitSequenceReader.html" title="type parameter in TwoBitSequenceReader">C</a> extends <a href="org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implementation of the 2bit encoding.</div>
</th>
</tr>
<tr id="i443" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="class in org.biojava3.core.sequence.storage">TwoBitSequenceReader.TwoBitArrayWorker</a>&lt;<a href="org/biojava3/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</a> extends <a href="org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Extension of the BitArrayWorker which provides the 2bit implementation code.</div>
</th>
</tr>
<tr id="i444" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/technology/UGiessen.html" title="class in bacnet.scripts.listeriomics.technology">UGiessen</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i445" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/core/UNAfold.html" title="class in bacnet.scripts.core">UNAfold</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i446" class="altColor">
<td class="colFirst"><a href="bacnet/utils/UNAfold.html" title="class in bacnet.utils">UNAfold</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i447" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/util/UncompressInputStream.html" title="class in org.biojava3.core.util">UncompressInputStream</a></td>
<th class="colLast" scope="row">
<div class="block">This class decompresses an input stream containing data compressed with the unix "compress"
 utility (LZC, a LZW variant).</div>
</th>
</tr>
<tr id="i448" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava3.core.sequence.loader">UniprotProxySequenceReader</a>&lt;<a href="org/biojava3/core/sequence/loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Pass in a Uniprot ID and this ProxySequenceReader when passed to a ProteinSequence will get the
 sequence data and other data elements associated with the ProteinSequence by Uniprot.</div>
</th>
</tr>
<tr id="i449" class="rowColor">
<td class="colFirst"><a href="org/biojava3/genome/uniprot/UniprotToFasta.html" title="class in org.biojava3.genome.uniprot">UniprotToFasta</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i450" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/core/normalization/VarianceNormalization.html" title="class in bacnet.scripts.core.normalization">VarianceNormalization</a></td>
<th class="colLast" scope="row">
<div class="block">Variance normalization methods which will reduce the total variance of each data to 1<br>
 For each column of the ExpressionMatrix, multiply each value by: Math.sqr(1/var(x))<br>
 So final variance of each column is equal to zero</div>
</th>
</tr>
<tr id="i451" class="rowColor">
<td class="colFirst"><a href="bacnet/utils/VectorUtils.html" title="class in bacnet.utils">VectorUtils</a></td>
<th class="colLast" scope="row">
<div class="block">Vector are array with one dimension like : <code>String[]</code> or <code>double[]</code></div>
</th>
</tr>
<tr id="i452" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/core/vennDiagram/VennDiagram.html" title="class in bacnet.scripts.core.vennDiagram">VennDiagram</a></td>
<th class="colLast" scope="row">
<div class="block">Class for creating sets necessary in Venn diagrams<br>
 Providing a list of sets (through <code>ArrayList<String></code>) it will find intersections,
 unions and differences between all sets.</div>
</th>
</tr>
<tr id="i453" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/views/WindowedSequence.html" title="class in org.biojava3.core.sequence.views">WindowedSequence</a>&lt;<a href="org/biojava3/core/sequence/views/WindowedSequence.html" title="type parameter in WindowedSequence">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A sliding window view of a sequence which does not implement any interfaces like <a href="org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template"><code>Sequence</code></a>
 because they do not fit how this works.</div>
</th>
</tr>
<tr id="i454" class="altColor">
<td class="colFirst"><a href="org/biojava3/core/sequence/views/WindowedSequence.WindowedSequenceIterator.html" title="class in org.biojava3.core.sequence.views">WindowedSequence.WindowedSequenceIterator</a>&lt;<a href="org/biojava3/core/sequence/views/WindowedSequence.WindowedSequenceIterator.html" title="type parameter in WindowedSequence.WindowedSequenceIterator">C</a> extends <a href="org/biojava3/core/sequence/template/Compound.html" title="interface in org.biojava3.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Iterator of all List of compounds available in a windowed sequence.</div>
</th>
</tr>
<tr id="i455" class="rowColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/technology/Wuerzburg.html" title="class in bacnet.scripts.listeriomics.technology">Wuerzburg</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i456" class="altColor">
<td class="colFirst"><a href="bacnet/scripts/listeriomics/technology/WUSTL.html" title="class in bacnet.scripts.listeriomics.technology">WUSTL</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i457" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/util/XMLHelper.html" title="class in org.biojava3.core.util">XMLHelper</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i458" class="altColor">
<td class="colFirst"><a href="bacnet/utils/XMLUtils.html" title="class in bacnet.utils">XMLUtils</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i459" class="rowColor">
<td class="colFirst"><a href="org/biojava3/core/util/XMLWriter.html" title="interface in org.biojava3.core.util">XMLWriter</a></td>
<th class="colLast" scope="row">
<div class="block">Simple interface for building XML documents.</div>
</th>
</tr>
<tr id="i460" class="altColor">
<td class="colFirst"><a href="bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
</table>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="overview-tree.html">Tree</a></li>
<li><a href="deprecated-list.html">Deprecated</a></li>
<li><a href="index-files/index-1.html">Index</a></li>
<li><a href="help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
