<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 14:20:51 CEST 2019 -->
<title>Serialized Form</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="script.js"></script>
<script type="text/javascript" src="jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Serialized Form";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "./";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="overview-tree.html">Tree</a></li>
<li><a href="deprecated-list.html">Deprecated</a></li>
<li><a href="index-files/index-1.html">Index</a></li>
<li><a href="help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h1 title="Serialized Form" class="title">Serialized Form</h1>
</div>
<div class="serializedFormContainer">
<ul class="blockList">
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.datamodel.annotation</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.datamodel.annotation.Annotation">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">bacnet.datamodel.annotation.Annotation</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>6308650364233188961L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>annotation</h4>
<pre>java.lang.String[][] annotation</pre>
<div class="block">Array containing all elements ordered by their begin position</div>
</li>
<li class="blockList">
<h4>chromosomeID</h4>
<pre>java.lang.String chromosomeID</pre>
<div class="block">Corresponding chromosome</div>
</li>
<li class="blockList">
<h4>genome</h4>
<pre>java.lang.String genome</pre>
<div class="block">Corresponding Genome</div>
</li>
<li class="blockListLast">
<h4>nbElementType</h4>
<pre>int nbElementType</pre>
<div class="block">Number of element type</div>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.datamodel.annotation.Signature">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">bacnet.datamodel.annotation.Signature</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-5588137900648373339L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>description</h4>
<pre>java.lang.String description</pre>
</li>
<li class="blockList">
<h4>elements</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; elements</pre>
</li>
<li class="blockList">
<h4>genome</h4>
<pre>java.lang.String genome</pre>
</li>
<li class="blockList">
<h4>ID</h4>
<pre>java.lang.String ID</pre>
</li>
<li class="blockList">
<h4>name</h4>
<pre>java.lang.String name</pre>
</li>
<li class="blockList">
<h4>reference</h4>
<pre>java.lang.String reference</pre>
</li>
<li class="blockList">
<h4>size</h4>
<pre>int size</pre>
</li>
<li class="blockListLast">
<h4>type</h4>
<pre>java.lang.String type</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.datamodel.dataset</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.datamodel.dataset.ExpressionData">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">bacnet.datamodel.dataset.ExpressionData</a> extends <a href="bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a> implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-7625087278597156645L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>alreadyRead</h4>
<pre>boolean alreadyRead</pre>
</li>
<li class="blockList">
<h4>chromosomeID</h4>
<pre>java.lang.String chromosomeID</pre>
</li>
<li class="blockList">
<h4>genomeName</h4>
<pre>java.lang.String genomeName</pre>
</li>
<li class="blockList">
<h4>length</h4>
<pre>int length</pre>
</li>
<li class="blockList">
<h4>max</h4>
<pre>double max</pre>
</li>
<li class="blockList">
<h4>mean</h4>
<pre>double mean</pre>
</li>
<li class="blockList">
<h4>median</h4>
<pre>double median</pre>
</li>
<li class="blockList">
<h4>min</h4>
<pre>double min</pre>
</li>
<li class="blockListLast">
<h4>variance</h4>
<pre>double variance</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.datamodel.dataset.ExpressionMatrix">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">bacnet.datamodel.dataset.ExpressionMatrix</a> extends <a href="bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a> implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-6000672805382130897L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>annotations</h4>
<pre>java.lang.String[][] annotations</pre>
<div class="block">String[][] for all annotation information</div>
</li>
<li class="blockList">
<h4>firstRowName</h4>
<pre>java.lang.String firstRowName</pre>
<div class="block">The name of first row used when converted to an array</div>
</li>
<li class="blockList">
<h4>headerAnnotation</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; headerAnnotation</pre>
<div class="block">we put all important informations in a table (which should be of the size of
 "values")</div>
</li>
<li class="blockList">
<h4>headers</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; headers</pre>
<div class="block">Column names are in an ArrayList<String> so we can navigate easily, and have
 access to the index of the column just by giving an header</div>
</li>
<li class="blockList">
<h4>ordered</h4>
<pre>boolean ordered</pre>
<div class="block">Indicates wether or not rows need to be ordered thanks to ArrayList<String>
 orderedRowNames</div>
</li>
<li class="blockList">
<h4>orderedRowNames</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; orderedRowNames</pre>
<div class="block">In some cases (clustering for example) we need the row to be specifically
 organized We use a List to force the organization of the row A boolean
 indicate if this matrix as to be used.</div>
</li>
<li class="blockList">
<h4>rowNames</h4>
<pre>java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.Integer&gt; rowNames</pre>
<div class="block">the rowNames are in a TreeMap to have a matching between names and index in
 values array, we navigate easily in the TreeMap which is sorted (using a
 modifyed Comparator)</div>
</li>
<li class="blockList">
<h4>secondRowName</h4>
<pre>java.lang.String secondRowName</pre>
<div class="block">The name of second row used when converted to an array</div>
</li>
<li class="blockListLast">
<h4>values</h4>
<pre>double[][] values</pre>
<div class="block">I use an array of double to gain memory and increase access time</div>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.datamodel.dataset.GeneExpression">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">bacnet.datamodel.dataset.GeneExpression</a> extends <a href="bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a> implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-947282303133711974L</dd>
</dl>
</li>
<li class="blockList"><a id="bacnet.datamodel.dataset.Network">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">bacnet.datamodel.dataset.Network</a> extends <a href="bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a> implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>991824259100059499L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>edges</h4>
<pre>java.util.HashMap&lt;java.lang.String,&#8203;java.util.HashMap&lt;java.lang.String,&#8203;java.lang.Double&gt;&gt; edges</pre>
<div class="block">An hashMap which map each vertex (=genome element) to the correlated
 vertex.<br>
 Vertex -> HashMap<Vetrex,PearsonCorrelation></div>
</li>
<li class="blockListLast">
<h4>vertices</h4>
<pre>java.util.HashMap&lt;java.lang.String,&#8203;java.lang.String&gt; vertices</pre>
<div class="block">List of vertices = genome elements + their attributes = genome position,
 type, annotation</div>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.datamodel.dataset.NGS">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">bacnet.datamodel.dataset.NGS</a> extends <a href="bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a> implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-3283911438157029331L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>datasets</h4>
<pre>java.util.HashMap&lt;java.lang.String,&#8203;<a href="bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt; datasets</pre>
</li>
<li class="blockList">
<h4>genomeName</h4>
<pre>java.lang.String genomeName</pre>
</li>
<li class="blockListLast">
<h4>logTransformed</h4>
<pre>boolean logTransformed</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.datamodel.dataset.NTermData">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">bacnet.datamodel.dataset.NTermData</a> extends <a href="bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a> implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>497243595006646884L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>annotation</h4>
<pre><a href="bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a> annotation</pre>
<div class="block">An <code>Annotation</code> object to order NTerm by position in the Genome</div>
</li>
<li class="blockList">
<h4>aTISMap</h4>
<pre>java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt; aTISMap</pre>
<div class="block">Link gene to they corresponding aTIS if it exists</div>
</li>
<li class="blockList">
<h4>elements</h4>
<pre>java.util.HashMap&lt;java.lang.String,&#8203;<a href="bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt; elements</pre>
<div class="block">Link NTerm name to <code>NTerm</code><br>
 This correspond to the list of <code>NTerm</code></div>
</li>
<li class="blockList">
<h4>geneModification</h4>
<pre>java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="bacnet/datamodel/proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a>&gt;&gt; geneModification</pre>
<div class="block">Link gene to a list of all <code>TypeModif</code> encountered for aTIS</div>
</li>
<li class="blockList">
<h4>genes</h4>
<pre>java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt; genes</pre>
<div class="block">Link lmo genes to NTerm overlapping it<br>
 NTerm with no association to an lmo are associated to "unknown" key</div>
</li>
<li class="blockList">
<h4>modifications</h4>
<pre>java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt; modifications</pre>
<div class="block">Link NTerm position (coded as a <code>String</code> to
 <code>ArrayList(NTerm)</code><br>
 This correspond to the list of mapped sequence which are equivalent to other
 NTerm but with different modification type</div>
</li>
<li class="blockList">
<h4>peptides</h4>
<pre>java.util.HashMap&lt;java.lang.String,&#8203;java.util.ArrayList&lt;<a href="bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt;&gt; peptides</pre>
<div class="block">Link peptideSequence to <code>ArrayList(NTerm)</code><br>
 This correspond to the list of <code>NTerm</code> having same peptideSequence</div>
</li>
<li class="blockList">
<h4>tisList</h4>
<pre>java.util.ArrayList&lt;<a href="bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt; tisList</pre>
</li>
<li class="blockList">
<h4>tisMap</h4>
<pre>java.util.HashMap&lt;<a href="bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>,&#8203;<a href="bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a>&gt; tisMap</pre>
<div class="block">Link <code>Nterm</code> to a list of all <code>Nterm</code> which might be
 related to it (e.g.: almost same sequence but proteolyse cut it)</div>
</li>
<li class="blockListLast">
<h4>typeOverlaps</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; typeOverlaps</pre>
<div class="block">List of Overlap Type used for TIS classification = aTIS, internal TIS, etc...</div>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.datamodel.dataset.OmicsData">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">bacnet.datamodel.dataset.OmicsData</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-3093089920089151952L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>bioCondName</h4>
<pre>java.lang.String bioCondName</pre>
<div class="block">The BioCondition to which its related</div>
</li>
<li class="blockList">
<h4>date</h4>
<pre>java.lang.String date</pre>
</li>
<li class="blockList">
<h4>name</h4>
<pre>java.lang.String name</pre>
</li>
<li class="blockList">
<h4>note</h4>
<pre>java.lang.String note</pre>
</li>
<li class="blockList">
<h4>rawDatas</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; rawDatas</pre>
<div class="block">List of rawData from which it has been created</div>
</li>
<li class="blockListLast">
<h4>type</h4>
<pre><a href="bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a> type</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.datamodel.dataset.ProteomicsData">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">bacnet.datamodel.dataset.ProteomicsData</a> extends <a href="bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a> implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-7625087278597156645L</dd>
</dl>
</li>
<li class="blockList"><a id="bacnet.datamodel.dataset.Tiling">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">bacnet.datamodel.dataset.Tiling</a> extends <a href="bacnet/datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a> implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-7904822758820760928L</dd>
</dl>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.datamodel.expdesign</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.datamodel.expdesign.BioCondition">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">bacnet.datamodel.expdesign.BioCondition</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-8545052328831545661L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>antiComparisons</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; antiComparisons</pre>
<div class="block">List of Data for which the BioCondition is the reference bioCodntion for the
 comparison: BioCondComparison vs This Biocondition</div>
</li>
<li class="blockList">
<h4>arrayExpressId</h4>
<pre>java.lang.String arrayExpressId</pre>
<div class="block">ArrayExpressId if this data is public</div>
</li>
<li class="blockList">
<h4>arrayExpressTechnoId</h4>
<pre>java.lang.String arrayExpressTechnoId</pre>
<div class="block">ArrayExpressId of the technology used</div>
</li>
<li class="blockList">
<h4>comment</h4>
<pre>java.lang.String comment</pre>
<div class="block">Different comment</div>
</li>
<li class="blockList">
<h4>comparisons</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; comparisons</pre>
<div class="block">List of Data for which we have comparison data: This Biocondition vs
 BioCondComparison</div>
</li>
<li class="blockList">
<h4>date</h4>
<pre>java.lang.String date</pre>
<div class="block">Date of creation or data of publication</div>
</li>
<li class="blockList">
<h4>experienceNb</h4>
<pre>int experienceNb</pre>
<div class="block">For experiment from Cossart lab, ID of the experiment</div>
</li>
<li class="blockList">
<h4>geneExprs</h4>
<pre>java.util.ArrayList&lt;<a href="bacnet/datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt; geneExprs</pre>
<div class="block">All Listeria GeneExpression arrays associated to this BioCondition</div>
</li>
<li class="blockList">
<h4>genomeName</h4>
<pre>java.lang.String genomeName</pre>
<div class="block">Reference genome ID to used for this BioCondition</div>
</li>
<li class="blockList">
<h4>genomeUsed</h4>
<pre>java.lang.String genomeUsed</pre>
<div class="block">Actual strain used to create this BioCondition, might be different from
 genomeName</div>
</li>
<li class="blockList">
<h4>growth</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; growth</pre>
<div class="block">Growth conditions are specified here</div>
</li>
<li class="blockList">
<h4>linkedBioCondition</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; linkedBioCondition</pre>
<div class="block">Data associated to this BioCondition to use in the case of timepoint
 experiment, or same mutant in different condition</div>
</li>
<li class="blockList">
<h4>localization</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; localization</pre>
<div class="block">If it is a proteome, this parameters tells you where in the bacteria cell the
 proteins have been extracted</div>
</li>
<li class="blockList">
<h4>matrices</h4>
<pre>java.util.ArrayList&lt;<a href="bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&gt; matrices</pre>
<div class="block">All data in <code>ExpressioMatrix</code> associated to this BioCondition</div>
</li>
<li class="blockList">
<h4>media</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; media</pre>
<div class="block">Growth media is specified here</div>
</li>
<li class="blockList">
<h4>mediaGrowthProperties</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; mediaGrowthProperties</pre>
<div class="block">Quantity/properties of drugs added to the media if specified: quantity of
 glucose, type of Broth</div>
</li>
<li class="blockList">
<h4>mutant</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; mutant</pre>
<div class="block">If it is a mutant, the locus is indicated (example: prfA, lmo0200 or rliA),
 leave empty otherwise</div>
</li>
<li class="blockList">
<h4>name</h4>
<pre>java.lang.String name</pre>
<div class="block">Name of the BioCondition</div>
</li>
<li class="blockList">
<h4>ngss</h4>
<pre>java.util.ArrayList&lt;<a href="bacnet/datamodel/dataset/NGS.html" title="class in bacnet.datamodel.dataset">NGS</a>&gt; ngss</pre>
<div class="block">All RNASeq data associated to this BioCondition</div>
</li>
<li class="blockList">
<h4>noData</h4>
<pre>boolean noData</pre>
<div class="block">True if no data are associated to this BioCondition<br>
 It is used principaly to indicate that the BioCondition is a reference
 condition for another BioCondition, and thus no data are available for it, as
 the data are in the other BioCondition</div>
</li>
<li class="blockList">
<h4>nTerms</h4>
<pre>java.util.ArrayList&lt;<a href="bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&gt; nTerms</pre>
<div class="block">All NTermData data associated to this BioCondition</div>
</li>
<li class="blockList">
<h4>proteomes</h4>
<pre>java.util.ArrayList&lt;<a href="bacnet/datamodel/dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a>&gt; proteomes</pre>
<div class="block">All ProteomicsData data associated to this BioCondition</div>
</li>
<li class="blockList">
<h4>reference</h4>
<pre>java.lang.String reference</pre>
<div class="block">Article in which data is published</div>
</li>
<li class="blockList">
<h4>studyName</h4>
<pre>java.lang.String studyName</pre>
<div class="block">Name of the study from which the data is extracted</div>
</li>
<li class="blockList">
<h4>temperature</h4>
<pre>java.lang.String temperature</pre>
<div class="block">Temperature is specified here</div>
</li>
<li class="blockList">
<h4>tilings</h4>
<pre>java.util.ArrayList&lt;<a href="bacnet/datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a>&gt; tilings</pre>
<div class="block">All Listeria Tiling arrays associated to this BioCondition</div>
</li>
<li class="blockList">
<h4>time</h4>
<pre>java.lang.String time</pre>
<div class="block">time point if necessary</div>
</li>
<li class="blockList">
<h4>typeDataContained</h4>
<pre>java.util.TreeSet&lt;<a href="bacnet/datamodel/dataset/OmicsData.TypeData.html" title="enum in bacnet.datamodel.dataset">OmicsData.TypeData</a>&gt; typeDataContained</pre>
<div class="block">Type of bioCondition : TSS only, TIS only, RNASeq + Tiling + GeneExpr, RNASeq
 only ...</div>
</li>
<li class="blockListLast">
<h4>wildType</h4>
<pre>boolean wildType</pre>
<div class="block">True if this BioCondition is a wild-type strain, to be used as a reference
 condition</div>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.datamodel.expdesign.Experiment">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/expdesign/Experiment.html" title="class in bacnet.datamodel.expdesign">bacnet.datamodel.expdesign.Experiment</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>993350014891502366L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>bioConds</h4>
<pre>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="bacnet/datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&gt; bioConds</pre>
</li>
<li class="blockListLast">
<h4>name</h4>
<pre>java.lang.String name</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.datamodel.proteomics</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.datamodel.proteomics.NTerm">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">bacnet.datamodel.proteomics.NTerm</a> extends <a href="bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-7191352336233310587L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>antiSDBindingFreeEnergy</h4>
<pre>double antiSDBindingFreeEnergy</pre>
<div class="block">Value of the binding between anti-SD sequence and the region -20bp upstream
 to the peptide</div>
</li>
<li class="blockList">
<h4>codonUsage</h4>
<pre>double codonUsage</pre>
<div class="block">CodonUsage of the peptideSequence</div>
</li>
<li class="blockList">
<h4>duplicates</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; duplicates</pre>
<div class="block">Number of duplicates having same sequencePeptide (e.g. peptide mapped on
 different regions)</div>
</li>
<li class="blockList">
<h4>insoluble</h4>
<pre>boolean insoluble</pre>
<div class="block">Indicate if it was found in soluble part</div>
</li>
<li class="blockList">
<h4>mappingFrame</h4>
<pre><a href="bacnet/datamodel/proteomics/NTerm.MappingFrame.html" title="enum in bacnet.datamodel.proteomics">NTerm.MappingFrame</a> mappingFrame</pre>
<div class="block">Frame where the peptide has been mapped</div>
</li>
<li class="blockList">
<h4>maxQuant</h4>
<pre>java.lang.String maxQuant</pre>
<div class="block">If the peptide has also been found with MaxQuant analysis software, we write
 something in this attribute, otherwise its blank</div>
</li>
<li class="blockList">
<h4>maxQuantum</h4>
<pre>java.lang.String[] maxQuantum</pre>
<div class="block">If the peptide has also been found with MaxQuant analysis software, we write
 something in this attribute, otherwise its blank, each column is for one type
 of experiments.</div>
</li>
<li class="blockList">
<h4>modifSequence</h4>
<pre>java.lang.String modifSequence</pre>
<div class="block">Modify sequence provided by Francis</div>
</li>
<li class="blockList">
<h4>nextCodon</h4>
<pre>java.lang.String nextCodon</pre>
<div class="block">Codon downstream to first codon of mapped sequence</div>
</li>
<li class="blockList">
<h4>overlap</h4>
<pre>java.lang.String overlap</pre>
<div class="block">Name of the genome element which <code>NTerm</code> overlaps</div>
</li>
<li class="blockList">
<h4>overlapInFrame</h4>
<pre>boolean overlapInFrame</pre>
<div class="block">Check if the peptide is in frame with genome element which it overlaps</div>
</li>
<li class="blockList">
<h4>previousCodon</h4>
<pre>java.lang.String previousCodon</pre>
<div class="block">Codon upstream to mapped sequence</div>
</li>
<li class="blockList">
<h4>score</h4>
<pre>int score</pre>
</li>
<li class="blockList">
<h4>sequenceMap</h4>
<pre>java.lang.String sequenceMap</pre>
<div class="block">Translated sequence of the mapped peptide</div>
</li>
<li class="blockList">
<h4>sequencePeptide</h4>
<pre>java.lang.String sequencePeptide</pre>
<div class="block">Peptide sequence extracted from modified Sequence</div>
</li>
<li class="blockList">
<h4>soluble</h4>
<pre>boolean soluble</pre>
<div class="block">Indicate if it was found in soluble part</div>
</li>
<li class="blockList">
<h4>spectra</h4>
<pre>int spectra</pre>
<div class="block">Total number of spectra found</div>
</li>
<li class="blockList">
<h4>spectrum</h4>
<pre>int[] spectrum</pre>
<div class="block">Each column of this vector give the number of spectra for each condition</div>
</li>
<li class="blockList">
<h4>startCode</h4>
<pre>java.lang.String startCode</pre>
<div class="block">Indicates if the first codon or previousCodon is a Start codon</div>
</li>
<li class="blockList">
<h4>startCodon</h4>
<pre>java.lang.String startCodon</pre>
<div class="block">First codon of the mapped sequence</div>
</li>
<li class="blockList">
<h4>threshold</h4>
<pre>int threshold</pre>
</li>
<li class="blockList">
<h4>TISName</h4>
<pre>java.lang.String TISName</pre>
<div class="block">TIS Name : A TIS is a group of NTerm have same position but a slightly
 different position</div>
</li>
<li class="blockList">
<h4>tssDownCoverage</h4>
<pre>int tssDownCoverage</pre>
<div class="block">Coverage of the closest TSS downstream to this NTerm</div>
</li>
<li class="blockList">
<h4>tssDownDistance</h4>
<pre>int tssDownDistance</pre>
<div class="block">Distance where can be found a TSS downstream to this NTerm</div>
</li>
<li class="blockList">
<h4>tssDowntype</h4>
<pre>java.lang.String tssDowntype</pre>
<div class="block">Type of TSS downstream to this NTerm : i.e. internal, intergenic, ...</div>
</li>
<li class="blockList">
<h4>tssUpCoverage</h4>
<pre>int tssUpCoverage</pre>
<div class="block">Coverage of the closest TSS upstream to this NTerm</div>
</li>
<li class="blockList">
<h4>tssUpDistance</h4>
<pre>int tssUpDistance</pre>
<div class="block">Distance where can be found a TSS upstream to this NTerm</div>
</li>
<li class="blockList">
<h4>tssUptype</h4>
<pre>java.lang.String tssUptype</pre>
<div class="block">Type of TSS upstream to this NTerm : i.e. internal, intergenic, ...</div>
</li>
<li class="blockList">
<h4>typeModif</h4>
<pre><a href="bacnet/datamodel/proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a> typeModif</pre>
<div class="block">Type of modification</div>
</li>
<li class="blockListLast">
<h4>typeOverlap</h4>
<pre>java.lang.String typeOverlap</pre>
<div class="block">Potential type of TIS of this NTerm</div>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.datamodel.proteomics.TIS">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">bacnet.datamodel.proteomics.TIS</a> extends <a href="bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>3452878896031108870L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>modifs</h4>
<pre>java.util.ArrayList&lt;<a href="bacnet/datamodel/proteomics/NTerm.TypeModif.html" title="enum in bacnet.datamodel.proteomics">NTerm.TypeModif</a>&gt; modifs</pre>
</li>
<li class="blockList">
<h4>nTermRef</h4>
<pre><a href="bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a> nTermRef</pre>
</li>
<li class="blockList">
<h4>nTerms</h4>
<pre>java.util.ArrayList&lt;<a href="bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a>&gt; nTerms</pre>
</li>
<li class="blockList">
<h4>overlapingGene</h4>
<pre>java.lang.String overlapingGene</pre>
</li>
<li class="blockList">
<h4>overlaps</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; overlaps</pre>
</li>
<li class="blockList">
<h4>refSequence</h4>
<pre>java.lang.String refSequence</pre>
</li>
<li class="blockListLast">
<h4>totalSpectra</h4>
<pre>int totalSpectra</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.datamodel.sequence</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.datamodel.sequence.Gene">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">bacnet.datamodel.sequence.Gene</a> extends <a href="bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>7176949538355147984L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>cog</h4>
<pre>java.lang.String cog</pre>
</li>
<li class="blockList">
<h4>geneName</h4>
<pre>java.lang.String geneName</pre>
</li>
<li class="blockList">
<h4>lengthAA</h4>
<pre>int lengthAA</pre>
</li>
<li class="blockList">
<h4>molMass</h4>
<pre>double molMass</pre>
</li>
<li class="blockList">
<h4>newLocusTag</h4>
<pre>java.lang.String newLocusTag</pre>
</li>
<li class="blockList">
<h4>oldLocusTag</h4>
<pre>java.lang.String oldLocusTag</pre>
</li>
<li class="blockList">
<h4>operon</h4>
<pre>java.lang.String operon</pre>
</li>
<li class="blockList">
<h4>product</h4>
<pre>java.lang.String product</pre>
</li>
<li class="blockList">
<h4>protein_id</h4>
<pre>java.lang.String protein_id</pre>
</li>
<li class="blockListLast">
<h4>pseudogene</h4>
<pre>boolean pseudogene</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.datamodel.sequence.NcRNA">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/sequence/NcRNA.html" title="class in bacnet.datamodel.sequence">bacnet.datamodel.sequence.NcRNA</a> extends <a href="bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a> implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>3968704131603938471L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>product</h4>
<pre>java.lang.String product</pre>
</li>
<li class="blockListLast">
<h4>typeNcRNA</h4>
<pre><a href="bacnet/datamodel/sequence/NcRNA.TypeNcRNA.html" title="enum in bacnet.datamodel.sequence">NcRNA.TypeNcRNA</a> typeNcRNA</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.datamodel.sequence.Operon">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">bacnet.datamodel.sequence.Operon</a> extends <a href="bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-8863494706241798280L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockListLast">
<h4>genes</h4>
<pre>java.util.TreeSet&lt;java.lang.String&gt; genes</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.datamodel.sequence.Sequence">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">bacnet.datamodel.sequence.Sequence</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>4818676464640692972L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>begin</h4>
<pre>int begin</pre>
</li>
<li class="blockList">
<h4>chromosomeID</h4>
<pre>java.lang.String chromosomeID</pre>
</li>
<li class="blockList">
<h4>comment</h4>
<pre>java.lang.String comment</pre>
</li>
<li class="blockList">
<h4>conservation</h4>
<pre>int conservation</pre>
</li>
<li class="blockList">
<h4>conservationHashMap</h4>
<pre>java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt; conservationHashMap</pre>
<div class="block">HashMap linking genomeName to conserved gene<br>
 Give information about the genome in which this gene is conserved</div>
</li>
<li class="blockList">
<h4>end</h4>
<pre>int end</pre>
</li>
<li class="blockList">
<h4>features</h4>
<pre>java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt; features</pre>
<div class="block">List of features</div>
</li>
<li class="blockList">
<h4>genomeName</h4>
<pre>java.lang.String genomeName</pre>
</li>
<li class="blockList">
<h4>id</h4>
<pre>java.lang.String id</pre>
</li>
<li class="blockList">
<h4>length</h4>
<pre>int length</pre>
</li>
<li class="blockList">
<h4>name</h4>
<pre>java.lang.String name</pre>
</li>
<li class="blockList">
<h4>ref</h4>
<pre>java.lang.String ref</pre>
</li>
<li class="blockList">
<h4>signatures</h4>
<pre>java.util.TreeSet&lt;java.lang.String&gt; signatures</pre>
<div class="block">List of signatures ID which this Sequence is part of</div>
</li>
<li class="blockList">
<h4>strand</h4>
<pre>char strand</pre>
</li>
<li class="blockList">
<h4>synonym</h4>
<pre>java.util.TreeSet&lt;java.lang.String&gt; synonym</pre>
</li>
<li class="blockListLast">
<h4>type</h4>
<pre><a href="bacnet/datamodel/sequence/Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a> type</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.datamodel.sequence.Srna">
<!--   -->
</a>
<h3>Class <a href="bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">bacnet.datamodel.sequence.Srna</a> extends <a href="bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>7964059340593114897L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>foundIn</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; foundIn</pre>
</li>
<li class="blockListLast">
<h4>typeSrna</h4>
<pre><a href="bacnet/datamodel/sequence/Srna.TypeSrna.html" title="enum in bacnet.datamodel.sequence">Srna.TypeSrna</a> typeSrna</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.e4.rap</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.e4.rap.BannerView">
<!--   -->
</a>
<h3>Class <a href="bacnet/e4/rap/BannerView.html" title="class in bacnet.e4.rap">bacnet.e4.rap.BannerView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>8818752955596747659L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>canvas</h4>
<pre>org.eclipse.swt.widgets.Canvas canvas</pre>
<div class="block">In this Canvas we will draw the Banner of the website/software</div>
</li>
<li class="blockList">
<h4>modelService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EModelService modelService</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>tltmAbout</h4>
<pre>org.eclipse.swt.widgets.ToolItem tltmAbout</pre>
</li>
<li class="blockList">
<h4>tltmGenomics</h4>
<pre>org.eclipse.swt.widgets.ToolItem tltmGenomics</pre>
</li>
<li class="blockList">
<h4>tltmGitlab</h4>
<pre>org.eclipse.swt.widgets.ToolItem tltmGitlab</pre>
</li>
<li class="blockList">
<h4>tltmPrintScreen</h4>
<pre>org.eclipse.swt.widgets.ToolItem tltmPrintScreen</pre>
</li>
<li class="blockList">
<h4>tltmProteomics</h4>
<pre>org.eclipse.swt.widgets.ToolItem tltmProteomics</pre>
</li>
<li class="blockList">
<h4>tltmTranscriptomics</h4>
<pre>org.eclipse.swt.widgets.ToolItem tltmTranscriptomics</pre>
</li>
<li class="blockListLast">
<h4>toolBar</h4>
<pre>org.eclipse.swt.widgets.ToolBar toolBar</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.e4.rap.InitViewBacnet">
<!--   -->
</a>
<h3>Class <a href="bacnet/e4/rap/InitViewBacnet.html" title="class in bacnet.e4.rap">bacnet.e4.rap.InitViewBacnet</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-3252983689419871498L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnCRISPR</h4>
<pre>org.eclipse.swt.widgets.Button btnCRISPR</pre>
</li>
<li class="blockList">
<h4>btnLeishomics</h4>
<pre>org.eclipse.swt.widgets.Button btnLeishomics</pre>
</li>
<li class="blockList">
<h4>btnListeriomics</h4>
<pre>org.eclipse.swt.widgets.Button btnListeriomics</pre>
</li>
<li class="blockList">
<h4>btnYersiniomics</h4>
<pre>org.eclipse.swt.widgets.Button btnYersiniomics</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>lblTzest</h4>
<pre>org.eclipse.swt.widgets.Label lblTzest</pre>
</li>
<li class="blockList">
<h4>linkHUB</h4>
<pre>org.eclipse.swt.widgets.Link linkHUB</pre>
</li>
<li class="blockList">
<h4>linkLicenceField</h4>
<pre>org.eclipse.swt.widgets.Link linkLicenceField</pre>
</li>
<li class="blockList">
<h4>linkPubli</h4>
<pre>org.eclipse.swt.widgets.Link linkPubli</pre>
</li>
<li class="blockList">
<h4>linkUIBC</h4>
<pre>org.eclipse.swt.widgets.Link linkUIBC</pre>
</li>
<li class="blockList">
<h4>modelService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EModelService modelService</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockListLast">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.e4.rap.InitViewCRISPR">
<!--   -->
</a>
<h3>Class <a href="bacnet/e4/rap/InitViewCRISPR.html" title="class in bacnet.e4.rap">bacnet.e4.rap.InitViewCRISPR</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>5364834050007950311L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnLoadCrisprSeq</h4>
<pre>org.eclipse.swt.widgets.Button btnLoadCrisprSeq</pre>
</li>
<li class="blockList">
<h4>btnPredictCrisprGuide</h4>
<pre>org.eclipse.swt.widgets.Button btnPredictCrisprGuide</pre>
</li>
<li class="blockList">
<h4>btnTableGuide</h4>
<pre>org.eclipse.swt.widgets.Button btnTableGuide</pre>
</li>
<li class="blockList">
<h4>linkHUB</h4>
<pre>org.eclipse.swt.widgets.Link linkHUB</pre>
</li>
<li class="blockList">
<h4>linkLicenceField</h4>
<pre>org.eclipse.swt.widgets.Link linkLicenceField</pre>
</li>
<li class="blockList">
<h4>linkPubli</h4>
<pre>org.eclipse.swt.widgets.Link linkPubli</pre>
</li>
<li class="blockList">
<h4>linkUIBC</h4>
<pre>org.eclipse.swt.widgets.Link linkUIBC</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>scrolledComposite</h4>
<pre>org.eclipse.swt.custom.ScrolledComposite scrolledComposite</pre>
</li>
<li class="blockListLast">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.e4.rap.InitViewLeishmania">
<!--   -->
</a>
<h3>Class <a href="bacnet/e4/rap/InitViewLeishmania.html" title="class in bacnet.e4.rap">bacnet.e4.rap.InitViewLeishmania</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-3252983689419871498L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnLeisheild</h4>
<pre>org.eclipse.swt.widgets.Button btnLeisheild</pre>
</li>
<li class="blockList">
<h4>btnStat</h4>
<pre>org.eclipse.swt.widgets.Button btnStat</pre>
</li>
<li class="blockList">
<h4>composite_1</h4>
<pre>org.eclipse.swt.widgets.Composite composite_1</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>lblTzest</h4>
<pre>org.eclipse.swt.widgets.Label lblTzest</pre>
</li>
<li class="blockList">
<h4>modelService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EModelService modelService</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>scrolledComposite</h4>
<pre>org.eclipse.swt.custom.ScrolledComposite scrolledComposite</pre>
</li>
<li class="blockListLast">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.e4.rap.InitViewListeria">
<!--   -->
</a>
<h3>Class <a href="bacnet/e4/rap/InitViewListeria.html" title="class in bacnet.e4.rap">bacnet.e4.rap.InitViewListeria</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-3252983689419871498L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnAccessWiki</h4>
<pre>org.eclipse.swt.widgets.Button btnAccessWiki</pre>
</li>
<li class="blockList">
<h4>btnBHI37</h4>
<pre>org.eclipse.swt.widgets.Button btnBHI37</pre>
</li>
<li class="blockList">
<h4>btnCoExpression</h4>
<pre>org.eclipse.swt.widgets.Button btnCoExpression</pre>
</li>
<li class="blockList">
<h4>btnGeneView</h4>
<pre>org.eclipse.swt.widgets.Button btnGeneView</pre>
</li>
<li class="blockList">
<h4>btnGenomics</h4>
<pre>org.eclipse.swt.widgets.Button btnGenomics</pre>
</li>
<li class="blockList">
<h4>btnIntracellular</h4>
<pre>org.eclipse.swt.widgets.Button btnIntracellular</pre>
</li>
<li class="blockList">
<h4>btnLoadData</h4>
<pre>org.eclipse.swt.widgets.Button btnLoadData</pre>
</li>
<li class="blockList">
<h4>btnNTerm</h4>
<pre>org.eclipse.swt.widgets.Button btnNTerm</pre>
</li>
<li class="blockList">
<h4>btnNTermTable</h4>
<pre>org.eclipse.swt.widgets.Button btnNTermTable</pre>
</li>
<li class="blockList">
<h4>btnProteomics</h4>
<pre>org.eclipse.swt.widgets.Button btnProteomics</pre>
</li>
<li class="blockList">
<h4>btnSrnas</h4>
<pre>org.eclipse.swt.widgets.Button btnSrnas</pre>
</li>
<li class="blockList">
<h4>btnStat</h4>
<pre>org.eclipse.swt.widgets.Button btnStat</pre>
</li>
<li class="blockList">
<h4>btnSubmitData</h4>
<pre>org.eclipse.swt.widgets.Button btnSubmitData</pre>
</li>
<li class="blockList">
<h4>btnTranscriptomics</h4>
<pre>org.eclipse.swt.widgets.Button btnTranscriptomics</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>lblTzest</h4>
<pre>org.eclipse.swt.widgets.Label lblTzest</pre>
</li>
<li class="blockList">
<h4>link_NTerm</h4>
<pre>org.eclipse.swt.widgets.Link link_NTerm</pre>
</li>
<li class="blockList">
<h4>linkHUB</h4>
<pre>org.eclipse.swt.widgets.Link linkHUB</pre>
</li>
<li class="blockList">
<h4>linkLicenceField</h4>
<pre>org.eclipse.swt.widgets.Link linkLicenceField</pre>
</li>
<li class="blockList">
<h4>linkPubli</h4>
<pre>org.eclipse.swt.widgets.Link linkPubli</pre>
</li>
<li class="blockList">
<h4>linkUIBC</h4>
<pre>org.eclipse.swt.widgets.Link linkUIBC</pre>
</li>
<li class="blockList">
<h4>modelService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EModelService modelService</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockListLast">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.e4.rap.InitViewListeriaSample">
<!--   -->
</a>
<h3>Class <a href="bacnet/e4/rap/InitViewListeriaSample.html" title="class in bacnet.e4.rap">bacnet.e4.rap.InitViewListeriaSample</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-3252983689419871498L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnAccessWiki</h4>
<pre>org.eclipse.swt.widgets.Button btnAccessWiki</pre>
</li>
<li class="blockList">
<h4>btnBHI37</h4>
<pre>org.eclipse.swt.widgets.Button btnBHI37</pre>
</li>
<li class="blockList">
<h4>btnGeneView</h4>
<pre>org.eclipse.swt.widgets.Button btnGeneView</pre>
</li>
<li class="blockList">
<h4>btnGenomics</h4>
<pre>org.eclipse.swt.widgets.Button btnGenomics</pre>
</li>
<li class="blockList">
<h4>btnLoadData</h4>
<pre>org.eclipse.swt.widgets.Button btnLoadData</pre>
</li>
<li class="blockList">
<h4>btnProteomics</h4>
<pre>org.eclipse.swt.widgets.Button btnProteomics</pre>
</li>
<li class="blockList">
<h4>btnSubmitData</h4>
<pre>org.eclipse.swt.widgets.Button btnSubmitData</pre>
</li>
<li class="blockList">
<h4>btnTranscriptomics</h4>
<pre>org.eclipse.swt.widgets.Button btnTranscriptomics</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>lblTzest</h4>
<pre>org.eclipse.swt.widgets.Label lblTzest</pre>
</li>
<li class="blockList">
<h4>linkHUB</h4>
<pre>org.eclipse.swt.widgets.Link linkHUB</pre>
</li>
<li class="blockList">
<h4>linkLicenceField</h4>
<pre>org.eclipse.swt.widgets.Link linkLicenceField</pre>
</li>
<li class="blockList">
<h4>linkPubli</h4>
<pre>org.eclipse.swt.widgets.Link linkPubli</pre>
</li>
<li class="blockList">
<h4>linkUIBC</h4>
<pre>org.eclipse.swt.widgets.Link linkUIBC</pre>
</li>
<li class="blockList">
<h4>modelService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EModelService modelService</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockListLast">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.e4.rap.InitViewYersinia">
<!--   -->
</a>
<h3>Class <a href="bacnet/e4/rap/InitViewYersinia.html" title="class in bacnet.e4.rap">bacnet.e4.rap.InitViewYersinia</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-9052983689419871498L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnAccessWiki</h4>
<pre>org.eclipse.swt.widgets.Button btnAccessWiki</pre>
</li>
<li class="blockList">
<h4>btnBHI37</h4>
<pre>org.eclipse.swt.widgets.Button btnBHI37</pre>
</li>
<li class="blockList">
<h4>btnCoExpression</h4>
<pre>org.eclipse.swt.widgets.Button btnCoExpression</pre>
</li>
<li class="blockList">
<h4>btnGeneView</h4>
<pre>org.eclipse.swt.widgets.Button btnGeneView</pre>
</li>
<li class="blockList">
<h4>btnGenomics</h4>
<pre>org.eclipse.swt.widgets.Button btnGenomics</pre>
</li>
<li class="blockList">
<h4>btnIntracellular</h4>
<pre>org.eclipse.swt.widgets.Button btnIntracellular</pre>
</li>
<li class="blockList">
<h4>btnLoadData</h4>
<pre>org.eclipse.swt.widgets.Button btnLoadData</pre>
</li>
<li class="blockList">
<h4>btnProteomics</h4>
<pre>org.eclipse.swt.widgets.Button btnProteomics</pre>
</li>
<li class="blockList">
<h4>btnSubmitData</h4>
<pre>org.eclipse.swt.widgets.Button btnSubmitData</pre>
</li>
<li class="blockList">
<h4>btnTranscriptomics</h4>
<pre>org.eclipse.swt.widgets.Button btnTranscriptomics</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>lblTzest</h4>
<pre>org.eclipse.swt.widgets.Label lblTzest</pre>
</li>
<li class="blockList">
<h4>linkHUB</h4>
<pre>org.eclipse.swt.widgets.Link linkHUB</pre>
</li>
<li class="blockList">
<h4>linkLicenceField</h4>
<pre>org.eclipse.swt.widgets.Link linkLicenceField</pre>
</li>
<li class="blockList">
<h4>linkPubli</h4>
<pre>org.eclipse.swt.widgets.Link linkPubli</pre>
</li>
<li class="blockList">
<h4>linkUIBC</h4>
<pre>org.eclipse.swt.widgets.Link linkUIBC</pre>
</li>
<li class="blockList">
<h4>modelService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EModelService modelService</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockListLast">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.e4.rap.SubmitDataView">
<!--   -->
</a>
<h3>Class <a href="bacnet/e4/rap/SubmitDataView.html" title="class in bacnet.e4.rap">bacnet.e4.rap.SubmitDataView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-2315247992526407384L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnHelp</h4>
<pre>org.eclipse.swt.widgets.Button btnHelp</pre>
</li>
<li class="blockList">
<h4>btnPreview</h4>
<pre>org.eclipse.swt.widgets.Button btnPreview</pre>
</li>
<li class="blockList">
<h4>btnSubmit</h4>
<pre>org.eclipse.swt.widgets.Button btnSubmit</pre>
</li>
<li class="blockList">
<h4>composite</h4>
<pre>org.eclipse.swt.widgets.Composite composite</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>textExpName</h4>
<pre>org.eclipse.swt.widgets.Text textExpName</pre>
</li>
<li class="blockList">
<h4>textNbDataset</h4>
<pre>org.eclipse.swt.widgets.Text textNbDataset</pre>
</li>
<li class="blockList">
<h4>textSubmiInstitution</h4>
<pre>org.eclipse.swt.widgets.Text textSubmiInstitution</pre>
</li>
<li class="blockList">
<h4>textSubmit</h4>
<pre>org.eclipse.swt.widgets.Text textSubmit</pre>
</li>
<li class="blockList">
<h4>textSubmitterName</h4>
<pre>org.eclipse.swt.widgets.Text textSubmitterName</pre>
</li>
<li class="blockList">
<h4>textTypeData</h4>
<pre>org.eclipse.swt.widgets.Text textTypeData</pre>
</li>
<li class="blockList">
<h4>txtDataID</h4>
<pre>org.eclipse.swt.widgets.Text txtDataID</pre>
</li>
<li class="blockList">
<h4>txtDescriptionExp</h4>
<pre>org.eclipse.swt.widgets.Text txtDescriptionExp</pre>
</li>
<li class="blockList">
<h4>txtEmail</h4>
<pre>org.eclipse.swt.widgets.Text txtEmail</pre>
</li>
<li class="blockListLast">
<h4>txtpublication</h4>
<pre>org.eclipse.swt.widgets.Text txtpublication</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.e4.rap.SynTView">
<!--   -->
</a>
<h3>Class <a href="bacnet/e4/rap/SynTView.html" title="class in bacnet.e4.rap">bacnet.e4.rap.SynTView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-4429010895850129342L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>browserGraph</h4>
<pre>org.eclipse.swt.browser.Browser browserGraph</pre>
</li>
<li class="blockList">
<h4>btnHelp</h4>
<pre>org.eclipse.swt.widgets.Button btnHelp</pre>
</li>
<li class="blockList">
<h4>btnSelectGenomeElements</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectGenomeElements</pre>
</li>
<li class="blockList">
<h4>compositeTitle</h4>
<pre>org.eclipse.swt.widgets.Composite compositeTitle</pre>
</li>
<li class="blockList">
<h4>filteredNetwork</h4>
<pre><a href="bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a> filteredNetwork</pre>
</li>
<li class="blockList">
<h4>generalNetwork</h4>
<pre><a href="bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a> generalNetwork</pre>
</li>
<li class="blockList">
<h4>genome</h4>
<pre><a href="bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> genome</pre>
</li>
<li class="blockList">
<h4>genomeName</h4>
<pre>java.lang.String genomeName</pre>
</li>
<li class="blockList">
<h4>lblSearch</h4>
<pre>org.eclipse.swt.widgets.Label lblSearch</pre>
</li>
<li class="blockList">
<h4>listGenomeElements</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; listGenomeElements</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>svgName</h4>
<pre>java.lang.String svgName</pre>
</li>
<li class="blockList">
<h4>tableGenes</h4>
<pre>org.eclipse.swt.widgets.Table tableGenes</pre>
</li>
<li class="blockListLast">
<h4>txtSearch</h4>
<pre>org.eclipse.swt.widgets.Text txtSearch</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.e4.rap.setup</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.e4.rap.setup.SetupPart">
<!--   -->
</a>
<h3>Class <a href="bacnet/e4/rap/setup/SetupPart.html" title="class in bacnet.e4.rap.setup">bacnet.e4.rap.setup.SetupPart</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-3563942149356313184L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnAddBiologicalConditions</h4>
<pre>org.eclipse.swt.widgets.Button btnAddBiologicalConditions</pre>
</li>
<li class="blockList">
<h4>btnAddComparisonsFrom</h4>
<pre>org.eclipse.swt.widgets.Button btnAddComparisonsFrom</pre>
</li>
<li class="blockList">
<h4>btnAddGenomes</h4>
<pre>org.eclipse.swt.widgets.Button btnAddGenomes</pre>
</li>
<li class="blockList">
<h4>btnAddProteomesFrom</h4>
<pre>org.eclipse.swt.widgets.Button btnAddProteomesFrom</pre>
</li>
<li class="blockList">
<h4>btnCreateExprPrteome</h4>
<pre>org.eclipse.swt.widgets.Button btnCreateExprPrteome</pre>
</li>
<li class="blockList">
<h4>btnCreateLogfcTranscriptome</h4>
<pre>org.eclipse.swt.widgets.Button btnCreateLogfcTranscriptome</pre>
</li>
<li class="blockList">
<h4>btnCreateProteomicsTable</h4>
<pre>org.eclipse.swt.widgets.Button btnCreateProteomicsTable</pre>
</li>
<li class="blockList">
<h4>btnCreateTranscriptomicsComparisons</h4>
<pre>org.eclipse.swt.widgets.Button btnCreateTranscriptomicsComparisons</pre>
</li>
<li class="blockList">
<h4>btnCreateTranscriptomicsTable</h4>
<pre>org.eclipse.swt.widgets.Button btnCreateTranscriptomicsTable</pre>
</li>
<li class="blockList">
<h4>btnDownloadGenomesFrom</h4>
<pre>org.eclipse.swt.widgets.Button btnDownloadGenomesFrom</pre>
</li>
<li class="blockList">
<h4>btnValidateBioconditionsDatabase</h4>
<pre>org.eclipse.swt.widgets.Button btnValidateBioconditionsDatabase</pre>
</li>
<li class="blockList">
<h4>btnValidateComparisonsDatabase</h4>
<pre>org.eclipse.swt.widgets.Button btnValidateComparisonsDatabase</pre>
</li>
<li class="blockList">
<h4>btnValidateGenomicsDatabase</h4>
<pre>org.eclipse.swt.widgets.Button btnValidateGenomicsDatabase</pre>
</li>
<li class="blockList">
<h4>btnValidateProteomics</h4>
<pre>org.eclipse.swt.widgets.Button btnValidateProteomics</pre>
</li>
<li class="blockList">
<h4>button_3</h4>
<pre>org.eclipse.swt.widgets.Button button_3</pre>
</li>
<li class="blockList">
<h4>buttonAddTranscriptomics</h4>
<pre>org.eclipse.swt.widgets.Button buttonAddTranscriptomics</pre>
</li>
<li class="blockList">
<h4>buttonValidateTrancriptomics</h4>
<pre>org.eclipse.swt.widgets.Button buttonValidateTrancriptomics</pre>
</li>
<li class="blockList">
<h4>comboProject</h4>
<pre>org.eclipse.swt.widgets.Combo comboProject</pre>
</li>
<li class="blockList">
<h4>composite_8</h4>
<pre>org.eclipse.swt.widgets.Composite composite_8</pre>
</li>
<li class="blockList">
<h4>console</h4>
<pre>org.eclipse.swt.widgets.Text console</pre>
</li>
<li class="blockList">
<h4>dataValidation</h4>
<pre><a href="bacnet/scripts/database/DataValidation.html" title="class in bacnet.scripts.database">DataValidation</a> dataValidation</pre>
</li>
<li class="blockList">
<h4>label_Transcriptomics</h4>
<pre>org.eclipse.swt.widgets.Label label_Transcriptomics</pre>
</li>
<li class="blockList">
<h4>labelGenomes</h4>
<pre>org.eclipse.swt.widgets.Label labelGenomes</pre>
</li>
<li class="blockList">
<h4>labelProteomics</h4>
<pre>org.eclipse.swt.widgets.Label labelProteomics</pre>
</li>
<li class="blockList">
<h4>lblBioconditions</h4>
<pre>org.eclipse.swt.widgets.Label lblBioconditions</pre>
</li>
<li class="blockList">
<h4>lblBioCondPath</h4>
<pre>org.eclipse.swt.widgets.Label lblBioCondPath</pre>
</li>
<li class="blockList">
<h4>lblComparisons</h4>
<pre>org.eclipse.swt.widgets.Label lblComparisons</pre>
</li>
<li class="blockList">
<h4>lblComparisonsPath</h4>
<pre>org.eclipse.swt.widgets.Label lblComparisonsPath</pre>
</li>
<li class="blockList">
<h4>lblHaveYouCreated</h4>
<pre>org.eclipse.swt.widgets.Label lblHaveYouCreated</pre>
</li>
<li class="blockList">
<h4>lblNewLabel_1</h4>
<pre>org.eclipse.swt.widgets.Label lblNewLabel_1</pre>
</li>
<li class="blockList">
<h4>lblPathData</h4>
<pre>org.eclipse.swt.widgets.Label lblPathData</pre>
</li>
<li class="blockList">
<h4>lblPathGenomics</h4>
<pre>org.eclipse.swt.widgets.Label lblPathGenomics</pre>
</li>
<li class="blockList">
<h4>lblProteomePath</h4>
<pre>org.eclipse.swt.widgets.Label lblProteomePath</pre>
</li>
<li class="blockList">
<h4>lblTechno</h4>
<pre>org.eclipse.swt.widgets.Label lblTechno</pre>
</li>
<li class="blockList">
<h4>lblTranscriPath</h4>
<pre>org.eclipse.swt.widgets.Label lblTranscriPath</pre>
</li>
<li class="blockList">
<h4>logs</h4>
<pre>java.lang.String logs</pre>
</li>
<li class="blockList">
<h4>modelService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EModelService modelService</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>tableBioCondition</h4>
<pre>org.eclipse.swt.widgets.Table tableBioCondition</pre>
</li>
<li class="blockList">
<h4>tableComparisons</h4>
<pre>org.eclipse.swt.widgets.Table tableComparisons</pre>
</li>
<li class="blockList">
<h4>tableGenome</h4>
<pre>org.eclipse.swt.widgets.Table tableGenome</pre>
</li>
<li class="blockList">
<h4>tableProteomics</h4>
<pre>org.eclipse.swt.widgets.Table tableProteomics</pre>
</li>
<li class="blockList">
<h4>tableTranscriptomes</h4>
<pre>org.eclipse.swt.widgets.Table tableTranscriptomes</pre>
</li>
<li class="blockListLast">
<h4>tbtmComparisons</h4>
<pre>org.eclipse.swt.widgets.TabItem tbtmComparisons</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.expressionAtlas</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.expressionAtlas.HeatMapMultiOmicsView">
<!--   -->
</a>
<h3>Class <a href="bacnet/expressionAtlas/HeatMapMultiOmicsView.html" title="class in bacnet.expressionAtlas">bacnet.expressionAtlas.HeatMapMultiOmicsView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-2640442866422682301L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>allButtons</h4>
<pre>java.util.ArrayList&lt;org.eclipse.swt.widgets.Button&gt; allButtons</pre>
</li>
<li class="blockList">
<h4>browserGraph</h4>
<pre>org.eclipse.swt.browser.Browser browserGraph</pre>
</li>
<li class="blockList">
<h4>btnDnDNA</h4>
<pre>org.eclipse.swt.widgets.Button btnDnDNA</pre>
</li>
<li class="blockList">
<h4>btnDnProt</h4>
<pre>org.eclipse.swt.widgets.Button btnDnProt</pre>
</li>
<li class="blockList">
<h4>btnDnRNA</h4>
<pre>org.eclipse.swt.widgets.Button btnDnRNA</pre>
</li>
<li class="blockList">
<h4>btnListIntersection</h4>
<pre>org.eclipse.swt.widgets.Button btnListIntersection</pre>
</li>
<li class="blockList">
<h4>btnListUnion</h4>
<pre>org.eclipse.swt.widgets.Button btnListUnion</pre>
</li>
<li class="blockList">
<h4>btnNoDNA</h4>
<pre>org.eclipse.swt.widgets.Button btnNoDNA</pre>
</li>
<li class="blockList">
<h4>btnNoProt</h4>
<pre>org.eclipse.swt.widgets.Button btnNoProt</pre>
</li>
<li class="blockList">
<h4>btnNoRNA</h4>
<pre>org.eclipse.swt.widgets.Button btnNoRNA</pre>
</li>
<li class="blockList">
<h4>btnPrint</h4>
<pre>org.eclipse.swt.widgets.Button btnPrint</pre>
</li>
<li class="blockList">
<h4>btnSavePNG</h4>
<pre>org.eclipse.swt.widgets.Button btnSavePNG</pre>
</li>
<li class="blockList">
<h4>btnUpdateCutoff</h4>
<pre>org.eclipse.swt.widgets.Button btnUpdateCutoff</pre>
</li>
<li class="blockList">
<h4>btnUpDNA</h4>
<pre>org.eclipse.swt.widgets.Button btnUpDNA</pre>
</li>
<li class="blockList">
<h4>btnUpProt</h4>
<pre>org.eclipse.swt.widgets.Button btnUpProt</pre>
</li>
<li class="blockList">
<h4>btnUpRNA</h4>
<pre>org.eclipse.swt.widgets.Button btnUpRNA</pre>
</li>
<li class="blockList">
<h4>cmbXaxis</h4>
<pre>org.eclipse.swt.widgets.Combo cmbXaxis</pre>
</li>
<li class="blockList">
<h4>cmbYaxis</h4>
<pre>org.eclipse.swt.widgets.Combo cmbYaxis</pre>
</li>
<li class="blockList">
<h4>compFilter</h4>
<pre>org.eclipse.swt.widgets.Composite compFilter</pre>
</li>
<li class="blockList">
<h4>composite</h4>
<pre>org.eclipse.swt.widgets.Composite composite</pre>
</li>
<li class="blockList">
<h4>compositeSummary</h4>
<pre>org.eclipse.swt.widgets.Composite compositeSummary</pre>
</li>
<li class="blockList">
<h4>downDNAList</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; downDNAList</pre>
<div class="block">List of genes down for DNA</div>
</li>
<li class="blockList">
<h4>downProtList</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; downProtList</pre>
<div class="block">List of genes down for Prot</div>
</li>
<li class="blockList">
<h4>downRNAList</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; downRNAList</pre>
<div class="block">List of genes down for RNA</div>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>genomeName</h4>
<pre>java.lang.String genomeName</pre>
</li>
<li class="blockList">
<h4>intersect</h4>
<pre>boolean intersect</pre>
<div class="block">Describe comportment of list overlap</div>
</li>
<li class="blockList">
<h4>logfcOmicsMatrix</h4>
<pre><a href="bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a> logfcOmicsMatrix</pre>
<div class="block">LogFC Matrix = Multiomics with only the logFC values</div>
</li>
<li class="blockList">
<h4>noDNAList</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; noDNAList</pre>
<div class="block">List of genes no change for DNA</div>
</li>
<li class="blockList">
<h4>noProtList</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; noProtList</pre>
<div class="block">List of genes no change for Prot</div>
</li>
<li class="blockList">
<h4>noRNAList</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; noRNAList</pre>
<div class="block">List of genes no change for RNA</div>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>sequence</h4>
<pre>java.lang.String sequence</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>tabFolder</h4>
<pre>org.eclipse.swt.widgets.TabFolder tabFolder</pre>
</li>
<li class="blockList">
<h4>tableComposite</h4>
<pre><a href="bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a> tableComposite</pre>
</li>
<li class="blockList">
<h4>tbtmHeatmap</h4>
<pre>org.eclipse.swt.widgets.TabItem tbtmHeatmap</pre>
</li>
<li class="blockList">
<h4>tbtmScatterPlot</h4>
<pre>org.eclipse.swt.widgets.TabItem tbtmScatterPlot</pre>
</li>
<li class="blockList">
<h4>txtCutoffLogFC</h4>
<pre>org.eclipse.swt.widgets.Text txtCutoffLogFC</pre>
</li>
<li class="blockList">
<h4>upDNAList</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; upDNAList</pre>
<div class="block">List of genes up for DNA</div>
</li>
<li class="blockList">
<h4>upProtList</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; upProtList</pre>
<div class="block">List of genes up for Prot</div>
</li>
<li class="blockList">
<h4>upRNAList</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; upRNAList</pre>
<div class="block">List of genes up for RNA</div>
</li>
<li class="blockListLast">
<h4>viewId</h4>
<pre>java.lang.String viewId</pre>
<div class="block">Current viewId = ID + math.random()</div>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.expressionAtlas.HeatMapProteomicsView">
<!--   -->
</a>
<h3>Class <a href="bacnet/expressionAtlas/HeatMapProteomicsView.html" title="class in bacnet.expressionAtlas">bacnet.expressionAtlas.HeatMapProteomicsView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>7806377909856153015L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>bioConditions</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; bioConditions</pre>
</li>
<li class="blockList">
<h4>compositeSummary</h4>
<pre>org.eclipse.swt.widgets.Composite compositeSummary</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>genomeName</h4>
<pre>java.lang.String genomeName</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>tableComposite</h4>
<pre><a href="bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a> tableComposite</pre>
</li>
<li class="blockListLast">
<h4>viewId</h4>
<pre>java.lang.String viewId</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.expressionAtlas.HeatMapTranscriptomicsView">
<!--   -->
</a>
<h3>Class <a href="bacnet/expressionAtlas/HeatMapTranscriptomicsView.html" title="class in bacnet.expressionAtlas">bacnet.expressionAtlas.HeatMapTranscriptomicsView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-2640442866422682301L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnPrint</h4>
<pre>org.eclipse.swt.widgets.Button btnPrint</pre>
</li>
<li class="blockList">
<h4>btnUpdateCutoff</h4>
<pre>org.eclipse.swt.widgets.Button btnUpdateCutoff</pre>
</li>
<li class="blockList">
<h4>cmbDataUsed</h4>
<pre>org.eclipse.swt.widgets.Combo cmbDataUsed</pre>
</li>
<li class="blockList">
<h4>compositeSummary</h4>
<pre>org.eclipse.swt.widgets.Composite compositeSummary</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>genomeName</h4>
<pre>java.lang.String genomeName</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>sequence</h4>
<pre>java.lang.String sequence</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>tableComposite</h4>
<pre><a href="bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">TableCompositeHeatMap</a> tableComposite</pre>
</li>
<li class="blockList">
<h4>txtCutoffLogFC</h4>
<pre>org.eclipse.swt.widgets.Text txtCutoffLogFC</pre>
</li>
<li class="blockListLast">
<h4>viewId</h4>
<pre>java.lang.String viewId</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.expressionAtlas.ProteomicsDataFilterComposite">
<!--   -->
</a>
<h3>Class <a href="bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">bacnet.expressionAtlas.ProteomicsDataFilterComposite</a> extends org.eclipse.swt.widgets.Composite implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-4827398812522159453L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>allButtons</h4>
<pre>java.util.ArrayList&lt;org.eclipse.swt.widgets.Button&gt; allButtons</pre>
</li>
<li class="blockList">
<h4>btnBhiBroth</h4>
<pre>org.eclipse.swt.widgets.Button btnBhiBroth</pre>
</li>
<li class="blockList">
<h4>btnCellWall</h4>
<pre>org.eclipse.swt.widgets.Button btnCellWall</pre>
</li>
<li class="blockList">
<h4>btnCytoplasm</h4>
<pre>org.eclipse.swt.widgets.Button btnCytoplasm</pre>
</li>
<li class="blockList">
<h4>btnExponential</h4>
<pre>org.eclipse.swt.widgets.Button btnExponential</pre>
</li>
<li class="blockList">
<h4>btnHumanBloodCells</h4>
<pre>org.eclipse.swt.widgets.Button btnHumanBloodCells</pre>
</li>
<li class="blockList">
<h4>btnLbBroth</h4>
<pre>org.eclipse.swt.widgets.Button btnLbBroth</pre>
</li>
<li class="blockList">
<h4>btnMembrane</h4>
<pre>org.eclipse.swt.widgets.Button btnMembrane</pre>
</li>
<li class="blockList">
<h4>btnMinimalMediaBroth</h4>
<pre>org.eclipse.swt.widgets.Button btnMinimalMediaBroth</pre>
</li>
<li class="blockList">
<h4>btnMouseMacrophagesCells</h4>
<pre>org.eclipse.swt.widgets.Button btnMouseMacrophagesCells</pre>
</li>
<li class="blockList">
<h4>btnSecretome</h4>
<pre>org.eclipse.swt.widgets.Button btnSecretome</pre>
</li>
<li class="blockList">
<h4>btnStationnary</h4>
<pre>org.eclipse.swt.widgets.Button btnStationnary</pre>
</li>
<li class="blockList">
<h4>btnUnpublished</h4>
<pre>org.eclipse.swt.widgets.Button btnUnpublished</pre>
</li>
<li class="blockList">
<h4>comboGenome</h4>
<pre>org.eclipse.swt.widgets.Combo comboGenome</pre>
</li>
<li class="blockList">
<h4>textSearch</h4>
<pre>org.eclipse.swt.widgets.Text textSearch</pre>
</li>
<li class="blockListLast">
<h4>view</h4>
<pre><a href="bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">ProteomicsView</a> view</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.expressionAtlas.ProteomicsView">
<!--   -->
</a>
<h3>Class <a href="bacnet/expressionAtlas/ProteomicsView.html" title="class in bacnet.expressionAtlas">bacnet.expressionAtlas.ProteomicsView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>3273618417542386059L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>bioConds</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioConds</pre>
<div class="block">List used to init data</div>
</li>
<li class="blockList">
<h4>bioCondsArray</h4>
<pre>java.lang.String[][] bioCondsArray</pre>
<div class="block">Array used to init data</div>
</li>
<li class="blockList">
<h4>bioCondsToDisplay</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioCondsToDisplay</pre>
<div class="block">Array which will be displayed</div>
</li>
<li class="blockList">
<h4>btnGenomeViewer</h4>
<pre>org.eclipse.swt.widgets.Button btnGenomeViewer</pre>
</li>
<li class="blockList">
<h4>btnHeatmap</h4>
<pre>org.eclipse.swt.widgets.Button btnHeatmap</pre>
</li>
<li class="blockList">
<h4>btnHelp</h4>
<pre>org.eclipse.swt.widgets.Button btnHelp</pre>
</li>
<li class="blockList">
<h4>btnSaveTxt</h4>
<pre>org.eclipse.swt.widgets.Button btnSaveTxt</pre>
</li>
<li class="blockList">
<h4>btnSelectall</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectall</pre>
</li>
<li class="blockList">
<h4>btnUnselectall</h4>
<pre>org.eclipse.swt.widgets.Button btnUnselectall</pre>
</li>
<li class="blockList">
<h4>columnNames</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; columnNames</pre>
</li>
<li class="blockList">
<h4>comparatorBioCondition</h4>
<pre><a href="bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a> comparatorBioCondition</pre>
</li>
<li class="blockList">
<h4>compositeDataFilter</h4>
<pre><a href="bacnet/expressionAtlas/ProteomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">ProteomicsDataFilterComposite</a> compositeDataFilter</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>imageGeneExpr</h4>
<pre>org.eclipse.swt.graphics.Image imageGeneExpr</pre>
</li>
<li class="blockList">
<h4>imageTSS</h4>
<pre>org.eclipse.swt.graphics.Image imageTSS</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>selectedProteomes</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; selectedProteomes</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>tableProteome</h4>
<pre>org.eclipse.swt.widgets.Table tableProteome</pre>
</li>
<li class="blockListLast">
<h4>tableProteomeViewer</h4>
<pre>org.eclipse.jface.viewers.TableViewer tableProteomeViewer</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.expressionAtlas.TableCompositeHeatMap">
<!--   -->
</a>
<h3>Class <a href="bacnet/expressionAtlas/TableCompositeHeatMap.html" title="class in bacnet.expressionAtlas">bacnet.expressionAtlas.TableCompositeHeatMap</a> extends org.eclipse.swt.widgets.Composite implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-5898609236718602892L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnColormapper</h4>
<pre>org.eclipse.swt.widgets.Button btnColormapper</pre>
</li>
<li class="blockList">
<h4>btnDisplayValues</h4>
<pre>org.eclipse.swt.widgets.Button btnDisplayValues</pre>
</li>
<li class="blockList">
<h4>btnHelp</h4>
<pre>org.eclipse.swt.widgets.Button btnHelp</pre>
</li>
<li class="blockList">
<h4>btnHide</h4>
<pre>org.eclipse.swt.widgets.Button btnHide</pre>
</li>
<li class="blockList">
<h4>btnSaveText</h4>
<pre>org.eclipse.swt.widgets.Button btnSaveText</pre>
</li>
<li class="blockList">
<h4>btnZoomIn</h4>
<pre>org.eclipse.swt.widgets.Button btnZoomIn</pre>
</li>
<li class="blockList">
<h4>btnZoomOut</h4>
<pre>org.eclipse.swt.widgets.Button btnZoomOut</pre>
</li>
<li class="blockList">
<h4>colorMapperList</h4>
<pre><a href="bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a> colorMapperList</pre>
</li>
<li class="blockList">
<h4>columnNames</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; columnNames</pre>
</li>
<li class="blockList">
<h4>comparator</h4>
<pre><a href="bacnet/table/core/TableViewerComparator.html" title="class in bacnet.table.core">TableViewerComparator</a> comparator</pre>
</li>
<li class="blockList">
<h4>composite</h4>
<pre>org.eclipse.swt.widgets.Composite composite</pre>
</li>
<li class="blockList">
<h4>composite_4</h4>
<pre>org.eclipse.swt.widgets.Composite composite_4</pre>
</li>
<li class="blockList">
<h4>displayValues</h4>
<pre>boolean displayValues</pre>
</li>
<li class="blockList">
<h4>excludeColumn</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; excludeColumn</pre>
</li>
<li class="blockList">
<h4>excludeRow</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; excludeRow</pre>
</li>
<li class="blockList">
<h4>genomeName</h4>
<pre>java.lang.String genomeName</pre>
</li>
<li class="blockList">
<h4>labelColumn</h4>
<pre>org.eclipse.swt.widgets.Label labelColumn</pre>
</li>
<li class="blockList">
<h4>labelRow</h4>
<pre>org.eclipse.swt.widgets.Label labelRow</pre>
</li>
<li class="blockList">
<h4>lblX</h4>
<pre>org.eclipse.swt.widgets.Label lblX</pre>
</li>
<li class="blockList">
<h4>lblYeah</h4>
<pre>org.eclipse.swt.widgets.Label lblYeah</pre>
</li>
<li class="blockList">
<h4>matrix</h4>
<pre><a href="bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a> matrix</pre>
</li>
<li class="blockList">
<h4>matrixDisplayed</h4>
<pre><a href="bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a> matrixDisplayed</pre>
</li>
<li class="blockList">
<h4>matrixSize</h4>
<pre><a href="bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a> matrixSize</pre>
</li>
<li class="blockList">
<h4>packColumns</h4>
<pre>boolean packColumns</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>searchFilter</h4>
<pre><a href="bacnet/table/gui/TableFilter.html" title="class in bacnet.table.gui">TableFilter</a> searchFilter</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>table</h4>
<pre>org.eclipse.swt.widgets.Table table</pre>
</li>
<li class="blockList">
<h4>tableViewer</h4>
<pre>org.eclipse.jface.viewers.TableViewer tableViewer</pre>
</li>
<li class="blockListLast">
<h4>transcriptomics</h4>
<pre>boolean transcriptomics</pre>
<div class="block">Boolean describing if it is a transcriptomics or proteomic heatmap viewer</div>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.expressionAtlas.TranscriptomicsDataFilterComposite">
<!--   -->
</a>
<h3>Class <a href="bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">bacnet.expressionAtlas.TranscriptomicsDataFilterComposite</a> extends org.eclipse.swt.widgets.Composite implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>7804472924985182424L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>allButtons</h4>
<pre>java.util.ArrayList&lt;org.eclipse.swt.widgets.Button&gt; allButtons</pre>
</li>
<li class="blockList">
<h4>btnAllMutant</h4>
<pre>org.eclipse.swt.widgets.Button btnAllMutant</pre>
</li>
<li class="blockList">
<h4>btnBhiBroth</h4>
<pre>org.eclipse.swt.widgets.Button btnBhiBroth</pre>
</li>
<li class="blockList">
<h4>btnCacoCells</h4>
<pre>org.eclipse.swt.widgets.Button btnCacoCells</pre>
</li>
<li class="blockList">
<h4>btnChooseOneMutant</h4>
<pre>org.eclipse.swt.widgets.Button btnChooseOneMutant</pre>
</li>
<li class="blockList">
<h4>btnDeath</h4>
<pre>org.eclipse.swt.widgets.Button btnDeath</pre>
</li>
<li class="blockList">
<h4>btnExponential</h4>
<pre>org.eclipse.swt.widgets.Button btnExponential</pre>
</li>
<li class="blockList">
<h4>btnGeneExpression</h4>
<pre>org.eclipse.swt.widgets.Button btnGeneExpression</pre>
</li>
<li class="blockList">
<h4>btnHumanBloodCells</h4>
<pre>org.eclipse.swt.widgets.Button btnHumanBloodCells</pre>
</li>
<li class="blockList">
<h4>btnLagPhase</h4>
<pre>org.eclipse.swt.widgets.Button btnLagPhase</pre>
</li>
<li class="blockList">
<h4>btnLbBroth</h4>
<pre>org.eclipse.swt.widgets.Button btnLbBroth</pre>
</li>
<li class="blockList">
<h4>btnMinimalMediaBroth</h4>
<pre>org.eclipse.swt.widgets.Button btnMinimalMediaBroth</pre>
</li>
<li class="blockList">
<h4>btnMouseIntestineCells</h4>
<pre>org.eclipse.swt.widgets.Button btnMouseIntestineCells</pre>
</li>
<li class="blockList">
<h4>btnMouseMacrophagesCells</h4>
<pre>org.eclipse.swt.widgets.Button btnMouseMacrophagesCells</pre>
</li>
<li class="blockList">
<h4>btnMouseSpleenCells</h4>
<pre>org.eclipse.swt.widgets.Button btnMouseSpleenCells</pre>
</li>
<li class="blockList">
<h4>btnNoneMutant</h4>
<pre>org.eclipse.swt.widgets.Button btnNoneMutant</pre>
</li>
<li class="blockList">
<h4>btnPorcineBileCells</h4>
<pre>org.eclipse.swt.widgets.Button btnPorcineBileCells</pre>
</li>
<li class="blockList">
<h4>btnRegrowth</h4>
<pre>org.eclipse.swt.widgets.Button btnRegrowth</pre>
</li>
<li class="blockList">
<h4>btnRiboSeq</h4>
<pre>org.eclipse.swt.widgets.Button btnRiboSeq</pre>
</li>
<li class="blockList">
<h4>btnRnaseq</h4>
<pre>org.eclipse.swt.widgets.Button btnRnaseq</pre>
</li>
<li class="blockList">
<h4>btnStationnary</h4>
<pre>org.eclipse.swt.widgets.Button btnStationnary</pre>
</li>
<li class="blockList">
<h4>btnSurvival</h4>
<pre>org.eclipse.swt.widgets.Button btnSurvival</pre>
</li>
<li class="blockList">
<h4>btnTermSeq</h4>
<pre>org.eclipse.swt.widgets.Button btnTermSeq</pre>
</li>
<li class="blockList">
<h4>btnTiling</h4>
<pre>org.eclipse.swt.widgets.Button btnTiling</pre>
</li>
<li class="blockList">
<h4>btnTrypticSoyBroth</h4>
<pre>org.eclipse.swt.widgets.Button btnTrypticSoyBroth</pre>
</li>
<li class="blockList">
<h4>btnTss</h4>
<pre>org.eclipse.swt.widgets.Button btnTss</pre>
</li>
<li class="blockList">
<h4>btnTurkeyDeliMeat</h4>
<pre>org.eclipse.swt.widgets.Button btnTurkeyDeliMeat</pre>
</li>
<li class="blockList">
<h4>btnUnpublished</h4>
<pre>org.eclipse.swt.widgets.Button btnUnpublished</pre>
</li>
<li class="blockList">
<h4>comboGenome</h4>
<pre>org.eclipse.swt.widgets.Combo comboGenome</pre>
</li>
<li class="blockList">
<h4>comboMutant</h4>
<pre>org.eclipse.swt.widgets.Combo comboMutant</pre>
</li>
<li class="blockList">
<h4>textSearch</h4>
<pre>org.eclipse.swt.widgets.Text textSearch</pre>
</li>
<li class="blockListLast">
<h4>view</h4>
<pre><a href="bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">TranscriptomicsView</a> view</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.expressionAtlas.TranscriptomicsView">
<!--   -->
</a>
<h3>Class <a href="bacnet/expressionAtlas/TranscriptomicsView.html" title="class in bacnet.expressionAtlas">bacnet.expressionAtlas.TranscriptomicsView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-5118476676200457964L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>bioConds</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioConds</pre>
<div class="block">List used to init data</div>
</li>
<li class="blockList">
<h4>bioCondsArray</h4>
<pre>java.lang.String[][] bioCondsArray</pre>
<div class="block">Array used to init data</div>
</li>
<li class="blockList">
<h4>bioCondsToDisplay</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioCondsToDisplay</pre>
<div class="block">Array which will be displayed</div>
</li>
<li class="blockList">
<h4>btnExpressionAtlas</h4>
<pre>org.eclipse.swt.widgets.Button btnExpressionAtlas</pre>
</li>
<li class="blockList">
<h4>btnGenomeViewer</h4>
<pre>org.eclipse.swt.widgets.Button btnGenomeViewer</pre>
</li>
<li class="blockList">
<h4>btnHelp</h4>
<pre>org.eclipse.swt.widgets.Button btnHelp</pre>
</li>
<li class="blockList">
<h4>btnSaveTxt</h4>
<pre>org.eclipse.swt.widgets.Button btnSaveTxt</pre>
</li>
<li class="blockList">
<h4>btnSelectall</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectall</pre>
</li>
<li class="blockList">
<h4>btnUnselectall</h4>
<pre>org.eclipse.swt.widgets.Button btnUnselectall</pre>
</li>
<li class="blockList">
<h4>columnNames</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; columnNames</pre>
</li>
<li class="blockList">
<h4>comparatorBioCondition</h4>
<pre><a href="bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a> comparatorBioCondition</pre>
</li>
<li class="blockList">
<h4>compositeDataFilter</h4>
<pre><a href="bacnet/expressionAtlas/TranscriptomicsDataFilterComposite.html" title="class in bacnet.expressionAtlas">TranscriptomicsDataFilterComposite</a> compositeDataFilter</pre>
<div class="block">Composite to filter list of data</div>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>imageGeneExpr</h4>
<pre>org.eclipse.swt.graphics.Image imageGeneExpr</pre>
</li>
<li class="blockList">
<h4>imageRNASeq</h4>
<pre>org.eclipse.swt.graphics.Image imageRNASeq</pre>
</li>
<li class="blockList">
<h4>imageTiling</h4>
<pre>org.eclipse.swt.graphics.Image imageTiling</pre>
</li>
<li class="blockList">
<h4>imageTilingGeneExpr</h4>
<pre>org.eclipse.swt.graphics.Image imageTilingGeneExpr</pre>
</li>
<li class="blockList">
<h4>imageTSS</h4>
<pre>org.eclipse.swt.graphics.Image imageTSS</pre>
</li>
<li class="blockList">
<h4>imageTSSTilingGeneExpr</h4>
<pre>org.eclipse.swt.graphics.Image imageTSSTilingGeneExpr</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>selectedTranscriptomes</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; selectedTranscriptomes</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>tableBioCondition</h4>
<pre>org.eclipse.swt.widgets.Table tableBioCondition</pre>
</li>
<li class="blockListLast">
<h4>tableBioConditionViewer</h4>
<pre>org.eclipse.jface.viewers.TableViewer tableBioConditionViewer</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.expressionAtlas.core</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.expressionAtlas.core.ComparisonAtlas">
<!--   -->
</a>
<h3>Class <a href="bacnet/expressionAtlas/core/ComparisonAtlas.html" title="class in bacnet.expressionAtlas.core">bacnet.expressionAtlas.core.ComparisonAtlas</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>3461420201661554414L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>excludeRows</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; excludeRows</pre>
</li>
<li class="blockList">
<h4>name</h4>
<pre>java.lang.String name</pre>
</li>
<li class="blockList">
<h4>notDiffExpressedASrnas</h4>
<pre>java.lang.String[] notDiffExpressedASrnas</pre>
</li>
<li class="blockList">
<h4>notDiffExpressedCisRegs</h4>
<pre>java.lang.String[] notDiffExpressedCisRegs</pre>
</li>
<li class="blockList">
<h4>notDiffExpressedGenes</h4>
<pre>java.lang.String[] notDiffExpressedGenes</pre>
</li>
<li class="blockList">
<h4>notDiffExpressedSrnas</h4>
<pre>java.lang.String[] notDiffExpressedSrnas</pre>
</li>
<li class="blockList">
<h4>overASrnas</h4>
<pre>java.lang.String[] overASrnas</pre>
</li>
<li class="blockList">
<h4>overCisRegs</h4>
<pre>java.lang.String[] overCisRegs</pre>
</li>
<li class="blockList">
<h4>overExpressed</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; overExpressed</pre>
</li>
<li class="blockList">
<h4>overGenes</h4>
<pre>java.lang.String[] overGenes</pre>
</li>
<li class="blockList">
<h4>overSrnas</h4>
<pre>java.lang.String[] overSrnas</pre>
</li>
<li class="blockList">
<h4>underASrnas</h4>
<pre>java.lang.String[] underASrnas</pre>
</li>
<li class="blockList">
<h4>underCisRegs</h4>
<pre>java.lang.String[] underCisRegs</pre>
</li>
<li class="blockList">
<h4>underExpressed</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; underExpressed</pre>
</li>
<li class="blockList">
<h4>underGenes</h4>
<pre>java.lang.String[] underGenes</pre>
</li>
<li class="blockListLast">
<h4>underSrnas</h4>
<pre>java.lang.String[] underSrnas</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.expressionAtlas.core.GenomeElementAtlas">
<!--   -->
</a>
<h3>Class <a href="bacnet/expressionAtlas/core/GenomeElementAtlas.html" title="class in bacnet.expressionAtlas.core">bacnet.expressionAtlas.core.GenomeElementAtlas</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>6277013816841128158L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>name</h4>
<pre>java.lang.String name</pre>
</li>
<li class="blockList">
<h4>notDiffExpresseds</h4>
<pre>java.util.TreeSet&lt;java.lang.String&gt; notDiffExpresseds</pre>
</li>
<li class="blockList">
<h4>overBioConds</h4>
<pre>java.util.TreeSet&lt;java.lang.String&gt; overBioConds</pre>
</li>
<li class="blockListLast">
<h4>underBioConds</h4>
<pre>java.util.TreeSet&lt;java.lang.String&gt; underBioConds</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.expressionAtlas.core.SelectGenomeElementDialog">
<!--   -->
</a>
<h3>Class <a href="bacnet/expressionAtlas/core/SelectGenomeElementDialog.html" title="class in bacnet.expressionAtlas.core">bacnet.expressionAtlas.core.SelectGenomeElementDialog</a> extends org.eclipse.jface.dialogs.TitleAreaDialog implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>3481072975746172059L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnDeselAS</h4>
<pre>org.eclipse.swt.widgets.Button btnDeselAS</pre>
</li>
<li class="blockList">
<h4>btnDeselGene</h4>
<pre>org.eclipse.swt.widgets.Button btnDeselGene</pre>
</li>
<li class="blockList">
<h4>btnDeselRNA</h4>
<pre>org.eclipse.swt.widgets.Button btnDeselRNA</pre>
</li>
<li class="blockList">
<h4>btnLoad</h4>
<pre>org.eclipse.swt.widgets.Button btnLoad</pre>
</li>
<li class="blockList">
<h4>btnSave</h4>
<pre>org.eclipse.swt.widgets.Button btnSave</pre>
</li>
<li class="blockList">
<h4>btnSelectAS</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectAS</pre>
</li>
<li class="blockList">
<h4>btnSelectGene</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectGene</pre>
</li>
<li class="blockList">
<h4>btnSelectRNA</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectRNA</pre>
</li>
<li class="blockList">
<h4>btnSelectSpecificList</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectSpecificList</pre>
</li>
<li class="blockList">
<h4>composite</h4>
<pre>org.eclipse.swt.widgets.Composite composite</pre>
</li>
<li class="blockList">
<h4>compositeButton</h4>
<pre>org.eclipse.swt.widgets.Composite compositeButton</pre>
</li>
<li class="blockList">
<h4>egde</h4>
<pre>boolean egde</pre>
</li>
<li class="blockList">
<h4>excludedElements</h4>
<pre>java.util.TreeSet&lt;java.lang.String&gt; excludedElements</pre>
</li>
<li class="blockList">
<h4>includeElements</h4>
<pre>java.util.TreeSet&lt;java.lang.String&gt; includeElements</pre>
</li>
<li class="blockList">
<h4>listasRNAs</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; listasRNAs</pre>
</li>
<li class="blockList">
<h4>listcisRegs</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; listcisRegs</pre>
</li>
<li class="blockList">
<h4>listGenes</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; listGenes</pre>
</li>
<li class="blockList">
<h4>listsRNAs</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; listsRNAs</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>rFamAnnot</h4>
<pre>boolean rFamAnnot</pre>
</li>
<li class="blockList">
<h4>scrolledComposite</h4>
<pre>org.eclipse.swt.custom.ScrolledComposite scrolledComposite</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>tableASRNAs</h4>
<pre>org.eclipse.swt.widgets.Table tableASRNAs</pre>
</li>
<li class="blockList">
<h4>tableGenes</h4>
<pre>org.eclipse.swt.widgets.Table tableGenes</pre>
</li>
<li class="blockListLast">
<h4>tableSRNAs</h4>
<pre>org.eclipse.swt.widgets.Table tableSRNAs</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.expressionAtlas.core.SignatureSelectionDialog">
<!--   -->
</a>
<h3>Class <a href="bacnet/expressionAtlas/core/SignatureSelectionDialog.html" title="class in bacnet.expressionAtlas.core">bacnet.expressionAtlas.core.SignatureSelectionDialog</a> extends org.eclipse.jface.dialogs.TitleAreaDialog implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>1941320621284175887L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>composite_1</h4>
<pre>org.eclipse.swt.widgets.Composite composite_1</pre>
</li>
<li class="blockList">
<h4>lblDescription</h4>
<pre>org.eclipse.swt.widgets.Label lblDescription</pre>
</li>
<li class="blockList">
<h4>lblGenesInclude</h4>
<pre>org.eclipse.swt.widgets.Label lblGenesInclude</pre>
</li>
<li class="blockList">
<h4>lblGenome</h4>
<pre>org.eclipse.swt.widgets.Label lblGenome</pre>
</li>
<li class="blockList">
<h4>lblID</h4>
<pre>org.eclipse.swt.widgets.Label lblID</pre>
</li>
<li class="blockList">
<h4>lblName</h4>
<pre>org.eclipse.swt.widgets.Label lblName</pre>
</li>
<li class="blockList">
<h4>lblRef</h4>
<pre>org.eclipse.swt.widgets.Label lblRef</pre>
</li>
<li class="blockList">
<h4>lblSelectYourList</h4>
<pre>org.eclipse.swt.widgets.Label lblSelectYourList</pre>
</li>
<li class="blockList">
<h4>lblSize</h4>
<pre>org.eclipse.swt.widgets.Label lblSize</pre>
</li>
<li class="blockList">
<h4>listGenes</h4>
<pre>org.eclipse.swt.widgets.List listGenes</pre>
</li>
<li class="blockList">
<h4>signature</h4>
<pre><a href="bacnet/datamodel/annotation/Signature.html" title="class in bacnet.datamodel.annotation">Signature</a> signature</pre>
</li>
<li class="blockList">
<h4>signatures</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; signatures</pre>
</li>
<li class="blockList">
<h4>tableSignature</h4>
<pre>org.eclipse.swt.widgets.Table tableSignature</pre>
</li>
<li class="blockListLast">
<h4>txtDescription</h4>
<pre>org.eclipse.swt.widgets.Text txtDescription</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.genomeBrowser</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.genomeBrowser.CompositeNTermInfo">
<!--   -->
</a>
<h3>Class <a href="bacnet/genomeBrowser/CompositeNTermInfo.html" title="class in bacnet.genomeBrowser">bacnet.genomeBrowser.CompositeNTermInfo</a> extends org.eclipse.swt.widgets.Composite implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-9089207362586510739L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>lblMaxquant</h4>
<pre>org.eclipse.swt.widgets.Label lblMaxquant</pre>
</li>
<li class="blockList">
<h4>lblModifv</h4>
<pre>org.eclipse.swt.widgets.Label lblModifv</pre>
</li>
<li class="blockList">
<h4>lblOverlap</h4>
<pre>org.eclipse.swt.widgets.Label lblOverlap</pre>
</li>
<li class="blockList">
<h4>lblPosition</h4>
<pre>org.eclipse.swt.widgets.Label lblPosition</pre>
</li>
<li class="blockList">
<h4>lblSize</h4>
<pre>org.eclipse.swt.widgets.Label lblSize</pre>
</li>
<li class="blockList">
<h4>lblSpectrav</h4>
<pre>org.eclipse.swt.widgets.Label lblSpectrav</pre>
</li>
<li class="blockList">
<h4>lblType</h4>
<pre>org.eclipse.swt.widgets.Label lblType</pre>
</li>
<li class="blockListLast">
<h4>txtSequence</h4>
<pre>org.eclipse.swt.widgets.Text txtSequence</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.genomeBrowser.CRISPRGuideView">
<!--   -->
</a>
<h3>Class <a href="bacnet/genomeBrowser/CRISPRGuideView.html" title="class in bacnet.genomeBrowser">bacnet.genomeBrowser.CRISPRGuideView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-4058433027371889806L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>bioConds</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioConds</pre>
</li>
<li class="blockList">
<h4>bioCondsArray</h4>
<pre>java.lang.String[][] bioCondsArray</pre>
<div class="block">Array used to init data</div>
</li>
<li class="blockList">
<h4>bioCondsToDisplay</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioCondsToDisplay</pre>
</li>
<li class="blockList">
<h4>btnHide</h4>
<pre>org.eclipse.swt.widgets.Button btnHide</pre>
</li>
<li class="blockList">
<h4>btnSaveTxt</h4>
<pre>org.eclipse.swt.widgets.Button btnSaveTxt</pre>
</li>
<li class="blockList">
<h4>columnNames</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; columnNames</pre>
</li>
<li class="blockList">
<h4>comparatorBioCondition</h4>
<pre><a href="bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a> comparatorBioCondition</pre>
</li>
<li class="blockList">
<h4>compositeSummary</h4>
<pre>org.eclipse.swt.widgets.Composite compositeSummary</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
</li>
<li class="blockList">
<h4>lblXxSrnas</h4>
<pre>org.eclipse.swt.widgets.Label lblXxSrnas</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>selectedNTerms</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; selectedNTerms</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>tableBioCondition</h4>
<pre>org.eclipse.jface.viewers.TableViewer tableBioCondition</pre>
</li>
<li class="blockListLast">
<h4>txtSearch</h4>
<pre>org.eclipse.swt.widgets.Text txtSearch</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.genomeBrowser.NTerminomicsView">
<!--   -->
</a>
<h3>Class <a href="bacnet/genomeBrowser/NTerminomicsView.html" title="class in bacnet.genomeBrowser">bacnet.genomeBrowser.NTerminomicsView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-4058433027371889806L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>bioConds</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioConds</pre>
</li>
<li class="blockList">
<h4>bioCondsArray</h4>
<pre>java.lang.String[][] bioCondsArray</pre>
<div class="block">Array used to init data</div>
</li>
<li class="blockList">
<h4>bioCondsToDisplay</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioCondsToDisplay</pre>
</li>
<li class="blockList">
<h4>btnHide</h4>
<pre>org.eclipse.swt.widgets.Button btnHide</pre>
</li>
<li class="blockList">
<h4>btnSaveTxt</h4>
<pre>org.eclipse.swt.widgets.Button btnSaveTxt</pre>
</li>
<li class="blockList">
<h4>columnNames</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; columnNames</pre>
</li>
<li class="blockList">
<h4>comparatorBioCondition</h4>
<pre><a href="bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a> comparatorBioCondition</pre>
</li>
<li class="blockList">
<h4>compositeSummary</h4>
<pre>org.eclipse.swt.widgets.Composite compositeSummary</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
</li>
<li class="blockList">
<h4>lblXxSrnas</h4>
<pre>org.eclipse.swt.widgets.Label lblXxSrnas</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>selectedNTerms</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; selectedNTerms</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>tableBioCondition</h4>
<pre>org.eclipse.jface.viewers.TableViewer tableBioCondition</pre>
</li>
<li class="blockListLast">
<h4>txtSearch</h4>
<pre>org.eclipse.swt.widgets.Text txtSearch</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.genomeBrowser.core</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.genomeBrowser.core.Region">
<!--   -->
</a>
<h3>Class <a href="bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">bacnet.genomeBrowser.core.Region</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-1103556086987455111L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>height</h4>
<pre>int height</pre>
</li>
<li class="blockList">
<h4>middleH</h4>
<pre>int middleH</pre>
</li>
<li class="blockList">
<h4>middleV</h4>
<pre>int middleV</pre>
</li>
<li class="blockList">
<h4>regionChanged</h4>
<pre>boolean regionChanged</pre>
</li>
<li class="blockList">
<h4>width</h4>
<pre>int width</pre>
</li>
<li class="blockList">
<h4>x1</h4>
<pre>int x1</pre>
</li>
<li class="blockList">
<h4>x2</h4>
<pre>int x2</pre>
</li>
<li class="blockList">
<h4>y1</h4>
<pre>int y1</pre>
</li>
<li class="blockListLast">
<h4>y2</h4>
<pre>int y2</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.genomeBrowser.core.Track">
<!--   -->
</a>
<h3>Class <a href="bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">bacnet.genomeBrowser.core.Track</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>4061565768191225185L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>chromosomeID</h4>
<pre>java.lang.String chromosomeID</pre>
</li>
<li class="blockList">
<h4>dataRegion</h4>
<pre><a href="bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a> dataRegion</pre>
</li>
<li class="blockList">
<h4>datas</h4>
<pre><a href="bacnet/genomeBrowser/core/DataTrack.html" title="class in bacnet.genomeBrowser.core">DataTrack</a> datas</pre>
</li>
<li class="blockList">
<h4>displayRegion</h4>
<pre><a href="bacnet/genomeBrowser/core/Region.html" title="class in bacnet.genomeBrowser.core">Region</a> displayRegion</pre>
</li>
<li class="blockList">
<h4>displaySequence</h4>
<pre>boolean displaySequence</pre>
</li>
<li class="blockList">
<h4>displayType</h4>
<pre><a href="bacnet/genomeBrowser/core/Track.DisplayType.html" title="enum in bacnet.genomeBrowser.core">Track.DisplayType</a> displayType</pre>
</li>
<li class="blockList">
<h4>genomeName</h4>
<pre>java.lang.String genomeName</pre>
</li>
<li class="blockList">
<h4>trackChangedFlag</h4>
<pre>boolean trackChangedFlag</pre>
</li>
<li class="blockListLast">
<h4>zoom</h4>
<pre><a href="bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">Zoom</a> zoom</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.genomeBrowser.core.Zoom">
<!--   -->
</a>
<h3>Class <a href="bacnet/genomeBrowser/core/Zoom.html" title="class in bacnet.genomeBrowser.core">bacnet.genomeBrowser.core.Zoom</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>5926414787821636159L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>totalBP</h4>
<pre>int totalBP</pre>
</li>
<li class="blockList">
<h4>zoomChangedFlag</h4>
<pre>boolean zoomChangedFlag</pre>
</li>
<li class="blockList">
<h4>zoomHeightMax</h4>
<pre>int zoomHeightMax</pre>
<div class="block">Maximal value for the zoom</div>
</li>
<li class="blockList">
<h4>zoomHeightPosition</h4>
<pre>int zoomHeightPosition</pre>
<div class="block">Position of the zoom for height</div>
</li>
<li class="blockList">
<h4>zoomNumber</h4>
<pre>int zoomNumber</pre>
</li>
<li class="blockList">
<h4>zoomPosition</h4>
<pre>int zoomPosition</pre>
</li>
<li class="blockListLast">
<h4>zoomRatio</h4>
<pre>double zoomRatio</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.genomeBrowser.dialog</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.genomeBrowser.dialog.AddProteomicsDataDialog">
<!--   -->
</a>
<h3>Class <a href="bacnet/genomeBrowser/dialog/AddProteomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">bacnet.genomeBrowser.dialog.AddProteomicsDataDialog</a> extends org.eclipse.jface.dialogs.TitleAreaDialog implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-8202496409867913492L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>bioConds</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioConds</pre>
<div class="block">List used to init data</div>
</li>
<li class="blockList">
<h4>bioCondsArray</h4>
<pre>java.lang.String[][] bioCondsArray</pre>
<div class="block">Array used to init data</div>
</li>
<li class="blockList">
<h4>bioCondsToDisplay</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioCondsToDisplay</pre>
<div class="block">Array which will be displayed</div>
</li>
<li class="blockList">
<h4>btnBhiBroth</h4>
<pre>org.eclipse.swt.widgets.Button btnBhiBroth</pre>
</li>
<li class="blockList">
<h4>btnCellWall</h4>
<pre>org.eclipse.swt.widgets.Button btnCellWall</pre>
</li>
<li class="blockList">
<h4>btnCytoplasm</h4>
<pre>org.eclipse.swt.widgets.Button btnCytoplasm</pre>
</li>
<li class="blockList">
<h4>btnExponential</h4>
<pre>org.eclipse.swt.widgets.Button btnExponential</pre>
</li>
<li class="blockList">
<h4>btnHumanBloodCells</h4>
<pre>org.eclipse.swt.widgets.Button btnHumanBloodCells</pre>
</li>
<li class="blockList">
<h4>btnLbBroth</h4>
<pre>org.eclipse.swt.widgets.Button btnLbBroth</pre>
</li>
<li class="blockList">
<h4>btnMembrane</h4>
<pre>org.eclipse.swt.widgets.Button btnMembrane</pre>
</li>
<li class="blockList">
<h4>btnMinimalMediaBroth</h4>
<pre>org.eclipse.swt.widgets.Button btnMinimalMediaBroth</pre>
</li>
<li class="blockList">
<h4>btnMouseMacrophagesCells</h4>
<pre>org.eclipse.swt.widgets.Button btnMouseMacrophagesCells</pre>
</li>
<li class="blockList">
<h4>btnSecretome</h4>
<pre>org.eclipse.swt.widgets.Button btnSecretome</pre>
</li>
<li class="blockList">
<h4>btnStationnary</h4>
<pre>org.eclipse.swt.widgets.Button btnStationnary</pre>
</li>
<li class="blockList">
<h4>btnUnpublished</h4>
<pre>org.eclipse.swt.widgets.Button btnUnpublished</pre>
</li>
<li class="blockList">
<h4>columnNames</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; columnNames</pre>
</li>
<li class="blockList">
<h4>comboGenome</h4>
<pre>org.eclipse.swt.widgets.Combo comboGenome</pre>
</li>
<li class="blockList">
<h4>comparatorBioCondition</h4>
<pre><a href="bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a> comparatorBioCondition</pre>
</li>
<li class="blockList">
<h4>imageGeneExpr</h4>
<pre>org.eclipse.swt.graphics.Image imageGeneExpr</pre>
</li>
<li class="blockList">
<h4>imageRNASeq</h4>
<pre>org.eclipse.swt.graphics.Image imageRNASeq</pre>
</li>
<li class="blockList">
<h4>imageTilingGeneExpr</h4>
<pre>org.eclipse.swt.graphics.Image imageTilingGeneExpr</pre>
</li>
<li class="blockList">
<h4>imageTSS</h4>
<pre>org.eclipse.swt.graphics.Image imageTSS</pre>
</li>
<li class="blockList">
<h4>imageTSSTilingGeneExpr</h4>
<pre>org.eclipse.swt.graphics.Image imageTSSTilingGeneExpr</pre>
</li>
<li class="blockList">
<h4>lblTssAvailable</h4>
<pre>org.eclipse.swt.widgets.Label lblTssAvailable</pre>
</li>
<li class="blockList">
<h4>lblTSSImage</h4>
<pre>org.eclipse.swt.widgets.Label lblTSSImage</pre>
</li>
<li class="blockList">
<h4>tableBioCondition</h4>
<pre>org.eclipse.jface.viewers.TableViewer tableBioCondition</pre>
</li>
<li class="blockList">
<h4>textSearch</h4>
<pre>org.eclipse.swt.widgets.Text textSearch</pre>
</li>
<li class="blockListLast">
<h4>track</h4>
<pre><a href="bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a> track</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.genomeBrowser.dialog.AddTranscriptomicsDataDialog">
<!--   -->
</a>
<h3>Class <a href="bacnet/genomeBrowser/dialog/AddTranscriptomicsDataDialog.html" title="class in bacnet.genomeBrowser.dialog">bacnet.genomeBrowser.dialog.AddTranscriptomicsDataDialog</a> extends org.eclipse.jface.dialogs.TitleAreaDialog implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-2962428673771724278L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>bioConds</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioConds</pre>
<div class="block">List used to init data</div>
</li>
<li class="blockList">
<h4>bioCondsArray</h4>
<pre>java.lang.String[][] bioCondsArray</pre>
<div class="block">Array used to init data</div>
</li>
<li class="blockList">
<h4>bioCondsToDisplay</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioCondsToDisplay</pre>
<div class="block">Array which will be displayed</div>
</li>
<li class="blockList">
<h4>btnAllMutant</h4>
<pre>org.eclipse.swt.widgets.Button btnAllMutant</pre>
</li>
<li class="blockList">
<h4>btnBhiBroth</h4>
<pre>org.eclipse.swt.widgets.Button btnBhiBroth</pre>
</li>
<li class="blockList">
<h4>btnCacoCells</h4>
<pre>org.eclipse.swt.widgets.Button btnCacoCells</pre>
</li>
<li class="blockList">
<h4>btnChooseOneMutant</h4>
<pre>org.eclipse.swt.widgets.Button btnChooseOneMutant</pre>
</li>
<li class="blockList">
<h4>btnDeath</h4>
<pre>org.eclipse.swt.widgets.Button btnDeath</pre>
</li>
<li class="blockList">
<h4>btnExponential</h4>
<pre>org.eclipse.swt.widgets.Button btnExponential</pre>
</li>
<li class="blockList">
<h4>btnGeneExpression</h4>
<pre>org.eclipse.swt.widgets.Button btnGeneExpression</pre>
</li>
<li class="blockList">
<h4>btnHumanBloodCells</h4>
<pre>org.eclipse.swt.widgets.Button btnHumanBloodCells</pre>
</li>
<li class="blockList">
<h4>btnLagPhase</h4>
<pre>org.eclipse.swt.widgets.Button btnLagPhase</pre>
</li>
<li class="blockList">
<h4>btnLbBroth</h4>
<pre>org.eclipse.swt.widgets.Button btnLbBroth</pre>
</li>
<li class="blockList">
<h4>btnMinimalMediaBroth</h4>
<pre>org.eclipse.swt.widgets.Button btnMinimalMediaBroth</pre>
</li>
<li class="blockList">
<h4>btnMouseIntestineCells</h4>
<pre>org.eclipse.swt.widgets.Button btnMouseIntestineCells</pre>
</li>
<li class="blockList">
<h4>btnMouseMacrophagesCells</h4>
<pre>org.eclipse.swt.widgets.Button btnMouseMacrophagesCells</pre>
</li>
<li class="blockList">
<h4>btnMouseSpleenCells</h4>
<pre>org.eclipse.swt.widgets.Button btnMouseSpleenCells</pre>
</li>
<li class="blockList">
<h4>btnNoneMutant</h4>
<pre>org.eclipse.swt.widgets.Button btnNoneMutant</pre>
</li>
<li class="blockList">
<h4>btnPorcineBileCells</h4>
<pre>org.eclipse.swt.widgets.Button btnPorcineBileCells</pre>
</li>
<li class="blockList">
<h4>btnRegrowth</h4>
<pre>org.eclipse.swt.widgets.Button btnRegrowth</pre>
</li>
<li class="blockList">
<h4>btnRiboSeq</h4>
<pre>org.eclipse.swt.widgets.Button btnRiboSeq</pre>
</li>
<li class="blockList">
<h4>btnRnaseq</h4>
<pre>org.eclipse.swt.widgets.Button btnRnaseq</pre>
</li>
<li class="blockList">
<h4>btnStationnary</h4>
<pre>org.eclipse.swt.widgets.Button btnStationnary</pre>
</li>
<li class="blockList">
<h4>btnSurvival</h4>
<pre>org.eclipse.swt.widgets.Button btnSurvival</pre>
</li>
<li class="blockList">
<h4>btnTermSeq</h4>
<pre>org.eclipse.swt.widgets.Button btnTermSeq</pre>
</li>
<li class="blockList">
<h4>btnTiling</h4>
<pre>org.eclipse.swt.widgets.Button btnTiling</pre>
</li>
<li class="blockList">
<h4>btnTrypticSoyBroth</h4>
<pre>org.eclipse.swt.widgets.Button btnTrypticSoyBroth</pre>
</li>
<li class="blockList">
<h4>btnTss</h4>
<pre>org.eclipse.swt.widgets.Button btnTss</pre>
</li>
<li class="blockList">
<h4>btnTurkeyDeliMeat</h4>
<pre>org.eclipse.swt.widgets.Button btnTurkeyDeliMeat</pre>
</li>
<li class="blockList">
<h4>btnUnpublished</h4>
<pre>org.eclipse.swt.widgets.Button btnUnpublished</pre>
</li>
<li class="blockList">
<h4>columnNames</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; columnNames</pre>
</li>
<li class="blockList">
<h4>comboGenome</h4>
<pre>org.eclipse.swt.widgets.Combo comboGenome</pre>
</li>
<li class="blockList">
<h4>comboMutant</h4>
<pre>org.eclipse.swt.widgets.Combo comboMutant</pre>
</li>
<li class="blockList">
<h4>comparatorBioCondition</h4>
<pre><a href="bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a> comparatorBioCondition</pre>
</li>
<li class="blockList">
<h4>imageGeneExpr</h4>
<pre>org.eclipse.swt.graphics.Image imageGeneExpr</pre>
</li>
<li class="blockList">
<h4>imageRNASeq</h4>
<pre>org.eclipse.swt.graphics.Image imageRNASeq</pre>
</li>
<li class="blockList">
<h4>imageTilingGeneExpr</h4>
<pre>org.eclipse.swt.graphics.Image imageTilingGeneExpr</pre>
</li>
<li class="blockList">
<h4>imageTSS</h4>
<pre>org.eclipse.swt.graphics.Image imageTSS</pre>
</li>
<li class="blockList">
<h4>imageTSSTilingGeneExpr</h4>
<pre>org.eclipse.swt.graphics.Image imageTSSTilingGeneExpr</pre>
</li>
<li class="blockList">
<h4>tableBioCondition</h4>
<pre>org.eclipse.jface.viewers.TableViewer tableBioCondition</pre>
</li>
<li class="blockList">
<h4>textSearch</h4>
<pre>org.eclipse.swt.widgets.Text textSearch</pre>
</li>
<li class="blockListLast">
<h4>track</h4>
<pre><a href="bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a> track</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.genomeBrowser.dialog.LegendDialog">
<!--   -->
</a>
<h3>Class <a href="bacnet/genomeBrowser/dialog/LegendDialog.html" title="class in bacnet.genomeBrowser.dialog">bacnet.genomeBrowser.dialog.LegendDialog</a> extends org.eclipse.jface.dialogs.TitleAreaDialog implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>3331773901037609884L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnRemoveSelectedBiological</h4>
<pre>org.eclipse.swt.widgets.Button btnRemoveSelectedBiological</pre>
</li>
<li class="blockList">
<h4>imageChecked</h4>
<pre>org.eclipse.swt.graphics.Image imageChecked</pre>
</li>
<li class="blockList">
<h4>imageGeneExpr</h4>
<pre>org.eclipse.swt.graphics.Image imageGeneExpr</pre>
</li>
<li class="blockList">
<h4>imageRNASeq</h4>
<pre>org.eclipse.swt.graphics.Image imageRNASeq</pre>
</li>
<li class="blockList">
<h4>imageTilingGeneExpr</h4>
<pre>org.eclipse.swt.graphics.Image imageTilingGeneExpr</pre>
</li>
<li class="blockList">
<h4>imageTSS</h4>
<pre>org.eclipse.swt.graphics.Image imageTSS</pre>
</li>
<li class="blockList">
<h4>imageTSSTilingGeneExpr</h4>
<pre>org.eclipse.swt.graphics.Image imageTSSTilingGeneExpr</pre>
</li>
<li class="blockList">
<h4>imageUnchecked</h4>
<pre>org.eclipse.swt.graphics.Image imageUnchecked</pre>
</li>
<li class="blockList">
<h4>scrolledComposite</h4>
<pre>org.eclipse.swt.custom.ScrolledComposite scrolledComposite</pre>
</li>
<li class="blockList">
<h4>tableBioCondition</h4>
<pre>org.eclipse.swt.widgets.Table tableBioCondition</pre>
</li>
<li class="blockList">
<h4>tableData</h4>
<pre>org.eclipse.swt.widgets.Table tableData</pre>
</li>
<li class="blockListLast">
<h4>track</h4>
<pre><a href="bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a> track</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.genomeBrowser.tracksGUI</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.genomeBrowser.tracksGUI.TrackCanvasData">
<!--   -->
</a>
<h3>Class <a href="bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">bacnet.genomeBrowser.tracksGUI.TrackCanvasData</a> extends org.eclipse.swt.widgets.Canvas implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>1080025663659839957L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>annotation</h4>
<pre><a href="bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a> annotation</pre>
<div class="block"><code>Annotation</code> data for all Genome information</div>
</li>
<li class="blockList">
<h4>bpSizeH</h4>
<pre>double bpSizeH</pre>
<div class="block">Width in pixel of one base pair (fundamental unit for the rest of the
 display)</div>
</li>
<li class="blockList">
<h4>chromoID</h4>
<pre>java.lang.String chromoID</pre>
</li>
<li class="blockList">
<h4>dataSizeReference</h4>
<pre>int dataSizeReference</pre>
<div class="block">Height of on data</div>
</li>
<li class="blockList">
<h4>dataWindowSize</h4>
<pre>int dataWindowSize</pre>
<div class="block">Height of all the data window = all data combine</div>
</li>
<li class="blockList">
<h4>decaySliderVBar</h4>
<pre>int decaySliderVBar</pre>
<div class="block">Vertical decay imposed by the vertical slider</div>
</li>
<li class="blockList">
<h4>displayMouseLine</h4>
<pre>boolean displayMouseLine</pre>
<div class="block">If true we display a vertical line at the pointer of the mouse</div>
</li>
<li class="blockList">
<h4>genomeName</h4>
<pre>java.lang.String genomeName</pre>
</li>
<li class="blockList">
<h4>heightPix</h4>
<pre>int heightPix</pre>
<div class="block">Height in pixels of the drawing area</div>
</li>
<li class="blockList">
<h4>massSpecData</h4>
<pre><a href="bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a> massSpecData</pre>
<div class="block">MassSpecData where the selected NTerm is</div>
</li>
<li class="blockList">
<h4>MINIMUM_DATA_SIZE</h4>
<pre>int MINIMUM_DATA_SIZE</pre>
<div class="block">Minimal height of a data track</div>
</li>
<li class="blockList">
<h4>mouseXPosition</h4>
<pre>int mouseXPosition</pre>
<div class="block">When mouse is clicked we displayed a line is displayed to show synteny:
 mouseYPosition gives the X position</div>
</li>
<li class="blockList">
<h4>mouseYPosition</h4>
<pre>int mouseYPosition</pre>
<div class="block">When mouse is clicked we displayed a line is displayed to show synteny:
 mouseYPosition gives the Y position</div>
</li>
<li class="blockList">
<h4>Name</h4>
<pre>java.lang.String Name</pre>
</li>
<li class="blockList">
<h4>nTermHighlight</h4>
<pre><a href="bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a> nTermHighlight</pre>
<div class="block">NTerm which will be highlighted</div>
</li>
<li class="blockList">
<h4>nTermTypeOverlap</h4>
<pre>java.lang.String nTermTypeOverlap</pre>
<div class="block">Type of NTerm to display, if = All display all types</div>
</li>
<li class="blockList">
<h4>saveData</h4>
<pre>java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.String[][]&gt; saveData</pre>
</li>
<li class="blockList">
<h4>sequenceHeight</h4>
<pre>int sequenceHeight</pre>
<div class="block">Height of the sequence track</div>
</li>
<li class="blockList">
<h4>testData</h4>
<pre>boolean testData</pre>
</li>
<li class="blockList">
<h4>track</h4>
<pre><a href="bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a> track</pre>
<div class="block">All information about the data to display</div>
</li>
<li class="blockList">
<h4>trackComposite</h4>
<pre>org.eclipse.swt.widgets.Composite trackComposite</pre>
</li>
<li class="blockList">
<h4>widthBP</h4>
<pre>int widthBP</pre>
<div class="block">Width in Base pairs of the drawing area</div>
</li>
<li class="blockListLast">
<h4>zoomVertical</h4>
<pre>double zoomVertical</pre>
<div class="block">Vertical zoom</div>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.genomeBrowser.tracksGUI.TrackCanvasGenome">
<!--   -->
</a>
<h3>Class <a href="bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">bacnet.genomeBrowser.tracksGUI.TrackCanvasGenome</a> extends org.eclipse.swt.widgets.Canvas implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>1080025663659839957L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>annotation</h4>
<pre><a href="bacnet/datamodel/annotation/Annotation.html" title="class in bacnet.datamodel.annotation">Annotation</a> annotation</pre>
<div class="block"><code>Annotation</code> data for all Genome information</div>
</li>
<li class="blockList">
<h4>annotElementHeight</h4>
<pre>int annotElementHeight</pre>
<div class="block">Height of one annnotation element</div>
</li>
<li class="blockList">
<h4>annotNumber</h4>
<pre>float annotNumber</pre>
</li>
<li class="blockList">
<h4>annotSeparationHeight</h4>
<pre>int annotSeparationHeight</pre>
<div class="block">Height of the separation between two annotation element</div>
</li>
<li class="blockList">
<h4>annotSeparationNumber</h4>
<pre>int annotSeparationNumber</pre>
</li>
<li class="blockList">
<h4>bpSizeH</h4>
<pre>double bpSizeH</pre>
<div class="block">Width in pixel of one base pair (fundamental unit for the rest of the
 display)</div>
</li>
<li class="blockList">
<h4>displayMouseLine</h4>
<pre>boolean displayMouseLine</pre>
<div class="block">If true we display a vertical line at the pointer of the mouse</div>
</li>
<li class="blockList">
<h4>heightPix</h4>
<pre>int heightPix</pre>
</li>
<li class="blockList">
<h4>mouseXPosition</h4>
<pre>int mouseXPosition</pre>
<div class="block">When mouse is clicked we displayed a line is displayed to show synteny:
 mouseYPosition gives the X position</div>
</li>
<li class="blockList">
<h4>mouseYPosition</h4>
<pre>int mouseYPosition</pre>
<div class="block">When mouse is clicked we displayed a line is displayed to show synteny:
 mouseYPosition gives the Y position</div>
</li>
<li class="blockList">
<h4>track</h4>
<pre><a href="bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a> track</pre>
<div class="block">data to display</div>
</li>
<li class="blockListLast">
<h4>widthBP</h4>
<pre>int widthBP</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.genomeBrowser.tracksGUI.TracksComposite">
<!--   -->
</a>
<h3>Class <a href="bacnet/genomeBrowser/tracksGUI/TracksComposite.html" title="class in bacnet.genomeBrowser.tracksGUI">bacnet.genomeBrowser.tracksGUI.TracksComposite</a> extends org.eclipse.swt.widgets.Composite implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-1957427699848660677L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnAddProteomicsData</h4>
<pre>org.eclipse.swt.widgets.Button btnAddProteomicsData</pre>
</li>
<li class="blockList">
<h4>btnAddTranscriptomicsData</h4>
<pre>org.eclipse.swt.widgets.Button btnAddTranscriptomicsData</pre>
</li>
<li class="blockList">
<h4>btnAnimate</h4>
<pre>org.eclipse.swt.widgets.Button btnAnimate</pre>
</li>
<li class="blockList">
<h4>btnHelp</h4>
<pre>org.eclipse.swt.widgets.Button btnHelp</pre>
</li>
<li class="blockList">
<h4>btnHideNterminomicsPanel</h4>
<pre>org.eclipse.swt.widgets.Button btnHideNterminomicsPanel</pre>
</li>
<li class="blockList">
<h4>btnLegendManagement</h4>
<pre>org.eclipse.swt.widgets.Button btnLegendManagement</pre>
</li>
<li class="blockList">
<h4>btnLoadData</h4>
<pre>org.eclipse.swt.widgets.Button btnLoadData</pre>
</li>
<li class="blockList">
<h4>btnNextnterm</h4>
<pre>org.eclipse.swt.widgets.Button btnNextnterm</pre>
</li>
<li class="blockList">
<h4>btnPreviousnterm</h4>
<pre>org.eclipse.swt.widgets.Button btnPreviousnterm</pre>
</li>
<li class="blockList">
<h4>btnSaveDataSelection</h4>
<pre>org.eclipse.swt.widgets.Button btnSaveDataSelection</pre>
</li>
<li class="blockList">
<h4>btnSequence</h4>
<pre>org.eclipse.swt.widgets.Button btnSequence</pre>
</li>
<li class="blockList">
<h4>btnZoomIn</h4>
<pre>org.eclipse.swt.widgets.Button btnZoomIn</pre>
</li>
<li class="blockList">
<h4>btnZoomInVertical</h4>
<pre>org.eclipse.swt.widgets.Button btnZoomInVertical</pre>
</li>
<li class="blockList">
<h4>btnZoomOut</h4>
<pre>org.eclipse.swt.widgets.Button btnZoomOut</pre>
</li>
<li class="blockList">
<h4>btnZoomOutVertical</h4>
<pre>org.eclipse.swt.widgets.Button btnZoomOutVertical</pre>
</li>
<li class="blockList">
<h4>canvasData</h4>
<pre><a href="bacnet/genomeBrowser/tracksGUI/TrackCanvasData.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasData</a> canvasData</pre>
</li>
<li class="blockList">
<h4>canvasGenome</h4>
<pre><a href="bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a> canvasGenome</pre>
</li>
<li class="blockList">
<h4>cmpNTermInfo</h4>
<pre>org.eclipse.swt.widgets.Composite cmpNTermInfo</pre>
</li>
<li class="blockList">
<h4>comboAbsoluteRelative</h4>
<pre>org.eclipse.swt.widgets.Combo comboAbsoluteRelative</pre>
</li>
<li class="blockList">
<h4>comboChromosome</h4>
<pre>org.eclipse.swt.widgets.Combo comboChromosome</pre>
</li>
<li class="blockList">
<h4>comboNTermData</h4>
<pre>org.eclipse.swt.widgets.Combo comboNTermData</pre>
</li>
<li class="blockList">
<h4>comboNTermFilter</h4>
<pre>org.eclipse.swt.widgets.Combo comboNTermFilter</pre>
</li>
<li class="blockList">
<h4>comboTypeDisplay</h4>
<pre>org.eclipse.swt.widgets.Combo comboTypeDisplay</pre>
</li>
<li class="blockList">
<h4>compositeAddData</h4>
<pre>org.eclipse.swt.widgets.Composite compositeAddData</pre>
</li>
<li class="blockList">
<h4>compositeChromosome</h4>
<pre>org.eclipse.swt.widgets.Composite compositeChromosome</pre>
</li>
<li class="blockList">
<h4>compositeGenome_1</h4>
<pre>org.eclipse.swt.widgets.Composite compositeGenome_1</pre>
</li>
<li class="blockList">
<h4>compositeNterm</h4>
<pre><a href="bacnet/genomeBrowser/CompositeNTermInfo.html" title="class in bacnet.genomeBrowser">CompositeNTermInfo</a> compositeNterm</pre>
</li>
<li class="blockList">
<h4>compositeTypeDisplay</h4>
<pre>org.eclipse.swt.widgets.Composite compositeTypeDisplay</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>gridDataCmpInfo</h4>
<pre>org.eclipse.swt.layout.GridData gridDataCmpInfo</pre>
</li>
<li class="blockList">
<h4>hideInfo</h4>
<pre>org.eclipse.swt.layout.GridLayout hideInfo</pre>
</li>
<li class="blockList">
<h4>lblTutorial</h4>
<pre>org.eclipse.swt.widgets.Label lblTutorial</pre>
</li>
<li class="blockList">
<h4>massSpecData</h4>
<pre><a href="bacnet/datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a> massSpecData</pre>
<div class="block">MassSpecData where the selected NTerm is</div>
</li>
<li class="blockList">
<h4>nTerm</h4>
<pre><a href="bacnet/datamodel/proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a> nTerm</pre>
<div class="block">Selected NTerm</div>
</li>
<li class="blockList">
<h4>parentViewId</h4>
<pre>java.lang.String parentViewId</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>showInfo</h4>
<pre>org.eclipse.swt.layout.GridLayout showInfo</pre>
</li>
<li class="blockList">
<h4>sliderHbar</h4>
<pre>org.eclipse.swt.widgets.Slider sliderHbar</pre>
</li>
<li class="blockList">
<h4>sliderVBar</h4>
<pre>org.eclipse.swt.widgets.Slider sliderVBar</pre>
</li>
<li class="blockList">
<h4>tableNtermInfo</h4>
<pre>org.eclipse.swt.widgets.Table tableNtermInfo</pre>
</li>
<li class="blockList">
<h4>track</h4>
<pre><a href="bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a> track</pre>
</li>
<li class="blockListLast">
<h4>transcriptome</h4>
<pre>boolean transcriptome</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.scripts.blast</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.scripts.blast.BlastResultView">
<!--   -->
</a>
<h3>Class <a href="bacnet/scripts/blast/BlastResultView.html" title="class in bacnet.scripts.blast">bacnet.scripts.blast.BlastResultView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>3579398886352775140L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>archiveFile</h4>
<pre>java.lang.String archiveFile</pre>
</li>
<li class="blockList">
<h4>browser</h4>
<pre>org.eclipse.swt.browser.Browser browser</pre>
</li>
<li class="blockList">
<h4>btnExport</h4>
<pre>org.eclipse.swt.widgets.Button btnExport</pre>
</li>
<li class="blockList">
<h4>btnLoad</h4>
<pre>org.eclipse.swt.widgets.Button btnLoad</pre>
</li>
<li class="blockList">
<h4>btnRefreshDisplay</h4>
<pre>org.eclipse.swt.widgets.Button btnRefreshDisplay</pre>
</li>
<li class="blockList">
<h4>btnSave</h4>
<pre>org.eclipse.swt.widgets.Button btnSave</pre>
</li>
<li class="blockList">
<h4>combo</h4>
<pre>org.eclipse.swt.widgets.Combo combo</pre>
</li>
<li class="blockList">
<h4>lblSave</h4>
<pre>org.eclipse.swt.widgets.Label lblSave</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockListLast">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.scripts.blast.gui</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.scripts.blast.gui.BlastGenomeSelectionPage">
<!--   -->
</a>
<h3>Class <a href="bacnet/scripts/blast/gui/BlastGenomeSelectionPage.html" title="class in bacnet.scripts.blast.gui">bacnet.scripts.blast.gui.BlastGenomeSelectionPage</a> extends org.eclipse.jface.wizard.WizardPage implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-5405627500164523922L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnBac</h4>
<pre>org.eclipse.swt.widgets.Button btnBac</pre>
</li>
<li class="blockList">
<h4>btnBac2</h4>
<pre>org.eclipse.swt.widgets.Button btnBac2</pre>
</li>
<li class="blockList">
<h4>btnMono</h4>
<pre>org.eclipse.swt.widgets.Button btnMono</pre>
</li>
<li class="blockList">
<h4>btnMono2</h4>
<pre>org.eclipse.swt.widgets.Button btnMono2</pre>
</li>
<li class="blockList">
<h4>btnNonMono</h4>
<pre>org.eclipse.swt.widgets.Button btnNonMono</pre>
</li>
<li class="blockList">
<h4>btnNonMono2</h4>
<pre>org.eclipse.swt.widgets.Button btnNonMono2</pre>
</li>
<li class="blockList">
<h4>btnOther</h4>
<pre>org.eclipse.swt.widgets.Button btnOther</pre>
</li>
<li class="blockList">
<h4>btnOther2</h4>
<pre>org.eclipse.swt.widgets.Button btnOther2</pre>
</li>
<li class="blockList">
<h4>composite</h4>
<pre>org.eclipse.swt.widgets.Composite composite</pre>
</li>
<li class="blockList">
<h4>composite_1</h4>
<pre>org.eclipse.swt.widgets.Composite composite_1</pre>
</li>
<li class="blockList">
<h4>label</h4>
<pre>org.eclipse.swt.widgets.Label label</pre>
</li>
<li class="blockList">
<h4>label_1</h4>
<pre>org.eclipse.swt.widgets.Label label_1</pre>
</li>
<li class="blockList">
<h4>listBac</h4>
<pre>org.eclipse.swt.widgets.List listBac</pre>
</li>
<li class="blockList">
<h4>listBac2</h4>
<pre>org.eclipse.swt.widgets.List listBac2</pre>
</li>
<li class="blockList">
<h4>listMono</h4>
<pre>org.eclipse.swt.widgets.List listMono</pre>
</li>
<li class="blockList">
<h4>listMono2</h4>
<pre>org.eclipse.swt.widgets.List listMono2</pre>
</li>
<li class="blockList">
<h4>listNonMono</h4>
<pre>org.eclipse.swt.widgets.List listNonMono</pre>
</li>
<li class="blockList">
<h4>listNonMono2</h4>
<pre>org.eclipse.swt.widgets.List listNonMono2</pre>
</li>
<li class="blockList">
<h4>listOther</h4>
<pre>org.eclipse.swt.widgets.List listOther</pre>
</li>
<li class="blockListLast">
<h4>listOther2</h4>
<pre>org.eclipse.swt.widgets.List listOther2</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.scripts.blast.gui.BlastParametersPage">
<!--   -->
</a>
<h3>Class <a href="bacnet/scripts/blast/gui/BlastParametersPage.html" title="class in bacnet.scripts.blast.gui">bacnet.scripts.blast.gui.BlastParametersPage</a> extends org.eclipse.jface.wizard.WizardPage implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-3655573289402306938L</dd>
</dl>
</li>
<li class="blockList"><a id="bacnet.scripts.blast.gui.BlastQueryPage">
<!--   -->
</a>
<h3>Class <a href="bacnet/scripts/blast/gui/BlastQueryPage.html" title="class in bacnet.scripts.blast.gui">bacnet.scripts.blast.gui.BlastQueryPage</a> extends org.eclipse.jface.wizard.WizardPage implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-5725807534323834458L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnLoad</h4>
<pre>org.eclipse.swt.widgets.Button btnLoad</pre>
</li>
<li class="blockList">
<h4>btnSave</h4>
<pre>org.eclipse.swt.widgets.Button btnSave</pre>
</li>
<li class="blockList">
<h4>cmbFileFormat</h4>
<pre>org.eclipse.swt.widgets.Combo cmbFileFormat</pre>
</li>
<li class="blockList">
<h4>lblLoadFile</h4>
<pre>org.eclipse.swt.widgets.Label lblLoadFile</pre>
</li>
<li class="blockList">
<h4>lblSave</h4>
<pre>org.eclipse.swt.widgets.Label lblSave</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockListLast">
<h4>textQuery</h4>
<pre>org.eclipse.swt.widgets.Text textQuery</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.scripts.genome</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.scripts.genome.CircularGenomeJPanel">
<!--   -->
</a>
<h3>Class <a href="bacnet/scripts/genome/CircularGenomeJPanel.html" title="class in bacnet.scripts.genome">bacnet.scripts.genome.CircularGenomeJPanel</a> extends javax.swing.JPanel implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>1L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>cgview</h4>
<pre>ca.ualberta.stothard.cgview.Cgview cgview</pre>
</li>
<li class="blockList">
<h4>genome</h4>
<pre><a href="bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> genome</pre>
</li>
<li class="blockList">
<h4>operons</h4>
<pre>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="bacnet/datamodel/sequence/Operon.html" title="class in bacnet.datamodel.sequence">Operon</a>&gt; operons</pre>
</li>
<li class="blockListLast">
<h4>sRNAs</h4>
<pre>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt; sRNAs</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.sequenceTools</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.sequenceTools.AnnotationView">
<!--   -->
</a>
<h3>Class <a href="bacnet/sequenceTools/AnnotationView.html" title="class in bacnet.sequenceTools">bacnet.sequenceTools.AnnotationView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-4058433027371889806L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>bioConds</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioConds</pre>
</li>
<li class="blockList">
<h4>bioCondsArray</h4>
<pre>java.lang.String[][] bioCondsArray</pre>
<div class="block">Array used to init data</div>
</li>
<li class="blockList">
<h4>bioCondsToDisplay</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioCondsToDisplay</pre>
</li>
<li class="blockList">
<h4>btnHide</h4>
<pre>org.eclipse.swt.widgets.Button btnHide</pre>
</li>
<li class="blockList">
<h4>btnSaveTxt</h4>
<pre>org.eclipse.swt.widgets.Button btnSaveTxt</pre>
</li>
<li class="blockList">
<h4>btnSelectall</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectall</pre>
</li>
<li class="blockList">
<h4>btnUnselectall</h4>
<pre>org.eclipse.swt.widgets.Button btnUnselectall</pre>
</li>
<li class="blockList">
<h4>columnNames</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; columnNames</pre>
</li>
<li class="blockList">
<h4>comparatorBioCondition</h4>
<pre><a href="bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a> comparatorBioCondition</pre>
</li>
<li class="blockList">
<h4>compositeSummary</h4>
<pre>org.eclipse.swt.widgets.Composite compositeSummary</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>genome</h4>
<pre><a href="bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> genome</pre>
</li>
<li class="blockList">
<h4>lblXxSrnas</h4>
<pre>org.eclipse.swt.widgets.Label lblXxSrnas</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>selectedGenes</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; selectedGenes</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>tableBioCondition</h4>
<pre>org.eclipse.jface.viewers.TableViewer tableBioCondition</pre>
</li>
<li class="blockList">
<h4>txtSearch</h4>
<pre>org.eclipse.swt.widgets.Text txtSearch</pre>
</li>
<li class="blockListLast">
<h4>viewID</h4>
<pre>java.lang.String viewID</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.sequenceTools.GeneView">
<!--   -->
</a>
<h3>Class <a href="bacnet/sequenceTools/GeneView.html" title="class in bacnet.sequenceTools">bacnet.sequenceTools.GeneView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-5632436594051134260L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>arrayDataList</h4>
<pre>java.lang.String[][] arrayDataList</pre>
</li>
<li class="blockList">
<h4>arrayGeneToLocalization</h4>
<pre>java.lang.String[][] arrayGeneToLocalization</pre>
</li>
<li class="blockList">
<h4>arrayProteomeList</h4>
<pre>java.lang.String[][] arrayProteomeList</pre>
</li>
<li class="blockList">
<h4>bioConds</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioConds</pre>
<div class="block">List used to init data</div>
</li>
<li class="blockList">
<h4>bioCondsArray</h4>
<pre>java.lang.String[][] bioCondsArray</pre>
<div class="block">Array used to init data</div>
</li>
<li class="blockList">
<h4>bioCondsToDisplay</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioCondsToDisplay</pre>
<div class="block">Array which will be displayed</div>
</li>
<li class="blockList">
<h4>browserHomolog</h4>
<pre>org.eclipse.swt.browser.Browser browserHomolog</pre>
</li>
<li class="blockList">
<h4>browserIsFocus</h4>
<pre>boolean browserIsFocus</pre>
</li>
<li class="blockList">
<h4>browserLocalization</h4>
<pre>org.eclipse.swt.browser.Browser browserLocalization</pre>
</li>
<li class="blockList">
<h4>browserSynteny</h4>
<pre>org.eclipse.swt.browser.Browser browserSynteny</pre>
</li>
<li class="blockList">
<h4>btnAminoAcidSequence</h4>
<pre>org.eclipse.swt.widgets.Button btnAminoAcidSequence</pre>
</li>
<li class="blockList">
<h4>btnDonwloadtxt</h4>
<pre>org.eclipse.swt.widgets.Button btnDonwloadtxt</pre>
</li>
<li class="blockList">
<h4>btnExportToFasta</h4>
<pre>org.eclipse.swt.widgets.Button btnExportToFasta</pre>
</li>
<li class="blockList">
<h4>btnGenomeViewer</h4>
<pre>org.eclipse.swt.widgets.Button btnGenomeViewer</pre>
</li>
<li class="blockList">
<h4>btnGetAnnotationInformation</h4>
<pre>org.eclipse.swt.widgets.Button btnGetAnnotationInformation</pre>
</li>
<li class="blockList">
<h4>btnHeatmapview</h4>
<pre>org.eclipse.swt.widgets.Button btnHeatmapview</pre>
</li>
<li class="blockList">
<h4>btnHelp</h4>
<pre>org.eclipse.swt.widgets.Button btnHelp</pre>
</li>
<li class="blockList">
<h4>btnLocalization</h4>
<pre>org.eclipse.swt.widgets.Button btnLocalization</pre>
</li>
<li class="blockList">
<h4>btnNcbiChromosome</h4>
<pre>org.eclipse.swt.widgets.Button btnNcbiChromosome</pre>
</li>
<li class="blockList">
<h4>btnNucleotideSequence</h4>
<pre>org.eclipse.swt.widgets.Button btnNucleotideSequence</pre>
</li>
<li class="blockList">
<h4>btnSaveAsPng</h4>
<pre>org.eclipse.swt.widgets.Button btnSaveAsPng</pre>
</li>
<li class="blockList">
<h4>btnSaveAsSvg</h4>
<pre>org.eclipse.swt.widgets.Button btnSaveAsSvg</pre>
</li>
<li class="blockList">
<h4>btnSelectall</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectall</pre>
</li>
<li class="blockList">
<h4>btnUnselectall</h4>
<pre>org.eclipse.swt.widgets.Button btnUnselectall</pre>
</li>
<li class="blockList">
<h4>btnUpdateCutoff</h4>
<pre>org.eclipse.swt.widgets.Button btnUpdateCutoff</pre>
</li>
<li class="blockList">
<h4>btnZoomminus</h4>
<pre>org.eclipse.swt.widgets.Button btnZoomminus</pre>
</li>
<li class="blockList">
<h4>btnZoomplus</h4>
<pre>org.eclipse.swt.widgets.Button btnZoomplus</pre>
</li>
<li class="blockList">
<h4>canvasGenome</h4>
<pre><a href="bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a> canvasGenome</pre>
</li>
<li class="blockList">
<h4>chromoID</h4>
<pre>java.lang.String chromoID</pre>
</li>
<li class="blockList">
<h4>columnNames</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; columnNames</pre>
</li>
<li class="blockList">
<h4>comboChromosome</h4>
<pre>org.eclipse.swt.widgets.Combo comboChromosome</pre>
</li>
<li class="blockList">
<h4>comboGenome</h4>
<pre>org.eclipse.swt.widgets.Combo comboGenome</pre>
</li>
<li class="blockList">
<h4>comparatorBioCondition</h4>
<pre><a href="bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a> comparatorBioCondition</pre>
</li>
<li class="blockList">
<h4>composite</h4>
<pre>org.eclipse.swt.widgets.Composite composite</pre>
</li>
<li class="blockList">
<h4>composite_12</h4>
<pre>org.eclipse.swt.widgets.Composite composite_12</pre>
</li>
<li class="blockList">
<h4>composite_13</h4>
<pre>org.eclipse.swt.widgets.Composite composite_13</pre>
</li>
<li class="blockList">
<h4>composite_14</h4>
<pre>org.eclipse.swt.widgets.Composite composite_14</pre>
</li>
<li class="blockList">
<h4>composite_15</h4>
<pre>org.eclipse.swt.widgets.Composite composite_15</pre>
</li>
<li class="blockList">
<h4>composite_16</h4>
<pre>org.eclipse.swt.widgets.Composite composite_16</pre>
</li>
<li class="blockList">
<h4>composite_17</h4>
<pre>org.eclipse.swt.widgets.Composite composite_17</pre>
</li>
<li class="blockList">
<h4>composite_18</h4>
<pre>org.eclipse.swt.widgets.Composite composite_18</pre>
</li>
<li class="blockList">
<h4>composite_6</h4>
<pre>org.eclipse.swt.widgets.Composite composite_6</pre>
</li>
<li class="blockList">
<h4>composite_7</h4>
<pre>org.eclipse.swt.widgets.Composite composite_7</pre>
</li>
<li class="blockList">
<h4>composite_9</h4>
<pre>org.eclipse.swt.widgets.Composite composite_9</pre>
</li>
<li class="blockList">
<h4>composite_localization</h4>
<pre>org.eclipse.swt.widgets.Composite composite_localization</pre>
</li>
<li class="blockList">
<h4>composite_pvalue</h4>
<pre>org.eclipse.swt.widgets.Composite composite_pvalue</pre>
</li>
<li class="blockList">
<h4>compositeGenome</h4>
<pre>org.eclipse.swt.widgets.Composite compositeGenome</pre>
</li>
<li class="blockList">
<h4>compSuppInfo</h4>
<pre>org.eclipse.swt.widgets.Composite compSuppInfo</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>genome</h4>
<pre><a href="bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> genome</pre>
<div class="block">Egde Genome loaded</div>
</li>
<li class="blockList">
<h4>lblBegin</h4>
<pre>org.eclipse.swt.widgets.Text lblBegin</pre>
</li>
<li class="blockList">
<h4>lblCog</h4>
<pre>org.eclipse.swt.widgets.Text lblCog</pre>
</li>
<li class="blockList">
<h4>lblConservation</h4>
<pre>org.eclipse.swt.widgets.Text lblConservation</pre>
</li>
<li class="blockList">
<h4>lblConservation2</h4>
<pre>org.eclipse.swt.widgets.Label lblConservation2</pre>
</li>
<li class="blockList">
<h4>lblEnd</h4>
<pre>org.eclipse.swt.widgets.Text lblEnd</pre>
</li>
<li class="blockList">
<h4>lblGene</h4>
<pre>org.eclipse.swt.widgets.Label lblGene</pre>
</li>
<li class="blockList">
<h4>lblLocus</h4>
<pre>org.eclipse.swt.widgets.Text lblLocus</pre>
</li>
<li class="blockList">
<h4>lblName</h4>
<pre>org.eclipse.swt.widgets.Text lblName</pre>
</li>
<li class="blockList">
<h4>lblNodiff</h4>
<pre>org.eclipse.swt.widgets.Label lblNodiff</pre>
</li>
<li class="blockList">
<h4>lblOperon</h4>
<pre>org.eclipse.swt.widgets.Text lblOperon</pre>
</li>
<li class="blockList">
<h4>lblOver</h4>
<pre>org.eclipse.swt.widgets.Label lblOver</pre>
</li>
<li class="blockList">
<h4>lblOverProteomes</h4>
<pre>org.eclipse.swt.widgets.Label lblOverProteomes</pre>
</li>
<li class="blockList">
<h4>lblPredictedSubcellularLocalization</h4>
<pre>org.eclipse.swt.widgets.Label lblPredictedSubcellularLocalization</pre>
</li>
<li class="blockList">
<h4>lblProduct</h4>
<pre>org.eclipse.swt.widgets.Text lblProduct</pre>
</li>
<li class="blockList">
<h4>lblProtID</h4>
<pre>org.eclipse.swt.widgets.Label lblProtID</pre>
</li>
<li class="blockList">
<h4>lblSizeaa</h4>
<pre>org.eclipse.swt.widgets.Text lblSizeaa</pre>
</li>
<li class="blockList">
<h4>lblSizeBP</h4>
<pre>org.eclipse.swt.widgets.Text lblSizeBP</pre>
</li>
<li class="blockList">
<h4>lblStrand</h4>
<pre>org.eclipse.swt.widgets.Text lblStrand</pre>
</li>
<li class="blockList">
<h4>lblTranscriptomesData</h4>
<pre>org.eclipse.swt.widgets.Label lblTranscriptomesData</pre>
</li>
<li class="blockList">
<h4>lblUnder</h4>
<pre>org.eclipse.swt.widgets.Label lblUnder</pre>
</li>
<li class="blockList">
<h4>listGenes</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; listGenes</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>scrolledComposite</h4>
<pre>org.eclipse.swt.custom.ScrolledComposite scrolledComposite</pre>
</li>
<li class="blockList">
<h4>selectedGenomes</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; selectedGenomes</pre>
</li>
<li class="blockList">
<h4>sequence</h4>
<pre><a href="bacnet/datamodel/sequence/Gene.html" title="class in bacnet.datamodel.sequence">Gene</a> sequence</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>tabFolder</h4>
<pre>org.eclipse.swt.widgets.TabFolder tabFolder</pre>
</li>
<li class="blockList">
<h4>tableGenes</h4>
<pre>org.eclipse.swt.widgets.Table tableGenes</pre>
</li>
<li class="blockList">
<h4>tableHomolog</h4>
<pre>org.eclipse.swt.widgets.Table tableHomolog</pre>
</li>
<li class="blockList">
<h4>tableHomologViewer</h4>
<pre>org.eclipse.jface.viewers.TableViewer tableHomologViewer</pre>
</li>
<li class="blockList">
<h4>tableNodiff</h4>
<pre>org.eclipse.swt.widgets.Table tableNodiff</pre>
</li>
<li class="blockList">
<h4>tableOver</h4>
<pre>org.eclipse.swt.widgets.Table tableOver</pre>
</li>
<li class="blockList">
<h4>tableProteomes</h4>
<pre>org.eclipse.swt.widgets.Table tableProteomes</pre>
</li>
<li class="blockList">
<h4>tableUnder</h4>
<pre>org.eclipse.swt.widgets.Table tableUnder</pre>
</li>
<li class="blockList">
<h4>tbtmExpressionData</h4>
<pre>org.eclipse.swt.widgets.TabItem tbtmExpressionData</pre>
</li>
<li class="blockList">
<h4>tbtmGeneralInformation</h4>
<pre>org.eclipse.swt.widgets.TabItem tbtmGeneralInformation</pre>
</li>
<li class="blockList">
<h4>tbtmHomologs</h4>
<pre>org.eclipse.swt.widgets.TabItem tbtmHomologs</pre>
</li>
<li class="blockList">
<h4>tbtmProteomes</h4>
<pre>org.eclipse.swt.widgets.TabItem tbtmProteomes</pre>
</li>
<li class="blockList">
<h4>tbtmSynteny</h4>
<pre>org.eclipse.swt.widgets.TabItem tbtmSynteny</pre>
</li>
<li class="blockList">
<h4>textFeature</h4>
<pre>org.eclipse.swt.widgets.Text textFeature</pre>
</li>
<li class="blockList">
<h4>trackGenome</h4>
<pre><a href="bacnet/genomeBrowser/core/Track.html" title="class in bacnet.genomeBrowser.core">Track</a> trackGenome</pre>
</li>
<li class="blockList">
<h4>txtCutoffLogFC</h4>
<pre>org.eclipse.swt.widgets.Text txtCutoffLogFC</pre>
</li>
<li class="blockList">
<h4>txtCutoffPvalue</h4>
<pre>org.eclipse.swt.widgets.Text txtCutoffPvalue</pre>
</li>
<li class="blockList">
<h4>txtSearch</h4>
<pre>org.eclipse.swt.widgets.Text txtSearch</pre>
</li>
<li class="blockList">
<h4>txtSearchGenome</h4>
<pre>org.eclipse.swt.widgets.Text txtSearchGenome</pre>
</li>
<li class="blockListLast">
<h4>viewID</h4>
<pre>java.lang.String viewID</pre>
<div class="block">Current viewId = ID + math.random()</div>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.sequenceTools.GenomicsView">
<!--   -->
</a>
<h3>Class <a href="bacnet/sequenceTools/GenomicsView.html" title="class in bacnet.sequenceTools">bacnet.sequenceTools.GenomicsView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>497572638968902304L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>bioCondsArray</h4>
<pre>java.lang.String[][] bioCondsArray</pre>
<div class="block">Array used to init data</div>
</li>
<li class="blockList">
<h4>bioCondsToDisplay</h4>
<pre>java.util.ArrayList&lt;java.lang.String[]&gt; bioCondsToDisplay</pre>
</li>
<li class="blockList">
<h4>browserPhylo</h4>
<pre>org.eclipse.swt.browser.Browser browserPhylo</pre>
</li>
<li class="blockList">
<h4>btnHelp</h4>
<pre>org.eclipse.swt.widgets.Button btnHelp</pre>
</li>
<li class="blockList">
<h4>btnSavePng</h4>
<pre>org.eclipse.swt.widgets.Button btnSavePng</pre>
</li>
<li class="blockList">
<h4>btnSaveSVG</h4>
<pre>org.eclipse.swt.widgets.Button btnSaveSVG</pre>
</li>
<li class="blockList">
<h4>btnSaveTxt</h4>
<pre>org.eclipse.swt.widgets.Button btnSaveTxt</pre>
</li>
<li class="blockList">
<h4>btnSelectall</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectall</pre>
</li>
<li class="blockList">
<h4>btnUnselectall</h4>
<pre>org.eclipse.swt.widgets.Button btnUnselectall</pre>
</li>
<li class="blockList">
<h4>columnNames</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; columnNames</pre>
</li>
<li class="blockList">
<h4>comparatorBioCondition</h4>
<pre><a href="bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">BioConditionComparator</a> comparatorBioCondition</pre>
</li>
<li class="blockList">
<h4>compositeSummary</h4>
<pre>org.eclipse.swt.widgets.Composite compositeSummary</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>lblXxSrnas</h4>
<pre>org.eclipse.swt.widgets.Label lblXxSrnas</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>selectedGenomes</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; selectedGenomes</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>tableGenome</h4>
<pre>org.eclipse.swt.widgets.Table tableGenome</pre>
</li>
<li class="blockList">
<h4>tableGenomeViewer</h4>
<pre>org.eclipse.jface.viewers.TableViewer tableGenomeViewer</pre>
</li>
<li class="blockList">
<h4>txtSearch</h4>
<pre>org.eclipse.swt.widgets.Text txtSearch</pre>
</li>
<li class="blockListLast">
<h4>url</h4>
<pre>java.lang.String url</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.sequenceTools.LocalizationDialog">
<!--   -->
</a>
<h3>Class <a href="bacnet/sequenceTools/LocalizationDialog.html" title="class in bacnet.sequenceTools">bacnet.sequenceTools.LocalizationDialog</a> extends org.eclipse.jface.dialogs.Dialog implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>8150068655518237507L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>arrayGeneToLocalization</h4>
<pre>java.lang.String[][] arrayGeneToLocalization</pre>
</li>
<li class="blockList">
<h4>bioCondsArray</h4>
<pre>java.lang.String[][] bioCondsArray</pre>
</li>
<li class="blockList">
<h4>genome</h4>
<pre><a href="bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> genome</pre>
</li>
<li class="blockListLast">
<h4>sequence</h4>
<pre><a href="bacnet/datamodel/sequence/Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a> sequence</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.sequenceTools.SecondStructureRNADialog">
<!--   -->
</a>
<h3>Class <a href="bacnet/sequenceTools/SecondStructureRNADialog.html" title="class in bacnet.sequenceTools">bacnet.sequenceTools.SecondStructureRNADialog</a> extends org.eclipse.jface.dialogs.Dialog implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-8357071260167781918L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockListLast">
<h4>fileName</h4>
<pre>java.lang.String fileName</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.sequenceTools.SequenceDisplayDialog">
<!--   -->
</a>
<h3>Class <a href="bacnet/sequenceTools/SequenceDisplayDialog.html" title="class in bacnet.sequenceTools">bacnet.sequenceTools.SequenceDisplayDialog</a> extends org.eclipse.jface.dialogs.Dialog implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-3418095326728894710L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>begin</h4>
<pre>int begin</pre>
</li>
<li class="blockList">
<h4>btnAminoAcidSequence</h4>
<pre>org.eclipse.swt.widgets.Button btnAminoAcidSequence</pre>
</li>
<li class="blockList">
<h4>btnNucleotideSequence</h4>
<pre>org.eclipse.swt.widgets.Button btnNucleotideSequence</pre>
</li>
<li class="blockList">
<h4>btnSaveToFasta</h4>
<pre>org.eclipse.swt.widgets.Button btnSaveToFasta</pre>
</li>
<li class="blockList">
<h4>btnSendToBlast</h4>
<pre>org.eclipse.swt.widgets.Button btnSendToBlast</pre>
</li>
<li class="blockList">
<h4>btnStrandMinus</h4>
<pre>org.eclipse.swt.widgets.Button btnStrandMinus</pre>
</li>
<li class="blockList">
<h4>btnStrandplus</h4>
<pre>org.eclipse.swt.widgets.Button btnStrandplus</pre>
</li>
<li class="blockList">
<h4>chromoId</h4>
<pre>java.lang.String chromoId</pre>
</li>
<li class="blockList">
<h4>cmbChromo</h4>
<pre>org.eclipse.swt.widgets.Combo cmbChromo</pre>
</li>
<li class="blockList">
<h4>comboGenome</h4>
<pre>org.eclipse.swt.widgets.Combo comboGenome</pre>
</li>
<li class="blockList">
<h4>end</h4>
<pre>int end</pre>
</li>
<li class="blockList">
<h4>genome</h4>
<pre><a href="bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> genome</pre>
</li>
<li class="blockList">
<h4>lblInfo</h4>
<pre>org.eclipse.swt.widgets.Label lblInfo</pre>
</li>
<li class="blockList">
<h4>nucleotide</h4>
<pre>boolean nucleotide</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>seqFasta</h4>
<pre>java.lang.String seqFasta</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>strand</h4>
<pre><a href="org/biojava3/core/sequence/Strand.html" title="enum in org.biojava3.core.sequence">Strand</a> strand</pre>
</li>
<li class="blockList">
<h4>textBegin</h4>
<pre>org.eclipse.swt.widgets.Text textBegin</pre>
</li>
<li class="blockList">
<h4>textEnd</h4>
<pre>org.eclipse.swt.widgets.Text textEnd</pre>
</li>
<li class="blockListLast">
<h4>textSequence</h4>
<pre>org.eclipse.swt.widgets.Text textSequence</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.sequenceTools.SrnaSummaryView">
<!--   -->
</a>
<h3>Class <a href="bacnet/sequenceTools/SrnaSummaryView.html" title="class in bacnet.sequenceTools">bacnet.sequenceTools.SrnaSummaryView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>5574893108478572129L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>array</h4>
<pre>java.lang.String[][] array</pre>
</li>
<li class="blockList">
<h4>btnAsrnas</h4>
<pre>org.eclipse.swt.widgets.Button btnAsrnas</pre>
</li>
<li class="blockList">
<h4>btnCisRegs</h4>
<pre>org.eclipse.swt.widgets.Button btnCisRegs</pre>
</li>
<li class="blockList">
<h4>btnDownloadtext</h4>
<pre>org.eclipse.swt.widgets.Button btnDownloadtext</pre>
</li>
<li class="blockList">
<h4>btnSrnas</h4>
<pre>org.eclipse.swt.widgets.Button btnSrnas</pre>
</li>
<li class="blockList">
<h4>composite</h4>
<pre>org.eclipse.swt.widgets.Composite composite</pre>
</li>
<li class="blockList">
<h4>composite_1</h4>
<pre>org.eclipse.swt.widgets.Composite composite_1</pre>
</li>
<li class="blockList">
<h4>composite_2</h4>
<pre>org.eclipse.swt.widgets.Composite composite_2</pre>
</li>
<li class="blockList">
<h4>composite_3</h4>
<pre>org.eclipse.swt.widgets.Composite composite_3</pre>
</li>
<li class="blockList">
<h4>displayType</h4>
<pre><a href="bacnet/datamodel/sequence/Srna.TypeSrna.html" title="enum in bacnet.datamodel.sequence">Srna.TypeSrna</a> displayType</pre>
</li>
<li class="blockList">
<h4>label</h4>
<pre>org.eclipse.swt.widgets.Label label</pre>
</li>
<li class="blockList">
<h4>lblImage</h4>
<pre>org.eclipse.swt.widgets.Label lblImage</pre>
</li>
<li class="blockList">
<h4>lblSelectAType</h4>
<pre>org.eclipse.swt.widgets.Label lblSelectAType</pre>
</li>
<li class="blockList">
<h4>lblXxSrnas</h4>
<pre>org.eclipse.swt.widgets.Label lblXxSrnas</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>scrolledComposite</h4>
<pre>org.eclipse.swt.custom.ScrolledComposite scrolledComposite</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>table</h4>
<pre>org.eclipse.swt.widgets.Table table</pre>
</li>
<li class="blockListLast">
<h4>url</h4>
<pre>java.lang.String url</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.sequenceTools.SrnaView">
<!--   -->
</a>
<h3>Class <a href="bacnet/sequenceTools/SrnaView.html" title="class in bacnet.sequenceTools">bacnet.sequenceTools.SrnaView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-2943857063947928839L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>arrayDataList</h4>
<pre>java.lang.String[][] arrayDataList</pre>
</li>
<li class="blockList">
<h4>btnDisplayNucleotideSequence</h4>
<pre>org.eclipse.swt.widgets.Button btnDisplayNucleotideSequence</pre>
</li>
<li class="blockList">
<h4>btnGenomeViewer</h4>
<pre>org.eclipse.swt.widgets.Button btnGenomeViewer</pre>
</li>
<li class="blockList">
<h4>btnHeatmapview</h4>
<pre>org.eclipse.swt.widgets.Button btnHeatmapview</pre>
</li>
<li class="blockList">
<h4>btnHelp</h4>
<pre>org.eclipse.swt.widgets.Button btnHelp</pre>
</li>
<li class="blockList">
<h4>btnOpenStructure</h4>
<pre>org.eclipse.swt.widgets.Button btnOpenStructure</pre>
</li>
<li class="blockList">
<h4>btnUpdateCutoff</h4>
<pre>org.eclipse.swt.widgets.Button btnUpdateCutoff</pre>
</li>
<li class="blockList">
<h4>btnZoomminus</h4>
<pre>org.eclipse.swt.widgets.Button btnZoomminus</pre>
</li>
<li class="blockList">
<h4>btnZoomplus</h4>
<pre>org.eclipse.swt.widgets.Button btnZoomplus</pre>
</li>
<li class="blockList">
<h4>canvasGenome</h4>
<pre><a href="bacnet/genomeBrowser/tracksGUI/TrackCanvasGenome.html" title="class in bacnet.genomeBrowser.tracksGUI">TrackCanvasGenome</a> canvasGenome</pre>
</li>
<li class="blockList">
<h4>composite</h4>
<pre>org.eclipse.swt.widgets.Composite composite</pre>
</li>
<li class="blockList">
<h4>composite_2</h4>
<pre>org.eclipse.swt.widgets.Composite composite_2</pre>
</li>
<li class="blockList">
<h4>composite_3</h4>
<pre>org.eclipse.swt.widgets.Composite composite_3</pre>
</li>
<li class="blockList">
<h4>composite_6</h4>
<pre>org.eclipse.swt.widgets.Composite composite_6</pre>
</li>
<li class="blockList">
<h4>composite_7</h4>
<pre>org.eclipse.swt.widgets.Composite composite_7</pre>
</li>
<li class="blockList">
<h4>composite_8_1</h4>
<pre>org.eclipse.swt.widgets.Composite composite_8_1</pre>
</li>
<li class="blockList">
<h4>composite_pvalue</h4>
<pre>org.eclipse.swt.widgets.Composite composite_pvalue</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>genome</h4>
<pre><a href="bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> genome</pre>
<div class="block">Egde Genome loaded</div>
</li>
<li class="blockList">
<h4>label_2</h4>
<pre>org.eclipse.swt.widgets.Label label_2</pre>
</li>
<li class="blockList">
<h4>label_3</h4>
<pre>org.eclipse.swt.widgets.Label label_3</pre>
</li>
<li class="blockList">
<h4>lblBegin</h4>
<pre>org.eclipse.swt.widgets.Text lblBegin</pre>
</li>
<li class="blockList">
<h4>lblDescribedIn</h4>
<pre>org.eclipse.swt.widgets.Label lblDescribedIn</pre>
</li>
<li class="blockList">
<h4>lblEnd</h4>
<pre>org.eclipse.swt.widgets.Text lblEnd</pre>
</li>
<li class="blockList">
<h4>lblGene</h4>
<pre>org.eclipse.swt.widgets.Label lblGene</pre>
</li>
<li class="blockList">
<h4>lblNewLabel</h4>
<pre>org.eclipse.swt.widgets.Label lblNewLabel</pre>
</li>
<li class="blockList">
<h4>lblNodiff</h4>
<pre>org.eclipse.swt.widgets.Label lblNodiff</pre>
</li>
<li class="blockList">
<h4>lblNote</h4>
<pre>org.eclipse.swt.widgets.Text lblNote</pre>
</li>
<li class="blockList">
<h4>lblOrAnyOther</h4>
<pre>org.eclipse.swt.widgets.Label lblOrAnyOther</pre>
</li>
<li class="blockList">
<h4>lblOver</h4>
<pre>org.eclipse.swt.widgets.Label lblOver</pre>
</li>
<li class="blockList">
<h4>lblPredictedSecondaryStructure</h4>
<pre>org.eclipse.swt.widgets.Label lblPredictedSecondaryStructure</pre>
</li>
<li class="blockList">
<h4>lblRef</h4>
<pre>org.eclipse.swt.widgets.Label lblRef</pre>
</li>
<li class="blockList">
<h4>lblSizeBP</h4>
<pre>org.eclipse.swt.widgets.Text lblSizeBP</pre>
</li>
<li class="blockList">
<h4>lblStrand</h4>
<pre>org.eclipse.swt.widgets.Text lblStrand</pre>
</li>
<li class="blockList">
<h4>lblStructure</h4>
<pre>org.eclipse.swt.widgets.Label lblStructure</pre>
</li>
<li class="blockList">
<h4>lblSynonims</h4>
<pre>org.eclipse.swt.widgets.Text lblSynonims</pre>
</li>
<li class="blockList">
<h4>lblUnder</h4>
<pre>org.eclipse.swt.widgets.Label lblUnder</pre>
</li>
<li class="blockList">
<h4>listSrnas</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; listSrnas</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>scrolledComposite</h4>
<pre>org.eclipse.swt.custom.ScrolledComposite scrolledComposite</pre>
</li>
<li class="blockList">
<h4>seq</h4>
<pre><a href="bacnet/datamodel/sequence/Srna.html" title="class in bacnet.datamodel.sequence">Srna</a> seq</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>sRNARefToLink</h4>
<pre>java.util.HashMap&lt;java.lang.String,&#8203;java.lang.String&gt; sRNARefToLink</pre>
</li>
<li class="blockList">
<h4>tabFolder</h4>
<pre>org.eclipse.swt.widgets.TabFolder tabFolder</pre>
</li>
<li class="blockList">
<h4>tableNoDiff</h4>
<pre>org.eclipse.swt.widgets.Table tableNoDiff</pre>
</li>
<li class="blockList">
<h4>tableOtherInfo</h4>
<pre>org.eclipse.swt.widgets.Table tableOtherInfo</pre>
</li>
<li class="blockList">
<h4>tableOver</h4>
<pre>org.eclipse.swt.widgets.Table tableOver</pre>
</li>
<li class="blockList">
<h4>tableSrnas</h4>
<pre>org.eclipse.swt.widgets.Table tableSrnas</pre>
</li>
<li class="blockList">
<h4>tableUnder</h4>
<pre>org.eclipse.swt.widgets.Table tableUnder</pre>
</li>
<li class="blockList">
<h4>tbtmExpressionAtlas</h4>
<pre>org.eclipse.swt.widgets.TabItem tbtmExpressionAtlas</pre>
</li>
<li class="blockList">
<h4>tbtmGeneralInformation</h4>
<pre>org.eclipse.swt.widgets.TabItem tbtmGeneralInformation</pre>
</li>
<li class="blockList">
<h4>txtCutoffLogFC</h4>
<pre>org.eclipse.swt.widgets.Text txtCutoffLogFC</pre>
</li>
<li class="blockList">
<h4>txtCutoffPvalue</h4>
<pre>org.eclipse.swt.widgets.Text txtCutoffPvalue</pre>
</li>
<li class="blockList">
<h4>txtSearch</h4>
<pre>org.eclipse.swt.widgets.Text txtSearch</pre>
</li>
<li class="blockList">
<h4>txtSeqnucleotide</h4>
<pre>org.eclipse.swt.widgets.Text txtSeqnucleotide</pre>
</li>
<li class="blockListLast">
<h4>viewId</h4>
<pre>java.lang.String viewId</pre>
<div class="block">Current viewId = ID + math.random()</div>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.table</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.table.TableSWTComposite">
<!--   -->
</a>
<h3>Class <a href="bacnet/table/TableSWTComposite.html" title="class in bacnet.table">bacnet.table.TableSWTComposite</a> extends org.eclipse.swt.widgets.Composite implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-5019287630134430710L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnColormapper</h4>
<pre>org.eclipse.swt.widgets.Button btnColormapper</pre>
</li>
<li class="blockList">
<h4>btnDisplayValues</h4>
<pre>org.eclipse.swt.widgets.Button btnDisplayValues</pre>
</li>
<li class="blockList">
<h4>btnFilter</h4>
<pre>org.eclipse.swt.widgets.Button btnFilter</pre>
</li>
<li class="blockList">
<h4>btnHide</h4>
<pre>org.eclipse.swt.widgets.Button btnHide</pre>
</li>
<li class="blockList">
<h4>btnSavepng</h4>
<pre>org.eclipse.swt.widgets.Button btnSavepng</pre>
</li>
<li class="blockList">
<h4>btnSaveText</h4>
<pre>org.eclipse.swt.widgets.Button btnSaveText</pre>
</li>
<li class="blockList">
<h4>btnZoomIn</h4>
<pre>org.eclipse.swt.widgets.Button btnZoomIn</pre>
</li>
<li class="blockList">
<h4>btnZoomOut</h4>
<pre>org.eclipse.swt.widgets.Button btnZoomOut</pre>
</li>
<li class="blockList">
<h4>colorMapperList</h4>
<pre><a href="bacnet/table/core/ColorMapperList.html" title="class in bacnet.table.core">ColorMapperList</a> colorMapperList</pre>
</li>
<li class="blockList">
<h4>columnNames</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; columnNames</pre>
</li>
<li class="blockList">
<h4>comparator</h4>
<pre><a href="bacnet/table/core/TableViewerComparator.html" title="class in bacnet.table.core">TableViewerComparator</a> comparator</pre>
</li>
<li class="blockList">
<h4>composite</h4>
<pre>org.eclipse.swt.widgets.Composite composite</pre>
</li>
<li class="blockList">
<h4>composite_4</h4>
<pre>org.eclipse.swt.widgets.Composite composite_4</pre>
</li>
<li class="blockList">
<h4>displayValues</h4>
<pre>boolean displayValues</pre>
</li>
<li class="blockList">
<h4>excludeColumn</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; excludeColumn</pre>
</li>
<li class="blockList">
<h4>excludeRow</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; excludeRow</pre>
</li>
<li class="blockList">
<h4>filters</h4>
<pre><a href="bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a> filters</pre>
</li>
<li class="blockList">
<h4>hideDialogExcludeColumn</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; hideDialogExcludeColumn</pre>
</li>
<li class="blockList">
<h4>hideDialogExcludeRow</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; hideDialogExcludeRow</pre>
</li>
<li class="blockList">
<h4>labelColumn</h4>
<pre>org.eclipse.swt.widgets.Label labelColumn</pre>
</li>
<li class="blockList">
<h4>labelRow</h4>
<pre>org.eclipse.swt.widgets.Label labelRow</pre>
</li>
<li class="blockList">
<h4>labelVoid</h4>
<pre>org.eclipse.swt.widgets.Label labelVoid</pre>
</li>
<li class="blockList">
<h4>lblSearch</h4>
<pre>org.eclipse.swt.widgets.Label lblSearch</pre>
</li>
<li class="blockList">
<h4>lblX</h4>
<pre>org.eclipse.swt.widgets.Label lblX</pre>
</li>
<li class="blockList">
<h4>matrix</h4>
<pre><a href="bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a> matrix</pre>
</li>
<li class="blockList">
<h4>matrixDisplayed</h4>
<pre><a href="bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a> matrixDisplayed</pre>
</li>
<li class="blockList">
<h4>matrixSize</h4>
<pre><a href="bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">MatrixSize</a> matrixSize</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>searchFilter</h4>
<pre><a href="bacnet/table/gui/TableFilter.html" title="class in bacnet.table.gui">TableFilter</a> searchFilter</pre>
</li>
<li class="blockList">
<h4>searchText</h4>
<pre>org.eclipse.swt.widgets.Text searchText</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>table</h4>
<pre>org.eclipse.swt.widgets.Table table</pre>
</li>
<li class="blockListLast">
<h4>tableViewer</h4>
<pre>org.eclipse.jface.viewers.TableViewer tableViewer</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.table.core</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.table.core.BioConditionComparator">
<!--   -->
</a>
<h3>Class <a href="bacnet/table/core/BioConditionComparator.html" title="class in bacnet.table.core">bacnet.table.core.BioConditionComparator</a> extends org.eclipse.jface.viewers.ViewerComparator implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>4101481773521242366L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>columnNames</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; columnNames</pre>
</li>
<li class="blockList">
<h4>direction</h4>
<pre>int direction</pre>
</li>
<li class="blockListLast">
<h4>propertyIndex</h4>
<pre>int propertyIndex</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.table.core.ColorMapper">
<!--   -->
</a>
<h3>Class <a href="bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">bacnet.table.core.ColorMapper</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>26986650775300526L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>centerColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> centerColorSWT</pre>
</li>
<li class="blockList">
<h4>centerPos</h4>
<pre>double centerPos</pre>
</li>
<li class="blockList">
<h4>columnNames</h4>
<pre>java.util.TreeSet&lt;java.lang.String&gt; columnNames</pre>
</li>
<li class="blockList">
<h4>displayValues</h4>
<pre>boolean displayValues</pre>
</li>
<li class="blockList">
<h4>fontDouble</h4>
<pre>org.eclipse.swt.graphics.Font fontDouble</pre>
</li>
<li class="blockList">
<h4>fontRowName</h4>
<pre>org.eclipse.swt.graphics.Font fontRowName</pre>
</li>
<li class="blockList">
<h4>fontText</h4>
<pre>org.eclipse.swt.graphics.Font fontText</pre>
</li>
<li class="blockList">
<h4>headerColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> headerColorSWT</pre>
</li>
<li class="blockList">
<h4>maxColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> maxColorSWT</pre>
</li>
<li class="blockList">
<h4>maxPos</h4>
<pre>double maxPos</pre>
</li>
<li class="blockList">
<h4>midLeftColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> midLeftColorSWT</pre>
</li>
<li class="blockList">
<h4>midLeftPos</h4>
<pre>double midLeftPos</pre>
</li>
<li class="blockList">
<h4>midRightColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> midRightColorSWT</pre>
</li>
<li class="blockList">
<h4>midRightPos</h4>
<pre>double midRightPos</pre>
</li>
<li class="blockList">
<h4>minColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> minColorSWT</pre>
</li>
<li class="blockList">
<h4>minPos</h4>
<pre>double minPos</pre>
</li>
<li class="blockList">
<h4>textColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> textColorSWT</pre>
</li>
<li class="blockListLast">
<h4>textHeaderColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> textHeaderColorSWT</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.table.core.ColorMapperRCP">
<!--   -->
</a>
<h3>Class <a href="bacnet/table/core/ColorMapperRCP.html" title="class in bacnet.table.core">bacnet.table.core.ColorMapperRCP</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>26986650775300526L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>centerColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> centerColorSWT</pre>
</li>
<li class="blockList">
<h4>centerPos</h4>
<pre>double centerPos</pre>
</li>
<li class="blockList">
<h4>deviation</h4>
<pre>double deviation</pre>
</li>
<li class="blockList">
<h4>displayValues</h4>
<pre>boolean displayValues</pre>
</li>
<li class="blockList">
<h4>fontDouble</h4>
<pre>java.awt.Font fontDouble</pre>
</li>
<li class="blockList">
<h4>fontString</h4>
<pre>java.awt.Font fontString</pre>
</li>
<li class="blockList">
<h4>headerColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> headerColorSWT</pre>
</li>
<li class="blockList">
<h4>max</h4>
<pre>double max</pre>
</li>
<li class="blockList">
<h4>maxColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> maxColorSWT</pre>
</li>
<li class="blockList">
<h4>maxPos</h4>
<pre>double maxPos</pre>
</li>
<li class="blockList">
<h4>mean</h4>
<pre>double mean</pre>
</li>
<li class="blockList">
<h4>median</h4>
<pre>double median</pre>
</li>
<li class="blockList">
<h4>midLeftColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> midLeftColorSWT</pre>
</li>
<li class="blockList">
<h4>midLeftPos</h4>
<pre>double midLeftPos</pre>
</li>
<li class="blockList">
<h4>midRightColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> midRightColorSWT</pre>
</li>
<li class="blockList">
<h4>midRightPos</h4>
<pre>double midRightPos</pre>
</li>
<li class="blockList">
<h4>min</h4>
<pre>double min</pre>
</li>
<li class="blockList">
<h4>minColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> minColorSWT</pre>
</li>
<li class="blockList">
<h4>minPos</h4>
<pre>double minPos</pre>
</li>
<li class="blockList">
<h4>nbElements</h4>
<pre>int nbElements</pre>
</li>
<li class="blockList">
<h4>textColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> textColorSWT</pre>
</li>
<li class="blockList">
<h4>textHeaderColorSWT</h4>
<pre><a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">ColorSWT</a> textHeaderColorSWT</pre>
</li>
<li class="blockListLast">
<h4>var</h4>
<pre>double var</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.table.core.ColorSWT">
<!--   -->
</a>
<h3>Class <a href="bacnet/table/core/ColorSWT.html" title="class in bacnet.table.core">bacnet.table.core.ColorSWT</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>3693115797571798995L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>blue</h4>
<pre>int blue</pre>
</li>
<li class="blockList">
<h4>green</h4>
<pre>int green</pre>
</li>
<li class="blockListLast">
<h4>red</h4>
<pre>int red</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.table.core.Filter">
<!--   -->
</a>
<h3>Class <a href="bacnet/table/core/Filter.html" title="class in bacnet.table.core">bacnet.table.core.Filter</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>6834197254054420424L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>cutOff1</h4>
<pre>double cutOff1</pre>
</li>
<li class="blockList">
<h4>cutOff2</h4>
<pre>double cutOff2</pre>
</li>
<li class="blockList">
<h4>filterColumn</h4>
<pre>boolean filterColumn</pre>
</li>
<li class="blockList">
<h4>name</h4>
<pre>java.lang.String name</pre>
</li>
<li class="blockList">
<h4>tableElementName</h4>
<pre>java.lang.String tableElementName</pre>
</li>
<li class="blockListLast">
<h4>typeFilter</h4>
<pre><a href="bacnet/table/core/Filter.TypeFilter.html" title="enum in bacnet.table.core">Filter.TypeFilter</a> typeFilter</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.table.core.MatrixSize">
<!--   -->
</a>
<h3>Class <a href="bacnet/table/core/MatrixSize.html" title="class in bacnet.table.core">bacnet.table.core.MatrixSize</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>3440001575014228874L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>height</h4>
<pre>java.util.HashMap&lt;java.lang.String,&#8203;java.lang.Integer&gt; height</pre>
</li>
<li class="blockList">
<h4>heightDefault</h4>
<pre>int heightDefault</pre>
</li>
<li class="blockList">
<h4>width</h4>
<pre>java.util.HashMap&lt;java.lang.String,&#8203;java.lang.Integer&gt; width</pre>
</li>
<li class="blockList">
<h4>widthDefault</h4>
<pre>int widthDefault</pre>
</li>
<li class="blockListLast">
<h4>widthRowNameDefault</h4>
<pre>int widthRowNameDefault</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.table.core.TableViewerComparator">
<!--   -->
</a>
<h3>Class <a href="bacnet/table/core/TableViewerComparator.html" title="class in bacnet.table.core">bacnet.table.core.TableViewerComparator</a> extends org.eclipse.jface.viewers.ViewerComparator implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-9021641281635843621L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>direction</h4>
<pre>int direction</pre>
</li>
<li class="blockListLast">
<h4>propertyIndex</h4>
<pre>int propertyIndex</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.table.gui</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.table.gui.ColorMapperWizardPage">
<!--   -->
</a>
<h3>Class <a href="bacnet/table/gui/ColorMapperWizardPage.html" title="class in bacnet.table.gui">bacnet.table.gui.ColorMapperWizardPage</a> extends org.eclipse.jface.wizard.WizardPage implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-3829149015717789437L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnFont</h4>
<pre>org.eclipse.swt.widgets.Button btnFont</pre>
</li>
<li class="blockList">
<h4>btnFontNumber</h4>
<pre>org.eclipse.swt.widgets.Button btnFontNumber</pre>
</li>
<li class="blockList">
<h4>btnLoadColorMapper</h4>
<pre>org.eclipse.swt.widgets.Button btnLoadColorMapper</pre>
</li>
<li class="blockList">
<h4>btnNumberTextColor</h4>
<pre>org.eclipse.swt.widgets.Button btnNumberTextColor</pre>
</li>
<li class="blockList">
<h4>btnSaveColorMapper</h4>
<pre>org.eclipse.swt.widgets.Button btnSaveColorMapper</pre>
</li>
<li class="blockList">
<h4>btnSelectCenterColor</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectCenterColor</pre>
</li>
<li class="blockList">
<h4>btnSelectHeaderColor</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectHeaderColor</pre>
</li>
<li class="blockList">
<h4>btnSelectMaxColor</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectMaxColor</pre>
</li>
<li class="blockList">
<h4>btnSelectMidleftColor</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectMidleftColor</pre>
</li>
<li class="blockList">
<h4>btnSelectMidRightColor</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectMidRightColor</pre>
</li>
<li class="blockList">
<h4>btnSelectMinColor</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectMinColor</pre>
</li>
<li class="blockList">
<h4>btnTextHeaderColor</h4>
<pre>org.eclipse.swt.widgets.Button btnTextHeaderColor</pre>
</li>
<li class="blockList">
<h4>canvas</h4>
<pre>org.eclipse.swt.widgets.Canvas canvas</pre>
</li>
<li class="blockList">
<h4>colorMapper</h4>
<pre><a href="bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a> colorMapper</pre>
</li>
<li class="blockList">
<h4>composite</h4>
<pre>org.eclipse.swt.widgets.Composite composite</pre>
</li>
<li class="blockList">
<h4>composite_10</h4>
<pre>org.eclipse.swt.widgets.Composite composite_10</pre>
</li>
<li class="blockList">
<h4>composite_11</h4>
<pre>org.eclipse.swt.widgets.Composite composite_11</pre>
</li>
<li class="blockList">
<h4>composite_12</h4>
<pre>org.eclipse.swt.widgets.Composite composite_12</pre>
</li>
<li class="blockList">
<h4>composite_13</h4>
<pre>org.eclipse.swt.widgets.Composite composite_13</pre>
</li>
<li class="blockList">
<h4>composite_8</h4>
<pre>org.eclipse.swt.widgets.Composite composite_8</pre>
</li>
<li class="blockList">
<h4>composite_9</h4>
<pre>org.eclipse.swt.widgets.Composite composite_9</pre>
</li>
<li class="blockList">
<h4>label_10</h4>
<pre>org.eclipse.swt.widgets.Label label_10</pre>
</li>
<li class="blockList">
<h4>label_11</h4>
<pre>org.eclipse.swt.widgets.Label label_11</pre>
</li>
<li class="blockList">
<h4>label_4</h4>
<pre>org.eclipse.swt.widgets.Label label_4</pre>
</li>
<li class="blockList">
<h4>label_9</h4>
<pre>org.eclipse.swt.widgets.Label label_9</pre>
</li>
<li class="blockList">
<h4>lblBoundleft</h4>
<pre>org.eclipse.swt.widgets.Label lblBoundleft</pre>
</li>
<li class="blockList">
<h4>lblBoundRight</h4>
<pre>org.eclipse.swt.widgets.Label lblBoundRight</pre>
</li>
<li class="blockList">
<h4>lblCenter</h4>
<pre>org.eclipse.swt.widgets.Label lblCenter</pre>
</li>
<li class="blockList">
<h4>lblMax</h4>
<pre>org.eclipse.swt.widgets.Label lblMax</pre>
</li>
<li class="blockList">
<h4>lblMidleft</h4>
<pre>org.eclipse.swt.widgets.Label lblMidleft</pre>
</li>
<li class="blockList">
<h4>lblMidright</h4>
<pre>org.eclipse.swt.widgets.Label lblMidright</pre>
</li>
<li class="blockList">
<h4>lblMin_1</h4>
<pre>org.eclipse.swt.widgets.Label lblMin_1</pre>
</li>
<li class="blockList">
<h4>lblSetColor</h4>
<pre>org.eclipse.swt.widgets.Label lblSetColor</pre>
</li>
<li class="blockList">
<h4>lblsliderCenter</h4>
<pre>org.eclipse.swt.widgets.Text lblsliderCenter</pre>
</li>
<li class="blockList">
<h4>lblsliderMidLeft</h4>
<pre>org.eclipse.swt.widgets.Text lblsliderMidLeft</pre>
</li>
<li class="blockList">
<h4>lblsliderMidRight</h4>
<pre>org.eclipse.swt.widgets.Text lblsliderMidRight</pre>
</li>
<li class="blockList">
<h4>previousColorMapper</h4>
<pre><a href="bacnet/table/core/ColorMapper.html" title="class in bacnet.table.core">ColorMapper</a> previousColorMapper</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>text</h4>
<pre>java.lang.String text</pre>
</li>
<li class="blockList">
<h4>textMax</h4>
<pre>org.eclipse.swt.widgets.Text textMax</pre>
</li>
<li class="blockListLast">
<h4>textMin</h4>
<pre>org.eclipse.swt.widgets.Text textMin</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.table.gui.FilterDialog">
<!--   -->
</a>
<h3>Class <a href="bacnet/table/gui/FilterDialog.html" title="class in bacnet.table.gui">bacnet.table.gui.FilterDialog</a> extends org.eclipse.jface.dialogs.Dialog implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>112312911659116561L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnColumn</h4>
<pre>org.eclipse.swt.widgets.Button btnColumn</pre>
</li>
<li class="blockList">
<h4>btnRow</h4>
<pre>org.eclipse.swt.widgets.Button btnRow</pre>
</li>
<li class="blockList">
<h4>cmbColumn</h4>
<pre>org.eclipse.swt.widgets.Combo cmbColumn</pre>
</li>
<li class="blockList">
<h4>cmbRow</h4>
<pre>org.eclipse.swt.widgets.Combo cmbRow</pre>
</li>
<li class="blockList">
<h4>cmbType</h4>
<pre>org.eclipse.swt.widgets.Combo cmbType</pre>
</li>
<li class="blockList">
<h4>composite</h4>
<pre>org.eclipse.swt.widgets.Composite composite</pre>
</li>
<li class="blockList">
<h4>composite_1</h4>
<pre>org.eclipse.swt.widgets.Composite composite_1</pre>
</li>
<li class="blockList">
<h4>composite_2</h4>
<pre>org.eclipse.swt.widgets.Composite composite_2</pre>
</li>
<li class="blockList">
<h4>composite_3</h4>
<pre>org.eclipse.swt.widgets.Composite composite_3</pre>
</li>
<li class="blockList">
<h4>composite_4</h4>
<pre>org.eclipse.swt.widgets.Composite composite_4</pre>
</li>
<li class="blockList">
<h4>composite_5</h4>
<pre>org.eclipse.swt.widgets.Composite composite_5</pre>
</li>
<li class="blockList">
<h4>composite_6</h4>
<pre>org.eclipse.swt.widgets.Composite composite_6</pre>
</li>
<li class="blockList">
<h4>filter</h4>
<pre><a href="bacnet/table/core/Filter.html" title="class in bacnet.table.core">Filter</a> filter</pre>
</li>
<li class="blockList">
<h4>label</h4>
<pre>org.eclipse.swt.widgets.Label label</pre>
</li>
<li class="blockList">
<h4>lblCutoff</h4>
<pre>org.eclipse.swt.widgets.Label lblCutoff</pre>
</li>
<li class="blockList">
<h4>lblNewLabel_1</h4>
<pre>org.eclipse.swt.widgets.Label lblNewLabel_1</pre>
</li>
<li class="blockList">
<h4>lblSetTheProperties</h4>
<pre>org.eclipse.swt.widgets.Label lblSetTheProperties</pre>
</li>
<li class="blockList">
<h4>matrix</h4>
<pre><a href="bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a> matrix</pre>
</li>
<li class="blockList">
<h4>textName</h4>
<pre>org.eclipse.swt.widgets.Text textName</pre>
</li>
<li class="blockList">
<h4>txtCutoff1</h4>
<pre>org.eclipse.swt.widgets.Text txtCutoff1</pre>
</li>
<li class="blockListLast">
<h4>txtCutoff2</h4>
<pre>org.eclipse.swt.widgets.Text txtCutoff2</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.table.gui.FilterExpressionMatrixDialog">
<!--   -->
</a>
<h3>Class <a href="bacnet/table/gui/FilterExpressionMatrixDialog.html" title="class in bacnet.table.gui">bacnet.table.gui.FilterExpressionMatrixDialog</a> extends org.eclipse.jface.dialogs.TitleAreaDialog implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>8526381730648269264L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnCreateNew</h4>
<pre>org.eclipse.swt.widgets.Button btnCreateNew</pre>
</li>
<li class="blockList">
<h4>btnEdit</h4>
<pre>org.eclipse.swt.widgets.Button btnEdit</pre>
</li>
<li class="blockList">
<h4>btnRemove</h4>
<pre>org.eclipse.swt.widgets.Button btnRemove</pre>
</li>
<li class="blockList">
<h4>composite_1</h4>
<pre>org.eclipse.swt.widgets.Composite composite_1</pre>
</li>
<li class="blockList">
<h4>filters</h4>
<pre><a href="bacnet/table/core/FilterList.html" title="class in bacnet.table.core">FilterList</a> filters</pre>
</li>
<li class="blockList">
<h4>label</h4>
<pre>org.eclipse.swt.widgets.Label label</pre>
</li>
<li class="blockList">
<h4>lblElementname</h4>
<pre>org.eclipse.swt.widgets.Label lblElementname</pre>
</li>
<li class="blockList">
<h4>lblName</h4>
<pre>org.eclipse.swt.widgets.Label lblName</pre>
</li>
<li class="blockList">
<h4>lblRoworcolumn</h4>
<pre>org.eclipse.swt.widgets.Label lblRoworcolumn</pre>
</li>
<li class="blockList">
<h4>lblType</h4>
<pre>org.eclipse.swt.widgets.Label lblType</pre>
</li>
<li class="blockList">
<h4>listFilter</h4>
<pre>org.eclipse.swt.widgets.List listFilter</pre>
</li>
<li class="blockListLast">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.table.gui.HideExpressionMatrixDialog">
<!--   -->
</a>
<h3>Class <a href="bacnet/table/gui/HideExpressionMatrixDialog.html" title="class in bacnet.table.gui">bacnet.table.gui.HideExpressionMatrixDialog</a> extends org.eclipse.jface.dialogs.TitleAreaDialog implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-6890471316308295477L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnAddColumn</h4>
<pre>org.eclipse.swt.widgets.Button btnAddColumn</pre>
</li>
<li class="blockList">
<h4>btnAddRow</h4>
<pre>org.eclipse.swt.widgets.Button btnAddRow</pre>
</li>
<li class="blockList">
<h4>btnHideColumn</h4>
<pre>org.eclipse.swt.widgets.Button btnHideColumn</pre>
</li>
<li class="blockList">
<h4>btnHideRow</h4>
<pre>org.eclipse.swt.widgets.Button btnHideRow</pre>
</li>
<li class="blockList">
<h4>btnLoadColumn</h4>
<pre>org.eclipse.swt.widgets.Button btnLoadColumn</pre>
</li>
<li class="blockList">
<h4>btnLoadRow</h4>
<pre>org.eclipse.swt.widgets.Button btnLoadRow</pre>
</li>
<li class="blockList">
<h4>btnSelectGenes</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectGenes</pre>
</li>
<li class="blockList">
<h4>btnSignaturesLoader</h4>
<pre>org.eclipse.swt.widgets.Button btnSignaturesLoader</pre>
</li>
<li class="blockList">
<h4>hideDialogExcludeColumn</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; hideDialogExcludeColumn</pre>
</li>
<li class="blockList">
<h4>hideRow</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; hideRow</pre>
</li>
<li class="blockList">
<h4>listAddColumn</h4>
<pre>org.eclipse.swt.widgets.List listAddColumn</pre>
</li>
<li class="blockList">
<h4>listAddRow</h4>
<pre>org.eclipse.swt.widgets.Table listAddRow</pre>
</li>
<li class="blockList">
<h4>listHideColumn</h4>
<pre>org.eclipse.swt.widgets.List listHideColumn</pre>
</li>
<li class="blockList">
<h4>listHideRow</h4>
<pre>org.eclipse.swt.widgets.Table listHideRow</pre>
</li>
<li class="blockList">
<h4>matrix</h4>
<pre><a href="bacnet/datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a> matrix</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockListLast">
<h4>showRow</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; showRow</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.table.gui.TableFilter">
<!--   -->
</a>
<h3>Class <a href="bacnet/table/gui/TableFilter.html" title="class in bacnet.table.gui">bacnet.table.gui.TableFilter</a> extends org.eclipse.jface.viewers.ViewerFilter implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-7621715116380436406L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockListLast">
<h4>searchString</h4>
<pre>java.lang.String searchString</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.utils</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.utils.Filter">
<!--   -->
</a>
<h3>Class <a href="bacnet/utils/Filter.html" title="class in bacnet.utils">bacnet.utils.Filter</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>6834197254054420424L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>cutOff1</h4>
<pre>double cutOff1</pre>
</li>
<li class="blockList">
<h4>cutOff2</h4>
<pre>double cutOff2</pre>
</li>
<li class="blockList">
<h4>filterColumn</h4>
<pre>boolean filterColumn</pre>
</li>
<li class="blockList">
<h4>name</h4>
<pre>java.lang.String name</pre>
</li>
<li class="blockList">
<h4>tableElementName</h4>
<pre>java.lang.String tableElementName</pre>
</li>
<li class="blockListLast">
<h4>typeFilter</h4>
<pre><a href="bacnet/utils/Filter.TypeFilter.html" title="enum in bacnet.utils">Filter.TypeFilter</a> typeFilter</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;bacnet.views</h2>
<ul class="blockList">
<li class="blockList"><a id="bacnet.views.BannerView">
<!--   -->
</a>
<h3>Class <a href="bacnet/views/BannerView.html" title="class in bacnet.views">bacnet.views.BannerView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>3803453895078156914L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>canvas</h4>
<pre>org.eclipse.swt.widgets.Canvas canvas</pre>
<div class="block">In this Canvas we will draw the Banner of the website/software</div>
</li>
<li class="blockList">
<h4>tltmAbout</h4>
<pre>org.eclipse.swt.widgets.ToolItem tltmAbout</pre>
</li>
<li class="blockList">
<h4>tltmGenomics</h4>
<pre>org.eclipse.swt.widgets.ToolItem tltmGenomics</pre>
</li>
<li class="blockList">
<h4>tltmProteomics</h4>
<pre>org.eclipse.swt.widgets.ToolItem tltmProteomics</pre>
</li>
<li class="blockListLast">
<h4>tltmTranscriptomics</h4>
<pre>org.eclipse.swt.widgets.ToolItem tltmTranscriptomics</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.views.CoExprNetworkView">
<!--   -->
</a>
<h3>Class <a href="bacnet/views/CoExprNetworkView.html" title="class in bacnet.views">bacnet.views.CoExprNetworkView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-4429010895850129342L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>browserGraph</h4>
<pre>org.eclipse.swt.browser.Browser browserGraph</pre>
</li>
<li class="blockList">
<h4>browserRadial</h4>
<pre>org.eclipse.swt.browser.Browser browserRadial</pre>
</li>
<li class="blockList">
<h4>btnCorrMinus</h4>
<pre>org.eclipse.swt.widgets.Button btnCorrMinus</pre>
</li>
<li class="blockList">
<h4>btnCorrPlus</h4>
<pre>org.eclipse.swt.widgets.Button btnCorrPlus</pre>
</li>
<li class="blockList">
<h4>btnHelp</h4>
<pre>org.eclipse.swt.widgets.Button btnHelp</pre>
</li>
<li class="blockList">
<h4>btnSelectGenomeElements</h4>
<pre>org.eclipse.swt.widgets.Button btnSelectGenomeElements</pre>
</li>
<li class="blockList">
<h4>composite</h4>
<pre>org.eclipse.swt.widgets.Composite composite</pre>
</li>
<li class="blockList">
<h4>filteredNetwork</h4>
<pre><a href="bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a> filteredNetwork</pre>
</li>
<li class="blockList">
<h4>focused</h4>
<pre>boolean focused</pre>
<div class="block">Indicates if we focus the view, so we can pushState navigation</div>
</li>
<li class="blockList">
<h4>generalNetwork</h4>
<pre><a href="bacnet/datamodel/dataset/Network.html" title="class in bacnet.datamodel.dataset">Network</a> generalNetwork</pre>
</li>
<li class="blockList">
<h4>genome</h4>
<pre><a href="bacnet/datamodel/sequence/Genome.html" title="class in bacnet.datamodel.sequence">Genome</a> genome</pre>
</li>
<li class="blockList">
<h4>genomeName</h4>
<pre>java.lang.String genomeName</pre>
</li>
<li class="blockList">
<h4>listGenomeElements</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; listGenomeElements</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>svgName</h4>
<pre>java.lang.String svgName</pre>
</li>
<li class="blockList">
<h4>tabFolder</h4>
<pre>org.eclipse.swt.widgets.TabFolder tabFolder</pre>
</li>
<li class="blockList">
<h4>tableGenes</h4>
<pre>org.eclipse.swt.widgets.Table tableGenes</pre>
</li>
<li class="blockList">
<h4>tbtmGraphVisualization</h4>
<pre>org.eclipse.swt.widgets.TabItem tbtmGraphVisualization</pre>
</li>
<li class="blockList">
<h4>tbtmRadialNetwork</h4>
<pre>org.eclipse.swt.widgets.TabItem tbtmRadialNetwork</pre>
</li>
<li class="blockList">
<h4>textCutOff</h4>
<pre>org.eclipse.swt.widgets.Text textCutOff</pre>
</li>
<li class="blockListLast">
<h4>txtSearch</h4>
<pre>org.eclipse.swt.widgets.Text txtSearch</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.views.CRISPRPredictView">
<!--   -->
</a>
<h3>Class <a href="bacnet/views/CRISPRPredictView.html" title="class in bacnet.views">bacnet.views.CRISPRPredictView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>4263354745027835111L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnRunPredicition</h4>
<pre>org.eclipse.swt.widgets.Button btnRunPredicition</pre>
</li>
<li class="blockList">
<h4>btnRunTest</h4>
<pre>org.eclipse.swt.widgets.Button btnRunTest</pre>
</li>
<li class="blockList">
<h4>btnTxt</h4>
<pre>org.eclipse.swt.widgets.Button btnTxt</pre>
</li>
<li class="blockList">
<h4>btnUploadFastaFile</h4>
<pre>org.eclipse.swt.widgets.Button btnUploadFastaFile</pre>
</li>
<li class="blockList">
<h4>columnNames</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; columnNames</pre>
</li>
<li class="blockList">
<h4>combo</h4>
<pre>org.eclipse.swt.widgets.Combo combo</pre>
</li>
<li class="blockList">
<h4>DEFAULT_MODEL</h4>
<pre>java.lang.String DEFAULT_MODEL</pre>
</li>
<li class="blockList">
<h4>MODEL_CAS9</h4>
<pre>java.lang.String MODEL_CAS9</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>PYTHON_PATH</h4>
<pre>java.lang.String PYTHON_PATH</pre>
</li>
<li class="blockList">
<h4>PYTHON_SERVER</h4>
<pre>java.lang.String PYTHON_SERVER</pre>
</li>
<li class="blockList">
<h4>results</h4>
<pre>java.lang.String[][] results</pre>
</li>
<li class="blockList">
<h4>SCRIPT_PATH</h4>
<pre>java.lang.String SCRIPT_PATH</pre>
</li>
<li class="blockList">
<h4>SCRIPT_SERVER</h4>
<pre>java.lang.String SCRIPT_SERVER</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>SINGULARITY_PATH</h4>
<pre>java.lang.String SINGULARITY_PATH</pre>
</li>
<li class="blockList">
<h4>table</h4>
<pre>org.eclipse.swt.widgets.Table table</pre>
</li>
<li class="blockList">
<h4>tableViewer</h4>
<pre>org.eclipse.jface.viewers.TableViewer tableViewer</pre>
</li>
<li class="blockListLast">
<h4>text</h4>
<pre>org.eclipse.swt.widgets.Text text</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.views.CRISPRPredictView2">
<!--   -->
</a>
<h3>Class <a href="bacnet/views/CRISPRPredictView2.html" title="class in bacnet.views">bacnet.views.CRISPRPredictView2</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>3931172856313756769L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnRunPredicition</h4>
<pre>org.eclipse.swt.widgets.Button btnRunPredicition</pre>
</li>
<li class="blockList">
<h4>btnRunTest</h4>
<pre>org.eclipse.swt.widgets.Button btnRunTest</pre>
</li>
<li class="blockList">
<h4>btnUploadFastaFile</h4>
<pre>org.eclipse.swt.widgets.Button btnUploadFastaFile</pre>
</li>
<li class="blockList">
<h4>columnNames</h4>
<pre>java.util.ArrayList&lt;java.lang.String&gt; columnNames</pre>
</li>
<li class="blockList">
<h4>combo</h4>
<pre>org.eclipse.swt.widgets.Combo combo</pre>
</li>
<li class="blockList">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
<li class="blockList">
<h4>PYTHON_PATH</h4>
<pre>java.lang.String PYTHON_PATH</pre>
</li>
<li class="blockList">
<h4>PYTHON_SERVER</h4>
<pre>java.lang.String PYTHON_SERVER</pre>
</li>
<li class="blockList">
<h4>SCRIPT_PATH</h4>
<pre>java.lang.String SCRIPT_PATH</pre>
</li>
<li class="blockList">
<h4>shell</h4>
<pre>org.eclipse.swt.widgets.Shell shell</pre>
</li>
<li class="blockList">
<h4>SINGULARITY_PATH</h4>
<pre>java.lang.String SINGULARITY_PATH</pre>
</li>
<li class="blockList">
<h4>table</h4>
<pre>org.eclipse.swt.widgets.Table table</pre>
</li>
<li class="blockList">
<h4>tableViewer</h4>
<pre>org.eclipse.jface.viewers.TableViewer tableViewer</pre>
</li>
<li class="blockListLast">
<h4>text</h4>
<pre>org.eclipse.swt.widgets.Text text</pre>
</li>
</ul>
</li>
</ul>
</li>
<li class="blockList"><a id="bacnet.views.InitView">
<!--   -->
</a>
<h3>Class <a href="bacnet/views/InitView.html" title="class in bacnet.views">bacnet.views.InitView</a> extends java.lang.Object implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-2416453633652904396L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockList">
<h4>btnBHI37</h4>
<pre>org.eclipse.swt.widgets.Button btnBHI37</pre>
</li>
<li class="blockList">
<h4>btnIntracellular</h4>
<pre>org.eclipse.swt.widgets.Button btnIntracellular</pre>
</li>
<li class="blockList">
<h4>btnSrnas</h4>
<pre>org.eclipse.swt.widgets.Button btnSrnas</pre>
</li>
<li class="blockList">
<h4>btnStat</h4>
<pre>org.eclipse.swt.widgets.Button btnStat</pre>
</li>
<li class="blockList">
<h4>lblTzest</h4>
<pre>org.eclipse.swt.widgets.Label lblTzest</pre>
</li>
<li class="blockListLast">
<h4>partService</h4>
<pre>org.eclipse.e4.ui.workbench.modeling.EPartService partService</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;org.biojava3.core.exceptions</h2>
<ul class="blockList">
<li class="blockList"><a id="org.biojava3.core.exceptions.CompoundNotFoundError">
<!--   -->
</a>
<h3>Class <a href="org/biojava3/core/exceptions/CompoundNotFoundError.html" title="class in org.biojava3.core.exceptions">org.biojava3.core.exceptions.CompoundNotFoundError</a> extends java.lang.Error implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>1L</dd>
</dl>
</li>
<li class="blockList"><a id="org.biojava3.core.exceptions.FileAccessError">
<!--   -->
</a>
<h3>Class <a href="org/biojava3/core/exceptions/FileAccessError.html" title="class in org.biojava3.core.exceptions">org.biojava3.core.exceptions.FileAccessError</a> extends java.lang.Error implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>6513440232428438424L</dd>
</dl>
</li>
<li class="blockList"><a id="org.biojava3.core.exceptions.HeaderParseException">
<!--   -->
</a>
<h3>Class <a href="org/biojava3/core/exceptions/HeaderParseException.html" title="class in org.biojava3.core.exceptions">org.biojava3.core.exceptions.HeaderParseException</a> extends java.lang.Error implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-8356845980320906455L</dd>
</dl>
</li>
<li class="blockList"><a id="org.biojava3.core.exceptions.ParserException">
<!--   -->
</a>
<h3>Class <a href="org/biojava3/core/exceptions/ParserException.html" title="class in org.biojava3.core.exceptions">org.biojava3.core.exceptions.ParserException</a> extends java.lang.RuntimeException implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-4101924035353204493L</dd>
</dl>
</li>
<li class="blockList"><a id="org.biojava3.core.exceptions.SequenceLengthError">
<!--   -->
</a>
<h3>Class <a href="org/biojava3/core/exceptions/SequenceLengthError.html" title="class in org.biojava3.core.exceptions">org.biojava3.core.exceptions.SequenceLengthError</a> extends java.lang.Error implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-5486504706601790351L</dd>
</dl>
</li>
<li class="blockList"><a id="org.biojava3.core.exceptions.TranslationException">
<!--   -->
</a>
<h3>Class <a href="org/biojava3/core/exceptions/TranslationException.html" title="class in org.biojava3.core.exceptions">org.biojava3.core.exceptions.TranslationException</a> extends java.lang.RuntimeException implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>-3017433758219757440L</dd>
</dl>
</li>
</ul>
</section>
</li>
<li class="blockList">
<section role="region">
<h2 title="Package">Package&nbsp;org.biojava3.genome.parsers.gff</h2>
<ul class="blockList">
<li class="blockList"><a id="org.biojava3.genome.parsers.gff.FeatureList">
<!--   -->
</a>
<h3>Class <a href="org/biojava3/genome/parsers/gff/FeatureList.html" title="class in org.biojava3.genome.parsers.gff">org.biojava3.genome.parsers.gff.FeatureList</a> extends java.util.ArrayList&lt;<a href="org/biojava3/genome/parsers/gff/FeatureI.html" title="interface in org.biojava3.genome.parsers.gff">FeatureI</a>&gt; implements Serializable</h3>
<dl class="nameValue">
<dt>serialVersionUID:</dt>
<dd>1604195493715169313L</dd>
</dl>
<ul class="blockList">
<li class="blockList">
<h3>Serialized Fields</h3>
<ul class="blockList">
<li class="blockListLast">
<h4>mLocation</h4>
<pre><a href="org/biojava3/genome/parsers/gff/Location.html" title="class in org.biojava3.genome.parsers.gff">Location</a> mLocation</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="overview-tree.html">Tree</a></li>
<li><a href="deprecated-list.html">Deprecated</a></li>
<li><a href="index-files/index-1.html">Index</a></li>
<li><a href="help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
