<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>DataTrack</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="DataTrack";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10,"i19":10,"i20":10,"i21":10,"i22":10,"i23":10,"i24":10,"i25":10,"i26":10,"i27":10,"i28":10,"i29":10,"i30":10,"i31":10,"i32":10,"i33":10,"i34":10,"i35":10,"i36":10,"i37":10,"i38":10,"i39":10,"i40":10,"i41":10,"i42":10,"i43":10,"i44":10,"i45":10,"i46":10,"i47":10,"i48":10,"i49":10,"i50":10,"i51":10,"i52":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/DataTrack.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.genomeBrowser.core</a></div>
<h2 title="Class DataTrack" class="title">Class DataTrack</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>bacnet.genomeBrowser.core.DataTrack</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<pre>public class <span class="typeNameLabel">DataTrack</span>
extends java.lang.Object</pre>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#bioConditionHashMaps">bioConditionHashMaps</a></span></code></th>
<td class="colLast">
<div class="block">HashMap used to quickly access a specific BioCondition</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.TreeMap&lt;java.lang.String,&#8203;org.eclipse.swt.graphics.Color&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#dataColors">dataColors</a></span></code></th>
<td class="colLast">
<div class="block">Indicate the color to display for each data</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#dataNOTDisplayed">dataNOTDisplayed</a></span></code></th>
<td class="colLast">
<div class="block">List of data that should be displayed</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.Float&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#dataSizes">dataSizes</a></span></code></th>
<td class="colLast">
<div class="block">Indicate the size factor of each data</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private boolean[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#display">display</a></span></code></th>
<td class="colLast">
<div class="block">Indicate if at a specific position of the genome : data are displayed, or just leave in grey
 color</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#displayAbsoluteValue">displayAbsoluteValue</a></span></code></th>
<td class="colLast">
<div class="block">True if we display Absolute value expression (Log(Value)) False if we display Relative value
 expression (Log(FC))</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private <a href="Track.html" title="class in bacnet.genomeBrowser.core">Track</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#track">track</a></span></code></th>
<td class="colLast">
<div class="block">Parent Track</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(bacnet.genomeBrowser.core.Track)">DataTrack</a></span>&#8203;(<a href="Track.html" title="class in bacnet.genomeBrowser.core">Track</a>&nbsp;track)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#absoluteTOrelativeValue()">absoluteTOrelativeValue</a></span>()</code></th>
<td class="colLast">
<div class="block">Changed all BioCondition to display when switching from "ABSOLUTE VALUE DISPLAY3 to "RELATIVE
 VALUE DISPLAY"</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addBioCondition(java.lang.String)">addBioCondition</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">
<div class="block">Read every ExpressionData contained in bioCondName</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getBioConditionHashMaps()">getBioConditionHashMaps</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getBioCondMax(java.lang.String)">getBioCondMax</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getBioCondMin(java.lang.String)">getBioCondMin</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>float</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getBioCondSize(java.lang.String)">getBioCondSize</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">
<div class="block">Return the size ratio that have a given BioCondition</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>java.util.TreeMap&lt;java.lang.String,&#8203;org.eclipse.swt.graphics.Color&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDataColors()">getDataColors</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDataCount()">getDataCount</a></span>()</code></th>
<td class="colLast">
<div class="block">Count the number of data available, without duplicate and filtered</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDataDisplayed()">getDataDisplayed</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the list of data to display</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDataDisplayedNames()">getDataDisplayedNames</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the names of the data which will be displayd</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDataNames()">getDataNames</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the list of all the data available</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDataNOTDisplayed()">getDataNOTDisplayed</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.Float&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDataSize()">getDataSize</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.Float&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDataSizes()">getDataSizes</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>float</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDataTotalSize(bacnet.genomeBrowser.core.Track.DisplayType)">getDataTotalSize</a></span>&#8203;(<a href="Track.DisplayType.html" title="enum in bacnet.genomeBrowser.core">Track.DisplayType</a>&nbsp;type)</code></th>
<td class="colLast">
<div class="block">Add all the size of the Tracks to know the total size of the display window</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>boolean[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDisplay()">getDisplay</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDNASeqDatas()">getDNASeqDatas</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the list of SeqDatas</div>
</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDNASeqDatas(java.lang.String)">getDNASeqDatas</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">
<div class="block">Return the list of RiboSeqData given a bioCondName</div>
</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGeneExprs()">getGeneExprs</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the list of GeneExpressions</div>
</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGeneExprs(java.lang.String)">getGeneExprs</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">
<div class="block">Return the list of GeneExpression given a bioCondName</div>
</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMatrices()">getMatrices</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the list of SeqData given a bioCondName</div>
</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMatrices(java.lang.String)">getMatrices</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">
<div class="block">Return the list of SeqData given a bioCondName</div>
</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMax()">getMax</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMin()">getMin</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getNTermDatas()">getNTermDatas</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the list of NTermData given a bioCondName<br></div>
</td>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getNTermDatas(java.lang.String)">getNTermDatas</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">
<div class="block">Return the list of NTermData given a bioCondName<br></div>
</td>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteomes()">getProteomes</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the list of MassSpecData given a bioCondName<br></div>
</td>
</tr>
<tr id="i27" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteomes(java.lang.String)">getProteomes</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">
<div class="block">Return the list of MassSpecData given a bioCondName<br></div>
</td>
</tr>
<tr id="i28" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getRiboSeqDatas()">getRiboSeqDatas</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the list of SeqDatas</div>
</td>
</tr>
<tr id="i29" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getRiboSeqDatas(java.lang.String)">getRiboSeqDatas</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">
<div class="block">Return the list of RiboSeqData given a bioCondName</div>
</td>
</tr>
<tr id="i30" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getRNASeqDatas()">getRNASeqDatas</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the list of SeqDatas</div>
</td>
</tr>
<tr id="i31" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getRNASeqDatas(java.lang.String)">getRNASeqDatas</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">
<div class="block">Return the list of RNASeqData given a bioCondName</div>
</td>
</tr>
<tr id="i32" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSeqDatas(java.lang.String)">getSeqDatas</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">
<div class="block">Return the list of RNASeqData given a bioCondName</div>
</td>
</tr>
<tr id="i33" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTermSeqDatas()">getTermSeqDatas</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the list of SeqDatas</div>
</td>
</tr>
<tr id="i34" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTermSeqDatas(java.lang.String)">getTermSeqDatas</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">
<div class="block">Return the list of Term given a bioCondName</div>
</td>
</tr>
<tr id="i35" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTilings()">getTilings</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the list of Tilings</div>
</td>
</tr>
<tr id="i36" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTilings(java.lang.String)">getTilings</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">
<div class="block">Return the list of Tiling given a bioCondName</div>
</td>
</tr>
<tr id="i37" class="rowColor">
<td class="colFirst"><code><a href="Track.html" title="class in bacnet.genomeBrowser.core">Track</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTrack()">getTrack</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i38" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTSSDatas()">getTSSDatas</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the list of SeqDatas</div>
</td>
</tr>
<tr id="i39" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTSSDatas(java.lang.String)">getTSSDatas</a></span>&#8203;(java.lang.String&nbsp;bioCondName)</code></th>
<td class="colLast">
<div class="block">Return the list of TSS given a bioCondName</div>
</td>
</tr>
<tr id="i40" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#initDisplayBoolean()">initDisplayBoolean</a></span>()</code></th>
<td class="colLast">
<div class="block">Initiate Display boolean Vector when the first is added</div>
</td>
</tr>
<tr id="i41" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isDisplayAbsoluteValue()">isDisplayAbsoluteValue</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i42" class="altColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadData()">loadData</a></span>()</code></th>
<td class="colLast">
<div class="block">Load all data included in the Biological Condition (bioConditionHashMaps)</div>
</td>
</tr>
<tr id="i43" class="rowColor">
<td class="colFirst"><code>private void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#orderBioCondition()">orderBioCondition</a></span>()</code></th>
<td class="colLast">
<div class="block">Add BioCondition in BioConditionHashMaps respecting a certain order: 1 - RNASeq 2 - Tiling 3 -
 Gene Expression 4 - ExpressionMatrix 5 - TSS 6 - Proteome 7 - Nterm 8 - Unknown</div>
</td>
</tr>
<tr id="i44" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#relativeTOabsoluteValue()">relativeTOabsoluteValue</a></span>()</code></th>
<td class="colLast">
<div class="block">Changed all BioCondition to display when switching from "RELATIVE VALUE DISPLAY3 to "ABSOLUTE
 VALUE DISPLAY"</div>
</td>
</tr>
<tr id="i45" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setBioConditionHashMaps(java.util.LinkedHashMap)">setBioConditionHashMaps</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&gt;&nbsp;bioConditionHashMaps)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i46" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDataColors()">setDataColors</a></span>()</code></th>
<td class="colLast">
<div class="block">For each data assign a Color <br>
 <br>
 OVERLAYBC : Color are always the same, blue and red for tiling, black for GeneExpr, other color
 for the rest <br>
 SUPERIMPOSED, OVERLAYDATA : Color.getColors(index) is used</div>
</td>
</tr>
<tr id="i47" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDataColors(java.util.TreeMap)">setDataColors</a></span>&#8203;(java.util.TreeMap&lt;java.lang.String,&#8203;org.eclipse.swt.graphics.Color&gt;&nbsp;dataColors)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i48" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDataNOTDisplayed(java.util.ArrayList)">setDataNOTDisplayed</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;dataNOTDisplayed)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i49" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDataSizes()">setDataSizes</a></span>()</code></th>
<td class="colLast">
<div class="block">For each data assign a Size <br>
 <br>
 1.5 for Tiling array, GeneExpression array and RNASeq
 1 for TSS
 0.5 for NTerm and Proteomics data</div>
</td>
</tr>
<tr id="i50" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDisplay(boolean%5B%5D)">setDisplay</a></span>&#8203;(boolean[]&nbsp;display)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i51" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDisplayAbsoluteValue(boolean)">setDisplayAbsoluteValue</a></span>&#8203;(boolean&nbsp;displayAbsoluteValue)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i52" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setTrack(bacnet.genomeBrowser.core.Track)">setTrack</a></span>&#8203;(<a href="Track.html" title="class in bacnet.genomeBrowser.core">Track</a>&nbsp;track)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="track">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>track</h4>
<pre>private&nbsp;<a href="Track.html" title="class in bacnet.genomeBrowser.core">Track</a> track</pre>
<div class="block">Parent Track</div>
</li>
</ul>
<a id="displayAbsoluteValue">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>displayAbsoluteValue</h4>
<pre>private&nbsp;boolean displayAbsoluteValue</pre>
<div class="block">True if we display Absolute value expression (Log(Value)) False if we display Relative value
 expression (Log(FC))</div>
</li>
</ul>
<a id="display">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>display</h4>
<pre>private&nbsp;boolean[] display</pre>
<div class="block">Indicate if at a specific position of the genome : data are displayed, or just leave in grey
 color</div>
</li>
</ul>
<a id="bioConditionHashMaps">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>bioConditionHashMaps</h4>
<pre>private&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&gt; bioConditionHashMaps</pre>
<div class="block">HashMap used to quickly access a specific BioCondition</div>
</li>
</ul>
<a id="dataNOTDisplayed">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>dataNOTDisplayed</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.String&gt; dataNOTDisplayed</pre>
<div class="block">List of data that should be displayed</div>
</li>
</ul>
<a id="dataColors">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>dataColors</h4>
<pre>private&nbsp;java.util.TreeMap&lt;java.lang.String,&#8203;org.eclipse.swt.graphics.Color&gt; dataColors</pre>
<div class="block">Indicate the color to display for each data</div>
</li>
</ul>
<a id="dataSizes">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>dataSizes</h4>
<pre>private final&nbsp;java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.Float&gt; dataSizes</pre>
<div class="block">Indicate the size factor of each data</div>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;(bacnet.genomeBrowser.core.Track)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>DataTrack</h4>
<pre>public&nbsp;DataTrack&#8203;(<a href="Track.html" title="class in bacnet.genomeBrowser.core">Track</a>&nbsp;track)</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="getDataCount()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDataCount</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getDataCount()</pre>
<div class="block">Count the number of data available, without duplicate and filtered</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>filteredData</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getDataDisplayedNames()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDataDisplayedNames</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getDataDisplayedNames()</pre>
<div class="block">Return the names of the data which will be displayd</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getDataDisplayed()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDataDisplayed</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/OmicsData.html" title="class in bacnet.datamodel.dataset">OmicsData</a>&gt;&nbsp;getDataDisplayed()</pre>
<div class="block">Get the list of data to display</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getDataNames()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDataNames</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getDataNames()</pre>
<div class="block">Return the list of all the data available</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>filteredData</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="setDataColors()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDataColors</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDataColors()</pre>
<div class="block">For each data assign a Color <br>
 <br>
 OVERLAYBC : Color are always the same, blue and red for tiling, black for GeneExpr, other color
 for the rest <br>
 SUPERIMPOSED, OVERLAYDATA : Color.getColors(index) is used</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>type</code> - </dd>
</dl>
</li>
</ul>
<a id="setDataSizes()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDataSizes</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDataSizes()</pre>
<div class="block">For each data assign a Size <br>
 <br>
 <li>1.5 for Tiling array, GeneExpression array and RNASeq
 <li>1 for TSS
 <li>0.5 for NTerm and Proteomics data</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>type</code> - </dd>
</dl>
</li>
</ul>
<a id="getDataTotalSize(bacnet.genomeBrowser.core.Track.DisplayType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDataTotalSize</h4>
<pre class="methodSignature">public&nbsp;float&nbsp;getDataTotalSize&#8203;(<a href="Track.DisplayType.html" title="enum in bacnet.genomeBrowser.core">Track.DisplayType</a>&nbsp;type)</pre>
<div class="block">Add all the size of the Tracks to know the total size of the display window</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>type</code> - </dd>
</dl>
</li>
</ul>
<a id="getBioCondSize(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBioCondSize</h4>
<pre class="methodSignature">public&nbsp;float&nbsp;getBioCondSize&#8203;(java.lang.String&nbsp;bioCondName)</pre>
<div class="block">Return the size ratio that have a given BioCondition</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="initDisplayBoolean()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>initDisplayBoolean</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;initDisplayBoolean()</pre>
<div class="block">Initiate Display boolean Vector when the first is added</div>
</li>
</ul>
<a id="addBioCondition(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addBioCondition</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;addBioCondition&#8203;(java.lang.String&nbsp;bioCondName)</pre>
<div class="block">Read every ExpressionData contained in bioCondName</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dd><code>genomeSize</code> - </dd>
</dl>
</li>
</ul>
<a id="absoluteTOrelativeValue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>absoluteTOrelativeValue</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;absoluteTOrelativeValue()</pre>
<div class="block">Changed all BioCondition to display when switching from "ABSOLUTE VALUE DISPLAY3 to "RELATIVE
 VALUE DISPLAY"</div>
</li>
</ul>
<a id="relativeTOabsoluteValue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>relativeTOabsoluteValue</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;relativeTOabsoluteValue()</pre>
<div class="block">Changed all BioCondition to display when switching from "RELATIVE VALUE DISPLAY3 to "ABSOLUTE
 VALUE DISPLAY"</div>
</li>
</ul>
<a id="loadData()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadData</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;loadData()</pre>
<div class="block">Load all data included in the Biological Condition (bioConditionHashMaps)</div>
</li>
</ul>
<a id="orderBioCondition()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>orderBioCondition</h4>
<pre class="methodSignature">private&nbsp;void&nbsp;orderBioCondition()</pre>
<div class="block">Add BioCondition in BioConditionHashMaps respecting a certain order: 1 - RNASeq 2 - Tiling 3 -
 Gene Expression 4 - ExpressionMatrix 5 - TSS 6 - Proteome 7 - Nterm 8 - Unknown</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCond</code> - </dd>
</dl>
</li>
</ul>
<a id="getTilings(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTilings</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a>&gt;&nbsp;getTilings&#8203;(java.lang.String&nbsp;bioCondName)</pre>
<div class="block">Return the list of Tiling given a bioCondName</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getGeneExprs(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneExprs</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;&nbsp;getGeneExprs&#8203;(java.lang.String&nbsp;bioCondName)</pre>
<div class="block">Return the list of GeneExpression given a bioCondName</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getSeqDatas(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSeqDatas</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;getSeqDatas&#8203;(java.lang.String&nbsp;bioCondName)</pre>
<div class="block">Return the list of RNASeqData given a bioCondName</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getRNASeqDatas(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRNASeqDatas</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;getRNASeqDatas&#8203;(java.lang.String&nbsp;bioCondName)</pre>
<div class="block">Return the list of RNASeqData given a bioCondName</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getDNASeqDatas(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDNASeqDatas</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;getDNASeqDatas&#8203;(java.lang.String&nbsp;bioCondName)</pre>
<div class="block">Return the list of RiboSeqData given a bioCondName</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getRiboSeqDatas(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRiboSeqDatas</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;getRiboSeqDatas&#8203;(java.lang.String&nbsp;bioCondName)</pre>
<div class="block">Return the list of RiboSeqData given a bioCondName</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getTSSDatas(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTSSDatas</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;getTSSDatas&#8203;(java.lang.String&nbsp;bioCondName)</pre>
<div class="block">Return the list of TSS given a bioCondName</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getTermSeqDatas(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTermSeqDatas</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;getTermSeqDatas&#8203;(java.lang.String&nbsp;bioCondName)</pre>
<div class="block">Return the list of Term given a bioCondName</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getMatrices(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMatrices</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&gt;&nbsp;getMatrices&#8203;(java.lang.String&nbsp;bioCondName)</pre>
<div class="block">Return the list of SeqData given a bioCondName</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getNTermDatas(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNTermDatas</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&gt;&nbsp;getNTermDatas&#8203;(java.lang.String&nbsp;bioCondName)</pre>
<div class="block">Return the list of NTermData given a bioCondName<br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>null if no BiCond was found</dd>
</dl>
</li>
</ul>
<a id="getProteomes(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteomes</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a>&gt;&nbsp;getProteomes&#8203;(java.lang.String&nbsp;bioCondName)</pre>
<div class="block">Return the list of MassSpecData given a bioCondName<br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>null if no BiCond was found</dd>
</dl>
</li>
</ul>
<a id="getTilings()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTilings</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/Tiling.html" title="class in bacnet.datamodel.dataset">Tiling</a>&gt;&nbsp;getTilings()</pre>
<div class="block">Return the list of Tilings</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getGeneExprs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneExprs</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/GeneExpression.html" title="class in bacnet.datamodel.dataset">GeneExpression</a>&gt;&nbsp;getGeneExprs()</pre>
<div class="block">Return the list of GeneExpressions</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getRNASeqDatas()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRNASeqDatas</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;getRNASeqDatas()</pre>
<div class="block">Return the list of SeqDatas</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getDNASeqDatas()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDNASeqDatas</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;getDNASeqDatas()</pre>
<div class="block">Return the list of SeqDatas</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getRiboSeqDatas()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRiboSeqDatas</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;getRiboSeqDatas()</pre>
<div class="block">Return the list of SeqDatas</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getTSSDatas()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTSSDatas</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;getTSSDatas()</pre>
<div class="block">Return the list of SeqDatas</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getTermSeqDatas()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTermSeqDatas</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionData.html" title="class in bacnet.datamodel.dataset">ExpressionData</a>&gt;&nbsp;getTermSeqDatas()</pre>
<div class="block">Return the list of SeqDatas</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getMatrices()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMatrices</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/ExpressionMatrix.html" title="class in bacnet.datamodel.dataset">ExpressionMatrix</a>&gt;&nbsp;getMatrices()</pre>
<div class="block">Return the list of SeqData given a bioCondName</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getNTermDatas()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNTermDatas</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/NTermData.html" title="class in bacnet.datamodel.dataset">NTermData</a>&gt;&nbsp;getNTermDatas()</pre>
<div class="block">Return the list of NTermData given a bioCondName<br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>null if no BiCond was found</dd>
</dl>
</li>
</ul>
<a id="getProteomes()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteomes</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../datamodel/dataset/ProteomicsData.html" title="class in bacnet.datamodel.dataset">ProteomicsData</a>&gt;&nbsp;getProteomes()</pre>
<div class="block">Return the list of MassSpecData given a bioCondName<br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>bioCondName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>null if no BiCond was found</dd>
</dl>
</li>
</ul>
<a id="getMax()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMax</h4>
<pre class="methodSignature">public&nbsp;double&nbsp;getMax()</pre>
</li>
</ul>
<a id="getBioCondMax(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBioCondMax</h4>
<pre class="methodSignature">public&nbsp;double&nbsp;getBioCondMax&#8203;(java.lang.String&nbsp;bioCondName)</pre>
</li>
</ul>
<a id="getMin()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMin</h4>
<pre class="methodSignature">public&nbsp;double&nbsp;getMin()</pre>
</li>
</ul>
<a id="getBioCondMin(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBioCondMin</h4>
<pre class="methodSignature">public&nbsp;double&nbsp;getBioCondMin&#8203;(java.lang.String&nbsp;bioCondName)</pre>
</li>
</ul>
<a id="getTrack()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTrack</h4>
<pre class="methodSignature">public&nbsp;<a href="Track.html" title="class in bacnet.genomeBrowser.core">Track</a>&nbsp;getTrack()</pre>
</li>
</ul>
<a id="setTrack(bacnet.genomeBrowser.core.Track)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setTrack</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setTrack&#8203;(<a href="Track.html" title="class in bacnet.genomeBrowser.core">Track</a>&nbsp;track)</pre>
</li>
</ul>
<a id="getDisplay()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDisplay</h4>
<pre class="methodSignature">public&nbsp;boolean[]&nbsp;getDisplay()</pre>
</li>
</ul>
<a id="setDisplay(boolean[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDisplay</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDisplay&#8203;(boolean[]&nbsp;display)</pre>
</li>
</ul>
<a id="getDataColors()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDataColors</h4>
<pre class="methodSignature">public&nbsp;java.util.TreeMap&lt;java.lang.String,&#8203;org.eclipse.swt.graphics.Color&gt;&nbsp;getDataColors()</pre>
</li>
</ul>
<a id="setDataColors(java.util.TreeMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDataColors</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDataColors&#8203;(java.util.TreeMap&lt;java.lang.String,&#8203;org.eclipse.swt.graphics.Color&gt;&nbsp;dataColors)</pre>
</li>
</ul>
<a id="getDataSize()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDataSize</h4>
<pre class="methodSignature">public&nbsp;java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.Float&gt;&nbsp;getDataSize()</pre>
</li>
</ul>
<a id="getBioConditionHashMaps()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBioConditionHashMaps</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&gt;&nbsp;getBioConditionHashMaps()</pre>
</li>
</ul>
<a id="isDisplayAbsoluteValue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isDisplayAbsoluteValue</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;isDisplayAbsoluteValue()</pre>
</li>
</ul>
<a id="setDisplayAbsoluteValue(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDisplayAbsoluteValue</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDisplayAbsoluteValue&#8203;(boolean&nbsp;displayAbsoluteValue)</pre>
</li>
</ul>
<a id="setBioConditionHashMaps(java.util.LinkedHashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setBioConditionHashMaps</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setBioConditionHashMaps&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../datamodel/expdesign/BioCondition.html" title="class in bacnet.datamodel.expdesign">BioCondition</a>&gt;&nbsp;bioConditionHashMaps)</pre>
</li>
</ul>
<a id="getDataNOTDisplayed()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDataNOTDisplayed</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getDataNOTDisplayed()</pre>
</li>
</ul>
<a id="setDataNOTDisplayed(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDataNOTDisplayed</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDataNOTDisplayed&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;dataNOTDisplayed)</pre>
</li>
</ul>
<a id="getDataSizes()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getDataSizes</h4>
<pre class="methodSignature">public&nbsp;java.util.TreeMap&lt;java.lang.String,&#8203;java.lang.Float&gt;&nbsp;getDataSizes()</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/DataTrack.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
