<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 18:10:05 CEST 2019 -->
<title>ChromosomeBacteriaSequence</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="ChromosomeBacteriaSequence";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10,"i19":10,"i20":10,"i21":10,"i22":10,"i23":10,"i24":10,"i25":10,"i26":10,"i27":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/ChromosomeBacteriaSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.sequence</a></div>
<h2 title="Class ChromosomeBacteriaSequence" class="title">Class ChromosomeBacteriaSequence</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li><a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">org.biojava3.core.sequence.template.AbstractSequence</a>&lt;<a href="../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</li>
<li>
<ul class="inheritance">
<li><a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">org.biojava3.core.sequence.DNASequence</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">org.biojava3.core.sequence.ChromosomeSequence</a></li>
<li>
<ul class="inheritance">
<li>bacnet.datamodel.sequence.ChromosomeBacteriaSequence</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><code>java.lang.Iterable&lt;<a href="../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code>, <code><a href="../../../org/biojava3/core/sequence/template/Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</a></code>, <code><a href="../../../org/biojava3/core/sequence/template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
</dl>
<hr>
<pre>public class <span class="typeNameLabel">ChromosomeBacteriaSequence</span>
extends <a href="../../../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></pre>
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Scooter Willis
 
         Here I modified the original Chromosome class to one which include:
         coding and noncoding regions + proteins, Christophe Bï¿½cavin</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a id="nested.classes.inherited.from.class.org.biojava3.core.sequence.DNASequence">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;org.biojava3.core.sequence.<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></h3>
<code><a href="../../../org/biojava3/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="nested.classes.inherited.from.class.org.biojava3.core.sequence.template.AbstractSequence">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;org.biojava3.core.sequence.template.<a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></h3>
<code><a href="../../../org/biojava3/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></code></li>
</ul>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">ChromosomeBacteriaSequence</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.lang.String)">ChromosomeBacteriaSequence</a></span>&#8203;(java.lang.String&nbsp;seqString)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">ChromosomeBacteriaSequence</a></span>&#8203;(java.lang.String&nbsp;seqString,
                          <a href="../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader)">ChromosomeBacteriaSequence</a></span>&#8203;(<a href="../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">ChromosomeBacteriaSequence</a></span>&#8203;(<a href="../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
                          <a href="../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAsRNAs()">getAsRNAs</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getChromosomeID()">getChromosomeID</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCisRegs()">getCisRegs</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code><a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCodingRegion(java.lang.String)">getCodingRegion</a></span>&#8203;(java.lang.String&nbsp;accession)</code></th>
<td class="colLast">
<div class="block">Get the gene based on accession.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCodingSequenceHashMap()">getCodingSequenceHashMap</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getElements()">getElements</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGeneNameToLocusTagMap()">getGeneNameToLocusTagMap</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getLocusTagToGeneNameMap()">getLocusTagToGeneNameMap</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getName()">getName</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code><a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getNonCodingRegion(java.lang.String)">getNonCodingRegion</a></span>&#8203;(java.lang.String&nbsp;accession)</code></th>
<td class="colLast">
<div class="block">Get the gene based on accession.</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getNoncodingSequenceHashMap()">getNoncodingSequenceHashMap</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="Operon.html" title="class in bacnet.datamodel.sequence">Operon</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getOperons()">getOperons</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code><a href="../sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getParentGenome()">getParentGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the gene based on accession.</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getsRNAs()">getsRNAs</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadOperonsAndTerminator()">loadOperonsAndTerminator</a></span>()</code></th>
<td class="colLast">
<div class="block">Add Operons and Terminator from Annotation.EGDE_SUPPTABLE <br>
 This table comes directly from Wurtzel et al. annotation table</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadSrnas()">loadSrnas</a></span>()</code></th>
<td class="colLast">
<div class="block">Load all serialized Srna object found in Srna.PATHSerialize</div>
</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setAsRNAs(java.util.ArrayList)">setAsRNAs</a></span>&#8203;(java.util.ArrayList&lt;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;&nbsp;asRNAs)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setChromosomeID(java.lang.String)">setChromosomeID</a></span>&#8203;(java.lang.String&nbsp;chromosomeID)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setCisRegs(java.util.ArrayList)">setCisRegs</a></span>&#8203;(java.util.ArrayList&lt;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;&nbsp;cisRegs)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setCodingSequenceHashMap(java.util.LinkedHashMap)">setCodingSequenceHashMap</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;codingSequenceHashMap)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setElements(java.util.ArrayList)">setElements</a></span>&#8203;(java.util.ArrayList&lt;<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;elements)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGeneNameToLocusTagMap(java.util.LinkedHashMap)">setGeneNameToLocusTagMap</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;&nbsp;geneNameToLocusTagMap)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setLocusTagToGeneNameMap(java.util.LinkedHashMap)">setLocusTagToGeneNameMap</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;&nbsp;locusTagToGeneNameMap)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setName(java.lang.String)">setName</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setNoncodingSequenceHashMap(java.util.LinkedHashMap)">setNoncodingSequenceHashMap</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;noncodingSequenceHashMap)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setOperons(java.util.ArrayList)">setOperons</a></span>&#8203;(java.util.ArrayList&lt;<a href="Operon.html" title="class in bacnet.datamodel.sequence">Operon</a>&gt;&nbsp;operons)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setParentGenome(bacnet.datamodel.sequenceNCBI.GenomeNCBI)">setParentGenome</a></span>&#8203;(<a href="../sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;parentGenome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i27" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setsRNAs(java.util.ArrayList)">setsRNAs</a></span>&#8203;(java.util.ArrayList&lt;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;&nbsp;sRNAs)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.org.biojava3.core.sequence.ChromosomeSequence">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.biojava3.core.sequence.<a href="../../../org/biojava3/core/sequence/ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></h3>
<code><a href="../../../org/biojava3/core/sequence/ChromosomeSequence.html#addGene(org.biojava3.core.sequence.AccessionID,int,int,org.biojava3.core.sequence.Strand)">addGene</a>, <a href="../../../org/biojava3/core/sequence/ChromosomeSequence.html#getChromosomeNumber()">getChromosomeNumber</a>, <a href="../../../org/biojava3/core/sequence/ChromosomeSequence.html#getGene(java.lang.String)">getGene</a>, <a href="../../../org/biojava3/core/sequence/ChromosomeSequence.html#getGeneSequences()">getGeneSequences</a>, <a href="../../../org/biojava3/core/sequence/ChromosomeSequence.html#removeGeneSequence(java.lang.String)">removeGeneSequence</a>, <a href="../../../org/biojava3/core/sequence/ChromosomeSequence.html#setChromosomeNumber(int)">setChromosomeNumber</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.org.biojava3.core.sequence.DNASequence">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.biojava3.core.sequence.<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></h3>
<code><a href="../../../org/biojava3/core/sequence/DNASequence.html#getComplement()">getComplement</a>, <a href="../../../org/biojava3/core/sequence/DNASequence.html#getDNAType()">getDNAType</a>, <a href="../../../org/biojava3/core/sequence/DNASequence.html#getGCCount()">getGCCount</a>, <a href="../../../org/biojava3/core/sequence/DNASequence.html#getReverse()">getReverse</a>, <a href="../../../org/biojava3/core/sequence/DNASequence.html#getReverseComplement()">getReverseComplement</a>, <a href="../../../org/biojava3/core/sequence/DNASequence.html#getRNASequence()">getRNASequence</a>, <a href="../../../org/biojava3/core/sequence/DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.Frame)">getRNASequence</a>, <a href="../../../org/biojava3/core/sequence/DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)">getRNASequence</a>, <a href="../../../org/biojava3/core/sequence/DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine,org.biojava3.core.sequence.transcription.Frame)">getRNASequence</a>, <a href="../../../org/biojava3/core/sequence/DNASequence.html#main(java.lang.String%5B%5D)">main</a>, <a href="../../../org/biojava3/core/sequence/DNASequence.html#setDNAType(org.biojava3.core.sequence.DNASequence.DNAType)">setDNAType</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.org.biojava3.core.sequence.template.AbstractSequence">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.biojava3.core.sequence.template.<a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></h3>
<code><a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#addFeature(int,int,org.biojava3.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#addFeature(org.biojava3.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#addNote(java.lang.String)">addNote</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#countCompounds(C...)">countCompounds</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getAccession()">getAccession</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getAnnotationType()">getAnnotationType</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getAsList()">getAsList</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getBioBegin()">getBioBegin</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getBioEnd()">getBioEnd</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getCompoundAt(int)">getCompoundAt</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getCompoundSet()">getCompoundSet</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getDatabaseReferences()">getDatabaseReferences</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getDescription()">getDescription</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures()">getFeatures</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures(int)">getFeatures</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeatures(java.lang.String,int)">getFeatures</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getFeaturesKeyWord()">getFeaturesKeyWord</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getIndexOf(C)">getIndexOf</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getInverse()">getInverse</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getLastIndexOf(C)">getLastIndexOf</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getLength()">getLength</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getNotesList()">getNotesList</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getOriginalHeader()">getOriginalHeader</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getParentSequence()">getParentSequence</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getProxySequenceReader()">getProxySequenceReader</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceAsString()">getSequenceAsString</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)">getSequenceAsString</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSequenceScore()">getSequenceScore</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSource()">getSource</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getTaxonomy()">getTaxonomy</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#getUserCollection()">getUserCollection</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#iterator()">iterator</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#removeFeature(org.biojava3.core.sequence.features.FeatureInterface)">removeFeature</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#removeNote(java.lang.String)">removeNote</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setAccession(org.biojava3.core.sequence.AccessionID)">setAccession</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setBioBegin(java.lang.Integer)">setBioBegin</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setBioEnd(java.lang.Integer)">setBioEnd</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setDescription(java.lang.String)">setDescription</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setNotesList(java.util.ArrayList)">setNotesList</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setOriginalHeader(java.lang.String)">setOriginalHeader</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)">setParentSequence</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)">setProxySequenceReader</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setSequenceScore(java.lang.Double)">setSequenceScore</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setSource(java.lang.String)">setSource</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setTaxonomy(org.biojava3.core.sequence.TaxonomyID)">setTaxonomy</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#setUserCollection(java.util.Collection)">setUserCollection</a>, <a href="../../../org/biojava3/core/sequence/template/AbstractSequence.html#toString()">toString</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>equals, getClass, hashCode, notify, notifyAll, wait, wait, wait</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Iterable">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;java.lang.Iterable</h3>
<code>forEach, spliterator</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>ChromosomeBacteriaSequence</h4>
<pre>public&nbsp;ChromosomeBacteriaSequence()</pre>
</li>
</ul>
<a id="&lt;init&gt;(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>ChromosomeBacteriaSequence</h4>
<pre>public&nbsp;ChromosomeBacteriaSequence&#8203;(java.lang.String&nbsp;seqString)</pre>
</li>
</ul>
<a id="&lt;init&gt;(org.biojava3.core.sequence.template.SequenceReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>ChromosomeBacteriaSequence</h4>
<pre>public&nbsp;ChromosomeBacteriaSequence&#8203;(<a href="../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader)</pre>
</li>
</ul>
<a id="&lt;init&gt;(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>ChromosomeBacteriaSequence</h4>
<pre>public&nbsp;ChromosomeBacteriaSequence&#8203;(java.lang.String&nbsp;seqString,
                                  <a href="../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</pre>
</li>
</ul>
<a id="&lt;init&gt;(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>ChromosomeBacteriaSequence</h4>
<pre>public&nbsp;ChromosomeBacteriaSequence&#8203;(<a href="../../../org/biojava3/core/sequence/template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
                                  <a href="../../../org/biojava3/core/sequence/template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="../../../org/biojava3/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="loadSrnas()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadSrnas</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;loadSrnas()</pre>
<div class="block">Load all serialized Srna object found in Srna.PATHSerialize</div>
</li>
</ul>
<a id="loadOperonsAndTerminator()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadOperonsAndTerminator</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;loadOperonsAndTerminator()
                              throws java.io.IOException</pre>
<div class="block">Add Operons and Terminator from Annotation.EGDE_SUPPTABLE <br>
 This table comes directly from Wurtzel et al. annotation table</div>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.io.IOException</code></dd>
</dl>
</li>
</ul>
<a id="getCodingRegion(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCodingRegion</h4>
<pre class="methodSignature">public&nbsp;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;getCodingRegion&#8203;(java.lang.String&nbsp;accession)</pre>
<div class="block">Get the gene based on accession. Will return null if not found</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getNonCodingRegion(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNonCodingRegion</h4>
<pre class="methodSignature">public&nbsp;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;getNonCodingRegion&#8203;(java.lang.String&nbsp;accession)</pre>
<div class="block">Get the gene based on accession. Will return null if not found</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getParentGenome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getParentGenome</h4>
<pre class="methodSignature">public&nbsp;<a href="../sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;getParentGenome()</pre>
<div class="block">Get the gene based on accession. Will return null if not found</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="setParentGenome(bacnet.datamodel.sequenceNCBI.GenomeNCBI)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setParentGenome</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setParentGenome&#8203;(<a href="../sequenceNCBI/GenomeNCBI.html" title="class in bacnet.datamodel.sequenceNCBI">GenomeNCBI</a>&nbsp;parentGenome)</pre>
</li>
</ul>
<a id="getCodingSequenceHashMap()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCodingSequenceHashMap</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;getCodingSequenceHashMap()</pre>
</li>
</ul>
<a id="setCodingSequenceHashMap(java.util.LinkedHashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setCodingSequenceHashMap</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setCodingSequenceHashMap&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;codingSequenceHashMap)</pre>
</li>
</ul>
<a id="getNoncodingSequenceHashMap()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNoncodingSequenceHashMap</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;getNoncodingSequenceHashMap()</pre>
</li>
</ul>
<a id="setNoncodingSequenceHashMap(java.util.LinkedHashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setNoncodingSequenceHashMap</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setNoncodingSequenceHashMap&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;noncodingSequenceHashMap)</pre>
</li>
</ul>
<a id="getGeneNameToLocusTagMap()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneNameToLocusTagMap</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;&nbsp;getGeneNameToLocusTagMap()</pre>
</li>
</ul>
<a id="setGeneNameToLocusTagMap(java.util.LinkedHashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGeneNameToLocusTagMap</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGeneNameToLocusTagMap&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;&nbsp;geneNameToLocusTagMap)</pre>
</li>
</ul>
<a id="getLocusTagToGeneNameMap()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLocusTagToGeneNameMap</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;&nbsp;getLocusTagToGeneNameMap()</pre>
</li>
</ul>
<a id="setLocusTagToGeneNameMap(java.util.LinkedHashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setLocusTagToGeneNameMap</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setLocusTagToGeneNameMap&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;&nbsp;locusTagToGeneNameMap)</pre>
</li>
</ul>
<a id="getsRNAs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getsRNAs</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;&nbsp;getsRNAs()</pre>
</li>
</ul>
<a id="setsRNAs(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setsRNAs</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setsRNAs&#8203;(java.util.ArrayList&lt;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;&nbsp;sRNAs)</pre>
</li>
</ul>
<a id="getAsRNAs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAsRNAs</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;&nbsp;getAsRNAs()</pre>
</li>
</ul>
<a id="setAsRNAs(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAsRNAs</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setAsRNAs&#8203;(java.util.ArrayList&lt;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;&nbsp;asRNAs)</pre>
</li>
</ul>
<a id="getCisRegs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCisRegs</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;&nbsp;getCisRegs()</pre>
</li>
</ul>
<a id="setCisRegs(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setCisRegs</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setCisRegs&#8203;(java.util.ArrayList&lt;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;&nbsp;cisRegs)</pre>
</li>
</ul>
<a id="getOperons()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getOperons</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="Operon.html" title="class in bacnet.datamodel.sequence">Operon</a>&gt;&nbsp;getOperons()</pre>
</li>
</ul>
<a id="setOperons(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setOperons</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setOperons&#8203;(java.util.ArrayList&lt;<a href="Operon.html" title="class in bacnet.datamodel.sequence">Operon</a>&gt;&nbsp;operons)</pre>
</li>
</ul>
<a id="getElements()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getElements</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;getElements()</pre>
</li>
</ul>
<a id="setElements(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setElements</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setElements&#8203;(java.util.ArrayList&lt;<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;elements)</pre>
</li>
</ul>
<a id="getName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getName</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getName()</pre>
</li>
</ul>
<a id="setName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setName</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setName&#8203;(java.lang.String&nbsp;name)</pre>
</li>
</ul>
<a id="getChromosomeID()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromosomeID</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getChromosomeID()</pre>
</li>
</ul>
<a id="setChromosomeID(java.lang.String)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setChromosomeID</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setChromosomeID&#8203;(java.lang.String&nbsp;chromosomeID)</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/ChromosomeBacteriaSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
