<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 14:20:51 CEST 2019 -->
<title>GenomeNCBI</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="GenomeNCBI";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10,"i11":10,"i12":10,"i13":9,"i14":10,"i15":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GenomeNCBI.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.sequence</a></div>
<h2 title="Class GenomeNCBI" class="title">Class GenomeNCBI</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>bacnet.datamodel.sequence.GenomeNCBI</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<pre>public class <span class="typeNameLabel">GenomeNCBI</span>
extends java.lang.Object</pre>
<div class="block">Object containing an entire genome, created from NCBI RefSeq data<br>
 Data for this genome come from: <it> NCBI.fna (for the sequence) <it>
 NCBI.gff (for Gene, CDS, and ncRNA lists) If it exists some information are
 added: <it> Rfam.gff (other info) <it> supp info in a .txt file <it> supp
 info about RAST results in a .txt file For EGD-e only: <it> Srna.xml are read
 to add to the genome</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>UIBC</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#PATH_ANNOTATION">PATH_ANNOTATION</a></span></code></th>
<td class="colLast">
<div class="block">Path for all supplementary information for genome annotation :
 Database.getInstance().getPath() + "/GenomeNCBI/Annotation/";</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#PATH_GENOMES">PATH_GENOMES</a></span></code></th>
<td class="colLast">
<div class="block">Path for all Genomes : Database.getInstance().getPath() +
 "/GenomeNCBI/Genomes/";</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#PATH_GENOMES_NEW">PATH_GENOMES_NEW</a></span></code></th>
<td class="colLast">
<div class="block">Path for all new Genomes : Database.getInstance().getPath() +
 "/GenomeNCBI/GenomesNew/"; This might be used if all IDs have been changed on
 NCBI servers like it was the cas in Sept 2016</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#PATH_HOMOLOGS">PATH_HOMOLOGS</a></span></code></th>
<td class="colLast">
<div class="block">Path for all homologs calculation file : Database.getInstance().getPath() +
 "/GenomeNCBI/Homologs/";</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#PATH_PHYLOGENY">PATH_PHYLOGENY</a></span></code></th>
<td class="colLast">
<div class="block">Path for all new Genomes : Database.getInstance().getPath() +
 "/GenomeNCBI/GenomesNew/"; This might be used if all IDs have been changed on
 NCBI servers like it was the cas in Sept 2016</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#PATH_RAW">PATH_RAW</a></span></code></th>
<td class="colLast">
<div class="block">Path for all raw data for Genomes : Database.getInstance().getPath() +
 "/GenomeNCBI/";</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">GenomeNCBI</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.lang.String,boolean)">GenomeNCBI</a></span>&#8203;(java.lang.String&nbsp;genomePath,
          boolean&nbsp;annotation)</code></th>
<td class="colLast">
<div class="block">Create a genome with all chromosomes found If annotation = TRUE , we had the
 annotation we can found in other fasta files annotation = FALSE, we read only
 the complete genome + eventual plasmids</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#equals(java.lang.Object)">equals</a></span>&#8203;(java.lang.Object&nbsp;o)</code></th>
<td class="colLast">
<div class="block">Override of equals method from Object
 
 To be equal two genome has to come from the same species.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getChromosomes()">getChromosomes</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCodingSequencesList(boolean)">getCodingSequencesList</a></span>&#8203;(boolean&nbsp;allChromosomes)</code></th>
<td class="colLast">
<div class="block">Put in a list all the CodingSequence found</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code><a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGeneFromName(java.lang.String)">getGeneFromName</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Go through codingSequence and nonCodingSequence searching for corresponding
 accession or locus</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGeneName(java.lang.String)">getGeneName</a></span>&#8203;(java.lang.String&nbsp;locusTag)</code></th>
<td class="colLast">
<div class="block">Look in LocusTagToGeneName if a geneName exists</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getLocusTag(java.lang.String)">getLocusTag</a></span>&#8203;(java.lang.String&nbsp;geneName)</code></th>
<td class="colLast">
<div class="block">Look in all LocusTagToGeneName exists</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getLocusTagCodingList(boolean)">getLocusTagCodingList</a></span>&#8203;(boolean&nbsp;allChromosomes)</code></th>
<td class="colLast">
<div class="block">Put in a list all the CodingSequence found</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getLocusTagKnownCodingList(boolean)">getLocusTagKnownCodingList</a></span>&#8203;(boolean&nbsp;allChromosomes)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getLocusTagList(boolean)">getLocusTagList</a></span>&#8203;(boolean&nbsp;allChromosomes)</code></th>
<td class="colLast">
<div class="block">Put in a list all the locusTag found in the list of genes</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getLocusTagNonCodingList(boolean)">getLocusTagNonCodingList</a></span>&#8203;(boolean&nbsp;allChromosomes)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getNonCodingSequencesList(boolean)">getNonCodingSequencesList</a></span>&#8203;(boolean&nbsp;allChromosomes)</code></th>
<td class="colLast">
<div class="block">Put in a list all the NonCodingSequence found</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSpecies()">getSpecies</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSubCodingSequencesList(java.util.ArrayList,boolean)">getSubCodingSequencesList</a></span>&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;locusTag,
                         boolean&nbsp;allChromosomes)</code></th>
<td class="colLast">
<div class="block">Put in a sublist all the CodingSequence found in locuTag list</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#initStaticVariables()">initStaticVariables</a></span>()</code></th>
<td class="colLast">
<div class="block">Initi static variables after Database change</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setChromosomes(java.util.LinkedHashMap)">setChromosomes</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;&nbsp;chromosomes)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setSpecies(java.lang.String)">setSpecies</a></span>&#8203;(java.lang.String&nbsp;species)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="PATH_RAW">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PATH_RAW</h4>
<pre>public static&nbsp;java.lang.String PATH_RAW</pre>
<div class="block">Path for all raw data for Genomes : Database.getInstance().getPath() +
 "/GenomeNCBI/";</div>
</li>
</ul>
<a id="PATH_GENOMES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PATH_GENOMES</h4>
<pre>public static&nbsp;java.lang.String PATH_GENOMES</pre>
<div class="block">Path for all Genomes : Database.getInstance().getPath() +
 "/GenomeNCBI/Genomes/";</div>
</li>
</ul>
<a id="PATH_GENOMES_NEW">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PATH_GENOMES_NEW</h4>
<pre>public static&nbsp;java.lang.String PATH_GENOMES_NEW</pre>
<div class="block">Path for all new Genomes : Database.getInstance().getPath() +
 "/GenomeNCBI/GenomesNew/"; This might be used if all IDs have been changed on
 NCBI servers like it was the cas in Sept 2016</div>
</li>
</ul>
<a id="PATH_PHYLOGENY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PATH_PHYLOGENY</h4>
<pre>public static&nbsp;java.lang.String PATH_PHYLOGENY</pre>
<div class="block">Path for all new Genomes : Database.getInstance().getPath() +
 "/GenomeNCBI/GenomesNew/"; This might be used if all IDs have been changed on
 NCBI servers like it was the cas in Sept 2016</div>
</li>
</ul>
<a id="PATH_HOMOLOGS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PATH_HOMOLOGS</h4>
<pre>public static&nbsp;java.lang.String PATH_HOMOLOGS</pre>
<div class="block">Path for all homologs calculation file : Database.getInstance().getPath() +
 "/GenomeNCBI/Homologs/";</div>
</li>
</ul>
<a id="PATH_ANNOTATION">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>PATH_ANNOTATION</h4>
<pre>public static&nbsp;java.lang.String PATH_ANNOTATION</pre>
<div class="block">Path for all supplementary information for genome annotation :
 Database.getInstance().getPath() + "/GenomeNCBI/Annotation/";</div>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>GenomeNCBI</h4>
<pre>public&nbsp;GenomeNCBI()</pre>
</li>
</ul>
<a id="&lt;init&gt;(java.lang.String,boolean)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>GenomeNCBI</h4>
<pre>public&nbsp;GenomeNCBI&#8203;(java.lang.String&nbsp;genomePath,
                  boolean&nbsp;annotation)
           throws java.lang.Exception</pre>
<div class="block">Create a genome with all chromosomes found If annotation = TRUE , we had the
 annotation we can found in other fasta files annotation = FALSE, we read only
 the complete genome + eventual plasmids</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genomeName</code> - </dd>
<dd><code>annotation</code> - </dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="getCodingSequencesList(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCodingSequencesList</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;getCodingSequencesList&#8203;(boolean&nbsp;allChromosomes)</pre>
<div class="block">Put in a list all the CodingSequence found</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>allChromosomes</code> - true if we include all chromosomes</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getSubCodingSequencesList(java.util.ArrayList,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSubCodingSequencesList</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;getSubCodingSequencesList&#8203;(java.util.ArrayList&lt;java.lang.String&gt;&nbsp;locusTag,
                                                                  boolean&nbsp;allChromosomes)</pre>
<div class="block">Put in a sublist all the CodingSequence found in locuTag list</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>allChromosomes</code> - true if we include all chromosomes</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getLocusTagList(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLocusTagList</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getLocusTagList&#8203;(boolean&nbsp;allChromosomes)</pre>
<div class="block">Put in a list all the locusTag found in the list of genes</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>allChromosomes</code> - true if we include all chromosomes</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getLocusTagCodingList(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLocusTagCodingList</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getLocusTagCodingList&#8203;(boolean&nbsp;allChromosomes)</pre>
<div class="block">Put in a list all the CodingSequence found</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>allChromosomes</code> - true if we include all chromosomes</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getLocusTagKnownCodingList(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLocusTagKnownCodingList</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getLocusTagKnownCodingList&#8203;(boolean&nbsp;allChromosomes)</pre>
</li>
</ul>
<a id="getLocusTagNonCodingList(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLocusTagNonCodingList</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getLocusTagNonCodingList&#8203;(boolean&nbsp;allChromosomes)</pre>
</li>
</ul>
<a id="getNonCodingSequencesList(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNonCodingSequencesList</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&gt;&nbsp;getNonCodingSequencesList&#8203;(boolean&nbsp;allChromosomes)</pre>
<div class="block">Put in a list all the NonCodingSequence found</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>allChromosomes</code> - true if we include all chromosomes or not</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getGeneFromName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneFromName</h4>
<pre class="methodSignature">public&nbsp;<a href="../../../org/biojava3/core/sequence/DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a>&nbsp;getGeneFromName&#8203;(java.lang.String&nbsp;name)</pre>
<div class="block">Go through codingSequence and nonCodingSequence searching for corresponding
 accession or locus</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>name</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getGeneName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneName</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getGeneName&#8203;(java.lang.String&nbsp;locusTag)</pre>
<div class="block">Look in LocusTagToGeneName if a geneName exists</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>name</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getLocusTag(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLocusTag</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getLocusTag&#8203;(java.lang.String&nbsp;geneName)</pre>
<div class="block">Look in all LocusTagToGeneName exists</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>name</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="initStaticVariables()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>initStaticVariables</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;initStaticVariables()</pre>
<div class="block">Initi static variables after Database change</div>
</li>
</ul>
<a id="equals(java.lang.Object)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>equals</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;equals&#8203;(java.lang.Object&nbsp;o)</pre>
<div class="block">Override of equals method from Object
 
 To be equal two genome has to come from the same species.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code>equals</code>&nbsp;in class&nbsp;<code>java.lang.Object</code></dd>
</dl>
</li>
</ul>
<a id="getSpecies()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSpecies</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getSpecies()</pre>
</li>
</ul>
<a id="setSpecies(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSpecies</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setSpecies&#8203;(java.lang.String&nbsp;species)</pre>
</li>
</ul>
<a id="getChromosomes()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromosomes</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;&nbsp;getChromosomes()</pre>
</li>
</ul>
<a id="setChromosomes(java.util.LinkedHashMap)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setChromosomes</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setChromosomes&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="ChromosomeBacteriaSequence.html" title="class in bacnet.datamodel.sequence">ChromosomeBacteriaSequence</a>&gt;&nbsp;chromosomes)</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GenomeNCBI.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
