<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>Genome</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Genome";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":9,"i5":10,"i6":10,"i7":10,"i8":9,"i9":10,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10,"i19":9,"i20":10,"i21":10,"i22":9,"i23":9,"i24":9,"i25":9,"i26":9,"i27":9,"i28":10,"i29":10,"i30":10,"i31":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Genome.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.sequence</a></div>
<h2 title="Class Genome" class="title">Class Genome</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>bacnet.datamodel.sequence.Genome</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<pre>public class <span class="typeNameLabel">Genome</span>
extends java.lang.Object</pre>
<div class="block">Class for managing Genome<br>
 <li>Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Christophe BÃ©cavin</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="memberSummary">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="Genome.GetMultiFastaThread.html" title="class in bacnet.datamodel.sequence">Genome.GetMultiFastaThread</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="Genome.OpenGenomesThread.html" title="class in bacnet.datamodel.sequence">Genome.OpenGenomesThread</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#BACSUBTILIS_NAME">BACSUBTILIS_NAME</a></span></code></th>
<td class="colLast">
<div class="block">Name for bacillus genome</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#chromosomes">chromosomes</a></span></code></th>
<td class="colLast">
<div class="block">List of chromosomes forming the genome</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#Donovani_NAME">Donovani_NAME</a></span></code></th>
<td class="colLast">
<div class="block">Name of Leishmania reference genome</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#DP10403S_NAME">DP10403S_NAME</a></span></code></th>
<td class="colLast">
<div class="block">Name of 10403S genome</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#DP10403S_NAME_BIS">DP10403S_NAME_BIS</a></span></code></th>
<td class="colLast">
<div class="block">Other name for 10403S genome</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#ECOLI_NAME">ECOLI_NAME</a></span></code></th>
<td class="colLast">
<div class="block">Name for Ecoli genome</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#EGDC_NAME">EGDC_NAME</a></span></code></th>
<td class="colLast">
<div class="block">Name of EGD genome</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#EGDE_CHROMO_NAME">EGDE_CHROMO_NAME</a></span></code></th>
<td class="colLast">
<div class="block">Name of EGD-e chromosomegenome</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#EGDE_NAME">EGDE_NAME</a></span></code></th>
<td class="colLast">
<div class="block">Name of EGD-e genome</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#EGDE_RAST_NAME">EGDE_RAST_NAME</a></span></code></th>
<td class="colLast">
<div class="block">Name of RAST EGD-e genome</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#species">species</a></span></code></th>
<td class="colLast">
<div class="block">Name of the bacteria</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#YERSINIA_NAME">YERSINIA_NAME</a></span></code></th>
<td class="colLast">
<div class="block">Default genome for Yersinia</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">Genome</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.lang.String,boolean)">Genome</a></span>&#8203;(java.lang.String&nbsp;genomePath,
      boolean&nbsp;annotation)</code></th>
<td class="colLast">
<div class="block">Create a Genome from .fna, .annot, and all serialized Sequence object:<br>
 Read Genome sequence information in the different fna files <br>
 Load annotation information<br>
 <br>
 As we are dealing with bacteria, we are sure that the chromosome with maximum length is THE
 chromosome, other are plasmids.</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#clearGenome()">clearGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Clear current genome from JVM by setting it up to a 4 base genome</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAllElementNames()">getAllElementNames</a></span>()</code></th>
<td class="colLast">
<div class="block">Return all element names contained in all chromosomes</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAllElements()">getAllElements</a></span>()</code></th>
<td class="colLast">
<div class="block">Return all elements contained in all chromosomes</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAsRNAs()">getAsRNAs</a></span>()</code></th>
<td class="colLast">
<div class="block">Return all AsRNas in all chromosomes</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAvailableGenomes()">getAvailableGenomes</a></span>()</code></th>
<td class="colLast">
<div class="block">Read a text file containing the different genome ID available for Listeria</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getChromosomeNames()">getChromosomeNames</a></span>()</code></th>
<td class="colLast">
<div class="block">Get an ordered list of all chromosomes The order is fixed by the order of the chromosomes in the
 FNA (fasta) file used for reading the genome</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getChromosomes()">getChromosomes</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCisRegs()">getCisRegs</a></span>()</code></th>
<td class="colLast">
<div class="block">Return all CisRegs in all chromosomes</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDefautGenome()">getDefautGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Return defaut genome for a databse <br> For the moment -> first element of getAvailableGenomes()</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code><a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getElement(java.lang.String)">getElement</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Search an element by its name in the genome</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code><a href="Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getFirstChromosome()">getFirstChromosome</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the first chromosome in the HashMap
 
 Is Used to replace: Genome.loadGenome(genomeName).getChromosomes().get(0)</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code><a href="Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGeneFromName(java.lang.String)">getGeneFromName</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Go through all chromosomes and each Genes map and GeneName map to find the corresponding gene</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code><a href="Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGeneFromName(java.lang.String,java.lang.String)">getGeneFromName</a></span>&#8203;(java.lang.String&nbsp;name,
               java.lang.String&nbsp;accessionChromo)</code></th>
<td class="colLast">
<div class="block">Go through all chromosomes and each Genes map and GeneName map to find the corresponding gene</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code><a href="Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGeneFromProteinId(java.lang.String)">getGeneFromProteinId</a></span>&#8203;(java.lang.String&nbsp;proteinId)</code></th>
<td class="colLast">
<div class="block">Go through all chromosomes and Gene from corresponding proteinid</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGeneNames()">getGeneNames</a></span>()</code></th>
<td class="colLast">
<div class="block">Return all the Gene locus tag name available in every chromosome</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGenes()">getGenes</a></span>()</code></th>
<td class="colLast">
<div class="block">Return all Genes in all chromosomes</div>
</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getLengthGenome()">getLengthGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the total length of the genome by adding all chromosomes lengths</div>
</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getNcRNAs()">getNcRNAs</a></span>()</code></th>
<td class="colLast">
<div class="block">Return all NcRNAs in all chromosomes</div>
</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Operon.html" title="class in bacnet.datamodel.sequence">Operon</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getOperons()">getOperons</a></span>()</code></th>
<td class="colLast">
<div class="block">Return all Operon in all chromosomes</div>
</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteinSequenceFromFasta(java.lang.String,java.lang.String)">getProteinSequenceFromFasta</a></span>&#8203;(java.lang.String&nbsp;genomeName,
                           java.lang.String&nbsp;gene)</code></th>
<td class="colLast">
<div class="block">Read all accessible FAA files to extract protein sequence of a gene</div>
</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSpecies()">getSpecies</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getsRNAs()">getsRNAs</a></span>()</code></th>
<td class="colLast">
<div class="block">Return all Srnas in all chromosomes</div>
</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code>static <a href="Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadEgdeGenome()">loadEgdeGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Find in ModelProvider if Egde Genome is already loaded If not search EGDE_NAME in GENOMEDIR</div>
</td>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><code>static java.util.HashMap&lt;java.lang.String,&#8203;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadGeneFromProteinId(java.lang.String)">loadGeneFromProteinId</a></span>&#8203;(java.lang.String&nbsp;genomeName)</code></th>
<td class="colLast">
<div class="block">Load proteinIdnToLocusTag hashmap for HomologCreation</div>
</td>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><code>static <a href="Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadGenome(java.lang.String)">loadGenome</a></span>&#8203;(java.lang.String&nbsp;genomeAccession)</code></th>
<td class="colLast">
<div class="block">Find in ModelProvider if Genome Accession is already loaded If not search load it</div>
</td>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><code>static <a href="Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadGenome(java.lang.String,boolean)">loadGenome</a></span>&#8203;(java.lang.String&nbsp;genomeAccession,
          boolean&nbsp;keepInMemory)</code></th>
<td class="colLast">
<div class="block">Find in ModelProvider if Genome Accession is already loaded If not search load it
 If keepInMemory is true : Find in ModelProvider if Genome Accession is already loaded if not load
 it <br>
 If keepInMemory is false : Only load the Genome without saving it <br>
 <br></div>
</td>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><code>static <a href="Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#loadGenome(java.lang.String,java.lang.String,boolean,boolean)">loadGenome</a></span>&#8203;(java.lang.String&nbsp;genomeAccession,
          java.lang.String&nbsp;path,
          boolean&nbsp;keepInMemory,
          boolean&nbsp;annotation)</code></th>
<td class="colLast">
<div class="block">Load/return a genome given by a genomeAccession ID<br>
 <br>
 
 If keepInMemory is true : Find in ModelProvider if Genome Accession is already loaded if not load
 it <br>
 If keepInMemory is false : Only load the Genome without saving it <br>
 <br>
 If annotation is true : Load annotation information of the genome <br>
 If anotation is false : Only load the sequence of the genome</div>
</td>
</tr>
<tr id="i27" class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#modifyID(java.lang.String,java.lang.String)">modifyID</a></span>&#8203;(java.lang.String&nbsp;name,
        java.lang.String&nbsp;prefix)</code></th>
<td class="colLast">
<div class="block">Modify id of gene, to be sure that the suffix number is always of 4 numbers<br>
 ex: replace lmpc1 by lmpc0001, lmpc12 by lmpc0012<br></div>
</td>
</tr>
<tr id="i28" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#saveProteinIdToLocusTag()">saveProteinIdToLocusTag</a></span>()</code></th>
<td class="colLast">
<div class="block">Save proteinIdnToLocusTag in a file for HomologCreation</div>
</td>
</tr>
<tr id="i29" class="rowColor">
<td class="colFirst"><code><a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#searchElement(java.lang.String)">searchElement</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Search in locustag names, gene name, and newlocustag if a gene is present or not</div>
</td>
</tr>
<tr id="i30" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setChromosomes(java.util.LinkedHashMap)">setChromosomes</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&gt;&nbsp;chromosomes)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i31" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setSpecies(java.lang.String)">setSpecies</a></span>&#8203;(java.lang.String&nbsp;species)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="EGDE_NAME">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>EGDE_NAME</h4>
<pre>public static&nbsp;java.lang.String EGDE_NAME</pre>
<div class="block">Name of EGD-e genome</div>
</li>
</ul>
<a id="EGDE_CHROMO_NAME">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>EGDE_CHROMO_NAME</h4>
<pre>public static&nbsp;java.lang.String EGDE_CHROMO_NAME</pre>
<div class="block">Name of EGD-e chromosomegenome</div>
</li>
</ul>
<a id="EGDE_RAST_NAME">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>EGDE_RAST_NAME</h4>
<pre>public static&nbsp;java.lang.String EGDE_RAST_NAME</pre>
<div class="block">Name of RAST EGD-e genome</div>
</li>
</ul>
<a id="YERSINIA_NAME">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>YERSINIA_NAME</h4>
<pre>public static&nbsp;java.lang.String YERSINIA_NAME</pre>
<div class="block">Default genome for Yersinia</div>
</li>
</ul>
<a id="EGDC_NAME">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>EGDC_NAME</h4>
<pre>public static&nbsp;java.lang.String EGDC_NAME</pre>
<div class="block">Name of EGD genome</div>
</li>
</ul>
<a id="DP10403S_NAME">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DP10403S_NAME</h4>
<pre>public static&nbsp;java.lang.String DP10403S_NAME</pre>
<div class="block">Name of 10403S genome</div>
</li>
</ul>
<a id="DP10403S_NAME_BIS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DP10403S_NAME_BIS</h4>
<pre>public static&nbsp;java.lang.String DP10403S_NAME_BIS</pre>
<div class="block">Other name for 10403S genome</div>
</li>
</ul>
<a id="BACSUBTILIS_NAME">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>BACSUBTILIS_NAME</h4>
<pre>public static&nbsp;java.lang.String BACSUBTILIS_NAME</pre>
<div class="block">Name for bacillus genome</div>
</li>
</ul>
<a id="ECOLI_NAME">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>ECOLI_NAME</h4>
<pre>public static&nbsp;java.lang.String ECOLI_NAME</pre>
<div class="block">Name for Ecoli genome</div>
</li>
</ul>
<a id="Donovani_NAME">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>Donovani_NAME</h4>
<pre>public static&nbsp;java.lang.String Donovani_NAME</pre>
<div class="block">Name of Leishmania reference genome</div>
</li>
</ul>
<a id="species">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>species</h4>
<pre>private&nbsp;java.lang.String species</pre>
<div class="block">Name of the bacteria</div>
</li>
</ul>
<a id="chromosomes">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>chromosomes</h4>
<pre>private&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&gt; chromosomes</pre>
<div class="block">List of chromosomes forming the genome</div>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>Genome</h4>
<pre>public&nbsp;Genome()</pre>
</li>
</ul>
<a id="&lt;init&gt;(java.lang.String,boolean)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>Genome</h4>
<pre>public&nbsp;Genome&#8203;(java.lang.String&nbsp;genomePath,
              boolean&nbsp;annotation)
       throws java.lang.Exception</pre>
<div class="block">Create a Genome from .fna, .annot, and all serialized Sequence object:<br>
 <li>Read Genome sequence information in the different fna files <br>
 <li>Load annotation information<br>
 <br>
 As we are dealing with bacteria, we are sure that the chromosome with maximum length is THE
 chromosome, other are plasmids. So we search this chromosome and classify the other as plasmid by
 ordering the list.<br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genomePath</code> - </dd>
<dd><code>annotation</code> - </dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="clearGenome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>clearGenome</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;clearGenome()</pre>
<div class="block">Clear current genome from JVM by setting it up to a 4 base genome</div>
</li>
</ul>
<a id="getGeneFromName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneFromName</h4>
<pre class="methodSignature">public&nbsp;<a href="Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;getGeneFromName&#8203;(java.lang.String&nbsp;name)</pre>
<div class="block">Go through all chromosomes and each Genes map and GeneName map to find the corresponding gene</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>name</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getGeneFromName(java.lang.String,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneFromName</h4>
<pre class="methodSignature">public&nbsp;<a href="Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;getGeneFromName&#8203;(java.lang.String&nbsp;name,
                            java.lang.String&nbsp;accessionChromo)</pre>
<div class="block">Go through all chromosomes and each Genes map and GeneName map to find the corresponding gene</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>name</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getGeneFromProteinId(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneFromProteinId</h4>
<pre class="methodSignature">public&nbsp;<a href="Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&nbsp;getGeneFromProteinId&#8203;(java.lang.String&nbsp;proteinId)</pre>
<div class="block">Go through all chromosomes and Gene from corresponding proteinid</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>name</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="saveProteinIdToLocusTag()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>saveProteinIdToLocusTag</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;saveProteinIdToLocusTag()</pre>
<div class="block">Save proteinIdnToLocusTag in a file for HomologCreation</div>
</li>
</ul>
<a id="loadGeneFromProteinId(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadGeneFromProteinId</h4>
<pre class="methodSignature">public static&nbsp;java.util.HashMap&lt;java.lang.String,&#8203;java.lang.String&gt;&nbsp;loadGeneFromProteinId&#8203;(java.lang.String&nbsp;genomeName)</pre>
<div class="block">Load proteinIdnToLocusTag hashmap for HomologCreation</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genomeName</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getAllElementNames()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAllElementNames</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getAllElementNames()</pre>
<div class="block">Return all element names contained in all chromosomes</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>name</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the list of Element names</dd>
</dl>
</li>
</ul>
<a id="getAllElements()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAllElements</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&gt;&nbsp;getAllElements()</pre>
<div class="block">Return all elements contained in all chromosomes</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>name</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the list of Elements</dd>
</dl>
</li>
</ul>
<a id="getElement(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getElement</h4>
<pre class="methodSignature">public&nbsp;<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;getElement&#8203;(java.lang.String&nbsp;name)</pre>
<div class="block">Search an element by its name in the genome</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>name</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getGeneNames()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneNames</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getGeneNames()</pre>
<div class="block">Return all the Gene locus tag name available in every chromosome</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>name</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getGenes()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenes</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Gene.html" title="class in bacnet.datamodel.sequence">Gene</a>&gt;&nbsp;getGenes()</pre>
<div class="block">Return all Genes in all chromosomes</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the LinkedHashMap<String, Sequence></dd>
</dl>
</li>
</ul>
<a id="getNcRNAs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNcRNAs</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="NcRNA.html" title="class in bacnet.datamodel.sequence">NcRNA</a>&gt;&nbsp;getNcRNAs()</pre>
<div class="block">Return all NcRNAs in all chromosomes</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the LinkedHashMap<String, Sequence></dd>
</dl>
</li>
</ul>
<a id="getOperons()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getOperons</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Operon.html" title="class in bacnet.datamodel.sequence">Operon</a>&gt;&nbsp;getOperons()</pre>
<div class="block">Return all Operon in all chromosomes</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the LinkedHashMap<String, Sequence></dd>
</dl>
</li>
</ul>
<a id="getsRNAs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getsRNAs</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;&nbsp;getsRNAs()</pre>
<div class="block">Return all Srnas in all chromosomes</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the LinkedHashMap<String, Sequence></dd>
</dl>
</li>
</ul>
<a id="getAsRNAs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAsRNAs</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;&nbsp;getAsRNAs()</pre>
<div class="block">Return all AsRNas in all chromosomes</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the LinkedHashMap<String, Sequence></dd>
</dl>
</li>
</ul>
<a id="getCisRegs()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCisRegs</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a>&gt;&nbsp;getCisRegs()</pre>
<div class="block">Return all CisRegs in all chromosomes</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the LinkedHashMap<String, Sequence></dd>
</dl>
</li>
</ul>
<a id="searchElement(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>searchElement</h4>
<pre class="methodSignature">public&nbsp;<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;searchElement&#8203;(java.lang.String&nbsp;name)</pre>
<div class="block">Search in locustag names, gene name, and newlocustag if a gene is present or not</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>name</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getLengthGenome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLengthGenome</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getLengthGenome()</pre>
<div class="block">Return the total length of the genome by adding all chromosomes lengths</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="loadEgdeGenome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadEgdeGenome</h4>
<pre class="methodSignature">public static&nbsp;<a href="Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;loadEgdeGenome()</pre>
<div class="block">Find in ModelProvider if Egde Genome is already loaded If not search EGDE_NAME in GENOMEDIR</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="loadGenome(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadGenome</h4>
<pre class="methodSignature">public static&nbsp;<a href="Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;loadGenome&#8203;(java.lang.String&nbsp;genomeAccession)</pre>
<div class="block">Find in ModelProvider if Genome Accession is already loaded If not search load it</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="loadGenome(java.lang.String,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadGenome</h4>
<pre class="methodSignature">public static&nbsp;<a href="Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;loadGenome&#8203;(java.lang.String&nbsp;genomeAccession,
                                boolean&nbsp;keepInMemory)</pre>
<div class="block">Find in ModelProvider if Genome Accession is already loaded If not search load it
 If keepInMemory is true : Find in ModelProvider if Genome Accession is already loaded if not load
 it <br>
 If keepInMemory is false : Only load the Genome without saving it <br>
 <br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genomeAccession</code> - </dd>
<dd><code>keepInMemory</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.Exception</code></dd>
</dl>
</li>
</ul>
<a id="loadGenome(java.lang.String,java.lang.String,boolean,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadGenome</h4>
<pre class="methodSignature">public static&nbsp;<a href="Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;loadGenome&#8203;(java.lang.String&nbsp;genomeAccession,
                                java.lang.String&nbsp;path,
                                boolean&nbsp;keepInMemory,
                                boolean&nbsp;annotation)</pre>
<div class="block">Load/return a genome given by a genomeAccession ID<br>
 <br>
 
 If keepInMemory is true : Find in ModelProvider if Genome Accession is already loaded if not load
 it <br>
 If keepInMemory is false : Only load the Genome without saving it <br>
 <br>
 If annotation is true : Load annotation information of the genome <br>
 If anotation is false : Only load the sequence of the genome</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genomeAccession</code> - </dd>
<dd><code>keepInMemory</code> - </dd>
<dd><code>annotation</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getChromosomeNames()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromosomeNames</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getChromosomeNames()</pre>
<div class="block">Get an ordered list of all chromosomes The order is fixed by the order of the chromosomes in the
 FNA (fasta) file used for reading the genome</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getAvailableGenomes()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAvailableGenomes</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getAvailableGenomes()</pre>
<div class="block">Read a text file containing the different genome ID available for Listeria</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getDefautGenome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDefautGenome</h4>
<pre class="methodSignature">public static&nbsp;java.lang.String&nbsp;getDefautGenome()</pre>
<div class="block">Return defaut genome for a databse <br> For the moment -> first element of getAvailableGenomes()</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="modifyID(java.lang.String,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>modifyID</h4>
<pre class="methodSignature">public static&nbsp;java.lang.String&nbsp;modifyID&#8203;(java.lang.String&nbsp;name,
                                        java.lang.String&nbsp;prefix)</pre>
<div class="block">Modify id of gene, to be sure that the suffix number is always of 4 numbers<br>
 ex: replace lmpc1 by lmpc0001, lmpc12 by lmpc0012<br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>name</code> - </dd>
<dd><code>suffix</code> - used for gene name, ex: lmo, lmpc, LMRG_</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getProteinSequenceFromFasta(java.lang.String,java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinSequenceFromFasta</h4>
<pre class="methodSignature">public static&nbsp;java.lang.String&nbsp;getProteinSequenceFromFasta&#8203;(java.lang.String&nbsp;genomeName,
                                                           java.lang.String&nbsp;gene)</pre>
<div class="block">Read all accessible FAA files to extract protein sequence of a gene</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genomeName</code> - </dd>
<dd><code>gene</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getSpecies()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSpecies</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getSpecies()</pre>
</li>
</ul>
<a id="setSpecies(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSpecies</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setSpecies&#8203;(java.lang.String&nbsp;species)</pre>
</li>
</ul>
<a id="getFirstChromosome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFirstChromosome</h4>
<pre class="methodSignature">public&nbsp;<a href="Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;getFirstChromosome()</pre>
<div class="block">Get the first chromosome in the HashMap
 
 Is Used to replace: Genome.loadGenome(genomeName).getChromosomes().get(0)</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getChromosomes()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromosomes</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&gt;&nbsp;getChromosomes()</pre>
</li>
</ul>
<a id="setChromosomes(java.util.LinkedHashMap)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setChromosomes</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setChromosomes&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;<a href="Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&gt;&nbsp;chromosomes)</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Genome.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
