<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>Uses of Package bacnet.datamodel.sequence</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Package bacnet.datamodel.sequence";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h1 title="Uses of Package bacnet.datamodel.sequence" class="title">Uses of Package<br>bacnet.datamodel.sequence</h1>
</div>
<div class="contentContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="package-summary.html">bacnet.datamodel.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet">bacnet</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.annotation">bacnet.datamodel.annotation</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.dataset">bacnet.datamodel.dataset</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.expdesign">bacnet.datamodel.expdesign</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.proteomics">bacnet.datamodel.proteomics</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.sequence">bacnet.datamodel.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.datamodel.sequenceNCBI">bacnet.datamodel.sequenceNCBI</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.e4.rap">bacnet.e4.rap</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.expressionAtlas.core">bacnet.expressionAtlas.core</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.genomeBrowser">bacnet.genomeBrowser</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.genomeBrowser.core">bacnet.genomeBrowser.core</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.genomeBrowser.tracksGUI">bacnet.genomeBrowser.tracksGUI</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.reader">bacnet.reader</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.blast">bacnet.scripts.blast</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.core">bacnet.scripts.core</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.database">bacnet.scripts.database</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.genome">bacnet.scripts.genome</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.listeriomics">bacnet.scripts.listeriomics</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.listeriomics.nterm">bacnet.scripts.listeriomics.nterm</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.listeriomics.srna">bacnet.scripts.listeriomics.srna</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.scripts.phylogeny">bacnet.scripts.phylogeny</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.sequenceTools">bacnet.sequenceTools</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.table">bacnet.table</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.table.gui">bacnet.table.gui</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#bacnet.utils">bacnet.utils</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#bacnet.views">bacnet.views</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../package-summary.html">bacnet</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.datamodel.annotation">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../annotation/package-summary.html">bacnet.datamodel.annotation</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Chromosome.html#bacnet.datamodel.annotation">Chromosome</a></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/ChromosomeBacteriaSequence.html#bacnet.datamodel.annotation">ChromosomeBacteriaSequence</a></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.datamodel.annotation">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Sequence.html#bacnet.datamodel.annotation">Sequence</a></th>
<td class="colLast">
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.datamodel.dataset">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../dataset/package-summary.html">bacnet.datamodel.dataset</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.datamodel.dataset">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.datamodel.expdesign">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../expdesign/package-summary.html">bacnet.datamodel.expdesign</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.datamodel.expdesign">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.datamodel.proteomics">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../proteomics/package-summary.html">bacnet.datamodel.proteomics</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Chromosome.html#bacnet.datamodel.proteomics">Chromosome</a></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.datamodel.proteomics">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Sequence.html#bacnet.datamodel.proteomics">Sequence</a></th>
<td class="colLast">
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.datamodel.sequence">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="package-summary.html">bacnet.datamodel.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Chromosome.html#bacnet.datamodel.sequence">Chromosome</a></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Gene.html#bacnet.datamodel.sequence">Gene</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.datamodel.sequence">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/NcRNA.html#bacnet.datamodel.sequence">NcRNA</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/NcRNA.TypeNcRNA.html#bacnet.datamodel.sequence">NcRNA.TypeNcRNA</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Operon.html#bacnet.datamodel.sequence">Operon</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Sequence.html#bacnet.datamodel.sequence">Sequence</a></th>
<td class="colLast">
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Sequence.SeqType.html#bacnet.datamodel.sequence">Sequence.SeqType</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Srna.html#bacnet.datamodel.sequence">Srna</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Srna.TypeSrna.html#bacnet.datamodel.sequence">Srna.TypeSrna</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.datamodel.sequenceNCBI">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../sequenceNCBI/package-summary.html">bacnet.datamodel.sequenceNCBI</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/ChromosomeBacteriaSequence.html#bacnet.datamodel.sequenceNCBI">ChromosomeBacteriaSequence</a></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Gene.html#bacnet.datamodel.sequenceNCBI">Gene</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.datamodel.sequenceNCBI">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/NcRNA.html#bacnet.datamodel.sequenceNCBI">NcRNA</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Operon.html#bacnet.datamodel.sequenceNCBI">Operon</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Sequence.html#bacnet.datamodel.sequenceNCBI">Sequence</a></th>
<td class="colLast">
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Srna.html#bacnet.datamodel.sequenceNCBI">Srna</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.e4.rap">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../e4/rap/package-summary.html">bacnet.e4.rap</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.e4.rap">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.expressionAtlas.core">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../expressionAtlas/core/package-summary.html">bacnet.expressionAtlas.core</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.expressionAtlas.core">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Sequence.html#bacnet.expressionAtlas.core">Sequence</a></th>
<td class="colLast">
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.genomeBrowser">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../genomeBrowser/package-summary.html">bacnet.genomeBrowser</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.genomeBrowser">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.genomeBrowser.core">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../genomeBrowser/core/package-summary.html">bacnet.genomeBrowser.core</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Chromosome.html#bacnet.genomeBrowser.core">Chromosome</a></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.genomeBrowser.core">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.genomeBrowser.tracksGUI">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../genomeBrowser/tracksGUI/package-summary.html">bacnet.genomeBrowser.tracksGUI</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Chromosome.html#bacnet.genomeBrowser.tracksGUI">Chromosome</a></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Sequence.html#bacnet.genomeBrowser.tracksGUI">Sequence</a></th>
<td class="colLast">
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.reader">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../reader/package-summary.html">bacnet.reader</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Chromosome.html#bacnet.reader">Chromosome</a></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/ChromosomeBacteriaSequence.html#bacnet.reader">ChromosomeBacteriaSequence</a></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Gene.html#bacnet.reader">Gene</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.reader">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.scripts.blast">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../scripts/blast/package-summary.html">bacnet.scripts.blast</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Sequence.html#bacnet.scripts.blast">Sequence</a></th>
<td class="colLast">
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.scripts.core">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../scripts/core/package-summary.html">bacnet.scripts.core</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.scripts.core">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Sequence.html#bacnet.scripts.core">Sequence</a></th>
<td class="colLast">
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.scripts.database">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../scripts/database/package-summary.html">bacnet.scripts.database</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.scripts.database">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.scripts.genome">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../scripts/genome/package-summary.html">bacnet.scripts.genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Gene.html#bacnet.scripts.genome">Gene</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.scripts.genome">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Operon.html#bacnet.scripts.genome">Operon</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Sequence.html#bacnet.scripts.genome">Sequence</a></th>
<td class="colLast">
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Srna.html#bacnet.scripts.genome">Srna</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Srna.TypeSrna.html#bacnet.scripts.genome">Srna.TypeSrna</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.scripts.listeriomics">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../scripts/listeriomics/package-summary.html">bacnet.scripts.listeriomics</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.scripts.listeriomics">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.scripts.listeriomics.nterm">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../scripts/listeriomics/nterm/package-summary.html">bacnet.scripts.listeriomics.nterm</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Gene.html#bacnet.scripts.listeriomics.nterm">Gene</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.scripts.listeriomics.nterm">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Sequence.html#bacnet.scripts.listeriomics.nterm">Sequence</a></th>
<td class="colLast">
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.scripts.listeriomics.srna">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../scripts/listeriomics/srna/package-summary.html">bacnet.scripts.listeriomics.srna</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Gene.html#bacnet.scripts.listeriomics.srna">Gene</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.scripts.listeriomics.srna">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Srna.html#bacnet.scripts.listeriomics.srna">Srna</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.scripts.phylogeny">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../scripts/phylogeny/package-summary.html">bacnet.scripts.phylogeny</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Sequence.html#bacnet.scripts.phylogeny">Sequence</a></th>
<td class="colLast">
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.sequenceTools">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../sequenceTools/package-summary.html">bacnet.sequenceTools</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Gene.html#bacnet.sequenceTools">Gene</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.sequenceTools">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Sequence.html#bacnet.sequenceTools">Sequence</a></th>
<td class="colLast">
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Srna.html#bacnet.sequenceTools">Srna</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Srna.TypeSrna.html#bacnet.sequenceTools">Srna.TypeSrna</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.table">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../table/package-summary.html">bacnet.table</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.table">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.table.gui">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../table/gui/package-summary.html">bacnet.table.gui</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.table.gui">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.utils">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../utils/package-summary.html">bacnet.utils</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Sequence.html#bacnet.utils">Sequence</a></th>
<td class="colLast">
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="bacnet.views">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">bacnet.datamodel.sequence</a> used by <a href="../../views/package-summary.html">bacnet.views</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Genome.html#bacnet.views">Genome</a></th>
<td class="colLast">
<div class="block">Class for managing Genome<br>
 Contains all Genes, non-coding RNA and all other elements found in the genome GFF
 <a href="../../reader/GFFNCBIReader.html" title="class in bacnet.reader"><code>GFFNCBIReader</code></a></div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
