<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>Sequence</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Sequence";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10,"i19":10,"i20":10,"i21":10,"i22":10,"i23":10,"i24":9,"i25":10,"i26":9,"i27":10,"i28":10,"i29":10,"i30":10,"i31":10,"i32":10,"i33":10,"i34":9,"i35":9,"i36":9,"i37":9,"i38":10,"i39":9,"i40":9,"i41":10,"i42":10,"i43":10,"i44":10,"i45":10,"i46":10,"i47":10,"i48":10,"i49":10,"i50":10,"i51":10,"i52":10,"i53":10,"i54":10,"i55":10,"i56":10,"i57":10,"i58":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Sequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.datamodel.sequence</a></div>
<h2 title="Class Sequence" class="title">Class Sequence</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>bacnet.datamodel.sequence.Sequence</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><code>java.io.Serializable</code></dd>
</dl>
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><code><a href="Gene.html" title="class in bacnet.datamodel.sequence">Gene</a></code>, <code><a href="../proteomics/NTerm.html" title="class in bacnet.datamodel.proteomics">NTerm</a></code>, <code><a href="Operon.html" title="class in bacnet.datamodel.sequence">Operon</a></code>, <code><a href="Srna.html" title="class in bacnet.datamodel.sequence">Srna</a></code>, <code><a href="../proteomics/TIS.html" title="class in bacnet.datamodel.proteomics">TIS</a></code></dd>
</dl>
<hr>
<pre>public class <span class="typeNameLabel">Sequence</span>
extends java.lang.Object
implements java.io.Serializable</pre>
<div class="block">Object for manipulating part of Sequence.<br>
 Main attributes are <code>name, begin, end, strand, genome, and type</code><br>
 <br>
 
 SeqType {Gene,NcRNA,Operon,Srna,ASrna,terminator,unknown}<br>
 <br>
 
 for each <code>Sequence</code> we force the condition: <br>
 <code>this.begin inf_to this.end</code></div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Christophe BÃ©cavin</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../serialized-form.html#bacnet.datamodel.sequence.Sequence">Serialized Form</a></dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="memberSummary">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#ANTI_SD_SEQ">ANTI_SD_SEQ</a></span></code></th>
<td class="colLast">
<div class="block">Anti-SD sequence taken from Correlations between Shine-Dalgarno Sequences and Gene Features Such
 as Predicted Expression Levels and Operon Structures Jiong Ma, Allan Campbell, and Samuel Karlin
 J.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#begin">begin</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#chromosomeID">chromosomeID</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#comment">comment</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#conservation">conservation</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#conservationHashMap">conservationHashMap</a></span></code></th>
<td class="colLast">
<div class="block">HashMap linking genomeName to conserved gene<br>
 Give information about the genome in which this gene is conserved</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#DEFAULT_LENGTH">DEFAULT_LENGTH</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#end">end</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#features">features</a></span></code></th>
<td class="colLast">
<div class="block">List of features</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#genomeName">genomeName</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#id">id</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#length">length</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#name">name</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#ref">ref</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private static long</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#serialVersionUID">serialVersionUID</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.TreeSet&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#signatures">signatures</a></span></code></th>
<td class="colLast">
<div class="block">List of signatures ID which this Sequence is part of</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private char</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#strand">strand</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.TreeSet&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#synonym">synonym</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private <a href="Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#type">type</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">Sequence</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.lang.String,int,int)">Sequence</a></span>&#8203;(java.lang.String&nbsp;name,
        int&nbsp;begin,
        int&nbsp;end)</code></th>
<td class="colLast">
<div class="block">Create a Sequence object<br>
 if(begin>end) <br>
 this.end = begin <br>
 this.begin = end <br>
 this.strand = '-' <br></div>
</td>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.lang.String,int,int,char)">Sequence</a></span>&#8203;(java.lang.String&nbsp;name,
        int&nbsp;begin,
        int&nbsp;end,
        char&nbsp;strand)</code></th>
<td class="colLast">
<div class="block">Create a Sequence with a specific name, from, to and strand <br>
 CAREFUL: <br>
 from < to ALWAYS<br>
 sense is given by strand<br>
 If to < from it will correct automatically the position and strand, to have from < to</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#foundSDSequence()">foundSDSequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Search the consensus SD sequence in the 20bp upstream to a <code>Sequence</code></div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#get3UTRSequence(int)">get3UTRSequence</a></span>&#8203;(int&nbsp;sizeUTR)</code></th>
<td class="colLast">
<div class="block">Get the nucleotide sequence n bp downstream to the <code>Sequence</code>, +3bp for including
 start codon</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getBegin()">getBegin</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code><a href="Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getChromosome()">getChromosome</a></span>()</code></th>
<td class="colLast">
<div class="block">Return corresponding Chromosome given by this.getChromosomeNb<br>
 Load it if necessary</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getChromosomeID()">getChromosomeID</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getComment()">getComment</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getConservation()">getConservation</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getConservationHashMap()">getConservationHashMap</a></span>()</code></th>
<td class="colLast">
<div class="block">Get HashMap linking genomeName to conserved gene<br>
 Give information about the genome in which this gene is conserved</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getEnd()">getEnd</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getFeature(java.lang.String)">getFeature</a></span>&#8203;(java.lang.String&nbsp;attribute)</code></th>
<td class="colLast">
<div class="block">Get the feature in the HashMap, pointed by the key attribute<br>
 return "" if not found</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getFeatures()">getFeatures</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getFeaturesText()">getFeaturesText</a></span>()</code></th>
<td class="colLast">
<div class="block">Create a String containing all features and their attribute</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getFeaturesTextForTable(java.lang.String)">getFeaturesTextForTable</a></span>&#8203;(java.lang.String&nbsp;ref)</code></th>
<td class="colLast">
<div class="block">Used when displaying information in <code>SrnaView</code><br>
 Display only supplementary information of each Srna publication</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code><a href="Genome.html" title="class in bacnet.datamodel.sequence">Genome</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGenome()">getGenome</a></span>()</code></th>
<td class="colLast">
<div class="block">Return corresponding Genome given by this.genomeName<br>
 Load it if necessary</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGenomeName()">getGenomeName</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getId()">getId</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getLength()">getLength</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getName()">getName</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getRef()">getRef</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSDSequence()">getSDSequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the nucleotide sequence 20bp upstream to the <code>Sequence</code>, +3bp for including start
 codon</div>
</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSDSequence(int)">getSDSequence</a></span>&#8203;(int&nbsp;sizeSD)</code></th>
<td class="colLast">
<div class="block">Get the nucleotide sequence n bp upstream to the <code>Sequence</code>, +3bp for including start
 codon</div>
</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSequence()">getSequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Read chromosome and extract the corresponding sequence as a String</div>
</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSequenceAA()">getSequenceAA</a></span>()</code></th>
<td class="colLast">
<div class="block">Get Amino Acid sequence of a Sequence</div>
</td>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSequenceAACurated()">getSequenceAACurated</a></span>()</code></th>
<td class="colLast">
<div class="block">When an Amino Acid sequence contains Stop codon ('*'), this method search the first start codon
 and remove sequence before, and then create different sequence, from one stop codon to the other.</div>
</td>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><code>static java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSequenceAACurated(java.lang.String)">getSequenceAACurated</a></span>&#8203;(java.lang.String&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">When an Amino Acid sequence contains Stop codon ('*'), this method search the first start codon
 and remove sequence before, and then create different sequence, from one stop codon to the other.</div>
</td>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSequenceRNA()">getSequenceRNA</a></span>()</code></th>
<td class="colLast">
<div class="block">Return RNA sequence of the <code>Sequence</code></div>
</td>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><code>static long</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSerialversionuid()">getSerialversionuid</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i27" class="rowColor">
<td class="colFirst"><code>java.util.TreeSet&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSignatures()">getSignatures</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i28" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSignaturesToString()">getSignaturesToString</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i29" class="rowColor">
<td class="colFirst"><code>char</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getStrand()">getStrand</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i30" class="altColor">
<td class="colFirst"><code>java.util.TreeSet&lt;java.lang.String&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSynonym()">getSynonym</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i31" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSynonymsText()">getSynonymsText</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i32" class="altColor">
<td class="colFirst"><code><a href="Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getType()">getType</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i33" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isEqual(bacnet.datamodel.sequence.Sequence)">isEqual</a></span>&#8203;(<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">If :
 sequence.getBegin()==this.getBegin()
 && sequence.getEnd()==this.getEnd()
 && sequence.getStrand()==this.getStrand()<br>
 return true</div>
</td>
</tr>
<tr id="i34" class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isOverlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)">isOverlap</a></span>&#8203;(<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
         <a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2)</code></th>
<td class="colLast">
<div class="block">Decide if two genes are overlapping <br>
 First, calculate the overlapping size of two genes, finding the gene upstream to the other<br>
 If overlap is negative it will mean that we have no overlap (gene1.end < gene2.begin) <br></div>
</td>
</tr>
<tr id="i35" class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isOverlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence,int)">isOverlap</a></span>&#8203;(<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
         <a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2,
         int&nbsp;overlapCutoff)</code></th>
<td class="colLast">
<div class="block">Decide if two genes are overlapping <br>
 First, calculate the overlapping size of two genes, finding the gene upstream to the other<br>
 If overlap is negative it will mean that we have no overlap (gene1.end < gene2.begin) <br></div>
</td>
</tr>
<tr id="i36" class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isOverlapStart(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence,int)">isOverlapStart</a></span>&#8203;(<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
              <a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2,
              int&nbsp;sizeStart)</code></th>
<td class="colLast">
<div class="block">Decide if first Sequence is overlapping the start region of the second <br></div>
</td>
</tr>
<tr id="i37" class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isSameStrand(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)">isSameStrand</a></span>&#8203;(<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
            <a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2)</code></th>
<td class="colLast">
<div class="block">Decide if two sequences have same strand</div>
</td>
</tr>
<tr id="i38" class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isStrand()">isStrand</a></span>()</code></th>
<td class="colLast">
<div class="block">Return True if strand=='+'</div>
</td>
</tr>
<tr id="i39" class="rowColor">
<td class="colFirst"><code>static <a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#load(java.lang.String)">load</a></span>&#8203;(java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Read compressed, serialized data with a FileInputStream.</div>
</td>
</tr>
<tr id="i40" class="altColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#overlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)">overlap</a></span>&#8203;(<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
       <a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2)</code></th>
<td class="colLast">
<div class="block">Calculate the overlapping size of two genes <br>
 First, find the gene upstream to the other<br>
 Then, as we got gene1.begin < gene2.begin, overlap = gene1.end - gene2.begin<br>
 overlap can be negative, it will mean that we have no overlap (gene1.end < gene2.begin) <br>
 or overlap can be higher than gene2.length, in that case overlap = gene2.length</div>
</td>
</tr>
<tr id="i41" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#save(java.lang.String)">save</a></span>&#8203;(java.lang.String&nbsp;fileName)</code></th>
<td class="colLast">
<div class="block">Serialize the vector of lines with an ObjectOutputStream.</div>
</td>
</tr>
<tr id="i42" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setBegin(int)">setBegin</a></span>&#8203;(int&nbsp;begin)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i43" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setChromosomeID(java.lang.String)">setChromosomeID</a></span>&#8203;(java.lang.String&nbsp;chromosomeID)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i44" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setComment(java.lang.String)">setComment</a></span>&#8203;(java.lang.String&nbsp;comment)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i45" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setConservation(int)">setConservation</a></span>&#8203;(int&nbsp;conservation)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i46" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setConservationHashMap(java.util.LinkedHashMap)">setConservationHashMap</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;&nbsp;conservationHashMap)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i47" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setEnd(int)">setEnd</a></span>&#8203;(int&nbsp;end)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i48" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setFeatures(java.util.LinkedHashMap)">setFeatures</a></span>&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;&nbsp;features)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i49" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setGenomeName(java.lang.String)">setGenomeName</a></span>&#8203;(java.lang.String&nbsp;genome)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i50" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setId(java.lang.String)">setId</a></span>&#8203;(java.lang.String&nbsp;id)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i51" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setLength(int)">setLength</a></span>&#8203;(int&nbsp;length)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i52" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setName(java.lang.String)">setName</a></span>&#8203;(java.lang.String&nbsp;name)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i53" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setRef(java.lang.String)">setRef</a></span>&#8203;(java.lang.String&nbsp;ref)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i54" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setSignatures(java.util.TreeSet)">setSignatures</a></span>&#8203;(java.util.TreeSet&lt;java.lang.String&gt;&nbsp;signatures)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i55" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setStrand(char)">setStrand</a></span>&#8203;(char&nbsp;strand)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i56" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setSynonym(java.util.TreeSet)">setSynonym</a></span>&#8203;(java.util.TreeSet&lt;java.lang.String&gt;&nbsp;synonym)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i57" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setType(bacnet.datamodel.sequence.Sequence.SeqType)">setType</a></span>&#8203;(<a href="Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a>&nbsp;type)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i58" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#toString()">toString</a></span>()</code></th>
<td class="colLast">
<div class="block">Print different <br></div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="serialVersionUID">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>serialVersionUID</h4>
<pre>private static final&nbsp;long serialVersionUID</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../constant-values.html#bacnet.datamodel.sequence.Sequence.serialVersionUID">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a id="DEFAULT_LENGTH">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DEFAULT_LENGTH</h4>
<pre>public static&nbsp;int DEFAULT_LENGTH</pre>
</li>
</ul>
<a id="ANTI_SD_SEQ">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>ANTI_SD_SEQ</h4>
<pre>public static&nbsp;java.lang.String ANTI_SD_SEQ</pre>
<div class="block">Anti-SD sequence taken from Correlations between Shine-Dalgarno Sequences and Gene Features Such
 as Predicted Expression Levels and Operon Structures Jiong Ma, Allan Campbell, and Samuel Karlin
 J. Bacteriol. October 2002 vol. 184 no. 20 5733-5745</div>
</li>
</ul>
<a id="name">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>name</h4>
<pre>private&nbsp;java.lang.String name</pre>
</li>
</ul>
<a id="type">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>type</h4>
<pre>private&nbsp;<a href="Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a> type</pre>
</li>
</ul>
<a id="id">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>id</h4>
<pre>private&nbsp;java.lang.String id</pre>
</li>
</ul>
<a id="begin">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>begin</h4>
<pre>private&nbsp;int begin</pre>
</li>
</ul>
<a id="end">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>end</h4>
<pre>private&nbsp;int end</pre>
</li>
</ul>
<a id="strand">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>strand</h4>
<pre>private&nbsp;char strand</pre>
</li>
</ul>
<a id="length">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>length</h4>
<pre>private&nbsp;int length</pre>
</li>
</ul>
<a id="conservation">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>conservation</h4>
<pre>private&nbsp;int conservation</pre>
</li>
</ul>
<a id="comment">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>comment</h4>
<pre>private&nbsp;java.lang.String comment</pre>
</li>
</ul>
<a id="ref">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>ref</h4>
<pre>private&nbsp;java.lang.String ref</pre>
</li>
</ul>
<a id="genomeName">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>genomeName</h4>
<pre>private&nbsp;java.lang.String genomeName</pre>
</li>
</ul>
<a id="chromosomeID">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>chromosomeID</h4>
<pre>private&nbsp;java.lang.String chromosomeID</pre>
</li>
</ul>
<a id="synonym">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>synonym</h4>
<pre>private&nbsp;java.util.TreeSet&lt;java.lang.String&gt; synonym</pre>
</li>
</ul>
<a id="features">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>features</h4>
<pre>private&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt; features</pre>
<div class="block">List of features</div>
</li>
</ul>
<a id="conservationHashMap">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>conservationHashMap</h4>
<pre>private&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt; conservationHashMap</pre>
<div class="block">HashMap linking genomeName to conserved gene<br>
 Give information about the genome in which this gene is conserved</div>
</li>
</ul>
<a id="signatures">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>signatures</h4>
<pre>private&nbsp;java.util.TreeSet&lt;java.lang.String&gt; signatures</pre>
<div class="block">List of signatures ID which this Sequence is part of</div>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>Sequence</h4>
<pre>public&nbsp;Sequence()</pre>
</li>
</ul>
<a id="&lt;init&gt;(java.lang.String,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>Sequence</h4>
<pre>public&nbsp;Sequence&#8203;(java.lang.String&nbsp;name,
                int&nbsp;begin,
                int&nbsp;end)</pre>
<div class="block">Create a Sequence object<br>
 if(begin>end) <br>
 this.end = begin <br>
 this.begin = end <br>
 this.strand = '-' <br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>name</code> - of the sequence</dd>
<dd><code>begin</code> - </dd>
<dd><code>end</code> - </dd>
</dl>
</li>
</ul>
<a id="&lt;init&gt;(java.lang.String,int,int,char)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>Sequence</h4>
<pre>public&nbsp;Sequence&#8203;(java.lang.String&nbsp;name,
                int&nbsp;begin,
                int&nbsp;end,
                char&nbsp;strand)</pre>
<div class="block">Create a Sequence with a specific name, from, to and strand <br>
 CAREFUL: <br>
 from < to ALWAYS<br>
 sense is given by strand<br>
 If to < from it will correct automatically the position and strand, to have from < to</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>name</code> - </dd>
<dd><code>begin</code> - </dd>
<dd><code>end</code> - </dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="isStrand()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isStrand</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;isStrand()</pre>
<div class="block">Return True if strand=='+'</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequence</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getSequence()</pre>
<div class="block">Read chromosome and extract the corresponding sequence as a String</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getSequenceRNA()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceRNA</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getSequenceRNA()</pre>
<div class="block">Return RNA sequence of the <code>Sequence</code></div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getSequenceAA()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAA</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getSequenceAA()</pre>
<div class="block">Get Amino Acid sequence of a Sequence</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getSequenceAACurated()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAACurated</h4>
<pre class="methodSignature">public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getSequenceAACurated()</pre>
<div class="block">When an Amino Acid sequence contains Stop codon ('*'), this method search the first start codon
 and remove sequence before, and then create different sequence, from one stop codon to the other.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getSequenceAACurated(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAACurated</h4>
<pre class="methodSignature">public static&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;getSequenceAACurated&#8203;(java.lang.String&nbsp;sequence)</pre>
<div class="block">When an Amino Acid sequence contains Stop codon ('*'), this method search the first start codon
 and remove sequence before, and then create different sequence, from one stop codon to the other.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getGenome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenome</h4>
<pre class="methodSignature">public&nbsp;<a href="Genome.html" title="class in bacnet.datamodel.sequence">Genome</a>&nbsp;getGenome()</pre>
<div class="block">Return corresponding Genome given by this.genomeName<br>
 Load it if necessary</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getChromosome()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromosome</h4>
<pre class="methodSignature">public&nbsp;<a href="Chromosome.html" title="class in bacnet.datamodel.sequence">Chromosome</a>&nbsp;getChromosome()</pre>
<div class="block">Return corresponding Chromosome given by this.getChromosomeNb<br>
 Load it if necessary</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getFeaturesText()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeaturesText</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getFeaturesText()</pre>
<div class="block">Create a String containing all features and their attribute</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getFeaturesTextForTable(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeaturesTextForTable</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getFeaturesTextForTable&#8203;(java.lang.String&nbsp;ref)</pre>
<div class="block">Used when displaying information in <code>SrnaView</code><br>
 Display only supplementary information of each Srna publication</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>ref</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getSynonymsText()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSynonymsText</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getSynonymsText()</pre>
</li>
</ul>
<a id="getSDSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSDSequence</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getSDSequence()</pre>
<div class="block">Get the nucleotide sequence 20bp upstream to the <code>Sequence</code>, +3bp for including start
 codon</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getSDSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSDSequence</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getSDSequence&#8203;(int&nbsp;sizeSD)</pre>
<div class="block">Get the nucleotide sequence n bp upstream to the <code>Sequence</code>, +3bp for including start
 codon</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sizeSD</code> - the number of base pair to include before start codon</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="get3UTRSequence(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>get3UTRSequence</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;get3UTRSequence&#8203;(int&nbsp;sizeUTR)</pre>
<div class="block">Get the nucleotide sequence n bp downstream to the <code>Sequence</code>, +3bp for including
 start codon</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sizeSD</code> - the number of base pair to include before start codon</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="foundSDSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>foundSDSequence</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;foundSDSequence()</pre>
<div class="block">Search the consensus SD sequence in the 20bp upstream to a <code>Sequence</code></div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="overlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>overlap</h4>
<pre class="methodSignature">public static&nbsp;int&nbsp;overlap&#8203;(<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
                          <a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2)</pre>
<div class="block">Calculate the overlapping size of two genes <br>
 First, find the gene upstream to the other<br>
 Then, as we got gene1.begin < gene2.begin, overlap = gene1.end - gene2.begin<br>
 overlap can be negative, it will mean that we have no overlap (gene1.end < gene2.begin) <br>
 or overlap can be higher than gene2.length, in that case overlap = gene2.length</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gene1</code> - </dd>
<dd><code>gene2</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="isOverlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isOverlap</h4>
<pre class="methodSignature">public static&nbsp;boolean&nbsp;isOverlap&#8203;(<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
                                <a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2)</pre>
<div class="block">Decide if two genes are overlapping <br>
 First, calculate the overlapping size of two genes, finding the gene upstream to the other<br>
 If overlap is negative it will mean that we have no overlap (gene1.end < gene2.begin) <br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gene1</code> - </dd>
<dd><code>gene2</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="isOverlap(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isOverlap</h4>
<pre class="methodSignature">public static&nbsp;boolean&nbsp;isOverlap&#8203;(<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
                                <a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2,
                                int&nbsp;overlapCutoff)</pre>
<div class="block">Decide if two genes are overlapping <br>
 First, calculate the overlapping size of two genes, finding the gene upstream to the other<br>
 If overlap is negative it will mean that we have no overlap (gene1.end < gene2.begin) <br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gene1</code> - </dd>
<dd><code>gene2</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="isOverlapStart(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isOverlapStart</h4>
<pre class="methodSignature">public static&nbsp;boolean&nbsp;isOverlapStart&#8203;(<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
                                     <a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2,
                                     int&nbsp;sizeStart)</pre>
<div class="block">Decide if first Sequence is overlapping the start region of the second <br></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gene1</code> - <code>Sequence</code> to test for overlapping</dd>
<dd><code>gene2</code> - <code>Sequence</code> which start region will be extracted</dd>
<dd><code>sizeStart</code> - number of base pair to integrate in the "start region of gene2</dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="isSameStrand(bacnet.datamodel.sequence.Sequence,bacnet.datamodel.sequence.Sequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isSameStrand</h4>
<pre class="methodSignature">public static&nbsp;boolean&nbsp;isSameStrand&#8203;(<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene1,
                                   <a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;gene2)</pre>
<div class="block">Decide if two sequences have same strand</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>gene1</code> - </dd>
<dd><code>gene2</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="isEqual(bacnet.datamodel.sequence.Sequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isEqual</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;isEqual&#8203;(<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;sequence)</pre>
<div class="block">If :
 <li>sequence.getBegin()==this.getBegin()
 <li>&& sequence.getEnd()==this.getEnd()
 <li>&& sequence.getStrand()==this.getStrand()<br>
 return true</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sequence</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;toString()</pre>
<div class="block">Print different <br></div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code>toString</code>&nbsp;in class&nbsp;<code>java.lang.Object</code></dd>
</dl>
</li>
</ul>
<a id="load(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>load</h4>
<pre class="methodSignature">public static&nbsp;<a href="Sequence.html" title="class in bacnet.datamodel.sequence">Sequence</a>&nbsp;load&#8203;(java.lang.String&nbsp;fileName)</pre>
<div class="block">Read compressed, serialized data with a FileInputStream. Uncompress that data with a
 GZIPInputStream. Deserialize the vector of lines with a ObjectInputStream. Replace current data
 with new data, and redraw everything.</div>
</li>
</ul>
<a id="save(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>save</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;save&#8203;(java.lang.String&nbsp;fileName)</pre>
<div class="block">Serialize the vector of lines with an ObjectOutputStream. Compress the serialized objects with a
 GZIPOutputStream. Write the compressed, serialized data to a file with a FileOutputStream. Don't
 forget to flush and close the stream.</div>
</li>
</ul>
<a id="getName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getName</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getName()</pre>
</li>
</ul>
<a id="setName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setName</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setName&#8203;(java.lang.String&nbsp;name)</pre>
</li>
</ul>
<a id="getType()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getType</h4>
<pre class="methodSignature">public&nbsp;<a href="Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a>&nbsp;getType()</pre>
</li>
</ul>
<a id="setType(bacnet.datamodel.sequence.Sequence.SeqType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setType</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setType&#8203;(<a href="Sequence.SeqType.html" title="enum in bacnet.datamodel.sequence">Sequence.SeqType</a>&nbsp;type)</pre>
</li>
</ul>
<a id="getId()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getId</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getId()</pre>
</li>
</ul>
<a id="setId(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setId</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setId&#8203;(java.lang.String&nbsp;id)</pre>
</li>
</ul>
<a id="getBegin()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBegin</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getBegin()</pre>
</li>
</ul>
<a id="setBegin(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setBegin</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setBegin&#8203;(int&nbsp;begin)</pre>
</li>
</ul>
<a id="getEnd()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getEnd</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getEnd()</pre>
</li>
</ul>
<a id="setEnd(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setEnd</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setEnd&#8203;(int&nbsp;end)</pre>
</li>
</ul>
<a id="getStrand()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getStrand</h4>
<pre class="methodSignature">public&nbsp;char&nbsp;getStrand()</pre>
</li>
</ul>
<a id="setStrand(char)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setStrand</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setStrand&#8203;(char&nbsp;strand)</pre>
</li>
</ul>
<a id="getLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLength</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getLength()</pre>
</li>
</ul>
<a id="setLength(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setLength</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setLength&#8203;(int&nbsp;length)</pre>
</li>
</ul>
<a id="getConservation()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getConservation</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getConservation()</pre>
</li>
</ul>
<a id="setConservation(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setConservation</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setConservation&#8203;(int&nbsp;conservation)</pre>
</li>
</ul>
<a id="getComment()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getComment</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getComment()</pre>
</li>
</ul>
<a id="setComment(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setComment</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setComment&#8203;(java.lang.String&nbsp;comment)</pre>
</li>
</ul>
<a id="getRef()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRef</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getRef()</pre>
</li>
</ul>
<a id="setRef(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setRef</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setRef&#8203;(java.lang.String&nbsp;ref)</pre>
</li>
</ul>
<a id="getGenomeName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomeName</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getGenomeName()</pre>
</li>
</ul>
<a id="setGenomeName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGenomeName</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setGenomeName&#8203;(java.lang.String&nbsp;genome)</pre>
</li>
</ul>
<a id="getChromosomeID()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromosomeID</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getChromosomeID()</pre>
</li>
</ul>
<a id="setChromosomeID(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setChromosomeID</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setChromosomeID&#8203;(java.lang.String&nbsp;chromosomeID)</pre>
</li>
</ul>
<a id="getSynonym()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSynonym</h4>
<pre class="methodSignature">public&nbsp;java.util.TreeSet&lt;java.lang.String&gt;&nbsp;getSynonym()</pre>
</li>
</ul>
<a id="setSynonym(java.util.TreeSet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSynonym</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setSynonym&#8203;(java.util.TreeSet&lt;java.lang.String&gt;&nbsp;synonym)</pre>
</li>
</ul>
<a id="getConservationHashMap()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getConservationHashMap</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;&nbsp;getConservationHashMap()</pre>
<div class="block">Get HashMap linking genomeName to conserved gene<br>
 Give information about the genome in which this gene is conserved</div>
</li>
</ul>
<a id="setConservationHashMap(java.util.LinkedHashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setConservationHashMap</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setConservationHashMap&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;&nbsp;conservationHashMap)</pre>
</li>
</ul>
<a id="getFeatures()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeatures</h4>
<pre class="methodSignature">public&nbsp;java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;&nbsp;getFeatures()</pre>
</li>
</ul>
<a id="getFeature(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeature</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getFeature&#8203;(java.lang.String&nbsp;attribute)</pre>
<div class="block">Get the feature in the HashMap, pointed by the key attribute<br>
 return "" if not found</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>attribute</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="setFeatures(java.util.LinkedHashMap)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setFeatures</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setFeatures&#8203;(java.util.LinkedHashMap&lt;java.lang.String,&#8203;java.lang.String&gt;&nbsp;features)</pre>
</li>
</ul>
<a id="getSignatures()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSignatures</h4>
<pre class="methodSignature">public&nbsp;java.util.TreeSet&lt;java.lang.String&gt;&nbsp;getSignatures()</pre>
</li>
</ul>
<a id="getSignaturesToString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSignaturesToString</h4>
<pre class="methodSignature">public&nbsp;java.lang.String&nbsp;getSignaturesToString()</pre>
</li>
</ul>
<a id="setSignatures(java.util.TreeSet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSignatures</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setSignatures&#8203;(java.util.TreeSet&lt;java.lang.String&gt;&nbsp;signatures)</pre>
</li>
</ul>
<a id="getSerialversionuid()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getSerialversionuid</h4>
<pre class="methodSignature">public static&nbsp;long&nbsp;getSerialversionuid()</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Sequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
