<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Fri Dec 10 09:15:34 CET 2021 -->
<title>ProteinTools</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-10">
<meta name="description" content="declaration: package: bacnet.scripts.blast, class: ProteinTools">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var evenRowColor = "even-row-color";
var oddRowColor = "odd-row-color";
var tableTab = "table-tab";
var activeTableTab = "active-table-tab";
var pathtoroot = "../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/ProteinTools.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html#class">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div>
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field-summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor-summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li><a href="#field-detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor-detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">bacnet.scripts.blast</a></div>
<h1 title="Class ProteinTools" class="title">Class ProteinTools</h1>
</div>
<div class="inheritance" title="Inheritance Tree"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">java.lang.Object</a>
<div class="inheritance">bacnet.scripts.blast.ProteinTools</div>
</div>
<section class="class-description" id="class-description">
<hr>
<div class="type-signature"><span class="modifiers">public class </span><span class="element-name type-name-label">ProteinTools</span>
<span class="extends-implements">extends <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></span></div>
</section>
<section class="summary">
<ul class="summary-list">
<!-- =========== FIELD SUMMARY =========== -->
<li>
<section class="field-summary" id="field-summary">
<h2>Field Summary</h2>
<div class="caption"><span>Fields</span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Field</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><code><a href="#DATABASE" class="member-name-link">DATABASE</a></code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static int</code></div>
<div class="col-second odd-row-color"><code><a href="#LENGTH_CUTOFF" class="member-name-link">LENGTH_CUTOFF</a></code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<li>
<section class="constructor-summary" id="constructor-summary">
<h2>Constructor Summary</h2>
<div class="caption"><span>Constructors</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E()" class="member-name-link">ProteinTools</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab1" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab1', 3)" class="table-tab">Static Methods</button><button id="method-summary-table-tab4" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab4', 3)" class="table-tab">Concrete Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel">
<div class="summary-table three-column-summary" aria-labelledby="method-summary-table-tab0">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#createDataBases()" class="member-name-link">createDataBases</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Create:<br>
 Protein DB
 UTR DB
 fusion Protein DB and Protein DB
 Genome DB
 TIS DB</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#createGenomeDB()" class="member-name-link">createGenomeDB</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">lists of small peptides contained in the genome of EGDe on 6frames.<br>
 The small peptides include in the list are the one only of size more than 5aa, and surrounded by
 at least one R.<br>
 example: Start - peptide - R<br>
 Stop - peptide - R<br>
 <br>
 Element like: Start - peptide - Stop are rejected from the analysis<br>
 <br>
 frame 1 = EGDe genome, ID=genome
 frame 2 = EGDe genome 1bp on the left, ID=genomeM
 frame 3 = EGDe genome 1bp on the right, ID=genomeP
 frame 4 = EGDe reverse complement genome, ID=genomeCompl
 frame 5 = EGDe reverse complement genome 1bp on the left, ID=genomeComplM
 frame 6 = EGDe reverse complement genome 1bp on the right, ID=genomeComplP <br></div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#createMDB()" class="member-name-link">createMDB</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#createProteinDB()" class="member-name-link">createProteinDB</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Create a fasta file which will be used has a database<br>
 This file contains:
 all genes
 all new genes described in RAST
 all Srna on 3 frames
 all ASrna on 3 frames (300bp long) lists of Genes, ID=lmo, directly taken from NCBI lists of
 "new"Gene, ID=rastlmo, from the analysis of annotation I just did using RAST software lists of
 Srna, ID=rli, rliM, rliP, we don't know where is the begining so we have 3 frame for each rli, we
 start at the first start codon, plus we cut when we found stop codon and peptide smaller than
 5aa. lists of ASrna, ID=anti, antiM, antiP, we don't know where is the begining so we have 3
 frame for each asRNA, we start at the first start codon, plus we cut when we found stop codon and
 peptide smaller than 5aa.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#createTISDB()" class="member-name-link">createTISDB</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Create database of all peptide from a start codon (M=ATG, L=TTG, V=GTG), to the next stop codon
 or R</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#createUTRDB()" class="member-name-link">createUTRDB</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Create a database with all UTR from Wurtzel et al. 2012<br>
 Found here : DATABASE+"table_s1_tss_table.txt"<br>
 We found the correct frame so we are sure to have the start codon of the gene at the exact
 position of NCBI annotation We start at the first start codon, then if stop codon are found in
 the UTR region we cut in different peptides, remove ones &lt; 5aa, and paste to the last part of the
 UTR the whole gene amino acide sequence all gene, ID=lmo_n n being the id of the UTR peptide
 extracted</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#divideTISDB()" class="member-name-link">divideTISDB</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Divide TISDB into sequence starting with a M (TISMDB), and other not starting with an M
 (TISnoMDB).</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#fusionProteinUTRDB()" class="member-name-link">fusionProteinUTRDB</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Fusion ProteinDB and UTRDB in one file</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getAllPeptides(java.lang.String,java.util.ArrayList,java.util.ArrayList,char,java.lang.String,int,boolean,int)" class="member-name-link">getAllPeptides</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;startOccurences,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;&nbsp;peptides,
 char&nbsp;codon,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;codonNucleotide,
 int&nbsp;genomeSize,
 boolean&nbsp;strand,
 int&nbsp;cutoff)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Given a genome sequence, and a list of position of start codon, update a list of peptides
 accordingly<br>
 The new peptide, are the peptide from a start codon to the next R or stop codon</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#possibleSmallRNAPeptides()" class="member-name-link">possibleSmallRNAPeptides</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Get all possible peptides from EGD-e: <br>
 get all proteins sequence from Small RNAs
 get peptide sequence from M start to first R, or first E
 cut-off every peptide of size under 5aa
 export in a table
 use <code>http://web.expasy.org/compute_pi/</code> to calculate mass and isoelectric point
 save the different mass in NTermUtils.getPATH()+EGDePeptideMass.txt</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#statDatabase()" class="member-name-link">statDatabase</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Extract statistics from existing databases, number of lmos, sRNA, utr...</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#verifyTISDB()" class="member-name-link">verifyTISDB</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Verify TISDB peptides:<br>
 Each peptide might have a R at the end, or no R at all
 Every EGD-e gene should be in this database</div>
</div>
</div>
</div>
</div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Object">Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)" title="class or interface in java.lang" class="external-link">equals</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#getClass()" title="class or interface in java.lang" class="external-link">getClass</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode()" title="class or interface in java.lang" class="external-link">hashCode</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notify()" title="class or interface in java.lang" class="external-link">notify</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notifyAll()" title="class or interface in java.lang" class="external-link">notifyAll</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#toString()" title="class or interface in java.lang" class="external-link">toString</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait()" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long)" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long,int)" title="class or interface in java.lang" class="external-link">wait</a></code></div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ============ FIELD DETAIL =========== -->
<li>
<section class="field-details" id="field-detail">
<h2>Field Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="DATABASE">
<h3>DATABASE</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name">DATABASE</span></div>
</section>
</li>
<li>
<section class="detail" id="LENGTH_CUTOFF">
<h3>LENGTH_CUTOFF</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name">LENGTH_CUTOFF</span></div>
</section>
</li>
</ul>
</section>
</li>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<li>
<section class="constructor-details" id="constructor-detail">
<h2>Constructor Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="&lt;init&gt;()">
<h3>ProteinTools</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name">ProteinTools</span>()</div>
</section>
</li>
</ul>
</section>
</li>
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="createDataBases()">
<h3>createDataBases</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">createDataBases</span>()</div>
<div class="block">Create:<br>
 <li>Protein DB
 <li>UTR DB
 <li>fusion Protein DB and Protein DB
 <li>Genome DB
 <li>TIS DB</div>
</section>
</li>
<li>
<section class="detail" id="fusionProteinUTRDB()">
<h3>fusionProteinUTRDB</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">fusionProteinUTRDB</span>()</div>
<div class="block">Fusion ProteinDB and UTRDB in one file</div>
</section>
</li>
<li>
<section class="detail" id="createProteinDB()">
<h3>createProteinDB</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">createProteinDB</span>()</div>
<div class="block">Create a fasta file which will be used has a database<br>
 This file contains:
 <li>all genes
 <li>all new genes described in RAST
 <li>all Srna on 3 frames
 <li>all ASrna on 3 frames (300bp long) lists of Genes, ID=lmo, directly taken from NCBI lists of
 "new"Gene, ID=rastlmo, from the analysis of annotation I just did using RAST software lists of
 Srna, ID=rli, rliM, rliP, we don't know where is the begining so we have 3 frame for each rli, we
 start at the first start codon, plus we cut when we found stop codon and peptide smaller than
 5aa. lists of ASrna, ID=anti, antiM, antiP, we don't know where is the begining so we have 3
 frame for each asRNA, we start at the first start codon, plus we cut when we found stop codon and
 peptide smaller than 5aa.</div>
</section>
</li>
<li>
<section class="detail" id="createUTRDB()">
<h3>createUTRDB</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">createUTRDB</span>()</div>
<div class="block">Create a database with all UTR from Wurtzel et al. 2012<br>
 Found here : DATABASE+"table_s1_tss_table.txt"<br>
 We found the correct frame so we are sure to have the start codon of the gene at the exact
 position of NCBI annotation We start at the first start codon, then if stop codon are found in
 the UTR region we cut in different peptides, remove ones &lt; 5aa, and paste to the last part of the
 UTR the whole gene amino acide sequence all gene, ID=lmo_n n being the id of the UTR peptide
 extracted</div>
</section>
</li>
<li>
<section class="detail" id="createGenomeDB()">
<h3>createGenomeDB</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">createGenomeDB</span>()</div>
<div class="block">lists of small peptides contained in the genome of EGDe on 6frames.<br>
 The small peptides include in the list are the one only of size more than 5aa, and surrounded by
 at least one R.<br>
 example: Start - peptide - R<br>
 Stop - peptide - R<br>
 <br>
 Element like: Start - peptide - Stop are rejected from the analysis<br>
 <br>
 <li>frame 1 = EGDe genome, ID=genome
 <li>frame 2 = EGDe genome 1bp on the left, ID=genomeM
 <li>frame 3 = EGDe genome 1bp on the right, ID=genomeP
 <li>frame 4 = EGDe reverse complement genome, ID=genomeCompl
 <li>frame 5 = EGDe reverse complement genome 1bp on the left, ID=genomeComplM
 <li>frame 6 = EGDe reverse complement genome 1bp on the right, ID=genomeComplP <br></div>
</section>
</li>
<li>
<section class="detail" id="createTISDB()">
<h3>createTISDB</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">createTISDB</span>()</div>
<div class="block">Create database of all peptide from a start codon (M=ATG, L=TTG, V=GTG), to the next stop codon
 or R</div>
</section>
</li>
<li>
<section class="detail" id="divideTISDB()">
<h3>divideTISDB</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">divideTISDB</span>()</div>
<div class="block">Divide TISDB into sequence starting with a M (TISMDB), and other not starting with an M
 (TISnoMDB).</div>
</section>
</li>
<li>
<section class="detail" id="getAllPeptides(java.lang.String,java.util.ArrayList,java.util.ArrayList,char,java.lang.String,int,boolean,int)">
<h3>getAllPeptides</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">getAllPeptides</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;startOccurences,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;&nbsp;peptides,
 char&nbsp;codon,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;codonNucleotide,
 int&nbsp;genomeSize,
 boolean&nbsp;strand,
 int&nbsp;cutoff)</span></div>
<div class="block">Given a genome sequence, and a list of position of start codon, update a list of peptides
 accordingly<br>
 The new peptide, are the peptide from a start codon to the next R or stop codon</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>sequence</code> - </dd>
<dd><code>startOccurences</code> - </dd>
<dd><code>peptides</code> - </dd>
<dd><code>codon</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="verifyTISDB()">
<h3>verifyTISDB</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">verifyTISDB</span>()</div>
<div class="block">Verify TISDB peptides:<br>
 <li>Each peptide might have a R at the end, or no R at all
 <li>Every EGD-e gene should be in this database</div>
</section>
</li>
<li>
<section class="detail" id="createMDB()">
<h3>createMDB</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">createMDB</span>()</div>
</section>
</li>
<li>
<section class="detail" id="possibleSmallRNAPeptides()">
<h3>possibleSmallRNAPeptides</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">possibleSmallRNAPeptides</span>()</div>
<div class="block">Get all possible peptides from EGD-e: <br>
 <li>get all proteins sequence from Small RNAs
 <li>get peptide sequence from M start to first R, or first E
 <li>cut-off every peptide of size under 5aa
 <li>export in a table
 <li>use <code>http://web.expasy.org/compute_pi/</code> to calculate mass and isoelectric point
 <li>save the different mass in NTermUtils.getPATH()+EGDePeptideMass.txt</div>
</section>
</li>
<li>
<section class="detail" id="statDatabase()">
<h3>statDatabase</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">statDatabase</span>()</div>
<div class="block">Extract statistics from existing databases, number of lmos, sRNA, utr...</div>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
</div>
</div>
</body>
</html>
