<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Thu Apr 11 18:10:05 CEST 2019 -->
<title>ProteinTools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2019-04-11">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
<script type="text/javascript" src="../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="ProteinTools";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":9,"i1":9,"i2":9,"i3":9,"i4":9,"i5":9,"i6":9,"i7":9,"i8":9,"i9":9,"i10":9,"i11":9};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/ProteinTools.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">bacnet.scripts.blast</a></div>
<h2 title="Class ProteinTools" class="title">Class ProteinTools</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>bacnet.scripts.blast.ProteinTools</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<pre>public class <span class="typeNameLabel">ProteinTools</span>
extends java.lang.Object</pre>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#DATABASE">DATABASE</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#LENGTH_CUTOFF">LENGTH_CUTOFF</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">ProteinTools</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createDataBases()">createDataBases</a></span>()</code></th>
<td class="colLast">
<div class="block">Create:<br>
 Protein DB
 UTR DB
 fusion Protein DB and Protein DB
 Genome DB
 TIS DB</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createGenomeDB()">createGenomeDB</a></span>()</code></th>
<td class="colLast">
<div class="block">lists of small peptides contained in the genome of EGDe on 6frames.<br>
 The small peptides include in the list are the one only of size more than
 5aa, and surrounded by at least one R.<br>
 example: Start - peptide - R<br>
 Stop - peptide - R<br>
 <br>
 Element like: Start - peptide - Stop are rejected from the analysis<br>
 <br>
 frame 1 = EGDe genome, ID=genome
 frame 2 = EGDe genome 1bp on the left, ID=genomeM
 frame 3 = EGDe genome 1bp on the right, ID=genomeP
 frame 4 = EGDe reverse complement genome, ID=genomeCompl
 frame 5 = EGDe reverse complement genome 1bp on the left, ID=genomeComplM
 frame 6 = EGDe reverse complement genome 1bp on the right,
 ID=genomeComplP <br></div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createMDB()">createMDB</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createProteinDB()">createProteinDB</a></span>()</code></th>
<td class="colLast">
<div class="block">Create a fasta file which will be used has a database<br>
 This file contains:
 all genes
 all new genes described in RAST
 all Srna on 3 frames
 all ASrna on 3 frames (300bp long) lists of Genes, ID=lmo, directly taken
 from NCBI lists of "new"Gene, ID=rastlmo, from the analysis of annotation I
 just did using RAST software lists of Srna, ID=rli, rliM, rliP, we don't know
 where is the begining so we have 3 frame for each rli, we start at the first
 start codon, plus we cut when we found stop codon and peptide smaller than
 5aa. lists of ASrna, ID=anti, antiM, antiP, we don't know where is the
 begining so we have 3 frame for each asRNA, we start at the first start
 codon, plus we cut when we found stop codon and peptide smaller than 5aa.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createTISDB()">createTISDB</a></span>()</code></th>
<td class="colLast">
<div class="block">Create database of all peptide from a start codon (M=ATG, L=TTG, V=GTG), to
 the next stop codon or R</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#createUTRDB()">createUTRDB</a></span>()</code></th>
<td class="colLast">
<div class="block">Create a database with all UTR from Wurtzel et al. 2012<br>
 Found here : DATABASE+"table_s1_tss_table.txt"<br>
 We found the correct frame so we are sure to have the start codon of the gene
 at the exact position of NCBI annotation We start at the first start codon,
 then if stop codon are found in the UTR region we cut in different peptides,
 remove ones < 5aa, and paste to the last part of the UTR the whole gene amino
 acide sequence all gene, ID=lmo_n n being the id of the UTR peptide extracted</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#divideTISDB()">divideTISDB</a></span>()</code></th>
<td class="colLast">
<div class="block">Divide TISDB into sequence starting with a M (TISMDB), and other not starting
 with an M (TISnoMDB).</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#fusionProteinUTRDB()">fusionProteinUTRDB</a></span>()</code></th>
<td class="colLast">
<div class="block">Fusion ProteinDB and UTRDB in one file</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAllPeptides(java.lang.String,java.util.ArrayList,java.util.ArrayList,char,java.lang.String,int,boolean,int)">getAllPeptides</a></span>&#8203;(java.lang.String&nbsp;sequence,
              java.util.ArrayList&lt;java.lang.Integer&gt;&nbsp;startOccurences,
              java.util.ArrayList&lt;java.lang.String&gt;&nbsp;peptides,
              char&nbsp;codon,
              java.lang.String&nbsp;codonNucleotide,
              int&nbsp;genomeSize,
              boolean&nbsp;strand,
              int&nbsp;cutoff)</code></th>
<td class="colLast">
<div class="block">Given a genome sequence, and a list of position of start codon, update a list
 of peptides accordingly<br>
 The new peptide, are the peptide from a start codon to the next R or stop
 codon</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#possibleSmallRNAPeptides()">possibleSmallRNAPeptides</a></span>()</code></th>
<td class="colLast">
<div class="block">Get all possible peptides from EGD-e: <br>
 get all proteins sequence from Small RNAs
 get peptide sequence from M start to first R, or first E
 cut-off every peptide of size under 5aa
 export in a table
 use <code>http://web.expasy.org/compute_pi/</code> to calculate mass and
 isoelectric point
 save the different mass in NTermUtils.getPATH()+EGDePeptideMass.txt</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#statDatabase()">statDatabase</a></span>()</code></th>
<td class="colLast">
<div class="block">Extract statistics from existing databases, number of lmos, sRNA, utr...</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#verifyTISDB()">verifyTISDB</a></span>()</code></th>
<td class="colLast">
<div class="block">Verify TISDB peptides:<br>
 Each peptide might have a R at the end, or no R at all
 Every EGD-e gene should be in this database</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>equals, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="DATABASE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DATABASE</h4>
<pre>public static&nbsp;java.lang.String DATABASE</pre>
</li>
</ul>
<a id="LENGTH_CUTOFF">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>LENGTH_CUTOFF</h4>
<pre>public static&nbsp;int LENGTH_CUTOFF</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>ProteinTools</h4>
<pre>public&nbsp;ProteinTools()</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="createDataBases()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createDataBases</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;createDataBases()</pre>
<div class="block">Create:<br>
 <li>Protein DB
 <li>UTR DB
 <li>fusion Protein DB and Protein DB
 <li>Genome DB
 <li>TIS DB</div>
</li>
</ul>
<a id="fusionProteinUTRDB()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fusionProteinUTRDB</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;fusionProteinUTRDB()</pre>
<div class="block">Fusion ProteinDB and UTRDB in one file</div>
</li>
</ul>
<a id="createProteinDB()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createProteinDB</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;createProteinDB()</pre>
<div class="block">Create a fasta file which will be used has a database<br>
 This file contains:
 <li>all genes
 <li>all new genes described in RAST
 <li>all Srna on 3 frames
 <li>all ASrna on 3 frames (300bp long) lists of Genes, ID=lmo, directly taken
 from NCBI lists of "new"Gene, ID=rastlmo, from the analysis of annotation I
 just did using RAST software lists of Srna, ID=rli, rliM, rliP, we don't know
 where is the begining so we have 3 frame for each rli, we start at the first
 start codon, plus we cut when we found stop codon and peptide smaller than
 5aa. lists of ASrna, ID=anti, antiM, antiP, we don't know where is the
 begining so we have 3 frame for each asRNA, we start at the first start
 codon, plus we cut when we found stop codon and peptide smaller than 5aa.</div>
</li>
</ul>
<a id="createUTRDB()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createUTRDB</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;createUTRDB()</pre>
<div class="block">Create a database with all UTR from Wurtzel et al. 2012<br>
 Found here : DATABASE+"table_s1_tss_table.txt"<br>
 We found the correct frame so we are sure to have the start codon of the gene
 at the exact position of NCBI annotation We start at the first start codon,
 then if stop codon are found in the UTR region we cut in different peptides,
 remove ones < 5aa, and paste to the last part of the UTR the whole gene amino
 acide sequence all gene, ID=lmo_n n being the id of the UTR peptide extracted</div>
</li>
</ul>
<a id="createGenomeDB()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createGenomeDB</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;createGenomeDB()</pre>
<div class="block">lists of small peptides contained in the genome of EGDe on 6frames.<br>
 The small peptides include in the list are the one only of size more than
 5aa, and surrounded by at least one R.<br>
 example: Start - peptide - R<br>
 Stop - peptide - R<br>
 <br>
 Element like: Start - peptide - Stop are rejected from the analysis<br>
 <br>
 <li>frame 1 = EGDe genome, ID=genome
 <li>frame 2 = EGDe genome 1bp on the left, ID=genomeM
 <li>frame 3 = EGDe genome 1bp on the right, ID=genomeP
 <li>frame 4 = EGDe reverse complement genome, ID=genomeCompl
 <li>frame 5 = EGDe reverse complement genome 1bp on the left, ID=genomeComplM
 <li>frame 6 = EGDe reverse complement genome 1bp on the right,
 ID=genomeComplP <br></div>
</li>
</ul>
<a id="createTISDB()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createTISDB</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;createTISDB()</pre>
<div class="block">Create database of all peptide from a start codon (M=ATG, L=TTG, V=GTG), to
 the next stop codon or R</div>
</li>
</ul>
<a id="divideTISDB()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>divideTISDB</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;divideTISDB()</pre>
<div class="block">Divide TISDB into sequence starting with a M (TISMDB), and other not starting
 with an M (TISnoMDB).</div>
</li>
</ul>
<a id="getAllPeptides(java.lang.String,java.util.ArrayList,java.util.ArrayList,char,java.lang.String,int,boolean,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAllPeptides</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;getAllPeptides&#8203;(java.lang.String&nbsp;sequence,
                                  java.util.ArrayList&lt;java.lang.Integer&gt;&nbsp;startOccurences,
                                  java.util.ArrayList&lt;java.lang.String&gt;&nbsp;peptides,
                                  char&nbsp;codon,
                                  java.lang.String&nbsp;codonNucleotide,
                                  int&nbsp;genomeSize,
                                  boolean&nbsp;strand,
                                  int&nbsp;cutoff)</pre>
<div class="block">Given a genome sequence, and a list of position of start codon, update a list
 of peptides accordingly<br>
 The new peptide, are the peptide from a start codon to the next R or stop
 codon</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sequence</code> - </dd>
<dd><code>startOccurences</code> - </dd>
<dd><code>peptides</code> - </dd>
<dd><code>codon</code> - </dd>
</dl>
</li>
</ul>
<a id="verifyTISDB()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>verifyTISDB</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;verifyTISDB()</pre>
<div class="block">Verify TISDB peptides:<br>
 <li>Each peptide might have a R at the end, or no R at all
 <li>Every EGD-e gene should be in this database</div>
</li>
</ul>
<a id="createMDB()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createMDB</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;createMDB()</pre>
</li>
</ul>
<a id="possibleSmallRNAPeptides()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>possibleSmallRNAPeptides</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;possibleSmallRNAPeptides()</pre>
<div class="block">Get all possible peptides from EGD-e: <br>
 <li>get all proteins sequence from Small RNAs
 <li>get peptide sequence from M start to first R, or first E
 <li>cut-off every peptide of size under 5aa
 <li>export in a table
 <li>use <code>http://web.expasy.org/compute_pi/</code> to calculate mass and
 isoelectric point
 <li>save the different mass in NTermUtils.getPATH()+EGDePeptideMass.txt</div>
</li>
</ul>
<a id="statDatabase()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>statDatabase</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;statDatabase()</pre>
<div class="block">Extract statistics from existing databases, number of lmos, sRNA, utr...</div>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/ProteinTools.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
