<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="fr">
<head>
<!-- Generated by javadoc (11.0.2) on Sun May 24 16:09:24 CEST 2020 -->
<title>DNASequence</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2020-05-24">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="DNASequence";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":9,"i10":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/DNASequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">org.biojava3.core.sequence</a></div>
<h2 title="Class DNASequence" class="title">Class DNASequence</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li><a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">org.biojava3.core.sequence.template.AbstractSequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</li>
<li>
<ul class="inheritance">
<li>org.biojava3.core.sequence.DNASequence</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><code>java.lang.Iterable&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code>, <code><a href="template/Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</a></code>, <code><a href="template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
</dl>
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><code><a href="CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></code>, <code><a href="ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></code>, <code><a href="ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</a></code>, <code><a href="GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a></code>, <code><a href="IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</a></code>, <code><a href="StartCodonSequence.html" title="class in org.biojava3.core.sequence">StartCodonSequence</a></code>, <code><a href="StopCodonSequence.html" title="class in org.biojava3.core.sequence">StopCodonSequence</a></code>, <code><a href="TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></code></dd>
</dl>
<hr>
<pre>public class <span class="typeNameLabel">DNASequence</span>
extends <a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</pre>
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Scooter Willis</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="memberSummary">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></span></code></th>
<td class="colLast">
<div class="block">The type of DNA sequence</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="nested.classes.inherited.from.class.org.biojava3.core.sequence.template.AbstractSequence">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;org.biojava3.core.sequence.template.<a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></h3>
<code><a href="template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></code></li>
</ul>
</li>
</ul>
</section>
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private <a href="DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#dnaType">dnaType</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">DNASequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Shouldn't be used but makes it bean happy</div>
</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.lang.String)">DNASequence</a></span>&#8203;(java.lang.String&nbsp;seqString)</code></th>
<td class="colLast">
<div class="block">String is king and create a sequence from DNA with default DNA compound set</div>
</td>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">DNASequence</a></span>&#8203;(java.lang.String&nbsp;seqString,
           <a href="template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Create a sequence from a string with user defined compound set</div>
</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader)">DNASequence</a></span>&#8203;(<a href="template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader)</code></th>
<td class="colLast">
<div class="block">Create a sequence where the actual storage of the sequence data is somewhere else</div>
</td>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">DNASequence</a></span>&#8203;(<a href="template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
           <a href="template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Create a sequence from a ProxySequencereader and user defined compound set</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code><a href="template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getComplement()">getComplement</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns a Sequence which will complement every base</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code><a href="DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDNAType()">getDNAType</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGCCount()">getGCCount</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the GC count in the DNA Sequence</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code><a href="template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getReverse()">getReverse</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns a Sequence which runs in the current reverse order</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code><a href="template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getReverseComplement()">getReverseComplement</a></span>()</code></th>
<td class="colLast">
<div class="block">Delegates to <a href="template/AbstractSequence.html#getInverse()"><code>AbstractSequence.getInverse()</code></a> for the reverse complement</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code><a href="RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getRNASequence()">getRNASequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the RNASequence equivalent of the DNASequence using default Transcription Engine.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code><a href="RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getRNASequence(org.biojava3.core.sequence.transcription.Frame)">getRNASequence</a></span>&#8203;(<a href="transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a>&nbsp;frame)</code></th>
<td class="colLast">
<div class="block">Allows the user to pass in the Frame shift.</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code><a href="RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)">getRNASequence</a></span>&#8203;(<a href="transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a>&nbsp;engine)</code></th>
<td class="colLast">
<div class="block">Allow a user to pass in a rules engine to do the DNA to RNA translation</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code><a href="RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine,org.biojava3.core.sequence.transcription.Frame)">getRNASequence</a></span>&#8203;(<a href="transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a>&nbsp;engine,
              <a href="transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a>&nbsp;frame)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#main(java.lang.String%5B%5D)">main</a></span>&#8203;(java.lang.String[]&nbsp;args)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setDNAType(org.biojava3.core.sequence.DNASequence.DNAType)">setDNAType</a></span>&#8203;(<a href="DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a>&nbsp;dnaType)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.org.biojava3.core.sequence.template.AbstractSequence">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.biojava3.core.sequence.template.<a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></h3>
<code><a href="template/AbstractSequence.html#addFeature(int,int,org.biojava3.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="template/AbstractSequence.html#addFeature(org.biojava3.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="template/AbstractSequence.html#addNote(java.lang.String)">addNote</a>, <a href="template/AbstractSequence.html#countCompounds(C...)">countCompounds</a>, <a href="template/AbstractSequence.html#getAccession()">getAccession</a>, <a href="template/AbstractSequence.html#getAnnotationType()">getAnnotationType</a>, <a href="template/AbstractSequence.html#getAsList()">getAsList</a>, <a href="template/AbstractSequence.html#getBioBegin()">getBioBegin</a>, <a href="template/AbstractSequence.html#getBioEnd()">getBioEnd</a>, <a href="template/AbstractSequence.html#getCompoundAt(int)">getCompoundAt</a>, <a href="template/AbstractSequence.html#getCompoundSet()">getCompoundSet</a>, <a href="template/AbstractSequence.html#getDatabaseReferences()">getDatabaseReferences</a>, <a href="template/AbstractSequence.html#getDescription()">getDescription</a>, <a href="template/AbstractSequence.html#getFeatures()">getFeatures</a>, <a href="template/AbstractSequence.html#getFeatures(int)">getFeatures</a>, <a href="template/AbstractSequence.html#getFeatures(java.lang.String,int)">getFeatures</a>, <a href="template/AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType</a>, <a href="template/AbstractSequence.html#getFeaturesKeyWord()">getFeaturesKeyWord</a>, <a href="template/AbstractSequence.html#getIndexOf(C)">getIndexOf</a>, <a href="template/AbstractSequence.html#getInverse()">getInverse</a>, <a href="template/AbstractSequence.html#getLastIndexOf(C)">getLastIndexOf</a>, <a href="template/AbstractSequence.html#getLength()">getLength</a>, <a href="template/AbstractSequence.html#getNotesList()">getNotesList</a>, <a href="template/AbstractSequence.html#getOriginalHeader()">getOriginalHeader</a>, <a href="template/AbstractSequence.html#getParentSequence()">getParentSequence</a>, <a href="template/AbstractSequence.html#getProxySequenceReader()">getProxySequenceReader</a>, <a href="template/AbstractSequence.html#getSequenceAsString()">getSequenceAsString</a>, <a href="template/AbstractSequence.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)">getSequenceAsString</a>, <a href="template/AbstractSequence.html#getSequenceScore()">getSequenceScore</a>, <a href="template/AbstractSequence.html#getSource()">getSource</a>, <a href="template/AbstractSequence.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence</a>, <a href="template/AbstractSequence.html#getTaxonomy()">getTaxonomy</a>, <a href="template/AbstractSequence.html#getUserCollection()">getUserCollection</a>, <a href="template/AbstractSequence.html#iterator()">iterator</a>, <a href="template/AbstractSequence.html#removeFeature(org.biojava3.core.sequence.features.FeatureInterface)">removeFeature</a>, <a href="template/AbstractSequence.html#removeNote(java.lang.String)">removeNote</a>, <a href="template/AbstractSequence.html#setAccession(org.biojava3.core.sequence.AccessionID)">setAccession</a>, <a href="template/AbstractSequence.html#setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType</a>, <a href="template/AbstractSequence.html#setBioBegin(java.lang.Integer)">setBioBegin</a>, <a href="template/AbstractSequence.html#setBioEnd(java.lang.Integer)">setBioEnd</a>, <a href="template/AbstractSequence.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a>, <a href="template/AbstractSequence.html#setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences</a>, <a href="template/AbstractSequence.html#setDescription(java.lang.String)">setDescription</a>, <a href="template/AbstractSequence.html#setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord</a>, <a href="template/AbstractSequence.html#setNotesList(java.util.ArrayList)">setNotesList</a>, <a href="template/AbstractSequence.html#setOriginalHeader(java.lang.String)">setOriginalHeader</a>, <a href="template/AbstractSequence.html#setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)">setParentSequence</a>, <a href="template/AbstractSequence.html#setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)">setProxySequenceReader</a>, <a href="template/AbstractSequence.html#setSequenceScore(java.lang.Double)">setSequenceScore</a>, <a href="template/AbstractSequence.html#setSource(java.lang.String)">setSource</a>, <a href="template/AbstractSequence.html#setTaxonomy(org.biojava3.core.sequence.TaxonomyID)">setTaxonomy</a>, <a href="template/AbstractSequence.html#setUserCollection(java.util.Collection)">setUserCollection</a>, <a href="template/AbstractSequence.html#toString()">toString</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, wait, wait, wait</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Iterable">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;java.lang.Iterable</h3>
<code>forEach, spliterator</code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="dnaType">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>dnaType</h4>
<pre>private&nbsp;<a href="DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a> dnaType</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DNASequence</h4>
<pre>public&nbsp;DNASequence()</pre>
<div class="block">Shouldn't be used but makes it bean happy</div>
</li>
</ul>
<a id="&lt;init&gt;(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DNASequence</h4>
<pre>public&nbsp;DNASequence&#8203;(java.lang.String&nbsp;seqString)</pre>
<div class="block">String is king and create a sequence from DNA with default DNA compound set</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>seqString</code> - </dd>
</dl>
</li>
</ul>
<a id="&lt;init&gt;(org.biojava3.core.sequence.template.SequenceReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DNASequence</h4>
<pre>public&nbsp;DNASequence&#8203;(<a href="template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader)</pre>
<div class="block">Create a sequence where the actual storage of the sequence data is somewhere else</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>proxyLoader</code> - </dd>
</dl>
</li>
</ul>
<a id="&lt;init&gt;(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DNASequence</h4>
<pre>public&nbsp;DNASequence&#8203;(java.lang.String&nbsp;seqString,
                   <a href="template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</pre>
<div class="block">Create a sequence from a string with user defined compound set</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>seqString</code> - </dd>
<dd><code>compoundSet</code> - </dd>
</dl>
</li>
</ul>
<a id="&lt;init&gt;(org.biojava3.core.sequence.template.SequenceReader,org.biojava3.core.sequence.template.CompoundSet)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>DNASequence</h4>
<pre>public&nbsp;DNASequence&#8203;(<a href="template/SequenceReader.html" title="interface in org.biojava3.core.sequence.template">SequenceReader</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
                   <a href="template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</pre>
<div class="block">Create a sequence from a ProxySequencereader and user defined compound set</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>proxyLoader</code> - </dd>
<dd><code>compoundSet</code> - </dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="getRNASequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRNASequence</h4>
<pre class="methodSignature">public&nbsp;<a href="RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a>&nbsp;getRNASequence()</pre>
<div class="block">Return the RNASequence equivalent of the DNASequence using default Transcription Engine. Not all
 species follow the same rules. If you don't know better use this method</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>RNA sequence</dd>
</dl>
</li>
</ul>
<a id="getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRNASequence</h4>
<pre class="methodSignature">public&nbsp;<a href="RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a>&nbsp;getRNASequence&#8203;(<a href="transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a>&nbsp;engine)</pre>
<div class="block">Allow a user to pass in a rules engine to do the DNA to RNA translation</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>engine</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>RNA sequence</dd>
</dl>
</li>
</ul>
<a id="getRNASequence(org.biojava3.core.sequence.transcription.Frame)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRNASequence</h4>
<pre class="methodSignature">public&nbsp;<a href="RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a>&nbsp;getRNASequence&#8203;(<a href="transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a>&nbsp;frame)</pre>
<div class="block">Allows the user to pass in the Frame shift.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>frame</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>rna sequence</dd>
</dl>
</li>
</ul>
<a id="getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine,org.biojava3.core.sequence.transcription.Frame)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRNASequence</h4>
<pre class="methodSignature">public&nbsp;<a href="RNASequence.html" title="class in org.biojava3.core.sequence">RNASequence</a>&nbsp;getRNASequence&#8203;(<a href="transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a>&nbsp;engine,
                                  <a href="transcription/Frame.html" title="enum in org.biojava3.core.sequence.transcription">Frame</a>&nbsp;frame)</pre>
</li>
</ul>
<a id="getGCCount()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGCCount</h4>
<pre class="methodSignature">public&nbsp;int&nbsp;getGCCount()</pre>
<div class="block">Get the GC count in the DNA Sequence</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>GC count</dd>
</dl>
</li>
</ul>
<a id="getReverse()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReverse</h4>
<pre class="methodSignature">public&nbsp;<a href="template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;getReverse()</pre>
<div class="block">Returns a Sequence which runs in the current reverse order</div>
</li>
</ul>
<a id="getComplement()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getComplement</h4>
<pre class="methodSignature">public&nbsp;<a href="template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;getComplement()</pre>
<div class="block">Returns a Sequence which will complement every base</div>
</li>
</ul>
<a id="getReverseComplement()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReverseComplement</h4>
<pre class="methodSignature">public&nbsp;<a href="template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;getReverseComplement()</pre>
<div class="block">Delegates to <a href="template/AbstractSequence.html#getInverse()"><code>AbstractSequence.getInverse()</code></a> for the reverse complement</div>
</li>
</ul>
<a id="getDNAType()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDNAType</h4>
<pre class="methodSignature">public&nbsp;<a href="DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a>&nbsp;getDNAType()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the dnaType</dd>
</dl>
</li>
</ul>
<a id="setDNAType(org.biojava3.core.sequence.DNASequence.DNAType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDNAType</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;setDNAType&#8203;(<a href="DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a>&nbsp;dnaType)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>dnaType</code> - the dnaType to set</dd>
</dl>
</li>
</ul>
<a id="main(java.lang.String[])">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>main</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;main&#8203;(java.lang.String[]&nbsp;args)</pre>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/DNASequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
</footer>
</body>
</html>
