<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>GeneSequence</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="declaration: package: org.biojava3.core.sequence, class: GeneSequence">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var evenRowColor = "even-row-color";
var oddRowColor = "odd-row-color";
var tableTab = "table-tab";
var activeTableTab = "active-table-tab";
var pathtoroot = "../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/GeneSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html#class">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div>
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li><a href="#nested-class-summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">org.biojava3.core.sequence</a></div>
<h1 title="Class GeneSequence" class="title">Class GeneSequence</h1>
</div>
<div class="inheritance" title="Inheritance Tree"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">java.lang.Object</a>
<div class="inheritance"><a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">org.biojava3.core.sequence.template.AbstractSequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;
<div class="inheritance"><a href="DNASequence.html" title="class in org.biojava3.core.sequence">org.biojava3.core.sequence.DNASequence</a>
<div class="inheritance">org.biojava3.core.sequence.GeneSequence</div>
</div>
</div>
</div>
<section class="class-description" id="class-description">
<dl class="notes">
<dt>All Implemented Interfaces:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html" title="class or interface in java.lang" class="external-link">Iterable</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code>, <code><a href="template/Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</a></code>, <code><a href="template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
</dl>
<hr>
<div class="type-signature"><span class="modifiers">public class </span><span class="element-name type-name-label">GeneSequence</span>
<span class="extends-implements">extends <a href="DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span></div>
<dl class="notes">
<dt>Author:</dt>
<dd>Scooter Willis</dd>
</dl>
</section>
<section class="summary">
<ul class="summary-list">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<li>
<section class="nested-class-summary" id="nested-class-summary">
<h2>Nested Class Summary</h2>
<div class="inherited-list">
<h2 id="nested-classes-inherited-from-class-org.biojava3.core.sequence.DNASequence">Nested classes/interfaces inherited from class&nbsp;org.biojava3.core.sequence.<a href="DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></h2>
<code><a href="DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></code></div>
<div class="inherited-list">
<h2 id="nested-classes-inherited-from-class-org.biojava3.core.sequence.template.AbstractSequence">Nested classes/interfaces inherited from class&nbsp;org.biojava3.core.sequence.template.<a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></h2>
<code><a href="template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></code></div>
</section>
</li>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<li>
<section class="constructor-summary" id="constructor-summary">
<h2>Constructor Summary</h2>
<div class="caption"><span>Constructors</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E(org.biojava3.core.sequence.ChromosomeSequence,int,int,org.biojava3.core.sequence.Strand)" class="member-name-link">GeneSequence</a><wbr>(<a href="ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&nbsp;parentSequence,
 int&nbsp;begin,
 int&nbsp;end,
 <a href="Strand.html" title="enum in org.biojava3.core.sequence">Strand</a>&nbsp;strand)</code></div>
<div class="col-last even-row-color">
<div class="block">A class that keeps track of the details of a GeneSequence which is difficult to properly model.</div>
</div>
</div>
</section>
</li>
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab2" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab2', 3)" class="table-tab">Instance Methods</button><button id="method-summary-table-tab4" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab4', 3)" class="table-tab">Concrete Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel">
<div class="summary-table three-column-summary" aria-labelledby="method-summary-table-tab0">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#addExon(org.biojava3.core.sequence.AccessionID,int,int)" class="member-name-link">addExon</a><wbr>(<a href="AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession,
 int&nbsp;begin,
 int&nbsp;end)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Add an ExonSequence mainly used to mark as a feature</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#addIntron(org.biojava3.core.sequence.AccessionID,int,int)" class="member-name-link">addIntron</a><wbr>(<a href="AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession,
 int&nbsp;begin,
 int&nbsp;end)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Add an Intron Currently used to mark an IntronSequence as a feature</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#addIntronsUsingExons()" class="member-name-link">addIntronsUsingExons</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Once everything has been added to the gene sequence where you might have added exon sequences
 only then you can infer the intron sequences and add them.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#addTranscript(org.biojava3.core.sequence.AccessionID,int,int)" class="member-name-link">addTranscript</a><wbr>(<a href="AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession,
 int&nbsp;begin,
 int&nbsp;end)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Add a transcription sequence to a gene which describes a ProteinSequence</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getExonSequences()" class="member-name-link">getExonSequences</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get the exons as an ArrayList</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</a>&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getIntronSequences()" class="member-name-link">getIntronSequences</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get the introns as an ArrayList</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>int</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getLength()" class="member-name-link">getLength</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Returns the length of the Sequence</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getParentChromosomeSequence()" class="member-name-link">getParentChromosomeSequence</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">The parent ChromosomeSequence which contains the actual DNA sequence data</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getSequence5PrimeTo3Prime()" class="member-name-link">getSequence5PrimeTo3Prime</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Try to give method clarity where you want a DNASequence coding in the 5' to 3' direction Returns
 the DNASequence representative of the 5' and 3' reading based on strand</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="Strand.html" title="enum in org.biojava3.core.sequence">Strand</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getStrand()" class="member-name-link">getStrand</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">A gene should have Strand</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getTranscript(java.lang.String)" class="member-name-link">getTranscript</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get the transcript sequence by accession</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/LinkedHashMap.html" title="class or interface in java.util" class="external-link">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a>&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getTranscripts()" class="member-name-link">getTranscripts</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get the collection of transcription sequences assigned to this gene</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#removeExon(java.lang.String)" class="member-name-link">removeExon</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Remove the exon sequence</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#removeIntron(java.lang.String)" class="member-name-link">removeIntron</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Remove the intron by accession</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#removeTranscript(java.lang.String)" class="member-name-link">removeTranscript</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Remove the transcript sequence from the gene</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setStrand(org.biojava3.core.sequence.Strand)" class="member-name-link">setStrand</a><wbr>(<a href="Strand.html" title="enum in org.biojava3.core.sequence">Strand</a>&nbsp;strand)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
</div>
</div>
</div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-org.biojava3.core.sequence.DNASequence">Methods inherited from class&nbsp;org.biojava3.core.sequence.<a href="DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></h3>
<code><a href="DNASequence.html#getComplement()">getComplement</a>, <a href="DNASequence.html#getDNAType()">getDNAType</a>, <a href="DNASequence.html#getGCCount()">getGCCount</a>, <a href="DNASequence.html#getReverse()">getReverse</a>, <a href="DNASequence.html#getReverseComplement()">getReverseComplement</a>, <a href="DNASequence.html#getRNASequence()">getRNASequence</a>, <a href="DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.Frame)">getRNASequence</a>, <a href="DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)">getRNASequence</a>, <a href="DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine,org.biojava3.core.sequence.transcription.Frame)">getRNASequence</a>, <a href="DNASequence.html#main(java.lang.String%5B%5D)">main</a>, <a href="DNASequence.html#setDNAType(org.biojava3.core.sequence.DNASequence.DNAType)">setDNAType</a></code></div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-org.biojava3.core.sequence.template.AbstractSequence">Methods inherited from class&nbsp;org.biojava3.core.sequence.template.<a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></h3>
<code><a href="template/AbstractSequence.html#addFeature(int,int,org.biojava3.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="template/AbstractSequence.html#addFeature(org.biojava3.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="template/AbstractSequence.html#addNote(java.lang.String)">addNote</a>, <a href="template/AbstractSequence.html#countCompounds(C...)">countCompounds</a>, <a href="template/AbstractSequence.html#getAccession()">getAccession</a>, <a href="template/AbstractSequence.html#getAnnotationType()">getAnnotationType</a>, <a href="template/AbstractSequence.html#getAsList()">getAsList</a>, <a href="template/AbstractSequence.html#getBioBegin()">getBioBegin</a>, <a href="template/AbstractSequence.html#getBioEnd()">getBioEnd</a>, <a href="template/AbstractSequence.html#getCompoundAt(int)">getCompoundAt</a>, <a href="template/AbstractSequence.html#getCompoundSet()">getCompoundSet</a>, <a href="template/AbstractSequence.html#getDatabaseReferences()">getDatabaseReferences</a>, <a href="template/AbstractSequence.html#getDescription()">getDescription</a>, <a href="template/AbstractSequence.html#getFeatures()">getFeatures</a>, <a href="template/AbstractSequence.html#getFeatures(int)">getFeatures</a>, <a href="template/AbstractSequence.html#getFeatures(java.lang.String,int)">getFeatures</a>, <a href="template/AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType</a>, <a href="template/AbstractSequence.html#getFeaturesKeyWord()">getFeaturesKeyWord</a>, <a href="template/AbstractSequence.html#getIndexOf(C)">getIndexOf</a>, <a href="template/AbstractSequence.html#getInverse()">getInverse</a>, <a href="template/AbstractSequence.html#getLastIndexOf(C)">getLastIndexOf</a>, <a href="template/AbstractSequence.html#getNotesList()">getNotesList</a>, <a href="template/AbstractSequence.html#getOriginalHeader()">getOriginalHeader</a>, <a href="template/AbstractSequence.html#getParentSequence()">getParentSequence</a>, <a href="template/AbstractSequence.html#getProxySequenceReader()">getProxySequenceReader</a>, <a href="template/AbstractSequence.html#getSequenceAsString()">getSequenceAsString</a>, <a href="template/AbstractSequence.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)">getSequenceAsString</a>, <a href="template/AbstractSequence.html#getSequenceScore()">getSequenceScore</a>, <a href="template/AbstractSequence.html#getSource()">getSource</a>, <a href="template/AbstractSequence.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence</a>, <a href="template/AbstractSequence.html#getTaxonomy()">getTaxonomy</a>, <a href="template/AbstractSequence.html#getUserCollection()">getUserCollection</a>, <a href="template/AbstractSequence.html#iterator()">iterator</a>, <a href="template/AbstractSequence.html#removeFeature(org.biojava3.core.sequence.features.FeatureInterface)">removeFeature</a>, <a href="template/AbstractSequence.html#removeNote(java.lang.String)">removeNote</a>, <a href="template/AbstractSequence.html#setAccession(org.biojava3.core.sequence.AccessionID)">setAccession</a>, <a href="template/AbstractSequence.html#setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType</a>, <a href="template/AbstractSequence.html#setBioBegin(java.lang.Integer)">setBioBegin</a>, <a href="template/AbstractSequence.html#setBioEnd(java.lang.Integer)">setBioEnd</a>, <a href="template/AbstractSequence.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a>, <a href="template/AbstractSequence.html#setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences</a>, <a href="template/AbstractSequence.html#setDescription(java.lang.String)">setDescription</a>, <a href="template/AbstractSequence.html#setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord</a>, <a href="template/AbstractSequence.html#setNotesList(java.util.ArrayList)">setNotesList</a>, <a href="template/AbstractSequence.html#setOriginalHeader(java.lang.String)">setOriginalHeader</a>, <a href="template/AbstractSequence.html#setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)">setParentSequence</a>, <a href="template/AbstractSequence.html#setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)">setProxySequenceReader</a>, <a href="template/AbstractSequence.html#setSequenceScore(java.lang.Double)">setSequenceScore</a>, <a href="template/AbstractSequence.html#setSource(java.lang.String)">setSource</a>, <a href="template/AbstractSequence.html#setTaxonomy(org.biojava3.core.sequence.TaxonomyID)">setTaxonomy</a>, <a href="template/AbstractSequence.html#setUserCollection(java.util.Collection)">setUserCollection</a>, <a href="template/AbstractSequence.html#toString()">toString</a></code></div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Object">Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)" title="class or interface in java.lang" class="external-link">equals</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#getClass()" title="class or interface in java.lang" class="external-link">getClass</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode()" title="class or interface in java.lang" class="external-link">hashCode</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notify()" title="class or interface in java.lang" class="external-link">notify</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notifyAll()" title="class or interface in java.lang" class="external-link">notifyAll</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait()" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long)" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long,int)" title="class or interface in java.lang" class="external-link">wait</a></code></div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Iterable">Methods inherited from interface&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html" title="class or interface in java.lang" class="external-link">Iterable</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html#forEach(java.util.function.Consumer)" title="class or interface in java.lang" class="external-link">forEach</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html#spliterator()" title="class or interface in java.lang" class="external-link">spliterator</a></code></div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<li>
<section class="constructor-details" id="constructor-detail">
<h2>Constructor Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="&lt;init&gt;(org.biojava3.core.sequence.ChromosomeSequence,int,int,org.biojava3.core.sequence.Strand)">
<h3>GeneSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name">GeneSequence</span><wbr><span class="parameters">(<a href="ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a>&nbsp;parentSequence,
 int&nbsp;begin,
 int&nbsp;end,
 <a href="Strand.html" title="enum in org.biojava3.core.sequence">Strand</a>&nbsp;strand)</span></div>
<div class="block">A class that keeps track of the details of a GeneSequence which is difficult to properly model.
 Two important concepts that is difficult to make everything flexible but still work. You can have
 GFF features that only describe Exons or Exons/Introns or CDS regions and one or more
 Transcriptions. You can have exon sequences but that does not imply transcription to the actual
 protein.

 The GeneSequence will keep track of Exons and Introns but to get a Protein sequence you need to
 start with a TranscriptSequence and then add CDS sequences.

 This is also a key class in the biojava-3-genome module for reading and writing GFF3 files</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>parentDNASequence</code> - </dd>
<dd><code>begin</code> - </dd>
<dd><code>end</code> - inclusive of end</dd>
<dd><code>strand</code> - force a gene to have strand and transcription sequence will inherit</dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="getParentChromosomeSequence()">
<h3>getParentChromosomeSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="ChromosomeSequence.html" title="class in org.biojava3.core.sequence">ChromosomeSequence</a></span>&nbsp;<span class="element-name">getParentChromosomeSequence</span>()</div>
<div class="block">The parent ChromosomeSequence which contains the actual DNA sequence data</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>Chromosome sequence</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getLength()">
<h3>getLength</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name">getLength</span>()</div>
<div class="block"><span class="descfrm-type-label">Description copied from interface:&nbsp;<code><a href="template/Sequence.html#getLength()">Sequence</a></code></span></div>
<div class="block">Returns the length of the Sequence</div>
<dl class="notes">
<dt>Specified by:</dt>
<dd><code><a href="template/Sequence.html#getLength()">getLength</a></code>&nbsp;in interface&nbsp;<code><a href="template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
<dt>Overrides:</dt>
<dd><code><a href="template/AbstractSequence.html#getLength()">getLength</a></code>&nbsp;in class&nbsp;<code><a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="addIntronsUsingExons()">
<h3>addIntronsUsingExons</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">addIntronsUsingExons</span>()
                          throws <span class="exceptions"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Exception.html" title="class or interface in java.lang" class="external-link">Exception</a></span></div>
<div class="block">Once everything has been added to the gene sequence where you might have added exon sequences
 only then you can infer the intron sequences and add them. You may also have the case where you
 only added one or more TranscriptSequences and from that you can infer the exon sequences and
 intron sequences. Currently not implement</div>
<dl class="notes">
<dt>Throws:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Exception.html" title="class or interface in java.lang" class="external-link">Exception</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getStrand()">
<h3>getStrand</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="Strand.html" title="enum in org.biojava3.core.sequence">Strand</a></span>&nbsp;<span class="element-name">getStrand</span>()</div>
<div class="block">A gene should have Strand</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the strand</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setStrand(org.biojava3.core.sequence.Strand)">
<h3>setStrand</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">setStrand</span><wbr><span class="parameters">(<a href="Strand.html" title="enum in org.biojava3.core.sequence">Strand</a>&nbsp;strand)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>strand</code> - the strand to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getTranscript(java.lang.String)">
<h3>getTranscript</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></span>&nbsp;<span class="element-name">getTranscript</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession)</span></div>
<div class="block">Get the transcript sequence by accession</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>accession</code> - </dd>
<dt>Returns:</dt>
<dd>the transcript</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getTranscripts()">
<h3>getTranscripts</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/LinkedHashMap.html" title="class or interface in java.util" class="external-link">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a>&gt;</span>&nbsp;<span class="element-name">getTranscripts</span>()</div>
<div class="block">Get the collection of transcription sequences assigned to this gene</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>transcripts</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="removeTranscript(java.lang.String)">
<h3>removeTranscript</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></span>&nbsp;<span class="element-name">removeTranscript</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession)</span></div>
<div class="block">Remove the transcript sequence from the gene</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>accession</code> - </dd>
<dt>Returns:</dt>
<dd>transcriptsequence</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="addTranscript(org.biojava3.core.sequence.AccessionID,int,int)">
<h3>addTranscript</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="TranscriptSequence.html" title="class in org.biojava3.core.sequence">TranscriptSequence</a></span>&nbsp;<span class="element-name">addTranscript</span><wbr><span class="parameters">(<a href="AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession,
 int&nbsp;begin,
 int&nbsp;end)</span>
                                 throws <span class="exceptions"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Exception.html" title="class or interface in java.lang" class="external-link">Exception</a></span></div>
<div class="block">Add a transcription sequence to a gene which describes a ProteinSequence</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>accession</code> - </dd>
<dd><code>begin</code> - </dd>
<dd><code>end</code> - </dd>
<dt>Returns:</dt>
<dd>transcript sequence</dd>
<dt>Throws:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Exception.html" title="class or interface in java.lang" class="external-link">Exception</a></code> - If the accession id is already used</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="removeIntron(java.lang.String)">
<h3>removeIntron</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</a></span>&nbsp;<span class="element-name">removeIntron</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession)</span></div>
<div class="block">Remove the intron by accession</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>accession</code> - </dd>
<dt>Returns:</dt>
<dd>intron sequence</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="addIntron(org.biojava3.core.sequence.AccessionID,int,int)">
<h3>addIntron</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</a></span>&nbsp;<span class="element-name">addIntron</span><wbr><span class="parameters">(<a href="AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession,
 int&nbsp;begin,
 int&nbsp;end)</span>
                         throws <span class="exceptions"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Exception.html" title="class or interface in java.lang" class="external-link">Exception</a></span></div>
<div class="block">Add an Intron Currently used to mark an IntronSequence as a feature</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>accession</code> - </dd>
<dd><code>begin</code> - </dd>
<dd><code>end</code> - </dd>
<dt>Returns:</dt>
<dd>intron sequence</dd>
<dt>Throws:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Exception.html" title="class or interface in java.lang" class="external-link">Exception</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="removeExon(java.lang.String)">
<h3>removeExon</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</a></span>&nbsp;<span class="element-name">removeExon</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession)</span></div>
<div class="block">Remove the exon sequence</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>accession</code> - </dd>
<dt>Returns:</dt>
<dd>exon sequence</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="addExon(org.biojava3.core.sequence.AccessionID,int,int)">
<h3>addExon</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</a></span>&nbsp;<span class="element-name">addExon</span><wbr><span class="parameters">(<a href="AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession,
 int&nbsp;begin,
 int&nbsp;end)</span>
                     throws <span class="exceptions"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Exception.html" title="class or interface in java.lang" class="external-link">Exception</a></span></div>
<div class="block">Add an ExonSequence mainly used to mark as a feature</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>accession</code> - </dd>
<dd><code>begin</code> - </dd>
<dd><code>end</code> - </dd>
<dt>Returns:</dt>
<dd>exon sequence</dd>
<dt>Throws:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Exception.html" title="class or interface in java.lang" class="external-link">Exception</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getExonSequences()">
<h3>getExonSequences</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="ExonSequence.html" title="class in org.biojava3.core.sequence">ExonSequence</a>&gt;</span>&nbsp;<span class="element-name">getExonSequences</span>()</div>
<div class="block">Get the exons as an ArrayList</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>exons</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getIntronSequences()">
<h3>getIntronSequences</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="IntronSequence.html" title="class in org.biojava3.core.sequence">IntronSequence</a>&gt;</span>&nbsp;<span class="element-name">getIntronSequences</span>()</div>
<div class="block">Get the introns as an ArrayList</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>introns</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getSequence5PrimeTo3Prime()">
<h3>getSequence5PrimeTo3Prime</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span>&nbsp;<span class="element-name">getSequence5PrimeTo3Prime</span>()</div>
<div class="block">Try to give method clarity where you want a DNASequence coding in the 5' to 3' direction Returns
 the DNASequence representative of the 5' and 3' reading based on strand</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>dna sequence</dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
</div>
</div>
</body>
</html>
