<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>RNASequence</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="declaration: package: org.biojava3.core.sequence, class: RNASequence">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var evenRowColor = "even-row-color";
var oddRowColor = "odd-row-color";
var tableTab = "table-tab";
var activeTableTab = "active-table-tab";
var pathtoroot = "../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/RNASequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html#class">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div>
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li><a href="#nested-class-summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">org.biojava3.core.sequence</a></div>
<h1 title="Class RNASequence" class="title">Class RNASequence</h1>
</div>
<div class="inheritance" title="Inheritance Tree"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">java.lang.Object</a>
<div class="inheritance"><a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">org.biojava3.core.sequence.template.AbstractSequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;
<div class="inheritance">org.biojava3.core.sequence.RNASequence</div>
</div>
</div>
<section class="class-description" id="class-description">
<dl class="notes">
<dt>All Implemented Interfaces:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html" title="class or interface in java.lang" class="external-link">Iterable</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code>, <code><a href="template/Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</a></code>, <code><a href="template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
</dl>
<hr>
<div class="type-signature"><span class="modifiers">public class </span><span class="element-name type-name-label">RNASequence</span>
<span class="extends-implements">extends <a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</span></div>
<div class="block">RNASequence where RNACompoundSet are the allowed values</div>
<dl class="notes">
<dt>Author:</dt>
<dd>Scooter Willis <willishf at gmail dot com></dd>
</dl>
</section>
<section class="summary">
<ul class="summary-list">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<li>
<section class="nested-class-summary" id="nested-class-summary">
<h2>Nested Class Summary</h2>
<div class="inherited-list">
<h2 id="nested-classes-inherited-from-class-org.biojava3.core.sequence.template.AbstractSequence">Nested classes/interfaces inherited from class&nbsp;org.biojava3.core.sequence.template.<a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></h2>
<code><a href="template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></code></div>
</section>
</li>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<li>
<section class="constructor-summary" id="constructor-summary">
<h2>Constructor Summary</h2>
<div class="caption"><span>Constructors</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E(java.lang.String)" class="member-name-link">RNASequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString)</code></div>
<div class="col-last even-row-color">
<div class="block">Create a RNA sequence from a String</div>
</div>
<div class="col-constructor-name odd-row-color"><code><a href="#%3Cinit%3E(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">RNASequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create a RNA sequence from a string with a user defined RNA compound set</div>
</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader)" class="member-name-link">RNASequence</a><wbr>(<a href="template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader)</code></div>
<div class="col-last even-row-color">
<div class="block">Create a RNA aequence from a proxy reader</div>
</div>
<div class="col-constructor-name odd-row-color"><code><a href="#%3Cinit%3E(org.biojava3.core.sequence.template.ProxySequenceReader,org.biojava3.core.sequence.template.CompoundSet)" class="member-name-link">RNASequence</a><wbr>(<a href="template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
 <a href="template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create a RNA sequence from a proxy reader and user defined RNA compound set</div>
</div>
</div>
</section>
</li>
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab2" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab2', 3)" class="table-tab">Instance Methods</button><button id="method-summary-table-tab4" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab4', 3)" class="table-tab">Concrete Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel">
<div class="summary-table three-column-summary" aria-labelledby="method-summary-table-tab0">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getComplement()" class="member-name-link">getComplement</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get the complement view of the RNA sequence</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>double</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getGC()" class="member-name-link">getGC</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getInverse()" class="member-name-link">getInverse</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get the inverse view of the sequence.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getProteinSequence()" class="member-name-link">getProteinSequence</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get the ProteinSequence from the RNA sequence</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getProteinSequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)" class="member-name-link">getProteinSequence</a><wbr>(<a href="transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a>&nbsp;engine)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get the ProteinSequene from the RNA sequence with user defined transcription engine</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getReverseComplement()" class="member-name-link">getReverseComplement</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get reverse complement view of the sequence</div>
</div>
</div>
</div>
</div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-org.biojava3.core.sequence.template.AbstractSequence">Methods inherited from class&nbsp;org.biojava3.core.sequence.template.<a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></h3>
<code><a href="template/AbstractSequence.html#addFeature(int,int,org.biojava3.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="template/AbstractSequence.html#addFeature(org.biojava3.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="template/AbstractSequence.html#addNote(java.lang.String)">addNote</a>, <a href="template/AbstractSequence.html#countCompounds(C...)">countCompounds</a>, <a href="template/AbstractSequence.html#getAccession()">getAccession</a>, <a href="template/AbstractSequence.html#getAnnotationType()">getAnnotationType</a>, <a href="template/AbstractSequence.html#getAsList()">getAsList</a>, <a href="template/AbstractSequence.html#getBioBegin()">getBioBegin</a>, <a href="template/AbstractSequence.html#getBioEnd()">getBioEnd</a>, <a href="template/AbstractSequence.html#getCompoundAt(int)">getCompoundAt</a>, <a href="template/AbstractSequence.html#getCompoundSet()">getCompoundSet</a>, <a href="template/AbstractSequence.html#getDatabaseReferences()">getDatabaseReferences</a>, <a href="template/AbstractSequence.html#getDescription()">getDescription</a>, <a href="template/AbstractSequence.html#getFeatures()">getFeatures</a>, <a href="template/AbstractSequence.html#getFeatures(int)">getFeatures</a>, <a href="template/AbstractSequence.html#getFeatures(java.lang.String,int)">getFeatures</a>, <a href="template/AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType</a>, <a href="template/AbstractSequence.html#getFeaturesKeyWord()">getFeaturesKeyWord</a>, <a href="template/AbstractSequence.html#getIndexOf(C)">getIndexOf</a>, <a href="template/AbstractSequence.html#getLastIndexOf(C)">getLastIndexOf</a>, <a href="template/AbstractSequence.html#getLength()">getLength</a>, <a href="template/AbstractSequence.html#getNotesList()">getNotesList</a>, <a href="template/AbstractSequence.html#getOriginalHeader()">getOriginalHeader</a>, <a href="template/AbstractSequence.html#getParentSequence()">getParentSequence</a>, <a href="template/AbstractSequence.html#getProxySequenceReader()">getProxySequenceReader</a>, <a href="template/AbstractSequence.html#getSequenceAsString()">getSequenceAsString</a>, <a href="template/AbstractSequence.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)">getSequenceAsString</a>, <a href="template/AbstractSequence.html#getSequenceScore()">getSequenceScore</a>, <a href="template/AbstractSequence.html#getSource()">getSource</a>, <a href="template/AbstractSequence.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence</a>, <a href="template/AbstractSequence.html#getTaxonomy()">getTaxonomy</a>, <a href="template/AbstractSequence.html#getUserCollection()">getUserCollection</a>, <a href="template/AbstractSequence.html#iterator()">iterator</a>, <a href="template/AbstractSequence.html#removeFeature(org.biojava3.core.sequence.features.FeatureInterface)">removeFeature</a>, <a href="template/AbstractSequence.html#removeNote(java.lang.String)">removeNote</a>, <a href="template/AbstractSequence.html#setAccession(org.biojava3.core.sequence.AccessionID)">setAccession</a>, <a href="template/AbstractSequence.html#setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType</a>, <a href="template/AbstractSequence.html#setBioBegin(java.lang.Integer)">setBioBegin</a>, <a href="template/AbstractSequence.html#setBioEnd(java.lang.Integer)">setBioEnd</a>, <a href="template/AbstractSequence.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a>, <a href="template/AbstractSequence.html#setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences</a>, <a href="template/AbstractSequence.html#setDescription(java.lang.String)">setDescription</a>, <a href="template/AbstractSequence.html#setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord</a>, <a href="template/AbstractSequence.html#setNotesList(java.util.ArrayList)">setNotesList</a>, <a href="template/AbstractSequence.html#setOriginalHeader(java.lang.String)">setOriginalHeader</a>, <a href="template/AbstractSequence.html#setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)">setParentSequence</a>, <a href="template/AbstractSequence.html#setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)">setProxySequenceReader</a>, <a href="template/AbstractSequence.html#setSequenceScore(java.lang.Double)">setSequenceScore</a>, <a href="template/AbstractSequence.html#setSource(java.lang.String)">setSource</a>, <a href="template/AbstractSequence.html#setTaxonomy(org.biojava3.core.sequence.TaxonomyID)">setTaxonomy</a>, <a href="template/AbstractSequence.html#setUserCollection(java.util.Collection)">setUserCollection</a>, <a href="template/AbstractSequence.html#toString()">toString</a></code></div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Object">Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)" title="class or interface in java.lang" class="external-link">equals</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#getClass()" title="class or interface in java.lang" class="external-link">getClass</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode()" title="class or interface in java.lang" class="external-link">hashCode</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notify()" title="class or interface in java.lang" class="external-link">notify</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notifyAll()" title="class or interface in java.lang" class="external-link">notifyAll</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait()" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long)" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long,int)" title="class or interface in java.lang" class="external-link">wait</a></code></div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Iterable">Methods inherited from interface&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html" title="class or interface in java.lang" class="external-link">Iterable</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html#forEach(java.util.function.Consumer)" title="class or interface in java.lang" class="external-link">forEach</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html#spliterator()" title="class or interface in java.lang" class="external-link">spliterator</a></code></div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<li>
<section class="constructor-details" id="constructor-detail">
<h2>Constructor Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="&lt;init&gt;(java.lang.String)">
<h3>RNASequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name">RNASequence</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString)</span></div>
<div class="block">Create a RNA sequence from a String</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>seqString</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="&lt;init&gt;(org.biojava3.core.sequence.template.ProxySequenceReader)">
<h3>RNASequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name">RNASequence</span><wbr><span class="parameters">(<a href="template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader)</span></div>
<div class="block">Create a RNA aequence from a proxy reader</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>proxyLoader</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="&lt;init&gt;(java.lang.String,org.biojava3.core.sequence.template.CompoundSet)">
<h3>RNASequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name">RNASequence</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</span></div>
<div class="block">Create a RNA sequence from a string with a user defined RNA compound set</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>seqString</code> - </dd>
<dd><code>compoundSet</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="&lt;init&gt;(org.biojava3.core.sequence.template.ProxySequenceReader,org.biojava3.core.sequence.template.CompoundSet)">
<h3>RNASequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name">RNASequence</span><wbr><span class="parameters">(<a href="template/ProxySequenceReader.html" title="interface in org.biojava3.core.sequence.template">ProxySequenceReader</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
 <a href="template/CompoundSet.html" title="interface in org.biojava3.core.sequence.template">CompoundSet</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</span></div>
<div class="block">Create a RNA sequence from a proxy reader and user defined RNA compound set</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>proxyLoader</code> - </dd>
<dd><code>compoundSet</code> - </dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="getReverseComplement()">
<h3>getReverseComplement</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</span>&nbsp;<span class="element-name">getReverseComplement</span>()</div>
<div class="block">Get reverse complement view of the sequence</div>
<dl class="notes">
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getInverse()">
<h3>getInverse</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</span>&nbsp;<span class="element-name">getInverse</span>()</div>
<div class="block">Get the inverse view of the sequence. It is the reverse sequence from end to begin where use
 reverse could imply complement. Called getInverse() in the hopes of making less confusing.</div>
<dl class="notes">
<dt>Specified by:</dt>
<dd><code><a href="template/Sequence.html#getInverse()">getInverse</a></code>&nbsp;in interface&nbsp;<code><a href="template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
<dt>Overrides:</dt>
<dd><code><a href="template/AbstractSequence.html#getInverse()">getInverse</a></code>&nbsp;in class&nbsp;<code><a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getComplement()">
<h3>getComplement</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="template/SequenceView.html" title="interface in org.biojava3.core.sequence.template">SequenceView</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</span>&nbsp;<span class="element-name">getComplement</span>()</div>
<div class="block">Get the complement view of the RNA sequence</div>
<dl class="notes">
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getProteinSequence()">
<h3>getProteinSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></span>&nbsp;<span class="element-name">getProteinSequence</span>()</div>
<div class="block">Get the ProteinSequence from the RNA sequence</div>
<dl class="notes">
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getProteinSequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)">
<h3>getProteinSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></span>&nbsp;<span class="element-name">getProteinSequence</span><wbr><span class="parameters">(<a href="transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a>&nbsp;engine)</span></div>
<div class="block">Get the ProteinSequene from the RNA sequence with user defined transcription engine</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>engine</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getGC()">
<h3>getGC</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">double</span>&nbsp;<span class="element-name">getGC</span>()</div>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
</div>
</div>
</body>
</html>
