<!DOCTYPE HTML>
<html lang="fr">
<head>
<!-- Generated by javadoc (17) on Sun Dec 19 23:37:40 CET 2021 -->
<title>TranscriptSequence</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2021-12-19">
<meta name="description" content="declaration: package: org.biojava3.core.sequence, class: TranscriptSequence">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var evenRowColor = "even-row-color";
var oddRowColor = "odd-row-color";
var tableTab = "table-tab";
var activeTableTab = "active-table-tab";
var pathtoroot = "../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top">
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/TranscriptSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html#class">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div>
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li><a href="#nested-class-summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" value="search" disabled="disabled">
<input type="reset" id="reset-button" value="reset" disabled="disabled">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">org.biojava3.core.sequence</a></div>
<h1 title="Class TranscriptSequence" class="title">Class TranscriptSequence</h1>
</div>
<div class="inheritance" title="Inheritance Tree"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">java.lang.Object</a>
<div class="inheritance"><a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">org.biojava3.core.sequence.template.AbstractSequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;
<div class="inheritance"><a href="DNASequence.html" title="class in org.biojava3.core.sequence">org.biojava3.core.sequence.DNASequence</a>
<div class="inheritance">org.biojava3.core.sequence.TranscriptSequence</div>
</div>
</div>
</div>
<section class="class-description" id="class-description">
<dl class="notes">
<dt>All Implemented Interfaces:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html" title="class or interface in java.lang" class="external-link">Iterable</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code>, <code><a href="template/Accessioned.html" title="interface in org.biojava3.core.sequence.template">Accessioned</a></code>, <code><a href="template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
</dl>
<hr>
<div class="type-signature"><span class="modifiers">public class </span><span class="element-name type-name-label">TranscriptSequence</span>
<span class="extends-implements">extends <a href="DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span></div>
<div class="block">This is the sequence if you want to go from a gene sequence to a protein sequence. Need to start
 with a ChromosomeSequence then getting a GeneSequence and then a TranscriptSequence</div>
<dl class="notes">
<dt>Author:</dt>
<dd>Scooter Willis</dd>
</dl>
</section>
<section class="summary">
<ul class="summary-list">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<li>
<section class="nested-class-summary" id="nested-class-summary">
<h2>Nested Class Summary</h2>
<div class="inherited-list">
<h2 id="nested-classes-inherited-from-class-org.biojava3.core.sequence.DNASequence">Nested classes/interfaces inherited from class&nbsp;org.biojava3.core.sequence.<a href="DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></h2>
<code><a href="DNASequence.DNAType.html" title="enum in org.biojava3.core.sequence">DNASequence.DNAType</a></code></div>
<div class="inherited-list">
<h2 id="nested-classes-inherited-from-class-org.biojava3.core.sequence.template.AbstractSequence">Nested classes/interfaces inherited from class&nbsp;org.biojava3.core.sequence.template.<a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></h2>
<code><a href="template/AbstractSequence.AnnotationType.html" title="enum in org.biojava3.core.sequence.template">AbstractSequence.AnnotationType</a></code></div>
</section>
</li>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<li>
<section class="constructor-summary" id="constructor-summary">
<h2>Constructor Summary</h2>
<div class="caption"><span>Constructors</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E(org.biojava3.core.sequence.GeneSequence,int,int)" class="member-name-link">TranscriptSequence</a><wbr>(<a href="GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a>&nbsp;parentDNASequence,
 int&nbsp;begin,
 int&nbsp;end)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab2" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab2', 3)" class="table-tab">Instance Methods</button><button id="method-summary-table-tab4" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab4', 3)" class="table-tab">Concrete Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel">
<div class="summary-table three-column-summary" aria-labelledby="method-summary-table-tab0">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#addCDS(org.biojava3.core.sequence.AccessionID,int,int,int)" class="member-name-link">addCDS</a><wbr>(<a href="AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession,
 int&nbsp;begin,
 int&nbsp;end,
 int&nbsp;phase)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Add a Coding Sequence region with phase to the transcript sequence</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#addStartCodonSequence(org.biojava3.core.sequence.AccessionID,int,int)" class="member-name-link">addStartCodonSequence</a><wbr>(<a href="AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession,
 int&nbsp;begin,
 int&nbsp;end)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#addStopCodonSequence(org.biojava3.core.sequence.AccessionID,int,int)" class="member-name-link">addStopCodonSequence</a><wbr>(<a href="AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession,
 int&nbsp;begin,
 int&nbsp;end)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/LinkedHashMap.html" title="class or interface in java.util" class="external-link">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a>&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getCDSSequences()" class="member-name-link">getCDSSequences</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get the CDS sequences that have been added to the TranscriptSequences</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getDNACodingSequence()" class="member-name-link">getDNACodingSequence</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get the stitched together CDS sequences then maps to the cDNA</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>int</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getLength()" class="member-name-link">getLength</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Returns the length of the Sequence</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getProteinCDSSequences()" class="member-name-link">getProteinCDSSequences</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Return a list of protein sequences based on each CDS sequence where the phase shift between two
 CDS sequences is assigned to the CDS sequence that starts the triplet.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getProteinSequence()" class="member-name-link">getProteinSequence</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get the protein sequence</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getProteinSequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)" class="member-name-link">getProteinSequence</a><wbr>(<a href="transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a>&nbsp;engine)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get the protein sequence with user defined TranscriptEngine</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="StartCodonSequence.html" title="class in org.biojava3.core.sequence">StartCodonSequence</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getStartCodonSequence()" class="member-name-link">getStartCodonSequence</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="StopCodonSequence.html" title="class in org.biojava3.core.sequence">StopCodonSequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getStopCodonSequence()" class="member-name-link">getStopCodonSequence</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="Strand.html" title="enum in org.biojava3.core.sequence">Strand</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getStrand()" class="member-name-link">getStrand</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#removeCDS(java.lang.String)" class="member-name-link">removeCDS</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Remove a CDS or coding sequence from the transcript sequence</div>
</div>
</div>
</div>
</div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-org.biojava3.core.sequence.DNASequence">Methods inherited from class&nbsp;org.biojava3.core.sequence.<a href="DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></h3>
<code><a href="DNASequence.html#getComplement()">getComplement</a>, <a href="DNASequence.html#getDNAType()">getDNAType</a>, <a href="DNASequence.html#getGCCount()">getGCCount</a>, <a href="DNASequence.html#getReverse()">getReverse</a>, <a href="DNASequence.html#getReverseComplement()">getReverseComplement</a>, <a href="DNASequence.html#getRNASequence()">getRNASequence</a>, <a href="DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.Frame)">getRNASequence</a>, <a href="DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)">getRNASequence</a>, <a href="DNASequence.html#getRNASequence(org.biojava3.core.sequence.transcription.TranscriptionEngine,org.biojava3.core.sequence.transcription.Frame)">getRNASequence</a>, <a href="DNASequence.html#main(java.lang.String%5B%5D)">main</a>, <a href="DNASequence.html#setDNAType(org.biojava3.core.sequence.DNASequence.DNAType)">setDNAType</a></code></div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-org.biojava3.core.sequence.template.AbstractSequence">Methods inherited from class&nbsp;org.biojava3.core.sequence.template.<a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a></h3>
<code><a href="template/AbstractSequence.html#addFeature(int,int,org.biojava3.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="template/AbstractSequence.html#addFeature(org.biojava3.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="template/AbstractSequence.html#addNote(java.lang.String)">addNote</a>, <a href="template/AbstractSequence.html#countCompounds(C...)">countCompounds</a>, <a href="template/AbstractSequence.html#getAccession()">getAccession</a>, <a href="template/AbstractSequence.html#getAnnotationType()">getAnnotationType</a>, <a href="template/AbstractSequence.html#getAsList()">getAsList</a>, <a href="template/AbstractSequence.html#getBioBegin()">getBioBegin</a>, <a href="template/AbstractSequence.html#getBioEnd()">getBioEnd</a>, <a href="template/AbstractSequence.html#getCompoundAt(int)">getCompoundAt</a>, <a href="template/AbstractSequence.html#getCompoundSet()">getCompoundSet</a>, <a href="template/AbstractSequence.html#getDatabaseReferences()">getDatabaseReferences</a>, <a href="template/AbstractSequence.html#getDescription()">getDescription</a>, <a href="template/AbstractSequence.html#getFeatures()">getFeatures</a>, <a href="template/AbstractSequence.html#getFeatures(int)">getFeatures</a>, <a href="template/AbstractSequence.html#getFeatures(java.lang.String,int)">getFeatures</a>, <a href="template/AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType</a>, <a href="template/AbstractSequence.html#getFeaturesKeyWord()">getFeaturesKeyWord</a>, <a href="template/AbstractSequence.html#getIndexOf(C)">getIndexOf</a>, <a href="template/AbstractSequence.html#getInverse()">getInverse</a>, <a href="template/AbstractSequence.html#getLastIndexOf(C)">getLastIndexOf</a>, <a href="template/AbstractSequence.html#getNotesList()">getNotesList</a>, <a href="template/AbstractSequence.html#getOriginalHeader()">getOriginalHeader</a>, <a href="template/AbstractSequence.html#getParentSequence()">getParentSequence</a>, <a href="template/AbstractSequence.html#getProxySequenceReader()">getProxySequenceReader</a>, <a href="template/AbstractSequence.html#getSequenceAsString()">getSequenceAsString</a>, <a href="template/AbstractSequence.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava3.core.sequence.Strand)">getSequenceAsString</a>, <a href="template/AbstractSequence.html#getSequenceScore()">getSequenceScore</a>, <a href="template/AbstractSequence.html#getSource()">getSource</a>, <a href="template/AbstractSequence.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence</a>, <a href="template/AbstractSequence.html#getTaxonomy()">getTaxonomy</a>, <a href="template/AbstractSequence.html#getUserCollection()">getUserCollection</a>, <a href="template/AbstractSequence.html#iterator()">iterator</a>, <a href="template/AbstractSequence.html#removeFeature(org.biojava3.core.sequence.features.FeatureInterface)">removeFeature</a>, <a href="template/AbstractSequence.html#removeNote(java.lang.String)">removeNote</a>, <a href="template/AbstractSequence.html#setAccession(org.biojava3.core.sequence.AccessionID)">setAccession</a>, <a href="template/AbstractSequence.html#setAnnotationType(org.biojava3.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType</a>, <a href="template/AbstractSequence.html#setBioBegin(java.lang.Integer)">setBioBegin</a>, <a href="template/AbstractSequence.html#setBioEnd(java.lang.Integer)">setBioEnd</a>, <a href="template/AbstractSequence.html#setCompoundSet(org.biojava3.core.sequence.template.CompoundSet)">setCompoundSet</a>, <a href="template/AbstractSequence.html#setDatabaseReferences(org.biojava3.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences</a>, <a href="template/AbstractSequence.html#setDescription(java.lang.String)">setDescription</a>, <a href="template/AbstractSequence.html#setFeaturesKeyWord(org.biojava3.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord</a>, <a href="template/AbstractSequence.html#setNotesList(java.util.ArrayList)">setNotesList</a>, <a href="template/AbstractSequence.html#setOriginalHeader(java.lang.String)">setOriginalHeader</a>, <a href="template/AbstractSequence.html#setParentSequence(org.biojava3.core.sequence.template.AbstractSequence)">setParentSequence</a>, <a href="template/AbstractSequence.html#setProxySequenceReader(org.biojava3.core.sequence.template.SequenceReader)">setProxySequenceReader</a>, <a href="template/AbstractSequence.html#setSequenceScore(java.lang.Double)">setSequenceScore</a>, <a href="template/AbstractSequence.html#setSource(java.lang.String)">setSource</a>, <a href="template/AbstractSequence.html#setTaxonomy(org.biojava3.core.sequence.TaxonomyID)">setTaxonomy</a>, <a href="template/AbstractSequence.html#setUserCollection(java.util.Collection)">setUserCollection</a>, <a href="template/AbstractSequence.html#toString()">toString</a></code></div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Object">Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)" title="class or interface in java.lang" class="external-link">equals</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#getClass()" title="class or interface in java.lang" class="external-link">getClass</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode()" title="class or interface in java.lang" class="external-link">hashCode</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notify()" title="class or interface in java.lang" class="external-link">notify</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notifyAll()" title="class or interface in java.lang" class="external-link">notifyAll</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait()" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long)" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long,int)" title="class or interface in java.lang" class="external-link">wait</a></code></div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Iterable">Methods inherited from interface&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html" title="class or interface in java.lang" class="external-link">Iterable</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html#forEach(java.util.function.Consumer)" title="class or interface in java.lang" class="external-link">forEach</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html#spliterator()" title="class or interface in java.lang" class="external-link">spliterator</a></code></div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<li>
<section class="constructor-details" id="constructor-detail">
<h2>Constructor Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="&lt;init&gt;(org.biojava3.core.sequence.GeneSequence,int,int)">
<h3>TranscriptSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name">TranscriptSequence</span><wbr><span class="parameters">(<a href="GeneSequence.html" title="class in org.biojava3.core.sequence">GeneSequence</a>&nbsp;parentDNASequence,
 int&nbsp;begin,
 int&nbsp;end)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>parentDNASequence</code> - </dd>
<dd><code>begin</code> - </dd>
<dd><code>end</code> - inclusive of end</dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="getLength()">
<h3>getLength</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name">getLength</span>()</div>
<div class="block"><span class="descfrm-type-label">Description copied from interface:&nbsp;<code><a href="template/Sequence.html#getLength()">Sequence</a></code></span></div>
<div class="block">Returns the length of the Sequence</div>
<dl class="notes">
<dt>Specified by:</dt>
<dd><code><a href="template/Sequence.html#getLength()">getLength</a></code>&nbsp;in interface&nbsp;<code><a href="template/Sequence.html" title="interface in org.biojava3.core.sequence.template">Sequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
<dt>Overrides:</dt>
<dd><code><a href="template/AbstractSequence.html#getLength()">getLength</a></code>&nbsp;in class&nbsp;<code><a href="template/AbstractSequence.html" title="class in org.biojava3.core.sequence.template">AbstractSequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava3.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getStrand()">
<h3>getStrand</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="Strand.html" title="enum in org.biojava3.core.sequence">Strand</a></span>&nbsp;<span class="element-name">getStrand</span>()</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the strand</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="removeCDS(java.lang.String)">
<h3>removeCDS</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></span>&nbsp;<span class="element-name">removeCDS</span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession)</span></div>
<div class="block">Remove a CDS or coding sequence from the transcript sequence</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>accession</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getCDSSequences()">
<h3>getCDSSequences</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/LinkedHashMap.html" title="class or interface in java.util" class="external-link">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a>&gt;</span>&nbsp;<span class="element-name">getCDSSequences</span>()</div>
<div class="block">Get the CDS sequences that have been added to the TranscriptSequences</div>
<dl class="notes">
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="addCDS(org.biojava3.core.sequence.AccessionID,int,int,int)">
<h3>addCDS</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="CDSSequence.html" title="class in org.biojava3.core.sequence">CDSSequence</a></span>&nbsp;<span class="element-name">addCDS</span><wbr><span class="parameters">(<a href="AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession,
 int&nbsp;begin,
 int&nbsp;end,
 int&nbsp;phase)</span>
                   throws <span class="exceptions"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Exception.html" title="class or interface in java.lang" class="external-link">Exception</a></span></div>
<div class="block">Add a Coding Sequence region with phase to the transcript sequence</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>accession</code> - </dd>
<dd><code>begin</code> - </dd>
<dd><code>end</code> - </dd>
<dd><code>phase</code> - 0,1,2</dd>
<dt>Returns:</dt>
<dt>Throws:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Exception.html" title="class or interface in java.lang" class="external-link">Exception</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getProteinCDSSequences()">
<h3>getProteinCDSSequences</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html" title="class or interface in java.util" class="external-link">ArrayList</a>&lt;<a href="ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a>&gt;</span>&nbsp;<span class="element-name">getProteinCDSSequences</span>()</div>
<div class="block">Return a list of protein sequences based on each CDS sequence where the phase shift between two
 CDS sequences is assigned to the CDS sequence that starts the triplet. This can be used to map a
 CDS/exon region of a protein sequence back to the DNA sequence If you have a protein sequence and
 a predicted gene you can take the predict CDS protein sequences and align back to the protein
 sequence. If you have errors in mapping the predicted protein CDS regions to an the known protein
 sequence then you can identify possible errors in the prediction</div>
<dl class="notes">
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getDNACodingSequence()">
<h3>getDNACodingSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="DNASequence.html" title="class in org.biojava3.core.sequence">DNASequence</a></span>&nbsp;<span class="element-name">getDNACodingSequence</span>()</div>
<div class="block">Get the stitched together CDS sequences then maps to the cDNA</div>
<dl class="notes">
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getProteinSequence()">
<h3>getProteinSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></span>&nbsp;<span class="element-name">getProteinSequence</span>()</div>
<div class="block">Get the protein sequence</div>
<dl class="notes">
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getProteinSequence(org.biojava3.core.sequence.transcription.TranscriptionEngine)">
<h3>getProteinSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="ProteinSequence.html" title="class in org.biojava3.core.sequence">ProteinSequence</a></span>&nbsp;<span class="element-name">getProteinSequence</span><wbr><span class="parameters">(<a href="transcription/TranscriptionEngine.html" title="class in org.biojava3.core.sequence.transcription">TranscriptionEngine</a>&nbsp;engine)</span></div>
<div class="block">Get the protein sequence with user defined TranscriptEngine</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>engine</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getStartCodonSequence()">
<h3>getStartCodonSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="StartCodonSequence.html" title="class in org.biojava3.core.sequence">StartCodonSequence</a></span>&nbsp;<span class="element-name">getStartCodonSequence</span>()</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the startCodonSequence</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="addStartCodonSequence(org.biojava3.core.sequence.AccessionID,int,int)">
<h3>addStartCodonSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">addStartCodonSequence</span><wbr><span class="parameters">(<a href="AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession,
 int&nbsp;begin,
 int&nbsp;end)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>startCodonSequence</code> - the startCodonSequence to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getStopCodonSequence()">
<h3>getStopCodonSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="StopCodonSequence.html" title="class in org.biojava3.core.sequence">StopCodonSequence</a></span>&nbsp;<span class="element-name">getStopCodonSequence</span>()</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the stopCodonSequence</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="addStopCodonSequence(org.biojava3.core.sequence.AccessionID,int,int)">
<h3>addStopCodonSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name">addStopCodonSequence</span><wbr><span class="parameters">(<a href="AccessionID.html" title="class in org.biojava3.core.sequence">AccessionID</a>&nbsp;accession,
 int&nbsp;begin,
 int&nbsp;end)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>stopCodonSequence</code> - the stopCodonSequence to set</dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
</div>
</div>
</body>
</html>
